// @salutejs/plasma-new-hope/cjs/components/Slider/Slider.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var base$2 = require('./variations/_disabled/base.js');
var Single = require('./components/Single/Single.js');
var Double = require('./components/Double/Double.js');

var isSingleValueProps = function isSingleValueProps(props) {
  return typeof props.value === 'number';
};
var sliderRoot = function sliderRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (props, ref) {
    return /*#__PURE__*/React.createElement(Root, _rollupPluginBabelHelpers.extends({
      ref: ref
    }, props), isSingleValueProps(props) && /*#__PURE__*/React.createElement(Single.SingleSlider, props), !isSingleValueProps(props) && /*#__PURE__*/React.createElement(Double.DoubleSlider, props));
  });
};
var sliderConfig = {
  name: 'Slider',
  tag: 'div',
  layout: sliderRoot,
  base: '',
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    },
    disabled: {
      css: base$2.base,
      attrs: true
    }
  },
  defaults: {
    view: 'default',
    size: 'm'
  }
};

exports.sliderConfig = sliderConfig;
exports.sliderRoot = sliderRoot;
