// @salutejs/plasma-ui/es/components/Badge/Badge.perftest.js
var _IconSettings;
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React from 'react';
import { IconSettings } from '@salutejs/plasma-icons';
import { Badge } from '.';
var defaultProps = {
  text: 'Badge',
  size: 'l',
  view: 'primary',
  enableIcon: false
};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Badge, _extends({
    contentLeft: _IconSettings || (_IconSettings = /*#__PURE__*/React.createElement(IconSettings, {
      color: "inherit",
      size: "xs"
    }))
  }, defaultProps));
};
var quantityProps = {
  text: '11',
  size: 's',
  view: 'secondary',
  circled: true
};
export var Quantity = function Quantity() {
  return /*#__PURE__*/React.createElement(Badge, quantityProps);
};
