// @salutejs/plasma-new-hope/es/components/Notification/Notification.styles.js
import './Notification.styles_7h8hp0.css';
import { styled } from '@linaria/react';
import { buttonConfig } from '../Button/Button.js';
import '@linaria/core';
import { component, mergeConfig } from '../../engines/common.js';
import { tokens } from './Notification.tokens.js';
import { placements } from './Notification.types.js';

var StyledNotification = "sb12p";
var mergedButtonConfig = /*#__PURE__*/mergeConfig(buttonConfig);
var Button = /*#__PURE__*/component(mergedButtonConfig);
var _exp = function _exp() {
  return Button;
};
var CloseIconWrapper = /*#__PURE__*/styled(_exp())({
  name: "CloseIconWrapper",
  "class": "crvd7hm",
  propsAsIs: true
});
var Wrapper = /*#__PURE__*/styled('div')({
  name: "Wrapper",
  "class": "wg1ga5v",
  propsAsIs: false
});
var _exp10 = function _exp10() {
  return function (_ref) {
    var iconPlacement = _ref.iconPlacement;
    return iconPlacement === placements.left ? "var(".concat(tokens.buttonsMarginLeft, ")") : 'unset';
  };
};
var ButtonsWrapper = /*#__PURE__*/styled('div')({
  name: "ButtonsWrapper",
  "class": "b1amsn8a",
  propsAsIs: false,
  vars: {
    "b1amsn8a-0": [/*#__PURE__*/_exp10()]
  }
});
var _exp14 = function _exp14() {
  return function (_ref2) {
    var iconPlacement = _ref2.iconPlacement;
    return iconPlacement === placements.left ? "var(".concat(tokens.contentLeftIconMargin, ")") : 'unset';
  };
};
var _exp15 = function _exp15() {
  return function (_ref3) {
    var iconPlacement = _ref3.iconPlacement;
    return iconPlacement === placements.top ? "var(".concat(tokens.contentTopIconMargin, ")") : 'unset';
  };
};
var IconWrapper = /*#__PURE__*/styled('div')({
  name: "IconWrapper",
  "class": "i14raqr9",
  propsAsIs: false,
  vars: {
    "i14raqr9-0": [/*#__PURE__*/_exp14()],
    "i14raqr9-1": [/*#__PURE__*/_exp15()]
  }
});
var _exp18 = function _exp18() {
  return function (_ref4) {
    var iconPlacement = _ref4.iconPlacement;
    return iconPlacement === placements.left ? 'flex' : 'block';
  };
};
var _exp19 = function _exp19() {
  return function (_ref5) {
    var iconPlacement = _ref5.iconPlacement;
    return iconPlacement ? "var(".concat(tokens.contentPaddingTop, ")") : "var(".concat(tokens.contentPaddingTopWithoutIcon, ")");
  };
};
var ContentBox = /*#__PURE__*/styled('div')({
  name: "ContentBox",
  "class": "c157peez",
  propsAsIs: false,
  vars: {
    "c157peez-0": [/*#__PURE__*/_exp18()],
    "c157peez-1": [/*#__PURE__*/_exp19()]
  }
});
var _exp25 = function _exp25() {
  return function (_ref6) {
    var iconPlacement = _ref6.iconPlacement;
    return !iconPlacement || iconPlacement === placements.top ? "var(".concat(tokens.textboxPaddingTopWithTopIcon, ")") : "var(".concat(tokens.textboxPaddingTop, ")");
  };
};
var _exp26 = function _exp26() {
  return function (_ref7) {
    var showCloseIcon = _ref7.showCloseIcon,
      iconPlacement = _ref7.iconPlacement;
    return showCloseIcon && iconPlacement !== placements.top ? "var(".concat(tokens.textboxPaddingRightWithCloseIcon, ")") : "var(".concat(tokens.textboxPaddingRight, ")");
  };
};
var TextBox = /*#__PURE__*/styled('div')({
  name: "TextBox",
  "class": "t1vmb0f2",
  propsAsIs: false,
  vars: {
    "t1vmb0f2-0": [/*#__PURE__*/_exp25()],
    "t1vmb0f2-1": [/*#__PURE__*/_exp26()]
  }
});
var StyledTitle = /*#__PURE__*/styled('div')({
  name: "StyledTitle",
  "class": "s1k4e58v",
  propsAsIs: false
});
var StyledContent = /*#__PURE__*/styled('div')({
  name: "StyledContent",
  "class": "s92kmpm",
  propsAsIs: false
});
var StyledItemWrapper = /*#__PURE__*/styled('div')({
  name: "StyledItemWrapper",
  "class": "s18m0x6v",
  propsAsIs: false
});
var StyledRoot = /*#__PURE__*/styled('div')({
  name: "StyledRoot",
  "class": "sfe5ql2",
  propsAsIs: false
});

export { ButtonsWrapper, CloseIconWrapper, ContentBox, IconWrapper, StyledContent, StyledItemWrapper, StyledNotification, StyledRoot, StyledTitle, TextBox, Wrapper };
