// @salutejs/plasma-new-hope/styled-components/cjs/components/Combobox/Combobox.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledRoot = exports.StyledPopover = exports.StyledNativeSelect = exports.StyledCombobox = exports.ComboboxItemNotFound = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _engines = /*#__PURE__*/require("../../engines");
var _Popover = /*#__PURE__*/require("../Popover");
var _Combobox = /*#__PURE__*/require("./Combobox.tokens");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var StyledRoot = exports.StyledRoot = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-gt7m6a-0"
})(["width:100%;position:relative;& .", "{display:block;}"], _Popover.popoverClasses.wrapper);
var StyledCombobox = exports.StyledCombobox = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-gt7m6a-1"
})(["box-sizing:border-box;background:var(", ");box-shadow:var(", ");border-radius:var(", ");width:var(", ");height:var(", ");padding:var(", ") var(", ") var(", ") var(", ");"], _Combobox.tokens.background, _Combobox.tokens.boxShadow, _Combobox.tokens.borderRadius, _Combobox.tokens.width, _Combobox.tokens.height, _Combobox.tokens.paddingTop, _Combobox.tokens.paddingRight, _Combobox.tokens.paddingBottom, _Combobox.tokens.paddingLeft);

// issue #823
var Popover = /*#__PURE__*/(0, _engines.component)(_Popover.popoverConfig);
var StyledPopover = exports.StyledPopover = /*#__PURE__*/(0, _styledComponents["default"])(Popover).withConfig({
  componentId: "plasma-new-hope__sc-gt7m6a-2"
})(["display:block;.", ",.", "{width:", ";}"], /*#__PURE__*/String(_Popover.popoverClasses.root), /*#__PURE__*/String(_Popover.popoverClasses.target), function (_ref) {
  var usePortal = _ref.usePortal,
    selectWidth = _ref.selectWidth;
  return usePortal ? "".concat(selectWidth, "px") : '100%';
});
var ComboboxItemNotFound = exports.ComboboxItemNotFound = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-gt7m6a-3"
})(["text-align:center;padding:0.5rem;font-family:var(", ");font-size:var(", ");font-style:var(", ");font-weight:var(", ");letter-spacing:var(", ");line-height:var(", ");"], _Combobox.tokens.itemFontFamily, _Combobox.tokens.itemFontSize, _Combobox.tokens.itemFontStyle, _Combobox.tokens.itemFontWeightBold, _Combobox.tokens.itemFontLetterSpacing, _Combobox.tokens.itemFontLineHeight);
var StyledNativeSelect = exports.StyledNativeSelect = /*#__PURE__*/_styledComponents["default"].select.withConfig({
  componentId: "plasma-new-hope__sc-gt7m6a-4"
})(["display:none;appearance:none;-moz-appearance:none;-webkit-appearance:none;background:transparent;color:transparent;border:none;outline:none;opacity:0;position:absolute;inset:0;"]);
