// @salutejs/plasma-web/components/Upload/Upload.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _ = /*#__PURE__*/require(".");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var defaultProps = {
  disabled: false,
  content: 'Загрузите файл формата любого'
};
var state = {
  status: undefined,
  progress: undefined,
  message: 'Подсказывающее сообщение'
};
var valid = function valid() {
  return {
    message: 'Добавлен файл формата ',
    status: 'success',
    data: null
  };
};
var func = function func() {};
var Default = exports.Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(_.Upload, _extends({
    status: state.status,
    progress: state.progress,
    message: state.message,
    loader: /*#__PURE__*/_react["default"].createElement("div", null, "\u041A\u0430\u0441\u0442\u043E\u043C\u043D\u0430\u044F \u0437\u0430\u0433\u0440\u0443\u0437\u043A\u0430 ", state.progress, "%"),
    onChange: func,
    onValidation: func,
    validate: valid
  }, defaultProps));
};
