// @salutejs/plasma-new-hope/styled-components/cjs/components/Dropdown/ui/DropdownItem/DropdownItem.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Wrapper = exports.StyledText = exports.StyledContentRight = exports.StyledContentLeft = exports.DisclosureIconWrapper = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _Dropdown = /*#__PURE__*/require("../../Dropdown.tokens");
var _mixins = /*#__PURE__*/require("../../../../mixins");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var StyledContentLeft = exports.StyledContentLeft = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-hs0ek1-0"
})(["display:inline-flex;"]);
var StyledContentRight = exports.StyledContentRight = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-hs0ek1-1"
})(["margin-left:auto;display:inline-flex;"]);
var DisclosureIconWrapper = exports.DisclosureIconWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-hs0ek1-2"
})(["line-height:0;color:var(", ");"], _Dropdown.tokens.disclosureIconColor);
var StyledText = exports.StyledText = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-hs0ek1-3"
})(["", ";flex:1;"], /*#__PURE__*/(0, _mixins.applyEllipsis)());
var Wrapper = exports.Wrapper = /*#__PURE__*/_styledComponents["default"].li.withConfig({
  componentId: "plasma-new-hope__sc-hs0ek1-4"
})(["display:flex;align-items:center;margin:0;box-sizing:content-box;padding-top:", ";padding-right:var(", ");padding-bottom:", ";padding-left:var(", ");font-family:var(", ");font-size:var(", ");font-style:var(", ");font-weight:var(", ");letter-spacing:var(", ");line-height:var(", ");background:var(", ");color:var(", ");border-radius:var(", ");user-select:none;&:hover:not(.", "){cursor:pointer;background:var(", ");}&.", "{background:var(", ");}&.", "{opacity:var(", ");cursor:not-allowed;}:focus{outline:none;}", ";"], function (_ref) {
  var variant = _ref.variant;
  return "var(".concat(variant === 'tight' ? _Dropdown.tokens.itemPaddingTopTight : _Dropdown.tokens.itemPaddingTop, ")");
}, _Dropdown.tokens.itemPaddingRight, function (_ref2) {
  var variant = _ref2.variant;
  return "var(".concat(variant === 'tight' ? _Dropdown.tokens.itemPaddingBottomTight : _Dropdown.tokens.itemPaddingBottom, ")");
}, _Dropdown.tokens.itemPaddingLeft, _Dropdown.tokens.itemFontFamily, _Dropdown.tokens.itemFontSize, _Dropdown.tokens.itemFontStyle, _Dropdown.tokens.itemFontWeightBold, _Dropdown.tokens.itemFontLetterSpacing, _Dropdown.tokens.itemFontLineHeight, _Dropdown.tokens.itemBackground, _Dropdown.tokens.itemColor, _Dropdown.tokens.itemBorderRadius, _Dropdown.classes.dropdownItemIsDisabled, _Dropdown.tokens.itemBackgroundHover, _Dropdown.classes.dropdownItemIsActive, _Dropdown.tokens.itemBackgroundHover, _Dropdown.classes.dropdownItemIsDisabled, _Dropdown.tokens.disabledOpacity, /*#__PURE__*/(0, _mixins.addFocus)({
  outlineSize: '0.0625rem',
  outlineOffset: '0',
  outlineColor: /*#__PURE__*/"var(".concat(_Dropdown.tokens.focusColor, ")"),
  outlineRadius: /*#__PURE__*/"var(".concat(_Dropdown.tokens.itemBorderRadius, ")"),
  hasTransition: false,
  customFocusRules: /*#__PURE__*/"\n            &.".concat(_Dropdown.classes.dropdownItemIsFocused, ":before {\n                outline: none;\n                box-shadow: 0 0 0 0.0625rem var(").concat(_Dropdown.tokens.focusColor, ");\n            }\n        ")
}));
