// @salutejs/plasma-new-hope/styled-components/cjs/components/Dropdown/ui/DropdownItem/DropdownItem.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DropdownItem = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _Dropdown = /*#__PURE__*/require("../../Dropdown.tokens");
var _utils = /*#__PURE__*/require("../../../../utils");
var _Icon = /*#__PURE__*/require("../../../_Icon");
var _DropdownItem = /*#__PURE__*/require("./DropdownItem.styles");
var _DisclosureIconWrappe;
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
var DropdownItem = exports.DropdownItem = function DropdownItem(_ref) {
  var item = _ref.item,
    path = _ref.path,
    focusedPath = _ref.focusedPath,
    currentLevel = _ref.currentLevel,
    index = _ref.index,
    itemRole = _ref.itemRole,
    closeOnSelect = _ref.closeOnSelect,
    handleGlobalToggle = _ref.handleGlobalToggle,
    onHover = _ref.onHover,
    onItemSelect = _ref.onItemSelect,
    onItemClick = _ref.onItemClick,
    ariaControls = _ref.ariaControls,
    ariaExpanded = _ref.ariaExpanded,
    ariaHasPopup = _ref.ariaHasPopup,
    ariaLevel = _ref.ariaLevel,
    ariaLabel = _ref.ariaLabel,
    variant = _ref.variant,
    hasArrow = _ref.hasArrow;
  var value = item.value,
    label = item.label,
    disabled = item.disabled,
    isDisabled = item.isDisabled,
    contentLeft = item.contentLeft,
    contentRight = item.contentRight;
  var ref = (0, _react.useRef)(null);
  var isDisabledClassName = disabled || isDisabled ? _Dropdown.classes.dropdownItemIsDisabled : undefined;
  var focusedClass = currentLevel === focusedPath.length - 1 && index === (focusedPath === null || focusedPath === void 0 ? void 0 : focusedPath[currentLevel]) ? _Dropdown.classes.dropdownItemIsFocused : undefined;
  var activeClass = value === (path === null || path === void 0 ? void 0 : path[currentLevel + 1]) ? _Dropdown.classes.dropdownItemIsActive : undefined;
  (0, _react.useEffect)(function () {
    if (focusedClass && ref !== null && ref !== void 0 && ref.current) {
      ref.current.scrollIntoView({
        behavior: 'smooth',
        block: 'center',
        inline: 'center'
      });
    }
  }, [focusedClass]);
  var handleClick = function handleClick(event) {
    if (disabled || isDisabled) {
      return;
    }
    if (onItemSelect) {
      onItemSelect(item, event);
    }
    if (onItemClick) {
      onItemClick(item, event);
    }
    if (handleGlobalToggle && closeOnSelect) {
      handleGlobalToggle(false, event);
    }
  };
  var handleHover = function handleHover() {
    if (onHover) {
      onHover(index);
    }
  };
  return /*#__PURE__*/_react["default"].createElement(_DropdownItem.Wrapper, {
    className: (0, _utils.cx)(isDisabledClassName, focusedClass, activeClass),
    id: value.toString(),
    role: itemRole,
    ref: ref,
    "aria-disabled": disabled || isDisabled,
    onClick: handleClick,
    onMouseEnter: handleHover,
    variant: variant,
    "aria-controls": ariaControls,
    "aria-expanded": ariaExpanded,
    "aria-haspopup": ariaHasPopup,
    "aria-level": ariaLevel,
    "aria-label": ariaLabel
  }, contentLeft && /*#__PURE__*/_react["default"].createElement(_DropdownItem.StyledContentLeft, null, contentLeft), /*#__PURE__*/_react["default"].createElement(_DropdownItem.StyledText, null, label), contentRight && /*#__PURE__*/_react["default"].createElement(_DropdownItem.StyledContentRight, null, contentRight), item.items && hasArrow && (_DisclosureIconWrappe || (_DisclosureIconWrappe = /*#__PURE__*/_react["default"].createElement(_DropdownItem.DisclosureIconWrapper, null, /*#__PURE__*/_react["default"].createElement(_Icon.IconDisclosureRight, {
    size: "xs",
    color: "inherit"
  })))));
};
