// @salutejs/plasma-new-hope/cjs/components/Switch/Switch.js
import './Switch_svqw4q.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var base$1 = require('./_size/base.js');
var base$2 = require('./_view/base.js');
var base$4 = require('./_focused/base.js');
var base$3 = require('./_disabled/base.js');
var base$5 = require('./_label-position/base.js');
var Switch_styles = require('./Switch.styles.js');

var _StyledTrigger;
var _excluded = ["size", "view", "focused", "disabled", "labelPosition", "label", "id", "style", "className", "checked", "defaultChecked"];
var base = "bagv7wj";
var switchRoot = function switchRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (props, ref) {
    var size = props.size,
      view = props.view,
      focused = props.focused,
      disabled = props.disabled,
      labelPosition = props.labelPosition,
      label = props.label,
      id = props.id,
      style = props.style,
      className = props.className,
      checked = props.checked,
      defaultChecked = props.defaultChecked,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(props, _excluded);
    var exactChecked = Boolean(checked !== undefined ? checked : defaultChecked);
    // const singleLineClass = singleLine ? 'single-line' : '';

    return /*#__PURE__*/React.createElement(Root, {
      view: view,
      size: size,
      disabled: disabled,
      focused: focused,
      labelPosition: labelPosition,
      id: id,
      style: style,
      className: className
    }, /*#__PURE__*/React.createElement(Switch_styles.StyledInput, _rollupPluginBabelHelpers.extends({}, rest, {
      ref: ref,
      role: "switch",
      "aria-checked": exactChecked,
      type: "checkbox",
      checked: checked,
      defaultChecked: defaultChecked,
      disabled: disabled
    })), /*#__PURE__*/React.createElement(Switch_styles.StyledLabel, {
      tabIndex: -1
    }, label), _StyledTrigger || (_StyledTrigger = /*#__PURE__*/React.createElement(Switch_styles.StyledTrigger, {
      "aria-hidden": true
    })));
  });
};
var switchConfig = {
  name: 'Switch',
  tag: 'label',
  base: base,
  layout: switchRoot,
  variations: {
    size: {
      css: base$1.base
    },
    view: {
      css: base$2.base
    },
    disabled: {
      css: base$3.base,
      attrs: true
    },
    focused: {
      css: base$4.base
    },
    labelPosition: {
      css: base$5.base
    }
  },
  defaults: {
    size: 's',
    view: 'default'
  }
};

exports.switchConfig = switchConfig;
exports.switchRoot = switchRoot;
