// @salutejs/plasma-tokens/brands/stylesSalute/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.textAccentMainActive = exports.textAccentMainHover = exports.textGradientMain = exports.textGradientMainActive = exports.textGradientMainHover = exports.textGradientBrand = exports.textGradientBrandActive = exports.textGradientBrandHover = exports.textGradientB2E = exports.textGradientB2EActive = exports.textGradientB2EHover = exports.textGradientJoy = exports.textGradientJoyActive = exports.textGradientJoyHover = exports.textGradientAthena = exports.textGradientAthenaActive = exports.textGradientAthenaHover = exports.textAccentBrand = exports.textAccentBrandActive = exports.textAccentBrandHover = exports.textAccentB2E = exports.textAccentB2EActive = exports.textAccentB2EHover = exports.textAccentJoy = exports.textAccentJoyActive = exports.textAccentJoyHover = exports.textAccentAthena = exports.textAccentAthenaActive = exports.textAccentAthenaHover = exports.textNegative = exports.textNegativeActive = exports.textNegativeHover = exports.textWarning = exports.textWarningActive = exports.textWarningHover = exports.textPositive = exports.textPositiveActive = exports.textPositiveHover = exports.textParagraph = exports.textParagraphActive = exports.textParagraphHover = exports.textTertiary = exports.textTertiaryActive = exports.textTertiaryHover = exports.textSecondary = exports.textSecondaryActive = exports.textSecondaryHover = exports.textPrimary = exports.textPrimaryActive = exports.textPrimaryHover = void 0;
exports.onDarkTextAccentMainHover = exports.onDarkTextGradientMain = exports.onDarkTextGradientMainActive = exports.onDarkTextGradientMainHover = exports.onDarkTextGradientBrand = exports.onDarkTextGradientBrandActive = exports.onDarkTextGradientBrandHover = exports.onDarkTextGradientB2E = exports.onDarkTextGradientB2EActive = exports.onDarkTextGradientB2EHover = exports.onDarkTextGradientJoy = exports.onDarkTextGradientJoyActive = exports.onDarkTextGradientJoyHover = exports.onDarkTextGradientAthena = exports.onDarkTextGradientAthenaActive = exports.onDarkTextGradientAthenaHover = exports.onDarkTextAccentBrand = exports.onDarkTextAccentBrandActive = exports.onDarkTextAccentBrandHover = exports.onDarkTextAccentB2E = exports.onDarkTextAccentB2EActive = exports.onDarkTextAccentB2EHover = exports.onDarkTextAccentJoy = exports.onDarkTextAccentJoyActive = exports.onDarkTextAccentJoyHover = exports.onDarkTextAccentAthena = exports.onDarkTextAccentAthenaActive = exports.onDarkTextAccentAthenaHover = exports.onDarkTextNegative = exports.onDarkTextNegativeActive = exports.onDarkTextNegativeHover = exports.onDarkTextWarning = exports.onDarkTextWarningActive = exports.onDarkTextWarningHover = exports.onDarkTextPositive = exports.onDarkTextPositiveActive = exports.onDarkTextPositiveHover = exports.onDarkTextParagraph = exports.onDarkTextParagraphActive = exports.onDarkTextParagraphHover = exports.onDarkTextTertiary = exports.onDarkTextTertiaryActive = exports.onDarkTextTertiaryHover = exports.onDarkTextSecondary = exports.onDarkTextSecondaryActive = exports.onDarkTextSecondaryHover = exports.onDarkTextPrimary = exports.onDarkTextPrimaryActive = exports.onDarkTextPrimaryHover = exports.textAccentMain = void 0;
exports.onLightTextGradientMain = exports.onLightTextGradientMainActive = exports.onLightTextGradientMainHover = exports.onLightTextGradientBrand = exports.onLightTextGradientBrandActive = exports.onLightTextGradientBrandHover = exports.onLightTextGradientB2E = exports.onLightTextGradientB2EActive = exports.onLightTextGradientB2EHover = exports.onLightTextGradientJoy = exports.onLightTextGradientJoyActive = exports.onLightTextGradientJoyHover = exports.onLightTextGradientAthena = exports.onLightTextGradientAthenaActive = exports.onLightTextGradientAthenaHover = exports.onLightTextAccentBrand = exports.onLightTextAccentBrandActive = exports.onLightTextAccentBrandHover = exports.onLightTextAccentB2E = exports.onLightTextAccentB2EActive = exports.onLightTextAccentB2EHover = exports.onLightTextAccentJoy = exports.onLightTextAccentJoyActive = exports.onLightTextAccentJoyHover = exports.onLightTextAccentAthena = exports.onLightTextAccentAthenaActive = exports.onLightTextAccentAthenaHover = exports.onLightTextNegative = exports.onLightTextNegativeActive = exports.onLightTextNegativeHover = exports.onLightTextWarning = exports.onLightTextWarningActive = exports.onLightTextWarningHover = exports.onLightTextPositive = exports.onLightTextPositiveActive = exports.onLightTextPositiveHover = exports.onLightTextParagraph = exports.onLightTextParagraphActive = exports.onLightTextParagraphHover = exports.onLightTextTertiary = exports.onLightTextTertiaryActive = exports.onLightTextTertiaryHover = exports.onLightTextSecondary = exports.onLightTextSecondaryActive = exports.onLightTextSecondaryHover = exports.onLightTextPrimary = exports.onLightTextPrimaryActive = exports.onLightTextPrimaryHover = exports.onDarkTextAccentMain = exports.onDarkTextAccentMainActive = void 0;
exports.surfaceWarningActive = exports.surfaceWarningHover = exports.surfacePositive = exports.surfacePositiveActive = exports.surfacePositiveHover = exports.surfaceClear = exports.surfaceClearActive = exports.surfaceClearHover = exports.surfaceTransparentCard = exports.surfaceTransparentCardActive = exports.surfaceTransparentCardHover = exports.surfaceTransparentTertiary = exports.surfaceTransparentTertiaryActive = exports.surfaceTransparentTertiaryHover = exports.surfaceTransparentSecondary = exports.surfaceTransparentSecondaryActive = exports.surfaceTransparentSecondaryHover = exports.surfaceTransparentPrimary = exports.surfaceTransparentPrimaryActive = exports.surfaceTransparentPrimaryHover = exports.surfaceSolidDefault = exports.surfaceSolidDefaultActive = exports.surfaceSolidDefaultHover = exports.surfaceSolidCard = exports.surfaceSolidCardActive = exports.surfaceSolidCardHover = exports.surfaceSolidTertiary = exports.surfaceSolidTertiaryActive = exports.surfaceSolidTertiaryHover = exports.surfaceSolidSecondary = exports.surfaceSolidSecondaryActive = exports.surfaceSolidSecondaryHover = exports.surfaceSolidPrimary = exports.surfaceSolidPrimaryActive = exports.surfaceSolidPrimaryHover = exports.inverseTextParagraph = exports.inverseTextParagraphActive = exports.inverseTextParagraphHover = exports.inverseTextTertiary = exports.inverseTextTertiaryActive = exports.inverseTextTertiaryHover = exports.inverseTextSecondary = exports.inverseTextSecondaryActive = exports.inverseTextSecondaryHover = exports.inverseTextPrimary = exports.inverseTextPrimaryActive = exports.inverseTextPrimaryHover = exports.onLightTextAccentMain = exports.onLightTextAccentMainActive = exports.onLightTextAccentMainHover = void 0;
exports.onDarkSurfaceTransparentSecondaryHover = exports.onDarkSurfaceTransparentPrimary = exports.onDarkSurfaceTransparentPrimaryActive = exports.onDarkSurfaceTransparentPrimaryHover = exports.onDarkSurfaceSolidDefault = exports.onDarkSurfaceSolidDefaultActive = exports.onDarkSurfaceSolidDefaultHover = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidCardActive = exports.onDarkSurfaceSolidCardHover = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidTertiaryActive = exports.onDarkSurfaceSolidTertiaryHover = exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidSecondaryActive = exports.onDarkSurfaceSolidSecondaryHover = exports.onDarkSurfaceSolidPrimary = exports.onDarkSurfaceSolidPrimaryActive = exports.onDarkSurfaceSolidPrimaryHover = exports.surfaceGradientMain = exports.surfaceGradientMainActive = exports.surfaceGradientMainHover = exports.surfaceAccentMain = exports.surfaceAccentMainActive = exports.surfaceAccentMainHover = exports.surfaceSkeleton = exports.surfaceSkeletonActive = exports.surfaceSkeletonHover = exports.surfaceSpeechBubbleReceived = exports.surfaceSpeechBubbleReceivedActive = exports.surfaceSpeechBubbleReceivedHover = exports.surfaceSpeechBubbleSent = exports.surfaceSpeechBubbleSentActive = exports.surfaceSpeechBubbleSentHover = exports.surfaceAccentBrand = exports.surfaceAccentBrandActive = exports.surfaceAccentBrandHover = exports.surfaceAccentB2E = exports.surfaceAccentB2EActive = exports.surfaceAccentB2EHover = exports.surfaceAccentJoy = exports.surfaceAccentJoyActive = exports.surfaceAccentJoyHover = exports.surfaceAccentAthena = exports.surfaceAccentAthenaActive = exports.surfaceAccentAthenaHover = exports.surfaceNegative = exports.surfaceNegativeActive = exports.surfaceNegativeHover = exports.surfaceWarning = void 0;
exports.onLightSurfaceSolidDefault = exports.onLightSurfaceSolidDefaultActive = exports.onLightSurfaceSolidDefaultHover = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidCardActive = exports.onLightSurfaceSolidCardHover = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidTertiaryActive = exports.onLightSurfaceSolidTertiaryHover = exports.onLightSurfaceSolidSecondary = exports.onLightSurfaceSolidSecondaryActive = exports.onLightSurfaceSolidSecondaryHover = exports.onLightSurfaceSolidPrimary = exports.onLightSurfaceSolidPrimaryActive = exports.onLightSurfaceSolidPrimaryHover = exports.onDarkSurfaceGradientMain = exports.onDarkSurfaceGradientMainActive = exports.onDarkSurfaceGradientMainHover = exports.onDarkSurfaceAccentMain = exports.onDarkSurfaceAccentMainActive = exports.onDarkSurfaceAccentMainHover = exports.onDarkSurfaceAccentBrand = exports.onDarkSurfaceAccentBrandActive = exports.onDarkSurfaceAccentBrandHover = exports.onDarkSurfaceAccentB2E = exports.onDarkSurfaceAccentB2EActive = exports.onDarkSurfaceAccentB2EHover = exports.onDarkSurfaceAccentJoy = exports.onDarkSurfaceAccentJoyActive = exports.onDarkSurfaceAccentJoyHover = exports.onDarkSurfaceAccentAthena = exports.onDarkSurfaceAccentAthenaActive = exports.onDarkSurfaceAccentAthenaHover = exports.onDarkSurfaceNegative = exports.onDarkSurfaceNegativeActive = exports.onDarkSurfaceNegativeHover = exports.onDarkSurfaceWarning = exports.onDarkSurfaceWarningActive = exports.onDarkSurfaceWarningHover = exports.onDarkSurfacePositive = exports.onDarkSurfacePositiveActive = exports.onDarkSurfacePositiveHover = exports.onDarkSurfaceTransparentCard = exports.onDarkSurfaceTransparentCardActive = exports.onDarkSurfaceTransparentCardHover = exports.onDarkSurfaceTransparentTertiary = exports.onDarkSurfaceTransparentTertiaryActive = exports.onDarkSurfaceTransparentTertiaryHover = exports.onDarkSurfaceTransparentSecondary = exports.onDarkSurfaceTransparentSecondaryActive = void 0;
exports.backgroundMobileAssistantMain = exports.backgroundMobileAssistantBrand = exports.backgroundMobileAssistantB2E = exports.backgroundMobileAssistantJoy = exports.backgroundMobileAssistantAthena = exports.backgroundModalSheet = exports.backgroundSecondary = exports.backgroundPrimary = exports.onLightSurfaceGradientMain = exports.onLightSurfaceGradientMainActive = exports.onLightSurfaceGradientMainHover = exports.onLightSurfaceAccentMain = exports.onLightSurfaceAccentMainActive = exports.onLightSurfaceAccentMainHover = exports.onLightSurfaceAccentBrand = exports.onLightSurfaceAccentBrandActive = exports.onLightSurfaceAccentBrandHover = exports.onLightSurfaceAccentB2E = exports.onLightSurfaceAccentB2EActive = exports.onLightSurfaceAccentB2EHover = exports.onLightSurfaceAccentJoy = exports.onLightSurfaceAccentJoyActive = exports.onLightSurfaceAccentJoyHover = exports.onLightSurfaceAccentAthena = exports.onLightSurfaceAccentAthenaActive = exports.onLightSurfaceAccentAthenaHover = exports.onLightSurfaceNegative = exports.onLightSurfaceNegativeActive = exports.onLightSurfaceNegativeHover = exports.onLightSurfaceWarning = exports.onLightSurfaceWarningActive = exports.onLightSurfaceWarningHover = exports.onLightSurfacePositive = exports.onLightSurfacePositiveActive = exports.onLightSurfacePositiveHover = exports.onLightSurfaceTransparentCard = exports.onLightSurfaceTransparentCardActive = exports.onLightSurfaceTransparentCardHover = exports.onLightSurfaceTransparentDeep = exports.onLightSurfaceTransparentDeepActive = exports.onLightSurfaceTransparentDeepHover = exports.onLightSurfaceTransparentTertiary = exports.onLightSurfaceTransparentTertiaryActive = exports.onLightSurfaceTransparentTertiaryHover = exports.onLightSurfaceTransparentSecondary = exports.onLightSurfaceTransparentSecondaryActive = exports.onLightSurfaceTransparentSecondaryHover = exports.onLightSurfaceTransparentPrimary = exports.onLightSurfaceTransparentPrimaryActive = exports.onLightSurfaceTransparentPrimaryHover = void 0;
exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonSecondary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.critical = exports.warning = exports.success = exports.background = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackSecondary = exports.buttonBlack = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.skeletonGradientLighter = exports.skeletonGradient = exports.onDarkOverlayBlur = exports.onDarkOverlayHard = exports.onDarkOverlaySoft = void 0;
/** Основной цвет текста */
exports.textPrimaryHover = 'var(--text-primary-hover, #FFFFFF93)';
/** Основной цвет текста */
exports.textPrimaryActive = 'var(--text-primary-active, #FFFFFFC4)';
/** Основной цвет текста */
exports.textPrimary = 'var(--text-primary, #FFFFFFF5)';
/** Вторичный цвет текста */
exports.textSecondaryHover = 'var(--text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста */
exports.textSecondaryActive = 'var(--text-secondary-active, #FFFFFFAB)';
/** Вторичный цвет текста */
exports.textSecondary = 'var(--text-secondary, #FFFFFF8F)';
/** Третичный цвет текста */
exports.textTertiaryHover = 'var(--text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста */
exports.textTertiaryActive = 'var(--text-tertiary-active, #FFFFFF56)';
/** Третичный цвет текста */
exports.textTertiary = 'var(--text-tertiary, #FFFFFF47)';
/** Сплошной наборный текст */
exports.textParagraphHover = 'var(--text-paragraph-hover, #FFFFFF7A)';
/** Сплошной наборный текст */
exports.textParagraphActive = 'var(--text-paragraph-active, #FFFFFFA3)';
/** Сплошной наборный текст */
exports.textParagraph = 'var(--text-paragraph, #FFFFFFCC)';
/** Цвет успеха */
exports.textPositiveHover = 'var(--text-positive-hover, #2ACB47FF)';
/** Цвет успеха */
exports.textPositiveActive = 'var(--text-positive-active, #1F9835FF)';
/** Цвет успеха */
exports.textPositive = 'var(--text-positive, #24B23E)';
/** Цвет предупреждения */
exports.textWarningHover = 'var(--text-warning-hover, #FF8442FF)';
/** Цвет предупреждения */
exports.textWarningActive = 'var(--text-warning-active, #FF5D05FF)';
/** Цвет предупреждения */
exports.textWarning = 'var(--text-warning, #FF7024)';
/** Цвет ошибки */
exports.textNegativeHover = 'var(--text-negative-hover, #FF475AFF)';
/** Цвет ошибки */
exports.textNegativeActive = 'var(--text-negative-active, #FF0A23FF)';
/** Цвет ошибки */
exports.textNegative = 'var(--text-negative, #FF293E)';
/** Акцентный цвет Афины */
exports.textAccentAthenaHover = 'var(--text-accent-athena-hover, #66BCF5FF)';
/** Акцентный цвет Афины */
exports.textAccentAthenaActive = 'var(--text-accent-athena-active, #0E8ADDFF)';
/** Акцентный цвет Афины */
exports.textAccentAthena = 'var(--text-accent-athena, #199AF0)';
/** Акцентный цвет Джой */
exports.textAccentJoyHover = 'var(--text-accent-joy-hover, #E4BDFFFF)';
/** Акцентный цвет Джой */
exports.textAccentJoyActive = 'var(--text-accent-joy-active, #B84DFFFF)';
/** Акцентный цвет Джой */
exports.textAccentJoy = 'var(--text-accent-joy, #C46BFF)';
/** Акцентный цвет B2E-ассистента */
exports.textAccentB2EHover = 'var(--text-accent-b2-e-hover, #2ACB47FF)';
/** Акцентный цвет B2E-ассистента */
exports.textAccentB2EActive = 'var(--text-accent-b2-e-active, #1F9835FF)';
/** Акцентный цвет B2E-ассистента */
exports.textAccentB2E = 'var(--text-accent-b2-e, #24B23E)';
/** Акцентный цвет бренда */
exports.textAccentBrandHover = 'var(--text-accent-brand-hover, #A3C3FFFF)';
/** Акцентный цвет бренда */
exports.textAccentBrandActive = 'var(--text-accent-brand-active, #337AFFFF)';
/** Акцентный цвет бренда */
exports.textAccentBrand = 'var(--text-accent-brand, #528EFF)';
/** Градиент Афины */
exports.textGradientAthenaHover = 'var(--text-gradient-athena-hover, #CCCCCCFF)';
/** Градиент Афины */
exports.textGradientAthenaActive = 'var(--text-gradient-athena-active, #E6E6E6FF)';
/** Градиент Афины */
exports.textGradientAthena = 'var(--text-gradient-athena, linear-gradient(93.97deg, #52BAFF 6.49%, #ADADFF 93.51%))';
/** Градиент Джой */
exports.textGradientJoyHover = 'var(--text-gradient-joy-hover, #CCCCCCFF)';
/** Градиент Джой */
exports.textGradientJoyActive = 'var(--text-gradient-joy-active, #E6E6E6FF)';
/** Градиент Джой */
exports.textGradientJoy = 'var(--text-gradient-joy, linear-gradient(93.97deg, #FD9C68 6.49%, #F391E2 93.51%))';
/** Градиент B2E-ассистента */
exports.textGradientB2EHover = 'var(--text-gradient-b2-e-hover, #CCCCCCFF)';
/** Градиент B2E-ассистента */
exports.textGradientB2EActive = 'var(--text-gradient-b2-e-active, #E6E6E6FF)';
/** Градиент B2E-ассистента */
exports.textGradientB2E = 'var(--text-gradient-b2-e, linear-gradient(89.83deg, #24B23E 0%, #1CC9C9 100%))';
/** Градиент бренда */
exports.textGradientBrandHover = 'var(--text-gradient-brand-hover, #CCCCCCFF)';
/** Градиент бренда */
exports.textGradientBrandActive = 'var(--text-gradient-brand-active, #E6E6E6FF)';
/** Градиент бренда */
exports.textGradientBrand = 'var(--text-gradient-brand, linear-gradient(89.83deg, #528EFF 0%, #04C6C9 100%))';
/** Градиент */
exports.textGradientMainHover = 'var(--text-gradient-main-hover, #CCCCCCFF)';
/** Градиент */
exports.textGradientMainActive = 'var(--text-gradient-main-active, #E6E6E6FF)';
/** Градиент */
exports.textGradientMain = 'var(--text-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #1CC9C9 100%))';
/** Акцентный цвет */
exports.textAccentMainHover = 'var(--text-accent-main-hover, #2ACB47FF)';
/** Акцентный цвет */
exports.textAccentMainActive = 'var(--text-accent-main-active, #1F9835FF)';
/** Акцентный цвет */
exports.textAccentMain = 'var(--text-accent-main, #24B23E)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimaryHover = 'var(--on-dark-text-primary-hover, #FFFFFF93)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimaryActive = 'var(--on-dark-text-primary-active, #FFFFFFC4)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = 'var(--on-dark-text-primary, #FFFFFFF5)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondaryHover = 'var(--on-dark-text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondaryActive = 'var(--on-dark-text-secondary-active, #FFFFFFAB)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = 'var(--on-dark-text-secondary, #FFFFFF8F)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiaryHover = 'var(--on-dark-text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiaryActive = 'var(--on-dark-text-tertiary-active, #FFFFFF56)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = 'var(--on-dark-text-tertiary, #FFFFFF47)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraphHover = 'var(--on-dark-text-paragraph-hover, #FFFFFF7A)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraphActive = 'var(--on-dark-text-paragraph-active, #FFFFFFA3)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraph = 'var(--on-dark-text-paragraph, #FFFFFFCC)';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositiveHover = 'var(--on-dark-text-positive-hover, #2ACB47FF)';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositiveActive = 'var(--on-dark-text-positive-active, #1F9835FF)';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositive = 'var(--on-dark-text-positive, #24B23E)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarningHover = 'var(--on-dark-text-warning-hover, #FF8442FF)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarningActive = 'var(--on-dark-text-warning-active, #FF5D05FF)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarning = 'var(--on-dark-text-warning, #FF7024)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegativeHover = 'var(--on-dark-text-negative-hover, #FF475AFF)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegativeActive = 'var(--on-dark-text-negative-active, #FF0A23FF)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegative = 'var(--on-dark-text-negative, #FF293E)';
/** Акцентный цвет Афины на темном */
exports.onDarkTextAccentAthenaHover = 'var(--on-dark-text-accent-athena-hover, #66BCF5FF)';
/** Акцентный цвет Афины на темном */
exports.onDarkTextAccentAthenaActive = 'var(--on-dark-text-accent-athena-active, #0E8ADDFF)';
/** Акцентный цвет Афины на темном */
exports.onDarkTextAccentAthena = 'var(--on-dark-text-accent-athena, #199AF0)';
/** Акцентный цвет Джой на темном фоне */
exports.onDarkTextAccentJoyHover = 'var(--on-dark-text-accent-joy-hover, #E4BDFFFF)';
/** Акцентный цвет Джой на темном фоне */
exports.onDarkTextAccentJoyActive = 'var(--on-dark-text-accent-joy-active, #B84DFFFF)';
/** Акцентный цвет Джой на темном фоне */
exports.onDarkTextAccentJoy = 'var(--on-dark-text-accent-joy, #C46BFF)';
/** Акцентный цвет B2E-ассистента на темном фоне */
exports.onDarkTextAccentB2EHover = 'var(--on-dark-text-accent-b2-e-hover, #2ACB47FF)';
/** Акцентный цвет B2E-ассистента на темном фоне */
exports.onDarkTextAccentB2EActive = 'var(--on-dark-text-accent-b2-e-active, #1F9835FF)';
/** Акцентный цвет B2E-ассистента на темном фоне */
exports.onDarkTextAccentB2E = 'var(--on-dark-text-accent-b2-e, #24B23E)';
/** Акцентный цвет бренда на темном фоне */
exports.onDarkTextAccentBrandHover = 'var(--on-dark-text-accent-brand-hover, #A3C3FFFF)';
/** Акцентный цвет бренда на темном фоне */
exports.onDarkTextAccentBrandActive = 'var(--on-dark-text-accent-brand-active, #337AFFFF)';
/** Акцентный цвет бренда на темном фоне */
exports.onDarkTextAccentBrand = 'var(--on-dark-text-accent-brand, #528EFF)';
/** Градиент Афины на темном фоне */
exports.onDarkTextGradientAthenaHover = 'var(--on-dark-text-gradient-athena-hover, #CCCCCCFF)';
/** Градиент Афины на темном фоне */
exports.onDarkTextGradientAthenaActive = 'var(--on-dark-text-gradient-athena-active, #E6E6E6FF)';
/** Градиент Афины на темном фоне */
exports.onDarkTextGradientAthena = 'var(--on-dark-text-gradient-athena, linear-gradient(93.97deg, #52BAFF 6.49%, #ADADFF 93.51%))';
/** Градиент Джой на темном фоне */
exports.onDarkTextGradientJoyHover = 'var(--on-dark-text-gradient-joy-hover, #CCCCCCFF)';
/** Градиент Джой на темном фоне */
exports.onDarkTextGradientJoyActive = 'var(--on-dark-text-gradient-joy-active, #E6E6E6FF)';
/** Градиент Джой на темном фоне */
exports.onDarkTextGradientJoy = 'var(--on-dark-text-gradient-joy, linear-gradient(93.97deg, #FD9C68 6.49%, #F391E2 93.51%))';
/** Градиент B2E-ассистента на темном фоне */
exports.onDarkTextGradientB2EHover = 'var(--on-dark-text-gradient-b2-e-hover, #CCCCCCFF)';
/** Градиент B2E-ассистента на темном фоне */
exports.onDarkTextGradientB2EActive = 'var(--on-dark-text-gradient-b2-e-active, #E6E6E6FF)';
/** Градиент B2E-ассистента на темном фоне */
exports.onDarkTextGradientB2E = 'var(--on-dark-text-gradient-b2-e, linear-gradient(89.83deg, #24B23E 0%, #1CC9C9 100%))';
/** Градиент бренда на темном фоне */
exports.onDarkTextGradientBrandHover = 'var(--on-dark-text-gradient-brand-hover, #CCCCCCFF)';
/** Градиент бренда на темном фоне */
exports.onDarkTextGradientBrandActive = 'var(--on-dark-text-gradient-brand-active, #E6E6E6FF)';
/** Градиент бренда на темном фоне */
exports.onDarkTextGradientBrand = 'var(--on-dark-text-gradient-brand, linear-gradient(89.83deg, #528EFF 0%, #04C6C9 100%))';
/** Градиент на темном фоне */
exports.onDarkTextGradientMainHover = 'var(--on-dark-text-gradient-main-hover, #CCCCCCFF)';
/** Градиент на темном фоне */
exports.onDarkTextGradientMainActive = 'var(--on-dark-text-gradient-main-active, #E6E6E6FF)';
/** Градиент на темном фоне */
exports.onDarkTextGradientMain = 'var(--on-dark-text-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #1CC9C9 100%))';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccentMainHover = 'var(--on-dark-text-accent-main-hover, #2ACB47FF)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccentMainActive = 'var(--on-dark-text-accent-main-active, #1F9835FF)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccentMain = 'var(--on-dark-text-accent-main, #24B23E)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimaryHover = 'var(--on-light-text-primary-hover, #08080893)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimaryActive = 'var(--on-light-text-primary-active, #080808C4)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = 'var(--on-light-text-primary, rgba(8, 8, 8, 0.96))';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondaryHover = 'var(--on-light-text-secondary-hover, #080808FF)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondaryActive = 'var(--on-light-text-secondary-active, #080808AB)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = 'var(--on-light-text-secondary, rgba(8, 8, 8, 0.56))';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiaryHover = 'var(--on-light-text-tertiary-hover, #080808FF)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiaryActive = 'var(--on-light-text-tertiary-active, #08080856)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = 'var(--on-light-text-tertiary, #08080847)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraphHover = 'var(--on-light-text-paragraph-hover, #0808087A)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraphActive = 'var(--on-light-text-paragraph-active, #080808A3)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraph = 'var(--on-light-text-paragraph, rgba(8, 8, 8, 0.8))';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositiveHover = 'var(--on-light-text-positive-hover, #1EB83AFF)';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositiveActive = 'var(--on-light-text-positive-active, #15842AFF)';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositive = 'var(--on-light-text-positive, #1A9E32)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarningHover = 'var(--on-light-text-warning-hover, #FB7223FF)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarningActive = 'var(--on-light-text-warning-active, #DC5304FF)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarning = 'var(--on-light-text-warning, #FA5F05)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegativeHover = 'var(--on-light-text-negative-hover, #FF475AFF)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegativeActive = 'var(--on-light-text-negative-active, #FF0A23FF)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegative = 'var(--on-light-text-negative, #FF293E)';
/** Акцентный цвет Афины на светлом фоне */
exports.onLightTextAccentAthenaHover = 'var(--on-light-text-accent-athena-hover, #50B1F2FF)';
/** Акцентный цвет Афины на светлом фоне */
exports.onLightTextAccentAthenaActive = 'var(--on-light-text-accent-athena-active, #0F7BC2FF)';
/** Акцентный цвет Афины на светлом фоне */
exports.onLightTextAccentAthena = 'var(--on-light-text-accent-athena, #118CDF)';
/** Акцентный цвет Джой на светлом фоне */
exports.onLightTextAccentJoyHover = 'var(--on-light-text-accent-joy-hover, #D7A4F9FF)';
/** Акцентный цвет Джой на светлом фоне */
exports.onLightTextAccentJoyActive = 'var(--on-light-text-accent-joy-active, #A93BF1FF)';
/** Акцентный цвет Джой на светлом фоне */
exports.onLightTextAccentJoy = 'var(--on-light-text-accent-joy, #B559F3)';
/** Акцентный цвет B2E-ассистента на светлом фоне */
exports.onLightTextAccentB2EHover = 'var(--on-light-text-accent-b2-e-hover, #1EB83AFF)';
/** Акцентный цвет B2E-ассистента на светлом фоне */
exports.onLightTextAccentB2EActive = 'var(--on-light-text-accent-b2-e-active, #15842AFF)';
/** Акцентный цвет B2E-ассистента на светлом фоне */
exports.onLightTextAccentB2E = 'var(--on-light-text-accent-b2-e, #1A9E32)';
/** Акцентный цвет бренда на светлом фоне */
exports.onLightTextAccentBrandHover = 'var(--on-light-text-accent-brand-hover, #90B6FEFF)';
/** Акцентный цвет бренда на светлом фоне */
exports.onLightTextAccentBrandActive = 'var(--on-light-text-accent-brand-active, #216EFDFF)';
/** Акцентный цвет бренда на светлом фоне */
exports.onLightTextAccentBrand = 'var(--on-light-text-accent-brand, #3F81FD)';
/** Градиент Афины на светлом фоне */
exports.onLightTextGradientAthenaHover = 'var(--on-light-text-gradient-athena-hover, #CCCCCCFF)';
/** Градиент Афины на светлом фоне */
exports.onLightTextGradientAthenaActive = 'var(--on-light-text-gradient-athena-active, #E6E6E6FF)';
/** Градиент Афины на светлом фоне */
exports.onLightTextGradientAthena = 'var(--on-light-text-gradient-athena, linear-gradient(89.83deg, #118CDF 0%, #7A7AFF 100%))';
/** Градиент Джой на светлом фоне */
exports.onLightTextGradientJoyHover = 'var(--on-light-text-gradient-joy-hover, #CCCCCCFF)';
/** Градиент Джой на светлом фоне */
exports.onLightTextGradientJoyActive = 'var(--on-light-text-gradient-joy-active, #E6E6E6FF)';
/** Градиент Джой на светлом фоне */
exports.onLightTextGradientJoy = 'var(--on-light-text-gradient-joy, linear-gradient(89.83deg, #FCA349 0%, #FF52C5 100%))';
/** Градиент B2E-ассистента на светлом фоне */
exports.onLightTextGradientB2EHover = 'var(--on-light-text-gradient-b2-e-hover, #CCCCCCFF)';
/** Градиент B2E-ассистента на светлом фоне */
exports.onLightTextGradientB2EActive = 'var(--on-light-text-gradient-b2-e-active, #E6E6E6FF)';
/** Градиент B2E-ассистента на светлом фоне */
exports.onLightTextGradientB2E = 'var(--on-light-text-gradient-b2-e, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Акцентный цвет бренда на светлом фоне */
exports.onLightTextGradientBrandHover = 'var(--on-light-text-gradient-brand-hover, #CCCCCCFF)';
/** Акцентный цвет бренда на светлом фоне */
exports.onLightTextGradientBrandActive = 'var(--on-light-text-gradient-brand-active, #E6E6E6FF)';
/** Акцентный цвет бренда на светлом фоне */
exports.onLightTextGradientBrand = 'var(--on-light-text-gradient-brand, linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%))';
/** Градиент на светлом фоне */
exports.onLightTextGradientMainHover = 'var(--on-light-text-gradient-main-hover, #CCCCCCFF)';
/** Градиент на светлом фоне */
exports.onLightTextGradientMainActive = 'var(--on-light-text-gradient-main-active, #E6E6E6FF)';
/** Градиент на светлом фоне */
exports.onLightTextGradientMain = 'var(--on-light-text-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccentMainHover = 'var(--on-light-text-accent-main-hover, #1EB83AFF)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccentMainActive = 'var(--on-light-text-accent-main-active, #15842AFF)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccentMain = 'var(--on-light-text-accent-main, #1A9E32)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimaryHover = 'var(--inverse-text-primary-hover, #08080893)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimaryActive = 'var(--inverse-text-primary-active, #080808C4)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimary = 'var(--inverse-text-primary, rgba(8, 8, 8, 0.96))';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondaryHover = 'var(--inverse-text-secondary-hover, #080808FF)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondaryActive = 'var(--inverse-text-secondary-active, #080808AB)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondary = 'var(--inverse-text-secondary, rgba(8, 8, 8, 0.56))';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiaryHover = 'var(--inverse-text-tertiary-hover, #080808FF)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiaryActive = 'var(--inverse-text-tertiary-active, #08080856)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiary = 'var(--inverse-text-tertiary, #08080847)';
/** Инвертированный cплошной наборный текст */
exports.inverseTextParagraphHover = 'var(--inverse-text-paragraph-hover, #0808087A)';
/** Инвертированный cплошной наборный текст */
exports.inverseTextParagraphActive = 'var(--inverse-text-paragraph-active, #080808A3)';
/** Инвертированный cплошной наборный текст */
exports.inverseTextParagraph = 'var(--inverse-text-paragraph, rgba(8, 8, 8, 0.8))';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimaryHover = 'var(--surface-solid-primary-hover, #1C1C1CFF)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimaryActive = 'var(--surface-solid-primary-active, #121212FF)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimary = 'var(--surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondaryHover = 'var(--surface-solid-secondary-hover, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondaryActive = 'var(--surface-solid-secondary-active, #212121FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondary = 'var(--surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiaryHover = 'var(--surface-solid-tertiary-hover, #404040FF)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiaryActive = 'var(--surface-solid-tertiary-active, #303030FF)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = 'var(--surface-solid-tertiary, #363636)';
/** Основной фон для карточек */
exports.surfaceSolidCardHover = 'var(--surface-solid-card-hover, #1C1C1CFF)';
/** Основной фон для карточек */
exports.surfaceSolidCardActive = 'var(--surface-solid-card-active, #121212FF)';
/** Основной фон для карточек */
exports.surfaceSolidCard = 'var(--surface-solid-card, #171717)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefaultHover = 'var(--surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefaultActive = 'var(--surface-solid-default-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefault = 'var(--surface-solid-default, #FFFFFF)';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimaryHover = 'var(--surface-transparent-primary-hover, #FFFFFF1F)';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimaryActive = 'var(--surface-transparent-primary-active, #FFFFFF0A)';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimary = 'var(--surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondaryHover = 'var(--surface-transparent-secondary-hover, #FFFFFF3D)';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondaryActive = 'var(--surface-transparent-secondary-active, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondary = 'var(--surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiaryHover = 'var(--surface-transparent-tertiary-hover, #FFFFFF52)';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiaryActive = 'var(--surface-transparent-tertiary-active, #FFFFFF24)';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCardHover = 'var(--surface-transparent-card-hover, #FFFFFF1F)';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCardActive = 'var(--surface-transparent-card-active, #FFFFFF0A)';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCard = 'var(--surface-transparent-card, rgba(255,255,255,0.06))';
/** Фон поверхности/контрола без заливки */
exports.surfaceClearHover = 'var(--surface-clear-hover, #00000000)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClearActive = 'var(--surface-clear-active, #00000000)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClear = 'var(--surface-clear, rgba(0, 0, 0, 0))';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositiveHover = 'var(--surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositiveActive = 'var(--surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositive = 'var(--surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarningHover = 'var(--surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarningActive = 'var(--surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarning = 'var(--surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeHover = 'var(--surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeActive = 'var(--surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = 'var(--surface-negative, #FF293E)';
/** Акцентный цвет поверхности/контролла Афины */
exports.surfaceAccentAthenaHover = 'var(--surface-accent-athena-hover, #1798EEFF)';
/** Акцентный цвет поверхности/контролла Афины */
exports.surfaceAccentAthenaActive = 'var(--surface-accent-athena-active, #1086D5FF)';
/** Акцентный цвет поверхности/контролла Афины */
exports.surfaceAccentAthena = 'var(--surface-accent-athena, #118CDF)';
/** Акцентный цвет поверхности/контролла Джой */
exports.surfaceAccentJoyHover = 'var(--surface-accent-joy-hover, #C275F5FF)';
/** Акцентный цвет поверхности/контролла Джой */
exports.surfaceAccentJoyActive = 'var(--surface-accent-joy-active, #B14FF3FF)';
/** Акцентный цвет поверхности/контролла Джой */
exports.surfaceAccentJoy = 'var(--surface-accent-joy, #B559F3)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
exports.surfaceAccentB2EHover = 'var(--surface-accent-b2-e-hover, #1DAF37FF)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
exports.surfaceAccentB2EActive = 'var(--surface-accent-b2-e-active, #18952FFF)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
exports.surfaceAccentB2E = 'var(--surface-accent-b2-e, #1A9E32)';
/** Акцентный цвет поверхности/контролла бренда */
exports.surfaceAccentBrandHover = 'var(--surface-accent-brand-hover, #5D95FDFF)';
/** Акцентный цвет поверхности/контролла бренда */
exports.surfaceAccentBrandActive = 'var(--surface-accent-brand-active, #357BFDFF)';
/** Акцентный цвет поверхности/контролла бренда */
exports.surfaceAccentBrand = 'var(--surface-accent-brand, #3F81FD)';
/** Цвет баббла сообщения от пользователя */
exports.surfaceSpeechBubbleSentHover = 'var(--surface-speech-bubble-sent-hover, #08080866)';
/** Цвет баббла сообщения от пользователя */
exports.surfaceSpeechBubbleSentActive = 'var(--surface-speech-bubble-sent-active, #08080838)';
/** Цвет баббла сообщения от пользователя */
exports.surfaceSpeechBubbleSent = 'var(--surface-speech-bubble-sent, #08080847)';
/** Цвет баббла сообщения от ассистента */
exports.surfaceSpeechBubbleReceivedHover = 'var(--surface-speech-bubble-received-hover, #FFFFFF1F)';
/** Цвет баббла сообщения от ассистента */
exports.surfaceSpeechBubbleReceivedActive = 'var(--surface-speech-bubble-received-active, #FFFFFF0A)';
/** Цвет баббла сообщения от ассистента */
exports.surfaceSpeechBubbleReceived = 'var(--surface-speech-bubble-received, #FFFFFF0F)';
/** Фон для скелетона */
exports.surfaceSkeletonHover = 'var(--surface-skeleton-hover, #FFFFFFFF)';
/** Фон для скелетона */
exports.surfaceSkeletonActive = 'var(--surface-skeleton-active, #FFFFFFFF)';
/** Фон для скелетона */
exports.surfaceSkeleton = 'var(--surface-skeleton, linear-gradient(90deg, rgba(255, 255, 255, 0.00) 0%, rgba(255, 255, 255, 0.01) 17.19%, rgba(255, 255, 255, 0.05) 40.24%, rgba(255, 255, 255, 0.06) 51.17%, rgba(255, 255, 255, 0.05) 62.12%, rgba(255, 255, 255, 0.01) 82.81%, rgba(255, 255, 255, 0.00) 100%))';
/** Акцентный цвет поверхности/контролла */
exports.surfaceAccentMainHover = 'var(--surface-accent-main-hover, #1DAF37FF)';
/** Акцентный цвет поверхности/контролла */
exports.surfaceAccentMainActive = 'var(--surface-accent-main-active, #18952FFF)';
/** Акцентный цвет поверхности/контролла */
exports.surfaceAccentMain = 'var(--surface-accent-main, #1A9E32)';
exports.surfaceGradientMainHover = 'var(--surface-gradient-main-hover, #FFFFFFFF)';
exports.surfaceGradientMainActive = 'var(--surface-gradient-main-active, #FFFFFFFF)';
exports.surfaceGradientMain = 'var(--surface-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimaryHover = 'var(--on-dark-surface-solid-primary-hover, #1C1C1CFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimaryActive = 'var(--on-dark-surface-solid-primary-active, #121212FF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности/контрола на темном */
exports.onDarkSurfaceSolidSecondaryHover = 'var(--on-dark-surface-solid-secondary-hover, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном */
exports.onDarkSurfaceSolidSecondaryActive = 'var(--on-dark-surface-solid-secondary-active, #212121FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном */
exports.onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiaryHover = 'var(--on-dark-surface-solid-tertiary-hover, #404040FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiaryActive = 'var(--on-dark-surface-solid-tertiary-active, #303030FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #363636)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCardHover = 'var(--on-dark-surface-solid-card-hover, #1C1C1CFF)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCardActive = 'var(--on-dark-surface-solid-card-active, #121212FF)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #171717)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefaultHover = 'var(--on-dark-surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefaultActive = 'var(--on-dark-surface-solid-default-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #FFFFFF)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimaryHover = 'var(--on-dark-surface-transparent-primary-hover, #FFFFFF1F)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimaryActive = 'var(--on-dark-surface-transparent-primary-active, #FFFFFF0A)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondaryHover = 'var(--on-dark-surface-transparent-secondary-hover, #FFFFFF3D)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondaryActive = 'var(--on-dark-surface-transparent-secondary-active, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiaryHover = 'var(--on-dark-surface-transparent-tertiary-hover, #FFFFFF52)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiaryActive = 'var(--on-dark-surface-transparent-tertiary-active, #FFFFFF24)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCardHover = 'var(--on-dark-surface-transparent-card-hover, #FFFFFF1F)';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCardActive = 'var(--on-dark-surface-transparent-card-active, #FFFFFF0A)';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, rgba(255,255,255,0.06))';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositiveHover = 'var(--on-dark-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositiveActive = 'var(--on-dark-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositive = 'var(--on-dark-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение на темном */
exports.onDarkSurfaceWarningHover = 'var(--on-dark-surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение на темном */
exports.onDarkSurfaceWarningActive = 'var(--on-dark-surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола предупреждение на темном */
exports.onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка на темном */
exports.onDarkSurfaceNegativeHover = 'var(--on-dark-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на темном */
exports.onDarkSurfaceNegativeActive = 'var(--on-dark-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка на темном */
exports.onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293E)';
/** Акцентный цвет поверхности/контролла Афины */
exports.onDarkSurfaceAccentAthenaHover = 'var(--on-dark-surface-accent-athena-hover, #1798EEFF)';
/** Акцентный цвет поверхности/контролла Афины */
exports.onDarkSurfaceAccentAthenaActive = 'var(--on-dark-surface-accent-athena-active, #1086D5FF)';
/** Акцентный цвет поверхности/контролла Афины */
exports.onDarkSurfaceAccentAthena = 'var(--on-dark-surface-accent-athena, #118CDF)';
/** Акцентный цвет поверхности/контролла Джой */
exports.onDarkSurfaceAccentJoyHover = 'var(--on-dark-surface-accent-joy-hover, #C275F5FF)';
/** Акцентный цвет поверхности/контролла Джой */
exports.onDarkSurfaceAccentJoyActive = 'var(--on-dark-surface-accent-joy-active, #B14FF3FF)';
/** Акцентный цвет поверхности/контролла Джой */
exports.onDarkSurfaceAccentJoy = 'var(--on-dark-surface-accent-joy, #B559F3)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
exports.onDarkSurfaceAccentB2EHover = 'var(--on-dark-surface-accent-b2-e-hover, #1DAF37FF)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
exports.onDarkSurfaceAccentB2EActive = 'var(--on-dark-surface-accent-b2-e-active, #18952FFF)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
exports.onDarkSurfaceAccentB2E = 'var(--on-dark-surface-accent-b2-e, #1A9E32)';
/** Акцентный цвет поверхности/контролла бренда */
exports.onDarkSurfaceAccentBrandHover = 'var(--on-dark-surface-accent-brand-hover, #5D95FDFF)';
/** Акцентный цвет поверхности/контролла бренда */
exports.onDarkSurfaceAccentBrandActive = 'var(--on-dark-surface-accent-brand-active, #357BFDFF)';
/** Акцентный цвет поверхности/контролла бренда */
exports.onDarkSurfaceAccentBrand = 'var(--on-dark-surface-accent-brand, #3F81FD)';
/** Акцентный цвет поверхности/контролла */
exports.onDarkSurfaceAccentMainHover = 'var(--on-dark-surface-accent-main-hover, #1DAF37FF)';
/** Акцентный цвет поверхности/контролла */
exports.onDarkSurfaceAccentMainActive = 'var(--on-dark-surface-accent-main-active, #18952FFF)';
/** Акцентный цвет поверхности/контролла */
exports.onDarkSurfaceAccentMain = 'var(--on-dark-surface-accent-main, #1A9E32)';
exports.onDarkSurfaceGradientMainHover = 'var(--on-dark-surface-gradient-main-hover, #FFFFFFFF)';
exports.onDarkSurfaceGradientMainActive = 'var(--on-dark-surface-gradient-main-active, #FFFFFFFF)';
exports.onDarkSurfaceGradientMain = 'var(--on-dark-surface-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Основной непрозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceSolidPrimaryHover = 'var(--on-light-surface-solid-primary-hover, #FAFAFAFF)';
/** Основной непрозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceSolidPrimaryActive = 'var(--on-light-surface-solid-primary-active, #F0F0F0FF)';
/** Основной непрозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F5F5F5)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceSolidSecondaryHover = 'var(--on-light-surface-solid-secondary-hover, #F7F7F7FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceSolidSecondaryActive = 'var(--on-light-surface-solid-secondary-active, #E8E8E8FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #ECECEC)';
/** Третичный непрозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceSolidTertiaryHover = 'var(--on-light-surface-solid-tertiary-hover, #E8E8E8FF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceSolidTertiaryActive = 'var(--on-light-surface-solid-tertiary-active, #D9D9D9FF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #DDDDDD)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCardHover = 'var(--on-light-surface-solid-card-hover, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCardActive = 'var(--on-light-surface-solid-card-active, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefaultHover = 'var(--on-light-surface-solid-default-hover, #0D0D0DFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefaultActive = 'var(--on-light-surface-solid-default-active, #030303FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #080808)';
/** Основной прозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceTransparentPrimaryHover = 'var(--on-light-surface-transparent-primary-hover, #08080812)';
/** Основной прозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceTransparentPrimaryActive = 'var(--on-light-surface-transparent-primary-active, #08080803)';
/** Основной прозрачный фон поверхности/контрола на светлом */
exports.onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, #08080808)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondaryHover = 'var(--on-light-surface-transparent-secondary-hover, #0808081F)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondaryActive = 'var(--on-light-surface-transparent-secondary-active, #0808080A)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiaryHover = 'var(--on-light-surface-transparent-tertiary-hover, #0808083D)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiaryActive = 'var(--on-light-surface-transparent-tertiary-active, #0808080F)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Прозрачный цвет поверхности/контролла на светлом фоне */
exports.onLightSurfaceTransparentDeepHover = 'var(--on-light-surface-transparent-deep-hover, #080808C2)';
/** Прозрачный цвет поверхности/контролла на светлом фоне */
exports.onLightSurfaceTransparentDeepActive = 'var(--on-light-surface-transparent-deep-active, #08080894)';
/** Прозрачный цвет поверхности/контролла на светлом фоне */
exports.onLightSurfaceTransparentDeep = 'var(--on-light-surface-transparent-deep, #080808A2)';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCardHover = 'var(--on-light-surface-transparent-card-hover, #FFFFFFFF)';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCardActive = 'var(--on-light-surface-transparent-card-active, #FFFFFFFF)';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, #FFFFFFFF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositiveHover = 'var(--on-light-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositiveActive = 'var(--on-light-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositive = 'var(--on-light-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение на светлом */
exports.onLightSurfaceWarningHover = 'var(--on-light-surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение на светлом */
exports.onLightSurfaceWarningActive = 'var(--on-light-surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола предупреждение на светлом */
exports.onLightSurfaceWarning = 'var(--on-light-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegativeHover = 'var(--on-light-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegativeActive = 'var(--on-light-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegative = 'var(--on-light-surface-negative, #FF293E)';
/** Акцентный цвет поверхности/контролла Афины на светлом фоне */
exports.onLightSurfaceAccentAthenaHover = 'var(--on-light-surface-accent-athena-hover, #1798EEFF)';
/** Акцентный цвет поверхности/контролла Афины на светлом фоне */
exports.onLightSurfaceAccentAthenaActive = 'var(--on-light-surface-accent-athena-active, #1086D5FF)';
/** Акцентный цвет поверхности/контролла Афины на светлом фоне */
exports.onLightSurfaceAccentAthena = 'var(--on-light-surface-accent-athena, #118CDF)';
/** Акцентный цвет поверхности/контролла Джой на светлом */
exports.onLightSurfaceAccentJoyHover = 'var(--on-light-surface-accent-joy-hover, #C275F5FF)';
/** Акцентный цвет поверхности/контролла Джой на светлом */
exports.onLightSurfaceAccentJoyActive = 'var(--on-light-surface-accent-joy-active, #B14FF3FF)';
/** Акцентный цвет поверхности/контролла Джой на светлом */
exports.onLightSurfaceAccentJoy = 'var(--on-light-surface-accent-joy, #B559F3)';
/** Акцентный цвет поверхности B2E-ассистента на светлом */
exports.onLightSurfaceAccentB2EHover = 'var(--on-light-surface-accent-b2-e-hover, #1DAF37FF)';
/** Акцентный цвет поверхности B2E-ассистента на светлом */
exports.onLightSurfaceAccentB2EActive = 'var(--on-light-surface-accent-b2-e-active, #18952FFF)';
/** Акцентный цвет поверхности B2E-ассистента на светлом */
exports.onLightSurfaceAccentB2E = 'var(--on-light-surface-accent-b2-e, #1A9E32)';
/** Акцентный цвет поверхности/контролла бренда на светлом */
exports.onLightSurfaceAccentBrandHover = 'var(--on-light-surface-accent-brand-hover, #5D95FDFF)';
/** Акцентный цвет поверхности/контролла бренда на светлом */
exports.onLightSurfaceAccentBrandActive = 'var(--on-light-surface-accent-brand-active, #357BFDFF)';
/** Акцентный цвет поверхности/контролла бренда на светлом */
exports.onLightSurfaceAccentBrand = 'var(--on-light-surface-accent-brand, #3F81FD)';
/** Акцентный цвет поверхности/контролла на светлом фоне */
exports.onLightSurfaceAccentMainHover = 'var(--on-light-surface-accent-main-hover, #1DAF37FF)';
/** Акцентный цвет поверхности/контролла на светлом фоне */
exports.onLightSurfaceAccentMainActive = 'var(--on-light-surface-accent-main-active, #18952FFF)';
/** Акцентный цвет поверхности/контролла на светлом фоне */
exports.onLightSurfaceAccentMain = 'var(--on-light-surface-accent-main, #1A9E32)';
exports.onLightSurfaceGradientMainHover = 'var(--on-light-surface-gradient-main-hover, #FFFFFFFF)';
exports.onLightSurfaceGradientMainActive = 'var(--on-light-surface-gradient-main-active, #FFFFFFFF)';
exports.onLightSurfaceGradientMain = 'var(--on-light-surface-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Первичный фон */
exports.backgroundPrimary = 'var(--background-primary, #000000FF)';
/** Вторичный фон */
exports.backgroundSecondary = 'var(--background-secondary, #171717)';
/** Фон для модальной шторки */
exports.backgroundModalSheet = 'var(--background-modal-sheet, linear-gradient(180deg, rgba(255, 255, 255, 0.1) 0%, rgba(1, 182, 255, 0) 50%)), var(--background-modal-sheet, #080808FF)';
/** Фон Афины на мобиле */
exports.backgroundMobileAssistantAthena = 'var(--background-mobile-assistant-athena, linear-gradient(336.9deg, #143787 0%, rgba(8, 8, 8, 0) 64.95%)), var(--background-mobile-assistant-athena, radial-gradient(66.53% 100% at 73.33% 100%, rgba(110, 5, 193, 0.44) 0%, rgba(53, 19, 149, 0.1) 99.69%)), var(--background-mobile-assistant-athena, linear-gradient(180.03deg, rgba(8, 8, 8, 0) 50%, rgba(25, 63, 152, 0.41) 99.97%)), var(--background-mobile-assistant-athena, linear-gradient(270deg, rgba(39, 15, 107, 0.3) 0%, rgba(8, 8, 8, 0) 100%)), var(--background-mobile-assistant-athena, #080808FF)';
/** Фон Джой на мобиле */
exports.backgroundMobileAssistantJoy = 'var(--background-mobile-assistant-joy, linear-gradient(336.9deg, rgba(255, 156, 101, 0.24) 0%, rgba(8, 8, 8, 0) 64.95%)), var(--background-mobile-assistant-joy, radial-gradient(66.53% 100% at 73.33% 100%, rgba(61, 19, 149, 0.34) 0%, rgba(19, 24, 149, 0.1) 99.69%)), var(--background-mobile-assistant-joy, linear-gradient(180.03deg, rgba(8, 8, 8, 0) 50%, rgba(255, 215, 179, 0.15) 99.97%)), var(--background-mobile-assistant-joy, linear-gradient(270deg, rgba(107, 15, 87, 0.2) 0%, rgba(8, 8, 8, 0) 100%)), var(--background-mobile-assistant-joy, #080808FF)';
/** Фоновый градиент B2E-ассистента на мобиле */
exports.backgroundMobileAssistantB2E = 'var(--background-mobile-assistant-b2-e, linear-gradient(336.84deg, rgba(20, 116, 70, 0.6) 0%, rgba(8, 8, 8, 0) 64.88%)), var(--background-mobile-assistant-b2-e, radial-gradient(100% 100% at 75.89% 100%, rgba(0, 133, 255, 0.24) 0%, rgba(0, 71, 255, 0.03) 100%)), var(--background-mobile-assistant-b2-e, linear-gradient(180deg, rgba(8, 8, 8, 0) 50%, rgba(7, 71, 33, 0.3) 100%)), var(--background-mobile-assistant-b2-e, linear-gradient(270deg, #061621 0%, rgba(8, 8, 8, 0) 100%)), var(--background-mobile-assistant-b2-e, #080808FF)';
/** Фоновый градиент бренда на мобиле */
exports.backgroundMobileAssistantBrand = 'var(--background-mobile-assistant-brand, linear-gradient(336.9deg, #184BAB 0%, rgba(8, 8, 8, 0) 64.95%)), var(--background-mobile-assistant-brand, radial-gradient(66.53% 100% at 73.33% 100%, rgba(22, 69, 164, 0.44) 0%, rgba(11, 39, 45, 0.1) 100%)), var(--background-mobile-assistant-brand, linear-gradient(180.03deg, rgba(8, 8, 8, 0) 50%, rgba(22, 69, 164, 0.41) 99.97%)), var(--background-mobile-assistant-brand, linear-gradient(269.47deg, rgba(22, 69, 164, 0.3) 0.52%, rgba(8, 8, 8, 0) 99.61%)), var(--background-mobile-assistant-brand, #080808FF)';
/** Фон ассистента на мобиле */
exports.backgroundMobileAssistantMain = 'var(--background-mobile-assistant-main, linear-gradient(336.84deg, rgba(20, 116, 70, 0.6) 0%, rgba(8, 8, 8, 0) 64.88%)), var(--background-mobile-assistant-main, radial-gradient(100% 100% at 75.89% 100%, rgba(0, 133, 255, 0.24) 0%, rgba(0, 71, 255, 0.03) 100%)), var(--background-mobile-assistant-main, linear-gradient(180deg, rgba(8, 8, 8, 0) 50%, rgba(7, 71, 33, 0.3) 100%)), var(--background-mobile-assistant-main, linear-gradient(270deg, #061621 0%, rgba(8, 8, 8, 0) 100%)), var(--background-mobile-assistant-main, #080808FF)';
/** Цвет фона паранжи светлый на темном фоне */
exports.onDarkOverlaySoft = 'var(--on-dark-overlay-soft, rgba(0, 0, 0, 0.56))';
/** Цвет фона паранжи темный на темном фоне */
exports.onDarkOverlayHard = 'var(--on-dark-overlay-hard, #000000EB)';
/** Оверлей с блюром */
exports.onDarkOverlayBlur = 'var(--on-dark-overlay-blur, #26262647)';
exports.skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
exports.skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use onLightSurfaceSolidDefault */
exports.buttonBlack = 'var(--plasma-colors-button-black, var(--on-light-surface-solid-default))';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
exports.buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, var(--on-light-surface-transparent-secondary))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
exports.buttonWhite = 'var(--plasma-colors-button-white, var(--on-dark-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
exports.buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, var(--on-dark-surface-transparent-secondary))';
/** @deprecated instead use textPrimary */
exports.text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
exports.primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
exports.secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
exports.tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use textParagraph */
exports.paragraph = 'var(--plasma-colors-paragraph, var(--text-paragraph))';
/** @deprecated instead use backgroundPrimary */
exports.background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textPositive */
exports.success = 'var(--plasma-colors-success, var(--text-positive))';
/** @deprecated instead use textWarning */
exports.warning = 'var(--plasma-colors-warning, var(--text-warning))';
/** @deprecated instead use textNegative */
exports.critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use surfaceTransparentPrimary */
exports.surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, var(--surface-transparent-primary))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, var(--surface-transparent-secondary))';
/** @deprecated instead use surfaceTransparentTertiary */
exports.surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, var(--surface-transparent-tertiary))';
/** @deprecated instead use surfaceSolidPrimary */
exports.surfaceSolid01 = 'var(--plasma-colors-surface-solid01, var(--surface-solid-primary))';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use surfaceTransparentCard */
exports.surfaceCard = 'var(--plasma-colors-surface-card, var(--surface-transparent-card))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.buttonSecondary = 'var(--plasma-colors-button-secondary, var(--surface-transparent-secondary))';
/** @deprecated instead use surfacePositive */
exports.buttonSuccess = 'var(--plasma-colors-button-success, var(--surface-positive))';
/** @deprecated instead use surfaceWarning */
exports.buttonWarning = 'var(--plasma-colors-button-warning, var(--surface-warning))';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
