// @salutejs/plasma-tokens-b2c/typography/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.underline = exports.caption = exports.button2 = exports.button1 = exports.footnote2 = exports.footnote1 = exports.paragraph2 = exports.paragraph1 = exports.body3 = exports.body2 = exports.body1 = exports.headline4 = exports.headline3 = exports.headline2 = exports.headline1 = exports.display3 = exports.display2 = exports.display1 = void 0;
var display1_1 = require("./display1");
Object.defineProperty(exports, "display1", { enumerable: true, get: function () { return display1_1.display1; } });
var display2_1 = require("./display2");
Object.defineProperty(exports, "display2", { enumerable: true, get: function () { return display2_1.display2; } });
var display3_1 = require("./display3");
Object.defineProperty(exports, "display3", { enumerable: true, get: function () { return display3_1.display3; } });
var headline1_1 = require("./headline1");
Object.defineProperty(exports, "headline1", { enumerable: true, get: function () { return headline1_1.headline1; } });
var headline2_1 = require("./headline2");
Object.defineProperty(exports, "headline2", { enumerable: true, get: function () { return headline2_1.headline2; } });
var headline3_1 = require("./headline3");
Object.defineProperty(exports, "headline3", { enumerable: true, get: function () { return headline3_1.headline3; } });
var headline4_1 = require("./headline4");
Object.defineProperty(exports, "headline4", { enumerable: true, get: function () { return headline4_1.headline4; } });
var body1_1 = require("./body1");
Object.defineProperty(exports, "body1", { enumerable: true, get: function () { return body1_1.body1; } });
var body2_1 = require("./body2");
Object.defineProperty(exports, "body2", { enumerable: true, get: function () { return body2_1.body2; } });
var body3_1 = require("./body3");
Object.defineProperty(exports, "body3", { enumerable: true, get: function () { return body3_1.body3; } });
var paragraph1_1 = require("./paragraph1");
Object.defineProperty(exports, "paragraph1", { enumerable: true, get: function () { return paragraph1_1.paragraph1; } });
var paragraph2_1 = require("./paragraph2");
Object.defineProperty(exports, "paragraph2", { enumerable: true, get: function () { return paragraph2_1.paragraph2; } });
var footnote1_1 = require("./footnote1");
Object.defineProperty(exports, "footnote1", { enumerable: true, get: function () { return footnote1_1.footnote1; } });
var footnote2_1 = require("./footnote2");
Object.defineProperty(exports, "footnote2", { enumerable: true, get: function () { return footnote2_1.footnote2; } });
var button1_1 = require("./button1");
Object.defineProperty(exports, "button1", { enumerable: true, get: function () { return button1_1.button1; } });
var button2_1 = require("./button2");
Object.defineProperty(exports, "button2", { enumerable: true, get: function () { return button2_1.button2; } });
var caption_1 = require("./caption");
Object.defineProperty(exports, "caption", { enumerable: true, get: function () { return caption_1.caption; } });
var underline_1 = require("./underline");
Object.defineProperty(exports, "underline", { enumerable: true, get: function () { return underline_1.underline; } });
