// @salutejs/plasma-new-hope/styled-components/es/components/Toast/Toast.styles.js
var _templateObject;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css } from 'styled-components';
import styled from 'styled-components';
import { Overlay } from '../Overlay';
import { buttonTokens } from '../Button';
import { Button, TOAST_Z_INDEX } from './utils';
import { classes, tokens } from './Toast.tokens';
export var base = /*#__PURE__*/css(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n    display: inline-flex;\n    align-items: center;\n\n    flex-shrink: 0;\n\n    user-select: none;\n    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);\n\n    ", ": 1rem;\n"])), buttonTokens.buttonWidth);
export var StyledRoot = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-19l9ekl-0"
})(["position:fixed;left:50%;transform:translateX(-50%);z-index:", ";display:flex;width:0;top:", ";bottom:", ";&.", "{animation:0.3s showToastAnimation ease-out forwards;}&.", "{animation:0.3s hideToastAnimation ease-out forwards;}@keyframes showToastAnimation{0%{transform:", ";opacity:0;}80%{transform:translate(-50%,0);opacity:0.7;}100%{opacity:1;}}@keyframes hideToastAnimation{0%{opacity:1;}20%{transform:translate(-50%,0);opacity:0.7;}100%{transform:", ";opacity:0;}}"], TOAST_Z_INDEX, function (_ref) {
  var position = _ref.position,
    _ref$offset = _ref.offset,
    offset = _ref$offset === void 0 ? 0 : _ref$offset;
  return position === 'top' ? "".concat(Number(offset) + 5, "rem") : 'unset';
}, function (_ref2) {
  var position = _ref2.position,
    _ref2$offset = _ref2.offset,
    offset = _ref2$offset === void 0 ? 0 : _ref2$offset;
  return position === 'bottom' ? "".concat(Number(offset) + 5, "rem") : 'unset';
}, /*#__PURE__*/String(classes.toastShowed), /*#__PURE__*/String(classes.toastHidden), function (_ref3) {
  var position = _ref3.position;
  return "translate(-50%, ".concat(position === 'top' ? '-' : '', "5rem)");
}, function (_ref4) {
  var position = _ref4.position;
  return "translate(-50%, ".concat(position === 'top' ? '-' : '', "5rem)");
});
export var CloseIconWrapper = /*#__PURE__*/styled(Button).withConfig({
  componentId: "plasma-new-hope__sc-19l9ekl-1"
})(["height:1rem;--plasma_private-close-icon-margin:var(", ");margin:var(--plasma_private-close-icon-margin);color:var(", ");:hover{color:var(", ");}"], tokens.closeIconMargin, tokens.closeIconColor, tokens.closeIconColorOnHover);
export var StyledOverlay = /*#__PURE__*/styled(Overlay).withConfig({
  componentId: "plasma-new-hope__sc-19l9ekl-2"
})(["&&.", "{animation:0.3s showFadeAnimation ease-out forwards;}&&.", "{animation:0.3s hideFadeAnimation ease-out forwards;}@keyframes showFadeAnimation{0%{opacity:0;}100%{opacity:1;}}@keyframes hideFadeAnimation{0%{opacity:1;}100%{opacity:0;}}"], /*#__PURE__*/String(classes.toastFadeShowed), /*#__PURE__*/String(classes.toastFadeHidden));
export var StyledContentLeft = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-19l9ekl-3"
})(["display:flex;--plasma_private-left-content-margin:var(", ");margin:var(--plasma_private-left-content-margin);"], tokens.leftContentMargin);
export var StyledContent = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-19l9ekl-4"
})([""]);
