// @salutejs/plasma-ui/components/Radiobox/Radiobox.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Squeeze = exports.Default = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _ = /*#__PURE__*/require(".");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var func = function func() {};
var items = [{
  id: 'radio-1-1',
  name: 'radio-1',
  value: 1,
  label: 'Radiobox with a very very very very very long label',
  disabled: false
}, {
  id: 'radio-1-2',
  name: 'radio-1',
  value: 2,
  label: 'Radiobox 2',
  disabled: false
}, {
  id: 'radio-2-1',
  name: 'radio-2',
  value: 3,
  label: 'Radiobox 3',
  disabled: true
}, {
  id: 'radio-2-2',
  name: 'radio-2',
  value: 4,
  label: 'Radiobox 4',
  disabled: true,
  checked: true
}];
var StyledGriddyRadioGroup = /*#__PURE__*/(0, _styledComponents["default"])(_.RadioGroup).withConfig({
  componentId: "plasma-ui__sc-34e0eu-0"
})(["display:inline-grid;grid-template-columns:repeat(2,50%);gap:1rem;align-items:flex-start;& > * + *{margin-top:0 !important;}"]);
var Default = exports.Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(StyledGriddyRadioGroup, null, items.map(function (item) {
    return /*#__PURE__*/_react["default"].createElement(_.Radiobox, {
      key: item.id,
      id: item.id,
      name: item.name,
      value: item.value,
      label: item.label,
      disabled: item.disabled,
      checked: item.checked,
      onChange: func,
      onFocus: func,
      onBlur: func
    });
  }));
};
var squeezeStyle = {
  resize: 'horizontal',
  overflow: 'hidden',
  border: '3px solid',
  display: 'inline-block'
};
var Squeeze = exports.Squeeze = function Squeeze() {
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: squeezeStyle
  }, items.map(function (item) {
    return /*#__PURE__*/_react["default"].createElement(_.Radiobox, {
      key: item.value,
      name: "radiobox",
      value: item.value,
      label: item.label,
      disabled: item.disabled,
      checked: true,
      description: "Description of ".concat(item.label),
      onChange: func,
      onFocus: func,
      onBlur: func
    });
  }));
};
