// @salutejs/plasma-ui/components/Radiobox/Radiobox.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Radiobox = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _plasmaTokens = /*#__PURE__*/require("@salutejs/plasma-tokens");
var _Checkbox = /*#__PURE__*/require("../Checkbox/Checkbox");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _StyledEllipse;
var _excluded = ["id", "label", "description", "disabled", "focused", "scaleOnInteraction", "style", "className", "singleLine"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('Radiobox/Radiobox');
var StyledTrigger = /*#__PURE__*/(0, _styledComponents["default"])(_Checkbox.StyledTrigger).withConfig({
  componentId: "plasma-ui__sc-1ol40bw-0"
})(["--plasma-trigger-border-radius:1.25rem;--plasma-trigger-outline-radius:1.375rem;"]);

// TODO: https://github.com/salute-developers/plasma/issues/232
var StyledEllipse = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-1ol40bw-1"
})(["position:absolute;top:0;left:0;right:0;bottom:0;margin:auto;width:0.625rem;height:0.625rem;border-radius:0.625rem;background-color:", ";transition:", ";transform:scale(0);input:checked + label ", " &{transform:scale(1);}"], _plasmaTokens.white, function (_ref) {
  var theme = _ref.theme;
  return theme.lowPerformance ? 'unset' : 'transform 0.3s ease-in-out';
}, StyledTrigger);

/**
 * Переключатель, или *радиокнопка*.
 */
// eslint-disable-next-line prefer-arrow-callback
var Radiobox = exports.Radiobox = /*#__PURE__*/(0, _react.forwardRef)(function Radiobox(_ref2, ref) {
  var id = _ref2.id,
    label = _ref2.label,
    description = _ref2.description,
    disabled = _ref2.disabled,
    focused = _ref2.focused,
    scaleOnInteraction = _ref2.scaleOnInteraction,
    style = _ref2.style,
    className = _ref2.className,
    _ref2$singleLine = _ref2.singleLine,
    singleLine = _ref2$singleLine === void 0 ? false : _ref2$singleLine,
    rest = _objectWithoutProperties(_ref2, _excluded);
  var uniqId = (0, _plasmaCore.useUniqId)();
  var uniqLabelId = (0, _plasmaCore.useUniqId)();
  var uniqDescriptionId = (0, _plasmaCore.useUniqId)();
  var radioboxId = id || uniqId;
  return /*#__PURE__*/_react["default"].createElement(_Checkbox.StyledRoot, {
    $disabled: disabled,
    style: style,
    className: className,
    tabIndex: -1
  }, /*#__PURE__*/_react["default"].createElement(_Checkbox.StyledInput, _extends({
    "aria-labelledby": uniqLabelId,
    "aria-describedby": uniqDescriptionId,
    id: radioboxId,
    ref: ref,
    type: "radio",
    disabled: disabled
  }, rest)), /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxContentWrapper, {
    htmlFor: radioboxId
  }, /*#__PURE__*/_react["default"].createElement(StyledTrigger, {
    $focused: focused,
    $scaleOnInteraction: scaleOnInteraction
  }, _StyledEllipse || (_StyledEllipse = /*#__PURE__*/_react["default"].createElement(StyledEllipse, null))), label && /*#__PURE__*/_react["default"].createElement(_Checkbox.StyledContent, null, /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxLabel, {
    as: "span",
    singleLine: singleLine,
    id: uniqLabelId
  }, label), description && /*#__PURE__*/_react["default"].createElement(_plasmaCore.BaseboxDescription, {
    mt: 4,
    singleLine: singleLine,
    id: uniqDescriptionId
  }, description))));
});
