// @salutejs/plasma-new-hope/es/components/Indicator/Indicator.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef } from 'react';
import { base } from './Indicator.styles.js';
import { base as base$2 } from './variations/_view/base.js';
import { base as base$1 } from './variations/_size/base.js';

var _excluded = ["size", "view"];
var indicatorRoot = function indicatorRoot(Root) {
  return /*#__PURE__*/forwardRef(function (props, ref) {
    var size = props.size,
      view = props.view,
      rest = _objectWithoutProperties(props, _excluded);
    return /*#__PURE__*/React.createElement(Root, _extends({
      ref: ref,
      size: size,
      view: view
    }, rest));
  });
};
var indicatorConfig = {
  name: 'Indicator',
  tag: 'div',
  layout: indicatorRoot,
  base: base,
  variations: {
    size: {
      css: base$1
    },
    view: {
      css: base$2
    }
  },
  defaults: {
    size: 's',
    view: 'default'
  }
};

export { indicatorConfig, indicatorRoot };
