// @salutejs/plasma-new-hope/es/components/Popup/PopupRoot.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef, useRef, useCallback } from 'react';
import { useForkRef } from '@salutejs/plasma-core';
import { usePopupContext } from './PopupContext.js';
import { PopupRootContainer, PopupView } from './Popup.styles.js';
import { classes } from './Popup.tokens.js';

var _excluded = ["id", "placement", "offset", "frame", "setVisible", "children", "role", "zIndex", "animationInfo", "className"];

/**
 * Корень Popup.
 * Управляет показом/скрытием и анимацией всплывающего окна.
 */
var PopupRoot = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var id = _ref.id,
    placement = _ref.placement,
    offset = _ref.offset,
    frame = _ref.frame,
    setVisible = _ref.setVisible,
    children = _ref.children,
    role = _ref.role,
    zIndex = _ref.zIndex,
    animationInfo = _ref.animationInfo,
    rest = _objectWithoutProperties(_ref, _excluded);
  var contentRef = useRef(null);
  var innerRef = useForkRef(contentRef, ref);
  var popupController = usePopupContext();
  var handleAnimationEnd = useCallback(function (e) {
    if (!contentRef || e.target !== contentRef.current) {
      return;
    }
    e.stopPropagation();
    if (animationInfo !== null && animationInfo !== void 0 && animationInfo.endAnimation) {
      popupController.unregister(id);
      setVisible(false);
      animationInfo.setEndAnimation(false);
    }
  }, [popupController.unregister, animationInfo === null || animationInfo === void 0 ? void 0 : animationInfo.endAnimation, setVisible]);
  return /*#__PURE__*/React.createElement(PopupRootContainer, _extends({
    className: classes.root,
    ref: innerRef,
    placement: placement,
    frame: frame,
    offset: offset,
    zIndex: zIndex,
    onAnimationEnd: handleAnimationEnd,
    onTransitionEnd: handleAnimationEnd
  }, rest), /*#__PURE__*/React.createElement(PopupView, {
    role: role
  }, children));
});

export { PopupRoot };
