// @salutejs/plasma-web/components/PreviewGallery/PreviewGallery.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Selectable = exports.Draggable = exports.AddButton = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _plasmaTokensWeb = /*#__PURE__*/require("@salutejs/plasma-tokens-web");
var _ = /*#__PURE__*/require(".");
var _StyledAddButton, _PreviewGallery, _PreviewGallery2;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var StyledAddButton = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-yzjbem-0"
})(["width:100%;background:", ";display:flex;justify-content:center;align-items:center;border-radius:0.75rem;user-select:none;padding-top:56.25%;"], _plasmaTokensWeb.surfaceSolid01);
var AddButton = exports.AddButton = function AddButton() {
  return _StyledAddButton || (_StyledAddButton = /*#__PURE__*/_react["default"].createElement(StyledAddButton, null));
};
var images = [{
  id: /*#__PURE__*/Math.random(),
  image: './images/320_320_0.jpg',
  caption: '3:24',
  tooltip: {
    text: 'Высокое качество воспроизведения',
    placement: 'top'
  }
}, {
  id: 2 * /*#__PURE__*/Math.random(),
  image: './images/320_320_1.jpg',
  isSelected: true
}, {
  id: 3 * /*#__PURE__*/Math.random(),
  image: './images/320_320_2.jpg',
  actionDisabled: true
}, {
  id: 4 * /*#__PURE__*/Math.random(),
  image: './images/320_320_3.jpg'
}, {
  id: 5 * /*#__PURE__*/Math.random(),
  image: './images/320_320_4.jpg'
}, {
  id: 6 * /*#__PURE__*/Math.random(),
  image: './images/320_320_4.jpg',
  status: 'error'
}, {
  id: 7 * /*#__PURE__*/Math.random(),
  tooltip: {
    placement: 'bottom',
    text: 'Не сильно высокое качество воспроизведения',
    animated: false
  },
  component: /*#__PURE__*/_react["default"].createElement(AddButton, null),
  customClickHandle: function customClickHandle() {}
}];
var selectableProps = {};
var func = function func() {};
var Selectable = exports.Selectable = function Selectable() {
  return _PreviewGallery || (_PreviewGallery = /*#__PURE__*/_react["default"].createElement(_.PreviewGallery, {
    items: images,
    onItemAction: func,
    actionIcon: /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconTrashFilled, {
      size: "xs",
      color: "inherit"
    }),
    onItemClick: func
  }));
};
var draggableProps = {};
var Draggable = exports.Draggable = function Draggable() {
  return _PreviewGallery2 || (_PreviewGallery2 = /*#__PURE__*/_react["default"].createElement(_.PreviewGallery, {
    interactionType: "draggable",
    items: images,
    actionIcon: /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconTrash, {
      size: "xs",
      color: "inherit"
    }),
    onItemAction: func,
    onItemsSortEnd: func
  }));
};
