// @salutejs/plasma-tokens-b2c/themesValues/dark.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.buttonPrimary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.critical = exports.warning = exports.success = exports.accent = exports.inputErrorBackground = exports.inputBorderFocus = exports.inputBorderHover = exports.inputBorder = exports.backgroundTertiary = exports.backgroundSecondary = exports.backgroundPrimary = exports.background = exports.inverse = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.linkVisitedActive = exports.linkVisitedHover = exports.linkVisited = exports.linkActive = exports.linkHover = exports.link = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackTransparent = exports.buttonBlackSecondary = exports.buttonBlack = exports.buttonClear = exports.transparent = exports.dark03 = exports.dark02 = exports.dark01 = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.black = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.white = void 0;
exports.textPrimary = exports.textPrimaryActive = exports.textPrimaryHover = exports.inverseOverlayBlur = exports.inverseOverlayHard = exports.inverseOverlaySoft = exports.lightOverlayBlur = exports.lightOverlayHard = exports.lightOverlaySoft = exports.darkOverlayBlur = exports.darkOverlayHard = exports.darkOverlaySoft = exports.overlayBlur = exports.overlayHard = exports.overlaySoft = exports.overlay = exports.speechBubbleReceived = exports.speechBubbleSent = exports.skeletonGradientLighter = exports.skeletonGradient = exports.voicePhraseGradient = exports.gradientDevice = exports.gradient = exports.buttonFocused = exports.buttonCheckedActiveColor = exports.buttonCheckedActive = exports.buttonCheckedHoverColor = exports.buttonCheckedHover = exports.buttonChecked = exports.buttonCriticalActiveTransparent = exports.buttonCriticalActive = exports.buttonCriticalHover = exports.buttonCriticalTransparent = exports.buttonCritical = exports.buttonWarningActiveTransparent = exports.buttonWarningActive = exports.buttonWarningHover = exports.buttonWarningTransparent = exports.buttonWarning = exports.buttonSuccessActiveTransparent = exports.buttonSuccessActive = exports.buttonSuccessHover = exports.buttonSuccessTransparent = exports.buttonSuccess = exports.buttonAccent = exports.buttonSecondaryActive = exports.buttonSecondaryHover = exports.buttonSecondary = exports.buttonPrimaryActive = exports.buttonPrimaryHover = void 0;
exports.onDarkTextNegativeActive = exports.onDarkTextNegativeHover = exports.onDarkTextWarning = exports.onDarkTextWarningActive = exports.onDarkTextWarningHover = exports.onDarkTextPositive = exports.onDarkTextPositiveActive = exports.onDarkTextPositiveHover = exports.onDarkTextAccentGradient = exports.onDarkTextAccentGradientActive = exports.onDarkTextAccentGradientHover = exports.onDarkTextAccent = exports.onDarkTextAccentActive = exports.onDarkTextAccentHover = exports.onDarkTextParagraph = exports.onDarkTextParagraphActive = exports.onDarkTextParagraphHover = exports.onDarkTextTertiary = exports.onDarkTextTertiaryActive = exports.onDarkTextTertiaryHover = exports.onDarkTextSecondary = exports.onDarkTextSecondaryActive = exports.onDarkTextSecondaryHover = exports.onDarkTextPrimary = exports.onDarkTextPrimaryActive = exports.onDarkTextPrimaryHover = exports.textNegative = exports.textNegativeActive = exports.textNegativeHover = exports.textWarning = exports.textWarningActive = exports.textWarningHover = exports.textPositive = exports.textPositiveActive = exports.textPositiveHover = exports.textAccentGradient = exports.textAccentGradientActive = exports.textAccentGradientHover = exports.textAccent = exports.textAccentActive = exports.textAccentHover = exports.textParagraph = exports.textParagraphActive = exports.textParagraphHover = exports.textTertiary = exports.textTertiaryActive = exports.textTertiaryHover = exports.textSecondary = exports.textSecondaryActive = exports.textSecondaryHover = void 0;
exports.surfaceSolidCardHover = exports.surfaceSolidTertiary = exports.surfaceSolidTertiaryActive = exports.surfaceSolidTertiaryHover = exports.surfaceSolidSecondary = exports.surfaceSolidSecondaryActive = exports.surfaceSolidSecondaryHover = exports.surfaceSolidPrimary = exports.surfaceSolidPrimaryActive = exports.surfaceSolidPrimaryHover = exports.inverseTextParagraph = exports.inverseTextParagraphActive = exports.inverseTextParagraphHover = exports.inverseTextTertiary = exports.inverseTextTertiaryActive = exports.inverseTextTertiaryHover = exports.inverseTextSecondary = exports.inverseTextSecondaryActive = exports.inverseTextSecondaryHover = exports.inverseTextPrimary = exports.inverseTextPrimaryActive = exports.inverseTextPrimaryHover = exports.onLightTextNegative = exports.onLightTextNegativeActive = exports.onLightTextNegativeHover = exports.onLightTextWarning = exports.onLightTextWarningActive = exports.onLightTextWarningHover = exports.onLightTextPositive = exports.onLightTextPositiveActive = exports.onLightTextPositiveHover = exports.onLightTextAccentGradient = exports.onLightTextAccentGradientActive = exports.onLightTextAccentGradientHover = exports.onLightTextAccent = exports.onLightTextAccentActive = exports.onLightTextAccentHover = exports.onLightTextParagraph = exports.onLightTextParagraphActive = exports.onLightTextParagraphHover = exports.onLightTextTertiary = exports.onLightTextTertiaryActive = exports.onLightTextTertiaryHover = exports.onLightTextSecondary = exports.onLightTextSecondaryActive = exports.onLightTextSecondaryHover = exports.onLightTextPrimary = exports.onLightTextPrimaryActive = exports.onLightTextPrimaryHover = exports.onDarkTextNegative = void 0;
exports.surfaceTransparentNegative = exports.surfaceTransparentWarning = exports.surfaceTransparentPositive = exports.surfaceTransparentAccent = exports.surfaceTransparentNegativeActive = exports.surfaceTransparentNegativeHover = exports.surfaceTransparentWarningActive = exports.surfaceTransparentWarningHover = exports.surfaceTransparentPositiveActive = exports.surfaceTransparentPositiveHover = exports.surfaceNegative = exports.surfaceNegativeActive = exports.surfaceNegativeHover = exports.surfaceWarning = exports.surfaceWarningActive = exports.surfaceWarningHover = exports.surfacePositive = exports.surfacePositiveActive = exports.surfacePositiveHover = exports.surfaceTransparentAccentActive = exports.surfaceTransparentAccentHover = exports.surfaceAccentGradient = exports.surfaceAccentGradientActive = exports.surfaceAccentGradientHover = exports.surfaceAccent = exports.surfaceAccentActive = exports.surfaceAccentHover = exports.surfaceClear = exports.surfaceClearActive = exports.surfaceClearHover = exports.surfaceTransparentCard = exports.surfaceTransparentCardActive = exports.surfaceTransparentCardHover = exports.surfaceTransparentDeep = exports.surfaceTransparentDeepActive = exports.surfaceTransparentDeepHover = exports.surfaceTransparentTertiary = exports.surfaceTransparentTertiaryActive = exports.surfaceTransparentTertiaryHover = exports.surfaceTransparentSecondary = exports.surfaceTransparentSecondaryActive = exports.surfaceTransparentSecondaryHover = exports.surfaceTransparentPrimary = exports.surfaceTransparentPrimaryActive = exports.surfaceTransparentPrimaryHover = exports.surfaceSolidDefault = exports.surfaceSolidDefaultActive = exports.surfaceSolidDefaultHover = exports.surfaceSolidCard = exports.surfaceSolidCardActive = void 0;
exports.onLightSurfaceSolidSecondaryActive = exports.onLightSurfaceSolidSecondaryHover = exports.onLightSurfaceSolidPrimary = exports.onLightSurfaceSolidPrimaryActive = exports.onLightSurfaceSolidPrimaryHover = exports.onDarkSurfaceNegative = exports.onDarkSurfaceNegativeActive = exports.onDarkSurfaceNegativeHover = exports.onDarkSurfaceWarning = exports.onDarkSurfaceWarningActive = exports.onDarkSurfaceWarningHover = exports.onDarkSurfacePositive = exports.onDarkSurfacePositiveActive = exports.onDarkSurfacePositiveHover = exports.onDarkSurfaceAccentGradient = exports.onDarkSurfaceAccentGradientActive = exports.onDarkSurfaceAccentGradientHover = exports.onDarkSurfaceAccent = exports.onDarkSurfaceAccentActive = exports.onDarkSurfaceAccentHover = exports.onDarkSurfaceTransparentCard = exports.onDarkSurfaceTransparentCardActive = exports.onDarkSurfaceTransparentCardHover = exports.onDarkSurfaceTransparentDeep = exports.onDarkSurfaceTransparentDeepActive = exports.onDarkSurfaceTransparentDeepHover = exports.onDarkSurfaceTransparentTertiary = exports.onDarkSurfaceTransparentTertiaryActive = exports.onDarkSurfaceTransparentTertiaryHover = exports.onDarkSurfaceTransparentSecondary = exports.onDarkSurfaceTransparentSecondaryActive = exports.onDarkSurfaceTransparentSecondaryHover = exports.onDarkSurfaceTransparentPrimary = exports.onDarkSurfaceTransparentPrimaryActive = exports.onDarkSurfaceTransparentPrimaryHover = exports.onDarkSurfaceSolidDefault = exports.onDarkSurfaceSolidDefaultActive = exports.onDarkSurfaceSolidDefaultHover = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidCardActive = exports.onDarkSurfaceSolidCardHover = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidTertiaryActive = exports.onDarkSurfaceSolidTertiaryHover = exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidSecondaryActive = exports.onDarkSurfaceSolidSecondaryHover = exports.onDarkSurfaceSolidPrimary = exports.onDarkSurfaceSolidPrimaryActive = exports.onDarkSurfaceSolidPrimaryHover = void 0;
exports.onLightOverlayBlur = exports.onLightOverlayHard = exports.onLightOverlaySoft = exports.onDarkOverlayBlur = exports.onDarkOverlayHard = exports.onDarkOverlaySoft = exports.lightBackgroundPrimary = exports.darkBackgroundPrimary = exports.onLightSurfaceNegative = exports.onLightSurfaceNegativeActive = exports.onLightSurfaceNegativeHover = exports.onLightSurfaceWarning = exports.onLightSurfaceWarningActive = exports.onLightSurfaceWarningHover = exports.onLightSurfacePositive = exports.onLightSurfacePositiveActive = exports.onLightSurfacePositiveHover = exports.onLightSurfaceAccentGradient = exports.onLightSurfaceAccentGradientActive = exports.onLightSurfaceAccentGradientHover = exports.onLightSurfaceAccent = exports.onLightSurfaceAccentActive = exports.onLightSurfaceAccentHover = exports.onLightSurfaceTransparentCard = exports.onLightSurfaceTransparentCardActive = exports.onLightSurfaceTransparentCardHover = exports.onLightSurfaceTransparentDeep = exports.onLightSurfaceTransparentDeepActive = exports.onLightSurfaceTransparentDeepHover = exports.onLightSurfaceTransparentTertiary = exports.onLightSurfaceTransparentTertiaryActive = exports.onLightSurfaceTransparentTertiaryHover = exports.onLightSurfaceTransparentSecondary = exports.onLightSurfaceTransparentSecondaryActive = exports.onLightSurfaceTransparentSecondaryHover = exports.onLightSurfaceTransparentPrimary = exports.onLightSurfaceTransparentPrimaryActive = exports.onLightSurfaceTransparentPrimaryHover = exports.onLightSurfaceSolidDefault = exports.onLightSurfaceSolidDefaultActive = exports.onLightSurfaceSolidDefaultHover = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidCardActive = exports.onLightSurfaceSolidCardHover = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidTertiaryActive = exports.onLightSurfaceSolidTertiaryHover = exports.onLightSurfaceSolidSecondary = void 0;
/** Базовый белый цвет совпадает с whitePrimary */
exports.white = 'rgba(255, 255, 255, 0.96)';
/** Первичный белый цвет */
exports.whitePrimary = 'rgba(255, 255, 255, 0.96)';
/** Вторичный белый цвет */
exports.whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный белый цвет */
exports.whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** Базовый черный цвет совпадает с blackPrimary */
exports.black = '#080808';
/** Первичный черный цвет */
exports.blackPrimary = '#080808';
/** Вторичный черныйцвет */
exports.blackSecondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный черный цвет */
exports.blackTertiary = 'rgba(8, 8, 8, 0.28)';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark01 = '#171717';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark02 = '#232323';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark03 = '#363636';
/** Прозрачный цвет */
exports.transparent = 'rgba(0, 0, 0, 0)';
/** Второстепенная прозрачная */
exports.buttonClear = 'rgba(0, 0, 0, 0)';
/** Дефолтная чёрная, не зависит от темы */
exports.buttonBlack = '#080808';
/** Второстепенная чёрная, не зависит от темы */
exports.buttonBlackSecondary = 'rgba(8, 8, 8, 0.12)';
/** Чёрная для использования поверх картинок */
exports.buttonBlackTransparent = 'rgba(8, 8, 8, 0.56)';
/** Дефолтная белая, не зависит от темы */
exports.buttonWhite = '#FFFFFF';
/** Второстепенная белая, не зависит от темы */
exports.buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** Базовый цвет текста, совпадает с primary */
exports.text = 'rgba(255, 255, 255, 0.96)';
exports.link = '#3F81FD';
exports.linkHover = '#6298FD';
exports.linkActive = '#1767FD';
exports.linkVisited = '#5993FF';
exports.linkVisitedHover = '#7DAAFF';
exports.linkVisitedActive = '#3078FF';
/** Основной текст в интерфейсе */
exports.primary = 'rgba(255, 255, 255, 0.96)';
/** Второстепенный текст */
exports.secondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный цвет текста */
exports.tertiary = 'rgba(255, 255, 255, 0.28)';
exports.paragraph = 'rgba(255, 255, 255, 0.96)';
exports.inverse = '#080808';
/** Основной бэкграунд всех приложений и смартапов */
exports.background = '#080808';
exports.backgroundPrimary = '#080808';
/** Вторичный цвет фона */
exports.backgroundSecondary = '#232323';
/** Третичный цвет фона */
exports.backgroundTertiary = '#363636';
exports.inputBorder = 'rgba(0, 0, 0, 0)';
exports.inputBorderHover = 'rgba(0, 0, 0, 0)';
exports.inputBorderFocus = 'rgba(0, 0, 0, 0)';
exports.inputErrorBackground = 'rgba(255, 46, 67, 0.12)';
/** Акцентный цвет бренда */
exports.accent = '#5993FF';
/** Обозначение успешного сценария */
exports.success = '#24B23E';
/** Цвет предупреждения */
exports.warning = '#FA6D20';
/** Цвет ошибки */
exports.critical = '#FF4D5F';
/** Полупрозрачные поверхности */
exports.surfaceLiquid01 = 'rgba(255, 255, 255, 0.06)';
/** Полупрозрачные поверхности */
exports.surfaceLiquid02 = 'rgba(255, 255, 255, 0.12)';
/** Полупрозрачные поверхности */
exports.surfaceLiquid03 = 'rgba(255, 255, 255, 0.2)';
/** Непрозрачные поверхности */
exports.surfaceSolid01 = '#171717';
/** Непрозрачные поверхности */
exports.surfaceSolid02 = '#232323';
/** Непрозрачные поверхности */
exports.surfaceSolid03 = '#363636';
/** Основной фон для карточек */
exports.surfaceCard = 'rgba(255, 255, 255, 0.12)';
/** Первичный цвет контролов */
exports.buttonPrimary = '#3F82FD';
exports.buttonPrimaryHover = '#538efe';
exports.buttonPrimaryActive = '#3a75e5';
/** Второстепенная кнопка */
exports.buttonSecondary = 'rgba(255, 255, 255, 0.12)';
exports.buttonSecondaryHover = 'rgba(255, 255, 255, 0.16)';
exports.buttonSecondaryActive = 'rgba(255, 255, 255, 0.1)';
/** Акцентный цвет у контролов */
exports.buttonAccent = '#3F82FD';
/** Кнопка для успешного сценария */
exports.buttonSuccess = '#21A038';
exports.buttonSuccessTransparent = 'rgba(33, 160, 56, 0.3)';
exports.buttonSuccessHover = '#3eaa4d';
exports.buttonSuccessActive = '#1d9233';
exports.buttonSuccessActiveTransparent = 'rgba(97, 181, 103, 0.384)';
/** Цвет предупреждения у контролов */
exports.buttonWarning = '#E35502';
exports.buttonWarningTransparent = 'rgba(227, 85, 2, 0.3)';
exports.buttonWarningHover = '#f76d2f';
exports.buttonWarningActive = '#de550a';
exports.buttonWarningActiveTransparent = 'rgba(238, 127, 83, 0.384)';
/** Цвет ошибки у контролов */
exports.buttonCritical = '#FF2E43';
exports.buttonCriticalTransparent = 'rgba(255, 46, 67, 0.3)';
exports.buttonCriticalHover = '#ff4757';
exports.buttonCriticalActive = '#e82a3e';
exports.buttonCriticalActiveTransparent = 'rgba(255, 114, 113, 0.384)';
/** Цвет зажатого контрола */
exports.buttonChecked = '#FFFFFF';
exports.buttonCheckedHover = '#FFFFFF';
exports.buttonCheckedHoverColor = 'hsla(0, 0%, 3.1372549019607843%, 1)';
exports.buttonCheckedActive = '#FFFFFF';
exports.buttonCheckedActiveColor = 'hsla(0, 0%, 3.1372549019607843%, 1)';
/** Цвет рамки фокуса у контрола */
exports.buttonFocused = '#3F82FD';
/** Градиент для заливки основного фона */
exports.gradient = 'linear-gradient(45deg, hsla(220.1123595505618, 85.57692307692307%, 59.21568627450981%, 1) 0%, hsla(189.78947368421052, 78.51239669421487%, 52.54901960784314%, 1) 100%)';
exports.gradientDevice = '';
exports.voicePhraseGradient = '';
exports.skeletonGradient = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% )';
exports.skeletonGradientLighter = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% )';
exports.speechBubbleSent = '';
exports.speechBubbleReceived = '';
/** Цвет фона паранжи */
exports.overlay = 'rgba(0, 0, 0, 0.8)';
exports.overlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.overlayHard = 'rgba(8, 8, 8, 0.96)';
exports.overlayBlur = 'rgba(8, 8, 8, 0.28)';
exports.darkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.darkOverlayHard = 'rgba(8, 8, 8, 0.9)';
exports.darkOverlayBlur = 'rgba(23, 23, 23, 0.28)';
exports.lightOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.lightOverlayHard = 'rgba(8, 8, 8, 0.9)';
exports.lightOverlayBlur = 'rgba(23, 23, 23, 0.28)';
exports.inverseOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.inverseOverlayHard = 'rgba(8, 8, 8, 0.9)';
exports.inverseOverlayBlur = 'rgba(23, 23, 23, 0.28)';
exports.textPrimaryHover = '#FFFFFF93';
exports.textPrimaryActive = '#FFFFFFC4';
exports.textPrimary = 'rgba(255, 255, 255, 0.96)';
exports.textSecondaryHover = '#FFFFFFFF';
exports.textSecondaryActive = '#FFFFFFAB';
exports.textSecondary = 'rgba(255, 255, 255, 0.56)';
exports.textTertiaryHover = '#FFFFFFFF';
exports.textTertiaryActive = '#FFFFFF56';
exports.textTertiary = 'rgba(255, 255, 255, 0.28)';
exports.textParagraphHover = '#FFFFFF7A';
exports.textParagraphActive = '#FFFFFFA3';
exports.textParagraph = 'rgba(255, 255, 255, 0.8)';
exports.textAccentHover = '#90B6FEFF';
exports.textAccentActive = '#216EFDFF';
exports.textAccent = '#3F81FD';
exports.textAccentGradientHover = '#CCCCCCFF';
exports.textAccentGradientActive = '#E6E6E6FF';
exports.textAccentGradient = 'linear-gradient(89.83deg, #5E94FF 0%, #43DBFA 100%)';
exports.textPositiveHover = '#1EB83AFF';
exports.textPositiveActive = '#15842AFF';
exports.textPositive = '#1A9E32';
exports.textWarningHover = '#FB7223FF';
exports.textWarningActive = '#DC5304FF';
exports.textWarning = '#FA5F05';
exports.textNegativeHover = '#FF475AFF';
exports.textNegativeActive = '#FF0A23FF';
exports.textNegative = '#FF293E';
exports.onDarkTextPrimaryHover = '#FFFFFF93';
exports.onDarkTextPrimaryActive = '#FFFFFFC4';
exports.onDarkTextPrimary = 'rgba(255, 255, 255, 0.96)';
exports.onDarkTextSecondaryHover = '#FFFFFFFF';
exports.onDarkTextSecondaryActive = '#FFFFFFAB';
exports.onDarkTextSecondary = 'rgba(255, 255, 255, 0.56)';
exports.onDarkTextTertiaryHover = '#FFFFFFFF';
exports.onDarkTextTertiaryActive = '#FFFFFF56';
exports.onDarkTextTertiary = 'rgba(255, 255, 255, 0.28)';
exports.onDarkTextParagraphHover = '#FFFFFF7A';
exports.onDarkTextParagraphActive = '#FFFFFFA3';
exports.onDarkTextParagraph = 'rgba(255, 255, 255, 0.8)';
exports.onDarkTextAccentHover = '#90B6FEFF';
exports.onDarkTextAccentActive = '#216EFDFF';
exports.onDarkTextAccent = '#3F81FD';
exports.onDarkTextAccentGradientHover = '#CCCCCCFF';
exports.onDarkTextAccentGradientActive = '#E6E6E6FF';
exports.onDarkTextAccentGradient = 'linear-gradient(89.83deg, #5E94FF 0%, #43DBFA 100%)';
exports.onDarkTextPositiveHover = '#1EB83AFF';
exports.onDarkTextPositiveActive = '#15842AFF';
exports.onDarkTextPositive = '#1A9E32';
exports.onDarkTextWarningHover = '#FB7223FF';
exports.onDarkTextWarningActive = '#DC5304FF';
exports.onDarkTextWarning = '#FA5F05';
exports.onDarkTextNegativeHover = '#FF475AFF';
exports.onDarkTextNegativeActive = '#FF0A23FF';
exports.onDarkTextNegative = '#FF293E';
exports.onLightTextPrimaryHover = '#08080893';
exports.onLightTextPrimaryActive = '#080808C4';
exports.onLightTextPrimary = 'rgba(8, 8, 8, 0.96)';
exports.onLightTextSecondaryHover = '#080808FF';
exports.onLightTextSecondaryActive = '#080808AB';
exports.onLightTextSecondary = 'rgba(8, 8, 8, 0.56)';
exports.onLightTextTertiaryHover = '#080808FF';
exports.onLightTextTertiaryActive = '#08080856';
exports.onLightTextTertiary = 'rgba(8, 8, 8, 0.28)';
exports.onLightTextParagraphHover = '#0808087A';
exports.onLightTextParagraphActive = '#080808A3';
exports.onLightTextParagraph = 'rgba(8, 8, 8, 0.8)';
exports.onLightTextAccentHover = '#79A7FBFF';
exports.onLightTextAccentActive = '#0D5FF8FF';
exports.onLightTextAccent = '#2A72F8';
exports.onLightTextAccentGradientHover = '#CCCCCCFF';
exports.onLightTextAccentGradientActive = '#E6E6E6FF';
exports.onLightTextAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
exports.onLightTextPositiveHover = '#1EB83AFF';
exports.onLightTextPositiveActive = '#15842AFF';
exports.onLightTextPositive = '#1A9E32';
exports.onLightTextWarningHover = '#FB7223FF';
exports.onLightTextWarningActive = '#DC5304FF';
exports.onLightTextWarning = '#FA5F05';
exports.onLightTextNegativeHover = '#F5384BFF';
exports.onLightTextNegativeActive = '#E40C22FF';
exports.onLightTextNegative = '#F31B31';
exports.inverseTextPrimaryHover = '#08080893';
exports.inverseTextPrimaryActive = '#080808C4';
exports.inverseTextPrimary = 'rgba(8, 8, 8, 0.96)';
exports.inverseTextSecondaryHover = '#080808FF';
exports.inverseTextSecondaryActive = '#080808AB';
exports.inverseTextSecondary = 'rgba(8, 8, 8, 0.56)';
exports.inverseTextTertiaryHover = '#080808FF';
exports.inverseTextTertiaryActive = '#08080856';
exports.inverseTextTertiary = 'rgba(8, 8, 8, 0.28)';
exports.inverseTextParagraphHover = '#0808087A';
exports.inverseTextParagraphActive = '#080808A3';
exports.inverseTextParagraph = 'rgba(8, 8, 8, 0.8)';
exports.surfaceSolidPrimaryHover = '#1C1C1CFF';
exports.surfaceSolidPrimaryActive = '#121212FF';
exports.surfaceSolidPrimary = '#171717';
exports.surfaceSolidSecondaryHover = '#303030FF';
exports.surfaceSolidSecondaryActive = '#212121FF';
exports.surfaceSolidSecondary = '#262626';
exports.surfaceSolidTertiaryHover = '#404040FF';
exports.surfaceSolidTertiaryActive = '#303030FF';
exports.surfaceSolidTertiary = '#363636';
exports.surfaceSolidCardHover = '#1C1C1CFF';
exports.surfaceSolidCardActive = '#121212FF';
exports.surfaceSolidCard = '#171717';
exports.surfaceSolidDefaultHover = '#FFFFFFFF';
exports.surfaceSolidDefaultActive = '#FFFFFFFF';
exports.surfaceSolidDefault = 'rgba(255, 255, 255, 1)';
exports.surfaceTransparentPrimaryHover = '#FFFFFF1F';
exports.surfaceTransparentPrimaryActive = '#FFFFFF0A';
exports.surfaceTransparentPrimary = 'rgba(255, 255, 255, 0.06)';
exports.surfaceTransparentSecondaryHover = '#FFFFFF3D';
exports.surfaceTransparentSecondaryActive = '#FFFFFF0F';
exports.surfaceTransparentSecondary = 'rgba(255, 255, 255, 0.12)';
exports.surfaceTransparentTertiaryHover = '#FFFFFF52';
exports.surfaceTransparentTertiaryActive = '#FFFFFF24';
exports.surfaceTransparentTertiary = 'rgba(255, 255, 255, 0.2)';
exports.surfaceTransparentDeepHover = '#FFFFFFC2';
exports.surfaceTransparentDeepActive = '#FFFFFF94';
exports.surfaceTransparentDeep = 'rgba(255, 255, 255, 0.64)';
exports.surfaceTransparentCardHover = '#FFFFFF3D';
exports.surfaceTransparentCardActive = '#FFFFFF0F';
exports.surfaceTransparentCard = 'rgba(255, 255, 255, 0.12)';
exports.surfaceClearHover = '#FFFFFF00';
exports.surfaceClearActive = '#FFFFFF00';
exports.surfaceClear = 'rgba(255, 255, 255, 0)';
exports.surfaceAccentHover = '#5D95FDFF';
exports.surfaceAccentActive = '#357BFDFF';
exports.surfaceAccent = '#3F81FD';
exports.surfaceAccentGradientHover = '#FFFFFFFF';
exports.surfaceAccentGradientActive = '#FFFFFFFF';
exports.surfaceAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%), linear-gradient(89.83deg, #5E94FF 0%, #43DBFA 100%)';
exports.surfaceTransparentAccentHover = '#3F82FD3D';
exports.surfaceTransparentAccentActive = '#3F82FD0F';
exports.surfacePositiveHover = '#1DAF37FF';
exports.surfacePositiveActive = '#18952FFF';
exports.surfacePositive = '#1A9E32';
exports.surfaceWarningHover = '#FB7223FF';
exports.surfaceWarningActive = '#F05B05FF';
exports.surfaceWarning = '#FA5F05';
exports.surfaceNegativeHover = '#FF475AFF';
exports.surfaceNegativeActive = '#FF1F35FF';
exports.surfaceNegative = '#FF293E';
exports.surfaceTransparentPositiveHover = '#1A9E323D';
exports.surfaceTransparentPositiveActive = '#1A9E320F';
exports.surfaceTransparentWarningHover = '#FA5F053D';
exports.surfaceTransparentWarningActive = '#FA5F050F';
exports.surfaceTransparentNegativeHover = '#FF293E3D';
exports.surfaceTransparentNegativeActive = '#FF293E0F';
exports.surfaceTransparentAccent = 'rgba(63,129,253,0.12)';
exports.surfaceTransparentPositive = 'rgba(26,158,50,0.12)';
exports.surfaceTransparentWarning = 'rgba(250,95,5,0.12)';
exports.surfaceTransparentNegative = 'rgba(255,41,62,0.12)';
exports.onDarkSurfaceSolidPrimaryHover = '#1C1C1CFF';
exports.onDarkSurfaceSolidPrimaryActive = '#121212FF';
exports.onDarkSurfaceSolidPrimary = '#171717';
exports.onDarkSurfaceSolidSecondaryHover = '#303030FF';
exports.onDarkSurfaceSolidSecondaryActive = '#212121FF';
exports.onDarkSurfaceSolidSecondary = '#262626';
exports.onDarkSurfaceSolidTertiaryHover = '#404040FF';
exports.onDarkSurfaceSolidTertiaryActive = '#303030FF';
exports.onDarkSurfaceSolidTertiary = '#363636';
exports.onDarkSurfaceSolidCardHover = '#1C1C1CFF';
exports.onDarkSurfaceSolidCardActive = '#121212FF';
exports.onDarkSurfaceSolidCard = '#171717';
exports.onDarkSurfaceSolidDefaultHover = '#FFFFFFFF';
exports.onDarkSurfaceSolidDefaultActive = '#FFFFFFFF';
exports.onDarkSurfaceSolidDefault = 'rgba(255, 255, 255, 1)';
exports.onDarkSurfaceTransparentPrimaryHover = '#FFFFFF1F';
exports.onDarkSurfaceTransparentPrimaryActive = '#FFFFFF0A';
exports.onDarkSurfaceTransparentPrimary = 'rgba(255, 255, 255, 0.06)';
exports.onDarkSurfaceTransparentSecondaryHover = '#FFFFFF3D';
exports.onDarkSurfaceTransparentSecondaryActive = '#FFFFFF0F';
exports.onDarkSurfaceTransparentSecondary = 'rgba(255, 255, 255, 0.12)';
exports.onDarkSurfaceTransparentTertiaryHover = '#FFFFFF52';
exports.onDarkSurfaceTransparentTertiaryActive = '#FFFFFF24';
exports.onDarkSurfaceTransparentTertiary = 'rgba(255, 255, 255, 0.2)';
exports.onDarkSurfaceTransparentDeepHover = '#FFFFFFC2';
exports.onDarkSurfaceTransparentDeepActive = '#FFFFFF94';
exports.onDarkSurfaceTransparentDeep = 'rgba(255, 255, 255, 0.64)';
exports.onDarkSurfaceTransparentCardHover = '#FFFFFF3D';
exports.onDarkSurfaceTransparentCardActive = '#FFFFFF0F';
exports.onDarkSurfaceTransparentCard = 'rgba(255, 255, 255, 0.12)';
exports.onDarkSurfaceAccentHover = '#5D95FDFF';
exports.onDarkSurfaceAccentActive = '#357BFDFF';
exports.onDarkSurfaceAccent = '#3F81FD';
exports.onDarkSurfaceAccentGradientHover = '#FFFFFFFF';
exports.onDarkSurfaceAccentGradientActive = '#FFFFFFFF';
exports.onDarkSurfaceAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
exports.onDarkSurfacePositiveHover = '#1DAF37FF';
exports.onDarkSurfacePositiveActive = '#18952FFF';
exports.onDarkSurfacePositive = '#1A9E32';
exports.onDarkSurfaceWarningHover = '#FB7223FF';
exports.onDarkSurfaceWarningActive = '#F05B05FF';
exports.onDarkSurfaceWarning = '#FA5F05';
exports.onDarkSurfaceNegativeHover = '#FF475AFF';
exports.onDarkSurfaceNegativeActive = '#FF1F35FF';
exports.onDarkSurfaceNegative = '#FF293E';
exports.onLightSurfaceSolidPrimaryHover = '#FAFAFAFF';
exports.onLightSurfaceSolidPrimaryActive = '#F0F0F0FF';
exports.onLightSurfaceSolidPrimary = '#F5F5F5';
exports.onLightSurfaceSolidSecondaryHover = '#F7F7F7FF';
exports.onLightSurfaceSolidSecondaryActive = '#E8E8E8FF';
exports.onLightSurfaceSolidSecondary = '#ECECEC';
exports.onLightSurfaceSolidTertiaryHover = '#E8E8E8FF';
exports.onLightSurfaceSolidTertiaryActive = '#D9D9D9FF';
exports.onLightSurfaceSolidTertiary = '#DDDDDD';
exports.onLightSurfaceSolidCardHover = '#FFFFFFFF';
exports.onLightSurfaceSolidCardActive = '#FFFFFFFF';
exports.onLightSurfaceSolidCard = 'rgba(255, 255, 255, 1)';
exports.onLightSurfaceSolidDefaultHover = '#0D0D0DFF';
exports.onLightSurfaceSolidDefaultActive = '#030303FF';
exports.onLightSurfaceSolidDefault = '#080808';
exports.onLightSurfaceTransparentPrimaryHover = '#08080812';
exports.onLightSurfaceTransparentPrimaryActive = '#08080803';
exports.onLightSurfaceTransparentPrimary = 'rgba(8, 8, 8, 0.03)';
exports.onLightSurfaceTransparentSecondaryHover = '#0808081F';
exports.onLightSurfaceTransparentSecondaryActive = '#0808080A';
exports.onLightSurfaceTransparentSecondary = 'rgba(8, 8, 8, 0.06)';
exports.onLightSurfaceTransparentTertiaryHover = '#0808083D';
exports.onLightSurfaceTransparentTertiaryActive = '#0808080F';
exports.onLightSurfaceTransparentTertiary = 'rgba(8, 8, 8, 0.12)';
exports.onLightSurfaceTransparentDeepHover = '#080808C2';
exports.onLightSurfaceTransparentDeepActive = '#08080894';
exports.onLightSurfaceTransparentDeep = 'rgba(8, 8, 8, 0.64)';
exports.onLightSurfaceTransparentCardHover = '#FFFFFFFF';
exports.onLightSurfaceTransparentCardActive = '#FFFFFFFF';
exports.onLightSurfaceTransparentCard = 'rgba(255, 255, 255, 1)';
exports.onLightSurfaceAccentHover = '#4886F9FF';
exports.onLightSurfaceAccentActive = '#206CF8FF';
exports.onLightSurfaceAccent = '#2A72F8';
exports.onLightSurfaceAccentGradientHover = '#FFFFFFFF';
exports.onLightSurfaceAccentGradientActive = '#FFFFFFFF';
exports.onLightSurfaceAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
exports.onLightSurfacePositiveHover = '#1DAF37FF';
exports.onLightSurfacePositiveActive = '#18952FFF';
exports.onLightSurfacePositive = '#1A9E32';
exports.onLightSurfaceWarningHover = '#FB7223FF';
exports.onLightSurfaceWarningActive = '#F05B05FF';
exports.onLightSurfaceWarning = '#FA5F05';
exports.onLightSurfaceNegativeHover = '#F5384BFF';
exports.onLightSurfaceNegativeActive = '#F31228FF';
exports.onLightSurfaceNegative = '#F31B31';
exports.darkBackgroundPrimary = '#080808';
exports.lightBackgroundPrimary = '#F9F9F9';
exports.onDarkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.onDarkOverlayHard = 'rgba(8, 8, 8, 0.96)';
exports.onDarkOverlayBlur = 'rgba(8, 8, 8, 0.28)';
exports.onLightOverlaySoft = 'rgba(249, 249, 249, 0.56)';
exports.onLightOverlayHard = 'rgba(249, 249, 249, 0.96)';
exports.onLightOverlayBlur = 'rgba(249, 249, 249, 0.28)';
