// @salutejs/plasma-core/components/PopupBase/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "PopupBase", {
  enumerable: true,
  get: function get() {
    return _PopupBase.PopupBase;
  }
});
Object.defineProperty(exports, "PopupBaseProvider", {
  enumerable: true,
  get: function get() {
    return _PopupBaseContext.PopupBaseProvider;
  }
});
Object.defineProperty(exports, "endAnimationClass", {
  enumerable: true,
  get: function get() {
    return _utils.endAnimationClass;
  }
});
Object.defineProperty(exports, "endTransitionClass", {
  enumerable: true,
  get: function get() {
    return _utils.endTransitionClass;
  }
});
Object.defineProperty(exports, "popupBaseRootClass", {
  enumerable: true,
  get: function get() {
    return _utils.popupBaseRootClass;
  }
});
Object.defineProperty(exports, "usePopupBaseContext", {
  enumerable: true,
  get: function get() {
    return _PopupBaseContext.usePopupBaseContext;
  }
});
var _PopupBaseContext = /*#__PURE__*/require("./PopupBaseContext");
var _PopupBase = /*#__PURE__*/require("./PopupBase");
var _utils = /*#__PURE__*/require("./utils");
