// @salutejs/plasma-core/components/PopupBase/PopupBaseRoot.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PopupBaseRoot = exports.DEFAULT_Z_INDEX = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _hooks = /*#__PURE__*/require("../../hooks");
var _PopupBaseContext = /*#__PURE__*/require("./PopupBaseContext");
var _utils = /*#__PURE__*/require("./utils");
var _excluded = ["id", "placement", "offset", "frame", "setVisible", "children", "role", "zIndex", "animationInfo"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var DEFAULT_Z_INDEX = exports.DEFAULT_Z_INDEX = 9000;
var PopupBaseView = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-14025jn-0"
})(["position:relative;max-width:100%;pointer-events:all;"]);
var PopupRootContainer = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-14025jn-1"
})(["", " ", " ", ";"], function (_ref) {
  var frame = _ref.frame;
  return (0, _styledComponents.css)(["position:", ";"], frame === 'document' ? 'fixed' : 'absolute');
}, function (_ref2) {
  var zIndex = _ref2.zIndex;
  return (0, _styledComponents.css)(["z-index:", ";"], zIndex || DEFAULT_Z_INDEX);
}, function (_ref3) {
  var placement = _ref3.placement,
    offset = _ref3.offset;
  return (0, _utils.handlePosition)(placement, offset);
});

/**
 * Корень PopupBase.
 * Управляет показом/скрытием и анимацией всплывающего окна.
 */
var PopupBaseRoot = exports.PopupBaseRoot = /*#__PURE__*/_react["default"].forwardRef(function (_ref4, ref) {
  var id = _ref4.id,
    placement = _ref4.placement,
    offset = _ref4.offset,
    frame = _ref4.frame,
    setVisible = _ref4.setVisible,
    children = _ref4.children,
    role = _ref4.role,
    zIndex = _ref4.zIndex,
    animationInfo = _ref4.animationInfo,
    rest = _objectWithoutProperties(_ref4, _excluded);
  var contentRef = (0, _react.useRef)(null);
  var innerRef = (0, _hooks.useForkRef)(contentRef, ref);
  var popupController = (0, _PopupBaseContext.usePopupBaseContext)();
  var handleAnimationEnd = (0, _react.useCallback)(function (e) {
    if (!contentRef || e.target !== contentRef.current) {
      return;
    }
    e.stopPropagation();
    if (animationInfo !== null && animationInfo !== void 0 && animationInfo.endAnimation) {
      popupController.unregister(id);
      setVisible(false);
      animationInfo.setEndAnimation(false);
    }
  }, [popupController.unregister, animationInfo === null || animationInfo === void 0 ? void 0 : animationInfo.endAnimation, setVisible]);
  return /*#__PURE__*/_react["default"].createElement(PopupRootContainer, _extends({
    className: _utils.popupBaseRootClass,
    ref: innerRef,
    placement: placement,
    frame: frame,
    offset: offset,
    zIndex: zIndex,
    onAnimationEnd: handleAnimationEnd,
    onTransitionEnd: handleAnimationEnd
  }, rest), /*#__PURE__*/_react["default"].createElement(PopupBaseView, {
    role: role
  }, children));
});
