// @salutejs/plasma-new-hope/styled-components/es/components/Segment/ui/SegmentGroup/SegmentGroup.styles.js
var _templateObject;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css } from 'styled-components';
import styled from 'styled-components';
import { addFocus } from '../../../../mixins';
import { tokens } from '../../tokens';
export var base = /*#__PURE__*/css(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n    display: inline-flex;\n    align-items: center;\n"])));
export var StyledContent = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1kj8wok-0"
})(["display:inline-flex;flex-direction:row;align-items:center;"]);
export var StyledContentWrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1kj8wok-1"
})(["position:relative;height:100%;margin:0;padding:0.25rem;::-webkit-scrollbar{display:none;}overflow-x:auto;overflow-y:hidden;scroll-behavior:smooth;scroll-snap-type:x mandatory;"]);
export var StyledArrow = /*#__PURE__*/styled.button.withConfig({
  componentId: "plasma-new-hope__sc-1kj8wok-2"
})(["display:flex;cursor:pointer;border:none;background-color:transparent;padding:0;outline:none;", ";&[disabled]{cursor:not-allowed;}"], /*#__PURE__*/addFocus({
  outlineSize: '0.063rem',
  outlineOffset: '-0.125rem',
  outlineColor: /*#__PURE__*/"var(".concat(tokens.outlineFocusColor, ")"),
  outlineRadius: 'calc(var(--plasma_private-outline-radius) - 0.063rem)'
}));
