// @salutejs/plasma-web/es/components/Tabs/Tabs.perftest.js
var _IconClock;
import React from 'react';
import { IconClock } from '@salutejs/plasma-icons';
import { Tabs, TabItem, TabsController } from '.';
var defaultProps = {
  itemsNumber: 4,
  disabled: false,
  stretch: true,
  label: 'Label'
};
var items = [0, 0, 0, 0];
var func = function func() {};
export var Default = function Default() {
  var disabled = defaultProps.disabled,
    stretch = defaultProps.stretch,
    label = defaultProps.label;
  return /*#__PURE__*/React.createElement(Tabs, {
    stretch: stretch,
    disabled: disabled
  }, items.map(function (_, i) {
    return /*#__PURE__*/React.createElement(TabItem, {
      key: "item:".concat(i),
      isActive: i === 0,
      tabIndex: -1,
      contentLeft: _IconClock || (_IconClock = /*#__PURE__*/React.createElement(IconClock, {
        color: "inherit"
      })),
      onClick: func,
      onFocus: func,
      onBlur: func
    }, label);
  }));
};
var arrowsProps = {
  disabled: false,
  stretch: true
};
var items2 = [{
  label: 'Label',
  contentLeft: /*#__PURE__*/React.createElement(IconClock, {
    color: "inherit"
  })
}];
export var Arrows = function Arrows() {
  var disabled = arrowsProps.disabled,
    stretch = arrowsProps.stretch;
  return /*#__PURE__*/React.createElement(TabsController, {
    items: items2,
    index: 0,
    onIndexChange: func,
    stretch: stretch,
    disabled: disabled
  });
};
