// @salutejs/plasma-new-hope/styled-components/cjs/components/Segment/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "SegmentProvider", {
  enumerable: true,
  get: function get() {
    return _SegmentProvider.SegmentProvider;
  }
});
Object.defineProperty(exports, "segmentClasses", {
  enumerable: true,
  get: function get() {
    return _tokens.classes;
  }
});
Object.defineProperty(exports, "segmentGroupConfig", {
  enumerable: true,
  get: function get() {
    return _ui.segmentGroupConfig;
  }
});
Object.defineProperty(exports, "segmentGroupRoot", {
  enumerable: true,
  get: function get() {
    return _ui.segmentGroupRoot;
  }
});
Object.defineProperty(exports, "segmentItemConfig", {
  enumerable: true,
  get: function get() {
    return _ui.segmentItemConfig;
  }
});
Object.defineProperty(exports, "segmentItemRoot", {
  enumerable: true,
  get: function get() {
    return _ui.segmentItemRoot;
  }
});
Object.defineProperty(exports, "segmentTokens", {
  enumerable: true,
  get: function get() {
    return _tokens.tokens;
  }
});
Object.defineProperty(exports, "useSegment", {
  enumerable: true,
  get: function get() {
    return _SegmentProvider.useSegment;
  }
});
var _ui = /*#__PURE__*/require("./ui");
var _SegmentProvider = /*#__PURE__*/require("./SegmentProvider");
var _tokens = /*#__PURE__*/require("./tokens");
