// @salutejs/plasma-tokens/es/brands/stylesSalute/index.js
// Generated by robots, do not change this manually!
/** Основной цвет текста */
export var textPrimaryHover = 'var(--text-primary-hover, #FFFFFF93)';
/** Основной цвет текста */
export var textPrimaryActive = 'var(--text-primary-active, #FFFFFFC4)';
/** Основной цвет текста */
export var textPrimary = 'var(--text-primary, #FFFFFFF5)';
/** Вторичный цвет текста */
export var textSecondaryHover = 'var(--text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста */
export var textSecondaryActive = 'var(--text-secondary-active, #FFFFFFAB)';
/** Вторичный цвет текста */
export var textSecondary = 'var(--text-secondary, #FFFFFF8F)';
/** Третичный цвет текста */
export var textTertiaryHover = 'var(--text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста */
export var textTertiaryActive = 'var(--text-tertiary-active, #FFFFFF56)';
/** Третичный цвет текста */
export var textTertiary = 'var(--text-tertiary, #FFFFFF47)';
/** Сплошной наборный текст */
export var textParagraphHover = 'var(--text-paragraph-hover, #FFFFFF7A)';
/** Сплошной наборный текст */
export var textParagraphActive = 'var(--text-paragraph-active, #FFFFFFA3)';
/** Сплошной наборный текст */
export var textParagraph = 'var(--text-paragraph, #FFFFFFCC)';
/** Цвет успеха */
export var textPositiveHover = 'var(--text-positive-hover, #2ACB47FF)';
/** Цвет успеха */
export var textPositiveActive = 'var(--text-positive-active, #1F9835FF)';
/** Цвет успеха */
export var textPositive = 'var(--text-positive, #24B23E)';
/** Цвет предупреждения */
export var textWarningHover = 'var(--text-warning-hover, #FF8442FF)';
/** Цвет предупреждения */
export var textWarningActive = 'var(--text-warning-active, #FF5D05FF)';
/** Цвет предупреждения */
export var textWarning = 'var(--text-warning, #FF7024)';
/** Цвет ошибки */
export var textNegativeHover = 'var(--text-negative-hover, #FF475AFF)';
/** Цвет ошибки */
export var textNegativeActive = 'var(--text-negative-active, #FF0A23FF)';
/** Цвет ошибки */
export var textNegative = 'var(--text-negative, #FF293E)';
/** Акцентный цвет Афины */
export var textAccentAthenaHover = 'var(--text-accent-athena-hover, #66BCF5FF)';
/** Акцентный цвет Афины */
export var textAccentAthenaActive = 'var(--text-accent-athena-active, #0E8ADDFF)';
/** Акцентный цвет Афины */
export var textAccentAthena = 'var(--text-accent-athena, #199AF0)';
/** Акцентный цвет Джой */
export var textAccentJoyHover = 'var(--text-accent-joy-hover, #E4BDFFFF)';
/** Акцентный цвет Джой */
export var textAccentJoyActive = 'var(--text-accent-joy-active, #B84DFFFF)';
/** Акцентный цвет Джой */
export var textAccentJoy = 'var(--text-accent-joy, #C46BFF)';
/** Акцентный цвет B2E-ассистента */
export var textAccentB2EHover = 'var(--text-accent-b2-e-hover, #2ACB47FF)';
/** Акцентный цвет B2E-ассистента */
export var textAccentB2EActive = 'var(--text-accent-b2-e-active, #1F9835FF)';
/** Акцентный цвет B2E-ассистента */
export var textAccentB2E = 'var(--text-accent-b2-e, #24B23E)';
/** Акцентный цвет бренда */
export var textAccentBrandHover = 'var(--text-accent-brand-hover, #A3C3FFFF)';
/** Акцентный цвет бренда */
export var textAccentBrandActive = 'var(--text-accent-brand-active, #337AFFFF)';
/** Акцентный цвет бренда */
export var textAccentBrand = 'var(--text-accent-brand, #528EFF)';
/** Градиент Афины */
export var textGradientAthenaHover = 'var(--text-gradient-athena-hover, #CCCCCCFF)';
/** Градиент Афины */
export var textGradientAthenaActive = 'var(--text-gradient-athena-active, #E6E6E6FF)';
/** Градиент Афины */
export var textGradientAthena = 'var(--text-gradient-athena, linear-gradient(93.97deg, #52BAFF 6.49%, #ADADFF 93.51%))';
/** Градиент Джой */
export var textGradientJoyHover = 'var(--text-gradient-joy-hover, #CCCCCCFF)';
/** Градиент Джой */
export var textGradientJoyActive = 'var(--text-gradient-joy-active, #E6E6E6FF)';
/** Градиент Джой */
export var textGradientJoy = 'var(--text-gradient-joy, linear-gradient(93.97deg, #FD9C68 6.49%, #F391E2 93.51%))';
/** Градиент B2E-ассистента */
export var textGradientB2EHover = 'var(--text-gradient-b2-e-hover, #CCCCCCFF)';
/** Градиент B2E-ассистента */
export var textGradientB2EActive = 'var(--text-gradient-b2-e-active, #E6E6E6FF)';
/** Градиент B2E-ассистента */
export var textGradientB2E = 'var(--text-gradient-b2-e, linear-gradient(89.83deg, #24B23E 0%, #1CC9C9 100%))';
/** Градиент бренда */
export var textGradientBrandHover = 'var(--text-gradient-brand-hover, #CCCCCCFF)';
/** Градиент бренда */
export var textGradientBrandActive = 'var(--text-gradient-brand-active, #E6E6E6FF)';
/** Градиент бренда */
export var textGradientBrand = 'var(--text-gradient-brand, linear-gradient(89.83deg, #528EFF 0%, #04C6C9 100%))';
/** Градиент */
export var textGradientMainHover = 'var(--text-gradient-main-hover, #CCCCCCFF)';
/** Градиент */
export var textGradientMainActive = 'var(--text-gradient-main-active, #E6E6E6FF)';
/** Градиент */
export var textGradientMain = 'var(--text-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #1CC9C9 100%))';
/** Акцентный цвет */
export var textAccentMainHover = 'var(--text-accent-main-hover, #2ACB47FF)';
/** Акцентный цвет */
export var textAccentMainActive = 'var(--text-accent-main-active, #1F9835FF)';
/** Акцентный цвет */
export var textAccentMain = 'var(--text-accent-main, #24B23E)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimaryHover = 'var(--on-dark-text-primary-hover, #FFFFFF93)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimaryActive = 'var(--on-dark-text-primary-active, #FFFFFFC4)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimary = 'var(--on-dark-text-primary, #FFFFFFF5)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondaryHover = 'var(--on-dark-text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondaryActive = 'var(--on-dark-text-secondary-active, #FFFFFFAB)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondary = 'var(--on-dark-text-secondary, #FFFFFF8F)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiaryHover = 'var(--on-dark-text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiaryActive = 'var(--on-dark-text-tertiary-active, #FFFFFF56)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiary = 'var(--on-dark-text-tertiary, #FFFFFF47)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraphHover = 'var(--on-dark-text-paragraph-hover, #FFFFFF7A)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraphActive = 'var(--on-dark-text-paragraph-active, #FFFFFFA3)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraph = 'var(--on-dark-text-paragraph, #FFFFFFCC)';
/** Цвет успеха на темном фоне */
export var onDarkTextPositiveHover = 'var(--on-dark-text-positive-hover, #2ACB47FF)';
/** Цвет успеха на темном фоне */
export var onDarkTextPositiveActive = 'var(--on-dark-text-positive-active, #1F9835FF)';
/** Цвет успеха на темном фоне */
export var onDarkTextPositive = 'var(--on-dark-text-positive, #24B23E)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarningHover = 'var(--on-dark-text-warning-hover, #FF8442FF)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarningActive = 'var(--on-dark-text-warning-active, #FF5D05FF)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarning = 'var(--on-dark-text-warning, #FF7024)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegativeHover = 'var(--on-dark-text-negative-hover, #FF475AFF)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegativeActive = 'var(--on-dark-text-negative-active, #FF0A23FF)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegative = 'var(--on-dark-text-negative, #FF293E)';
/** Акцентный цвет Афины на темном */
export var onDarkTextAccentAthenaHover = 'var(--on-dark-text-accent-athena-hover, #66BCF5FF)';
/** Акцентный цвет Афины на темном */
export var onDarkTextAccentAthenaActive = 'var(--on-dark-text-accent-athena-active, #0E8ADDFF)';
/** Акцентный цвет Афины на темном */
export var onDarkTextAccentAthena = 'var(--on-dark-text-accent-athena, #199AF0)';
/** Акцентный цвет Джой на темном фоне */
export var onDarkTextAccentJoyHover = 'var(--on-dark-text-accent-joy-hover, #E4BDFFFF)';
/** Акцентный цвет Джой на темном фоне */
export var onDarkTextAccentJoyActive = 'var(--on-dark-text-accent-joy-active, #B84DFFFF)';
/** Акцентный цвет Джой на темном фоне */
export var onDarkTextAccentJoy = 'var(--on-dark-text-accent-joy, #C46BFF)';
/** Акцентный цвет B2E-ассистента на темном фоне */
export var onDarkTextAccentB2EHover = 'var(--on-dark-text-accent-b2-e-hover, #2ACB47FF)';
/** Акцентный цвет B2E-ассистента на темном фоне */
export var onDarkTextAccentB2EActive = 'var(--on-dark-text-accent-b2-e-active, #1F9835FF)';
/** Акцентный цвет B2E-ассистента на темном фоне */
export var onDarkTextAccentB2E = 'var(--on-dark-text-accent-b2-e, #24B23E)';
/** Акцентный цвет бренда на темном фоне */
export var onDarkTextAccentBrandHover = 'var(--on-dark-text-accent-brand-hover, #A3C3FFFF)';
/** Акцентный цвет бренда на темном фоне */
export var onDarkTextAccentBrandActive = 'var(--on-dark-text-accent-brand-active, #337AFFFF)';
/** Акцентный цвет бренда на темном фоне */
export var onDarkTextAccentBrand = 'var(--on-dark-text-accent-brand, #528EFF)';
/** Градиент Афины на темном фоне */
export var onDarkTextGradientAthenaHover = 'var(--on-dark-text-gradient-athena-hover, #CCCCCCFF)';
/** Градиент Афины на темном фоне */
export var onDarkTextGradientAthenaActive = 'var(--on-dark-text-gradient-athena-active, #E6E6E6FF)';
/** Градиент Афины на темном фоне */
export var onDarkTextGradientAthena = 'var(--on-dark-text-gradient-athena, linear-gradient(93.97deg, #52BAFF 6.49%, #ADADFF 93.51%))';
/** Градиент Джой на темном фоне */
export var onDarkTextGradientJoyHover = 'var(--on-dark-text-gradient-joy-hover, #CCCCCCFF)';
/** Градиент Джой на темном фоне */
export var onDarkTextGradientJoyActive = 'var(--on-dark-text-gradient-joy-active, #E6E6E6FF)';
/** Градиент Джой на темном фоне */
export var onDarkTextGradientJoy = 'var(--on-dark-text-gradient-joy, linear-gradient(93.97deg, #FD9C68 6.49%, #F391E2 93.51%))';
/** Градиент B2E-ассистента на темном фоне */
export var onDarkTextGradientB2EHover = 'var(--on-dark-text-gradient-b2-e-hover, #CCCCCCFF)';
/** Градиент B2E-ассистента на темном фоне */
export var onDarkTextGradientB2EActive = 'var(--on-dark-text-gradient-b2-e-active, #E6E6E6FF)';
/** Градиент B2E-ассистента на темном фоне */
export var onDarkTextGradientB2E = 'var(--on-dark-text-gradient-b2-e, linear-gradient(89.83deg, #24B23E 0%, #1CC9C9 100%))';
/** Градиент бренда на темном фоне */
export var onDarkTextGradientBrandHover = 'var(--on-dark-text-gradient-brand-hover, #CCCCCCFF)';
/** Градиент бренда на темном фоне */
export var onDarkTextGradientBrandActive = 'var(--on-dark-text-gradient-brand-active, #E6E6E6FF)';
/** Градиент бренда на темном фоне */
export var onDarkTextGradientBrand = 'var(--on-dark-text-gradient-brand, linear-gradient(89.83deg, #528EFF 0%, #04C6C9 100%))';
/** Градиент на темном фоне */
export var onDarkTextGradientMainHover = 'var(--on-dark-text-gradient-main-hover, #CCCCCCFF)';
/** Градиент на темном фоне */
export var onDarkTextGradientMainActive = 'var(--on-dark-text-gradient-main-active, #E6E6E6FF)';
/** Градиент на темном фоне */
export var onDarkTextGradientMain = 'var(--on-dark-text-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #1CC9C9 100%))';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccentMainHover = 'var(--on-dark-text-accent-main-hover, #2ACB47FF)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccentMainActive = 'var(--on-dark-text-accent-main-active, #1F9835FF)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccentMain = 'var(--on-dark-text-accent-main, #24B23E)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimaryHover = 'var(--on-light-text-primary-hover, #08080893)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimaryActive = 'var(--on-light-text-primary-active, #080808C4)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimary = 'var(--on-light-text-primary, rgba(8, 8, 8, 0.96))';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondaryHover = 'var(--on-light-text-secondary-hover, #080808FF)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondaryActive = 'var(--on-light-text-secondary-active, #080808AB)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondary = 'var(--on-light-text-secondary, rgba(8, 8, 8, 0.56))';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiaryHover = 'var(--on-light-text-tertiary-hover, #080808FF)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiaryActive = 'var(--on-light-text-tertiary-active, #08080856)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiary = 'var(--on-light-text-tertiary, #08080847)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraphHover = 'var(--on-light-text-paragraph-hover, #0808087A)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraphActive = 'var(--on-light-text-paragraph-active, #080808A3)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraph = 'var(--on-light-text-paragraph, rgba(8, 8, 8, 0.8))';
/** Цвет успеха на светлом фоне */
export var onLightTextPositiveHover = 'var(--on-light-text-positive-hover, #1EB83AFF)';
/** Цвет успеха на светлом фоне */
export var onLightTextPositiveActive = 'var(--on-light-text-positive-active, #15842AFF)';
/** Цвет успеха на светлом фоне */
export var onLightTextPositive = 'var(--on-light-text-positive, #1A9E32)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarningHover = 'var(--on-light-text-warning-hover, #FB7223FF)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarningActive = 'var(--on-light-text-warning-active, #DC5304FF)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarning = 'var(--on-light-text-warning, #FA5F05)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegativeHover = 'var(--on-light-text-negative-hover, #FF475AFF)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegativeActive = 'var(--on-light-text-negative-active, #FF0A23FF)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegative = 'var(--on-light-text-negative, #FF293E)';
/** Акцентный цвет Афины на светлом фоне */
export var onLightTextAccentAthenaHover = 'var(--on-light-text-accent-athena-hover, #50B1F2FF)';
/** Акцентный цвет Афины на светлом фоне */
export var onLightTextAccentAthenaActive = 'var(--on-light-text-accent-athena-active, #0F7BC2FF)';
/** Акцентный цвет Афины на светлом фоне */
export var onLightTextAccentAthena = 'var(--on-light-text-accent-athena, #118CDF)';
/** Акцентный цвет Джой на светлом фоне */
export var onLightTextAccentJoyHover = 'var(--on-light-text-accent-joy-hover, #D7A4F9FF)';
/** Акцентный цвет Джой на светлом фоне */
export var onLightTextAccentJoyActive = 'var(--on-light-text-accent-joy-active, #A93BF1FF)';
/** Акцентный цвет Джой на светлом фоне */
export var onLightTextAccentJoy = 'var(--on-light-text-accent-joy, #B559F3)';
/** Акцентный цвет B2E-ассистента на светлом фоне */
export var onLightTextAccentB2EHover = 'var(--on-light-text-accent-b2-e-hover, #1EB83AFF)';
/** Акцентный цвет B2E-ассистента на светлом фоне */
export var onLightTextAccentB2EActive = 'var(--on-light-text-accent-b2-e-active, #15842AFF)';
/** Акцентный цвет B2E-ассистента на светлом фоне */
export var onLightTextAccentB2E = 'var(--on-light-text-accent-b2-e, #1A9E32)';
/** Акцентный цвет бренда на светлом фоне */
export var onLightTextAccentBrandHover = 'var(--on-light-text-accent-brand-hover, #90B6FEFF)';
/** Акцентный цвет бренда на светлом фоне */
export var onLightTextAccentBrandActive = 'var(--on-light-text-accent-brand-active, #216EFDFF)';
/** Акцентный цвет бренда на светлом фоне */
export var onLightTextAccentBrand = 'var(--on-light-text-accent-brand, #3F81FD)';
/** Градиент Афины на светлом фоне */
export var onLightTextGradientAthenaHover = 'var(--on-light-text-gradient-athena-hover, #CCCCCCFF)';
/** Градиент Афины на светлом фоне */
export var onLightTextGradientAthenaActive = 'var(--on-light-text-gradient-athena-active, #E6E6E6FF)';
/** Градиент Афины на светлом фоне */
export var onLightTextGradientAthena = 'var(--on-light-text-gradient-athena, linear-gradient(89.83deg, #118CDF 0%, #7A7AFF 100%))';
/** Градиент Джой на светлом фоне */
export var onLightTextGradientJoyHover = 'var(--on-light-text-gradient-joy-hover, #CCCCCCFF)';
/** Градиент Джой на светлом фоне */
export var onLightTextGradientJoyActive = 'var(--on-light-text-gradient-joy-active, #E6E6E6FF)';
/** Градиент Джой на светлом фоне */
export var onLightTextGradientJoy = 'var(--on-light-text-gradient-joy, linear-gradient(89.83deg, #FCA349 0%, #FF52C5 100%))';
/** Градиент B2E-ассистента на светлом фоне */
export var onLightTextGradientB2EHover = 'var(--on-light-text-gradient-b2-e-hover, #CCCCCCFF)';
/** Градиент B2E-ассистента на светлом фоне */
export var onLightTextGradientB2EActive = 'var(--on-light-text-gradient-b2-e-active, #E6E6E6FF)';
/** Градиент B2E-ассистента на светлом фоне */
export var onLightTextGradientB2E = 'var(--on-light-text-gradient-b2-e, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Акцентный цвет бренда на светлом фоне */
export var onLightTextGradientBrandHover = 'var(--on-light-text-gradient-brand-hover, #CCCCCCFF)';
/** Акцентный цвет бренда на светлом фоне */
export var onLightTextGradientBrandActive = 'var(--on-light-text-gradient-brand-active, #E6E6E6FF)';
/** Акцентный цвет бренда на светлом фоне */
export var onLightTextGradientBrand = 'var(--on-light-text-gradient-brand, linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%))';
/** Градиент на светлом фоне */
export var onLightTextGradientMainHover = 'var(--on-light-text-gradient-main-hover, #CCCCCCFF)';
/** Градиент на светлом фоне */
export var onLightTextGradientMainActive = 'var(--on-light-text-gradient-main-active, #E6E6E6FF)';
/** Градиент на светлом фоне */
export var onLightTextGradientMain = 'var(--on-light-text-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccentMainHover = 'var(--on-light-text-accent-main-hover, #1EB83AFF)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccentMainActive = 'var(--on-light-text-accent-main-active, #15842AFF)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccentMain = 'var(--on-light-text-accent-main, #1A9E32)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimaryHover = 'var(--inverse-text-primary-hover, #08080893)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimaryActive = 'var(--inverse-text-primary-active, #080808C4)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimary = 'var(--inverse-text-primary, rgba(8, 8, 8, 0.96))';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondaryHover = 'var(--inverse-text-secondary-hover, #080808FF)';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondaryActive = 'var(--inverse-text-secondary-active, #080808AB)';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondary = 'var(--inverse-text-secondary, rgba(8, 8, 8, 0.56))';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiaryHover = 'var(--inverse-text-tertiary-hover, #080808FF)';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiaryActive = 'var(--inverse-text-tertiary-active, #08080856)';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiary = 'var(--inverse-text-tertiary, #08080847)';
/** Инвертированный cплошной наборный текст */
export var inverseTextParagraphHover = 'var(--inverse-text-paragraph-hover, #0808087A)';
/** Инвертированный cплошной наборный текст */
export var inverseTextParagraphActive = 'var(--inverse-text-paragraph-active, #080808A3)';
/** Инвертированный cплошной наборный текст */
export var inverseTextParagraph = 'var(--inverse-text-paragraph, rgba(8, 8, 8, 0.8))';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimaryHover = 'var(--surface-solid-primary-hover, #1C1C1CFF)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimaryActive = 'var(--surface-solid-primary-active, #121212FF)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimary = 'var(--surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondaryHover = 'var(--surface-solid-secondary-hover, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondaryActive = 'var(--surface-solid-secondary-active, #212121FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondary = 'var(--surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiaryHover = 'var(--surface-solid-tertiary-hover, #404040FF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiaryActive = 'var(--surface-solid-tertiary-active, #303030FF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiary = 'var(--surface-solid-tertiary, #363636)';
/** Основной фон для карточек */
export var surfaceSolidCardHover = 'var(--surface-solid-card-hover, #1C1C1CFF)';
/** Основной фон для карточек */
export var surfaceSolidCardActive = 'var(--surface-solid-card-active, #121212FF)';
/** Основной фон для карточек */
export var surfaceSolidCard = 'var(--surface-solid-card, #171717)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefaultHover = 'var(--surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefaultActive = 'var(--surface-solid-default-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefault = 'var(--surface-solid-default, #FFFFFF)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimaryHover = 'var(--surface-transparent-primary-hover, #FFFFFF1F)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimaryActive = 'var(--surface-transparent-primary-active, #FFFFFF0A)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimary = 'var(--surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondaryHover = 'var(--surface-transparent-secondary-hover, #FFFFFF3D)';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondaryActive = 'var(--surface-transparent-secondary-active, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondary = 'var(--surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiaryHover = 'var(--surface-transparent-tertiary-hover, #FFFFFF52)';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiaryActive = 'var(--surface-transparent-tertiary-active, #FFFFFF24)';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек */
export var surfaceTransparentCardHover = 'var(--surface-transparent-card-hover, #FFFFFF1F)';
/** Прозрачный фон для карточек */
export var surfaceTransparentCardActive = 'var(--surface-transparent-card-active, #FFFFFF0A)';
/** Прозрачный фон для карточек */
export var surfaceTransparentCard = 'var(--surface-transparent-card, rgba(255,255,255,0.06))';
/** Фон поверхности/контрола без заливки */
export var surfaceClearHover = 'var(--surface-clear-hover, #00000000)';
/** Фон поверхности/контрола без заливки */
export var surfaceClearActive = 'var(--surface-clear-active, #00000000)';
/** Фон поверхности/контрола без заливки */
export var surfaceClear = 'var(--surface-clear, rgba(0, 0, 0, 0))';
/** Цвет фона поверхности/контрола успех */
export var surfacePositiveHover = 'var(--surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositiveActive = 'var(--surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositive = 'var(--surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarningHover = 'var(--surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarningActive = 'var(--surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarning = 'var(--surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegativeHover = 'var(--surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegativeActive = 'var(--surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegative = 'var(--surface-negative, #FF293E)';
/** Акцентный цвет поверхности/контролла Афины */
export var surfaceAccentAthenaHover = 'var(--surface-accent-athena-hover, #1798EEFF)';
/** Акцентный цвет поверхности/контролла Афины */
export var surfaceAccentAthenaActive = 'var(--surface-accent-athena-active, #1086D5FF)';
/** Акцентный цвет поверхности/контролла Афины */
export var surfaceAccentAthena = 'var(--surface-accent-athena, #118CDF)';
/** Акцентный цвет поверхности/контролла Джой */
export var surfaceAccentJoyHover = 'var(--surface-accent-joy-hover, #C275F5FF)';
/** Акцентный цвет поверхности/контролла Джой */
export var surfaceAccentJoyActive = 'var(--surface-accent-joy-active, #B14FF3FF)';
/** Акцентный цвет поверхности/контролла Джой */
export var surfaceAccentJoy = 'var(--surface-accent-joy, #B559F3)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
export var surfaceAccentB2EHover = 'var(--surface-accent-b2-e-hover, #1DAF37FF)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
export var surfaceAccentB2EActive = 'var(--surface-accent-b2-e-active, #18952FFF)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
export var surfaceAccentB2E = 'var(--surface-accent-b2-e, #1A9E32)';
/** Акцентный цвет поверхности/контролла бренда */
export var surfaceAccentBrandHover = 'var(--surface-accent-brand-hover, #5D95FDFF)';
/** Акцентный цвет поверхности/контролла бренда */
export var surfaceAccentBrandActive = 'var(--surface-accent-brand-active, #357BFDFF)';
/** Акцентный цвет поверхности/контролла бренда */
export var surfaceAccentBrand = 'var(--surface-accent-brand, #3F81FD)';
/** Цвет баббла сообщения от пользователя */
export var surfaceSpeechBubbleSentHover = 'var(--surface-speech-bubble-sent-hover, #08080866)';
/** Цвет баббла сообщения от пользователя */
export var surfaceSpeechBubbleSentActive = 'var(--surface-speech-bubble-sent-active, #08080838)';
/** Цвет баббла сообщения от пользователя */
export var surfaceSpeechBubbleSent = 'var(--surface-speech-bubble-sent, #08080847)';
/** Цвет баббла сообщения от ассистента */
export var surfaceSpeechBubbleReceivedHover = 'var(--surface-speech-bubble-received-hover, #FFFFFF1F)';
/** Цвет баббла сообщения от ассистента */
export var surfaceSpeechBubbleReceivedActive = 'var(--surface-speech-bubble-received-active, #FFFFFF0A)';
/** Цвет баббла сообщения от ассистента */
export var surfaceSpeechBubbleReceived = 'var(--surface-speech-bubble-received, #FFFFFF0F)';
/** Фон для скелетона */
export var surfaceSkeletonHover = 'var(--surface-skeleton-hover, #FFFFFFFF)';
/** Фон для скелетона */
export var surfaceSkeletonActive = 'var(--surface-skeleton-active, #FFFFFFFF)';
/** Фон для скелетона */
export var surfaceSkeleton = 'var(--surface-skeleton, linear-gradient(90deg, rgba(255, 255, 255, 0.00) 0%, rgba(255, 255, 255, 0.01) 17.19%, rgba(255, 255, 255, 0.05) 40.24%, rgba(255, 255, 255, 0.06) 51.17%, rgba(255, 255, 255, 0.05) 62.12%, rgba(255, 255, 255, 0.01) 82.81%, rgba(255, 255, 255, 0.00) 100%))';
/** Акцентный цвет поверхности/контролла */
export var surfaceAccentMainHover = 'var(--surface-accent-main-hover, #1DAF37FF)';
/** Акцентный цвет поверхности/контролла */
export var surfaceAccentMainActive = 'var(--surface-accent-main-active, #18952FFF)';
/** Акцентный цвет поверхности/контролла */
export var surfaceAccentMain = 'var(--surface-accent-main, #1A9E32)';
export var surfaceGradientMainHover = 'var(--surface-gradient-main-hover, #FFFFFFFF)';
export var surfaceGradientMainActive = 'var(--surface-gradient-main-active, #FFFFFFFF)';
export var surfaceGradientMain = 'var(--surface-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimaryHover = 'var(--on-dark-surface-solid-primary-hover, #1C1C1CFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimaryActive = 'var(--on-dark-surface-solid-primary-active, #121212FF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности/контрола на темном */
export var onDarkSurfaceSolidSecondaryHover = 'var(--on-dark-surface-solid-secondary-hover, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном */
export var onDarkSurfaceSolidSecondaryActive = 'var(--on-dark-surface-solid-secondary-active, #212121FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном */
export var onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiaryHover = 'var(--on-dark-surface-solid-tertiary-hover, #404040FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiaryActive = 'var(--on-dark-surface-solid-tertiary-active, #303030FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #363636)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCardHover = 'var(--on-dark-surface-solid-card-hover, #1C1C1CFF)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCardActive = 'var(--on-dark-surface-solid-card-active, #121212FF)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #171717)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefaultHover = 'var(--on-dark-surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefaultActive = 'var(--on-dark-surface-solid-default-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #FFFFFF)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimaryHover = 'var(--on-dark-surface-transparent-primary-hover, #FFFFFF1F)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimaryActive = 'var(--on-dark-surface-transparent-primary-active, #FFFFFF0A)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondaryHover = 'var(--on-dark-surface-transparent-secondary-hover, #FFFFFF3D)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondaryActive = 'var(--on-dark-surface-transparent-secondary-active, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiaryHover = 'var(--on-dark-surface-transparent-tertiary-hover, #FFFFFF52)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiaryActive = 'var(--on-dark-surface-transparent-tertiary-active, #FFFFFF24)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCardHover = 'var(--on-dark-surface-transparent-card-hover, #FFFFFF1F)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCardActive = 'var(--on-dark-surface-transparent-card-active, #FFFFFF0A)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, rgba(255,255,255,0.06))';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositiveHover = 'var(--on-dark-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositiveActive = 'var(--on-dark-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositive = 'var(--on-dark-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение на темном */
export var onDarkSurfaceWarningHover = 'var(--on-dark-surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение на темном */
export var onDarkSurfaceWarningActive = 'var(--on-dark-surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола предупреждение на темном */
export var onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка на темном */
export var onDarkSurfaceNegativeHover = 'var(--on-dark-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на темном */
export var onDarkSurfaceNegativeActive = 'var(--on-dark-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка на темном */
export var onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293E)';
/** Акцентный цвет поверхности/контролла Афины */
export var onDarkSurfaceAccentAthenaHover = 'var(--on-dark-surface-accent-athena-hover, #1798EEFF)';
/** Акцентный цвет поверхности/контролла Афины */
export var onDarkSurfaceAccentAthenaActive = 'var(--on-dark-surface-accent-athena-active, #1086D5FF)';
/** Акцентный цвет поверхности/контролла Афины */
export var onDarkSurfaceAccentAthena = 'var(--on-dark-surface-accent-athena, #118CDF)';
/** Акцентный цвет поверхности/контролла Джой */
export var onDarkSurfaceAccentJoyHover = 'var(--on-dark-surface-accent-joy-hover, #C275F5FF)';
/** Акцентный цвет поверхности/контролла Джой */
export var onDarkSurfaceAccentJoyActive = 'var(--on-dark-surface-accent-joy-active, #B14FF3FF)';
/** Акцентный цвет поверхности/контролла Джой */
export var onDarkSurfaceAccentJoy = 'var(--on-dark-surface-accent-joy, #B559F3)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
export var onDarkSurfaceAccentB2EHover = 'var(--on-dark-surface-accent-b2-e-hover, #1DAF37FF)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
export var onDarkSurfaceAccentB2EActive = 'var(--on-dark-surface-accent-b2-e-active, #18952FFF)';
/** Акцентный цвет поверхности/контролла B2E-ассистента */
export var onDarkSurfaceAccentB2E = 'var(--on-dark-surface-accent-b2-e, #1A9E32)';
/** Акцентный цвет поверхности/контролла бренда */
export var onDarkSurfaceAccentBrandHover = 'var(--on-dark-surface-accent-brand-hover, #5D95FDFF)';
/** Акцентный цвет поверхности/контролла бренда */
export var onDarkSurfaceAccentBrandActive = 'var(--on-dark-surface-accent-brand-active, #357BFDFF)';
/** Акцентный цвет поверхности/контролла бренда */
export var onDarkSurfaceAccentBrand = 'var(--on-dark-surface-accent-brand, #3F81FD)';
/** Акцентный цвет поверхности/контролла */
export var onDarkSurfaceAccentMainHover = 'var(--on-dark-surface-accent-main-hover, #1DAF37FF)';
/** Акцентный цвет поверхности/контролла */
export var onDarkSurfaceAccentMainActive = 'var(--on-dark-surface-accent-main-active, #18952FFF)';
/** Акцентный цвет поверхности/контролла */
export var onDarkSurfaceAccentMain = 'var(--on-dark-surface-accent-main, #1A9E32)';
export var onDarkSurfaceGradientMainHover = 'var(--on-dark-surface-gradient-main-hover, #FFFFFFFF)';
export var onDarkSurfaceGradientMainActive = 'var(--on-dark-surface-gradient-main-active, #FFFFFFFF)';
export var onDarkSurfaceGradientMain = 'var(--on-dark-surface-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Основной непрозрачный фон поверхности/контрола на светлом */
export var onLightSurfaceSolidPrimaryHover = 'var(--on-light-surface-solid-primary-hover, #FAFAFAFF)';
/** Основной непрозрачный фон поверхности/контрола на светлом */
export var onLightSurfaceSolidPrimaryActive = 'var(--on-light-surface-solid-primary-active, #F0F0F0FF)';
/** Основной непрозрачный фон поверхности/контрола на светлом */
export var onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F5F5F5)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом */
export var onLightSurfaceSolidSecondaryHover = 'var(--on-light-surface-solid-secondary-hover, #F7F7F7FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом */
export var onLightSurfaceSolidSecondaryActive = 'var(--on-light-surface-solid-secondary-active, #E8E8E8FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом */
export var onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #ECECEC)';
/** Третичный непрозрачный фон поверхности/контрола на светлом */
export var onLightSurfaceSolidTertiaryHover = 'var(--on-light-surface-solid-tertiary-hover, #E8E8E8FF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом */
export var onLightSurfaceSolidTertiaryActive = 'var(--on-light-surface-solid-tertiary-active, #D9D9D9FF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом */
export var onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #DDDDDD)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCardHover = 'var(--on-light-surface-solid-card-hover, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCardActive = 'var(--on-light-surface-solid-card-active, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefaultHover = 'var(--on-light-surface-solid-default-hover, #0D0D0DFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefaultActive = 'var(--on-light-surface-solid-default-active, #030303FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #080808)';
/** Основной прозрачный фон поверхности/контрола на светлом */
export var onLightSurfaceTransparentPrimaryHover = 'var(--on-light-surface-transparent-primary-hover, #08080812)';
/** Основной прозрачный фон поверхности/контрола на светлом */
export var onLightSurfaceTransparentPrimaryActive = 'var(--on-light-surface-transparent-primary-active, #08080803)';
/** Основной прозрачный фон поверхности/контрола на светлом */
export var onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, #08080808)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondaryHover = 'var(--on-light-surface-transparent-secondary-hover, #0808081F)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondaryActive = 'var(--on-light-surface-transparent-secondary-active, #0808080A)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiaryHover = 'var(--on-light-surface-transparent-tertiary-hover, #0808083D)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiaryActive = 'var(--on-light-surface-transparent-tertiary-active, #0808080F)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Прозрачный цвет поверхности/контролла на светлом фоне */
export var onLightSurfaceTransparentDeepHover = 'var(--on-light-surface-transparent-deep-hover, #080808C2)';
/** Прозрачный цвет поверхности/контролла на светлом фоне */
export var onLightSurfaceTransparentDeepActive = 'var(--on-light-surface-transparent-deep-active, #08080894)';
/** Прозрачный цвет поверхности/контролла на светлом фоне */
export var onLightSurfaceTransparentDeep = 'var(--on-light-surface-transparent-deep, #080808A2)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCardHover = 'var(--on-light-surface-transparent-card-hover, #FFFFFFFF)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCardActive = 'var(--on-light-surface-transparent-card-active, #FFFFFFFF)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, #FFFFFFFF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositiveHover = 'var(--on-light-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositiveActive = 'var(--on-light-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositive = 'var(--on-light-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение на светлом */
export var onLightSurfaceWarningHover = 'var(--on-light-surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение на светлом */
export var onLightSurfaceWarningActive = 'var(--on-light-surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола предупреждение на светлом */
export var onLightSurfaceWarning = 'var(--on-light-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeHover = 'var(--on-light-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeActive = 'var(--on-light-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegative = 'var(--on-light-surface-negative, #FF293E)';
/** Акцентный цвет поверхности/контролла Афины на светлом фоне */
export var onLightSurfaceAccentAthenaHover = 'var(--on-light-surface-accent-athena-hover, #1798EEFF)';
/** Акцентный цвет поверхности/контролла Афины на светлом фоне */
export var onLightSurfaceAccentAthenaActive = 'var(--on-light-surface-accent-athena-active, #1086D5FF)';
/** Акцентный цвет поверхности/контролла Афины на светлом фоне */
export var onLightSurfaceAccentAthena = 'var(--on-light-surface-accent-athena, #118CDF)';
/** Акцентный цвет поверхности/контролла Джой на светлом */
export var onLightSurfaceAccentJoyHover = 'var(--on-light-surface-accent-joy-hover, #C275F5FF)';
/** Акцентный цвет поверхности/контролла Джой на светлом */
export var onLightSurfaceAccentJoyActive = 'var(--on-light-surface-accent-joy-active, #B14FF3FF)';
/** Акцентный цвет поверхности/контролла Джой на светлом */
export var onLightSurfaceAccentJoy = 'var(--on-light-surface-accent-joy, #B559F3)';
/** Акцентный цвет поверхности B2E-ассистента на светлом */
export var onLightSurfaceAccentB2EHover = 'var(--on-light-surface-accent-b2-e-hover, #1DAF37FF)';
/** Акцентный цвет поверхности B2E-ассистента на светлом */
export var onLightSurfaceAccentB2EActive = 'var(--on-light-surface-accent-b2-e-active, #18952FFF)';
/** Акцентный цвет поверхности B2E-ассистента на светлом */
export var onLightSurfaceAccentB2E = 'var(--on-light-surface-accent-b2-e, #1A9E32)';
/** Акцентный цвет поверхности/контролла бренда на светлом */
export var onLightSurfaceAccentBrandHover = 'var(--on-light-surface-accent-brand-hover, #5D95FDFF)';
/** Акцентный цвет поверхности/контролла бренда на светлом */
export var onLightSurfaceAccentBrandActive = 'var(--on-light-surface-accent-brand-active, #357BFDFF)';
/** Акцентный цвет поверхности/контролла бренда на светлом */
export var onLightSurfaceAccentBrand = 'var(--on-light-surface-accent-brand, #3F81FD)';
/** Акцентный цвет поверхности/контролла на светлом фоне */
export var onLightSurfaceAccentMainHover = 'var(--on-light-surface-accent-main-hover, #1DAF37FF)';
/** Акцентный цвет поверхности/контролла на светлом фоне */
export var onLightSurfaceAccentMainActive = 'var(--on-light-surface-accent-main-active, #18952FFF)';
/** Акцентный цвет поверхности/контролла на светлом фоне */
export var onLightSurfaceAccentMain = 'var(--on-light-surface-accent-main, #1A9E32)';
export var onLightSurfaceGradientMainHover = 'var(--on-light-surface-gradient-main-hover, #FFFFFFFF)';
export var onLightSurfaceGradientMainActive = 'var(--on-light-surface-gradient-main-active, #FFFFFFFF)';
export var onLightSurfaceGradientMain = 'var(--on-light-surface-gradient-main, linear-gradient(89.83deg, #24B23E 0%, #00A4D1 100%))';
/** Первичный фон */
export var backgroundPrimary = 'var(--background-primary, #000000FF)';
/** Вторичный фон */
export var backgroundSecondary = 'var(--background-secondary, #171717)';
/** Фон для модальной шторки */
export var backgroundModalSheet = 'var(--background-modal-sheet, linear-gradient(180deg, rgba(255, 255, 255, 0.1) 0%, rgba(1, 182, 255, 0) 50%)), var(--background-modal-sheet, #080808FF)';
/** Фон Афины на мобиле */
export var backgroundMobileAssistantAthena = 'var(--background-mobile-assistant-athena, linear-gradient(336.9deg, #143787 0%, rgba(8, 8, 8, 0) 64.95%)), var(--background-mobile-assistant-athena, radial-gradient(66.53% 100% at 73.33% 100%, rgba(110, 5, 193, 0.44) 0%, rgba(53, 19, 149, 0.1) 99.69%)), var(--background-mobile-assistant-athena, linear-gradient(180.03deg, rgba(8, 8, 8, 0) 50%, rgba(25, 63, 152, 0.41) 99.97%)), var(--background-mobile-assistant-athena, linear-gradient(270deg, rgba(39, 15, 107, 0.3) 0%, rgba(8, 8, 8, 0) 100%)), var(--background-mobile-assistant-athena, #080808FF)';
/** Фон Джой на мобиле */
export var backgroundMobileAssistantJoy = 'var(--background-mobile-assistant-joy, linear-gradient(336.9deg, rgba(255, 156, 101, 0.24) 0%, rgba(8, 8, 8, 0) 64.95%)), var(--background-mobile-assistant-joy, radial-gradient(66.53% 100% at 73.33% 100%, rgba(61, 19, 149, 0.34) 0%, rgba(19, 24, 149, 0.1) 99.69%)), var(--background-mobile-assistant-joy, linear-gradient(180.03deg, rgba(8, 8, 8, 0) 50%, rgba(255, 215, 179, 0.15) 99.97%)), var(--background-mobile-assistant-joy, linear-gradient(270deg, rgba(107, 15, 87, 0.2) 0%, rgba(8, 8, 8, 0) 100%)), var(--background-mobile-assistant-joy, #080808FF)';
/** Фоновый градиент B2E-ассистента на мобиле */
export var backgroundMobileAssistantB2E = 'var(--background-mobile-assistant-b2-e, linear-gradient(336.84deg, rgba(20, 116, 70, 0.6) 0%, rgba(8, 8, 8, 0) 64.88%)), var(--background-mobile-assistant-b2-e, radial-gradient(100% 100% at 75.89% 100%, rgba(0, 133, 255, 0.24) 0%, rgba(0, 71, 255, 0.03) 100%)), var(--background-mobile-assistant-b2-e, linear-gradient(180deg, rgba(8, 8, 8, 0) 50%, rgba(7, 71, 33, 0.3) 100%)), var(--background-mobile-assistant-b2-e, linear-gradient(270deg, #061621 0%, rgba(8, 8, 8, 0) 100%)), var(--background-mobile-assistant-b2-e, #080808FF)';
/** Фоновый градиент бренда на мобиле */
export var backgroundMobileAssistantBrand = 'var(--background-mobile-assistant-brand, linear-gradient(336.9deg, #184BAB 0%, rgba(8, 8, 8, 0) 64.95%)), var(--background-mobile-assistant-brand, radial-gradient(66.53% 100% at 73.33% 100%, rgba(22, 69, 164, 0.44) 0%, rgba(11, 39, 45, 0.1) 100%)), var(--background-mobile-assistant-brand, linear-gradient(180.03deg, rgba(8, 8, 8, 0) 50%, rgba(22, 69, 164, 0.41) 99.97%)), var(--background-mobile-assistant-brand, linear-gradient(269.47deg, rgba(22, 69, 164, 0.3) 0.52%, rgba(8, 8, 8, 0) 99.61%)), var(--background-mobile-assistant-brand, #080808FF)';
/** Фон ассистента на мобиле */
export var backgroundMobileAssistantMain = 'var(--background-mobile-assistant-main, linear-gradient(336.84deg, rgba(20, 116, 70, 0.6) 0%, rgba(8, 8, 8, 0) 64.88%)), var(--background-mobile-assistant-main, radial-gradient(100% 100% at 75.89% 100%, rgba(0, 133, 255, 0.24) 0%, rgba(0, 71, 255, 0.03) 100%)), var(--background-mobile-assistant-main, linear-gradient(180deg, rgba(8, 8, 8, 0) 50%, rgba(7, 71, 33, 0.3) 100%)), var(--background-mobile-assistant-main, linear-gradient(270deg, #061621 0%, rgba(8, 8, 8, 0) 100%)), var(--background-mobile-assistant-main, #080808FF)';
/** Цвет фона паранжи светлый на темном фоне */
export var onDarkOverlaySoft = 'var(--on-dark-overlay-soft, rgba(0, 0, 0, 0.56))';
/** Цвет фона паранжи темный на темном фоне */
export var onDarkOverlayHard = 'var(--on-dark-overlay-hard, #000000EB)';
/** Оверлей с блюром */
export var onDarkOverlayBlur = 'var(--on-dark-overlay-blur, #26262647)';
export var skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
export var skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
export var whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
export var whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
export var whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
export var blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
export var blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
export var blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use onLightSurfaceSolidDefault */
export var buttonBlack = 'var(--plasma-colors-button-black, var(--on-light-surface-solid-default))';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
export var buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, var(--on-light-surface-transparent-secondary))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
export var buttonWhite = 'var(--plasma-colors-button-white, var(--on-dark-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
export var buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, var(--on-dark-surface-transparent-secondary))';
/** @deprecated instead use textPrimary */
export var text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
export var primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
export var secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
export var tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use textParagraph */
export var paragraph = 'var(--plasma-colors-paragraph, var(--text-paragraph))';
/** @deprecated instead use backgroundPrimary */
export var background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textPositive */
export var success = 'var(--plasma-colors-success, var(--text-positive))';
/** @deprecated instead use textWarning */
export var warning = 'var(--plasma-colors-warning, var(--text-warning))';
/** @deprecated instead use textNegative */
export var critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use surfaceTransparentPrimary */
export var surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, var(--surface-transparent-primary))';
/** @deprecated instead use surfaceTransparentSecondary */
export var surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, var(--surface-transparent-secondary))';
/** @deprecated instead use surfaceTransparentTertiary */
export var surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, var(--surface-transparent-tertiary))';
/** @deprecated instead use surfaceSolidPrimary */
export var surfaceSolid01 = 'var(--plasma-colors-surface-solid01, var(--surface-solid-primary))';
/** @deprecated instead use surfaceSolidSecondary */
export var surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
export var surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use surfaceTransparentCard */
export var surfaceCard = 'var(--plasma-colors-surface-card, var(--surface-transparent-card))';
/** @deprecated instead use surfaceTransparentSecondary */
export var buttonSecondary = 'var(--plasma-colors-button-secondary, var(--surface-transparent-secondary))';
/** @deprecated instead use surfacePositive */
export var buttonSuccess = 'var(--plasma-colors-button-success, var(--surface-positive))';
/** @deprecated instead use surfaceWarning */
export var buttonWarning = 'var(--plasma-colors-button-warning, var(--surface-warning))';
/** @deprecated instead use surfaceNegative */
export var buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
