// @salutejs/plasma-tokens-b2b/es/themesValues/light.js
// Generated by robots, do not change this manually!
/** Базовый белый цвет совпадает с whitePrimary */
export var white = '#FFFFFF';
/** Первичный белый цвет */
export var whitePrimary = '#FFFFFF';
/** Вторичный белый цвет */
export var whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный белый цвет */
export var whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** Базовый черный цвет совпадает с blackPrimary */
export var black = '#080808';
/** Первичный черный цвет */
export var blackPrimary = '#080808';
/** Вторичный черныйцвет */
export var blackSecondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный черный цвет */
export var blackTertiary = 'rgba(8, 8, 8, 0.28)';
/** Чёрный непрозрачный, не зависит от темы */
export var dark01 = '#171717';
/** Чёрный непрозрачный, не зависит от темы */
export var dark02 = '#232323';
/** Чёрный непрозрачный, не зависит от темы */
export var dark03 = '#363636';
/** Прозрачный цвет */
export var transparent = 'rgba(0, 0, 0, 0)';
/** Второстепенная прозрачная */
export var buttonClear = 'rgba(0, 0, 0, 0)';
/** Дефолтная чёрная, не зависит от темы */
export var buttonBlack = '#080808';
/** Второстепенная чёрная, не зависит от темы */
export var buttonBlackSecondary = 'rgba(8, 8, 8, 0.12)';
/** Чёрная для использования поверх картинок */
export var buttonBlackTransparent = 'rgba(8, 8, 8, 0.56)';
/** Дефолтная белая, не зависит от темы */
export var buttonWhite = '#FFFFFF';
/** Второстепенная белая, не зависит от темы */
export var buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** Базовый цвет текста, совпадает с primary */
export var text = '#080808';
export var link = '#2A72F8';
export var linkHover = '#4D89F9';
export var linkActive = '#0859F1';
export var linkVisited = '#2364DE';
export var linkVisitedHover = '#427AE3';
export var linkVisitedActive = '#1C54BC';
/** Основной текст в интерфейсе */
export var primary = '#080808';
/** Второстепенный текст */
export var secondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный цвет текста */
export var tertiary = 'rgba(8, 8, 8, 0.28)';
export var paragraph = 'rgba(8, 8, 8, 0.8)';
export var inverse = 'rgba(255, 255, 255, 0.96)';
/** Основной бэкграунд всех приложений и смартапов */
export var background = '#FFFFFF';
export var backgroundPrimary = '#F9F9F9';
/** Вторичный цвет фона */
export var backgroundSecondary = '#FFFFFF';
/** Третичный цвет фона */
export var backgroundTertiary = '#FFFFFF';
export var inputBorder = 'rgba(8, 8, 8, 0.28)';
export var inputBorderHover = 'rgba(8, 8, 8, 0.56)';
export var inputBorderFocus = '#2A72F8';
/** Акцентный цвет бренда */
export var accent = '#2A72F8';
/** Обозначение успешного сценария */
export var success = '#118C28';
/** Цвет предупреждения */
export var warning = '#E35502';
/** Цвет ошибки */
export var critical = '#F31B30';
/** Полупрозрачные поверхности */
export var surfaceLiquid01 = 'rgba(8, 8, 8, 0.02)';
/** Полупрозрачные поверхности */
export var surfaceLiquid02 = 'rgba(8, 8, 8, 0.06)';
/** Полупрозрачные поверхности */
export var surfaceLiquid03 = 'rgba(8, 8, 8, 0.12)';
/** Непрозрачные поверхности */
export var surfaceSolid01 = '#FAFAFA';
/** Непрозрачные поверхности */
export var surfaceSolid02 = '#F0F0F0';
/** Непрозрачные поверхности */
export var surfaceSolid03 = '#E1E1E1';
/** Основной фон для карточек */
export var surfaceCard = '#FFFFFF';
/** Первичный цвет контролов */
export var buttonPrimary = '#2A72F8';
export var buttonPrimaryHover = '#397CF8';
export var buttonPrimaryActive = '#1B68F8';
/** Второстепенная кнопка */
export var buttonSecondary = 'rgba(8, 8, 8, 0.06)';
export var buttonSecondaryHover = 'rgba(8, 8, 8, 0.04)';
export var buttonSecondaryActive = 'rgba(8, 8, 8, 0.08)';
/** Акцентный цвет у контролов */
export var buttonAccent = '#2A72F8';
/** Кнопка для успешного сценария */
export var buttonSuccess = '#118C28';
export var buttonSuccessHover = '#3eaa4d';
export var buttonSuccessActive = '#1d9233';
/** Цвет предупреждения у контролов */
export var buttonWarning = '#F55D05';
export var buttonWarningHover = '#FA650F';
export var buttonWarningActive = '#E65705';
/** Цвет ошибки у контролов */
export var buttonCritical = '#F31B30';
export var buttonCriticalHover = '#F42A3E';
export var buttonCriticalActive = '#F20D24';
/** Цвет зажатого контрола */
export var buttonChecked = '#080808';
export var buttonCheckedHover = '#171717';
export var buttonCheckedHoverColor = 'hsla(0, 0%, 100%, 0.9599999785423279)';
export var buttonCheckedActive = '#080808';
export var buttonCheckedActiveColor = 'rgba(255, 255, 255, 0.6)';
/** Цвет рамки фокуса у контрола */
export var buttonFocused = '#2A72F8';
/** Градиент для заливки основного фона */
export var gradient = 'linear-gradient(45deg, hsla(220.1123595505618, 85.57692307692307%, 59.21568627450981%, 1) 0%, hsla(189.78947368421052, 78.51239669421487%, 52.54901960784314%, 1) 100%)';
export var gradientDevice = '';
export var voicePhraseGradient = '';
export var skeletonGradient = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.09) 0%, rgba(8, 8, 8, 0.08) 6.25%, rgba(8, 8, 8, 0.05) 12.5%, rgba(8, 8, 8, 0.01) 25%, rgba(8, 8, 8, 0.05) 37.5%, rgba(8, 8, 8, 0.08) 43.75%, rgba(8, 8, 8, 0.09) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.05) 62.5%, rgba(8, 8, 8, 0.01) 75%, rgba(8, 8, 8, 0.05) 87.5%, rgba(8, 8, 8, 0.08) 93.75%, rgba(8, 8, 8, 0.09) 100% )';
export var skeletonGradientLighter = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.36) 0%, rgba(8, 8, 8, 0.32) 6.25%, rgba(8, 8, 8, 0.20) 12.5%, rgba(8, 8, 8, 0.04) 25%, rgba(8, 8, 8, 0.20) 37.5%, rgba(8, 8, 8, 0.32) 43.75%, rgba(8, 8, 8, 0.36) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.20) 62.5%, rgba(8, 8, 8, 0.04) 75%, rgba(8, 8, 8, 0.20) 87.5%, rgba(8, 8, 8, 0.32) 93.75%, rgba(8, 8, 8, 0.36) 100% )';
export var speechBubbleSent = '';
export var speechBubbleReceived = '';
/** Цвет фона паранжи */
export var overlay = 'rgba(0, 0, 0, 0.8)';
export var overlaySoft = 'rgba(249, 249, 249, 0.56)';
export var overlayHard = 'rgba(249, 249, 249, 0.96)';
export var overlayBlur = 'rgba(249, 249, 249, 0.28)';
export var darkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var darkOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var darkOverlayBlur = 'rgba(35, 35, 35, 0.2)';
export var lightOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var lightOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var lightOverlayBlur = 'rgba(35, 35, 35, 0.2)';
export var inverseOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var inverseOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var inverseOverlayBlur = 'rgba(35, 35, 35, 0.2)';
export var textPrimaryHover = '#08080893';
export var textPrimaryActive = '#080808C4';
export var textPrimary = 'rgba(8, 8, 8, 0.96)';
export var textSecondaryHover = '#080808FF';
export var textSecondaryActive = '#080808AB';
export var textSecondary = 'rgba(8, 8, 8, 0.56)';
export var textTertiaryHover = '#080808FF';
export var textTertiaryActive = '#08080856';
export var textTertiary = 'rgba(8, 8, 8, 0.28)';
export var textParagraphHover = '#0808087A';
export var textParagraphActive = '#080808A3';
export var textParagraph = 'rgba(8, 8, 8, 0.8)';
export var textAccentHover = '#528DFAFF';
export var textAccentActive = '#075AF2FF';
export var textAccent = '#2A72F8';
export var textAccentGradientHover = '#000000FF';
export var textAccentGradientActive = '#CCCCCCFF';
export var textAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
export var textPositiveHover = '#1FC13DFF';
export var textPositiveActive = '#147B27FF';
export var textPositive = '#1A9E32';
export var textWarningHover = '#FB782DFF';
export var textWarningActive = '#D25004FF';
export var textWarning = '#FA5F05';
export var textNegativeHover = '#F54254FF';
export var textNegativeActive = '#DA0B20FF';
export var textNegative = '#F31B31';
export var onDarkTextPrimaryHover = '#FFFFFF93';
export var onDarkTextPrimaryActive = '#FFFFFFC4';
export var onDarkTextPrimary = 'rgba(255, 255, 255, 0.96)';
export var onDarkTextSecondaryHover = '#FFFFFFFF';
export var onDarkTextSecondaryActive = '#FFFFFFAB';
export var onDarkTextSecondary = 'rgba(255, 255, 255, 0.56)';
export var onDarkTextTertiaryHover = '#FFFFFFFF';
export var onDarkTextTertiaryActive = '#FFFFFF56';
export var onDarkTextTertiary = 'rgba(255, 255, 255, 0.28)';
export var onDarkTextParagraphHover = '#FFFFFF7A';
export var onDarkTextParagraphActive = '#FFFFFFA3';
export var onDarkTextParagraph = 'rgba(255, 255, 255, 0.8)';
export var onDarkTextAccentHover = '#689CFDFF';
export var onDarkTextAccentActive = '#1767FDFF';
export var onDarkTextAccent = '#3F81FD';
export var onDarkTextAccentGradientHover = '#000000FF';
export var onDarkTextAccentGradientActive = '#CCCCCCFF';
export var onDarkTextAccentGradient = 'linear-gradient(89.83deg, #5E94FF 0%, #43DBFA 100%)';
export var onDarkTextPositiveHover = '#1FC13DFF';
export var onDarkTextPositiveActive = '#147B27FF';
export var onDarkTextPositive = '#1A9E32';
export var onDarkTextWarningHover = '#FB782DFF';
export var onDarkTextWarningActive = '#D25004FF';
export var onDarkTextWarning = '#FA5F05';
export var onDarkTextNegativeHover = '#FF5263FF';
export var onDarkTextNegativeActive = '#FF001AFF';
export var onDarkTextNegative = '#FF293E';
export var onLightTextPrimaryHover = '#08080893';
export var onLightTextPrimaryActive = '#080808C4';
export var onLightTextPrimary = 'rgba(8, 8, 8, 0.96)';
export var onLightTextSecondaryHover = '#080808FF';
export var onLightTextSecondaryActive = '#080808AB';
export var onLightTextSecondary = 'rgba(8, 8, 8, 0.56)';
export var onLightTextTertiaryHover = '#080808FF';
export var onLightTextTertiaryActive = '#08080856';
export var onLightTextTertiary = 'rgba(8, 8, 8, 0.28)';
export var onLightTextParagraphHover = '#0808087A';
export var onLightTextParagraphActive = '#080808A3';
export var onLightTextParagraph = 'rgba(8, 8, 8, 0.8)';
export var onLightTextAccentHover = '#528DFAFF';
export var onLightTextAccentActive = '#075AF2FF';
export var onLightTextAccent = '#2A72F8';
export var onLightTextAccentGradientHover = '#000000FF';
export var onLightTextAccentGradientActive = '#CCCCCCFF';
export var onLightTextAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
export var onLightTextPositiveHover = '#1FC13DFF';
export var onLightTextPositiveActive = '#147B27FF';
export var onLightTextPositive = '#1A9E32';
export var onLightTextWarningHover = '#FB782DFF';
export var onLightTextWarningActive = '#D25004FF';
export var onLightTextWarning = '#FA5F05';
export var onLightTextNegativeHover = '#F54254FF';
export var onLightTextNegativeActive = '#DA0B20FF';
export var onLightTextNegative = '#F31B31';
export var inverseTextPrimaryHover = '#FFFFFF93';
export var inverseTextPrimaryActive = '#FFFFFFC4';
export var inverseTextPrimary = 'rgba(255, 255, 255, 0.96)';
export var inverseTextSecondaryHover = '#FFFFFFFF';
export var inverseTextSecondaryActive = '#FFFFFFAB';
export var inverseTextSecondary = 'rgba(255, 255, 255, 0.56)';
export var inverseTextTertiaryHover = '#FFFFFFFF';
export var inverseTextTertiaryActive = '#FFFFFF56';
export var inverseTextTertiary = 'rgba(255, 255, 255, 0.28)';
export var inverseTextParagraphHover = '#FFFFFF7A';
export var inverseTextParagraphActive = '#FFFFFFA3';
export var inverseTextParagraph = 'rgba(255, 255, 255, 0.8)';
export var surfaceSolidPrimaryHover = '#FAFAFAFF';
export var surfaceSolidPrimaryActive = '#F0F0F0FF';
export var surfaceSolidPrimary = '#F5F5F5';
export var surfaceSolidSecondaryHover = '#F7F7F7FF';
export var surfaceSolidSecondaryActive = '#E8E8E8FF';
export var surfaceSolidSecondary = '#ECECEC';
export var surfaceSolidTertiaryHover = '#EDEDEDFF';
export var surfaceSolidTertiaryActive = '#D4D4D4FF';
export var surfaceSolidTertiary = '#DDDDDD';
export var surfaceSolidCardHover = '#FFFFFFFF';
export var surfaceSolidCardActive = '#FFFFFFFF';
export var surfaceSolidCard = 'rgba(255, 255, 255, 1)';
export var surfaceSolidDefaultHover = '#262626FF';
export var surfaceSolidDefaultActive = '#030303FF';
export var surfaceSolidDefault = '#080808';
export var surfaceTransparentPrimaryHover = '#08080803';
export var surfaceTransparentPrimaryActive = '#0808080D';
export var surfaceTransparentPrimary = 'rgba(8, 8, 8, 0.03)';
export var surfaceTransparentSecondaryHover = '#08080805';
export var surfaceTransparentSecondaryActive = '#0808081A';
export var surfaceTransparentSecondary = 'rgba(8, 8, 8, 0.06)';
export var surfaceTransparentTertiaryHover = '#0808080A';
export var surfaceTransparentTertiaryActive = '#08080829';
export var surfaceTransparentTertiary = 'rgba(8, 8, 8, 0.12)';
export var surfaceTransparentDeepHover = '#0808088F';
export var surfaceTransparentDeepActive = '#080808AD';
export var surfaceTransparentDeep = 'rgba(8, 8, 8, 0.64)';
export var surfaceTransparentCardHover = '#FFFFFFFF';
export var surfaceTransparentCardActive = '#FFFFFFFF';
export var surfaceTransparentCard = 'rgba(255, 255, 255, 1)';
export var surfaceClearHover = '#FFFFFF00';
export var surfaceClearActive = '#FFFFFF00';
export var surfaceClear = 'rgba(255, 255, 255, 0)';
export var surfaceAccentHover = '#528DFAFF';
export var surfaceAccentActive = '#1665F8FF';
export var surfaceAccent = '#2A72F8';
export var surfaceAccentGradientHover = '#FFFFFFFF';
export var surfaceAccentGradientActive = '#FFFFFFFF';
export var surfaceAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
export var surfaceTransparentAccentHover = '#2A72F805';
export var surfaceTransparentAccentActive = '#2A72F81A';
export var surfacePositiveHover = '#1EB83AFF';
export var surfacePositiveActive = '#178C2CFF';
export var surfacePositive = '#1A9E32';
export var surfaceWarningHover = '#FB782DFF';
export var surfaceWarningActive = '#E65705FF';
export var surfaceWarning = '#FA5F05';
export var surfaceNegativeHover = '#F54254FF';
export var surfaceNegativeActive = '#ED0C23FF';
export var surfaceNegative = '#F31B31';
export var surfaceTransparentPositiveHover = '#1A9E3205';
export var surfaceTransparentPositiveActive = '#1A9E321A';
export var surfaceTransparentWarningHover = '#FA5F0505';
export var surfaceTransparentWarningActive = '#FA5F051A';
export var surfaceTransparentNegativeHover = '#F31B3105';
export var surfaceTransparentNegativeActive = '#F31B311A';
export var surfaceTransparentAccent = 'rgba(42,114,248,0.06)';
export var surfaceTransparentPositive = 'rgba(26,158,50,0.06)';
export var surfaceTransparentWarning = 'rgba(250,95,5,0.06)';
export var surfaceTransparentNegative = 'rgba(243,27,49,0.06)';
export var onDarkSurfaceSolidPrimaryHover = '#363636FF';
export var onDarkSurfaceSolidPrimaryActive = '#0D0D0DFF';
export var onDarkSurfaceSolidPrimary = '#171717';
export var onDarkSurfaceSolidSecondaryHover = '#3B3B3BFF';
export var onDarkSurfaceSolidSecondaryActive = '#1C1C1CFF';
export var onDarkSurfaceSolidSecondary = '#262626';
export var onDarkSurfaceSolidTertiaryHover = '#4A4A4AFF';
export var onDarkSurfaceSolidTertiaryActive = '#2B2B2BFF';
export var onDarkSurfaceSolidTertiary = '#363636';
export var onDarkSurfaceSolidCardHover = '#363636FF';
export var onDarkSurfaceSolidCardActive = '#0D0D0DFF';
export var onDarkSurfaceSolidCard = '#171717';
export var onDarkSurfaceSolidDefaultHover = '#FFFFFFFF';
export var onDarkSurfaceSolidDefaultActive = '#FFFFFFFF';
export var onDarkSurfaceSolidDefault = 'rgba(255, 255, 255, 1)';
export var onDarkSurfaceTransparentPrimaryHover = '#363636FF';
export var onDarkSurfaceTransparentPrimaryActive = '#0D0D0DFF';
export var onDarkSurfaceTransparentPrimary = '#171717';
export var onDarkSurfaceTransparentSecondaryHover = '#3B3B3BFF';
export var onDarkSurfaceTransparentSecondaryActive = '#1C1C1CFF';
export var onDarkSurfaceTransparentSecondary = '#262626';
export var onDarkSurfaceTransparentTertiaryHover = '#4A4A4AFF';
export var onDarkSurfaceTransparentTertiaryActive = '#2B2B2BFF';
export var onDarkSurfaceTransparentTertiary = '#363636';
export var onDarkSurfaceTransparentDeepHover = '#FFFFFF8F';
export var onDarkSurfaceTransparentDeepActive = '#FFFFFFAD';
export var onDarkSurfaceTransparentDeep = 'rgba(255, 255, 255, 0.64)';
export var onDarkSurfaceTransparentCardHover = '#FFFFFF0A';
export var onDarkSurfaceTransparentCardActive = '#FFFFFF29';
export var onDarkSurfaceTransparentCard = 'rgba(255, 255, 255, 0.12)';
export var onDarkSurfaceAccentHover = '#689CFDFF';
export var onDarkSurfaceAccentActive = '#2B74FDFF';
export var onDarkSurfaceAccent = '#3F81FD';
export var onDarkSurfaceAccentGradientHover = '#FFFFFFFF';
export var onDarkSurfaceAccentGradientActive = '#FFFFFFFF';
export var onDarkSurfaceAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
export var onDarkSurfacePositiveHover = '#1EB83AFF';
export var onDarkSurfacePositiveActive = '#178C2CFF';
export var onDarkSurfacePositive = '#1A9E32';
export var onDarkSurfaceWarningHover = '#FB782DFF';
export var onDarkSurfaceWarningActive = '#E65705FF';
export var onDarkSurfaceWarning = '#FA5F05';
export var onDarkSurfaceNegativeHover = '#FF5263FF';
export var onDarkSurfaceNegativeActive = '#FF142CFF';
export var onDarkSurfaceNegative = '#FF293E';
export var onLightSurfaceSolidPrimaryHover = '#FAFAFAFF';
export var onLightSurfaceSolidPrimaryActive = '#F0F0F0FF';
export var onLightSurfaceSolidPrimary = '#F5F5F5';
export var onLightSurfaceSolidSecondaryHover = '#F7F7F7FF';
export var onLightSurfaceSolidSecondaryActive = '#E8E8E8FF';
export var onLightSurfaceSolidSecondary = '#ECECEC';
export var onLightSurfaceSolidTertiaryHover = '#EDEDEDFF';
export var onLightSurfaceSolidTertiaryActive = '#D4D4D4FF';
export var onLightSurfaceSolidTertiary = '#DDDDDD';
export var onLightSurfaceSolidCardHover = '#FFFFFFFF';
export var onLightSurfaceSolidCardActive = '#FFFFFFFF';
export var onLightSurfaceSolidCard = 'rgba(255, 255, 255, 1)';
export var onLightSurfaceSolidDefaultHover = '#262626FF';
export var onLightSurfaceSolidDefaultActive = '#030303FF';
export var onLightSurfaceSolidDefault = '#080808';
export var onLightSurfaceTransparentPrimaryHover = '#08080803';
export var onLightSurfaceTransparentPrimaryActive = '#0808080D';
export var onLightSurfaceTransparentPrimary = 'rgba(8, 8, 8, 0.03)';
export var onLightSurfaceTransparentSecondaryHover = '#08080805';
export var onLightSurfaceTransparentSecondaryActive = '#0808081A';
export var onLightSurfaceTransparentSecondary = 'rgba(8, 8, 8, 0.06)';
export var onLightSurfaceTransparentTertiaryHover = '#0808080A';
export var onLightSurfaceTransparentTertiaryActive = '#08080829';
export var onLightSurfaceTransparentTertiary = 'rgba(8, 8, 8, 0.12)';
export var onLightSurfaceTransparentDeepHover = '#0808088F';
export var onLightSurfaceTransparentDeepActive = '#080808AD';
export var onLightSurfaceTransparentDeep = 'rgba(8, 8, 8, 0.64)';
export var onLightSurfaceTransparentCardHover = '#FFFFFFFF';
export var onLightSurfaceTransparentCardActive = '#FFFFFFFF';
export var onLightSurfaceTransparentCard = 'rgba(255, 255, 255, 1)';
export var onLightSurfaceAccentHover = '#528DFAFF';
export var onLightSurfaceAccentActive = '#1665F8FF';
export var onLightSurfaceAccent = '#2A72F8';
export var onLightSurfaceAccentGradientHover = '#FFFFFFFF';
export var onLightSurfaceAccentGradientActive = '#FFFFFFFF';
export var onLightSurfaceAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
export var onLightSurfacePositiveHover = '#1EB83AFF';
export var onLightSurfacePositiveActive = '#178C2CFF';
export var onLightSurfacePositive = '#1A9E32';
export var onLightSurfaceWarningHover = '#FB782DFF';
export var onLightSurfaceWarningActive = '#E65705FF';
export var onLightSurfaceWarning = '#FA5F05';
export var onLightSurfaceNegativeHover = '#F54254FF';
export var onLightSurfaceNegativeActive = '#ED0C23FF';
export var onLightSurfaceNegative = '#F31B31';
export var darkBackgroundPrimary = '#080808';
export var lightBackgroundPrimary = '#F9F9F9';
export var onDarkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var onDarkOverlayHard = 'rgba(8, 8, 8, 0.96)';
export var onDarkOverlayBlur = 'rgba(8, 8, 8, 0.28)';
export var onLightOverlaySoft = 'rgba(249, 249, 249, 0.56)';
export var onLightOverlayHard = 'rgba(249, 249, 249, 0.96)';
export var onLightOverlayBlur = 'rgba(249, 249, 249, 0.28)';
