// @salutejs/plasma-ui/components/Card/Card.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ListCard = exports.BasicValue = exports.BasicDefault = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _Card = /*#__PURE__*/require("./Card");
var _CardBody = /*#__PURE__*/require("./CardBody");
var _CardContent = /*#__PURE__*/require("./CardContent");
var _CardMedia = /*#__PURE__*/require("./CardMedia");
var _ = /*#__PURE__*/require(".");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var longText = "\u041A\u0430\u043D\u0430\u0434\u0441\u043A\u0438\u0439 \u0430\u043A\u0442\u0451\u0440, \u043A\u0438\u043D\u043E\u043F\u0440\u043E\u0434\u044E\u0441\u0435\u0440, \u0438 \u043C\u0443\u0437\u044B\u043A\u0430\u043D\u0442. \u041D\u0430\u0438\u0431\u043E\u043B\u0435\u0435 \u0438\u0437\u0432\u0435\u0441\u0442\u0435\u043D \u0441\u0432\u043E\u0438\u043C\u0438 \u0440\u043E\u043B\u044F\u043C\u0438 \u0432 \u043A\u0438\u043D\u043E\u0441\u0435\u0440\u0438\u044F\u0445 \xAB\u041C\u0430\u0442\u0440\u0438\u0446\u0430\xBB \u0438\n\xAB\u0414\u0436\u043E\u043D \u0423\u0438\u043A\xBB, \u0430 \u0442\u0430\u043A\u0436\u0435 \u0432 \u0444\u0438\u043B\u044C\u043C\u0430\u0445 \xAB\u041D\u0430 \u0433\u0440\u0435\u0431\u043D\u0435 \u0432\u043E\u043B\u043D\u044B\xBB, \xAB\u041C\u043E\u0439 \u043B\u0438\u0447\u043D\u044B\u0439 \u0448\u0442\u0430\u0442 \u0410\u0439\u0434\u0430\u0445\u043E\xBB, \xAB\u0414\u0440\u0430\u043A\u0443\u043B\u0430\xBB, \xAB\u0421\u043A\u043E\u0440\u043E\u0441\u0442\u044C\xBB,\n\xAB\u0414\u0436\u043E\u043D\u043D\u0438-\u043C\u043D\u0435\u043C\u043E\u043D\u0438\u043A\xBB, \xAB\u0410\u0434\u0432\u043E\u043A\u0430\u0442 \u0434\u044C\u044F\u0432\u043E\u043B\u0430\xBB, \xAB\u041A\u043E\u043D\u0441\u0442\u0430\u043D\u0442\u0438\u043D: \u041F\u043E\u0432\u0435\u043B\u0438\u0442\u0435\u043B\u044C \u0442\u044C\u043C\u044B\xBB \u0438 \xAB\u041A\u043E\u0440\u043E\u043B\u0438 \u0443\u043B\u0438\u0446\xBB.";
var basicCardStyle = {
  width: '22.5rem'
};
var basicValueProps = {
  cover: false,
  coverGradient: true,
  outlined: true,
  scaleOnFocus: true,
  subTitle: 'Потребительский кредит',
  title: 'до 230 000 ₽',
  description: 'На 18 месяцев, ставка 13,9%'
};
var BasicValue = exports.BasicValue = function BasicValue() {
  var outlined = basicValueProps.outlined,
    coverGradient = basicValueProps.coverGradient,
    scaleOnFocus = basicValueProps.scaleOnFocus,
    cover = basicValueProps.cover,
    subTitle = basicValueProps.subTitle,
    title = basicValueProps.title,
    description = basicValueProps.description;
  return /*#__PURE__*/_react["default"].createElement(_Card.Card, {
    style: basicCardStyle,
    tabIndex: 0,
    outlined: outlined,
    scaleOnFocus: scaleOnFocus
  }, /*#__PURE__*/_react["default"].createElement(_CardBody.CardBody, null, /*#__PURE__*/_react["default"].createElement(_CardMedia.CardMedia, {
    src: "./images/320_320_0.jpg",
    placeholder: "./images/320_320_1.jpg",
    ratio: cover ? '1 / 1' : '16 / 9'
  }), /*#__PURE__*/_react["default"].createElement(_CardContent.CardContent, {
    cover: cover,
    coverGradient: cover && coverGradient
  }, subTitle, title, description)));
};
var defaultCardStyle = {
  width: '20rem'
};
var defaultParagraphStyle = {
  marginTop: '0.75rem'
};
var basicDefaultProps = {
  cover: false,
  title: 'Киану Ривз',
  footer1: 'Актёр',
  footer2: 'ru.wikipedia.org',
  description: longText
};
var BasicDefault = exports.BasicDefault = function BasicDefault() {
  var cover = basicDefaultProps.cover,
    title = basicDefaultProps.title,
    footer1 = basicDefaultProps.footer1,
    footer2 = basicDefaultProps.footer2,
    description = basicDefaultProps.description;
  return /*#__PURE__*/_react["default"].createElement(_Card.Card, {
    style: defaultCardStyle
  }, /*#__PURE__*/_react["default"].createElement(_CardBody.CardBody, null, /*#__PURE__*/_react["default"].createElement(_CardMedia.CardMedia, {
    src: "./images/320_320_2.jpg",
    placeholder: "./images/320_320_3.jpg",
    ratio: cover ? '1 / 1' : '16 / 9'
  }), /*#__PURE__*/_react["default"].createElement(_CardContent.CardContent, {
    cover: cover
  }, title, footer1, /*#__PURE__*/_react["default"].createElement(_.CardParagraph1, {
    style: defaultParagraphStyle,
    lines: 4
  }, description), footer2)));
};
var listCardStyle = {
  width: '20rem'
};
var listCardProps = {
  title: 'Title',
  subtitle: 'Subtitle'
};
var ListCard = exports.ListCard = function ListCard() {
  var title = listCardProps.title,
    subtitle = listCardProps.subtitle;
  return /*#__PURE__*/_react["default"].createElement(_Card.Card, {
    style: listCardStyle
  }, /*#__PURE__*/_react["default"].createElement(_CardContent.CardContent, {
    compact: true
  }, title, subtitle));
};
