// @salutejs/plasma-ui/components/Card/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Card", {
  enumerable: true,
  get: function get() {
    return _Card.Card;
  }
});
Object.defineProperty(exports, "CardBadge", {
  enumerable: true,
  get: function get() {
    return _CardBadge.CardBadge;
  }
});
Object.defineProperty(exports, "CardBody", {
  enumerable: true,
  get: function get() {
    return _CardBody.CardBody;
  }
});
Object.defineProperty(exports, "CardBody1", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.CardBody1;
  }
});
Object.defineProperty(exports, "CardBody2", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.CardBody2;
  }
});
Object.defineProperty(exports, "CardContent", {
  enumerable: true,
  get: function get() {
    return _CardContent.CardContent;
  }
});
Object.defineProperty(exports, "CardFootnote1", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.CardFootnote1;
  }
});
Object.defineProperty(exports, "CardFootnote2", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.CardFootnote2;
  }
});
Object.defineProperty(exports, "CardHeadline1", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.CardHeadline1;
  }
});
Object.defineProperty(exports, "CardHeadline2", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.CardHeadline2;
  }
});
Object.defineProperty(exports, "CardHeadline3", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.CardHeadline3;
  }
});
Object.defineProperty(exports, "CardMedia", {
  enumerable: true,
  get: function get() {
    return _CardMedia.CardMedia;
  }
});
Object.defineProperty(exports, "CardParagraph1", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.CardParagraph1;
  }
});
Object.defineProperty(exports, "CardParagraph2", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.CardParagraph2;
  }
});
Object.defineProperty(exports, "CardPrice", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.CardPrice;
  }
});
Object.defineProperty(exports, "StyledCard", {
  enumerable: true,
  get: function get() {
    return _Card.StyledCard;
  }
});
var _Card = /*#__PURE__*/require("./Card");
var _CardBadge = /*#__PURE__*/require("./CardBadge");
var _CardBody = /*#__PURE__*/require("./CardBody");
var _CardContent = /*#__PURE__*/require("./CardContent");
var _CardMedia = /*#__PURE__*/require("./CardMedia");
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
