// @salutejs/plasma-tokens-b2c/es/themesValues/dark.js
// Generated by robots, do not change this manually!
/** Базовый белый цвет совпадает с whitePrimary */
export var white = 'rgba(255, 255, 255, 0.96)';
/** Первичный белый цвет */
export var whitePrimary = 'rgba(255, 255, 255, 0.96)';
/** Вторичный белый цвет */
export var whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный белый цвет */
export var whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** Базовый черный цвет совпадает с blackPrimary */
export var black = '#080808';
/** Первичный черный цвет */
export var blackPrimary = '#080808';
/** Вторичный черныйцвет */
export var blackSecondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный черный цвет */
export var blackTertiary = 'rgba(8, 8, 8, 0.28)';
/** Чёрный непрозрачный, не зависит от темы */
export var dark01 = '#171717';
/** Чёрный непрозрачный, не зависит от темы */
export var dark02 = '#232323';
/** Чёрный непрозрачный, не зависит от темы */
export var dark03 = '#363636';
/** Прозрачный цвет */
export var transparent = 'rgba(0, 0, 0, 0)';
/** Второстепенная прозрачная */
export var buttonClear = 'rgba(0, 0, 0, 0)';
/** Дефолтная чёрная, не зависит от темы */
export var buttonBlack = '#080808';
/** Второстепенная чёрная, не зависит от темы */
export var buttonBlackSecondary = 'rgba(8, 8, 8, 0.12)';
/** Чёрная для использования поверх картинок */
export var buttonBlackTransparent = 'rgba(8, 8, 8, 0.56)';
/** Дефолтная белая, не зависит от темы */
export var buttonWhite = '#FFFFFF';
/** Второстепенная белая, не зависит от темы */
export var buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** Базовый цвет текста, совпадает с primary */
export var text = 'rgba(255, 255, 255, 0.96)';
export var link = '#3F81FD';
export var linkHover = '#6298FD';
export var linkActive = '#1767FD';
export var linkVisited = '#5993FF';
export var linkVisitedHover = '#7DAAFF';
export var linkVisitedActive = '#3078FF';
/** Основной текст в интерфейсе */
export var primary = 'rgba(255, 255, 255, 0.96)';
/** Второстепенный текст */
export var secondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный цвет текста */
export var tertiary = 'rgba(255, 255, 255, 0.28)';
export var paragraph = 'rgba(255, 255, 255, 0.96)';
export var inverse = '#080808';
/** Основной бэкграунд всех приложений и смартапов */
export var background = '#080808';
export var backgroundPrimary = '#080808';
/** Вторичный цвет фона */
export var backgroundSecondary = '#232323';
/** Третичный цвет фона */
export var backgroundTertiary = '#363636';
export var inputBorder = 'rgba(0, 0, 0, 0)';
export var inputBorderHover = 'rgba(0, 0, 0, 0)';
export var inputBorderFocus = 'rgba(0, 0, 0, 0)';
export var inputErrorBackground = 'rgba(255, 46, 67, 0.12)';
/** Акцентный цвет бренда */
export var accent = '#5993FF';
/** Обозначение успешного сценария */
export var success = '#24B23E';
/** Цвет предупреждения */
export var warning = '#FA6D20';
/** Цвет ошибки */
export var critical = '#FF4D5F';
/** Полупрозрачные поверхности */
export var surfaceLiquid01 = 'rgba(255, 255, 255, 0.06)';
/** Полупрозрачные поверхности */
export var surfaceLiquid02 = 'rgba(255, 255, 255, 0.12)';
/** Полупрозрачные поверхности */
export var surfaceLiquid03 = 'rgba(255, 255, 255, 0.2)';
/** Непрозрачные поверхности */
export var surfaceSolid01 = '#171717';
/** Непрозрачные поверхности */
export var surfaceSolid02 = '#232323';
/** Непрозрачные поверхности */
export var surfaceSolid03 = '#363636';
/** Основной фон для карточек */
export var surfaceCard = 'rgba(255, 255, 255, 0.12)';
/** Первичный цвет контролов */
export var buttonPrimary = '#3F82FD';
export var buttonPrimaryHover = '#538efe';
export var buttonPrimaryActive = '#3a75e5';
/** Второстепенная кнопка */
export var buttonSecondary = 'rgba(255, 255, 255, 0.12)';
export var buttonSecondaryHover = 'rgba(255, 255, 255, 0.16)';
export var buttonSecondaryActive = 'rgba(255, 255, 255, 0.1)';
/** Акцентный цвет у контролов */
export var buttonAccent = '#3F82FD';
/** Кнопка для успешного сценария */
export var buttonSuccess = '#21A038';
export var buttonSuccessTransparent = 'rgba(33, 160, 56, 0.3)';
export var buttonSuccessHover = '#3eaa4d';
export var buttonSuccessActive = '#1d9233';
export var buttonSuccessActiveTransparent = 'rgba(97, 181, 103, 0.384)';
/** Цвет предупреждения у контролов */
export var buttonWarning = '#E35502';
export var buttonWarningTransparent = 'rgba(227, 85, 2, 0.3)';
export var buttonWarningHover = '#f76d2f';
export var buttonWarningActive = '#de550a';
export var buttonWarningActiveTransparent = 'rgba(238, 127, 83, 0.384)';
/** Цвет ошибки у контролов */
export var buttonCritical = '#FF2E43';
export var buttonCriticalTransparent = 'rgba(255, 46, 67, 0.3)';
export var buttonCriticalHover = '#ff4757';
export var buttonCriticalActive = '#e82a3e';
export var buttonCriticalActiveTransparent = 'rgba(255, 114, 113, 0.384)';
/** Цвет зажатого контрола */
export var buttonChecked = '#FFFFFF';
export var buttonCheckedHover = '#FFFFFF';
export var buttonCheckedHoverColor = 'hsla(0, 0%, 3.1372549019607843%, 1)';
export var buttonCheckedActive = '#FFFFFF';
export var buttonCheckedActiveColor = 'hsla(0, 0%, 3.1372549019607843%, 1)';
/** Цвет рамки фокуса у контрола */
export var buttonFocused = '#3F82FD';
/** Градиент для заливки основного фона */
export var gradient = 'linear-gradient(45deg, hsla(220.1123595505618, 85.57692307692307%, 59.21568627450981%, 1) 0%, hsla(189.78947368421052, 78.51239669421487%, 52.54901960784314%, 1) 100%)';
export var gradientDevice = '';
export var voicePhraseGradient = '';
export var skeletonGradient = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% )';
export var skeletonGradientLighter = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% )';
export var speechBubbleSent = '';
export var speechBubbleReceived = '';
/** Цвет фона паранжи */
export var overlay = 'rgba(0, 0, 0, 0.8)';
export var overlaySoft = 'rgba(8, 8, 8, 0.56)';
export var overlayHard = 'rgba(8, 8, 8, 0.96)';
export var overlayBlur = 'rgba(8, 8, 8, 0.28)';
export var darkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var darkOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var darkOverlayBlur = 'rgba(23, 23, 23, 0.28)';
export var lightOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var lightOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var lightOverlayBlur = 'rgba(23, 23, 23, 0.28)';
export var inverseOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var inverseOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var inverseOverlayBlur = 'rgba(23, 23, 23, 0.28)';
export var textPrimaryHover = '#FFFFFF93';
export var textPrimaryActive = '#FFFFFFC4';
export var textPrimary = 'rgba(255, 255, 255, 0.96)';
export var textSecondaryHover = '#FFFFFFFF';
export var textSecondaryActive = '#FFFFFFAB';
export var textSecondary = 'rgba(255, 255, 255, 0.56)';
export var textTertiaryHover = '#FFFFFFFF';
export var textTertiaryActive = '#FFFFFF56';
export var textTertiary = 'rgba(255, 255, 255, 0.28)';
export var textParagraphHover = '#FFFFFF7A';
export var textParagraphActive = '#FFFFFFA3';
export var textParagraph = 'rgba(255, 255, 255, 0.8)';
export var textAccentHover = '#90B6FEFF';
export var textAccentActive = '#216EFDFF';
export var textAccent = '#3F81FD';
export var textAccentGradientHover = '#CCCCCCFF';
export var textAccentGradientActive = '#E6E6E6FF';
export var textAccentGradient = 'linear-gradient(89.83deg, #5E94FF 0%, #43DBFA 100%)';
export var textPositiveHover = '#1EB83AFF';
export var textPositiveActive = '#15842AFF';
export var textPositive = '#1A9E32';
export var textWarningHover = '#FB7223FF';
export var textWarningActive = '#DC5304FF';
export var textWarning = '#FA5F05';
export var textNegativeHover = '#FF475AFF';
export var textNegativeActive = '#FF0A23FF';
export var textNegative = '#FF293E';
export var onDarkTextPrimaryHover = '#FFFFFF93';
export var onDarkTextPrimaryActive = '#FFFFFFC4';
export var onDarkTextPrimary = 'rgba(255, 255, 255, 0.96)';
export var onDarkTextSecondaryHover = '#FFFFFFFF';
export var onDarkTextSecondaryActive = '#FFFFFFAB';
export var onDarkTextSecondary = 'rgba(255, 255, 255, 0.56)';
export var onDarkTextTertiaryHover = '#FFFFFFFF';
export var onDarkTextTertiaryActive = '#FFFFFF56';
export var onDarkTextTertiary = 'rgba(255, 255, 255, 0.28)';
export var onDarkTextParagraphHover = '#FFFFFF7A';
export var onDarkTextParagraphActive = '#FFFFFFA3';
export var onDarkTextParagraph = 'rgba(255, 255, 255, 0.8)';
export var onDarkTextAccentHover = '#90B6FEFF';
export var onDarkTextAccentActive = '#216EFDFF';
export var onDarkTextAccent = '#3F81FD';
export var onDarkTextAccentGradientHover = '#CCCCCCFF';
export var onDarkTextAccentGradientActive = '#E6E6E6FF';
export var onDarkTextAccentGradient = 'linear-gradient(89.83deg, #5E94FF 0%, #43DBFA 100%)';
export var onDarkTextPositiveHover = '#1EB83AFF';
export var onDarkTextPositiveActive = '#15842AFF';
export var onDarkTextPositive = '#1A9E32';
export var onDarkTextWarningHover = '#FB7223FF';
export var onDarkTextWarningActive = '#DC5304FF';
export var onDarkTextWarning = '#FA5F05';
export var onDarkTextNegativeHover = '#FF475AFF';
export var onDarkTextNegativeActive = '#FF0A23FF';
export var onDarkTextNegative = '#FF293E';
export var onLightTextPrimaryHover = '#08080893';
export var onLightTextPrimaryActive = '#080808C4';
export var onLightTextPrimary = 'rgba(8, 8, 8, 0.96)';
export var onLightTextSecondaryHover = '#080808FF';
export var onLightTextSecondaryActive = '#080808AB';
export var onLightTextSecondary = 'rgba(8, 8, 8, 0.56)';
export var onLightTextTertiaryHover = '#080808FF';
export var onLightTextTertiaryActive = '#08080856';
export var onLightTextTertiary = 'rgba(8, 8, 8, 0.28)';
export var onLightTextParagraphHover = '#0808087A';
export var onLightTextParagraphActive = '#080808A3';
export var onLightTextParagraph = 'rgba(8, 8, 8, 0.8)';
export var onLightTextAccentHover = '#79A7FBFF';
export var onLightTextAccentActive = '#0D5FF8FF';
export var onLightTextAccent = '#2A72F8';
export var onLightTextAccentGradientHover = '#CCCCCCFF';
export var onLightTextAccentGradientActive = '#E6E6E6FF';
export var onLightTextAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
export var onLightTextPositiveHover = '#1EB83AFF';
export var onLightTextPositiveActive = '#15842AFF';
export var onLightTextPositive = '#1A9E32';
export var onLightTextWarningHover = '#FB7223FF';
export var onLightTextWarningActive = '#DC5304FF';
export var onLightTextWarning = '#FA5F05';
export var onLightTextNegativeHover = '#F5384BFF';
export var onLightTextNegativeActive = '#E40C22FF';
export var onLightTextNegative = '#F31B31';
export var inverseTextPrimaryHover = '#08080893';
export var inverseTextPrimaryActive = '#080808C4';
export var inverseTextPrimary = 'rgba(8, 8, 8, 0.96)';
export var inverseTextSecondaryHover = '#080808FF';
export var inverseTextSecondaryActive = '#080808AB';
export var inverseTextSecondary = 'rgba(8, 8, 8, 0.56)';
export var inverseTextTertiaryHover = '#080808FF';
export var inverseTextTertiaryActive = '#08080856';
export var inverseTextTertiary = 'rgba(8, 8, 8, 0.28)';
export var inverseTextParagraphHover = '#0808087A';
export var inverseTextParagraphActive = '#080808A3';
export var inverseTextParagraph = 'rgba(8, 8, 8, 0.8)';
export var surfaceSolidPrimaryHover = '#1C1C1CFF';
export var surfaceSolidPrimaryActive = '#121212FF';
export var surfaceSolidPrimary = '#171717';
export var surfaceSolidSecondaryHover = '#303030FF';
export var surfaceSolidSecondaryActive = '#212121FF';
export var surfaceSolidSecondary = '#262626';
export var surfaceSolidTertiaryHover = '#404040FF';
export var surfaceSolidTertiaryActive = '#303030FF';
export var surfaceSolidTertiary = '#363636';
export var surfaceSolidCardHover = '#1C1C1CFF';
export var surfaceSolidCardActive = '#121212FF';
export var surfaceSolidCard = '#171717';
export var surfaceSolidDefaultHover = '#FFFFFFFF';
export var surfaceSolidDefaultActive = '#FFFFFFFF';
export var surfaceSolidDefault = 'rgba(255, 255, 255, 1)';
export var surfaceTransparentPrimaryHover = '#FFFFFF1F';
export var surfaceTransparentPrimaryActive = '#FFFFFF0A';
export var surfaceTransparentPrimary = 'rgba(255, 255, 255, 0.06)';
export var surfaceTransparentSecondaryHover = '#FFFFFF3D';
export var surfaceTransparentSecondaryActive = '#FFFFFF0F';
export var surfaceTransparentSecondary = 'rgba(255, 255, 255, 0.12)';
export var surfaceTransparentTertiaryHover = '#FFFFFF52';
export var surfaceTransparentTertiaryActive = '#FFFFFF24';
export var surfaceTransparentTertiary = 'rgba(255, 255, 255, 0.2)';
export var surfaceTransparentDeepHover = '#FFFFFFC2';
export var surfaceTransparentDeepActive = '#FFFFFF94';
export var surfaceTransparentDeep = 'rgba(255, 255, 255, 0.64)';
export var surfaceTransparentCardHover = '#FFFFFF3D';
export var surfaceTransparentCardActive = '#FFFFFF0F';
export var surfaceTransparentCard = 'rgba(255, 255, 255, 0.12)';
export var surfaceClearHover = '#FFFFFF00';
export var surfaceClearActive = '#FFFFFF00';
export var surfaceClear = 'rgba(255, 255, 255, 0)';
export var surfaceAccentHover = '#5D95FDFF';
export var surfaceAccentActive = '#357BFDFF';
export var surfaceAccent = '#3F81FD';
export var surfaceAccentGradientHover = '#FFFFFFFF';
export var surfaceAccentGradientActive = '#FFFFFFFF';
export var surfaceAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%), linear-gradient(89.83deg, #5E94FF 0%, #43DBFA 100%)';
export var surfaceTransparentAccentHover = '#3F82FD3D';
export var surfaceTransparentAccentActive = '#3F82FD0F';
export var surfacePositiveHover = '#1DAF37FF';
export var surfacePositiveActive = '#18952FFF';
export var surfacePositive = '#1A9E32';
export var surfaceWarningHover = '#FB7223FF';
export var surfaceWarningActive = '#F05B05FF';
export var surfaceWarning = '#FA5F05';
export var surfaceNegativeHover = '#FF475AFF';
export var surfaceNegativeActive = '#FF1F35FF';
export var surfaceNegative = '#FF293E';
export var surfaceTransparentPositiveHover = '#1A9E323D';
export var surfaceTransparentPositiveActive = '#1A9E320F';
export var surfaceTransparentWarningHover = '#FA5F053D';
export var surfaceTransparentWarningActive = '#FA5F050F';
export var surfaceTransparentNegativeHover = '#FF293E3D';
export var surfaceTransparentNegativeActive = '#FF293E0F';
export var surfaceTransparentAccent = 'rgba(63,129,253,0.12)';
export var surfaceTransparentPositive = 'rgba(26,158,50,0.12)';
export var surfaceTransparentWarning = 'rgba(250,95,5,0.12)';
export var surfaceTransparentNegative = 'rgba(255,41,62,0.12)';
export var onDarkSurfaceSolidPrimaryHover = '#1C1C1CFF';
export var onDarkSurfaceSolidPrimaryActive = '#121212FF';
export var onDarkSurfaceSolidPrimary = '#171717';
export var onDarkSurfaceSolidSecondaryHover = '#303030FF';
export var onDarkSurfaceSolidSecondaryActive = '#212121FF';
export var onDarkSurfaceSolidSecondary = '#262626';
export var onDarkSurfaceSolidTertiaryHover = '#404040FF';
export var onDarkSurfaceSolidTertiaryActive = '#303030FF';
export var onDarkSurfaceSolidTertiary = '#363636';
export var onDarkSurfaceSolidCardHover = '#1C1C1CFF';
export var onDarkSurfaceSolidCardActive = '#121212FF';
export var onDarkSurfaceSolidCard = '#171717';
export var onDarkSurfaceSolidDefaultHover = '#FFFFFFFF';
export var onDarkSurfaceSolidDefaultActive = '#FFFFFFFF';
export var onDarkSurfaceSolidDefault = 'rgba(255, 255, 255, 1)';
export var onDarkSurfaceTransparentPrimaryHover = '#FFFFFF1F';
export var onDarkSurfaceTransparentPrimaryActive = '#FFFFFF0A';
export var onDarkSurfaceTransparentPrimary = 'rgba(255, 255, 255, 0.06)';
export var onDarkSurfaceTransparentSecondaryHover = '#FFFFFF3D';
export var onDarkSurfaceTransparentSecondaryActive = '#FFFFFF0F';
export var onDarkSurfaceTransparentSecondary = 'rgba(255, 255, 255, 0.12)';
export var onDarkSurfaceTransparentTertiaryHover = '#FFFFFF52';
export var onDarkSurfaceTransparentTertiaryActive = '#FFFFFF24';
export var onDarkSurfaceTransparentTertiary = 'rgba(255, 255, 255, 0.2)';
export var onDarkSurfaceTransparentDeepHover = '#FFFFFFC2';
export var onDarkSurfaceTransparentDeepActive = '#FFFFFF94';
export var onDarkSurfaceTransparentDeep = 'rgba(255, 255, 255, 0.64)';
export var onDarkSurfaceTransparentCardHover = '#FFFFFF3D';
export var onDarkSurfaceTransparentCardActive = '#FFFFFF0F';
export var onDarkSurfaceTransparentCard = 'rgba(255, 255, 255, 0.12)';
export var onDarkSurfaceAccentHover = '#5D95FDFF';
export var onDarkSurfaceAccentActive = '#357BFDFF';
export var onDarkSurfaceAccent = '#3F81FD';
export var onDarkSurfaceAccentGradientHover = '#FFFFFFFF';
export var onDarkSurfaceAccentGradientActive = '#FFFFFFFF';
export var onDarkSurfaceAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
export var onDarkSurfacePositiveHover = '#1DAF37FF';
export var onDarkSurfacePositiveActive = '#18952FFF';
export var onDarkSurfacePositive = '#1A9E32';
export var onDarkSurfaceWarningHover = '#FB7223FF';
export var onDarkSurfaceWarningActive = '#F05B05FF';
export var onDarkSurfaceWarning = '#FA5F05';
export var onDarkSurfaceNegativeHover = '#FF475AFF';
export var onDarkSurfaceNegativeActive = '#FF1F35FF';
export var onDarkSurfaceNegative = '#FF293E';
export var onLightSurfaceSolidPrimaryHover = '#FAFAFAFF';
export var onLightSurfaceSolidPrimaryActive = '#F0F0F0FF';
export var onLightSurfaceSolidPrimary = '#F5F5F5';
export var onLightSurfaceSolidSecondaryHover = '#F7F7F7FF';
export var onLightSurfaceSolidSecondaryActive = '#E8E8E8FF';
export var onLightSurfaceSolidSecondary = '#ECECEC';
export var onLightSurfaceSolidTertiaryHover = '#E8E8E8FF';
export var onLightSurfaceSolidTertiaryActive = '#D9D9D9FF';
export var onLightSurfaceSolidTertiary = '#DDDDDD';
export var onLightSurfaceSolidCardHover = '#FFFFFFFF';
export var onLightSurfaceSolidCardActive = '#FFFFFFFF';
export var onLightSurfaceSolidCard = 'rgba(255, 255, 255, 1)';
export var onLightSurfaceSolidDefaultHover = '#0D0D0DFF';
export var onLightSurfaceSolidDefaultActive = '#030303FF';
export var onLightSurfaceSolidDefault = '#080808';
export var onLightSurfaceTransparentPrimaryHover = '#08080812';
export var onLightSurfaceTransparentPrimaryActive = '#08080803';
export var onLightSurfaceTransparentPrimary = 'rgba(8, 8, 8, 0.03)';
export var onLightSurfaceTransparentSecondaryHover = '#0808081F';
export var onLightSurfaceTransparentSecondaryActive = '#0808080A';
export var onLightSurfaceTransparentSecondary = 'rgba(8, 8, 8, 0.06)';
export var onLightSurfaceTransparentTertiaryHover = '#0808083D';
export var onLightSurfaceTransparentTertiaryActive = '#0808080F';
export var onLightSurfaceTransparentTertiary = 'rgba(8, 8, 8, 0.12)';
export var onLightSurfaceTransparentDeepHover = '#080808C2';
export var onLightSurfaceTransparentDeepActive = '#08080894';
export var onLightSurfaceTransparentDeep = 'rgba(8, 8, 8, 0.64)';
export var onLightSurfaceTransparentCardHover = '#FFFFFFFF';
export var onLightSurfaceTransparentCardActive = '#FFFFFFFF';
export var onLightSurfaceTransparentCard = 'rgba(255, 255, 255, 1)';
export var onLightSurfaceAccentHover = '#4886F9FF';
export var onLightSurfaceAccentActive = '#206CF8FF';
export var onLightSurfaceAccent = '#2A72F8';
export var onLightSurfaceAccentGradientHover = '#FFFFFFFF';
export var onLightSurfaceAccentGradientActive = '#FFFFFFFF';
export var onLightSurfaceAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
export var onLightSurfacePositiveHover = '#1DAF37FF';
export var onLightSurfacePositiveActive = '#18952FFF';
export var onLightSurfacePositive = '#1A9E32';
export var onLightSurfaceWarningHover = '#FB7223FF';
export var onLightSurfaceWarningActive = '#F05B05FF';
export var onLightSurfaceWarning = '#FA5F05';
export var onLightSurfaceNegativeHover = '#F5384BFF';
export var onLightSurfaceNegativeActive = '#F31228FF';
export var onLightSurfaceNegative = '#F31B31';
export var darkBackgroundPrimary = '#080808';
export var lightBackgroundPrimary = '#F9F9F9';
export var onDarkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var onDarkOverlayHard = 'rgba(8, 8, 8, 0.96)';
export var onDarkOverlayBlur = 'rgba(8, 8, 8, 0.28)';
export var onLightOverlaySoft = 'rgba(249, 249, 249, 0.56)';
export var onLightOverlayHard = 'rgba(249, 249, 249, 0.96)';
export var onLightOverlayBlur = 'rgba(249, 249, 249, 0.28)';
