// @salutejs/plasma-new-hope/es/components/Switch/Switch.js
import './Switch_svqw4q.css';
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef } from 'react';
import { base as base$1 } from './_size/base.js';
import { base as base$2 } from './_view/base.js';
import { base as base$4 } from './_focused/base.js';
import { base as base$3 } from './_disabled/base.js';
import { base as base$5 } from './_label-position/base.js';
import { StyledInput, StyledLabel, StyledTrigger } from './Switch.styles.js';

var _StyledTrigger;
var _excluded = ["size", "view", "focused", "disabled", "labelPosition", "label", "id", "style", "className", "checked", "defaultChecked"];
var base = "bagv7wj";
var switchRoot = function switchRoot(Root) {
  return /*#__PURE__*/forwardRef(function (props, ref) {
    var size = props.size,
      view = props.view,
      focused = props.focused,
      disabled = props.disabled,
      labelPosition = props.labelPosition,
      label = props.label,
      id = props.id,
      style = props.style,
      className = props.className,
      checked = props.checked,
      defaultChecked = props.defaultChecked,
      rest = _objectWithoutProperties(props, _excluded);
    var exactChecked = Boolean(checked !== undefined ? checked : defaultChecked);
    // const singleLineClass = singleLine ? 'single-line' : '';

    return /*#__PURE__*/React.createElement(Root, {
      view: view,
      size: size,
      disabled: disabled,
      focused: focused,
      labelPosition: labelPosition,
      id: id,
      style: style,
      className: className
    }, /*#__PURE__*/React.createElement(StyledInput, _extends({}, rest, {
      ref: ref,
      role: "switch",
      "aria-checked": exactChecked,
      type: "checkbox",
      checked: checked,
      defaultChecked: defaultChecked,
      disabled: disabled
    })), /*#__PURE__*/React.createElement(StyledLabel, {
      tabIndex: -1
    }, label), _StyledTrigger || (_StyledTrigger = /*#__PURE__*/React.createElement(StyledTrigger, {
      "aria-hidden": true
    })));
  });
};
var switchConfig = {
  name: 'Switch',
  tag: 'label',
  base: base,
  layout: switchRoot,
  variations: {
    size: {
      css: base$1
    },
    view: {
      css: base$2
    },
    disabled: {
      css: base$3,
      attrs: true
    },
    focused: {
      css: base$4
    },
    labelPosition: {
      css: base$5
    }
  },
  defaults: {
    size: 's',
    view: 'default'
  }
};

export { switchConfig, switchRoot };
