// @salutejs/plasma-typo/lib/cjs/archetypes/plasma.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.plasma = exports.plasmaTypoObject = exports.plasmaTypoProperties = void 0;
var helpers_1 = require("../helpers");
var mage_1 = require("./mage");
var typo = {
    'dspl-l': {
        'font-weight': '400',
    },
    'dspl-l-bold': {
        'font-weight': '700',
    },
    'dspl-m': {
        'font-weight': '400',
    },
    'dspl-m-bold': {
        'font-weight': '700',
    },
    'dspl-s': {
        'font-weight': '400',
    },
    'dspl-s-bold': {
        'font-weight': '700',
    },
    h1: {
        'font-weight': '400',
    },
    'h1-bold': {
        'font-weight': '700',
    },
    h2: {
        'font-weight': '400',
    },
    'h2-bold': {
        'font-weight': '700',
    },
    h3: {
        'font-weight': '400',
    },
    'h3-bold': {
        'font-weight': '700',
    },
    h4: {
        'font-weight': '400',
    },
    'h4-bold': {
        'font-weight': '700',
    },
    h5: {
        'font-weight': '400',
    },
    'h5-bold': {
        'font-weight': '700',
    },
    'body-l': {
        'font-weight': '400',
    },
    'body-l-bold': {
        'font-weight': '700',
    },
    'body-m': {
        'font-weight': '400',
    },
    'body-m-bold': {
        'font-weight': '700',
    },
    'body-s': {
        'font-weight': '400',
    },
    'body-s-bold': {
        'font-weight': '700',
    },
    'body-xs': {
        'font-weight': '400',
    },
    'body-xs-bold': {
        'font-weight': '700',
    },
    'body-xxs': {
        'font-weight': '400',
    },
    'body-xxs-bold': {
        'font-weight': '700',
    },
    'text-l': {
        'font-weight': '400',
    },
    'text-l-bold': {
        'font-weight': '700',
    },
    'text-m': {
        'font-weight': '400',
    },
    'text-m-bold': {
        'font-weight': '700',
    },
    'text-s': {
        'font-weight': '400',
    },
    'text-s-bold': {
        'font-weight': '700',
    },
    'text-xs': {
        'font-weight': '400',
    },
    'text-xs-bold': {
        'font-weight': '700',
    },
};
var displayFontFamily = 'SB Sans Cond Mono';
var textFontFamily = 'SB Sans Text Mono';
exports.plasmaTypoProperties = {
    s: helpers_1.mergeTypoProps(mage_1.baseTypoS, typo, mage_1.typoCommonProps(displayFontFamily, textFontFamily)),
    m: helpers_1.mergeTypoProps(mage_1.baseTypoM, typo, mage_1.typoCommonProps(displayFontFamily, textFontFamily)),
    l: helpers_1.mergeTypoProps(mage_1.baseTypoL, typo, mage_1.typoCommonProps(displayFontFamily, textFontFamily)),
};
exports.plasmaTypoObject = {
    displayFontFamily: displayFontFamily,
    textFontFamily: textFontFamily,
    typoS: helpers_1.mergeTypoProps(mage_1.baseTypoS, typo),
    typoM: helpers_1.mergeTypoProps(mage_1.baseTypoM, typo),
    typoL: helpers_1.mergeTypoProps(mage_1.baseTypoL, typo),
    getTypoCommonProps: mage_1.typoCommonProps,
};
exports.plasma = helpers_1.createVariablesByArcheType(exports.plasmaTypoObject);
