// @salutejs/plasma-core/components/Badge/Badge.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.badgeSizes = exports.badgeRootSizes = exports.Badge = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _tokens = /*#__PURE__*/require("../../tokens");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _excluded = ["size", "text", "contentLeft"];
/**
 * Размеры в ремах.
 */
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('Badge/Badge');
var badgeSizes = exports.badgeSizes = {
  l: {
    textMarginX: '0.25rem',
    textMarginLeftAfterContent: '0.25rem'
  },
  s: {
    textMarginX: '0.125rem',
    textMarginLeftAfterContent: '0.125rem'
  }
};
var badgeRootSizes = exports.badgeRootSizes = {
  l: {
    height: '1.5rem',
    padding: '0.25rem',
    minWidth: '1.5rem',
    borderRadius: '0.75rem',
    fontSize: '0.75rem'
  },
  s: {
    height: '1rem',
    padding: '0.125rem',
    minWidth: '1rem',
    borderRadius: '0.5rem',
    fontSize: '0.625rem'
  }
};
var StyledText = /*#__PURE__*/_styledComponents["default"].span.withConfig({
  componentId: "plasma__sc-16llsfu-0"
})([""]);
var StyledContent = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-16llsfu-1"
})(["display:flex;"]);
var StyledBadge = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-16llsfu-2"
})(["", ";display:inline-flex;align-items:center;box-sizing:border-box;justify-content:center;width:max-content;", ";"], _tokens.caption, function (_ref) {
  var size = _ref.size,
    circled = _ref.circled;
  return (0, _styledComponents.css)(["", ";", " ", "{margin-left:", ";margin-right:", ";}", " ~ ", "{margin-left:", ";}"], badgeRootSizes[size], circled && (0, _styledComponents.css)(["width:", ";padding-left:0;padding-right:0;"], badgeRootSizes[size].minWidth), StyledText, badgeSizes[size].textMarginX, badgeSizes[size].textMarginX, StyledContent, StyledText, badgeSizes[size].textMarginLeftAfterContent);
});
/**
 * Небольшая бирка для ячеек и карточек.
 * Компонент может отображаться в нескольких размерах и цветах, может содержать текст и/или иконку.
 */
var Badge = exports.Badge = function Badge(_ref2) {
  var _ref2$size = _ref2.size,
    size = _ref2$size === void 0 ? 'l' : _ref2$size,
    text = _ref2.text,
    contentLeft = _ref2.contentLeft,
    rest = _objectWithoutProperties(_ref2, _excluded);
  return /*#__PURE__*/_react["default"].createElement(StyledBadge, _extends({
    size: size
  }, rest), contentLeft && /*#__PURE__*/_react["default"].createElement(StyledContent, null, contentLeft), text && /*#__PURE__*/_react["default"].createElement(StyledText, null, text));
};
