// @salutejs/plasma-tokens/es/themesValues/salutejs_B2E__light.js
// Generated by robots, do not change this manually!
/** Базовый белый цвет совпадает с whitePrimary */
export var white = '#FFFFFF';
/** Первичный белый цвет */
export var whitePrimary = '#FFFFFF';
/** Вторичный белый цвет */
export var whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный белый цвет */
export var whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** Базовый черный цвет совпадает с blackPrimary */
export var black = '#080808';
/** Первичный черный цвет */
export var blackPrimary = '#080808';
/** Вторичный черныйцвет */
export var blackSecondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный черный цвет */
export var blackTertiary = 'rgba(8, 8, 8, 0.28)';
/** Чёрный непрозрачный, не зависит от темы */
export var dark01 = '#171717';
/** Чёрный непрозрачный, не зависит от темы */
export var dark02 = '#232323';
/** Чёрный непрозрачный, не зависит от темы */
export var dark03 = '#363636';
/** Прозрачный цвет */
export var transparent = 'rgba(0, 0, 0, 0)';
/** Второстепенная прозрачная */
export var buttonClear = 'rgba(0, 0, 0, 0)';
/** Дефолтная чёрная, не зависит от темы */
export var buttonBlack = '#080808';
/** Второстепенная чёрная, не зависит от темы */
export var buttonBlackSecondary = 'rgba(8, 8, 8, 0.12)';
/** Чёрная для использования поверх картинок */
export var buttonBlackTransparent = 'rgba(8, 8, 8, 0.56)';
/** Дефолтная белая, не зависит от темы */
export var buttonWhite = '#FFFFFF';
/** Второстепенная белая, не зависит от темы */
export var buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** Базовый цвет текста, совпадает с primary */
export var text = '#080808';
/** Основной текст в интерфейсе */
export var primary = '#080808';
/** Второстепенный текст */
export var secondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный цвет текста */
export var tertiary = 'rgba(8, 8, 8, 0.28)';
/** Сплошной наборный текст */
export var paragraph = 'rgba(8, 8, 8, 0.8)';
/** Белый в светлой теме / Чёрный в тёмной теме */
export var inverse = 'rgba(255, 255, 255, 0.96)';
/** Основной бэкграунд всех приложений и смартапов */
export var background = '#FFFFFF';
/** Первичный цвет фона */
export var backgroundPrimary = '#FFFFFF';
/** Вторичный цвет фона */
export var backgroundSecondary = '#FFFFFF';
/** Третичный цвет фона */
export var backgroundTertiary = '#FFFFFF';
/** Обозначение успешного сценария */
export var success = '#0D8523';
/** Цвет предупреждения */
export var warning = '#D14D00';
/** Цвет ошибки */
export var critical = '#E31227';
/** Полупрозрачные поверхности */
export var surfaceLiquid01 = 'rgba(8, 8, 8, 0.02)';
/** Полупрозрачные поверхности */
export var surfaceLiquid02 = 'rgba(8, 8, 8, 0.06)';
/** Полупрозрачные поверхности */
export var surfaceLiquid03 = 'rgba(8, 8, 8, 0.12)';
/** Непрозрачные поверхности */
export var surfaceSolid01 = '#FAFAFA';
/** Непрозрачные поверхности */
export var surfaceSolid02 = '#F0F0F0';
/** Непрозрачные поверхности */
export var surfaceSolid03 = '#E1E1E1';
/** Основной фон для карточек */
export var surfaceCard = '#FFFFFF';
/** Первичный цвет контролов */
export var buttonPrimary = '#080808';
/** Второстепенная кнопка */
export var buttonSecondary = 'rgba(8, 8, 8, 0.06)';
/** Кнопка для успешного сценария */
export var buttonSuccess = '#148F2B';
/** Цвет предупреждения у контролов */
export var buttonWarning = '#E35502';
/** Цвет ошибки у контролов */
export var buttonCritical = '#F31B30';
/** Цвет зажатого контрола */
export var buttonChecked = '#080808';
export var skeletonGradient = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.09) 0%, rgba(8, 8, 8, 0.08) 6.25%, rgba(8, 8, 8, 0.05) 12.5%, rgba(8, 8, 8, 0.01) 25%, rgba(8, 8, 8, 0.05) 37.5%, rgba(8, 8, 8, 0.08) 43.75%, rgba(8, 8, 8, 0.09) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.05) 62.5%, rgba(8, 8, 8, 0.01) 75%, rgba(8, 8, 8, 0.05) 87.5%, rgba(8, 8, 8, 0.08) 93.75%, rgba(8, 8, 8, 0.09) 100% )';
export var skeletonGradientLighter = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.36) 0%, rgba(8, 8, 8, 0.32) 6.25%, rgba(8, 8, 8, 0.20) 12.5%, rgba(8, 8, 8, 0.04) 25%, rgba(8, 8, 8, 0.20) 37.5%, rgba(8, 8, 8, 0.32) 43.75%, rgba(8, 8, 8, 0.36) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.20) 62.5%, rgba(8, 8, 8, 0.04) 75%, rgba(8, 8, 8, 0.20) 87.5%, rgba(8, 8, 8, 0.32) 93.75%, rgba(8, 8, 8, 0.36) 100% )';
/** Цвет фона баблов отправленный сообщений */
export var speechBubbleSent = 'rgba(8, 8, 8, 0.4)';
/** Цвет фона баблов получнных сообщений */
export var speechBubbleReceived = 'rgba(0, 0, 0, 0.02)';
/** Цвет фона паранжи */
export var overlay = 'rgba(0, 0, 0, 0.8)';
export var overlaySoft = 'rgba(8, 8, 8, 0.56)';
export var overlayHard = 'rgba(8, 8, 8, 0.9)';
export var overlayBlur = 'rgba(35, 35, 35, 0.2)';
export var darkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var darkOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var darkOverlayBlur = 'rgba(35, 35, 35, 0.2)';
export var lightOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var lightOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var lightOverlayBlur = 'rgba(35, 35, 35, 0.2)';
export var inverseOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var inverseOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var inverseOverlayBlur = 'rgba(35, 35, 35, 0.2)';
/** Акцентный цвет бренда */
export var accent = '#0D8523';
/** Акцентный цвет у контролов */
export var buttonAccent = '#148F2B';
/** Цвет рамки фокуса у контрола */
export var buttonFocused = '#148F2B';
/** Градиент для заливки основного фона */
export var gradient = 'linear-gradient(336.9deg, rgba(4, 255, 44, 0.02) 0%, rgba(255, 255, 255, 0) 64.95%), radial-gradient(66.53% 100% at 73.33% 100%, rgba(0, 179, 255, 0.06) 0%, rgba(0, 209, 255, 0.01) 99.69%), linear-gradient(180deg, rgba(255, 255, 255, 0) 50%, rgba(0, 251, 59, 0.05) 100%), linear-gradient(270deg, rgba(6, 195, 255, 0.05) 0%, rgba(255, 255, 255, 0) 100%)';
/** Градиент для заливки основного фона */
export var gradientDevice = 'linear-gradient(26.05deg, rgba(24, 242, 61, 0.06) 0%, rgba(255, 255, 255, 0) 72.24%), radial-gradient(100% 100% at 0% 100%, rgba(0, 170, 255, 0.04) 0%, rgba(255, 255, 255, 0) 99.69%), radial-gradient(74.68% 149.35% at 50% 149.35%, rgba(0, 170, 255, 0.12) 0%, rgba(255, 255, 255, 0) 99.69%)';
/** Градиент подсказок о голосовых запросах */
export var voicePhraseGradient = 'linear-gradient(45deg, hsla(130, 85.26315789473684%, 37.254901960784316%, 1) 0%, hsla(189.94974874371857, 100%, 39.01960784313726%, 1) 100%)';
