// @salutejs/plasma-new-hope/styled-components/es/examples/sds_engineer/components/Divider/Divider.config.js
import { css } from 'styled-components';
import { dividerTokens } from '../../../../components/Divider';
export var config = {
  defaults: {
    view: 'default',
    size: 'm',
    orientation: 'horizontal'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/css(["", ":var(--surface-transparent-tertiary);"], dividerTokens.background),
      dark: /*#__PURE__*/css(["", ":var(--on-dark-surface-transparent-tertiary);"], dividerTokens.background),
      light: /*#__PURE__*/css(["", ":var(--on-light-surface-transparent-tertiary);"], dividerTokens.background),
      // TODO поменять на корректный токен --inverse-surface-transparent-tertiary когда он появится
      inverse: /*#__PURE__*/css(["", ":var(--inverse-text-tertiary);"], dividerTokens.background)
    },
    size: {
      m: /*#__PURE__*/css(["", ":0.0625rem;"], dividerTokens.borderRadius)
    },
    orientation: {
      horizontal: /*#__PURE__*/css(["", ":0.0625rem;"], dividerTokens.baseSideSize),
      vertical: /*#__PURE__*/css(["", ":0.0625rem;"], dividerTokens.baseSideSize)
    }
  }
};
