// @salutejs/plasma-new-hope/styled-components/cjs/components/Checkbox/variations/_focused/base.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.base = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _mixins = /*#__PURE__*/require("../../../../mixins");
var _Checkbox = /*#__PURE__*/require("../../Checkbox.tokens");
/**
 * INFO: на 36 строке нет возможности использвать .${classes.checkboxTrigger},
 * потому что линтер правит селектор и он применяется некорректно
 */

var base = exports.base = /*#__PURE__*/(0, _styledComponents.css)([".", "{", "}input[data-focus-visible-added] + label .", "{border-color:transparent;", "}input[data-focus-visible-added]:checked + label .checkbox-trigger,input[type='checkbox'][data-focus-visible-added]:indeterminate + label .", "{", "}"], _Checkbox.classes.checkboxTrigger, /*#__PURE__*/(0, _mixins.addFocus)({
  outlineOffset: '0',
  outlineSize: '0.125rem',
  outlineRadius: /*#__PURE__*/"var(".concat(_Checkbox.tokens.triggerBorderRadius, ")"),
  outlineColor: 'transparent'
}), _Checkbox.classes.checkboxTrigger, /*#__PURE__*/(0, _mixins.addFocus)({
  outlineOffset: '-0.125rem',
  outlineSize: '0.125rem',
  outlineRadius: /*#__PURE__*/"var(".concat(_Checkbox.tokens.triggerBorderRadius, ")"),
  outlineColor: /*#__PURE__*/"var(".concat(_Checkbox.tokens.focusColor, ")"),
  customFocusRules: /*#__PURE__*/"\n                &::before {\n                    box-shadow: 0 0 0 0.125rem var(".concat(_Checkbox.tokens.focusColor, ");\n                }\n            ")
}), _Checkbox.classes.checkboxTrigger, /*#__PURE__*/(0, _mixins.addFocus)({
  outlineOffset: '-0.125rem',
  outlineSize: '0.063rem',
  outlineRadius: /*#__PURE__*/"calc(var(".concat(_Checkbox.tokens.triggerBorderRadius, ") + 0.125rem)"),
  outlineColor: /*#__PURE__*/"var(".concat(_Checkbox.tokens.focusColor, ")"),
  customFocusRules: /*#__PURE__*/"\n                &::before {\n                    box-shadow: 0 0 0 0.063rem var(".concat(_Checkbox.tokens.focusColor, ");\n                }\n            ")
}));
