// @salutejs/plasma-tokens-b2b/colors/values.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.buttonPrimaryHover = exports.buttonPrimary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.critical = exports.warning = exports.success = exports.accent = exports.inputBorderFocus = exports.inputBorderHover = exports.inputBorder = exports.backgroundTertiary = exports.backgroundSecondary = exports.backgroundPrimary = exports.background = exports.inverse = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.linkVisitedActive = exports.linkVisitedHover = exports.linkVisited = exports.linkActive = exports.linkHover = exports.link = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackTransparent = exports.buttonBlackSecondary = exports.buttonBlack = exports.buttonClear = exports.transparent = exports.dark03 = exports.dark02 = exports.dark01 = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.black = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.white = void 0;
exports.inverseOverlayBlur = exports.inverseOverlayHard = exports.inverseOverlaySoft = exports.lightOverlayBlur = exports.lightOverlayHard = exports.lightOverlaySoft = exports.darkOverlayBlur = exports.darkOverlayHard = exports.darkOverlaySoft = exports.overlayBlur = exports.overlayHard = exports.overlaySoft = exports.overlay = exports.speechBubbleReceived = exports.speechBubbleSent = exports.skeletonGradientLighter = exports.skeletonGradient = exports.voicePhraseGradient = exports.gradientDevice = exports.gradient = exports.buttonFocused = exports.buttonCheckedActiveColor = exports.buttonCheckedActive = exports.buttonCheckedHoverColor = exports.buttonCheckedHover = exports.buttonChecked = exports.buttonCriticalActive = exports.buttonCriticalHover = exports.buttonCritical = exports.buttonWarningActive = exports.buttonWarningHover = exports.buttonWarning = exports.buttonSuccessActive = exports.buttonSuccessHover = exports.buttonSuccess = exports.buttonAccent = exports.buttonSecondaryActive = exports.buttonSecondaryHover = exports.buttonSecondary = exports.buttonPrimaryActive = void 0;
/** Базовый белый цвет совпадает с whitePrimary */
exports.white = '#FFFFFF';
/** Первичный белый цвет */
exports.whitePrimary = '#FFFFFF';
/** Вторичный белый цвет */
exports.whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный белый цвет */
exports.whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** Базовый черный цвет совпадает с blackPrimary */
exports.black = '#080808';
/** Первичный черный цвет */
exports.blackPrimary = '#080808';
/** Вторичный черныйцвет */
exports.blackSecondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный черный цвет */
exports.blackTertiary = 'rgba(8, 8, 8, 0.28)';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark01 = '#171717';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark02 = '#232323';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark03 = '#363636';
/** Прозрачный цвет */
exports.transparent = 'rgba(0, 0, 0, 0)';
/** Второстепенная прозрачная */
exports.buttonClear = 'rgba(0, 0, 0, 0)';
/** Дефолтная чёрная, не зависит от темы */
exports.buttonBlack = '#080808';
/** Второстепенная чёрная, не зависит от темы */
exports.buttonBlackSecondary = 'rgba(8, 8, 8, 0.12)';
/** Чёрная для использования поверх картинок */
exports.buttonBlackTransparent = 'rgba(8, 8, 8, 0.56)';
/** Дефолтная белая, не зависит от темы */
exports.buttonWhite = '#FFFFFF';
/** Второстепенная белая, не зависит от темы */
exports.buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** Базовый цвет текста, совпадает с primary */
exports.text = '#080808';
exports.link = '#2A72F8';
exports.linkHover = '#4D89F9';
exports.linkActive = '#0859F1';
exports.linkVisited = '#2364DE';
exports.linkVisitedHover = '#427AE3';
exports.linkVisitedActive = '#1C54BC';
/** Основной текст в интерфейсе */
exports.primary = '#080808';
/** Второстепенный текст */
exports.secondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный цвет текста */
exports.tertiary = 'rgba(8, 8, 8, 0.28)';
exports.paragraph = 'rgba(8, 8, 8, 0.8)';
exports.inverse = 'rgba(255, 255, 255, 0.96)';
/** Основной бэкграунд всех приложений и смартапов */
exports.background = '#FFFFFF';
/** Первичный цвет фона */
exports.backgroundPrimary = '#FFFFFF';
/** Вторичный цвет фона */
exports.backgroundSecondary = '#FFFFFF';
/** Третичный цвет фона */
exports.backgroundTertiary = '#FFFFFF';
exports.inputBorder = 'rgba(8, 8, 8, 0.28)';
exports.inputBorderHover = 'rgba(8, 8, 8, 0.56)';
exports.inputBorderFocus = '#2A72F8';
/** Акцентный цвет бренда */
exports.accent = '#2A72F8';
/** Обозначение успешного сценария */
exports.success = '#118C28';
/** Цвет предупреждения */
exports.warning = '#E35502';
/** Цвет ошибки */
exports.critical = '#F31B30';
/** Полупрозрачные поверхности */
exports.surfaceLiquid01 = 'rgba(8, 8, 8, 0.02)';
/** Полупрозрачные поверхности */
exports.surfaceLiquid02 = 'rgba(8, 8, 8, 0.06)';
/** Полупрозрачные поверхности */
exports.surfaceLiquid03 = 'rgba(8, 8, 8, 0.12)';
/** Непрозрачные поверхности */
exports.surfaceSolid01 = '#FAFAFA';
/** Непрозрачные поверхности */
exports.surfaceSolid02 = '#F0F0F0';
/** Непрозрачные поверхности */
exports.surfaceSolid03 = '#E1E1E1';
/** Основной фон для карточек */
exports.surfaceCard = '#FFFFFF';
/** Первичный цвет контролов */
exports.buttonPrimary = '#2A72F8';
exports.buttonPrimaryHover = '#397CF8';
exports.buttonPrimaryActive = '#1B68F8';
/** Второстепенная кнопка */
exports.buttonSecondary = 'rgba(8, 8, 8, 0.06)';
exports.buttonSecondaryHover = 'rgba(8, 8, 8, 0.04)';
exports.buttonSecondaryActive = 'rgba(8, 8, 8, 0.08)';
/** Акцентный цвет у контролов */
exports.buttonAccent = '#2A72F8';
/** Кнопка для успешного сценария */
exports.buttonSuccess = '#118C28';
exports.buttonSuccessHover = '#3eaa4d';
exports.buttonSuccessActive = '#1d9233';
/** Цвет предупреждения у контролов */
exports.buttonWarning = '#F55D05';
exports.buttonWarningHover = '#FA650F';
exports.buttonWarningActive = '#E65705';
/** Цвет ошибки у контролов */
exports.buttonCritical = '#F31B30';
exports.buttonCriticalHover = '#F42A3E';
exports.buttonCriticalActive = '#F20D24';
/** Цвет зажатого контрола */
exports.buttonChecked = '#080808';
exports.buttonCheckedHover = '#171717';
exports.buttonCheckedHoverColor = 'hsla(0, 0%, 100%, 0.9599999785423279)';
exports.buttonCheckedActive = '#080808';
exports.buttonCheckedActiveColor = 'rgba(255, 255, 255, 0.6)';
/** Цвет рамки фокуса у контрола */
exports.buttonFocused = '#2A72F8';
/** Градиент для заливки основного фона */
exports.gradient = 'linear-gradient(45deg, hsla(220.1123595505618, 85.57692307692307%, 59.21568627450981%, 1) 0%, hsla(189.78947368421052, 78.51239669421487%, 52.54901960784314%, 1) 100%)';
exports.gradientDevice = '';
exports.voicePhraseGradient = '';
exports.skeletonGradient = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.09) 0%, rgba(8, 8, 8, 0.08) 6.25%, rgba(8, 8, 8, 0.05) 12.5%, rgba(8, 8, 8, 0.01) 25%, rgba(8, 8, 8, 0.05) 37.5%, rgba(8, 8, 8, 0.08) 43.75%, rgba(8, 8, 8, 0.09) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.05) 62.5%, rgba(8, 8, 8, 0.01) 75%, rgba(8, 8, 8, 0.05) 87.5%, rgba(8, 8, 8, 0.08) 93.75%, rgba(8, 8, 8, 0.09) 100% )';
exports.skeletonGradientLighter = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.36) 0%, rgba(8, 8, 8, 0.32) 6.25%, rgba(8, 8, 8, 0.20) 12.5%, rgba(8, 8, 8, 0.04) 25%, rgba(8, 8, 8, 0.20) 37.5%, rgba(8, 8, 8, 0.32) 43.75%, rgba(8, 8, 8, 0.36) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.20) 62.5%, rgba(8, 8, 8, 0.04) 75%, rgba(8, 8, 8, 0.20) 87.5%, rgba(8, 8, 8, 0.32) 93.75%, rgba(8, 8, 8, 0.36) 100% )';
exports.speechBubbleSent = '';
exports.speechBubbleReceived = '';
/** Цвет фона паранжи */
exports.overlay = 'rgba(0, 0, 0, 0.8)';
exports.overlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.overlayHard = 'rgba(8, 8, 8, 0.9)';
exports.overlayBlur = 'rgba(35, 35, 35, 0.2)';
exports.darkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.darkOverlayHard = 'rgba(8, 8, 8, 0.9)';
exports.darkOverlayBlur = 'rgba(35, 35, 35, 0.2)';
exports.lightOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.lightOverlayHard = 'rgba(8, 8, 8, 0.9)';
exports.lightOverlayBlur = 'rgba(35, 35, 35, 0.2)';
exports.inverseOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.inverseOverlayHard = 'rgba(8, 8, 8, 0.9)';
exports.inverseOverlayBlur = 'rgba(35, 35, 35, 0.2)';
