// @salutejs/plasma-new-hope/cjs/components/Popover/Popover.styles.js
import './Popover.styles_rmr9za.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
var index = require('../Popup/utils/index.js');

var StyledWrapper = /*#__PURE__*/react.styled('div')({
  name: "StyledWrapper",
  "class": "s1g1xbt5",
  propsAsIs: false
});
var StyledRoot = /*#__PURE__*/react.styled('div')({
  name: "StyledRoot",
  "class": "s12xwbyh",
  propsAsIs: false
});
var StyledArrow = /*#__PURE__*/react.styled('div')({
  name: "StyledArrow",
  "class": "s1pd4d8a",
  propsAsIs: false
});
var _exp7 = function _exp7() {
  return function (_ref) {
    var zIndex = _ref.zIndex;
    return zIndex || index.DEFAULT_Z_INDEX;
  };
};
var StyledPopover = /*#__PURE__*/react.styled('div')({
  name: "StyledPopover",
  "class": "s16xlixz",
  propsAsIs: false,
  vars: {
    "s16xlixz-0": [/*#__PURE__*/_exp7()]
  }
});

exports.StyledArrow = StyledArrow;
exports.StyledPopover = StyledPopover;
exports.StyledRoot = StyledRoot;
exports.StyledWrapper = StyledWrapper;
