// @salutejs/plasma-new-hope/styled-components/es/components/Combobox/Combobox.styles.js
import styled from 'styled-components';
import { component } from '../../engines';
import { popoverClasses, popoverConfig } from '../Popover';
import { tokens } from './Combobox.tokens';
export var StyledRoot = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-gt7m6a-0"
})(["width:100%;position:relative;& .", "{display:block;}"], popoverClasses.wrapper);
export var StyledCombobox = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-gt7m6a-1"
})(["box-sizing:border-box;background:var(", ");box-shadow:var(", ");border-radius:var(", ");width:var(", ");height:var(", ");padding:var(", ") var(", ") var(", ") var(", ");"], tokens.background, tokens.boxShadow, tokens.borderRadius, tokens.width, tokens.height, tokens.paddingTop, tokens.paddingRight, tokens.paddingBottom, tokens.paddingLeft);

// issue #823
var Popover = /*#__PURE__*/component(popoverConfig);
export var StyledPopover = /*#__PURE__*/styled(Popover).withConfig({
  componentId: "plasma-new-hope__sc-gt7m6a-2"
})(["display:block;.", ",.", "{width:", ";}"], /*#__PURE__*/String(popoverClasses.root), /*#__PURE__*/String(popoverClasses.target), function (_ref) {
  var usePortal = _ref.usePortal,
    selectWidth = _ref.selectWidth;
  return usePortal ? "".concat(selectWidth, "px") : '100%';
});
export var ComboboxItemNotFound = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-gt7m6a-3"
})(["text-align:center;padding:0.5rem;font-family:var(", ");font-size:var(", ");font-style:var(", ");font-weight:var(", ");letter-spacing:var(", ");line-height:var(", ");"], tokens.itemFontFamily, tokens.itemFontSize, tokens.itemFontStyle, tokens.itemFontWeightBold, tokens.itemFontLetterSpacing, tokens.itemFontLineHeight);
export var StyledNativeSelect = /*#__PURE__*/styled.select.withConfig({
  componentId: "plasma-new-hope__sc-gt7m6a-4"
})(["display:none;appearance:none;-moz-appearance:none;-webkit-appearance:none;background:transparent;color:transparent;border:none;outline:none;opacity:0;position:absolute;inset:0;"]);
