// @salutejs/plasma-new-hope/styled-components/es/examples/sds_engineer/components/Toolbar/Toolbar.config.js
import { css } from 'styled-components';
import { toolbarTokens } from '../../../../components/Toolbar';
export var config = {
  defaults: {
    view: 'default',
    size: 'm'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/css(["", ":var(--shadow-down-soft-s);", ":var(--surface-solid-card);", ":var(--surface-transparent-tertiary);"], toolbarTokens.boxShadow, toolbarTokens.background, toolbarTokens.dividerColor)
    },
    size: {
      xs: /*#__PURE__*/css(["", ":2.5rem;", ":0.25rem;", ":0.75rem;", ":1.25rem;", ":0.5rem;", ":0.0625rem;"], toolbarTokens.size, toolbarTokens.padding, toolbarTokens.borderRadius, toolbarTokens.dividerSize, toolbarTokens.dividerOffset, toolbarTokens.dividerBorderRadius),
      s: /*#__PURE__*/css(["", ":3rem;", ":0.25rem;", ":0.875rem;", ":1.5rem;", ":0.5rem;", ":0.0625rem;"], toolbarTokens.size, toolbarTokens.padding, toolbarTokens.borderRadius, toolbarTokens.dividerSize, toolbarTokens.dividerOffset, toolbarTokens.dividerBorderRadius),
      m: /*#__PURE__*/css(["", ":3.75rem;", ":0.375rem;", ":1rem;", ":1.75rem;", ":0.625rem;", ":0.0625rem;"], toolbarTokens.size, toolbarTokens.padding, toolbarTokens.borderRadius, toolbarTokens.dividerSize, toolbarTokens.dividerOffset, toolbarTokens.dividerBorderRadius),
      l: /*#__PURE__*/css(["", ":4.5rem;", ":0.5rem;", ":1.25rem;", ":2rem;", ":0.75rem;", ":0.0625rem;"], toolbarTokens.size, toolbarTokens.padding, toolbarTokens.borderRadius, toolbarTokens.dividerSize, toolbarTokens.dividerOffset, toolbarTokens.dividerBorderRadius)
    }
  }
};
