// @salutejs/plasma-ui/es/components/Tabs/Tabs.perftest.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import React, { useMemo } from 'react';
import styled from 'styled-components';
import { Tabs, TabItem } from '.';
var StorybookTabsWrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-ui__sc-u6wu2m-0"
})(["margin-top:0.5rem;"]);
var defaultProps = {
  itemsNumber: 4,
  size: 'l',
  view: 'secondary',
  stretch: true,
  pilled: false,
  scaleOnInteraction: true,
  outlined: false,
  disabled: false,
  text: 'Label'
};
var icon = /*#__PURE__*/React.createElement("div", {
  style: {
    width: 10,
    height: 10
  }
});
export var Default = function Default() {
  var itemsNumber = defaultProps.itemsNumber,
    size = defaultProps.size,
    view = defaultProps.view,
    stretch = defaultProps.stretch,
    pilled = defaultProps.pilled,
    scaleOnInteraction = defaultProps.scaleOnInteraction,
    text = defaultProps.text,
    disabled = defaultProps.disabled,
    outlined = defaultProps.outlined;
  var id = 'tabs-example';
  var _React$useState = React.useState(0),
    _React$useState2 = _slicedToArray(_React$useState, 2),
    index = _React$useState2[0],
    setIndex = _React$useState2[1];
  var items = useMemo(function () {
    var res = [];
    var _loop = function _loop(i) {
      res.push({
        label: "".concat(text, " ").concat(i, " "),
        onFocus: function onFocus() {},
        onBlur: function onBlur() {},
        onClick: function onClick() {
          !disabled && setIndex(i);
        }
      });
    };
    for (var i = 0; i < itemsNumber; i++) {
      _loop(i);
    }
    return res;
  }, [itemsNumber, text]);
  var tabIndex = disabled ? -1 : 0;
  return /*#__PURE__*/React.createElement(StorybookTabsWrapper, null, /*#__PURE__*/React.createElement(Tabs, {
    id: id,
    size: size,
    view: view,
    stretch: stretch,
    pilled: pilled,
    scaleOnPress: scaleOnInteraction,
    outlined: !disabled && outlined,
    disabled: disabled,
    index: index,
    animated: true,
    forwardedAs: "ul"
  }, items.map(function (elem, i) {
    return /*#__PURE__*/React.createElement(TabItem, {
      key: "item:".concat(i),
      forwardedAs: "li",
      isActive: i === index,
      "aria-controls": id,
      tabIndex: tabIndex,
      contentLeft: icon,
      onClick: elem.onClick,
      onFocus: elem.onFocus,
      onBlur: elem.onBlur
    }, elem.label);
  })));
};
