// @salutejs/plasma-new-hope/es/components/Modal/Modal.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef, useMemo, useCallback } from 'react';
import { useFocusTrap, useForkRef, safeUseId } from '@salutejs/plasma-core';
import '@linaria/core';
import { component } from '../../engines/common.js';
import { usePopupContext } from '../Popup/PopupContext.js';
import { popupConfig } from '../Popup/Popup.js';
import { Overlay } from '../Overlay/Overlay.js';
import { DEFAULT_Z_INDEX } from '../Popup/utils/index.js';
import { classes, tokens } from './Modal.tokens.js';
import { useModal } from './hooks/useModal.js';
import { base } from './variations/_view/base.js';
import { getIdLastModal } from './ModalContext.js';

var _excluded = ["id", "withAnimation", "onClose", "onOverlayClick", "onEscKeyDown", "closeOnEsc", "closeOnOverlayClick", "withBlur", "initialFocusRef", "focusAfterRef", "zIndex", "popupInfo", "children", "view", "isOpen"];

// issue #823
var Popup = /*#__PURE__*/component(popupConfig);

/**
 * Modal.
 * Управляет показом/скрытием, подложкой и анимацией визуальной части модального окна.
 */
var modalRoot = function modalRoot(Root) {
  return /*#__PURE__*/forwardRef(function (_ref, outerRootRef) {
    var id = _ref.id,
      withAnimation = _ref.withAnimation,
      onClose = _ref.onClose,
      onOverlayClick = _ref.onOverlayClick,
      onEscKeyDown = _ref.onEscKeyDown,
      _ref$closeOnEsc = _ref.closeOnEsc,
      closeOnEsc = _ref$closeOnEsc === void 0 ? true : _ref$closeOnEsc,
      _ref$closeOnOverlayCl = _ref.closeOnOverlayClick,
      closeOnOverlayClick = _ref$closeOnOverlayCl === void 0 ? true : _ref$closeOnOverlayCl,
      withBlur = _ref.withBlur,
      initialFocusRef = _ref.initialFocusRef,
      focusAfterRef = _ref.focusAfterRef,
      zIndex = _ref.zIndex,
      popupInfo = _ref.popupInfo,
      children = _ref.children,
      view = _ref.view,
      isOpen = _ref.isOpen,
      rest = _objectWithoutProperties(_ref, _excluded);
    var trapRef = useFocusTrap(true, initialFocusRef, focusAfterRef);
    var popupController = usePopupContext();
    var innerRef = useForkRef(trapRef, outerRootRef);
    var uniqId = safeUseId();
    var innerId = id || uniqId;
    var overlayBackgroundToken = withBlur ? "var(".concat(tokens.modalOverlayWithBlurColor, ")") : "var(".concat(tokens.modalOverlayColor, ")");
    var _useModal = useModal({
        id: innerId,
        isOpen: isOpen,
        closeOnEsc: closeOnEsc,
        onEscKeyDown: onEscKeyDown,
        onClose: onClose,
        popupInfo: popupInfo
      }),
      modalInfo = _useModal.modalInfo;
    var transparent = useMemo(function () {
      return getIdLastModal(popupController.items) !== innerId;
    }, [innerId, popupController.items]);
    var onModalOverlayKeyDown = useCallback(function (event) {
      if (onOverlayClick) {
        onOverlayClick(event);
        return;
      }
      if (onClose) {
        onClose();
      }
    }, [closeOnOverlayClick, onOverlayClick, onClose]);
    return /*#__PURE__*/React.createElement(Popup, _extends({
      id: innerId,
      isOpen: isOpen,
      ref: innerRef,
      popupInfo: modalInfo,
      withAnimation: withAnimation,
      zIndex: zIndex,
      overlay: /*#__PURE__*/React.createElement(Root, {
        view: view
      }, /*#__PURE__*/React.createElement(Overlay, {
        className: classes.overlay,
        zIndex: zIndex || DEFAULT_Z_INDEX,
        backgroundColorProperty: overlayBackgroundToken,
        withBlur: withBlur,
        transparent: transparent,
        isClickable: closeOnOverlayClick,
        onOverlayClick: onModalOverlayKeyDown
      }))
    }, rest), children);
  });
};
var modalConfig = {
  name: 'Modal',
  tag: 'div',
  layout: modalRoot,
  base: '',
  variations: {
    view: {
      css: base
    }
  },
  defaults: {
    view: 'default'
  }
};

export { modalConfig, modalRoot };
