// @salutejs/plasma-new-hope/cjs/components/Pagination/ui/PaginationQuickJumpToPage/PaginationQuickJumpToPage.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var index = require('../../utils/index.js');
var PaginationQuickJumpToPage_styles = require('./PaginationQuickJumpToPage.styles.js');

var _excluded = ["placeholderQuickJump", "textQuickJump", "onChangeValue"];
var PaginationQuickJumpToPage = function PaginationQuickJumpToPage(_ref) {
  var _ref$placeholderQuick = _ref.placeholderQuickJump,
    placeholderQuickJump = _ref$placeholderQuick === void 0 ? index.defaultValues.placeholderQuickJump : _ref$placeholderQuick,
    _ref$textQuickJump = _ref.textQuickJump,
    textQuickJump = _ref$textQuickJump === void 0 ? index.defaultValues.textQuickJump : _ref$textQuickJump,
    onChangeValue = _ref.onChangeValue,
    rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
  var _useState = React.useState(),
    _useState2 = _rollupPluginBabelHelpers.slicedToArray(_useState, 2),
    pageValue = _useState2[0],
    setPageValue = _useState2[1];
  var handleKeyPress = function handleKeyPress(event) {
    if (event.key === 'Enter') {
      onChangeValue === null || onChangeValue === void 0 || onChangeValue(Number(pageValue));
      setPageValue('');
    }
  };
  var handleChangeInput = function handleChangeInput(event) {
    var result = event.currentTarget.value.replace(/\D/g, '');
    if (result === '') return setPageValue('');
    setPageValue(Number(result));
  };
  return /*#__PURE__*/React.createElement(PaginationQuickJumpToPage_styles.QuickJumpToPageRoot, rest, /*#__PURE__*/React.createElement(PaginationQuickJumpToPage_styles.QuickJumpToPageTypography, null, textQuickJump), /*#__PURE__*/React.createElement(PaginationQuickJumpToPage_styles.QuickJumpToPageInput, {
    value: pageValue,
    placeholder: placeholderQuickJump,
    onKeyUp: handleKeyPress,
    onChange: handleChangeInput
  }));
};

exports.PaginationQuickJumpToPage = PaginationQuickJumpToPage;
