// @salutejs/plasma-new-hope/styled-components/cjs/components/Drawer/Drawer.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledPopup = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _engines = /*#__PURE__*/require("../../engines");
var _Popup = /*#__PURE__*/require("../Popup");
var _Drawer = /*#__PURE__*/require("./Drawer.tokens");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var Popup = /*#__PURE__*/(0, _engines.component)(_Popup.popupConfig);
var animationVariants = {
  right: {
    show: {
      start: 'translate(100%, -50%)',
      end: 'translate(0, -50%)'
    },
    hide: {
      start: 'translate(0, -50%)',
      end: 'translate(100%, -50%)'
    }
  },
  bottom: {
    show: {
      start: 'translate(-50%, 100%)',
      end: 'translate(-50%, 0)'
    },
    hide: {
      start: 'translate(-50%, 0)',
      end: 'translate(-50%, 100%)'
    }
  },
  left: {
    show: {
      start: 'translate(-100%, -50%)',
      end: 'translate(0, -50%)'
    },
    hide: {
      start: 'translate(0, -50%)',
      end: 'translate(-100%, -50%)'
    }
  },
  top: {
    show: {
      start: 'translate(-50%, -100%)',
      end: 'translate(-50%, 0)'
    },
    hide: {
      start: 'translate(-50%, 0)',
      end: 'translate(-50%, -100%)'
    }
  }
};
var StyledPopup = exports.StyledPopup = /*#__PURE__*/(0, _styledComponents["default"])(Popup).withConfig({
  componentId: "plasma-new-hope__sc-cup208-0"
})(["&&.", "{animation:fadeIn 0.2s forwards;}&&.", " .", "{animation:fadeOut 0.2s forwards;}@keyframes fadeIn{from{opacity:0;}to{opacity:1;}}@keyframes fadeOut{from{opacity:1;}to{opacity:0;}}&& .", ",&& .", " > div{width:", ";height:", ";}&& .", "{animation:showAnimation 0.2s forwards;}&&.", " .", "{animation:hideAnimation 0.2s forwards;}@keyframes showAnimation{0%{transform:", ";opacity:0;}100%{transform:", ";opacity:1;}}@keyframes hideAnimation{0%{transform:", ";opacity:1;}100%{transform:", ";opacity:0;}}"], _Drawer.classes.overlay, _Popup.popupClasses.endAnimation, _Drawer.classes.overlay, _Popup.popupClasses.root, _Popup.popupClasses.root, function (_ref) {
  var width = _ref.width;
  return width;
}, function (_ref2) {
  var height = _ref2.height;
  return height;
}, _Popup.popupClasses.root, _Popup.popupClasses.endAnimation, _Popup.popupClasses.root, function (_ref3) {
  var placement = _ref3.placement;
  return animationVariants[placement].show.start;
}, function (_ref4) {
  var placement = _ref4.placement;
  return animationVariants[placement].show.end;
}, function (_ref5) {
  var placement = _ref5.placement;
  return animationVariants[placement].hide.start;
}, function (_ref6) {
  var placement = _ref6.placement;
  return animationVariants[placement].hide.end;
});
