// @salutejs/plasma-ui/es/components/ProductCard/ProductCardPrice.js
import React, { memo } from 'react';
import styled, { css } from 'styled-components';
import { black, success } from '@salutejs/plasma-core';
import { Price } from '../Price';
import { Body2, Caption } from '../Typography';
import { PriceTypes } from './types';
var StyledPrices = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-ui__sc-eqx89j-0"
})(["display:flex;flex-direction:column;justify-content:center;width:100%;min-height:2.375rem;"]);
var StyledPrice = /*#__PURE__*/styled(Price).withConfig({
  componentId: "plasma-ui__sc-eqx89j-1"
})(["color:", ";", ""], function (_ref) {
  var $type = _ref.$type;
  switch ($type) {
    case PriceTypes.NEW:
      return success;
    default:
      return black;
  }
}, function (_ref2) {
  var $type = _ref2.$type;
  return $type === PriceTypes.OLD && css(["opacity:0.56;"]);
});
export var ProductCardPrice = /*#__PURE__*/memo(function (_ref3) {
  var price = _ref3.price,
    oldPrice = _ref3.oldPrice,
    periodicity = _ref3.periodicity;
  if (!price) {
    return null;
  }
  return /*#__PURE__*/React.createElement(StyledPrices, null, /*#__PURE__*/React.createElement(StyledPrice, {
    $type: oldPrice !== undefined ? PriceTypes.NEW : undefined,
    forwardedAs: Body2,
    periodicity: periodicity
  }, price), oldPrice && /*#__PURE__*/React.createElement(StyledPrice, {
    $type: PriceTypes.OLD,
    stroke: true,
    forwardedAs: Caption,
    periodicity: periodicity
  }, oldPrice));
});
