// @salutejs/plasma-new-hope/styled-components/es/components/Calendar/ui/CalendarDayItem/CalendarDayItem.styles.js
import styled from 'styled-components';
import { flexCenter, selectedMixin } from '../../mixins';
import { classes, tokens } from '../../Calendar.tokens';
import { addFocus } from '../../../../mixins';
var inRange = function inRange() {
  return "\n    &:before {\n        content: '';\n        z-index: -1;\n        position: absolute;\n        width: 2.5rem;\n        height: 1.875rem;\n        background: var(".concat(tokens.calendarRangeBackground, ");\n    }\n");
};
var sideInRange = function sideInRange() {
  return "\n    &:before {\n        content: '';\n        z-index: -1;\n        position: absolute;\n        width: 0.313rem;\n        height: 1.875rem;\n        background: var(".concat(tokens.calendarRangeBackground, ");\n    }\n");
};
export var StyledDay = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-2y5uj0-0"
})(["border-radius:0.375rem;", ";&.", "{", ";}&.", "{", ";}&.", "{&:before{left:", ";}}&.", "{&:before{right:", ";}}"], /*#__PURE__*/String(flexCenter), /*#__PURE__*/String(classes.dayInRange), /*#__PURE__*/inRange(), /*#__PURE__*/String(classes.daySideInRange), /*#__PURE__*/sideInRange(), /*#__PURE__*/String(classes.daySideLeft), function (_ref) {
  var offset = _ref.offset;
  return "".concat(offset, "px");
}, /*#__PURE__*/String(classes.daySideRight), function (_ref2) {
  var offset = _ref2.offset;
  return "".concat(offset, "px");
});
var dayOfWeek = function dayOfWeek() {
  return "\n    color: var(--plasma-colors-tertiary);\n";
};
var disabledDay = function disabledDay() {
  return "\n    cursor: not-allowed;\n    opacity: 0.4;\n\n    ".concat(addFocus({
    outlineSize: '0.063rem',
    outlineOffset: '-0.063rem',
    outlineColor: "var(".concat(tokens.calendarOutlineFocusColor, ")"),
    outlineRadius: '0.563rem'
  }), ";\n");
};
var disabledCurrentDay = function disabledCurrentDay() {
  return "\n    ".concat(disabledDay(), ";\n    --calendar_private-add-focus-outline-offset: 0.125rem;\n");
};
export var StyledDayRoot = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-2y5uj0-1"
})(["font-family:var(", ");font-size:var(", ");font-style:var(", ");font-weight:var(", ");letter-spacing:var(", ");line-height:var(", ");position:relative;box-sizing:border-box;min-width:2.5rem;min-height:2rem;border-radius:var(", ");", ";color:", ";visibility:", ";&.", "{--calendar_private-selected-min-width:2.25rem;--calendar_private-selected-min-height:1.75rem;--calendar_private-selected-background:var(", ");--calendar_private-selected-color:var(", ");--calendar_private-selectable-background-hover:var(", ");--calendar_private-current-border-color:var(", ");--calendar_private-current-background-hover:var(", ");--calendar_private-current-color-hover:var(", ");--calendar_private-current-child-background-hover:var(", ");--calendar_private-active-background:var(", ");--calendar_private-active-color:var(", ");--calendar_private-hovered-background:var(", ");--calendar_private-hovered-color:var(", ");", ";", ";}&.", "{", ";}&.", "{", ";}&.", "{", ";}"], tokens.calendarDayFontFamily, tokens.calendarDayFontSize, tokens.calendarDayFontStyle, tokens.calendarDayFontWeight, tokens.calendarDayFontLetterSpacing, tokens.calendarDayFontLineHeight, tokens.calendarItemBorderRadius, /*#__PURE__*/String(flexCenter), function (_ref3) {
  var isDayInCurrentMonth = _ref3.isDayInCurrentMonth;
  return isDayInCurrentMonth ? "var(".concat(tokens.calendarContentPrimaryColor, ")") : "var(".concat(tokens.calendarContentSecondaryColor, ")");
}, function (_ref4) {
  var isDayInCurrentMonth = _ref4.isDayInCurrentMonth,
    isDouble = _ref4.isDouble;
  return !isDayInCurrentMonth && isDouble ? 'hidden' : 'visible';
}, /*#__PURE__*/String(classes.selectableItem), tokens.calendarSelectedItemBackground, tokens.calendarSelectedItemColor, tokens.calendarSelectableItemBackgroundHover, tokens.calendarCurrentItemBorderColor, tokens.calendarCurrentItemBackgroundHover, tokens.calendarCurrentItemColorHover, tokens.calendarCurrentItemChildBackgroundHover, tokens.calendarActiveItemBackground, tokens.calendarActiveItemColor, tokens.calendarHoveredItemBackground, tokens.calendarHoveredItemColor, /*#__PURE__*/addFocus({
  outlineSize: '0.063rem',
  outlineOffset: '-0.063rem',
  outlineColor: /*#__PURE__*/"var(".concat(tokens.calendarOutlineFocusColor, ")"),
  outlineRadius: '0.563rem'
}), /*#__PURE__*/String(selectedMixin), /*#__PURE__*/String(classes.dayOfWeek), /*#__PURE__*/dayOfWeek(), /*#__PURE__*/String(classes.dayDisabled), /*#__PURE__*/disabledDay(), /*#__PURE__*/String(classes.dayDisabledCurrent), /*#__PURE__*/disabledCurrentDay());
export var StyledEvents = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-2y5uj0-2"
})(["display:flex;position:absolute;bottom:0.25rem;"]);
export var StyledEvent = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-2y5uj0-3"
})(["margin:0 0.063rem;width:0.188rem;height:0.188rem;border-radius:50%;background-color:", ";"], function (_ref5) {
  var _ref5$color = _ref5.color,
    color = _ref5$color === void 0 ? 'var(--plasma-colors-accent)' : _ref5$color;
  return color;
});
