// @salutejs/plasma-new-hope/styled-components/es/components/Calendar/ui/CalendarDayItem/CalendarDayItem.js
var _excluded = ["isFocused", "dayOfWeek", "disabled", "isCurrent", "isSelected", "isDayInCurrentMonth", "isDouble", "inRange", "isHovered", "sideInRange", "eventList", "day", "monthIndex", "year", "onClick", "onMouseOver", "onFocus", "disabledArrowKey", "disabledMonths"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React, { forwardRef, memo } from 'react';
import { cx } from '../../../../utils';
import { classes } from '../../Calendar.tokens';
import { StyledDay, StyledDayRoot, StyledEvent, StyledEvents } from './CalendarDayItem.styles';

/**
 * Компонент дня в календаре.
 */
export var CalendarDayItem = /*#__PURE__*/memo( /*#__PURE__*/forwardRef(function (_ref, outerRef) {
  var isFocused = _ref.isFocused,
    dayOfWeek = _ref.dayOfWeek,
    disabled = _ref.disabled,
    isCurrent = _ref.isCurrent,
    isSelected = _ref.isSelected,
    isDayInCurrentMonth = _ref.isDayInCurrentMonth,
    isDouble = _ref.isDouble,
    inRange = _ref.inRange,
    isHovered = _ref.isHovered,
    sideInRange = _ref.sideInRange,
    _ref$eventList = _ref.eventList,
    eventList = _ref$eventList === void 0 ? [] : _ref$eventList,
    day = _ref.day,
    monthIndex = _ref.monthIndex,
    year = _ref.year,
    onClick = _ref.onClick,
    onMouseOver = _ref.onMouseOver,
    onFocus = _ref.onFocus,
    disabledArrowKey = _ref.disabledArrowKey,
    disabledMonths = _ref.disabledMonths,
    rest = _objectWithoutProperties(_ref, _excluded);
  var selectableClass = !dayOfWeek && !disabled ? classes.selectableItem : undefined;
  var selectedClass = isSelected ? classes.selectedItem : undefined;
  var currentClass = isCurrent ? classes.currentItem : undefined;
  var hoveredClass = isHovered ? classes.hoveredItem : undefined;
  var dayOfWeekClass = dayOfWeek ? classes.dayOfWeek : undefined;
  var disabledClass = disabled && !isCurrent ? classes.dayDisabled : undefined;
  var disabledCurrentClass = disabled && isCurrent ? classes.dayDisabledCurrent : undefined;
  var sideInRangeLeft = sideInRange === 'left' ? classes.daySideLeft : undefined;
  var sideInRangeRight = sideInRange === 'right' ? classes.daySideRight : undefined;
  var sideInRangeClass = sideInRange ? classes.daySideInRange : undefined;
  var inRangeStyle = inRange ? classes.dayInRange : undefined;
  var sideOffset = !isSelected && isCurrent ? -1 : 0;
  return /*#__PURE__*/React.createElement(StyledDayRoot, _extends({
    ref: outerRef,
    className: cx(dayOfWeekClass, selectableClass, selectedClass, currentClass, disabledClass, disabledCurrentClass, hoveredClass),
    tabIndex: isFocused ? 0 : -1,
    dayOfWeek: dayOfWeek,
    isCurrent: isCurrent,
    isSelected: isSelected,
    isDayInCurrentMonth: isDayInCurrentMonth,
    isDouble: isDouble,
    isHovered: isHovered,
    sideInRange: sideInRange,
    onClick: disabled ? undefined : onClick,
    onMouseOver: onMouseOver,
    onFocus: onFocus,
    "data-day": day,
    "data-month-index": monthIndex,
    "data-year": year,
    "data-disabled-arrow-key": disabledArrowKey || undefined,
    "data-disabled-months": disabledMonths || undefined,
    "aria-selected": isSelected,
    "aria-disabled": disabled,
    "aria-describedby": disabled ? 'withShift' : undefined
  }, rest), /*#__PURE__*/React.createElement(StyledDay, {
    className: cx(sideInRangeClass, sideInRangeLeft, sideInRangeRight, inRangeStyle),
    offset: sideOffset
  }, day), /*#__PURE__*/React.createElement(StyledEvents, {
    "aria-hidden": true
  }, [eventList[0], eventList[1], eventList[2]].map(function (event, i) {
    return event && /*#__PURE__*/React.createElement(StyledEvent, _extends({
      key: "".concat(event.date, "-").concat(i)
    }, event));
  })));
}));
