// @salutejs/plasma-web/css/es/components/Combobox/Combobox.js
import { component, mergeConfig, comboboxConfig, comboboxItemConfig, comboboxDividerConfig, comboboxFooterConfig, comboboxGroupConfig, comboboxHeaderConfig } from '@salutejs/plasma-new-hope';
import { config } from './Combobox.config.js';

var mergedConfig = /*#__PURE__*/mergeConfig(comboboxConfig, config);
var Combobox = /*#__PURE__*/component(mergedConfig);
var mergedItemConfig = /*#__PURE__*/mergeConfig(comboboxItemConfig);
var ComboboxItem = /*#__PURE__*/component(mergedItemConfig);
var mergedDividerConfig = /*#__PURE__*/mergeConfig(comboboxDividerConfig);
var ComboboxDivider = /*#__PURE__*/component(mergedDividerConfig);
var mergedFooterConfig = /*#__PURE__*/mergeConfig(comboboxFooterConfig);
var ComboboxFooter = /*#__PURE__*/component(mergedFooterConfig);
var mergedGroupConfig = /*#__PURE__*/mergeConfig(comboboxGroupConfig);
var ComboboxGroup = /*#__PURE__*/component(mergedGroupConfig);
var mergedHeaderConfig = /*#__PURE__*/mergeConfig(comboboxHeaderConfig);
var ComboboxHeader = /*#__PURE__*/component(mergedHeaderConfig);

export { Combobox, ComboboxDivider, ComboboxFooter, ComboboxGroup, ComboboxHeader, ComboboxItem };
