// @salutejs/plasma-tokens-b2b/new/values.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.onDarkTextWarningActive = exports.onDarkTextWarningHover = exports.onDarkTextPositive = exports.onDarkTextPositiveActive = exports.onDarkTextPositiveHover = exports.onDarkTextAccentGradient = exports.onDarkTextAccentGradientActive = exports.onDarkTextAccentGradientHover = exports.onDarkTextAccent = exports.onDarkTextAccentActive = exports.onDarkTextAccentHover = exports.onDarkTextParagraph = exports.onDarkTextParagraphActive = exports.onDarkTextParagraphHover = exports.onDarkTextTertiary = exports.onDarkTextTertiaryActive = exports.onDarkTextTertiaryHover = exports.onDarkTextSecondary = exports.onDarkTextSecondaryActive = exports.onDarkTextSecondaryHover = exports.onDarkTextPrimary = exports.onDarkTextPrimaryActive = exports.onDarkTextPrimaryHover = exports.textNegative = exports.textNegativeActive = exports.textNegativeHover = exports.textWarning = exports.textWarningActive = exports.textWarningHover = exports.textPositive = exports.textPositiveActive = exports.textPositiveHover = exports.textAccentGradient = exports.textAccentGradientActive = exports.textAccentGradientHover = exports.textAccent = exports.textAccentActive = exports.textAccentHover = exports.textParagraph = exports.textParagraphActive = exports.textParagraphHover = exports.textTertiary = exports.textTertiaryActive = exports.textTertiaryHover = exports.textSecondary = exports.textSecondaryActive = exports.textSecondaryHover = exports.textPrimary = exports.textPrimaryActive = exports.textPrimaryHover = void 0;
exports.surfaceSolidTertiaryHover = exports.surfaceSolidSecondary = exports.surfaceSolidSecondaryActive = exports.surfaceSolidSecondaryHover = exports.surfaceSolidPrimary = exports.surfaceSolidPrimaryActive = exports.surfaceSolidPrimaryHover = exports.inverseTextParagraph = exports.inverseTextParagraphActive = exports.inverseTextParagraphHover = exports.inverseTextTertiary = exports.inverseTextTertiaryActive = exports.inverseTextTertiaryHover = exports.inverseTextSecondary = exports.inverseTextSecondaryActive = exports.inverseTextSecondaryHover = exports.inverseTextPrimary = exports.inverseTextPrimaryActive = exports.inverseTextPrimaryHover = exports.onLightTextNegative = exports.onLightTextNegativeActive = exports.onLightTextNegativeHover = exports.onLightTextWarning = exports.onLightTextWarningActive = exports.onLightTextWarningHover = exports.onLightTextPositive = exports.onLightTextPositiveActive = exports.onLightTextPositiveHover = exports.onLightTextAccentGradient = exports.onLightTextAccentGradientActive = exports.onLightTextAccentGradientHover = exports.onLightTextAccent = exports.onLightTextAccentActive = exports.onLightTextAccentHover = exports.onLightTextParagraph = exports.onLightTextParagraphActive = exports.onLightTextParagraphHover = exports.onLightTextTertiary = exports.onLightTextTertiaryActive = exports.onLightTextTertiaryHover = exports.onLightTextSecondary = exports.onLightTextSecondaryActive = exports.onLightTextSecondaryHover = exports.onLightTextPrimary = exports.onLightTextPrimaryActive = exports.onLightTextPrimaryHover = exports.onDarkTextNegative = exports.onDarkTextNegativeActive = exports.onDarkTextNegativeHover = exports.onDarkTextWarning = void 0;
exports.surfaceTransparentAccent = exports.surfaceTransparentNegativeActive = exports.surfaceTransparentNegativeHover = exports.surfaceTransparentWarningActive = exports.surfaceTransparentWarningHover = exports.surfaceTransparentPositiveActive = exports.surfaceTransparentPositiveHover = exports.surfaceNegative = exports.surfaceNegativeActive = exports.surfaceNegativeHover = exports.surfaceWarning = exports.surfaceWarningActive = exports.surfaceWarningHover = exports.surfacePositive = exports.surfacePositiveActive = exports.surfacePositiveHover = exports.surfaceTransparentAccentActive = exports.surfaceTransparentAccentHover = exports.surfaceAccentGradient = exports.surfaceAccentGradientActive = exports.surfaceAccentGradientHover = exports.surfaceAccent = exports.surfaceAccentActive = exports.surfaceAccentHover = exports.surfaceClear = exports.surfaceClearActive = exports.surfaceClearHover = exports.surfaceTransparentCard = exports.surfaceTransparentCardActive = exports.surfaceTransparentCardHover = exports.surfaceTransparentDeep = exports.surfaceTransparentDeepActive = exports.surfaceTransparentDeepHover = exports.surfaceTransparentTertiary = exports.surfaceTransparentTertiaryActive = exports.surfaceTransparentTertiaryHover = exports.surfaceTransparentSecondary = exports.surfaceTransparentSecondaryActive = exports.surfaceTransparentSecondaryHover = exports.surfaceTransparentPrimary = exports.surfaceTransparentPrimaryActive = exports.surfaceTransparentPrimaryHover = exports.surfaceSolidDefault = exports.surfaceSolidDefaultActive = exports.surfaceSolidDefaultHover = exports.surfaceSolidCard = exports.surfaceSolidCardActive = exports.surfaceSolidCardHover = exports.surfaceSolidTertiary = exports.surfaceSolidTertiaryActive = void 0;
exports.onLightSurfaceSolidPrimaryActive = exports.onLightSurfaceSolidPrimaryHover = exports.onDarkSurfaceNegative = exports.onDarkSurfaceNegativeActive = exports.onDarkSurfaceNegativeHover = exports.onDarkSurfaceWarning = exports.onDarkSurfaceWarningActive = exports.onDarkSurfaceWarningHover = exports.onDarkSurfacePositive = exports.onDarkSurfacePositiveActive = exports.onDarkSurfacePositiveHover = exports.onDarkSurfaceAccentGradient = exports.onDarkSurfaceAccentGradientActive = exports.onDarkSurfaceAccentGradientHover = exports.onDarkSurfaceAccent = exports.onDarkSurfaceAccentActive = exports.onDarkSurfaceAccentHover = exports.onDarkSurfaceTransparentCard = exports.onDarkSurfaceTransparentCardActive = exports.onDarkSurfaceTransparentCardHover = exports.onDarkSurfaceTransparentDeep = exports.onDarkSurfaceTransparentDeepActive = exports.onDarkSurfaceTransparentDeepHover = exports.onDarkSurfaceTransparentTertiary = exports.onDarkSurfaceTransparentTertiaryActive = exports.onDarkSurfaceTransparentTertiaryHover = exports.onDarkSurfaceTransparentSecondary = exports.onDarkSurfaceTransparentSecondaryActive = exports.onDarkSurfaceTransparentSecondaryHover = exports.onDarkSurfaceTransparentPrimary = exports.onDarkSurfaceTransparentPrimaryActive = exports.onDarkSurfaceTransparentPrimaryHover = exports.onDarkSurfaceSolidDefault = exports.onDarkSurfaceSolidDefaultActive = exports.onDarkSurfaceSolidDefaultHover = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidCardActive = exports.onDarkSurfaceSolidCardHover = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidTertiaryActive = exports.onDarkSurfaceSolidTertiaryHover = exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidSecondaryActive = exports.onDarkSurfaceSolidSecondaryHover = exports.onDarkSurfaceSolidPrimary = exports.onDarkSurfaceSolidPrimaryActive = exports.onDarkSurfaceSolidPrimaryHover = exports.surfaceTransparentNegative = exports.surfaceTransparentWarning = exports.surfaceTransparentPositive = void 0;
exports.onDarkOverlaySoft = exports.overlayBlur = exports.overlayHard = exports.overlaySoft = exports.lightBackgroundPrimary = exports.darkBackgroundPrimary = exports.backgroundPrimary = exports.onLightSurfaceNegative = exports.onLightSurfaceNegativeActive = exports.onLightSurfaceNegativeHover = exports.onLightSurfaceWarning = exports.onLightSurfaceWarningActive = exports.onLightSurfaceWarningHover = exports.onLightSurfacePositive = exports.onLightSurfacePositiveActive = exports.onLightSurfacePositiveHover = exports.onLightSurfaceAccentGradient = exports.onLightSurfaceAccentGradientActive = exports.onLightSurfaceAccentGradientHover = exports.onLightSurfaceAccent = exports.onLightSurfaceAccentActive = exports.onLightSurfaceAccentHover = exports.onLightSurfaceTransparentCard = exports.onLightSurfaceTransparentCardActive = exports.onLightSurfaceTransparentCardHover = exports.onLightSurfaceTransparentDeep = exports.onLightSurfaceTransparentDeepActive = exports.onLightSurfaceTransparentDeepHover = exports.onLightSurfaceTransparentTertiary = exports.onLightSurfaceTransparentTertiaryActive = exports.onLightSurfaceTransparentTertiaryHover = exports.onLightSurfaceTransparentSecondary = exports.onLightSurfaceTransparentSecondaryActive = exports.onLightSurfaceTransparentSecondaryHover = exports.onLightSurfaceTransparentPrimary = exports.onLightSurfaceTransparentPrimaryActive = exports.onLightSurfaceTransparentPrimaryHover = exports.onLightSurfaceSolidDefault = exports.onLightSurfaceSolidDefaultActive = exports.onLightSurfaceSolidDefaultHover = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidCardActive = exports.onLightSurfaceSolidCardHover = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidTertiaryActive = exports.onLightSurfaceSolidTertiaryHover = exports.onLightSurfaceSolidSecondary = exports.onLightSurfaceSolidSecondaryActive = exports.onLightSurfaceSolidSecondaryHover = exports.onLightSurfaceSolidPrimary = void 0;
exports.skeletonGradientLighter = exports.skeletonGradient = exports.onLightOverlayBlur = exports.onLightOverlayHard = exports.onLightOverlaySoft = exports.onDarkOverlayBlur = exports.onDarkOverlayHard = void 0;
exports.textPrimaryHover = '#08080893';
exports.textPrimaryActive = '#080808C4';
exports.textPrimary = 'rgba(8, 8, 8, 0.96)';
exports.textSecondaryHover = '#080808FF';
exports.textSecondaryActive = '#080808AB';
exports.textSecondary = 'rgba(8, 8, 8, 0.56)';
exports.textTertiaryHover = '#080808FF';
exports.textTertiaryActive = '#08080856';
exports.textTertiary = 'rgba(8, 8, 8, 0.28)';
exports.textParagraphHover = '#0808087A';
exports.textParagraphActive = '#080808A3';
exports.textParagraph = 'rgba(8, 8, 8, 0.8)';
exports.textAccentHover = '#528DFAFF';
exports.textAccentActive = '#075AF2FF';
exports.textAccent = '#2A72F8';
exports.textAccentGradientHover = '#000000FF';
exports.textAccentGradientActive = '#CCCCCCFF';
exports.textAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
exports.textPositiveHover = '#1FC13DFF';
exports.textPositiveActive = '#147B27FF';
exports.textPositive = '#1A9E32';
exports.textWarningHover = '#FB782DFF';
exports.textWarningActive = '#D25004FF';
exports.textWarning = '#FA5F05';
exports.textNegativeHover = '#F54254FF';
exports.textNegativeActive = '#DA0B20FF';
exports.textNegative = '#F31B31';
exports.onDarkTextPrimaryHover = '#FFFFFF93';
exports.onDarkTextPrimaryActive = '#FFFFFFC4';
exports.onDarkTextPrimary = 'rgba(255, 255, 255, 0.96)';
exports.onDarkTextSecondaryHover = '#FFFFFFFF';
exports.onDarkTextSecondaryActive = '#FFFFFFAB';
exports.onDarkTextSecondary = 'rgba(255, 255, 255, 0.56)';
exports.onDarkTextTertiaryHover = '#FFFFFFFF';
exports.onDarkTextTertiaryActive = '#FFFFFF56';
exports.onDarkTextTertiary = 'rgba(255, 255, 255, 0.28)';
exports.onDarkTextParagraphHover = '#FFFFFF7A';
exports.onDarkTextParagraphActive = '#FFFFFFA3';
exports.onDarkTextParagraph = 'rgba(255, 255, 255, 0.8)';
exports.onDarkTextAccentHover = '#689CFDFF';
exports.onDarkTextAccentActive = '#1767FDFF';
exports.onDarkTextAccent = '#3F81FD';
exports.onDarkTextAccentGradientHover = '#000000FF';
exports.onDarkTextAccentGradientActive = '#CCCCCCFF';
exports.onDarkTextAccentGradient = 'linear-gradient(89.83deg, #5E94FF 0%, #43DBFA 100%)';
exports.onDarkTextPositiveHover = '#1FC13DFF';
exports.onDarkTextPositiveActive = '#147B27FF';
exports.onDarkTextPositive = '#1A9E32';
exports.onDarkTextWarningHover = '#FB782DFF';
exports.onDarkTextWarningActive = '#D25004FF';
exports.onDarkTextWarning = '#FA5F05';
exports.onDarkTextNegativeHover = '#FF5263FF';
exports.onDarkTextNegativeActive = '#FF001AFF';
exports.onDarkTextNegative = '#FF293E';
exports.onLightTextPrimaryHover = '#08080893';
exports.onLightTextPrimaryActive = '#080808C4';
exports.onLightTextPrimary = 'rgba(8, 8, 8, 0.96)';
exports.onLightTextSecondaryHover = '#080808FF';
exports.onLightTextSecondaryActive = '#080808AB';
exports.onLightTextSecondary = 'rgba(8, 8, 8, 0.56)';
exports.onLightTextTertiaryHover = '#080808FF';
exports.onLightTextTertiaryActive = '#08080856';
exports.onLightTextTertiary = 'rgba(8, 8, 8, 0.28)';
exports.onLightTextParagraphHover = '#0808087A';
exports.onLightTextParagraphActive = '#080808A3';
exports.onLightTextParagraph = 'rgba(8, 8, 8, 0.8)';
exports.onLightTextAccentHover = '#528DFAFF';
exports.onLightTextAccentActive = '#075AF2FF';
exports.onLightTextAccent = '#2A72F8';
exports.onLightTextAccentGradientHover = '#000000FF';
exports.onLightTextAccentGradientActive = '#CCCCCCFF';
exports.onLightTextAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
exports.onLightTextPositiveHover = '#1FC13DFF';
exports.onLightTextPositiveActive = '#147B27FF';
exports.onLightTextPositive = '#1A9E32';
exports.onLightTextWarningHover = '#FB782DFF';
exports.onLightTextWarningActive = '#D25004FF';
exports.onLightTextWarning = '#FA5F05';
exports.onLightTextNegativeHover = '#F54254FF';
exports.onLightTextNegativeActive = '#DA0B20FF';
exports.onLightTextNegative = '#F31B31';
exports.inverseTextPrimaryHover = '#FFFFFF93';
exports.inverseTextPrimaryActive = '#FFFFFFC4';
exports.inverseTextPrimary = 'rgba(255, 255, 255, 0.96)';
exports.inverseTextSecondaryHover = '#FFFFFFFF';
exports.inverseTextSecondaryActive = '#FFFFFFAB';
exports.inverseTextSecondary = 'rgba(255, 255, 255, 0.56)';
exports.inverseTextTertiaryHover = '#FFFFFFFF';
exports.inverseTextTertiaryActive = '#FFFFFF56';
exports.inverseTextTertiary = 'rgba(255, 255, 255, 0.28)';
exports.inverseTextParagraphHover = '#FFFFFF7A';
exports.inverseTextParagraphActive = '#FFFFFFA3';
exports.inverseTextParagraph = 'rgba(255, 255, 255, 0.8)';
exports.surfaceSolidPrimaryHover = '#FAFAFAFF';
exports.surfaceSolidPrimaryActive = '#F0F0F0FF';
exports.surfaceSolidPrimary = '#F5F5F5';
exports.surfaceSolidSecondaryHover = '#F7F7F7FF';
exports.surfaceSolidSecondaryActive = '#E8E8E8FF';
exports.surfaceSolidSecondary = '#ECECEC';
exports.surfaceSolidTertiaryHover = '#EDEDEDFF';
exports.surfaceSolidTertiaryActive = '#D4D4D4FF';
exports.surfaceSolidTertiary = '#DDDDDD';
exports.surfaceSolidCardHover = '#FFFFFFFF';
exports.surfaceSolidCardActive = '#FFFFFFFF';
exports.surfaceSolidCard = 'rgba(255, 255, 255, 1)';
exports.surfaceSolidDefaultHover = '#262626FF';
exports.surfaceSolidDefaultActive = '#030303FF';
exports.surfaceSolidDefault = '#080808';
exports.surfaceTransparentPrimaryHover = '#08080803';
exports.surfaceTransparentPrimaryActive = '#0808080D';
exports.surfaceTransparentPrimary = 'rgba(8, 8, 8, 0.03)';
exports.surfaceTransparentSecondaryHover = '#08080805';
exports.surfaceTransparentSecondaryActive = '#0808081A';
exports.surfaceTransparentSecondary = 'rgba(8, 8, 8, 0.06)';
exports.surfaceTransparentTertiaryHover = '#0808080A';
exports.surfaceTransparentTertiaryActive = '#08080829';
exports.surfaceTransparentTertiary = 'rgba(8, 8, 8, 0.12)';
exports.surfaceTransparentDeepHover = '#0808088F';
exports.surfaceTransparentDeepActive = '#080808AD';
exports.surfaceTransparentDeep = 'rgba(8, 8, 8, 0.64)';
exports.surfaceTransparentCardHover = '#FFFFFFFF';
exports.surfaceTransparentCardActive = '#FFFFFFFF';
exports.surfaceTransparentCard = 'rgba(255, 255, 255, 1)';
exports.surfaceClearHover = '#FFFFFF00';
exports.surfaceClearActive = '#FFFFFF00';
exports.surfaceClear = 'rgba(255, 255, 255, 0)';
exports.surfaceAccentHover = '#528DFAFF';
exports.surfaceAccentActive = '#1665F8FF';
exports.surfaceAccent = '#2A72F8';
exports.surfaceAccentGradientHover = '#FFFFFFFF';
exports.surfaceAccentGradientActive = '#FFFFFFFF';
exports.surfaceAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
exports.surfaceTransparentAccentHover = '#2A72F805';
exports.surfaceTransparentAccentActive = '#2A72F81A';
exports.surfacePositiveHover = '#1EB83AFF';
exports.surfacePositiveActive = '#178C2CFF';
exports.surfacePositive = '#1A9E32';
exports.surfaceWarningHover = '#FB782DFF';
exports.surfaceWarningActive = '#E65705FF';
exports.surfaceWarning = '#FA5F05';
exports.surfaceNegativeHover = '#F54254FF';
exports.surfaceNegativeActive = '#ED0C23FF';
exports.surfaceNegative = '#F31B31';
exports.surfaceTransparentPositiveHover = '#1A9E3205';
exports.surfaceTransparentPositiveActive = '#1A9E321A';
exports.surfaceTransparentWarningHover = '#FA5F0505';
exports.surfaceTransparentWarningActive = '#FA5F051A';
exports.surfaceTransparentNegativeHover = '#F31B3105';
exports.surfaceTransparentNegativeActive = '#F31B311A';
exports.surfaceTransparentAccent = 'rgba(42,114,248,0.06)';
exports.surfaceTransparentPositive = 'rgba(26,158,50,0.06)';
exports.surfaceTransparentWarning = 'rgba(250,95,5,0.06)';
exports.surfaceTransparentNegative = 'rgba(243,27,49,0.06)';
exports.onDarkSurfaceSolidPrimaryHover = '#363636FF';
exports.onDarkSurfaceSolidPrimaryActive = '#0D0D0DFF';
exports.onDarkSurfaceSolidPrimary = '#171717';
exports.onDarkSurfaceSolidSecondaryHover = '#3B3B3BFF';
exports.onDarkSurfaceSolidSecondaryActive = '#1C1C1CFF';
exports.onDarkSurfaceSolidSecondary = '#262626';
exports.onDarkSurfaceSolidTertiaryHover = '#4A4A4AFF';
exports.onDarkSurfaceSolidTertiaryActive = '#2B2B2BFF';
exports.onDarkSurfaceSolidTertiary = '#363636';
exports.onDarkSurfaceSolidCardHover = '#363636FF';
exports.onDarkSurfaceSolidCardActive = '#0D0D0DFF';
exports.onDarkSurfaceSolidCard = '#171717';
exports.onDarkSurfaceSolidDefaultHover = '#FFFFFFFF';
exports.onDarkSurfaceSolidDefaultActive = '#FFFFFFFF';
exports.onDarkSurfaceSolidDefault = 'rgba(255, 255, 255, 1)';
exports.onDarkSurfaceTransparentPrimaryHover = '#363636FF';
exports.onDarkSurfaceTransparentPrimaryActive = '#0D0D0DFF';
exports.onDarkSurfaceTransparentPrimary = '#171717';
exports.onDarkSurfaceTransparentSecondaryHover = '#3B3B3BFF';
exports.onDarkSurfaceTransparentSecondaryActive = '#1C1C1CFF';
exports.onDarkSurfaceTransparentSecondary = '#262626';
exports.onDarkSurfaceTransparentTertiaryHover = '#4A4A4AFF';
exports.onDarkSurfaceTransparentTertiaryActive = '#2B2B2BFF';
exports.onDarkSurfaceTransparentTertiary = '#363636';
exports.onDarkSurfaceTransparentDeepHover = '#FFFFFF8F';
exports.onDarkSurfaceTransparentDeepActive = '#FFFFFFAD';
exports.onDarkSurfaceTransparentDeep = 'rgba(255, 255, 255, 0.64)';
exports.onDarkSurfaceTransparentCardHover = '#FFFFFF0A';
exports.onDarkSurfaceTransparentCardActive = '#FFFFFF29';
exports.onDarkSurfaceTransparentCard = 'rgba(255, 255, 255, 0.12)';
exports.onDarkSurfaceAccentHover = '#689CFDFF';
exports.onDarkSurfaceAccentActive = '#2B74FDFF';
exports.onDarkSurfaceAccent = '#3F81FD';
exports.onDarkSurfaceAccentGradientHover = '#FFFFFFFF';
exports.onDarkSurfaceAccentGradientActive = '#FFFFFFFF';
exports.onDarkSurfaceAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
exports.onDarkSurfacePositiveHover = '#1EB83AFF';
exports.onDarkSurfacePositiveActive = '#178C2CFF';
exports.onDarkSurfacePositive = '#1A9E32';
exports.onDarkSurfaceWarningHover = '#FB782DFF';
exports.onDarkSurfaceWarningActive = '#E65705FF';
exports.onDarkSurfaceWarning = '#FA5F05';
exports.onDarkSurfaceNegativeHover = '#FF5263FF';
exports.onDarkSurfaceNegativeActive = '#FF142CFF';
exports.onDarkSurfaceNegative = '#FF293E';
exports.onLightSurfaceSolidPrimaryHover = '#FAFAFAFF';
exports.onLightSurfaceSolidPrimaryActive = '#F0F0F0FF';
exports.onLightSurfaceSolidPrimary = '#F5F5F5';
exports.onLightSurfaceSolidSecondaryHover = '#F7F7F7FF';
exports.onLightSurfaceSolidSecondaryActive = '#E8E8E8FF';
exports.onLightSurfaceSolidSecondary = '#ECECEC';
exports.onLightSurfaceSolidTertiaryHover = '#EDEDEDFF';
exports.onLightSurfaceSolidTertiaryActive = '#D4D4D4FF';
exports.onLightSurfaceSolidTertiary = '#DDDDDD';
exports.onLightSurfaceSolidCardHover = '#FFFFFFFF';
exports.onLightSurfaceSolidCardActive = '#FFFFFFFF';
exports.onLightSurfaceSolidCard = 'rgba(255, 255, 255, 1)';
exports.onLightSurfaceSolidDefaultHover = '#262626FF';
exports.onLightSurfaceSolidDefaultActive = '#030303FF';
exports.onLightSurfaceSolidDefault = '#080808';
exports.onLightSurfaceTransparentPrimaryHover = '#08080803';
exports.onLightSurfaceTransparentPrimaryActive = '#0808080D';
exports.onLightSurfaceTransparentPrimary = 'rgba(8, 8, 8, 0.03)';
exports.onLightSurfaceTransparentSecondaryHover = '#08080805';
exports.onLightSurfaceTransparentSecondaryActive = '#0808081A';
exports.onLightSurfaceTransparentSecondary = 'rgba(8, 8, 8, 0.06)';
exports.onLightSurfaceTransparentTertiaryHover = '#0808080A';
exports.onLightSurfaceTransparentTertiaryActive = '#08080829';
exports.onLightSurfaceTransparentTertiary = 'rgba(8, 8, 8, 0.12)';
exports.onLightSurfaceTransparentDeepHover = '#0808088F';
exports.onLightSurfaceTransparentDeepActive = '#080808AD';
exports.onLightSurfaceTransparentDeep = 'rgba(8, 8, 8, 0.64)';
exports.onLightSurfaceTransparentCardHover = '#FFFFFFFF';
exports.onLightSurfaceTransparentCardActive = '#FFFFFFFF';
exports.onLightSurfaceTransparentCard = 'rgba(255, 255, 255, 1)';
exports.onLightSurfaceAccentHover = '#528DFAFF';
exports.onLightSurfaceAccentActive = '#1665F8FF';
exports.onLightSurfaceAccent = '#2A72F8';
exports.onLightSurfaceAccentGradientHover = '#FFFFFFFF';
exports.onLightSurfaceAccentGradientActive = '#FFFFFFFF';
exports.onLightSurfaceAccentGradient = 'linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%)';
exports.onLightSurfacePositiveHover = '#1EB83AFF';
exports.onLightSurfacePositiveActive = '#178C2CFF';
exports.onLightSurfacePositive = '#1A9E32';
exports.onLightSurfaceWarningHover = '#FB782DFF';
exports.onLightSurfaceWarningActive = '#E65705FF';
exports.onLightSurfaceWarning = '#FA5F05';
exports.onLightSurfaceNegativeHover = '#F54254FF';
exports.onLightSurfaceNegativeActive = '#ED0C23FF';
exports.onLightSurfaceNegative = '#F31B31';
exports.backgroundPrimary = '#F9F9F9';
exports.darkBackgroundPrimary = '#080808';
exports.lightBackgroundPrimary = '#F9F9F9';
exports.overlaySoft = 'rgba(249, 249, 249, 0.56)';
exports.overlayHard = 'rgba(249, 249, 249, 0.96)';
exports.overlayBlur = 'rgba(249, 249, 249, 0.28)';
exports.onDarkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.onDarkOverlayHard = 'rgba(8, 8, 8, 0.96)';
exports.onDarkOverlayBlur = 'rgba(8, 8, 8, 0.28)';
exports.onLightOverlaySoft = 'rgba(249, 249, 249, 0.56)';
exports.onLightOverlayHard = 'rgba(249, 249, 249, 0.96)';
exports.onLightOverlayBlur = 'rgba(249, 249, 249, 0.28)';
exports.skeletonGradient = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.09) 0%, rgba(8, 8, 8, 0.08) 6.25%, rgba(8, 8, 8, 0.05) 12.5%, rgba(8, 8, 8, 0.01) 25%, rgba(8, 8, 8, 0.05) 37.5%, rgba(8, 8, 8, 0.08) 43.75%, rgba(8, 8, 8, 0.09) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.05) 62.5%, rgba(8, 8, 8, 0.01) 75%, rgba(8, 8, 8, 0.05) 87.5%, rgba(8, 8, 8, 0.08) 93.75%, rgba(8, 8, 8, 0.09) 100% )';
exports.skeletonGradientLighter = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.36) 0%, rgba(8, 8, 8, 0.32) 6.25%, rgba(8, 8, 8, 0.20) 12.5%, rgba(8, 8, 8, 0.04) 25%, rgba(8, 8, 8, 0.20) 37.5%, rgba(8, 8, 8, 0.32) 43.75%, rgba(8, 8, 8, 0.36) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.20) 62.5%, rgba(8, 8, 8, 0.04) 75%, rgba(8, 8, 8, 0.20) 87.5%, rgba(8, 8, 8, 0.32) 93.75%, rgba(8, 8, 8, 0.36) 100% )';
