// @salutejs/plasma-new-hope/cjs/components/Counter/Counter.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var Counter_styles = require('./Counter.styles.js');

var _excluded = ["count", "maxCount", "size", "view"];
var counterRoot = function counterRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (props, ref) {
    var count = props.count,
      maxCount = props.maxCount,
      size = props.size,
      view = props.view,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(props, _excluded);
    var content = maxCount && count > maxCount ? "".concat(maxCount, "+") : count;
    return /*#__PURE__*/React.createElement(Root, _rollupPluginBabelHelpers.extends({
      ref: ref,
      view: view,
      size: size,
      count: count
    }, rest), content);
  });
};
var counterConfig = {
  name: 'Counter',
  tag: 'div',
  layout: counterRoot,
  base: Counter_styles.base,
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'xs'
  }
};

exports.counterConfig = counterConfig;
exports.counterRoot = counterRoot;
