// @salutejs/plasma-new-hope/cjs/components/Link/Link.js
import './Link_1oiwct5.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var base$1 = require('./_view/base.js');
var base$2 = require('./_disabled/base.js');
var base$3 = require('./_focused/base.js');

var _excluded = ["children", "disabled", "target", "href", "rel"];
var base = "b1j91s21";
var linkRoot = function linkRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (props, ref) {
    var children = props.children,
      disabled = props.disabled,
      target = props.target,
      href = props.href,
      rel = props.rel,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(props, _excluded);
    var forwardProps = {
      disabled: disabled,
      target: target,
      href: disabled ? undefined : href,
      // NOTE: https://developer.chrome.com/docs/lighthouse/best-practices/external-anchors-use-rel-noopener/
      rel: target === '_blank' ? 'noopener noreferrer' : rel
    };
    return /*#__PURE__*/React.createElement(Root, _rollupPluginBabelHelpers.extends({
      ref: ref
    }, forwardProps, rest), children);
  });
};
var linkConfig = {
  name: 'Link',
  tag: 'a',
  layout: linkRoot,
  base: base,
  variations: {
    view: {
      css: base$1.base
    },
    disabled: {
      css: base$2.base,
      attrs: true
    },
    focused: {
      css: base$3.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

exports.linkConfig = linkConfig;
exports.linkRoot = linkRoot;
