// @salutejs/plasma-new-hope/styled-components/es/examples/plasma_b2c/components/Select/Select.js
import { selectConfig, selectItemConfig, selectDividerConfig, selectFooterConfig, selectGroupConfig, selectHeaderConfig } from '../../../../components/Select';
import { component, mergeConfig } from '../../../../engines';
import { config } from './Select.config';
var mergedConfig = /*#__PURE__*/mergeConfig(selectConfig, config);
export var Select = /*#__PURE__*/component(mergedConfig);
var mergedItemConfig = /*#__PURE__*/mergeConfig(selectItemConfig);
export var SelectItem = /*#__PURE__*/component(mergedItemConfig);
var mergedDividerConfig = /*#__PURE__*/mergeConfig(selectDividerConfig);
export var SelectDivider = /*#__PURE__*/component(mergedDividerConfig);
var mergedFooterConfig = /*#__PURE__*/mergeConfig(selectFooterConfig);
export var SelectFooter = /*#__PURE__*/component(mergedFooterConfig);
var mergedGroupConfig = /*#__PURE__*/mergeConfig(selectGroupConfig);
export var SelectGroup = /*#__PURE__*/component(mergedGroupConfig);
var mergedHeaderConfig = /*#__PURE__*/mergeConfig(selectHeaderConfig);
export var SelectHeader = /*#__PURE__*/component(mergedHeaderConfig);
