// @salutejs/plasma-new-hope/styled-components/es/components/Slider/ui/Thumb/Thumb.styles.js
import styled from 'styled-components';
import { addFocus } from '../../../../mixins';
import { tokens } from '../../Slider.tokens';
export var ThumbBase = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-13botku-0"
})(["width:var(", ");height:var(", ");position:relative;left:-0.125rem;top:-0.125rem;border-radius:50%;box-sizing:border-box;background:var(", ");margin:0.125rem;transition:border-color 0.1s ease-in-out;&:after{background:var(", ");margin:-0.125rem;content:'';position:absolute;inset:0;z-index:-1;border-radius:inherit;}&:not([disabled]):hover:after,&:not([disabled]):active:after{background:var(", ");}&[disabled]{cursor:not-allowed;}&:focus{outline:none;}", ""], tokens.thumbSize, tokens.thumbSize, tokens.thumbBackgroundColor, tokens.thumbBorderColor, tokens.thumbFocusBorderColor, /*#__PURE__*/addFocus({
  outlineOffset: '0.125rem',
  outlineSize: '0.125rem',
  outlineRadius: '50%',
  outlineColor: /*#__PURE__*/"var(".concat(tokens.thumbFocusBorderColor, ")")
}));
