// @salutejs/plasma-new-hope/styled-components/cjs/components/Tabs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "TabItemRefs", {
  enumerable: true,
  get: function get() {
    return _TabsContext.TabItemRefs;
  }
});
Object.defineProperty(exports, "TabsContext", {
  enumerable: true,
  get: function get() {
    return _TabsContext.TabsContext;
  }
});
Object.defineProperty(exports, "createTabsController", {
  enumerable: true,
  get: function get() {
    return _createTabsController.createTabsController;
  }
});
Object.defineProperty(exports, "tabItemConfig", {
  enumerable: true,
  get: function get() {
    return _TabItem.tabItemConfig;
  }
});
Object.defineProperty(exports, "tabItemRoot", {
  enumerable: true,
  get: function get() {
    return _TabItem.tabItemRoot;
  }
});
Object.defineProperty(exports, "tabsConfig", {
  enumerable: true,
  get: function get() {
    return _Tabs.tabsConfig;
  }
});
Object.defineProperty(exports, "tabsRoot", {
  enumerable: true,
  get: function get() {
    return _Tabs.tabsRoot;
  }
});
Object.defineProperty(exports, "tabsTokens", {
  enumerable: true,
  get: function get() {
    return _tokens.tokens;
  }
});
var _Tabs = /*#__PURE__*/require("./ui/Tabs/Tabs");
var _TabItem = /*#__PURE__*/require("./ui/TabItem/TabItem");
var _tokens = /*#__PURE__*/require("./tokens");
var _TabsContext = /*#__PURE__*/require("./TabsContext");
var _createTabsController = /*#__PURE__*/require("./createTabsController");
