// @salutejs/plasma-new-hope/cjs/components/Dropdown/ui/DropdownFooter/DropdownFooter.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var plasmaCore = require('@salutejs/plasma-core');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var DropdownFooter_styles = require('./DropdownFooter.styles.js');

var _excluded = ["id", "className", "children", "view", "size"];
/**
 * Элемент выпадающего списка
 */
var dropdownFooterRoot = function dropdownFooterRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (_ref, outerRootRef) {
    var id = _ref.id,
      className = _ref.className,
      children = _ref.children,
      view = _ref.view,
      size = _ref.size,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
    var uniqId = plasmaCore.useUniqId();
    var innerId = id || uniqId;
    return /*#__PURE__*/React.createElement(Root, {
      id: innerId,
      ref: outerRootRef,
      view: view,
      size: size
    }, /*#__PURE__*/React.createElement(DropdownFooter_styles.StyledDropdownFooter, _rollupPluginBabelHelpers.extends({
      className: className
    }, rest), children));
  });
};
var dropdownFooterConfig = {
  name: 'DropdownFooter',
  tag: 'div',
  layout: dropdownFooterRoot,
  base: '',
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

exports.dropdownFooterConfig = dropdownFooterConfig;
exports.dropdownFooterRoot = dropdownFooterRoot;
