// @salutejs/plasma-web/css/cjs/components/Button/Button.config.js
import './Button.config_d2euf6.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var config = {
  defaults: {
    view: 'default',
    focused: 'true',
    size: 'm'
  },
  variations: {
    view: {
      "default": "d1fnkof5",
      /**
       * @deprecated
       * использовать `default`
       */
      primary: "p1129msw",
      accent: "abaqjet",
      secondary: "smxfxlw",
      clear: "cleczrv",
      success: "slekduy",
      warning: "w1yfa47j",
      critical: "csb1lec",
      dark: "d123wpoi",
      black: "b1w797f8",
      white: "w1toqou7"
    },
    size: {
      l: "ly229gi",
      lr: "l1joiaye",
      m: "m1i1eza5",
      mr: "m1r1gaby",
      s: "sqhwg07",
      sr: "s1bf1tsb",
      xs: "x1ca3q6l",
      xsr: "xrk6508",
      xxs: "x2s1r71"
    },
    disabled: {
      "true": "tf053fi"
    },
    focused: {
      "true": "t17x5rnm"
    },
    stretching: {
      auto: "appmfmz",
      filled: "faa8u42",
      fixed: "f1y67d5i"
    }
  }
};

exports.config = config;
