// @salutejs/plasma-new-hope/cjs/components/Select/ui/SelectGroup/SelectGroup.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('react');
require('../../../../utils/react.js');
require('@salutejs/plasma-core');
require('@linaria/react');
var DropdownGroup = require('../../../Dropdown/ui/DropdownGroup/DropdownGroup.js');
require('@linaria/core');
require('focus-visible');
require('react-dom');
require('react-popper');
require('styled-components');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var SelectGroup_styles = require('./SelectGroup.styles.js');

var selectGroupRoot = DropdownGroup.dropdownGroupRoot;
var selectGroupConfig = {
  name: 'SelectGroup',
  tag: 'div',
  layout: selectGroupRoot,
  base: SelectGroup_styles.base,
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

exports.selectGroupConfig = selectGroupConfig;
exports.selectGroupRoot = selectGroupRoot;
