// @salutejs/plasma-new-hope/styled-components/cjs/components/Slider/ui/Thumb/Thumb.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ThumbBase = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _mixins = /*#__PURE__*/require("../../../../mixins");
var _Slider = /*#__PURE__*/require("../../Slider.tokens");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var ThumbBase = exports.ThumbBase = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-13botku-0"
})(["width:var(", ");height:var(", ");position:relative;left:-0.125rem;top:-0.125rem;border-radius:50%;box-sizing:border-box;background:var(", ");margin:0.125rem;transition:border-color 0.1s ease-in-out;&:after{background:var(", ");margin:-0.125rem;content:'';position:absolute;inset:0;z-index:-1;border-radius:inherit;}&:not([disabled]):hover:after,&:not([disabled]):active:after{background:var(", ");}&[disabled]{cursor:not-allowed;}&:focus{outline:none;}", ""], _Slider.tokens.thumbSize, _Slider.tokens.thumbSize, _Slider.tokens.thumbBackgroundColor, _Slider.tokens.thumbBorderColor, _Slider.tokens.thumbFocusBorderColor, /*#__PURE__*/(0, _mixins.addFocus)({
  outlineOffset: '0.125rem',
  outlineSize: '0.125rem',
  outlineRadius: '50%',
  outlineColor: /*#__PURE__*/"var(".concat(_Slider.tokens.thumbFocusBorderColor, ")")
}));
