// @salutejs/plasma-new-hope/es/components/Dropdown/ui/DropdownGroup/DropdownGroup.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef } from 'react';
import { useUniqId } from '@salutejs/plasma-core';
import { base } from './variations/_view/base.js';
import { base as base$1 } from './variations/_size/base.js';
import { StyledDropdownLabel, StyledDropdownGroup } from './DropdownGroup.styles.js';

var _excluded = ["id", "labelClassName", "groupClassName", "label", "children", "role", "view", "size"];
/**
 * Элемент группы выпадающего списка
 */
var dropdownGroupRoot = function dropdownGroupRoot(Root) {
  return /*#__PURE__*/forwardRef(function (_ref, outerRootRef) {
    var id = _ref.id,
      labelClassName = _ref.labelClassName,
      groupClassName = _ref.groupClassName,
      label = _ref.label,
      children = _ref.children,
      role = _ref.role,
      view = _ref.view,
      size = _ref.size,
      rest = _objectWithoutProperties(_ref, _excluded);
    var uniqId = useUniqId();
    var innerId = id || uniqId;
    return /*#__PURE__*/React.createElement(Root, {
      id: innerId,
      ref: outerRootRef,
      view: view,
      size: size,
      role: role
    }, /*#__PURE__*/React.createElement(StyledDropdownLabel, {
      className: labelClassName
    }, label), /*#__PURE__*/React.createElement(StyledDropdownGroup, _extends({
      className: groupClassName
    }, rest), children));
  });
};
var dropdownGroupConfig = {
  name: 'DropdownGroup',
  tag: 'div',
  layout: dropdownGroupRoot,
  base: '',
  variations: {
    view: {
      css: base
    },
    size: {
      css: base$1
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

export { dropdownGroupConfig, dropdownGroupRoot };
