// @salutejs/plasma-web/tokens/typography.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "body1", {
  enumerable: true,
  get: function get() {
    return _styledComponents.body1;
  }
});
Object.defineProperty(exports, "body2", {
  enumerable: true,
  get: function get() {
    return _styledComponents.body2;
  }
});
Object.defineProperty(exports, "body3", {
  enumerable: true,
  get: function get() {
    return _styledComponents.body3;
  }
});
Object.defineProperty(exports, "bodyL", {
  enumerable: true,
  get: function get() {
    return _styledComponents.bodyL;
  }
});
Object.defineProperty(exports, "bodyLBold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.bodyLBold;
  }
});
Object.defineProperty(exports, "bodyM", {
  enumerable: true,
  get: function get() {
    return _styledComponents.bodyM;
  }
});
Object.defineProperty(exports, "bodyMBold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.bodyMBold;
  }
});
Object.defineProperty(exports, "bodyS", {
  enumerable: true,
  get: function get() {
    return _styledComponents.bodyS;
  }
});
Object.defineProperty(exports, "bodySBold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.bodySBold;
  }
});
Object.defineProperty(exports, "bodyXS", {
  enumerable: true,
  get: function get() {
    return _styledComponents.bodyXS;
  }
});
Object.defineProperty(exports, "bodyXSBold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.bodyXSBold;
  }
});
Object.defineProperty(exports, "bodyXXS", {
  enumerable: true,
  get: function get() {
    return _styledComponents.bodyXXS;
  }
});
Object.defineProperty(exports, "bodyXXSBold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.bodyXXSBold;
  }
});
Object.defineProperty(exports, "button1", {
  enumerable: true,
  get: function get() {
    return _styledComponents.button1;
  }
});
Object.defineProperty(exports, "button2", {
  enumerable: true,
  get: function get() {
    return _styledComponents.button2;
  }
});
Object.defineProperty(exports, "caption", {
  enumerable: true,
  get: function get() {
    return _styledComponents.caption;
  }
});
Object.defineProperty(exports, "display1", {
  enumerable: true,
  get: function get() {
    return _styledComponents.display1;
  }
});
Object.defineProperty(exports, "display2", {
  enumerable: true,
  get: function get() {
    return _styledComponents.display2;
  }
});
Object.defineProperty(exports, "display3", {
  enumerable: true,
  get: function get() {
    return _styledComponents.display3;
  }
});
Object.defineProperty(exports, "dsplL", {
  enumerable: true,
  get: function get() {
    return _styledComponents.dsplL;
  }
});
Object.defineProperty(exports, "dsplLBold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.dsplLBold;
  }
});
Object.defineProperty(exports, "dsplM", {
  enumerable: true,
  get: function get() {
    return _styledComponents.dsplM;
  }
});
Object.defineProperty(exports, "dsplMBold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.dsplMBold;
  }
});
Object.defineProperty(exports, "dsplS", {
  enumerable: true,
  get: function get() {
    return _styledComponents.dsplS;
  }
});
Object.defineProperty(exports, "dsplSBold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.dsplSBold;
  }
});
exports.fieldTypo = exports.fieldHelperTextTypo = void 0;
Object.defineProperty(exports, "footnote1", {
  enumerable: true,
  get: function get() {
    return _styledComponents.footnote1;
  }
});
Object.defineProperty(exports, "footnote2", {
  enumerable: true,
  get: function get() {
    return _styledComponents.footnote2;
  }
});
Object.defineProperty(exports, "h1", {
  enumerable: true,
  get: function get() {
    return _styledComponents.h1;
  }
});
Object.defineProperty(exports, "h1Bold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.h1Bold;
  }
});
Object.defineProperty(exports, "h2", {
  enumerable: true,
  get: function get() {
    return _styledComponents.h2;
  }
});
Object.defineProperty(exports, "h2Bold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.h2Bold;
  }
});
Object.defineProperty(exports, "h3", {
  enumerable: true,
  get: function get() {
    return _styledComponents.h3;
  }
});
Object.defineProperty(exports, "h3Bold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.h3Bold;
  }
});
Object.defineProperty(exports, "h4", {
  enumerable: true,
  get: function get() {
    return _styledComponents.h4;
  }
});
Object.defineProperty(exports, "h4Bold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.h4Bold;
  }
});
Object.defineProperty(exports, "h5", {
  enumerable: true,
  get: function get() {
    return _styledComponents.h5;
  }
});
Object.defineProperty(exports, "h5Bold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.h5Bold;
  }
});
Object.defineProperty(exports, "headline1", {
  enumerable: true,
  get: function get() {
    return _styledComponents.headline1;
  }
});
Object.defineProperty(exports, "headline2", {
  enumerable: true,
  get: function get() {
    return _styledComponents.headline2;
  }
});
Object.defineProperty(exports, "headline3", {
  enumerable: true,
  get: function get() {
    return _styledComponents.headline3;
  }
});
Object.defineProperty(exports, "headline4", {
  enumerable: true,
  get: function get() {
    return _styledComponents.headline4;
  }
});
exports.inputTypo = exports.inputLabelTypo = exports.headline5 = void 0;
Object.defineProperty(exports, "paragraph1", {
  enumerable: true,
  get: function get() {
    return _styledComponents.paragraph1;
  }
});
Object.defineProperty(exports, "paragraph2", {
  enumerable: true,
  get: function get() {
    return _styledComponents.paragraph2;
  }
});
exports.subtitle = void 0;
Object.defineProperty(exports, "textL", {
  enumerable: true,
  get: function get() {
    return _styledComponents.textL;
  }
});
Object.defineProperty(exports, "textLBold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.textLBold;
  }
});
Object.defineProperty(exports, "textM", {
  enumerable: true,
  get: function get() {
    return _styledComponents.textM;
  }
});
Object.defineProperty(exports, "textMBold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.textMBold;
  }
});
Object.defineProperty(exports, "textS", {
  enumerable: true,
  get: function get() {
    return _styledComponents.textS;
  }
});
Object.defineProperty(exports, "textSBold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.textSBold;
  }
});
Object.defineProperty(exports, "textXS", {
  enumerable: true,
  get: function get() {
    return _styledComponents.textXS;
  }
});
Object.defineProperty(exports, "textXSBold", {
  enumerable: true,
  get: function get() {
    return _styledComponents.textXSBold;
  }
});
Object.defineProperty(exports, "underline", {
  enumerable: true,
  get: function get() {
    return _styledComponents.underline;
  }
});
var _styledComponents = /*#__PURE__*/require("@salutejs/plasma-new-hope/styled-components");
var fieldTypo = exports.fieldTypo = {
  fontFamily: 'var(--plasma-field-font-family)',
  fontSize: 'var(--plasma-field-font-size)',
  fontStyle: 'var(--plasma-field-font-style)',
  fontWeight: 'var(--plasma-field-font-weight)',
  letterSpacing: 'var(--plasma-field-letter-spacing)',
  lineHeight: 'var(--plasma-field-line-height)'
};
var fieldHelperTextTypo = exports.fieldHelperTextTypo = {
  fontFamily: 'var(--plasma-field-helper-text-font-family)',
  fontSize: 'var(--plasma-field-helper-text-font-size)',
  fontStyle: 'var(--plasma-field-helper-text-font-style)',
  fontWeight: 'var(--plasma-field-helper-text-font-weight)',
  letterSpacing: 'var(--plasma-field-helper-text-letter-spacing)',
  lineHeight: 'var(--plasma-field-helper-text-line-height)'
};
var headline5 = exports.headline5 = {
  fontFamily: "var(--plasma-typo-headline5-font-family, 'SB Sans Text',sans-serif)",
  fontWeight: 'var(--plasma-typo-headline5-font-weight, 600)',
  fontStyle: 'var(--plasma-typo-headline5-font-style, normal)',
  fontSize: 'var(--plasma-typo-headline5-font-size, 1rem)',
  letterSpacing: 'var(--plasma-typo-headline5-letter-spacing, -0.0190em)',
  lineHeight: 'var(--plasma-typo-headline5-line-height, 1.25rem)'
};
var inputTypo = exports.inputTypo = {
  fontFamily: 'var(--plasma-input-font-family)',
  fontSize: 'var(--plasma-input-font-size)',
  fontStyle: 'var(--plasma-input-font-style)',
  fontWeight: 'var(--plasma-input-font-weight)',
  letterSpacing: 'var(--plasma-input-letter-spacing)',
  lineHeight: 'var(--plasma-input-line-height)'
};
var inputLabelTypo = exports.inputLabelTypo = {
  fontFamily: 'var(--plasma-input-label-font-family)',
  fontSize: 'var(--plasma-input-label-font-size)',
  fontStyle: 'var(--plasma-input-label-font-style)',
  fontWeight: 'var(--plasma-input-label-font-weight)',
  letterSpacing: 'var(--plasma-input-label-letter-spacing)',
  lineHeight: 'var(--plasma-input-label-line-height)'
};
var subtitle = exports.subtitle = {
  fontFamily: "var(--plasma-typo-subtitle-font-family, 'SB Sans Text',sans-serif)",
  fontWeight: 'var(--plasma-typo-subtitle-font-weight, 400)',
  fontStyle: 'var(--plasma-typo-subtitle-font-style, normal)',
  fontSize: 'var(--plasma-typo-subtitle-font-size, 1.25rem)',
  letterSpacing: 'var(--plasma-typo-subtitle-letter-spacing, -0.0190em)',
  lineHeight: 'var(--plasma-typo-subtitle-line-height, 1.5rem)'
};
