// @salutejs/plasma-ui/components/TextArea/TextArea.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _helpers = /*#__PURE__*/require("../../helpers");
var _ = /*#__PURE__*/require(".");
var _IconPlaceholder;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var func = function func() {};
var defaultProps = {
  id: 'example-textarea',
  placeholder: 'Заполните многострочное поле',
  helperText: 'Подсказка к полю',
  enableRightIcon: true,
  resize: 'vertical',
  disabled: false,
  readOnly: false
};
var status = '';
var Default = exports.Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(_.TextArea, _extends({
    value: "\u0417\u043D\u0430\u0447\u0435\u043D\u0438\u0435 \u043F\u043E\u043B\u044F",
    contentRight: _IconPlaceholder || (_IconPlaceholder = /*#__PURE__*/_react["default"].createElement(_helpers.IconPlaceholder, null)),
    status: status,
    onChange: func,
    onFocus: func,
    onBlur: func
  }, defaultProps));
};
