// @salutejs/plasma-ui/utils/deviceDetection.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isTV = exports.isSberPortal = exports.isSberBoxTop = exports.isSberBox = exports.deviceScales = exports.detectDevice = exports.DeviceKindList = void 0;
var deviceScales = exports.deviceScales = {
  sberPortal: 2,
  sberBox: 2,
  mobile: 1
};
var DeviceKindList = exports.DeviceKindList = /*#__PURE__*/function (DeviceKindList) {
  DeviceKindList["sberBox"] = "sberBox";
  DeviceKindList["mobile"] = "mobile";
  DeviceKindList["sberPortal"] = "sberPortal";
  return DeviceKindList;
}({});
/**
 * Проверка в браузере на устройство "SberPortal".
 * @return {boolean}
 */
var isSberPortal = exports.isSberPortal = function isSberPortal() {
  if (typeof navigator === 'undefined') {
    return false;
  }
  var ua = navigator.userAgent.toLowerCase();
  return ua.includes('sberportal') || ua.includes('stargate');
};

/**
 * Проверка в браузере на устройство "SberBox".
 * @return {boolean}
 */
var isSberBox = exports.isSberBox = function isSberBox() {
  if (typeof navigator === 'undefined') {
    return false;
  }
  var ua = navigator.userAgent.toLowerCase();
  return ua.includes('sberbox');
};

/**
 * Проверка в браузере на устройство "SberBoxTop".
 * @return {boolean}
 */
var isSberBoxTop = exports.isSberBoxTop = function isSberBoxTop() {
  if (typeof navigator === 'undefined') {
    return false;
  }
  var ua = navigator.userAgent.toLowerCase();
  return ua.includes('satellite') || ua.includes('sberbox top');
};

/**
 * Проверка в браузере на устройство "SberBoxTV".
 * @return {boolean}
 */
var isTV = exports.isTV = function isTV() {
  if (typeof navigator === 'undefined') {
    return false;
  }
  var ua = navigator.userAgent.toLowerCase();
  return ua.includes('(tv; tv)') || ua.includes('(tv; huawei)') || ua.includes('(huawei-tv; huawei)') || ua.includes('(huawei-tv; huawei tv)');
};

/**
 * Вернет тип устройства, под которым запустилось приложение.
 * В случае запуска в серверном окружении, возвращает "sberBox".
 * @return {DeviceKind}
 */
var detectDevice = exports.detectDevice = function detectDevice() {
  if (typeof navigator === 'undefined') {
    return DeviceKindList.sberBox;
  }
  switch (true) {
    case isSberPortal():
      return DeviceKindList.sberPortal;
    case isSberBoxTop():
    case isSberBox():
    case isTV():
      return DeviceKindList.sberBox;
    default:
      return DeviceKindList.mobile;
  }
};
