// @salutejs/plasma-ui/utils/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "DeviceKindList", {
  enumerable: true,
  get: function get() {
    return _deviceDetection.DeviceKindList;
  }
});
Object.defineProperty(exports, "IS_REACT_18", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.IS_REACT_18;
  }
});
Object.defineProperty(exports, "animatedScrollToX", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.animatedScrollToX;
  }
});
Object.defineProperty(exports, "animatedScrollToY", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.animatedScrollToY;
  }
});
Object.defineProperty(exports, "convertRoundnessMatrix", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.convertRoundnessMatrix;
  }
});
Object.defineProperty(exports, "detectDevice", {
  enumerable: true,
  get: function get() {
    return _deviceDetection.detectDevice;
  }
});
Object.defineProperty(exports, "deviceScales", {
  enumerable: true,
  get: function get() {
    return _deviceDetection.deviceScales;
  }
});
Object.defineProperty(exports, "formatCurrency", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.formatCurrency;
  }
});
Object.defineProperty(exports, "gridBreakpoints", {
  enumerable: true,
  get: function get() {
    return _mediaQuery.breakpoints;
  }
});
Object.defineProperty(exports, "gridColumns", {
  enumerable: true,
  get: function get() {
    return _mediaQuery.columns;
  }
});
Object.defineProperty(exports, "gridGutters", {
  enumerable: true,
  get: function get() {
    return _mediaQuery.gutters;
  }
});
Object.defineProperty(exports, "gridMargins", {
  enumerable: true,
  get: function get() {
    return _mediaQuery.margins;
  }
});
Object.defineProperty(exports, "gridSizes", {
  enumerable: true,
  get: function get() {
    return _mediaQuery.sizes;
  }
});
Object.defineProperty(exports, "isSberBox", {
  enumerable: true,
  get: function get() {
    return _deviceDetection.isSberBox;
  }
});
Object.defineProperty(exports, "isSberPortal", {
  enumerable: true,
  get: function get() {
    return _deviceDetection.isSberPortal;
  }
});
Object.defineProperty(exports, "mediaQuery", {
  enumerable: true,
  get: function get() {
    return _mediaQuery.mediaQuery;
  }
});
Object.defineProperty(exports, "monthLongName", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.monthLongName;
  }
});
Object.defineProperty(exports, "padZeroNumber", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.padZeroNumber;
  }
});
Object.defineProperty(exports, "safeFlushSync", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.safeFlushSync;
  }
});
Object.defineProperty(exports, "setRef", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.setRef;
  }
});
Object.defineProperty(exports, "spatnavClassNameAttrs", {
  enumerable: true,
  get: function get() {
    return _spatnavClassNameAttrs.spatnavClassNameAttrs;
  }
});
Object.defineProperty(exports, "toCssSize", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.toCssSize;
  }
});
Object.defineProperty(exports, "transformStyles", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.transformStyles;
  }
});
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _mediaQuery = /*#__PURE__*/require("./mediaQuery");
var _deviceDetection = /*#__PURE__*/require("./deviceDetection");
var _spatnavClassNameAttrs = /*#__PURE__*/require("./spatnavClassNameAttrs");
