// @salutejs/plasma-ui/utils/mediaQuery.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.sizes = exports.mediaQuery = exports.margins = exports.gutters = exports.columns = exports.breakpoints = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var breakpoints = exports.breakpoints = {
  XL: 960,
  // TV
  L: 769,
  // Portal 8 cols
  M: 560,
  // Portal 6 cols
  S: 0 // Mobile
};
var columns = exports.columns = {
  XL: 12,
  L: 8,
  M: 6,
  S: 4
};
var margins = exports.margins = {
  XL: 4,
  L: 3.5,
  M: 3.5,
  S: 1
};
var gutters = exports.gutters = {
  XL: 1,
  L: 1,
  M: 1,
  S: 0.5
};
var sizes = exports.sizes = /*#__PURE__*/Object.keys(breakpoints);

/**
 * Обертка над css-медиазапросами с заранее определенными брейкпоинтами.
 */
var mediaQuery = exports.mediaQuery = function mediaQuery(breakpoint) {
  var deviceScale = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 2;
  var index = sizes.indexOf(breakpoint);
  var nextBreakpoint = sizes[index - 1];
  var min = breakpoints[breakpoint] > 0 ? breakpoints[breakpoint] * deviceScale : null;
  var max = breakpoints[nextBreakpoint] ? (breakpoints[nextBreakpoint] - 1) * deviceScale : null;
  if (min === null && max !== null) {
    return function (content) {
      return (0, _styledComponents.css)(["@media (max-width:", "px){", "}"], max, content);
    };
  }
  if (min !== null && max !== null) {
    return function (content) {
      return (0, _styledComponents.css)(["@media screen and (min-width:", "px) and (max-width:", "px){", "}"], min, max, content);
    };
  }
  return function (content) {
    return (0, _styledComponents.css)(["@media (min-width:", "px){", "}"], min, content);
  };
};
