// @salutejs/plasma-new-hope/es/components/Calendar/CalendarDouble/CalendarDouble.js
import { objectWithoutProperties as _objectWithoutProperties, slicedToArray as _slicedToArray, extends as _extends } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef, useMemo, useState, useCallback } from 'react';
import { getDateFromValue, getPrevDate, getNextDate, isValueUpdate } from '../utils/index.js';
import { useKeyNavigation } from '../hooks/useKeyNavigation.js';
import { CalendarHeader } from '../ui/CalendarHeader/CalendarHeader.js';
import { CalendarDays } from '../ui/CalendarDays/CalendarDays.js';
import '../../../utils/react.js';
import '@linaria/react';
import { base } from './variations/_view/base.js';
import { StyledWrapper, StyledSeparator, StyledCalendar } from './CalendarDouble.styles.js';

var _StyledSeparator;
var _excluded = ["value", "min", "max", "includeEdgeDates", "eventList", "disabledList", "onChangeValue"];
/**
 * Компонент двойного календаря.
 */
var calendarDoubleRoot = function calendarDoubleRoot(Root) {
  return /*#__PURE__*/forwardRef(function (_ref, outerRootRef) {
    var externalValue = _ref.value,
      min = _ref.min,
      max = _ref.max,
      includeEdgeDates = _ref.includeEdgeDates,
      eventList = _ref.eventList,
      disabledList = _ref.disabledList,
      onChangeValue = _ref.onChangeValue,
      rest = _objectWithoutProperties(_ref, _excluded);
    var _useMemo = useMemo(function () {
        return Array.isArray(externalValue) ? externalValue : [externalValue];
      }, [externalValue]),
      _useMemo2 = _slicedToArray(_useMemo, 2),
      firstValue = _useMemo2[0],
      secondValue = _useMemo2[1];
    var value = secondValue || firstValue;
    var _useState = useState(),
      _useState2 = _slicedToArray(_useState, 2),
      hoveredDay = _useState2[0],
      setHoveredDay = _useState2[1];
    var _useState3 = useState(getDateFromValue(value)),
      _useState4 = _slicedToArray(_useState3, 2),
      date = _useState4[0],
      setDate = _useState4[1];
    var _useState5 = useState(value),
      _useState6 = _slicedToArray(_useState5, 2),
      prevValue = _useState6[0],
      setPrevValue = _useState6[1];
    var _useState7 = useState(function () {
        var nextDate = getDateFromValue(firstValue);
        var _getNextDate = getNextDate(nextDate.year, nextDate.monthIndex),
          _getNextDate2 = _slicedToArray(_getNextDate, 2),
          initialYear = _getNextDate2[0],
          initialMonth = _getNextDate2[1];
        return {
          monthIndex: [nextDate.monthIndex, initialMonth],
          year: [nextDate.year, initialYear]
        };
      }),
      _useState8 = _slicedToArray(_useState7, 2),
      doubleDate = _useState8[0],
      setMonths = _useState8[1];
    var handleMonth = useCallback(function (getDate) {
      var _getDate = getDate(doubleDate.year[0], doubleDate.monthIndex[0]),
        _getDate2 = _slicedToArray(_getDate, 2),
        newCurrentYear = _getDate2[0],
        newCurrentMonth = _getDate2[1];
      var _getDate3 = getDate(doubleDate.year[1], doubleDate.monthIndex[1]),
        _getDate4 = _slicedToArray(_getDate3, 2),
        newNextYear = _getDate4[0],
        newNextMonth = _getDate4[1];
      setMonths({
        monthIndex: [newCurrentMonth, newNextMonth],
        year: [newCurrentYear, newNextYear]
      });
    }, [doubleDate]);
    var handlePrev = useCallback(function () {
      handleMonth(getPrevDate);
    }, [handleMonth]);
    var handleNext = useCallback(function () {
      handleMonth(getNextDate);
    }, [handleMonth]);
    var _useKeyNavigation = useKeyNavigation({
        isDouble: true,
        size: [11, 6],
        onNext: handleNext,
        onPrev: handlePrev
      }),
      _useKeyNavigation2 = _slicedToArray(_useKeyNavigation, 4),
      selectIndexes = _useKeyNavigation2[0],
      onKeyDown = _useKeyNavigation2[1],
      onSelectIndexes = _useKeyNavigation2[2],
      outerRefs = _useKeyNavigation2[3];
    var handleOnChangeDay = useCallback(function (newDate, coord) {
      var newDay = new Date(newDate.year, newDate.monthIndex, newDate.day);
      onChangeValue === null || onChangeValue === void 0 || onChangeValue(newDay);
      onSelectIndexes(coord);
    }, [onChangeValue, onSelectIndexes]);
    var firstDate = useMemo(function () {
      return {
        day: date.day,
        year: doubleDate.year[0],
        monthIndex: doubleDate.monthIndex[0]
      };
    }, [date, doubleDate]);
    var secondDate = useMemo(function () {
      return {
        day: date.day,
        year: doubleDate.year[1],
        monthIndex: doubleDate.monthIndex[1]
      };
    }, [date, doubleDate]);
    if (value && prevValue && isValueUpdate(value, prevValue)) {
      var newDate = getDateFromValue(value);
      var year = newDate.year,
        monthIndex = newDate.monthIndex;
      var _doubleDate$monthInde = _slicedToArray(doubleDate.monthIndex, 2),
        prevMonthIndex = _doubleDate$monthInde[1],
        _doubleDate$year = _slicedToArray(doubleDate.year, 2),
        prevYear = _doubleDate$year[1];
      if (prevMonthIndex !== monthIndex || prevYear !== year) {
        var _getNextDate3 = getNextDate(year, monthIndex),
          _getNextDate4 = _slicedToArray(_getNextDate3, 2),
          nextYear = _getNextDate4[0],
          nextMonthIndex = _getNextDate4[1];
        setDate(newDate);
        setMonths({
          monthIndex: [monthIndex, nextMonthIndex],
          year: [year, nextYear]
        });
      }
      setPrevValue(value);
    }
    return /*#__PURE__*/React.createElement(Root, _extends({
      ref: outerRootRef,
      "aria-label": "\u0412\u044B\u0431\u043E\u0440 \u0434\u0430\u0442\u044B"
    }, rest), /*#__PURE__*/React.createElement(CalendarHeader, {
      isDouble: true,
      firstDate: firstDate,
      secondDate: secondDate,
      onPrev: handlePrev,
      onNext: handleNext
    }), /*#__PURE__*/React.createElement(StyledWrapper, null, /*#__PURE__*/React.createElement(CalendarDays, {
      isDouble: true,
      eventList: eventList,
      disabledList: disabledList,
      min: min,
      max: max,
      includeEdgeDates: includeEdgeDates,
      value: externalValue,
      date: firstDate,
      hoveredDay: hoveredDay,
      selectIndexes: selectIndexes,
      onChangeDay: handleOnChangeDay,
      onHoverDay: setHoveredDay,
      onSetSelected: onSelectIndexes,
      onKeyDown: onKeyDown,
      outerRefs: outerRefs
    }), _StyledSeparator || (_StyledSeparator = /*#__PURE__*/React.createElement(StyledSeparator, null)), /*#__PURE__*/React.createElement(CalendarDays, {
      isDouble: true,
      isSecond: true,
      eventList: eventList,
      disabledList: disabledList,
      min: min,
      max: max,
      includeEdgeDates: includeEdgeDates,
      value: externalValue,
      date: secondDate,
      hoveredDay: hoveredDay,
      selectIndexes: selectIndexes,
      onChangeDay: handleOnChangeDay,
      onHoverDay: setHoveredDay,
      onSetSelected: onSelectIndexes,
      onKeyDown: onKeyDown,
      outerRefs: outerRefs
    })));
  });
};
var calendarDoubleConfig = {
  name: 'CalendarDouble',
  tag: 'div',
  layout: calendarDoubleRoot,
  base: StyledCalendar,
  variations: {
    view: {
      css: base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

export { calendarDoubleConfig, calendarDoubleRoot };
