// @salutejs/plasma-new-hope/cjs/components/Cell/Cell.styles.js
import './Cell.styles_ygfo1w.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var styled = require('styled-components');
var Cell_tokens = require('./Cell.tokens.js');

var alignToFlex = {
  center: 'center',
  top: 'flex-start',
  bottom: 'flex-end'
};
var base = "b1wikybr";
var CellContentWrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-taup5i-0"
})(["display:flex;flex:auto;align-items:stretch;justify-content:space-between;gap:var(", ");height:auto;"], Cell_tokens.tokens.cellGap);
var CellContent = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-taup5i-1"
})(["display:flex;align-items:center;height:auto;"]);
var CellLeft = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-taup5i-2"
})(["display:flex;align-items:", ";"], function (_ref) {
  var align = _ref.align;
  return alignToFlex[align];
});
var CellRight = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-taup5i-3"
})(["display:flex;align-items:", ";text-align:right;"], function (_ref2) {
  var align = _ref2.align;
  return alignToFlex[align];
});

exports.CellContent = CellContent;
exports.CellContentWrapper = CellContentWrapper;
exports.CellLeft = CellLeft;
exports.CellRight = CellRight;
exports.base = base;
