// @salutejs/plasma-new-hope/styled-components/cjs/components/Calendar/ui/CalendarYears/CalendarYears.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CalendarYears = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _utils = /*#__PURE__*/require("../../../../utils");
var _hooks = /*#__PURE__*/require("../../hooks");
var _Calendar = /*#__PURE__*/require("../../Calendar.tokens");
var _CalendarYears = /*#__PURE__*/require("./CalendarYears.styles");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
/**
 * Компонент годов в календаре.
 */
var CalendarYears = exports.CalendarYears = function CalendarYears(_ref) {
  var currentDate = _ref.date,
    startYear = _ref.startYear,
    selectIndexes = _ref.selectIndexes,
    outerRefs = _ref.outerRefs,
    onChangeYear = _ref.onChangeYear,
    onSetSelected = _ref.onSetSelected,
    onKeyDown = _ref.onKeyDown;
  var _useYears = (0, _hooks.useYears)(currentDate, startYear),
    _useYears2 = _slicedToArray(_useYears, 2),
    years = _useYears2[0],
    selected = _useYears2[1];
  var selectedRef = (0, _react.useRef)(selected);
  var onSetSelectedRef = (0, _react.useRef)(onSetSelected);
  var handleOnChangeYear = (0, _react.useCallback)(function (event) {
    /**
     * нужно вызвать stopImmediatePropagation для случаев, когда
     * обработчик события onClick навешивается снаружи.
     * Как, например, в компоненте Popup
     */
    event.nativeEvent.stopImmediatePropagation();
    var year = event.currentTarget.dataset.year;
    onChangeYear(Number(year));
  }, [onChangeYear]);
  var getRefs = (0, _react.useCallback)(function (element, i, j) {
    outerRefs.current[i][j] = element;
  }, [outerRefs]);
  (0, _react.useEffect)(function () {
    if (selectedRef.current) {
      var _onSetSelectedRef$cur;
      (_onSetSelectedRef$cur = onSetSelectedRef.current) === null || _onSetSelectedRef$cur === void 0 || _onSetSelectedRef$cur.call(onSetSelectedRef, selectedRef.current);
    }
  }, []);
  return /*#__PURE__*/_react["default"].createElement(_CalendarYears.StyledCalendarYears, {
    role: "grid",
    "aria-labelledby": "id-grid-label",
    onKeyDown: onKeyDown
  }, years.map(function (year, i) {
    return /*#__PURE__*/_react["default"].createElement(_CalendarYears.StyledFlex, {
      role: "row",
      key: i
    }, year.map(function (_ref2, j) {
      var yearValue = _ref2.yearValue,
        isSelected = _ref2.isSelected,
        isCurrent = _ref2.isCurrent;
      var selectedClass = isSelected ? _Calendar.classes.selectedItem : undefined;
      var currentClass = !isSelected && isCurrent ? _Calendar.classes.currentItem : undefined;
      return /*#__PURE__*/_react["default"].createElement(_CalendarYears.StyledYearRoot, {
        className: (0, _utils.cx)(selectedClass, currentClass, _Calendar.classes.selectableItem),
        ref: function ref(element) {
          return getRefs(element, i, j);
        },
        tabIndex: i === (selectIndexes === null || selectIndexes === void 0 ? void 0 : selectIndexes[0]) && j === (selectIndexes === null || selectIndexes === void 0 ? void 0 : selectIndexes[1]) ? 0 : -1,
        onClick: handleOnChangeYear,
        "data-year": yearValue,
        "aria-selected": isSelected,
        role: "gridcell",
        key: "StyledYear-".concat(i, "-").concat(j),
        "aria-label": String(yearValue)
      }, /*#__PURE__*/_react["default"].createElement(_CalendarYears.StyledYear, null, yearValue));
    }));
  }));
};
