// @salutejs/plasma-core/components/Basebox/Basebox.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledTrigger = exports.StyledRoot = exports.StyledLabel = exports.StyledInput = exports.StyledDescription = exports.StyledContentWrapper = exports.StyledContent = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _Typography = /*#__PURE__*/require("../Typography");
var _mixins = /*#__PURE__*/require("../../mixins");
var _tokens = /*#__PURE__*/require("../../tokens");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
(0, _collectPackageInfo.collectPackageInfo)('Basebox/Basebox');
var StyledRoot = exports.StyledRoot = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-11r48g6-0"
})(["", " position:relative;align-items:flex-start;display:flex;cursor:pointer;"], _mixins.applyNoSelect);
var StyledContentWrapper = exports.StyledContentWrapper = /*#__PURE__*/_styledComponents["default"].label.withConfig({
  componentId: "plasma__sc-11r48g6-1"
})(["position:relative;align-items:flex-start;display:flex;width:100%;cursor:inherit;"]);
var StyledInput = exports.StyledInput = /*#__PURE__*/_styledComponents["default"].input.withConfig({
  componentId: "plasma__sc-11r48g6-2"
})(["position:absolute;&:focus{outline:0 none;}"]);
var StyledTrigger = exports.StyledTrigger = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-11r48g6-3"
})(["box-sizing:border-box;position:relative;transition:", ";"], function (_ref) {
  var theme = _ref.theme;
  return theme.lowPerformance ? 'unset' : 'all 0.1s ease-in-out';
});
var StyledContent = exports.StyledContent = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-11r48g6-4"
})(["display:flex;flex-direction:column;margin-left:0.75rem;"]);
var StyledLabel = exports.StyledLabel = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Body1).withConfig({
  componentId: "plasma__sc-11r48g6-5"
})(["", ""], function (_ref2) {
  var singleLine = _ref2.singleLine;
  return singleLine && (0, _styledComponents.css)(["", ""], _mixins.applyEllipsis);
});
var StyledDescription = exports.StyledDescription = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Footnote1).withConfig({
  componentId: "plasma__sc-11r48g6-6"
})(["color:", ";", ""], _tokens.secondary, function (_ref3) {
  var singleLine = _ref3.singleLine;
  return singleLine && (0, _styledComponents.css)(["", ""], _mixins.applyEllipsis);
});
