// @salutejs/plasma-core/components/Basebox/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "BaseboxContent", {
  enumerable: true,
  get: function get() {
    return _Basebox.StyledContent;
  }
});
Object.defineProperty(exports, "BaseboxContentWrapper", {
  enumerable: true,
  get: function get() {
    return _Basebox.StyledContentWrapper;
  }
});
Object.defineProperty(exports, "BaseboxDescription", {
  enumerable: true,
  get: function get() {
    return _Basebox.StyledDescription;
  }
});
Object.defineProperty(exports, "BaseboxInput", {
  enumerable: true,
  get: function get() {
    return _Basebox.StyledInput;
  }
});
Object.defineProperty(exports, "BaseboxLabel", {
  enumerable: true,
  get: function get() {
    return _Basebox.StyledLabel;
  }
});
Object.defineProperty(exports, "BaseboxRoot", {
  enumerable: true,
  get: function get() {
    return _Basebox.StyledRoot;
  }
});
Object.defineProperty(exports, "BaseboxTrigger", {
  enumerable: true,
  get: function get() {
    return _Basebox.StyledTrigger;
  }
});
var _Basebox = /*#__PURE__*/require("./Basebox");
