// @salutejs/plasma-new-hope/cjs/components/Calendar/ui/CalendarMonths/CalendarMonths.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var index = require('../../../../utils/index.js');
var useMonths = require('../../hooks/useMonths.js');
var Calendar_tokens = require('../../Calendar.tokens.js');
var CalendarMonths_styles = require('./CalendarMonths.styles.js');

/**
 * Компонент месяцев в календаре.
 */
var CalendarMonths = function CalendarMonths(_ref) {
  var currentDate = _ref.date,
    selectIndexes = _ref.selectIndexes,
    outerRefs = _ref.outerRefs,
    onChangeMonth = _ref.onChangeMonth,
    onSetSelected = _ref.onSetSelected,
    onKeyDown = _ref.onKeyDown;
  var _useMonths = useMonths.useMonths(currentDate),
    _useMonths2 = _rollupPluginBabelHelpers.slicedToArray(_useMonths, 2),
    months = _useMonths2[0],
    selected = _useMonths2[1];
  var selectedRef = React.useRef(selected);
  var onSetSelectedRef = React.useRef(onSetSelected);
  var handleOnChangeMonth = React.useCallback(function (event) {
    /**
     * нужно вызвать stopImmediatePropagation для случаев, когда
     * обработчик события onClick навешивается снаружи.
     * Как, например, в компоненте Popup
     */
    event.nativeEvent.stopImmediatePropagation();
    var monthIndex = event.currentTarget.dataset.monthIndex;
    onChangeMonth(Number(monthIndex));
  }, [onChangeMonth]);
  var getRefs = React.useCallback(function (element, i, j) {
    outerRefs.current[i][j] = element;
  }, [outerRefs]);
  React.useEffect(function () {
    if (selectedRef.current) {
      var _onSetSelectedRef$cur;
      (_onSetSelectedRef$cur = onSetSelectedRef.current) === null || _onSetSelectedRef$cur === void 0 || _onSetSelectedRef$cur.call(onSetSelectedRef, selectedRef.current);
    }
  }, []);
  return /*#__PURE__*/React.createElement(CalendarMonths_styles.StyledCalendarMonths, {
    role: "grid",
    "aria-labelledby": "id-grid-label",
    onKeyDown: onKeyDown
  }, months.map(function (month, i) {
    return /*#__PURE__*/React.createElement(CalendarMonths_styles.StyledFlex, {
      role: "row",
      key: i
    }, month.map(function (_ref2, j) {
      var monthName = _ref2.monthName,
        monthIndex = _ref2.monthIndex,
        isSelected = _ref2.isSelected,
        isCurrent = _ref2.isCurrent,
        monthFullName = _ref2.monthFullName;
      var selectedClass = isSelected ? Calendar_tokens.classes.selectedItem : undefined;
      var currentClass = !isSelected && isCurrent ? Calendar_tokens.classes.currentItem : undefined;
      return /*#__PURE__*/React.createElement(CalendarMonths_styles.StyledMonthRoot, {
        id: "month-test-".concat(i, "-").concat(j),
        className: index.cx(selectedClass, currentClass, Calendar_tokens.classes.selectableItem),
        ref: function ref(element) {
          return getRefs(element, i, j);
        },
        tabIndex: i === (selectIndexes === null || selectIndexes === void 0 ? void 0 : selectIndexes[0]) && j === (selectIndexes === null || selectIndexes === void 0 ? void 0 : selectIndexes[1]) ? 0 : -1,
        onClick: handleOnChangeMonth,
        "data-month-index": monthIndex,
        "aria-selected": isSelected,
        role: "gridcell",
        key: "StyledMonth-".concat(i, "-").concat(j),
        "aria-label": monthFullName
      }, /*#__PURE__*/React.createElement(CalendarMonths_styles.StyledMonth, null, monthName));
    }));
  }));
};

exports.CalendarMonths = CalendarMonths;
