// @salutejs/plasma-new-hope/es/components/Button/Button.styles.js
import './Button.styles_1oz1dew.css';
import { styled } from '@linaria/react';
import 'focus-visible';
import 'styled-components';
import '@linaria/core';
import { component, mergeConfig } from '../../engines/common.js';
import { spinnerConfig } from '../Spinner/Spinner.js';

// issue #823
var mergedConfig = /*#__PURE__*/mergeConfig(spinnerConfig);
var Spinner = /*#__PURE__*/component(mergedConfig);
var ButtonText = /*#__PURE__*/styled('span')({
  name: "ButtonText",
  "class": "bzw1898",
  propsAsIs: false
});
var ButtonValue = /*#__PURE__*/styled('span')({
  name: "ButtonValue",
  "class": "bdfuqwe",
  propsAsIs: false
});

// TODO: #720
var _exp6 = function _exp6() {
  return function (_ref) {
    var isLoading = _ref.isLoading;
    return isLoading ? '0' : '1';
  };
};
var _exp7 = function _exp7() {
  return function (_ref2) {
    var contentPlacing = _ref2.contentPlacing;
    return contentPlacing;
  };
};
var LoadWrap = /*#__PURE__*/styled('div')({
  name: "LoadWrap",
  "class": "l9ypozh",
  propsAsIs: false,
  vars: {
    "l9ypozh-0": [/*#__PURE__*/_exp6()],
    "l9ypozh-1": [/*#__PURE__*/_exp7()]
  }
});
var Loader = /*#__PURE__*/styled('div')({
  name: "Loader",
  "class": "l4dfhbg",
  propsAsIs: false
});
var _exp8 = function _exp8() {
  return Spinner;
};
var StyledSpinner = /*#__PURE__*/styled(_exp8())({
  name: "StyledSpinner",
  "class": "srdlazt",
  propsAsIs: true
});
var StyledContentLeft = /*#__PURE__*/styled('div')({
  name: "StyledContentLeft",
  "class": "s7rggv0",
  propsAsIs: false
});
var StyledContentRight = /*#__PURE__*/styled('div')({
  name: "StyledContentRight",
  "class": "s1rjh7ra",
  propsAsIs: false
});
var base = "b9ga1dj";

export { ButtonText, ButtonValue, LoadWrap, Loader, StyledContentLeft, StyledContentRight, StyledSpinner, base };
