// @salutejs/plasma-new-hope/cjs/components/Dropdown/Dropdown.styles.js
import './Dropdown.styles_1tihqkp.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
require('@linaria/core');
var common = require('../../engines/common.js');
var Popover$1 = require('../Popover/Popover.js');
var index = require('./utils/index.js');
var Dropdown_tokens = require('./Dropdown.tokens.js');

var Popover = /*#__PURE__*/common.component(Popover$1.popoverConfig);
var _exp = function _exp() {
  return Popover;
};
var StyledPopover = /*#__PURE__*/react.styled(_exp())({
  name: "StyledPopover",
  "class": "sn7uamp",
  propsAsIs: true
});
var _exp7 = function _exp7() {
  return function (_ref) {
    var listWidth = _ref.listWidth;
    return listWidth || "var(".concat(Dropdown_tokens.tokens.width, ")");
  };
};
var _exp8 = function _exp8() {
  return function (_ref2) {
    var listHeight = _ref2.listHeight;
    return listHeight ? index.getCorrectHeight(listHeight) : 'auto';
  };
};
var _exp9 = function _exp9() {
  return function (_ref3) {
    var listOverflow = _ref3.listOverflow;
    return listOverflow || 'initial';
  };
};
var _exp10 = function _exp10() {
  return function (_ref4) {
    var isInnerUl = _ref4.isInnerUl;
    return isInnerUl ? "calc(var(".concat(Dropdown_tokens.tokens.padding, ") * -1) 0 0 var(").concat(Dropdown_tokens.tokens.padding, ")") : 0;
  };
};
var Ul = /*#__PURE__*/react.styled('ul')({
  name: "Ul",
  "class": "ujr7fmv",
  propsAsIs: false,
  vars: {
    "ujr7fmv-0": [/*#__PURE__*/_exp7()],
    "ujr7fmv-1": [/*#__PURE__*/_exp8()],
    "ujr7fmv-2": [/*#__PURE__*/_exp9()],
    "ujr7fmv-3": [/*#__PURE__*/_exp10()]
  }
});
var base = "byfocd2";

exports.StyledPopover = StyledPopover;
exports.Ul = Ul;
exports.base = base;
