// @salutejs/plasma-web/es/components/Drawer/Drawer.config.js
var _templateObject, _templateObject2, _templateObject3, _templateObject4;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css, drawerTokens } from '@salutejs/plasma-new-hope/styled-components';
export var config = {
  defaults: {
    view: 'default',
    size: 'm'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/css(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--surface-solid-card);\n                ", ": 0 3.75rem 7rem -0.5rem rgba(0, 0, 0, 0.08);\n                ", ": var(--surface-transparent-primary);\n                ", ": var(--overlay-blur);\n                ", ": var(--overlay-soft);\n            "])), drawerTokens.background, drawerTokens.shadow, drawerTokens.contentBackgroundColor, drawerTokens.drawerOverlayWithBlurColor, drawerTokens.drawerOverlayColor)
    },
    size: {
      m: /*#__PURE__*/css(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": 1.5rem;\n            "])), drawerTokens.padding)
    },
    borderRadius: {
      none: /*#__PURE__*/css(_templateObject3 || (_templateObject3 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": 0;\n            "])), drawerTokens.borderRadius),
      "default": /*#__PURE__*/css(_templateObject4 || (_templateObject4 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": 1.25rem;\n            "])), drawerTokens.borderRadius)
    }
  }
};
