// @salutejs/plasma-new-hope/es/components/Slider/Slider.js
import { extends as _extends } from '../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef } from 'react';
import { base } from './variations/_view/base.js';
import { base as base$1 } from './variations/_size/base.js';
import { base as base$2 } from './variations/_disabled/base.js';
import { SingleSlider } from './components/Single/Single.js';
import { DoubleSlider } from './components/Double/Double.js';

var isSingleValueProps = function isSingleValueProps(props) {
  return typeof props.value === 'number';
};
var sliderRoot = function sliderRoot(Root) {
  return /*#__PURE__*/forwardRef(function (props, ref) {
    return /*#__PURE__*/React.createElement(Root, _extends({
      ref: ref
    }, props), isSingleValueProps(props) && /*#__PURE__*/React.createElement(SingleSlider, props), !isSingleValueProps(props) && /*#__PURE__*/React.createElement(DoubleSlider, props));
  });
};
var sliderConfig = {
  name: 'Slider',
  tag: 'div',
  layout: sliderRoot,
  base: '',
  variations: {
    view: {
      css: base
    },
    size: {
      css: base$1
    },
    disabled: {
      css: base$2,
      attrs: true
    }
  },
  defaults: {
    view: 'default',
    size: 'm'
  }
};

export { sliderConfig, sliderRoot };
