// @salutejs/plasma-new-hope/styled-components/cjs/components/Drawer/ui/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "drawerContentConfig", {
  enumerable: true,
  get: function get() {
    return _DrawerContent.drawerContentConfig;
  }
});
Object.defineProperty(exports, "drawerContentRoot", {
  enumerable: true,
  get: function get() {
    return _DrawerContent.drawerContentRoot;
  }
});
Object.defineProperty(exports, "drawerFooterConfig", {
  enumerable: true,
  get: function get() {
    return _DrawerFooter.drawerFooterConfig;
  }
});
Object.defineProperty(exports, "drawerFooterRoot", {
  enumerable: true,
  get: function get() {
    return _DrawerFooter.drawerFooterRoot;
  }
});
Object.defineProperty(exports, "drawerHeaderConfig", {
  enumerable: true,
  get: function get() {
    return _DrawerHeader.drawerHeaderConfig;
  }
});
Object.defineProperty(exports, "drawerHeaderRoot", {
  enumerable: true,
  get: function get() {
    return _DrawerHeader.drawerHeaderRoot;
  }
});
var _DrawerContent = /*#__PURE__*/require("./DrawerContent/DrawerContent");
var _DrawerFooter = /*#__PURE__*/require("./DrawerFooter/DrawerFooter");
var _DrawerHeader = /*#__PURE__*/require("./DrawerHeader/DrawerHeader");
