// @salutejs/plasma-new-hope/styled-components/es/examples/plasma_web/components/Spinner/Spinner.config.js
import { css } from 'styled-components';
import { spinnerTokens } from '../../../../components/Spinner';
export var config = {
  defaults: {
    view: 'accent',
    size: 'm'
  },
  variations: {
    size: {
      s: /*#__PURE__*/css(["", ":16px;"], spinnerTokens.size),
      m: /*#__PURE__*/css(["", ":24px;"], spinnerTokens.size),
      l: /*#__PURE__*/css(["", ":36px;"], spinnerTokens.size)
    },
    view: {
      "default": /*#__PURE__*/css(["", ":var(--text-primary);"], spinnerTokens.color),
      secondary: /*#__PURE__*/css(["", ":var(--text-secondary);"], spinnerTokens.color),
      tertiary: /*#__PURE__*/css(["", ":var(--text-tertiary);"], spinnerTokens.color),
      paragraph: /*#__PURE__*/css(["", ":var(--text-paragraph);"], spinnerTokens.color),
      accent: /*#__PURE__*/css(["", ":var(--text-accent);"], spinnerTokens.color),
      positive: /*#__PURE__*/css(["", ":var(--text-positive);"], spinnerTokens.color),
      warning: /*#__PURE__*/css(["", ":var(--text-warning);"], spinnerTokens.color),
      negative: /*#__PURE__*/css(["", ":var(--text-negative);"], spinnerTokens.color)
    }
  }
};
