// @salutejs/plasma-new-hope/cjs/components/Typography/Old/Subtitle/Subtitle.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var applySpacing = require('../../../../mixins/applySpacing.js');
var base = require('../../variations/_size/base.js');
var Subtitle_styles = require('./Subtitle.styles.js');

var _excluded = ["size", "children", "style"];
var subtitleRoot = function subtitleRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (props, ref) {
    var size = props.size,
      children = props.children,
      style = props.style,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(props, _excluded);
    return /*#__PURE__*/React.createElement(Root, _rollupPluginBabelHelpers.extends({
      size: size,
      ref: ref,
      style: _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, style), applySpacing.applySpacing(rest))
    }, rest), children);
  });
};
var subtitleConfig = {
  name: 'Subtitle',
  tag: 'div',
  layout: subtitleRoot,
  base: Subtitle_styles.base,
  variations: {
    size: {
      css: base.base
    }
  },
  defaults: {
    size: 'm'
  }
};

exports.subtitleConfig = subtitleConfig;
exports.subtitleRoot = subtitleRoot;
