// @salutejs/plasma-web/css/es/components/TextField/TextField.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef } from 'react';
import { component, mergeConfig, textFieldConfig } from '@salutejs/plasma-new-hope';
import { config } from './TextField.config.js';

var _excluded = ["status", "label", "placeholder", "animatedHint", "helperText", "enumerationType", "chips", "onSearch", "onChangeChips", "size"];
var mergedConfig = /*#__PURE__*/mergeConfig(textFieldConfig, config);
var TextFieldComponent = /*#__PURE__*/component(mergedConfig);
var statusToView = {
  success: 'positive',
  warning: 'warning',
  error: 'negative'
};
var animatedHintToLabelPlacement = {
  placeholder: 'outer',
  label: 'inner'
};

/**
 * Поле ввода текста.
 */
var TextField = /*#__PURE__*/forwardRef(function (props, ref) {
  var status = props.status,
    label = props.label,
    placeholder = props.placeholder,
    animatedHint = props.animatedHint,
    helperText = props.helperText,
    enumerationType = props.enumerationType,
    chips = props.chips,
    onSearch = props.onSearch,
    onChangeChips = props.onChangeChips,
    _props$size = props.size,
    size = _props$size === void 0 ? 'l' : _props$size,
    rest = _objectWithoutProperties(props, _excluded);

  /** TODO: #1064 Remove In plasma-2.0 */
  /* eslint-disable no-underscore-dangle */
  var _view = status ? statusToView[status] : 'default';
  var _labelPlacement = animatedHint ? animatedHintToLabelPlacement[animatedHint] : 'outer';
  var _label = label ? String(label) : undefined;
  if (size !== 'l' && _labelPlacement === 'inner') {
    _label = undefined;
    _labelPlacement = 'outer';
  }
  if (size === 'l' && animatedHint === 'placeholder' && !label) {
    _label = placeholder;
    _labelPlacement = 'inner';
  }
  var _size = size === 'xs' ? 's' : size;
  if (enumerationType === 'chip') {
    return /*#__PURE__*/React.createElement(TextFieldComponent, _extends({}, rest, {
      view: _view,
      size: _size,
      labelPlacement: _labelPlacement,
      label: _label,
      placeholder: placeholder,
      leftHelper: helperText,
      ref: ref,
      enumerationType: "chip",
      chips: chips,
      onChangeChips: onChangeChips
    }));
  }
  return /*#__PURE__*/React.createElement(TextFieldComponent, _extends({}, rest, {
    view: _view,
    size: _size,
    labelPlacement: _labelPlacement,
    label: _label,
    placeholder: placeholder,
    leftHelper: helperText,
    ref: ref,
    enumerationType: "plain",
    onSearch: onSearch
  }));
});

export { TextField };
