// @salutejs/plasma-new-hope/cjs/components/Header/ui/HeaderTitleWrapper/HeaderTitleWrapper.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var base$1 = require('./variations/_size/base.js');
var base = require('./variations/_view/base.js');
var HeaderTitleWrapper_styles = require('./HeaderTitleWrapper.styles.js');

var _excluded = ["label", "title", "subTitle", "caption", "withBigTitle", "children"];

/**
 * Компонент для отображения текста в скомпанованном блоке.
 */
var headerTitleWrapperRoot = function headerTitleWrapperRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (props, outerRootRef) {
    var label = props.label,
      title = props.title,
      subTitle = props.subTitle,
      caption = props.caption,
      withBigTitle = props.withBigTitle,
      children = props.children,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(props, _excluded);
    if (children) {
      return /*#__PURE__*/React.createElement(Root, {
        ref: outerRootRef
      }, /*#__PURE__*/React.createElement(HeaderTitleWrapper_styles.TextBoxRoot, rest, children));
    }
    var T = withBigTitle ? HeaderTitleWrapper_styles.BigTitle : HeaderTitleWrapper_styles.Title;
    return /*#__PURE__*/React.createElement(Root, {
      ref: outerRootRef
    }, /*#__PURE__*/React.createElement(HeaderTitleWrapper_styles.TextBoxRoot, rest, label && /*#__PURE__*/React.createElement(HeaderTitleWrapper_styles.Label, null, label), title && /*#__PURE__*/React.createElement(T, null, title), subTitle && /*#__PURE__*/React.createElement(HeaderTitleWrapper_styles.SubTitle, null, subTitle), caption && /*#__PURE__*/React.createElement(HeaderTitleWrapper_styles.Caption, null, caption)));
  });
};
var headerTitleWrapperConfig = {
  name: 'HeaderTitleWrapper',
  tag: 'div',
  layout: headerTitleWrapperRoot,
  base: '',
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

exports.headerTitleWrapperConfig = headerTitleWrapperConfig;
exports.headerTitleWrapperRoot = headerTitleWrapperRoot;
