// @salutejs/plasma-new-hope/cjs/components/Header/ui/HeaderTitleWrapper/HeaderTitleWrapper.styles.js
import './HeaderTitleWrapper.styles_1lra53t.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');

var Title = /*#__PURE__*/react.styled('div')({
  name: "Title",
  "class": "t7wn5f0",
  propsAsIs: false
});
var BigTitle = /*#__PURE__*/react.styled('div')({
  name: "BigTitle",
  "class": "bkd5o5i",
  propsAsIs: false
});
var Label = /*#__PURE__*/react.styled('div')({
  name: "Label",
  "class": "l1216phq",
  propsAsIs: false
});
var SubTitle = /*#__PURE__*/react.styled('div')({
  name: "SubTitle",
  "class": "st5x6gi",
  propsAsIs: false
});
var Caption = /*#__PURE__*/react.styled('div')({
  name: "Caption",
  "class": "cig18vd",
  propsAsIs: false
});
var TextBoxRoot = /*#__PURE__*/react.styled('div')({
  name: "TextBoxRoot",
  "class": "tnlilk7",
  propsAsIs: false
});

exports.BigTitle = BigTitle;
exports.Caption = Caption;
exports.Label = Label;
exports.SubTitle = SubTitle;
exports.TextBoxRoot = TextBoxRoot;
exports.Title = Title;
