// @salutejs/plasma-web/components/Carousel/Carousel.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _Carousel = /*#__PURE__*/require("./Carousel.examples");
var _ = /*#__PURE__*/require(".");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var items = /*#__PURE__*/Array(25).fill({
  title: 'Слайд',
  subtitle: 'Описание слайда',
  imageSrc: 'images/320_320_n.jpg',
  imageAlt: 'Картинка'
}).map(function (_ref, i) {
  var title = _ref.title,
    subtitle = _ref.subtitle,
    imageSrc = _ref.imageSrc,
    imageAlt = _ref.imageAlt;
  return {
    id: "slide_".concat(i),
    title: "".concat(title, " ").concat(i),
    subtitle: "".concat(subtitle, " ").concat(i),
    imageSrc: imageSrc.replace('n', i % 12),
    imageAlt: "".concat(imageAlt, " ").concat(i)
  };
});
var defaultStyle = {
  margin: '0 -0.5rem'
};
var defaultCarouselItemStyle = {
  width: '20rem',
  padding: '0 0.5rem'
};
var Default = exports.Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(_.Carousel, {
    index: 0,
    style: defaultStyle
  }, items.map(function (item) {
    return /*#__PURE__*/_react["default"].createElement(_.CarouselItem, {
      key: item.id,
      style: defaultCarouselItemStyle
    }, /*#__PURE__*/_react["default"].createElement(_Carousel.CarouselCard, {
      id: item.id,
      title: item.title,
      subtitle: item.subtitle,
      imageSrc: item.imageSrc,
      imageAlt: item.imageAlt
    }));
  }));
};
