// @salutejs/plasma-web/components/Carousel/Carousel.examples.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CarouselCard = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _mixins = /*#__PURE__*/require("../../mixins");
var _Image = /*#__PURE__*/require("../Image");
var _Typography = /*#__PURE__*/require("../Typography");
var _excluded = ["title", "subtitle", "imageSrc", "imageAlt", "imageBase"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var StyledCard = /*#__PURE__*/_styledComponents["default"].a.withConfig({
  componentId: "plasma-web__sc-18sk49o-0"
})(["display:flex;position:relative;border-radius:1rem;", ""], /*#__PURE__*/(0, _mixins.addFocus)({
  outlined: true,
  outlineRadius: '1.125rem'
}));
var StyledCardContent = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-18sk49o-1"
})(["position:absolute;left:0;right:0;bottom:0;padding:1.72rem;color:#fff;"]);

/**
 * Карточка под примеры с каруселью.
 * @private
 */
var CarouselCard = exports.CarouselCard = function CarouselCard(_ref) {
  var title = _ref.title,
    subtitle = _ref.subtitle,
    imageSrc = _ref.imageSrc,
    imageAlt = _ref.imageAlt,
    _ref$imageBase = _ref.imageBase,
    imageBase = _ref$imageBase === void 0 ? 'div' : _ref$imageBase,
    rest = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/_react["default"].createElement(StyledCard, rest, /*#__PURE__*/_react["default"].createElement(_Image.Image, {
    src: imageSrc,
    ratio: "16 / 9",
    base: imageBase,
    alt: imageAlt
  }), /*#__PURE__*/_react["default"].createElement(StyledCardContent, null, /*#__PURE__*/_react["default"].createElement(_Typography.Headline4, null, title), /*#__PURE__*/_react["default"].createElement(_Typography.Footnote1, null, subtitle)));
};
