// @salutejs/plasma-ui/es/components/TextField/TextField.perftest.js
var _IconSleep, _ActionButton;
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React from 'react';
import { IconSleep, IconEye } from '@salutejs/plasma-icons';
import { ActionButton } from '../Button';
import { TextField } from '.';
var defaultProps = {
  id: 'example-text-field',
  type: 'text',
  placeholder: 'Заполните поле',
  helperText: 'Подсказка к полю',
  disabled: false,
  readOnly: false,
  size: 'l',
  enableLeftIcon: true,
  enableRightIcon: true
};
var status = '';
var func = function func() {};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(TextField, _extends({
    value: "\u0417\u043D\u0430\u0447\u0435\u043D\u0438\u0435 \u043F\u043E\u043B\u044F",
    status: status,
    contentLeft: _IconSleep || (_IconSleep = /*#__PURE__*/React.createElement(IconSleep, {
      color: "inherit",
      size: "s"
    })),
    contentRight: _ActionButton || (_ActionButton = /*#__PURE__*/React.createElement(ActionButton, {
      view: "clear"
    }, /*#__PURE__*/React.createElement(IconEye, {
      color: "inherit",
      size: "s"
    }))),
    onChange: func,
    onFocus: func,
    onBlur: func,
    onSearch: func
  }, defaultProps));
};
