// @salutejs/plasma-ui/components/Cell/CellDisclosure.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Disclosure = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _Button = /*#__PURE__*/require("../Button");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _IconDisclosureRight;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
(0, _collectPackageInfo.collectPackageInfo)('Cell/CellDisclosure');
var DisclosureBtn = /*#__PURE__*/(0, _styledComponents["default"])(_Button.Button).withConfig({
  componentId: "plasma-ui__sc-1hbfhpz-0"
})(["left:1rem;margin-left:-1rem;"]);
var Disclosure = exports.Disclosure = function Disclosure(props) {
  return /*#__PURE__*/_react["default"].createElement(DisclosureBtn, _extends({}, props, {
    size: "s",
    view: "clear",
    outlined: false,
    scaleOnInteraction: false,
    contentRight: _IconDisclosureRight || (_IconDisclosureRight = /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconDisclosureRight, {
      size: "s"
    }))
  }));
};
