// @salutejs/plasma-new-hope/cjs/components/Checkbox/Checkbox.styles.js
import './Checkbox.styles_1266095.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');

var base = "b1udywkg";
var StyledInput = /*#__PURE__*/react.styled('input')({
  name: "StyledInput",
  "class": "s165w87d",
  propsAsIs: false
});
var StyledTrigger = /*#__PURE__*/react.styled('div')({
  name: "StyledTrigger",
  "class": "sft3zxv",
  propsAsIs: false
});
var StyledContentWrapper = /*#__PURE__*/react.styled('label')({
  name: "StyledContentWrapper",
  "class": "s1u3nd7e",
  propsAsIs: false
});
var StyledContent = /*#__PURE__*/react.styled('div')({
  name: "StyledContent",
  "class": "shwv1in",
  propsAsIs: false
});
var StyledLabel = /*#__PURE__*/react.styled('span')({
  name: "StyledLabel",
  "class": "s1my1ay",
  propsAsIs: false
});
var StyledDescription = /*#__PURE__*/react.styled('div')({
  name: "StyledDescription",
  "class": "s14u8fnn",
  propsAsIs: false
});

exports.StyledContent = StyledContent;
exports.StyledContentWrapper = StyledContentWrapper;
exports.StyledDescription = StyledDescription;
exports.StyledInput = StyledInput;
exports.StyledLabel = StyledLabel;
exports.StyledTrigger = StyledTrigger;
exports.base = base;
