// @salutejs/plasma-new-hope/cjs/components/Skeleton/LineSkeleton/LineSkeleton.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
require('focus-visible');
var applyRoundness = require('../../../mixins/applyRoundness.js');
var applySkeletonGradient = require('../../../mixins/applySkeletonGradient.js');
require('styled-components');
var base = require('./variations/_size/base.js');
var LineSkeleton_styles = require('./LineSkeleton.styles.js');

var _excluded = ["size", "lighter", "customGradientColor", "roundness"];
/**
 * Копонент-составляющая Скелетона
 */
var lineSkeletonRoot = function lineSkeletonRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (_ref, outerRootRef) {
    var size = _ref.size,
      lighter = _ref.lighter,
      customGradientColor = _ref.customGradientColor,
      _ref$roundness = _ref.roundness,
      roundness = _ref$roundness === void 0 ? '16' : _ref$roundness,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
    var roundnessValue = applyRoundness.getRoundness({
      roundness: roundness
    });
    var skeletonGradientColor = applySkeletonGradient.getSkeletonColor({
      lighter: lighter,
      customGradientColor: customGradientColor
    });
    return /*#__PURE__*/React.createElement(Root, _rollupPluginBabelHelpers.extends({
      ref: outerRootRef,
      size: size
    }, rest), /*#__PURE__*/React.createElement(LineSkeleton_styles.StyledVisibleLine, {
      roundness: roundnessValue,
      gradientColor: skeletonGradientColor
    }));
  });
};
var lineSkeletonConfig = {
  name: 'LineSkeleton',
  tag: 'div',
  layout: lineSkeletonRoot,
  base: LineSkeleton_styles.base,
  variations: {
    size: {
      css: base.base
    }
  },
  defaults: {}
};

exports.lineSkeletonConfig = lineSkeletonConfig;
exports.lineSkeletonRoot = lineSkeletonRoot;
