// @salutejs/plasma-web/css/es/components/Skeleton/LineSkeleton.config.js
import './LineSkeleton.config_1fkfzjy.css';
var config = {
  defaults: {
    size: 'body1'
  },
  variations: {
    size: {
      body1: "b19llrr0",
      body2: "bownd2t",
      body3: "b1vlwnuz",
      button1: "b72vt9m",
      button2: "b7xroe3",
      caption: "c6j6lo5",
      display1: "dvy7k5l",
      display2: "d17f8ptj",
      display3: "d2hkci6",
      footnote1: "fytodi8",
      footnote2: "fl0wrbi",
      headline1: "hzwkeo",
      headline2: "hwfajcs",
      headline3: "h1cqdap2",
      headline4: "h7a1pe3",
      paragraph1: "pcgpodh",
      paragraph2: "p1f4baxc",
      underline: "uku2ojo",
      bodyL: "bfwhw7s",
      bodyM: "bhijpf2",
      bodyS: "b1l8tqez",
      bodyXS: "b50tsgh",
      bodyXXS: "b7s7htb",
      dsplL: "d1n6dpqc",
      dsplM: "dgf070x",
      dsplS: "d10pnghs",
      h1: "h1yg90yy",
      h2: "hc1hunq",
      h3: "hj6pc49",
      h4: "h1fm37xw",
      h5: "h1n0slms",
      textL: "tel2znt",
      textM: "tgrv5t3",
      textS: "t1wqk19",
      textXS: "t1mynjfh"
    }
  }
};

export { config };
