// @salutejs/plasma-web/components/Combobox/Combobox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ComboboxItem = exports.ComboboxHeader = exports.ComboboxGroup = exports.ComboboxFooter = exports.ComboboxDivider = exports.Combobox = void 0;
var _styledComponents = /*#__PURE__*/require("@salutejs/plasma-new-hope/styled-components");
var _Combobox = /*#__PURE__*/require("./Combobox.config");
var mergedConfig = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.comboboxConfig, _Combobox.config);
var Combobox = exports.Combobox = /*#__PURE__*/(0, _styledComponents.component)(mergedConfig);
var mergedItemConfig = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.comboboxItemConfig);
var ComboboxItem = exports.ComboboxItem = /*#__PURE__*/(0, _styledComponents.component)(mergedItemConfig);
var mergedDividerConfig = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.comboboxDividerConfig);
var ComboboxDivider = exports.ComboboxDivider = /*#__PURE__*/(0, _styledComponents.component)(mergedDividerConfig);
var mergedFooterConfig = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.comboboxFooterConfig);
var ComboboxFooter = exports.ComboboxFooter = /*#__PURE__*/(0, _styledComponents.component)(mergedFooterConfig);
var mergedGroupConfig = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.comboboxGroupConfig);
var ComboboxGroup = exports.ComboboxGroup = /*#__PURE__*/(0, _styledComponents.component)(mergedGroupConfig);
var mergedHeaderConfig = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.comboboxHeaderConfig);
var ComboboxHeader = exports.ComboboxHeader = /*#__PURE__*/(0, _styledComponents.component)(mergedHeaderConfig);
