// @salutejs/plasma-web/css/cjs/components/Calendar/Calendar.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var plasmaNewHope = require('@salutejs/plasma-new-hope');
var CalendarBase_config = require('./CalendarBase.config.js');
var CalendarDouble_config = require('./CalendarDouble.config.js');

var mergedCalendarBaseConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.calendarBaseConfig, CalendarBase_config.config);
var mergedCalendarDoubleConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.calendarDoubleConfig, CalendarDouble_config.config);
var CalendarBase = /*#__PURE__*/plasmaNewHope.component(mergedCalendarBaseConfig);
var CalendarDouble = /*#__PURE__*/plasmaNewHope.component(mergedCalendarDoubleConfig);
var CalendarBaseRange = /*#__PURE__*/plasmaNewHope.withRange(CalendarBase);
var CalendarDoubleRange = /*#__PURE__*/plasmaNewHope.withRange(CalendarDouble);
var Calendar = /*#__PURE__*/plasmaNewHope.boundCalendar({
  base: CalendarBase,
  baseRange: CalendarBaseRange,
  "double": CalendarDouble,
  doubleRange: CalendarDoubleRange
});

exports.Calendar = Calendar;
exports.CalendarBase = CalendarBase;
exports.CalendarBaseRange = CalendarBaseRange;
exports.CalendarDouble = CalendarDouble;
exports.CalendarDoubleRange = CalendarDoubleRange;
