// @salutejs/plasma-tokens-b2b/es/colors/values.js
// Generated by robots, do not change this manually!
/** Базовый белый цвет совпадает с whitePrimary */
export var white = '#FFFFFF';
/** Первичный белый цвет */
export var whitePrimary = '#FFFFFF';
/** Вторичный белый цвет */
export var whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный белый цвет */
export var whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** Базовый черный цвет совпадает с blackPrimary */
export var black = '#080808';
/** Первичный черный цвет */
export var blackPrimary = '#080808';
/** Вторичный черныйцвет */
export var blackSecondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный черный цвет */
export var blackTertiary = 'rgba(8, 8, 8, 0.28)';
/** Чёрный непрозрачный, не зависит от темы */
export var dark01 = '#171717';
/** Чёрный непрозрачный, не зависит от темы */
export var dark02 = '#232323';
/** Чёрный непрозрачный, не зависит от темы */
export var dark03 = '#363636';
/** Прозрачный цвет */
export var transparent = 'rgba(0, 0, 0, 0)';
/** Второстепенная прозрачная */
export var buttonClear = 'rgba(0, 0, 0, 0)';
/** Дефолтная чёрная, не зависит от темы */
export var buttonBlack = '#080808';
/** Второстепенная чёрная, не зависит от темы */
export var buttonBlackSecondary = 'rgba(8, 8, 8, 0.12)';
/** Чёрная для использования поверх картинок */
export var buttonBlackTransparent = 'rgba(8, 8, 8, 0.56)';
/** Дефолтная белая, не зависит от темы */
export var buttonWhite = '#FFFFFF';
/** Второстепенная белая, не зависит от темы */
export var buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** Базовый цвет текста, совпадает с primary */
export var text = '#080808';
export var link = '#2A72F8';
export var linkHover = '#4D89F9';
export var linkActive = '#0859F1';
export var linkVisited = '#2364DE';
export var linkVisitedHover = '#427AE3';
export var linkVisitedActive = '#1C54BC';
/** Основной текст в интерфейсе */
export var primary = '#080808';
/** Второстепенный текст */
export var secondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный цвет текста */
export var tertiary = 'rgba(8, 8, 8, 0.28)';
export var paragraph = 'rgba(8, 8, 8, 0.8)';
export var inverse = 'rgba(255, 255, 255, 0.96)';
/** Основной бэкграунд всех приложений и смартапов */
export var background = '#FFFFFF';
/** Первичный цвет фона */
export var backgroundPrimary = '#FFFFFF';
/** Вторичный цвет фона */
export var backgroundSecondary = '#FFFFFF';
/** Третичный цвет фона */
export var backgroundTertiary = '#FFFFFF';
export var inputBorder = 'rgba(8, 8, 8, 0.28)';
export var inputBorderHover = 'rgba(8, 8, 8, 0.56)';
export var inputBorderFocus = '#2A72F8';
/** Акцентный цвет бренда */
export var accent = '#2A72F8';
/** Обозначение успешного сценария */
export var success = '#118C28';
/** Цвет предупреждения */
export var warning = '#E35502';
/** Цвет ошибки */
export var critical = '#F31B30';
/** Полупрозрачные поверхности */
export var surfaceLiquid01 = 'rgba(8, 8, 8, 0.02)';
/** Полупрозрачные поверхности */
export var surfaceLiquid02 = 'rgba(8, 8, 8, 0.06)';
/** Полупрозрачные поверхности */
export var surfaceLiquid03 = 'rgba(8, 8, 8, 0.12)';
/** Непрозрачные поверхности */
export var surfaceSolid01 = '#FAFAFA';
/** Непрозрачные поверхности */
export var surfaceSolid02 = '#F0F0F0';
/** Непрозрачные поверхности */
export var surfaceSolid03 = '#E1E1E1';
/** Основной фон для карточек */
export var surfaceCard = '#FFFFFF';
/** Первичный цвет контролов */
export var buttonPrimary = '#2A72F8';
export var buttonPrimaryHover = '#397CF8';
export var buttonPrimaryActive = '#1B68F8';
/** Второстепенная кнопка */
export var buttonSecondary = 'rgba(8, 8, 8, 0.06)';
export var buttonSecondaryHover = 'rgba(8, 8, 8, 0.04)';
export var buttonSecondaryActive = 'rgba(8, 8, 8, 0.08)';
/** Акцентный цвет у контролов */
export var buttonAccent = '#2A72F8';
/** Кнопка для успешного сценария */
export var buttonSuccess = '#118C28';
export var buttonSuccessHover = '#3eaa4d';
export var buttonSuccessActive = '#1d9233';
/** Цвет предупреждения у контролов */
export var buttonWarning = '#F55D05';
export var buttonWarningHover = '#FA650F';
export var buttonWarningActive = '#E65705';
/** Цвет ошибки у контролов */
export var buttonCritical = '#F31B30';
export var buttonCriticalHover = '#F42A3E';
export var buttonCriticalActive = '#F20D24';
/** Цвет зажатого контрола */
export var buttonChecked = '#080808';
export var buttonCheckedHover = '#171717';
export var buttonCheckedHoverColor = 'hsla(0, 0%, 100%, 0.9599999785423279)';
export var buttonCheckedActive = '#080808';
export var buttonCheckedActiveColor = 'rgba(255, 255, 255, 0.6)';
/** Цвет рамки фокуса у контрола */
export var buttonFocused = '#2A72F8';
/** Градиент для заливки основного фона */
export var gradient = 'linear-gradient(45deg, hsla(220.1123595505618, 85.57692307692307%, 59.21568627450981%, 1) 0%, hsla(189.78947368421052, 78.51239669421487%, 52.54901960784314%, 1) 100%)';
export var gradientDevice = '';
export var voicePhraseGradient = '';
export var skeletonGradient = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.09) 0%, rgba(8, 8, 8, 0.08) 6.25%, rgba(8, 8, 8, 0.05) 12.5%, rgba(8, 8, 8, 0.01) 25%, rgba(8, 8, 8, 0.05) 37.5%, rgba(8, 8, 8, 0.08) 43.75%, rgba(8, 8, 8, 0.09) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.05) 62.5%, rgba(8, 8, 8, 0.01) 75%, rgba(8, 8, 8, 0.05) 87.5%, rgba(8, 8, 8, 0.08) 93.75%, rgba(8, 8, 8, 0.09) 100% )';
export var skeletonGradientLighter = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.36) 0%, rgba(8, 8, 8, 0.32) 6.25%, rgba(8, 8, 8, 0.20) 12.5%, rgba(8, 8, 8, 0.04) 25%, rgba(8, 8, 8, 0.20) 37.5%, rgba(8, 8, 8, 0.32) 43.75%, rgba(8, 8, 8, 0.36) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.20) 62.5%, rgba(8, 8, 8, 0.04) 75%, rgba(8, 8, 8, 0.20) 87.5%, rgba(8, 8, 8, 0.32) 93.75%, rgba(8, 8, 8, 0.36) 100% )';
export var speechBubbleSent = '';
export var speechBubbleReceived = '';
/** Цвет фона паранжи */
export var overlay = 'rgba(0, 0, 0, 0.8)';
export var overlaySoft = 'rgba(8, 8, 8, 0.56)';
export var overlayHard = 'rgba(8, 8, 8, 0.9)';
export var overlayBlur = 'rgba(35, 35, 35, 0.2)';
export var darkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var darkOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var darkOverlayBlur = 'rgba(35, 35, 35, 0.2)';
export var lightOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var lightOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var lightOverlayBlur = 'rgba(35, 35, 35, 0.2)';
export var inverseOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var inverseOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var inverseOverlayBlur = 'rgba(35, 35, 35, 0.2)';
