// @salutejs/plasma-new-hope/styled-components/cjs/examples/plasma_web/components/Combobox/Combobox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ComboboxItem = exports.ComboboxHeader = exports.ComboboxGroup = exports.ComboboxFooter = exports.ComboboxDivider = exports.Combobox = void 0;
var _Combobox = /*#__PURE__*/require("../../../../components/Combobox");
var _engines = /*#__PURE__*/require("../../../../engines");
var _Combobox2 = /*#__PURE__*/require("./Combobox.config");
var mergedConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Combobox.comboboxConfig, _Combobox2.config);
var Combobox = exports.Combobox = /*#__PURE__*/(0, _engines.component)(mergedConfig);
var mergedItemConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Combobox.comboboxItemConfig);
var ComboboxItem = exports.ComboboxItem = /*#__PURE__*/(0, _engines.component)(mergedItemConfig);
var mergedDividerConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Combobox.comboboxDividerConfig);
var ComboboxDivider = exports.ComboboxDivider = /*#__PURE__*/(0, _engines.component)(mergedDividerConfig);
var mergedFooterConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Combobox.comboboxFooterConfig);
var ComboboxFooter = exports.ComboboxFooter = /*#__PURE__*/(0, _engines.component)(mergedFooterConfig);
var mergedGroupConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Combobox.comboboxGroupConfig);
var ComboboxGroup = exports.ComboboxGroup = /*#__PURE__*/(0, _engines.component)(mergedGroupConfig);
var mergedHeaderConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Combobox.comboboxHeaderConfig);
var ComboboxHeader = exports.ComboboxHeader = /*#__PURE__*/(0, _engines.component)(mergedHeaderConfig);
