// @salutejs/plasma-new-hope/styled-components/cjs/components/Header/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "HeaderContent", {
  enumerable: true,
  get: function get() {
    return _HeaderContent.HeaderContent;
  }
});
Object.defineProperty(exports, "HeaderTitle", {
  enumerable: true,
  get: function get() {
    return _ui.HeaderTitle;
  }
});
Object.defineProperty(exports, "headerArrowConfig", {
  enumerable: true,
  get: function get() {
    return _ui.headerArrowConfig;
  }
});
Object.defineProperty(exports, "headerClasses", {
  enumerable: true,
  get: function get() {
    return _Header.classes;
  }
});
Object.defineProperty(exports, "headerLogoConfig", {
  enumerable: true,
  get: function get() {
    return _ui.headerLogoConfig;
  }
});
Object.defineProperty(exports, "headerRootConfig", {
  enumerable: true,
  get: function get() {
    return _HeaderRoot.headerRootConfig;
  }
});
Object.defineProperty(exports, "headerTitleWrapperConfig", {
  enumerable: true,
  get: function get() {
    return _ui.headerTitleWrapperConfig;
  }
});
Object.defineProperty(exports, "headerTokens", {
  enumerable: true,
  get: function get() {
    return _Header.tokens;
  }
});
var _ui = /*#__PURE__*/require("./ui");
var _HeaderContent = /*#__PURE__*/require("./HeaderContent/HeaderContent");
var _HeaderRoot = /*#__PURE__*/require("./HeaderRoot/HeaderRoot");
var _Header = /*#__PURE__*/require("./Header.tokens");
