// @salutejs/plasma-new-hope/styled-components/cjs/components/Slider/components/SliderBase/SliderBase.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SliderBase = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _hooks = /*#__PURE__*/require("../../../../hooks");
var _SliderBase = /*#__PURE__*/require("./SliderBase.styles");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
var SliderBase = exports.SliderBase = function SliderBase(_ref) {
  var max = _ref.max,
    min = _ref.min,
    setStepSize = _ref.setStepSize,
    railFillWidth = _ref.railFillWidth,
    children = _ref.children,
    _ref$railFillXPositio = _ref.railFillXPosition,
    railFillXPosition = _ref$railFillXPositio === void 0 ? 0 : _ref$railFillXPositio,
    disabled = _ref.disabled,
    labelPlacement = _ref.labelPlacement,
    rangeValuesPlacement = _ref.rangeValuesPlacement,
    onChange = _ref.onChange,
    _ref$settings = _ref.settings,
    settings = _ref$settings === void 0 ? {} : _ref$settings;
  var _settings$indent = settings.indent,
    indent = _settings$indent === void 0 ? 0.75 : _settings$indent,
    _settings$fontSizeMul = settings.fontSizeMultiplier,
    fontSizeMultiplier = _settings$fontSizeMul === void 0 ? 16 : _settings$fontSizeMul;
  var ref = (0, _react.useRef)(null);
  var gap = indent * fontSizeMultiplier * 2;
  (0, _react.useEffect)(function () {
    var resizeHandler = function resizeHandler() {
      if (ref.current) {
        var railSize = ref.current.offsetWidth - gap;
        var totalSteps = max - min;
        setStepSize(railSize / totalSteps);
      }
    };
    resizeHandler();
  }, [labelPlacement, rangeValuesPlacement, ref.current]);
  var onHandleChange = function onHandleChange(e) {
    if (!onChange || disabled) {
      return;
    }
    var _e$currentTarget$getB = e.currentTarget.getBoundingClientRect(),
      x = _e$currentTarget$getB.x,
      width = _e$currentTarget$getB.width;
    var lastX = e.clientX - x;
    var position = min + lastX / (width - gap) * (max - min);
    var result = Math.max(min, Math.min(max, position));
    onChange(result, {
      lastX: lastX
    });
  };
  (0, _hooks.useIsomorphicLayoutEffect)(function () {
    var resizeHandler = function resizeHandler() {
      if (ref.current) {
        var railSize = ref.current.offsetWidth - gap;
        var totalSteps = max - min;
        setStepSize(railSize / totalSteps);
      }
    };
    resizeHandler();
    window.addEventListener('resize', resizeHandler);
    return function () {
      return window.removeEventListener('resize', resizeHandler);
    };
  }, [min, max, setStepSize, ref.current, gap, labelPlacement, rangeValuesPlacement]);
  var fillStyle = {
    left: "".concat(railFillXPosition, "px"),
    width: "".concat(railFillWidth, "px")
  };
  return /*#__PURE__*/_react["default"].createElement(_SliderBase.Slider, {
    ref: ref
  }, /*#__PURE__*/_react["default"].createElement(_SliderBase.RailWrap, {
    "aria-hidden": "true",
    onMouseDown: onHandleChange
  }, /*#__PURE__*/_react["default"].createElement(_SliderBase.Rail, null, /*#__PURE__*/_react["default"].createElement(_SliderBase.Fill, {
    style: fillStyle
  }))), children);
};
