// @salutejs/plasma-tokens/brands/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.onDarkTextTertiaryHover = exports.onDarkTextTertiary = exports.onDarkTextSecondaryActive = exports.onDarkTextSecondaryHover = exports.onDarkTextSecondary = exports.onDarkTextPrimaryActive = exports.onDarkTextPrimaryHover = exports.onDarkTextPrimary = exports.textInfoMinorActive = exports.textInfoMinorHover = exports.textInfoMinor = exports.textNegativeMinorActive = exports.textNegativeMinorHover = exports.textNegativeMinor = exports.textWarningMinorActive = exports.textWarningMinorHover = exports.textWarningMinor = exports.textPositiveMinorActive = exports.textPositiveMinorHover = exports.textPositiveMinor = exports.textInfoActive = exports.textInfoHover = exports.textInfo = exports.textNegativeActive = exports.textNegativeHover = exports.textNegative = exports.textWarningActive = exports.textWarningHover = exports.textWarning = exports.textPositiveActive = exports.textPositiveHover = exports.textPositive = exports.textAccentMinorActive = exports.textAccentMinorHover = exports.textAccentMinor = exports.textAccentActive = exports.textAccentHover = exports.textAccent = exports.textParagraphActive = exports.textParagraphHover = exports.textParagraph = exports.textTertiaryActive = exports.textTertiaryHover = exports.textTertiary = exports.textSecondaryActive = exports.textSecondaryHover = exports.textSecondary = exports.textPrimaryActive = exports.textPrimaryHover = exports.textPrimary = void 0;
exports.onLightTextAccentMinor = exports.onLightTextAccentActive = exports.onLightTextAccentHover = exports.onLightTextAccent = exports.onLightTextParagraphActive = exports.onLightTextParagraphHover = exports.onLightTextParagraph = exports.onLightTextTertiaryActive = exports.onLightTextTertiaryHover = exports.onLightTextTertiary = exports.onLightTextSecondaryActive = exports.onLightTextSecondaryHover = exports.onLightTextSecondary = exports.onLightTextPrimaryActive = exports.onLightTextPrimaryHover = exports.onLightTextPrimary = exports.onDarkTextInfoMinorActive = exports.onDarkTextInfoMinorHover = exports.onDarkTextInfoMinor = exports.onDarkTextNegativeMinorActive = exports.onDarkTextNegativeMinorHover = exports.onDarkTextNegativeMinor = exports.onDarkTextWarningMinorActive = exports.onDarkTextWarningMinorHover = exports.onDarkTextWarningMinor = exports.onDarkTextPositiveMinorActive = exports.onDarkTextPositiveMinorHover = exports.onDarkTextPositiveMinor = exports.onDarkTextInfoActive = exports.onDarkTextInfoHover = exports.onDarkTextInfo = exports.onDarkTextNegativeActive = exports.onDarkTextNegativeHover = exports.onDarkTextNegative = exports.onDarkTextWarningActive = exports.onDarkTextWarningHover = exports.onDarkTextWarning = exports.onDarkTextPositiveActive = exports.onDarkTextPositiveHover = exports.onDarkTextPositive = exports.onDarkTextAccentMinorActive = exports.onDarkTextAccentMinorHover = exports.onDarkTextAccentMinor = exports.onDarkTextAccentActive = exports.onDarkTextAccentHover = exports.onDarkTextAccent = exports.onDarkTextParagraphActive = exports.onDarkTextParagraphHover = exports.onDarkTextParagraph = exports.onDarkTextTertiaryActive = void 0;
exports.inverseTextWarningActive = exports.inverseTextWarningHover = exports.inverseTextWarning = exports.inverseTextPositiveActive = exports.inverseTextPositiveHover = exports.inverseTextPositive = exports.inverseTextAccentMinorActive = exports.inverseTextAccentMinorHover = exports.inverseTextAccentMinor = exports.inverseTextAccentActive = exports.inverseTextAccentHover = exports.inverseTextAccent = exports.inverseTextParagraphActive = exports.inverseTextParagraphHover = exports.inverseTextParagraph = exports.inverseTextTertiaryActive = exports.inverseTextTertiaryHover = exports.inverseTextTertiary = exports.inverseTextSecondaryActive = exports.inverseTextSecondaryHover = exports.inverseTextSecondary = exports.inverseTextPrimaryActive = exports.inverseTextPrimaryHover = exports.inverseTextPrimary = exports.onLightTextInfoMinorActive = exports.onLightTextInfoMinorHover = exports.onLightTextInfoMinor = exports.onLightTextNegativeMinorActive = exports.onLightTextNegativeMinorHover = exports.onLightTextNegativeMinor = exports.onLightTextWarningMinorActive = exports.onLightTextWarningMinorHover = exports.onLightTextWarningMinor = exports.onLightTextPositiveMinorActive = exports.onLightTextPositiveMinorHover = exports.onLightTextPositiveMinor = exports.onLightTextInfoActive = exports.onLightTextInfoHover = exports.onLightTextInfo = exports.onLightTextNegativeActive = exports.onLightTextNegativeHover = exports.onLightTextNegative = exports.onLightTextWarningActive = exports.onLightTextWarningHover = exports.onLightTextWarning = exports.onLightTextPositiveActive = exports.onLightTextPositiveHover = exports.onLightTextPositive = exports.onLightTextAccentMinorActive = exports.onLightTextAccentMinorHover = void 0;
exports.surfaceClearHover = exports.surfaceClear = exports.surfaceTransparentCardActive = exports.surfaceTransparentCardHover = exports.surfaceTransparentCard = exports.surfaceTransparentDeepActive = exports.surfaceTransparentDeepHover = exports.surfaceTransparentDeep = exports.surfaceTransparentTertiaryActive = exports.surfaceTransparentTertiaryHover = exports.surfaceTransparentTertiary = exports.surfaceTransparentSecondaryActive = exports.surfaceTransparentSecondaryHover = exports.surfaceTransparentSecondary = exports.surfaceTransparentPrimaryActive = exports.surfaceTransparentPrimaryHover = exports.surfaceTransparentPrimary = exports.surfaceSolidDefaultActive = exports.surfaceSolidDefaultHover = exports.surfaceSolidDefault = exports.surfaceSolidCardActive = exports.surfaceSolidCardHover = exports.surfaceSolidCard = exports.surfaceSolidTertiaryActive = exports.surfaceSolidTertiaryHover = exports.surfaceSolidTertiary = exports.surfaceSolidSecondaryActive = exports.surfaceSolidSecondaryHover = exports.surfaceSolidSecondary = exports.surfaceSolidPrimaryActive = exports.surfaceSolidPrimaryHover = exports.surfaceSolidPrimary = exports.inverseTextInfoMinorActive = exports.inverseTextInfoMinorHover = exports.inverseTextInfoMinor = exports.inverseTextNegativeMinorActive = exports.inverseTextNegativeMinorHover = exports.inverseTextNegativeMinor = exports.inverseTextWarningMinorActive = exports.inverseTextWarningMinorHover = exports.inverseTextWarningMinor = exports.inverseTextPositiveMinorActive = exports.inverseTextPositiveMinorHover = exports.inverseTextPositiveMinor = exports.inverseTextInfoActive = exports.inverseTextInfoHover = exports.inverseTextInfo = exports.inverseTextNegativeActive = exports.inverseTextNegativeHover = exports.inverseTextNegative = void 0;
exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidPrimaryActive = exports.onDarkSurfaceSolidPrimaryHover = exports.onDarkSurfaceSolidPrimary = exports.surfaceTransparentInfoActive = exports.surfaceTransparentInfoHover = exports.surfaceTransparentInfo = exports.surfaceTransparentNegativeActive = exports.surfaceTransparentNegativeHover = exports.surfaceTransparentNegative = exports.surfaceTransparentWarningActive = exports.surfaceTransparentWarningHover = exports.surfaceTransparentWarning = exports.surfaceTransparentPositiveActive = exports.surfaceTransparentPositiveHover = exports.surfaceTransparentPositive = exports.surfaceInfoMinorActive = exports.surfaceInfoMinorHover = exports.surfaceInfoMinor = exports.surfaceNegativeMinorActive = exports.surfaceNegativeMinorHover = exports.surfaceNegativeMinor = exports.surfaceWarningMinorActive = exports.surfaceWarningMinorHover = exports.surfaceWarningMinor = exports.surfacePositiveMinorActive = exports.surfacePositiveMinorHover = exports.surfacePositiveMinor = exports.surfaceInfoActive = exports.surfaceInfoHover = exports.surfaceInfo = exports.surfaceNegativeActive = exports.surfaceNegativeHover = exports.surfaceNegative = exports.surfaceWarningActive = exports.surfaceWarningHover = exports.surfaceWarning = exports.surfacePositiveActive = exports.surfacePositiveHover = exports.surfacePositive = exports.surfaceTransparentAccentActive = exports.surfaceTransparentAccentHover = exports.surfaceTransparentAccent = exports.surfaceAccentMinorActive = exports.surfaceAccentMinorHover = exports.surfaceAccentMinor = exports.surfaceAccentActive = exports.surfaceAccentHover = exports.surfaceAccent = exports.surfaceClearActive = void 0;
exports.onDarkSurfaceInfoActive = exports.onDarkSurfaceInfoHover = exports.onDarkSurfaceInfo = exports.onDarkSurfaceNegativeActive = exports.onDarkSurfaceNegativeHover = exports.onDarkSurfaceNegative = exports.onDarkSurfaceWarningActive = exports.onDarkSurfaceWarningHover = exports.onDarkSurfaceWarning = exports.onDarkSurfacePositiveActive = exports.onDarkSurfacePositiveHover = exports.onDarkSurfacePositive = exports.onDarkSurfaceTransparentAccentActive = exports.onDarkSurfaceTransparentAccentHover = exports.onDarkSurfaceTransparentAccent = exports.onDarkSurfaceAccentMinorActive = exports.onDarkSurfaceAccentMinorHover = exports.onDarkSurfaceAccentMinor = exports.onDarkSurfaceAccentActive = exports.onDarkSurfaceAccentHover = exports.onDarkSurfaceAccent = exports.onDarkSurfaceClearActive = exports.onDarkSurfaceClearHover = exports.onDarkSurfaceClear = exports.onDarkSurfaceTransparentCardActive = exports.onDarkSurfaceTransparentCardHover = exports.onDarkSurfaceTransparentCard = exports.onDarkSurfaceTransparentDeepActive = exports.onDarkSurfaceTransparentDeepHover = exports.onDarkSurfaceTransparentDeep = exports.onDarkSurfaceTransparentTertiaryActive = exports.onDarkSurfaceTransparentTertiaryHover = exports.onDarkSurfaceTransparentTertiary = exports.onDarkSurfaceTransparentSecondaryActive = exports.onDarkSurfaceTransparentSecondaryHover = exports.onDarkSurfaceTransparentSecondary = exports.onDarkSurfaceTransparentPrimaryActive = exports.onDarkSurfaceTransparentPrimaryHover = exports.onDarkSurfaceTransparentPrimary = exports.onDarkSurfaceSolidDefaultActive = exports.onDarkSurfaceSolidDefaultHover = exports.onDarkSurfaceSolidDefault = exports.onDarkSurfaceSolidCardActive = exports.onDarkSurfaceSolidCardHover = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidTertiaryActive = exports.onDarkSurfaceSolidTertiaryHover = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidSecondaryActive = exports.onDarkSurfaceSolidSecondaryHover = void 0;
exports.onLightSurfaceTransparentDeepHover = exports.onLightSurfaceTransparentDeep = exports.onLightSurfaceTransparentTertiaryActive = exports.onLightSurfaceTransparentTertiaryHover = exports.onLightSurfaceTransparentTertiary = exports.onLightSurfaceTransparentSecondaryActive = exports.onLightSurfaceTransparentSecondaryHover = exports.onLightSurfaceTransparentSecondary = exports.onLightSurfaceTransparentPrimaryActive = exports.onLightSurfaceTransparentPrimaryHover = exports.onLightSurfaceTransparentPrimary = exports.onLightSurfaceSolidDefaultActive = exports.onLightSurfaceSolidDefaultHover = exports.onLightSurfaceSolidDefault = exports.onLightSurfaceSolidCardActive = exports.onLightSurfaceSolidCardHover = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidTertiaryActive = exports.onLightSurfaceSolidTertiaryHover = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidSecondaryActive = exports.onLightSurfaceSolidSecondaryHover = exports.onLightSurfaceSolidSecondary = exports.onLightSurfaceSolidPrimaryActive = exports.onLightSurfaceSolidPrimaryHover = exports.onLightSurfaceSolidPrimary = exports.onDarkSurfaceTransparentInfoActive = exports.onDarkSurfaceTransparentInfoHover = exports.onDarkSurfaceTransparentInfo = exports.onDarkSurfaceTransparentNegativeActive = exports.onDarkSurfaceTransparentNegativeHover = exports.onDarkSurfaceTransparentNegative = exports.onDarkSurfaceTransparentWarningActive = exports.onDarkSurfaceTransparentWarningHover = exports.onDarkSurfaceTransparentWarning = exports.onDarkSurfaceTransparentPositiveActive = exports.onDarkSurfaceTransparentPositiveHover = exports.onDarkSurfaceTransparentPositive = exports.onDarkSurfaceInfoMinorActive = exports.onDarkSurfaceInfoMinorHover = exports.onDarkSurfaceInfoMinor = exports.onDarkSurfaceNegativeMinorActive = exports.onDarkSurfaceNegativeMinorHover = exports.onDarkSurfaceNegativeMinor = exports.onDarkSurfaceWarningMinorActive = exports.onDarkSurfaceWarningMinorHover = exports.onDarkSurfaceWarningMinor = exports.onDarkSurfacePositiveMinorActive = exports.onDarkSurfacePositiveMinorHover = exports.onDarkSurfacePositiveMinor = void 0;
exports.onLightSurfaceTransparentInfo = exports.onLightSurfaceTransparentNegativeActive = exports.onLightSurfaceTransparentNegativeHover = exports.onLightSurfaceTransparentNegative = exports.onLightSurfaceTransparentWarningActive = exports.onLightSurfaceTransparentWarningHover = exports.onLightSurfaceTransparentWarning = exports.onLightSurfaceTransparentPositiveActive = exports.onLightSurfaceTransparentPositiveHover = exports.onLightSurfaceTransparentPositive = exports.onLightSurfaceInfoMinorActive = exports.onLightSurfaceInfoMinorHover = exports.onLightSurfaceInfoMinor = exports.onLightSurfaceNegativeMinorActive = exports.onLightSurfaceNegativeMinorHover = exports.onLightSurfaceNegativeMinor = exports.onLightSurfaceWarningMinorActive = exports.onLightSurfaceWarningMinorHover = exports.onLightSurfaceWarningMinor = exports.onLightSurfacePositiveMinorActive = exports.onLightSurfacePositiveMinorHover = exports.onLightSurfacePositiveMinor = exports.onLightSurfaceInfoActive = exports.onLightSurfaceInfoHover = exports.onLightSurfaceInfo = exports.onLightSurfaceNegativeActive = exports.onLightSurfaceNegativeHover = exports.onLightSurfaceNegative = exports.onLightSurfaceWarningActive = exports.onLightSurfaceWarningHover = exports.onLightSurfaceWarning = exports.onLightSurfacePositiveActive = exports.onLightSurfacePositiveHover = exports.onLightSurfacePositive = exports.onLightSurfaceTransparentAccentActive = exports.onLightSurfaceTransparentAccentHover = exports.onLightSurfaceTransparentAccent = exports.onLightSurfaceAccentMinorActive = exports.onLightSurfaceAccentMinorHover = exports.onLightSurfaceAccentMinor = exports.onLightSurfaceAccentActive = exports.onLightSurfaceAccentHover = exports.onLightSurfaceAccent = exports.onLightSurfaceClearActive = exports.onLightSurfaceClearHover = exports.onLightSurfaceClear = exports.onLightSurfaceTransparentCardActive = exports.onLightSurfaceTransparentCardHover = exports.onLightSurfaceTransparentCard = exports.onLightSurfaceTransparentDeepActive = void 0;
exports.inverseSurfaceWarningActive = exports.inverseSurfaceWarningHover = exports.inverseSurfaceWarning = exports.inverseSurfacePositiveActive = exports.inverseSurfacePositiveHover = exports.inverseSurfacePositive = exports.inverseSurfaceTransparentAccentActive = exports.inverseSurfaceTransparentAccentHover = exports.inverseSurfaceTransparentAccent = exports.inverseSurfaceAccentMinorActive = exports.inverseSurfaceAccentMinorHover = exports.inverseSurfaceAccentMinor = exports.inverseSurfaceAccentActive = exports.inverseSurfaceAccentHover = exports.inverseSurfaceAccent = exports.inverseSurfaceClearActive = exports.inverseSurfaceClearHover = exports.inverseSurfaceClear = exports.inverseSurfaceTransparentCardActive = exports.inverseSurfaceTransparentCardHover = exports.inverseSurfaceTransparentCard = exports.inverseSurfaceTransparentDeepActive = exports.inverseSurfaceTransparentDeepHover = exports.inverseSurfaceTransparentDeep = exports.inverseSurfaceTransparentTertiaryActive = exports.inverseSurfaceTransparentTertiaryHover = exports.inverseSurfaceTransparentTertiary = exports.inverseSurfaceTransparentSecondaryActive = exports.inverseSurfaceTransparentSecondaryHover = exports.inverseSurfaceTransparentSecondary = exports.inverseSurfaceTransparentPrimaryActive = exports.inverseSurfaceTransparentPrimaryHover = exports.inverseSurfaceTransparentPrimary = exports.inverseSurfaceSolidDefaultActive = exports.inverseSurfaceSolidDefaultHover = exports.inverseSurfaceSolidDefault = exports.inverseSurfaceSolidCardActive = exports.inverseSurfaceSolidCardHover = exports.inverseSurfaceSolidCard = exports.inverseSurfaceSolidTertiaryActive = exports.inverseSurfaceSolidTertiaryHover = exports.inverseSurfaceSolidTertiary = exports.inverseSurfaceSolidSecondaryActive = exports.inverseSurfaceSolidSecondaryHover = exports.inverseSurfaceSolidSecondary = exports.inverseSurfaceSolidPrimaryActive = exports.inverseSurfaceSolidPrimaryHover = exports.inverseSurfaceSolidPrimary = exports.onLightSurfaceTransparentInfoActive = exports.onLightSurfaceTransparentInfoHover = void 0;
exports.outlineDefaultOutlineSolidSecondary = exports.outlineDefaultOutlineSolidPrimaryActive = exports.outlineDefaultOutlineSolidPrimaryHover = exports.outlineDefaultOutlineSolidPrimary = exports.inverseOverlayBlur = exports.inverseOverlayHard = exports.inverseOverlaySoft = exports.onLightOverlayBlur = exports.onLightOverlayHard = exports.onLightOverlaySoft = exports.onDarkOverlayBlur = exports.onDarkOverlayHard = exports.onDarkOverlaySoft = exports.overlayBlur = exports.overlayHard = exports.overlaySoft = exports.inverseBackgroundPrimary = exports.lightBackgroundPrimary = exports.darkBackgroundPrimary = exports.backgroundPrimary = exports.inverseSurfaceTransparentInfoActive = exports.inverseSurfaceTransparentInfoHover = exports.inverseSurfaceTransparentInfo = exports.inverseSurfaceTransparentNegativeActive = exports.inverseSurfaceTransparentNegativeHover = exports.inverseSurfaceTransparentNegative = exports.inverseSurfaceTransparentWarningActive = exports.inverseSurfaceTransparentWarningHover = exports.inverseSurfaceTransparentWarning = exports.inverseSurfaceTransparentPositiveActive = exports.inverseSurfaceTransparentPositiveHover = exports.inverseSurfaceTransparentPositive = exports.inverseSurfaceInfoMinorActive = exports.inverseSurfaceInfoMinorHover = exports.inverseSurfaceInfoMinor = exports.inverseSurfaceNegativeMinorActive = exports.inverseSurfaceNegativeMinorHover = exports.inverseSurfaceNegativeMinor = exports.inverseSurfaceWarningMinorActive = exports.inverseSurfaceWarningMinorHover = exports.inverseSurfaceWarningMinor = exports.inverseSurfacePositiveMinorActive = exports.inverseSurfacePositiveMinorHover = exports.inverseSurfacePositiveMinor = exports.inverseSurfaceInfoActive = exports.inverseSurfaceInfoHover = exports.inverseSurfaceInfo = exports.inverseSurfaceNegativeActive = exports.inverseSurfaceNegativeHover = exports.inverseSurfaceNegative = void 0;
exports.outlineDefaultOutlineNegativeMinorActive = exports.outlineDefaultOutlineNegativeMinorHover = exports.outlineDefaultOutlineNegativeMinor = exports.outlineDefaultOutlineWarningMinorActive = exports.outlineDefaultOutlineWarningMinorHover = exports.outlineDefaultOutlineWarningMinor = exports.outlineDefaultOutlinePositiveMinorActive = exports.outlineDefaultOutlinePositiveMinorHover = exports.outlineDefaultOutlinePositiveMinor = exports.outlineDefaultOutlineInfoActive = exports.outlineDefaultOutlineInfoHover = exports.outlineDefaultOutlineInfo = exports.outlineDefaultOutlineNegativeActive = exports.outlineDefaultOutlineNegativeHover = exports.outlineDefaultOutlineNegative = exports.outlineDefaultOutlineWarningActive = exports.outlineDefaultOutlineWarningHover = exports.outlineDefaultOutlineWarning = exports.outlineDefaultOutlinePositiveActive = exports.outlineDefaultOutlinePositiveHover = exports.outlineDefaultOutlinePositive = exports.outlineDefaultOutlineTransparentAccentActive = exports.outlineDefaultOutlineTransparentAccentHover = exports.outlineDefaultOutlineTransparentAccent = exports.outlineDefaultOutlineAccentMinorActive = exports.outlineDefaultOutlineAccentMinorHover = exports.outlineDefaultOutlineAccentMinor = exports.outlineDefaultOutlineAccentActive = exports.outlineDefaultOutlineAccentHover = exports.outlineDefaultOutlineAccent = exports.outlineDefaultOutlineClearActive = exports.outlineDefaultOutlineClearHover = exports.outlineDefaultOutlineClear = exports.outlineDefaultOutlineTransparentTertiaryActive = exports.outlineDefaultOutlineTransparentTertiaryHover = exports.outlineDefaultOutlineTransparentTertiary = exports.outlineDefaultOutlineTransparentSecondaryActive = exports.outlineDefaultOutlineTransparentSecondaryHover = exports.outlineDefaultOutlineTransparentSecondary = exports.outlineDefaultOutlineTransparentPrimaryActive = exports.outlineDefaultOutlineTransparentPrimaryHover = exports.outlineDefaultOutlineTransparentPrimary = exports.outlineDefaultOutlineTransparentDefaultActive = exports.outlineDefaultOutlineTransparentDefaultHover = exports.outlineDefaultOutlineTransparentDefault = exports.outlineDefaultOutlineSolidTertiaryActive = exports.outlineDefaultOutlineSolidTertiaryHover = exports.outlineDefaultOutlineSolidTertiary = exports.outlineDefaultOutlineSolidSecondaryActive = exports.outlineDefaultOutlineSolidSecondaryHover = void 0;
exports.outlineOnDarkOutlinePositiveHover = exports.outlineOnDarkOutlinePositive = exports.outlineOnDarkOutlineTransparentAccentActive = exports.outlineOnDarkOutlineTransparentAccentHover = exports.outlineOnDarkOutlineTransparentAccent = exports.outlineOnDarkOutlineAccentMinorActive = exports.outlineOnDarkOutlineAccentMinorHover = exports.outlineOnDarkOutlineAccentMinor = exports.outlineOnDarkOutlineAccentActive = exports.outlineOnDarkOutlineAccentHover = exports.outlineOnDarkOutlineAccent = exports.outlineOnDarkOutlineClearActive = exports.outlineOnDarkOutlineClearHover = exports.outlineOnDarkOutlineClear = exports.outlineOnDarkOutlineTransparentTertiaryActive = exports.outlineOnDarkOutlineTransparentTertiaryHover = exports.outlineOnDarkOutlineTransparentTertiary = exports.outlineOnDarkOutlineTransparentSecondaryActive = exports.outlineOnDarkOutlineTransparentSecondaryHover = exports.outlineOnDarkOutlineTransparentSecondary = exports.outlineOnDarkOutlineTransparentPrimaryActive = exports.outlineOnDarkOutlineTransparentPrimaryHover = exports.outlineOnDarkOutlineTransparentPrimary = exports.outlineOnDarkOutlineTransparentDefaultActive = exports.outlineOnDarkOutlineTransparentDefaultHover = exports.outlineOnDarkOutlineTransparentDefault = exports.outlineOnDarkOutlineSolidTertiaryActive = exports.outlineOnDarkOutlineSolidTertiaryHover = exports.outlineOnDarkOutlineSolidTertiary = exports.outlineOnDarkOutlineSolidSecondaryActive = exports.outlineOnDarkOutlineSolidSecondaryHover = exports.outlineOnDarkOutlineSolidSecondary = exports.outlineOnDarkOutlineSolidPrimaryActive = exports.outlineOnDarkOutlineSolidPrimaryHover = exports.outlineOnDarkOutlineSolidPrimary = exports.outlineDefaultOutlineTransparentInfoActive = exports.outlineDefaultOutlineTransparentInfoHover = exports.outlineDefaultOutlineTransparentInfo = exports.outlineDefaultOutlineTransparentNegativeActive = exports.outlineDefaultOutlineTransparentNegativeHover = exports.outlineDefaultOutlineTransparentNegative = exports.outlineDefaultOutlineTransparentWarningActive = exports.outlineDefaultOutlineTransparentWarningHover = exports.outlineDefaultOutlineTransparentWarning = exports.outlineDefaultOutlineTransparentPositiveActive = exports.outlineDefaultOutlineTransparentPositiveHover = exports.outlineDefaultOutlineTransparentPositive = exports.outlineDefaultOutlineInfoMinorActive = exports.outlineDefaultOutlineInfoMinorHover = exports.outlineDefaultOutlineInfoMinor = void 0;
exports.outlineOnLightOutlineTransparentSecondary = exports.outlineOnLightOutlineTransparentPrimaryActive = exports.outlineOnLightOutlineTransparentPrimaryHover = exports.outlineOnLightOutlineTransparentPrimary = exports.outlineOnLightOutlineTransparentDefaultActive = exports.outlineOnLightOutlineTransparentDefaultHover = exports.outlineOnLightOutlineTransparentDefault = exports.outlineOnLightOutlineSolidTertiaryActive = exports.outlineOnLightOutlineSolidTertiaryHover = exports.outlineOnLightOutlineSolidTertiary = exports.outlineOnLightOutlineSolidSecondaryActive = exports.outlineOnLightOutlineSolidSecondaryHover = exports.outlineOnLightOutlineSolidSecondary = exports.outlineOnLightOutlineSolidPrimaryActive = exports.outlineOnLightOutlineSolidPrimaryHover = exports.outlineOnLightOutlineSolidPrimary = exports.outlineOnDarkOutlineTransparentInfoActive = exports.outlineOnDarkOutlineTransparentInfoHover = exports.outlineOnDarkOutlineTransparentInfo = exports.outlineOnDarkOutlineTransparentNegativeActive = exports.outlineOnDarkOutlineTransparentNegativeHover = exports.outlineOnDarkOutlineTransparentNegative = exports.outlineOnDarkOutlineTransparentWarningActive = exports.outlineOnDarkOutlineTransparentWarningHover = exports.outlineOnDarkOutlineTransparentWarning = exports.outlineOnDarkOutlineTransparentPositiveActive = exports.outlineOnDarkOutlineTransparentPositiveHover = exports.outlineOnDarkOutlineTransparentPositive = exports.outlineOnDarkOutlineInfoMinorActive = exports.outlineOnDarkOutlineInfoMinorHover = exports.outlineOnDarkOutlineInfoMinor = exports.outlineOnDarkOutlineNegativeMinorActive = exports.outlineOnDarkOutlineNegativeMinorHover = exports.outlineOnDarkOutlineNegativeMinor = exports.outlineOnDarkOutlineWarningMinorActive = exports.outlineOnDarkOutlineWarningMinorHover = exports.outlineOnDarkOutlineWarningMinor = exports.outlineOnDarkOutlinePositiveMinorActive = exports.outlineOnDarkOutlinePositiveMinorHover = exports.outlineOnDarkOutlinePositiveMinor = exports.outlineOnDarkOutlineInfoActive = exports.outlineOnDarkOutlineInfoHover = exports.outlineOnDarkOutlineInfo = exports.outlineOnDarkOutlineNegativeActive = exports.outlineOnDarkOutlineNegativeHover = exports.outlineOnDarkOutlineNegative = exports.outlineOnDarkOutlineWarningActive = exports.outlineOnDarkOutlineWarningHover = exports.outlineOnDarkOutlineWarning = exports.outlineOnDarkOutlinePositiveActive = void 0;
exports.outlineOnLightOutlineTransparentNegativeActive = exports.outlineOnLightOutlineTransparentNegativeHover = exports.outlineOnLightOutlineTransparentNegative = exports.outlineOnLightOutlineTransparentWarningActive = exports.outlineOnLightOutlineTransparentWarningHover = exports.outlineOnLightOutlineTransparentWarning = exports.outlineOnLightOutlineTransparentPositiveActive = exports.outlineOnLightOutlineTransparentPositiveHover = exports.outlineOnLightOutlineTransparentPositive = exports.outlineOnLightOutlineInfoMinorActive = exports.outlineOnLightOutlineInfoMinorHover = exports.outlineOnLightOutlineInfoMinor = exports.outlineOnLightOutlineNegativeMinorActive = exports.outlineOnLightOutlineNegativeMinorHover = exports.outlineOnLightOutlineNegativeMinor = exports.outlineOnLightOutlineWarningMinorActive = exports.outlineOnLightOutlineWarningMinorHover = exports.outlineOnLightOutlineWarningMinor = exports.outlineOnLightOutlinePositiveMinorActive = exports.outlineOnLightOutlinePositiveMinorHover = exports.outlineOnLightOutlinePositiveMinor = exports.outlineOnLightOutlineInfoActive = exports.outlineOnLightOutlineInfoHover = exports.outlineOnLightOutlineInfo = exports.outlineOnLightOutlineNegativeActive = exports.outlineOnLightOutlineNegativeHover = exports.outlineOnLightOutlineNegative = exports.outlineOnLightOutlineWarningActive = exports.outlineOnLightOutlineWarningHover = exports.outlineOnLightOutlineWarning = exports.outlineOnLightOutlinePositiveActive = exports.outlineOnLightOutlinePositiveHover = exports.outlineOnLightOutlinePositive = exports.outlineOnLightOutlineTransparentAccentActive = exports.outlineOnLightOutlineTransparentAccentHover = exports.outlineOnLightOutlineTransparentAccent = exports.outlineOnLightOutlineAccentMinorActive = exports.outlineOnLightOutlineAccentMinorHover = exports.outlineOnLightOutlineAccentMinor = exports.outlineOnLightOutlineAccentActive = exports.outlineOnLightOutlineAccentHover = exports.outlineOnLightOutlineAccent = exports.outlineOnLightOutlineClearActive = exports.outlineOnLightOutlineClearHover = exports.outlineOnLightOutlineClear = exports.outlineOnLightOutlineTransparentTertiaryActive = exports.outlineOnLightOutlineTransparentTertiaryHover = exports.outlineOnLightOutlineTransparentTertiary = exports.outlineOnLightOutlineTransparentSecondaryActive = exports.outlineOnLightOutlineTransparentSecondaryHover = void 0;
exports.outlineInverseOutlinePositiveMinorHover = exports.outlineInverseOutlinePositiveMinor = exports.outlineInverseOutlineInfoActive = exports.outlineInverseOutlineInfoHover = exports.outlineInverseOutlineInfo = exports.outlineInverseOutlineNegativeActive = exports.outlineInverseOutlineNegativeHover = exports.outlineInverseOutlineNegative = exports.outlineInverseOutlineWarningActive = exports.outlineInverseOutlineWarningHover = exports.outlineInverseOutlineWarning = exports.outlineInverseOutlinePositiveActive = exports.outlineInverseOutlinePositiveHover = exports.outlineInverseOutlinePositive = exports.outlineInverseOutlineTransparentAccentActive = exports.outlineInverseOutlineTransparentAccentHover = exports.outlineInverseOutlineTransparentAccent = exports.outlineInverseOutlineAccentMinorActive = exports.outlineInverseOutlineAccentMinorHover = exports.outlineInverseOutlineAccentMinor = exports.outlineInverseOutlineAccentActive = exports.outlineInverseOutlineAccentHover = exports.outlineInverseOutlineAccent = exports.outlineInverseOutlineClearActive = exports.outlineInverseOutlineClearHover = exports.outlineInverseOutlineClear = exports.outlineInverseOutlineTransparentTertiaryActive = exports.outlineInverseOutlineTransparentTertiaryHover = exports.outlineInverseOutlineTransparentTertiary = exports.outlineInverseOutlineTransparentSecondaryActive = exports.outlineInverseOutlineTransparentSecondaryHover = exports.outlineInverseOutlineTransparentSecondary = exports.outlineInverseOutlineTransparentPrimaryActive = exports.outlineInverseOutlineTransparentPrimaryHover = exports.outlineInverseOutlineTransparentPrimary = exports.outlineInverseOutlineTransparentDefaultActive = exports.outlineInverseOutlineTransparentDefaultHover = exports.outlineInverseOutlineTransparentDefault = exports.outlineInverseOutlineSolidTertiaryActive = exports.outlineInverseOutlineSolidTertiaryHover = exports.outlineInverseOutlineSolidTertiary = exports.outlineInverseOutlineSolidSecondaryActive = exports.outlineInverseOutlineSolidSecondaryHover = exports.outlineInverseOutlineSolidSecondary = exports.outlineInverseOutlineSolidPrimaryActive = exports.outlineInverseOutlineSolidPrimaryHover = exports.outlineInverseOutlineSolidPrimary = exports.outlineOnLightOutlineTransparentInfoActive = exports.outlineOnLightOutlineTransparentInfoHover = exports.outlineOnLightOutlineTransparentInfo = void 0;
exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.overlay = exports.critical = exports.warning = exports.success = exports.accent = exports.background = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackSecondary = exports.buttonBlack = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.skeletonGradientLighter = exports.skeletonGradient = exports.outlineInverseOutlineTransparentInfoActive = exports.outlineInverseOutlineTransparentInfoHover = exports.outlineInverseOutlineTransparentInfo = exports.outlineInverseOutlineTransparentNegativeActive = exports.outlineInverseOutlineTransparentNegativeHover = exports.outlineInverseOutlineTransparentNegative = exports.outlineInverseOutlineTransparentWarningActive = exports.outlineInverseOutlineTransparentWarningHover = exports.outlineInverseOutlineTransparentWarning = exports.outlineInverseOutlineTransparentPositiveActive = exports.outlineInverseOutlineTransparentPositiveHover = exports.outlineInverseOutlineTransparentPositive = exports.outlineInverseOutlineInfoMinorActive = exports.outlineInverseOutlineInfoMinorHover = exports.outlineInverseOutlineInfoMinor = exports.outlineInverseOutlineNegativeMinorActive = exports.outlineInverseOutlineNegativeMinorHover = exports.outlineInverseOutlineNegativeMinor = exports.outlineInverseOutlineWarningMinorActive = exports.outlineInverseOutlineWarningMinorHover = exports.outlineInverseOutlineWarningMinor = exports.outlineInverseOutlinePositiveMinorActive = void 0;
exports.borderRadiusXxl = exports.borderRadiusXl = exports.borderRadiusL = exports.borderRadiusM = exports.borderRadiusS = exports.borderRadiusXs = exports.borderRadiusXxs = exports.shadowUpHardL = exports.shadowUpHardM = exports.shadowUpHardS = exports.shadowUpSoftL = exports.shadowUpSoftM = exports.shadowUpSoftS = exports.shadowDownHardL = exports.shadowDownHardM = exports.shadowDownHardS = exports.shadowDownSoftL = exports.shadowDownSoftM = exports.shadowDownSoftS = exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonAccent = exports.buttonSecondary = exports.surfaceCard = exports.surfaceSolid03 = void 0;
/** Основной цвет текста */
exports.textPrimary = 'var(--text-primary, rgba(255, 255, 255, 0.96))';
/** Основной цвет текста */
exports.textPrimaryHover = 'var(--text-primary-hover, #FFFFFF93)';
/** Основной цвет текста */
exports.textPrimaryActive = 'var(--text-primary-active, #FFFFFFC4)';
/** Вторичный цвет текста */
exports.textSecondary = 'var(--text-secondary, rgba(255, 255, 255, 0.56))';
/** Вторичный цвет текста */
exports.textSecondaryHover = 'var(--text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста */
exports.textSecondaryActive = 'var(--text-secondary-active, #FFFFFFAB)';
/** Третичный цвет текста */
exports.textTertiary = 'var(--text-tertiary, rgba(255, 255, 255, 0.28))';
/** Третичный цвет текста */
exports.textTertiaryHover = 'var(--text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста */
exports.textTertiaryActive = 'var(--text-tertiary-active, #FFFFFF56)';
/** Сплошной наборный текст */
exports.textParagraph = 'var(--text-paragraph, rgba(255, 255, 255, 0.8))';
/** Сплошной наборный текст */
exports.textParagraphHover = 'var(--text-paragraph-hover, #FFFFFF7A)';
/** Сплошной наборный текст */
exports.textParagraphActive = 'var(--text-paragraph-active, #FFFFFFA3)';
/** Акцентный цвет */
exports.textAccent = 'var(--text-accent, #1A9E32)';
/** Акцентный цвет */
exports.textAccentHover = 'var(--text-accent-hover, #1EB83AFF)';
/** Акцентный цвет */
exports.textAccentActive = 'var(--text-accent-active, #15842AFF)';
/** Акцентный минорный цвет */
exports.textAccentMinor = 'var(--text-accent-minor, #9EFAAF)';
/** Акцентный минорный цвет */
exports.textAccentMinorHover = 'var(--text-accent-minor-hover, #64F77FFF)';
/** Акцентный минорный цвет */
exports.textAccentMinorActive = 'var(--text-accent-minor-active, #81F897FF)';
/** Цвет успеха */
exports.textPositive = 'var(--text-positive, #1A9E32)';
/** Цвет успеха */
exports.textPositiveHover = 'var(--text-positive-hover, #1EB83AFF)';
/** Цвет успеха */
exports.textPositiveActive = 'var(--text-positive-active, #15842AFF)';
/** Цвет предупреждения */
exports.textWarning = 'var(--text-warning, #FA5F05)';
/** Цвет предупреждения */
exports.textWarningHover = 'var(--text-warning-hover, #FB7223FF)';
/** Цвет предупреждения */
exports.textWarningActive = 'var(--text-warning-active, #DC5304FF)';
/** Цвет ошибки */
exports.textNegative = 'var(--text-negative, #FF293E)';
/** Цвет ошибки */
exports.textNegativeHover = 'var(--text-negative-hover, #FF475AFF)';
/** Цвет ошибки */
exports.textNegativeActive = 'var(--text-negative-active, #FF0A23FF)';
/** Цвет информации */
exports.textInfo = 'var(--text-info, #118CDF)';
/** Цвет информации */
exports.textInfoHover = 'var(--text-info-hover, #50B1F2FF)';
/** Цвет информации */
exports.textInfoActive = 'var(--text-info-active, #0F7BC2FF)';
/** Минорный цвет успеха */
exports.textPositiveMinor = 'var(--text-positive-minor, #9EFAAF)';
/** Минорный цвет успеха */
exports.textPositiveMinorHover = 'var(--text-positive-minor-hover, #64F77FFF)';
/** Минорный цвет успеха */
exports.textPositiveMinorActive = 'var(--text-positive-minor-active, #81F897FF)';
/** Минорный цвет предупреждения */
exports.textWarningMinor = 'var(--text-warning-minor, #FEE2D2)';
/** Минорный цвет предупреждения */
exports.textWarningMinorHover = 'var(--text-warning-minor-hover, #FDBC96FF)';
/** Минорный цвет предупреждения */
exports.textWarningMinorActive = 'var(--text-warning-minor-active, #FDCFB4FF)';
/** Минорный цвет ошибки */
exports.textNegativeMinor = 'var(--text-negative-minor, #FFE0E3)';
/** Минорный цвет ошибки */
exports.textNegativeMinorHover = 'var(--text-negative-minor-hover, #FFA3ACFF)';
/** Минорный цвет ошибки */
exports.textNegativeMinorActive = 'var(--text-negative-minor-active, #FFC2C8FF)';
/** Минорный цвет информации */
exports.textInfoMinor = 'var(--text-info-minor, #CFECFF)';
/** Минорный цвет информации */
exports.textInfoMinorHover = 'var(--text-info-minor-hover, #94D4FFFF)';
/** Минорный цвет информации */
exports.textInfoMinorActive = 'var(--text-info-minor-active, #B3E0FFFF)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = 'var(--on-dark-text-primary, rgba(255, 255, 255, 0.96))';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimaryHover = 'var(--on-dark-text-primary-hover, #FFFFFF93)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimaryActive = 'var(--on-dark-text-primary-active, #FFFFFFC4)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = 'var(--on-dark-text-secondary, rgba(255, 255, 255, 0.56))';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondaryHover = 'var(--on-dark-text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondaryActive = 'var(--on-dark-text-secondary-active, #FFFFFFAB)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = 'var(--on-dark-text-tertiary, rgba(255, 255, 255, 0.28))';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiaryHover = 'var(--on-dark-text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiaryActive = 'var(--on-dark-text-tertiary-active, #FFFFFF56)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraph = 'var(--on-dark-text-paragraph, rgba(255, 255, 255, 0.8))';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraphHover = 'var(--on-dark-text-paragraph-hover, #FFFFFF7A)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraphActive = 'var(--on-dark-text-paragraph-active, #FFFFFFA3)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccent = 'var(--on-dark-text-accent, #1A9E32)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccentHover = 'var(--on-dark-text-accent-hover, #1EB83AFF)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccentActive = 'var(--on-dark-text-accent-active, #15842AFF)';
/** Акцентный минорный цвет на темном фоне */
exports.onDarkTextAccentMinor = 'var(--on-dark-text-accent-minor, #9EFAAF)';
/** Акцентный минорный цвет на темном фоне */
exports.onDarkTextAccentMinorHover = 'var(--on-dark-text-accent-minor-hover, #64F77FFF)';
/** Акцентный минорный цвет на темном фоне */
exports.onDarkTextAccentMinorActive = 'var(--on-dark-text-accent-minor-active, #81F897FF)';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositive = 'var(--on-dark-text-positive, #1A9E32)';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositiveHover = 'var(--on-dark-text-positive-hover, #1EB83AFF)';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositiveActive = 'var(--on-dark-text-positive-active, #15842AFF)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarning = 'var(--on-dark-text-warning, #FA5F05)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarningHover = 'var(--on-dark-text-warning-hover, #FB7223FF)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarningActive = 'var(--on-dark-text-warning-active, #DC5304FF)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegative = 'var(--on-dark-text-negative, #FF293E)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegativeHover = 'var(--on-dark-text-negative-hover, #FF475AFF)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegativeActive = 'var(--on-dark-text-negative-active, #FF0A23FF)';
/** Цвет информации на темном фоне */
exports.onDarkTextInfo = 'var(--on-dark-text-info, #118CDF)';
/** Цвет информации на темном фоне */
exports.onDarkTextInfoHover = 'var(--on-dark-text-info-hover, #50B1F2FF)';
/** Цвет информации на темном фоне */
exports.onDarkTextInfoActive = 'var(--on-dark-text-info-active, #0F7BC2FF)';
/** Минорный цвет успеха на темном фоне */
exports.onDarkTextPositiveMinor = 'var(--on-dark-text-positive-minor, #9EFAAF)';
/** Минорный цвет успеха на темном фоне */
exports.onDarkTextPositiveMinorHover = 'var(--on-dark-text-positive-minor-hover, #64F77FFF)';
/** Минорный цвет успеха на темном фоне */
exports.onDarkTextPositiveMinorActive = 'var(--on-dark-text-positive-minor-active, #81F897FF)';
/** Минорный цвет предупреждения на темном фоне */
exports.onDarkTextWarningMinor = 'var(--on-dark-text-warning-minor, #FEE2D2)';
/** Минорный цвет предупреждения на темном фоне */
exports.onDarkTextWarningMinorHover = 'var(--on-dark-text-warning-minor-hover, #FDBC96FF)';
/** Минорный цвет предупреждения на темном фоне */
exports.onDarkTextWarningMinorActive = 'var(--on-dark-text-warning-minor-active, #FDCFB4FF)';
/** Минорный цвет ошибки на темном фоне */
exports.onDarkTextNegativeMinor = 'var(--on-dark-text-negative-minor, #FFE0E3)';
/** Минорный цвет ошибки на темном фоне */
exports.onDarkTextNegativeMinorHover = 'var(--on-dark-text-negative-minor-hover, #FFA3ACFF)';
/** Минорный цвет ошибки на темном фоне */
exports.onDarkTextNegativeMinorActive = 'var(--on-dark-text-negative-minor-active, #FFC2C8FF)';
/** Минорный цвет информации на темном фоне */
exports.onDarkTextInfoMinor = 'var(--on-dark-text-info-minor, #CFECFF)';
/** Минорный цвет информации на темном фоне */
exports.onDarkTextInfoMinorHover = 'var(--on-dark-text-info-minor-hover, #94D4FFFF)';
/** Минорный цвет информации на темном фоне */
exports.onDarkTextInfoMinorActive = 'var(--on-dark-text-info-minor-active, #B3E0FFFF)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = 'var(--on-light-text-primary, #080808F5)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimaryHover = 'var(--on-light-text-primary-hover, #08080893)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimaryActive = 'var(--on-light-text-primary-active, #080808C4)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = 'var(--on-light-text-secondary, #0808088F)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondaryHover = 'var(--on-light-text-secondary-hover, #080808FF)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondaryActive = 'var(--on-light-text-secondary-active, #080808AB)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = 'var(--on-light-text-tertiary, #08080847)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiaryHover = 'var(--on-light-text-tertiary-hover, #080808FF)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiaryActive = 'var(--on-light-text-tertiary-active, #08080856)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraph = 'var(--on-light-text-paragraph, #080808CC)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraphHover = 'var(--on-light-text-paragraph-hover, #0808087A)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraphActive = 'var(--on-light-text-paragraph-active, #080808A3)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccent = 'var(--on-light-text-accent, #108E26)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccentHover = 'var(--on-light-text-accent-hover, #13AA2CFF)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccentActive = 'var(--on-light-text-accent-active, #0D731EFF)';
/** Акцентный минорный цвет на светлом фоне */
exports.onLightTextAccentMinor = 'var(--on-light-text-accent-minor, #0A2B10)';
/** Акцентный минорный цвет на светлом фоне */
exports.onLightTextAccentMinorHover = 'var(--on-light-text-accent-minor-hover, #155B22FF)';
/** Акцентный минорный цвет на светлом фоне */
exports.onLightTextAccentMinorActive = 'var(--on-light-text-accent-minor-active, #104219FF)';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositive = 'var(--on-light-text-positive, #108E26)';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositiveHover = 'var(--on-light-text-positive-hover, #13AA2CFF)';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositiveActive = 'var(--on-light-text-positive-active, #0D731EFF)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarning = 'var(--on-light-text-warning, #E85702)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarningHover = 'var(--on-light-text-warning-hover, #FD650DFF)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarningActive = 'var(--on-light-text-warning-active, #CA4B02FF)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegative = 'var(--on-light-text-negative, #F31B31)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegativeHover = 'var(--on-light-text-negative-hover, #F5384BFF)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegativeActive = 'var(--on-light-text-negative-active, #E40C22FF)';
/** Цвет информации на светлом фоне */
exports.onLightTextInfo = 'var(--on-light-text-info, #0B7ECB)';
/** Цвет информации на светлом фоне */
exports.onLightTextInfoHover = 'var(--on-light-text-info-hover, #34A7F4FF)';
/** Цвет информации на светлом фоне */
exports.onLightTextInfoActive = 'var(--on-light-text-info-active, #096CAEFF)';
/** Минорный цвет успеха на светлом фоне */
exports.onLightTextPositiveMinor = 'var(--on-light-text-positive-minor, #0A2B10)';
/** Минорный цвет успеха на светлом фоне */
exports.onLightTextPositiveMinorHover = 'var(--on-light-text-positive-minor-hover, #155B22FF)';
/** Минорный цвет успеха на светлом фоне */
exports.onLightTextPositiveMinorActive = 'var(--on-light-text-positive-minor-active, #104219FF)';
/** Минорный цвет предупреждения на светлом фоне */
exports.onLightTextWarningMinor = 'var(--on-light-text-warning-minor, #3D1D0A)';
/** Минорный цвет предупреждения на светлом фоне */
exports.onLightTextWarningMinorHover = 'var(--on-light-text-warning-minor-hover, #723613FF)';
/** Минорный цвет предупреждения на светлом фоне */
exports.onLightTextWarningMinorActive = 'var(--on-light-text-warning-minor-active, #58290EFF)';
/** Минорный цвет ошибки на светлом фоне */
exports.onLightTextNegativeMinor = 'var(--on-light-text-negative-minor, #4A0D13)';
/** Минорный цвет ошибки на светлом фоне */
exports.onLightTextNegativeMinorHover = 'var(--on-light-text-negative-minor-hover, #7E1621FF)';
/** Минорный цвет ошибки на светлом фоне */
exports.onLightTextNegativeMinorActive = 'var(--on-light-text-negative-minor-active, #64121AFF)';
/** Минорный цвет информации на светлом фоне */
exports.onLightTextInfoMinor = 'var(--on-light-text-info-minor, #0C283B)';
/** Минорный цвет информации на светлом фоне */
exports.onLightTextInfoMinorHover = 'var(--on-light-text-info-minor-hover, #174B6EFF)';
/** Минорный цвет информации на светлом фоне */
exports.onLightTextInfoMinorActive = 'var(--on-light-text-info-minor-active, #113A55FF)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimary = 'var(--inverse-text-primary, #080808F5)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimaryHover = 'var(--inverse-text-primary-hover, #08080893)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimaryActive = 'var(--inverse-text-primary-active, #080808C4)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondary = 'var(--inverse-text-secondary, #0808088F)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondaryHover = 'var(--inverse-text-secondary-hover, #080808FF)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondaryActive = 'var(--inverse-text-secondary-active, #080808AB)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiary = 'var(--inverse-text-tertiary, #08080847)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiaryHover = 'var(--inverse-text-tertiary-hover, #080808FF)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiaryActive = 'var(--inverse-text-tertiary-active, #08080856)';
/** Инвертированный сплошной наборный текст */
exports.inverseTextParagraph = 'var(--inverse-text-paragraph, #080808CC)';
/** Инвертированный сплошной наборный текст */
exports.inverseTextParagraphHover = 'var(--inverse-text-paragraph-hover, #0808087A)';
/** Инвертированный сплошной наборный текст */
exports.inverseTextParagraphActive = 'var(--inverse-text-paragraph-active, #080808A3)';
/** Инвертированный акцентный цвет */
exports.inverseTextAccent = 'var(--inverse-text-accent, #108E26)';
/** Инвертированный акцентный цвет */
exports.inverseTextAccentHover = 'var(--inverse-text-accent-hover, #13AA2CFF)';
/** Инвертированный акцентный цвет */
exports.inverseTextAccentActive = 'var(--inverse-text-accent-active, #0D731EFF)';
/** Инвертированный минорный акцентный цвет */
exports.inverseTextAccentMinor = 'var(--inverse-text-accent-minor, #0A2B10)';
/** Инвертированный минорный акцентный цвет */
exports.inverseTextAccentMinorHover = 'var(--inverse-text-accent-minor-hover, #155B22FF)';
/** Инвертированный минорный акцентный цвет */
exports.inverseTextAccentMinorActive = 'var(--inverse-text-accent-minor-active, #104219FF)';
/** Инвертированный цвет успеха */
exports.inverseTextPositive = 'var(--inverse-text-positive, #108E26)';
/** Инвертированный цвет успеха */
exports.inverseTextPositiveHover = 'var(--inverse-text-positive-hover, #13AA2CFF)';
/** Инвертированный цвет успеха */
exports.inverseTextPositiveActive = 'var(--inverse-text-positive-active, #0D731EFF)';
/** Инвертированный цвет предупреждения */
exports.inverseTextWarning = 'var(--inverse-text-warning, #E85702)';
/** Инвертированный цвет предупреждения */
exports.inverseTextWarningHover = 'var(--inverse-text-warning-hover, #FD650DFF)';
/** Инвертированный цвет предупреждения */
exports.inverseTextWarningActive = 'var(--inverse-text-warning-active, #CA4B02FF)';
/** Инвертированный цвет ошибки */
exports.inverseTextNegative = 'var(--inverse-text-negative, #F31B31)';
/** Инвертированный цвет ошибки */
exports.inverseTextNegativeHover = 'var(--inverse-text-negative-hover, #F5384BFF)';
/** Инвертированный цвет ошибки */
exports.inverseTextNegativeActive = 'var(--inverse-text-negative-active, #E40C22FF)';
/** Инвертированный цвет информации */
exports.inverseTextInfo = 'var(--inverse-text-info, #0B7ECB)';
/** Инвертированный цвет информации */
exports.inverseTextInfoHover = 'var(--inverse-text-info-hover, #34A7F4FF)';
/** Инвертированный цвет информации */
exports.inverseTextInfoActive = 'var(--inverse-text-info-active, #096CAEFF)';
/** Инвертированный минорный цвет успеха */
exports.inverseTextPositiveMinor = 'var(--inverse-text-positive-minor, #0A2B10)';
/** Инвертированный минорный цвет успеха */
exports.inverseTextPositiveMinorHover = 'var(--inverse-text-positive-minor-hover, #155B22FF)';
/** Инвертированный минорный цвет успеха */
exports.inverseTextPositiveMinorActive = 'var(--inverse-text-positive-minor-active, #104219FF)';
/** Инвертированный минорный цвет предупреждения */
exports.inverseTextWarningMinor = 'var(--inverse-text-warning-minor, #3D1D0A)';
/** Инвертированный минорный цвет предупреждения */
exports.inverseTextWarningMinorHover = 'var(--inverse-text-warning-minor-hover, #723613FF)';
/** Инвертированный минорный цвет предупреждения */
exports.inverseTextWarningMinorActive = 'var(--inverse-text-warning-minor-active, #58290EFF)';
/** Инвертированный минорный цвет ошибки */
exports.inverseTextNegativeMinor = 'var(--inverse-text-negative-minor, #4A0D13)';
/** Инвертированный минорный цвет ошибки */
exports.inverseTextNegativeMinorHover = 'var(--inverse-text-negative-minor-hover, #7E1621FF)';
/** Инвертированный минорный цвет ошибки */
exports.inverseTextNegativeMinorActive = 'var(--inverse-text-negative-minor-active, #64121AFF)';
/** Инвертированный минорный цвет информации */
exports.inverseTextInfoMinor = 'var(--inverse-text-info-minor, #0C283B)';
/** Инвертированный минорный цвет информации */
exports.inverseTextInfoMinorHover = 'var(--inverse-text-info-minor-hover, #174B6EFF)';
/** Инвертированный минорный цвет информации */
exports.inverseTextInfoMinorActive = 'var(--inverse-text-info-minor-active, #113A55FF)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimary = 'var(--surface-solid-primary, #171717)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimaryHover = 'var(--surface-solid-primary-hover, #1C1C1CFF)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimaryActive = 'var(--surface-solid-primary-active, #121212FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondary = 'var(--surface-solid-secondary, #262626)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondaryHover = 'var(--surface-solid-secondary-hover, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondaryActive = 'var(--surface-solid-secondary-active, #212121FF)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = 'var(--surface-solid-tertiary, #363636)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiaryHover = 'var(--surface-solid-tertiary-hover, #404040FF)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiaryActive = 'var(--surface-solid-tertiary-active, #303030FF)';
/** Основной фон для карточек */
exports.surfaceSolidCard = 'var(--surface-solid-card, #171717)';
/** Основной фон для карточек */
exports.surfaceSolidCardHover = 'var(--surface-solid-card-hover, #1C1C1CFF)';
/** Основной фон для карточек */
exports.surfaceSolidCardActive = 'var(--surface-solid-card-active, #121212FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefault = 'var(--surface-solid-default, #F9F9F9)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefaultHover = 'var(--surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefaultActive = 'var(--surface-solid-default-active, #F5F5F5FF)';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimary = 'var(--surface-transparent-primary, #F9F9F90F)';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimaryHover = 'var(--surface-transparent-primary-hover, #FAFAFA1F)';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimaryActive = 'var(--surface-transparent-primary-active, #FAFAFA0A)';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondary = 'var(--surface-transparent-secondary, #F9F9F91F)';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondaryHover = 'var(--surface-transparent-secondary-hover, #FAFAFA3D)';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondaryActive = 'var(--surface-transparent-secondary-active, #FAFAFA0F)';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, #F9F9F933)';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiaryHover = 'var(--surface-transparent-tertiary-hover, #FAFAFA52)';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiaryActive = 'var(--surface-transparent-tertiary-active, #FAFAFA24)';
/** Глубокий прозрачный фон поверхности/контрола */
exports.surfaceTransparentDeep = 'var(--surface-transparent-deep, #F9F9F9A3)';
/** Глубокий прозрачный фон поверхности/контрола */
exports.surfaceTransparentDeepHover = 'var(--surface-transparent-deep-hover, #FAFAFAC2)';
/** Глубокий прозрачный фон поверхности/контрола */
exports.surfaceTransparentDeepActive = 'var(--surface-transparent-deep-active, #FAFAFA94)';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCard = 'var(--surface-transparent-card, #F9F9F90F)';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCardHover = 'var(--surface-transparent-card-hover, #FAFAFA1F)';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCardActive = 'var(--surface-transparent-card-active, #FAFAFA0A)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClear = 'var(--surface-clear, #FFFFFF00)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClearHover = 'var(--surface-clear-hover, #FFFFFF00)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClearActive = 'var(--surface-clear-active, #FFFFFF00)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccent = 'var(--surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentHover = 'var(--surface-accent-hover, #1DAF37FF)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentActive = 'var(--surface-accent-active, #18952FFF)';
/** Акцентный минорный непрозрачный фон поверхности/контрола */
exports.surfaceAccentMinor = 'var(--surface-accent-minor, #0A2B10)';
/** Акцентный минорный непрозрачный фон поверхности/контрола */
exports.surfaceAccentMinorHover = 'var(--surface-accent-minor-hover, #0E3A16FF)';
/** Акцентный минорный непрозрачный фон поверхности/контрола */
exports.surfaceAccentMinorActive = 'var(--surface-accent-minor-active, #08210CFF)';
/** Прозрачный акцентный фон поверхности/контрола */
exports.surfaceTransparentAccent = 'var(--surface-transparent-accent, #1A9E3247)';
/** Прозрачный акцентный фон поверхности/контрола */
exports.surfaceTransparentAccentHover = 'var(--surface-transparent-accent-hover, #1A9E3266)';
/** Прозрачный акцентный фон поверхности/контрола */
exports.surfaceTransparentAccentActive = 'var(--surface-transparent-accent-active, #1A9E3238)';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositive = 'var(--surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositiveHover = 'var(--surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositiveActive = 'var(--surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarning = 'var(--surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarningHover = 'var(--surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarningActive = 'var(--surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = 'var(--surface-negative, #FF293E)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeHover = 'var(--surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeActive = 'var(--surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола информация */
exports.surfaceInfo = 'var(--surface-info, #118CDF)';
/** Цвет фона поверхности/контрола информация */
exports.surfaceInfoHover = 'var(--surface-info-hover, #1798EEFF)';
/** Цвет фона поверхности/контрола информация */
exports.surfaceInfoActive = 'var(--surface-info-active, #1086D5FF)';
/** Минорный цвет фона поверхности/контрола успех */
exports.surfacePositiveMinor = 'var(--surface-positive-minor, #0A2B10)';
/** Минорный цвет фона поверхности/контрола успех */
exports.surfacePositiveMinorHover = 'var(--surface-positive-minor-hover, #0E3A16FF)';
/** Минорный цвет фона поверхности/контрола успех */
exports.surfacePositiveMinorActive = 'var(--surface-positive-minor-active, #08210CFF)';
/** Минорный цвет фона поверхности/контрола предупреждение */
exports.surfaceWarningMinor = 'var(--surface-warning-minor, #3D1D0A)';
/** Минорный цвет фона поверхности/контрола предупреждение */
exports.surfaceWarningMinorHover = 'var(--surface-warning-minor-hover, #4F250DFF)';
/** Минорный цвет фона поверхности/контрола предупреждение */
exports.surfaceWarningMinorActive = 'var(--surface-warning-minor-active, #351909FF)';
/** Минорный цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeMinor = 'var(--surface-negative-minor, #4A0D13)';
/** Минорный цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeMinorHover = 'var(--surface-negative-minor-hover, #5B1018FF)';
/** Минорный цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeMinorActive = 'var(--surface-negative-minor-active, #410B11FF)';
/** Минорный цвет фона поверхности/контрола информация */
exports.surfaceInfoMinor = 'var(--surface-info-minor, #0C283B)';
/** Минорный цвет фона поверхности/контрола информация */
exports.surfaceInfoMinorHover = 'var(--surface-info-minor-hover, #10344CFF)';
/** Минорный цвет фона поверхности/контрола информация */
exports.surfaceInfoMinorActive = 'var(--surface-info-minor-active, #0A2333FF)';
/** Прозрачный цвет фона поверхности/контрола успех */
exports.surfaceTransparentPositive = 'var(--surface-transparent-positive, #1A9E3247)';
/** Прозрачный цвет фона поверхности/контрола успех */
exports.surfaceTransparentPositiveHover = 'var(--surface-transparent-positive-hover, #1A9E3266)';
/** Прозрачный цвет фона поверхности/контрола успех */
exports.surfaceTransparentPositiveActive = 'var(--surface-transparent-positive-active, #1A9E3238)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
exports.surfaceTransparentWarning = 'var(--surface-transparent-warning, #FA5F0547)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
exports.surfaceTransparentWarningHover = 'var(--surface-transparent-warning-hover, #FA5F0566)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
exports.surfaceTransparentWarningActive = 'var(--surface-transparent-warning-active, #FA5F0538)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
exports.surfaceTransparentNegative = 'var(--surface-transparent-negative, #FF293E47)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
exports.surfaceTransparentNegativeHover = 'var(--surface-transparent-negative-hover, #FF293E66)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
exports.surfaceTransparentNegativeActive = 'var(--surface-transparent-negative-active, #FF293E38)';
/** Прозрачный цвет фона поверхности/контрола информация */
exports.surfaceTransparentInfo = 'var(--surface-transparent-info, #118CDF47)';
/** Прозрачный цвет фона поверхности/контрола информация */
exports.surfaceTransparentInfoHover = 'var(--surface-transparent-info-hover, #118CDF66)';
/** Прозрачный цвет фона поверхности/контрола информация */
exports.surfaceTransparentInfoActive = 'var(--surface-transparent-info-active, #118CDF38)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #171717)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimaryHover = 'var(--on-dark-surface-solid-primary-hover, #1C1C1CFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimaryActive = 'var(--on-dark-surface-solid-primary-active, #121212FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #262626)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondaryHover = 'var(--on-dark-surface-solid-secondary-hover, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondaryActive = 'var(--on-dark-surface-solid-secondary-active, #212121FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #363636)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiaryHover = 'var(--on-dark-surface-solid-tertiary-hover, #404040FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiaryActive = 'var(--on-dark-surface-solid-tertiary-active, #303030FF)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #171717)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCardHover = 'var(--on-dark-surface-solid-card-hover, #1C1C1CFF)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCardActive = 'var(--on-dark-surface-solid-card-active, #121212FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #F9F9F9)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefaultHover = 'var(--on-dark-surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefaultActive = 'var(--on-dark-surface-solid-default-active, #F5F5F5FF)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, #F9F9F90F)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimaryHover = 'var(--on-dark-surface-transparent-primary-hover, #FAFAFA1F)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimaryActive = 'var(--on-dark-surface-transparent-primary-active, #FAFAFA0A)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, #F9F9F91F)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondaryHover = 'var(--on-dark-surface-transparent-secondary-hover, #FAFAFA3D)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondaryActive = 'var(--on-dark-surface-transparent-secondary-active, #FAFAFA0F)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, #F9F9F933)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiaryHover = 'var(--on-dark-surface-transparent-tertiary-hover, #FAFAFA52)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiaryActive = 'var(--on-dark-surface-transparent-tertiary-active, #FAFAFA24)';
/** Глубокий прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentDeep = 'var(--on-dark-surface-transparent-deep, #F9F9F9A3)';
/** Глубокий прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentDeepHover = 'var(--on-dark-surface-transparent-deep-hover, #FAFAFAC2)';
/** Глубокий прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentDeepActive = 'var(--on-dark-surface-transparent-deep-active, #FAFAFA94)';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, #F9F9F90F)';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCardHover = 'var(--on-dark-surface-transparent-card-hover, #FAFAFA1F)';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCardActive = 'var(--on-dark-surface-transparent-card-active, #FAFAFA0A)';
/** Фон поверхности/контрола без заливки на темном фоне */
exports.onDarkSurfaceClear = 'var(--on-dark-surface-clear, #FFFFFF00)';
/** Фон поверхности/контрола без заливки на темном фоне */
exports.onDarkSurfaceClearHover = 'var(--on-dark-surface-clear-hover, #FFFFFF00)';
/** Фон поверхности/контрола без заливки на темном фоне */
exports.onDarkSurfaceClearActive = 'var(--on-dark-surface-clear-active, #FFFFFF00)';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccentHover = 'var(--on-dark-surface-accent-hover, #1DAF37FF)';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccentActive = 'var(--on-dark-surface-accent-active, #18952FFF)';
/** Акцентный минорный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccentMinor = 'var(--on-dark-surface-accent-minor, #0A2B10)';
/** Акцентный минорный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccentMinorHover = 'var(--on-dark-surface-accent-minor-hover, #0E3A16FF)';
/** Акцентный минорный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccentMinorActive = 'var(--on-dark-surface-accent-minor-active, #08210CFF)';
/** Прозрачный акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentAccent = 'var(--on-dark-surface-transparent-accent, #1A9E3247)';
/** Прозрачный акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentAccentHover = 'var(--on-dark-surface-transparent-accent-hover, #1A9E3266)';
/** Прозрачный акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentAccentActive = 'var(--on-dark-surface-transparent-accent-active, #1A9E3238)';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositive = 'var(--on-dark-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositiveHover = 'var(--on-dark-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositiveActive = 'var(--on-dark-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceWarningHover = 'var(--on-dark-surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceWarningActive = 'var(--on-dark-surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293E)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegativeHover = 'var(--on-dark-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegativeActive = 'var(--on-dark-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола информация на темном фоне */
exports.onDarkSurfaceInfo = 'var(--on-dark-surface-info, #118CDF)';
/** Цвет фона поверхности/контрола информация на темном фоне */
exports.onDarkSurfaceInfoHover = 'var(--on-dark-surface-info-hover, #1798EEFF)';
/** Цвет фона поверхности/контрола информация на темном фоне */
exports.onDarkSurfaceInfoActive = 'var(--on-dark-surface-info-active, #1086D5FF)';
/** Минорный цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositiveMinor = 'var(--on-dark-surface-positive-minor, #0A2B10)';
/** Минорный цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositiveMinorHover = 'var(--on-dark-surface-positive-minor-hover, #0E3A16FF)';
/** Минорный цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositiveMinorActive = 'var(--on-dark-surface-positive-minor-active, #08210CFF)';
/** Минорный цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceWarningMinor = 'var(--on-dark-surface-warning-minor, #3D1D0A)';
/** Минорный цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceWarningMinorHover = 'var(--on-dark-surface-warning-minor-hover, #4F250DFF)';
/** Минорный цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceWarningMinorActive = 'var(--on-dark-surface-warning-minor-active, #351909FF)';
/** Минорный цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegativeMinor = 'var(--on-dark-surface-negative-minor, #4A0D13)';
/** Минорный цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegativeMinorHover = 'var(--on-dark-surface-negative-minor-hover, #5B1018FF)';
/** Минорный цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegativeMinorActive = 'var(--on-dark-surface-negative-minor-active, #410B11FF)';
/** Минорный цвет фона поверхности/контрола информация на темном фоне */
exports.onDarkSurfaceInfoMinor = 'var(--on-dark-surface-info-minor, #0C283B)';
/** Минорный цвет фона поверхности/контрола информация на темном фоне */
exports.onDarkSurfaceInfoMinorHover = 'var(--on-dark-surface-info-minor-hover, #10344CFF)';
/** Минорный цвет фона поверхности/контрола информация на темном фоне */
exports.onDarkSurfaceInfoMinorActive = 'var(--on-dark-surface-info-minor-active, #0A2333FF)';
/** Прозрачный цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfaceTransparentPositive = 'var(--on-dark-surface-transparent-positive, #1A9E3247)';
/** Прозрачный цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfaceTransparentPositiveHover = 'var(--on-dark-surface-transparent-positive-hover, #1A9E3266)';
/** Прозрачный цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfaceTransparentPositiveActive = 'var(--on-dark-surface-transparent-positive-active, #1A9E3238)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceTransparentWarning = 'var(--on-dark-surface-transparent-warning, #FA5F0547)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceTransparentWarningHover = 'var(--on-dark-surface-transparent-warning-hover, #FA5F0566)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceTransparentWarningActive = 'var(--on-dark-surface-transparent-warning-active, #FA5F0538)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceTransparentNegative = 'var(--on-dark-surface-transparent-negative, #FF293E47)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceTransparentNegativeHover = 'var(--on-dark-surface-transparent-negative-hover, #FF293E66)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
exports.onDarkSurfaceTransparentNegativeActive = 'var(--on-dark-surface-transparent-negative-active, #FF293E38)';
/** Прозрачный цвет фона поверхности/контрола информация на темном фоне */
exports.onDarkSurfaceTransparentInfo = 'var(--on-dark-surface-transparent-info, #118CDF47)';
/** Прозрачный цвет фона поверхности/контрола информация на темном фоне */
exports.onDarkSurfaceTransparentInfoHover = 'var(--on-dark-surface-transparent-info-hover, #118CDF66)';
/** Прозрачный цвет фона поверхности/контрола информация на темном фоне */
exports.onDarkSurfaceTransparentInfoActive = 'var(--on-dark-surface-transparent-info-active, #118CDF38)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F5F5F5)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidPrimaryHover = 'var(--on-light-surface-solid-primary-hover, #FAFAFAFF)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidPrimaryActive = 'var(--on-light-surface-solid-primary-active, #F0F0F0FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #ECECEC)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondaryHover = 'var(--on-light-surface-solid-secondary-hover, #F7F7F7FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondaryActive = 'var(--on-light-surface-solid-secondary-active, #E8E8E8FF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #DDDDDD)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidTertiaryHover = 'var(--on-light-surface-solid-tertiary-hover, #E8E8E8FF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidTertiaryActive = 'var(--on-light-surface-solid-tertiary-active, #D9D9D9FF)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFF)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCardHover = 'var(--on-light-surface-solid-card-hover, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCardActive = 'var(--on-light-surface-solid-card-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #080808)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefaultHover = 'var(--on-light-surface-solid-default-hover, #0D0D0DFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefaultActive = 'var(--on-light-surface-solid-default-active, #030303FF)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, #08080808)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentPrimaryHover = 'var(--on-light-surface-transparent-primary-hover, #08080812)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentPrimaryActive = 'var(--on-light-surface-transparent-primary-active, #08080803)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, #0808080F)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondaryHover = 'var(--on-light-surface-transparent-secondary-hover, #0808081F)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondaryActive = 'var(--on-light-surface-transparent-secondary-active, #0808080A)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, #0808081F)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiaryHover = 'var(--on-light-surface-transparent-tertiary-hover, #0808083D)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiaryActive = 'var(--on-light-surface-transparent-tertiary-active, #0808080F)';
/** Глубокий прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentDeep = 'var(--on-light-surface-transparent-deep, #080808A3)';
/** Глубокий прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentDeepHover = 'var(--on-light-surface-transparent-deep-hover, #080808C2)';
/** Глубокий прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentDeepActive = 'var(--on-light-surface-transparent-deep-active, #08080894)';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, #0808080F)';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCardHover = 'var(--on-light-surface-transparent-card-hover, #0808081F)';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCardActive = 'var(--on-light-surface-transparent-card-active, #0808080A)';
/** Фон поверхности/контрола без заливки на светлом фоне */
exports.onLightSurfaceClear = 'var(--on-light-surface-clear, #FFFFFF00)';
/** Фон поверхности/контрола без заливки на светлом фоне */
exports.onLightSurfaceClearHover = 'var(--on-light-surface-clear-hover, #FFFFFF00)';
/** Фон поверхности/контрола без заливки на светлом фоне */
exports.onLightSurfaceClearActive = 'var(--on-light-surface-clear-active, #FFFFFF00)';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccent = 'var(--on-light-surface-accent, #108E26)';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccentHover = 'var(--on-light-surface-accent-hover, #12A12AFF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccentActive = 'var(--on-light-surface-accent-active, #0F8523FF)';
/** Акцентный минорный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccentMinor = 'var(--on-light-surface-accent-minor, #77F38E)';
/** Акцентный минорный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccentMinorHover = 'var(--on-light-surface-accent-minor-hover, #8AF59DFF)';
/** Акцентный минорный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccentMinorActive = 'var(--on-light-surface-accent-minor-active, #6EF286FF)';
/** Прозрачный акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentAccent = 'var(--on-light-surface-transparent-accent, #108E2647)';
/** Прозрачный акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentAccentHover = 'var(--on-light-surface-transparent-accent-hover, #108E2566)';
/** Прозрачный акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentAccentActive = 'var(--on-light-surface-transparent-accent-active, #108E2538)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositive = 'var(--on-light-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositiveHover = 'var(--on-light-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositiveActive = 'var(--on-light-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceWarning = 'var(--on-light-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceWarningHover = 'var(--on-light-surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceWarningActive = 'var(--on-light-surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegative = 'var(--on-light-surface-negative, #FF293E)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegativeHover = 'var(--on-light-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegativeActive = 'var(--on-light-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола информация на светлом фоне */
exports.onLightSurfaceInfo = 'var(--on-light-surface-info, #118CDF)';
/** Цвет фона поверхности/контрола информация на светлом фоне */
exports.onLightSurfaceInfoHover = 'var(--on-light-surface-info-hover, #1798EEFF)';
/** Цвет фона поверхности/контрола информация на светлом фоне */
exports.onLightSurfaceInfoActive = 'var(--on-light-surface-info-active, #1086D5FF)';
/** Минорный цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositiveMinor = 'var(--on-light-surface-positive-minor, #77F38E)';
/** Минорный цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositiveMinorHover = 'var(--on-light-surface-positive-minor-hover, #8AF59DFF)';
/** Минорный цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositiveMinorActive = 'var(--on-light-surface-positive-minor-active, #6EF286FF)';
/** Минорный цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceWarningMinor = 'var(--on-light-surface-warning-minor, #FFD1B8)';
/** Минорный цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceWarningMinorHover = 'var(--on-light-surface-warning-minor-hover, #FFDECCFF)';
/** Минорный цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceWarningMinorActive = 'var(--on-light-surface-warning-minor-active, #FFCAADFF)';
/** Минорный цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegativeMinor = 'var(--on-light-surface-negative-minor, #FECDD2)';
/** Минорный цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegativeMinorHover = 'var(--on-light-surface-negative-minor-hover, #FED7DBFF)';
/** Минорный цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegativeMinorActive = 'var(--on-light-surface-negative-minor-active, #FEC3C9FF)';
/** Минорный цвет фона поверхности/контрола информация на светлом фоне */
exports.onLightSurfaceInfoMinor = 'var(--on-light-surface-info-minor, #B0DFFF)';
/** Минорный цвет фона поверхности/контрола информация на светлом фоне */
exports.onLightSurfaceInfoMinorHover = 'var(--on-light-surface-info-minor-hover, #C7E9FFFF)';
/** Минорный цвет фона поверхности/контрола информация на светлом фоне */
exports.onLightSurfaceInfoMinorActive = 'var(--on-light-surface-info-minor-active, #A8DCFFFF)';
/** Прозрачный цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfaceTransparentPositive = 'var(--on-light-surface-transparent-positive, #1A9E3247)';
/** Прозрачный цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfaceTransparentPositiveHover = 'var(--on-light-surface-transparent-positive-hover, #1A9E3266)';
/** Прозрачный цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfaceTransparentPositiveActive = 'var(--on-light-surface-transparent-positive-active, #1A9E3238)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceTransparentWarning = 'var(--on-light-surface-transparent-warning, #FA5F0547)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceTransparentWarningHover = 'var(--on-light-surface-transparent-warning-hover, #FA5F0566)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceTransparentWarningActive = 'var(--on-light-surface-transparent-warning-active, #FA5F0538)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceTransparentNegative = 'var(--on-light-surface-transparent-negative, #FF293E47)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceTransparentNegativeHover = 'var(--on-light-surface-transparent-negative-hover, #FF293E66)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
exports.onLightSurfaceTransparentNegativeActive = 'var(--on-light-surface-transparent-negative-active, #FF293E38)';
/** Прозрачный цвет фона поверхности/контрола информация на светлом фоне */
exports.onLightSurfaceTransparentInfo = 'var(--on-light-surface-transparent-info, #118CDF47)';
/** Прозрачный цвет фона поверхности/контрола информация на светлом фоне */
exports.onLightSurfaceTransparentInfoHover = 'var(--on-light-surface-transparent-info-hover, #118CDF66)';
/** Прозрачный цвет фона поверхности/контрола информация на светлом фоне */
exports.onLightSurfaceTransparentInfoActive = 'var(--on-light-surface-transparent-info-active, #118CDF38)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidPrimary = 'var(--inverse-surface-solid-primary, #F5F5F5)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidPrimaryHover = 'var(--inverse-surface-solid-primary-hover, #FAFAFAFF)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidPrimaryActive = 'var(--inverse-surface-solid-primary-active, #F0F0F0FF)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidSecondary = 'var(--inverse-surface-solid-secondary, #ECECEC)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidSecondaryHover = 'var(--inverse-surface-solid-secondary-hover, #F7F7F7FF)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidSecondaryActive = 'var(--inverse-surface-solid-secondary-active, #E8E8E8FF)';
/** Инвертированный третичный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidTertiary = 'var(--inverse-surface-solid-tertiary, #DDDDDD)';
/** Инвертированный третичный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidTertiaryHover = 'var(--inverse-surface-solid-tertiary-hover, #E8E8E8FF)';
/** Инвертированный третичный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidTertiaryActive = 'var(--inverse-surface-solid-tertiary-active, #D9D9D9FF)';
/** Инвертированный основной фон для карточек */
exports.inverseSurfaceSolidCard = 'var(--inverse-surface-solid-card, #FFFFFF)';
/** Инвертированный основной фон для карточек */
exports.inverseSurfaceSolidCardHover = 'var(--inverse-surface-solid-card-hover, #FFFFFFFF)';
/** Инвертированный основной фон для карточек */
exports.inverseSurfaceSolidCardActive = 'var(--inverse-surface-solid-card-active, #FFFFFFFF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
exports.inverseSurfaceSolidDefault = 'var(--inverse-surface-solid-default, #080808)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
exports.inverseSurfaceSolidDefaultHover = 'var(--inverse-surface-solid-default-hover, #0D0D0DFF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
exports.inverseSurfaceSolidDefaultActive = 'var(--inverse-surface-solid-default-active, #030303FF)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentPrimary = 'var(--inverse-surface-transparent-primary, #08080808)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentPrimaryHover = 'var(--inverse-surface-transparent-primary-hover, #08080812)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentPrimaryActive = 'var(--inverse-surface-transparent-primary-active, #08080803)';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentSecondary = 'var(--inverse-surface-transparent-secondary, #0808080F)';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentSecondaryHover = 'var(--inverse-surface-transparent-secondary-hover, #0808081F)';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentSecondaryActive = 'var(--inverse-surface-transparent-secondary-active, #0808080A)';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentTertiary = 'var(--inverse-surface-transparent-tertiary, #0808081F)';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentTertiaryHover = 'var(--inverse-surface-transparent-tertiary-hover, #0808083D)';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentTertiaryActive = 'var(--inverse-surface-transparent-tertiary-active, #0808080F)';
/** Инвертированный глубокий прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentDeep = 'var(--inverse-surface-transparent-deep, #080808A3)';
/** Инвертированный глубокий прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentDeepHover = 'var(--inverse-surface-transparent-deep-hover, #080808C2)';
/** Инвертированный глубокий прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentDeepActive = 'var(--inverse-surface-transparent-deep-active, #08080894)';
/** Инвертированный прозрачный фон для карточек */
exports.inverseSurfaceTransparentCard = 'var(--inverse-surface-transparent-card, #0808080F)';
/** Инвертированный прозрачный фон для карточек */
exports.inverseSurfaceTransparentCardHover = 'var(--inverse-surface-transparent-card-hover, #0808081F)';
/** Инвертированный прозрачный фон для карточек */
exports.inverseSurfaceTransparentCardActive = 'var(--inverse-surface-transparent-card-active, #0808080A)';
/** Инвертированный фон поверхности/контрола без заливки */
exports.inverseSurfaceClear = 'var(--inverse-surface-clear, #FFFFFF00)';
/** Инвертированный фон поверхности/контрола без заливки */
exports.inverseSurfaceClearHover = 'var(--inverse-surface-clear-hover, #FFFFFF00)';
/** Инвертированный фон поверхности/контрола без заливки */
exports.inverseSurfaceClearActive = 'var(--inverse-surface-clear-active, #FFFFFF00)';
/** Инвертированный акцентный фон поверхности/контрола */
exports.inverseSurfaceAccent = 'var(--inverse-surface-accent, #108E26)';
/** Инвертированный акцентный фон поверхности/контрола */
exports.inverseSurfaceAccentHover = 'var(--inverse-surface-accent-hover, #12A12AFF)';
/** Инвертированный акцентный фон поверхности/контрола */
exports.inverseSurfaceAccentActive = 'var(--inverse-surface-accent-active, #0F8523FF)';
/** Инвертированный акцентный минорный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceAccentMinor = 'var(--inverse-surface-accent-minor, #77F38E)';
/** Инвертированный акцентный минорный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceAccentMinorHover = 'var(--inverse-surface-accent-minor-hover, #8AF59DFF)';
/** Инвертированный акцентный минорный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceAccentMinorActive = 'var(--inverse-surface-accent-minor-active, #6EF286FF)';
/** Прозрачный инвертированный акцентный фон поверхности/контрола */
exports.inverseSurfaceTransparentAccent = 'var(--inverse-surface-transparent-accent, #108E2647)';
/** Прозрачный инвертированный акцентный фон поверхности/контрола */
exports.inverseSurfaceTransparentAccentHover = 'var(--inverse-surface-transparent-accent-hover, #108E2566)';
/** Прозрачный инвертированный акцентный фон поверхности/контрола */
exports.inverseSurfaceTransparentAccentActive = 'var(--inverse-surface-transparent-accent-active, #108E2538)';
/** Инвертированный цвет фона поверхности/контрола успех */
exports.inverseSurfacePositive = 'var(--inverse-surface-positive, #1A9E32)';
/** Инвертированный цвет фона поверхности/контрола успех */
exports.inverseSurfacePositiveHover = 'var(--inverse-surface-positive-hover, #1DAF37FF)';
/** Инвертированный цвет фона поверхности/контрола успех */
exports.inverseSurfacePositiveActive = 'var(--inverse-surface-positive-active, #18952FFF)';
/** Инвертированный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceWarning = 'var(--inverse-surface-warning, #FA5F05)';
/** Инвертированный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceWarningHover = 'var(--inverse-surface-warning-hover, #FB7223FF)';
/** Инвертированный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceWarningActive = 'var(--inverse-surface-warning-active, #F05B05FF)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
exports.inverseSurfaceNegative = 'var(--inverse-surface-negative, #FF293E)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
exports.inverseSurfaceNegativeHover = 'var(--inverse-surface-negative-hover, #FF475AFF)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
exports.inverseSurfaceNegativeActive = 'var(--inverse-surface-negative-active, #FF1F35FF)';
/** Инвертированный цвет фона поверхности/контрола информация */
exports.inverseSurfaceInfo = 'var(--inverse-surface-info, #118CDF)';
/** Инвертированный цвет фона поверхности/контрола информация */
exports.inverseSurfaceInfoHover = 'var(--inverse-surface-info-hover, #1798EEFF)';
/** Инвертированный цвет фона поверхности/контрола информация */
exports.inverseSurfaceInfoActive = 'var(--inverse-surface-info-active, #1086D5FF)';
/** Инвертированный минорный цвет фона поверхности/контрола успех */
exports.inverseSurfacePositiveMinor = 'var(--inverse-surface-positive-minor, #77F38E)';
/** Инвертированный минорный цвет фона поверхности/контрола успех */
exports.inverseSurfacePositiveMinorHover = 'var(--inverse-surface-positive-minor-hover, #8AF59DFF)';
/** Инвертированный минорный цвет фона поверхности/контрола успех */
exports.inverseSurfacePositiveMinorActive = 'var(--inverse-surface-positive-minor-active, #6EF286FF)';
/** Инвертированный минорный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceWarningMinor = 'var(--inverse-surface-warning-minor, #FFD1B8)';
/** Инвертированный минорный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceWarningMinorHover = 'var(--inverse-surface-warning-minor-hover, #FFDECCFF)';
/** Инвертированный минорный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceWarningMinorActive = 'var(--inverse-surface-warning-minor-active, #FFCAADFF)';
/** Инвертированный минорный цвет фона поверхности/контрола ошибка */
exports.inverseSurfaceNegativeMinor = 'var(--inverse-surface-negative-minor, #FECDD2)';
/** Инвертированный минорный цвет фона поверхности/контрола ошибка */
exports.inverseSurfaceNegativeMinorHover = 'var(--inverse-surface-negative-minor-hover, #FED7DBFF)';
/** Инвертированный минорный цвет фона поверхности/контрола ошибка */
exports.inverseSurfaceNegativeMinorActive = 'var(--inverse-surface-negative-minor-active, #FEC3C9FF)';
/** Инвертированный минорный цвет фона поверхности/контрола информация */
exports.inverseSurfaceInfoMinor = 'var(--inverse-surface-info-minor, #B0DFFF)';
/** Инвертированный минорный цвет фона поверхности/контрола информация */
exports.inverseSurfaceInfoMinorHover = 'var(--inverse-surface-info-minor-hover, #C7E9FFFF)';
/** Инвертированный минорный цвет фона поверхности/контрола информация */
exports.inverseSurfaceInfoMinorActive = 'var(--inverse-surface-info-minor-active, #A8DCFFFF)';
/** Прозрачный инвертированный цвет фона поверхности/контрола успех */
exports.inverseSurfaceTransparentPositive = 'var(--inverse-surface-transparent-positive, #1A9E3247)';
/** Прозрачный инвертированный цвет фона поверхности/контрола успех */
exports.inverseSurfaceTransparentPositiveHover = 'var(--inverse-surface-transparent-positive-hover, #1A9E3266)';
/** Прозрачный инвертированный цвет фона поверхности/контрола успех */
exports.inverseSurfaceTransparentPositiveActive = 'var(--inverse-surface-transparent-positive-active, #1A9E3238)';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceTransparentWarning = 'var(--inverse-surface-transparent-warning, #FA5F0547)';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceTransparentWarningHover = 'var(--inverse-surface-transparent-warning-hover, #FA5F0566)';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceTransparentWarningActive = 'var(--inverse-surface-transparent-warning-active, #FA5F0538)';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceTransparentNegative = 'var(--inverse-surface-transparent-negative, #FF293E47)';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceTransparentNegativeHover = 'var(--inverse-surface-transparent-negative-hover, #FF293E66)';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
exports.inverseSurfaceTransparentNegativeActive = 'var(--inverse-surface-transparent-negative-active, #FF293E38)';
/** Прозрачный инвертированный цвет фона поверхности/контрола информация */
exports.inverseSurfaceTransparentInfo = 'var(--inverse-surface-transparent-info, #118CDF47)';
/** Прозрачный инвертированный цвет фона поверхности/контрола информация */
exports.inverseSurfaceTransparentInfoHover = 'var(--inverse-surface-transparent-info-hover, #118CDF66)';
/** Прозрачный инвертированный цвет фона поверхности/контрола информация */
exports.inverseSurfaceTransparentInfoActive = 'var(--inverse-surface-transparent-info-active, #118CDF38)';
/** Основной фон */
exports.backgroundPrimary = 'var(--background-primary, #080808)';
/** Основной фон на темном фоне */
exports.darkBackgroundPrimary = 'var(--dark-background-primary, #080808)';
/** Основной фон на светлом фоне */
exports.lightBackgroundPrimary = 'var(--light-background-primary, #F9F9F9)';
/** Инвертированный основной фон */
exports.inverseBackgroundPrimary = 'var(--inverse-background-primary, #F9F9F9)';
/** Цвет фона паранжи светлый */
exports.overlaySoft = 'var(--overlay-soft, #0808088F)';
/** Цвет фона паранжи темный */
exports.overlayHard = 'var(--overlay-hard, #080808F5)';
/** Цвет фона паранжи размытый */
exports.overlayBlur = 'var(--overlay-blur, #08080833)';
/** Цвет фона паранжи светлый на темном фоне */
exports.onDarkOverlaySoft = 'var(--on-dark-overlay-soft, #0808088F)';
/** Цвет фона паранжи темный на темном фоне */
exports.onDarkOverlayHard = 'var(--on-dark-overlay-hard, #080808F5)';
/** Цвет фона паранжи размытый на темном фоне */
exports.onDarkOverlayBlur = 'var(--on-dark-overlay-blur, #08080833)';
/** Цвет фона паранжи светлый на светлом фоне */
exports.onLightOverlaySoft = 'var(--on-light-overlay-soft, #F9F9F98F)';
/** Цвет фона паранжи темный на светлом фоне */
exports.onLightOverlayHard = 'var(--on-light-overlay-hard, #F9F9F9F5)';
/** Цвет фона паранжи размытый на светлом фоне */
exports.onLightOverlayBlur = 'var(--on-light-overlay-blur, #F9F9F933)';
/** Инвертированный цвет фона паранжи светлый */
exports.inverseOverlaySoft = 'var(--inverse-overlay-soft, #F9F9F98F)';
/** Инвертированный цвет фона паранжи темный */
exports.inverseOverlayHard = 'var(--inverse-overlay-hard, #F9F9F9F5)';
/** Инвертированный цвет фона паранжи размытый */
exports.inverseOverlayBlur = 'var(--inverse-overlay-blur, #F9F9F933)';
/** Основной непрозрачный цвет обводки */
exports.outlineDefaultOutlineSolidPrimary = 'var(--outline-default-outline-solid-primary, #262626)';
/** Основной непрозрачный цвет обводки */
exports.outlineDefaultOutlineSolidPrimaryHover = 'var(--outline-default-outline-solid-primary-hover, #FFFFFFFF)';
/** Основной непрозрачный цвет обводки */
exports.outlineDefaultOutlineSolidPrimaryActive = 'var(--outline-default-outline-solid-primary-active, #ADADADFF)';
/** Вторичный непрозрачный цвет обводки */
exports.outlineDefaultOutlineSolidSecondary = 'var(--outline-default-outline-solid-secondary, #363636)';
/** Вторичный непрозрачный цвет обводки */
exports.outlineDefaultOutlineSolidSecondaryHover = 'var(--outline-default-outline-solid-secondary-hover, #FFFFFFFF)';
/** Вторичный непрозрачный цвет обводки */
exports.outlineDefaultOutlineSolidSecondaryActive = 'var(--outline-default-outline-solid-secondary-active, #A1A1A1FF)';
/** Третичный непрозрачный цвет обводки */
exports.outlineDefaultOutlineSolidTertiary = 'var(--outline-default-outline-solid-tertiary, #4E4E4E)';
/** Третичный непрозрачный цвет обводки */
exports.outlineDefaultOutlineSolidTertiaryHover = 'var(--outline-default-outline-solid-tertiary-hover, #FFFFFFFF)';
/** Третичный непрозрачный цвет обводки */
exports.outlineDefaultOutlineSolidTertiaryActive = 'var(--outline-default-outline-solid-tertiary-active, #8C8C8CFF)';
/** Прозрачный цвет обводки по умолчанию */
exports.outlineDefaultOutlineTransparentDefault = 'var(--outline-default-outline-transparent-default, #F9F9F9F5)';
/** Прозрачный цвет обводки по умолчанию */
exports.outlineDefaultOutlineTransparentDefaultHover = 'var(--outline-default-outline-transparent-default-hover, #FAFAFA93)';
/** Прозрачный цвет обводки по умолчанию */
exports.outlineDefaultOutlineTransparentDefaultActive = 'var(--outline-default-outline-transparent-default-active, #FAFAFAC4)';
/** Основной прозрачный цвет обводки */
exports.outlineDefaultOutlineTransparentPrimary = 'var(--outline-default-outline-transparent-primary, #F9F9F91F)';
/** Основной прозрачный цвет обводки */
exports.outlineDefaultOutlineTransparentPrimaryHover = 'var(--outline-default-outline-transparent-primary-hover, #FAFAFAFF)';
/** Основной прозрачный цвет обводки */
exports.outlineDefaultOutlineTransparentPrimaryActive = 'var(--outline-default-outline-transparent-primary-active, #FAFAFA25)';
/** Вторичный прозрачный цвет обводки */
exports.outlineDefaultOutlineTransparentSecondary = 'var(--outline-default-outline-transparent-secondary, #F9F9F933)';
/** Вторичный прозрачный цвет обводки */
exports.outlineDefaultOutlineTransparentSecondaryHover = 'var(--outline-default-outline-transparent-secondary-hover, #FAFAFAFF)';
/** Вторичный прозрачный цвет обводки */
exports.outlineDefaultOutlineTransparentSecondaryActive = 'var(--outline-default-outline-transparent-secondary-active, #FAFAFA3D)';
/** Третичный прозрачный цвет обводки */
exports.outlineDefaultOutlineTransparentTertiary = 'var(--outline-default-outline-transparent-tertiary, #F9F9F947)';
/** Третичный прозрачный цвет обводки */
exports.outlineDefaultOutlineTransparentTertiaryHover = 'var(--outline-default-outline-transparent-tertiary-hover, #FAFAFAFF)';
/** Третичный прозрачный цвет обводки */
exports.outlineDefaultOutlineTransparentTertiaryActive = 'var(--outline-default-outline-transparent-tertiary-active, #FAFAFA56)';
/** Бесцветная обводка */
exports.outlineDefaultOutlineClear = 'var(--outline-default-outline-clear, #FFFFFF00)';
/** Бесцветная обводка */
exports.outlineDefaultOutlineClearHover = 'var(--outline-default-outline-clear-hover, #FFFFFF00)';
/** Бесцветная обводка */
exports.outlineDefaultOutlineClearActive = 'var(--outline-default-outline-clear-active, #FFFFFF00)';
/** Акцентный цвет обводки */
exports.outlineDefaultOutlineAccent = 'var(--outline-default-outline-accent, #1A9E32)';
/** Акцентный цвет обводки */
exports.outlineDefaultOutlineAccentHover = 'var(--outline-default-outline-accent-hover, #1EB83AFF)';
/** Акцентный цвет обводки */
exports.outlineDefaultOutlineAccentActive = 'var(--outline-default-outline-accent-active, #15842AFF)';
/** Акцентный минорный непрозрачный цвет обводки */
exports.outlineDefaultOutlineAccentMinor = 'var(--outline-default-outline-accent-minor, #0A4014)';
/** Акцентный минорный непрозрачный цвет обводки */
exports.outlineDefaultOutlineAccentMinorHover = 'var(--outline-default-outline-accent-minor-hover, #137725FF)';
/** Акцентный минорный непрозрачный цвет обводки */
exports.outlineDefaultOutlineAccentMinorActive = 'var(--outline-default-outline-accent-minor-active, #0E5D1DFF)';
/** Прозрачный акцентный цвет обводки */
exports.outlineDefaultOutlineTransparentAccent = 'var(--outline-default-outline-transparent-accent, #1A9E3266)';
/** Прозрачный акцентный цвет обводки */
exports.outlineDefaultOutlineTransparentAccentHover = 'var(--outline-default-outline-transparent-accent-hover, #1A9E32FF)';
/** Прозрачный акцентный цвет обводки */
exports.outlineDefaultOutlineTransparentAccentActive = 'var(--outline-default-outline-transparent-accent-active, #1A9E327A)';
/** Цвет обводки успех */
exports.outlineDefaultOutlinePositive = 'var(--outline-default-outline-positive, #1A9E32)';
/** Цвет обводки успех */
exports.outlineDefaultOutlinePositiveHover = 'var(--outline-default-outline-positive-hover, #1EB83AFF)';
/** Цвет обводки успех */
exports.outlineDefaultOutlinePositiveActive = 'var(--outline-default-outline-positive-active, #15842AFF)';
/** Цвет обводки предупреждение */
exports.outlineDefaultOutlineWarning = 'var(--outline-default-outline-warning, #FA5F05)';
/** Цвет обводки предупреждение */
exports.outlineDefaultOutlineWarningHover = 'var(--outline-default-outline-warning-hover, #FB7223FF)';
/** Цвет обводки предупреждение */
exports.outlineDefaultOutlineWarningActive = 'var(--outline-default-outline-warning-active, #DC5304FF)';
/** Цвет обводки ошибка */
exports.outlineDefaultOutlineNegative = 'var(--outline-default-outline-negative, #FF293E)';
/** Цвет обводки ошибка */
exports.outlineDefaultOutlineNegativeHover = 'var(--outline-default-outline-negative-hover, #FF475AFF)';
/** Цвет обводки ошибка */
exports.outlineDefaultOutlineNegativeActive = 'var(--outline-default-outline-negative-active, #FF0A23FF)';
/** Цвет обводки информация */
exports.outlineDefaultOutlineInfo = 'var(--outline-default-outline-info, #118CDF)';
/** Цвет обводки информация */
exports.outlineDefaultOutlineInfoHover = 'var(--outline-default-outline-info-hover, #50B1F2FF)';
/** Цвет обводки информация */
exports.outlineDefaultOutlineInfoActive = 'var(--outline-default-outline-info-active, #0F7BC2FF)';
/** Минорный цвет обводки успех */
exports.outlineDefaultOutlinePositiveMinor = 'var(--outline-default-outline-positive-minor, #0A4014)';
/** Минорный цвет обводки успех */
exports.outlineDefaultOutlinePositiveMinorHover = 'var(--outline-default-outline-positive-minor-hover, #137725FF)';
/** Минорный цвет обводки успех */
exports.outlineDefaultOutlinePositiveMinorActive = 'var(--outline-default-outline-positive-minor-active, #0E5D1DFF)';
/** Минорный цвет обводки предупреждение */
exports.outlineDefaultOutlineWarningMinor = 'var(--outline-default-outline-warning-minor, #5C2709)';
/** Минорный цвет обводки предупреждение */
exports.outlineDefaultOutlineWarningMinorHover = 'var(--outline-default-outline-warning-minor-hover, #95400FFF)';
/** Минорный цвет обводки предупреждение */
exports.outlineDefaultOutlineWarningMinorActive = 'var(--outline-default-outline-warning-minor-active, #79340CFF)';
/** Минорный цвет обводки ошибка */
exports.outlineDefaultOutlineNegativeMinor = 'var(--outline-default-outline-negative-minor, #6B1019)';
/** Минорный цвет обводки ошибка */
exports.outlineDefaultOutlineNegativeMinorHover = 'var(--outline-default-outline-negative-minor-hover, #A01825FF)';
/** Минорный цвет обводки ошибка */
exports.outlineDefaultOutlineNegativeMinorActive = 'var(--outline-default-outline-negative-minor-active, #85141FFF)';
/** Минорный цвет обводки информация */
exports.outlineDefaultOutlineInfoMinor = 'var(--outline-default-outline-info-minor, #0D3B59)';
/** Минорный цвет обводки информация */
exports.outlineDefaultOutlineInfoMinorHover = 'var(--outline-default-outline-info-minor-hover, #145E8FFF)';
/** Минорный цвет обводки информация */
exports.outlineDefaultOutlineInfoMinorActive = 'var(--outline-default-outline-info-minor-active, #114C74FF)';
/** Прозрачный цвет обводки успех */
exports.outlineDefaultOutlineTransparentPositive = 'var(--outline-default-outline-transparent-positive, #1A9E3247)';
/** Прозрачный цвет обводки успех */
exports.outlineDefaultOutlineTransparentPositiveHover = 'var(--outline-default-outline-transparent-positive-hover, #1A9E32FF)';
/** Прозрачный цвет обводки успех */
exports.outlineDefaultOutlineTransparentPositiveActive = 'var(--outline-default-outline-transparent-positive-active, #1A9E3256)';
/** Прозрачный цвет обводки предупреждение */
exports.outlineDefaultOutlineTransparentWarning = 'var(--outline-default-outline-transparent-warning, #FA5F0547)';
/** Прозрачный цвет обводки предупреждение */
exports.outlineDefaultOutlineTransparentWarningHover = 'var(--outline-default-outline-transparent-warning-hover, #FA5F05FF)';
/** Прозрачный цвет обводки предупреждение */
exports.outlineDefaultOutlineTransparentWarningActive = 'var(--outline-default-outline-transparent-warning-active, #FA5F0556)';
/** Прозрачный цвет обводки предупреждение */
exports.outlineDefaultOutlineTransparentNegative = 'var(--outline-default-outline-transparent-negative, #FF293E47)';
/** Прозрачный цвет обводки предупреждение */
exports.outlineDefaultOutlineTransparentNegativeHover = 'var(--outline-default-outline-transparent-negative-hover, #FF293EFF)';
/** Прозрачный цвет обводки предупреждение */
exports.outlineDefaultOutlineTransparentNegativeActive = 'var(--outline-default-outline-transparent-negative-active, #FF293E56)';
/** Прозрачный цвет обводки информация */
exports.outlineDefaultOutlineTransparentInfo = 'var(--outline-default-outline-transparent-info, #118CDF47)';
/** Прозрачный цвет обводки информация */
exports.outlineDefaultOutlineTransparentInfoHover = 'var(--outline-default-outline-transparent-info-hover, #118CDFFF)';
/** Прозрачный цвет обводки информация */
exports.outlineDefaultOutlineTransparentInfoActive = 'var(--outline-default-outline-transparent-info-active, #118CDF56)';
/** Основной непрозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineSolidPrimary = 'var(--outline-on-dark-outline-solid-primary, #262626)';
/** Основной непрозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineSolidPrimaryHover = 'var(--outline-on-dark-outline-solid-primary-hover, #FFFFFFFF)';
/** Основной непрозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineSolidPrimaryActive = 'var(--outline-on-dark-outline-solid-primary-active, #ADADADFF)';
/** Вторичный непрозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineSolidSecondary = 'var(--outline-on-dark-outline-solid-secondary, #363636)';
/** Вторичный непрозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineSolidSecondaryHover = 'var(--outline-on-dark-outline-solid-secondary-hover, #FFFFFFFF)';
/** Вторичный непрозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineSolidSecondaryActive = 'var(--outline-on-dark-outline-solid-secondary-active, #A1A1A1FF)';
/** Третичный непрозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineSolidTertiary = 'var(--outline-on-dark-outline-solid-tertiary, #4E4E4E)';
/** Третичный непрозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineSolidTertiaryHover = 'var(--outline-on-dark-outline-solid-tertiary-hover, #FFFFFFFF)';
/** Третичный непрозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineSolidTertiaryActive = 'var(--outline-on-dark-outline-solid-tertiary-active, #8C8C8CFF)';
/** Прозрачный цвет обводки по умолчанию на темном фоне */
exports.outlineOnDarkOutlineTransparentDefault = 'var(--outline-on-dark-outline-transparent-default, #F9F9F9F5)';
/** Прозрачный цвет обводки по умолчанию на темном фоне */
exports.outlineOnDarkOutlineTransparentDefaultHover = 'var(--outline-on-dark-outline-transparent-default-hover, #FAFAFA93)';
/** Прозрачный цвет обводки по умолчанию на темном фоне */
exports.outlineOnDarkOutlineTransparentDefaultActive = 'var(--outline-on-dark-outline-transparent-default-active, #FAFAFAC4)';
/** Основной прозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineTransparentPrimary = 'var(--outline-on-dark-outline-transparent-primary, #F9F9F91F)';
/** Основной прозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineTransparentPrimaryHover = 'var(--outline-on-dark-outline-transparent-primary-hover, #FAFAFAFF)';
/** Основной прозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineTransparentPrimaryActive = 'var(--outline-on-dark-outline-transparent-primary-active, #FAFAFA25)';
/** Вторичный прозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineTransparentSecondary = 'var(--outline-on-dark-outline-transparent-secondary, #F9F9F933)';
/** Вторичный прозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineTransparentSecondaryHover = 'var(--outline-on-dark-outline-transparent-secondary-hover, #FAFAFAFF)';
/** Вторичный прозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineTransparentSecondaryActive = 'var(--outline-on-dark-outline-transparent-secondary-active, #FAFAFA3D)';
/** Третичный прозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineTransparentTertiary = 'var(--outline-on-dark-outline-transparent-tertiary, #F9F9F947)';
/** Третичный прозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineTransparentTertiaryHover = 'var(--outline-on-dark-outline-transparent-tertiary-hover, #FAFAFAFF)';
/** Третичный прозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineTransparentTertiaryActive = 'var(--outline-on-dark-outline-transparent-tertiary-active, #FAFAFA56)';
/** Бесцветная обводка на темном фоне */
exports.outlineOnDarkOutlineClear = 'var(--outline-on-dark-outline-clear, #FFFFFF00)';
/** Бесцветная обводка на темном фоне */
exports.outlineOnDarkOutlineClearHover = 'var(--outline-on-dark-outline-clear-hover, #FFFFFF00)';
/** Бесцветная обводка на темном фоне */
exports.outlineOnDarkOutlineClearActive = 'var(--outline-on-dark-outline-clear-active, #FFFFFF00)';
/** Акцентный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineAccent = 'var(--outline-on-dark-outline-accent, #1A9E32)';
/** Акцентный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineAccentHover = 'var(--outline-on-dark-outline-accent-hover, #1EB83AFF)';
/** Акцентный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineAccentActive = 'var(--outline-on-dark-outline-accent-active, #15842AFF)';
/** Акцентный минорный непрозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineAccentMinor = 'var(--outline-on-dark-outline-accent-minor, #0A4014)';
/** Акцентный минорный непрозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineAccentMinorHover = 'var(--outline-on-dark-outline-accent-minor-hover, #137725FF)';
/** Акцентный минорный непрозрачный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineAccentMinorActive = 'var(--outline-on-dark-outline-accent-minor-active, #0E5D1DFF)';
/** Прозрачный акцентный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineTransparentAccent = 'var(--outline-on-dark-outline-transparent-accent, #1A9E3266)';
/** Прозрачный акцентный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineTransparentAccentHover = 'var(--outline-on-dark-outline-transparent-accent-hover, #1A9E32FF)';
/** Прозрачный акцентный цвет обводки на темном фоне */
exports.outlineOnDarkOutlineTransparentAccentActive = 'var(--outline-on-dark-outline-transparent-accent-active, #1A9E327A)';
/** Цвет обводки успех на темном фоне */
exports.outlineOnDarkOutlinePositive = 'var(--outline-on-dark-outline-positive, #1A9E32)';
/** Цвет обводки успех на темном фоне */
exports.outlineOnDarkOutlinePositiveHover = 'var(--outline-on-dark-outline-positive-hover, #1EB83AFF)';
/** Цвет обводки успех на темном фоне */
exports.outlineOnDarkOutlinePositiveActive = 'var(--outline-on-dark-outline-positive-active, #15842AFF)';
/** Цвет обводки предупреждение на темном фоне */
exports.outlineOnDarkOutlineWarning = 'var(--outline-on-dark-outline-warning, #FA5F05)';
/** Цвет обводки предупреждение на темном фоне */
exports.outlineOnDarkOutlineWarningHover = 'var(--outline-on-dark-outline-warning-hover, #FB7223FF)';
/** Цвет обводки предупреждение на темном фоне */
exports.outlineOnDarkOutlineWarningActive = 'var(--outline-on-dark-outline-warning-active, #DC5304FF)';
/** Цвет обводки ошибка на темном фоне */
exports.outlineOnDarkOutlineNegative = 'var(--outline-on-dark-outline-negative, #FF293E)';
/** Цвет обводки ошибка на темном фоне */
exports.outlineOnDarkOutlineNegativeHover = 'var(--outline-on-dark-outline-negative-hover, #FF475AFF)';
/** Цвет обводки ошибка на темном фоне */
exports.outlineOnDarkOutlineNegativeActive = 'var(--outline-on-dark-outline-negative-active, #FF0A23FF)';
/** Цвет обводки информация на темном фоне */
exports.outlineOnDarkOutlineInfo = 'var(--outline-on-dark-outline-info, #118CDF)';
/** Цвет обводки информация на темном фоне */
exports.outlineOnDarkOutlineInfoHover = 'var(--outline-on-dark-outline-info-hover, #50B1F2FF)';
/** Цвет обводки информация на темном фоне */
exports.outlineOnDarkOutlineInfoActive = 'var(--outline-on-dark-outline-info-active, #0F7BC2FF)';
/** Минорный цвет обводки успех на темном фоне */
exports.outlineOnDarkOutlinePositiveMinor = 'var(--outline-on-dark-outline-positive-minor, #0A4014)';
/** Минорный цвет обводки успех на темном фоне */
exports.outlineOnDarkOutlinePositiveMinorHover = 'var(--outline-on-dark-outline-positive-minor-hover, #137725FF)';
/** Минорный цвет обводки успех на темном фоне */
exports.outlineOnDarkOutlinePositiveMinorActive = 'var(--outline-on-dark-outline-positive-minor-active, #0E5D1DFF)';
/** Минорный цвет обводки предупреждение на темном фоне */
exports.outlineOnDarkOutlineWarningMinor = 'var(--outline-on-dark-outline-warning-minor, #5C2709)';
/** Минорный цвет обводки предупреждение на темном фоне */
exports.outlineOnDarkOutlineWarningMinorHover = 'var(--outline-on-dark-outline-warning-minor-hover, #95400FFF)';
/** Минорный цвет обводки предупреждение на темном фоне */
exports.outlineOnDarkOutlineWarningMinorActive = 'var(--outline-on-dark-outline-warning-minor-active, #79340CFF)';
/** Минорный цвет обводки ошибка на темном фоне */
exports.outlineOnDarkOutlineNegativeMinor = 'var(--outline-on-dark-outline-negative-minor, #6B1019)';
/** Минорный цвет обводки ошибка на темном фоне */
exports.outlineOnDarkOutlineNegativeMinorHover = 'var(--outline-on-dark-outline-negative-minor-hover, #A01825FF)';
/** Минорный цвет обводки ошибка на темном фоне */
exports.outlineOnDarkOutlineNegativeMinorActive = 'var(--outline-on-dark-outline-negative-minor-active, #85141FFF)';
/** Минорный цвет обводки информация на темном фоне */
exports.outlineOnDarkOutlineInfoMinor = 'var(--outline-on-dark-outline-info-minor, #0D3B59)';
/** Минорный цвет обводки информация на темном фоне */
exports.outlineOnDarkOutlineInfoMinorHover = 'var(--outline-on-dark-outline-info-minor-hover, #145E8FFF)';
/** Минорный цвет обводки информация на темном фоне */
exports.outlineOnDarkOutlineInfoMinorActive = 'var(--outline-on-dark-outline-info-minor-active, #114C74FF)';
/** Прозрачный цвет обводки успех на темном фоне */
exports.outlineOnDarkOutlineTransparentPositive = 'var(--outline-on-dark-outline-transparent-positive, #1A9E3247)';
/** Прозрачный цвет обводки успех на темном фоне */
exports.outlineOnDarkOutlineTransparentPositiveHover = 'var(--outline-on-dark-outline-transparent-positive-hover, #1A9E32FF)';
/** Прозрачный цвет обводки успех на темном фоне */
exports.outlineOnDarkOutlineTransparentPositiveActive = 'var(--outline-on-dark-outline-transparent-positive-active, #1A9E3256)';
/** Прозрачный цвет обводки предупреждение на темном фоне */
exports.outlineOnDarkOutlineTransparentWarning = 'var(--outline-on-dark-outline-transparent-warning, #FA5F0547)';
/** Прозрачный цвет обводки предупреждение на темном фоне */
exports.outlineOnDarkOutlineTransparentWarningHover = 'var(--outline-on-dark-outline-transparent-warning-hover, #FA5F05FF)';
/** Прозрачный цвет обводки предупреждение на темном фоне */
exports.outlineOnDarkOutlineTransparentWarningActive = 'var(--outline-on-dark-outline-transparent-warning-active, #FA5F0556)';
/** Прозрачный цвет обводки предупреждение на темном фоне */
exports.outlineOnDarkOutlineTransparentNegative = 'var(--outline-on-dark-outline-transparent-negative, #FF293E47)';
/** Прозрачный цвет обводки предупреждение на темном фоне */
exports.outlineOnDarkOutlineTransparentNegativeHover = 'var(--outline-on-dark-outline-transparent-negative-hover, #FF293EFF)';
/** Прозрачный цвет обводки предупреждение на темном фоне */
exports.outlineOnDarkOutlineTransparentNegativeActive = 'var(--outline-on-dark-outline-transparent-negative-active, #FF293E56)';
/** Прозрачный цвет обводки информация на темном фоне */
exports.outlineOnDarkOutlineTransparentInfo = 'var(--outline-on-dark-outline-transparent-info, #118CDF47)';
/** Прозрачный цвет обводки информация на темном фоне */
exports.outlineOnDarkOutlineTransparentInfoHover = 'var(--outline-on-dark-outline-transparent-info-hover, #118CDFFF)';
/** Прозрачный цвет обводки информация на темном фоне */
exports.outlineOnDarkOutlineTransparentInfoActive = 'var(--outline-on-dark-outline-transparent-info-active, #118CDF56)';
/** Основной непрозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineSolidPrimary = 'var(--outline-on-light-outline-solid-primary, #DDDDDD)';
/** Основной непрозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineSolidPrimaryHover = 'var(--outline-on-light-outline-solid-primary-hover, #ABABABFF)';
/** Основной непрозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineSolidPrimaryActive = 'var(--outline-on-light-outline-solid-primary-active, #C4C4C4FF)';
/** Вторичный непрозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineSolidSecondary = 'var(--outline-on-light-outline-solid-secondary, #B3B3B3)';
/** Вторичный непрозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineSolidSecondaryHover = 'var(--outline-on-light-outline-solid-secondary-hover, #FFFFFFFF)';
/** Вторичный непрозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineSolidSecondaryActive = 'var(--outline-on-light-outline-solid-secondary-active, #3D3D3DFF)';
/** Третичный непрозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineSolidTertiary = 'var(--outline-on-light-outline-solid-tertiary, #707070)';
/** Третичный непрозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineSolidTertiaryHover = 'var(--outline-on-light-outline-solid-tertiary-hover, #FFFFFFFF)';
/** Третичный непрозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineSolidTertiaryActive = 'var(--outline-on-light-outline-solid-tertiary-active, #737373FF)';
/** Прозрачный цвет обводки по умолчанию на светлом фоне */
exports.outlineOnLightOutlineTransparentDefault = 'var(--outline-on-light-outline-transparent-default, #080808F5)';
/** Прозрачный цвет обводки по умолчанию на светлом фоне */
exports.outlineOnLightOutlineTransparentDefaultHover = 'var(--outline-on-light-outline-transparent-default-hover, #08080893)';
/** Прозрачный цвет обводки по умолчанию на светлом фоне */
exports.outlineOnLightOutlineTransparentDefaultActive = 'var(--outline-on-light-outline-transparent-default-active, #080808C4)';
/** Основной прозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineTransparentPrimary = 'var(--outline-on-light-outline-transparent-primary, #0808081F)';
/** Основной прозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineTransparentPrimaryHover = 'var(--outline-on-light-outline-transparent-primary-hover, #080808FF)';
/** Основной прозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineTransparentPrimaryActive = 'var(--outline-on-light-outline-transparent-primary-active, #08080825)';
/** Вторичный прозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineTransparentSecondary = 'var(--outline-on-light-outline-transparent-secondary, #08080847)';
/** Вторичный прозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineTransparentSecondaryHover = 'var(--outline-on-light-outline-transparent-secondary-hover, #080808FF)';
/** Вторичный прозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineTransparentSecondaryActive = 'var(--outline-on-light-outline-transparent-secondary-active, #08080856)';
/** Третичный прозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineTransparentTertiary = 'var(--outline-on-light-outline-transparent-tertiary, #0808088F)';
/** Третичный прозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineTransparentTertiaryHover = 'var(--outline-on-light-outline-transparent-tertiary-hover, #080808FF)';
/** Третичный прозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineTransparentTertiaryActive = 'var(--outline-on-light-outline-transparent-tertiary-active, #080808AB)';
/** Бесцветная обводка на светлом фоне */
exports.outlineOnLightOutlineClear = 'var(--outline-on-light-outline-clear, #FFFFFF00)';
/** Бесцветная обводка на светлом фоне */
exports.outlineOnLightOutlineClearHover = 'var(--outline-on-light-outline-clear-hover, #FFFFFF00)';
/** Бесцветная обводка на светлом фоне */
exports.outlineOnLightOutlineClearActive = 'var(--outline-on-light-outline-clear-active, #FFFFFF00)';
/** Акцентный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineAccent = 'var(--outline-on-light-outline-accent, #108E26)';
/** Акцентный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineAccentHover = 'var(--outline-on-light-outline-accent-hover, #13AA2CFF)';
/** Акцентный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineAccentActive = 'var(--outline-on-light-outline-accent-active, #0D731EFF)';
/** Акцентный минорный непрозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineAccentMinor = 'var(--outline-on-light-outline-accent-minor, #45E362)';
/** Акцентный минорный непрозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineAccentMinorHover = 'var(--outline-on-light-outline-accent-minor-hover, #5FE778FF)';
/** Акцентный минорный непрозрачный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineAccentMinorActive = 'var(--outline-on-light-outline-accent-minor-active, #2ADF4BFF)';
/** Прозрачный акцентный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineTransparentAccent = 'var(--outline-on-light-outline-transparent-accent, #108E2666)';
/** Прозрачный акцентный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineTransparentAccentHover = 'var(--outline-on-light-outline-transparent-accent-hover, #108E25FF)';
/** Прозрачный акцентный цвет обводки на светлом фоне */
exports.outlineOnLightOutlineTransparentAccentActive = 'var(--outline-on-light-outline-transparent-accent-active, #108E257A)';
/** Цвет обводки успех на светлом фоне */
exports.outlineOnLightOutlinePositive = 'var(--outline-on-light-outline-positive, #108E26)';
/** Цвет обводки успех на светлом фоне */
exports.outlineOnLightOutlinePositiveHover = 'var(--outline-on-light-outline-positive-hover, #13AA2CFF)';
/** Цвет обводки успех на светлом фоне */
exports.outlineOnLightOutlinePositiveActive = 'var(--outline-on-light-outline-positive-active, #0D731EFF)';
/** Цвет обводки предупреждение на светлом фоне */
exports.outlineOnLightOutlineWarning = 'var(--outline-on-light-outline-warning, #E85702)';
/** Цвет обводки предупреждение на светлом фоне */
exports.outlineOnLightOutlineWarningHover = 'var(--outline-on-light-outline-warning-hover, #FD650DFF)';
/** Цвет обводки предупреждение на светлом фоне */
exports.outlineOnLightOutlineWarningActive = 'var(--outline-on-light-outline-warning-active, #CA4B02FF)';
/** Цвет обводки ошибка на светлом фоне */
exports.outlineOnLightOutlineNegative = 'var(--outline-on-light-outline-negative, #F31B31)';
/** Цвет обводки ошибка на светлом фоне */
exports.outlineOnLightOutlineNegativeHover = 'var(--outline-on-light-outline-negative-hover, #F5384BFF)';
/** Цвет обводки ошибка на светлом фоне */
exports.outlineOnLightOutlineNegativeActive = 'var(--outline-on-light-outline-negative-active, #E40C22FF)';
/** Цвет обводки информация на светлом фоне */
exports.outlineOnLightOutlineInfo = 'var(--outline-on-light-outline-info, #0B7ECB)';
/** Цвет обводки информация на светлом фоне */
exports.outlineOnLightOutlineInfoHover = 'var(--outline-on-light-outline-info-hover, #34A7F4FF)';
/** Цвет обводки информация на светлом фоне */
exports.outlineOnLightOutlineInfoActive = 'var(--outline-on-light-outline-info-active, #096CAEFF)';
/** Минорный цвет обводки успех на светлом фоне */
exports.outlineOnLightOutlinePositiveMinor = 'var(--outline-on-light-outline-positive-minor, #45E362)';
/** Минорный цвет обводки успех на светлом фоне */
exports.outlineOnLightOutlinePositiveMinorHover = 'var(--outline-on-light-outline-positive-minor-hover, #5FE778FF)';
/** Минорный цвет обводки успех на светлом фоне */
exports.outlineOnLightOutlinePositiveMinorActive = 'var(--outline-on-light-outline-positive-minor-active, #2ADF4BFF)';
/** Минорный цвет обводки предупреждение на светлом фоне */
exports.outlineOnLightOutlineWarningMinor = 'var(--outline-on-light-outline-warning-minor, #FFB38A)';
/** Минорный цвет обводки предупреждение на светлом фоне */
exports.outlineOnLightOutlineWarningMinorHover = 'var(--outline-on-light-outline-warning-minor-hover, #FFC7A8FF)';
/** Минорный цвет обводки предупреждение на светлом фоне */
exports.outlineOnLightOutlineWarningMinorActive = 'var(--outline-on-light-outline-warning-minor-active, #FF9F6BFF)';
/** Минорный цвет обводки ошибка на светлом фоне */
exports.outlineOnLightOutlineNegativeMinor = 'var(--outline-on-light-outline-negative-minor, #FEB3BB)';
/** Минорный цвет обводки ошибка на светлом фоне */
exports.outlineOnLightOutlineNegativeMinorHover = 'var(--outline-on-light-outline-negative-minor-hover, #FD7785FF)';
/** Минорный цвет обводки ошибка на светлом фоне */
exports.outlineOnLightOutlineNegativeMinorActive = 'var(--outline-on-light-outline-negative-minor-active, #FD96A0FF)';
/** Минорный цвет обводки информация на светлом фоне */
exports.outlineOnLightOutlineInfoMinor = 'var(--outline-on-light-outline-info-minor, #85CEFF)';
/** Минорный цвет обводки информация на светлом фоне */
exports.outlineOnLightOutlineInfoMinorHover = 'var(--outline-on-light-outline-info-minor-hover, #D6EFFFFF)';
/** Минорный цвет обводки информация на светлом фоне */
exports.outlineOnLightOutlineInfoMinorActive = 'var(--outline-on-light-outline-info-minor-active, #66C2FFFF)';
/** Прозрачный цвет обводки успех на светлом фоне */
exports.outlineOnLightOutlineTransparentPositive = 'var(--outline-on-light-outline-transparent-positive, #108E2647)';
/** Прозрачный цвет обводки успех на светлом фоне */
exports.outlineOnLightOutlineTransparentPositiveHover = 'var(--outline-on-light-outline-transparent-positive-hover, #108E25FF)';
/** Прозрачный цвет обводки успех на светлом фоне */
exports.outlineOnLightOutlineTransparentPositiveActive = 'var(--outline-on-light-outline-transparent-positive-active, #108E2556)';
/** Прозрачный цвет обводки предупреждение на светлом фоне */
exports.outlineOnLightOutlineTransparentWarning = 'var(--outline-on-light-outline-transparent-warning, #E8570247)';
/** Прозрачный цвет обводки предупреждение на светлом фоне */
exports.outlineOnLightOutlineTransparentWarningHover = 'var(--outline-on-light-outline-transparent-warning-hover, #E85702FF)';
/** Прозрачный цвет обводки предупреждение на светлом фоне */
exports.outlineOnLightOutlineTransparentWarningActive = 'var(--outline-on-light-outline-transparent-warning-active, #E8570256)';
/** Прозрачный цвет обводки предупреждение на светлом фоне */
exports.outlineOnLightOutlineTransparentNegative = 'var(--outline-on-light-outline-transparent-negative, #F31B3147)';
/** Прозрачный цвет обводки предупреждение на светлом фоне */
exports.outlineOnLightOutlineTransparentNegativeHover = 'var(--outline-on-light-outline-transparent-negative-hover, #F31B31FF)';
/** Прозрачный цвет обводки предупреждение на светлом фоне */
exports.outlineOnLightOutlineTransparentNegativeActive = 'var(--outline-on-light-outline-transparent-negative-active, #F31B3156)';
/** Прозрачный цвет обводки информация на светлом фоне */
exports.outlineOnLightOutlineTransparentInfo = 'var(--outline-on-light-outline-transparent-info, #0B7ECB47)';
/** Прозрачный цвет обводки информация на светлом фоне */
exports.outlineOnLightOutlineTransparentInfoHover = 'var(--outline-on-light-outline-transparent-info-hover, #0B7ECBFF)';
/** Прозрачный цвет обводки информация на светлом фоне */
exports.outlineOnLightOutlineTransparentInfoActive = 'var(--outline-on-light-outline-transparent-info-active, #0B7ECB56)';
/** Инвертированный основной непрозрачный цвет обводки */
exports.outlineInverseOutlineSolidPrimary = 'var(--outline-inverse-outline-solid-primary, #DDDDDD)';
/** Инвертированный основной непрозрачный цвет обводки */
exports.outlineInverseOutlineSolidPrimaryHover = 'var(--outline-inverse-outline-solid-primary-hover, #ABABABFF)';
/** Инвертированный основной непрозрачный цвет обводки */
exports.outlineInverseOutlineSolidPrimaryActive = 'var(--outline-inverse-outline-solid-primary-active, #C4C4C4FF)';
/** Инвертированный вторичный непрозрачный цвет обводки */
exports.outlineInverseOutlineSolidSecondary = 'var(--outline-inverse-outline-solid-secondary, #B3B3B3)';
/** Инвертированный вторичный непрозрачный цвет обводки */
exports.outlineInverseOutlineSolidSecondaryHover = 'var(--outline-inverse-outline-solid-secondary-hover, #FFFFFFFF)';
/** Инвертированный вторичный непрозрачный цвет обводки */
exports.outlineInverseOutlineSolidSecondaryActive = 'var(--outline-inverse-outline-solid-secondary-active, #3D3D3DFF)';
/** Инвертированный третичный непрозрачный цвет обводки */
exports.outlineInverseOutlineSolidTertiary = 'var(--outline-inverse-outline-solid-tertiary, #707070)';
/** Инвертированный третичный непрозрачный цвет обводки */
exports.outlineInverseOutlineSolidTertiaryHover = 'var(--outline-inverse-outline-solid-tertiary-hover, #FFFFFFFF)';
/** Инвертированный третичный непрозрачный цвет обводки */
exports.outlineInverseOutlineSolidTertiaryActive = 'var(--outline-inverse-outline-solid-tertiary-active, #737373FF)';
/** Инвертированный прозрачный цвет обводки по умолчанию */
exports.outlineInverseOutlineTransparentDefault = 'var(--outline-inverse-outline-transparent-default, #080808F5)';
/** Инвертированный прозрачный цвет обводки по умолчанию */
exports.outlineInverseOutlineTransparentDefaultHover = 'var(--outline-inverse-outline-transparent-default-hover, #08080893)';
/** Инвертированный прозрачный цвет обводки по умолчанию */
exports.outlineInverseOutlineTransparentDefaultActive = 'var(--outline-inverse-outline-transparent-default-active, #080808C4)';
/** Инвертированный основной прозрачный цвет обводки */
exports.outlineInverseOutlineTransparentPrimary = 'var(--outline-inverse-outline-transparent-primary, #0808081F)';
/** Инвертированный основной прозрачный цвет обводки */
exports.outlineInverseOutlineTransparentPrimaryHover = 'var(--outline-inverse-outline-transparent-primary-hover, #080808FF)';
/** Инвертированный основной прозрачный цвет обводки */
exports.outlineInverseOutlineTransparentPrimaryActive = 'var(--outline-inverse-outline-transparent-primary-active, #08080825)';
/** Инвертированный вторичный прозрачный цвет обводки */
exports.outlineInverseOutlineTransparentSecondary = 'var(--outline-inverse-outline-transparent-secondary, #08080847)';
/** Инвертированный вторичный прозрачный цвет обводки */
exports.outlineInverseOutlineTransparentSecondaryHover = 'var(--outline-inverse-outline-transparent-secondary-hover, #080808FF)';
/** Инвертированный вторичный прозрачный цвет обводки */
exports.outlineInverseOutlineTransparentSecondaryActive = 'var(--outline-inverse-outline-transparent-secondary-active, #08080856)';
/** Инвертированный третичный прозрачный цвет обводки */
exports.outlineInverseOutlineTransparentTertiary = 'var(--outline-inverse-outline-transparent-tertiary, #0808088F)';
/** Инвертированный третичный прозрачный цвет обводки */
exports.outlineInverseOutlineTransparentTertiaryHover = 'var(--outline-inverse-outline-transparent-tertiary-hover, #080808FF)';
/** Инвертированный третичный прозрачный цвет обводки */
exports.outlineInverseOutlineTransparentTertiaryActive = 'var(--outline-inverse-outline-transparent-tertiary-active, #080808AB)';
/** Инвертированная бесцветная обводка */
exports.outlineInverseOutlineClear = 'var(--outline-inverse-outline-clear, #FFFFFF00)';
/** Инвертированная бесцветная обводка */
exports.outlineInverseOutlineClearHover = 'var(--outline-inverse-outline-clear-hover, #FFFFFF00)';
/** Инвертированная бесцветная обводка */
exports.outlineInverseOutlineClearActive = 'var(--outline-inverse-outline-clear-active, #FFFFFF00)';
/** Инвертированный акцентный цвет обводки */
exports.outlineInverseOutlineAccent = 'var(--outline-inverse-outline-accent, #108E26)';
/** Инвертированный акцентный цвет обводки */
exports.outlineInverseOutlineAccentHover = 'var(--outline-inverse-outline-accent-hover, #13AA2CFF)';
/** Инвертированный акцентный цвет обводки */
exports.outlineInverseOutlineAccentActive = 'var(--outline-inverse-outline-accent-active, #0D731EFF)';
/** Инвертированный акцентный минорный непрозрачный цвет обводки */
exports.outlineInverseOutlineAccentMinor = 'var(--outline-inverse-outline-accent-minor, #45E362)';
/** Инвертированный акцентный минорный непрозрачный цвет обводки */
exports.outlineInverseOutlineAccentMinorHover = 'var(--outline-inverse-outline-accent-minor-hover, #5FE778FF)';
/** Инвертированный акцентный минорный непрозрачный цвет обводки */
exports.outlineInverseOutlineAccentMinorActive = 'var(--outline-inverse-outline-accent-minor-active, #2ADF4BFF)';
/** Прозрачный инвертированный акцентный цвет обводки */
exports.outlineInverseOutlineTransparentAccent = 'var(--outline-inverse-outline-transparent-accent, #108E2666)';
/** Прозрачный инвертированный акцентный цвет обводки */
exports.outlineInverseOutlineTransparentAccentHover = 'var(--outline-inverse-outline-transparent-accent-hover, #108E25FF)';
/** Прозрачный инвертированный акцентный цвет обводки */
exports.outlineInverseOutlineTransparentAccentActive = 'var(--outline-inverse-outline-transparent-accent-active, #108E257A)';
/** Инвертированный цвет обводки успех */
exports.outlineInverseOutlinePositive = 'var(--outline-inverse-outline-positive, #108E26)';
/** Инвертированный цвет обводки успех */
exports.outlineInverseOutlinePositiveHover = 'var(--outline-inverse-outline-positive-hover, #13AA2CFF)';
/** Инвертированный цвет обводки успех */
exports.outlineInverseOutlinePositiveActive = 'var(--outline-inverse-outline-positive-active, #0D731EFF)';
/** Инвертированный цвет обводки предупреждение */
exports.outlineInverseOutlineWarning = 'var(--outline-inverse-outline-warning, #E85702)';
/** Инвертированный цвет обводки предупреждение */
exports.outlineInverseOutlineWarningHover = 'var(--outline-inverse-outline-warning-hover, #FD650DFF)';
/** Инвертированный цвет обводки предупреждение */
exports.outlineInverseOutlineWarningActive = 'var(--outline-inverse-outline-warning-active, #CA4B02FF)';
/** Инвертированный цвет обводки ошибка */
exports.outlineInverseOutlineNegative = 'var(--outline-inverse-outline-negative, #F31B31)';
/** Инвертированный цвет обводки ошибка */
exports.outlineInverseOutlineNegativeHover = 'var(--outline-inverse-outline-negative-hover, #F5384BFF)';
/** Инвертированный цвет обводки ошибка */
exports.outlineInverseOutlineNegativeActive = 'var(--outline-inverse-outline-negative-active, #E40C22FF)';
/** Инвертированный цвет обводки информация */
exports.outlineInverseOutlineInfo = 'var(--outline-inverse-outline-info, #0B7ECB)';
/** Инвертированный цвет обводки информация */
exports.outlineInverseOutlineInfoHover = 'var(--outline-inverse-outline-info-hover, #34A7F4FF)';
/** Инвертированный цвет обводки информация */
exports.outlineInverseOutlineInfoActive = 'var(--outline-inverse-outline-info-active, #096CAEFF)';
/** Инвертированный минорный цвет обводки успех */
exports.outlineInverseOutlinePositiveMinor = 'var(--outline-inverse-outline-positive-minor, #45E362)';
/** Инвертированный минорный цвет обводки успех */
exports.outlineInverseOutlinePositiveMinorHover = 'var(--outline-inverse-outline-positive-minor-hover, #5FE778FF)';
/** Инвертированный минорный цвет обводки успех */
exports.outlineInverseOutlinePositiveMinorActive = 'var(--outline-inverse-outline-positive-minor-active, #2ADF4BFF)';
/** Инвертированный минорный цвет обводки предупреждение */
exports.outlineInverseOutlineWarningMinor = 'var(--outline-inverse-outline-warning-minor, #FFB38A)';
/** Инвертированный минорный цвет обводки предупреждение */
exports.outlineInverseOutlineWarningMinorHover = 'var(--outline-inverse-outline-warning-minor-hover, #FFC7A8FF)';
/** Инвертированный минорный цвет обводки предупреждение */
exports.outlineInverseOutlineWarningMinorActive = 'var(--outline-inverse-outline-warning-minor-active, #FF9F6BFF)';
/** Инвертированный минорный цвет обводки ошибка */
exports.outlineInverseOutlineNegativeMinor = 'var(--outline-inverse-outline-negative-minor, #FEB3BB)';
/** Инвертированный минорный цвет обводки ошибка */
exports.outlineInverseOutlineNegativeMinorHover = 'var(--outline-inverse-outline-negative-minor-hover, #FD7785FF)';
/** Инвертированный минорный цвет обводки ошибка */
exports.outlineInverseOutlineNegativeMinorActive = 'var(--outline-inverse-outline-negative-minor-active, #FD96A0FF)';
/** Инвертированный минорный цвет обводки информация */
exports.outlineInverseOutlineInfoMinor = 'var(--outline-inverse-outline-info-minor, #85CEFF)';
/** Инвертированный минорный цвет обводки информация */
exports.outlineInverseOutlineInfoMinorHover = 'var(--outline-inverse-outline-info-minor-hover, #D6EFFFFF)';
/** Инвертированный минорный цвет обводки информация */
exports.outlineInverseOutlineInfoMinorActive = 'var(--outline-inverse-outline-info-minor-active, #66C2FFFF)';
/** Прозрачный инвертированный цвет обводки успех */
exports.outlineInverseOutlineTransparentPositive = 'var(--outline-inverse-outline-transparent-positive, #108E2647)';
/** Прозрачный инвертированный цвет обводки успех */
exports.outlineInverseOutlineTransparentPositiveHover = 'var(--outline-inverse-outline-transparent-positive-hover, #108E25FF)';
/** Прозрачный инвертированный цвет обводки успех */
exports.outlineInverseOutlineTransparentPositiveActive = 'var(--outline-inverse-outline-transparent-positive-active, #108E2556)';
/** Прозрачный инвертированный цвет обводки предупреждение */
exports.outlineInverseOutlineTransparentWarning = 'var(--outline-inverse-outline-transparent-warning, #E8570247)';
/** Прозрачный инвертированный цвет обводки предупреждение */
exports.outlineInverseOutlineTransparentWarningHover = 'var(--outline-inverse-outline-transparent-warning-hover, #E85702FF)';
/** Прозрачный инвертированный цвет обводки предупреждение */
exports.outlineInverseOutlineTransparentWarningActive = 'var(--outline-inverse-outline-transparent-warning-active, #E8570256)';
/** Прозрачный инвертированный цвет обводки предупреждение */
exports.outlineInverseOutlineTransparentNegative = 'var(--outline-inverse-outline-transparent-negative, #F31B3147)';
/** Прозрачный инвертированный цвет обводки предупреждение */
exports.outlineInverseOutlineTransparentNegativeHover = 'var(--outline-inverse-outline-transparent-negative-hover, #F31B31FF)';
/** Прозрачный инвертированный цвет обводки предупреждение */
exports.outlineInverseOutlineTransparentNegativeActive = 'var(--outline-inverse-outline-transparent-negative-active, #F31B3156)';
/** Прозрачный инвертированный цвет обводки информация */
exports.outlineInverseOutlineTransparentInfo = 'var(--outline-inverse-outline-transparent-info, #0B7ECB47)';
/** Прозрачный инвертированный цвет обводки информация */
exports.outlineInverseOutlineTransparentInfoHover = 'var(--outline-inverse-outline-transparent-info-hover, #0B7ECBFF)';
/** Прозрачный инвертированный цвет обводки информация */
exports.outlineInverseOutlineTransparentInfoActive = 'var(--outline-inverse-outline-transparent-info-active, #0B7ECB56)';
exports.skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
exports.skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use onLightSurfaceSolidDefault */
exports.buttonBlack = 'var(--plasma-colors-button-black, var(--on-light-surface-solid-default))';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
exports.buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, var(--on-light-surface-transparent-secondary))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
exports.buttonWhite = 'var(--plasma-colors-button-white, var(--on-dark-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
exports.buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, var(--on-dark-surface-transparent-secondary))';
/** @deprecated instead use textPrimary */
exports.text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
exports.primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
exports.secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
exports.tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use textParagraph */
exports.paragraph = 'var(--plasma-colors-paragraph, var(--text-paragraph))';
/** @deprecated instead use backgroundPrimary */
exports.background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textAccent */
exports.accent = 'var(--plasma-colors-accent, var(--text-accent))';
/** @deprecated instead use textPositive */
exports.success = 'var(--plasma-colors-success, var(--text-positive))';
/** @deprecated instead use textWarning */
exports.warning = 'var(--plasma-colors-warning, var(--text-warning))';
/** @deprecated instead use textNegative */
exports.critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use overlaySoft */
exports.overlay = 'var(--plasma-colors-overlay, var(--overlay-soft))';
/** @deprecated instead use surfaceTransparentPrimary */
exports.surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, var(--surface-transparent-primary))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, var(--surface-transparent-secondary))';
/** @deprecated instead use surfaceTransparentTertiary */
exports.surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, var(--surface-transparent-tertiary))';
/** @deprecated instead use surfaceSolidPrimary */
exports.surfaceSolid01 = 'var(--plasma-colors-surface-solid01, var(--surface-solid-primary))';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use surfaceTransparentCard */
exports.surfaceCard = 'var(--plasma-colors-surface-card, var(--surface-transparent-card))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.buttonSecondary = 'var(--plasma-colors-button-secondary, var(--surface-transparent-secondary))';
/** @deprecated instead use textAccent */
exports.buttonAccent = 'var(--plasma-colors-button-accent, var(--text-accent))';
/** @deprecated instead use surfacePositive */
exports.buttonSuccess = 'var(--plasma-colors-button-success, var(--surface-positive))';
/** @deprecated instead use surfaceWarning */
exports.buttonWarning = 'var(--plasma-colors-button-warning, var(--surface-warning))';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
exports.shadowDownSoftS = 'var(--shadow-down-soft-s, 0px 4px 14px -4px rgba(8, 8, 8, 0.08)), var(--shadow-down-soft-s, 0px 1px 4px -1px rgba(0, 0, 0, 0.04))';
exports.shadowDownSoftM = 'var(--shadow-down-soft-m, 0px 24px 48px -8px rgba(0, 0, 0, 0.08))';
exports.shadowDownSoftL = 'var(--shadow-down-soft-l, 0px 60px 112px -8px rgba(0, 0, 0, 0.08))';
exports.shadowDownHardS = 'var(--shadow-down-hard-s, 0px 4px 12px -3px rgba(8, 8, 8, 0.16)), var(--shadow-down-hard-s, 0px 1px 4px -2px rgba(0, 0, 0, 0.08))';
exports.shadowDownHardM = 'var(--shadow-down-hard-m, 0px 16px 32px -8px rgba(0, 0, 0, 0.24))';
exports.shadowDownHardL = 'var(--shadow-down-hard-l, 0px 60px 112px -8px rgba(0, 0, 0, 0.40))';
exports.shadowUpSoftS = 'var(--shadow-up-soft-s, 0px -4px 14px -4px rgba(8, 8, 8, 0.08)), var(--shadow-up-soft-s, 0px -1px 4px -1px rgba(0, 0, 0, 0.03))';
exports.shadowUpSoftM = 'var(--shadow-up-soft-m, 0px -24px 48px -8px rgba(0, 0, 0, 0.08))';
exports.shadowUpSoftL = 'var(--shadow-up-soft-l, 0px -60px 112px -8px rgba(0, 0, 0, 0.08))';
exports.shadowUpHardS = 'var(--shadow-up-hard-s, 0px -4px 12px -3px rgba(8, 8, 8, 0.20)), var(--shadow-up-hard-s, 0px -1px 4px -1px rgba(0, 0, 0, 0.03))';
exports.shadowUpHardM = 'var(--shadow-up-hard-m, 0px -16px 32px -8px rgba(0, 0, 0, 0.24))';
exports.shadowUpHardL = 'var(--shadow-up-hard-l, 0px -60px 112px -8px rgba(0, 0, 0, 0.40))';
exports.borderRadiusXxs = 'var(--border-radius-xxs, 4px)';
exports.borderRadiusXs = 'var(--border-radius-xs, 6px)';
exports.borderRadiusS = 'var(--border-radius-s, 8px)';
exports.borderRadiusM = 'var(--border-radius-m, 12px)';
exports.borderRadiusL = 'var(--border-radius-l, 16px)';
exports.borderRadiusXl = 'var(--border-radius-xl, 20px)';
exports.borderRadiusXxl = 'var(--border-radius-xxl, 40px)';
