// @salutejs/plasma-ui/components/Button/Button.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Button = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _mixins = /*#__PURE__*/require("../../mixins");
var _hooks = /*#__PURE__*/require("../../hooks");
var _utils = /*#__PURE__*/require("../../utils");
var _Button = /*#__PURE__*/require("./Button.mixins");
var _Button2 = /*#__PURE__*/require("./Button.sizes");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
(0, _collectPackageInfo.collectPackageInfo)('Button/Button');
var StyledButtonRoot = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.ButtonRoot).attrs(_utils.spatnavClassNameAttrs).withConfig({
  componentId: "plasma-ui__sc-vm1boz-0"
})(["", " ", " ", " ", " ", ""], _Button2.sizes, _Button.applySizes, _Button.applyViews, _mixins.applyInteraction, _plasmaCore.applyNoSelect);

/**
 * Основной компонент для отображения кнопок.
 * Поддерживает несколько режимов отображения (`view`) и размеров (`size`).
 */
var ButtonBase = /*#__PURE__*/(0, _plasmaCore.createButton)(StyledButtonRoot);
var Button = exports.Button = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  var _useThemeContext = (0, _hooks.useThemeContext)(),
    deviceScale = _useThemeContext.deviceScale;
  return /*#__PURE__*/_react["default"].createElement(ButtonBase, _extends({}, props, {
    ref: ref,
    deviceScale: deviceScale
  }));
});
Button.defaultProps = {
  size: 'l',
  view: 'secondary',
  pin: 'square-square',
  outlined: true,
  scaleOnInteraction: true
};
