// @salutejs/plasma-core/components/Button/Button.props.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.buttonViews = exports.buttonTypography = exports.buttonSizes = void 0;
var _mixins = /*#__PURE__*/require("../../mixins");
var _tokens = /*#__PURE__*/require("../../tokens");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
/**
 * Цветовые виды.
 */
(0, _collectPackageInfo.collectPackageInfo)('Button/Button.props');
var buttonViews = exports.buttonViews = {
  primary: _mixins.views.primary,
  secondary: _mixins.views.secondary,
  success: _mixins.views.success,
  warning: _mixins.views.warning,
  critical: _mixins.views.critical,
  checked: _mixins.views.checked,
  overlay: _mixins.views.overlay,
  clear: _mixins.views.clear
};

/**
 * Размеры в rem в соответствии с дизайном.
 */
var buttonSizes = exports.buttonSizes = {
  l: {
    height: 'var(--plasma-button-l-height)',
    paddingY: 'var(--plasma-button-l-padding-y)',
    paddingX: 'var(--plasma-button-l-padding-x)',
    paddingContentX: 'var(--plasma-button-l-padding-content-x)',
    paddingStretchX: 'var(--plasma-button-l-padding-stretch-x)',
    radius: 'var(--plasma-button-l-radius)',
    radiusCircle: 'var(--plasma-button-l-radius-circle)'
  },
  m: {
    height: 'var(--plasma-button-m-height)',
    paddingY: 'var(--plasma-button-m-padding-y)',
    paddingX: 'var(--plasma-button-m-padding-x)',
    paddingContentX: 'var(--plasma-button-m-padding-content-x)',
    paddingStretchX: 'var(--plasma-button-m-padding-stretch-x)',
    radius: 'var(--plasma-button-m-radius)',
    radiusCircle: 'var(--plasma-button-m-radius-circle)'
  },
  s: {
    height: 'var(--plasma-button-s-height)',
    paddingY: 'var(--plasma-button-s-padding-y)',
    paddingX: 'var(--plasma-button-s-padding-x)',
    paddingContentX: 'var(--plasma-button-s-padding-content-x)',
    paddingStretchX: 'var(--plasma-button-s-padding-stretch-x)',
    radius: 'var(--plasma-button-s-radius)',
    radiusCircle: 'var(--plasma-button-s-radius-circle)'
  }
};

/**
 * Размеры типографики кнопки.
 */
var buttonTypography = exports.buttonTypography = {
  l: _tokens.button1,
  m: _tokens.button1,
  s: _tokens.button2
};
