// @salutejs/plasma-core/components/Button/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Button", {
  enumerable: true,
  get: function get() {
    return _Button.Button;
  }
});
Object.defineProperty(exports, "ButtonRoot", {
  enumerable: true,
  get: function get() {
    return _Button.ButtonRoot;
  }
});
Object.defineProperty(exports, "buttonSizes", {
  enumerable: true,
  get: function get() {
    return _Button2.buttonSizes;
  }
});
Object.defineProperty(exports, "buttonTypography", {
  enumerable: true,
  get: function get() {
    return _Button2.buttonTypography;
  }
});
Object.defineProperty(exports, "buttonViews", {
  enumerable: true,
  get: function get() {
    return _Button2.buttonViews;
  }
});
Object.defineProperty(exports, "createButton", {
  enumerable: true,
  get: function get() {
    return _createButton.createButton;
  }
});
Object.defineProperty(exports, "getButtonSizesMixin", {
  enumerable: true,
  get: function get() {
    return _Button3.getButtonSizesMixin;
  }
});
var _Button = /*#__PURE__*/require("./Button");
var _Button2 = /*#__PURE__*/require("./Button.props");
var _Button3 = /*#__PURE__*/require("./Button.mixins");
var _createButton = /*#__PURE__*/require("./createButton");
