// @salutejs/plasma-core/components/ModalBase/ModalBase.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalBase = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/require("styled-components");
var _PopupBase = /*#__PURE__*/require("../PopupBase/PopupBase");
var _hooks = /*#__PURE__*/require("../../hooks");
var _ModalOverlay = /*#__PURE__*/require("./ModalOverlay");
var _hooks2 = /*#__PURE__*/require("./hooks");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _NoScroll;
var _excluded = ["id", "withAnimation", "onClose", "onOverlayClick", "onEscKeyDown", "closeOnEsc", "closeOnOverlayClick", "withBlur", "initialFocusRef", "focusAfterRef", "zIndex", "popupInfo", "children"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('ModalBase/ModalBase');
var NoScroll = /*#__PURE__*/(0, _styledComponents.createGlobalStyle)(["body{overflow-y:hidden;}"]);

/**
 * ModalBase.
 * Управляет показом/скрытием, подложкой и анимацией визуальной части модального окна.
 */
var ModalBase = exports.ModalBase = function ModalBase(_ref) {
  var id = _ref.id,
    withAnimation = _ref.withAnimation,
    onClose = _ref.onClose,
    onOverlayClick = _ref.onOverlayClick,
    onEscKeyDown = _ref.onEscKeyDown,
    _ref$closeOnEsc = _ref.closeOnEsc,
    closeOnEsc = _ref$closeOnEsc === void 0 ? true : _ref$closeOnEsc,
    _ref$closeOnOverlayCl = _ref.closeOnOverlayClick,
    closeOnOverlayClick = _ref$closeOnOverlayCl === void 0 ? true : _ref$closeOnOverlayCl,
    withBlur = _ref.withBlur,
    initialFocusRef = _ref.initialFocusRef,
    focusAfterRef = _ref.focusAfterRef,
    zIndex = _ref.zIndex,
    popupInfo = _ref.popupInfo,
    children = _ref.children,
    rest = _objectWithoutProperties(_ref, _excluded);
  var trapRef = (0, _hooks.useFocusTrap)(true, initialFocusRef, focusAfterRef);
  var uniqId = (0, _hooks.useUniqId)();
  var innerId = id || uniqId;
  var _useModal = (0, _hooks2.useModal)({
      id: innerId,
      closeOnEsc: closeOnEsc,
      onEscKeyDown: onEscKeyDown,
      onClose: onClose,
      popupInfo: popupInfo
    }),
    modalInfo = _useModal.modalInfo;
  return /*#__PURE__*/_react["default"].createElement(_PopupBase.PopupBase, _extends({
    id: id,
    ref: trapRef,
    popupInfo: modalInfo,
    withAnimation: withAnimation,
    zIndex: zIndex,
    overlay: /*#__PURE__*/_react["default"].createElement(_ModalOverlay.ModalOverlay, {
      id: innerId,
      withBlur: withBlur,
      onOverlayClick: onOverlayClick,
      onClose: onClose,
      zIndex: zIndex,
      closeOnOverlayClick: closeOnOverlayClick
    })
  }, rest), _NoScroll || (_NoScroll = /*#__PURE__*/_react["default"].createElement(NoScroll, null)), children);
};
