// @salutejs/plasma-ui/components/Sheet/Sheet.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Sheet = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _plasmaTokens = /*#__PURE__*/require("@salutejs/plasma-tokens");
var _useSheetSwipe = /*#__PURE__*/require("./useSheetSwipe");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _NoScroll;
var _excluded = ["isOpen", "children", "onClose", "withOverlay", "withTransition", "throttleMs"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('Sheet/Sheet');
var handleTransition = function handleTransition(transition) {
  return function (props) {
    var lowPerformance = props.theme.lowPerformance,
      withTransition = props.withTransition;
    return !withTransition || lowPerformance ? 'unset' : transition;
  };
};
var NoScroll = /*#__PURE__*/(0, _styledComponents.createGlobalStyle)(["body{overflow:hidden;}"]);
var StyledWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-1c9dnkw-0"
})(["position:fixed;top:0;bottom:0;left:0;right:0;opacity:1;transition:", ";z-index:1000;", ""], /*#__PURE__*/handleTransition('all 0.5s 0.1s'), function (_ref) {
  var isOpen = _ref.isOpen;
  return !isOpen && (0, _styledComponents.css)(["opacity:0;visibility:hidden;"]);
});
var StyledContentWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-1c9dnkw-1"
})(["position:absolute;left:0;bottom:0;width:100%;max-height:100%;z-index:1;display:flex;flex-direction:column;justify-content:flex-end;transition:", ";", ""], /*#__PURE__*/handleTransition('transform 0.5s'), function (_ref2) {
  var isOpen = _ref2.isOpen;
  return !isOpen && (0, _styledComponents.css)(["transform:translateY(100%);"]);
});
var StyledOverlay = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-1c9dnkw-2"
})(["position:absolute;top:0;left:0;width:100%;height:100%;background-color:", ";"], function (_ref3) {
  var withOverlay = _ref3.withOverlay;
  return withOverlay ? _plasmaTokens.overlay : 'transparent';
});
var StyledSheetContent = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-1c9dnkw-3"
})(["border-radius:1.5rem 1.5rem 0 0;background-color:", ";padding:1rem;max-height:100%;overflow:auto;overscroll-behavior:contain;&::-webkit-scrollbar{display:none;}"], _plasmaTokens.backgroundPrimary);
var StyledSheetHandle = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-1c9dnkw-4"
})(["width:4rem;padding:1.75rem 0;margin:-1.5rem 0;opacity:0.65;align-self:center;&::before{content:'';display:block;border-radius:0.625rem;background-color:", ";opacity:0.96;height:0.25rem;}"], _plasmaTokens.white);

/**
 * Открывает окно-шторку поверх основного экрана.
 * Только для приложения Салют.
 */
var Sheet = exports.Sheet = function Sheet(_ref4) {
  var isOpen = _ref4.isOpen,
    children = _ref4.children,
    onClose = _ref4.onClose,
    _ref4$withOverlay = _ref4.withOverlay,
    withOverlay = _ref4$withOverlay === void 0 ? true : _ref4$withOverlay,
    _ref4$withTransition = _ref4.withTransition,
    withTransition = _ref4$withTransition === void 0 ? true : _ref4$withTransition,
    throttleMs = _ref4.throttleMs,
    restProps = _objectWithoutProperties(_ref4, _excluded);
  var contentWrapperRef = _react["default"].useRef(null);
  var contentRef = _react["default"].useRef(null);
  var handleRef = _react["default"].useRef(null);
  (0, _useSheetSwipe.useSheetSwipe)({
    contentWrapperRef: contentWrapperRef,
    contentRef: contentRef,
    handleRef: handleRef,
    throttleMs: throttleMs,
    onClose: onClose
  });
  return /*#__PURE__*/_react["default"].createElement(StyledWrapper, _extends({
    isOpen: isOpen,
    withTransition: withTransition
  }, restProps), isOpen && (_NoScroll || (_NoScroll = /*#__PURE__*/_react["default"].createElement(NoScroll, null))), /*#__PURE__*/_react["default"].createElement(StyledContentWrapper, {
    isOpen: isOpen,
    withTransition: withTransition,
    ref: contentWrapperRef
  }, /*#__PURE__*/_react["default"].createElement(StyledSheetHandle, {
    ref: handleRef
  }), /*#__PURE__*/_react["default"].createElement(StyledSheetContent, {
    ref: contentRef
  }, children)), /*#__PURE__*/_react["default"].createElement(StyledOverlay, {
    withOverlay: withOverlay,
    onClick: onClose
  }));
};
