// @salutejs/plasma-new-hope/styled-components/es/components/Cell/Cell.styles.js
import styled from 'styled-components';
import { css } from 'styled-components';
import { tokens, classes } from './Cell.tokens';
var alignToFlex = {
  center: 'center',
  top: 'flex-start',
  bottom: 'flex-end'
};
export var base = /*#__PURE__*/css(["&&.", "{width:var(", ");}&&.", "{width:100%;}&&.", "{width:auto;}"], classes.fixedStretching, tokens.cellWidth, classes.filledStretching, classes.autoStretching);
export var CellContentWrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-16emz3n-0"
})(["display:flex;flex:auto;align-items:stretch;justify-content:space-between;gap:var(", ");height:auto;"], tokens.cellGap);
export var CellContent = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-16emz3n-1"
})(["display:flex;align-items:center;height:auto;"]);
export var CellLeft = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-16emz3n-2"
})(["display:flex;align-items:", ";"], function (_ref) {
  var align = _ref.align;
  return alignToFlex[align];
});
export var CellRight = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-16emz3n-3"
})(["display:flex;align-items:", ";text-align:right;"], function (_ref2) {
  var align = _ref2.align;
  return alignToFlex[align];
});
