// @salutejs/plasma-new-hope/styled-components/es/components/Cell/Cell.js
import React, { forwardRef } from 'react';
import { cx } from '../../utils';
import { base, CellLeft, CellRight, CellContentWrapper, CellContent } from './Cell.styles';
import { classes } from './Cell.tokens';
import { base as viewCSS } from './variations/_view/base';
import { base as sizeCSS } from './variations/_size/base';
import { CellTextbox } from './ui/CellTextbox/CellTextbox';
export var cellRoot = function cellRoot(Root) {
  return /*#__PURE__*/forwardRef(function (_ref, outerRootRef) {
    var size = _ref.size,
      view = _ref.view,
      contentLeft = _ref.contentLeft,
      contentRight = _ref.contentRight,
      _ref$alignContentLeft = _ref.alignContentLeft,
      alignContentLeft = _ref$alignContentLeft === void 0 ? 'center' : _ref$alignContentLeft,
      _ref$alignContentRigh = _ref.alignContentRight,
      alignContentRight = _ref$alignContentRigh === void 0 ? 'center' : _ref$alignContentRigh,
      children = _ref.children,
      title = _ref.title,
      subtitle = _ref.subtitle,
      label = _ref.label,
      description = _ref.description,
      content = _ref.content,
      _ref$stretching = _ref.stretching,
      stretching = _ref$stretching === void 0 ? 'filled' : _ref$stretching;
    var stretchingClass = classes["".concat(stretching, "Stretching")];

    // Импорт  старых свойств
    var titleText = title || description;
    var contentLeftDeprecated = contentLeft || content;
    return /*#__PURE__*/React.createElement(Root, {
      ref: outerRootRef,
      size: size,
      view: view,
      className: cx(stretchingClass, classes.cellRoot)
    }, contentLeftDeprecated && /*#__PURE__*/React.createElement(CellLeft, {
      align: alignContentLeft
    }, contentLeftDeprecated), /*#__PURE__*/React.createElement(CellContentWrapper, null, /*#__PURE__*/React.createElement(CellContent, null, titleText || subtitle || label ? /*#__PURE__*/React.createElement(CellTextbox, {
      label: label,
      title: titleText,
      subtitle: subtitle
    }) : children), contentRight && /*#__PURE__*/React.createElement(CellRight, {
      align: alignContentRight
    }, contentRight)));
  });
};
export var cellConfig = {
  name: 'Cell',
  tag: 'div',
  layout: cellRoot,
  base: base,
  variations: {
    view: {
      css: viewCSS
    },
    size: {
      css: sizeCSS
    }
  },
  defaults: {
    view: 'default',
    size: 'm'
  }
};
