// @salutejs/plasma-new-hope/cjs/components/Drawer/Drawer.styles.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var styled = require('styled-components');
require('@linaria/core');
var common = require('../../engines/common.js');
require('react');
var Popup$1 = require('../Popup/Popup.js');
var Popup_tokens = require('../Popup/Popup.tokens.js');
var Drawer_tokens = require('./Drawer.tokens.js');

var Popup = /*#__PURE__*/common.component(Popup$1.popupConfig);
var animationVariants = {
  right: {
    show: {
      start: 'translate(100%, -50%)',
      end: 'translate(0, -50%)'
    },
    hide: {
      start: 'translate(0, -50%)',
      end: 'translate(100%, -50%)'
    }
  },
  bottom: {
    show: {
      start: 'translate(-50%, 100%)',
      end: 'translate(-50%, 0)'
    },
    hide: {
      start: 'translate(-50%, 0)',
      end: 'translate(-50%, 100%)'
    }
  },
  left: {
    show: {
      start: 'translate(-100%, -50%)',
      end: 'translate(0, -50%)'
    },
    hide: {
      start: 'translate(0, -50%)',
      end: 'translate(-100%, -50%)'
    }
  },
  top: {
    show: {
      start: 'translate(-50%, -100%)',
      end: 'translate(-50%, 0)'
    },
    hide: {
      start: 'translate(-50%, 0)',
      end: 'translate(-50%, -100%)'
    }
  }
};
var StyledPopup = /*#__PURE__*/styled(Popup).withConfig({
  componentId: "plasma-new-hope__sc-rwu6sj-0"
})(["&&.", "{animation:fadeIn 0.2s forwards;}&&.", " .", "{animation:fadeOut 0.2s forwards;}@keyframes fadeIn{from{opacity:0;}to{opacity:1;}}@keyframes fadeOut{from{opacity:1;}to{opacity:0;}}&& .", ",&& .", " > div{width:", ";height:", ";}&& .", "{animation:showAnimation 0.2s forwards;}&&.", " .", "{animation:hideAnimation 0.2s forwards;}@keyframes showAnimation{0%{transform:", ";opacity:0;}100%{transform:", ";opacity:1;}}@keyframes hideAnimation{0%{transform:", ";opacity:1;}100%{transform:", ";opacity:0;}}"], Drawer_tokens.classes.overlay, Popup_tokens.classes.endAnimation, Drawer_tokens.classes.overlay, Popup_tokens.classes.root, Popup_tokens.classes.root, function (_ref) {
  var width = _ref.width;
  return width;
}, function (_ref2) {
  var height = _ref2.height;
  return height;
}, Popup_tokens.classes.root, Popup_tokens.classes.endAnimation, Popup_tokens.classes.root, function (_ref3) {
  var placement = _ref3.placement;
  return animationVariants[placement].show.start;
}, function (_ref4) {
  var placement = _ref4.placement;
  return animationVariants[placement].show.end;
}, function (_ref5) {
  var placement = _ref5.placement;
  return animationVariants[placement].hide.start;
}, function (_ref6) {
  var placement = _ref6.placement;
  return animationVariants[placement].hide.end;
});

exports.StyledPopup = StyledPopup;
