// @salutejs/plasma-new-hope/styled-components/cjs/examples/plasma_b2c/components/Spinner/Spinner.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _Spinner = /*#__PURE__*/require("../../../../components/Spinner");
var config = exports.config = {
  defaults: {
    view: 'accent',
    size: 'm'
  },
  variations: {
    size: {
      s: /*#__PURE__*/(0, _styledComponents.css)(["", ":16px;"], _Spinner.spinnerTokens.size),
      m: /*#__PURE__*/(0, _styledComponents.css)(["", ":24px;"], _Spinner.spinnerTokens.size),
      l: /*#__PURE__*/(0, _styledComponents.css)(["", ":36px;"], _Spinner.spinnerTokens.size)
    },
    view: {
      "default": /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--text-primary);"], _Spinner.spinnerTokens.color),
      secondary: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--text-secondary);"], _Spinner.spinnerTokens.color),
      tertiary: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--text-tertiary);"], _Spinner.spinnerTokens.color),
      paragraph: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--text-paragraph);"], _Spinner.spinnerTokens.color),
      accent: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--text-accent);"], _Spinner.spinnerTokens.color),
      positive: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--text-positive);"], _Spinner.spinnerTokens.color),
      warning: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--text-warning);"], _Spinner.spinnerTokens.color),
      negative: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--text-negative);"], _Spinner.spinnerTokens.color)
    }
  }
};
