// @salutejs/plasma-new-hope/styled-components/cjs/components/Select/Select.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.selectRoot = exports.selectConfig = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _utils = /*#__PURE__*/require("../../utils");
var _utils2 = /*#__PURE__*/require("../Dropdown/utils");
var _hooks = /*#__PURE__*/require("../../hooks");
var _base = /*#__PURE__*/require("./variations/_target/base");
var _base2 = /*#__PURE__*/require("./variations/_view/base");
var _base3 = /*#__PURE__*/require("./variations/_size/base");
var _ui = /*#__PURE__*/require("./ui");
var _utils3 = /*#__PURE__*/require("./utils");
var _Select = /*#__PURE__*/require("./Select.tokens");
var _Select2 = /*#__PURE__*/require("./Select.styles");
var _hooks2 = /*#__PURE__*/require("./hooks");
var _excluded = ["name", "form", "readOnly", "disabled", "label", "id", "children", "role", "target", "view", "size", "frame", "usePortal", "enumerationType", "opened", "placement", "onToggle"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
/**
 * Выпадающий список без внешнего контроля видимости.
 */
var selectRoot = exports.selectRoot = function selectRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (_ref, outerRootRef) {
    var _targetRef$current;
    var name = _ref.name,
      form = _ref.form,
      readOnly = _ref.readOnly,
      disabled = _ref.disabled,
      label = _ref.label,
      id = _ref.id,
      children = _ref.children,
      role = _ref.role,
      target = _ref.target,
      view = _ref.view,
      size = _ref.size,
      frame = _ref.frame,
      usePortal = _ref.usePortal,
      _ref$enumerationType = _ref.enumerationType,
      enumerationType = _ref$enumerationType === void 0 ? 'comma' : _ref$enumerationType,
      _ref$opened = _ref.opened,
      opened = _ref$opened === void 0 ? false : _ref$opened,
      _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? 'bottom' : _ref$placement,
      onToggle = _ref.onToggle,
      rest = _objectWithoutProperties(_ref, _excluded);
    var forceUpdate = (0, _hooks.useForceUpdate)();
    var uniqId = (0, _plasmaCore.safeUseId)();
    var innerId = id || uniqId;
    var _rest$valueType = rest.valueType,
      valueType = _rest$valueType === void 0 ? 'single' : _rest$valueType,
      value = rest.value;
    var offset = [0, 4];
    var _useState = (0, _react.useState)(null),
      _useState2 = _slicedToArray(_useState, 2),
      ref = _useState2[0],
      setRef = _useState2[1];
    var rootRef = (0, _react.useRef)(null);
    var handleRef = (0, _plasmaCore.useForkRef)(rootRef, outerRootRef);
    var targetRef = (0, _react.useRef)(null);
    var chipsRefs = (0, _react.useRef)([]);
    var selectRef = (0, _react.useRef)(null);
    var itemsRefs = (0, _react.useRef)([]);
    var controlledRefs = {
      targetRef: targetRef,
      chipsRefs: chipsRefs,
      selectRef: selectRef,
      itemsRefs: itemsRefs
    };
    var _useState3 = (0, _react.useState)(opened),
      _useState4 = _slicedToArray(_useState3, 2),
      innerOpened = _useState4[0],
      setOpened = _useState4[1];

    // INFO: нужно высчитывать для корректной работы портала
    var dynamicTargetWidth = (_targetRef$current = targetRef.current) === null || _targetRef$current === void 0 ? void 0 : _targetRef$current.offsetWidth;
    var withNativeSelectVisible = valueType === 'native' ? _Select.classes.nativeSelectVisible : undefined;

    // INFO: Из-за того, что классы передаются через ref,
    // состояние пропсов после изменения view, size, target
    // всегда предыдущее, поэтому нужно делать форс-ререндер
    (0, _hooks.useDidMountEffect)(function () {
      forceUpdate();
    }, [view, size, target]);
    (0, _react.useEffect)(function () {
      if (disabled || readOnly) {
        return;
      }
      setOpened(opened);
    }, [opened, disabled, readOnly]);
    var onInnerToggle = (0, _react.useCallback)(function (openValue, event) {
      if (disabled || readOnly) {
        return;
      }
      if (onToggle) {
        onToggle(openValue, event);
        return;
      }
      setOpened(openValue);
    }, [disabled, readOnly, onToggle]);
    var updateValue = (0, _react.useCallback)(function (item, event) {
      var _rest$onChangeValue2;
      var newValue = item.dataset.value;
      if (rest.valueType === 'multiple') {
        var _rest$onChangeValue;
        var _newSelected = (0, _utils3.getNewSelected)(rest.value, newValue, rest.valueType);
        (_rest$onChangeValue = rest.onChangeValue) === null || _rest$onChangeValue === void 0 || _rest$onChangeValue.call(rest, _newSelected);
        return;
      }
      var newSelected = (0, _utils3.getNewSelected)(rest.value, newValue, 'single');
      (_rest$onChangeValue2 = rest.onChangeValue) === null || _rest$onChangeValue2 === void 0 || _rest$onChangeValue2.call(rest, newSelected);
      onInnerToggle === null || onInnerToggle === void 0 || onInnerToggle(false, event);
    }, [rest.value, rest.valueType, rest.onChangeValue, onToggle, onInnerToggle]);
    var onClickChildrenItem = (0, _react.useCallback)(function (event) {
      // INFO: Для корректной работы компонента Popper
      event.stopPropagation();
      var item = event.currentTarget;
      updateValue(item, event);
    }, [updateValue]);
    var onNativeSelectChange = (0, _react.useCallback)(function (event) {
      var _rest$onChangeValue3;
      if (rest.valueType === 'multiple') {
        return;
      }
      var newValue = event.target.value;
      var newSelected = (0, _utils3.getNewSelected)(rest.value, newValue, 'native');
      (_rest$onChangeValue3 = rest.onChangeValue) === null || _rest$onChangeValue3 === void 0 || _rest$onChangeValue3.call(rest, newSelected);
    }, [rest.value, rest.valueType, rest.onChangeValue]);
    var onChangeChipValue = (0, _react.useCallback)(function (newValue) {
      var _rest$onChangeValue4;
      if (rest.valueType !== 'multiple') {
        return;
      }
      (_rest$onChangeValue4 = rest.onChangeValue) === null || _rest$onChangeValue4 === void 0 || _rest$onChangeValue4.call(rest, newValue);
    }, [rest.valueType]);
    var _useKeyNavigation = (0, _hooks2.useKeyNavigation)({
        controlledRefs: controlledRefs,
        opened: innerOpened,
        enumerationType: enumerationType,
        valueType: valueType,
        value: value,
        updateOpened: onInnerToggle,
        updateValue: updateValue
      }),
      onKeyDownTarget = _useKeyNavigation.onKeyDownTarget,
      onKeyDownSelect = _useKeyNavigation.onKeyDownSelect;
    var childrenArray = (0, _react.useMemo)(function () {
      return _react.Children.toArray(children);
    }, [children]);
    var childrenMemo = (0, _react.useMemo)(function () {
      return (0, _utils3.getChildren)(childrenArray, {
        onClick: onClickChildrenItem,
        childrenRefs: itemsRefs
      }, value);
    }, [childrenArray, value, innerOpened]);
    var values = (0, _react.useMemo)(function () {
      return (0, _utils3.getValues)(childrenArray, value);
    }, [childrenArray, value]);
    return /*#__PURE__*/_react["default"].createElement(_Select2.StyledRoot, {
      ref: selectRef
    }, /*#__PURE__*/_react["default"].createElement(_Select2.StyledNativeSelect, {
      "aria-invalid": "false",
      "aria-hidden": "true",
      "aria-readonly": readOnly,
      tabIndex: valueType === 'native' ? 0 : -1,
      form: form,
      name: name,
      ref: handleRef,
      value: "".concat(value),
      disabled: disabled || readOnly,
      onChange: onNativeSelectChange,
      className: (0, _utils.cx)(ref === null || ref === void 0 ? void 0 : ref.classList.toString(), withNativeSelectVisible)
    }, valueType === 'native' && childrenMemo), /*#__PURE__*/_react["default"].createElement(_Select2.StyledPopover, {
      role: role,
      isOpen: innerOpened,
      onToggle: onInnerToggle,
      target: /*#__PURE__*/_react["default"].createElement(_ui.SelectTarget, {
        tabIndex: valueType === 'native' ? -1 : 0,
        opened: innerOpened,
        target: target,
        values: values,
        label: label,
        enumerationType: enumerationType,
        readOnly: readOnly,
        disabled: disabled,
        size: size,
        ref: targetRef,
        chipsRefs: chipsRefs,
        id: innerId,
        onChange: onChangeChipValue,
        onKeyDown: onKeyDownTarget
      }),
      className: (0, _utils.cx)(ref === null || ref === void 0 ? void 0 : ref.classList.toString()) // INFO: Прокидываем стили для Popover
      ,
      id: innerId,
      offset: offset,
      frame: frame,
      placement: (0, _utils2.getPlacements)(placement),
      trigger: "click",
      usePortal: usePortal,
      selectWidth: dynamicTargetWidth,
      isFocusTrapped: false,
      preventOverflow: false,
      closeOnEsc: false,
      closeOnOverlayClick: true
    }, /*#__PURE__*/_react["default"].createElement(Root, _extends({
      ref: setRef,
      target: target,
      view: view,
      size: size
    }, rest), valueType !== 'native' && /*#__PURE__*/_react["default"].createElement(_Select2.StyledSelect, {
      onKeyDown: onKeyDownSelect
    }, childrenMemo))));
  });
};
var selectConfig = exports.selectConfig = {
  name: 'Select',
  tag: 'div',
  layout: selectRoot,
  base: '',
  variations: {
    target: {
      css: _base.base
    },
    view: {
      css: _base2.base
    },
    size: {
      css: _base3.base
    }
  },
  defaults: {
    view: 'default',
    size: 'm'
  }
};
