// @salutejs/plasma-new-hope/es/components/Select/ui/SelectTarget/SelectTarget.js
import { objectWithoutProperties as _objectWithoutProperties, slicedToArray as _slicedToArray, extends as _extends } from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef, useRef, useCallback } from 'react';
import { classes } from '../../Select.tokens.js';
import { cx } from '../../../../utils/index.js';
import { SelectChip } from '../SelectChip/SelectChip.js';
import { Keys } from '../../hooks/useKeyNavigation.js';
import { StyledSelectTarget, StyledText, StyledChipsWrapper, StyledChips, StyledLabel, StyledArrow } from './SelectTarget.styles.js';

var _excluded = ["values", "target", "label", "opened", "readOnly", "disabled", "enumerationType", "size", "id", "chipsRefs", "onChange", "onKeyDown"];
var hasNoFocus = classes.hasNoFocus,
  hasChips = classes.hasChips,
  innerLabelUp = classes.innerLabelUp,
  arrowInverse = classes.arrowInverse,
  selectTarget = classes.selectTarget,
  selectTargetArrow = classes.selectTargetArrow,
  selectTargeText = classes.selectTargeText,
  selectTargetLabel = classes.selectTargetLabel;

/**
 * Элемент для выпадающего списка
 */
var SelectTarget = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var values = _ref.values,
    _ref$target = _ref.target,
    target = _ref$target === void 0 ? 'textField-like' : _ref$target,
    label = _ref.label,
    opened = _ref.opened,
    readOnly = _ref.readOnly,
    disabled = _ref.disabled,
    enumerationType = _ref.enumerationType,
    size = _ref.size,
    id = _ref.id,
    chipsRefs = _ref.chipsRefs,
    onChange = _ref.onChange,
    onKeyDown = _ref.onKeyDown,
    rest = _objectWithoutProperties(_ref, _excluded);
  var hasText = Boolean(values === null || values === void 0 ? void 0 : values.some(function (_ref2) {
    var _ref3 = _slicedToArray(_ref2, 1),
      value = _ref3[0];
    return value;
  }));
  var hasLabel = label && size !== 'xs' && target === 'textField-like';
  var textContent = values === null || values === void 0 ? void 0 : values.map(function (_ref4) {
    var _ref5 = _slicedToArray(_ref4, 2),
      text = _ref5[1];
    return text;
  }).join(', ');
  var contentRef = useRef(null);
  var withArrowInverse = opened ? arrowInverse : undefined;
  var withInnerLabelUp = hasLabel && hasText ? innerLabelUp : undefined;
  var withHasChips = hasText && enumerationType === 'chip' && target === 'textField-like' ? hasChips : undefined;
  var withNoFocus = target === 'textField-like' ? hasNoFocus : undefined;
  var isLabelVisible = !hasText || hasLabel && enumerationType === 'comma';
  var onChipClick = useCallback(function (event) {
    event.stopPropagation();
  }, []);
  var onChipClear = useCallback(function (value, text, index) {
    var newValue = values === null || values === void 0 ? void 0 : values.filter(function (_ref6) {
      var _ref7 = _slicedToArray(_ref6, 2),
        itemValue = _ref7[0],
        itemText = _ref7[1];
      return !(itemValue === value && itemText === text);
    }).map(function (_ref8) {
      var _ref9 = _slicedToArray(_ref8, 1),
        itemValue = _ref9[0];
      return itemValue;
    });
    chipsRefs === null || chipsRefs === void 0 || chipsRefs.current.splice(index, 1);
    onChange === null || onChange === void 0 || onChange(newValue);
  }, [values, onChange]);
  var onChipKeyDown = useCallback(function (value, text, index, event) {
    var code = event.code,
      shiftKey = event.shiftKey;
    if (code === Keys.Tab || code === Keys.Tab && shiftKey) {
      event.preventDefault();
    }
    if (code === Keys.Backspace) {
      onChipClear(value, text, index);
    }
  }, [onChipClear]);
  var onWheel = useCallback(function (event) {
    var deltaY = event.deltaY;
    if (contentRef.current) {
      contentRef.current.scrollLeft += Math.round(deltaY);
    }
  }, []);
  var getRef = useCallback(function (element, index) {
    if (element && chipsRefs && chipsRefs.current) {
      chipsRefs.current[index] = element;
    }
  }, []);

  // INFO: Нужно очищать на каждый ререндер компонента для актуализации состояния рефов
  if (chipsRefs) {
    chipsRefs.current = [];
  }
  return /*#__PURE__*/React.createElement(StyledSelectTarget, _extends({}, rest, {
    ref: ref,
    opened: opened,
    target: target,
    readOnly: readOnly,
    disabled: disabled,
    title: textContent,
    "aria-label": label,
    className: cx(withInnerLabelUp, withHasChips, withNoFocus, selectTarget),
    onWheel: onWheel,
    onKeyDown: onKeyDown,
    stretching: "filled"
  }), hasText && (enumerationType === 'comma' || target === 'button-like' ? /*#__PURE__*/React.createElement(StyledText, {
    className: selectTargeText
  }, textContent) : /*#__PURE__*/React.createElement(StyledChipsWrapper, {
    ref: contentRef
  }, /*#__PURE__*/React.createElement(StyledChips, null, values === null || values === void 0 ? void 0 : values.map(function (_ref10, index) {
    var _ref11 = _slicedToArray(_ref10, 2),
      value = _ref11[0],
      text = _ref11[1];
    return /*#__PURE__*/React.createElement(SelectChip, {
      index: index,
      ref: function ref(element) {
        return getRef(element, index);
      },
      key: "".concat(value, "_").concat(text),
      disabled: disabled,
      readOnly: readOnly,
      value: value,
      text: text,
      onClear: function onClear(v, t) {
        return onChipClear(v, t, index);
      },
      onClick: onChipClick,
      onKeyDown: onChipKeyDown
    });
  })))), isLabelVisible && /*#__PURE__*/React.createElement(StyledLabel, {
    className: selectTargetLabel,
    htmlFor: id
  }, label), /*#__PURE__*/React.createElement(StyledArrow, {
    size: "s",
    color: "inherit",
    className: cx(selectTargetArrow, withArrowInverse)
  }));
});

export { SelectTarget };
