// @salutejs/plasma-new-hope/styled-components/cjs/components/Header/ui/HeaderTitleWrapper/HeaderTitleWrapper.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.headerTitleWrapperRoot = exports.headerTitleWrapperConfig = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _base = /*#__PURE__*/require("./variations/_size/base");
var _base2 = /*#__PURE__*/require("./variations/_view/base");
var _HeaderTitleWrapper = /*#__PURE__*/require("./HeaderTitleWrapper.styles");
var _excluded = ["label", "title", "subTitle", "caption", "withBigTitle", "children"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
/**
 * Компонент для отображения текста в скомпанованном блоке.
 */
var headerTitleWrapperRoot = exports.headerTitleWrapperRoot = function headerTitleWrapperRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (props, outerRootRef) {
    var label = props.label,
      title = props.title,
      subTitle = props.subTitle,
      caption = props.caption,
      withBigTitle = props.withBigTitle,
      children = props.children,
      rest = _objectWithoutProperties(props, _excluded);
    if (children) {
      return /*#__PURE__*/_react["default"].createElement(Root, {
        ref: outerRootRef
      }, /*#__PURE__*/_react["default"].createElement(_HeaderTitleWrapper.TextBoxRoot, rest, children));
    }
    var T = withBigTitle ? _HeaderTitleWrapper.BigTitle : _HeaderTitleWrapper.Title;
    return /*#__PURE__*/_react["default"].createElement(Root, {
      ref: outerRootRef
    }, /*#__PURE__*/_react["default"].createElement(_HeaderTitleWrapper.TextBoxRoot, rest, label && /*#__PURE__*/_react["default"].createElement(_HeaderTitleWrapper.Label, null, label), title && /*#__PURE__*/_react["default"].createElement(T, null, title), subTitle && /*#__PURE__*/_react["default"].createElement(_HeaderTitleWrapper.SubTitle, null, subTitle), caption && /*#__PURE__*/_react["default"].createElement(_HeaderTitleWrapper.Caption, null, caption)));
  });
};
var headerTitleWrapperConfig = exports.headerTitleWrapperConfig = {
  name: 'HeaderTitleWrapper',
  tag: 'div',
  layout: headerTitleWrapperRoot,
  base: '',
  variations: {
    view: {
      css: _base2.base
    },
    size: {
      css: _base.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};
