// @salutejs/plasma-new-hope/styled-components/cjs/components/Header/ui/HeaderTitleWrapper/HeaderTitleWrapper.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Title = exports.TextBoxRoot = exports.SubTitle = exports.Label = exports.Caption = exports.BiggerTitle = exports.BigTitle = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _Header = /*#__PURE__*/require("../../Header.tokens");
var _mixins = /*#__PURE__*/require("../../../../mixins");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var Title = exports.Title = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-0"
})(["", ";margin:var(", ");color:var(", ");", ";", ";"], _mixins.bodyM, _Header.tokens.headerTitleMargin, _Header.tokens.headerTitleColor, _mixins.applyHyphens, /*#__PURE__*/(0, _mixins.applyEllipsis)());
var BigTitle = exports.BigTitle = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-1"
})(["", ";font-weight:600;margin:var(", ");color:var(", ");", ";", ";"], _mixins.h3, _Header.tokens.headerBigTitleMargin, _Header.tokens.headerBiggerTitleColor, _mixins.applyHyphens, /*#__PURE__*/(0, _mixins.applyEllipsis)());
var BiggerTitle = exports.BiggerTitle = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-2"
})(["", ";font-weight:600;margin:var(", ");color:var(", ");", ";", ";"], _mixins.h1, _Header.tokens.headerBiggerTitleMargin, _Header.tokens.headerBiggerTitleColor, _mixins.applyHyphens, /*#__PURE__*/(0, _mixins.applyEllipsis)());
var Label = exports.Label = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-3"
})(["margin:var(", ");color:var(", ");", ";"], _Header.tokens.headerLabelMargin, _Header.tokens.headerLabelColor, /*#__PURE__*/(0, _mixins.applyEllipsis)());
var SubTitle = exports.SubTitle = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-4"
})(["", ";margin:var(", ");color:var(", ");", ";", ";"], _mixins.bodyS, _Header.tokens.headerSubTitleMargin, _Header.tokens.headerSubTitleColor, _mixins.applyHyphens, /*#__PURE__*/(0, _mixins.applyEllipsis)());
var Caption = exports.Caption = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-5"
})(["margin:var(", ");color:var(", ");", ";"], _Header.tokens.headerCaptionMargin, _Header.tokens.headerCaptionColor, /*#__PURE__*/(0, _mixins.applyEllipsis)());
var TextBoxRoot = exports.TextBoxRoot = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-6"
})(["& >:first-child{margin-top:0;}& >:last-child{margin-bottom:0;}"]);
