// @salutejs/plasma-new-hope/cjs/components/Select/ui/SelectDivider/SelectDivider.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('react');
require('../../../../utils/react.js');
var DropdownDivider = require('../../../Dropdown/ui/DropdownDivider/DropdownDivider.js');
require('@salutejs/plasma-core');
require('@linaria/react');
require('@linaria/core');
require('focus-visible');
require('react-dom');
require('react-popper');
require('styled-components');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var SelectDivider_styles = require('./SelectDivider.styles.js');

var selectDividerRoot = DropdownDivider.dropdownDividerRoot;
var selectDividerConfig = {
  name: 'SelectDivider',
  tag: 'div',
  layout: selectDividerRoot,
  base: SelectDivider_styles.base,
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

exports.selectDividerConfig = selectDividerConfig;
exports.selectDividerRoot = selectDividerRoot;
