// @salutejs/plasma-ui/es/components/Checkbox/Checkbox.perftest.js
import React from 'react';
import styled from 'styled-components';
import { Checkbox } from '.';
var func = function func() {};
var labelWithLink = /*#__PURE__*/React.createElement("div", null, "Label 5 with ", /*#__PURE__*/React.createElement("a", {
  href: "/#"
}, "link"));
var items = [{
  id: 'check-1-1',
  name: 'check',
  value: 1,
  label: 'Checkbox 1',
  disabled: false
}, {
  id: 'check-1-2',
  name: 'check',
  value: 2,
  label: 'Checkbox 2',
  disabled: false
}, {
  id: 'check-1-3',
  name: 'check',
  value: 3,
  label: 'Checkbox 3',
  disabled: true
}, {
  id: 'check-1-4',
  name: 'check',
  value: 4,
  label: 'Checkbox 4',
  disabled: true
}, {
  id: 'check-1-5',
  name: 'check',
  value: 5,
  label: labelWithLink
}];
var StyledGrid = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-ui__sc-czrehc-0"
})(["display:inline-grid;grid-template-columns:repeat(2,50%);gap:1rem;align-items:flex-start;& > * + *{margin-top:0 !important;}"]);
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(StyledGrid, null, items.map(function (item) {
    return /*#__PURE__*/React.createElement(Checkbox, {
      key: item.id,
      id: item.id,
      name: item.name,
      value: item.value,
      label: item.label,
      disabled: item.disabled,
      checked: [2, 4].includes(item.value),
      onChange: func,
      onFocus: func,
      onBlur: func
    });
  }));
};
