// @salutejs/plasma-web/css/es/components/Typography/Typography.js
import { component, mergeConfig, bodyConfig, dsplConfig, headingConfig, textConfig } from '@salutejs/plasma-new-hope';
import { configM, configL, configS, configXS, configXXS } from './Body.config.js';
import { configM as configM$1, configL as configL$1, configS as configS$1 } from './Dspl.config.js';
import { configH1, configH2, configH3, configH4, configH5 } from './Heading.config.js';
import { configM as configM$2, configL as configL$2, configS as configS$2, configXS as configXS$1 } from './Text.config.js';

var bodyMConfig = /*#__PURE__*/mergeConfig(bodyConfig, configM);
var BodyMComponent = /*#__PURE__*/component(bodyMConfig);
var bodyLConfig = /*#__PURE__*/mergeConfig(bodyConfig, configL);
var BodyLComponent = /*#__PURE__*/component(bodyLConfig);
var bodySConfig = /*#__PURE__*/mergeConfig(bodyConfig, configS);
var BodySComponent = /*#__PURE__*/component(bodySConfig);
var bodyXSConfig = /*#__PURE__*/mergeConfig(bodyConfig, configXS);
var BodyXSComponent = /*#__PURE__*/component(bodyXSConfig);
var bodyXXSConfig = /*#__PURE__*/mergeConfig(bodyConfig, configXXS);
var BodyXXSComponent = /*#__PURE__*/component(bodyXXSConfig);
var dsplMConfig = /*#__PURE__*/mergeConfig(dsplConfig, configM$1);
var DsplMComponent = /*#__PURE__*/component(dsplMConfig);
var dsplLConfig = /*#__PURE__*/mergeConfig(dsplConfig, configL$1);
var DsplLComponent = /*#__PURE__*/component(dsplLConfig);
var dsplSConfig = /*#__PURE__*/mergeConfig(dsplConfig, configS$1);
var DsplSComponent = /*#__PURE__*/component(dsplSConfig);
var heading1Config = /*#__PURE__*/mergeConfig(headingConfig, configH1);
var Heading1Component = /*#__PURE__*/component(heading1Config);
var heading2Config = /*#__PURE__*/mergeConfig(headingConfig, configH2);
var Heading2Component = /*#__PURE__*/component(heading2Config);
var heading3Config = /*#__PURE__*/mergeConfig(headingConfig, configH3);
var Heading3Component = /*#__PURE__*/component(heading3Config);
var heading4Config = /*#__PURE__*/mergeConfig(headingConfig, configH4);
var Heading4Component = /*#__PURE__*/component(heading4Config);
var heading5Config = /*#__PURE__*/mergeConfig(headingConfig, configH5);
var Heading5Component = /*#__PURE__*/component(heading5Config);
var textMConfig = /*#__PURE__*/mergeConfig(textConfig, configM$2);
var TextMComponent = /*#__PURE__*/component(textMConfig);
var textLConfig = /*#__PURE__*/mergeConfig(textConfig, configL$2);
var TextLComponent = /*#__PURE__*/component(textLConfig);
var textSConfig = /*#__PURE__*/mergeConfig(textConfig, configS$2);
var TextSComponent = /*#__PURE__*/component(textSConfig);
var textXSConfig = /*#__PURE__*/mergeConfig(textConfig, configXS$1);
var TextXSComponent = /*#__PURE__*/component(textXSConfig);
var BodyM = BodyMComponent;
var BodyL = BodyLComponent;
var BodyS = BodySComponent;
var BodyXS = BodyXSComponent;
var BodyXXS = BodyXXSComponent;
var DsplM = DsplMComponent;
var DsplL = DsplLComponent;
var DsplS = DsplSComponent;
var H1 = Heading1Component;
var H2 = Heading2Component;
var H3 = Heading3Component;
var H4 = Heading4Component;
var H5 = Heading5Component;
var TextM = TextMComponent;
var TextL = TextLComponent;
var TextS = TextSComponent;
var TextXS = TextXSComponent;

export { BodyL, BodyM, BodyS, BodyXS, BodyXXS, DsplL, DsplM, DsplS, H1, H2, H3, H4, H5, TextL, TextM, TextS, TextXS };
