// @salutejs/plasma-new-hope/cjs/components/Panel/ui/PanelHeader/PanelHeader.styles.js
import './PanelHeader.styles_1cid6t7.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
require('@linaria/core');
var common = require('../../../../engines/common.js');
var Button$1 = require('../../../Button/Button.js');
var PanelHeader_types = require('./PanelHeader.types.js');

var mergedButtonConfig = /*#__PURE__*/common.mergeConfig(Button$1.buttonConfig);
var Button = /*#__PURE__*/common.component(mergedButtonConfig);
var _exp2 = function _exp2() {
  return function (_ref) {
    var placement = _ref.placement;
    return placement === PanelHeader_types.placements.left ? -1 : 0;
  };
};
var _exp3 = function _exp3() {
  return function (_ref2) {
    var placement = _ref2.placement;
    return !placement || placement === PanelHeader_types.placements.right ? '0.5rem' : '';
  };
};
var _exp4 = function _exp4() {
  return function (_ref3) {
    var placement = _ref3.placement;
    return placement === PanelHeader_types.placements.left ? '0.5rem' : '';
  };
};
var ButtonWrapper = /*#__PURE__*/react.styled('div')({
  name: "ButtonWrapper",
  "class": "b5xb1jq",
  propsAsIs: false,
  vars: {
    "b5xb1jq-0": [/*#__PURE__*/_exp2()],
    "b5xb1jq-1": [/*#__PURE__*/_exp3()],
    "b5xb1jq-2": [/*#__PURE__*/_exp4()]
  }
});
var StyledHeader = /*#__PURE__*/react.styled('div')({
  name: "StyledHeader",
  "class": "s1hqmxl1",
  propsAsIs: false
});
var _exp5 = function _exp5() {
  return Button;
};
var CloseIconWrapper = /*#__PURE__*/react.styled(_exp5())({
  name: "CloseIconWrapper",
  "class": "c7hx8go",
  propsAsIs: true
});

exports.ButtonWrapper = ButtonWrapper;
exports.CloseIconWrapper = CloseIconWrapper;
exports.StyledHeader = StyledHeader;
