// @salutejs/plasma-new-hope/cjs/components/Dropdown/ui/DropdownHeader/DropdownHeader.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var plasmaCore = require('@salutejs/plasma-core');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var DropdownHeader_styles = require('./DropdownHeader.styles.js');

var _excluded = ["id", "className", "view", "size", "children"];
/**
 * Элемент выпадающего списка
 */
var dropdownHeaderRoot = function dropdownHeaderRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (_ref, outerRootRef) {
    var id = _ref.id,
      className = _ref.className,
      view = _ref.view,
      size = _ref.size,
      children = _ref.children,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
    var uniqId = plasmaCore.useUniqId();
    var innerId = id || uniqId;
    return /*#__PURE__*/React.createElement(Root, {
      id: innerId,
      ref: outerRootRef,
      view: view,
      size: size
    }, /*#__PURE__*/React.createElement(DropdownHeader_styles.StyledDropdownHeader, _rollupPluginBabelHelpers.extends({
      className: className
    }, rest), children));
  });
};
var dropdownHeaderConfig = {
  name: 'DropdownHeader',
  tag: 'div',
  layout: dropdownHeaderRoot,
  base: '',
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

exports.dropdownHeaderConfig = dropdownHeaderConfig;
exports.dropdownHeaderRoot = dropdownHeaderRoot;
