// @salutejs/plasma-core/es/tokens/colors.js
export var white = 'var(--plasma-colors-white)';
export var whitePrimary = 'var(--plasma-colors-white-primary)';
export var whiteSecondary = 'var(--plasma-colors-white-secondary)';
export var whiteTertiary = 'var(--plasma-colors-white-tertiary)';
export var black = 'var(--plasma-colors-black)';
export var blackPrimary = 'var(--plasma-colors-black-primary)';
export var blackSecondary = 'var(--plasma-colors-black-secondary)';
export var blackTertiary = 'var(--plasma-colors-black-tertiary)';
export var dark01 = 'var(--plasma-colors-dark01)';
export var dark02 = 'var(--plasma-colors-dark02)';
export var dark03 = 'var(--plasma-colors-dark03)';
export var transparent = 'var(--plasma-colors-transparent)';
export var text = 'var(--plasma-colors-text)';
export var primary = 'var(--plasma-colors-primary)';
export var secondary = 'var(--plasma-colors-secondary)';
export var tertiary = 'var(--plasma-colors-tertiary)';
export var inverse = 'var(--plasma-colors-inverse)';
export var paragraph = 'var(--plasma-colors-paragraph)';
export var background = 'var(--plasma-colors-background)';
export var backgroundPrimary = 'var(--plasma-colors-background-primary)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107
export var backgroundSecondary = 'var(--plasma-colors-background-secondary)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107
export var backgroundTertiary = 'var(--plasma-colors-background-tertiary)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107

export var success = 'var(--plasma-colors-success)';
export var warning = 'var(--plasma-colors-warning)';
export var critical = 'var(--plasma-colors-critical)';
export var overlay = 'var(--plasma-colors-overlay)';
export var surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107
export var surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107
export var surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107
export var surfaceSolid01 = 'var(--plasma-colors-surface-solid01)';
export var surfaceSolid02 = 'var(--plasma-colors-surface-solid02)';
export var surfaceSolid03 = 'var(--plasma-colors-surface-solid03)';
export var surfaceCard = 'var(--plasma-colors-surface-card)';
export var buttonPrimary = 'var(--plasma-colors-button-primary)';
export var buttonSecondary = 'var(--plasma-colors-button-secondary)';
export var buttonSuccess = 'var(--plasma-colors-button-success)';
export var buttonWarning = 'var(--plasma-colors-button-warning)';
export var buttonCritical = 'var(--plasma-colors-button-critical)';
export var buttonChecked = 'var(--plasma-colors-button-checked)';
export var speechBubbleSent = 'var(--plasma-colors-speech-bubble-sent)';
export var speechBubbleReceived = 'var(--plasma-colors-speech-bubble-received)';
export var accent = 'var(--plasma-colors-accent)';
export var buttonAccent = 'var(--plasma-colors-button-accent)';
export var buttonFocused = 'var(--plasma-colors-button-focused)';
export var gradient = 'var(--plasma-colors-gradient)';
export var voicePhraseGradient = 'var(--plasma-colors-voice-phrase-gradient)';
export var skeletonGradient = 'var(--plasma-colors-skeleton-gradient)';
export var skeletonGradientLighter = 'var(--plasma-colors-skeleton-gradient-lighter)';
