// @salutejs/plasma-new-hope/styled-components/es/engines/styled-components.js
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React, { forwardRef } from 'react';
import styled, { css } from 'styled-components';
import { getStaticVariants, getDynamicVariants } from './common';
export { styled, css };
var Root = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1xug4g9-0"
})(["", ";", ";", ";"], function (_ref) {
  var base = _ref.base;
  return base;
}, function (_ref2) {
  var staticVariants = _ref2.staticVariants;
  return staticVariants;
}, function (_ref3) {
  var dynamicVariants = _ref3.dynamicVariants;
  return dynamicVariants;
});

/* eslint-disable no-underscore-dangle */
export var _component = function _component(componentConfig) {
  var tag = componentConfig.tag,
    base = componentConfig.base;
  var staticVariants = getStaticVariants(componentConfig);
  var dynamicVariants = getDynamicVariants(componentConfig);
  return /*#__PURE__*/forwardRef(function (props, ref) {
    return /*#__PURE__*/React.createElement(Root, _extends({
      as: tag,
      base: base,
      staticVariants: staticVariants,
      dynamicVariants: dynamicVariants
    }, props, {
      ref: ref
    }));
  });
};
