// @salutejs/plasma-ui/es/components/Button/Button.perftest.js
var _IconMic, _IconMic2;
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React from 'react';
import { IconMic } from '@salutejs/plasma-icons';
import { Button, ActionButton as ActionButtonComponent } from '.';
var defaultProps = {
  size: 'm',
  view: 'primary',
  pin: 'square-square',
  enableIcon: true,
  scaleOnInteraction: true,
  outlined: true,
  focused: false,
  disabled: false,
  square: false,
  stretch: false,
  isLoading: false,
  onClick: function onClick() {},
  onFocus: function onFocus() {},
  onBlur: function onBlur() {}
};
var text = 'Hello Plasma';
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Button, _extends({
    text: text,
    contentLeft: _IconMic || (_IconMic = /*#__PURE__*/React.createElement(IconMic, {
      size: "s",
      color: "inherit"
    }))
  }, defaultProps));
};
var actionButton = {
  size: 'm',
  view: 'primary',
  pin: 'square-square',
  scaleOnInteraction: true,
  disabled: false,
  tabIndex: 0,
  onClick: function onClick() {},
  onFocus: function onFocus() {},
  onBlur: function onBlur() {}
};
export var ActionButton = function ActionButton() {
  return /*#__PURE__*/React.createElement(ActionButtonComponent, actionButton, _IconMic2 || (_IconMic2 = /*#__PURE__*/React.createElement(IconMic, {
    size: "xs",
    color: "inherit"
  })));
};
