// @salutejs/plasma-new-hope/cjs/components/Combobox/ui/ComboboxGroup/ComboboxGroup.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('react');
require('../../../../utils/react.js');
require('@salutejs/plasma-core');
require('@linaria/react');
var DropdownGroup = require('../../../Dropdown/ui/DropdownGroup/DropdownGroup.js');
require('@linaria/core');
require('focus-visible');
require('react-dom');
require('react-popper');
require('styled-components');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var ComboboxGroup_styles = require('./ComboboxGroup.styles.js');

var comboboxGroupRoot = DropdownGroup.dropdownGroupRoot;
var comboboxGroupConfig = {
  name: 'ComboboxGroup',
  tag: 'div',
  layout: comboboxGroupRoot,
  base: ComboboxGroup_styles.base,
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

exports.comboboxGroupConfig = comboboxGroupConfig;
exports.comboboxGroupRoot = comboboxGroupRoot;
