// @salutejs/plasma-hope/es/components/Upload/UploadButton.js
var _excluded = ["status", "disabled", "isProgress", "accept", "content", "loader", "onChange", "onValidation", "validate"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React, { useCallback, useState } from 'react';
import styled, { css } from 'styled-components';
import { surfaceLiquid01, surfaceLiquid02, surfaceLiquid03, buttonAccent, secondary, tertiary, transparent, success, critical, footnote1 } from '@salutejs/plasma-core';
import { FileDrop } from 'react-file-drop';
import { defaultValidate } from './utils';

// TODO: https://github.com/salute-developers/plasma/issues/364

var statuses = {
  error: {
    borderColor: critical
  },
  success: {
    borderColor: success
  }
};
var paints = {
  "default": {
    backgroundColor: surfaceLiquid01,
    borderColor: tertiary,
    color: secondary
  },
  progress: {
    backgroundColor: surfaceLiquid03,
    borderColor: transparent,
    color: secondary
  }
};
export var StyledButton = /*#__PURE__*/styled.button.withConfig({
  componentId: "plasma-web-core__sc-3ebw3y-0"
})(["", ";position:relative;display:flex;align-items:center;box-sizing:border-box;justify-content:center;width:100%;height:4rem;padding:1.25rem 1rem;appearance:none;border:1px dashed;border-radius:0.75rem;cursor:pointer;transition:all 0.3s ease-in-out;transition-property:transform,background-color,border-color,color;&:hover:not(:disabled){background-color:", ";transform:scale(1.04);}", " ", " ", " ", ""], footnote1, surfaceLiquid02, function (_ref) {
  var isGrabbing = _ref.isGrabbing;
  return isGrabbing && css(["border-color:", ";background-color:", ";transform:scale(1.04);"], buttonAccent, surfaceLiquid02);
}, function (_ref2) {
  var paint = _ref2.paint;
  return paints[paint];
}, function (_ref3) {
  var status = _ref3.status;
  return status && statuses[status];
}, function (_ref4) {
  var disabled = _ref4.disabled;
  return disabled && css(["opacity:0.4;cursor:not-allowed;"]);
});
var StyledInputLabel = /*#__PURE__*/styled.label.withConfig({
  componentId: "plasma-web-core__sc-3ebw3y-1"
})(["position:absolute;width:100%;height:100%;top:0;left:0;cursor:pointer;", ""], function (_ref5) {
  var disabled = _ref5.disabled;
  return disabled && css(["cursor:not-allowed;"]);
});
/**
 * Базовый компонент для загрузки файлов.
 */
export var UploadButton = function UploadButton(_ref6) {
  var status = _ref6.status,
    disabled = _ref6.disabled,
    isProgress = _ref6.isProgress,
    accept = _ref6.accept,
    content = _ref6.content,
    loader = _ref6.loader,
    _ref6$onChange = _ref6.onChange,
    onChange = _ref6$onChange === void 0 ? function () {} : _ref6$onChange,
    _ref6$onValidation = _ref6.onValidation,
    onValidation = _ref6$onValidation === void 0 ? function () {} : _ref6$onValidation,
    validate = _ref6.validate,
    rest = _objectWithoutProperties(_ref6, _excluded);
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    isGrabbing = _useState2[0],
    setGrabbing = _useState2[1];
  var paint = isProgress ? 'progress' : 'default';
  var onUpload = useCallback(function (files, event) {
    var result = (validate || defaultValidate)(files, accept);
    onValidation(result);
    if (result.data) {
      onChange(result.data, event);
    }
  }, [accept, onChange, onValidation, validate]);
  var dragStart = useCallback(function () {
    return !disabled && !isProgress && setGrabbing(true);
  }, [disabled, isProgress]);
  var dragEnd = useCallback(function () {
    return setGrabbing(false);
  }, []);
  var drop = useCallback(function (files, event) {
    if (disabled || isProgress) {
      return;
    }
    event.preventDefault();
    onUpload(files);
    setGrabbing(false);
  }, [onUpload, disabled, isProgress]);
  var inputChangeHandler = useCallback(function (event) {
    var files = event.target.files;
    onUpload(files, event);
  }, [onUpload]);
  return /*#__PURE__*/React.createElement(FileDrop, {
    onDrop: drop,
    onDragOver: dragStart,
    onDragLeave: dragEnd
  }, /*#__PURE__*/React.createElement(StyledButton, _extends({
    type: "button",
    isGrabbing: isGrabbing,
    paint: paint,
    status: status,
    disabled: disabled
  }, rest), /*#__PURE__*/React.createElement(StyledInputLabel, {
    disabled: disabled
  }, /*#__PURE__*/React.createElement("input", {
    disabled: disabled,
    type: "file",
    onChange: inputChangeHandler,
    accept: accept,
    hidden: true
  })), isProgress ? loader : content));
};
