// @salutejs/plasma-hope/es/components/Upload/Upload.js
var _excluded = ["content", "status", "message", "disabled", "progress", "accept", "loader", "validate", "onChange", "onValidation"];
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React from 'react';
import { UploadRoot } from './UploadRoot';
import { UploadButton } from './UploadButton';
import { UploadMessage } from './UploadMessage';
import { UploadProgress } from './UploadProgress';
import { collectPackageInfo } from '../../collectPackageInfo';
collectPackageInfo('Upload/Upload');
/**
 * Компонент для загрузки файлов.
 */
export var Upload = function Upload(_ref) {
  var content = _ref.content,
    status = _ref.status,
    message = _ref.message,
    disabled = _ref.disabled,
    progress = _ref.progress,
    accept = _ref.accept,
    loader = _ref.loader,
    validate = _ref.validate,
    onChange = _ref.onChange,
    onValidation = _ref.onValidation,
    rest = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/React.createElement(UploadRoot, rest, /*#__PURE__*/React.createElement(UploadButton, {
    status: status,
    disabled: disabled,
    accept: accept,
    content: content,
    isProgress: Boolean(progress !== undefined),
    loader: loader || /*#__PURE__*/React.createElement(UploadProgress, {
      progress: progress
    }),
    validate: validate,
    onValidation: onValidation,
    onChange: onChange
  }), message && /*#__PURE__*/React.createElement(UploadMessage, {
    status: status
  }, message));
};
