// @salutejs/plasma-ui/es/components/Stepper/Stepper.hooks.js
import { useEffect, useCallback, useRef } from 'react';
import { collectPackageInfo } from '../../collectPackageInfo';
collectPackageInfo('Stepper/Stepper.hooks');
export var useStepper = function useStepper(_ref) {
  var value = _ref.value,
    _ref$step = _ref.step,
    step = _ref$step === void 0 ? 1 : _ref$step,
    _ref$min = _ref.min,
    min = _ref$min === void 0 ? 0 : _ref$min,
    _ref$max = _ref.max,
    max = _ref$max === void 0 ? Infinity : _ref$max,
    onChange = _ref.onChange;
  var curValue = useRef(value);
  useEffect(function () {
    curValue.current = value;
  }, [value]);
  var onLessClick = useCallback(function (event) {
    return onChange === null || onChange === void 0 ? void 0 : onChange(curValue.current - step, event);
  }, [step, onChange]);
  var onMoreClick = useCallback(function (event) {
    return onChange === null || onChange === void 0 ? void 0 : onChange(curValue.current + step, event);
  }, [step, onChange]);
  var isMin = value <= min;
  var isMax = value >= max;
  var isLessDisabled = isMin || value - step < min;
  var isMoreDisabled = isMax || value + step > max;
  return {
    onLessClick: onLessClick,
    onMoreClick: onMoreClick,
    isMin: isMin,
    isMax: isMax,
    isLessDisabled: isLessDisabled,
    isMoreDisabled: isMoreDisabled
  };
};
