// @salutejs/plasma-new-hope/cjs/components/Combobox/ui/ComboboxDivider/ComboboxDivider.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('react');
require('../../../../utils/react.js');
var DropdownDivider = require('../../../Dropdown/ui/DropdownDivider/DropdownDivider.js');
require('@salutejs/plasma-core');
require('@linaria/react');
require('@linaria/core');
require('focus-visible');
require('react-dom');
require('react-popper');
require('styled-components');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var ComboboxDivider_styles = require('./ComboboxDivider.styles.js');

var comboboxDividerRoot = DropdownDivider.dropdownDividerRoot;
var comboboxDividerConfig = {
  name: 'ComboboxDivider',
  tag: 'div',
  layout: comboboxDividerRoot,
  base: ComboboxDivider_styles.base,
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

exports.comboboxDividerConfig = comboboxDividerConfig;
exports.comboboxDividerRoot = comboboxDividerRoot;
