// @salutejs/plasma-new-hope/cjs/components/TextField/ui/TextFieldChip/TextFieldChip.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var TextFieldChip_styles = require('./TextFieldChip.styles.js');

/**
 * Чип, отображающий выбранный элемент
 */
var TextFieldChip = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var id = _ref.id,
    text = _ref.text,
    disabled = _ref.disabled,
    readOnly = _ref.readOnly,
    onClick = _ref.onClick,
    onClear = _ref.onClear,
    onKeyDown = _ref.onKeyDown;
  var onClearHandle = function onClearHandle() {
    return onClear(id);
  };
  var onKeyDownHandle = function onKeyDownHandle(event) {
    onKeyDown === null || onKeyDown === void 0 || onKeyDown(event, id);
  };
  return /*#__PURE__*/React.createElement(TextFieldChip_styles.StyledChip, {
    ref: ref,
    tabIndex: -1,
    disabled: disabled,
    readOnly: readOnly,
    onClick: onClick,
    onClear: onClearHandle,
    onKeyDown: onKeyDownHandle,
    text: "".concat(text)
  });
});

exports.TextFieldChip = TextFieldChip;
