// @salutejs/plasma-tokens-web/es/new/index.js
// Generated by robots, do not change this manually!
export var textPrimaryHover = 'var(--text-primary-hover, #08080893)';
export var textPrimaryActive = 'var(--text-primary-active, #080808C4)';
export var textPrimary = 'var(--text-primary, rgba(8, 8, 8, 0.96))';
export var textSecondaryHover = 'var(--text-secondary-hover, #080808FF)';
export var textSecondaryActive = 'var(--text-secondary-active, #080808AB)';
export var textSecondary = 'var(--text-secondary, rgba(8, 8, 8, 0.56))';
export var textTertiaryHover = 'var(--text-tertiary-hover, #080808FF)';
export var textTertiaryActive = 'var(--text-tertiary-active, #08080856)';
export var textTertiary = 'var(--text-tertiary, rgba(8, 8, 8, 0.28))';
export var textParagraphHover = 'var(--text-paragraph-hover, #0808087A)';
export var textParagraphActive = 'var(--text-paragraph-active, #080808A3)';
export var textParagraph = 'var(--text-paragraph, rgba(8, 8, 8, 0.8))';
export var textAccentHover = 'var(--text-accent-hover, #528DFAFF)';
export var textAccentActive = 'var(--text-accent-active, #075AF2FF)';
export var textAccent = 'var(--text-accent, #2A72F8)';
export var textAccentGradientHover = 'var(--text-accent-gradient-hover, #000000FF)';
export var textAccentGradientActive = 'var(--text-accent-gradient-active, #CCCCCCFF)';
export var textAccentGradient = 'var(--text-accent-gradient, linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%))';
export var textPositiveHover = 'var(--text-positive-hover, #1FC13DFF)';
export var textPositiveActive = 'var(--text-positive-active, #147B27FF)';
export var textPositive = 'var(--text-positive, #1A9E32)';
export var textWarningHover = 'var(--text-warning-hover, #FB782DFF)';
export var textWarningActive = 'var(--text-warning-active, #D25004FF)';
export var textWarning = 'var(--text-warning, #FA5F05)';
export var textNegativeHover = 'var(--text-negative-hover, #F54254FF)';
export var textNegativeActive = 'var(--text-negative-active, #DA0B20FF)';
export var textNegative = 'var(--text-negative, #F31B31)';
export var onDarkTextPrimaryHover = 'var(--on-dark-text-primary-hover, #FFFFFF93)';
export var onDarkTextPrimaryActive = 'var(--on-dark-text-primary-active, #FFFFFFC4)';
export var onDarkTextPrimary = 'var(--on-dark-text-primary, rgba(255, 255, 255, 0.96))';
export var onDarkTextSecondaryHover = 'var(--on-dark-text-secondary-hover, #FFFFFFFF)';
export var onDarkTextSecondaryActive = 'var(--on-dark-text-secondary-active, #FFFFFFAB)';
export var onDarkTextSecondary = 'var(--on-dark-text-secondary, rgba(255, 255, 255, 0.56))';
export var onDarkTextTertiaryHover = 'var(--on-dark-text-tertiary-hover, #FFFFFFFF)';
export var onDarkTextTertiaryActive = 'var(--on-dark-text-tertiary-active, #FFFFFF56)';
export var onDarkTextTertiary = 'var(--on-dark-text-tertiary, rgba(255, 255, 255, 0.28))';
export var onDarkTextParagraphHover = 'var(--on-dark-text-paragraph-hover, #FFFFFF7A)';
export var onDarkTextParagraphActive = 'var(--on-dark-text-paragraph-active, #FFFFFFA3)';
export var onDarkTextParagraph = 'var(--on-dark-text-paragraph, rgba(255, 255, 255, 0.8))';
export var onDarkTextAccentHover = 'var(--on-dark-text-accent-hover, #689CFDFF)';
export var onDarkTextAccentActive = 'var(--on-dark-text-accent-active, #1767FDFF)';
export var onDarkTextAccent = 'var(--on-dark-text-accent, #3F81FD)';
export var onDarkTextAccentGradientHover = 'var(--on-dark-text-accent-gradient-hover, #000000FF)';
export var onDarkTextAccentGradientActive = 'var(--on-dark-text-accent-gradient-active, #CCCCCCFF)';
export var onDarkTextAccentGradient = 'var(--on-dark-text-accent-gradient, linear-gradient(89.83deg, #5E94FF 0%, #43DBFA 100%))';
export var onDarkTextPositiveHover = 'var(--on-dark-text-positive-hover, #1FC13DFF)';
export var onDarkTextPositiveActive = 'var(--on-dark-text-positive-active, #147B27FF)';
export var onDarkTextPositive = 'var(--on-dark-text-positive, #1A9E32)';
export var onDarkTextWarningHover = 'var(--on-dark-text-warning-hover, #FB782DFF)';
export var onDarkTextWarningActive = 'var(--on-dark-text-warning-active, #D25004FF)';
export var onDarkTextWarning = 'var(--on-dark-text-warning, #FA5F05)';
export var onDarkTextNegativeHover = 'var(--on-dark-text-negative-hover, #FF5263FF)';
export var onDarkTextNegativeActive = 'var(--on-dark-text-negative-active, #FF001AFF)';
export var onDarkTextNegative = 'var(--on-dark-text-negative, #FF293E)';
export var onLightTextPrimaryHover = 'var(--on-light-text-primary-hover, #08080893)';
export var onLightTextPrimaryActive = 'var(--on-light-text-primary-active, #080808C4)';
export var onLightTextPrimary = 'var(--on-light-text-primary, rgba(8, 8, 8, 0.96))';
export var onLightTextSecondaryHover = 'var(--on-light-text-secondary-hover, #080808FF)';
export var onLightTextSecondaryActive = 'var(--on-light-text-secondary-active, #080808AB)';
export var onLightTextSecondary = 'var(--on-light-text-secondary, rgba(8, 8, 8, 0.56))';
export var onLightTextTertiaryHover = 'var(--on-light-text-tertiary-hover, #080808FF)';
export var onLightTextTertiaryActive = 'var(--on-light-text-tertiary-active, #08080856)';
export var onLightTextTertiary = 'var(--on-light-text-tertiary, rgba(8, 8, 8, 0.28))';
export var onLightTextParagraphHover = 'var(--on-light-text-paragraph-hover, #0808087A)';
export var onLightTextParagraphActive = 'var(--on-light-text-paragraph-active, #080808A3)';
export var onLightTextParagraph = 'var(--on-light-text-paragraph, rgba(8, 8, 8, 0.8))';
export var onLightTextAccentHover = 'var(--on-light-text-accent-hover, #528DFAFF)';
export var onLightTextAccentActive = 'var(--on-light-text-accent-active, #075AF2FF)';
export var onLightTextAccent = 'var(--on-light-text-accent, #2A72F8)';
export var onLightTextAccentGradientHover = 'var(--on-light-text-accent-gradient-hover, #000000FF)';
export var onLightTextAccentGradientActive = 'var(--on-light-text-accent-gradient-active, #CCCCCCFF)';
export var onLightTextAccentGradient = 'var(--on-light-text-accent-gradient, linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%))';
export var onLightTextPositiveHover = 'var(--on-light-text-positive-hover, #1FC13DFF)';
export var onLightTextPositiveActive = 'var(--on-light-text-positive-active, #147B27FF)';
export var onLightTextPositive = 'var(--on-light-text-positive, #1A9E32)';
export var onLightTextWarningHover = 'var(--on-light-text-warning-hover, #FB782DFF)';
export var onLightTextWarningActive = 'var(--on-light-text-warning-active, #D25004FF)';
export var onLightTextWarning = 'var(--on-light-text-warning, #FA5F05)';
export var onLightTextNegativeHover = 'var(--on-light-text-negative-hover, #F54254FF)';
export var onLightTextNegativeActive = 'var(--on-light-text-negative-active, #DA0B20FF)';
export var onLightTextNegative = 'var(--on-light-text-negative, #F31B31)';
export var inverseTextPrimaryHover = 'var(--inverse-text-primary-hover, #FFFFFF93)';
export var inverseTextPrimaryActive = 'var(--inverse-text-primary-active, #FFFFFFC4)';
export var inverseTextPrimary = 'var(--inverse-text-primary, rgba(255, 255, 255, 0.96))';
export var inverseTextSecondaryHover = 'var(--inverse-text-secondary-hover, #FFFFFFFF)';
export var inverseTextSecondaryActive = 'var(--inverse-text-secondary-active, #FFFFFFAB)';
export var inverseTextSecondary = 'var(--inverse-text-secondary, rgba(255, 255, 255, 0.56))';
export var inverseTextTertiaryHover = 'var(--inverse-text-tertiary-hover, #FFFFFFFF)';
export var inverseTextTertiaryActive = 'var(--inverse-text-tertiary-active, #FFFFFF56)';
export var inverseTextTertiary = 'var(--inverse-text-tertiary, rgba(255, 255, 255, 0.28))';
export var inverseTextParagraphHover = 'var(--inverse-text-paragraph-hover, #FFFFFF7A)';
export var inverseTextParagraphActive = 'var(--inverse-text-paragraph-active, #FFFFFFA3)';
export var inverseTextParagraph = 'var(--inverse-text-paragraph, rgba(255, 255, 255, 0.8))';
export var surfaceSolidPrimaryHover = 'var(--surface-solid-primary-hover, #FAFAFAFF)';
export var surfaceSolidPrimaryActive = 'var(--surface-solid-primary-active, #F0F0F0FF)';
export var surfaceSolidPrimary = 'var(--surface-solid-primary, #F5F5F5)';
export var surfaceSolidSecondaryHover = 'var(--surface-solid-secondary-hover, #F7F7F7FF)';
export var surfaceSolidSecondaryActive = 'var(--surface-solid-secondary-active, #E8E8E8FF)';
export var surfaceSolidSecondary = 'var(--surface-solid-secondary, #ECECEC)';
export var surfaceSolidTertiaryHover = 'var(--surface-solid-tertiary-hover, #EDEDEDFF)';
export var surfaceSolidTertiaryActive = 'var(--surface-solid-tertiary-active, #D4D4D4FF)';
export var surfaceSolidTertiary = 'var(--surface-solid-tertiary, #DDDDDD)';
export var surfaceSolidCardHover = 'var(--surface-solid-card-hover, #FFFFFFFF)';
export var surfaceSolidCardActive = 'var(--surface-solid-card-active, #FFFFFFFF)';
export var surfaceSolidCard = 'var(--surface-solid-card, rgba(255, 255, 255, 1))';
export var surfaceSolidDefaultHover = 'var(--surface-solid-default-hover, #262626FF)';
export var surfaceSolidDefaultActive = 'var(--surface-solid-default-active, #030303FF)';
export var surfaceSolidDefault = 'var(--surface-solid-default, #080808)';
export var surfaceTransparentPrimaryHover = 'var(--surface-transparent-primary-hover, #08080803)';
export var surfaceTransparentPrimaryActive = 'var(--surface-transparent-primary-active, #0808080D)';
export var surfaceTransparentPrimary = 'var(--surface-transparent-primary, rgba(8, 8, 8, 0.03))';
export var surfaceTransparentSecondaryHover = 'var(--surface-transparent-secondary-hover, #08080805)';
export var surfaceTransparentSecondaryActive = 'var(--surface-transparent-secondary-active, #0808081A)';
export var surfaceTransparentSecondary = 'var(--surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
export var surfaceTransparentTertiaryHover = 'var(--surface-transparent-tertiary-hover, #0808080A)';
export var surfaceTransparentTertiaryActive = 'var(--surface-transparent-tertiary-active, #08080829)';
export var surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
export var surfaceTransparentDeepHover = 'var(--surface-transparent-deep-hover, #0808088F)';
export var surfaceTransparentDeepActive = 'var(--surface-transparent-deep-active, #080808AD)';
export var surfaceTransparentDeep = 'var(--surface-transparent-deep, rgba(8, 8, 8, 0.64))';
export var surfaceTransparentCardHover = 'var(--surface-transparent-card-hover, #FFFFFFFF)';
export var surfaceTransparentCardActive = 'var(--surface-transparent-card-active, #FFFFFFFF)';
export var surfaceTransparentCard = 'var(--surface-transparent-card, rgba(255, 255, 255, 1))';
export var surfaceClearHover = 'var(--surface-clear-hover, #FFFFFF00)';
export var surfaceClearActive = 'var(--surface-clear-active, #FFFFFF00)';
export var surfaceClear = 'var(--surface-clear, rgba(255, 255, 255, 0))';
export var surfaceAccentHover = 'var(--surface-accent-hover, #528DFAFF)';
export var surfaceAccentActive = 'var(--surface-accent-active, #1665F8FF)';
export var surfaceAccent = 'var(--surface-accent, #2A72F8)';
export var surfaceAccentGradientHover = 'var(--surface-accent-gradient-hover, #FFFFFFFF)';
export var surfaceAccentGradientActive = 'var(--surface-accent-gradient-active, #FFFFFFFF)';
export var surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%))';
export var surfaceTransparentAccentHover = 'var(--surface-transparent-accent-hover, #2A72F805)';
export var surfaceTransparentAccentActive = 'var(--surface-transparent-accent-active, #2A72F81A)';
export var surfacePositiveHover = 'var(--surface-positive-hover, #1EB83AFF)';
export var surfacePositiveActive = 'var(--surface-positive-active, #178C2CFF)';
export var surfacePositive = 'var(--surface-positive, #1A9E32)';
export var surfaceWarningHover = 'var(--surface-warning-hover, #FB782DFF)';
export var surfaceWarningActive = 'var(--surface-warning-active, #E65705FF)';
export var surfaceWarning = 'var(--surface-warning, #FA5F05)';
export var surfaceNegativeHover = 'var(--surface-negative-hover, #F54254FF)';
export var surfaceNegativeActive = 'var(--surface-negative-active, #ED0C23FF)';
export var surfaceNegative = 'var(--surface-negative, #F31B31)';
export var surfaceTransparentPositiveHover = 'var(--surface-transparent-positive-hover, #1A9E3205)';
export var surfaceTransparentPositiveActive = 'var(--surface-transparent-positive-active, #1A9E321A)';
export var surfaceTransparentWarningHover = 'var(--surface-transparent-warning-hover, #FA5F0505)';
export var surfaceTransparentWarningActive = 'var(--surface-transparent-warning-active, #FA5F051A)';
export var surfaceTransparentNegativeHover = 'var(--surface-transparent-negative-hover, #F31B3105)';
export var surfaceTransparentNegativeActive = 'var(--surface-transparent-negative-active, #F31B311A)';
export var surfaceTransparentAccent = 'var(--surface-transparent-accent, rgba(42,114,248,0.06))';
export var surfaceTransparentPositive = 'var(--surface-transparent-positive, rgba(26,158,50,0.06))';
export var surfaceTransparentWarning = 'var(--surface-transparent-warning, rgba(250,95,5,0.06))';
export var surfaceTransparentNegative = 'var(--surface-transparent-negative, rgba(243,27,49,0.06))';
export var onDarkSurfaceSolidPrimaryHover = 'var(--on-dark-surface-solid-primary-hover, #363636FF)';
export var onDarkSurfaceSolidPrimaryActive = 'var(--on-dark-surface-solid-primary-active, #0D0D0DFF)';
export var onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #171717)';
export var onDarkSurfaceSolidSecondaryHover = 'var(--on-dark-surface-solid-secondary-hover, #3B3B3BFF)';
export var onDarkSurfaceSolidSecondaryActive = 'var(--on-dark-surface-solid-secondary-active, #1C1C1CFF)';
export var onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #262626)';
export var onDarkSurfaceSolidTertiaryHover = 'var(--on-dark-surface-solid-tertiary-hover, #4A4A4AFF)';
export var onDarkSurfaceSolidTertiaryActive = 'var(--on-dark-surface-solid-tertiary-active, #2B2B2BFF)';
export var onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #363636)';
export var onDarkSurfaceSolidCardHover = 'var(--on-dark-surface-solid-card-hover, #363636FF)';
export var onDarkSurfaceSolidCardActive = 'var(--on-dark-surface-solid-card-active, #0D0D0DFF)';
export var onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #171717)';
export var onDarkSurfaceSolidDefaultHover = 'var(--on-dark-surface-solid-default-hover, #FFFFFFFF)';
export var onDarkSurfaceSolidDefaultActive = 'var(--on-dark-surface-solid-default-active, #FFFFFFFF)';
export var onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, rgba(255, 255, 255, 1))';
export var onDarkSurfaceTransparentPrimaryHover = 'var(--on-dark-surface-transparent-primary-hover, #363636FF)';
export var onDarkSurfaceTransparentPrimaryActive = 'var(--on-dark-surface-transparent-primary-active, #0D0D0DFF)';
export var onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, #171717)';
export var onDarkSurfaceTransparentSecondaryHover = 'var(--on-dark-surface-transparent-secondary-hover, #3B3B3BFF)';
export var onDarkSurfaceTransparentSecondaryActive = 'var(--on-dark-surface-transparent-secondary-active, #1C1C1CFF)';
export var onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, #262626)';
export var onDarkSurfaceTransparentTertiaryHover = 'var(--on-dark-surface-transparent-tertiary-hover, #4A4A4AFF)';
export var onDarkSurfaceTransparentTertiaryActive = 'var(--on-dark-surface-transparent-tertiary-active, #2B2B2BFF)';
export var onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, #363636)';
export var onDarkSurfaceTransparentDeepHover = 'var(--on-dark-surface-transparent-deep-hover, #FFFFFF8F)';
export var onDarkSurfaceTransparentDeepActive = 'var(--on-dark-surface-transparent-deep-active, #FFFFFFAD)';
export var onDarkSurfaceTransparentDeep = 'var(--on-dark-surface-transparent-deep, rgba(255, 255, 255, 0.64))';
export var onDarkSurfaceTransparentCardHover = 'var(--on-dark-surface-transparent-card-hover, #FFFFFF0A)';
export var onDarkSurfaceTransparentCardActive = 'var(--on-dark-surface-transparent-card-active, #FFFFFF29)';
export var onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, rgba(255, 255, 255, 0.12))';
export var onDarkSurfaceAccentHover = 'var(--on-dark-surface-accent-hover, #689CFDFF)';
export var onDarkSurfaceAccentActive = 'var(--on-dark-surface-accent-active, #2B74FDFF)';
export var onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #3F81FD)';
export var onDarkSurfaceAccentGradientHover = 'var(--on-dark-surface-accent-gradient-hover, #FFFFFFFF)';
export var onDarkSurfaceAccentGradientActive = 'var(--on-dark-surface-accent-gradient-active, #FFFFFFFF)';
export var onDarkSurfaceAccentGradient = 'var(--on-dark-surface-accent-gradient, linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%))';
export var onDarkSurfacePositiveHover = 'var(--on-dark-surface-positive-hover, #1EB83AFF)';
export var onDarkSurfacePositiveActive = 'var(--on-dark-surface-positive-active, #178C2CFF)';
export var onDarkSurfacePositive = 'var(--on-dark-surface-positive, #1A9E32)';
export var onDarkSurfaceWarningHover = 'var(--on-dark-surface-warning-hover, #FB782DFF)';
export var onDarkSurfaceWarningActive = 'var(--on-dark-surface-warning-active, #E65705FF)';
export var onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #FA5F05)';
export var onDarkSurfaceNegativeHover = 'var(--on-dark-surface-negative-hover, #FF5263FF)';
export var onDarkSurfaceNegativeActive = 'var(--on-dark-surface-negative-active, #FF142CFF)';
export var onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293E)';
export var onLightSurfaceSolidPrimaryHover = 'var(--on-light-surface-solid-primary-hover, #FAFAFAFF)';
export var onLightSurfaceSolidPrimaryActive = 'var(--on-light-surface-solid-primary-active, #F0F0F0FF)';
export var onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F5F5F5)';
export var onLightSurfaceSolidSecondaryHover = 'var(--on-light-surface-solid-secondary-hover, #F7F7F7FF)';
export var onLightSurfaceSolidSecondaryActive = 'var(--on-light-surface-solid-secondary-active, #E8E8E8FF)';
export var onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #ECECEC)';
export var onLightSurfaceSolidTertiaryHover = 'var(--on-light-surface-solid-tertiary-hover, #EDEDEDFF)';
export var onLightSurfaceSolidTertiaryActive = 'var(--on-light-surface-solid-tertiary-active, #D4D4D4FF)';
export var onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #DDDDDD)';
export var onLightSurfaceSolidCardHover = 'var(--on-light-surface-solid-card-hover, #FFFFFFFF)';
export var onLightSurfaceSolidCardActive = 'var(--on-light-surface-solid-card-active, #FFFFFFFF)';
export var onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, rgba(255, 255, 255, 1))';
export var onLightSurfaceSolidDefaultHover = 'var(--on-light-surface-solid-default-hover, #262626FF)';
export var onLightSurfaceSolidDefaultActive = 'var(--on-light-surface-solid-default-active, #030303FF)';
export var onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #080808)';
export var onLightSurfaceTransparentPrimaryHover = 'var(--on-light-surface-transparent-primary-hover, #08080803)';
export var onLightSurfaceTransparentPrimaryActive = 'var(--on-light-surface-transparent-primary-active, #0808080D)';
export var onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, rgba(8, 8, 8, 0.03))';
export var onLightSurfaceTransparentSecondaryHover = 'var(--on-light-surface-transparent-secondary-hover, #08080805)';
export var onLightSurfaceTransparentSecondaryActive = 'var(--on-light-surface-transparent-secondary-active, #0808081A)';
export var onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
export var onLightSurfaceTransparentTertiaryHover = 'var(--on-light-surface-transparent-tertiary-hover, #0808080A)';
export var onLightSurfaceTransparentTertiaryActive = 'var(--on-light-surface-transparent-tertiary-active, #08080829)';
export var onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
export var onLightSurfaceTransparentDeepHover = 'var(--on-light-surface-transparent-deep-hover, #0808088F)';
export var onLightSurfaceTransparentDeepActive = 'var(--on-light-surface-transparent-deep-active, #080808AD)';
export var onLightSurfaceTransparentDeep = 'var(--on-light-surface-transparent-deep, rgba(8, 8, 8, 0.64))';
export var onLightSurfaceTransparentCardHover = 'var(--on-light-surface-transparent-card-hover, #FFFFFFFF)';
export var onLightSurfaceTransparentCardActive = 'var(--on-light-surface-transparent-card-active, #FFFFFFFF)';
export var onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, rgba(255, 255, 255, 1))';
export var onLightSurfaceAccentHover = 'var(--on-light-surface-accent-hover, #528DFAFF)';
export var onLightSurfaceAccentActive = 'var(--on-light-surface-accent-active, #1665F8FF)';
export var onLightSurfaceAccent = 'var(--on-light-surface-accent, #2A72F8)';
export var onLightSurfaceAccentGradientHover = 'var(--on-light-surface-accent-gradient-hover, #FFFFFFFF)';
export var onLightSurfaceAccentGradientActive = 'var(--on-light-surface-accent-gradient-active, #FFFFFFFF)';
export var onLightSurfaceAccentGradient = 'var(--on-light-surface-accent-gradient, linear-gradient(93.97deg, #3E79F0 6.49%, #27C6E5 93.51%))';
export var onLightSurfacePositiveHover = 'var(--on-light-surface-positive-hover, #1EB83AFF)';
export var onLightSurfacePositiveActive = 'var(--on-light-surface-positive-active, #178C2CFF)';
export var onLightSurfacePositive = 'var(--on-light-surface-positive, #1A9E32)';
export var onLightSurfaceWarningHover = 'var(--on-light-surface-warning-hover, #FB782DFF)';
export var onLightSurfaceWarningActive = 'var(--on-light-surface-warning-active, #E65705FF)';
export var onLightSurfaceWarning = 'var(--on-light-surface-warning, #FA5F05)';
export var onLightSurfaceNegativeHover = 'var(--on-light-surface-negative-hover, #F54254FF)';
export var onLightSurfaceNegativeActive = 'var(--on-light-surface-negative-active, #ED0C23FF)';
export var onLightSurfaceNegative = 'var(--on-light-surface-negative, #F31B31)';
export var backgroundPrimary = 'var(--background-primary, #F9F9F9)';
export var darkBackgroundPrimary = 'var(--dark-background-primary, #080808)';
export var lightBackgroundPrimary = 'var(--light-background-primary, #F9F9F9)';
export var overlaySoft = 'var(--overlay-soft, rgba(249, 249, 249, 0.56))';
export var overlayHard = 'var(--overlay-hard, rgba(249, 249, 249, 0.96))';
export var overlayBlur = 'var(--overlay-blur, rgba(249, 249, 249, 0.28))';
export var onDarkOverlaySoft = 'var(--on-dark-overlay-soft, rgba(8, 8, 8, 0.56))';
export var onDarkOverlayHard = 'var(--on-dark-overlay-hard, rgba(8, 8, 8, 0.96))';
export var onDarkOverlayBlur = 'var(--on-dark-overlay-blur, rgba(8, 8, 8, 0.28))';
export var onLightOverlaySoft = 'var(--on-light-overlay-soft, rgba(249, 249, 249, 0.56))';
export var onLightOverlayHard = 'var(--on-light-overlay-hard, rgba(249, 249, 249, 0.96))';
export var onLightOverlayBlur = 'var(--on-light-overlay-blur, rgba(249, 249, 249, 0.28))';
export var skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(8, 8, 8, 0.09) 0%, rgba(8, 8, 8, 0.08) 6.25%, rgba(8, 8, 8, 0.05) 12.5%, rgba(8, 8, 8, 0.01) 25%, rgba(8, 8, 8, 0.05) 37.5%, rgba(8, 8, 8, 0.08) 43.75%, rgba(8, 8, 8, 0.09) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.05) 62.5%, rgba(8, 8, 8, 0.01) 75%, rgba(8, 8, 8, 0.05) 87.5%, rgba(8, 8, 8, 0.08) 93.75%, rgba(8, 8, 8, 0.09) 100% ))';
export var skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(8, 8, 8, 0.36) 0%, rgba(8, 8, 8, 0.32) 6.25%, rgba(8, 8, 8, 0.20) 12.5%, rgba(8, 8, 8, 0.04) 25%, rgba(8, 8, 8, 0.20) 37.5%, rgba(8, 8, 8, 0.32) 43.75%, rgba(8, 8, 8, 0.36) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.20) 62.5%, rgba(8, 8, 8, 0.04) 75%, rgba(8, 8, 8, 0.20) 87.5%, rgba(8, 8, 8, 0.32) 93.75%, rgba(8, 8, 8, 0.36) 100% ))';
