// @salutejs/plasma-new-hope/styled-components/cjs/components/Segment/ui/SegmentItem/SegmentItem.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.segmentItemRoot = exports.segmentItemConfig = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _utils = /*#__PURE__*/require("../../../../utils");
var _tokens = /*#__PURE__*/require("../../tokens");
var _SegmentProvider = /*#__PURE__*/require("../../SegmentProvider/SegmentProvider");
var _base = /*#__PURE__*/require("./variations/_size/base");
var _base2 = /*#__PURE__*/require("./variations/_view/base");
var _base3 = /*#__PURE__*/require("./variations/_disabled/base");
var _base4 = /*#__PURE__*/require("./variations/_pilled/base");
var _SegmentItem = /*#__PURE__*/require("./SegmentItem.styles");
var _excluded = ["size", "view", "id", "label", "value", "pilled", "customHandleSelect", "aria-label"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var segmentItemRoot = exports.segmentItemRoot = function segmentItemRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (props, outerRef) {
    var size = props.size,
      view = props.view,
      id = props.id,
      label = props.label,
      value = props.value,
      pilled = props.pilled,
      customHandleSelect = props.customHandleSelect,
      ariaLabelExternal = props['aria-label'],
      rest = _objectWithoutProperties(props, _excluded);
    var _useSegmentInner = (0, _SegmentProvider.useSegmentInner)(),
      selectionMode = _useSegmentInner.selectionMode,
      disabledGroup = _useSegmentInner.disabledGroup,
      handleSelect = _useSegmentInner.handleSelect,
      selectedSegmentItems = _useSegmentInner.selectedSegmentItems;
    var uniqId = (0, _plasmaCore.safeUseId)();
    var segmentId = id || "label-".concat(uniqId);
    var ariaLabelDefault = (0, _react.useMemo)(function () {
      return (0, _utils.extractTextFrom)(label);
    }, [label]);
    var pilledAttr = view !== 'clear' && pilled;
    var pilledClass = pilledAttr ? _tokens.classes.segmentPilled : undefined;
    var isSelected = selectedSegmentItems === null || selectedSegmentItems === void 0 ? void 0 : selectedSegmentItems.includes(value || ariaLabelDefault);
    var selectedClass = isSelected ? _tokens.classes.selectedSegmentItem : undefined;
    var handleSelectSegment = (0, _react.useCallback)(function (event) {
      if (disabledGroup) {
        return;
      }
      customHandleSelect === null || customHandleSelect === void 0 || customHandleSelect(event);
      handleSelect === null || handleSelect === void 0 || handleSelect(value || ariaLabelDefault);
    }, [selectionMode]);
    return /*#__PURE__*/_react["default"].createElement(Root, _extends({
      view: view,
      size: size,
      id: segmentId,
      ref: outerRef,
      "aria-label": ariaLabelExternal || ariaLabelDefault,
      value: value,
      pilled: pilledAttr,
      className: (0, _utils.cx)(selectedClass, pilledClass),
      onClick: handleSelectSegment,
      tabIndex: disabledGroup ? -1 : 0,
      disabled: disabledGroup
    }, rest), /*#__PURE__*/_react["default"].createElement(_SegmentItem.StyledContent, null, label || value));
  });
};
var segmentItemConfig = exports.segmentItemConfig = {
  name: 'SegmentItem',
  tag: 'button',
  layout: segmentItemRoot,
  base: _SegmentItem.base,
  variations: {
    size: {
      css: _base.base
    },
    view: {
      css: _base2.base
    },
    disabled: {
      css: _base3.base,
      attrs: true
    },
    pilled: {
      css: _base4.base,
      attrs: true
    }
  },
  defaults: {
    view: 'clear',
    size: 'xs'
  }
};
