// @salutejs/plasma-new-hope/styled-components/es/examples/plasma_web/components/Header/Header.config.js
import { css } from 'styled-components';
import { headerTokens } from '../../../../components/Header';
export var config = {
  defaults: {
    view: 'default',
    size: 'm'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/css(["", ":var(--text-primary);", ":var(--surface-clear);", ":transform 0.15s ease-in-out;", ":var(--plasma-colors-primary);", ":var(--plasma-colors-primary);", ":var(--plasma-colors-primary);", ":var(--plasma-colors-secondary);", ":var(--plasma-colors-secondary);", ":var(--plasma-colors-secondary);"], headerTokens.headerArrowButtonColor, headerTokens.headerArrowButtonBackgroundColor, headerTokens.headerArrowTransition, headerTokens.headerTitleColor, headerTokens.headerBigTitleColor, headerTokens.headerBiggerTitleColor, headerTokens.headerLabelColor, headerTokens.headerSubTitleColor, headerTokens.headerCaptionColor)
    },
    size: {
      m: /*#__PURE__*/css(["", ":2.25rem;", ":1.625rem;", ":0.625rem;", ":0.75rem;", ":1rem;", ":1.5rem;", ":0.125rem 0;", ":0.5rem 0;", ":0.5rem 0;", ":0.375rem 0;", ":0.125rem 0;", ":0.25rem 0;", ":2.25rem;", ":0.5rem;"], headerTokens.headerHeight, headerTokens.headerPaddingTop, headerTokens.headerPaddingBottom, headerTokens.headerLogoMarginRight, headerTokens.headerArrowMarginRight, headerTokens.headerArrowIconSize, headerTokens.headerTitleMargin, headerTokens.headerBigTitleMargin, headerTokens.headerBiggerTitleMargin, headerTokens.headerLabelMargin, headerTokens.headerSubTitleMargin, headerTokens.headerCaptionMargin, headerTokens.headerArrowHeight, headerTokens.headerArrowPadding)
    }
  }
};
