// @salutejs/plasma-new-hope/cjs/components/Select/ui/SelectChip/SelectChip.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var SelectChip_styles = require('./SelectChip.styles.js');

/**
 * Чип, отображающий выбранный элемент
 */
var SelectChip = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var value = _ref.value,
    text = _ref.text,
    index = _ref.index,
    disabled = _ref.disabled,
    readOnly = _ref.readOnly,
    onClick = _ref.onClick,
    onClear = _ref.onClear,
    onKeyDown = _ref.onKeyDown;
  var onClearHandle = React.useCallback(function () {
    onClear(value, text);
  }, [onClear, value, text]);
  var onKeyDownHandle = React.useCallback(function (event) {
    onKeyDown === null || onKeyDown === void 0 || onKeyDown(value, text, index, event);
  }, [onKeyDown]);
  return /*#__PURE__*/React.createElement(SelectChip_styles.StyledChip, {
    ref: ref,
    tabIndex: -1,
    disabled: disabled,
    readOnly: readOnly,
    onClick: onClick,
    onClear: onClearHandle,
    onKeyDown: onKeyDownHandle,
    text: "".concat(text)
  });
});

exports.SelectChip = SelectChip;
