// @salutejs/plasma-new-hope/es/mixins/typography.js
import { objectSpread2 as _objectSpread2 } from '../_virtual/_rollupPluginBabelHelpers.js';

var dsplL = {
  fontFamily: 'var(--plasma-typo-dspl-l-font-family)',
  fontSize: 'var(--plasma-typo-dspl-l-font-size)',
  fontStyle: 'var(--plasma-typo-dspl-l-font-style)',
  fontWeight: 'var(--plasma-typo-dspl-l-font-weight)',
  letterSpacing: 'var(--plasma-typo-dspl-l-letter-spacing)',
  lineHeight: 'var(--plasma-typo-dspl-l-line-height)'
};
var dsplLBold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, dsplL), {}, {
  fontWeight: 'var(--plasma-typo-dspl-l-bold-font-weight)'
});
var dsplM = {
  fontFamily: 'var(--plasma-typo-dspl-m-font-family)',
  fontSize: 'var(--plasma-typo-dspl-m-font-size)',
  fontStyle: 'var(--plasma-typo-dspl-m-font-style)',
  fontWeight: 'var(--plasma-typo-dspl-m-font-weight)',
  letterSpacing: 'var(--plasma-typo-dspl-m-letter-spacing)',
  lineHeight: 'var(--plasma-typo-dspl-m-line-height)'
};
var dsplMBold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, dsplM), {}, {
  fontWeight: 'var(--plasma-typo-dspl-m-bold-font-weight)'
});
var dsplS = {
  fontFamily: 'var(--plasma-typo-dspl-s-font-family)',
  fontSize: 'var(--plasma-typo-dspl-s-font-size)',
  fontStyle: 'var(--plasma-typo-dspl-s-font-style)',
  fontWeight: 'var(--plasma-typo-dspl-s-font-weight)',
  letterSpacing: 'var(--plasma-typo-dspl-s-letter-spacing)',
  lineHeight: 'var(--plasma-typo-dspl-s-line-height)'
};
var dsplSBold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, dsplS), {}, {
  fontWeight: 'var(--plasma-typo-dspl-s-bold-font-weight)'
});
var h1 = {
  fontFamily: 'var(--plasma-typo-h1-font-family)',
  fontSize: 'var(--plasma-typo-h1-font-size)',
  fontStyle: 'var(--plasma-typo-h1-font-style)',
  fontWeight: 'var(--plasma-typo-h1-font-weight)',
  letterSpacing: 'var(--plasma-typo-h1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-h1-line-height)'
};
var h1Bold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, h1), {}, {
  fontWeight: 'var(--plasma-typo-h1-bold-font-weight)'
});
var h2 = {
  fontFamily: 'var(--plasma-typo-h2-font-family)',
  fontSize: 'var(--plasma-typo-h2-font-size)',
  fontStyle: 'var(--plasma-typo-h2-font-style)',
  fontWeight: 'var(--plasma-typo-h2-font-weight)',
  letterSpacing: 'var(--plasma-typo-h2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-h2-line-height)'
};
var h2Bold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, h2), {}, {
  fontWeight: 'var(--plasma-typo-h2-bold-font-weight)'
});
var h3 = {
  fontFamily: 'var(--plasma-typo-h3-font-family)',
  fontSize: 'var(--plasma-typo-h3-font-size)',
  fontStyle: 'var(--plasma-typo-h3-font-style)',
  fontWeight: 'var(--plasma-typo-h3-font-weight)',
  letterSpacing: 'var(--plasma-typo-h3-letter-spacing)',
  lineHeight: 'var(--plasma-typo-h3-line-height)'
};
var h3Bold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, h3), {}, {
  fontWeight: 'var(--plasma-typo-h3-bold-font-weight)'
});
var h4 = {
  fontFamily: 'var(--plasma-typo-h4-font-family)',
  fontSize: 'var(--plasma-typo-h4-font-size)',
  fontStyle: 'var(--plasma-typo-h4-font-style)',
  fontWeight: 'var(--plasma-typo-h4-font-weight)',
  letterSpacing: 'var(--plasma-typo-h4-letter-spacing)',
  lineHeight: 'var(--plasma-typo-h4-line-height)'
};
var h4Bold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, h4), {}, {
  fontWeight: 'var(--plasma-typo-h4-bold-font-weight)'
});
var h5 = {
  fontFamily: 'var(--plasma-typo-h5-font-family)',
  fontSize: 'var(--plasma-typo-h5-font-size)',
  fontStyle: 'var(--plasma-typo-h5-font-style)',
  fontWeight: 'var(--plasma-typo-h5-font-weight)',
  letterSpacing: 'var(--plasma-typo-h5-letter-spacing)',
  lineHeight: 'var(--plasma-typo-h5-line-height)'
};
var h5Bold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, h5), {}, {
  fontWeight: 'var(--plasma-typo-h5-bold-font-weight)'
});
var bodyL = {
  fontFamily: 'var(--plasma-typo-body-l-font-family)',
  fontSize: 'var(--plasma-typo-body-l-font-size)',
  fontStyle: 'var(--plasma-typo-body-l-font-style)',
  fontWeight: 'var(--plasma-typo-body-l-font-weight)',
  letterSpacing: 'var(--plasma-typo-body-l-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body-l-line-height)'
};
var bodyLBold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, bodyL), {}, {
  fontWeight: 'var(--plasma-typo-body-l-bold-font-weight)'
});
var bodyM = {
  fontFamily: 'var(--plasma-typo-body-m-font-family)',
  fontSize: 'var(--plasma-typo-body-m-font-size)',
  fontStyle: 'var(--plasma-typo-body-m-font-style)',
  fontWeight: 'var(--plasma-typo-body-m-font-weight)',
  letterSpacing: 'var(--plasma-typo-body-m-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body-m-line-height)'
};
var bodyMBold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, bodyM), {}, {
  fontWeight: 'var(--plasma-typo-body-m-bold-font-weight)'
});
var bodyS = {
  fontFamily: 'var(--plasma-typo-body-s-font-family)',
  fontSize: 'var(--plasma-typo-body-s-font-size)',
  fontStyle: 'var(--plasma-typo-body-s-font-style)',
  fontWeight: 'var(--plasma-typo-body-s-font-weight)',
  letterSpacing: 'var(--plasma-typo-body-s-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body-s-line-height)'
};
var bodySBold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, bodyS), {}, {
  fontWeight: 'var(--plasma-typo-body-s-bold-font-weight)'
});
var bodyXS = {
  fontFamily: 'var(--plasma-typo-body-xs-font-family)',
  fontSize: 'var(--plasma-typo-body-xs-font-size)',
  fontStyle: 'var(--plasma-typo-body-xs-font-style)',
  fontWeight: 'var(--plasma-typo-body-xs-font-weight)',
  letterSpacing: 'var(--plasma-typo-body-xs-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body-xs-line-height)'
};
var bodyXSBold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, bodyXS), {}, {
  fontWeight: 'var(--plasma-typo-body-xs-bold-font-weight)'
});
var bodyXXS = {
  fontFamily: 'var(--plasma-typo-body-xxs-font-family)',
  fontSize: 'var(--plasma-typo-body-xxs-font-size)',
  fontStyle: 'var(--plasma-typo-body-xxs-font-style)',
  fontWeight: 'var(--plasma-typo-body-xxs-font-weight)',
  letterSpacing: 'var(--plasma-typo-body-xxs-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body-xxs-line-height)'
};
var bodyXXSBold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, bodyXXS), {}, {
  fontWeight: 'var(--plasma-typo-body-xxs-bold-font-weight)'
});
var textL = {
  fontFamily: 'var(--plasma-typo-text-l-font-family)',
  fontSize: 'var(--plasma-typo-text-l-font-size)',
  fontStyle: 'var(--plasma-typo-text-l-font-style)',
  fontWeight: 'var(--plasma-typo-text-l-font-weight)',
  letterSpacing: 'var(--plasma-typo-text-l-letter-spacing)',
  lineHeight: 'var(--plasma-typo-text-l-line-height)'
};
var textLBold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, textL), {}, {
  fontWeight: 'var(--plasma-typo-text-l-bold-font-weight)'
});
var textM = {
  fontFamily: 'var(--plasma-typo-text-m-font-family)',
  fontSize: 'var(--plasma-typo-text-m-font-size)',
  fontStyle: 'var(--plasma-typo-text-m-font-style)',
  fontWeight: 'var(--plasma-typo-text-m-font-weight)',
  letterSpacing: 'var(--plasma-typo-text-m-letter-spacing)',
  lineHeight: 'var(--plasma-typo-text-m-line-height)'
};
var textMBold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, textM), {}, {
  fontWeight: 'var(--plasma-typo-text-m-bold-font-weight)'
});
var textS = {
  fontFamily: 'var(--plasma-typo-text-s-font-family)',
  fontSize: 'var(--plasma-typo-text-s-font-size)',
  fontStyle: 'var(--plasma-typo-text-s-font-style)',
  fontWeight: 'var(--plasma-typo-text-s-font-weight)',
  letterSpacing: 'var(--plasma-typo-text-s-letter-spacing)',
  lineHeight: 'var(--plasma-typo-text-s-line-height)'
};
var textSBold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, textS), {}, {
  fontWeight: 'var(--plasma-typo-text-s-bold-font-weight)'
});
var textXS = {
  fontFamily: 'var(--plasma-typo-text-xs-font-family)',
  fontSize: 'var(--plasma-typo-text-xs-font-size)',
  fontStyle: 'var(--plasma-typo-text-xs-font-style)',
  fontWeight: 'var(--plasma-typo-text-xs-font-weight)',
  letterSpacing: 'var(--plasma-typo-text-xs-letter-spacing)',
  lineHeight: 'var(--plasma-typo-text-xs-line-height)'
};
var textXSBold = /*#__PURE__*/_objectSpread2( /*#__PURE__*/_objectSpread2({}, textXS), {}, {
  fontWeight: 'var(--plasma-typo-text-xs-bold-font-weight)'
});
var body1 = {
  fontFamily: 'var(--plasma-typo-body1-font-family)',
  fontWeight: 'var(--plasma-typo-body1-font-weight)',
  fontStyle: 'var(--plasma-typo-body1-font-style)',
  fontSize: 'var(--plasma-typo-body1-font-size)',
  letterSpacing: 'var(--plasma-typo-body1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body1-line-height)'
};
var body2 = {
  fontFamily: 'var(--plasma-typo-body2-font-family)',
  fontWeight: 'var(--plasma-typo-body2-font-weight)',
  fontStyle: 'var(--plasma-typo-body2-font-style)',
  fontSize: 'var(--plasma-typo-body2-font-size)',
  letterSpacing: 'var(--plasma-typo-body2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body2-line-height)'
};
var body3 = {
  fontFamily: 'var(--plasma-typo-body3-font-family)',
  fontWeight: 'var(--plasma-typo-body3-font-weight)',
  fontStyle: 'var(--plasma-typo-body3-font-style)',
  fontSize: 'var(--plasma-typo-body3-font-size)',
  letterSpacing: 'var(--plasma-typo-body3-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body3-line-height)'
};
var button1 = {
  fontFamily: 'var(--plasma-typo-button1-font-family)',
  fontWeight: 'var(--plasma-typo-button1-font-weight)',
  fontStyle: 'var(--plasma-typo-button1-font-style)',
  fontSize: 'var(--plasma-typo-button1-font-size)',
  letterSpacing: 'var(--plasma-typo-button1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-button1-line-height)'
};
var button2 = {
  fontFamily: 'var(--plasma-typo-button2-font-family)',
  fontWeight: 'var(--plasma-typo-button2-font-weight)',
  fontStyle: 'var(--plasma-typo-button2-font-style)',
  fontSize: 'var(--plasma-typo-button2-font-size)',
  letterSpacing: 'var(--plasma-typo-button2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-button2-line-height)'
};
var caption = {
  fontFamily: 'var(--plasma-typo-caption-font-family)',
  fontWeight: 'var(--plasma-typo-caption-font-weight)',
  fontStyle: 'var(--plasma-typo-caption-font-style)',
  fontSize: 'var(--plasma-typo-caption-font-size)',
  letterSpacing: 'var(--plasma-typo-caption-letter-spacing)',
  lineHeight: 'var(--plasma-typo-caption-line-height)'
};
var display1 = {
  fontFamily: 'var(--plasma-typo-display1-font-family)',
  fontWeight: 'var(--plasma-typo-display1-font-weight)',
  fontStyle: 'var(--plasma-typo-display1-font-style)',
  fontSize: 'var(--plasma-typo-display1-font-size)',
  letterSpacing: 'var(--plasma-typo-display1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-display1-line-height)'
};
var display2 = {
  fontFamily: 'var(--plasma-typo-display2-font-family)',
  fontWeight: 'var(--plasma-typo-display2-font-weight)',
  fontStyle: 'var(--plasma-typo-display2-font-style)',
  fontSize: 'var(--plasma-typo-display2-font-size)',
  letterSpacing: 'var(--plasma-typo-display2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-display2-line-height)'
};
var display3 = {
  fontFamily: 'var(--plasma-typo-display3-font-family)',
  fontWeight: 'var(--plasma-typo-display3-font-weight)',
  fontStyle: 'var(--plasma-typo-display3-font-style)',
  fontSize: 'var(--plasma-typo-display3-font-size)',
  letterSpacing: 'var(--plasma-typo-display3-letter-spacing)',
  lineHeight: 'var(--plasma-typo-display3-line-height)'
};
var footnote1 = {
  fontFamily: 'var(--plasma-typo-footnote1-font-family)',
  fontWeight: 'var(--plasma-typo-footnote1-font-weight)',
  fontStyle: 'var(--plasma-typo-footnote1-font-style)',
  fontSize: 'var(--plasma-typo-footnote1-font-size)',
  letterSpacing: 'var(--plasma-typo-footnote1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-footnote1-line-height)'
};
var footnote2 = {
  fontFamily: 'var(--plasma-typo-footnote2-font-family)',
  fontWeight: 'var(--plasma-typo-footnote2-font-weight)',
  fontStyle: 'var(--plasma-typo-footnote2-font-style)',
  fontSize: 'var(--plasma-typo-footnote2-font-size)',
  letterSpacing: 'var(--plasma-typo-footnote2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-footnote2-line-height)'
};
var headline1 = {
  fontFamily: 'var(--plasma-typo-headline1-font-family)',
  fontWeight: 'var(--plasma-typo-headline1-font-weight)',
  fontStyle: 'var(--plasma-typo-headline1-font-style)',
  fontSize: 'var(--plasma-typo-headline1-font-size)',
  letterSpacing: 'var(--plasma-typo-headline1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-headline1-line-height)'
};
var headline2 = {
  fontFamily: 'var(--plasma-typo-headline2-font-family)',
  fontWeight: 'var(--plasma-typo-headline2-font-weight)',
  fontStyle: 'var(--plasma-typo-headline2-font-style)',
  fontSize: 'var(--plasma-typo-headline2-font-size)',
  letterSpacing: 'var(--plasma-typo-headline2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-headline2-line-height)'
};
var headline3 = {
  fontFamily: 'var(--plasma-typo-headline3-font-family)',
  fontWeight: 'var(--plasma-typo-headline3-font-weight)',
  fontStyle: 'var(--plasma-typo-headline3-font-style)',
  fontSize: 'var(--plasma-typo-headline3-font-size)',
  letterSpacing: 'var(--plasma-typo-headline3-letter-spacing)',
  lineHeight: 'var(--plasma-typo-headline3-line-height)'
};
var headline4 = {
  fontFamily: 'var(--plasma-typo-headline4-font-family)',
  fontWeight: 'var(--plasma-typo-headline4-font-weight)',
  fontStyle: 'var(--plasma-typo-headline4-font-style)',
  fontSize: 'var(--plasma-typo-headline4-font-size)',
  letterSpacing: 'var(--plasma-typo-headline4-letter-spacing)',
  lineHeight: 'var(--plasma-typo-headline4-line-height)'
};
var paragraph1 = {
  fontFamily: 'var(--plasma-typo-paragraph1-font-family)',
  fontWeight: 'var(--plasma-typo-paragraph1-font-weight)',
  fontStyle: 'var(--plasma-typo-paragraph1-font-style)',
  fontSize: 'var(--plasma-typo-paragraph1-font-size)',
  letterSpacing: 'var(--plasma-typo-paragraph1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-paragraph1-line-height)'
};
var paragraph2 = {
  fontFamily: 'var(--plasma-typo-paragraph2-font-family)',
  fontWeight: 'var(--plasma-typo-paragraph2-font-weight)',
  fontStyle: 'var(--plasma-typo-paragraph2-font-style)',
  fontSize: 'var(--plasma-typo-paragraph2-font-size)',
  letterSpacing: 'var(--plasma-typo-paragraph2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-paragraph2-line-height)'
};
var underline = {
  fontFamily: 'var(--plasma-typo-underline-font-family)',
  fontWeight: 'var(--plasma-typo-underline-font-weight)',
  fontStyle: 'var(--plasma-typo-underline-font-style)',
  fontSize: 'var(--plasma-typo-underline-font-size)',
  letterSpacing: 'var(--plasma-typo-underline-letter-spacing)',
  lineHeight: 'var(--plasma-typo-underline-line-height)'
};

export { body1, body2, body3, bodyL, bodyLBold, bodyM, bodyMBold, bodyS, bodySBold, bodyXS, bodyXSBold, bodyXXS, bodyXXSBold, button1, button2, caption, display1, display2, display3, dsplL, dsplLBold, dsplM, dsplMBold, dsplS, dsplSBold, footnote1, footnote2, h1, h1Bold, h2, h2Bold, h3, h3Bold, h4, h4Bold, h5, h5Bold, headline1, headline2, headline3, headline4, paragraph1, paragraph2, textL, textLBold, textM, textMBold, textS, textSBold, textXS, textXSBold, underline };
