// @salutejs/plasma-web/es/components/Divider/Divider.config.js
var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5, _templateObject6, _templateObject7;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css, dividerTokens } from '@salutejs/plasma-new-hope/styled-components';
export var config = {
  defaults: {
    view: 'default',
    size: 'm',
    orientation: 'horizontal'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/css(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--surface-transparent-tertiary);\n            "])), dividerTokens.background),
      dark: /*#__PURE__*/css(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--on-dark-surface-transparent-tertiary);\n            "])), dividerTokens.background),
      light: /*#__PURE__*/css(_templateObject3 || (_templateObject3 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--on-light-surface-transparent-tertiary);\n            "])), dividerTokens.background),
      // TODO поменять на корректный токен --inverse-surface-transparent-tertiary когда он появится
      inverse: /*#__PURE__*/css(_templateObject4 || (_templateObject4 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--inverse-text-tertiary);\n            "])), dividerTokens.background)
    },
    size: {
      m: /*#__PURE__*/css(_templateObject5 || (_templateObject5 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": 0.0625rem;\n            "])), dividerTokens.borderRadius)
    },
    orientation: {
      horizontal: /*#__PURE__*/css(_templateObject6 || (_templateObject6 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": 0.0625rem;\n            "])), dividerTokens.baseSideSize),
      vertical: /*#__PURE__*/css(_templateObject7 || (_templateObject7 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": 0.0625rem;\n            "])), dividerTokens.baseSideSize)
    }
  }
};
