// @salutejs/plasma-hope/components/Modal/ModalView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalView = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _Button = /*#__PURE__*/require("../Button");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _IconClose;
var _excluded = ["role", "closeButtonAriaLabel", "children", "onClose", "showCloseButton"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('Modal/ModalView');
var StyledWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web-core__sc-92wujm-0"
})(["display:flex;align-items:center;height:100%;"]);
var StyledBody = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web-core__sc-92wujm-1"
})(["position:relative;width:25rem;max-width:100%;border-radius:1.25rem;background-color:", ";pointer-events:all;"], _plasmaCore.backgroundPrimary);
var StyledContent = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web-core__sc-92wujm-2"
})(["padding:2rem;"]);
var StyledButtonClose = /*#__PURE__*/(0, _styledComponents["default"])((0, _Button.Button)({
  design: 'web'
})).attrs(function () {
  return {
    view: 'clear'
  };
}).withConfig({
  componentId: "plasma-web-core__sc-92wujm-3"
})(["position:absolute;top:0;right:0;margin:2rem;width:auto;height:auto;padding:0;"]);

/**
 * Визуальная часть модального окна.
 */
var ModalView = exports.ModalView = /*#__PURE__*/_react["default"].forwardRef(function (_ref, ref) {
  var _ref$role = _ref.role,
    role = _ref$role === void 0 ? 'dialog' : _ref$role,
    closeButtonAriaLabel = _ref.closeButtonAriaLabel,
    children = _ref.children,
    onClose = _ref.onClose,
    _ref$showCloseButton = _ref.showCloseButton,
    showCloseButton = _ref$showCloseButton === void 0 ? true : _ref$showCloseButton,
    rest = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/_react["default"].createElement(StyledWrapper, null, /*#__PURE__*/_react["default"].createElement(StyledBody, _extends({}, rest, {
    ref: ref,
    role: role,
    "aria-modal": "true"
  }), /*#__PURE__*/_react["default"].createElement(StyledContent, null, children), showCloseButton && /*#__PURE__*/_react["default"].createElement(StyledButtonClose, {
    "aria-label": closeButtonAriaLabel,
    onClick: onClose,
    contentLeft: _IconClose || (_IconClose = /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconClose, {
      size: "s",
      color: "inherit"
    }))
  })));
});
