// @salutejs/plasma-hope/components/Modal/Modal.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Modal = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _reactDom = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react-dom"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _plasmaTokensWeb = /*#__PURE__*/require("@salutejs/plasma-tokens-web");
var _hooks = /*#__PURE__*/require("../../hooks");
var _ModalsContext = /*#__PURE__*/require("./ModalsContext");
var _ModalView = /*#__PURE__*/require("./ModalView");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _NoScroll;
var _excluded = ["id", "isOpen", "onClose", "onOverlayClick", "onEscKeyDown", "closeOnEsc", "closeOnOverlayClick", "withBlur", "initialFocusRef", "focusAfterRef"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('Modal/Modal');
var ESCAPE_KEYCODE = 27;
var wrapperHideAnimation = /*#__PURE__*/(0, _styledComponents.keyframes)(["from{opacity:1;}to{opacity:0;}"]);
var bodyShowAnimation = /*#__PURE__*/(0, _styledComponents.keyframes)(["from{opacity:0;transform:translateY(-100%);}to{opacity:1;transform:translateY(0);}"]);
var bodyHideAnimation = /*#__PURE__*/(0, _styledComponents.keyframes)(["from{opacity:1;transform:translateY(0);}to{opacity:0;transform:translateY(-100%);}"]);
var StyledWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web-core__sc-t11oy9-0"
})(["position:fixed;top:0;left:0;right:0;bottom:0;z-index:1;display:flex;align-items:center;justify-content:center;", ""], function (_ref) {
  var isHiding = _ref.isHiding;
  return isHiding && (0, _styledComponents.css)(["animation:0.1s ", " ease-out;"], wrapperHideAnimation);
});
var StyledOverlay = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web-core__sc-t11oy9-1"
})(["position:absolute;", ";width:100%;height:100%;background-color:", ";backdrop-filter:var(--backdrop-filter);cursor:", ";"], function (_ref2) {
  var $withBlur = _ref2.$withBlur;
  return (0, _styledComponents.css)(["--background-color:", ";--backdrop-filter:", ";"], $withBlur ? _plasmaTokensWeb.darkOverlayBlur : _plasmaTokensWeb.overlaySoft, $withBlur ? 'blur(1rem)' : 'none');
}, function (_ref3) {
  var transparent = _ref3.transparent;
  return transparent ? 'transparent' : 'var(--background-color)';
}, function (_ref4) {
  var clickable = _ref4.clickable;
  return clickable ? 'pointer' : 'default';
});
var StyledModal = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web-core__sc-t11oy9-2"
})(["position:absolute;height:100%;pointer-events:none;", ""], function (_ref5) {
  var isHiding = _ref5.isHiding;
  return (0, _styledComponents.css)(["animation:0.4s ", " ease-out;"], isHiding ? bodyHideAnimation : bodyShowAnimation);
});
var NoScroll = /*#__PURE__*/(0, _styledComponents.createGlobalStyle)(["body{overflow-y:hidden;}"]);

/**
 * Модальное окно.
 * Управляет показом/скрытием, подложкой и анимацией визуальной части модального окна.
 */
var Modal = exports.Modal = function Modal(_ref6) {
  var id = _ref6.id,
    isOpen = _ref6.isOpen,
    onClose = _ref6.onClose,
    onOverlayClick = _ref6.onOverlayClick,
    onEscKeyDown = _ref6.onEscKeyDown,
    _ref6$closeOnEsc = _ref6.closeOnEsc,
    closeOnEsc = _ref6$closeOnEsc === void 0 ? true : _ref6$closeOnEsc,
    _ref6$closeOnOverlayC = _ref6.closeOnOverlayClick,
    closeOnOverlayClick = _ref6$closeOnOverlayC === void 0 ? true : _ref6$closeOnOverlayC,
    withBlur = _ref6.withBlur,
    initialFocusRef = _ref6.initialFocusRef,
    focusAfterRef = _ref6.focusAfterRef,
    rest = _objectWithoutProperties(_ref6, _excluded);
  var uniqId = (0, _plasmaCore.useUniqId)();
  var innerId = id || uniqId;
  var wrapperRef = (0, _react.useRef)(null);
  var portalRef = (0, _react.useRef)(null);
  var trapRef = (0, _hooks.useFocusTrap)(true, initialFocusRef, focusAfterRef);
  var controller = (0, _react.useContext)(_ModalsContext.ModalsContext);
  var _useState = (0, _react.useState)(false),
    _useState2 = _slicedToArray(_useState, 2),
    forceRender = _useState2[1];
  var onOverlayKeyDown = (0, _react.useCallback)(function (event) {
    if (!closeOnOverlayClick) {
      return;
    }
    if (onOverlayClick) {
      onOverlayClick(event);
      return;
    }
    onClose === null || onClose === void 0 || onClose();
  }, [closeOnOverlayClick, onOverlayClick, onClose]);
  (0, _react.useEffect)(function () {
    var portal = document.getElementById(_ModalsContext.MODALS_PORTAL_ID);
    if (!portal) {
      portal = document.createElement('div');
      portal.setAttribute('id', _ModalsContext.MODALS_PORTAL_ID);
      portal.setAttribute('aria-live', 'off');
      portal.setAttribute('role', 'presentation');
      portal.style.position = 'relative';
      portal.style.zIndex = '9000';
      document.body.appendChild(portal);
    }
    portalRef.current = portal;

    /**
     * Изменение стейта нужно для того, чтобы модальное окно
     * отобразилось после записи DOM элемента в portalRef.current
     */
    forceRender(true);
    return function () {
      controller.unregister(innerId);
    };
  }, [controller, innerId]);
  (0, _react.useEffect)(function () {
    var onKeyDown = function onKeyDown(event) {
      if (closeOnEsc && event.keyCode === ESCAPE_KEYCODE && wrapperRef.current && portalRef.current && portalRef.current.contains(wrapperRef.current) && portalRef.current.children[portalRef.current.children.length - 1] === wrapperRef.current) {
        if (onEscKeyDown) {
          onEscKeyDown(event);
          return;
        }
        onClose === null || onClose === void 0 || onClose();
      }
    };
    window.addEventListener('keydown', onKeyDown);
    return function () {
      window.removeEventListener('keydown', onKeyDown);
    };
  }, [onClose, onEscKeyDown, closeOnEsc]);
  if (isOpen) {
    controller.register(innerId);
  } else {
    controller.unregister(innerId);
    return null;
  }
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, _NoScroll || (_NoScroll = /*#__PURE__*/_react["default"].createElement(NoScroll, null)), portalRef.current && /*#__PURE__*/_reactDom["default"].createPortal( /*#__PURE__*/_react["default"].createElement(StyledWrapper, {
    ref: wrapperRef
  }, /*#__PURE__*/_react["default"].createElement(StyledOverlay, {
    transparent: controller.items.indexOf(innerId) !== 0,
    clickable: closeOnOverlayClick,
    onClick: onOverlayKeyDown,
    $withBlur: withBlur
  }), /*#__PURE__*/_react["default"].createElement(StyledModal, null, /*#__PURE__*/_react["default"].createElement(_ModalView.ModalView, _extends({
    onClose: onClose,
    ref: trapRef
  }, rest)))), portalRef.current));
};
