// @salutejs/plasma-web/es/components/Typography/Old/Body.config.js
var _templateObject, _templateObject2, _templateObject3;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css, typographyTokens } from '@salutejs/plasma-new-hope/styled-components';
export var configBody1 = {
  defaults: {
    size: 'body1'
  },
  variations: {
    size: {
      body1: /*#__PURE__*/css(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-body1-font-family);\n                ", ": var(--plasma-typo-body1-font-size);\n                ", ": var(--plasma-typo-body1-font-style);\n                ", ": var(--plasma-typo-body1-font-weight);\n                ", ": var(--plasma-typo-body1-letter-spacing);\n                ", ": var(--plasma-typo-body1-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
export var configBody2 = {
  defaults: {
    size: 'body2'
  },
  variations: {
    size: {
      body2: /*#__PURE__*/css(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-body2-font-family);\n                ", ": var(--plasma-typo-body2-font-size);\n                ", ": var(--plasma-typo-body2-font-style);\n                ", ": var(--plasma-typo-body2-font-weight);\n                ", ": var(--plasma-typo-body2-letter-spacing);\n                ", ": var(--plasma-typo-body2-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
export var configBody3 = {
  defaults: {
    size: 'body3'
  },
  variations: {
    size: {
      body3: /*#__PURE__*/css(_templateObject3 || (_templateObject3 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-body3-font-family);\n                ", ": var(--plasma-typo-body3-font-size);\n                ", ": var(--plasma-typo-body3-font-style);\n                ", ": var(--plasma-typo-body3-font-weight);\n                ", ": var(--plasma-typo-body3-letter-spacing);\n                ", ": var(--plasma-typo-body3-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
