// @salutejs/plasma-web/es/components/Typography/Old/Button.config.js
var _templateObject, _templateObject2;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css, typographyTokens } from '@salutejs/plasma-new-hope/styled-components';
export var configButton1 = {
  defaults: {
    size: 'button1'
  },
  variations: {
    size: {
      button1: /*#__PURE__*/css(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-button1-font-family);\n                ", ": var(--plasma-typo-button1-font-size);\n                ", ": var(--plasma-typo-button1-font-style);\n                ", ": var(--plasma-typo-button1-font-weight);\n                ", ": var(--plasma-typo-button1-letter-spacing);\n                ", ": var(--plasma-typo-button1-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
export var configButton2 = {
  defaults: {
    size: 'button2'
  },
  variations: {
    size: {
      button2: /*#__PURE__*/css(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-button2-font-family);\n                ", ": var(--plasma-typo-button2-font-size);\n                ", ": var(--plasma-typo-button2-font-style);\n                ", ": var(--plasma-typo-button2-font-weight);\n                ", ": var(--plasma-typo-button2-letter-spacing);\n                ", ": var(--plasma-typo-button2-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
