// @salutejs/plasma-web/es/components/Typography/Old/Display.config.js
var _templateObject, _templateObject2, _templateObject3;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css, typographyTokens } from '@salutejs/plasma-new-hope/styled-components';
export var configDisplay1 = {
  defaults: {
    size: 'display1'
  },
  variations: {
    size: {
      display1: /*#__PURE__*/css(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-display1-font-family);\n                ", ": var(--plasma-typo-display1-font-size);\n                ", ": var(--plasma-typo-display1-font-style);\n                ", ": var(--plasma-typo-display1-font-weight);\n                ", ": var(--plasma-typo-display1-letter-spacing);\n                ", ": var(--plasma-typo-display1-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
export var configDisplay2 = {
  defaults: {
    size: 'display2'
  },
  variations: {
    size: {
      display2: /*#__PURE__*/css(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-display2-font-family);\n                ", ": var(--plasma-typo-display2-font-size);\n                ", ": var(--plasma-typo-display2-font-style);\n                ", ": var(--plasma-typo-display2-font-weight);\n                ", ": var(--plasma-typo-display2-letter-spacing);\n                ", ": var(--plasma-typo-display2-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
export var configDisplay3 = {
  defaults: {
    size: 'display3'
  },
  variations: {
    size: {
      display3: /*#__PURE__*/css(_templateObject3 || (_templateObject3 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-display3-font-family);\n                ", ": var(--plasma-typo-display3-font-size);\n                ", ": var(--plasma-typo-display3-font-style);\n                ", ": var(--plasma-typo-display3-font-weight);\n                ", ": var(--plasma-typo-display3-letter-spacing);\n                ", ": var(--plasma-typo-display3-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
