// @salutejs/plasma-web/es/components/Typography/Old/Headline.config.js
var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css, typographyTokens } from '@salutejs/plasma-new-hope/styled-components';
export var configHeadline1 = {
  defaults: {
    size: 'headline1'
  },
  variations: {
    size: {
      headline1: /*#__PURE__*/css(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-headline1-font-family);\n                ", ": var(--plasma-typo-headline1-font-size);\n                ", ": var(--plasma-typo-headline1-font-style);\n                ", ": var(--plasma-typo-headline1-font-weight);\n                ", ": var(--plasma-typo-headline1-letter-spacing);\n                ", ": var(--plasma-typo-headline1-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
export var configHeadline2 = {
  defaults: {
    size: 'headline2'
  },
  variations: {
    size: {
      headline2: /*#__PURE__*/css(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-headline2-font-family);\n                ", ": var(--plasma-typo-headline2-font-size);\n                ", ": var(--plasma-typo-headline2-font-style);\n                ", ": var(--plasma-typo-headline2-font-weight);\n                ", ": var(--plasma-typo-headline2-letter-spacing);\n                ", ": var(--plasma-typo-headline2-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
export var configHeadline3 = {
  defaults: {
    size: 'headline3'
  },
  variations: {
    size: {
      headline3: /*#__PURE__*/css(_templateObject3 || (_templateObject3 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-headline3-font-family);\n                ", ": var(--plasma-typo-headline3-font-size);\n                ", ": var(--plasma-typo-headline3-font-style);\n                ", ": var(--plasma-typo-headline3-font-weight);\n                ", ": var(--plasma-typo-headline3-letter-spacing);\n                ", ": var(--plasma-typo-headline3-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
export var configHeadline4 = {
  defaults: {
    size: 'headline4'
  },
  variations: {
    size: {
      headline4: /*#__PURE__*/css(_templateObject4 || (_templateObject4 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-headline4-font-family);\n                ", ": var(--plasma-typo-headline4-font-size);\n                ", ": var(--plasma-typo-headline4-font-style);\n                ", ": var(--plasma-typo-headline4-font-weight);\n                ", ": var(--plasma-typo-headline4-letter-spacing);\n                ", ": var(--plasma-typo-headline4-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
export var configHeadline5 = {
  defaults: {
    size: 'headline5'
  },
  variations: {
    size: {
      headline5: /*#__PURE__*/css(_templateObject5 || (_templateObject5 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-headline5-font-family, 'SB Sans Text',sans-serif);\n                ", ": var(--plasma-typo-headline5-font-size, 1rem);\n                ", ": var(--plasma-typo-headline5-font-style, normal);\n                ", ": var(--plasma-typo-headline5-font-weight, 600);\n                ", ": var(--plasma-typo-headline5-letter-spacing, -0.0190em);\n                ", ": var(--plasma-typo-headline5-line-height, 1.25rem);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
