// @salutejs/plasma-web/es/components/Typography/Old/index.js
import { bodyConfig, buttonTypographyConfig, captionConfig as captionDefaultConfig, displayConfig, footnoteConfig, headlineConfig, paragraphConfig, paragraphTextConfig, underlineConfig as underlineDefaultConfig, subtitleConfig as subtitleDefaultConfig, component, mergeConfig } from '@salutejs/plasma-new-hope/styled-components';
import { configBody1, configBody2, configBody3 } from './Body.config';
import { configButton1, configButton2 } from './Button.config';
import { config as captionConfigCustom } from './Caption.config';
import { configDisplay1, configDisplay2, configDisplay3 } from './Display.config';
import { configFootnote1, configFootnote2 } from './Footnote.config';
import { configHeadline1, configHeadline2, configHeadline3, configHeadline4, configHeadline5 } from './Headline.config';
import { configParagraph1, configParagraph2 } from './Paragraph.config';
import { configParagraphText1, configParagraphText2 } from './ParagraphText.config';
import { config as underlineConfigCustom } from './Underline.config';
import { config as subtitleConfigCustom } from './Subtitle.config';
var body1Config = /*#__PURE__*/mergeConfig(bodyConfig, configBody1);
var Body1Component = /*#__PURE__*/component(body1Config);
var body2Config = /*#__PURE__*/mergeConfig(bodyConfig, configBody2);
var Body2Component = /*#__PURE__*/component(body2Config);
var body3Config = /*#__PURE__*/mergeConfig(bodyConfig, configBody3);
var Body3Component = /*#__PURE__*/component(body3Config);
var button1Config = /*#__PURE__*/mergeConfig(buttonTypographyConfig, configButton1);
var Button1Component = /*#__PURE__*/component(button1Config);
var button2Config = /*#__PURE__*/mergeConfig(buttonTypographyConfig, configButton2);
var Button2Component = /*#__PURE__*/component(button2Config);
var captionConfig = /*#__PURE__*/mergeConfig(captionDefaultConfig, captionConfigCustom);
var CaptionComponent = /*#__PURE__*/component(captionConfig);
var display1Config = /*#__PURE__*/mergeConfig(displayConfig, configDisplay1);
var Display1Component = /*#__PURE__*/component(display1Config);
var display2Config = /*#__PURE__*/mergeConfig(displayConfig, configDisplay2);
var Display2Component = /*#__PURE__*/component(display2Config);
var display3Config = /*#__PURE__*/mergeConfig(displayConfig, configDisplay3);
var Display3Component = /*#__PURE__*/component(display3Config);
var footnote1Config = /*#__PURE__*/mergeConfig(footnoteConfig, configFootnote1);
var Footnote1Component = /*#__PURE__*/component(footnote1Config);
var footnote2Config = /*#__PURE__*/mergeConfig(footnoteConfig, configFootnote2);
var Footnote2Component = /*#__PURE__*/component(footnote2Config);
var headline1Config = /*#__PURE__*/mergeConfig(headlineConfig, configHeadline1);
var Headline1Component = /*#__PURE__*/component(headline1Config);
var headline2Config = /*#__PURE__*/mergeConfig(headlineConfig, configHeadline2);
var Headline2Component = /*#__PURE__*/component(headline2Config);
var headline3Config = /*#__PURE__*/mergeConfig(headlineConfig, configHeadline3);
var Headline3Component = /*#__PURE__*/component(headline3Config);
var headline4Config = /*#__PURE__*/mergeConfig(headlineConfig, configHeadline4);
var Headline4Component = /*#__PURE__*/component(headline4Config);
var headline5Config = /*#__PURE__*/mergeConfig(headlineConfig, configHeadline5);
var Headline5Component = /*#__PURE__*/component(headline5Config);
var underlineConfig = /*#__PURE__*/mergeConfig(underlineDefaultConfig, underlineConfigCustom);
var UnderlineComponent = /*#__PURE__*/component(underlineConfig);
var subtitleConfig = /*#__PURE__*/mergeConfig(subtitleDefaultConfig, subtitleConfigCustom);
var SubtitleComponent = /*#__PURE__*/component(subtitleConfig);
var paragraph1Config = /*#__PURE__*/mergeConfig(paragraphConfig, configParagraph1);
var Paragraph1Component = /*#__PURE__*/component(paragraph1Config);
var paragraph2Config = /*#__PURE__*/mergeConfig(paragraphConfig, configParagraph2);
var Paragraph2Component = /*#__PURE__*/component(paragraph2Config);
var paragraphText1Config = /*#__PURE__*/mergeConfig(paragraphTextConfig, configParagraphText1);
var ParagraphText1Component = /*#__PURE__*/component(paragraphText1Config);
var paragraphText2Config = /*#__PURE__*/mergeConfig(paragraphTextConfig, configParagraphText2);
var ParagraphText2Component = /*#__PURE__*/component(paragraphText2Config);
export var Body1 = Body1Component;
export var Body2 = Body2Component;
export var Body3 = Body3Component;
export var Button1 = Button1Component;
export var Button2 = Button2Component;
export var Caption = CaptionComponent;
export var Display1 = Display1Component;
export var Display2 = Display2Component;
export var Display3 = Display3Component;
export var Footnote1 = Footnote1Component;
export var Footnote2 = Footnote2Component;
export var Headline1 = Headline1Component;
export var Headline2 = Headline2Component;
export var Headline3 = Headline3Component;
export var Headline4 = Headline4Component;
export var Headline5 = Headline5Component;
export var P1 = Paragraph1Component;
export var P2 = Paragraph2Component;
export var ParagraphText1 = ParagraphText1Component;
export var ParagraphText2 = ParagraphText2Component;
export var Underline = UnderlineComponent;
export var Subtitle = SubtitleComponent;
