// @salutejs/plasma-web/es/components/Typography/Old/Paragraph.config.js
var _templateObject, _templateObject2;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css, typographyTokens } from '@salutejs/plasma-new-hope/styled-components';
export var configParagraph1 = {
  defaults: {
    size: 'paragraph1'
  },
  variations: {
    size: {
      paragraph1: /*#__PURE__*/css(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-paragraph1-font-family);\n                ", ": var(--plasma-typo-paragraph1-font-size);\n                ", ": var(--plasma-typo-paragraph1-font-style);\n                ", ": var(--plasma-typo-paragraph1-font-weight);\n                ", ": var(--plasma-typo-paragraph1-letter-spacing);\n                ", ": var(--plasma-typo-paragraph1-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
export var configParagraph2 = {
  defaults: {
    size: 'paragraph2'
  },
  variations: {
    size: {
      paragraph2: /*#__PURE__*/css(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-paragraph2-font-family);\n                ", ": var(--plasma-typo-paragraph2-font-size);\n                ", ": var(--plasma-typo-paragraph2-font-style);\n                ", ": var(--plasma-typo-paragraph2-font-weight);\n                ", ": var(--plasma-typo-paragraph2-letter-spacing);\n                ", ": var(--plasma-typo-paragraph2-line-height);\n            "])), typographyTokens.typoFontFamily, typographyTokens.typoFontSize, typographyTokens.typoFontStyle, typographyTokens.typoFontWeight, typographyTokens.typoFontLetterSpacing, typographyTokens.typoFontLineHeight)
    }
  }
};
