// @salutejs/plasma-new-hope/styled-components/es/components/Header/ui/HeaderTitleWrapper/HeaderTitleWrapper.styles.js
import styled from 'styled-components';
import { tokens } from '../../Header.tokens';
import { applyEllipsis, applyHyphens, bodyM, bodyS, h1, h3 } from '../../../../mixins';
export var Title = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-0"
})(["", ";margin:var(", ");color:var(", ");", ";", ";"], bodyM, tokens.headerTitleMargin, tokens.headerTitleColor, applyHyphens, /*#__PURE__*/applyEllipsis());
export var BigTitle = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-1"
})(["", ";font-weight:600;margin:var(", ");color:var(", ");", ";", ";"], h3, tokens.headerBigTitleMargin, tokens.headerBiggerTitleColor, applyHyphens, /*#__PURE__*/applyEllipsis());
export var BiggerTitle = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-2"
})(["", ";font-weight:600;margin:var(", ");color:var(", ");", ";", ";"], h1, tokens.headerBiggerTitleMargin, tokens.headerBiggerTitleColor, applyHyphens, /*#__PURE__*/applyEllipsis());
export var Label = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-3"
})(["margin:var(", ");color:var(", ");", ";"], tokens.headerLabelMargin, tokens.headerLabelColor, /*#__PURE__*/applyEllipsis());
export var SubTitle = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-4"
})(["", ";margin:var(", ");color:var(", ");", ";", ";"], bodyS, tokens.headerSubTitleMargin, tokens.headerSubTitleColor, applyHyphens, /*#__PURE__*/applyEllipsis());
export var Caption = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-5"
})(["margin:var(", ");color:var(", ");", ";"], tokens.headerCaptionMargin, tokens.headerCaptionColor, /*#__PURE__*/applyEllipsis());
export var TextBoxRoot = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1jzpuar-6"
})(["& >:first-child{margin-top:0;}& >:last-child{margin-bottom:0;}"]);
