// @salutejs/plasma-web/components/Tooltip/Tooltip.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.All = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _Button = /*#__PURE__*/require("../Button");
var _ = /*#__PURE__*/require(".");
var _StyledGrid;
/* eslint-disable */
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var StyledGrid = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-pnme77-0"
})(["display:grid;grid-template-columns:repeat(3,max-content);grid-gap:1rem 3.5rem;padding:2.5rem;"]);
var All = exports.All = function All() {
  return _StyledGrid || (_StyledGrid = /*#__PURE__*/_react["default"].createElement(StyledGrid, null, /*#__PURE__*/_react["default"].createElement(_.Tooltip, {
    target: /*#__PURE__*/_react["default"].createElement(_.Tooltip, {
      target: /*#__PURE__*/_react["default"].createElement(_Button.Button, {
        contentLeft: /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconDownload, null)
      }),
      placement: "left",
      isOpen: true,
      hasArrow: true,
      text: "Left"
    }),
    placement: "top-start",
    isOpen: true,
    hasArrow: true,
    text: "Top start"
  }), /*#__PURE__*/_react["default"].createElement(_.Tooltip, {
    target: /*#__PURE__*/_react["default"].createElement(_Button.Button, {
      contentLeft: /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconDownload, null)
    }),
    placement: "top",
    isOpen: true,
    hasArrow: true,
    text: "Top"
  }), /*#__PURE__*/_react["default"].createElement(_.Tooltip, {
    target: /*#__PURE__*/_react["default"].createElement(_.Tooltip, {
      target: /*#__PURE__*/_react["default"].createElement(_Button.Button, {
        contentLeft: /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconDownload, null)
      }),
      placement: "right",
      isOpen: true,
      hasArrow: true,
      text: "Right"
    }),
    placement: "top-end",
    isOpen: true,
    hasArrow: true,
    text: "Top end"
  }), /*#__PURE__*/_react["default"].createElement(_.Tooltip, {
    target: /*#__PURE__*/_react["default"].createElement(_Button.Button, {
      contentLeft: /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconDownload, null)
    }),
    placement: "bottom-start",
    isOpen: true,
    hasArrow: true,
    text: "Bottom start"
  }), /*#__PURE__*/_react["default"].createElement(_.Tooltip, {
    target: /*#__PURE__*/_react["default"].createElement(_Button.Button, {
      contentLeft: /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconDownload, null)
    }),
    placement: "bottom",
    isOpen: true,
    hasArrow: true,
    text: "Bottom"
  }), /*#__PURE__*/_react["default"].createElement(_.Tooltip, {
    target: /*#__PURE__*/_react["default"].createElement(_Button.Button, {
      contentLeft: /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconDownload, null)
    }),
    placement: "bottom-end",
    isOpen: true,
    hasArrow: true,
    text: "Bottom end"
  })));
};
