// @salutejs/plasma-web/es/components/Combobox/Combobox.js
import { comboboxConfig, comboboxItemConfig, comboboxDividerConfig, comboboxFooterConfig, comboboxGroupConfig, comboboxHeaderConfig, component, mergeConfig } from '@salutejs/plasma-new-hope/styled-components';
import { config } from './Combobox.config';
var mergedConfig = /*#__PURE__*/mergeConfig(comboboxConfig, config);
export var Combobox = /*#__PURE__*/component(mergedConfig);
var mergedItemConfig = /*#__PURE__*/mergeConfig(comboboxItemConfig);
export var ComboboxItem = /*#__PURE__*/component(mergedItemConfig);
var mergedDividerConfig = /*#__PURE__*/mergeConfig(comboboxDividerConfig);
export var ComboboxDivider = /*#__PURE__*/component(mergedDividerConfig);
var mergedFooterConfig = /*#__PURE__*/mergeConfig(comboboxFooterConfig);
export var ComboboxFooter = /*#__PURE__*/component(mergedFooterConfig);
var mergedGroupConfig = /*#__PURE__*/mergeConfig(comboboxGroupConfig);
export var ComboboxGroup = /*#__PURE__*/component(mergedGroupConfig);
var mergedHeaderConfig = /*#__PURE__*/mergeConfig(comboboxHeaderConfig);
export var ComboboxHeader = /*#__PURE__*/component(mergedHeaderConfig);
