// @salutejs/plasma-tokens/index.js
"use strict";
// Generated by robots, do not change this manually!
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.scalingPixelBasis = exports.mobileScale = exports.sberBoxScale = exports.sberPortalScale = exports.typographyValues = exports.typography = exports.colorValues = exports.colors = void 0;
var colors = __importStar(require("./colors"));
exports.colors = colors;
var colorValues = __importStar(require("./colors/values"));
exports.colorValues = colorValues;
var typography = __importStar(require("./typography"));
exports.typography = typography;
var typographyValues = __importStar(require("./typographyValues"));
exports.typographyValues = typographyValues;
exports.sberPortalScale = 2;
exports.sberBoxScale = 2;
exports.mobileScale = 1;
exports.scalingPixelBasis = 16;
__exportStar(require("./colors"), exports);
__exportStar(require("./typography"), exports);
__exportStar(require("./typographyValues"), exports);
__exportStar(require("./typo"), exports);
__exportStar(require("./themes"), exports);
