// @salutejs/plasma-tokens/brands/sberonline/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.onDarkTextWarningActive = exports.onDarkTextWarningHover = exports.onDarkTextPositive = exports.onDarkTextPositiveActive = exports.onDarkTextPositiveHover = exports.onDarkTextAccentGradient = exports.onDarkTextAccentGradientActive = exports.onDarkTextAccentGradientHover = exports.onDarkTextAccent = exports.onDarkTextAccentActive = exports.onDarkTextAccentHover = exports.onDarkTextParagraph = exports.onDarkTextParagraphActive = exports.onDarkTextParagraphHover = exports.onDarkTextTertiary = exports.onDarkTextTertiaryActive = exports.onDarkTextTertiaryHover = exports.onDarkTextSecondary = exports.onDarkTextSecondaryActive = exports.onDarkTextSecondaryHover = exports.onDarkTextPrimary = exports.onDarkTextPrimaryActive = exports.onDarkTextPrimaryHover = exports.textNegative = exports.textNegativeActive = exports.textNegativeHover = exports.textWarning = exports.textWarningActive = exports.textWarningHover = exports.textPositive = exports.textPositiveActive = exports.textPositiveHover = exports.textAccentGradient = exports.textAccentGradientActive = exports.textAccentGradientHover = exports.textAccent = exports.textAccentActive = exports.textAccentHover = exports.textParagraph = exports.textParagraphActive = exports.textParagraphHover = exports.textTertiary = exports.textTertiaryActive = exports.textTertiaryHover = exports.textSecondary = exports.textSecondaryActive = exports.textSecondaryHover = exports.textPrimary = exports.textPrimaryActive = exports.textPrimaryHover = void 0;
exports.inverseTextPositiveHover = exports.inverseTextAccentGradient = exports.inverseTextAccentGradientActive = exports.inverseTextAccentGradientHover = exports.inverseTextAccent = exports.inverseTextAccentActive = exports.inverseTextAccentHover = exports.inverseTextParagraph = exports.inverseTextParagraphActive = exports.inverseTextParagraphHover = exports.inverseTextTertiary = exports.inverseTextTertiaryActive = exports.inverseTextTertiaryHover = exports.inverseTextSecondary = exports.inverseTextSecondaryActive = exports.inverseTextSecondaryHover = exports.inverseTextPrimary = exports.inverseTextPrimaryActive = exports.inverseTextPrimaryHover = exports.onLightTextNegative = exports.onLightTextNegativeActive = exports.onLightTextNegativeHover = exports.onLightTextWarning = exports.onLightTextWarningActive = exports.onLightTextWarningHover = exports.onLightTextPositive = exports.onLightTextPositiveActive = exports.onLightTextPositiveHover = exports.onLightTextAccentGradient = exports.onLightTextAccentGradientActive = exports.onLightTextAccentGradientHover = exports.onLightTextAccent = exports.onLightTextAccentActive = exports.onLightTextAccentHover = exports.onLightTextParagraph = exports.onLightTextParagraphActive = exports.onLightTextParagraphHover = exports.onLightTextTertiary = exports.onLightTextTertiaryActive = exports.onLightTextTertiaryHover = exports.onLightTextSecondary = exports.onLightTextSecondaryActive = exports.onLightTextSecondaryHover = exports.onLightTextPrimary = exports.onLightTextPrimaryActive = exports.onLightTextPrimaryHover = exports.onDarkTextNegative = exports.onDarkTextNegativeActive = exports.onDarkTextNegativeHover = exports.onDarkTextWarning = void 0;
exports.surfacePositive = exports.surfacePositiveActive = exports.surfacePositiveHover = exports.surfaceAccentGradient = exports.surfaceAccentGradientActive = exports.surfaceAccentGradientHover = exports.surfaceAccent = exports.surfaceAccentActive = exports.surfaceAccentHover = exports.surfaceClear = exports.surfaceClearActive = exports.surfaceClearHover = exports.surfaceTransparentCard = exports.surfaceTransparentCardActive = exports.surfaceTransparentCardHover = exports.surfaceTransparentDeep = exports.surfaceTransparentDeepActive = exports.surfaceTransparentDeepHover = exports.surfaceTransparentTertiary = exports.surfaceTransparentTertiaryActive = exports.surfaceTransparentTertiaryHover = exports.surfaceTransparentSecondary = exports.surfaceTransparentSecondaryActive = exports.surfaceTransparentSecondaryHover = exports.surfaceTransparentPrimary = exports.surfaceTransparentPrimaryActive = exports.surfaceTransparentPrimaryHover = exports.surfaceSolidDefault = exports.surfaceSolidDefaultActive = exports.surfaceSolidDefaultHover = exports.surfaceSolidCard = exports.surfaceSolidCardActive = exports.surfaceSolidCardHover = exports.surfaceSolidTertiary = exports.surfaceSolidTertiaryActive = exports.surfaceSolidTertiaryHover = exports.surfaceSolidSecondary = exports.surfaceSolidSecondaryActive = exports.surfaceSolidSecondaryHover = exports.surfaceSolidPrimary = exports.surfaceSolidPrimaryActive = exports.surfaceSolidPrimaryHover = exports.inverseTextNegative = exports.inverseTextNegativeActive = exports.inverseTextNegativeHover = exports.inverseTextWarning = exports.inverseTextWarningActive = exports.inverseTextWarningHover = exports.inverseTextPositive = exports.inverseTextPositiveActive = void 0;
exports.onDarkSurfaceWarningActive = exports.onDarkSurfaceWarningHover = exports.onDarkSurfacePositive = exports.onDarkSurfacePositiveActive = exports.onDarkSurfacePositiveHover = exports.onDarkSurfaceAccentGradient = exports.onDarkSurfaceAccentGradientActive = exports.onDarkSurfaceAccentGradientHover = exports.onDarkSurfaceAccent = exports.onDarkSurfaceAccentActive = exports.onDarkSurfaceAccentHover = exports.onDarkSurfaceClear = exports.onDarkSurfaceClearActive = exports.onDarkSurfaceClearHover = exports.onDarkSurfaceTransparentCard = exports.onDarkSurfaceTransparentCardActive = exports.onDarkSurfaceTransparentCardHover = exports.onDarkSurfaceTransparentDeep = exports.onDarkSurfaceTransparentDeepActive = exports.onDarkSurfaceTransparentDeepHover = exports.onDarkSurfaceTransparentTertiary = exports.onDarkSurfaceTransparentTertiaryActive = exports.onDarkSurfaceTransparentTertiaryHover = exports.onDarkSurfaceTransparentSecondary = exports.onDarkSurfaceTransparentSecondaryActive = exports.onDarkSurfaceTransparentSecondaryHover = exports.onDarkSurfaceTransparentPrimary = exports.onDarkSurfaceTransparentPrimaryActive = exports.onDarkSurfaceTransparentPrimaryHover = exports.onDarkSurfaceSolidDefault = exports.onDarkSurfaceSolidDefaultActive = exports.onDarkSurfaceSolidDefaultHover = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidCardActive = exports.onDarkSurfaceSolidCardHover = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidTertiaryActive = exports.onDarkSurfaceSolidTertiaryHover = exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidSecondaryActive = exports.onDarkSurfaceSolidSecondaryHover = exports.onDarkSurfaceSolidPrimary = exports.onDarkSurfaceSolidPrimaryActive = exports.onDarkSurfaceSolidPrimaryHover = exports.surfaceNegative = exports.surfaceNegativeActive = exports.surfaceNegativeHover = exports.surfaceWarning = exports.surfaceWarningActive = exports.surfaceWarningHover = void 0;
exports.onLightSurfaceNegativeHover = exports.onLightSurfaceWarning = exports.onLightSurfaceWarningActive = exports.onLightSurfaceWarningHover = exports.onLightSurfacePositive = exports.onLightSurfacePositiveActive = exports.onLightSurfacePositiveHover = exports.onLightSurfaceAccentGradient = exports.onLightSurfaceAccentGradientActive = exports.onLightSurfaceAccentGradientHover = exports.onLightSurfaceAccent = exports.onLightSurfaceAccentActive = exports.onLightSurfaceAccentHover = exports.onLightSurfaceClear = exports.onLightSurfaceClearActive = exports.onLightSurfaceClearHover = exports.onLightSurfaceTransparentCard = exports.onLightSurfaceTransparentCardActive = exports.onLightSurfaceTransparentCardHover = exports.onLightSurfaceTransparentDeep = exports.onLightSurfaceTransparentDeepActive = exports.onLightSurfaceTransparentDeepHover = exports.onLightSurfaceTransparentTertiary = exports.onLightSurfaceTransparentTertiaryActive = exports.onLightSurfaceTransparentTertiaryHover = exports.onLightSurfaceTransparentSecondary = exports.onLightSurfaceTransparentSecondaryActive = exports.onLightSurfaceTransparentSecondaryHover = exports.onLightSurfaceTransparentPrimary = exports.onLightSurfaceTransparentPrimaryActive = exports.onLightSurfaceTransparentPrimaryHover = exports.onLightSurfaceSolidDefault = exports.onLightSurfaceSolidDefaultActive = exports.onLightSurfaceSolidDefaultHover = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidCardActive = exports.onLightSurfaceSolidCardHover = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidTertiaryActive = exports.onLightSurfaceSolidTertiaryHover = exports.onLightSurfaceSolidSecondary = exports.onLightSurfaceSolidSecondaryActive = exports.onLightSurfaceSolidSecondaryHover = exports.onLightSurfaceSolidPrimary = exports.onLightSurfaceSolidPrimaryActive = exports.onLightSurfaceSolidPrimaryHover = exports.onDarkSurfaceNegative = exports.onDarkSurfaceNegativeActive = exports.onDarkSurfaceNegativeHover = exports.onDarkSurfaceWarning = void 0;
exports.inverseSurfaceNegative = exports.inverseSurfaceNegativeActive = exports.inverseSurfaceNegativeHover = exports.inverseSurfaceWarning = exports.inverseSurfaceWarningActive = exports.inverseSurfaceWarningHover = exports.inverseSurfacePositive = exports.inverseSurfacePositiveActive = exports.inverseSurfacePositiveHover = exports.inverseSurfaceAccentGradient = exports.inverseSurfaceAccentGradientActive = exports.inverseSurfaceAccentGradientHover = exports.inverseSurfaceAccent = exports.inverseSurfaceAccentActive = exports.inverseSurfaceAccentHover = exports.inverseSurfaceClear = exports.inverseSurfaceClearActive = exports.inverseSurfaceClearHover = exports.inverseSurfaceTransparentCard = exports.inverseSurfaceTransparentCardActive = exports.inverseSurfaceTransparentCardHover = exports.inverseSurfaceTransparentDeep = exports.inverseSurfaceTransparentDeepActive = exports.inverseSurfaceTransparentDeepHover = exports.inverseSurfaceTransparentTertiary = exports.inverseSurfaceTransparentTertiaryActive = exports.inverseSurfaceTransparentTertiaryHover = exports.inverseSurfaceTransparentSecondary = exports.inverseSurfaceTransparentSecondaryActive = exports.inverseSurfaceTransparentSecondaryHover = exports.inverseSurfaceTransparentPrimary = exports.inverseSurfaceTransparentPrimaryActive = exports.inverseSurfaceTransparentPrimaryHover = exports.inverseSurfaceSolidDefault = exports.inverseSurfaceSolidDefaultActive = exports.inverseSurfaceSolidDefaultHover = exports.inverseSurfaceSolidCard = exports.inverseSurfaceSolidCardActive = exports.inverseSurfaceSolidCardHover = exports.inverseSurfaceSolidTertiary = exports.inverseSurfaceSolidTertiaryActive = exports.inverseSurfaceSolidTertiaryHover = exports.inverseSurfaceSolidSecondary = exports.inverseSurfaceSolidSecondaryActive = exports.inverseSurfaceSolidSecondaryHover = exports.inverseSurfaceSolidPrimary = exports.inverseSurfaceSolidPrimaryActive = exports.inverseSurfaceSolidPrimaryHover = exports.onLightSurfaceNegative = exports.onLightSurfaceNegativeActive = void 0;
exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonAccent = exports.buttonSecondary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.overlay = exports.critical = exports.warning = exports.success = exports.accent = exports.background = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackSecondary = exports.buttonBlack = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.skeletonGradientLighter = exports.skeletonGradient = exports.inverseOverlayHard = exports.inverseOverlaySoft = exports.onLightOverlayHard = exports.onLightOverlaySoft = exports.onDarkOverlayHard = exports.onDarkOverlaySoft = exports.overlayHard = exports.overlaySoft = exports.inverseBackgroundPrimary = exports.lightBackgroundPrimary = exports.darkBackgroundPrimary = exports.backgroundPrimary = void 0;
/** Основной цвет текста */
exports.textPrimaryHover = 'var(--text-primary-hover, #F5F5F593)';
/** Основной цвет текста */
exports.textPrimaryActive = 'var(--text-primary-active, #F5F5F5C4)';
/** Основной цвет текста */
exports.textPrimary = 'var(--text-primary, #f5f5f5F5)';
/** Вторичный цвет текста */
exports.textSecondaryHover = 'var(--text-secondary-hover, #F5F5F5FF)';
/** Вторичный цвет текста */
exports.textSecondaryActive = 'var(--text-secondary-active, #F5F5F5AB)';
/** Вторичный цвет текста */
exports.textSecondary = 'var(--text-secondary, #F5f5f58F)';
/** Третичный цвет текста */
exports.textTertiaryHover = 'var(--text-tertiary-hover, #F5F5F5FF)';
/** Третичный цвет текста */
exports.textTertiaryActive = 'var(--text-tertiary-active, #F5F5F556)';
/** Третичный цвет текста */
exports.textTertiary = 'var(--text-tertiary, #F5f5f548)';
/** Сплошной наборный текст */
exports.textParagraphHover = 'var(--text-paragraph-hover, #F5F5F57A)';
/** Сплошной наборный текст */
exports.textParagraphActive = 'var(--text-paragraph-active, #F5F5F5A3)';
/** Сплошной наборный текст */
exports.textParagraph = 'var(--text-paragraph, #f5f5f5cc)';
/** Акцентный цвет */
exports.textAccentHover = 'var(--text-accent-hover, #2ACB47FF)';
/** Акцентный цвет */
exports.textAccentActive = 'var(--text-accent-active, #1F9835FF)';
/** Акцентный цвет */
exports.textAccent = 'var(--text-accent, #24B23E)';
/** Акцентный цвет с градиентом */
exports.textAccentGradientHover = 'var(--text-accent-gradient-hover, #CCCCCCFF)';
/** Акцентный цвет с градиентом */
exports.textAccentGradientActive = 'var(--text-accent-gradient-active, #E6E6E6FF)';
/** Акцентный цвет с градиентом */
exports.textAccentGradient = 'var(--text-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет успеха */
exports.textPositiveHover = 'var(--text-positive-hover, #2ACB47FF)';
/** Цвет успеха */
exports.textPositiveActive = 'var(--text-positive-active, #1F9835FF)';
/** Цвет успеха */
exports.textPositive = 'var(--text-positive, #24B23E)';
/** Цвет предупреждения */
exports.textWarningHover = 'var(--text-warning-hover, #FF8442FF)';
/** Цвет предупреждения */
exports.textWarningActive = 'var(--text-warning-active, #FF5D05FF)';
/** Цвет предупреждения */
exports.textWarning = 'var(--text-warning, #FF7024)';
/** Цвет ошибки */
exports.textNegativeHover = 'var(--text-negative-hover, #FF5C6CFF)';
/** Цвет ошибки */
exports.textNegativeActive = 'var(--text-negative-active, #FF1F35FF)';
/** Цвет ошибки */
exports.textNegative = 'var(--text-negative, #FF3D51)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimaryHover = 'var(--on-dark-text-primary-hover, #F5F5F593)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimaryActive = 'var(--on-dark-text-primary-active, #F5F5F5C4)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = 'var(--on-dark-text-primary, #f5f5f5F5)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondaryHover = 'var(--on-dark-text-secondary-hover, #F5F5F5FF)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondaryActive = 'var(--on-dark-text-secondary-active, #F5F5F5AB)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = 'var(--on-dark-text-secondary, #F5f5f58F)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiaryHover = 'var(--on-dark-text-tertiary-hover, #F5F5F5FF)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiaryActive = 'var(--on-dark-text-tertiary-active, #F5F5F556)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = 'var(--on-dark-text-tertiary, #F5f5f548)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraphHover = 'var(--on-dark-text-paragraph-hover, #F5F5F57A)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraphActive = 'var(--on-dark-text-paragraph-active, #F5F5F5A3)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraph = 'var(--on-dark-text-paragraph, #f5f5f5CC)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccentHover = 'var(--on-dark-text-accent-hover, #2ACB47FF)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccentActive = 'var(--on-dark-text-accent-active, #1F9835FF)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccent = 'var(--on-dark-text-accent, #24B23E)';
/** Акцентный цвет с градиентом на темном фоне */
exports.onDarkTextAccentGradientHover = 'var(--on-dark-text-accent-gradient-hover, #CCCCCCFF)';
/** Акцентный цвет с градиентом на темном фоне */
exports.onDarkTextAccentGradientActive = 'var(--on-dark-text-accent-gradient-active, #E6E6E6FF)';
/** Акцентный цвет с градиентом на темном фоне */
exports.onDarkTextAccentGradient = 'var(--on-dark-text-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositiveHover = 'var(--on-dark-text-positive-hover, #2ACB47FF)';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositiveActive = 'var(--on-dark-text-positive-active, #1F9835FF)';
/** Цвет успеха на темном фоне */
exports.onDarkTextPositive = 'var(--on-dark-text-positive, #24B23E)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarningHover = 'var(--on-dark-text-warning-hover, #FF8442FF)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarningActive = 'var(--on-dark-text-warning-active, #FF5D05FF)';
/** Цвет предупреждения на темном фоне */
exports.onDarkTextWarning = 'var(--on-dark-text-warning, #FF7024)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegativeHover = 'var(--on-dark-text-negative-hover, #FF5C6CFF)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegativeActive = 'var(--on-dark-text-negative-active, #FF1F35FF)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegative = 'var(--on-dark-text-negative, #FF3D51)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimaryHover = 'var(--on-light-text-primary-hover, #17171793)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimaryActive = 'var(--on-light-text-primary-active, #171717C4)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = 'var(--on-light-text-primary, #171717F5)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondaryHover = 'var(--on-light-text-secondary-hover, #171717FF)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondaryActive = 'var(--on-light-text-secondary-active, #171717AB)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = 'var(--on-light-text-secondary, #1717178F)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiaryHover = 'var(--on-light-text-tertiary-hover, #171717FF)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiaryActive = 'var(--on-light-text-tertiary-active, #17171756)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = 'var(--on-light-text-tertiary, #17171748)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraphHover = 'var(--on-light-text-paragraph-hover, #1717177A)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraphActive = 'var(--on-light-text-paragraph-active, #171717A3)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraph = 'var(--on-light-text-paragraph, #171717CC)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccentHover = 'var(--on-light-text-accent-hover, #13AA2CFF)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccentActive = 'var(--on-light-text-accent-active, #0D731EFF)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccent = 'var(--on-light-text-accent, #108E26)';
/** Акцентный цвет с градиентом на светлом фоне */
exports.onLightTextAccentGradientHover = 'var(--on-light-text-accent-gradient-hover, #CCCCCCFF)';
/** Акцентный цвет с градиентом на светлом фоне */
exports.onLightTextAccentGradientActive = 'var(--on-light-text-accent-gradient-active, #E6E6E6FF)';
/** Акцентный цвет с градиентом на светлом фоне */
exports.onLightTextAccentGradient = 'var(--on-light-text-accent-gradient, linear-gradient(45deg, #0D8523 0%, #0DA8AB 100%))';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositiveHover = 'var(--on-light-text-positive-hover, #13AA2CFF)';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositiveActive = 'var(--on-light-text-positive-active, #0D731EFF)';
/** Цвет успеха на светлом фоне */
exports.onLightTextPositive = 'var(--on-light-text-positive, #108E26)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarningHover = 'var(--on-light-text-warning-hover, #FD650DFF)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarningActive = 'var(--on-light-text-warning-active, #CA4B02FF)';
/** Цвет предупреждения на светлом фоне */
exports.onLightTextWarning = 'var(--on-light-text-warning, #E85702)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegativeHover = 'var(--on-light-text-negative-hover, #F5384BFF)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegativeActive = 'var(--on-light-text-negative-active, #E40C22FF)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegative = 'var(--on-light-text-negative, #F31B31)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimaryHover = 'var(--inverse-text-primary-hover, #17171793)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimaryActive = 'var(--inverse-text-primary-active, #171717C4)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimary = 'var(--inverse-text-primary, #171717F5)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondaryHover = 'var(--inverse-text-secondary-hover, #171717FF)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondaryActive = 'var(--inverse-text-secondary-active, #171717AB)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondary = 'var(--inverse-text-secondary, #1717178F)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiaryHover = 'var(--inverse-text-tertiary-hover, #171717FF)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiaryActive = 'var(--inverse-text-tertiary-active, #17171756)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiary = 'var(--inverse-text-tertiary, #17171748)';
/** Инвертированный cплошной наборный текст */
exports.inverseTextParagraphHover = 'var(--inverse-text-paragraph-hover, #1717177A)';
/** Инвертированный cплошной наборный текст */
exports.inverseTextParagraphActive = 'var(--inverse-text-paragraph-active, #171717A3)';
/** Инвертированный cплошной наборный текст */
exports.inverseTextParagraph = 'var(--inverse-text-paragraph, #171717CC)';
/** Инвертированный акцентный цвет */
exports.inverseTextAccentHover = 'var(--inverse-text-accent-hover, #13AA2CFF)';
/** Инвертированный акцентный цвет */
exports.inverseTextAccentActive = 'var(--inverse-text-accent-active, #0D731EFF)';
/** Инвертированный акцентный цвет */
exports.inverseTextAccent = 'var(--inverse-text-accent, #108E26)';
/** Инвертированный акцентный цвет с градиентом */
exports.inverseTextAccentGradientHover = 'var(--inverse-text-accent-gradient-hover, #CCCCCCFF)';
/** Инвертированный акцентный цвет с градиентом */
exports.inverseTextAccentGradientActive = 'var(--inverse-text-accent-gradient-active, #E6E6E6FF)';
/** Инвертированный акцентный цвет с градиентом */
exports.inverseTextAccentGradient = 'var(--inverse-text-accent-gradient, linear-gradient(45deg, #0D8523 0%, #0DA8AB 100%))';
/** Инвертированный цвет успеха */
exports.inverseTextPositiveHover = 'var(--inverse-text-positive-hover, #13AA2CFF)';
/** Инвертированный цвет успеха */
exports.inverseTextPositiveActive = 'var(--inverse-text-positive-active, #0D731EFF)';
/** Инвертированный цвет успеха */
exports.inverseTextPositive = 'var(--inverse-text-positive, #108E26)';
/** Инвертированный цвет предупреждения */
exports.inverseTextWarningHover = 'var(--inverse-text-warning-hover, #FD650DFF)';
/** Инвертированный цвет предупреждения */
exports.inverseTextWarningActive = 'var(--inverse-text-warning-active, #CA4B02FF)';
/** Инвертированный цвет предупреждения */
exports.inverseTextWarning = 'var(--inverse-text-warning, #E85702)';
/** Инвертированный цвет ошибки */
exports.inverseTextNegativeHover = 'var(--inverse-text-negative-hover, #F5384BFF)';
/** Инвертированный цвет ошибки */
exports.inverseTextNegativeActive = 'var(--inverse-text-negative-active, #E40C22FF)';
/** Инвертированный цвет ошибки */
exports.inverseTextNegative = 'var(--inverse-text-negative, #F31B31)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimaryHover = 'var(--surface-solid-primary-hover, #1C1C1CFF)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimaryActive = 'var(--surface-solid-primary-active, #121212FF)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimary = 'var(--surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondaryHover = 'var(--surface-solid-secondary-hover, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondaryActive = 'var(--surface-solid-secondary-active, #212121FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondary = 'var(--surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiaryHover = 'var(--surface-solid-tertiary-hover, #404040FF)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiaryActive = 'var(--surface-solid-tertiary-active, #303030FF)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = 'var(--surface-solid-tertiary, #363636)';
/** Основной фон для карточек */
exports.surfaceSolidCardHover = 'var(--surface-solid-card-hover, #1C1C1CFF)';
/** Основной фон для карточек */
exports.surfaceSolidCardActive = 'var(--surface-solid-card-active, #121212FF)';
/** Основной фон для карточек */
exports.surfaceSolidCard = 'var(--surface-solid-card, #171717)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefaultHover = 'var(--surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefaultActive = 'var(--surface-solid-default-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefault = 'var(--surface-solid-default, #FFFFFFFF)';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimaryHover = 'var(--surface-transparent-primary-hover, #FFFFFF1F)';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimaryActive = 'var(--surface-transparent-primary-active, #FFFFFF0A)';
/** Основной прозрачный фон поверхности/контрола */
exports.surfaceTransparentPrimary = 'var(--surface-transparent-primary, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondaryHover = 'var(--surface-transparent-secondary-hover, #FFFFFF3D)';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondaryActive = 'var(--surface-transparent-secondary-active, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentSecondary = 'var(--surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiaryHover = 'var(--surface-transparent-tertiary-hover, #FFFFFF52)';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiaryActive = 'var(--surface-transparent-tertiary-active, #FFFFFF24)';
/** Третичный прозрачный фон поверхности/контрола */
exports.surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон поверхности/контрола по умолчанию */
exports.surfaceTransparentDeepHover = 'var(--surface-transparent-deep-hover, #FFFFFFC2)';
/** Прозрачный фон поверхности/контрола по умолчанию */
exports.surfaceTransparentDeepActive = 'var(--surface-transparent-deep-active, #FFFFFF94)';
/** Прозрачный фон поверхности/контрола по умолчанию */
exports.surfaceTransparentDeep = 'var(--surface-transparent-deep, #FFFFFFA3)';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCardHover = 'var(--surface-transparent-card-hover, #FFFFFF1F)';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCardActive = 'var(--surface-transparent-card-active, #FFFFFF0A)';
/** Прозрачный фон для карточек */
exports.surfaceTransparentCard = 'var(--surface-transparent-card, #FFFFFF0F)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClearHover = 'var(--surface-clear-hover, #00000000)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClearActive = 'var(--surface-clear-active, #00000000)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClear = 'var(--surface-clear, rgba(0, 0, 0, 0))';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentHover = 'var(--surface-accent-hover, #1DAF37FF)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentActive = 'var(--surface-accent-active, #18952FFF)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccent = 'var(--surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradientHover = 'var(--surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradientActive = 'var(--surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositiveHover = 'var(--surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositiveActive = 'var(--surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositive = 'var(--surface-positive, #1A9E32)';
/** Цвет фона поверхности предупреждение */
exports.surfaceWarningHover = 'var(--surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности предупреждение */
exports.surfaceWarningActive = 'var(--surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности предупреждение */
exports.surfaceWarning = 'var(--surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeHover = 'var(--surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeActive = 'var(--surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = 'var(--surface-negative, #FF293E)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimaryHover = 'var(--on-dark-surface-solid-primary-hover, #1C1C1CFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimaryActive = 'var(--on-dark-surface-solid-primary-active, #121212FF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности на темном фоне */
exports.onDarkSurfaceSolidSecondaryHover = 'var(--on-dark-surface-solid-secondary-hover, #303030FF)';
/** Вторичный непрозрачный фон поверхности на темном фоне */
exports.onDarkSurfaceSolidSecondaryActive = 'var(--on-dark-surface-solid-secondary-active, #212121FF)';
/** Вторичный непрозрачный фон поверхности на темном фоне */
exports.onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiaryHover = 'var(--on-dark-surface-solid-tertiary-hover, #404040FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiaryActive = 'var(--on-dark-surface-solid-tertiary-active, #303030FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #363636)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCardHover = 'var(--on-dark-surface-solid-card-hover, #1C1C1CFF)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCardActive = 'var(--on-dark-surface-solid-card-active, #121212FF)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #171717)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefaultHover = 'var(--on-dark-surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefaultActive = 'var(--on-dark-surface-solid-default-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #FFFFFF)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimaryHover = 'var(--on-dark-surface-transparent-primary-hover, #FFFFFF1F)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimaryActive = 'var(--on-dark-surface-transparent-primary-active, #FFFFFF0A)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondaryHover = 'var(--on-dark-surface-transparent-secondary-hover, #FFFFFF3D)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondaryActive = 'var(--on-dark-surface-transparent-secondary-active, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiaryHover = 'var(--on-dark-surface-transparent-tertiary-hover, #FFFFFF52)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiaryActive = 'var(--on-dark-surface-transparent-tertiary-active, #FFFFFF24)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон поверхности по умолчанию на темном фоне */
exports.onDarkSurfaceTransparentDeepHover = 'var(--on-dark-surface-transparent-deep-hover, #FFFFFFC2)';
/** Прозрачный фон поверхности по умолчанию на темном фоне */
exports.onDarkSurfaceTransparentDeepActive = 'var(--on-dark-surface-transparent-deep-active, #FFFFFF94)';
/** Прозрачный фон поверхности по умолчанию на темном фоне */
exports.onDarkSurfaceTransparentDeep = 'var(--on-dark-surface-transparent-deep, #FFFFFFA3)';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCardHover = 'var(--on-dark-surface-transparent-card-hover, #FFFFFF1F)';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCardActive = 'var(--on-dark-surface-transparent-card-active, #FFFFFF0A)';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, #FFFFFF0F)';
/** Фон поверхности/контрола без заливки на темном фоне */
exports.onDarkSurfaceClearHover = 'var(--on-dark-surface-clear-hover, #00000000)';
/** Фон поверхности/контрола без заливки на темном фоне */
exports.onDarkSurfaceClearActive = 'var(--on-dark-surface-clear-active, #00000000)';
/** Фон поверхности/контрола без заливки на темном фоне */
exports.onDarkSurfaceClear = 'var(--on-dark-surface-clear, rgba(0, 0, 0, 0))';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccentHover = 'var(--on-dark-surface-accent-hover, #1DAF37FF)';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccentActive = 'var(--on-dark-surface-accent-active, #18952FFF)';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradientHover = 'var(--on-dark-surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradientActive = 'var(--on-dark-surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradient = 'var(--on-dark-surface-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositiveHover = 'var(--on-dark-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositiveActive = 'var(--on-dark-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех на темном фоне */
exports.onDarkSurfacePositive = 'var(--on-dark-surface-positive, #1A9E32)';
/** Цвет фона поверхности предупреждение на темном фоне */
exports.onDarkSurfaceWarningHover = 'var(--on-dark-surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности предупреждение на темном фоне */
exports.onDarkSurfaceWarningActive = 'var(--on-dark-surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности предупреждение на темном фоне */
exports.onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegativeHover = 'var(--on-dark-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegativeActive = 'var(--on-dark-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293E)';
/** Основной непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidPrimaryHover = 'var(--on-light-surface-solid-primary-hover, #FAFAFAFF)';
/** Основной непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidPrimaryActive = 'var(--on-light-surface-solid-primary-active, #F0F0F0FF)';
/** Основной непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F5F5F5)';
/** Вторичный непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidSecondaryHover = 'var(--on-light-surface-solid-secondary-hover, #F7F7F7FF)';
/** Вторичный непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidSecondaryActive = 'var(--on-light-surface-solid-secondary-active, #E8E8E8FF)';
/** Вторичный непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #ECECEC)';
/** Третичный непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidTertiaryHover = 'var(--on-light-surface-solid-tertiary-hover, #E8E8E8FF)';
/** Третичный непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidTertiaryActive = 'var(--on-light-surface-solid-tertiary-active, #D9D9D9FF)';
/** Третичный непрозрачный фон поверхности на светлом фоне */
exports.onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #DDDDDD)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCardHover = 'var(--on-light-surface-solid-card-hover, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCardActive = 'var(--on-light-surface-solid-card-active, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefaultHover = 'var(--on-light-surface-solid-default-hover, #0D0D0DFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefaultActive = 'var(--on-light-surface-solid-default-active, #030303FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #080808)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentPrimaryHover = 'var(--on-light-surface-transparent-primary-hover, #0808080F)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentPrimaryActive = 'var(--on-light-surface-transparent-primary-active, #08080800)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, rgba(8, 8, 8, 0.02))';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondaryHover = 'var(--on-light-surface-transparent-secondary-hover, #0808081F)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondaryActive = 'var(--on-light-surface-transparent-secondary-active, #0808080A)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiaryHover = 'var(--on-light-surface-transparent-tertiary-hover, #0808083D)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiaryActive = 'var(--on-light-surface-transparent-tertiary-active, #0808080F)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Прозрачный фон поверхности по умолчанию на светлом фоне */
exports.onLightSurfaceTransparentDeepHover = 'var(--on-light-surface-transparent-deep-hover, #080808C2)';
/** Прозрачный фон поверхности по умолчанию на светлом фоне */
exports.onLightSurfaceTransparentDeepActive = 'var(--on-light-surface-transparent-deep-active, #08080894)';
/** Прозрачный фон поверхности по умолчанию на светлом фоне */
exports.onLightSurfaceTransparentDeep = 'var(--on-light-surface-transparent-deep, #080808A3)';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCardHover = 'var(--on-light-surface-transparent-card-hover, #0808081F)';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCardActive = 'var(--on-light-surface-transparent-card-active, #0808080A)';
/** Прозрачный фон для карточек на светлом фоне */
exports.onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, rgba(8, 8, 8, 0.06))';
/** Фон поверхности/контрола без заливки на светлом фоне */
exports.onLightSurfaceClearHover = 'var(--on-light-surface-clear-hover, #00000000)';
/** Фон поверхности/контрола без заливки на светлом фоне */
exports.onLightSurfaceClearActive = 'var(--on-light-surface-clear-active, #00000000)';
/** Фон поверхности/контрола без заливки на светлом фоне */
exports.onLightSurfaceClear = 'var(--on-light-surface-clear, rgba(0, 0, 0, 0))';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccentHover = 'var(--on-light-surface-accent-hover, #1DAF37FF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccentActive = 'var(--on-light-surface-accent-active, #18952FFF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccent = 'var(--on-light-surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradientHover = 'var(--on-light-surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradientActive = 'var(--on-light-surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradient = 'var(--on-light-surface-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositiveHover = 'var(--on-light-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositiveActive = 'var(--on-light-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
exports.onLightSurfacePositive = 'var(--on-light-surface-positive, #1A9E32)';
/** Цвет фона поверхности / предупреждение на светлом фоне */
exports.onLightSurfaceWarningHover = 'var(--on-light-surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности / предупреждение на светлом фоне */
exports.onLightSurfaceWarningActive = 'var(--on-light-surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности / предупреждение на светлом фоне */
exports.onLightSurfaceWarning = 'var(--on-light-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegativeHover = 'var(--on-light-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegativeActive = 'var(--on-light-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegative = 'var(--on-light-surface-negative, #FF293E)';
/** Инвертированный основной непрозрачный фон поверхности */
exports.inverseSurfaceSolidPrimaryHover = 'var(--inverse-surface-solid-primary-hover, #FAFAFAFF)';
/** Инвертированный основной непрозрачный фон поверхности */
exports.inverseSurfaceSolidPrimaryActive = 'var(--inverse-surface-solid-primary-active, #F0F0F0FF)';
/** Инвертированный основной непрозрачный фон поверхности */
exports.inverseSurfaceSolidPrimary = 'var(--inverse-surface-solid-primary, #F5F5F5)';
/** Инвертированный вторичный непрозрачный фон поверхности */
exports.inverseSurfaceSolidSecondaryHover = 'var(--inverse-surface-solid-secondary-hover, #F7F7F7FF)';
/** Инвертированный вторичный непрозрачный фон поверхности */
exports.inverseSurfaceSolidSecondaryActive = 'var(--inverse-surface-solid-secondary-active, #E8E8E8FF)';
/** Инвертированный вторичный непрозрачный фон поверхности */
exports.inverseSurfaceSolidSecondary = 'var(--inverse-surface-solid-secondary, #ECECEC)';
/** Инвертированный третичный непрозрачный фон поверхности */
exports.inverseSurfaceSolidTertiaryHover = 'var(--inverse-surface-solid-tertiary-hover, #E8E8E8FF)';
/** Инвертированный третичный непрозрачный фон поверхности */
exports.inverseSurfaceSolidTertiaryActive = 'var(--inverse-surface-solid-tertiary-active, #D9D9D9FF)';
/** Инвертированный третичный непрозрачный фон поверхности */
exports.inverseSurfaceSolidTertiary = 'var(--inverse-surface-solid-tertiary, #DDDDDD)';
/** Инвертированный основной фон для карточек */
exports.inverseSurfaceSolidCardHover = 'var(--inverse-surface-solid-card-hover, #FFFFFFFF)';
/** Инвертированный основной фон для карточек */
exports.inverseSurfaceSolidCardActive = 'var(--inverse-surface-solid-card-active, #FFFFFFFF)';
/** Инвертированный основной фон для карточек */
exports.inverseSurfaceSolidCard = 'var(--inverse-surface-solid-card, #FFFFFFFF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
exports.inverseSurfaceSolidDefaultHover = 'var(--inverse-surface-solid-default-hover, #0D0D0DFF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
exports.inverseSurfaceSolidDefaultActive = 'var(--inverse-surface-solid-default-active, #030303FF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
exports.inverseSurfaceSolidDefault = 'var(--inverse-surface-solid-default, #080808)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentPrimaryHover = 'var(--inverse-surface-transparent-primary-hover, #0808080F)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentPrimaryActive = 'var(--inverse-surface-transparent-primary-active, #08080800)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentPrimary = 'var(--inverse-surface-transparent-primary, rgba(8, 8, 8, 0.02))';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentSecondaryHover = 'var(--inverse-surface-transparent-secondary-hover, #0808081F)';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentSecondaryActive = 'var(--inverse-surface-transparent-secondary-active, #0808080A)';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentSecondary = 'var(--inverse-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentTertiaryHover = 'var(--inverse-surface-transparent-tertiary-hover, #0808083D)';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentTertiaryActive = 'var(--inverse-surface-transparent-tertiary-active, #0808080F)';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
exports.inverseSurfaceTransparentTertiary = 'var(--inverse-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Инвертированный прозрачный фон поверхности по умолчанию */
exports.inverseSurfaceTransparentDeepHover = 'var(--inverse-surface-transparent-deep-hover, #080808C2)';
/** Инвертированный прозрачный фон поверхности по умолчанию */
exports.inverseSurfaceTransparentDeepActive = 'var(--inverse-surface-transparent-deep-active, #08080894)';
/** Инвертированный прозрачный фон поверхности по умолчанию */
exports.inverseSurfaceTransparentDeep = 'var(--inverse-surface-transparent-deep, #080808A3)';
/** Инвертированный прозрачный фон для карточек */
exports.inverseSurfaceTransparentCardHover = 'var(--inverse-surface-transparent-card-hover, #FFFFFFFF)';
/** Инвертированный прозрачный фон для карточек */
exports.inverseSurfaceTransparentCardActive = 'var(--inverse-surface-transparent-card-active, #FFFFFFFF)';
/** Инвертированный прозрачный фон для карточек */
exports.inverseSurfaceTransparentCard = 'var(--inverse-surface-transparent-card, #FFFFFFFF)';
/** Инвертированный фон поверхности/контрола без заливки */
exports.inverseSurfaceClearHover = 'var(--inverse-surface-clear-hover, #00000000)';
/** Инвертированный фон поверхности/контрола без заливки */
exports.inverseSurfaceClearActive = 'var(--inverse-surface-clear-active, #00000000)';
/** Инвертированный фон поверхности/контрола без заливки */
exports.inverseSurfaceClear = 'var(--inverse-surface-clear, rgba(0, 0, 0, 0))';
/** Инвертированный акцентный фон поверхности/контрола */
exports.inverseSurfaceAccentHover = 'var(--inverse-surface-accent-hover, #1DAF37FF)';
/** Инвертированный акцентный фон поверхности/контрола */
exports.inverseSurfaceAccentActive = 'var(--inverse-surface-accent-active, #18952FFF)';
/** Инвертированный акцентный фон поверхности/контрола */
exports.inverseSurfaceAccent = 'var(--inverse-surface-accent, #1A9E32)';
/** Инвертированный акцентный фон поверхности/контрола с градиентом */
exports.inverseSurfaceAccentGradientHover = 'var(--inverse-surface-accent-gradient-hover, #FFFFFFFF)';
/** Инвертированный акцентный фон поверхности/контрола с градиентом */
exports.inverseSurfaceAccentGradientActive = 'var(--inverse-surface-accent-gradient-active, #FFFFFFFF)';
/** Инвертированный акцентный фон поверхности/контрола с градиентом */
exports.inverseSurfaceAccentGradient = 'var(--inverse-surface-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Инвертированный цвет фона поверхности/контрола успех */
exports.inverseSurfacePositiveHover = 'var(--inverse-surface-positive-hover, #1DAF37FF)';
/** Инвертированный цвет фона поверхности/контрола успех */
exports.inverseSurfacePositiveActive = 'var(--inverse-surface-positive-active, #18952FFF)';
/** Инвертированный цвет фона поверхности/контрола успех */
exports.inverseSurfacePositive = 'var(--inverse-surface-positive, #1A9E32)';
/** Инвертированный цвет фона поверхности / предупреждение */
exports.inverseSurfaceWarningHover = 'var(--inverse-surface-warning-hover, #FB7223FF)';
/** Инвертированный цвет фона поверхности / предупреждение */
exports.inverseSurfaceWarningActive = 'var(--inverse-surface-warning-active, #F05B05FF)';
/** Инвертированный цвет фона поверхности / предупреждение */
exports.inverseSurfaceWarning = 'var(--inverse-surface-warning, #FA5F05)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
exports.inverseSurfaceNegativeHover = 'var(--inverse-surface-negative-hover, #FF475AFF)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
exports.inverseSurfaceNegativeActive = 'var(--inverse-surface-negative-active, #FF1F35FF)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
exports.inverseSurfaceNegative = 'var(--inverse-surface-negative, #FF293E)';
/** Основной фон */
exports.backgroundPrimary = 'var(--background-primary, #080808)';
/** Основной фон на темном фоне */
exports.darkBackgroundPrimary = 'var(--dark-background-primary, #080808)';
/** Основной фон на светлом фоне */
exports.lightBackgroundPrimary = 'var(--light-background-primary, #F9F9F9)';
/** Инвертированный основной фон */
exports.inverseBackgroundPrimary = 'var(--inverse-background-primary, #F9F9F9)';
/** Цвет фона паранжи светлый */
exports.overlaySoft = 'var(--overlay-soft, #0808088F)';
/** Цвет фона паранжи темный */
exports.overlayHard = 'var(--overlay-hard, #080808f5)';
/** Цвет фона паранжи светлый на темном фоне */
exports.onDarkOverlaySoft = 'var(--on-dark-overlay-soft, #0808088f)';
/** Цвет фона паранжи темный на темном фоне */
exports.onDarkOverlayHard = 'var(--on-dark-overlay-hard, #080808f5)';
/** Цвет фона паранжи светлый на светлом фоне */
exports.onLightOverlaySoft = 'var(--on-light-overlay-soft, #f9f9f98f)';
/** Цвет фона паранжи темный на светлом фоне */
exports.onLightOverlayHard = 'var(--on-light-overlay-hard, #f9f9f9f5)';
/** Инвертированный цвет фона паранжи светлый */
exports.inverseOverlaySoft = 'var(--inverse-overlay-soft, #f9f9f98F)';
/** Инвертированный цвет фона паранжи темный */
exports.inverseOverlayHard = 'var(--inverse-overlay-hard, #f9f9f9f5)';
exports.skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
exports.skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use onLightSurfaceSolidDefault */
exports.buttonBlack = 'var(--plasma-colors-button-black, var(--on-light-surface-solid-default))';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
exports.buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, var(--on-light-surface-transparent-secondary))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
exports.buttonWhite = 'var(--plasma-colors-button-white, var(--on-dark-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
exports.buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, var(--on-dark-surface-transparent-secondary))';
/** @deprecated instead use textPrimary */
exports.text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
exports.primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
exports.secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
exports.tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use textParagraph */
exports.paragraph = 'var(--plasma-colors-paragraph, var(--text-paragraph))';
/** @deprecated instead use backgroundPrimary */
exports.background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textAccent */
exports.accent = 'var(--plasma-colors-accent, var(--text-accent))';
/** @deprecated instead use textPositive */
exports.success = 'var(--plasma-colors-success, var(--text-positive))';
/** @deprecated instead use textWarning */
exports.warning = 'var(--plasma-colors-warning, var(--text-warning))';
/** @deprecated instead use textNegative */
exports.critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use overlaySoft */
exports.overlay = 'var(--plasma-colors-overlay, var(--overlay-soft))';
/** @deprecated instead use surfaceTransparentPrimary */
exports.surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, var(--surface-transparent-primary))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, var(--surface-transparent-secondary))';
/** @deprecated instead use surfaceTransparentTertiary */
exports.surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, var(--surface-transparent-tertiary))';
/** @deprecated instead use surfaceSolidPrimary */
exports.surfaceSolid01 = 'var(--plasma-colors-surface-solid01, var(--surface-solid-primary))';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use surfaceTransparentCard */
exports.surfaceCard = 'var(--plasma-colors-surface-card, var(--surface-transparent-card))';
/** @deprecated instead use surfaceTransparentSecondary */
exports.buttonSecondary = 'var(--plasma-colors-button-secondary, var(--surface-transparent-secondary))';
/** @deprecated instead use textAccent */
exports.buttonAccent = 'var(--plasma-colors-button-accent, var(--text-accent))';
/** @deprecated instead use surfacePositive */
exports.buttonSuccess = 'var(--plasma-colors-button-success, var(--surface-positive))';
/** @deprecated instead use surfaceWarning */
exports.buttonWarning = 'var(--plasma-colors-button-warning, var(--surface-warning))';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
