// @salutejs/plasma-new-hope/styled-components/cjs/components/Slider/components/Double/Double.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DoubleSlider = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _SliderBase = /*#__PURE__*/require("../SliderBase/SliderBase");
var _ui = /*#__PURE__*/require("../../ui");
var _utils = /*#__PURE__*/require("../../utils");
var _utils2 = /*#__PURE__*/require("../../../../utils");
var _Slider = /*#__PURE__*/require("../../Slider.tokens");
var _Double = /*#__PURE__*/require("./Double.styles");
var _excluded = ["min", "max", "value", "disabled", "label", "labelContentLeft", "size", "onChangeCommitted", "onChangeTextField", "onBlurTextField", "onKeyDownTextField", "onChange", "ariaLabel", "multipleStepSize"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
function getXCenterHandle(handle) {
  var _handle$parentElement;
  var containerX = ((_handle$parentElement = handle.parentElement) === null || _handle$parentElement === void 0 || (_handle$parentElement = _handle$parentElement.getBoundingClientRect()) === null || _handle$parentElement === void 0 ? void 0 : _handle$parentElement.x) || 0;
  var handleRect = handle.getBoundingClientRect();
  var handlePosition = handleRect.x;
  return handlePosition - containerX;
}
var DoubleSlider = exports.DoubleSlider = function DoubleSlider(_ref) {
  var min = _ref.min,
    max = _ref.max,
    value = _ref.value,
    disabled = _ref.disabled,
    label = _ref.label,
    labelContentLeft = _ref.labelContentLeft,
    _ref$size = _ref.size,
    size = _ref$size === void 0 ? 'm' : _ref$size,
    onChangeCommitted = _ref.onChangeCommitted,
    onChangeTextField = _ref.onChangeTextField,
    onBlurTextField = _ref.onBlurTextField,
    onKeyDownTextField = _ref.onKeyDownTextField,
    onChange = _ref.onChange,
    ariaLabel = _ref.ariaLabel,
    _ref$multipleStepSize = _ref.multipleStepSize,
    multipleStepSize = _ref$multipleStepSize === void 0 ? 10 : _ref$multipleStepSize,
    rest = _objectWithoutProperties(_ref, _excluded);
  var _useState = (0, _react.useState)({
      stepSize: 0,
      railFillWidth: 0,
      railFillXPosition: 0,
      xFirstHandle: 0,
      xSecondHandle: 0,
      firstHandleZIndex: 100,
      secondHandleZIndex: 101,
      firstValue: value[0],
      secondValue: value[1]
    }),
    _useState2 = _slicedToArray(_useState, 2),
    state = _useState2[0],
    setState = _useState2[1];
  var _useState3 = (0, _react.useState)(false),
    _useState4 = _slicedToArray(_useState3, 2),
    firstInputActive = _useState4[0],
    setFirstInputActive = _useState4[1];
  var _useState5 = (0, _react.useState)(false),
    _useState6 = _slicedToArray(_useState5, 2),
    secondInputActive = _useState6[0],
    setSecondInputActive = _useState6[1];
  var _useState7 = (0, _react.useState)(value[0]),
    _useState8 = _slicedToArray(_useState7, 2),
    firstValue = _useState8[0],
    setFirstValue = _useState8[1];
  var _useState9 = (0, _react.useState)(value[1]),
    _useState10 = _slicedToArray(_useState9, 2),
    secondValue = _useState10[0],
    setSecondValue = _useState10[1];
  var firstHandleRef = (0, _react.useRef)(null);
  var secondHandleRef = (0, _react.useRef)(null);
  var firstHandleValue = (0, _react.useRef)(value[0]);
  var secondHandleValue = (0, _react.useRef)(value[1]);
  var stepSize = state.stepSize;
  var hasLabelContent = label || labelContentLeft;
  var firstInputActiveClass = firstInputActive && !disabled ? _Slider.classes.textFieldActive : undefined;
  var secondInputActiveClass = secondInputActive && !disabled ? _Slider.classes.textFieldActive : undefined;
  (0, _react.useEffect)(function () {
    var firstLocalValue = Math.min(Math.max(value[0], min), max) - min;
    var secondLocalValue = Math.min(Math.max(value[1], min), max) - min;
    setFirstValue(value[0]);
    setSecondValue(value[1]);
    setState(function (prevState) {
      return _objectSpread(_objectSpread({}, prevState), {}, {
        railFillXPosition: stepSize * firstLocalValue,
        railFillWidth: stepSize * secondLocalValue - stepSize * firstLocalValue,
        xFirstHandle: stepSize * firstLocalValue,
        xSecondHandle: stepSize * secondLocalValue
      });
    });
  }, [value, stepSize, min, max, setFirstValue, setSecondValue]);
  var setStepSize = (0, _react.useCallback)(function (newStepSize) {
    setState(function (prevState) {
      return _objectSpread(_objectSpread({}, prevState), {}, {
        stepSize: newStepSize
      });
    });
  }, [setState]);
  var onFirstHandleChange = function onFirstHandleChange(handleValue, data) {
    if (!(secondHandleRef !== null && secondHandleRef !== void 0 && secondHandleRef.current)) {
      return;
    }
    var newHandleXPosition = data.x;
    var secondHandleXPosition = getXCenterHandle(secondHandleRef.current);
    var fillWidth = secondHandleXPosition - newHandleXPosition;
    firstHandleValue.current = handleValue;
    setFirstValue(handleValue);
    setState(function (prevState) {
      return _objectSpread(_objectSpread({}, prevState), {}, {
        firstHandleZIndex: 101,
        secondHandleZIndex: 100,
        railFillWidth: fillWidth < 0 ? 0 : fillWidth,
        railFillXPosition: newHandleXPosition
      });
    });
    if (onChange) {
      onChange([handleValue, value[1]]);
    }
  };
  var onFirstHandleChangeCommitted = function onFirstHandleChangeCommitted(handleValue, data) {
    if (!(secondHandleRef !== null && secondHandleRef !== void 0 && secondHandleRef.current)) {
      return;
    }
    var newHandleXPosition = data.x;
    var secondHandleXPosition = getXCenterHandle(secondHandleRef.current);
    var fillWidth = secondHandleXPosition - newHandleXPosition;
    firstHandleValue.current = handleValue;
    setFirstValue(handleValue);
    if (onChangeCommitted) {
      onChangeCommitted([handleValue, value[1]]);
    }
    setState(function (prevState) {
      return _objectSpread(_objectSpread({}, prevState), {}, {
        firstValue: handleValue,
        xFirstHandle: data.x,
        railFillWidth: fillWidth < 0 ? 0 : fillWidth,
        railFillXPosition: newHandleXPosition
      });
    });
  };
  var onFirstTextfieldChange = function onFirstTextfieldChange(event) {
    if (!(0, _utils2.isNumber)(event.target.value)) {
      return;
    }
    var handleValue = Number(event.target.value);
    setFirstValue(handleValue);
    if (onChangeTextField) {
      onChangeTextField([handleValue, secondValue], event);
    }
  };
  var onFirstTextfieldBlur = function onFirstTextfieldBlur(event) {
    if (!(0, _utils2.isNumber)(event.target.value)) {
      return;
    }
    var handleValue = Number(event.target.value);
    setFirstValue(handleValue);
    if (onBlurTextField) {
      onBlurTextField && onBlurTextField([handleValue, secondValue], event);
    }
  };
  var onSecondHandleChange = function onSecondHandleChange(handleValue, data) {
    if (!(firstHandleRef !== null && firstHandleRef !== void 0 && firstHandleRef.current)) {
      return;
    }
    var firstXHandleXPosition = getXCenterHandle(firstHandleRef.current);
    var newHandleXPosition = data.x;
    var fillWidth = newHandleXPosition - firstXHandleXPosition;
    secondHandleValue.current = handleValue;
    setSecondValue(handleValue);
    setState(function (prevState) {
      return _objectSpread(_objectSpread({}, prevState), {}, {
        firstHandleZIndex: 100,
        secondHandleZIndex: 101,
        railFillWidth: fillWidth < 0 ? 0 : fillWidth,
        railFillXPosition: firstXHandleXPosition
      });
    });
    if (onChange) {
      onChange([value[0], handleValue]);
    }
  };
  var onSecondHandleChangeCommitted = function onSecondHandleChangeCommitted(handleValue, data) {
    if (!(firstHandleRef !== null && firstHandleRef !== void 0 && firstHandleRef.current)) {
      return;
    }
    var firstXHandleXPosition = getXCenterHandle(firstHandleRef.current);
    var newHandleXPosition = data.x;
    var fillWidth = newHandleXPosition - firstXHandleXPosition;
    secondHandleValue.current = handleValue;
    if (onChangeCommitted) {
      onChangeCommitted([value[0], handleValue]);
    }
    setSecondValue(handleValue);
    setState(function (prevState) {
      return _objectSpread(_objectSpread({}, prevState), {}, {
        secondValue: handleValue,
        xSecondHandle: data.x,
        railFillWidth: fillWidth < 0 ? 0 : fillWidth,
        railFillXPosition: firstXHandleXPosition
      });
    });
  };
  var onSecondTextfieldChange = function onSecondTextfieldChange(event) {
    if (!(0, _utils2.isNumber)(event.target.value)) {
      return;
    }
    var handleValue = Number(event.target.value);
    setSecondValue(handleValue);
    if (onChangeTextField) {
      onChangeTextField([firstValue, handleValue], event);
    }
  };
  var onSecondTextfieldBlur = function onSecondTextfieldBlur(event) {
    if (!(0, _utils2.isNumber)(event.target.value)) {
      return;
    }
    var handleValue = Number(event.target.value);
    setSecondValue(handleValue);
    if (onBlurTextField) {
      onBlurTextField([firstValue, handleValue], event);
    }
  };
  var onTextfieldKeyDown = function onTextfieldKeyDown(event) {
    if (onKeyDownTextField) {
      onKeyDownTextField([firstValue, secondValue], event);
    }
  };
  var _ref2 = ariaLabel || [],
    _ref3 = _slicedToArray(_ref2, 2),
    ariaLabelLeft = _ref3[0],
    ariaLabelRight = _ref3[1];
  var currentFirstSliderValue = Math.max(state.firstValue, min);
  return /*#__PURE__*/_react["default"].createElement(_Double.DoubleWrapper, null, hasLabelContent && /*#__PURE__*/_react["default"].createElement(_Double.LabelWrapper, null, labelContentLeft && /*#__PURE__*/_react["default"].createElement(_Double.LabelContentLeft, null, labelContentLeft), label && /*#__PURE__*/_react["default"].createElement(_Double.Label, null, label)), /*#__PURE__*/_react["default"].createElement(_Double.SliderWrapper, null, /*#__PURE__*/_react["default"].createElement(_SliderBase.SliderBase, _extends({
    min: min,
    max: max,
    disabled: disabled,
    setStepSize: setStepSize,
    railFillWidth: state.railFillWidth,
    settings: _utils.sizeData[size],
    railFillXPosition: state.railFillXPosition
  }, rest), /*#__PURE__*/_react["default"].createElement(_ui.Handler, {
    ref: firstHandleRef,
    stepSize: state.stepSize,
    multipleStepSize: multipleStepSize,
    onChangeCommitted: onFirstHandleChangeCommitted,
    onChange: onFirstHandleChange,
    min: min,
    max: state.secondValue,
    disabled: disabled,
    bounds: [min, state.secondValue],
    side: "left",
    xPosition: state.xFirstHandle,
    zIndex: state.firstHandleZIndex,
    value: currentFirstSliderValue,
    ariaLabel: ariaLabelLeft,
    onMouseEnter: function onMouseEnter() {
      return setFirstInputActive(true);
    },
    onMouseLeave: function onMouseLeave() {
      return setFirstInputActive(false);
    }
  }), /*#__PURE__*/_react["default"].createElement(_ui.Handler, {
    ref: secondHandleRef,
    stepSize: state.stepSize,
    multipleStepSize: multipleStepSize,
    onChangeCommitted: onSecondHandleChangeCommitted,
    onChange: onSecondHandleChange,
    min: min,
    ariaValueMin: currentFirstSliderValue,
    max: max,
    disabled: disabled,
    bounds: [state.firstValue, max],
    side: "right",
    xPosition: state.xSecondHandle,
    zIndex: state.secondHandleZIndex,
    value: Math.max(state.secondValue, min),
    ariaLabel: ariaLabelRight,
    onMouseEnter: function onMouseEnter() {
      return setSecondInputActive(true);
    },
    onMouseLeave: function onMouseLeave() {
      return setSecondInputActive(false);
    }
  })), /*#__PURE__*/_react["default"].createElement(_Double.InputsWrapper, null, /*#__PURE__*/_react["default"].createElement(_Double.StyledInput, {
    className: (0, _utils2.cx)(_Slider.classes.firstTextField, firstInputActiveClass),
    enumerationType: "plain",
    disabled: disabled,
    value: firstValue,
    onChange: onFirstTextfieldChange,
    onBlur: onFirstTextfieldBlur,
    onKeyDown: onTextfieldKeyDown
  }), /*#__PURE__*/_react["default"].createElement(_Double.StyledInput, {
    className: (0, _utils2.cx)(_Slider.classes.secondTextField, secondInputActiveClass),
    enumerationType: "plain",
    disabled: disabled,
    value: secondValue,
    onChange: onSecondTextfieldChange,
    onBlur: onSecondTextfieldBlur,
    onKeyDown: onTextfieldKeyDown
  }))));
};
