// @salutejs/plasma-new-hope/styled-components/es/examples/plasma_web/components/Switch/Switch.config.js
import { css } from 'styled-components';
import { switchTokens } from '../../../../components/Switch';
import { labelPosition_Before } from '../../../../components/Switch/_label-position/before';
import { labelPosition_After } from '../../../../components/Switch/_label-position/after';
export var config = {
  defaults: {
    view: 'default',
    size: 'm',
    labelPosition: 'before',
    focused: 'true'
  },
  variations: {
    size: {
      m: /*#__PURE__*/css(["", ":var(--plasma-typo-body1-font-family);", ":var(--plasma-typo-body1-font-style);", ":var(--plasma-typo-body1-font-weight);", ":var(--plasma-typo-body1-font-weight);", ":var(--plasma-typo-body1-line-height);", ":var(--plasma-typo-body1-font-size);", ":var(--plasma-typo-body1-letter-spacing);", ":2.75rem;", ":1.75rem;", ":calc(var(", ") / 2);", ":1.5rem;", ":calc(var(", ") / 2);", ":0.125rem;", ":1.25;", ":0.75rem;"], switchTokens.fontFamily, switchTokens.fontStyle, switchTokens.fontWeight, switchTokens.fontWeight, switchTokens.lineHeight, switchTokens.fontSize, switchTokens.letterSpacing, switchTokens.trackWidth, switchTokens.trackHeight, switchTokens.trackBorderRadius, switchTokens.trackHeight, switchTokens.thumbSize, switchTokens.thumbBorderRadius, switchTokens.thumbSize, switchTokens.thumbOffset, switchTokens.thumbPressScale, switchTokens.labelOffset)
    },
    view: {
      "default": /*#__PURE__*/css(["", ":var(--surface-accent);", ":var(--surface-transparent-tertiary-active);", ":var(--on-dark-surface-solid-default);", ":0 1px 1px rgba(0,0,0,0.11);"], switchTokens.trackBackgroundColorOn, switchTokens.trackBackgroundColorOff, switchTokens.thumbBackgroundColor, switchTokens.thumbBoxShadow)
    },
    labelPosition: {
      before: labelPosition_Before,
      after: labelPosition_After
    },
    disabled: {
      "true": /*#__PURE__*/css(["", ":0.4;"], switchTokens.disabledOpacity)
    },
    focused: {
      "true": /*#__PURE__*/css(["", ":var(--surface-accent);"], switchTokens.trackFocusColor)
    }
  }
};
