// @salutejs/plasma-new-hope/styled-components/cjs/examples/typograpy/components/Old/Display/Display.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _tokens = /*#__PURE__*/require("../../../../../components/Typography/tokens");
var config = exports.config = {
  defaults: {
    size: 'display1'
  },
  variations: {
    size: {
      display1: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--plasma-typo-display1-font-family);", ":var(--plasma-typo-display1-font-size);", ":var(--plasma-typo-display1-font-style);", ":var(--plasma-typo-display1-font-weight);", ":var(--plasma-typo-display1-letter-spacing);", ":var(--plasma-typo-display1-line-height);"], _tokens.tokens.typoFontFamily, _tokens.tokens.typoFontSize, _tokens.tokens.typoFontStyle, _tokens.tokens.typoFontWeight, _tokens.tokens.typoFontLetterSpacing, _tokens.tokens.typoFontLineHeight),
      display2: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--plasma-typo-display2-font-family);", ":var(--plasma-typo-display2-font-size);", ":var(--plasma-typo-display2-font-style);", ":var(--plasma-typo-display2-font-weight);", ":var(--plasma-typo-display2-letter-spacing);", ":var(--plasma-typo-display2-line-height);"], _tokens.tokens.typoFontFamily, _tokens.tokens.typoFontSize, _tokens.tokens.typoFontStyle, _tokens.tokens.typoFontWeight, _tokens.tokens.typoFontLetterSpacing, _tokens.tokens.typoFontLineHeight),
      display3: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--plasma-typo-display3-font-family);", ":var(--plasma-typo-display3-font-size);", ":var(--plasma-typo-display3-font-style);", ":var(--plasma-typo-display3-font-weight);", ":var(--plasma-typo-display3-letter-spacing);", ":var(--plasma-typo-display3-line-height);"], _tokens.tokens.typoFontFamily, _tokens.tokens.typoFontSize, _tokens.tokens.typoFontStyle, _tokens.tokens.typoFontWeight, _tokens.tokens.typoFontLetterSpacing, _tokens.tokens.typoFontLineHeight)
    }
  }
};
