// @salutejs/plasma-ui/es/components/Marquee/Marquee.js
var _excluded = ["textAlign", "children"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React, { useRef, useState } from 'react';
import styled, { css, keyframes } from 'styled-components';
import { useIsomorphicLayoutEffect } from '@salutejs/plasma-core';
import { collectPackageInfo } from '../../collectPackageInfo';
collectPackageInfo('Marquee/Marquee');
var marquee = /*#__PURE__*/keyframes(["0%{transform:translateX(0%)}100%{transform:translateX(-100%)}"]);
var MarqueeText = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-ui__sc-hrzlgo-0"
})(["padding-right:4rem;animation:", ";animation-delay:500ms;animation-duration:", "s;"], function (_ref) {
  var isPlaying = _ref.isPlaying;
  return isPlaying && css(["", " linear infinite;"], marquee);
}, function (_ref2) {
  var duration = _ref2.duration;
  return duration;
});
var Wrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-ui__sc-hrzlgo-1"
})(["display:flex;justify-content:", ";justify-content:", ";justify-content:", ";"], function (_ref3) {
  var textAlign = _ref3.textAlign;
  return textAlign === 'start' && css(["flex-start;"]);
}, function (_ref4) {
  var textAlign = _ref4.textAlign;
  return textAlign === 'center' && css(["center;"]);
}, function (_ref5) {
  var textAlign = _ref5.textAlign;
  return textAlign === 'end' && css(["flex-end;"]);
});
var StyledText = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-ui__sc-hrzlgo-2"
})(["white-space:nowrap;"]);
/**
 * Компонент для отображения бегущей строки
 */
export var Marquee = function Marquee(_ref6) {
  var textAlign = _ref6.textAlign,
    children = _ref6.children,
    rest = _objectWithoutProperties(_ref6, _excluded);
  var animationSpeed = 70;
  var textRef = useRef(null);
  var wrapperRef = useRef(null);
  var _useState = useState({
      isPlaying: false,
      animationDuration: 0
    }),
    _useState2 = _slicedToArray(_useState, 2),
    state = _useState2[0],
    setState = _useState2[1];
  useIsomorphicLayoutEffect(function () {
    var _wrapperRef$current, _textRef$current;
    var wrapperWidth = ((_wrapperRef$current = wrapperRef.current) === null || _wrapperRef$current === void 0 ? void 0 : _wrapperRef$current.getBoundingClientRect().width) || 0;
    var textWidth = ((_textRef$current = textRef.current) === null || _textRef$current === void 0 ? void 0 : _textRef$current.getBoundingClientRect().width) || 0;
    setState({
      isPlaying: wrapperWidth < textWidth,
      animationDuration: textWidth / animationSpeed
    });
  }, [children]);
  var textDiv = /*#__PURE__*/React.createElement(StyledText, {
    ref: textRef
  }, children);
  var marqueeText = /*#__PURE__*/React.createElement(MarqueeText, {
    isPlaying: state.isPlaying,
    duration: state.animationDuration
  }, textDiv);
  return /*#__PURE__*/React.createElement(Wrapper, _extends({
    ref: wrapperRef,
    textAlign: !state.isPlaying ? textAlign : undefined
  }, rest), state.isPlaying ? /*#__PURE__*/React.createElement(React.Fragment, null, marqueeText, marqueeText) : textDiv);
};
