// @salutejs/plasma-new-hope/styled-components/cjs/utils/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "IS_REACT_18", {
  enumerable: true,
  get: function get() {
    return _react.IS_REACT_18;
  }
});
Object.defineProperty(exports, "canUseDOM", {
  enumerable: true,
  get: function get() {
    return _canUseDOM.canUseDOM;
  }
});
exports.cx = exports.composableStyle = void 0;
Object.defineProperty(exports, "extractTextFrom", {
  enumerable: true,
  get: function get() {
    return _extractTextFrom.extractTextFrom;
  }
});
Object.defineProperty(exports, "getSizeValueFromProp", {
  enumerable: true,
  get: function get() {
    return _getSizeValueFromProp.getSizeValueFromProp;
  }
});
Object.defineProperty(exports, "isNumber", {
  enumerable: true,
  get: function get() {
    return _isNumber.isNumber;
  }
});
Object.defineProperty(exports, "safeUseId", {
  enumerable: true,
  get: function get() {
    return _react.safeUseId;
  }
});
var _canUseDOM = /*#__PURE__*/require("./canUseDOM");
var _extractTextFrom = /*#__PURE__*/require("./extractTextFrom");
var _getSizeValueFromProp = /*#__PURE__*/require("./getSizeValueFromProp");
var _react = /*#__PURE__*/require("./react");
var _isNumber = /*#__PURE__*/require("./isNumber");
var cx = exports.cx = function cx() {
  for (var _len = arguments.length, classes = new Array(_len), _key = 0; _key < _len; _key++) {
    classes[_key] = arguments[_key];
  }
  return classes.filter(function (classItem) {
    return classItem;
  }).join(' ');
};
var composableStyle = exports.composableStyle = function composableStyle(s) {
  var res = '';
  for (var i = 0; i < Math.max(s.length, arguments.length <= 1 ? 0 : arguments.length - 1); ++i) {
    var _s$i, _ref;
    res += (_s$i = s[i]) !== null && _s$i !== void 0 ? _s$i : '';
    res += (_ref = i + 1 < 1 || arguments.length <= i + 1 ? undefined : arguments[i + 1]) !== null && _ref !== void 0 ? _ref : '';
  }
  return res;
};
