// @salutejs/plasma-new-hope/cjs/components/Combobox/Combobox.styles.js
import './Combobox.styles_dw6hnk.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
require('@linaria/core');
var common = require('../../engines/common.js');
var Popover$1 = require('../Popover/Popover.js');

var StyledRoot = /*#__PURE__*/react.styled('div')({
  name: "StyledRoot",
  "class": "sd7salz",
  propsAsIs: false
});
var StyledCombobox = /*#__PURE__*/react.styled('div')({
  name: "StyledCombobox",
  "class": "s1tp4xgb",
  propsAsIs: false
});

// issue #823
var Popover = /*#__PURE__*/common.component(Popover$1.popoverConfig);
var _exp11 = function _exp11() {
  return Popover;
};
var _exp14 = function _exp14() {
  return function (_ref) {
    var usePortal = _ref.usePortal,
      selectWidth = _ref.selectWidth;
    return usePortal ? "".concat(selectWidth, "px") : '100%';
  };
};
var StyledPopover = /*#__PURE__*/react.styled(_exp11())({
  name: "StyledPopover",
  "class": "s1fuzbs4",
  propsAsIs: true,
  vars: {
    "s1fuzbs4-0": [/*#__PURE__*/_exp14()]
  }
});
var ComboboxItemNotFound = /*#__PURE__*/react.styled('div')({
  name: "ComboboxItemNotFound",
  "class": "c1bkz6yw",
  propsAsIs: false
});
var StyledNativeSelect = /*#__PURE__*/react.styled('select')({
  name: "StyledNativeSelect",
  "class": "s1mwm6am",
  propsAsIs: false
});

exports.ComboboxItemNotFound = ComboboxItemNotFound;
exports.StyledCombobox = StyledCombobox;
exports.StyledNativeSelect = StyledNativeSelect;
exports.StyledPopover = StyledPopover;
exports.StyledRoot = StyledRoot;
