// @salutejs/plasma-new-hope/cjs/components/Calendar/store/reducer.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../_virtual/_rollupPluginBabelHelpers.js');
var index = require('../utils/index.js');
var types = require('./types.js');

var getInitialState = function getInitialState(value, size, calendarState) {
  var initDate = value || new Date();
  var date = index.getDateFromValue(initDate);
  return {
    date: _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, date), {}, {
      day: value !== undefined ? date.day : 0
    }),
    startYear: index.getStartYear(date.year),
    calendarState: calendarState,
    size: size
  };
};
var reducer = function reducer(state, action) {
  switch (action.type) {
    case types.ActionType.PREVIOUS_MONTH:
      {
        var _action$payload = action.payload,
          year = _action$payload.year,
          monthIndex = _action$payload.monthIndex;
        var _getPrevDate = index.getPrevDate(year, monthIndex),
          _getPrevDate2 = _rollupPluginBabelHelpers.slicedToArray(_getPrevDate, 2),
          prevYear = _getPrevDate2[0],
          prevMonthIndex = _getPrevDate2[1];
        return _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, state), {}, {
          date: {
            day: state.date.day,
            monthIndex: prevMonthIndex,
            year: prevYear
          }
        });
      }
    case types.ActionType.NEXT_MONTH:
      {
        var _action$payload2 = action.payload,
          _monthIndex = _action$payload2.monthIndex,
          _year = _action$payload2.year;
        var _getNextDate = index.getNextDate(_year, _monthIndex),
          _getNextDate2 = _rollupPluginBabelHelpers.slicedToArray(_getNextDate, 2),
          nextYear = _getNextDate2[0],
          nextMonthIndex = _getNextDate2[1];
        return _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, state), {}, {
          date: {
            day: state.date.day,
            monthIndex: nextMonthIndex,
            year: nextYear
          }
        });
      }
    case types.ActionType.PREVIOUS_YEAR:
      {
        var step = action.payload.step;
        return _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, state), {}, {
          startYear: state.startYear - step,
          date: {
            day: state.date.day,
            monthIndex: state.date.monthIndex,
            year: state.date.year - step
          }
        });
      }
    case types.ActionType.NEXT_YEAR:
      {
        var _step = action.payload.step;
        return _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, state), {}, {
          startYear: state.startYear + _step,
          date: {
            day: state.date.day,
            monthIndex: state.date.monthIndex,
            year: state.date.year + _step
          }
        });
      }
    case types.ActionType.PREVIOUS_START_YEAR:
      {
        var yearsCount = action.payload.yearsCount;
        return _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, state), {}, {
          startYear: state.startYear - yearsCount
        });
      }
    case types.ActionType.NEXT_START_YEAR:
      {
        var _yearsCount = action.payload.yearsCount;
        return _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, state), {}, {
          startYear: state.startYear + _yearsCount
        });
      }
    case types.ActionType.UPDATE_DATE:
      {
        var value = action.payload.value;
        var date = index.getDateFromValue(value);
        var startYear = index.getStartYear(value.getFullYear());
        return _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, state), {}, {
          date: date,
          startYear: startYear
        });
      }
    case types.ActionType.UPDATE_MONTH:
      {
        var _action$payload3 = action.payload,
          calendarState = _action$payload3.calendarState,
          _monthIndex2 = _action$payload3.monthIndex,
          size = _action$payload3.size;
        return _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, state), {}, {
          size: size,
          calendarState: calendarState,
          date: {
            day: state.date.day,
            monthIndex: _monthIndex2,
            year: state.date.year
          }
        });
      }
    case types.ActionType.UPDATE_YEAR:
      {
        var _action$payload4 = action.payload,
          _calendarState = _action$payload4.calendarState,
          _year2 = _action$payload4.year;
        return _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, state), {}, {
          calendarState: _calendarState,
          date: {
            day: state.date.day,
            monthIndex: state.date.monthIndex,
            year: _year2
          }
        });
      }
    case types.ActionType.UPDATE_CALENDAR_STATE:
      {
        var _action$payload5 = action.payload,
          _calendarState2 = _action$payload5.calendarState,
          _size = _action$payload5.size;
        return _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, state), {}, {
          calendarState: _calendarState2,
          size: _size || state.size
        });
      }
    default:
      return state;
  }
};

exports.getInitialState = getInitialState;
exports.reducer = reducer;
