// @salutejs/plasma-new-hope/styled-components/cjs/components/Notification/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "NotificationsProvider", {
  enumerable: true,
  get: function get() {
    return _NotificationsProvider.NotificationsProvider;
  }
});
Object.defineProperty(exports, "addNotification", {
  enumerable: true,
  get: function get() {
    return _NotificationsStore.addNotification;
  }
});
Object.defineProperty(exports, "closeNotification", {
  enumerable: true,
  get: function get() {
    return _NotificationsStore.closeNotification;
  }
});
Object.defineProperty(exports, "noticationConfig", {
  enumerable: true,
  get: function get() {
    return _Notification.noticationConfig;
  }
});
Object.defineProperty(exports, "notificationClasses", {
  enumerable: true,
  get: function get() {
    return _Notification2.classes;
  }
});
Object.defineProperty(exports, "notificationRoot", {
  enumerable: true,
  get: function get() {
    return _Notification.notificationRoot;
  }
});
Object.defineProperty(exports, "notificationTokens", {
  enumerable: true,
  get: function get() {
    return _Notification2.tokens;
  }
});
var _NotificationsProvider = /*#__PURE__*/require("./NotificationsProvider");
var _Notification = /*#__PURE__*/require("./Notification");
var _NotificationsStore = /*#__PURE__*/require("./NotificationsStore");
var _Notification2 = /*#__PURE__*/require("./Notification.tokens");
