// @salutejs/plasma-new-hope/styled-components/cjs/components/Notification/Notification.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.notificationRoot = exports.noticationConfig = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _utils = /*#__PURE__*/require("../../utils");
var _IconCross2 = /*#__PURE__*/require("../_Icon/Icons/IconCross");
var _Notification = /*#__PURE__*/require("./Notification.tokens");
var _base = /*#__PURE__*/require("./variations/_view/base");
var _base2 = /*#__PURE__*/require("./variations/_layout/base");
var _base3 = /*#__PURE__*/require("./variations/_size/base");
var _Notification2 = /*#__PURE__*/require("./Notification.types");
var _Notification3 = /*#__PURE__*/require("./Notification.styles");
var _utils2 = /*#__PURE__*/require("./utils");
var _IconCross;
var _excluded = ["role", "title", "children", "actions", "view", "size", "iconPlacement", "showCloseIcon", "layout", "icon", "onCloseButtonClick"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
/**
 * Компонент для небольших уведомлений пользователя
 */
var notificationRoot = exports.notificationRoot = function notificationRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
    var _props$role = props.role,
      role = _props$role === void 0 ? 'status' : _props$role,
      title = props.title,
      content = props.children,
      actions = props.actions,
      view = props.view,
      size = props.size,
      iconPlacement = props.iconPlacement,
      _props$showCloseIcon = props.showCloseIcon,
      showCloseIcon = _props$showCloseIcon === void 0 ? true : _props$showCloseIcon,
      _props$layout = props.layout,
      layout = _props$layout === void 0 ? _Notification2.layouts.vertical : _props$layout,
      icon = props.icon,
      onCloseButtonClick = props.onCloseButtonClick,
      rest = _objectWithoutProperties(props, _excluded);
    var ariaLive = 'polite';
    var ariaAtomic = false;
    if (role === 'alert') {
      ariaLive = 'assertive';
    } else if (role === 'status') {
      ariaAtomic = true;
    }
    var isOneLine = !content || !title;
    var oneLineClass = isOneLine ? _Notification.classes.oneLine : undefined;
    var withoutIconClass = icon ? undefined : _Notification.classes.withoutIcon;
    var withoutCloseIconClass = showCloseIcon ? undefined : _Notification.classes.withoutCloseIcon;
    var IconPlacementInternal = (0, _react.useMemo)(function () {
      return icon ? iconPlacement : undefined;
    }, [icon, iconPlacement]);
    return /*#__PURE__*/_react["default"].createElement(Root, _extends({
      view: view,
      size: size,
      layout: layout,
      ref: ref,
      role: role,
      "aria-live": ariaLive,
      "aria-atomic": ariaAtomic
    }, rest), /*#__PURE__*/_react["default"].createElement(_Notification3.Wrapper, {
      className: (0, _utils.cx)(_Notification.classes.wrapper, (0, _utils2.getLayoutClass)(layout), oneLineClass, withoutCloseIconClass)
    }, /*#__PURE__*/_react["default"].createElement(_Notification3.ContentBox, {
      iconPlacement: IconPlacementInternal,
      className: (0, _utils.cx)(_Notification.classes.contentBox, (0, _utils2.getLayoutClass)(layout), withoutIconClass)
    }, icon && /*#__PURE__*/_react["default"].createElement(_Notification3.IconWrapper, {
      iconPlacement: IconPlacementInternal,
      className: (0, _utils.cx)(_Notification.classes.icon, (0, _utils2.getLayoutClass)(layout))
    }, icon), /*#__PURE__*/_react["default"].createElement(_Notification3.TextBox, {
      iconPlacement: IconPlacementInternal,
      showCloseIcon: showCloseIcon,
      className: (0, _utils.cx)(_Notification.classes.textbox, (0, _utils2.getLayoutClass)(layout))
    }, title && /*#__PURE__*/_react["default"].createElement(_Notification3.StyledTitle, {
      className: _Notification.classes.title
    }, title), content && /*#__PURE__*/_react["default"].createElement(_Notification3.StyledContent, {
      className: _Notification.classes.text
    }, content))), actions && /*#__PURE__*/_react["default"].createElement(_Notification3.ButtonsWrapper, {
      iconPlacement: IconPlacementInternal,
      className: (0, _utils.cx)(_Notification.classes.buttonsWrapper, (0, _utils2.getLayoutClass)(layout))
    }, actions), showCloseIcon && /*#__PURE__*/_react["default"].createElement(_Notification3.CloseIconWrapper, {
      view: "clear",
      size: "s",
      onClick: onCloseButtonClick,
      className: (0, _utils.cx)(_Notification.classes.closeIcon, (0, _utils2.getLayoutClass)(layout))
    }, _IconCross || (_IconCross = /*#__PURE__*/_react["default"].createElement(_IconCross2.IconCross, {
      size: "s",
      color: "inherit"
    })))));
  });
};
var noticationConfig = exports.noticationConfig = {
  name: 'Notification',
  tag: 'div',
  layout: notificationRoot,
  base: _Notification3.StyledNotification,
  variations: {
    layout: {
      css: _base2.base
    },
    view: {
      css: _base.base
    },
    size: {
      css: _base3.base
    }
  },
  defaults: {
    view: 'default',
    layout: _Notification2.layouts.vertical,
    size: 'xs'
  }
};
