// @salutejs/plasma-new-hope/styled-components/cjs/components/Notification/NotificationsPortal.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NotificationsPortal = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _react2 = /*#__PURE__*/require("storeon/react");
var _Popup = /*#__PURE__*/require("../Popup");
var _engines = /*#__PURE__*/require("../../engines");
var _utils = /*#__PURE__*/require("../../utils");
var _NotificationsStore = /*#__PURE__*/require("./NotificationsStore");
var _Notification = /*#__PURE__*/require("./Notification.styles");
var _Notification2 = /*#__PURE__*/require("./Notification.tokens");
var _excluded = ["id", "isHidden"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
// issue #823
var Popup = /*#__PURE__*/(0, _engines.component)(_Popup.popupConfig);

/**
 * Обертка для визуального представления уведомлений.
 */
var NotificationsPortal = exports.NotificationsPortal = function NotificationsPortal(_ref) {
  var config = _ref.config,
    frame = _ref.frame;
  var _useStoreon = (0, _react2.useStoreon)('notifications'),
    notifications = _useStoreon.notifications;
  var Notification = (0, _react.useMemo)(function () {
    return (0, _engines.component)(config);
  }, []);
  return /*#__PURE__*/_react["default"].createElement(_Popup.PopupProvider, null, notifications.length > 0 && /*#__PURE__*/_react["default"].createElement(Popup, {
    isOpen: true,
    frame: frame,
    placement: "bottom-right",
    zIndex: "9100"
  }, /*#__PURE__*/_react["default"].createElement(_Notification.StyledRoot, null, notifications.map(function (_ref2) {
    var id = _ref2.id,
      isHidden = _ref2.isHidden,
      rest = _objectWithoutProperties(_ref2, _excluded);
    return /*#__PURE__*/_react["default"].createElement(_Notification.StyledItemWrapper, {
      key: id,
      className: (0, _utils.cx)(isHidden ? _Notification2.classes.notificationItemHidden : _Notification2.classes.notificationItemOpened),
      isHidden: isHidden || false
    }, /*#__PURE__*/_react["default"].createElement(Notification, _extends({
      key: id,
      id: id,
      onCloseButtonClick: function onCloseButtonClick() {
        return (0, _NotificationsStore.closeNotification)(id);
      }
    }, rest)));
  }))));
};
