// @salutejs/plasma-tokens/themesValues/salutejs_joy__light.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.overlaySoft = exports.overlay = exports.speechBubbleReceived = exports.speechBubbleSent = exports.skeletonGradientLighter = exports.skeletonGradient = exports.buttonChecked = exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonSecondary = exports.buttonPrimary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.critical = exports.warning = exports.success = exports.backgroundTertiary = exports.backgroundSecondary = exports.backgroundPrimary = exports.background = exports.inverse = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackTransparent = exports.buttonBlackSecondary = exports.buttonBlack = exports.buttonClear = exports.transparent = exports.dark03 = exports.dark02 = exports.dark01 = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.black = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.white = void 0;
exports.voicePhraseGradient = exports.gradientDevice = exports.gradient = exports.buttonFocused = exports.buttonAccent = exports.accent = exports.inverseOverlayBlur = exports.inverseOverlayHard = exports.inverseOverlaySoft = exports.lightOverlayBlur = exports.lightOverlayHard = exports.lightOverlaySoft = exports.darkOverlayBlur = exports.darkOverlayHard = exports.darkOverlaySoft = exports.overlayBlur = exports.overlayHard = void 0;
/** Базовый белый цвет совпадает с whitePrimary */
exports.white = '#FFFFFF';
/** Первичный белый цвет */
exports.whitePrimary = '#FFFFFF';
/** Вторичный белый цвет */
exports.whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный белый цвет */
exports.whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** Базовый черный цвет совпадает с blackPrimary */
exports.black = '#080808';
/** Первичный черный цвет */
exports.blackPrimary = '#080808';
/** Вторичный черныйцвет */
exports.blackSecondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный черный цвет */
exports.blackTertiary = 'rgba(8, 8, 8, 0.28)';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark01 = '#171717';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark02 = '#232323';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark03 = '#363636';
/** Прозрачный цвет */
exports.transparent = 'rgba(0, 0, 0, 0)';
/** Второстепенная прозрачная */
exports.buttonClear = 'rgba(0, 0, 0, 0)';
/** Дефолтная чёрная, не зависит от темы */
exports.buttonBlack = '#080808';
/** Второстепенная чёрная, не зависит от темы */
exports.buttonBlackSecondary = 'rgba(8, 8, 8, 0.12)';
/** Чёрная для использования поверх картинок */
exports.buttonBlackTransparent = 'rgba(8, 8, 8, 0.56)';
/** Дефолтная белая, не зависит от темы */
exports.buttonWhite = '#FFFFFF';
/** Второстепенная белая, не зависит от темы */
exports.buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** Базовый цвет текста, совпадает с primary */
exports.text = '#080808';
/** Основной текст в интерфейсе */
exports.primary = '#080808';
/** Второстепенный текст */
exports.secondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный цвет текста */
exports.tertiary = 'rgba(8, 8, 8, 0.28)';
/** Сплошной наборный текст */
exports.paragraph = 'rgba(8, 8, 8, 0.8)';
/** Белый в светлой теме / Чёрный в тёмной теме */
exports.inverse = 'rgba(255, 255, 255, 0.96)';
/** Основной бэкграунд всех приложений и смартапов */
exports.background = '#FFFFFF';
/** Первичный цвет фона */
exports.backgroundPrimary = '#FFFFFF';
/** Вторичный цвет фона */
exports.backgroundSecondary = '#FFFFFF';
/** Третичный цвет фона */
exports.backgroundTertiary = '#FFFFFF';
/** Обозначение успешного сценария */
exports.success = '#0D8523';
/** Цвет предупреждения */
exports.warning = '#D14D00';
/** Цвет ошибки */
exports.critical = '#E31227';
/** Полупрозрачные поверхности */
exports.surfaceLiquid01 = 'rgba(8, 8, 8, 0.02)';
/** Полупрозрачные поверхности */
exports.surfaceLiquid02 = 'rgba(8, 8, 8, 0.06)';
/** Полупрозрачные поверхности */
exports.surfaceLiquid03 = 'rgba(8, 8, 8, 0.12)';
/** Непрозрачные поверхности */
exports.surfaceSolid01 = '#FAFAFA';
/** Непрозрачные поверхности */
exports.surfaceSolid02 = '#F0F0F0';
/** Непрозрачные поверхности */
exports.surfaceSolid03 = '#E1E1E1';
/** Основной фон для карточек */
exports.surfaceCard = '#FFFFFF';
/** Первичный цвет контролов */
exports.buttonPrimary = '#080808';
/** Второстепенная кнопка */
exports.buttonSecondary = 'rgba(8, 8, 8, 0.06)';
/** Кнопка для успешного сценария */
exports.buttonSuccess = '#148F2B';
/** Цвет предупреждения у контролов */
exports.buttonWarning = '#E35502';
/** Цвет ошибки у контролов */
exports.buttonCritical = '#F31B30';
/** Цвет зажатого контрола */
exports.buttonChecked = '#080808';
exports.skeletonGradient = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.09) 0%, rgba(8, 8, 8, 0.08) 6.25%, rgba(8, 8, 8, 0.05) 12.5%, rgba(8, 8, 8, 0.01) 25%, rgba(8, 8, 8, 0.05) 37.5%, rgba(8, 8, 8, 0.08) 43.75%, rgba(8, 8, 8, 0.09) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.05) 62.5%, rgba(8, 8, 8, 0.01) 75%, rgba(8, 8, 8, 0.05) 87.5%, rgba(8, 8, 8, 0.08) 93.75%, rgba(8, 8, 8, 0.09) 100% )';
exports.skeletonGradientLighter = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.36) 0%, rgba(8, 8, 8, 0.32) 6.25%, rgba(8, 8, 8, 0.20) 12.5%, rgba(8, 8, 8, 0.04) 25%, rgba(8, 8, 8, 0.20) 37.5%, rgba(8, 8, 8, 0.32) 43.75%, rgba(8, 8, 8, 0.36) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.20) 62.5%, rgba(8, 8, 8, 0.04) 75%, rgba(8, 8, 8, 0.20) 87.5%, rgba(8, 8, 8, 0.32) 93.75%, rgba(8, 8, 8, 0.36) 100% )';
/** Цвет фона баблов отправленный сообщений */
exports.speechBubbleSent = 'rgba(8, 8, 8, 0.4)';
/** Цвет фона баблов получнных сообщений */
exports.speechBubbleReceived = 'rgba(0, 0, 0, 0.02)';
/** Цвет фона паранжи */
exports.overlay = 'rgba(0, 0, 0, 0.8)';
exports.overlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.overlayHard = 'rgba(8, 8, 8, 0.9)';
exports.overlayBlur = 'rgba(35, 35, 35, 0.2)';
exports.darkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.darkOverlayHard = 'rgba(8, 8, 8, 0.9)';
exports.darkOverlayBlur = 'rgba(35, 35, 35, 0.2)';
exports.lightOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.lightOverlayHard = 'rgba(8, 8, 8, 0.9)';
exports.lightOverlayBlur = 'rgba(35, 35, 35, 0.2)';
exports.inverseOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.inverseOverlayHard = 'rgba(8, 8, 8, 0.9)';
exports.inverseOverlayBlur = 'rgba(35, 35, 35, 0.2)';
/** Акцентный цвет бренда */
exports.accent = '#A039E5';
/** Акцентный цвет у контролов */
exports.buttonAccent = '#AD42F5';
/** Цвет рамки фокуса у контрола */
exports.buttonFocused = '#080808';
/** Градиент для заливки основного фона */
exports.gradient = 'linear-gradient(336.9deg, rgba(255, 200, 3, 0.05) 0%, rgba(255, 255, 255, 0) 64.95%), radial-gradient(66.53% 100% at 73.33% 100%, rgba(148, 0, 238, 0.02) 0%, rgba(160, 4, 255, 0.01) 99.69%), linear-gradient(180.03deg, rgba(255, 255, 255, 0) 50%, rgba(255, 184, 0, 0.04) 99.97%), linear-gradient(270deg, rgba(240, 0, 187, 0.03) 0%, rgba(255, 255, 255, 0) 100%)';
/** Градиент для заливки основного фона */
exports.gradientDevice = 'linear-gradient(26.05deg, rgba(255, 219, 77, 0.1) 0%, rgba(255, 255, 255, 0) 72.24%), radial-gradient(100% 100% at 0% 100%, rgba(163, 71, 255, 0.04) 0%, rgba(255, 255, 255, 0) 99.69%), radial-gradient(74.68% 149.35% at 50% 149.35%, rgba(140, 26, 255, 0.1) 0%, rgba(255, 255, 255, 0) 99.69%)';
/** Градиент подсказок о голосовых запросах */
exports.voicePhraseGradient = 'linear-gradient(45deg, hsla(30.167597765363123, 96.75675675675677%, 63.725490196078425%, 1) 0%, hsla(320.11560693641616, 100%, 66.07843137254902%, 1) 100%)';
