// @salutejs/plasma-tokens/themesValues/salutejs_joy__dark.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.overlaySoft = exports.overlay = exports.speechBubbleReceived = exports.speechBubbleSent = exports.skeletonGradientLighter = exports.skeletonGradient = exports.buttonChecked = exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonSecondary = exports.buttonPrimary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.critical = exports.warning = exports.success = exports.backgroundTertiary = exports.backgroundSecondary = exports.backgroundPrimary = exports.background = exports.inverse = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackTransparent = exports.buttonBlackSecondary = exports.buttonBlack = exports.buttonClear = exports.transparent = exports.dark03 = exports.dark02 = exports.dark01 = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.black = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.white = void 0;
exports.voicePhraseGradient = exports.gradientDevice = exports.gradient = exports.buttonFocused = exports.buttonAccent = exports.accent = exports.inverseOverlayBlur = exports.inverseOverlayHard = exports.inverseOverlaySoft = exports.lightOverlayBlur = exports.lightOverlayHard = exports.lightOverlaySoft = exports.darkOverlayBlur = exports.darkOverlayHard = exports.darkOverlaySoft = exports.overlayBlur = exports.overlayHard = void 0;
/** Базовый белый цвет совпадает с whitePrimary */
exports.white = '#FFFFFF';
/** Первичный белый цвет */
exports.whitePrimary = '#FFFFFF';
/** Вторичный белый цвет */
exports.whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный белый цвет */
exports.whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** Базовый черный цвет совпадает с blackPrimary */
exports.black = '#080808';
/** Первичный черный цвет */
exports.blackPrimary = '#080808';
/** Вторичный черныйцвет */
exports.blackSecondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный черный цвет */
exports.blackTertiary = 'rgba(8, 8, 8, 0.28)';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark01 = '#171717';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark02 = '#232323';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark03 = '#363636';
/** Прозрачный цвет */
exports.transparent = 'rgba(0, 0, 0, 0)';
/** Второстепенная прозрачная */
exports.buttonClear = 'rgba(0, 0, 0, 0)';
/** Дефолтная чёрная, не зависит от темы */
exports.buttonBlack = '#080808';
/** Второстепенная чёрная, не зависит от темы */
exports.buttonBlackSecondary = 'rgba(8, 8, 8, 0.12)';
/** Чёрная для использования поверх картинок */
exports.buttonBlackTransparent = 'rgba(8, 8, 8, 0.56)';
/** Дефолтная белая, не зависит от темы */
exports.buttonWhite = '#FFFFFF';
/** Второстепенная белая, не зависит от темы */
exports.buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** Базовый цвет текста, совпадает с primary */
exports.text = 'rgba(255, 255, 255, 0.96)';
/** Основной текст в интерфейсе */
exports.primary = 'rgba(255, 255, 255, 0.96)';
/** Второстепенный текст */
exports.secondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный цвет текста */
exports.tertiary = 'rgba(255, 255, 255, 0.28)';
/** Сплошной наборный текст */
exports.paragraph = 'rgba(255, 255, 255, 0.96)';
/** Белый в светлой теме / Чёрный в тёмной теме */
exports.inverse = '#080808';
/** Основной бэкграунд всех приложений и смартапов */
exports.background = '#080808';
/** Первичный цвет фона */
exports.backgroundPrimary = '#171717';
/** Вторичный цвет фона */
exports.backgroundSecondary = '#232323';
/** Третичный цвет фона */
exports.backgroundTertiary = '#363636';
/** Обозначение успешного сценария */
exports.success = '#24B23E';
/** Цвет предупреждения */
exports.warning = '#FA6D20';
/** Цвет ошибки */
exports.critical = '#FF4D5F';
/** Полупрозрачные поверхности */
exports.surfaceLiquid01 = 'rgba(255, 255, 255, 0.06)';
/** Полупрозрачные поверхности */
exports.surfaceLiquid02 = 'rgba(255, 255, 255, 0.12)';
/** Полупрозрачные поверхности */
exports.surfaceLiquid03 = 'rgba(255, 255, 255, 0.2)';
/** Непрозрачные поверхности */
exports.surfaceSolid01 = '#171717';
/** Непрозрачные поверхности */
exports.surfaceSolid02 = '#232323';
/** Непрозрачные поверхности */
exports.surfaceSolid03 = '#363636';
/** Основной фон для карточек */
exports.surfaceCard = 'rgba(255, 255, 255, 0.12)';
/** Первичный цвет контролов */
exports.buttonPrimary = '#FFFFFF';
/** Второстепенная кнопка */
exports.buttonSecondary = 'rgba(255, 255, 255, 0.12)';
/** Кнопка для успешного сценария */
exports.buttonSuccess = '#21A038';
/** Цвет предупреждения у контролов */
exports.buttonWarning = '#E35502';
/** Цвет ошибки у контролов */
exports.buttonCritical = '#FF2E43';
/** Цвет зажатого контрола */
exports.buttonChecked = '#FFFFFF';
exports.skeletonGradient = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% )';
exports.skeletonGradientLighter = 'linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% )';
/** Цвет фона баблов отправленный сообщений */
exports.speechBubbleSent = 'rgba(0, 0, 0, 0.28)';
/** Цвет фона баблов получнных сообщений */
exports.speechBubbleReceived = 'rgba(255, 255, 255, 0.12)';
/** Цвет фона паранжи */
exports.overlay = 'rgba(0, 0, 0, 0.8)';
exports.overlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.overlayHard = 'rgba(8, 8, 8, 0.9)';
exports.overlayBlur = 'rgba(23, 23, 23, 0.28)';
exports.darkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.darkOverlayHard = 'rgba(8, 8, 8, 0.9)';
exports.darkOverlayBlur = 'rgba(23, 23, 23, 0.28)';
exports.lightOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.lightOverlayHard = 'rgba(8, 8, 8, 0.9)';
exports.lightOverlayBlur = 'rgba(23, 23, 23, 0.28)';
exports.inverseOverlaySoft = 'rgba(8, 8, 8, 0.56)';
exports.inverseOverlayHard = 'rgba(8, 8, 8, 0.9)';
exports.inverseOverlayBlur = 'rgba(23, 23, 23, 0.28)';
/** Акцентный цвет бренда */
exports.accent = '#C370FA';
/** Акцентный цвет у контролов */
exports.buttonAccent = '#B559F3';
/** Цвет рамки фокуса у контрола */
exports.buttonFocused = 'rgba(255, 255, 255, 0.96)';
/** Градиент для заливки основного фона */
exports.gradient = 'linear-gradient(336.9deg, rgba(255, 156, 101, 0.24) 0%, rgba(8, 8, 8, 0) 64.95%), radial-gradient(66.53% 100% at 73.33% 100%, rgba(61, 19, 149, 0.34) 0%, rgba(19, 24, 149, 0.1) 99.69%), linear-gradient(180.03deg, rgba(8, 8, 8, 0) 50%, rgba(255, 215, 179, 0.15) 99.97%), linear-gradient(270deg, rgba(107, 15, 87, 0.2) 0%, rgba(8, 8, 8, 0) 100%)';
/** Градиент для заливки основного фона */
exports.gradientDevice = 'linear-gradient(26.05deg, rgba(26, 140, 255, 0.16) 0%, rgba(8, 8, 8, 0) 72.24%), radial-gradient(100% 100% at 0% 100%, rgba(0, 85, 255, 0.28) 0%, rgba(8, 8, 8, 0) 99.69%), radial-gradient(74.68% 149.35% at 50% 149.35%, rgba(128, 0, 255, 0.48) 0%, rgba(8, 8, 8, 0) 99.69%)';
/** Градиент подсказок о голосовых запросах */
exports.voicePhraseGradient = 'linear-gradient(45deg, hsla(19.862068965517242, 100%, 71.56862745098039%, 1) 0%, hsla(319.8058252427184, 94.49541284403672%, 78.62745098039217%, 1) 100%)';
