// @salutejs/plasma-hope/es/components/TextField/views/web/Input.props.js
import { css } from 'styled-components';
import { inputBorderWidth, inputSuccessBackground, inputSuccessColor, inputSuccessCaretColor, inputSuccessBorderColor, inputWarningBackground, inputWarningColor, inputWarningCaretColor, inputWarningBorderColor, inputErrorBackground, inputErrorColor, inputErrorCaretColor, inputErrorBorderColor } from '../../../../tokens';
export var sizes = {
  l: {
    height: 'var(--plasma-input-l-height)',
    paddingY: 'var(--plasma-input-l-padding-y)',
    paddingX: 'var(--plasma-input-l-padding-x)',
    hasLabelPaddingTop: 'var(--plasma-input-l-has-label-padding-top)',
    hasLabelPaddingBottom: 'var(--plasma-input-l-has-label-padding-bottom)',
    hasContentPaddingLeft: 'var(--plasma-input-l-has-content-padding-left)',
    hasContentPaddingRight: 'var(--plasma-input-l-has-content-padding-right)'
  },
  m: {
    height: 'var(--plasma-input-m-height)',
    paddingY: 'var(--plasma-input-m-padding-y)',
    paddingX: 'var(--plasma-input-m-padding-x)',
    hasLabelPaddingTop: 'var(--plasma-input-m-padding-y)',
    hasLabelPaddingBottom: 'var(--plasma-input-m-padding-y)',
    hasContentPaddingLeft: 'var(--plasma-input-m-has-content-padding-left)',
    hasContentPaddingRight: 'var(--plasma-input-m-has-content-padding-right)'
  },
  s: {
    height: 'var(--plasma-input-s-height)',
    paddingY: 'var(--plasma-input-s-padding-y)',
    paddingX: 'var(--plasma-input-s-padding-x)',
    hasLabelPaddingTop: 'var(--plasma-input-s-padding-y)',
    hasLabelPaddingBottom: 'var(--plasma-input-s-padding-y)',
    hasContentPaddingLeft: 'var(--plasma-input-s-has-content-padding-left)',
    hasContentPaddingRight: 'var(--plasma-input-s-has-content-padding-right)'
  },
  // INFO: Временно s = xs, пока не произведен редизайн
  xs: {
    height: 'var(--plasma-input-s-height)',
    paddingY: 'var(--plasma-input-s-padding-y)',
    paddingX: 'var(--plasma-input-s-padding-x)',
    hasLabelPaddingTop: 'var(--plasma-input-s-padding-y)',
    hasLabelPaddingBottom: 'var(--plasma-input-s-padding-y)',
    hasContentPaddingLeft: 'var(--plasma-input-s-has-content-padding-left)',
    hasContentPaddingRight: 'var(--plasma-input-s-has-content-padding-right)'
  }
};
export var statuses = {
  success: /*#__PURE__*/css(["&,&:hover,&:focus,&:hover:focus{background-color:", ";color:", ";caret-color:", ";box-shadow:inset 0 0 0 ", " ", ";}"], inputSuccessBackground, inputSuccessColor, inputSuccessCaretColor, inputBorderWidth, inputSuccessBorderColor),
  warning: /*#__PURE__*/css(["&,&:hover,&:focus,&:hover:focus{background-color:", ";color:", ";caret-color:", ";box-shadow:inset 0 0 0 ", " ", ";}"], inputWarningBackground, inputWarningColor, inputWarningCaretColor, inputBorderWidth, inputWarningBorderColor),
  error: /*#__PURE__*/css(["&,&:hover,&:focus,&:hover:focus{background-color:", ";color:", ";caret-color:", ";box-shadow:inset 0 0 0 ", " ", ";}"], inputErrorBackground, inputErrorColor, inputErrorCaretColor, inputBorderWidth, inputErrorBorderColor)
};
