// @salutejs/plasma-web/es/components/TextField/TextField.perftest.js
var _IconPlaceholder, _IconPlaceholder2, _TextField;
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React from 'react';
import { IconPlaceholder } from '../../helpers';
import { TextField } from '.';
var func = function func() {};
var defaultProps = {
  id: 'example-text-field',
  size: 'l',
  label: 'Лейбл',
  animatedHint: undefined,
  placeholder: 'Заполните поле',
  helperText: 'Подсказка к полю',
  status: '',
  disabled: false,
  readOnly: false
};
var status = '';
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(TextField, _extends({}, defaultProps, {
    value: "\u0417\u043D\u0430\u0447\u0435\u043D\u0438\u0435 \u043F\u043E\u043B\u044F",
    contentLeft: _IconPlaceholder || (_IconPlaceholder = /*#__PURE__*/React.createElement(IconPlaceholder, null)),
    contentRight: _IconPlaceholder2 || (_IconPlaceholder2 = /*#__PURE__*/React.createElement(IconPlaceholder, null)),
    status: status,
    onChange: func,
    onFocus: func,
    onBlur: func,
    onSearch: func
  }));
};
export var DeferredValue = function DeferredValue() {
  return _TextField || (_TextField = /*#__PURE__*/React.createElement(TextField, {
    placeholder: "Wait three seconds...",
    defaultValue: "",
    readOnly: true
  }));
};
