// @salutejs/plasma-new-hope/es/components/Range/Range.styles.js
import './Range.styles_mc4w65.css';
import { styled } from '@linaria/react';
import '@linaria/core';
import { component, mergeConfig } from '../../engines/common.js';
import { textFieldConfig } from '../TextField/TextField.js';

var mergedConfig = /*#__PURE__*/mergeConfig(textFieldConfig);
var TextField = /*#__PURE__*/component(mergedConfig);

// NOTE: переопределение токенов TextField
var _exp = function _exp() {
  return TextField;
};
var StyledInput = /*#__PURE__*/styled(_exp())({
  name: "StyledInput",
  "class": "s1k2erwt",
  propsAsIs: true
});
var base = "b16kyj37";
var StyledLabel = /*#__PURE__*/styled('label')({
  name: "StyledLabel",
  "class": "s14q3826",
  propsAsIs: false
});
var LeftHelper = /*#__PURE__*/styled('div')({
  name: "LeftHelper",
  "class": "l1fwucr5",
  propsAsIs: false
});
var ContentWrapper = /*#__PURE__*/styled('div')({
  name: "ContentWrapper",
  "class": "czi0aji",
  propsAsIs: false
});
var StyledDivider = /*#__PURE__*/styled('div')({
  name: "StyledDivider",
  "class": "s1e6rwfy",
  propsAsIs: false
});
var StyledContentLeft = /*#__PURE__*/styled('div')({
  name: "StyledContentLeft",
  "class": "sfxtxl2",
  propsAsIs: false
});
var StyledContentRight = /*#__PURE__*/styled('div')({
  name: "StyledContentRight",
  "class": "s1hiyoiq",
  propsAsIs: false
});

export { ContentWrapper, LeftHelper, StyledContentLeft, StyledContentRight, StyledDivider, StyledInput, StyledLabel, base };
