// @salutejs/plasma-hope/es/components/Button/Button.props.js
import { css } from 'styled-components';
import { buttonViews as baseViews } from '@salutejs/plasma-core';
import { buttonPrimaryHover, buttonPrimaryActive, buttonSecondaryHover, buttonSecondaryActive, buttonSuccessHover, buttonSuccessActive, buttonWarningHover, buttonWarningActive, buttonCriticalHover, buttonCriticalActive, buttonCheckedHover, buttonCheckedHoverColor, buttonCheckedActive, buttonCheckedActiveColor } from '../../tokens';

/**
 * Views (colors) for both B2B and B2C
 */
import { collectPackageInfo } from '../../collectPackageInfo';
collectPackageInfo('Button/Button.props');
export var buttonViews = {
  primary: /*#__PURE__*/css(["", " &:hover{background-color:", ";color:", ";}&:active{background-color:", ";color:", ";}"], baseViews.primary, buttonPrimaryHover, baseViews.primary.color, buttonPrimaryActive, baseViews.primary.color),
  success: /*#__PURE__*/css(["", " &:hover{background-color:", ";color:", ";}&:active{background-color:", ";color:", ";}"], baseViews.success, buttonSuccessHover, baseViews.success.color, buttonSuccessActive, baseViews.success.color),
  warning: /*#__PURE__*/css(["", " &:hover{background-color:", ";color:", ";}&:active{background-color:", ";color:", ";}"], baseViews.warning, buttonWarningHover, baseViews.warning.color, buttonWarningActive, baseViews.warning.color),
  critical: /*#__PURE__*/css(["", " &:hover{background-color:", ";color:", ";}&:active{background-color:", ";color:", ";}"], baseViews.critical, buttonCriticalHover, baseViews.critical.color, buttonCriticalActive, baseViews.critical.color),
  secondary: /*#__PURE__*/css(["", " &:hover{background-color:", ";color:", ";}&:active{background-color:", ";color:", ";}"], baseViews.secondary, buttonSecondaryHover, baseViews.secondary.color, buttonSecondaryActive, baseViews.secondary.color),
  checked: /*#__PURE__*/css(["", " &:hover{background-color:", ";color:", ";}&:active{background-color:", ";color:", ";}"], baseViews.checked, buttonCheckedHover, buttonCheckedHoverColor, buttonCheckedActive, buttonCheckedActiveColor),
  overlay: baseViews.overlay,
  clear: baseViews.clear
};

/**
 * @private
 */
