// @salutejs/plasma-new-hope/cjs/components/Range/Range.styles.js
import './Range.styles_mc4w65.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
require('@linaria/core');
var common = require('../../engines/common.js');
var TextField$1 = require('../TextField/TextField.js');

var mergedConfig = /*#__PURE__*/common.mergeConfig(TextField$1.textFieldConfig);
var TextField = /*#__PURE__*/common.component(mergedConfig);

// NOTE: переопределение токенов TextField
var _exp = function _exp() {
  return TextField;
};
var StyledInput = /*#__PURE__*/react.styled(_exp())({
  name: "StyledInput",
  "class": "s1k2erwt",
  propsAsIs: true
});
var base = "b16kyj37";
var StyledLabel = /*#__PURE__*/react.styled('label')({
  name: "StyledLabel",
  "class": "s14q3826",
  propsAsIs: false
});
var LeftHelper = /*#__PURE__*/react.styled('div')({
  name: "LeftHelper",
  "class": "l1fwucr5",
  propsAsIs: false
});
var ContentWrapper = /*#__PURE__*/react.styled('div')({
  name: "ContentWrapper",
  "class": "czi0aji",
  propsAsIs: false
});
var StyledDivider = /*#__PURE__*/react.styled('div')({
  name: "StyledDivider",
  "class": "s1e6rwfy",
  propsAsIs: false
});
var StyledContentLeft = /*#__PURE__*/react.styled('div')({
  name: "StyledContentLeft",
  "class": "sfxtxl2",
  propsAsIs: false
});
var StyledContentRight = /*#__PURE__*/react.styled('div')({
  name: "StyledContentRight",
  "class": "s1hiyoiq",
  propsAsIs: false
});

exports.ContentWrapper = ContentWrapper;
exports.LeftHelper = LeftHelper;
exports.StyledContentLeft = StyledContentLeft;
exports.StyledContentRight = StyledContentRight;
exports.StyledDivider = StyledDivider;
exports.StyledInput = StyledInput;
exports.StyledLabel = StyledLabel;
exports.base = base;
