// @salutejs/plasma-new-hope/cjs/components/Range/Range.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var index = require('../../utils/index.js');
var base$1 = require('./variations/_size/base.js');
var base = require('./variations/_view/base.js');
var base$2 = require('./variations/_disabled/base.js');
var base$3 = require('./variations/_readonly/base.js');
var index$1 = require('./utils/index.js');
var Range_styles = require('./Range.styles.js');
var Range_tokens = require('./Range.tokens.js');

var _StyledDivider;
var _excluded = ["label", "leftHelper", "contentLeft", "contentRight", "view", "size", "readOnly", "disabled", "dividerVariant", "dividerIcon", "firstValue", "secondValue", "firstValueError", "secondValueError", "firstValueSuccess", "secondValueSuccess", "firstPlaceholder", "secondPlaceholder", "firstTextfieldContentLeft", "firstTextfieldContentRight", "secondTextfieldContentLeft", "secondTextfieldContentRight", "firstTextfieldTextBefore", "secondTextfieldTextBefore", "firstTextfieldTextAfter", "secondTextfieldTextAfter", "onChangeFirstValue", "onChangeSecondValue", "onSearchFirstValue", "onSearchSecondValue", "onFocusFirstTextfield", "onFocusSecondTextfield", "onBlurFirstTextfield", "onBlurSecondTextfield"];
var rangeRoot = function rangeRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (_ref, ref) {
    var _StyledDivider2;
    var label = _ref.label,
      leftHelper = _ref.leftHelper,
      contentLeft = _ref.contentLeft,
      contentRight = _ref.contentRight,
      view = _ref.view,
      size = _ref.size,
      _ref$readOnly = _ref.readOnly,
      readOnly = _ref$readOnly === void 0 ? false : _ref$readOnly,
      _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? false : _ref$disabled,
      _ref$dividerVariant = _ref.dividerVariant,
      dividerVariant = _ref$dividerVariant === void 0 ? 'dash' : _ref$dividerVariant,
      dividerIcon = _ref.dividerIcon,
      firstValue = _ref.firstValue,
      secondValue = _ref.secondValue,
      firstValueError = _ref.firstValueError,
      secondValueError = _ref.secondValueError,
      firstValueSuccess = _ref.firstValueSuccess,
      secondValueSuccess = _ref.secondValueSuccess,
      firstPlaceholder = _ref.firstPlaceholder,
      secondPlaceholder = _ref.secondPlaceholder,
      firstTextfieldContentLeft = _ref.firstTextfieldContentLeft,
      firstTextfieldContentRight = _ref.firstTextfieldContentRight,
      secondTextfieldContentLeft = _ref.secondTextfieldContentLeft,
      secondTextfieldContentRight = _ref.secondTextfieldContentRight,
      firstTextfieldTextBefore = _ref.firstTextfieldTextBefore,
      secondTextfieldTextBefore = _ref.secondTextfieldTextBefore,
      firstTextfieldTextAfter = _ref.firstTextfieldTextAfter,
      secondTextfieldTextAfter = _ref.secondTextfieldTextAfter,
      onChangeFirstValue = _ref.onChangeFirstValue,
      onChangeSecondValue = _ref.onChangeSecondValue,
      onSearchFirstValue = _ref.onSearchFirstValue,
      onSearchSecondValue = _ref.onSearchSecondValue,
      onFocusFirstTextfield = _ref.onFocusFirstTextfield,
      onFocusSecondTextfield = _ref.onFocusSecondTextfield,
      onBlurFirstTextfield = _ref.onBlurFirstTextfield,
      onBlurSecondTextfield = _ref.onBlurSecondTextfield,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
    var rangeErrorClass = firstValueError && secondValueError ? Range_tokens.classes.rangeError : undefined;
    var firstValueErrorClass = !rangeErrorClass && firstValueError ? Range_tokens.classes.rangeValueError : undefined;
    var secondValueErrorClass = !rangeErrorClass && secondValueError ? Range_tokens.classes.rangeValueError : undefined;
    var rangeSuccessClass = firstValueSuccess && secondValueSuccess ? Range_tokens.classes.rangeSuccess : undefined;
    var firstValueSuccessClass = !rangeSuccessClass && firstValueSuccess ? Range_tokens.classes.rangeValueSuccess : undefined;
    var secondValueSuccessClass = !rangeSuccessClass && secondValueSuccess ? Range_tokens.classes.rangeValueSuccess : undefined;
    var handleChangeFirstValue = index$1.getChangeHandler(onChangeFirstValue, disabled, readOnly);
    var handleChangeSecondValue = index$1.getChangeHandler(onChangeSecondValue, disabled, readOnly);
    var handleSearchFirstValue = index$1.getSearchHandler(onSearchFirstValue, disabled, readOnly);
    var handleSearchSecondValue = index$1.getSearchHandler(onSearchSecondValue, disabled, readOnly);
    var Divider = React.useMemo(function () {
      if (dividerVariant === 'none') {
        return null;
      }
      if (dividerVariant === 'dash') {
        return _StyledDivider || (_StyledDivider = /*#__PURE__*/React.createElement(Range_styles.StyledDivider, null, "\u2013"));
      }
      return _StyledDivider2 || (_StyledDivider2 = /*#__PURE__*/React.createElement(Range_styles.StyledDivider, null, dividerIcon));
    }, [dividerVariant, dividerIcon]);
    return /*#__PURE__*/React.createElement(Root, _rollupPluginBabelHelpers.extends({
      view: view,
      size: size,
      disabled: disabled,
      readOnly: !disabled && readOnly,
      ref: ref
    }, rest), label && /*#__PURE__*/React.createElement(Range_styles.StyledLabel, null, label), /*#__PURE__*/React.createElement(Range_styles.ContentWrapper, {
      className: index.cx(rangeErrorClass, rangeSuccessClass)
    }, contentLeft && /*#__PURE__*/React.createElement(Range_styles.StyledContentLeft, null, contentLeft), /*#__PURE__*/React.createElement(Range_styles.StyledInput, {
      className: index.cx(firstValueErrorClass, firstValueSuccessClass),
      value: firstValue,
      readOnly: readOnly,
      disabled: disabled,
      placeholder: firstPlaceholder,
      contentLeft: firstTextfieldContentLeft,
      contentRight: firstTextfieldContentRight,
      textBefore: firstTextfieldTextBefore,
      textAfter: firstTextfieldTextAfter,
      onChange: handleChangeFirstValue,
      onSearch: handleSearchFirstValue,
      onFocus: onFocusFirstTextfield,
      onBlur: onBlurFirstTextfield
    }), Divider, /*#__PURE__*/React.createElement(Range_styles.StyledInput, {
      className: index.cx(secondValueErrorClass, secondValueSuccessClass),
      value: secondValue,
      readOnly: readOnly,
      disabled: disabled,
      placeholder: secondPlaceholder,
      contentLeft: secondTextfieldContentLeft,
      contentRight: secondTextfieldContentRight,
      textBefore: secondTextfieldTextBefore,
      textAfter: secondTextfieldTextAfter,
      onChange: handleChangeSecondValue,
      onSearch: handleSearchSecondValue,
      onFocus: onFocusSecondTextfield,
      onBlur: onBlurSecondTextfield
    }), contentRight && /*#__PURE__*/React.createElement(Range_styles.StyledContentRight, null, contentRight)), leftHelper && /*#__PURE__*/React.createElement(Range_styles.LeftHelper, null, leftHelper));
  });
};
var rangeConfig = {
  name: 'Range',
  tag: 'div',
  layout: rangeRoot,
  base: Range_styles.base,
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    },
    disabled: {
      css: base$2.base,
      attrs: true
    },
    readOnly: {
      css: base$3.base,
      attrs: true
    }
  },
  defaults: {
    size: 'm',
    view: 'default'
  }
};

exports.rangeConfig = rangeConfig;
exports.rangeRoot = rangeRoot;
