// @salutejs/plasma-ui/components/Badge/Badge.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Quantity = exports.Default = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _ = /*#__PURE__*/require(".");
var _IconSettings;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var defaultProps = {
  text: 'Badge',
  size: 'l',
  view: 'primary',
  enableIcon: false
};
var Default = exports.Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(_.Badge, _extends({
    contentLeft: _IconSettings || (_IconSettings = /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconSettings, {
      color: "inherit",
      size: "xs"
    }))
  }, defaultProps));
};
var quantityProps = {
  text: '11',
  size: 's',
  view: 'secondary',
  circled: true
};
var Quantity = exports.Quantity = function Quantity() {
  return /*#__PURE__*/_react["default"].createElement(_.Badge, quantityProps);
};
