// @salutejs/plasma-ui/components/Badge/Badge.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.badgeViews = exports.Badge = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
(0, _collectPackageInfo.collectPackageInfo)('Badge/Badge');
var badgeViews = exports.badgeViews = {
  primary: _plasmaCore.views.primary,
  secondary: _plasmaCore.views.overlay,
  warning: _plasmaCore.views.warning
};
/**
 * Небольшая бирка для ячеек и карточек.
 * Компонент может отображаться в нескольких размерах и цветах, может содержать текст и/или иконку.
 */
var Badge = exports.Badge = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.Badge).withConfig({
  componentId: "plasma-ui__sc-vvsdz6-0"
})(["", ";"], function (_ref) {
  var _ref$view = _ref.view,
    view = _ref$view === void 0 ? 'primary' : _ref$view;
  return badgeViews[view];
});
