// @salutejs/plasma-web/es/components/Carousel/Carousel.perftest.js
import React from 'react';
import { CarouselCard } from './Carousel.examples';
import { Carousel, CarouselItem } from '.';
var items = /*#__PURE__*/Array(25).fill({
  title: 'Слайд',
  subtitle: 'Описание слайда',
  imageSrc: 'images/320_320_n.jpg',
  imageAlt: 'Картинка'
}).map(function (_ref, i) {
  var title = _ref.title,
    subtitle = _ref.subtitle,
    imageSrc = _ref.imageSrc,
    imageAlt = _ref.imageAlt;
  return {
    id: "slide_".concat(i),
    title: "".concat(title, " ").concat(i),
    subtitle: "".concat(subtitle, " ").concat(i),
    imageSrc: imageSrc.replace('n', i % 12),
    imageAlt: "".concat(imageAlt, " ").concat(i)
  };
});
var defaultStyle = {
  margin: '0 -0.5rem'
};
var defaultCarouselItemStyle = {
  width: '20rem',
  padding: '0 0.5rem'
};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Carousel, {
    index: 0,
    style: defaultStyle
  }, items.map(function (item) {
    return /*#__PURE__*/React.createElement(CarouselItem, {
      key: item.id,
      style: defaultCarouselItemStyle
    }, /*#__PURE__*/React.createElement(CarouselCard, {
      id: item.id,
      title: item.title,
      subtitle: item.subtitle,
      imageSrc: item.imageSrc,
      imageAlt: item.imageAlt
    }));
  }));
};
