// @salutejs/plasma-web/es/components/Carousel/Carousel.examples.js
var _excluded = ["title", "subtitle", "imageSrc", "imageAlt", "imageBase"];
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React from 'react';
import styled from 'styled-components';
import { addFocus } from '../../mixins';
import { Image } from '../Image';
import { Headline4, Footnote1 } from '../Typography';
var StyledCard = /*#__PURE__*/styled.a.withConfig({
  componentId: "plasma-web__sc-18sk49o-0"
})(["display:flex;position:relative;border-radius:1rem;", ""], /*#__PURE__*/addFocus({
  outlined: true,
  outlineRadius: '1.125rem'
}));
var StyledCardContent = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-web__sc-18sk49o-1"
})(["position:absolute;left:0;right:0;bottom:0;padding:1.72rem;color:#fff;"]);

/**
 * Карточка под примеры с каруселью.
 * @private
 */
export var CarouselCard = function CarouselCard(_ref) {
  var title = _ref.title,
    subtitle = _ref.subtitle,
    imageSrc = _ref.imageSrc,
    imageAlt = _ref.imageAlt,
    _ref$imageBase = _ref.imageBase,
    imageBase = _ref$imageBase === void 0 ? 'div' : _ref$imageBase,
    rest = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/React.createElement(StyledCard, rest, /*#__PURE__*/React.createElement(Image, {
    src: imageSrc,
    ratio: "16 / 9",
    base: imageBase,
    alt: imageAlt
  }), /*#__PURE__*/React.createElement(StyledCardContent, null, /*#__PURE__*/React.createElement(Headline4, null, title), /*#__PURE__*/React.createElement(Footnote1, null, subtitle)));
};
