// @salutejs/plasma-new-hope/es/components/TextField/TextField.styles.js
import './TextField.styles_aqmut.css';
import { styled } from '@linaria/react';

var InputWrapper = /*#__PURE__*/styled('div')({
  name: "InputWrapper",
  "class": "iq39zt5",
  propsAsIs: false
});
var InputLabelWrapper = /*#__PURE__*/styled('div')({
  name: "InputLabelWrapper",
  "class": "i1j28rg8",
  propsAsIs: false
});
var StyledChips = /*#__PURE__*/styled('div')({
  name: "StyledChips",
  "class": "s13zudoe",
  propsAsIs: false
});
var Input = /*#__PURE__*/styled('input')({
  name: "Input",
  "class": "io4bpie",
  propsAsIs: false
});
var Label = /*#__PURE__*/styled('label')({
  name: "Label",
  "class": "l13qvwoy",
  propsAsIs: false
});
var StyledContentLeft = /*#__PURE__*/styled('div')({
  name: "StyledContentLeft",
  "class": "s19ry60d",
  propsAsIs: false
});
var StyledContentRight = /*#__PURE__*/styled('div')({
  name: "StyledContentRight",
  "class": "sbt1ohz",
  propsAsIs: false
});
var LeftHelper = /*#__PURE__*/styled('div')({
  name: "LeftHelper",
  "class": "llv9av6",
  propsAsIs: false
});
var StyledTextBefore = /*#__PURE__*/styled('div')({
  name: "StyledTextBefore",
  "class": "s600610",
  propsAsIs: false
});
var StyledTextAfter = /*#__PURE__*/styled('div')({
  name: "StyledTextAfter",
  "class": "suk2zst",
  propsAsIs: false
});

export { Input, InputLabelWrapper, InputWrapper, Label, LeftHelper, StyledChips, StyledContentLeft, StyledContentRight, StyledTextAfter, StyledTextBefore };
