// @salutejs/plasma-web/es/components/PreviewGallery/PreviewGallery.perftest.js
var _StyledAddButton, _PreviewGallery, _PreviewGallery2;
import React from 'react';
import styled from 'styled-components';
import { IconTrash, IconTrashFilled } from '@salutejs/plasma-icons';
import { surfaceSolid01 } from '@salutejs/plasma-tokens-web';
import { PreviewGallery } from '.';
var StyledAddButton = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-web__sc-yzjbem-0"
})(["width:100%;background:", ";display:flex;justify-content:center;align-items:center;border-radius:0.75rem;user-select:none;padding-top:56.25%;"], surfaceSolid01);
export var AddButton = function AddButton() {
  return _StyledAddButton || (_StyledAddButton = /*#__PURE__*/React.createElement(StyledAddButton, null));
};
var images = [{
  id: /*#__PURE__*/Math.random(),
  image: './images/320_320_0.jpg',
  caption: '3:24',
  tooltip: {
    text: 'Высокое качество воспроизведения',
    placement: 'top'
  }
}, {
  id: 2 * /*#__PURE__*/Math.random(),
  image: './images/320_320_1.jpg',
  isSelected: true
}, {
  id: 3 * /*#__PURE__*/Math.random(),
  image: './images/320_320_2.jpg',
  actionDisabled: true
}, {
  id: 4 * /*#__PURE__*/Math.random(),
  image: './images/320_320_3.jpg'
}, {
  id: 5 * /*#__PURE__*/Math.random(),
  image: './images/320_320_4.jpg'
}, {
  id: 6 * /*#__PURE__*/Math.random(),
  image: './images/320_320_4.jpg',
  status: 'error'
}, {
  id: 7 * /*#__PURE__*/Math.random(),
  tooltip: {
    placement: 'bottom',
    text: 'Не сильно высокое качество воспроизведения',
    animated: false
  },
  component: /*#__PURE__*/React.createElement(AddButton, null),
  customClickHandle: function customClickHandle() {}
}];
var selectableProps = {};
var func = function func() {};
export var Selectable = function Selectable() {
  return _PreviewGallery || (_PreviewGallery = /*#__PURE__*/React.createElement(PreviewGallery, {
    items: images,
    onItemAction: func,
    actionIcon: /*#__PURE__*/React.createElement(IconTrashFilled, {
      size: "xs",
      color: "inherit"
    }),
    onItemClick: func
  }));
};
var draggableProps = {};
export var Draggable = function Draggable() {
  return _PreviewGallery2 || (_PreviewGallery2 = /*#__PURE__*/React.createElement(PreviewGallery, {
    interactionType: "draggable",
    items: images,
    actionIcon: /*#__PURE__*/React.createElement(IconTrash, {
      size: "xs",
      color: "inherit"
    }),
    onItemAction: func,
    onItemsSortEnd: func
  }));
};
