// @salutejs/plasma-new-hope/cjs/components/TextField/TextField.js
import './TextField_wctd2m.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var plasmaCore = require('@salutejs/plasma-core');
var index = require('../../utils/index.js');
var base$2 = require('./variations/_size/base.js');
var base$1 = require('./variations/_view/base.js');
var base$3 = require('./variations/_disabled/base.js');
var base$4 = require('./variations/_read-only/base.js');
var base$5 = require('./variations/_label-placement/base.js');
var TextField_styles = require('./TextField.styles.js');
var TextField_tokens = require('./TextField.tokens.js');
var TextFieldChip = require('./ui/TextFieldChip/TextFieldChip.js');
var useKeyNavigation = require('./hooks/useKeyNavigation.js');

var _excluded = ["id", "className", "style", "contentLeft", "contentRight", "label", "labelPlacement", "textBefore", "textAfter", "placeholder", "leftHelper", "enumerationType", "view", "size", "readOnly", "disabled", "chips", "onChange", "onChangeChips", "onSearch", "onKeyDown"];
var base = "bpunezx";
var textFieldRoot = function textFieldRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (_ref, ref) {
    var id = _ref.id,
      className = _ref.className,
      style = _ref.style,
      contentLeft = _ref.contentLeft,
      contentRight = _ref.contentRight,
      label = _ref.label,
      labelPlacement = _ref.labelPlacement,
      textBefore = _ref.textBefore,
      textAfter = _ref.textAfter,
      placeholder = _ref.placeholder,
      leftHelper = _ref.leftHelper,
      _ref$enumerationType = _ref.enumerationType,
      enumerationType = _ref$enumerationType === void 0 ? 'plain' : _ref$enumerationType,
      view = _ref.view,
      size = _ref.size,
      _ref$readOnly = _ref.readOnly,
      readOnly = _ref$readOnly === void 0 ? false : _ref$readOnly,
      _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? false : _ref$disabled,
      values = _ref.chips,
      onChange = _ref.onChange,
      onChangeChips = _ref.onChangeChips,
      onSearch = _ref.onSearch,
      onKeyDown = _ref.onKeyDown,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
    var contentRef = React.useRef(null);
    var inputRef = React.useRef(null);
    var inputForkRef = plasmaCore.useForkRef(inputRef, ref);
    var chipsRefs = React.useRef([]);
    var controlledRefs = {
      contentRef: contentRef,
      inputRef: inputRef,
      chipsRefs: chipsRefs
    };
    var _useState = React.useState([]),
      _useState2 = _rollupPluginBabelHelpers.slicedToArray(_useState, 2),
      chips = _useState2[0],
      setChips = _useState2[1];
    var uniqId = plasmaCore.safeUseId();
    var innerId = id || uniqId;
    var labelId = plasmaCore.safeUseId();
    var helperTextId = plasmaCore.safeUseId();
    var isChipEnumeration = enumerationType === 'chip';
    var hideLabel = (size === 'xs' || isChipEnumeration) && labelPlacement === 'inner';
    var labelInside = size !== 'xs' && labelPlacement === 'inner';
    var innerLabelPlacementValue = hideLabel ? 'outer' : labelPlacement;
    var innerPlaceholderValue = hideLabel ? label : placeholder;
    var innerLabelValue = hideLabel ? undefined : label;
    var isChipsVisible = isChipEnumeration && (chips === null || chips === void 0 ? void 0 : chips.length);
    var withHasChips = isChipsVisible ? TextField_tokens.classes.hasChips : undefined;
    var wrapperWithoutLeftContent = !contentLeft && isChipsVisible ? TextField_tokens.classes.hasEmptyContentLeft : undefined;
    var wrapperWithoutRightContent = !contentRight && isChipsVisible ? TextField_tokens.classes.hasEmptyContentRight : undefined;
    var handleChange = function handleChange(event) {
      if (disabled || readOnly) {
        return;
      }
      var _event$target = event.target,
        maxLength = _event$target.maxLength,
        value = _event$target.value;
      if (maxLength !== -1 && value.length > maxLength) {
        return;
      }
      onChange === null || onChange === void 0 || onChange(event);
    };
    var updateChips = function updateChips(newChips, newValues) {
      setChips(newChips);
      onChangeChips === null || onChangeChips === void 0 || onChangeChips(newValues);
    };
    var _useKeyNavigation = useKeyNavigation.useKeyNavigation({
        controlledRefs: controlledRefs,
        disabled: disabled,
        readOnly: readOnly,
        chips: chips,
        enumerationType: enumerationType,
        updateChips: updateChips,
        onSearch: onSearch,
        onChange: onChange
      }),
      handleInputKeydown = _useKeyNavigation.handleInputKeydown,
      handleChipKeyDown = _useKeyNavigation.handleChipKeyDown,
      onChipClear = _useKeyNavigation.onChipClear,
      handleContentKeyDown = _useKeyNavigation.handleContentKeyDown;
    var onChipClick = function onChipClick(event) {
      return event.stopPropagation();
    };
    var handleInputFocus = function handleInputFocus() {
      if (readOnly || disabled || !(inputRef !== null && inputRef !== void 0 && inputRef.current)) {
        return;
      }
      inputRef.current.scrollTo({
        top: 0,
        left: inputRef.current.offsetLeft,
        behavior: 'smooth'
      });
      inputRef.current.focus();
    };
    var getRef = function getRef(element, index) {
      if (element && chipsRefs !== null && chipsRefs !== void 0 && chipsRefs.current) {
        chipsRefs.current[index] = element;
      }
    };
    var handleOnKeyDown = function handleOnKeyDown(event) {
      handleInputKeydown(event);
      onKeyDown && onKeyDown(event);
    };
    React.useEffect(function () {
      if (!isChipEnumeration && !(values !== null && values !== void 0 && values.length)) {
        return;
      }
      var newChips = (values === null || values === void 0 ? void 0 : values.map(function (value, index) {
        return {
          id: "".concat(index, "_").concat(value),
          text: value
        };
      })) || [];
      setChips(newChips);
    }, [isChipEnumeration, values]);
    return /*#__PURE__*/React.createElement(Root, {
      view: view,
      size: size,
      disabled: disabled,
      readOnly: !disabled && readOnly,
      labelPlacement: innerLabelPlacementValue,
      onClick: handleInputFocus,
      className: className,
      style: style
    }, labelInside || innerLabelValue && /*#__PURE__*/React.createElement(TextField_styles.Label, {
      id: labelId,
      htmlFor: id
    }, innerLabelValue), /*#__PURE__*/React.createElement(TextField_styles.InputWrapper, {
      className: index.cx(withHasChips, wrapperWithoutLeftContent, wrapperWithoutRightContent)
    }, contentLeft && /*#__PURE__*/React.createElement(TextField_styles.StyledContentLeft, null, contentLeft), /*#__PURE__*/React.createElement(TextField_styles.InputLabelWrapper, {
      tabIndex: -1,
      ref: contentRef,
      onKeyDown: handleContentKeyDown,
      className: withHasChips
    }, textBefore && /*#__PURE__*/React.createElement(TextField_styles.StyledTextBefore, null, textBefore), isChipEnumeration && Boolean(chips === null || chips === void 0 ? void 0 : chips.length) && /*#__PURE__*/React.createElement(TextField_styles.StyledChips, null, chips === null || chips === void 0 ? void 0 : chips.map(function (_ref2, index) {
      var chipId = _ref2.id,
        text = _ref2.text;
      return /*#__PURE__*/React.createElement(TextFieldChip.TextFieldChip, {
        id: chipId,
        ref: function ref(element) {
          return getRef(element, index);
        },
        key: "".concat(chipId, "_").concat(index),
        disabled: disabled,
        readOnly: readOnly,
        value: text,
        text: text,
        onKeyDown: function onKeyDown(event) {
          return handleChipKeyDown(event, chipId, index);
        },
        onClear: function onClear() {
          return onChipClear(chipId, index);
        },
        onClick: onChipClick
      });
    })), /*#__PURE__*/React.createElement(TextField_styles.Input, _rollupPluginBabelHelpers.extends({}, rest, {
      ref: inputForkRef,
      id: innerId,
      "aria-labelledby": labelId,
      "aria-describedby": helperTextId,
      placeholder: innerPlaceholderValue,
      disabled: disabled,
      readOnly: !disabled && readOnly,
      onChange: handleChange,
      onKeyDown: handleOnKeyDown
    })), labelInside && /*#__PURE__*/React.createElement(TextField_styles.Label, {
      id: labelId,
      htmlFor: innerId
    }, innerLabelValue), textAfter && /*#__PURE__*/React.createElement(TextField_styles.StyledTextAfter, null, textAfter)), contentRight && /*#__PURE__*/React.createElement(TextField_styles.StyledContentRight, null, contentRight)), leftHelper && /*#__PURE__*/React.createElement(TextField_styles.LeftHelper, {
      id: helperTextId
    }, leftHelper));
  });
};
var textFieldConfig = {
  name: 'TextField',
  tag: 'div',
  layout: textFieldRoot,
  base: base,
  variations: {
    view: {
      css: base$1.base
    },
    size: {
      css: base$2.base
    },
    disabled: {
      css: base$3.base,
      attrs: true
    },
    readOnly: {
      css: base$4.base,
      attrs: true
    },
    labelPlacement: {
      css: base$5.base
    }
  },
  defaults: {
    size: 'm',
    view: 'default'
  }
};

exports.base = base;
exports.textFieldConfig = textFieldConfig;
exports.textFieldRoot = textFieldRoot;
