// @salutejs/plasma-new-hope/cjs/components/TextField/TextField.styles.js
import './TextField.styles_aqmut.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');

var InputWrapper = /*#__PURE__*/react.styled('div')({
  name: "InputWrapper",
  "class": "iq39zt5",
  propsAsIs: false
});
var InputLabelWrapper = /*#__PURE__*/react.styled('div')({
  name: "InputLabelWrapper",
  "class": "i1j28rg8",
  propsAsIs: false
});
var StyledChips = /*#__PURE__*/react.styled('div')({
  name: "StyledChips",
  "class": "s13zudoe",
  propsAsIs: false
});
var Input = /*#__PURE__*/react.styled('input')({
  name: "Input",
  "class": "io4bpie",
  propsAsIs: false
});
var Label = /*#__PURE__*/react.styled('label')({
  name: "Label",
  "class": "l13qvwoy",
  propsAsIs: false
});
var StyledContentLeft = /*#__PURE__*/react.styled('div')({
  name: "StyledContentLeft",
  "class": "s19ry60d",
  propsAsIs: false
});
var StyledContentRight = /*#__PURE__*/react.styled('div')({
  name: "StyledContentRight",
  "class": "sbt1ohz",
  propsAsIs: false
});
var LeftHelper = /*#__PURE__*/react.styled('div')({
  name: "LeftHelper",
  "class": "llv9av6",
  propsAsIs: false
});
var StyledTextBefore = /*#__PURE__*/react.styled('div')({
  name: "StyledTextBefore",
  "class": "s600610",
  propsAsIs: false
});
var StyledTextAfter = /*#__PURE__*/react.styled('div')({
  name: "StyledTextAfter",
  "class": "suk2zst",
  propsAsIs: false
});

exports.Input = Input;
exports.InputLabelWrapper = InputLabelWrapper;
exports.InputWrapper = InputWrapper;
exports.Label = Label;
exports.LeftHelper = LeftHelper;
exports.StyledChips = StyledChips;
exports.StyledContentLeft = StyledContentLeft;
exports.StyledContentRight = StyledContentRight;
exports.StyledTextAfter = StyledTextAfter;
exports.StyledTextBefore = StyledTextBefore;
