// @salutejs/plasma-new-hope/styled-components/es/components/Pagination/Pagination.js
var _PaginationShorter;
var _excluded = ["value", "perPage", "slots", "view", "viewCurrentPage", "type", "size", "pilled", "square", "disabledPages", "count", "hasQuickJump", "hasPerPage", "perPageList", "isCommonButtonStyles", "placeholderQuickJump", "textQuickJump", "textPerPage", "leftContent", "rightContent", "onChangePageValue", "onChangePerPageValue"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React, { useEffect, useState, forwardRef } from 'react';
import { cx, isNumber } from '../../utils';
import { PaginationPages, PaginationActions, PaginationButton, PaginationSection, PaginationButtonGroup, PaginationRoot, PaginationShorter } from './Pagination.styles';
import { getSections, defaultValues } from './utils';
import { base as viewCSS } from './variations/_view/base';
import { base as typeCSS } from './variations/_type/base';
import { base as sizeCSS } from './variations/_size/base';
import { classes } from './Pagination.tokens';
import { PaginationQuickJumpToPage } from './ui/PaginationQuickJumpToPage/PaginationQuickJumpToPage';
import { PaginationSelectPerPage } from './ui/PaginationSelectPerPage/PaginationSelectPerPage';
export var paginationRoot = function paginationRoot(Root) {
  return /*#__PURE__*/forwardRef(function (_ref, ref) {
    var _ref$value = _ref.value,
      value = _ref$value === void 0 ? defaultValues.value : _ref$value,
      perPage = _ref.perPage,
      _ref$slots = _ref.slots,
      slots = _ref$slots === void 0 ? defaultValues.slots : _ref$slots,
      view = _ref.view,
      viewCurrentPage = _ref.viewCurrentPage,
      _ref$type = _ref.type,
      type = _ref$type === void 0 ? defaultValues.type : _ref$type,
      size = _ref.size,
      _ref$pilled = _ref.pilled,
      pilled = _ref$pilled === void 0 ? defaultValues.pilled : _ref$pilled,
      _ref$square = _ref.square,
      square = _ref$square === void 0 ? defaultValues.square : _ref$square,
      _ref$disabledPages = _ref.disabledPages,
      disabledPages = _ref$disabledPages === void 0 ? defaultValues.disabledPages : _ref$disabledPages,
      _ref$count = _ref.count,
      count = _ref$count === void 0 ? defaultValues.value : _ref$count,
      _ref$hasQuickJump = _ref.hasQuickJump,
      hasQuickJump = _ref$hasQuickJump === void 0 ? defaultValues.hasQuickJump : _ref$hasQuickJump,
      _ref$hasPerPage = _ref.hasPerPage,
      hasPerPage = _ref$hasPerPage === void 0 ? defaultValues.hasPerPage : _ref$hasPerPage,
      _ref$perPageList = _ref.perPageList,
      perPageList = _ref$perPageList === void 0 ? defaultValues.perPageList : _ref$perPageList,
      _ref$isCommonButtonSt = _ref.isCommonButtonStyles,
      isCommonButtonStyles = _ref$isCommonButtonSt === void 0 ? defaultValues.isCommonButtonStyles : _ref$isCommonButtonSt,
      _ref$placeholderQuick = _ref.placeholderQuickJump,
      placeholderQuickJump = _ref$placeholderQuick === void 0 ? defaultValues.placeholderQuickJump : _ref$placeholderQuick,
      _ref$textQuickJump = _ref.textQuickJump,
      textQuickJump = _ref$textQuickJump === void 0 ? defaultValues.textQuickJump : _ref$textQuickJump,
      _ref$textPerPage = _ref.textPerPage,
      textPerPage = _ref$textPerPage === void 0 ? defaultValues.textPerPage : _ref$textPerPage,
      leftContent = _ref.leftContent,
      rightContent = _ref.rightContent,
      onChangePageValue = _ref.onChangePageValue,
      onChangePerPageValue = _ref.onChangePerPageValue,
      rest = _objectWithoutProperties(_ref, _excluded);
    var _useState = useState(1),
      _useState2 = _slicedToArray(_useState, 2),
      page = _useState2[0],
      setPageValue = _useState2[1];
    var _useState3 = useState(perPage),
      _useState4 = _slicedToArray(_useState3, 2),
      perPageValue = _useState4[0],
      setPerPageValue = _useState4[1];
    var _useState5 = useState(value),
      _useState6 = _slicedToArray(_useState5, 2),
      pages = _useState6[0],
      setPagesValue = _useState6[1];
    var _useState7 = useState(null),
      _useState8 = _slicedToArray(_useState7, 2),
      sections = _useState8[0],
      setSections = _useState8[1];
    var typeClass = classes["".concat(type, "Type")];
    var roundedClass = pilled ? 'circle-circle' : 'square-square';
    var withHasPerPageSelect = hasPerPage ? classes.withHasPerPageSelect : undefined;
    var withHasQuickJump = hasQuickJump ? classes.withHasQuickJump : undefined;
    var disabled = disabledPages.map(function (el) {
      return isNumber(el) ? Number(el) : null;
    });
    var handlerSetPages = function handlerSetPages(newPerPage) {
      setPagesValue(Math.ceil(count / (newPerPage || (hasPerPage ? defaultValues.perPage : 1))));
    };
    var handlerSetPage = function handlerSetPage(newPageValue) {
      var pageValue = newPageValue !== null && newPageValue !== void 0 ? newPageValue : defaultValues.value;
      if (pageValue > pages) {
        pageValue = pages;
      }
      if (pageValue < 1) {
        pageValue = 1;
      }
      if (disabled.includes(pageValue)) {
        return;
      }
      setPageValue(pageValue);
      onChangePageValue === null || onChangePageValue === void 0 || onChangePageValue(pageValue);
    };
    var handlerSetPerPage = function handlerSetPerPage(newPerPageValue) {
      handlerSetPage(1);
      setPerPageValue(newPerPageValue);
      onChangePerPageValue === null || onChangePerPageValue === void 0 || onChangePerPageValue(newPerPageValue);
      handlerSetPages(newPerPageValue);
    };
    var isActiveButton = function isActiveButton(checkPage) {
      return page === checkPage ? classes.paginationPageButtonActive : '';
    };
    useEffect(function () {
      handlerSetPages(perPage);
    }, [perPage, count, handlerSetPages]);
    useEffect(function () {
      handlerSetPage(value);
    }, [value]);
    useEffect(function () {
      setSections(getSections(page, pages, slots));
    }, [page, slots, pages, setSections, getSections]);
    return /*#__PURE__*/React.createElement(Root, _extends({
      size: size,
      view: view,
      viewCurrentPage: viewCurrentPage,
      className: cx(classes.wrapper, typeClass),
      ref: ref
    }, rest), /*#__PURE__*/React.createElement(PaginationRoot, {
      className: typeClass
    }, /*#__PURE__*/React.createElement(PaginationPages, null, leftContent && /*#__PURE__*/React.createElement(PaginationButtonGroup, {
      isCommonButtonStyles: isCommonButtonStyles
    }, leftContent), sections && sections.map(function (section, indexSection) {
      return /*#__PURE__*/React.createElement(React.Fragment, null, indexSection !== 0 && (_PaginationShorter || (_PaginationShorter = /*#__PURE__*/React.createElement(PaginationShorter, {
        stretching: "fixed",
        disabled: true
      }, "..."))), /*#__PURE__*/React.createElement(PaginationSection, {
        key: indexSection
      }, section.map(function (pageValue) {
        return /*#__PURE__*/React.createElement(PaginationButton, {
          square: square,
          pin: roundedClass,
          key: pageValue,
          onClick: function onClick() {
            return handlerSetPage(pageValue);
          },
          disabled: disabled.includes(pageValue),
          className: cx(classes.paginationPageButton, isActiveButton(pageValue))
        }, pageValue);
      })));
    }), rightContent && /*#__PURE__*/React.createElement(PaginationButtonGroup, {
      isCommonButtonStyles: isCommonButtonStyles
    }, rightContent)), /*#__PURE__*/React.createElement(PaginationActions, {
      className: cx(classes.actions, typeClass, withHasPerPageSelect, withHasQuickJump)
    }, hasQuickJump && /*#__PURE__*/React.createElement(PaginationQuickJumpToPage, {
      placeholderQuickJump: placeholderQuickJump,
      textQuickJump: textQuickJump,
      onChangeValue: handlerSetPage
    }), hasPerPage && /*#__PURE__*/React.createElement(PaginationSelectPerPage, {
      textPerPage: textPerPage,
      value: perPageValue,
      valuesList: perPageList,
      onChangeValue: handlerSetPerPage
    }))));
  });
};
export var paginationConfig = {
  name: 'Pagination',
  tag: 'div',
  layout: paginationRoot,
  base: '',
  variations: {
    view: {
      css: viewCSS
    },
    size: {
      css: sizeCSS
    },
    type: {
      css: typeCSS
    }
  },
  defaults: {
    view: 'clear',
    size: 'm'
  }
};
