// @salutejs/plasma-new-hope/cjs/components/Calendar/ui/CalendarDayItem/CalendarDayItem.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var index = require('../../../../utils/index.js');
var Calendar_tokens = require('../../Calendar.tokens.js');
var CalendarDayItem_styles = require('./CalendarDayItem.styles.js');

var _excluded = ["isFocused", "dayOfWeek", "disabled", "isCurrent", "isSelected", "isDayInCurrentMonth", "isDouble", "inRange", "isHovered", "sideInRange", "eventList", "day", "monthIndex", "year", "onClick", "onMouseOver", "onFocus", "disabledArrowKey", "disabledMonths"];

/**
 * Компонент дня в календаре.
 */
var CalendarDayItem = /*#__PURE__*/React.memo( /*#__PURE__*/React.forwardRef(function (_ref, outerRef) {
  var isFocused = _ref.isFocused,
    dayOfWeek = _ref.dayOfWeek,
    disabled = _ref.disabled,
    isCurrent = _ref.isCurrent,
    isSelected = _ref.isSelected,
    isDayInCurrentMonth = _ref.isDayInCurrentMonth,
    isDouble = _ref.isDouble,
    inRange = _ref.inRange,
    isHovered = _ref.isHovered,
    sideInRange = _ref.sideInRange,
    _ref$eventList = _ref.eventList,
    eventList = _ref$eventList === void 0 ? [] : _ref$eventList,
    day = _ref.day,
    monthIndex = _ref.monthIndex,
    year = _ref.year,
    onClick = _ref.onClick,
    onMouseOver = _ref.onMouseOver,
    onFocus = _ref.onFocus,
    disabledArrowKey = _ref.disabledArrowKey,
    disabledMonths = _ref.disabledMonths,
    rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
  var selectableClass = !dayOfWeek && !disabled ? Calendar_tokens.classes.selectableItem : undefined;
  var selectedClass = isSelected ? Calendar_tokens.classes.selectedItem : undefined;
  var currentClass = isCurrent ? Calendar_tokens.classes.currentItem : undefined;
  var hoveredClass = isHovered ? Calendar_tokens.classes.hoveredItem : undefined;
  var dayOfWeekClass = dayOfWeek ? Calendar_tokens.classes.dayOfWeek : undefined;
  var disabledClass = disabled && !isCurrent ? Calendar_tokens.classes.dayDisabled : undefined;
  var disabledCurrentClass = disabled && isCurrent ? Calendar_tokens.classes.dayDisabledCurrent : undefined;
  var sideInRangeLeft = sideInRange === 'left' ? Calendar_tokens.classes.daySideLeft : undefined;
  var sideInRangeRight = sideInRange === 'right' ? Calendar_tokens.classes.daySideRight : undefined;
  var sideInRangeClass = sideInRange ? Calendar_tokens.classes.daySideInRange : undefined;
  var inRangeStyle = inRange ? Calendar_tokens.classes.dayInRange : undefined;
  var sideOffset = !isSelected && isCurrent ? -1 : 0;
  return /*#__PURE__*/React.createElement(CalendarDayItem_styles.StyledDayRoot, _rollupPluginBabelHelpers.extends({
    ref: outerRef,
    className: index.cx(dayOfWeekClass, selectableClass, selectedClass, currentClass, disabledClass, disabledCurrentClass, hoveredClass),
    tabIndex: isFocused ? 0 : -1,
    dayOfWeek: dayOfWeek,
    isCurrent: isCurrent,
    isSelected: isSelected,
    isDayInCurrentMonth: isDayInCurrentMonth,
    isDouble: isDouble,
    isHovered: isHovered,
    sideInRange: sideInRange,
    onClick: disabled ? undefined : onClick,
    onMouseOver: onMouseOver,
    onFocus: onFocus,
    "data-day": day,
    "data-month-index": monthIndex,
    "data-year": year,
    "data-disabled-arrow-key": disabledArrowKey || undefined,
    "data-disabled-months": disabledMonths || undefined,
    "aria-selected": isSelected,
    "aria-disabled": disabled,
    "aria-describedby": disabled ? 'withShift' : undefined
  }, rest), /*#__PURE__*/React.createElement(CalendarDayItem_styles.StyledDay, {
    className: index.cx(sideInRangeClass, sideInRangeLeft, sideInRangeRight, inRangeStyle),
    offset: sideOffset
  }, day), /*#__PURE__*/React.createElement(CalendarDayItem_styles.StyledEvents, {
    "aria-hidden": true
  }, [eventList[0], eventList[1], eventList[2]].map(function (event, i) {
    return event && /*#__PURE__*/React.createElement(CalendarDayItem_styles.StyledEvent, _rollupPluginBabelHelpers.extends({
      key: "".concat(event.date, "-").concat(i)
    }, event));
  })));
}));

exports.CalendarDayItem = CalendarDayItem;
