// @salutejs/plasma-new-hope/styled-components/cjs/components/Switch/Switch.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.switchRoot = exports.switchConfig = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/require("styled-components");
var _base = /*#__PURE__*/require("../Switch/_size/base");
var _base2 = /*#__PURE__*/require("../Switch/_view/base");
var _base3 = /*#__PURE__*/require("../Switch/_focused/base");
var _base4 = /*#__PURE__*/require("../Switch/_disabled/base");
var _base5 = /*#__PURE__*/require("../Switch/_label-position/base");
var _Switch = /*#__PURE__*/require("./Switch.styles");
var _StyledTrigger;
var _excluded = ["size", "view", "focused", "disabled", "labelPosition", "label", "id", "style", "className", "checked", "defaultChecked"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var base = /*#__PURE__*/(0, _styledComponents.css)(["position:relative;display:flex;align-items:center;cursor:pointer;-webkit-tap-highlight-color:rgba(0,0,0,0);&:focus{outline:0 none;}"]);
var switchRoot = exports.switchRoot = function switchRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
    var size = props.size,
      view = props.view,
      focused = props.focused,
      disabled = props.disabled,
      labelPosition = props.labelPosition,
      label = props.label,
      id = props.id,
      style = props.style,
      className = props.className,
      checked = props.checked,
      defaultChecked = props.defaultChecked,
      rest = _objectWithoutProperties(props, _excluded);
    var exactChecked = Boolean(checked !== undefined ? checked : defaultChecked);
    // const singleLineClass = singleLine ? 'single-line' : '';

    return /*#__PURE__*/_react["default"].createElement(Root, {
      view: view,
      size: size,
      disabled: disabled,
      focused: focused,
      labelPosition: labelPosition,
      id: id,
      style: style,
      className: className
    }, /*#__PURE__*/_react["default"].createElement(_Switch.StyledInput, _extends({}, rest, {
      ref: ref,
      role: "switch",
      "aria-checked": exactChecked,
      type: "checkbox",
      checked: checked,
      defaultChecked: defaultChecked,
      disabled: disabled
    })), /*#__PURE__*/_react["default"].createElement(_Switch.StyledLabel, {
      tabIndex: -1
    }, label), _StyledTrigger || (_StyledTrigger = /*#__PURE__*/_react["default"].createElement(_Switch.StyledTrigger, {
      "aria-hidden": true
    })));
  });
};
var switchConfig = exports.switchConfig = {
  name: 'Switch',
  tag: 'label',
  base: base,
  layout: switchRoot,
  variations: {
    size: {
      css: _base.base
    },
    view: {
      css: _base2.base
    },
    disabled: {
      css: _base4.base,
      attrs: true
    },
    focused: {
      css: _base3.base
    },
    labelPosition: {
      css: _base5.base
    }
  },
  defaults: {
    size: 's',
    view: 'default'
  }
};
