// @salutejs/plasma-hope/es/tokens/colors.js
export var buttonPrimaryHover = 'var(--plasma-colors-button-primary-hover)';
export var buttonPrimaryActive = 'var(--plasma-colors-button-primary-active)';
export var buttonSecondaryHover = 'var(--plasma-colors-button-secondary-hover)';
export var buttonSecondaryActive = 'var(--plasma-colors-button-secondary-active)';
export var buttonSuccessHover = 'var(--plasma-colors-button-success-hover)';
export var buttonSuccessActive = 'var(--plasma-colors-button-success-active)';
export var buttonWarningHover = 'var(--plasma-colors-button-warning-hover)';
export var buttonWarningActive = 'var(--plasma-colors-button-warning-active)';
export var buttonCriticalHover = 'var(--plasma-colors-button-critical-hover)';
export var buttonCriticalActive = 'var(--plasma-colors-button-critical-active)';
export var buttonCheckedHover = 'var(--plasma-colors-button-checked-hover)';
export var buttonCheckedHoverColor = 'var(--plasma-colors-button-checked-hover-color)';
export var buttonCheckedActive = 'var(--plasma-colors-button-checked-active)';
export var buttonCheckedActiveColor = 'var(--plasma-colors-button-checked-active-color)';
export var fieldColor = 'var(--plasma-field-color)';
export var fieldLabelColor = 'var(--plasma-field-label-color)';
export var inputBackground = 'var(--plasma-input-background-color)';
export var inputBackgroundHover = 'var(--plasma-input-hover-background-color)';
export var inputBackgroundFocus = 'var(--plasma-input-focus-background-color)';
export var inputColor = 'var(--plasma-input-color)';
export var inputCaretColor = 'var(--plasma-input-caret-color)';
export var inputPlaceholderColor = 'var(--plasma-input-placeholder-color)';
export var inputFocusPlaceholderColor = 'var(--plasma-input-focus-placeholder-color)';
export var inputBorder = 'var(--plasma-colors-input-border)';
export var inputBorderHover = 'var(--plasma-colors-input-border-hover)';
export var inputBorderFocus = 'var(--plasma-colors-input-border-focus)';
export var inputSuccessBackground = 'var(--plasma-input-success-background)';
export var inputSuccessColor = 'var(--plasma-input-success-color)';
export var inputSuccessCaretColor = 'var(--plasma-input-success-caret-color)';
export var inputSuccessBorderColor = 'var(--plasma-input-success-border-color)';
export var inputWarningBackground = 'var(--plasma-input-warning-background)';
export var inputWarningColor = 'var(--plasma-input-warning-color)';
export var inputWarningCaretColor = 'var(--plasma-input-warning-caret-color)';
export var inputWarningBorderColor = 'var(--plasma-input-warning-border-color)';
export var inputErrorBackground = 'var(--plasma-input-error-background)';
export var inputErrorColor = 'var(--plasma-input-error-color)';
export var inputErrorCaretColor = 'var(--plasma-input-error-caret-color)';
export var inputErrorBorderColor = 'var(--plasma-input-error-border-color)';
export var inputLabelColor = 'var(--plasma-input-label-color)';
export var link = 'var(--plasma-colors-link)';
export var linkHover = 'var(--plasma-colors-link-hover)';
export var linkActive = 'var(--plasma-colors-link-active)';
export var linkVisited = 'var(--plasma-colors-link-visited)';
export var linkVisitedHover = 'var(--plasma-colors-link-visited-hover)';
export var linkVisitedActive = 'var(--plasma-colors-link-visited-active)';
