// @salutejs/plasma-new-hope/es/components/Dropdown/ui/DropdownItem/DropdownItem.styles.js
import './DropdownItem.styles_1o4szyu.css';
import { styled } from '@linaria/react';
import { tokens } from '../../Dropdown.tokens.js';

var StyledContentLeft = /*#__PURE__*/styled('div')({
  name: "StyledContentLeft",
  "class": "s1gkzlls",
  propsAsIs: false
});
var StyledContentRight = /*#__PURE__*/styled('div')({
  name: "StyledContentRight",
  "class": "sz6s14y",
  propsAsIs: false
});
var DisclosureIconWrapper = /*#__PURE__*/styled('div')({
  name: "DisclosureIconWrapper",
  "class": "d1m97l2m",
  propsAsIs: false
});
var StyledText = /*#__PURE__*/styled('div')({
  name: "StyledText",
  "class": "soib72h",
  propsAsIs: false
});
var _exp3 = function _exp3() {
  return function (_ref) {
    var variant = _ref.variant;
    return "var(".concat(variant === 'tight' ? tokens.itemPaddingTopTight : tokens.itemPaddingTop, ")");
  };
};
var _exp5 = function _exp5() {
  return function (_ref2) {
    var variant = _ref2.variant;
    return "var(".concat(variant === 'tight' ? tokens.itemPaddingBottomTight : tokens.itemPaddingBottom, ")");
  };
};
var Wrapper = /*#__PURE__*/styled('li')({
  name: "Wrapper",
  "class": "w6l2f0g",
  propsAsIs: false,
  vars: {
    "w6l2f0g-0": [/*#__PURE__*/_exp3()],
    "w6l2f0g-1": [/*#__PURE__*/_exp5()]
  }
});

export { DisclosureIconWrapper, StyledContentLeft, StyledContentRight, StyledText, Wrapper };
