// @salutejs/plasma-new-hope/es/components/Dropdown/ui/DropdownItem/DropdownItem.js
import React, { useRef, useEffect } from 'react';
import { classes } from '../../Dropdown.tokens.js';
import { cx } from '../../../../utils/index.js';
import '@linaria/react';
import { IconDisclosureRight } from '../../../_Icon/Icons/IconDisclosureRight.js';
import { Wrapper, StyledContentLeft, StyledText, StyledContentRight, DisclosureIconWrapper } from './DropdownItem.styles.js';

var _DisclosureIconWrappe;
var DropdownItem = function DropdownItem(_ref) {
  var item = _ref.item,
    path = _ref.path,
    focusedPath = _ref.focusedPath,
    currentLevel = _ref.currentLevel,
    index = _ref.index,
    itemRole = _ref.itemRole,
    closeOnSelect = _ref.closeOnSelect,
    handleGlobalToggle = _ref.handleGlobalToggle,
    onHover = _ref.onHover,
    onItemSelect = _ref.onItemSelect,
    onItemClick = _ref.onItemClick,
    ariaControls = _ref.ariaControls,
    ariaExpanded = _ref.ariaExpanded,
    ariaHasPopup = _ref.ariaHasPopup,
    ariaLevel = _ref.ariaLevel,
    ariaLabel = _ref.ariaLabel,
    variant = _ref.variant,
    hasArrow = _ref.hasArrow;
  var value = item.value,
    label = item.label,
    disabled = item.disabled,
    isDisabled = item.isDisabled,
    contentLeft = item.contentLeft,
    contentRight = item.contentRight;
  var ref = useRef(null);
  var isDisabledClassName = disabled || isDisabled ? classes.dropdownItemIsDisabled : undefined;
  var focusedClass = currentLevel === focusedPath.length - 1 && index === (focusedPath === null || focusedPath === void 0 ? void 0 : focusedPath[currentLevel]) ? classes.dropdownItemIsFocused : undefined;
  var activeClass = value === (path === null || path === void 0 ? void 0 : path[currentLevel + 1]) ? classes.dropdownItemIsActive : undefined;
  useEffect(function () {
    if (focusedClass && ref !== null && ref !== void 0 && ref.current) {
      ref.current.scrollIntoView({
        behavior: 'smooth',
        block: 'center',
        inline: 'center'
      });
    }
  }, [focusedClass]);
  var handleClick = function handleClick(event) {
    if (disabled || isDisabled) {
      return;
    }
    if (onItemSelect) {
      onItemSelect(item, event);
    }
    if (onItemClick) {
      onItemClick(item, event);
    }
    if (handleGlobalToggle && closeOnSelect) {
      handleGlobalToggle(false, event);
    }
  };
  var handleHover = function handleHover() {
    if (onHover) {
      onHover(index);
    }
  };
  return /*#__PURE__*/React.createElement(Wrapper, {
    className: cx(isDisabledClassName, focusedClass, activeClass),
    id: value.toString(),
    role: itemRole,
    ref: ref,
    "aria-disabled": disabled || isDisabled,
    onClick: handleClick,
    onMouseEnter: handleHover,
    variant: variant,
    "aria-controls": ariaControls,
    "aria-expanded": ariaExpanded,
    "aria-haspopup": ariaHasPopup,
    "aria-level": ariaLevel,
    "aria-label": ariaLabel
  }, contentLeft && /*#__PURE__*/React.createElement(StyledContentLeft, null, contentLeft), /*#__PURE__*/React.createElement(StyledText, null, label), contentRight && /*#__PURE__*/React.createElement(StyledContentRight, null, contentRight), item.items && hasArrow && (_DisclosureIconWrappe || (_DisclosureIconWrappe = /*#__PURE__*/React.createElement(DisclosureIconWrapper, null, /*#__PURE__*/React.createElement(IconDisclosureRight, {
    size: "xs",
    color: "inherit"
  })))));
};

export { DropdownItem };
