// @salutejs/plasma-core/components/Toast/ToastController.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ToastController = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _Fade = /*#__PURE__*/require("../Fade");
var _utils = /*#__PURE__*/require("../../utils");
var _Toast = /*#__PURE__*/require("./Toast");
var _useToast2 = /*#__PURE__*/require("./useToast");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var showAnimation = function showAnimation(position) {
  return (0, _styledComponents.keyframes)(["0%{transform:translate(-50%,", "5rem);opacity:0;}80%{transform:translate(-50%,0);opacity:0.7;}100%{opacity:1;}"], position === 'top' && '-');
};
var hideAnimation = function hideAnimation(position) {
  return (0, _styledComponents.keyframes)(["0%{opacity:1;}20%{transform:translate(-50%,0);opacity:0.7;}100%{transform:translate(-50%,", "5rem);opacity:0;}"], position === 'top' && '-');
};
var fadeIn = /*#__PURE__*/(0, _styledComponents.keyframes)(["from{opacity:0;}to{opacity:1;}"]);
var fadeOut = /*#__PURE__*/(0, _styledComponents.keyframes)(["from{opacity:1;}to{opacity:0;}"]);
var StyledFade = /*#__PURE__*/(0, _styledComponents["default"])(_Fade.Fade).withConfig({
  componentId: "plasma__sc-5qbx6t-0"
})(["", ";"], function (_ref) {
  var isVisible = _ref.isVisible;
  return (0, _styledComponents.css)(["animation:300ms ", ";"], isVisible ? fadeIn : fadeOut);
});
var StyledRoot = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-5qbx6t-1"
})(["position:fixed;left:50%;z-index:1000;transform:translateX(-50%);", ";display:flex;width:0;", "{transform:translateX(-50%);max-width:calc(100vw - 5rem);flex-shrink:0;}"], function (_ref2) {
  var $position = _ref2.$position,
    isVisible = _ref2.isVisible,
    _ref2$$offset = _ref2.$offset,
    $offset = _ref2$$offset === void 0 ? 0 : _ref2$$offset;
  return (0, _styledComponents.css)(["", ":", "rem;animation:300ms ", ";"], $position, $offset + 5, isVisible ? showAnimation($position) : hideAnimation($position));
}, _Toast.StyledRoot);

/**
 * Создаёт <div />, который внутри себя содержит тост.
 * Цикл: show => timeout => hide.
 */
var ToastController = exports.ToastController = function ToastController(_ref3) {
  var role = _ref3.role,
    text = _ref3.text,
    contentLeft = _ref3.contentLeft,
    position = _ref3.position,
    timeout = _ref3.timeout,
    fade = _ref3.fade,
    offset = _ref3.offset;
  var _useToast = (0, _useToast2.useToast)(),
    hideToast = _useToast.hideToast;
  var _useState = (0, _react.useState)(true),
    _useState2 = _slicedToArray(_useState, 2),
    isVisible = _useState2[0],
    setIsVisible = _useState2[1];
  var hideTimeout = (0, _react.useRef)(null);
  var toastKey = "".concat(text).concat(position);
  var animationEndHandler = (0, _react.useCallback)(function () {
    if (!isVisible) {
      hideToast();
      (0, _utils.safeFlushSync)(function () {
        setIsVisible(true); // Необходимо вернуть булево к следующему вызову тоста
      });
    }
    if (isVisible) {
      hideTimeout.current = setTimeout(function () {
        setIsVisible(false);
      }, timeout);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [timeout, isVisible]);
  (0, _react.useEffect)(function () {
    if (timeout === null && hideTimeout.current !== null) {
      clearTimeout(hideTimeout.current);
    }
  }, [timeout]);
  (0, _react.useEffect)(function () {
    return function () {
      if (hideTimeout.current !== null) {
        clearTimeout(hideTimeout.current);
      }
    };
  }, []);
  if (!text || !position) {
    return null;
  }
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, fade && /*#__PURE__*/_react["default"].createElement(StyledFade, {
    isVisible: isVisible,
    placement: position
  }), /*#__PURE__*/_react["default"].createElement(StyledRoot, {
    key: toastKey,
    $position: position,
    $offset: offset,
    isVisible: isVisible,
    onAnimationEnd: animationEndHandler
  }, /*#__PURE__*/_react["default"].createElement(_Toast.Toast, {
    role: role,
    text: text,
    contentLeft: contentLeft
  })));
};
