// @salutejs/plasma-new-hope/styled-components/cjs/components/Calendar/mixins/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.selectedMixin = exports.flexSpaceBetween = exports.flexCenter = exports.buttonFocus = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _Calendar = /*#__PURE__*/require("../Calendar.tokens");
var _mixins = /*#__PURE__*/require("../../../mixins");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var flexCenter = exports.flexCenter = "\n    display: flex;\n    justify-content: center;\n    align-items: center;\n";
var flexSpaceBetween = exports.flexSpaceBetween = "\n    display: flex;\n    justify-content: space-between;\n    align-items: center;\n";

/**
 * Миксин для установки синтетического фокуса на выбранной кнопке.
 */
var buttonFocus = exports.buttonFocus = /*#__PURE__*/_styledComponents["default"].button.withConfig({
  componentId: "plasma-new-hope__sc-1tntqjy-0"
})(["border:none;background-color:transparent;padding:0;outline:none;", ";"], /*#__PURE__*/(0, _mixins.addFocus)({
  outlineSize: '0.063rem',
  outlineOffset: '-0.125rem',
  outlineColor: /*#__PURE__*/"var(".concat(_Calendar.tokens.calendarOutlineFocusColor, ")"),
  outlineRadius: '0.563rem'
}));

/**
 * Миксины для изменения стиля выбранного дня.
 */
var active = "\n    &:active {\n        background-color: var(--calendar_private-active-background);\n        color: var(--calendar_private-active-color);\n        font-weight: var(--plasma-typo-body-s-bold-font-weight);\n    }\n";

/*
 *  Если использовать токены и классы из ./tokens rollup ругается на этот миксин
 */
var selectedMixin = exports.selectedMixin = /*#__PURE__*/"\n    outline: none;\n    cursor: pointer;\n\n    &:hover:not(.item-selected) {\n        background-color: var(--calendar_private-selectable-background-hover);\n        \n        ".concat(String(active), ";\n    };\n\n    &.item-selected {\n        border: 0;\n        background-color: var(--calendar_private-selected-background);\n        color: var(--calendar_private-selected-color);\n        cursor: default;\n        font-weight: var(--plasma-typo-body-s-bold-font-weight);\n    };\n\n    &.item-current {\n        border: 0.063rem solid var(--calendar_private-current-border-color);\n\n        &:not(.item-selected):hover {\n            background-color: var(--calendar_private-current-background-hover);\n            color: var(--calendar_private-current-color-hover);\n\n            & > div:first-of-type {\n                background-color: var(--calendar_private-current-child-background-hover);\n                min-width: var(--calendar_private-selected-min-width);\n                min-height: var(--calendar_private-selected-min-height);\n            }\n        \n            ").concat(String(active), ";\n        }\n    };\n\n    &.item-hovered, &.item-hovered:hover {\n        background-color: var(--calendar_private-hovered-background);\n        color: var(--calendar_private-hovered-color);\n        cursor: pointer;\n        font-weight: var(--plasma-typo-body-s-bold-font-weight);\n        ").concat(String(active), ";\n    }\n\n    ").concat( /*#__PURE__*/String(active), ";\n");
