// @salutejs/plasma-hope/components/PreviewGallery/PreviewGalleryItemBase.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.buttonMixin = exports.StyledTrashButton = exports.StyledSelectButton = exports.StyledItem = exports.StyledIconDrag = exports.StyledCaption = exports.PreviewGalleryItemBase = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _Typography = /*#__PURE__*/require("../Typography");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _StyledSelectButton, _StyledIconDrag;
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
(0, _collectPackageInfo.collectPackageInfo)('PreviewGallery/PreviewGalleryItemBase');
var buttonMixin = exports.buttonMixin = function buttonMixin(vPosition, bgColor, color) {
  return (0, _styledComponents.css)(["position:absolute;top:0.25rem;", ":0.25rem;display:flex;align-items:center;justify-content:center;width:1.25rem;height:1.25rem;padding:0;appearance:none;background-color:", ";border:0 none;border-radius:100%;color:", ";cursor:pointer;"], vPosition, bgColor, color);
};
var StyledSelectButton = exports.StyledSelectButton = /*#__PURE__*/_styledComponents["default"].button.withConfig({
  componentId: "plasma-web-core__sc-1hth9wg-0"
})(["", ";"], /*#__PURE__*/buttonMixin('left', _plasmaCore.buttonAccent, _plasmaCore.whitePrimary));
var StyledTrashButton = exports.StyledTrashButton = /*#__PURE__*/_styledComponents["default"].button.withConfig({
  componentId: "plasma-web-core__sc-1hth9wg-1"
})(["", ";&:hover{color:", ";}"], /*#__PURE__*/buttonMixin('right', _plasmaCore.critical, _plasmaCore.whiteSecondary), _plasmaCore.whitePrimary);
var StyledIconDrag = exports.StyledIconDrag = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaIcons.IconDrag).withConfig({
  componentId: "plasma-web-core__sc-1hth9wg-2"
})(["", ";svg{opacity:0.24;}"], /*#__PURE__*/buttonMixin('left', 'transparent', _plasmaCore.whitePrimary));

// NOTE: don't know how to fix ts error TS2742

var StyledCaption = exports.StyledCaption = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Footnote2).withConfig({
  componentId: "plasma-web-core__sc-1hth9wg-3"
})(["position:absolute;left:0.625rem;bottom:0.5rem;color:", ";"], _plasmaCore.whiteTertiary);
var StyledItem = exports.StyledItem = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web-core__sc-1hth9wg-4"
})(["position:relative;border-radius:0.75rem;height:100%;", " ", "{display:none;}&:hover{transform:scale(1.04);", "{display:flex;}}"], function (_ref) {
  var width = _ref.width;
  return width && (0, _styledComponents.css)(["width:", ";"], width);
}, StyledTrashButton, StyledTrashButton);
/**
 * Базовый компонент с превью.
 */
var PreviewGalleryItemBase = exports.PreviewGalleryItemBase = /*#__PURE__*/(0, _react.memo)(function (_ref2) {
  var id = _ref2.id,
    itemSize = _ref2.itemSize,
    _ref2$image = _ref2.image,
    image = _ref2$image === void 0 ? '' : _ref2$image,
    component = _ref2.component,
    customClickHandle = _ref2.customClickHandle,
    isSelected = _ref2.isSelected,
    caption = _ref2.caption,
    interactionType = _ref2.interactionType,
    actionIcon = _ref2.actionIcon,
    status = _ref2.status,
    onItemAction = _ref2.onItemAction,
    onItemClick = _ref2.onItemClick,
    actionDisabled = _ref2.actionDisabled;
  var itemActionHandle = function itemActionHandle(event) {
    event.stopPropagation();
    onItemAction === null || onItemAction === void 0 || onItemAction(id);
  };
  return !component ? /*#__PURE__*/_react["default"].createElement(StyledItem, {
    width: itemSize,
    key: id,
    onClick: customClickHandle ? function () {
      return customClickHandle(id);
    } : function () {
      return onItemClick === null || onItemClick === void 0 ? void 0 : onItemClick(id);
    }
  }, /*#__PURE__*/_react["default"].createElement(_plasmaCore.Image, {
    base: "div",
    src: image,
    ratio: "16 / 9"
  }), !actionDisabled && /*#__PURE__*/_react["default"].createElement(StyledTrashButton, {
    type: "button",
    onClick: itemActionHandle
  }, actionIcon), interactionType === 'selectable' && status !== 'error' ? isSelected && (_StyledSelectButton || (_StyledSelectButton = /*#__PURE__*/_react["default"].createElement(StyledSelectButton, {
    type: "button"
  }, /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconDone, {
    size: "xs",
    color: "inherit"
  })))) : _StyledIconDrag || (_StyledIconDrag = /*#__PURE__*/_react["default"].createElement(StyledIconDrag, {
    size: "xs",
    color: "inherit"
  })), caption && /*#__PURE__*/_react["default"].createElement(StyledCaption, null, caption)) : /*#__PURE__*/_react["default"].createElement(StyledItem, {
    width: itemSize,
    key: id,
    onClick: customClickHandle ? function () {
      return customClickHandle(id);
    } : function () {
      return onItemClick === null || onItemClick === void 0 ? void 0 : onItemClick(id);
    }
  }, component);
});
