// @salutejs/plasma-hope/components/PreviewGallery/PreviewGalleryItems.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledRoot = exports.PreviewGalleryListItems = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _reactSortableHoc = /*#__PURE__*/require("react-sortable-hoc");
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _PreviewGalleryItemError = /*#__PURE__*/require("./PreviewGalleryItemError");
var _PreviewGalleryItemBase = /*#__PURE__*/require("./PreviewGalleryItemBase");
var _PreviewGalleryItemWithTooltip = /*#__PURE__*/require("./PreviewGalleryItemWithTooltip");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _excluded = ["status"],
  _excluded2 = ["items", "interactionType", "actionIcon", "itemSize", "isGrabbing", "maxHeight", "onItemAction", "onItemClick"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('PreviewGallery/PreviewGalleryItems');
var StyledRoot = exports.StyledRoot = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web-core__sc-av4s1g-0"
})(["display:grid;grid-template-columns:repeat(3,1fr);gap:0.5rem;cursor:pointer;", " ", ""], function (_ref) {
  var isGrabbing = _ref.isGrabbing;
  return isGrabbing && (0, _styledComponents.css)(["cursor:grabbing;"]);
}, function (_ref2) {
  var maxHeight = _ref2.maxHeight;
  return maxHeight && (0, _styledComponents.css)(["max-height:", "px;overflow:auto;"], maxHeight);
});
var PreviewGalleryItem = /*#__PURE__*/(0, _react.memo)( /*#__PURE__*/(0, _reactSortableHoc.SortableElement)(function (_ref3) {
  var status = _ref3.status,
    itemRest = _objectWithoutProperties(_ref3, _excluded);
  return status === 'error' ? /*#__PURE__*/_react["default"].createElement(_PreviewGalleryItemError.PreviewGalleryItemError, itemRest) : /*#__PURE__*/_react["default"].createElement(_PreviewGalleryItemBase.PreviewGalleryItemBase, itemRest);
}));

/**
 * Компонент со списком превью изображений.
 */
var PreviewGalleryListItems = exports.PreviewGalleryListItems = /*#__PURE__*/(0, _reactSortableHoc.SortableContainer)(function (_ref4) {
  var _ref4$items = _ref4.items,
    items = _ref4$items === void 0 ? [] : _ref4$items,
    interactionType = _ref4.interactionType,
    actionIcon = _ref4.actionIcon,
    itemSize = _ref4.itemSize,
    isGrabbing = _ref4.isGrabbing,
    maxHeight = _ref4.maxHeight,
    onItemAction = _ref4.onItemAction,
    onItemClick = _ref4.onItemClick,
    rest = _objectWithoutProperties(_ref4, _excluded2);
  var isDragDisabled = interactionType === 'selectable';

  // deleteIcon не указан в зависимости, т.к. предполагается,
  // что данный пропс не будет меняться динамически
  var iconMemo = (0, _react.useMemo)(function () {
    return actionIcon;
  }, []);
  return /*#__PURE__*/_react["default"].createElement(StyledRoot, _extends({
    isGrabbing: isGrabbing,
    maxHeight: maxHeight
  }, rest), items.map(function (item, index) {
    return /*#__PURE__*/_react["default"].createElement(_PreviewGalleryItemWithTooltip.PreviewGalleryItemWithTooltip, {
      tooltip: item.tooltip
    }, /*#__PURE__*/_react["default"].createElement(PreviewGalleryItem, _extends({
      disabled: isDragDisabled,
      key: item.id,
      actionIcon: iconMemo,
      index: index,
      interactionType: interactionType,
      itemSize: itemSize,
      onItemAction: onItemAction,
      onItemClick: onItemClick
    }, item)));
  }));
});
