// @salutejs/plasma-hope/components/PreviewGallery/PreviewGalleryItemError.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PreviewGalleryItemError = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _PreviewGalleryItemBase = /*#__PURE__*/require("./PreviewGalleryItemBase");
var _StyledErrorImage, _StyledIconDrag;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
var StyledErrorImage = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web-core__sc-qjhbf7-0"
})(["position:relative;border-radius:0.75rem;padding-bottom:56.25%;background-color:", ";color:", ";"], _plasmaCore.surfaceSolid02, _plasmaCore.backgroundPrimary);
var StyledIconPictureFilled = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaIcons.IconPictureFilled).withConfig({
  componentId: "plasma-web-core__sc-qjhbf7-1"
})(["position:absolute;top:50%;left:50%;margin-right:-50%;transform:translate(-50%,-50%);"]);

/**
 * Базовый компонент с ошибкой.
 */
var PreviewGalleryItemError = exports.PreviewGalleryItemError = /*#__PURE__*/(0, _react.memo)(function (_ref) {
  var id = _ref.id,
    itemSize = _ref.itemSize,
    interactionType = _ref.interactionType,
    actionIcon = _ref.actionIcon,
    onItemAction = _ref.onItemAction;
  var itemRemoveHandle = function itemRemoveHandle(event) {
    event.stopPropagation();
    onItemAction === null || onItemAction === void 0 || onItemAction(id);
  };
  return /*#__PURE__*/_react["default"].createElement(_PreviewGalleryItemBase.StyledItem, {
    width: itemSize,
    key: id
  }, _StyledErrorImage || (_StyledErrorImage = /*#__PURE__*/_react["default"].createElement(StyledErrorImage, null, /*#__PURE__*/_react["default"].createElement(StyledIconPictureFilled, {
    size: "s",
    color: "inherit"
  }))), /*#__PURE__*/_react["default"].createElement(_PreviewGalleryItemBase.StyledTrashButton, {
    onClick: itemRemoveHandle
  }, actionIcon), interactionType !== 'selectable' && (_StyledIconDrag || (_StyledIconDrag = /*#__PURE__*/_react["default"].createElement(_PreviewGalleryItemBase.StyledIconDrag, {
    size: "xs",
    color: "inherit"
  }))));
});
