// @salutejs/plasma-new-hope/es/components/Modal/hooks/useModal.js
import { objectSpread2 as _objectSpread2 } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import { useRef, useCallback, useEffect } from 'react';
import { getIdLastModal, hasModals } from '../ModalContext.js';
import { usePopupContext } from '../../Popup/PopupContext.js';
import 'react-dom';
import '@salutejs/plasma-core';
import '../../../utils/react.js';
import '@linaria/react';

var ESCAPE_KEYCODE = 27;
var useModal = function useModal(_ref) {
  var id = _ref.id,
    isOpen = _ref.isOpen,
    popupInfo = _ref.popupInfo,
    onEscKeyDown = _ref.onEscKeyDown,
    onClose = _ref.onClose,
    _ref$closeOnEsc = _ref.closeOnEsc,
    closeOnEsc = _ref$closeOnEsc === void 0 ? true : _ref$closeOnEsc;
  var popupController = usePopupContext();
  var overflow = useRef(document.body.style.overflowY);

  // При ESC закрывает текущее окно, если это возможно
  var onKeyDown = useCallback(function (event) {
    if (!closeOnEsc) {
      return;
    }
    if (event.keyCode === ESCAPE_KEYCODE && getIdLastModal(popupController.items) === id) {
      if (onEscKeyDown) {
        onEscKeyDown(event);
        return;
      }
      if (onClose) {
        onClose();
      }
    }
  }, [onClose, onEscKeyDown, popupController.items, closeOnEsc]);
  useEffect(function () {
    window.addEventListener('keydown', onKeyDown);
    return function () {
      window.removeEventListener('keydown', onKeyDown);
    };
  }, [onClose, onEscKeyDown, popupController.items, closeOnEsc]);

  // linaria не поддерживает динамический global
  useEffect(function () {
    if (isOpen) {
      overflow.current = document.body.style.overflowY;
      document.body.style.overflowY = 'hidden';
    }
  }, [isOpen]);
  useEffect(function () {
    if (!isOpen && !hasModals(popupController.items)) {
      document.body.style.overflowY = overflow.current;
    }
  }, [isOpen, overflow.current, popupController.items]);
  var modalInfo = _objectSpread2({
    id: id,
    info: {
      isModal: true
    }
  }, popupInfo);
  return {
    modalInfo: modalInfo
  };
};

export { useModal };
