// @salutejs/plasma-hope/utils/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  animatedScrollToX: true,
  animatedScrollToY: true,
  canUseDOM: true,
  convertRoundnessMatrix: true,
  extractTextFrom: true,
  padZeroNumber: true,
  monthLongName: true,
  monthShortName: true,
  formatCurrency: true,
  setRef: true,
  toCssSize: true,
  transformStyles: true,
  mediaQuery: true,
  gridBreakpoints: true,
  gridColumns: true,
  gridMargins: true,
  gridGutters: true,
  gridSizes: true
};
Object.defineProperty(exports, "animatedScrollToX", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.animatedScrollToX;
  }
});
Object.defineProperty(exports, "animatedScrollToY", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.animatedScrollToY;
  }
});
Object.defineProperty(exports, "canUseDOM", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.canUseDOM;
  }
});
Object.defineProperty(exports, "convertRoundnessMatrix", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.convertRoundnessMatrix;
  }
});
Object.defineProperty(exports, "extractTextFrom", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.extractTextFrom;
  }
});
Object.defineProperty(exports, "formatCurrency", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.formatCurrency;
  }
});
Object.defineProperty(exports, "gridBreakpoints", {
  enumerable: true,
  get: function get() {
    return _mediaQuery.breakpoints;
  }
});
Object.defineProperty(exports, "gridColumns", {
  enumerable: true,
  get: function get() {
    return _mediaQuery.columns;
  }
});
Object.defineProperty(exports, "gridGutters", {
  enumerable: true,
  get: function get() {
    return _mediaQuery.gutters;
  }
});
Object.defineProperty(exports, "gridMargins", {
  enumerable: true,
  get: function get() {
    return _mediaQuery.margins;
  }
});
Object.defineProperty(exports, "gridSizes", {
  enumerable: true,
  get: function get() {
    return _mediaQuery.sizes;
  }
});
Object.defineProperty(exports, "mediaQuery", {
  enumerable: true,
  get: function get() {
    return _mediaQuery.mediaQuery;
  }
});
Object.defineProperty(exports, "monthLongName", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.monthLongName;
  }
});
Object.defineProperty(exports, "monthShortName", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.monthShortName;
  }
});
Object.defineProperty(exports, "padZeroNumber", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.padZeroNumber;
  }
});
Object.defineProperty(exports, "setRef", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.setRef;
  }
});
Object.defineProperty(exports, "toCssSize", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.toCssSize;
  }
});
Object.defineProperty(exports, "transformStyles", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.transformStyles;
  }
});
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _mediaQuery = /*#__PURE__*/require("./mediaQuery");
var _clearSelection = /*#__PURE__*/require("./clearSelection");
Object.keys(_clearSelection).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _clearSelection[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _clearSelection[key];
    }
  });
});
var _selectText = /*#__PURE__*/require("./selectText");
Object.keys(_selectText).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _selectText[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _selectText[key];
    }
  });
});
