// @salutejs/plasma-new-hope/styled-components/es/components/Tabs/ui/Tabs/Tabs.styles.js
var _templateObject;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css } from 'styled-components';
import styled from 'styled-components';
import { addFocus } from '../../../../mixins';
import { tokens } from '../../tokens';
export var base = /*#__PURE__*/css(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n    display: flex;\n    align-items: center;\n    position: relative;\n"])));
export var StyledContent = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1ibkmsl-0"
})(["display:inline-flex;flex-direction:row;align-items:center;"]);
export var StyledContentWrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1ibkmsl-1"
})(["margin:-0.125rem;padding:0.25rem;box-sizing:content-box;overflow:scroll;position:relative;height:100%;width:100%;display:flex;align-items:center;scrollbar-width:none;::-webkit-scrollbar{display:none;}overscroll-behavior:contain;"]);
export var StyledArrow = /*#__PURE__*/styled.button.withConfig({
  componentId: "plasma-new-hope__sc-1ibkmsl-2"
})(["display:flex;cursor:pointer;border:none;background-color:transparent;padding:0;outline:none;", ";&[disabled]{cursor:not-allowed;}--plasma_private-outer-padding:", ";padding-right:", ";padding-left:", ";"], /*#__PURE__*/addFocus({
  outlineSize: '0.063rem',
  outlineOffset: '-0.125rem',
  outlineColor: /*#__PURE__*/"var(".concat(tokens.outlineFocusColor, ")"),
  outlineRadius: 'calc(var(--plasma_private-outline-radius) - 0.063rem)'
}), function (_ref) {
  var isFilled = _ref.isFilled;
  return isFilled ? "var(".concat(tokens.arrowOuterPadding, ")") : '';
}, function (_ref2) {
  var isLeftArrow = _ref2.isLeftArrow;
  return isLeftArrow ? "var(".concat(tokens.arrowInnerPadding, ")") : 'var(--plasma_private-outer-padding)';
}, function (_ref3) {
  var isLeftArrow = _ref3.isLeftArrow;
  return isLeftArrow ? 'var(--plasma_private-outer-padding)' : "var(".concat(tokens.arrowInnerPadding, ")");
});
