// @salutejs/plasma-hope/components/Dropdown/DropdownItem.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DropdownItem = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _StyledDot, _StyledChevron, _StyledCheck;
var _excluded = ["value", "label", "isActive", "isDisabled", "isHovered", "color", "contentLeft", "items", "role", "index", "onClick", "onHover", "onFocus"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('Dropdown/DropdownItem');
var StyledDropdownItem = /*#__PURE__*/_styledComponents["default"].li.withConfig({
  componentId: "plasma-web-core__sc-sdbmry-0"
})(["display:flex;flex:1;align-items:center;justify-content:flex-start;box-sizing:border-box;text-decoration:none;height:var(--plasma-dropdown-item-height,3rem);padding:var(--plasma-dropdown-item-padding,1rem);border-radius:var(--plasma-dropdown-item-border-radius,0);font-family:var(--plasma-dropdown-item-font-family,", ");font-size:var(--plasma-dropdown-item-font-size,", ");font-weight:var(--plasma-dropdown-item-font-weight,", ");line-height:var(--plasma-dropdown-item-line-height,", ");letter-spacing:var(--plasma-dropdown-item-letter-spacing,", ");background-color:transparent;color:", ";transition:background-color 0.3s ease-in-out;cursor:pointer;", " &:focus{outline:0 none;}", " &:active{background-color:", ";}", ""], _plasmaCore.body1.fontFamily, _plasmaCore.body1.fontSize, _plasmaCore.body1.fontWeight, _plasmaCore.body1.lineHeight, _plasmaCore.body1.letterSpacing, function (_ref) {
  var $color = _ref.$color;
  return $color || _plasmaCore.primary;
}, function (_ref2) {
  var $withAssistive = _ref2.$withAssistive,
    $color = _ref2.$color;
  return !$withAssistive && (0, _styledComponents.css)(["&:hover,&:focus{color:", ";background-color:", ";}"], $color || _plasmaCore.primary, _plasmaCore.surfaceLiquid02);
}, function (_ref3) {
  var $hover = _ref3.$hover,
    $color = _ref3.$color;
  return $hover && (0, _styledComponents.css)(["color:", ";background-color:", ";"], $color || _plasmaCore.primary, _plasmaCore.surfaceLiquid02);
}, _plasmaCore.surfaceLiquid03, _plasmaCore.applyDisabled);
var StyledContent = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web-core__sc-sdbmry-1"
})(["display:inline-flex;margin-right:0.375rem;"]);
var StyledText = /*#__PURE__*/_styledComponents["default"].span.withConfig({
  componentId: "plasma-web-core__sc-sdbmry-2"
})(["", ";margin-right:0.625rem;pointer-events:none;user-select:none;"], _plasmaCore.applyEllipsis);
var StyledChevron = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaIcons.IconChevronRight).withConfig({
  componentId: "plasma-web-core__sc-sdbmry-3"
})(["margin-left:auto;pointer-events:none;margin-right:-0.25rem;"]);
var StyledCheck = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaIcons.IconDone).withConfig({
  componentId: "plasma-web-core__sc-sdbmry-4"
})(["margin-left:auto;pointer-events:none;margin-right:-0.5rem;"]);
var StyledDot = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web-core__sc-sdbmry-5"
})(["width:0.375rem;height:0.375rem;background-color:", ";border-radius:50%;"], _plasmaCore.accent);

/**
 * Элемент выпадающего списка.
 */
var DropdownItem = exports.DropdownItem = /*#__PURE__*/(0, _react.forwardRef)(function (props, outerRef) {
  var _items$filter;
  var value = props.value,
    label = props.label,
    isActive = props.isActive,
    isDisabled = props.isDisabled,
    isHovered = props.isHovered,
    color = props.color,
    contentLeft = props.contentLeft,
    _props$items = props.items,
    items = _props$items === void 0 ? [] : _props$items,
    _props$role = props.role,
    role = _props$role === void 0 ? 'menuitem' : _props$role,
    index = props.index,
    onClickExternal = props.onClick,
    onHover = props.onHover,
    onFocus = props.onFocus,
    rest = _objectWithoutProperties(props, _excluded);
  var ref = (0, _react.useRef)(null);
  var forkRef = (0, _plasmaCore.useForkRef)(outerRef, ref);
  var hasItems = Boolean(items.length);
  var isActiveAsSingleOrNode = Boolean(isActive || ((_items$filter = items.filter(function (item) {
    return item.isActive;
  })) === null || _items$filter === void 0 ? void 0 : _items$filter.length));
  var contentRight = (0, _react.useMemo)(function () {
    if (hasItems) {
      return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, isActiveAsSingleOrNode && (_StyledDot || (_StyledDot = /*#__PURE__*/_react["default"].createElement(StyledDot, null))), _StyledChevron || (_StyledChevron = /*#__PURE__*/_react["default"].createElement(StyledChevron, {
        size: "xs"
      })));
    }
    if (isActive) {
      return _StyledCheck || (_StyledCheck = /*#__PURE__*/_react["default"].createElement(StyledCheck, {
        size: "s",
        color: _plasmaCore.accent
      }));
    }
    return null;
  }, [isActive, isActiveAsSingleOrNode, hasItems]);
  var onClick = (0, _react.useCallback)(function (event) {
    var _ref$current;
    event.preventDefault();
    var targetIsItem = event.target === ref.current;
    var targetInItem = (_ref$current = ref.current) === null || _ref$current === void 0 ? void 0 : _ref$current.contains(event.target);
    if (value !== undefined && !isDisabled && (targetIsItem || targetInItem)) {
      onClickExternal === null || onClickExternal === void 0 || onClickExternal({
        value: value,
        label: label
      }, event);
    }
  }, [value, label, isDisabled, onClickExternal]);
  var onMouseOver = (0, _react.useCallback)(function () {
    onHover === null || onHover === void 0 || onHover(index);
  }, [value, label, isDisabled, onClickExternal]);
  return /*#__PURE__*/_react["default"].createElement(StyledDropdownItem, _extends({}, rest, {
    ref: forkRef,
    $withAssistive: Boolean(onHover),
    $hover: isHovered,
    $disabled: isDisabled,
    $color: color,
    role: role,
    "aria-label": label,
    "aria-disabled": isDisabled,
    "aria-selected": role === 'option' ? isActiveAsSingleOrNode : undefined,
    onClick: onClick,
    onMouseOver: onMouseOver,
    onFocus: onFocus
  }), contentLeft && /*#__PURE__*/_react["default"].createElement(StyledContent, null, contentLeft), label && /*#__PURE__*/_react["default"].createElement(StyledText, {
    role: "presentation"
  }, label), contentRight);
});
