// @salutejs/plasma-new-hope/styled-components/cjs/examples/sds_engineer/components/Divider/Divider.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _Divider = /*#__PURE__*/require("../../../../components/Divider");
var config = exports.config = {
  defaults: {
    view: 'default',
    size: 'm',
    orientation: 'horizontal'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--surface-transparent-tertiary);"], _Divider.dividerTokens.background),
      dark: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--on-dark-surface-transparent-tertiary);"], _Divider.dividerTokens.background),
      light: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--on-light-surface-transparent-tertiary);"], _Divider.dividerTokens.background),
      // TODO поменять на корректный токен --inverse-surface-transparent-tertiary когда он появится
      inverse: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--inverse-text-tertiary);"], _Divider.dividerTokens.background)
    },
    size: {
      m: /*#__PURE__*/(0, _styledComponents.css)(["", ":0.0625rem;"], _Divider.dividerTokens.borderRadius)
    },
    orientation: {
      horizontal: /*#__PURE__*/(0, _styledComponents.css)(["", ":0.0625rem;"], _Divider.dividerTokens.baseSideSize),
      vertical: /*#__PURE__*/(0, _styledComponents.css)(["", ":0.0625rem;"], _Divider.dividerTokens.baseSideSize)
    }
  }
};
