// @salutejs/plasma-tokens/es/brands/flamingo/index.js
// Generated by robots, do not change this manually!
/** Основной цвет текста */
export var textPrimaryHover = 'var(--text-primary-hover, #CCCCCCFF)';
/** Основной цвет текста */
export var textPrimaryActive = 'var(--text-primary-active, #E6E6E6FF)';
/** Основной цвет текста */
export var textPrimary = 'var(--text-primary, #FFFFFF)';
/** Вторичный цвет текста */
export var textSecondaryHover = 'var(--text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста */
export var textSecondaryActive = 'var(--text-secondary-active, #3D3D3DFF)';
/** Вторичный цвет текста */
export var textSecondary = 'var(--text-secondary, #B3B3B3)';
/** Третичный цвет текста */
export var textTertiaryHover = 'var(--text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста */
export var textTertiaryActive = 'var(--text-tertiary-active, #737373FF)';
/** Третичный цвет текста */
export var textTertiary = 'var(--text-tertiary, #707070)';
/** Акцентный цвет */
export var textAccentHover = 'var(--text-accent-hover, #2ACB47FF)';
/** Акцентный цвет */
export var textAccentActive = 'var(--text-accent-active, #1F9835FF)';
/** Акцентный цвет */
export var textAccent = 'var(--text-accent, #24B23E)';
/** Акцентный цвет с градиентом */
export var textAccentGradientHover = 'var(--text-accent-gradient-hover, #CCCCCCFF)';
/** Акцентный цвет с градиентом */
export var textAccentGradientActive = 'var(--text-accent-gradient-active, #E6E6E6FF)';
/** Акцентный цвет с градиентом */
export var textAccentGradient = 'var(--text-accent-gradient, linear-gradient(45deg, #0B8C0B 0%, #14CC98 100%))';
/** Цвет успеха */
export var textPositiveHover = 'var(--text-positive-hover, #1EB83AFF)';
/** Цвет успеха */
export var textPositiveActive = 'var(--text-positive-active, #15842AFF)';
/** Цвет успеха */
export var textPositive = 'var(--text-positive, #1A9E32)';
/** Цвет предупреждения */
export var textWarningHover = 'var(--text-warning-hover, #FB7223FF)';
/** Цвет предупреждения */
export var textWarningActive = 'var(--text-warning-active, #DC5304FF)';
/** Цвет предупреждения */
export var textWarning = 'var(--text-warning, #FA5F05)';
/** Цвет ошибки */
export var textNegativeHover = 'var(--text-negative-hover, #FF475AFF)';
/** Цвет ошибки */
export var textNegativeActive = 'var(--text-negative-active, #FF0A23FF)';
/** Цвет ошибки */
export var textNegative = 'var(--text-negative, #FF293E)';
export var textInfoHover = 'var(--text-info-hover, #50B1F2FF)';
export var textInfoActive = 'var(--text-info-active, #0F7BC2FF)';
export var textInfo = 'var(--text-info, #118CDF)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimaryHover = 'var(--on-dark-text-primary-hover, #CCCCCCFF)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimaryActive = 'var(--on-dark-text-primary-active, #E6E6E6FF)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimary = 'var(--on-dark-text-primary, #FFFFFF)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondaryHover = 'var(--on-dark-text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondaryActive = 'var(--on-dark-text-secondary-active, #3D3D3DFF)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondary = 'var(--on-dark-text-secondary, #B3B3B3)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiaryHover = 'var(--on-dark-text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiaryActive = 'var(--on-dark-text-tertiary-active, #737373FF)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiary = 'var(--on-dark-text-tertiary, #707070)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimaryHover = 'var(--on-light-text-primary-hover, #FFFFFFFF)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimaryActive = 'var(--on-light-text-primary-active, #ADADADFF)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimary = 'var(--on-light-text-primary, #262626)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondaryHover = 'var(--on-light-text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondaryActive = 'var(--on-light-text-secondary-active, #8C8C8CFF)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondary = 'var(--on-light-text-secondary, #4E4E4E)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiaryHover = 'var(--on-light-text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiaryActive = 'var(--on-light-text-tertiary-active, #575757FF)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiary = 'var(--on-light-text-tertiary, #949494)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimaryHover = 'var(--inverse-text-primary-hover, #FFFFFFFF)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimaryActive = 'var(--inverse-text-primary-active, #ADADADFF)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimary = 'var(--inverse-text-primary, #262626)';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondaryHover = 'var(--inverse-text-secondary-hover, #FFFFFFFF)';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondaryActive = 'var(--inverse-text-secondary-active, #8C8C8CFF)';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondary = 'var(--inverse-text-secondary, #4E4E4E)';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiaryHover = 'var(--inverse-text-tertiary-hover, #FFFFFFFF)';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiaryActive = 'var(--inverse-text-tertiary-active, #575757FF)';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiary = 'var(--inverse-text-tertiary, #949494)';
/** Инвертированный акцентный цвет */
export var inverseTextAccentHover = 'var(--inverse-text-accent-hover, #13AA2CFF)';
/** Инвертированный акцентный цвет */
export var inverseTextAccentActive = 'var(--inverse-text-accent-active, #0D731EFF)';
/** Инвертированный акцентный цвет */
export var inverseTextAccent = 'var(--inverse-text-accent, #108E26)';
/** Инвертированный акцентный цвет с градиентом */
export var inverseTextAccentGradientHover = 'var(--inverse-text-accent-gradient-hover, #CCCCCCFF)';
/** Инвертированный акцентный цвет с градиентом */
export var inverseTextAccentGradientActive = 'var(--inverse-text-accent-gradient-active, #E6E6E6FF)';
/** Инвертированный акцентный цвет с градиентом */
export var inverseTextAccentGradient = 'var(--inverse-text-accent-gradient, linear-gradient(45deg, #108210 0%, #14CC98 100%))';
/** Инвертированный цвет успеха */
export var inverseTextPositiveHover = 'var(--inverse-text-positive-hover, #13AA2CFF)';
/** Инвертированный цвет успеха */
export var inverseTextPositiveActive = 'var(--inverse-text-positive-active, #0D731EFF)';
/** Инвертированный цвет успеха */
export var inverseTextPositive = 'var(--inverse-text-positive, #108E26)';
/** Инвертированный цвет предупреждения */
export var inverseTextWarningHover = 'var(--inverse-text-warning-hover, #FD650DFF)';
/** Инвертированный цвет предупреждения */
export var inverseTextWarningActive = 'var(--inverse-text-warning-active, #CA4B02FF)';
/** Инвертированный цвет предупреждения */
export var inverseTextWarning = 'var(--inverse-text-warning, #E85702)';
/** Инвертированный цвет ошибки */
export var inverseTextNegativeHover = 'var(--inverse-text-negative-hover, #ED263AFF)';
/** Инвертированный цвет ошибки */
export var inverseTextNegativeActive = 'var(--inverse-text-negative-active, #C61022FF)';
/** Инвертированный цвет ошибки */
export var inverseTextNegative = 'var(--inverse-text-negative, #E31227)';
export var inverseTextInfoHover = 'var(--inverse-text-info-hover, #229EF1FF)';
export var inverseTextInfoActive = 'var(--inverse-text-info-active, #0A6099FF)';
export var inverseTextInfo = 'var(--inverse-text-info, #0C72B6)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimaryHover = 'var(--surface-solid-primary-hover, #303030FF)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimaryActive = 'var(--surface-solid-primary-active, #212121FF)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimary = 'var(--surface-solid-primary, #262626)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondaryHover = 'var(--surface-solid-secondary-hover, #404040FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondaryActive = 'var(--surface-solid-secondary-active, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondary = 'var(--surface-solid-secondary, #363636)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiaryHover = 'var(--surface-solid-tertiary-hover, #595959FF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiaryActive = 'var(--surface-solid-tertiary-active, #4A4A4AFF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiary = 'var(--surface-solid-tertiary, #4E4E4E)';
/** Основной фон для карточек */
export var surfaceSolidCardHover = 'var(--surface-solid-card-hover, #303030FF)';
/** Основной фон для карточек */
export var surfaceSolidCardActive = 'var(--surface-solid-card-active, #212121FF)';
/** Основной фон для карточек */
export var surfaceSolidCard = 'var(--surface-solid-card, #262626)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefaultHover = 'var(--surface-solid-default-hover, #0D0D0DFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefaultActive = 'var(--surface-solid-default-active, #030303FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefault = 'var(--surface-solid-default, #080808)';
/** Фон поверхности/контрола без заливки */
export var surfaceClearHover = 'var(--surface-clear-hover, #00000000)';
/** Фон поверхности/контрола без заливки */
export var surfaceClearActive = 'var(--surface-clear-active, #00000000)';
/** Фон поверхности/контрола без заливки */
export var surfaceClear = 'var(--surface-clear, rgba(0,0,0,0))';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentHover = 'var(--surface-accent-hover, #1DAF37FF)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentActive = 'var(--surface-accent-active, #18952FFF)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccent = 'var(--surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradientHover = 'var(--surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradientActive = 'var(--surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(45deg, #108210 0%, #14CC98 100%))';
/** Цвет фона поверхности/контрола успех */
export var surfacePositiveHover = 'var(--surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositiveActive = 'var(--surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositive = 'var(--surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarningHover = 'var(--surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarningActive = 'var(--surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarning = 'var(--surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegativeHover = 'var(--surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegativeActive = 'var(--surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegative = 'var(--surface-negative, #FF293E)';
export var surfaceInfoHover = 'var(--surface-info-hover, #1798EEFF)';
export var surfaceInfoActive = 'var(--surface-info-active, #1086D5FF)';
export var surfaceInfo = 'var(--surface-info, #118CDF)';
export var surfacePositiveMinorHover = 'var(--surface-positive-minor-hover, #0E3A16FF)';
export var surfacePositiveMinorActive = 'var(--surface-positive-minor-active, #08210CFF)';
export var surfacePositiveMinor = 'var(--surface-positive-minor, #0A2B10)';
export var surfaceWarningMinorHover = 'var(--surface-warning-minor-hover, #4F250DFF)';
export var surfaceWarningMinorActive = 'var(--surface-warning-minor-active, #351909FF)';
export var surfaceWarningMinor = 'var(--surface-warning-minor, #3D1D0A)';
export var surfaceNegativeMinorHover = 'var(--surface-negative-minor-hover, #5B1018FF)';
export var surfaceNegativeMinorActive = 'var(--surface-negative-minor-active, #410B11FF)';
export var surfaceNegativeMinor = 'var(--surface-negative-minor, #4A0D13)';
export var surfaceInfoMinorHover = 'var(--surface-info-minor-hover, #10344CFF)';
export var surfaceInfoMinorActive = 'var(--surface-info-minor-active, #0A2333FF)';
export var surfaceInfoMinor = 'var(--surface-info-minor, #0C283B)';
/** Прозрачный цвет фона поверхности/контрола успех */
export var surfaceTransparentPositiveHover = 'var(--surface-transparent-positive-hover, #1A9E3224)';
/** Прозрачный цвет фона поверхности/контрола успех */
export var surfaceTransparentPositiveActive = 'var(--surface-transparent-positive-active, #1A9E320F)';
/** Прозрачный цвет фона поверхности/контрола успех */
export var surfaceTransparentPositive = 'var(--surface-transparent-positive, rgba(26,158,50,0.08))';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
export var surfaceTransparentWarningHover = 'var(--surface-transparent-warning-hover, #FF702424)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
export var surfaceTransparentWarningActive = 'var(--surface-transparent-warning-active, #FF70240F)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
export var surfaceTransparentWarning = 'var(--surface-transparent-warning, rgba(255,112,36,0.08))';
/** Прозрачный цвет фона поверхности/контрола ошибка */
export var surfaceTransparentNegativeHover = 'var(--surface-transparent-negative-hover, #FF3D5124)';
/** Прозрачный цвет фона поверхности/контрола ошибка */
export var surfaceTransparentNegativeActive = 'var(--surface-transparent-negative-active, #FF3D510F)';
/** Прозрачный цвет фона поверхности/контрола ошибка */
export var surfaceTransparentNegative = 'var(--surface-transparent-negative, rgba(255,61,81,0.08))';
export var surfaceTransparentInfoHover = 'var(--surface-transparent-info-hover, #199AF024)';
export var surfaceTransparentInfoActive = 'var(--surface-transparent-info-active, #199AF00F)';
export var surfaceTransparentInfo = 'var(--surface-transparent-info, rgba(25,154,240,0.08))';
export var surfaceRatingHover = 'var(--surface-rating-hover, #FFDE5CFF)';
export var surfaceRatingActive = 'var(--surface-rating-active, #FFD633FF)';
export var surfaceRating = 'var(--surface-rating, #FFD83D)';
export var surfaceRatingMinorHover = 'var(--surface-rating-minor-hover, #3C320BFF)';
export var surfaceRatingMinorActive = 'var(--surface-rating-minor-active, #221D06FF)';
export var surfaceRatingMinor = 'var(--surface-rating-minor, #2B2408)';
export var surfaceTransparentRatingHover = 'var(--surface-transparent-rating-hover, #B7920124)';
export var surfaceTransparentRatingActive = 'var(--surface-transparent-rating-active, #B792010F)';
export var surfaceTransparentRating = 'var(--surface-transparent-rating, rgba(183,146,1,0.08))';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimaryHover = 'var(--on-dark-surface-solid-primary-hover, #FAFAFAFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimaryActive = 'var(--on-dark-surface-solid-primary-active, #F0F0F0FF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #F5F5F5)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondaryHover = 'var(--on-dark-surface-solid-secondary-hover, #F7F7F7FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondaryActive = 'var(--on-dark-surface-solid-secondary-active, #E8E8E8FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #ECECEC)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefaultHover = 'var(--on-dark-surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefaultActive = 'var(--on-dark-surface-solid-default-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #FFFFFF)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimaryHover = 'var(--on-dark-surface-transparent-primary-hover, #FFFFFF1F)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimaryActive = 'var(--on-dark-surface-transparent-primary-active, #FFFFFF0A)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, rgba(255,255,255,0.06))';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondaryHover = 'var(--on-dark-surface-transparent-secondary-hover, #FFFFFF3D)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondaryActive = 'var(--on-dark-surface-transparent-secondary-active, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, rgba(255,255,255,0.12))';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiaryHover = 'var(--on-dark-surface-transparent-tertiary-hover, #FFFFFF52)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiaryActive = 'var(--on-dark-surface-transparent-tertiary-active, #FFFFFF24)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, rgba(255,255,255,0.2))';
/** Непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentDeepHover = 'var(--on-dark-surface-transparent-deep-hover, #FFFFFFC2)';
/** Непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentDeepActive = 'var(--on-dark-surface-transparent-deep-active, #FFFFFF94)';
/** Непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentDeep = 'var(--on-dark-surface-transparent-deep, rgba(255,255,255,0.64))';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCardHover = 'var(--on-dark-surface-transparent-card-hover, #FFFFFF1F)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCardActive = 'var(--on-dark-surface-transparent-card-active, #FFFFFF0A)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, rgba(255,255,255,0.06))';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidPrimaryHover = 'var(--on-light-surface-solid-primary-hover, #303030FF)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidPrimaryActive = 'var(--on-light-surface-solid-primary-active, #212121FF)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #262626)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondaryHover = 'var(--on-light-surface-solid-secondary-hover, #404040FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondaryActive = 'var(--on-light-surface-solid-secondary-active, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #363636)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefaultHover = 'var(--on-light-surface-solid-default-hover, #0D0D0DFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefaultActive = 'var(--on-light-surface-solid-default-active, #030303FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #080808)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidPrimaryHover = 'var(--inverse-surface-solid-primary-hover, #FAFAFAFF)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidPrimaryActive = 'var(--inverse-surface-solid-primary-active, #F0F0F0FF)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidPrimary = 'var(--inverse-surface-solid-primary, #F5F5F5)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidSecondaryHover = 'var(--inverse-surface-solid-secondary-hover, #F7F7F7FF)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidSecondaryActive = 'var(--inverse-surface-solid-secondary-active, #E8E8E8FF)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidSecondary = 'var(--inverse-surface-solid-secondary, #ECECEC)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
export var inverseSurfaceSolidDefaultHover = 'var(--inverse-surface-solid-default-hover, #FFFFFFFF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
export var inverseSurfaceSolidDefaultActive = 'var(--inverse-surface-solid-default-active, #FFFFFFFF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
export var inverseSurfaceSolidDefault = 'var(--inverse-surface-solid-default, #FFFFFF)';
/** Основной фон */
export var backgroundPrimary = 'var(--background-primary, #080808)';
export var backgroundGradientGreen = 'var(--background-gradient-green, linear-gradient(36.09deg, #24B23E 1.152%, #AFED00 95.819%))';
export var backgroundGradientOrange = 'var(--background-gradient-orange, linear-gradient(36.09deg, #FA6D20 1.152%, #FA9D6B 95.819%))';
export var backgroundGradientYellow = 'var(--background-gradient-yellow, linear-gradient(36.09deg, #F2C202 1.152%, #FFD83D 95.819%))';
export var backgroundGradientBlue = 'var(--background-gradient-blue, linear-gradient(36.09deg, #129DFA 1.152%, #52BAFF 95.819%))';
export var backgroundGradientRed = 'var(--background-gradient-red, linear-gradient(36.09deg, #FF2E43 1.152%, #FD96A0 95.819%))';
export var backgroundGradientPink = 'var(--background-gradient-pink, linear-gradient(34.38deg, #FF1F78 7.846%, #FF8FBC 90.55%))';
export var backgroundGradientPurple = 'var(--background-gradient-purple, linear-gradient(34.38deg, #AD42F5 7.846%, #D39CF7 90.55%))';
/** Цвет фона паранжи светлый */
export var overlaySoft = 'var(--overlay-soft, rgba(8,8,8,0.56))';
/** Цвет фона паранжи темный */
export var overlayHard = 'var(--overlay-hard, rgba(8,8,8,0.96))';
export var overlayBlur = 'var(--overlay-blur, rgba(8,8,8,0.20))';
export var outlineDefaultOutlinePositiveHover = 'var(--outline-default-outline-positive-hover, #2ACB47FF)';
export var outlineDefaultOutlinePositiveActive = 'var(--outline-default-outline-positive-active, #1F9835FF)';
export var outlineDefaultOutlinePositive = 'var(--outline-default-outline-positive, #24B23E)';
export var outlineDefaultOutlineWarningHover = 'var(--outline-default-outline-warning-hover, #FF8442FF)';
export var outlineDefaultOutlineWarningActive = 'var(--outline-default-outline-warning-active, #FF5D05FF)';
export var outlineDefaultOutlineWarning = 'var(--outline-default-outline-warning, #FF7024)';
export var outlineDefaultOutlineNegativeHover = 'var(--outline-default-outline-negative-hover, #FF5C6CFF)';
export var outlineDefaultOutlineNegativeActive = 'var(--outline-default-outline-negative-active, #FF1F35FF)';
export var outlineDefaultOutlineNegative = 'var(--outline-default-outline-negative, #FF3D51)';
export var outlineDefaultOutlineInfoHover = 'var(--outline-default-outline-info-hover, #66BCF5FF)';
export var outlineDefaultOutlineInfoActive = 'var(--outline-default-outline-info-active, #0E8ADDFF)';
export var outlineDefaultOutlineInfo = 'var(--outline-default-outline-info, #199AF0)';
export var outlineDefaultOutlinePositiveMinorHover = 'var(--outline-default-outline-positive-minor-hover, #0F760FFF)';
export var outlineDefaultOutlinePositiveMinorActive = 'var(--outline-default-outline-positive-minor-active, #0B5B0BFF)';
export var outlineDefaultOutlinePositiveMinor = 'var(--outline-default-outline-positive-minor, #084008)';
export var outlineDefaultOutlineWarningMinorHover = 'var(--outline-default-outline-warning-minor-hover, #95400FFF)';
export var outlineDefaultOutlineWarningMinorActive = 'var(--outline-default-outline-warning-minor-active, #79340CFF)';
export var outlineDefaultOutlineWarningMinor = 'var(--outline-default-outline-warning-minor, #5C2709)';
export var outlineDefaultOutlineNegativeMinorHover = 'var(--outline-default-outline-negative-minor-hover, #A01825FF)';
export var outlineDefaultOutlineNegativeMinorActive = 'var(--outline-default-outline-negative-minor-active, #85141FFF)';
export var outlineDefaultOutlineNegativeMinor = 'var(--outline-default-outline-negative-minor, #6B1019)';
export var outlineDefaultOutlineInfoMinorHover = 'var(--outline-default-outline-info-minor-hover, #145E8FFF)';
export var outlineDefaultOutlineInfoMinorActive = 'var(--outline-default-outline-info-minor-active, #114C74FF)';
export var outlineDefaultOutlineInfoMinor = 'var(--outline-default-outline-info-minor, #0D3B59)';
export var outlineDefaultOutlinePrimaryHover = 'var(--outline-default-outline-primary-hover, #FFFFFFFF)';
export var outlineDefaultOutlinePrimaryActive = 'var(--outline-default-outline-primary-active, #8C8C8CFF)';
export var outlineDefaultOutlinePrimary = 'var(--outline-default-outline-primary, #4E4E4E)';
export var outlineDefaultOutlineSecondaryHover = 'var(--outline-default-outline-secondary-hover, #FFFFFFFF)';
export var outlineDefaultOutlineSecondaryActive = 'var(--outline-default-outline-secondary-active, #737373FF)';
export var outlineDefaultOutlineSecondary = 'var(--outline-default-outline-secondary, #707070)';
export var outlineDefaultOutlineTertiaryHover = 'var(--outline-default-outline-tertiary-hover, #FFFFFFFF)';
export var outlineDefaultOutlineTertiaryActive = 'var(--outline-default-outline-tertiary-active, #575757FF)';
export var outlineDefaultOutlineTertiary = 'var(--outline-default-outline-tertiary, #949494)';
export var outlineDefaultOutlineRatingHover = 'var(--outline-default-outline-rating-hover, #FFDE5CFF)';
export var outlineDefaultOutlineRatingActive = 'var(--outline-default-outline-rating-active, #FFD21FFF)';
export var outlineDefaultOutlineRating = 'var(--outline-default-outline-rating, #FFD83D)';
export var outlineDefaultOutlineRatingMinorHover = 'var(--outline-default-outline-rating-minor-hover, #77620DFF)';
export var outlineDefaultOutlineRatingMinorActive = 'var(--outline-default-outline-rating-minor-active, #5C4B0AFF)';
export var outlineDefaultOutlineRatingMinor = 'var(--outline-default-outline-rating-minor, #403507)';
export var outlineDefaultOutlinePositiveTransparentHover = 'var(--outline-default-outline-positive-transparent-hover, #24B23EFF)';
export var outlineDefaultOutlinePositiveTransparentActive = 'var(--outline-default-outline-positive-transparent-active, #24B23E31)';
export var outlineDefaultOutlinePositiveTransparent = 'var(--outline-default-outline-positive-transparent, rgba(36,178,62,0.16))';
export var outlineDefaultOutlineWarningTransparentHover = 'var(--outline-default-outline-warning-transparent-hover, #FF7024FF)';
export var outlineDefaultOutlineWarningTransparentActive = 'var(--outline-default-outline-warning-transparent-active, #FF702431)';
export var outlineDefaultOutlineWarningTransparent = 'var(--outline-default-outline-warning-transparent, rgba(255,112,36,0.16))';
export var outlineDefaultOutlineNegativeTransparentHover = 'var(--outline-default-outline-negative-transparent-hover, #FF3D51FF)';
export var outlineDefaultOutlineNegativeTransparentActive = 'var(--outline-default-outline-negative-transparent-active, #FF3D5131)';
export var outlineDefaultOutlineNegativeTransparent = 'var(--outline-default-outline-negative-transparent, rgba(255,61,81,0.16))';
export var outlineDefaultOutlineInfoTransparentHover = 'var(--outline-default-outline-info-transparent-hover, #528EFFFF)';
export var outlineDefaultOutlineInfoTransparentActive = 'var(--outline-default-outline-info-transparent-active, #528EFF31)';
export var outlineDefaultOutlineInfoTransparent = 'var(--outline-default-outline-info-transparent, rgba(82,142,255,0.16))';
export var outlineDefaultOutlineRatingTransparentHover = 'var(--outline-default-outline-rating-transparent-hover, #FFD83DFF)';
export var outlineDefaultOutlineRatingTransparentActive = 'var(--outline-default-outline-rating-transparent-active, #FFD83D31)';
export var outlineDefaultOutlineRatingTransparent = 'var(--outline-default-outline-rating-transparent, rgba(255,216,61,0.16))';
export var skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
export var skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
export var whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
export var whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
export var whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
export var blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
export var blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
export var blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use onLightSurfaceSolidDefault */
export var buttonBlack = 'var(--plasma-colors-button-black, var(--on-light-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
export var buttonWhite = 'var(--plasma-colors-button-white, var(--on-dark-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
export var buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, var(--on-dark-surface-transparent-secondary))';
/** @deprecated instead use textPrimary */
export var text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
export var primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
export var secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
export var tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use backgroundPrimary */
export var background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textAccent */
export var accent = 'var(--plasma-colors-accent, var(--text-accent))';
/** @deprecated instead use textPositive */
export var success = 'var(--plasma-colors-success, var(--text-positive))';
/** @deprecated instead use textWarning */
export var warning = 'var(--plasma-colors-warning, var(--text-warning))';
/** @deprecated instead use textNegative */
export var critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use overlaySoft */
export var overlay = 'var(--plasma-colors-overlay, var(--overlay-soft))';
/** @deprecated instead use surfaceSolidPrimary */
export var surfaceSolid01 = 'var(--plasma-colors-surface-solid01, var(--surface-solid-primary))';
/** @deprecated instead use surfaceSolidSecondary */
export var surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
export var surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use textAccent */
export var buttonAccent = 'var(--plasma-colors-button-accent, var(--text-accent))';
/** @deprecated instead use surfacePositive */
export var buttonSuccess = 'var(--plasma-colors-button-success, var(--surface-positive))';
/** @deprecated instead use surfaceWarning */
export var buttonWarning = 'var(--plasma-colors-button-warning, var(--surface-warning))';
/** @deprecated instead use surfaceNegative */
export var buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
