// @salutejs/plasma-new-hope/styled-components/cjs/components/Popup/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "PopupProvider", {
  enumerable: true,
  get: function get() {
    return _PopupContext.PopupProvider;
  }
});
Object.defineProperty(exports, "popupClasses", {
  enumerable: true,
  get: function get() {
    return _Popup2.classes;
  }
});
Object.defineProperty(exports, "popupConfig", {
  enumerable: true,
  get: function get() {
    return _Popup.popupConfig;
  }
});
Object.defineProperty(exports, "popupRoot", {
  enumerable: true,
  get: function get() {
    return _Popup.popupRoot;
  }
});
Object.defineProperty(exports, "usePopup", {
  enumerable: true,
  get: function get() {
    return _hooks.usePopup;
  }
});
Object.defineProperty(exports, "usePopupContext", {
  enumerable: true,
  get: function get() {
    return _PopupContext.usePopupContext;
  }
});
var _PopupContext = /*#__PURE__*/require("./PopupContext");
var _Popup = /*#__PURE__*/require("./Popup");
var _hooks = /*#__PURE__*/require("./hooks");
var _Popup2 = /*#__PURE__*/require("./Popup.tokens");
