// @salutejs/plasma-core/components/Typography/Headline.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Headline4 = exports.Headline3 = exports.Headline2 = exports.Headline1 = exports.H4 = exports.H3 = exports.H2 = exports.H1 = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _tokens = /*#__PURE__*/require("../../tokens");
var _mixins = /*#__PURE__*/require("../../mixins");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
(0, _collectPackageInfo.collectPackageInfo)('Typography/Headline');
var Headline1 = exports.Headline1 = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-13i65ks-0"
})(["", " ", " ", ""], _mixins.applyHyphens, _mixins.applySpacing, _tokens.headline1);
var Headline2 = exports.Headline2 = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-13i65ks-1"
})(["", " ", " ", ""], _mixins.applyHyphens, _mixins.applySpacing, _tokens.headline2);
var Headline3 = exports.Headline3 = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-13i65ks-2"
})(["", " ", " ", ""], _mixins.applyHyphens, _mixins.applySpacing, _tokens.headline3);
var Headline4 = exports.Headline4 = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-13i65ks-3"
})(["", " ", " ", ""], _mixins.applyHyphens, _mixins.applySpacing, _tokens.headline4);
var H1 = exports.H1 = /*#__PURE__*/_styledComponents["default"].h1.withConfig({
  componentId: "plasma__sc-13i65ks-4"
})(["margin:0;", " ", " ", ""], _mixins.applyHyphens, _mixins.applySpacing, _tokens.headline1);
var H2 = exports.H2 = /*#__PURE__*/_styledComponents["default"].h2.withConfig({
  componentId: "plasma__sc-13i65ks-5"
})(["margin:0;", " ", " ", ""], _mixins.applyHyphens, _mixins.applySpacing, _tokens.headline2);
var H3 = exports.H3 = /*#__PURE__*/_styledComponents["default"].h3.withConfig({
  componentId: "plasma__sc-13i65ks-6"
})(["margin:0;", " ", " ", ""], _mixins.applyHyphens, _mixins.applySpacing, _tokens.headline3);
var H4 = exports.H4 = /*#__PURE__*/_styledComponents["default"].h4.withConfig({
  componentId: "plasma__sc-13i65ks-7"
})(["margin:0;", " ", " ", ""], _mixins.applyHyphens, _mixins.applySpacing, _tokens.headline4);
