// @salutejs/plasma-ui/components/Tabs/TabsView.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TabsView = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _plasmaTokens = /*#__PURE__*/require("@salutejs/plasma-tokens");
var _mixins = /*#__PURE__*/require("../../mixins");
var _TabItem = /*#__PURE__*/require("./TabItem");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
/**
 * Размеры для контейнера и айтемов (в css vars).
 * Паддинги айтемов дополнительно обыгрываются в TabItem.
 * Так нужно поступать, потому что у айтемов есть своя логика паддингов,
 * а размеры (свойство size) задаются через компонент Tabs.
 */
var sizes = {
  l: function l(stretch) {
    return (0, _styledComponents.css)(["--tab-item-padding-x:", "rem;--tab-item-padding-y:0.875rem;--tab-item-padding-y-reduced:0.75rem;--tab-item-height:3rem;--tab-item-border-radius:1rem;height:3.25rem;border-radius:1.125rem;"], stretch ? 0.75 : 1.5);
  },
  m: function m(stretch) {
    return (0, _styledComponents.css)(["--tab-item-padding-x:", "rem;--tab-item-padding-y:0.75rem;--tab-item-padding-y-reduced:0.5rem;--tab-item-height:2.5rem;--tab-item-border-radius:0.75rem;height:2.75rem;border-radius:0.875rem;"], stretch ? 0.75 : 1.25);
  },
  s: function s(stretch) {
    return (0, _styledComponents.css)(["--tab-item-padding-x:", "rem;--tab-item-padding-y:0.625rem;--tab-item-padding-y-reduced:0.375rem;--tab-item-height:2.25rem;--tab-item-border-radius:0.75rem;height:2.5rem;border-radius:0.875rem;"], stretch ? 0.625 : 1);
  }
};
var shiftingSizes = {
  l: function l(stretch) {
    return "-".concat(stretch ? 0.875 : 1.625, "rem");
  },
  m: function m(stretch) {
    return "-".concat(stretch ? 0.875 : 1.375, "rem");
  },
  s: function s(stretch) {
    return "-".concat(stretch ? 0.75 : 1.125, "rem");
  }
};
var outlineSizes = {
  l: {
    radius: '1.125rem'
  },
  m: {
    radius: '0.875rem'
  },
  s: {
    radius: '0.875rem'
  }
};

/**
 * Цвета контейнера.
 */
var views = {
  secondary: _plasmaTokens.surfaceLiquid01,
  black: _plasmaTokens.blackSecondary,
  clear: _plasmaTokens.transparent
};
/**
 * Визуальная составляющая контейнера (списка) вкладок.
 */
var TabsView = exports.TabsView = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.Tabs).withConfig({
  componentId: "plasma-ui__sc-1aott2z-0"
})(["--tab-focus-border-size:0.125rem;--tabs-margin:var(--plasma-grid-margin,0rem);", ";", "{padding:var(--tab-focus-border-size);background-color:", ";&:focus{outline:0 none;}", ";", " ", " & > ", "{", "}}"], function (_ref) {
  var _ref$size = _ref.size,
    size = _ref$size === void 0 ? 'l' : _ref$size,
    _ref$stretch = _ref.stretch,
    stretch = _ref$stretch === void 0 ? false : _ref$stretch,
    shiftLeft = _ref.shiftLeft,
    shiftRight = _ref.shiftRight;
  return (0, _styledComponents.css)(["", " ", ""], shiftLeft && (0, _styledComponents.css)(["margin-left:calc(", " - var(--tab-focus-border-size) * 2);"], shiftingSizes[size](stretch)), shiftRight && (0, _styledComponents.css)(["margin-right:", ";"], shiftingSizes[size](stretch)));
}, _plasmaCore.StyledTabs, function (_ref2) {
  var _ref2$view = _ref2.view,
    view = _ref2$view === void 0 ? 'secondary' : _ref2$view;
  return views[view];
}, function (_ref3) {
  var _ref3$size = _ref3.size,
    size = _ref3$size === void 0 ? 'l' : _ref3$size,
    _ref3$stretch = _ref3.stretch,
    stretch = _ref3$stretch === void 0 ? false : _ref3$stretch;
  return (0, _styledComponents.css)(["", ";"], sizes[size](stretch));
}, function (_ref4) {
  var pilled = _ref4.pilled;
  return pilled && (0, _styledComponents.css)(["--tab-item-border-radius:6.25rem;border-radius:6.25rem;"]);
}, function (_ref5) {
  var _ref5$size = _ref5.size,
    size = _ref5$size === void 0 ? 'l' : _ref5$size,
    pilled = _ref5.pilled;
  return (0, _styledComponents.css)(["--tab-item-outline-radius:", ";"], pilled ? '6.375rem' : outlineSizes[size].radius);
}, _TabItem.StyledTabItem, function (props) {
  return !props.disabled && (0, _mixins.applyInteraction)(props);
});
