// @salutejs/plasma-new-hope/es/components/Calendar/ui/CalendarDayItem/CalendarDayItem.styles.js
import './CalendarDayItem.styles_xf9c5e.css';
import { styled } from '@linaria/react';
import { tokens } from '../../Calendar.tokens.js';

var _exp9 = function _exp9() {
  return function (_ref) {
    var offset = _ref.offset;
    return "".concat(offset, "px");
  };
};
var StyledDay = /*#__PURE__*/styled('div')({
  name: "StyledDay",
  "class": "sotsbos",
  propsAsIs: false,
  vars: {
    "sotsbos-0": [/*#__PURE__*/_exp9()]
  }
});
var _exp18 = function _exp18() {
  return function (_ref2) {
    var isDayInCurrentMonth = _ref2.isDayInCurrentMonth;
    return isDayInCurrentMonth ? "var(".concat(tokens.calendarContentPrimaryColor, ")") : "var(".concat(tokens.calendarContentSecondaryColor, ")");
  };
};
var _exp19 = function _exp19() {
  return function (_ref3) {
    var isDayInCurrentMonth = _ref3.isDayInCurrentMonth,
      isDouble = _ref3.isDouble;
    return !isDayInCurrentMonth && isDouble ? 'hidden' : 'visible';
  };
};
var StyledDayRoot = /*#__PURE__*/styled('div')({
  name: "StyledDayRoot",
  "class": "smv00b5",
  propsAsIs: false,
  vars: {
    "smv00b5-0": [/*#__PURE__*/_exp18()],
    "smv00b5-1": [/*#__PURE__*/_exp19()]
  }
});
var StyledEvents = /*#__PURE__*/styled('div')({
  name: "StyledEvents",
  "class": "s58etsc",
  propsAsIs: false
});
var _exp40 = function _exp40() {
  return function (_ref4) {
    var _ref4$color = _ref4.color,
      color = _ref4$color === void 0 ? 'var(--plasma-colors-accent)' : _ref4$color;
    return color;
  };
};
var StyledEvent = /*#__PURE__*/styled('div')({
  name: "StyledEvent",
  "class": "say0ota",
  propsAsIs: false,
  vars: {
    "say0ota-0": [/*#__PURE__*/_exp40()]
  }
});

export { StyledDay, StyledDayRoot, StyledEvent, StyledEvents };
