// @salutejs/plasma-new-hope/styled-components/cjs/components/Cell/Cell.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.cellRoot = exports.cellConfig = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _utils = /*#__PURE__*/require("../../utils");
var _Cell = /*#__PURE__*/require("./Cell.styles");
var _Cell2 = /*#__PURE__*/require("./Cell.tokens");
var _base = /*#__PURE__*/require("./variations/_view/base");
var _base2 = /*#__PURE__*/require("./variations/_size/base");
var _CellTextbox = /*#__PURE__*/require("./ui/CellTextbox/CellTextbox");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
var cellRoot = exports.cellRoot = function cellRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (_ref, outerRootRef) {
    var size = _ref.size,
      view = _ref.view,
      contentLeft = _ref.contentLeft,
      contentRight = _ref.contentRight,
      _ref$alignContentLeft = _ref.alignContentLeft,
      alignContentLeft = _ref$alignContentLeft === void 0 ? 'center' : _ref$alignContentLeft,
      _ref$alignContentRigh = _ref.alignContentRight,
      alignContentRight = _ref$alignContentRigh === void 0 ? 'center' : _ref$alignContentRigh,
      children = _ref.children,
      title = _ref.title,
      subtitle = _ref.subtitle,
      label = _ref.label,
      description = _ref.description,
      content = _ref.content,
      _ref$stretching = _ref.stretching,
      stretching = _ref$stretching === void 0 ? 'filled' : _ref$stretching;
    var stretchingClass = _Cell2.classes["".concat(stretching, "Stretching")];

    // Импорт  старых свойств
    var titleText = title || description;
    var contentLeftDeprecated = contentLeft || content;
    return /*#__PURE__*/_react["default"].createElement(Root, {
      ref: outerRootRef,
      size: size,
      view: view,
      className: (0, _utils.cx)(stretchingClass, _Cell2.classes.cellRoot)
    }, contentLeftDeprecated && /*#__PURE__*/_react["default"].createElement(_Cell.CellLeft, {
      align: alignContentLeft
    }, contentLeftDeprecated), /*#__PURE__*/_react["default"].createElement(_Cell.CellContentWrapper, null, /*#__PURE__*/_react["default"].createElement(_Cell.CellContent, null, titleText || subtitle || label ? /*#__PURE__*/_react["default"].createElement(_CellTextbox.CellTextbox, {
      label: label,
      title: titleText,
      subtitle: subtitle
    }) : children), contentRight && /*#__PURE__*/_react["default"].createElement(_Cell.CellRight, {
      align: alignContentRight
    }, contentRight)));
  });
};
var cellConfig = exports.cellConfig = {
  name: 'Cell',
  tag: 'div',
  layout: cellRoot,
  base: _Cell.base,
  variations: {
    view: {
      css: _base.base
    },
    size: {
      css: _base2.base
    }
  },
  defaults: {
    view: 'default',
    size: 'm'
  }
};
