// @salutejs/plasma-new-hope/styled-components/cjs/components/Cell/Cell.styles.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.base = exports.CellRight = exports.CellLeft = exports.CellContentWrapper = exports.CellContent = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _Cell = /*#__PURE__*/require("./Cell.tokens");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
var alignToFlex = {
  center: 'center',
  top: 'flex-start',
  bottom: 'flex-end'
};
var base = exports.base = /*#__PURE__*/(0, _styledComponents.css)(["&&.", "{width:var(", ");}&&.", "{width:100%;}&&.", "{width:auto;}"], _Cell.classes.fixedStretching, _Cell.tokens.cellWidth, _Cell.classes.filledStretching, _Cell.classes.autoStretching);
var CellContentWrapper = exports.CellContentWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-16emz3n-0"
})(["display:flex;flex:auto;align-items:stretch;justify-content:space-between;gap:var(", ");height:auto;"], _Cell.tokens.cellGap);
var CellContent = exports.CellContent = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-16emz3n-1"
})(["display:flex;align-items:center;height:auto;"]);
var CellLeft = exports.CellLeft = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-16emz3n-2"
})(["display:flex;align-items:", ";"], function (_ref) {
  var align = _ref.align;
  return alignToFlex[align];
});
var CellRight = exports.CellRight = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-16emz3n-3"
})(["display:flex;align-items:", ";text-align:right;"], function (_ref2) {
  var align = _ref2.align;
  return alignToFlex[align];
});
