// @salutejs/plasma-new-hope/styled-components/cjs/examples/plasma_web/components/Indicator/Indicator.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _Indicator = /*#__PURE__*/require("../../../../components/Indicator");
var config = exports.config = {
  defaults: {
    size: 's',
    view: 'default'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--surface-solid-default);"], _Indicator.indicatorTokens.color),
      accent: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--surface-accent);"], _Indicator.indicatorTokens.color),
      inactive: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--surface-solid-tertiary);"], _Indicator.indicatorTokens.color),
      positive: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--surface-positive);"], _Indicator.indicatorTokens.color),
      warning: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--surface-warning);"], _Indicator.indicatorTokens.color),
      negative: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--surface-negative);"], _Indicator.indicatorTokens.color),
      black: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--on-light-surface-solid-default);"], _Indicator.indicatorTokens.color),
      white: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--on-dark-surface-solid-default);"], _Indicator.indicatorTokens.color)
    },
    size: {
      l: /*#__PURE__*/(0, _styledComponents.css)(["", ":0.75rem;"], _Indicator.indicatorTokens.size),
      m: /*#__PURE__*/(0, _styledComponents.css)(["", ":0.5rem;"], _Indicator.indicatorTokens.size),
      s: /*#__PURE__*/(0, _styledComponents.css)(["", ":0.375rem;"], _Indicator.indicatorTokens.size)
    }
  }
};
