// @salutejs/plasma-new-hope/cjs/components/Tooltip/Tooltip.js
import './Tooltip_aay375.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var react = require('@linaria/react');
require('@linaria/core');
var common = require('../../engines/common.js');
var Popover$1 = require('../Popover/Popover.js');
var index = require('../../utils/index.js');
var Tooltip_styles = require('./Tooltip.styles.js');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var Tooltip_tokens = require('./Tooltip.tokens.js');

var _excluded = ["id", "text", "isOpen", "isVisible", "hasArrow", "arrow", "animated", "offset", "minWidth", "maxWidth", "placement", "usePortal", "target", "children", "onDismiss", "view", "size", "contentLeft", "zIndex", "className"];
var ESCAPE_KEYCODE = 27;
var getStringValue = function getStringValue(value) {
  return typeof value === 'number' ? "".concat(value, "rem") : value;
};
var Popover = /*#__PURE__*/common.component(Popover$1.popoverConfig);
var _exp = function _exp() {
  return Popover;
};
var StyledPopover = /*#__PURE__*/react.styled(_exp())({
  name: "StyledPopover",
  "class": "s1cz8cv8",
  propsAsIs: true
});

/**
 * Компонент для текстовых подсказок. Основное предназначение — подписи к блокам.
 */

var tooltipRoot = function tooltipRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (_ref, outerRef) {
    var id = _ref.id,
      text = _ref.text,
      isOpen = _ref.isOpen,
      isVisible = _ref.isVisible,
      _ref$hasArrow = _ref.hasArrow,
      hasArrow = _ref$hasArrow === void 0 ? true : _ref$hasArrow,
      arrow = _ref.arrow,
      animated = _ref.animated,
      _ref$offset = _ref.offset,
      offset = _ref$offset === void 0 ? [0, 8] : _ref$offset,
      minWidth = _ref.minWidth,
      maxWidth = _ref.maxWidth,
      _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? 'bottom' : _ref$placement,
      _ref$usePortal = _ref.usePortal,
      usePortal = _ref$usePortal === void 0 ? true : _ref$usePortal,
      target = _ref.target,
      children = _ref.children,
      onDismiss = _ref.onDismiss,
      view = _ref.view,
      size = _ref.size,
      contentLeft = _ref.contentLeft,
      _ref$zIndex = _ref.zIndex,
      zIndex = _ref$zIndex === void 0 ? '9200' : _ref$zIndex,
      className = _ref.className,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
    var _useState = React.useState(null),
      _useState2 = _rollupPluginBabelHelpers.slicedToArray(_useState, 2),
      ref = _useState2[0],
      setRef = _useState2[1];

    // TODO убрать после отказа от старого API
    var innerIsOpen = isVisible || isOpen;
    var innerHasArrow = arrow || hasArrow;
    var showTooltip = innerIsOpen && Boolean(text === null || text === void 0 ? void 0 : text.length);
    var animatedClass = animated ? Tooltip_tokens.classes.animated : undefined;
    React.useEffect(function () {
      var onKeyDown = function onKeyDown(event) {
        if (event.keyCode === ESCAPE_KEYCODE) {
          onDismiss === null || onDismiss === void 0 || onDismiss();
        }
      };
      window.addEventListener('keydown', onKeyDown);
      return function () {
        window.removeEventListener('keydown', onKeyDown);
      };
    }, []);
    var withContentLeft = contentLeft ? Tooltip_tokens.classes.hasContentLeft : undefined;
    return /*#__PURE__*/React.createElement(StyledPopover, _rollupPluginBabelHelpers.extends({
      isOpen: showTooltip,
      placement: placement,
      offset: offset,
      zIndex: zIndex,
      target: target || children,
      usePortal: usePortal,
      hasArrow: innerHasArrow,
      "aria-hidden": !innerIsOpen,
      "aria-live": "polite",
      role: "tooltip",
      className: index.cx(ref === null || ref === void 0 ? void 0 : ref.classList.toString(), animatedClass, className)
      // INFO: Прокидываем стили для Popover из Root Tooltip-а
    }, rest), /*#__PURE__*/React.createElement(Root, {
      view: view,
      size: size,
      ref: setRef
    }, /*#__PURE__*/React.createElement(Tooltip_styles.TooltipRoot, {
      ref: outerRef,
      id: id,
      className: index.cx(withContentLeft),
      maxWidth: getStringValue(maxWidth),
      minWidth: getStringValue(minWidth)
    }, contentLeft && /*#__PURE__*/React.createElement(Tooltip_styles.StyledContentLeft, null, contentLeft), text)));
  });
};
var tooltipConfig = {
  name: 'Tooltip',
  tag: 'div',
  layout: tooltipRoot,
  base: '',
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'default',
    size: 'm'
  }
};

exports.tooltipConfig = tooltipConfig;
exports.tooltipRoot = tooltipRoot;
