// @salutejs/plasma-new-hope/cjs/components/Combobox/ui/ComboboxChip/ComboboxChip.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var ComboboxChip_styles = require('./ComboboxChip.styles.js');

/**
 * Чип, отображающий выбранный элемент
 */
var ComboboxChip = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var value = _ref.value,
    text = _ref.text,
    index = _ref.index,
    disabled = _ref.disabled,
    readOnly = _ref.readOnly,
    onClick = _ref.onClick,
    onClear = _ref.onClear,
    onKeyDown = _ref.onKeyDown;
  var onClearHandle = function onClearHandle() {
    return onClear(value, text);
  };
  var onKeyDownHandle = function onKeyDownHandle(event) {
    onKeyDown === null || onKeyDown === void 0 || onKeyDown(value, text, index, event);
  };
  return /*#__PURE__*/React.createElement(ComboboxChip_styles.StyledChip, {
    ref: ref,
    tabIndex: -1,
    disabled: disabled,
    readOnly: readOnly,
    onClick: onClick,
    onClear: onClearHandle,
    onKeyDown: onKeyDownHandle,
    text: "".concat(text)
  });
});

exports.ComboboxChip = ComboboxChip;
