// @salutejs/plasma-new-hope/cjs/components/Header/HeaderRoot/HeaderRoot.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var Header_tokens = require('../Header.tokens.js');
var index = require('../../../utils/index.js');
var base = require('./variations/_size/base.js');
var HeaderRoot_styles = require('./HeaderRoot.styles.js');

var _excluded = ["children", "size", "view", "gradientColor"];

/**
 * Корневой узел для шапки.
 */
var headerRoot = function headerRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (props, outerRootRef) {
    var children = props.children,
      size = props.size,
      view = props.view,
      gradientColor = props.gradientColor,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(props, _excluded);
    var gradientColorClass = gradientColor ? Header_tokens.classes.headerWithGradient : undefined;
    return /*#__PURE__*/React.createElement(Root, {
      ref: outerRootRef,
      size: size,
      view: view
    }, /*#__PURE__*/React.createElement(HeaderRoot_styles.StyledHeaderRoot, _rollupPluginBabelHelpers.extends({
      className: index.cx(Header_tokens.classes.headerRoot, gradientColorClass)
    }, rest, {
      $gradientColor: gradientColor
    }), /*#__PURE__*/React.createElement(HeaderRoot_styles.StyledInner, null, children)));
  });
};
var headerRootConfig = {
  name: 'HeaderRoot',
  tag: 'div',
  layout: headerRoot,
  base: '',
  variations: {
    size: {
      css: base.base
    }
  },
  defaults: {
    size: 'm'
  }
};

exports.headerRoot = headerRoot;
exports.headerRootConfig = headerRootConfig;
