// @salutejs/plasma-web/utils/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "animatedScrollToX", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.animatedScrollToX;
  }
});
Object.defineProperty(exports, "animatedScrollToY", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.animatedScrollToY;
  }
});
Object.defineProperty(exports, "canUseDOM", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.canUseDOM;
  }
});
Object.defineProperty(exports, "clearSelection", {
  enumerable: true,
  get: function get() {
    return _plasmaHope.clearSelection;
  }
});
Object.defineProperty(exports, "convertRoundnessMatrix", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.convertRoundnessMatrix;
  }
});
Object.defineProperty(exports, "extractTextFrom", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.extractTextFrom;
  }
});
Object.defineProperty(exports, "formatCurrency", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.formatCurrency;
  }
});
Object.defineProperty(exports, "gridBreakpoints", {
  enumerable: true,
  get: function get() {
    return _plasmaHope.gridBreakpoints;
  }
});
Object.defineProperty(exports, "gridColumns", {
  enumerable: true,
  get: function get() {
    return _plasmaHope.gridColumns;
  }
});
Object.defineProperty(exports, "gridGutters", {
  enumerable: true,
  get: function get() {
    return _plasmaHope.gridGutters;
  }
});
Object.defineProperty(exports, "gridMargins", {
  enumerable: true,
  get: function get() {
    return _plasmaHope.gridMargins;
  }
});
Object.defineProperty(exports, "gridSizes", {
  enumerable: true,
  get: function get() {
    return _plasmaHope.gridSizes;
  }
});
Object.defineProperty(exports, "mediaQuery", {
  enumerable: true,
  get: function get() {
    return _plasmaHope.mediaQuery;
  }
});
Object.defineProperty(exports, "monthLongName", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.monthLongName;
  }
});
Object.defineProperty(exports, "monthShortName", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.monthShortName;
  }
});
Object.defineProperty(exports, "padZeroNumber", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.padZeroNumber;
  }
});
Object.defineProperty(exports, "selectText", {
  enumerable: true,
  get: function get() {
    return _plasmaHope.selectText;
  }
});
Object.defineProperty(exports, "setRef", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.setRef;
  }
});
Object.defineProperty(exports, "toCssSize", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.toCssSize;
  }
});
Object.defineProperty(exports, "transformStyles", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.transformStyles;
  }
});
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _plasmaHope = /*#__PURE__*/require("@salutejs/plasma-hope");
