// @salutejs/plasma-new-hope/styled-components/es/components/Notification/Notification.styles.js
var _templateObject;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css } from 'styled-components';
import styled from 'styled-components';
import { applyHyphens } from '../../mixins';
import { buttonConfig } from '../Button';
import { component, mergeConfig } from '../../engines';
import { classes, tokens } from './Notification.tokens';
import { placements } from './Notification.types';
export var StyledNotification = /*#__PURE__*/css(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n    width: fit-content;\n"])));
var mergedButtonConfig = /*#__PURE__*/mergeConfig(buttonConfig);
var Button = /*#__PURE__*/component(mergedButtonConfig);
export var CloseIconWrapper = /*#__PURE__*/styled(Button).withConfig({
  componentId: "plasma-new-hope__sc-1iufy9r-0"
})(["position:absolute;width:1.5rem;height:1.5rem;color:var(", ");:hover{color:var(", ");}&.", "{position:relative;}"], tokens.closeIconColor, tokens.closeIconColorOnHover, classes.horizontal);
export var Wrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1iufy9r-1"
})(["position:relative;box-sizing:border-box;background:var(", ");border-radius:var(", ");&.", "{display:flex;align-items:center;gap:var(", ");}"], tokens.background, tokens.borderRadius, classes.horizontal, tokens.horizontalLayoutGap);
export var ButtonsWrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1iufy9r-2"
})(["margin-top:var(", ");margin-left:", ";&.", "{margin-top:unset;margin-left:unset;}"], tokens.buttonsMarginTop, function (_ref) {
  var iconPlacement = _ref.iconPlacement;
  return iconPlacement === placements.left ? "var(".concat(tokens.buttonsMarginLeft, ")") : 'unset';
}, classes.horizontal);
export var IconWrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1iufy9r-3"
})(["width:var(", ");height:var(", ");margin-right:", ";margin-bottom:", ";&.", "{margin-right:var(", ");margin-bottom:unset;}"], tokens.contentLeftIconSize, tokens.contentLeftIconSize, function (_ref2) {
  var iconPlacement = _ref2.iconPlacement;
  return iconPlacement === placements.left ? "var(".concat(tokens.contentLeftIconMargin, ")") : 'unset';
}, function (_ref3) {
  var iconPlacement = _ref3.iconPlacement;
  return iconPlacement === placements.top ? "var(".concat(tokens.contentTopIconMargin, ")") : 'unset';
}, classes.horizontal, tokens.horizontalLayoutLeftIconMargin);
export var ContentBox = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1iufy9r-4"
})(["display:", ";padding-top:", ";&.", "{padding:unset;display:flex;align-items:center;flex-grow:1;}"], function (_ref4) {
  var iconPlacement = _ref4.iconPlacement;
  return iconPlacement === placements.left ? 'flex' : 'block';
}, function (_ref5) {
  var iconPlacement = _ref5.iconPlacement;
  return iconPlacement ? "var(".concat(tokens.contentPaddingTop, ")") : "var(".concat(tokens.contentPaddingTopWithoutIcon, ")");
}, classes.horizontal);
export var TextBox = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1iufy9r-5"
})(["padding:var(", ") var(", ") var(", ") var(", ");padding-top:", ";padding-right:", ";display:flex;flex-direction:column;row-gap:var(", ");word-break:break-word;&.", "{padding:unset;}"], tokens.textboxPaddingTop, tokens.textboxPaddingRight, tokens.textboxPaddingBottom, tokens.textboxPaddingLeft, function (_ref6) {
  var iconPlacement = _ref6.iconPlacement;
  return !iconPlacement || iconPlacement === placements.top ? "var(".concat(tokens.textboxPaddingTopWithTopIcon, ")") : "var(".concat(tokens.textboxPaddingTop, ")");
}, function (_ref7) {
  var showCloseIcon = _ref7.showCloseIcon,
    iconPlacement = _ref7.iconPlacement;
  return showCloseIcon && iconPlacement !== placements.top ? "var(".concat(tokens.textboxPaddingRightWithCloseIcon, ")") : "var(".concat(tokens.textboxPaddingRight, ")");
}, tokens.textboxGap, classes.horizontal);
export var StyledTitle = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1iufy9r-6"
})(["font-family:var(", ");font-size:var(", ");font-style:var(", ");font-weight:var(", ");letter-spacing:var(", ");line-height:var(", ");", ";"], tokens.titleFontFamily, tokens.titleFontSize, tokens.titleFontStyle, tokens.titleFontWeight, tokens.titleFontLetterSpacing, tokens.titleFontLineHeight, /*#__PURE__*/String(applyHyphens));
export var StyledContent = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1iufy9r-7"
})(["font-family:var(", ");font-size:var(", ");font-style:var(", ");font-weight:var(", ");letter-spacing:var(", ");line-height:var(", ");", ";color:var(", ");"], tokens.contentFontFamily, tokens.contentFontSize, tokens.contentFontStyle, tokens.contentFontWeight, tokens.contentFontLetterSpacing, tokens.contentFontLineHeight, /*#__PURE__*/String(applyHyphens), tokens.contentColor);
export var StyledItemWrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1iufy9r-8"
})(["margin-top:1rem;opacity:1;&&.", "{animation:0.4s showAnimation ease-out;}&&.", "{animation:0.4s hideAnimation ease-out;}@keyframes showAnimation{0%{transform:translateX(100%);opacity:0;}100%{transform:translateX(0);opacity:1;}}@keyframes hideAnimation{0%{transform:translateX(0);opacity:1;}100%{transform:translateX(100%);opacity:0;}}"], classes.notificationItemOpened, classes.notificationItemHidden);
export var StyledRoot = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-1iufy9r-9"
})(["display:flex;flex-direction:column-reverse;box-sizing:border-box;padding:0 1.5rem 1.5rem;max-height:100%;align-items:flex-end;"]);
