// @salutejs/plasma-new-hope/styled-components/es/components/Notification/Notification.js
var _IconCross;
var _excluded = ["role", "title", "children", "actions", "view", "size", "iconPlacement", "showCloseIcon", "layout", "icon", "onCloseButtonClick"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React, { forwardRef, useMemo } from 'react';
import { cx } from '../../utils';
import { IconCross } from '../_Icon/Icons/IconCross';
import { classes } from './Notification.tokens';
import { base as viewCSS } from './variations/_view/base';
import { base as layoutCSS } from './variations/_layout/base';
import { base as sizeCSS } from './variations/_size/base';
import { layouts } from './Notification.types';
import { ButtonsWrapper, CloseIconWrapper, ContentBox, IconWrapper, StyledContent, StyledNotification, StyledTitle, TextBox, Wrapper } from './Notification.styles';
import { getLayoutClass } from './utils';

/**
 * Компонент для небольших уведомлений пользователя
 */
export var notificationRoot = function notificationRoot(Root) {
  return /*#__PURE__*/forwardRef(function (props, ref) {
    var _props$role = props.role,
      role = _props$role === void 0 ? 'status' : _props$role,
      title = props.title,
      content = props.children,
      actions = props.actions,
      view = props.view,
      size = props.size,
      iconPlacement = props.iconPlacement,
      _props$showCloseIcon = props.showCloseIcon,
      showCloseIcon = _props$showCloseIcon === void 0 ? true : _props$showCloseIcon,
      _props$layout = props.layout,
      layout = _props$layout === void 0 ? layouts.vertical : _props$layout,
      icon = props.icon,
      onCloseButtonClick = props.onCloseButtonClick,
      rest = _objectWithoutProperties(props, _excluded);
    var ariaLive = 'polite';
    var ariaAtomic = false;
    if (role === 'alert') {
      ariaLive = 'assertive';
    } else if (role === 'status') {
      ariaAtomic = true;
    }
    var isOneLine = !content || !title;
    var oneLineClass = isOneLine ? classes.oneLine : undefined;
    var withoutIconClass = icon ? undefined : classes.withoutIcon;
    var withoutCloseIconClass = showCloseIcon ? undefined : classes.withoutCloseIcon;
    var IconPlacementInternal = useMemo(function () {
      return icon ? iconPlacement : undefined;
    }, [icon, iconPlacement]);
    return /*#__PURE__*/React.createElement(Root, _extends({
      view: view,
      size: size,
      layout: layout,
      ref: ref,
      role: role,
      "aria-live": ariaLive,
      "aria-atomic": ariaAtomic
    }, rest), /*#__PURE__*/React.createElement(Wrapper, {
      className: cx(classes.wrapper, getLayoutClass(layout), oneLineClass, withoutCloseIconClass)
    }, /*#__PURE__*/React.createElement(ContentBox, {
      iconPlacement: IconPlacementInternal,
      className: cx(classes.contentBox, getLayoutClass(layout), withoutIconClass)
    }, icon && /*#__PURE__*/React.createElement(IconWrapper, {
      iconPlacement: IconPlacementInternal,
      className: cx(classes.icon, getLayoutClass(layout))
    }, icon), /*#__PURE__*/React.createElement(TextBox, {
      iconPlacement: IconPlacementInternal,
      showCloseIcon: showCloseIcon,
      className: cx(classes.textbox, getLayoutClass(layout))
    }, title && /*#__PURE__*/React.createElement(StyledTitle, {
      className: classes.title
    }, title), content && /*#__PURE__*/React.createElement(StyledContent, {
      className: classes.text
    }, content))), actions && /*#__PURE__*/React.createElement(ButtonsWrapper, {
      iconPlacement: IconPlacementInternal,
      className: cx(classes.buttonsWrapper, getLayoutClass(layout))
    }, actions), showCloseIcon && /*#__PURE__*/React.createElement(CloseIconWrapper, {
      view: "clear",
      size: "s",
      onClick: onCloseButtonClick,
      className: cx(classes.closeIcon, getLayoutClass(layout))
    }, _IconCross || (_IconCross = /*#__PURE__*/React.createElement(IconCross, {
      size: "s",
      color: "inherit"
    })))));
  });
};
export var noticationConfig = {
  name: 'Notification',
  tag: 'div',
  layout: notificationRoot,
  base: StyledNotification,
  variations: {
    layout: {
      css: layoutCSS
    },
    view: {
      css: viewCSS
    },
    size: {
      css: sizeCSS
    }
  },
  defaults: {
    view: 'default',
    layout: layouts.vertical,
    size: 'xs'
  }
};
