// @salutejs/plasma-core/components/Tabs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "StyledTabs", {
  enumerable: true,
  get: function get() {
    return _Tabs.StyledTabs;
  }
});
Object.defineProperty(exports, "StyledWrapper", {
  enumerable: true,
  get: function get() {
    return _Tabs.StyledWrapper;
  }
});
Object.defineProperty(exports, "TabItem", {
  enumerable: true,
  get: function get() {
    return _TabItem.TabItem;
  }
});
Object.defineProperty(exports, "Tabs", {
  enumerable: true,
  get: function get() {
    return _Tabs.Tabs;
  }
});
Object.defineProperty(exports, "TabsContext", {
  enumerable: true,
  get: function get() {
    return _TabsContext.TabsContext;
  }
});
Object.defineProperty(exports, "createTabsController", {
  enumerable: true,
  get: function get() {
    return _createTabsController.createTabsController;
  }
});
var _createTabsController = /*#__PURE__*/require("./createTabsController");
var _TabsContext = /*#__PURE__*/require("./TabsContext");
var _Tabs = /*#__PURE__*/require("./Tabs");
var _TabItem = /*#__PURE__*/require("./TabItem");
