// @salutejs/plasma-core/components/Tabs/Tabs.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Tabs = exports.StyledWrapper = exports.StyledTabs = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _mixins = /*#__PURE__*/require("../../mixins");
var _TabItem = /*#__PURE__*/require("./TabItem");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _excluded = ["role", "as", "children", "stretch"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('Tabs/Tabs');
var StyledTabs = exports.StyledTabs = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-z15rbr-0"
})(["position:relative;display:inline-flex;align-items:center;box-sizing:border-box;flex-wrap:nowrap;justify-content:stretch;padding:0;margin:0;width:max-content;list-style-type:none;user-select:none;-webkit-tap-highlight-color:rgba(0,0,0,0);", " gap:inherit;"], function (_ref) {
  var _ref$stretch = _ref.stretch,
    stretch = _ref$stretch === void 0 ? false : _ref$stretch;
  return !stretch && (0, _styledComponents.css)(["margin:0 var(--tabs-margin);"]);
});
var StyledWrapper = exports.StyledWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-z15rbr-1"
})(["overflow-x:auto;margin-left:calc(var(--tab-focus-border-size) * -1);margin-top:calc(var(--tab-focus-border-size) * -2);padding:var(--tab-focus-border-size);transform:translateY(var(--tab-focus-border-size));&::-webkit-scrollbar{display:none;}", "{", "}", " ", " ", "{", "}gap:0.125rem;"], StyledTabs, _mixins.applyDisabled, function (_ref2) {
  var stretch = _ref2.stretch;
  return stretch && (0, _styledComponents.css)(["width:100%;", "{width:100%;}", "{flex:1 1 calc((100% / 4) - 0.125rem);}"], StyledTabs, _TabItem.StyledTabItem);
}, function (_ref3) {
  var _ref3$stretch = _ref3.stretch,
    stretch = _ref3$stretch === void 0 ? false : _ref3$stretch;
  return !stretch && (0, _styledComponents.css)(["margin-left:calc(var(--tabs-margin) * -1 + var(--tab-focus-border-size) * -1);margin-right:calc(var(--tabs-margin) * -1);"]);
}, _TabItem.StyledTabItem, function (_ref4) {
  var disabled = _ref4.disabled;
  return disabled && (0, _styledComponents.css)(["cursor:not-allowed;"]);
});

/**
 * Контейнер вкладок.
 */
var Tabs = exports.Tabs = /*#__PURE__*/_react["default"].forwardRef(function (_ref5, ref) {
  var _ref5$role = _ref5.role,
    role = _ref5$role === void 0 ? 'tablist' : _ref5$role,
    as = _ref5.as,
    children = _ref5.children,
    stretch = _ref5.stretch,
    rest = _objectWithoutProperties(_ref5, _excluded);
  return /*#__PURE__*/_react["default"].createElement(StyledWrapper, _extends({
    stretch: stretch
  }, rest), /*#__PURE__*/_react["default"].createElement(StyledTabs, {
    as: as,
    ref: ref,
    role: role,
    stretch: stretch
  }, children));
});
