// @salutejs/plasma-new-hope/styled-components/es/components/Avatar/Avatar.styles.js
var _templateObject;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { css } from 'styled-components';
import styled from 'styled-components';
import { tokens } from './Avatar.tokens';
export var base = /*#__PURE__*/css(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n    position: relative;\n"])));
export var Wrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-t16ccn-0"
})(["display:flex;justify-content:center;align-items:center;width:100%;height:100%;border-radius:50%;background:var(", ");overflow:hidden;&:hover{scale:", ";cursor:", ";}"], tokens.backgroundColor, function (_ref) {
  var isScalable = _ref.isScalable;
  return isScalable ? "var(".concat(tokens.scaleHover, ")") : '1';
}, function (_ref2) {
  var isScalable = _ref2.isScalable;
  return isScalable ? 'pointer' : 'inherit';
});
export var StatusIcon = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-t16ccn-1"
})(["position:absolute;top:calc(1.02 * 0.85 * var(", ") - var(", ") / 2);left:calc(1.02 * 0.85 * var(", ") - var(", ") / 2);"], tokens.avatarSize, tokens.statusIconSize, tokens.avatarSize, tokens.statusIconSize);
export var Image = /*#__PURE__*/styled.img.withConfig({
  componentId: "plasma-new-hope__sc-t16ccn-2"
})(["width:100%;height:100%;border-radius:50%;"]);
export var Text = /*#__PURE__*/styled.span.withConfig({
  componentId: "plasma-new-hope__sc-t16ccn-3"
})(["font-size:var(", ");font-family:var(", ");font-weight:var(", ");line-height:var(", ");background-image:var(", ");background-color:var(", ");background-size:100%;-webkit-background-clip:text;-webkit-text-fill-color:transparent;"], tokens.fontSize, tokens.fontFamily, tokens.fontWeight, tokens.lineHeight, tokens.color, tokens.color);
