// @salutejs/plasma-web/components/Dropdown/Dropdown.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Trigger = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _Button = /*#__PURE__*/require("../Button");
var _ = /*#__PURE__*/require(".");
var _Dropdown;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var items = [{
  value: 'north_america',
  label: 'Северная Америка',
  contentLeft: /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconLocation, {
    color: "inherit"
  }),
  contentRight: /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconLocation, {
    color: "inherit"
  })
}, {
  value: 'south_america',
  label: 'Южная Америка',
  items: [{
    value: 'brazil',
    label: 'Бразилия',
    items: [{
      value: 'rio_de_janeiro',
      label: 'Рио-де-Жанейро'
    }, {
      value: 'sao_paulo',
      label: 'Сан-Паулу'
    }]
  }, {
    value: 'argentina',
    label: 'Аргентина',
    items: [{
      value: 'buenos_aires',
      label: 'Буэнос-Айрес'
    }, {
      value: 'cordoba',
      label: 'Кордова'
    }]
  }, {
    value: 'colombia',
    label: 'Колумбия',
    items: [{
      value: 'bogota',
      label: 'Богота'
    }, {
      value: 'medellin',
      label: 'Медельин'
    }]
  }]
}, {
  value: 'europe',
  label: 'Европа',
  items: [{
    value: 'france',
    label: 'Франция',
    items: [{
      value: 'paris',
      label: 'Париж'
    }, {
      value: 'lyon',
      label: 'Лион'
    }]
  }, {
    value: 'germany',
    label: 'Германия',
    items: [{
      value: 'berlin',
      label: 'Берлин'
    }, {
      value: 'munich',
      label: 'Мюнхен'
    }]
  }, {
    value: 'italy',
    label: 'Италия',
    items: [{
      value: 'rome',
      label: 'Рим'
    }, {
      value: 'milan',
      label: 'Милан'
    }]
  }, {
    value: 'spain',
    label: 'Испания',
    items: [{
      value: 'madrid',
      label: 'Мадрид'
    }, {
      value: 'barcelona',
      label: 'Барселона'
    }]
  }, {
    value: 'united_kingdom',
    label: 'Великобритания',
    items: [{
      value: 'london',
      label: 'Лондон'
    }, {
      value: 'manchester',
      label: 'Манчестер'
    }]
  }]
}, {
  value: 'asia',
  label: 'Азия',
  items: [{
    value: 'china',
    label: 'Китай',
    items: [{
      value: 'beijing',
      label: 'Пекин'
    }, {
      value: 'shanghai',
      label: 'Шанхай'
    }]
  }, {
    value: 'japan',
    label: 'Япония',
    items: [{
      value: 'tokyo',
      label: 'Токио'
    }, {
      value: 'osaka',
      label: 'Осака'
    }]
  }, {
    value: 'india',
    label: 'Индия',
    items: [{
      value: 'delhi',
      label: 'Дели'
    }, {
      value: 'mumbai',
      label: 'Мумбаи'
    }]
  }, {
    value: 'south_korea',
    label: 'Южная Корея',
    items: [{
      value: 'seoul',
      label: 'Сеул'
    }, {
      value: 'busan',
      label: 'Пусан'
    }]
  }, {
    value: 'thailand',
    label: 'Таиланд',
    items: [{
      value: 'bangkok',
      label: 'Бангкок'
    }, {
      value: 'phuket',
      label: 'Пхукет'
    }]
  }]
}, {
  value: 'africa',
  label: 'Африка',
  disabled: true
}];
var Trigger = exports.Trigger = function Trigger() {
  return _Dropdown || (_Dropdown = /*#__PURE__*/_react["default"].createElement(_.Dropdown, {
    items: items
  }, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    text: "\u0421\u043F\u0438\u0441\u043E\u043A \u0441\u0442\u0440\u0430\u043D"
  })));
};
