// @salutejs/plasma-new-hope/es/components/Header/ui/HeaderTitleWrapper/HeaderTitleWrapper.js
import { objectWithoutProperties as _objectWithoutProperties } from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef } from 'react';
import { base as base$1 } from './variations/_size/base.js';
import { base } from './variations/_view/base.js';
import { TextBoxRoot, Label, SubTitle, Caption, BigTitle, Title } from './HeaderTitleWrapper.styles.js';

var _excluded = ["label", "title", "subTitle", "caption", "withBigTitle", "children"];

/**
 * Компонент для отображения текста в скомпанованном блоке.
 */
var headerTitleWrapperRoot = function headerTitleWrapperRoot(Root) {
  return /*#__PURE__*/forwardRef(function (props, outerRootRef) {
    var label = props.label,
      title = props.title,
      subTitle = props.subTitle,
      caption = props.caption,
      withBigTitle = props.withBigTitle,
      children = props.children,
      rest = _objectWithoutProperties(props, _excluded);
    if (children) {
      return /*#__PURE__*/React.createElement(Root, {
        ref: outerRootRef
      }, /*#__PURE__*/React.createElement(TextBoxRoot, rest, children));
    }
    var T = withBigTitle ? BigTitle : Title;
    return /*#__PURE__*/React.createElement(Root, {
      ref: outerRootRef
    }, /*#__PURE__*/React.createElement(TextBoxRoot, rest, label && /*#__PURE__*/React.createElement(Label, null, label), title && /*#__PURE__*/React.createElement(T, null, title), subTitle && /*#__PURE__*/React.createElement(SubTitle, null, subTitle), caption && /*#__PURE__*/React.createElement(Caption, null, caption)));
  });
};
var headerTitleWrapperConfig = {
  name: 'HeaderTitleWrapper',
  tag: 'div',
  layout: headerTitleWrapperRoot,
  base: '',
  variations: {
    view: {
      css: base
    },
    size: {
      css: base$1
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

export { headerTitleWrapperConfig, headerTitleWrapperRoot };
