// @salutejs/plasma-new-hope/cjs/components/Slider/components/Double/Double.styles.js
import './Double.styles_1jhda4u.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
require('@linaria/core');
var common = require('../../../../engines/common.js');
var TextField$1 = require('../../../TextField/TextField.js');

var mergedConfig = /*#__PURE__*/common.mergeConfig(TextField$1.textFieldConfig);
var TextField = /*#__PURE__*/common.component(mergedConfig);
var LabelWrapper = /*#__PURE__*/react.styled('div')({
  name: "LabelWrapper",
  "class": "lcs82cd",
  propsAsIs: false
});
var LabelContentLeft = /*#__PURE__*/react.styled('div')({
  name: "LabelContentLeft",
  "class": "l1tvgbgq",
  propsAsIs: false
});
var Label = /*#__PURE__*/react.styled('label')({
  name: "Label",
  "class": "l1at2q9b",
  propsAsIs: false
});
var InputsWrapper = /*#__PURE__*/react.styled('div')({
  name: "InputsWrapper",
  "class": "i11dc2og",
  propsAsIs: false
});

// NOTE: переопределение токенов TextField
var _exp12 = function _exp12() {
  return TextField;
};
var StyledInput = /*#__PURE__*/react.styled(_exp12())({
  name: "StyledInput",
  "class": "s4zvp7i",
  propsAsIs: true
});
var DoubleWrapper = /*#__PURE__*/react.styled('div')({
  name: "DoubleWrapper",
  "class": "d3kvcjo",
  propsAsIs: false
});
var SliderWrapper = /*#__PURE__*/react.styled('div')({
  name: "SliderWrapper",
  "class": "s11ntvbh",
  propsAsIs: false
});

exports.DoubleWrapper = DoubleWrapper;
exports.InputsWrapper = InputsWrapper;
exports.Label = Label;
exports.LabelContentLeft = LabelContentLeft;
exports.LabelWrapper = LabelWrapper;
exports.SliderWrapper = SliderWrapper;
exports.StyledInput = StyledInput;
