// @salutejs/plasma-new-hope/styled-components/es/components/Spinner/SpinnerSvg.js
var _excluded = ["id", "color", "width", "height"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React from 'react';
export var SpinnerSvg = function SpinnerSvg(_ref) {
  var id = _ref.id,
    color = _ref.color,
    width = _ref.width,
    height = _ref.height,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/React.createElement("svg", _extends({
    viewBox: "0 0 56 56",
    fill: "none",
    style: {
      width: width,
      height: height
    }
  }, props), /*#__PURE__*/React.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M28 0C12.536 0 0 12.536 0 28s12.536 28 28 28h.055H28v-4C14.745 52 4 41.255 4 28S14.745 4 28 4V0z",
    fill: "url(#prefix__paint0_linear".concat(id, ")")
  }), /*#__PURE__*/React.createElement("path", {
    d: "M56 28c0 14.791-11.47 26.904-26 27.93-1.102.077-2-.825-2-1.93s.899-1.991 2-2.082C42.318 50.902 52 40.58 52 28 52 14.745 41.255 4 28 4V0c15.464 0 28 12.536 28 28z",
    fill: "url(#prefix__paint1_linear".concat(id, ")")
  }), /*#__PURE__*/React.createElement("defs", null, /*#__PURE__*/React.createElement("linearGradient", {
    id: "prefix__paint0_linear".concat(id),
    x1: 0,
    y1: 56,
    x2: 0,
    y2: 0,
    gradientUnits: "userSpaceOnUse"
  }, /*#__PURE__*/React.createElement("stop", {
    stopColor: color,
    stopOpacity: 0
  }), /*#__PURE__*/React.createElement("stop", {
    offset: 0.048,
    stopColor: color,
    stopOpacity: 0
  }), /*#__PURE__*/React.createElement("stop", {
    offset: 0.952,
    stopColor: color,
    stopOpacity: 0.5
  }), /*#__PURE__*/React.createElement("stop", {
    offset: 1,
    stopColor: color,
    stopOpacity: 0.5
  })), /*#__PURE__*/React.createElement("linearGradient", {
    id: "prefix__paint1_linear".concat(id),
    x1: 28,
    y1: 56,
    x2: 28,
    y2: 0,
    gradientUnits: "userSpaceOnUse"
  }, /*#__PURE__*/React.createElement("stop", {
    stopColor: color
  }), /*#__PURE__*/React.createElement("stop", {
    offset: 0.048,
    stopColor: color
  }), /*#__PURE__*/React.createElement("stop", {
    offset: 0.952,
    stopColor: color,
    stopOpacity: 0.5
  }), /*#__PURE__*/React.createElement("stop", {
    offset: 1,
    stopColor: color,
    stopOpacity: 0.5
  }))));
};
