// @salutejs/plasma-new-hope/cjs/components/Combobox/ui/ComboboxTarget/ComboboxTarget.styles.js
import './ComboboxTarget.styles_1htd0dz.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
require('react');
var IconChevronDown = require('../../../_Icon/Icons/IconChevronDown.js');
var Combobox_tokens = require('../../Combobox.tokens.js');

var _exp10 = function _exp10() {
  return function (_ref) {
    var opened = _ref.opened;
    return opened ? "var(".concat(Combobox_tokens.tokens.targetBackgroundColorOpen, ")") : "var(".concat(Combobox_tokens.tokens.targetBackgroundColor, ")");
  };
};
var _exp11 = function _exp11() {
  return function (_ref2) {
    var opened = _ref2.opened;
    return opened ? "var(".concat(Combobox_tokens.tokens.targetBorderColorOpen, ")") : "var(".concat(Combobox_tokens.tokens.targetBorderColor, ")");
  };
};
var _exp12 = function _exp12() {
  return function (_ref3) {
    var opened = _ref3.opened;
    return opened ? "var(".concat(Combobox_tokens.tokens.targetBackgroundColorOpen, ")") : "var(".concat(Combobox_tokens.tokens.targetBackgroundColorHover, ")");
  };
};
var _exp13 = function _exp13() {
  return function (_ref4) {
    var opened = _ref4.opened;
    return opened ? "var(".concat(Combobox_tokens.tokens.targetBorderColorOpen, ")") : "var(".concat(Combobox_tokens.tokens.targetBorderColorHover, ")");
  };
};
var _exp14 = function _exp14() {
  return function (_ref5) {
    var opened = _ref5.opened;
    return opened ? "var(".concat(Combobox_tokens.tokens.targetBackgroundColorOpen, ")") : "var(".concat(Combobox_tokens.tokens.targetBackgroundColorActive, ")");
  };
};
var _exp15 = function _exp15() {
  return function (_ref6) {
    var opened = _ref6.opened;
    return opened ? "var(".concat(Combobox_tokens.tokens.targetBorderColorOpen, ")") : "var(".concat(Combobox_tokens.tokens.targetBorderColorActive, ")");
  };
};
var StyledComboboxTarget = /*#__PURE__*/react.styled('div')({
  name: "StyledComboboxTarget",
  "class": "s1f9n908",
  propsAsIs: false,
  vars: {
    "s1f9n908-0": [/*#__PURE__*/_exp10()],
    "s1f9n908-1": [/*#__PURE__*/_exp11()],
    "s1f9n908-2": [/*#__PURE__*/_exp12()],
    "s1f9n908-3": [/*#__PURE__*/_exp13()],
    "s1f9n908-4": [/*#__PURE__*/_exp14()],
    "s1f9n908-5": [/*#__PURE__*/_exp15()]
  }
});
var _exp35 = function _exp35() {
  return IconChevronDown.IconChevronDown;
};
var StyledArrow = /*#__PURE__*/react.styled(_exp35())({
  name: "StyledArrow",
  "class": "s16n9w7y",
  propsAsIs: true
});
var StyledContent = /*#__PURE__*/react.styled('div')({
  name: "StyledContent",
  "class": "sw8s5yo",
  propsAsIs: false
});
var StyledComboboxInput = /*#__PURE__*/react.styled('input')({
  name: "StyledComboboxInput",
  "class": "sys30py",
  propsAsIs: false
});
var StyledComboboxInputHelper = /*#__PURE__*/react.styled('div')({
  name: "StyledComboboxInputHelper",
  "class": "s16hup6g",
  propsAsIs: false
});
var StyledText = /*#__PURE__*/react.styled('span')({
  name: "StyledText",
  "class": "s3gbdjt",
  propsAsIs: false
});
var StyledChips = /*#__PURE__*/react.styled('div')({
  name: "StyledChips",
  "class": "s1camohz",
  propsAsIs: false
});
var StyledLabel = /*#__PURE__*/react.styled('label')({
  name: "StyledLabel",
  "class": "s16upk0a",
  propsAsIs: false
});

exports.StyledArrow = StyledArrow;
exports.StyledChips = StyledChips;
exports.StyledComboboxInput = StyledComboboxInput;
exports.StyledComboboxInputHelper = StyledComboboxInputHelper;
exports.StyledComboboxTarget = StyledComboboxTarget;
exports.StyledContent = StyledContent;
exports.StyledLabel = StyledLabel;
exports.StyledText = StyledText;
