// @salutejs/plasma-new-hope/cjs/components/Avatar/Avatar.styles.js
import './Avatar.styles_1gd252x.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
var Avatar_tokens = require('./Avatar.tokens.js');

var base = "b1d1rtx8";
var _exp2 = function _exp2() {
  return function (_ref) {
    var isScalable = _ref.isScalable;
    return isScalable ? "var(".concat(Avatar_tokens.tokens.scaleHover, ")") : '1';
  };
};
var _exp3 = function _exp3() {
  return function (_ref2) {
    var isScalable = _ref2.isScalable;
    return isScalable ? 'pointer' : 'inherit';
  };
};
var Wrapper = /*#__PURE__*/react.styled('div')({
  name: "Wrapper",
  "class": "wxwn3we",
  propsAsIs: false,
  vars: {
    "wxwn3we-0": [/*#__PURE__*/_exp2()],
    "wxwn3we-1": [/*#__PURE__*/_exp3()]
  }
});
var StatusIcon = /*#__PURE__*/react.styled('div')({
  name: "StatusIcon",
  "class": "sf2jhi1",
  propsAsIs: false
});
var Image = /*#__PURE__*/react.styled('img')({
  name: "Image",
  "class": "igq6ihg",
  propsAsIs: false
});
var Text = /*#__PURE__*/react.styled('span')({
  name: "Text",
  "class": "t1kfsxcq",
  propsAsIs: false
});

exports.Image = Image;
exports.StatusIcon = StatusIcon;
exports.Text = Text;
exports.Wrapper = Wrapper;
exports.base = base;
