// @salutejs/plasma-new-hope/es/components/Pagination/ui/PaginationQuickJumpToPage/PaginationQuickJumpToPage.js
import { objectWithoutProperties as _objectWithoutProperties, slicedToArray as _slicedToArray } from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { useState } from 'react';
import { defaultValues } from '../../utils/index.js';
import { QuickJumpToPageRoot, QuickJumpToPageTypography, QuickJumpToPageInput } from './PaginationQuickJumpToPage.styles.js';

var _excluded = ["placeholderQuickJump", "textQuickJump", "onChangeValue"];
var PaginationQuickJumpToPage = function PaginationQuickJumpToPage(_ref) {
  var _ref$placeholderQuick = _ref.placeholderQuickJump,
    placeholderQuickJump = _ref$placeholderQuick === void 0 ? defaultValues.placeholderQuickJump : _ref$placeholderQuick,
    _ref$textQuickJump = _ref.textQuickJump,
    textQuickJump = _ref$textQuickJump === void 0 ? defaultValues.textQuickJump : _ref$textQuickJump,
    onChangeValue = _ref.onChangeValue,
    rest = _objectWithoutProperties(_ref, _excluded);
  var _useState = useState(),
    _useState2 = _slicedToArray(_useState, 2),
    pageValue = _useState2[0],
    setPageValue = _useState2[1];
  var handleKeyPress = function handleKeyPress(event) {
    if (event.key === 'Enter') {
      onChangeValue === null || onChangeValue === void 0 || onChangeValue(Number(pageValue));
      setPageValue('');
    }
  };
  var handleChangeInput = function handleChangeInput(event) {
    var result = event.currentTarget.value.replace(/\D/g, '');
    if (result === '') return setPageValue('');
    setPageValue(Number(result));
  };
  return /*#__PURE__*/React.createElement(QuickJumpToPageRoot, rest, /*#__PURE__*/React.createElement(QuickJumpToPageTypography, null, textQuickJump), /*#__PURE__*/React.createElement(QuickJumpToPageInput, {
    value: pageValue,
    placeholder: placeholderQuickJump,
    onKeyUp: handleKeyPress,
    onChange: handleChangeInput
  }));
};

export { PaginationQuickJumpToPage };
