// @salutejs/plasma-new-hope/styled-components/cjs/examples/plasma_b2c/components/Calendar/Calendar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CalendarDoubleRange = exports.CalendarDouble = exports.CalendarBaseRange = exports.CalendarBase = exports.Calendar = void 0;
var _engines = /*#__PURE__*/require("../../../../engines");
var _Calendar = /*#__PURE__*/require("../../../../components/Calendar");
var _Calendar2 = /*#__PURE__*/require("../../../../components/Calendar/Calendar");
var _CalendarBase = /*#__PURE__*/require("./CalendarBase.config");
var _CalendarDouble = /*#__PURE__*/require("./CalendarDouble.config");
var mergedCalendarBaseConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Calendar.calendarBaseConfig, _CalendarBase.config);
var mergedCalendarDoubleConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Calendar.calendarDoubleConfig, _CalendarDouble.config);
var CalendarBase = exports.CalendarBase = /*#__PURE__*/(0, _engines.component)(mergedCalendarBaseConfig);
var CalendarDouble = exports.CalendarDouble = /*#__PURE__*/(0, _engines.component)(mergedCalendarDoubleConfig);
var CalendarBaseRange = exports.CalendarBaseRange = /*#__PURE__*/(0, _Calendar.withRange)(CalendarBase);
var CalendarDoubleRange = exports.CalendarDoubleRange = /*#__PURE__*/(0, _Calendar.withRange)(CalendarDouble);
var Calendar = exports.Calendar = /*#__PURE__*/(0, _Calendar2.boundCalendar)({
  base: CalendarBase,
  baseRange: CalendarBaseRange,
  "double": CalendarDouble,
  doubleRange: CalendarDoubleRange
});
