// @salutejs/plasma-ui/components/Checkbox/Checkbox.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _ = /*#__PURE__*/require(".");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var func = function func() {};
var labelWithLink = /*#__PURE__*/_react["default"].createElement("div", null, "Label 5 with ", /*#__PURE__*/_react["default"].createElement("a", {
  href: "/#"
}, "link"));
var items = [{
  id: 'check-1-1',
  name: 'check',
  value: 1,
  label: 'Checkbox 1',
  disabled: false
}, {
  id: 'check-1-2',
  name: 'check',
  value: 2,
  label: 'Checkbox 2',
  disabled: false
}, {
  id: 'check-1-3',
  name: 'check',
  value: 3,
  label: 'Checkbox 3',
  disabled: true
}, {
  id: 'check-1-4',
  name: 'check',
  value: 4,
  label: 'Checkbox 4',
  disabled: true
}, {
  id: 'check-1-5',
  name: 'check',
  value: 5,
  label: labelWithLink
}];
var StyledGrid = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-czrehc-0"
})(["display:inline-grid;grid-template-columns:repeat(2,50%);gap:1rem;align-items:flex-start;& > * + *{margin-top:0 !important;}"]);
var Default = exports.Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(StyledGrid, null, items.map(function (item) {
    return /*#__PURE__*/_react["default"].createElement(_.Checkbox, {
      key: item.id,
      id: item.id,
      name: item.name,
      value: item.value,
      label: item.label,
      disabled: item.disabled,
      checked: [2, 4].includes(item.value),
      onChange: func,
      onFocus: func,
      onBlur: func
    });
  }));
};
