// @salutejs/plasma-new-hope/styled-components/cjs/components/Tabs/ui/Tabs/Tabs.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.tabsRoot = exports.tabsConfig = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _Icon = /*#__PURE__*/require("../../../_Icon");
var _tokens = /*#__PURE__*/require("../../tokens");
var _utils = /*#__PURE__*/require("../../../../utils");
var _TabsContext = /*#__PURE__*/require("../../TabsContext");
var _base = /*#__PURE__*/require("./variations/_size/base");
var _base2 = /*#__PURE__*/require("./variations/_view/base");
var _base3 = /*#__PURE__*/require("./variations/_disabled/base");
var _base4 = /*#__PURE__*/require("./variations/_pilled/base");
var _base5 = /*#__PURE__*/require("./variations/_stretch/base");
var _Tabs = /*#__PURE__*/require("./Tabs.styles");
var _excluded = ["id", "stretch", "disabled", "size", "view", "children", "pilled", "index", "className"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var Keys = /*#__PURE__*/function (Keys) {
  Keys[Keys["end"] = 35] = "end";
  Keys[Keys["home"] = 36] = "home";
  Keys[Keys["left"] = 37] = "left";
  Keys[Keys["right"] = 39] = "right";
  return Keys;
}(Keys || {});
var tabsRoot = exports.tabsRoot = function tabsRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (props, outerRef) {
    var id = props.id,
      _props$stretch = props.stretch,
      stretch = _props$stretch === void 0 ? false : _props$stretch,
      _props$disabled = props.disabled,
      disabled = _props$disabled === void 0 ? false : _props$disabled,
      size = props.size,
      view = props.view,
      children = props.children,
      pilled = props.pilled,
      index = props.index,
      className = props.className,
      rest = _objectWithoutProperties(props, _excluded);
    var _useState = (0, _react.useState)(true),
      _useState2 = _slicedToArray(_useState, 2),
      firstItemVisible = _useState2[0],
      setFirstItemVisible = _useState2[1];
    var _useState3 = (0, _react.useState)(true),
      _useState4 = _slicedToArray(_useState3, 2),
      lastItemVisible = _useState4[0],
      setLastItemVisible = _useState4[1];
    var refs = (0, _react.useMemo)(function () {
      return new _TabsContext.TabItemRefs(index);
    }, []);
    var uniqId = (0, _plasmaCore.safeUseId)();
    var tabsId = id || uniqId;
    var isFilled = view === 'filled'; // outer padding is only for filled view
    var pilledAttr = view !== 'clear' && pilled;
    var pilledClass = pilledAttr ? _tokens.classes.tabsPilled : undefined;
    var stretchClass = firstItemVisible && lastItemVisible && stretch ? _tokens.classes.tabsStretch : undefined;
    var hasLeftArrowClass = !firstItemVisible ? _tokens.classes.tabsHasLeftArrow : undefined;
    var hasRightArrowClass = !lastItemVisible ? _tokens.classes.tabsHasRightArrow : undefined;
    var scrollRef = (0, _react.useRef)(null);
    var trackRef = (0, _react.useRef)(null);
    var onPrev = (0, _react.useCallback)(function () {
      !disabled && scrollRef.current && (0, _plasmaCore.animatedScrollToX)(scrollRef.current, scrollRef.current.scrollLeft - scrollRef.current.offsetWidth / 2);
    }, [disabled, scrollRef]);
    var onNext = (0, _react.useCallback)(function () {
      !disabled && scrollRef.current && (0, _plasmaCore.animatedScrollToX)(scrollRef.current, scrollRef.current.scrollLeft + scrollRef.current.offsetWidth / 2);
    }, [disabled, scrollRef]);
    var PreviousButton = (0, _react.useMemo)(function () {
      return /*#__PURE__*/_react["default"].createElement(_Tabs.StyledArrow, {
        type: "button",
        "aria-label": "\u041F\u0440\u0435\u0434\u044B\u0434\u0443\u0449\u0438\u0439 \u0442\u0430\u0431",
        onClick: onPrev,
        tabIndex: disabled ? -1 : 0,
        disabled: disabled,
        isFilled: isFilled,
        isLeftArrow: true
      }, /*#__PURE__*/_react["default"].createElement(_Icon.IconDisclosureLeft, {
        color: "var(".concat(_tokens.tokens.arrowColor, ")")
      }));
    }, [onPrev, disabled, isFilled]);
    var NextButton = (0, _react.useMemo)(function () {
      return /*#__PURE__*/_react["default"].createElement(_Tabs.StyledArrow, {
        type: "button",
        "aria-label": "\u0421\u043B\u0435\u0434\u0443\u044E\u0449\u0438\u0439 \u0442\u0430\u0431",
        onClick: onNext,
        tabIndex: disabled ? -1 : 0,
        disabled: disabled,
        isFilled: isFilled
      }, /*#__PURE__*/_react["default"].createElement(_Icon.IconDisclosureRight, {
        color: "var(".concat(_tokens.tokens.arrowColor, ")")
      }));
    }, [onNext, disabled, isFilled]);
    var handleScroll = (0, _react.useCallback)(function (event) {
      event.stopPropagation();
      var maxScrollLeft = event.currentTarget.scrollWidth - event.currentTarget.clientWidth;
      setFirstItemVisible(event.currentTarget.scrollLeft <= 0);
      setLastItemVisible(event.currentTarget.scrollLeft >= maxScrollLeft);
    }, [setFirstItemVisible, setLastItemVisible]);
    var onKeyDown = (0, _react.useCallback)(function (event) {
      if (index === undefined) {
        return;
      }
      var minIndex = 0;
      var maxIndex = refs.items.length - 1;
      var nextIndex;
      switch (event.keyCode) {
        case Keys.end:
          nextIndex = maxIndex;
          break;
        case Keys.left:
          nextIndex = index > minIndex ? index - 1 : index;
          break;
        case Keys.right:
          nextIndex = index < maxIndex ? index + 1 : index;
          break;
        case Keys.home:
          nextIndex = minIndex;
          break;
        default:
          return;
      }
      if (nextIndex !== index) {
        var _refs$items$nextIndex, _refs$items$nextIndex2;
        event.preventDefault();
        (_refs$items$nextIndex = refs.items[nextIndex].current) === null || _refs$items$nextIndex === void 0 || _refs$items$nextIndex.focus();
        (_refs$items$nextIndex2 = refs.items[nextIndex].current) === null || _refs$items$nextIndex2 === void 0 || _refs$items$nextIndex2.scrollIntoView({
          block: 'center',
          inline: 'center',
          behavior: 'smooth'
        });
      }
    }, [index]);
    (0, _react.useEffect)(function () {
      var _scrollRef$current, _scrollRef$current2;
      setLastItemVisible(((_scrollRef$current = scrollRef.current) === null || _scrollRef$current === void 0 ? void 0 : _scrollRef$current.scrollWidth) === ((_scrollRef$current2 = scrollRef.current) === null || _scrollRef$current2 === void 0 ? void 0 : _scrollRef$current2.clientWidth));
    }, []);
    return /*#__PURE__*/_react["default"].createElement(_TabsContext.TabsContext.Provider, {
      value: refs
    }, /*#__PURE__*/_react["default"].createElement(Root, _extends({
      view: view,
      role: "tablist",
      size: size,
      pilled: pilled,
      id: tabsId,
      ref: outerRef,
      disabled: disabled,
      className: (0, _utils.cx)(pilledClass, stretchClass, hasLeftArrowClass, hasRightArrowClass, className),
      onKeyDown: onKeyDown
    }, rest), !firstItemVisible && PreviousButton, /*#__PURE__*/_react["default"].createElement(_Tabs.StyledContentWrapper, {
      ref: scrollRef,
      onScroll: handleScroll
    }, /*#__PURE__*/_react["default"].createElement(_Tabs.StyledContent, {
      ref: trackRef
    }, children)), !lastItemVisible && NextButton));
  });
};
var tabsConfig = exports.tabsConfig = {
  name: 'Tabs',
  tag: 'div',
  layout: tabsRoot,
  base: _Tabs.base,
  variations: {
    size: {
      css: _base.base
    },
    view: {
      css: _base2.base
    },
    disabled: {
      css: _base3.base,
      attrs: true
    },
    stretch: {
      css: _base5.base,
      attrs: true
    },
    pilled: {
      css: _base4.base
    }
  },
  defaults: {
    view: 'clear',
    size: 'xs'
  }
};
