// @salutejs/plasma-new-hope/styled-components/cjs/components/Tabs/ui/Tabs/Tabs.styles.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.base = exports.StyledContentWrapper = exports.StyledContent = exports.StyledArrow = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _mixins = /*#__PURE__*/require("../../../../mixins");
var _tokens = /*#__PURE__*/require("../../tokens");
var _templateObject;
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
var base = exports.base = /*#__PURE__*/(0, _styledComponents.css)(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n    display: flex;\n    align-items: center;\n    position: relative;\n"])));
var StyledContent = exports.StyledContent = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-1ibkmsl-0"
})(["display:inline-flex;flex-direction:row;align-items:center;"]);
var StyledContentWrapper = exports.StyledContentWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-1ibkmsl-1"
})(["margin:-0.125rem;padding:0.25rem;box-sizing:content-box;overflow:scroll;position:relative;height:100%;width:100%;display:flex;align-items:center;scrollbar-width:none;::-webkit-scrollbar{display:none;}overscroll-behavior:contain;"]);
var StyledArrow = exports.StyledArrow = /*#__PURE__*/_styledComponents["default"].button.withConfig({
  componentId: "plasma-new-hope__sc-1ibkmsl-2"
})(["display:flex;cursor:pointer;border:none;background-color:transparent;padding:0;outline:none;", ";&[disabled]{cursor:not-allowed;}--plasma_private-outer-padding:", ";padding-right:", ";padding-left:", ";"], /*#__PURE__*/(0, _mixins.addFocus)({
  outlineSize: '0.063rem',
  outlineOffset: '-0.125rem',
  outlineColor: /*#__PURE__*/"var(".concat(_tokens.tokens.outlineFocusColor, ")"),
  outlineRadius: 'calc(var(--plasma_private-outline-radius) - 0.063rem)'
}), function (_ref) {
  var isFilled = _ref.isFilled;
  return isFilled ? "var(".concat(_tokens.tokens.arrowOuterPadding, ")") : '';
}, function (_ref2) {
  var isLeftArrow = _ref2.isLeftArrow;
  return isLeftArrow ? "var(".concat(_tokens.tokens.arrowInnerPadding, ")") : 'var(--plasma_private-outer-padding)';
}, function (_ref3) {
  var isLeftArrow = _ref3.isLeftArrow;
  return isLeftArrow ? 'var(--plasma_private-outer-padding)' : "var(".concat(_tokens.tokens.arrowInnerPadding, ")");
});
