// @salutejs/plasma-hope/components/Select/views/b2c/SelectView.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectView = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _SelectButton = /*#__PURE__*/require("../../SelectButton");
var _SelectDropdown = /*#__PURE__*/require("../../SelectDropdown");
var _Dropdown = /*#__PURE__*/require("../../../Dropdown");
var _SelectGroup = /*#__PURE__*/require("../../SelectGroup");
var _excluded = ["id", "placeholder", "value", "helperText", "disabled", "status", "className", "style", "items", "multiselect", "onItemSelect"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var SelectButton = /*#__PURE__*/(0, _styledComponents["default"])(_SelectButton.SelectButton).attrs({
  design: 'b2c'
}).withConfig({
  componentId: "plasma-web-core__sc-samcez-0"
})([""]);
var StyledRoot = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.TextFieldRoot).withConfig({
  componentId: "plasma-web-core__sc-samcez-1"
})(["", " &:not(:last-child){margin-right:0.125rem;}", " &:not(:first-child) ", "{border-top-left-radius:0;border-bottom-left-radius:0;}", " &:not(:last-child) ", "{border-top-right-radius:0;border-bottom-right-radius:0;}"], _SelectGroup.SelectGroup, _SelectGroup.SelectGroup, SelectButton, _SelectGroup.SelectGroup, SelectButton);
var SelectDropdown = /*#__PURE__*/(0, _styledComponents["default"])(_SelectDropdown.SelectDropdown).attrs({
  design: 'b2c'
}).withConfig({
  componentId: "plasma-web-core__sc-samcez-2"
})([""]);
var DropdownButton = /*#__PURE__*/(0, _Dropdown.withAssistiveDropdown)(SelectButton, SelectDropdown);

/**
 * Поле с выпадающим списком.
 */
var SelectView = exports.SelectView = /*#__PURE__*/(0, _react.forwardRef)(function (_ref, ref) {
  var id = _ref.id,
    placeholder = _ref.placeholder,
    value = _ref.value,
    helperText = _ref.helperText,
    disabled = _ref.disabled,
    status = _ref.status,
    className = _ref.className,
    style = _ref.style,
    items = _ref.items,
    multiselect = _ref.multiselect,
    onItemSelect = _ref.onItemSelect,
    rest = _objectWithoutProperties(_ref, _excluded);
  var hasItems = Array.isArray(items) && items.length > 0;
  return /*#__PURE__*/_react["default"].createElement(StyledRoot, {
    $size: "m",
    $disabled: disabled,
    $isContentRight: hasItems,
    $isHelper: Boolean(helperText),
    status: status,
    className: className,
    style: style
  }, /*#__PURE__*/_react["default"].createElement(DropdownButton, _extends({}, rest, {
    ref: ref,
    id: id ? "".concat(id, "-dropdown") : id,
    role: "combobox",
    menuRole: "listbox",
    menuItemRole: "option",
    value: value,
    placeholder: placeholder,
    hasItems: hasItems,
    status: status,
    items: items,
    disabled: disabled,
    closeOnSelect: !multiselect,
    onItemSelect: onItemSelect
  })), helperText && /*#__PURE__*/_react["default"].createElement(_plasmaCore.TextFieldHelper, null, helperText));
});
