// @salutejs/plasma-core/tokens/colors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.white = exports.warning = exports.voicePhraseGradient = exports.transparent = exports.text = exports.tertiary = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.surfaceCard = exports.success = exports.speechBubbleSent = exports.speechBubbleReceived = exports.skeletonGradientLighter = exports.skeletonGradient = exports.secondary = exports.primary = exports.paragraph = exports.overlay = exports.inverse = exports.gradient = exports.dark03 = exports.dark02 = exports.dark01 = exports.critical = exports.buttonWarning = exports.buttonSuccess = exports.buttonSecondary = exports.buttonPrimary = exports.buttonFocused = exports.buttonCritical = exports.buttonChecked = exports.buttonAccent = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.black = exports.backgroundTertiary = exports.backgroundSecondary = exports.backgroundPrimary = exports.background = exports.accent = void 0;
var white = exports.white = 'var(--plasma-colors-white)';
var whitePrimary = exports.whitePrimary = 'var(--plasma-colors-white-primary)';
var whiteSecondary = exports.whiteSecondary = 'var(--plasma-colors-white-secondary)';
var whiteTertiary = exports.whiteTertiary = 'var(--plasma-colors-white-tertiary)';
var black = exports.black = 'var(--plasma-colors-black)';
var blackPrimary = exports.blackPrimary = 'var(--plasma-colors-black-primary)';
var blackSecondary = exports.blackSecondary = 'var(--plasma-colors-black-secondary)';
var blackTertiary = exports.blackTertiary = 'var(--plasma-colors-black-tertiary)';
var dark01 = exports.dark01 = 'var(--plasma-colors-dark01)';
var dark02 = exports.dark02 = 'var(--plasma-colors-dark02)';
var dark03 = exports.dark03 = 'var(--plasma-colors-dark03)';
var transparent = exports.transparent = 'var(--plasma-colors-transparent)';
var text = exports.text = 'var(--plasma-colors-text)';
var primary = exports.primary = 'var(--plasma-colors-primary)';
var secondary = exports.secondary = 'var(--plasma-colors-secondary)';
var tertiary = exports.tertiary = 'var(--plasma-colors-tertiary)';
var inverse = exports.inverse = 'var(--plasma-colors-inverse)';
var paragraph = exports.paragraph = 'var(--plasma-colors-paragraph)';
var background = exports.background = 'var(--plasma-colors-background)';
var backgroundPrimary = exports.backgroundPrimary = 'var(--plasma-colors-background-primary)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107
var backgroundSecondary = exports.backgroundSecondary = 'var(--plasma-colors-background-secondary)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107
var backgroundTertiary = exports.backgroundTertiary = 'var(--plasma-colors-background-tertiary)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107

var success = exports.success = 'var(--plasma-colors-success)';
var warning = exports.warning = 'var(--plasma-colors-warning)';
var critical = exports.critical = 'var(--plasma-colors-critical)';
var overlay = exports.overlay = 'var(--plasma-colors-overlay)';
var surfaceLiquid01 = exports.surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107
var surfaceLiquid02 = exports.surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107
var surfaceLiquid03 = exports.surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03)'; // ToDo: https://github.com/salute-developers/plasma/issues/1107
var surfaceSolid01 = exports.surfaceSolid01 = 'var(--plasma-colors-surface-solid01)';
var surfaceSolid02 = exports.surfaceSolid02 = 'var(--plasma-colors-surface-solid02)';
var surfaceSolid03 = exports.surfaceSolid03 = 'var(--plasma-colors-surface-solid03)';
var surfaceCard = exports.surfaceCard = 'var(--plasma-colors-surface-card)';
var buttonPrimary = exports.buttonPrimary = 'var(--plasma-colors-button-primary)';
var buttonSecondary = exports.buttonSecondary = 'var(--plasma-colors-button-secondary)';
var buttonSuccess = exports.buttonSuccess = 'var(--plasma-colors-button-success)';
var buttonWarning = exports.buttonWarning = 'var(--plasma-colors-button-warning)';
var buttonCritical = exports.buttonCritical = 'var(--plasma-colors-button-critical)';
var buttonChecked = exports.buttonChecked = 'var(--plasma-colors-button-checked)';
var speechBubbleSent = exports.speechBubbleSent = 'var(--plasma-colors-speech-bubble-sent)';
var speechBubbleReceived = exports.speechBubbleReceived = 'var(--plasma-colors-speech-bubble-received)';
var accent = exports.accent = 'var(--plasma-colors-accent)';
var buttonAccent = exports.buttonAccent = 'var(--plasma-colors-button-accent)';
var buttonFocused = exports.buttonFocused = 'var(--plasma-colors-button-focused)';
var gradient = exports.gradient = 'var(--plasma-colors-gradient)';
var voicePhraseGradient = exports.voicePhraseGradient = 'var(--plasma-colors-voice-phrase-gradient)';
var skeletonGradient = exports.skeletonGradient = 'var(--plasma-colors-skeleton-gradient)';
var skeletonGradientLighter = exports.skeletonGradientLighter = 'var(--plasma-colors-skeleton-gradient-lighter)';
