// @salutejs/plasma-new-hope/cjs/components/Slider/utils/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

/**
 * Расчитать значение слайдера с учетом его координат и шага изменений.
 * @param {number} handleCenterXRelative
 * @param {number} stepSize
 * @param {number} min
 * @param {number} max
 * @return {number}
 */
function getSliderThumbValue(handleCenterXRelative, stepSize, min, max) {
  var newValue = Math.round(handleCenterXRelative / stepSize) + min;
  return Math.min(Math.max(newValue, min), max);
}

/**
 * Расчитывает значение отступа слайдера на основе его положения (справа, слева) на отрезке слайдера.
 * Значение используется для правильного расчета ограничения движения слайдера.
 * @param ref
 * @param {'left' | 'right'} side
 * @return Array<number>
 */
function getOffsets(ref, side) {
  if (!ref || !('current' in ref) || !ref.current || !side) {
    return [0, 0];
  }
  var size = ref.current.clientWidth;
  if (side === 'left') {
    return [0, size];
  }
  if (side === 'right') {
    return [size, 0];
  }
  return [0, 0];
}
var sizeData = {
  s: {
    indent: 0.5,
    fontSizeMultiplier: 16
  },
  m: {
    indent: 0.75,
    fontSizeMultiplier: 16
  },
  l: {
    indent: 0.75,
    fontSizeMultiplier: 16
  }
};

exports.getOffsets = getOffsets;
exports.getSliderThumbValue = getSliderThumbValue;
exports.sizeData = sizeData;
