// @salutejs/plasma-new-hope/es/components/Slider/components/Double/Double.styles.js
import './Double.styles_1jhda4u.css';
import { styled } from '@linaria/react';
import '@linaria/core';
import { component, mergeConfig } from '../../../../engines/common.js';
import { textFieldConfig } from '../../../TextField/TextField.js';

var mergedConfig = /*#__PURE__*/mergeConfig(textFieldConfig);
var TextField = /*#__PURE__*/component(mergedConfig);
var LabelWrapper = /*#__PURE__*/styled('div')({
  name: "LabelWrapper",
  "class": "lcs82cd",
  propsAsIs: false
});
var LabelContentLeft = /*#__PURE__*/styled('div')({
  name: "LabelContentLeft",
  "class": "l1tvgbgq",
  propsAsIs: false
});
var Label = /*#__PURE__*/styled('label')({
  name: "Label",
  "class": "l1at2q9b",
  propsAsIs: false
});
var InputsWrapper = /*#__PURE__*/styled('div')({
  name: "InputsWrapper",
  "class": "i11dc2og",
  propsAsIs: false
});

// NOTE: переопределение токенов TextField
var _exp12 = function _exp12() {
  return TextField;
};
var StyledInput = /*#__PURE__*/styled(_exp12())({
  name: "StyledInput",
  "class": "s4zvp7i",
  propsAsIs: true
});
var DoubleWrapper = /*#__PURE__*/styled('div')({
  name: "DoubleWrapper",
  "class": "d3kvcjo",
  propsAsIs: false
});
var SliderWrapper = /*#__PURE__*/styled('div')({
  name: "SliderWrapper",
  "class": "s11ntvbh",
  propsAsIs: false
});

export { DoubleWrapper, InputsWrapper, Label, LabelContentLeft, LabelWrapper, SliderWrapper, StyledInput };
