// @salutejs/plasma-hope/components/TextField/views/web/Input.props.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.statuses = exports.sizes = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _tokens = /*#__PURE__*/require("../../../../tokens");
var sizes = exports.sizes = {
  l: {
    height: 'var(--plasma-input-l-height)',
    paddingY: 'var(--plasma-input-l-padding-y)',
    paddingX: 'var(--plasma-input-l-padding-x)',
    hasLabelPaddingTop: 'var(--plasma-input-l-has-label-padding-top)',
    hasLabelPaddingBottom: 'var(--plasma-input-l-has-label-padding-bottom)',
    hasContentPaddingLeft: 'var(--plasma-input-l-has-content-padding-left)',
    hasContentPaddingRight: 'var(--plasma-input-l-has-content-padding-right)'
  },
  m: {
    height: 'var(--plasma-input-m-height)',
    paddingY: 'var(--plasma-input-m-padding-y)',
    paddingX: 'var(--plasma-input-m-padding-x)',
    hasLabelPaddingTop: 'var(--plasma-input-m-padding-y)',
    hasLabelPaddingBottom: 'var(--plasma-input-m-padding-y)',
    hasContentPaddingLeft: 'var(--plasma-input-m-has-content-padding-left)',
    hasContentPaddingRight: 'var(--plasma-input-m-has-content-padding-right)'
  },
  s: {
    height: 'var(--plasma-input-s-height)',
    paddingY: 'var(--plasma-input-s-padding-y)',
    paddingX: 'var(--plasma-input-s-padding-x)',
    hasLabelPaddingTop: 'var(--plasma-input-s-padding-y)',
    hasLabelPaddingBottom: 'var(--plasma-input-s-padding-y)',
    hasContentPaddingLeft: 'var(--plasma-input-s-has-content-padding-left)',
    hasContentPaddingRight: 'var(--plasma-input-s-has-content-padding-right)'
  },
  // INFO: Временно s = xs, пока не произведен редизайн
  xs: {
    height: 'var(--plasma-input-s-height)',
    paddingY: 'var(--plasma-input-s-padding-y)',
    paddingX: 'var(--plasma-input-s-padding-x)',
    hasLabelPaddingTop: 'var(--plasma-input-s-padding-y)',
    hasLabelPaddingBottom: 'var(--plasma-input-s-padding-y)',
    hasContentPaddingLeft: 'var(--plasma-input-s-has-content-padding-left)',
    hasContentPaddingRight: 'var(--plasma-input-s-has-content-padding-right)'
  }
};
var statuses = exports.statuses = {
  success: /*#__PURE__*/(0, _styledComponents.css)(["&,&:hover,&:focus,&:hover:focus{background-color:", ";color:", ";caret-color:", ";box-shadow:inset 0 0 0 ", " ", ";}"], _tokens.inputSuccessBackground, _tokens.inputSuccessColor, _tokens.inputSuccessCaretColor, _tokens.inputBorderWidth, _tokens.inputSuccessBorderColor),
  warning: /*#__PURE__*/(0, _styledComponents.css)(["&,&:hover,&:focus,&:hover:focus{background-color:", ";color:", ";caret-color:", ";box-shadow:inset 0 0 0 ", " ", ";}"], _tokens.inputWarningBackground, _tokens.inputWarningColor, _tokens.inputWarningCaretColor, _tokens.inputBorderWidth, _tokens.inputWarningBorderColor),
  error: /*#__PURE__*/(0, _styledComponents.css)(["&,&:hover,&:focus,&:hover:focus{background-color:", ";color:", ";caret-color:", ";box-shadow:inset 0 0 0 ", " ", ";}"], _tokens.inputErrorBackground, _tokens.inputErrorColor, _tokens.inputErrorCaretColor, _tokens.inputBorderWidth, _tokens.inputErrorBorderColor)
};
