// @salutejs/plasma-new-hope/styled-components/cjs/examples/typograpy/components/Old/Button/Button.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _tokens = /*#__PURE__*/require("../../../../../components/Typography/tokens");
var config = exports.config = {
  defaults: {
    size: 'button1'
  },
  variations: {
    size: {
      button1: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--plasma-typo-button1-font-family);", ":var(--plasma-typo-button1-font-size);", ":var(--plasma-typo-button1-font-style);", ":var(--plasma-typo-button1-font-weight);", ":var(--plasma-typo-button1-letter-spacing);", ":var(--plasma-typo-button1-line-height);"], _tokens.tokens.typoFontFamily, _tokens.tokens.typoFontSize, _tokens.tokens.typoFontStyle, _tokens.tokens.typoFontWeight, _tokens.tokens.typoFontLetterSpacing, _tokens.tokens.typoFontLineHeight),
      button2: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--plasma-typo-button2-font-family);", ":var(--plasma-typo-button2-font-size);", ":var(--plasma-typo-button2-font-style);", ":var(--plasma-typo-button2-font-weight);", ":var(--plasma-typo-button2-letter-spacing);", ":var(--plasma-typo-button2-line-height);"], _tokens.tokens.typoFontFamily, _tokens.tokens.typoFontSize, _tokens.tokens.typoFontStyle, _tokens.tokens.typoFontWeight, _tokens.tokens.typoFontLetterSpacing, _tokens.tokens.typoFontLineHeight)
    }
  }
};
