// @salutejs/plasma-new-hope/es/components/Radiobox/Radiobox.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef, useMemo } from 'react';
import { safeUseId, extractTextFrom } from '@salutejs/plasma-core';
import { StyledInput, StyledContentWrapper, StyledTrigger, StyledContent, StyledLabel, StyledDescription } from '../Checkbox/Checkbox.styles.js';
import { cx } from '../../utils/index.js';
import { base as base$2 } from './variations/_view/base.js';
import { base as base$1 } from './variations/_size/base.js';
import { base as base$3 } from './variations/_disabled/base.js';
import { base as base$4 } from './variations/_focused/base.js';
import { StyledEllipse, base } from './Radiobox.styles.js';
import { classes } from './Radiobox.tokens.js';

var _StyledEllipse;
var _excluded = ["size", "view", "focused", "disabled", "id", "label", "description", "style", "className", "singleLine", "aria-label"];
var radioboxRoot = function radioboxRoot(Root) {
  return /*#__PURE__*/forwardRef(function (props, ref) {
    var size = props.size,
      view = props.view,
      focused = props.focused,
      disabled = props.disabled,
      id = props.id,
      label = props.label,
      description = props.description,
      style = props.style,
      className = props.className,
      _props$singleLine = props.singleLine,
      singleLine = _props$singleLine === void 0 ? false : _props$singleLine,
      ariaLabelExternal = props['aria-label'],
      rest = _objectWithoutProperties(props, _excluded);
    var uniqId = safeUseId();
    var uniqLabelId = "label-".concat(uniqId);
    var uniqDescriptionId = "description-".concat(uniqId);
    var radioboxId = id || "input-".concat(uniqId);
    var ariaLabel = useMemo(function () {
      return ariaLabelExternal || extractTextFrom(label);
    }, [ariaLabelExternal, label]);
    var isLabelAriaHidden = typeof label === 'string';
    var canFocused = focused ? 0 : -1;
    var hasContent = label || description;
    var singleLineClass = singleLine ? classes.singleLine : undefined;
    var onlyDescriptionClass = !label && description ? classes.onlyDescription : undefined;
    return /*#__PURE__*/React.createElement(Root, {
      view: view,
      size: size,
      disabled: disabled,
      focused: focused,
      style: style,
      className: className,
      tabIndex: -1
    }, /*#__PURE__*/React.createElement(StyledInput, _extends({}, rest, {
      id: radioboxId,
      ref: ref,
      type: "radio",
      disabled: disabled,
      "aria-label": ariaLabel,
      "aria-describedby": uniqDescriptionId,
      tabIndex: canFocused
    })), /*#__PURE__*/React.createElement(StyledContentWrapper, {
      htmlFor: radioboxId
    }, /*#__PURE__*/React.createElement(StyledTrigger, {
      className: classes.radioboxTrigger
    }, _StyledEllipse || (_StyledEllipse = /*#__PURE__*/React.createElement(StyledEllipse, null))), hasContent && /*#__PURE__*/React.createElement(StyledContent, {
      className: singleLineClass
    }, label && /*#__PURE__*/React.createElement(StyledLabel, {
      className: cx(singleLineClass, onlyDescriptionClass),
      id: uniqLabelId,
      "aria-hidden": isLabelAriaHidden
    }, label), description && /*#__PURE__*/React.createElement(StyledDescription, {
      className: singleLineClass,
      id: uniqDescriptionId
    }, description))));
  });
};
var radioboxConfig = {
  name: 'Radiobox',
  tag: 'div',
  layout: radioboxRoot,
  base: base,
  variations: {
    size: {
      css: base$1
    },
    view: {
      css: base$2
    },
    disabled: {
      css: base$3
    },
    focused: {
      css: base$4
    }
  },
  defaults: {
    size: 'm',
    view: 'accent'
  }
};

export { radioboxConfig, radioboxRoot };
