// @salutejs/plasma-new-hope/cjs/components/Calendar/ui/CalendarHeader/CalendarHeader.styles.js
import './CalendarHeader.styles_1c4p417.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
var index = require('../../mixins/index.js');

var StyledCalendarHeader = /*#__PURE__*/react.styled('div')({
  name: "StyledCalendarHeader",
  "class": "sjaqipy",
  propsAsIs: false
});
var _exp7 = function _exp7() {
  return index.buttonFocus;
};
var StyledHeader = /*#__PURE__*/react.styled(_exp7())({
  name: "StyledHeader",
  "class": "s1k7jebr",
  propsAsIs: true
});
var StyledHeaderDouble = /*#__PURE__*/react.styled('h4')({
  name: "StyledHeaderDouble",
  "class": "s1b07tlr",
  propsAsIs: false
});
var StyledArrows = /*#__PURE__*/react.styled('div')({
  name: "StyledArrows",
  "class": "s13yvda3",
  propsAsIs: false
});
var _exp23 = function _exp23() {
  return index.buttonFocus;
};
var StyledArrow = /*#__PURE__*/react.styled(_exp23())({
  name: "StyledArrow",
  "class": "s1i8jkj2",
  propsAsIs: true
});
var StyledNavigation = /*#__PURE__*/react.styled('div')({
  name: "StyledNavigation",
  "class": "sw3by8z",
  propsAsIs: false
});

exports.StyledArrow = StyledArrow;
exports.StyledArrows = StyledArrows;
exports.StyledCalendarHeader = StyledCalendarHeader;
exports.StyledHeader = StyledHeader;
exports.StyledHeaderDouble = StyledHeaderDouble;
exports.StyledNavigation = StyledNavigation;
