// @salutejs/plasma-new-hope/cjs/components/Select/Select.styles.js
import './Select.styles_1q3i6gw.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
require('@linaria/core');
var common = require('../../engines/common.js');
var Popover$1 = require('../Popover/Popover.js');

var StyledRoot = /*#__PURE__*/react.styled('div')({
  name: "StyledRoot",
  "class": "spxbau6",
  propsAsIs: false
});
var StyledSelect = /*#__PURE__*/react.styled('div')({
  name: "StyledSelect",
  "class": "s1yey2x3",
  propsAsIs: false
});

// issue #823
var Popover = /*#__PURE__*/common.component(Popover$1.popoverConfig);
var _exp12 = function _exp12() {
  return Popover;
};
var _exp15 = function _exp15() {
  return function (_ref) {
    var usePortal = _ref.usePortal,
      selectWidth = _ref.selectWidth;
    return usePortal ? "".concat(selectWidth, "px") : '100%';
  };
};
var StyledPopover = /*#__PURE__*/react.styled(_exp12())({
  name: "StyledPopover",
  "class": "svro45w",
  propsAsIs: true,
  vars: {
    "svro45w-0": [/*#__PURE__*/_exp15()]
  }
});
var _exp16 = function _exp16() {
  return function (_ref2) {
    var disabled = _ref2.disabled;
    return disabled ? 'not-allowed' : 'pointer';
  };
};
var StyledNativeSelect = /*#__PURE__*/react.styled('select')({
  name: "StyledNativeSelect",
  "class": "sgpldc9",
  propsAsIs: false,
  vars: {
    "sgpldc9-0": [/*#__PURE__*/_exp16()]
  }
});

exports.StyledNativeSelect = StyledNativeSelect;
exports.StyledPopover = StyledPopover;
exports.StyledRoot = StyledRoot;
exports.StyledSelect = StyledSelect;
