// @salutejs/plasma-hope/components/Select/SelectButton.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectButton = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _tokens = /*#__PURE__*/require("../../tokens");
var _SelectButton = /*#__PURE__*/require("./views/b2c/SelectButton");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _StyledArrow;
var _excluded = ["design"],
  _excluded2 = ["value", "placeholder", "hasItems"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('Select/SelectButton');
var statuses = {
  success: _plasmaCore.success,
  warning: _plasmaCore.warning,
  error: _plasmaCore.critical
};
var StyledArrow = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaIcons.IconChevronDown).withConfig({
  componentId: "plasma-web-core__sc-wqinip-0"
})(["margin:var(--plasma-select-button-arrow-margin,0 0 0 0.75rem);transition:color 0.3s ease-in-out,transform 0.15s ease-in-out;pointer-events:none;user-select:none;"]);
var StyledText = /*#__PURE__*/_styledComponents["default"].span.withConfig({
  componentId: "plasma-web-core__sc-wqinip-1"
})(["", ";color:", ";transition:color 0.3s ease-in-out;pointer-events:none;user-select:none;"], _plasmaCore.applyEllipsis, _plasmaCore.primary);
var StyledPlaceholder = /*#__PURE__*/_styledComponents["default"].span.withConfig({
  componentId: "plasma-web-core__sc-wqinip-2"
})(["", ";color:", ";pointer-events:none;user-select:none;"], _plasmaCore.applyEllipsis, _plasmaCore.tertiary);
var StyledButton = /*#__PURE__*/_styledComponents["default"].button.withConfig({
  componentId: "plasma-web-core__sc-wqinip-3"
})(["", ";display:flex;align-items:center;justify-content:space-between;box-sizing:border-box;width:100%;height:3rem;padding:0.875rem 0.9375rem;border:1px solid ", ";border-radius:0.25rem;background-color:transparent;color:", ";transition:border-color 0.3s ease-in-out;", " &:disabled{cursor:inherit;}&:not(:disabled):focus{outline:0 none;border-color:", ";color:", ";}", " ", " ", " ", ""], _plasmaCore.body1, _tokens.inputBorder, _plasmaCore.secondary, function (_ref) {
  var hasItems = _ref.hasItems;
  return hasItems && (0, _styledComponents.css)(["&:not(:disabled):hover{cursor:pointer;border-color:", ";color:", ";}"], _tokens.inputBorderHover, _plasmaCore.secondary);
}, _plasmaCore.accent, _plasmaCore.accent, function (_ref2) {
  var focused = _ref2.focused;
  return focused && (0, _styledComponents.css)(["border-color:", ";color:", ";"], _plasmaCore.accent, _plasmaCore.accent);
}, function (_ref3) {
  var status = _ref3.status;
  return status && (0, _styledComponents.css)(["border-color:", ";color:", ";&:hover,&:focus{border-color:", ";}"], statuses[status], statuses[status], statuses[status]);
}, function (_ref4) {
  var isOpen = _ref4.isOpen;
  return isOpen && (0, _styledComponents.css)(["", "{transform:rotate(-180deg);}"], StyledArrow);
}, function (_ref5) {
  var design = _ref5.design,
    props = _objectWithoutProperties(_ref5, _excluded);
  return design === 'b2c' && (0, _SelectButton.applyB2CSelectButton)(props);
});

// eslint-disable-next-line prefer-arrow-callback
var SelectButton = exports.SelectButton = /*#__PURE__*/(0, _react.forwardRef)(function SelectButton(_ref6, ref) {
  var value = _ref6.value,
    placeholder = _ref6.placeholder,
    hasItems = _ref6.hasItems,
    rest = _objectWithoutProperties(_ref6, _excluded2);
  return /*#__PURE__*/_react["default"].createElement(StyledButton, _extends({
    type: "button"
  }, rest, {
    ref: ref,
    hasItems: hasItems,
    "aria-label": placeholder
  }), value && /*#__PURE__*/_react["default"].createElement(StyledText, null, value), placeholder && !value && /*#__PURE__*/_react["default"].createElement(StyledPlaceholder, {
    "aria-hidden": "true"
  }, placeholder), hasItems && (_StyledArrow || (_StyledArrow = /*#__PURE__*/_react["default"].createElement(StyledArrow, {
    size: "xs",
    color: "inherit"
  }))));
});
