// @salutejs/plasma-new-hope/es/components/Toast/Toast.styles.js
import './Toast.styles_1v9oj7f.css';
import { styled } from '@linaria/react';
import { Overlay } from '../Overlay/Overlay.js';
import { Button } from './utils.js';

var base = "b1xqk743";
var _exp3 = function _exp3() {
  return function (_ref) {
    var position = _ref.position,
      _ref$offset = _ref.offset,
      offset = _ref$offset === void 0 ? 0 : _ref$offset;
    return position === 'top' ? "".concat(Number(offset) + 5, "rem") : 'unset';
  };
};
var _exp4 = function _exp4() {
  return function (_ref2) {
    var position = _ref2.position,
      _ref2$offset = _ref2.offset,
      offset = _ref2$offset === void 0 ? 0 : _ref2$offset;
    return position === 'bottom' ? "".concat(Number(offset) + 5, "rem") : 'unset';
  };
};
var _exp8 = function _exp8() {
  return function (_ref3) {
    var position = _ref3.position;
    return "translate(-50%, ".concat(position === 'top' ? '-' : '', "5rem)");
  };
};
var StyledRoot = /*#__PURE__*/styled('div')({
  name: "StyledRoot",
  "class": "s1ajsqhx",
  propsAsIs: false,
  vars: {
    "s1ajsqhx-0": [/*#__PURE__*/_exp3()],
    "s1ajsqhx-1": [/*#__PURE__*/_exp4()],
    "s1ajsqhx-2": [/*#__PURE__*/_exp8()]
  }
});
var _exp9 = function _exp9() {
  return Button;
};
var CloseIconWrapper = /*#__PURE__*/styled(_exp9())({
  name: "CloseIconWrapper",
  "class": "cb0tpdq",
  propsAsIs: true
});
var _exp13 = function _exp13() {
  return Overlay;
};
var StyledOverlay = /*#__PURE__*/styled(_exp13())({
  name: "StyledOverlay",
  "class": "segu2tf",
  propsAsIs: true
});
var StyledContentLeft = /*#__PURE__*/styled('div')({
  name: "StyledContentLeft",
  "class": "s1bt6ppa",
  propsAsIs: false
});
var StyledContent = /*#__PURE__*/styled('div')({
  name: "StyledContent",
  "class": "s1h44nk1",
  propsAsIs: false
});

export { CloseIconWrapper, StyledContent, StyledContentLeft, StyledOverlay, StyledRoot, base };
