// @salutejs/plasma-ui/es/components/Card/Card.examples.js
var _excluded = ["title", "subtitle", "imageSrc"],
  _excluded2 = ["title", "subtitle", "roundness", "imageSrc", "imageRatio", "imageCustomRatio", "textAlign"],
  _excluded3 = ["title", "subtitle", "imageSrc", "imageRatio", "imageCustomRatio", "textAlign"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React from 'react';
import styled, { css } from 'styled-components';
import { primary, secondary } from '@salutejs/plasma-tokens';
import { applyMaxLines } from '@salutejs/plasma-core';
import { Body1, Footnote1 } from '../Typography';
import { Card } from './Card';
import { CardBody } from './CardBody';
import { CardContent } from './CardContent';
import { CardMedia } from './CardMedia';
var applyTextAlign = function applyTextAlign(_ref) {
  var textAlign = _ref.textAlign;
  return textAlign && css(["text-align:", ";"], textAlign);
};
var StyledProductIcon = /*#__PURE__*/styled(CardMedia).withConfig({
  componentId: "plasma-ui__sc-3b5xy8-0"
})(["width:3rem;height:3rem;padding:0;border-radius:1rem;"]);
var StyledItemDescr = /*#__PURE__*/styled(Footnote1).withConfig({
  componentId: "plasma-ui__sc-3b5xy8-1"
})(["", " ", " margin-top:0.25rem;color:", ";", " + &{margin-top:1rem;}"], applyMaxLines, applyTextAlign, secondary, StyledProductIcon);
var StyledItemTitle = /*#__PURE__*/styled(Body1).withConfig({
  componentId: "plasma-ui__sc-3b5xy8-2"
})(["", " ", " margin-top:0.75rem;color:", ";", " + &{margin-top:0.25rem;}"], applyMaxLines, applyTextAlign, primary, StyledItemDescr);
export var ProductCard = function ProductCard(_ref2) {
  var title = _ref2.title,
    subtitle = _ref2.subtitle,
    imageSrc = _ref2.imageSrc,
    rest = _objectWithoutProperties(_ref2, _excluded);
  return /*#__PURE__*/React.createElement(Card, rest, /*#__PURE__*/React.createElement(CardBody, null, /*#__PURE__*/React.createElement(CardContent, null, /*#__PURE__*/React.createElement(StyledProductIcon, {
    src: imageSrc
  }), subtitle && /*#__PURE__*/React.createElement(StyledItemDescr, {
    maxLines: 1
  }, subtitle), /*#__PURE__*/React.createElement(StyledItemTitle, {
    maxLines: 1
  }, title))));
};
export var MusicCard = function MusicCard(_ref3) {
  var title = _ref3.title,
    subtitle = _ref3.subtitle,
    _ref3$roundness = _ref3.roundness,
    roundness = _ref3$roundness === void 0 ? 12 : _ref3$roundness,
    imageSrc = _ref3.imageSrc,
    imageRatio = _ref3.imageRatio,
    imageCustomRatio = _ref3.imageCustomRatio,
    textAlign = _ref3.textAlign,
    rest = _objectWithoutProperties(_ref3, _excluded2);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Card, _extends({
    roundness: roundness
  }, rest), /*#__PURE__*/React.createElement(CardBody, null, /*#__PURE__*/React.createElement(CardMedia, {
    src: imageSrc,
    ratio: imageRatio,
    customRatio: imageCustomRatio
  }))), /*#__PURE__*/React.createElement(StyledItemTitle, {
    maxLines: 1,
    textAlign: textAlign
  }, title), subtitle && /*#__PURE__*/React.createElement(StyledItemDescr, {
    maxLines: 1,
    textAlign: textAlign
  }, subtitle));
};
export var GalleryCard = function GalleryCard(_ref4) {
  var title = _ref4.title,
    subtitle = _ref4.subtitle,
    imageSrc = _ref4.imageSrc,
    imageRatio = _ref4.imageRatio,
    imageCustomRatio = _ref4.imageCustomRatio,
    textAlign = _ref4.textAlign,
    rest = _objectWithoutProperties(_ref4, _excluded3);
  return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(Card, rest, /*#__PURE__*/React.createElement(CardBody, null, /*#__PURE__*/React.createElement(CardMedia, {
    src: imageSrc,
    ratio: imageRatio,
    customRatio: imageCustomRatio
  }), /*#__PURE__*/React.createElement(CardContent, {
    cover: true
  }, /*#__PURE__*/React.createElement(StyledItemTitle, {
    maxLines: 1,
    textAlign: textAlign
  }, title), subtitle && /*#__PURE__*/React.createElement(StyledItemDescr, {
    maxLines: 1,
    textAlign: textAlign
  }, subtitle)))));
};
