// @salutejs/plasma-new-hope/es/components/Combobox/Combobox.js
import { objectWithoutProperties as _objectWithoutProperties, slicedToArray as _slicedToArray, extends as _extends } from '../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef, useState, useRef, useEffect, Children } from 'react';
import { safeUseId, useForkRef } from '@salutejs/plasma-core';
import { cx } from '../../utils/index.js';
import { getPlacements } from '../Dropdown/utils/index.js';
import { getChildren, getValues, getNewSelected } from '../Select/utils/index.js';
import { useKeyNavigation } from '../Select/hooks/useKeyNavigation.js';
import { useForceUpdate } from '../../hooks/useForceUpdate.js';
import { useDidMountEffect } from '../../hooks/useDidMountEffect.js';
import { base } from './variations/_view/base.js';
import { base as base$1 } from './variations/_size/base.js';
import '@linaria/react';
import '@linaria/core';
import 'focus-visible';
import 'react-dom';
import 'react-popper';
import 'styled-components';
import { ComboboxTarget } from './ui/ComboboxTarget/ComboboxTarget.js';
import { StyledRoot, StyledNativeSelect, StyledPopover, StyledCombobox, ComboboxItemNotFound } from './Combobox.styles.js';

var _excluded = ["name", "form", "readOnly", "disabled", "label", "placeholder", "id", "children", "role", "view", "size", "frame", "usePortal", "negativeResultContent", "enumerationType", "opened", "placement", "onToggle", "onKeyDown", "filterFunction"];
/**
 * Выпадающий список без внешнего контроля видимости.
 */
var comboboxRoot = function comboboxRoot(Root) {
  return /*#__PURE__*/forwardRef(function (_ref, outerRootRef) {
    var _targetRef$current;
    var name = _ref.name,
      form = _ref.form,
      readOnly = _ref.readOnly,
      disabled = _ref.disabled,
      label = _ref.label,
      placeholder = _ref.placeholder,
      id = _ref.id,
      children = _ref.children,
      role = _ref.role,
      view = _ref.view,
      size = _ref.size,
      frame = _ref.frame,
      usePortal = _ref.usePortal,
      _ref$negativeResultCo = _ref.negativeResultContent,
      negativeResultContent = _ref$negativeResultCo === void 0 ? 'Элементы не найдены' : _ref$negativeResultCo,
      _ref$enumerationType = _ref.enumerationType,
      enumerationType = _ref$enumerationType === void 0 ? 'comma' : _ref$enumerationType,
      _ref$opened = _ref.opened,
      opened = _ref$opened === void 0 ? false : _ref$opened,
      _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? 'bottom' : _ref$placement,
      onToggle = _ref.onToggle,
      onKeyDown = _ref.onKeyDown,
      filterFunction = _ref.filterFunction,
      rest = _objectWithoutProperties(_ref, _excluded);
    var forceUpdate = useForceUpdate();
    var uniqId = safeUseId();
    var innerId = id || uniqId;
    var _rest$valueType = rest.valueType,
      valueType = _rest$valueType === void 0 ? 'single' : _rest$valueType,
      value = rest.value;
    var offset = [0, 4];
    var _useState = useState(null),
      _useState2 = _slicedToArray(_useState, 2),
      ref = _useState2[0],
      setRef = _useState2[1];
    var targetRef = useRef(null);
    var chipsRefs = useRef([]);
    var comboboxRef = useRef(null);
    var itemsRefs = useRef([]);
    var inputRef = useRef(null);
    var handleRef = useForkRef(targetRef, outerRootRef);
    var controlledRefs = {
      targetRef: targetRef,
      chipsRefs: chipsRefs,
      selectRef: comboboxRef,
      itemsRefs: itemsRefs,
      inputRef: inputRef
    };
    var _useState3 = useState(opened),
      _useState4 = _slicedToArray(_useState3, 2),
      innerOpened = _useState4[0],
      setInnerOpened = _useState4[1];
    var _useState5 = useState(''),
      _useState6 = _slicedToArray(_useState5, 2),
      search = _useState6[0],
      setSearch = _useState6[1];
    var _useState7 = useState(search),
      _useState8 = _slicedToArray(_useState7, 2),
      filterValue = _useState8[0],
      setFilterValue = _useState8[1];

    // INFO: нужно высчитывать для корректной работы портала
    var dynamicTargetWidth = (_targetRef$current = targetRef.current) === null || _targetRef$current === void 0 ? void 0 : _targetRef$current.offsetWidth;

    // INFO: Из-за того, что классы передаются через ref,
    // состояние пропсов после изменения view, size
    // всегда предыдущее, поэтому нужно делать форс-ререндер
    useDidMountEffect(function () {
      forceUpdate();
    }, [view, size]);
    useEffect(function () {
      if (disabled || readOnly) {
        return;
      }
      setInnerOpened(opened);
    }, [opened, disabled, readOnly]);
    useEffect(function () {
      // INFO: Для кейсов, когда значение выбрано и нужно вывести весь список
      if (opened) {
        setFilterValue('');
      }
    }, []);
    var onInnerToggle = function onInnerToggle(openValue, event) {
      if (disabled || readOnly) {
        return;
      }

      // INFO: Для кейсов, когда значение выбрано и нужно вывести весь список
      if (!innerOpened) {
        setFilterValue('');
      }
      if (onToggle) {
        onToggle(openValue, event);
        return;
      }
      setInnerOpened(openValue);
    };
    var closedWithoutChanges = useRef(true);
    var updateValue = function updateValue(item, event) {
      var _rest$onChangeValue2;
      var newValue = item.dataset.value;
      if (rest.valueType === 'multiple') {
        var _rest$onChangeValue;
        var _newSelected = getNewSelected(rest.value, newValue, rest.valueType);
        (_rest$onChangeValue = rest.onChangeValue) === null || _rest$onChangeValue === void 0 || _rest$onChangeValue.call(rest, _newSelected);
        return;
      }
      var newSelected = getNewSelected(rest.value, newValue, 'single');
      (_rest$onChangeValue2 = rest.onChangeValue) === null || _rest$onChangeValue2 === void 0 || _rest$onChangeValue2.call(rest, newSelected);
      onInnerToggle === null || onInnerToggle === void 0 || onInnerToggle(false, event);
      closedWithoutChanges.current = false;
    };
    var onClickChildrenItem = function onClickChildrenItem(event) {
      // INFO: Для корректной работы компонента Popper
      event.stopPropagation();
      var item = event.currentTarget;
      updateValue(item, event);
    };
    var onChangeChipValue = function onChangeChipValue(newValue) {
      var _rest$onChangeValue3;
      if (rest.valueType !== 'multiple') {
        return;
      }
      (_rest$onChangeValue3 = rest.onChangeValue) === null || _rest$onChangeValue3 === void 0 || _rest$onChangeValue3.call(rest, newValue);
    };
    var _useKeyNavigation = useKeyNavigation({
        controlledRefs: controlledRefs,
        opened: innerOpened,
        enumerationType: enumerationType,
        valueType: valueType,
        componentType: 'combobox',
        value: value,
        search: filterValue,
        updateOpened: onInnerToggle,
        updateValue: updateValue
      }),
      onKeyDownTarget = _useKeyNavigation.onKeyDownTarget,
      onKeyDownSelect = _useKeyNavigation.onKeyDownSelect;
    var childrenArray = Children.toArray(children);
    var childrenMemo = getChildren(childrenArray, {
      onClick: onClickChildrenItem,
      childrenRefs: itemsRefs
    }, value, filterValue, filterFunction).filter(Boolean);
    var values = getValues(childrenArray, value);
    var onSearch = function onSearch(newSearch, newOpened) {
      setSearch(newSearch);
      setFilterValue(newSearch);
      setInnerOpened(newOpened !== null && newOpened !== void 0 ? newOpened : true);
    };
    var onKeyDownHandle = function onKeyDownHandle(event) {
      onKeyDown === null || onKeyDown === void 0 || onKeyDown(event);
      onKeyDownTarget(event);
    };
    return /*#__PURE__*/React.createElement(StyledRoot, {
      ref: comboboxRef
    }, /*#__PURE__*/React.createElement(StyledNativeSelect, {
      "aria-invalid": "false",
      "aria-hidden": "true",
      "aria-readonly": readOnly,
      tabIndex: -1,
      form: form,
      name: name,
      value: "".concat(value),
      disabled: disabled || readOnly
    }), /*#__PURE__*/React.createElement(StyledPopover, {
      role: role,
      isOpen: innerOpened,
      onToggle: onInnerToggle,
      target: /*#__PURE__*/React.createElement(ComboboxTarget, _extends({
        opened: innerOpened,
        values: values,
        label: label,
        placeholder: placeholder,
        search: search,
        enumerationType: enumerationType,
        valueType: valueType,
        readOnly: readOnly,
        disabled: disabled,
        size: size,
        closedWithoutChanges: closedWithoutChanges,
        ref: handleRef,
        chipsRefs: chipsRefs,
        inputRef: inputRef,
        id: innerId,
        onChange: onChangeChipValue,
        onSearch: onSearch,
        onKeyDown: onKeyDownHandle
      }, rest)),
      className: cx(ref === null || ref === void 0 ? void 0 : ref.classList.toString()) // INFO: Прокидываем стили для Popover
      ,
      id: innerId,
      offset: offset,
      placement: getPlacements(placement),
      usePortal: usePortal,
      selectWidth: dynamicTargetWidth,
      frame: frame,
      trigger: "click",
      isFocusTrapped: false,
      preventOverflow: false,
      closeOnEsc: false,
      closeOnOverlayClick: true
    }, /*#__PURE__*/React.createElement(Root, {
      ref: setRef,
      view: view,
      size: size
    }, /*#__PURE__*/React.createElement(StyledCombobox, {
      onKeyDown: onKeyDownSelect
    }, childrenMemo.length ? childrenMemo : /*#__PURE__*/React.createElement(ComboboxItemNotFound, null, negativeResultContent)))));
  });
};
var comboboxConfig = {
  name: 'Combobox',
  tag: 'div',
  layout: comboboxRoot,
  base: '',
  variations: {
    view: {
      css: base
    },
    size: {
      css: base$1
    }
  },
  defaults: {
    view: 'default',
    size: 'm'
  }
};

export { comboboxConfig, comboboxRoot };
