// @salutejs/plasma-new-hope/es/components/Combobox/Combobox.styles.js
import './Combobox.styles_dw6hnk.css';
import { styled } from '@linaria/react';
import '@linaria/core';
import { component } from '../../engines/common.js';
import { popoverConfig } from '../Popover/Popover.js';

var StyledRoot = /*#__PURE__*/styled('div')({
  name: "StyledRoot",
  "class": "sd7salz",
  propsAsIs: false
});
var StyledCombobox = /*#__PURE__*/styled('div')({
  name: "StyledCombobox",
  "class": "s1tp4xgb",
  propsAsIs: false
});

// issue #823
var Popover = /*#__PURE__*/component(popoverConfig);
var _exp11 = function _exp11() {
  return Popover;
};
var _exp14 = function _exp14() {
  return function (_ref) {
    var usePortal = _ref.usePortal,
      selectWidth = _ref.selectWidth;
    return usePortal ? "".concat(selectWidth, "px") : '100%';
  };
};
var StyledPopover = /*#__PURE__*/styled(_exp11())({
  name: "StyledPopover",
  "class": "s1fuzbs4",
  propsAsIs: true,
  vars: {
    "s1fuzbs4-0": [/*#__PURE__*/_exp14()]
  }
});
var ComboboxItemNotFound = /*#__PURE__*/styled('div')({
  name: "ComboboxItemNotFound",
  "class": "c1bkz6yw",
  propsAsIs: false
});
var StyledNativeSelect = /*#__PURE__*/styled('select')({
  name: "StyledNativeSelect",
  "class": "s1mwm6am",
  propsAsIs: false
});

export { ComboboxItemNotFound, StyledCombobox, StyledNativeSelect, StyledPopover, StyledRoot };
