// @salutejs/plasma-new-hope/styled-components/es/examples/plasma_b2c/components/Indicator/Indicator.config.js
import { css } from 'styled-components';
import { indicatorTokens as tokens } from '../../../../components/Indicator';
export var config = {
  defaults: {
    size: 's',
    view: 'default'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/css(["", ":var(--surface-solid-default);"], tokens.color),
      accent: /*#__PURE__*/css(["", ":var(--surface-accent);"], tokens.color),
      inactive: /*#__PURE__*/css(["", ":var(--surface-solid-tertiary);"], tokens.color),
      positive: /*#__PURE__*/css(["", ":var(--surface-positive);"], tokens.color),
      warning: /*#__PURE__*/css(["", ":var(--surface-warning);"], tokens.color),
      negative: /*#__PURE__*/css(["", ":var(--surface-negative);"], tokens.color),
      black: /*#__PURE__*/css(["", ":var(--on-light-surface-solid-default);"], tokens.color),
      white: /*#__PURE__*/css(["", ":var(--on-dark-surface-solid-default);"], tokens.color)
    },
    size: {
      l: /*#__PURE__*/css(["", ":0.75rem;"], tokens.size),
      m: /*#__PURE__*/css(["", ":0.5rem;"], tokens.size),
      s: /*#__PURE__*/css(["", ":0.375rem;"], tokens.size)
    }
  }
};
