// @salutejs/plasma-new-hope/es/components/Calendar/ui/CalendarHeader/CalendarHeader.styles.js
import './CalendarHeader.styles_1c4p417.css';
import { styled } from '@linaria/react';
import { buttonFocus } from '../../mixins/index.js';

var StyledCalendarHeader = /*#__PURE__*/styled('div')({
  name: "StyledCalendarHeader",
  "class": "sjaqipy",
  propsAsIs: false
});
var _exp7 = function _exp7() {
  return buttonFocus;
};
var StyledHeader = /*#__PURE__*/styled(_exp7())({
  name: "StyledHeader",
  "class": "s1k7jebr",
  propsAsIs: true
});
var StyledHeaderDouble = /*#__PURE__*/styled('h4')({
  name: "StyledHeaderDouble",
  "class": "s1b07tlr",
  propsAsIs: false
});
var StyledArrows = /*#__PURE__*/styled('div')({
  name: "StyledArrows",
  "class": "s13yvda3",
  propsAsIs: false
});
var _exp23 = function _exp23() {
  return buttonFocus;
};
var StyledArrow = /*#__PURE__*/styled(_exp23())({
  name: "StyledArrow",
  "class": "s1i8jkj2",
  propsAsIs: true
});
var StyledNavigation = /*#__PURE__*/styled('div')({
  name: "StyledNavigation",
  "class": "sw3by8z",
  propsAsIs: false
});

export { StyledArrow, StyledArrows, StyledCalendarHeader, StyledHeader, StyledHeaderDouble, StyledNavigation };
