// @salutejs/plasma-new-hope/styled-components/cjs/components/Dropdown/utils/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getValidComponent = exports.getPlacements = exports.getPlacement = exports.getCorrectHeight = exports.childrenWithProps = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
var getPlacement = exports.getPlacement = function getPlacement(placement) {
  return "".concat(placement, "-start");
};
var getPlacements = exports.getPlacements = function getPlacements(placements) {
  if (!placements) {
    return;
  }
  var isArray = Array.isArray(placements);
  if (!isArray) {
    return getPlacement(placements);
  }
  return (placements || []).map(function (placement) {
    return getPlacement(placement);
  });
};
var getCorrectHeight = exports.getCorrectHeight = function getCorrectHeight(listHeight) {
  if (Number.isNaN(Number(listHeight))) {
    return listHeight.toString();
  }
  return "".concat(listHeight, "rem");
};
var childrenWithProps = exports.childrenWithProps = function childrenWithProps(children, additionalProps) {
  return _react.Children.map(children, function (child) {
    if ( /*#__PURE__*/_react["default"].isValidElement(child)) {
      return /*#__PURE__*/_react["default"].cloneElement(child, additionalProps);
    }
    return child;
  });
};
var isReactObject = function isReactObject(element) {
  return _typeof(element) === 'object' || typeof element === 'function';
};
var getValidComponent = exports.getValidComponent = function getValidComponent(element, props) {
  if ( /*#__PURE__*/(0, _react.isValidElement)(element)) {
    return /*#__PURE__*/(0, _react.cloneElement)(element, props);
  }
  if (isReactObject(element)) {
    var Component = element;
    return /*#__PURE__*/_react["default"].createElement(Component, props);
  }
  return element;
};
