// @salutejs/plasma-ui/hooks/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "useDebouncedFunction", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.useDebouncedFunction;
  }
});
Object.defineProperty(exports, "useDeviceKind", {
  enumerable: true,
  get: function get() {
    return _useDeviceKind.useDeviceKind;
  }
});
Object.defineProperty(exports, "useForkRef", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.useForkRef;
  }
});
Object.defineProperty(exports, "useRemoteListener", {
  enumerable: true,
  get: function get() {
    return _useRemoteListener.useRemoteListener;
  }
});
Object.defineProperty(exports, "useResizeObserver", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.useResizeObserver;
  }
});
Object.defineProperty(exports, "useThemeContext", {
  enumerable: true,
  get: function get() {
    return _useThemeContext.useThemeContext;
  }
});
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _useRemoteListener = /*#__PURE__*/require("./useRemoteListener");
var _useThemeContext = /*#__PURE__*/require("./useThemeContext");
var _useDeviceKind = /*#__PURE__*/require("./useDeviceKind");
