// @salutejs/plasma-new-hope/es/components/TextArea/TextArea.styles.js
import './TextArea.styles_1bv40vw.css';
import { styled } from '@linaria/react';
import { tokens } from './TextArea.tokens.js';

var StyledLabel = /*#__PURE__*/styled('div')({
  name: "StyledLabel",
  "class": "s1luq7vd",
  propsAsIs: false
});
var _exp10 = function _exp10() {
  return function (_ref) {
    var hasHelper = _ref.hasHelper;
    return hasHelper ? "var(".concat(tokens.inputPaddingBottomWithHelpers, ")") : "var(".concat(tokens.inputPaddingBottom, ")");
  };
};
var _exp11 = function _exp11() {
  return function (_ref2) {
    var hasHelper = _ref2.hasHelper;
    return hasHelper ? "var(".concat(tokens.borderRadiusWithHelpers, ")") : "var(".concat(tokens.borderRadius, ")");
  };
};
var StyledTextAreaWrapper = /*#__PURE__*/styled('div')({
  name: "StyledTextAreaWrapper",
  "class": "s1r1fvk8",
  propsAsIs: false,
  vars: {
    "s1r1fvk8-0": [/*#__PURE__*/_exp10()],
    "s1r1fvk8-1": [/*#__PURE__*/_exp11()]
  }
});
var StyledContent = /*#__PURE__*/styled('div')({
  name: "StyledContent",
  "class": "surmvwq",
  propsAsIs: false
});
var _exp15 = function _exp15() {
  return function (_ref3) {
    var width = _ref3.width;
    return width;
  };
};
var StyledContainer = /*#__PURE__*/styled('div')({
  name: "StyledContainer",
  "class": "s18bd513",
  propsAsIs: false,
  vars: {
    "s18bd513-0": [/*#__PURE__*/_exp15()]
  }
});
var _exp18 = function _exp18() {
  return function (_ref4) {
    var resize = _ref4.resize;
    return resize || 'none';
  };
};
var _exp19 = function _exp19() {
  return function (_ref5) {
    var height = _ref5.height;
    return !Number.isNaN(Number(height)) ? "calc(".concat(height, "rem - 0.875rem)") : height || "var(".concat(tokens.inputHeight, ")");
  };
};
var _exp20 = function _exp20() {
  return function (_ref6) {
    var width = _ref6.width;
    return !Number.isNaN(Number(width)) ? "".concat(width, "rem") : width || "var(".concat(tokens.inputWidth, ")");
  };
};
var _exp21 = function _exp21() {
  return function (_ref7) {
    var rows = _ref7.rows;
    return rows ? 'unset' : 'var(--plasma_private-textarea-height)';
  };
};
var _exp22 = function _exp22() {
  return function (_ref8) {
    var cols = _ref8.cols;
    return cols ? 'unset' : 'var(--plasma_private-textarea-width)';
  };
};
var _exp24 = function _exp24() {
  return function (_ref9) {
    var applyCustomWidth = _ref9.applyCustomWidth;
    return applyCustomWidth ? 'var(--plasma_private-textarea-width)' : 'auto';
  };
};
var _exp26 = function _exp26() {
  return function (_ref10) {
    var hasContentRight = _ref10.hasContentRight;
    return hasContentRight ? "var(".concat(tokens.inputPaddingRightWithRightContent, ")") : "var(".concat(tokens.inputPaddingRight, ")");
  };
};
var _exp33 = function _exp33() {
  return function (_ref11) {
    var hasHelper = _ref11.hasHelper,
      rows = _ref11.rows;
    return hasHelper && !rows ? 'var(--plasma_private-textarea-input-with-helpers-height)' : 'var(--plasma_private-textarea-input-without-helpers-height)';
  };
};
var StyledTextArea = /*#__PURE__*/styled('textarea')({
  name: "StyledTextArea",
  "class": "s7alcr2",
  propsAsIs: false,
  vars: {
    "s7alcr2-0": [/*#__PURE__*/_exp18()],
    "s7alcr2-1": [/*#__PURE__*/_exp19()],
    "s7alcr2-2": [/*#__PURE__*/_exp20()],
    "s7alcr2-3": [/*#__PURE__*/_exp21()],
    "s7alcr2-4": [/*#__PURE__*/_exp22()],
    "s7alcr2-5": [/*#__PURE__*/_exp24()],
    "s7alcr2-6": [/*#__PURE__*/_exp26()],
    "s7alcr2-7": [/*#__PURE__*/_exp33()]
  }
});
var StyledHelpers = /*#__PURE__*/styled('div')({
  name: "StyledHelpers",
  "class": "si9oz50",
  propsAsIs: false
});
var StyledLeftHelper = /*#__PURE__*/styled('span')({
  name: "StyledLeftHelper",
  "class": "s18tpn65",
  propsAsIs: false
});
var _exp56 = function _exp56() {
  return StyledLeftHelper;
};
var StyledRightHelper = /*#__PURE__*/styled(_exp56())({
  name: "StyledRightHelper",
  "class": "s19sfsrt",
  propsAsIs: true
});
var _exp61 = function _exp61() {
  return function (_ref12) {
    var hasContentRight = _ref12.hasContentRight;
    return hasContentRight ? "var(".concat(tokens.inputPaddingRightWithRightContent, ")") : "var(".concat(tokens.inputPaddingRight, ")");
  };
};
var StyledPlaceholder = /*#__PURE__*/styled('label')({
  name: "StyledPlaceholder",
  "class": "sfype23",
  propsAsIs: false,
  vars: {
    "sfype23-0": [/*#__PURE__*/_exp61()]
  }
});

export { StyledContainer, StyledContent, StyledHelpers, StyledLabel, StyledLeftHelper, StyledPlaceholder, StyledRightHelper, StyledTextArea, StyledTextAreaWrapper };
