// @salutejs/plasma-new-hope/cjs/components/Select/ui/SelectFooter/SelectFooter.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('react');
require('../../../../utils/react.js');
require('@salutejs/plasma-core');
require('@linaria/react');
var DropdownFooter = require('../../../Dropdown/ui/DropdownFooter/DropdownFooter.js');
require('@linaria/core');
require('focus-visible');
require('react-dom');
require('react-popper');
require('styled-components');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var SelectFooter_styles = require('./SelectFooter.styles.js');

var selectFooterRoot = DropdownFooter.dropdownFooterRoot;
var selectFooterConfig = {
  name: 'SelectFooter',
  tag: 'div',
  layout: selectFooterRoot,
  base: SelectFooter_styles.base,
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

exports.selectFooterConfig = selectFooterConfig;
exports.selectFooterRoot = selectFooterRoot;
