// @salutejs/plasma-new-hope/es/components/Header/ui/HeaderLogo/HeaderLogo.js
import { extends as _extends } from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef } from 'react';
import '@linaria/core';
import { component } from '../../../../engines/common.js';
import { imageConfig } from '../../../Image/Image.js';
import { classes } from '../../Header.tokens.js';
import { base } from './variations/_size/base.js';

// INFO: Omit 'onResize' | 'onResizeCapture' | 'nonce'
// because this types coming with @types/react@18.0.25 and breaks react@17.0.2 with @types/react@18.0.18

// issue #823
var Image = /*#__PURE__*/component(imageConfig);

/**
 * Компонент для размещения логотипа.
 */
var headerLogoRoot = function headerLogoRoot(Root) {
  return /*#__PURE__*/forwardRef(function (props, outerRootRef) {
    return /*#__PURE__*/React.createElement(Root, {
      ref: outerRootRef
    }, /*#__PURE__*/React.createElement(Image, _extends({
      className: classes.headerLogoImage
    }, props)));
  });
};
var headerLogoConfig = {
  name: 'HeaderLogo',
  tag: 'div',
  layout: headerLogoRoot,
  base: '',
  variations: {
    size: {
      css: base
    }
  },
  defaults: {
    size: 'm'
  }
};

export { headerLogoConfig, headerLogoRoot };
