// @salutejs/plasma-new-hope/cjs/components/Panel/ui/PanelContent/PanelContent.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var base = require('./variations/_view/base.js');

var _excluded = ["children", "view"];
var panelContentRoot = function panelContentRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (_ref, outerRef) {
    var children = _ref.children,
      view = _ref.view,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
    return /*#__PURE__*/React.createElement(Root, _rollupPluginBabelHelpers.extends({
      ref: outerRef,
      view: view
    }, rest), children);
  });
};
var panelContentConfig = {
  name: 'PanelContent',
  tag: 'div',
  layout: panelContentRoot,
  base: '',
  variations: {
    view: {
      css: base.base
    }
  },
  defaults: {
    view: 'default'
  }
};

exports.panelContentConfig = panelContentConfig;
exports.panelContentRoot = panelContentRoot;
