// @salutejs/plasma-hope/components/UploadAudio/UploadAudio.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UploadAudio = exports.StyledText = exports.StyledIconMusic = exports.StyledContent = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _Upload = /*#__PURE__*/require("../Upload");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _StyledIconMusic;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _objectDestructuringEmpty(obj) { if (obj == null) throw new TypeError("Cannot destructure " + obj); }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
(0, _collectPackageInfo.collectPackageInfo)('UploadAudio/UploadAudio');
var StyledContent = exports.StyledContent = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web-core__sc-va5v03-0"
})(["display:flex;align-items:center;box-sizing:border-box;"]);
var StyledText = exports.StyledText = /*#__PURE__*/_styledComponents["default"].span.withConfig({
  componentId: "plasma-web-core__sc-va5v03-1"
})(["margin-left:0.5rem;"]);
var StyledIconMusic = exports.StyledIconMusic = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaIcons.IconMusic).withConfig({
  componentId: "plasma-web-core__sc-va5v03-2"
})(["svg{opacity:0.28;}"]);

/**
 * Компонент для загрузки аудио файлов.
 */
var UploadAudio = exports.UploadAudio = function UploadAudio(_ref) {
  var rest = _extends({}, (_objectDestructuringEmpty(_ref), _ref));
  var acceptExtensions = '.mp3,.wav';
  var text = 'Перетащите трек в формате mp3';
  return /*#__PURE__*/_react["default"].createElement(_Upload.Upload, _extends({
    accept: acceptExtensions,
    content: /*#__PURE__*/_react["default"].createElement(StyledContent, null, _StyledIconMusic || (_StyledIconMusic = /*#__PURE__*/_react["default"].createElement(StyledIconMusic, {
      size: "s",
      color: "inherit"
    })), /*#__PURE__*/_react["default"].createElement(StyledText, null, text))
  }, rest));
};
