// @salutejs/plasma-web/css/cjs/components/Typography/Typography.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var plasmaNewHope = require('@salutejs/plasma-new-hope');
var Body_config = require('./Body.config.js');
var Dspl_config = require('./Dspl.config.js');
var Heading_config = require('./Heading.config.js');
var Text_config = require('./Text.config.js');

var bodyMConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.bodyConfig, Body_config.configM);
var BodyMComponent = /*#__PURE__*/plasmaNewHope.component(bodyMConfig);
var bodyLConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.bodyConfig, Body_config.configL);
var BodyLComponent = /*#__PURE__*/plasmaNewHope.component(bodyLConfig);
var bodySConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.bodyConfig, Body_config.configS);
var BodySComponent = /*#__PURE__*/plasmaNewHope.component(bodySConfig);
var bodyXSConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.bodyConfig, Body_config.configXS);
var BodyXSComponent = /*#__PURE__*/plasmaNewHope.component(bodyXSConfig);
var bodyXXSConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.bodyConfig, Body_config.configXXS);
var BodyXXSComponent = /*#__PURE__*/plasmaNewHope.component(bodyXXSConfig);
var dsplMConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.dsplConfig, Dspl_config.configM);
var DsplMComponent = /*#__PURE__*/plasmaNewHope.component(dsplMConfig);
var dsplLConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.dsplConfig, Dspl_config.configL);
var DsplLComponent = /*#__PURE__*/plasmaNewHope.component(dsplLConfig);
var dsplSConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.dsplConfig, Dspl_config.configS);
var DsplSComponent = /*#__PURE__*/plasmaNewHope.component(dsplSConfig);
var heading1Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.headingConfig, Heading_config.configH1);
var Heading1Component = /*#__PURE__*/plasmaNewHope.component(heading1Config);
var heading2Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.headingConfig, Heading_config.configH2);
var Heading2Component = /*#__PURE__*/plasmaNewHope.component(heading2Config);
var heading3Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.headingConfig, Heading_config.configH3);
var Heading3Component = /*#__PURE__*/plasmaNewHope.component(heading3Config);
var heading4Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.headingConfig, Heading_config.configH4);
var Heading4Component = /*#__PURE__*/plasmaNewHope.component(heading4Config);
var heading5Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.headingConfig, Heading_config.configH5);
var Heading5Component = /*#__PURE__*/plasmaNewHope.component(heading5Config);
var textMConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.textConfig, Text_config.configM);
var TextMComponent = /*#__PURE__*/plasmaNewHope.component(textMConfig);
var textLConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.textConfig, Text_config.configL);
var TextLComponent = /*#__PURE__*/plasmaNewHope.component(textLConfig);
var textSConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.textConfig, Text_config.configS);
var TextSComponent = /*#__PURE__*/plasmaNewHope.component(textSConfig);
var textXSConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.textConfig, Text_config.configXS);
var TextXSComponent = /*#__PURE__*/plasmaNewHope.component(textXSConfig);
var BodyM = BodyMComponent;
var BodyL = BodyLComponent;
var BodyS = BodySComponent;
var BodyXS = BodyXSComponent;
var BodyXXS = BodyXXSComponent;
var DsplM = DsplMComponent;
var DsplL = DsplLComponent;
var DsplS = DsplSComponent;
var H1 = Heading1Component;
var H2 = Heading2Component;
var H3 = Heading3Component;
var H4 = Heading4Component;
var H5 = Heading5Component;
var TextM = TextMComponent;
var TextL = TextLComponent;
var TextS = TextSComponent;
var TextXS = TextXSComponent;

exports.BodyL = BodyL;
exports.BodyM = BodyM;
exports.BodyS = BodyS;
exports.BodyXS = BodyXS;
exports.BodyXXS = BodyXXS;
exports.DsplL = DsplL;
exports.DsplM = DsplM;
exports.DsplS = DsplS;
exports.H1 = H1;
exports.H2 = H2;
exports.H3 = H3;
exports.H4 = H4;
exports.H5 = H5;
exports.TextL = TextL;
exports.TextM = TextM;
exports.TextS = TextS;
exports.TextXS = TextXS;
