// @salutejs/plasma-new-hope/cjs/components/Image/Image.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var Image_styles = require('./Image.styles.js');
var Image_utils = require('./Image.utils.js');

var _excluded = ["src", "srcSet", "sizes", "ratio", "customRatio", "alt", "width", "height", "base", "style"];

/**
 * Компонент для отображения картинок.
 */
var imageRoot = function imageRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (_ref, outerRootRef) {
    var src = _ref.src,
      srcSet = _ref.srcSet,
      sizes = _ref.sizes,
      ratio = _ref.ratio,
      customRatio = _ref.customRatio,
      alt = _ref.alt,
      width = _ref.width,
      height = _ref.height,
      _ref$base = _ref.base,
      base = _ref$base === void 0 ? 'img' : _ref$base,
      style = _ref.style,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
    var widthToken = {
      '--plasma_private-image-width': width || '100%'
    };
    var heightToken = {
      '--plasma_private-image-height': height || '0'
    };
    var rootCustomRatio = customRatio ? "".concat(customRatio, "%") : 'unset';
    var ratioToken = {
      '--plasma_private-image-pading-bottom': ratio ? "".concat(Image_utils.ratios[ratio], "%") : rootCustomRatio
    };
    return /*#__PURE__*/React.createElement(Root, _rollupPluginBabelHelpers.extends({
      ref: outerRootRef,
      style: _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, style), widthToken), heightToken), ratioToken)
    }, rest), base === 'img' && /*#__PURE__*/React.createElement(Image_styles.StyledImg, {
      src: src,
      srcSet: srcSet,
      sizes: sizes,
      alt: alt
    }), base === 'div' && /*#__PURE__*/React.createElement(Image_styles.StyledDivImg, {
      style: {
        backgroundImage: "url(".concat(src, ")")
      },
      role: "img",
      "aria-label": alt
    }));
  });
};
var imageConfig = {
  name: 'Image',
  tag: 'div',
  layout: imageRoot,
  base: Image_styles.base,
  variations: {},
  defaults: {}
};

exports.imageConfig = imageConfig;
exports.imageRoot = imageRoot;
