// @salutejs/plasma-new-hope/styled-components/cjs/components/Radiobox/variations/_focused/base.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.base = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _mixins = /*#__PURE__*/require("../../../../mixins");
var _Radiobox = /*#__PURE__*/require("../../Radiobox.tokens");
var base = exports.base = /*#__PURE__*/(0, _styledComponents.css)([".", "{", "}input[data-focus-visible-added] + label .", "{border-color:transparent;", "}input[data-focus-visible-added]:checked + label .", "{", "}"], _Radiobox.classes.radioboxTrigger, /*#__PURE__*/(0, _mixins.addFocus)({
  outlineOffset: '0',
  outlineSize: '0.125rem',
  outlineRadius: /*#__PURE__*/"var(".concat(_Radiobox.tokens.triggerBorderRadius, ")"),
  outlineColor: 'transparent'
}), _Radiobox.classes.radioboxTrigger, /*#__PURE__*/(0, _mixins.addFocus)({
  outlineOffset: '-0.125rem',
  outlineSize: '0.125rem',
  outlineRadius: /*#__PURE__*/"var(".concat(_Radiobox.tokens.triggerBorderRadius, ")"),
  outlineColor: /*#__PURE__*/"var(".concat(_Radiobox.tokens.focusColor, ")"),
  customFocusRules: /*#__PURE__*/"\n                &::before {\n                    box-shadow: 0 0 0 0.125rem var(".concat(_Radiobox.tokens.focusColor, ");\n                }\n            ")
}), _Radiobox.classes.radioboxTrigger, /*#__PURE__*/(0, _mixins.addFocus)({
  outlineOffset: '-0.125rem',
  outlineSize: '0.063rem',
  outlineRadius: /*#__PURE__*/"calc(var(".concat(_Radiobox.tokens.triggerBorderRadius, ") + 0.125rem)"),
  outlineColor: /*#__PURE__*/"var(".concat(_Radiobox.tokens.focusColor, ")"),
  customFocusRules: /*#__PURE__*/"\n                &::before {\n                    box-shadow: 0 0 0 0.063rem var(".concat(_Radiobox.tokens.focusColor, ");\n                }\n            ")
}));
