// @salutejs/plasma-new-hope/styled-components/cjs/examples/typograpy/components/Old/Body/Body.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _tokens = /*#__PURE__*/require("../../../../../components/Typography/tokens");
var config = exports.config = {
  defaults: {
    size: 'body1'
  },
  variations: {
    size: {
      body1: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--plasma-typo-body1-font-family);", ":var(--plasma-typo-body1-font-size);", ":var(--plasma-typo-body1-font-style);", ":var(--plasma-typo-body1-font-weight);", ":var(--plasma-typo-body1-letter-spacing);", ":var(--plasma-typo-body1-line-height);"], _tokens.tokens.typoFontFamily, _tokens.tokens.typoFontSize, _tokens.tokens.typoFontStyle, _tokens.tokens.typoFontWeight, _tokens.tokens.typoFontLetterSpacing, _tokens.tokens.typoFontLineHeight),
      body2: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--plasma-typo-body2-font-family);", ":var(--plasma-typo-body2-font-size);", ":var(--plasma-typo-body2-font-style);", ":var(--plasma-typo-body2-font-weight);", ":var(--plasma-typo-body2-letter-spacing);", ":var(--plasma-typo-body2-line-height);"], _tokens.tokens.typoFontFamily, _tokens.tokens.typoFontSize, _tokens.tokens.typoFontStyle, _tokens.tokens.typoFontWeight, _tokens.tokens.typoFontLetterSpacing, _tokens.tokens.typoFontLineHeight),
      body3: /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--plasma-typo-body3-font-family);", ":var(--plasma-typo-body3-font-size);", ":var(--plasma-typo-body3-font-style);", ":var(--plasma-typo-body3-font-weight);", ":var(--plasma-typo-body3-letter-spacing);", ":var(--plasma-typo-body3-line-height);"], _tokens.tokens.typoFontFamily, _tokens.tokens.typoFontSize, _tokens.tokens.typoFontStyle, _tokens.tokens.typoFontWeight, _tokens.tokens.typoFontLetterSpacing, _tokens.tokens.typoFontLineHeight)
    }
  }
};
