// @salutejs/plasma-ui/es/components/Header/Header.js
var _excluded = ["children"],
  _excluded2 = ["minimize", "back", "logo", "logoAlt", "title", "subtitle", "hideSubtitleOnMobile", "onMinimizeClick", "onBackClick"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React, { forwardRef } from 'react';
import { HeaderRoot } from './HeaderRoot';
import { HeaderLogo } from './HeaderLogo';
import { HeaderSubtitle } from './HeaderSubtitle';
import { HeaderTitle } from './HeaderTitle';
import { HeaderTitleWrapper } from './HeaderTitleWrapper';
import { HeaderContent } from './HeaderContent';
import { HeaderArrow } from './HeaderArrow';
import { collectPackageInfo } from '../../collectPackageInfo';
collectPackageInfo('Header/Header');
/**
 * Сборный компонент для отрисовки шапки страницы.
 * Уже включает в себя все составные части шапки.
 */
export var Header = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var children = _ref.children,
    props = _objectWithoutProperties(_ref, _excluded);
  var _ref2 = props,
    minimize = _ref2.minimize,
    back = _ref2.back,
    logo = _ref2.logo,
    logoAlt = _ref2.logoAlt,
    title = _ref2.title,
    subtitle = _ref2.subtitle,
    _ref2$hideSubtitleOnM = _ref2.hideSubtitleOnMobile,
    hideSubtitleOnMobile = _ref2$hideSubtitleOnM === void 0 ? true : _ref2$hideSubtitleOnM,
    onMinimizeClick = _ref2.onMinimizeClick,
    onBackClick = _ref2.onBackClick,
    rest = _objectWithoutProperties(_ref2, _excluded2);
  return /*#__PURE__*/React.createElement(HeaderRoot, _extends({
    ref: ref
  }, rest), (minimize || back) && /*#__PURE__*/React.createElement(HeaderArrow, {
    onClick: back ? onBackClick : onMinimizeClick,
    arrow: back ? 'back' : 'minimize'
  }), logo && /*#__PURE__*/React.createElement(HeaderLogo, {
    src: logo,
    alt: logoAlt
  }), title && /*#__PURE__*/React.createElement(HeaderTitleWrapper, null, /*#__PURE__*/React.createElement(HeaderTitle, null, title), subtitle && /*#__PURE__*/React.createElement(HeaderSubtitle, {
    hideOnMobile: hideSubtitleOnMobile
  }, subtitle)), children && /*#__PURE__*/React.createElement(HeaderContent, null, children));
});
