// @salutejs/plasma-web/es/components/Button/Button.perftest.js
var _IconPlaceholder, _IconPlaceholder2, _IconPlaceholder3, _IconPlaceholder4;
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React from 'react';
import { IconPlaceholder } from '@salutejs/plasma-sb-utils';
import { Button } from '.';
var defaultProps = {
  view: 'default',
  size: 'l',
  pin: 'square-square',
  disabled: false,
  outlined: true,
  focused: false,
  contentType: 'Text',
  isLoading: false,
  onClick: function onClick() {},
  onFocus: function onFocus() {},
  onBlur: function onBlur() {}
};
var text = 'Label';
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(Button, _extends({
    text: text,
    contentLeft: _IconPlaceholder || (_IconPlaceholder = /*#__PURE__*/React.createElement(IconPlaceholder, null)),
    contentRight: _IconPlaceholder2 || (_IconPlaceholder2 = /*#__PURE__*/React.createElement(IconPlaceholder, null))
  }, defaultProps));
};
export var Anchor = function Anchor() {
  return /*#__PURE__*/React.createElement(Button, _extends({
    as: "a",
    text: text,
    contentLeft: _IconPlaceholder3 || (_IconPlaceholder3 = /*#__PURE__*/React.createElement(IconPlaceholder, null)),
    contentRight: _IconPlaceholder4 || (_IconPlaceholder4 = /*#__PURE__*/React.createElement(IconPlaceholder, null))
  }, defaultProps));
};
