// @salutejs/plasma-ui/components/Header/Header.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Header = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _HeaderRoot = /*#__PURE__*/require("./HeaderRoot");
var _HeaderLogo = /*#__PURE__*/require("./HeaderLogo");
var _HeaderSubtitle = /*#__PURE__*/require("./HeaderSubtitle");
var _HeaderTitle = /*#__PURE__*/require("./HeaderTitle");
var _HeaderTitleWrapper = /*#__PURE__*/require("./HeaderTitleWrapper");
var _HeaderContent = /*#__PURE__*/require("./HeaderContent");
var _HeaderArrow = /*#__PURE__*/require("./HeaderArrow");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _excluded = ["children"],
  _excluded2 = ["minimize", "back", "logo", "logoAlt", "title", "subtitle", "hideSubtitleOnMobile", "onMinimizeClick", "onBackClick"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('Header/Header');
/**
 * Сборный компонент для отрисовки шапки страницы.
 * Уже включает в себя все составные части шапки.
 */
var Header = exports.Header = /*#__PURE__*/(0, _react.forwardRef)(function (_ref, ref) {
  var children = _ref.children,
    props = _objectWithoutProperties(_ref, _excluded);
  var _ref2 = props,
    minimize = _ref2.minimize,
    back = _ref2.back,
    logo = _ref2.logo,
    logoAlt = _ref2.logoAlt,
    title = _ref2.title,
    subtitle = _ref2.subtitle,
    _ref2$hideSubtitleOnM = _ref2.hideSubtitleOnMobile,
    hideSubtitleOnMobile = _ref2$hideSubtitleOnM === void 0 ? true : _ref2$hideSubtitleOnM,
    onMinimizeClick = _ref2.onMinimizeClick,
    onBackClick = _ref2.onBackClick,
    rest = _objectWithoutProperties(_ref2, _excluded2);
  return /*#__PURE__*/_react["default"].createElement(_HeaderRoot.HeaderRoot, _extends({
    ref: ref
  }, rest), (minimize || back) && /*#__PURE__*/_react["default"].createElement(_HeaderArrow.HeaderArrow, {
    onClick: back ? onBackClick : onMinimizeClick,
    arrow: back ? 'back' : 'minimize'
  }), logo && /*#__PURE__*/_react["default"].createElement(_HeaderLogo.HeaderLogo, {
    src: logo,
    alt: logoAlt
  }), title && /*#__PURE__*/_react["default"].createElement(_HeaderTitleWrapper.HeaderTitleWrapper, null, /*#__PURE__*/_react["default"].createElement(_HeaderTitle.HeaderTitle, null, title), subtitle && /*#__PURE__*/_react["default"].createElement(_HeaderSubtitle.HeaderSubtitle, {
    hideOnMobile: hideSubtitleOnMobile
  }, subtitle)), children && /*#__PURE__*/_react["default"].createElement(_HeaderContent.HeaderContent, null, children));
});
