// @salutejs/plasma-hope/components/Upload/UploadButton.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UploadButton = exports.StyledButton = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _reactFileDrop = /*#__PURE__*/require("react-file-drop");
var _utils = /*#__PURE__*/require("./utils");
var _excluded = ["status", "disabled", "isProgress", "accept", "content", "loader", "onChange", "onValidation", "validate"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
// TODO: https://github.com/salute-developers/plasma/issues/364

var statuses = {
  error: {
    borderColor: _plasmaCore.critical
  },
  success: {
    borderColor: _plasmaCore.success
  }
};
var paints = {
  "default": {
    backgroundColor: _plasmaCore.surfaceLiquid01,
    borderColor: _plasmaCore.tertiary,
    color: _plasmaCore.secondary
  },
  progress: {
    backgroundColor: _plasmaCore.surfaceLiquid03,
    borderColor: _plasmaCore.transparent,
    color: _plasmaCore.secondary
  }
};
var StyledButton = exports.StyledButton = /*#__PURE__*/_styledComponents["default"].button.withConfig({
  componentId: "plasma-web-core__sc-3ebw3y-0"
})(["", ";position:relative;display:flex;align-items:center;box-sizing:border-box;justify-content:center;width:100%;height:4rem;padding:1.25rem 1rem;appearance:none;border:1px dashed;border-radius:0.75rem;cursor:pointer;transition:all 0.3s ease-in-out;transition-property:transform,background-color,border-color,color;&:hover:not(:disabled){background-color:", ";transform:scale(1.04);}", " ", " ", " ", ""], _plasmaCore.footnote1, _plasmaCore.surfaceLiquid02, function (_ref) {
  var isGrabbing = _ref.isGrabbing;
  return isGrabbing && (0, _styledComponents.css)(["border-color:", ";background-color:", ";transform:scale(1.04);"], _plasmaCore.buttonAccent, _plasmaCore.surfaceLiquid02);
}, function (_ref2) {
  var paint = _ref2.paint;
  return paints[paint];
}, function (_ref3) {
  var status = _ref3.status;
  return status && statuses[status];
}, function (_ref4) {
  var disabled = _ref4.disabled;
  return disabled && (0, _styledComponents.css)(["opacity:0.4;cursor:not-allowed;"]);
});
var StyledInputLabel = /*#__PURE__*/_styledComponents["default"].label.withConfig({
  componentId: "plasma-web-core__sc-3ebw3y-1"
})(["position:absolute;width:100%;height:100%;top:0;left:0;cursor:pointer;", ""], function (_ref5) {
  var disabled = _ref5.disabled;
  return disabled && (0, _styledComponents.css)(["cursor:not-allowed;"]);
});
/**
 * Базовый компонент для загрузки файлов.
 */
var UploadButton = exports.UploadButton = function UploadButton(_ref6) {
  var status = _ref6.status,
    disabled = _ref6.disabled,
    isProgress = _ref6.isProgress,
    accept = _ref6.accept,
    content = _ref6.content,
    loader = _ref6.loader,
    _ref6$onChange = _ref6.onChange,
    onChange = _ref6$onChange === void 0 ? function () {} : _ref6$onChange,
    _ref6$onValidation = _ref6.onValidation,
    onValidation = _ref6$onValidation === void 0 ? function () {} : _ref6$onValidation,
    validate = _ref6.validate,
    rest = _objectWithoutProperties(_ref6, _excluded);
  var _useState = (0, _react.useState)(false),
    _useState2 = _slicedToArray(_useState, 2),
    isGrabbing = _useState2[0],
    setGrabbing = _useState2[1];
  var paint = isProgress ? 'progress' : 'default';
  var onUpload = (0, _react.useCallback)(function (files, event) {
    var result = (validate || _utils.defaultValidate)(files, accept);
    onValidation(result);
    if (result.data) {
      onChange(result.data, event);
    }
  }, [accept, onChange, onValidation, validate]);
  var dragStart = (0, _react.useCallback)(function () {
    return !disabled && !isProgress && setGrabbing(true);
  }, [disabled, isProgress]);
  var dragEnd = (0, _react.useCallback)(function () {
    return setGrabbing(false);
  }, []);
  var drop = (0, _react.useCallback)(function (files, event) {
    if (disabled || isProgress) {
      return;
    }
    event.preventDefault();
    onUpload(files);
    setGrabbing(false);
  }, [onUpload, disabled, isProgress]);
  var inputChangeHandler = (0, _react.useCallback)(function (event) {
    var files = event.target.files;
    onUpload(files, event);
  }, [onUpload]);
  return /*#__PURE__*/_react["default"].createElement(_reactFileDrop.FileDrop, {
    onDrop: drop,
    onDragOver: dragStart,
    onDragLeave: dragEnd
  }, /*#__PURE__*/_react["default"].createElement(StyledButton, _extends({
    type: "button",
    isGrabbing: isGrabbing,
    paint: paint,
    status: status,
    disabled: disabled
  }, rest), /*#__PURE__*/_react["default"].createElement(StyledInputLabel, {
    disabled: disabled
  }, /*#__PURE__*/_react["default"].createElement("input", {
    disabled: disabled,
    type: "file",
    onChange: inputChangeHandler,
    accept: accept,
    hidden: true
  })), isProgress ? loader : content));
};
