// @salutejs/plasma-new-hope/cjs/components/Button/Button.styles.js
import './Button.styles_1oz1dew.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
require('focus-visible');
require('styled-components');
require('@linaria/core');
var common = require('../../engines/common.js');
var Spinner$1 = require('../Spinner/Spinner.js');

// issue #823
var mergedConfig = /*#__PURE__*/common.mergeConfig(Spinner$1.spinnerConfig);
var Spinner = /*#__PURE__*/common.component(mergedConfig);
var ButtonText = /*#__PURE__*/react.styled('span')({
  name: "ButtonText",
  "class": "bzw1898",
  propsAsIs: false
});
var ButtonValue = /*#__PURE__*/react.styled('span')({
  name: "ButtonValue",
  "class": "bdfuqwe",
  propsAsIs: false
});

// TODO: #720
var _exp6 = function _exp6() {
  return function (_ref) {
    var isLoading = _ref.isLoading;
    return isLoading ? '0' : '1';
  };
};
var _exp7 = function _exp7() {
  return function (_ref2) {
    var contentPlacing = _ref2.contentPlacing;
    return contentPlacing;
  };
};
var LoadWrap = /*#__PURE__*/react.styled('div')({
  name: "LoadWrap",
  "class": "l9ypozh",
  propsAsIs: false,
  vars: {
    "l9ypozh-0": [/*#__PURE__*/_exp6()],
    "l9ypozh-1": [/*#__PURE__*/_exp7()]
  }
});
var Loader = /*#__PURE__*/react.styled('div')({
  name: "Loader",
  "class": "l4dfhbg",
  propsAsIs: false
});
var _exp8 = function _exp8() {
  return Spinner;
};
var StyledSpinner = /*#__PURE__*/react.styled(_exp8())({
  name: "StyledSpinner",
  "class": "srdlazt",
  propsAsIs: true
});
var StyledContentLeft = /*#__PURE__*/react.styled('div')({
  name: "StyledContentLeft",
  "class": "s7rggv0",
  propsAsIs: false
});
var StyledContentRight = /*#__PURE__*/react.styled('div')({
  name: "StyledContentRight",
  "class": "s1rjh7ra",
  propsAsIs: false
});
var base = "b9ga1dj";

exports.ButtonText = ButtonText;
exports.ButtonValue = ButtonValue;
exports.LoadWrap = LoadWrap;
exports.Loader = Loader;
exports.StyledContentLeft = StyledContentLeft;
exports.StyledContentRight = StyledContentRight;
exports.StyledSpinner = StyledSpinner;
exports.base = base;
