// @salutejs/plasma-web/es/components/Tooltip/Tooltip.perftest.js
var _StyledGrid;
/* eslint-disable */
import React from 'react';
import styled from 'styled-components';
import { IconDownload } from '@salutejs/plasma-icons';
import { Button } from '../Button';
import { Tooltip } from '.';
var StyledGrid = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-web__sc-pnme77-0"
})(["display:grid;grid-template-columns:repeat(3,max-content);grid-gap:1rem 3.5rem;padding:2.5rem;"]);
export var All = function All() {
  return _StyledGrid || (_StyledGrid = /*#__PURE__*/React.createElement(StyledGrid, null, /*#__PURE__*/React.createElement(Tooltip, {
    target: /*#__PURE__*/React.createElement(Tooltip, {
      target: /*#__PURE__*/React.createElement(Button, {
        contentLeft: /*#__PURE__*/React.createElement(IconDownload, null)
      }),
      placement: "left",
      isOpen: true,
      hasArrow: true,
      text: "Left"
    }),
    placement: "top-start",
    isOpen: true,
    hasArrow: true,
    text: "Top start"
  }), /*#__PURE__*/React.createElement(Tooltip, {
    target: /*#__PURE__*/React.createElement(Button, {
      contentLeft: /*#__PURE__*/React.createElement(IconDownload, null)
    }),
    placement: "top",
    isOpen: true,
    hasArrow: true,
    text: "Top"
  }), /*#__PURE__*/React.createElement(Tooltip, {
    target: /*#__PURE__*/React.createElement(Tooltip, {
      target: /*#__PURE__*/React.createElement(Button, {
        contentLeft: /*#__PURE__*/React.createElement(IconDownload, null)
      }),
      placement: "right",
      isOpen: true,
      hasArrow: true,
      text: "Right"
    }),
    placement: "top-end",
    isOpen: true,
    hasArrow: true,
    text: "Top end"
  }), /*#__PURE__*/React.createElement(Tooltip, {
    target: /*#__PURE__*/React.createElement(Button, {
      contentLeft: /*#__PURE__*/React.createElement(IconDownload, null)
    }),
    placement: "bottom-start",
    isOpen: true,
    hasArrow: true,
    text: "Bottom start"
  }), /*#__PURE__*/React.createElement(Tooltip, {
    target: /*#__PURE__*/React.createElement(Button, {
      contentLeft: /*#__PURE__*/React.createElement(IconDownload, null)
    }),
    placement: "bottom",
    isOpen: true,
    hasArrow: true,
    text: "Bottom"
  }), /*#__PURE__*/React.createElement(Tooltip, {
    target: /*#__PURE__*/React.createElement(Button, {
      contentLeft: /*#__PURE__*/React.createElement(IconDownload, null)
    }),
    placement: "bottom-end",
    isOpen: true,
    hasArrow: true,
    text: "Bottom end"
  })));
};
