// @salutejs/plasma-new-hope/es/components/Skeleton/TextSkeleton/TextSkeleton.js
import { objectWithoutProperties as _objectWithoutProperties, slicedToArray as _slicedToArray, objectSpread2 as _objectSpread2, defineProperty as _defineProperty } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { useState, useEffect } from 'react';
import { StyledTextSkeleton } from './TextSkeleton.styles.js';

var _excluded = ["lines", "width", "roundness", "customGradientColor", "lighter", "size"];
var variousWidth = [7.58, 5.27, 13.54, 6.63, 0.28, 14.8, 0.33, 11.26, 14.1, 10.59, 3.38, 13.5, 7.71, 3.34, 7.96];

/**
 * Хок для создания компонента плейсхолдера нескольких строк текста.
 * Размеры компонента задаются с помощью констант и соответствуют размерам [типографических элементов](/?path=/docs/).
 */
var textSkeleton = function textSkeleton(Component) {
  return function (_ref) {
    var lines = _ref.lines,
      width = _ref.width,
      roundness = _ref.roundness,
      customGradientColor = _ref.customGradientColor,
      lighter = _ref.lighter,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'body1' : _ref$size,
      props = _objectWithoutProperties(_ref, _excluded);
    var _useState = useState(null),
      _useState2 = _slicedToArray(_useState, 2),
      fixedWidth = _useState2[0],
      setFixedWidth = _useState2[1];
    var _useState3 = useState([]),
      _useState4 = _slicedToArray(_useState3, 2),
      linesWidth = _useState4[0],
      setLinesWidth = _useState4[1];
    useEffect(function () {
      if (!width) {
        setFixedWidth(null);
        var computedLengths = Array(lines).fill(0).reduce(function (acc, _, i) {
          var w;
          // Последняя строка
          if (i === lines - 1 && lines !== 1) {
            w = 45 - variousWidth[i % variousWidth.length];
          } else if (lines === 1 || lines === 2) {
            // Единственная или первая из двух
            w = 100;
          } else {
            w = 100 - variousWidth[i % variousWidth.length];
          }
          return _objectSpread2(_objectSpread2({}, acc), {}, _defineProperty({}, i, w));
        }, {});
        setLinesWidth(Object.values(computedLengths));
        return;
      }
      if (!Number.isNaN(Number(width))) {
        setFixedWidth("".concat(width, "%"));
      } else {
        setFixedWidth(String(width));
      }
    }, [width, lines]);
    return /*#__PURE__*/React.createElement(StyledTextSkeleton, props, Array.from(Array(lines), function (_, i) {
      return /*#__PURE__*/React.createElement(Component, {
        key: "line:".concat(i),
        size: size,
        roundness: roundness,
        customGradientColor: customGradientColor,
        lighter: lighter,
        style: {
          width: fixedWidth || "".concat(linesWidth[i], "%")
        }
      });
    }));
  };
};

export { textSkeleton };
