// @salutejs/plasma-new-hope/cjs/components/Skeleton/RectSkeleton/RectSkeleton.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
require('focus-visible');
var applyRoundness = require('../../../mixins/applyRoundness.js');
var applySkeletonGradient = require('../../../mixins/applySkeletonGradient.js');
require('styled-components');
var RectSkeleton_styles = require('./RectSkeleton.styles.js');

var _excluded = ["width", "height", "lighter", "customGradientColor", "roundness"];
/**
 * Компонент для создания прямоугольного плейсхолдера.
 */
var RectSkeleton = /*#__PURE__*/React.forwardRef(function (_ref, outerRootRef) {
  var width = _ref.width,
    height = _ref.height,
    lighter = _ref.lighter,
    customGradientColor = _ref.customGradientColor,
    _ref$roundness = _ref.roundness,
    roundness = _ref$roundness === void 0 ? '16' : _ref$roundness,
    rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
  var roundnessValue = applyRoundness.getRoundness({
    roundness: roundness
  });
  var skeletonGradientColor = applySkeletonGradient.getSkeletonColor({
    lighter: lighter,
    customGradientColor: customGradientColor
  });
  return /*#__PURE__*/React.createElement(RectSkeleton_styles.StyledRectSkeleton, _rollupPluginBabelHelpers.extends({
    ref: outerRootRef,
    roundness: roundnessValue,
    gradientColor: skeletonGradientColor,
    width: Number.isNaN(Number(width)) ? String(width) : "".concat(width, "px"),
    height: Number.isNaN(Number(height)) ? String(height) : "".concat(height, "px")
  }, rest));
});

exports.RectSkeleton = RectSkeleton;
