// @salutejs/plasma-ui/tokens/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "bodyL", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.bodyL;
  }
});
Object.defineProperty(exports, "bodyLBold", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.bodyLBold;
  }
});
Object.defineProperty(exports, "bodyM", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.bodyM;
  }
});
Object.defineProperty(exports, "bodyMBold", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.bodyMBold;
  }
});
Object.defineProperty(exports, "bodyS", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.bodyS;
  }
});
Object.defineProperty(exports, "bodySBold", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.bodySBold;
  }
});
Object.defineProperty(exports, "bodyXS", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.bodyXS;
  }
});
Object.defineProperty(exports, "bodyXSBold", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.bodyXSBold;
  }
});
Object.defineProperty(exports, "bodyXXS", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.bodyXXS;
  }
});
Object.defineProperty(exports, "bodyXXSBold", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.bodyXXSBold;
  }
});
Object.defineProperty(exports, "dsplL", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.dsplL;
  }
});
Object.defineProperty(exports, "dsplM", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.dsplM;
  }
});
Object.defineProperty(exports, "dsplS", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.dsplS;
  }
});
Object.defineProperty(exports, "h1", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.h1;
  }
});
Object.defineProperty(exports, "h2", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.h2;
  }
});
Object.defineProperty(exports, "h3", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.h3;
  }
});
Object.defineProperty(exports, "h4", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.h4;
  }
});
Object.defineProperty(exports, "h5", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.h5;
  }
});
Object.defineProperty(exports, "textL", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.textL;
  }
});
Object.defineProperty(exports, "textLBold", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.textLBold;
  }
});
Object.defineProperty(exports, "textM", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.textM;
  }
});
Object.defineProperty(exports, "textMBold", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.textMBold;
  }
});
Object.defineProperty(exports, "textS", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.textS;
  }
});
Object.defineProperty(exports, "textSBold", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.textSBold;
  }
});
Object.defineProperty(exports, "textXS", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.textXS;
  }
});
Object.defineProperty(exports, "textXSBold", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.textXSBold;
  }
});
var _plasmaTypo = /*#__PURE__*/require("@salutejs/plasma-typo");
