// @salutejs/plasma-tokens/es/brands/index.js
// Generated by robots, do not change this manually!
/** Основной цвет текста */
export var textPrimary = 'var(--text-primary, rgba(255, 255, 255, 0.96))';
/** Основной цвет текста */
export var textPrimaryHover = 'var(--text-primary-hover, #FFFFFF93)';
/** Основной цвет текста */
export var textPrimaryActive = 'var(--text-primary-active, #FFFFFFC4)';
/** Вторичный цвет текста */
export var textSecondary = 'var(--text-secondary, rgba(255, 255, 255, 0.56))';
/** Вторичный цвет текста */
export var textSecondaryHover = 'var(--text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста */
export var textSecondaryActive = 'var(--text-secondary-active, #FFFFFFAB)';
/** Третичный цвет текста */
export var textTertiary = 'var(--text-tertiary, rgba(255, 255, 255, 0.28))';
/** Третичный цвет текста */
export var textTertiaryHover = 'var(--text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста */
export var textTertiaryActive = 'var(--text-tertiary-active, #FFFFFF56)';
/** Сплошной наборный текст */
export var textParagraph = 'var(--text-paragraph, rgba(255, 255, 255, 0.8))';
/** Сплошной наборный текст */
export var textParagraphHover = 'var(--text-paragraph-hover, #FFFFFF7A)';
/** Сплошной наборный текст */
export var textParagraphActive = 'var(--text-paragraph-active, #FFFFFFA3)';
/** Акцентный цвет */
export var textAccent = 'var(--text-accent, #1A9E32)';
/** Акцентный цвет */
export var textAccentHover = 'var(--text-accent-hover, #1EB83AFF)';
/** Акцентный цвет */
export var textAccentActive = 'var(--text-accent-active, #15842AFF)';
/** Акцентный минорный цвет */
export var textAccentMinor = 'var(--text-accent-minor, #9EFAAF)';
/** Акцентный минорный цвет */
export var textAccentMinorHover = 'var(--text-accent-minor-hover, #64F77FFF)';
/** Акцентный минорный цвет */
export var textAccentMinorActive = 'var(--text-accent-minor-active, #81F897FF)';
/** Цвет успеха */
export var textPositive = 'var(--text-positive, #1A9E32)';
/** Цвет успеха */
export var textPositiveHover = 'var(--text-positive-hover, #1EB83AFF)';
/** Цвет успеха */
export var textPositiveActive = 'var(--text-positive-active, #15842AFF)';
/** Цвет предупреждения */
export var textWarning = 'var(--text-warning, #FA5F05)';
/** Цвет предупреждения */
export var textWarningHover = 'var(--text-warning-hover, #FB7223FF)';
/** Цвет предупреждения */
export var textWarningActive = 'var(--text-warning-active, #DC5304FF)';
/** Цвет ошибки */
export var textNegative = 'var(--text-negative, #FF293E)';
/** Цвет ошибки */
export var textNegativeHover = 'var(--text-negative-hover, #FF475AFF)';
/** Цвет ошибки */
export var textNegativeActive = 'var(--text-negative-active, #FF0A23FF)';
/** Цвет информации */
export var textInfo = 'var(--text-info, #118CDF)';
/** Цвет информации */
export var textInfoHover = 'var(--text-info-hover, #50B1F2FF)';
/** Цвет информации */
export var textInfoActive = 'var(--text-info-active, #0F7BC2FF)';
/** Минорный цвет успеха */
export var textPositiveMinor = 'var(--text-positive-minor, #9EFAAF)';
/** Минорный цвет успеха */
export var textPositiveMinorHover = 'var(--text-positive-minor-hover, #64F77FFF)';
/** Минорный цвет успеха */
export var textPositiveMinorActive = 'var(--text-positive-minor-active, #81F897FF)';
/** Минорный цвет предупреждения */
export var textWarningMinor = 'var(--text-warning-minor, #FEE2D2)';
/** Минорный цвет предупреждения */
export var textWarningMinorHover = 'var(--text-warning-minor-hover, #FDBC96FF)';
/** Минорный цвет предупреждения */
export var textWarningMinorActive = 'var(--text-warning-minor-active, #FDCFB4FF)';
/** Минорный цвет ошибки */
export var textNegativeMinor = 'var(--text-negative-minor, #FFE0E3)';
/** Минорный цвет ошибки */
export var textNegativeMinorHover = 'var(--text-negative-minor-hover, #FFA3ACFF)';
/** Минорный цвет ошибки */
export var textNegativeMinorActive = 'var(--text-negative-minor-active, #FFC2C8FF)';
/** Минорный цвет информации */
export var textInfoMinor = 'var(--text-info-minor, #CFECFF)';
/** Минорный цвет информации */
export var textInfoMinorHover = 'var(--text-info-minor-hover, #94D4FFFF)';
/** Минорный цвет информации */
export var textInfoMinorActive = 'var(--text-info-minor-active, #B3E0FFFF)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimary = 'var(--on-dark-text-primary, rgba(255, 255, 255, 0.96))';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimaryHover = 'var(--on-dark-text-primary-hover, #FFFFFF93)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimaryActive = 'var(--on-dark-text-primary-active, #FFFFFFC4)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondary = 'var(--on-dark-text-secondary, rgba(255, 255, 255, 0.56))';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondaryHover = 'var(--on-dark-text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondaryActive = 'var(--on-dark-text-secondary-active, #FFFFFFAB)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiary = 'var(--on-dark-text-tertiary, rgba(255, 255, 255, 0.28))';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiaryHover = 'var(--on-dark-text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiaryActive = 'var(--on-dark-text-tertiary-active, #FFFFFF56)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraph = 'var(--on-dark-text-paragraph, rgba(255, 255, 255, 0.8))';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraphHover = 'var(--on-dark-text-paragraph-hover, #FFFFFF7A)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraphActive = 'var(--on-dark-text-paragraph-active, #FFFFFFA3)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccent = 'var(--on-dark-text-accent, #1A9E32)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccentHover = 'var(--on-dark-text-accent-hover, #1EB83AFF)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccentActive = 'var(--on-dark-text-accent-active, #15842AFF)';
/** Акцентный минорный цвет на темном фоне */
export var onDarkTextAccentMinor = 'var(--on-dark-text-accent-minor, #9EFAAF)';
/** Акцентный минорный цвет на темном фоне */
export var onDarkTextAccentMinorHover = 'var(--on-dark-text-accent-minor-hover, #64F77FFF)';
/** Акцентный минорный цвет на темном фоне */
export var onDarkTextAccentMinorActive = 'var(--on-dark-text-accent-minor-active, #81F897FF)';
/** Цвет успеха на темном фоне */
export var onDarkTextPositive = 'var(--on-dark-text-positive, #1A9E32)';
/** Цвет успеха на темном фоне */
export var onDarkTextPositiveHover = 'var(--on-dark-text-positive-hover, #1EB83AFF)';
/** Цвет успеха на темном фоне */
export var onDarkTextPositiveActive = 'var(--on-dark-text-positive-active, #15842AFF)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarning = 'var(--on-dark-text-warning, #FA5F05)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarningHover = 'var(--on-dark-text-warning-hover, #FB7223FF)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarningActive = 'var(--on-dark-text-warning-active, #DC5304FF)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegative = 'var(--on-dark-text-negative, #FF293E)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegativeHover = 'var(--on-dark-text-negative-hover, #FF475AFF)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegativeActive = 'var(--on-dark-text-negative-active, #FF0A23FF)';
/** Цвет информации на темном фоне */
export var onDarkTextInfo = 'var(--on-dark-text-info, #118CDF)';
/** Цвет информации на темном фоне */
export var onDarkTextInfoHover = 'var(--on-dark-text-info-hover, #50B1F2FF)';
/** Цвет информации на темном фоне */
export var onDarkTextInfoActive = 'var(--on-dark-text-info-active, #0F7BC2FF)';
/** Минорный цвет успеха на темном фоне */
export var onDarkTextPositiveMinor = 'var(--on-dark-text-positive-minor, #9EFAAF)';
/** Минорный цвет успеха на темном фоне */
export var onDarkTextPositiveMinorHover = 'var(--on-dark-text-positive-minor-hover, #64F77FFF)';
/** Минорный цвет успеха на темном фоне */
export var onDarkTextPositiveMinorActive = 'var(--on-dark-text-positive-minor-active, #81F897FF)';
/** Минорный цвет предупреждения на темном фоне */
export var onDarkTextWarningMinor = 'var(--on-dark-text-warning-minor, #FEE2D2)';
/** Минорный цвет предупреждения на темном фоне */
export var onDarkTextWarningMinorHover = 'var(--on-dark-text-warning-minor-hover, #FDBC96FF)';
/** Минорный цвет предупреждения на темном фоне */
export var onDarkTextWarningMinorActive = 'var(--on-dark-text-warning-minor-active, #FDCFB4FF)';
/** Минорный цвет ошибки на темном фоне */
export var onDarkTextNegativeMinor = 'var(--on-dark-text-negative-minor, #FFE0E3)';
/** Минорный цвет ошибки на темном фоне */
export var onDarkTextNegativeMinorHover = 'var(--on-dark-text-negative-minor-hover, #FFA3ACFF)';
/** Минорный цвет ошибки на темном фоне */
export var onDarkTextNegativeMinorActive = 'var(--on-dark-text-negative-minor-active, #FFC2C8FF)';
/** Минорный цвет информации на темном фоне */
export var onDarkTextInfoMinor = 'var(--on-dark-text-info-minor, #CFECFF)';
/** Минорный цвет информации на темном фоне */
export var onDarkTextInfoMinorHover = 'var(--on-dark-text-info-minor-hover, #94D4FFFF)';
/** Минорный цвет информации на темном фоне */
export var onDarkTextInfoMinorActive = 'var(--on-dark-text-info-minor-active, #B3E0FFFF)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimary = 'var(--on-light-text-primary, #080808F5)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimaryHover = 'var(--on-light-text-primary-hover, #08080893)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimaryActive = 'var(--on-light-text-primary-active, #080808C4)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondary = 'var(--on-light-text-secondary, #0808088F)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondaryHover = 'var(--on-light-text-secondary-hover, #080808FF)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondaryActive = 'var(--on-light-text-secondary-active, #080808AB)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiary = 'var(--on-light-text-tertiary, #08080847)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiaryHover = 'var(--on-light-text-tertiary-hover, #080808FF)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiaryActive = 'var(--on-light-text-tertiary-active, #08080856)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraph = 'var(--on-light-text-paragraph, #080808CC)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraphHover = 'var(--on-light-text-paragraph-hover, #0808087A)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraphActive = 'var(--on-light-text-paragraph-active, #080808A3)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccent = 'var(--on-light-text-accent, #108E26)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccentHover = 'var(--on-light-text-accent-hover, #13AA2CFF)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccentActive = 'var(--on-light-text-accent-active, #0D731EFF)';
/** Акцентный минорный цвет на светлом фоне */
export var onLightTextAccentMinor = 'var(--on-light-text-accent-minor, #0A2B10)';
/** Акцентный минорный цвет на светлом фоне */
export var onLightTextAccentMinorHover = 'var(--on-light-text-accent-minor-hover, #155B22FF)';
/** Акцентный минорный цвет на светлом фоне */
export var onLightTextAccentMinorActive = 'var(--on-light-text-accent-minor-active, #104219FF)';
/** Цвет успеха на светлом фоне */
export var onLightTextPositive = 'var(--on-light-text-positive, #108E26)';
/** Цвет успеха на светлом фоне */
export var onLightTextPositiveHover = 'var(--on-light-text-positive-hover, #13AA2CFF)';
/** Цвет успеха на светлом фоне */
export var onLightTextPositiveActive = 'var(--on-light-text-positive-active, #0D731EFF)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarning = 'var(--on-light-text-warning, #E85702)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarningHover = 'var(--on-light-text-warning-hover, #FD650DFF)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarningActive = 'var(--on-light-text-warning-active, #CA4B02FF)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegative = 'var(--on-light-text-negative, #F31B31)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegativeHover = 'var(--on-light-text-negative-hover, #F5384BFF)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegativeActive = 'var(--on-light-text-negative-active, #E40C22FF)';
/** Цвет информации на светлом фоне */
export var onLightTextInfo = 'var(--on-light-text-info, #0B7ECB)';
/** Цвет информации на светлом фоне */
export var onLightTextInfoHover = 'var(--on-light-text-info-hover, #34A7F4FF)';
/** Цвет информации на светлом фоне */
export var onLightTextInfoActive = 'var(--on-light-text-info-active, #096CAEFF)';
/** Минорный цвет успеха на светлом фоне */
export var onLightTextPositiveMinor = 'var(--on-light-text-positive-minor, #0A2B10)';
/** Минорный цвет успеха на светлом фоне */
export var onLightTextPositiveMinorHover = 'var(--on-light-text-positive-minor-hover, #155B22FF)';
/** Минорный цвет успеха на светлом фоне */
export var onLightTextPositiveMinorActive = 'var(--on-light-text-positive-minor-active, #104219FF)';
/** Минорный цвет предупреждения на светлом фоне */
export var onLightTextWarningMinor = 'var(--on-light-text-warning-minor, #3D1D0A)';
/** Минорный цвет предупреждения на светлом фоне */
export var onLightTextWarningMinorHover = 'var(--on-light-text-warning-minor-hover, #723613FF)';
/** Минорный цвет предупреждения на светлом фоне */
export var onLightTextWarningMinorActive = 'var(--on-light-text-warning-minor-active, #58290EFF)';
/** Минорный цвет ошибки на светлом фоне */
export var onLightTextNegativeMinor = 'var(--on-light-text-negative-minor, #4A0D13)';
/** Минорный цвет ошибки на светлом фоне */
export var onLightTextNegativeMinorHover = 'var(--on-light-text-negative-minor-hover, #7E1621FF)';
/** Минорный цвет ошибки на светлом фоне */
export var onLightTextNegativeMinorActive = 'var(--on-light-text-negative-minor-active, #64121AFF)';
/** Минорный цвет информации на светлом фоне */
export var onLightTextInfoMinor = 'var(--on-light-text-info-minor, #0C283B)';
/** Минорный цвет информации на светлом фоне */
export var onLightTextInfoMinorHover = 'var(--on-light-text-info-minor-hover, #174B6EFF)';
/** Минорный цвет информации на светлом фоне */
export var onLightTextInfoMinorActive = 'var(--on-light-text-info-minor-active, #113A55FF)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimary = 'var(--inverse-text-primary, #080808F5)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimaryHover = 'var(--inverse-text-primary-hover, #08080893)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimaryActive = 'var(--inverse-text-primary-active, #080808C4)';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondary = 'var(--inverse-text-secondary, #0808088F)';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondaryHover = 'var(--inverse-text-secondary-hover, #080808FF)';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondaryActive = 'var(--inverse-text-secondary-active, #080808AB)';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiary = 'var(--inverse-text-tertiary, #08080847)';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiaryHover = 'var(--inverse-text-tertiary-hover, #080808FF)';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiaryActive = 'var(--inverse-text-tertiary-active, #08080856)';
/** Инвертированный сплошной наборный текст */
export var inverseTextParagraph = 'var(--inverse-text-paragraph, #080808CC)';
/** Инвертированный сплошной наборный текст */
export var inverseTextParagraphHover = 'var(--inverse-text-paragraph-hover, #0808087A)';
/** Инвертированный сплошной наборный текст */
export var inverseTextParagraphActive = 'var(--inverse-text-paragraph-active, #080808A3)';
/** Инвертированный акцентный цвет */
export var inverseTextAccent = 'var(--inverse-text-accent, #108E26)';
/** Инвертированный акцентный цвет */
export var inverseTextAccentHover = 'var(--inverse-text-accent-hover, #13AA2CFF)';
/** Инвертированный акцентный цвет */
export var inverseTextAccentActive = 'var(--inverse-text-accent-active, #0D731EFF)';
/** Инвертированный минорный акцентный цвет */
export var inverseTextAccentMinor = 'var(--inverse-text-accent-minor, #0A2B10)';
/** Инвертированный минорный акцентный цвет */
export var inverseTextAccentMinorHover = 'var(--inverse-text-accent-minor-hover, #155B22FF)';
/** Инвертированный минорный акцентный цвет */
export var inverseTextAccentMinorActive = 'var(--inverse-text-accent-minor-active, #104219FF)';
/** Инвертированный цвет успеха */
export var inverseTextPositive = 'var(--inverse-text-positive, #108E26)';
/** Инвертированный цвет успеха */
export var inverseTextPositiveHover = 'var(--inverse-text-positive-hover, #13AA2CFF)';
/** Инвертированный цвет успеха */
export var inverseTextPositiveActive = 'var(--inverse-text-positive-active, #0D731EFF)';
/** Инвертированный цвет предупреждения */
export var inverseTextWarning = 'var(--inverse-text-warning, #E85702)';
/** Инвертированный цвет предупреждения */
export var inverseTextWarningHover = 'var(--inverse-text-warning-hover, #FD650DFF)';
/** Инвертированный цвет предупреждения */
export var inverseTextWarningActive = 'var(--inverse-text-warning-active, #CA4B02FF)';
/** Инвертированный цвет ошибки */
export var inverseTextNegative = 'var(--inverse-text-negative, #F31B31)';
/** Инвертированный цвет ошибки */
export var inverseTextNegativeHover = 'var(--inverse-text-negative-hover, #F5384BFF)';
/** Инвертированный цвет ошибки */
export var inverseTextNegativeActive = 'var(--inverse-text-negative-active, #E40C22FF)';
/** Инвертированный цвет информации */
export var inverseTextInfo = 'var(--inverse-text-info, #0B7ECB)';
/** Инвертированный цвет информации */
export var inverseTextInfoHover = 'var(--inverse-text-info-hover, #34A7F4FF)';
/** Инвертированный цвет информации */
export var inverseTextInfoActive = 'var(--inverse-text-info-active, #096CAEFF)';
/** Инвертированный минорный цвет успеха */
export var inverseTextPositiveMinor = 'var(--inverse-text-positive-minor, #0A2B10)';
/** Инвертированный минорный цвет успеха */
export var inverseTextPositiveMinorHover = 'var(--inverse-text-positive-minor-hover, #155B22FF)';
/** Инвертированный минорный цвет успеха */
export var inverseTextPositiveMinorActive = 'var(--inverse-text-positive-minor-active, #104219FF)';
/** Инвертированный минорный цвет предупреждения */
export var inverseTextWarningMinor = 'var(--inverse-text-warning-minor, #3D1D0A)';
/** Инвертированный минорный цвет предупреждения */
export var inverseTextWarningMinorHover = 'var(--inverse-text-warning-minor-hover, #723613FF)';
/** Инвертированный минорный цвет предупреждения */
export var inverseTextWarningMinorActive = 'var(--inverse-text-warning-minor-active, #58290EFF)';
/** Инвертированный минорный цвет ошибки */
export var inverseTextNegativeMinor = 'var(--inverse-text-negative-minor, #4A0D13)';
/** Инвертированный минорный цвет ошибки */
export var inverseTextNegativeMinorHover = 'var(--inverse-text-negative-minor-hover, #7E1621FF)';
/** Инвертированный минорный цвет ошибки */
export var inverseTextNegativeMinorActive = 'var(--inverse-text-negative-minor-active, #64121AFF)';
/** Инвертированный минорный цвет информации */
export var inverseTextInfoMinor = 'var(--inverse-text-info-minor, #0C283B)';
/** Инвертированный минорный цвет информации */
export var inverseTextInfoMinorHover = 'var(--inverse-text-info-minor-hover, #174B6EFF)';
/** Инвертированный минорный цвет информации */
export var inverseTextInfoMinorActive = 'var(--inverse-text-info-minor-active, #113A55FF)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimary = 'var(--surface-solid-primary, #171717)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimaryHover = 'var(--surface-solid-primary-hover, #1C1C1CFF)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimaryActive = 'var(--surface-solid-primary-active, #121212FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondary = 'var(--surface-solid-secondary, #262626)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondaryHover = 'var(--surface-solid-secondary-hover, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondaryActive = 'var(--surface-solid-secondary-active, #212121FF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiary = 'var(--surface-solid-tertiary, #363636)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiaryHover = 'var(--surface-solid-tertiary-hover, #404040FF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiaryActive = 'var(--surface-solid-tertiary-active, #303030FF)';
/** Основной фон для карточек */
export var surfaceSolidCard = 'var(--surface-solid-card, #171717)';
/** Основной фон для карточек */
export var surfaceSolidCardHover = 'var(--surface-solid-card-hover, #1C1C1CFF)';
/** Основной фон для карточек */
export var surfaceSolidCardActive = 'var(--surface-solid-card-active, #121212FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefault = 'var(--surface-solid-default, #F9F9F9)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefaultHover = 'var(--surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefaultActive = 'var(--surface-solid-default-active, #F5F5F5FF)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimary = 'var(--surface-transparent-primary, #F9F9F90F)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimaryHover = 'var(--surface-transparent-primary-hover, #FAFAFA1F)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimaryActive = 'var(--surface-transparent-primary-active, #FAFAFA0A)';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondary = 'var(--surface-transparent-secondary, #F9F9F91F)';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondaryHover = 'var(--surface-transparent-secondary-hover, #FAFAFA3D)';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondaryActive = 'var(--surface-transparent-secondary-active, #FAFAFA0F)';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, #F9F9F933)';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiaryHover = 'var(--surface-transparent-tertiary-hover, #FAFAFA52)';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiaryActive = 'var(--surface-transparent-tertiary-active, #FAFAFA24)';
/** Глубокий прозрачный фон поверхности/контрола */
export var surfaceTransparentDeep = 'var(--surface-transparent-deep, #F9F9F9A3)';
/** Глубокий прозрачный фон поверхности/контрола */
export var surfaceTransparentDeepHover = 'var(--surface-transparent-deep-hover, #FAFAFAC2)';
/** Глубокий прозрачный фон поверхности/контрола */
export var surfaceTransparentDeepActive = 'var(--surface-transparent-deep-active, #FAFAFA94)';
/** Прозрачный фон для карточек */
export var surfaceTransparentCard = 'var(--surface-transparent-card, #F9F9F90F)';
/** Прозрачный фон для карточек */
export var surfaceTransparentCardHover = 'var(--surface-transparent-card-hover, #FAFAFA1F)';
/** Прозрачный фон для карточек */
export var surfaceTransparentCardActive = 'var(--surface-transparent-card-active, #FAFAFA0A)';
/** Фон поверхности/контрола без заливки */
export var surfaceClear = 'var(--surface-clear, #FFFFFF00)';
/** Фон поверхности/контрола без заливки */
export var surfaceClearHover = 'var(--surface-clear-hover, #FFFFFF00)';
/** Фон поверхности/контрола без заливки */
export var surfaceClearActive = 'var(--surface-clear-active, #FFFFFF00)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccent = 'var(--surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentHover = 'var(--surface-accent-hover, #1DAF37FF)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentActive = 'var(--surface-accent-active, #18952FFF)';
/** Акцентный минорный непрозрачный фон поверхности/контрола */
export var surfaceAccentMinor = 'var(--surface-accent-minor, #0A2B10)';
/** Акцентный минорный непрозрачный фон поверхности/контрола */
export var surfaceAccentMinorHover = 'var(--surface-accent-minor-hover, #0E3A16FF)';
/** Акцентный минорный непрозрачный фон поверхности/контрола */
export var surfaceAccentMinorActive = 'var(--surface-accent-minor-active, #08210CFF)';
/** Прозрачный акцентный фон поверхности/контрола */
export var surfaceTransparentAccent = 'var(--surface-transparent-accent, #1A9E3247)';
/** Прозрачный акцентный фон поверхности/контрола */
export var surfaceTransparentAccentHover = 'var(--surface-transparent-accent-hover, #1A9E3266)';
/** Прозрачный акцентный фон поверхности/контрола */
export var surfaceTransparentAccentActive = 'var(--surface-transparent-accent-active, #1A9E3238)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositive = 'var(--surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositiveHover = 'var(--surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositiveActive = 'var(--surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarning = 'var(--surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarningHover = 'var(--surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarningActive = 'var(--surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegative = 'var(--surface-negative, #FF293E)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegativeHover = 'var(--surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegativeActive = 'var(--surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола информация */
export var surfaceInfo = 'var(--surface-info, #118CDF)';
/** Цвет фона поверхности/контрола информация */
export var surfaceInfoHover = 'var(--surface-info-hover, #1798EEFF)';
/** Цвет фона поверхности/контрола информация */
export var surfaceInfoActive = 'var(--surface-info-active, #1086D5FF)';
/** Минорный цвет фона поверхности/контрола успех */
export var surfacePositiveMinor = 'var(--surface-positive-minor, #0A2B10)';
/** Минорный цвет фона поверхности/контрола успех */
export var surfacePositiveMinorHover = 'var(--surface-positive-minor-hover, #0E3A16FF)';
/** Минорный цвет фона поверхности/контрола успех */
export var surfacePositiveMinorActive = 'var(--surface-positive-minor-active, #08210CFF)';
/** Минорный цвет фона поверхности/контрола предупреждение */
export var surfaceWarningMinor = 'var(--surface-warning-minor, #3D1D0A)';
/** Минорный цвет фона поверхности/контрола предупреждение */
export var surfaceWarningMinorHover = 'var(--surface-warning-minor-hover, #4F250DFF)';
/** Минорный цвет фона поверхности/контрола предупреждение */
export var surfaceWarningMinorActive = 'var(--surface-warning-minor-active, #351909FF)';
/** Минорный цвет фона поверхности/контрола ошибка */
export var surfaceNegativeMinor = 'var(--surface-negative-minor, #4A0D13)';
/** Минорный цвет фона поверхности/контрола ошибка */
export var surfaceNegativeMinorHover = 'var(--surface-negative-minor-hover, #5B1018FF)';
/** Минорный цвет фона поверхности/контрола ошибка */
export var surfaceNegativeMinorActive = 'var(--surface-negative-minor-active, #410B11FF)';
/** Минорный цвет фона поверхности/контрола информация */
export var surfaceInfoMinor = 'var(--surface-info-minor, #0C283B)';
/** Минорный цвет фона поверхности/контрола информация */
export var surfaceInfoMinorHover = 'var(--surface-info-minor-hover, #10344CFF)';
/** Минорный цвет фона поверхности/контрола информация */
export var surfaceInfoMinorActive = 'var(--surface-info-minor-active, #0A2333FF)';
/** Прозрачный цвет фона поверхности/контрола успех */
export var surfaceTransparentPositive = 'var(--surface-transparent-positive, #1A9E3247)';
/** Прозрачный цвет фона поверхности/контрола успех */
export var surfaceTransparentPositiveHover = 'var(--surface-transparent-positive-hover, #1A9E3266)';
/** Прозрачный цвет фона поверхности/контрола успех */
export var surfaceTransparentPositiveActive = 'var(--surface-transparent-positive-active, #1A9E3238)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
export var surfaceTransparentWarning = 'var(--surface-transparent-warning, #FA5F0547)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
export var surfaceTransparentWarningHover = 'var(--surface-transparent-warning-hover, #FA5F0566)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
export var surfaceTransparentWarningActive = 'var(--surface-transparent-warning-active, #FA5F0538)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
export var surfaceTransparentNegative = 'var(--surface-transparent-negative, #FF293E47)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
export var surfaceTransparentNegativeHover = 'var(--surface-transparent-negative-hover, #FF293E66)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
export var surfaceTransparentNegativeActive = 'var(--surface-transparent-negative-active, #FF293E38)';
/** Прозрачный цвет фона поверхности/контрола информация */
export var surfaceTransparentInfo = 'var(--surface-transparent-info, #118CDF47)';
/** Прозрачный цвет фона поверхности/контрола информация */
export var surfaceTransparentInfoHover = 'var(--surface-transparent-info-hover, #118CDF66)';
/** Прозрачный цвет фона поверхности/контрола информация */
export var surfaceTransparentInfoActive = 'var(--surface-transparent-info-active, #118CDF38)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #171717)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimaryHover = 'var(--on-dark-surface-solid-primary-hover, #1C1C1CFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimaryActive = 'var(--on-dark-surface-solid-primary-active, #121212FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #262626)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondaryHover = 'var(--on-dark-surface-solid-secondary-hover, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondaryActive = 'var(--on-dark-surface-solid-secondary-active, #212121FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #363636)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiaryHover = 'var(--on-dark-surface-solid-tertiary-hover, #404040FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiaryActive = 'var(--on-dark-surface-solid-tertiary-active, #303030FF)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #171717)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCardHover = 'var(--on-dark-surface-solid-card-hover, #1C1C1CFF)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCardActive = 'var(--on-dark-surface-solid-card-active, #121212FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #F9F9F9)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefaultHover = 'var(--on-dark-surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefaultActive = 'var(--on-dark-surface-solid-default-active, #F5F5F5FF)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, #F9F9F90F)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimaryHover = 'var(--on-dark-surface-transparent-primary-hover, #FAFAFA1F)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimaryActive = 'var(--on-dark-surface-transparent-primary-active, #FAFAFA0A)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, #F9F9F91F)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondaryHover = 'var(--on-dark-surface-transparent-secondary-hover, #FAFAFA3D)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondaryActive = 'var(--on-dark-surface-transparent-secondary-active, #FAFAFA0F)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, #F9F9F933)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiaryHover = 'var(--on-dark-surface-transparent-tertiary-hover, #FAFAFA52)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiaryActive = 'var(--on-dark-surface-transparent-tertiary-active, #FAFAFA24)';
/** Глубокий прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentDeep = 'var(--on-dark-surface-transparent-deep, #F9F9F9A3)';
/** Глубокий прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentDeepHover = 'var(--on-dark-surface-transparent-deep-hover, #FAFAFAC2)';
/** Глубокий прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentDeepActive = 'var(--on-dark-surface-transparent-deep-active, #FAFAFA94)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, #F9F9F90F)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCardHover = 'var(--on-dark-surface-transparent-card-hover, #FAFAFA1F)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCardActive = 'var(--on-dark-surface-transparent-card-active, #FAFAFA0A)';
/** Фон поверхности/контрола без заливки на темном фоне */
export var onDarkSurfaceClear = 'var(--on-dark-surface-clear, #FFFFFF00)';
/** Фон поверхности/контрола без заливки на темном фоне */
export var onDarkSurfaceClearHover = 'var(--on-dark-surface-clear-hover, #FFFFFF00)';
/** Фон поверхности/контрола без заливки на темном фоне */
export var onDarkSurfaceClearActive = 'var(--on-dark-surface-clear-active, #FFFFFF00)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentHover = 'var(--on-dark-surface-accent-hover, #1DAF37FF)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentActive = 'var(--on-dark-surface-accent-active, #18952FFF)';
/** Акцентный минорный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentMinor = 'var(--on-dark-surface-accent-minor, #0A2B10)';
/** Акцентный минорный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentMinorHover = 'var(--on-dark-surface-accent-minor-hover, #0E3A16FF)';
/** Акцентный минорный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentMinorActive = 'var(--on-dark-surface-accent-minor-active, #08210CFF)';
/** Прозрачный акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentAccent = 'var(--on-dark-surface-transparent-accent, #1A9E3247)';
/** Прозрачный акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentAccentHover = 'var(--on-dark-surface-transparent-accent-hover, #1A9E3266)';
/** Прозрачный акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentAccentActive = 'var(--on-dark-surface-transparent-accent-active, #1A9E3238)';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositive = 'var(--on-dark-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositiveHover = 'var(--on-dark-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositiveActive = 'var(--on-dark-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceWarningHover = 'var(--on-dark-surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceWarningActive = 'var(--on-dark-surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293E)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegativeHover = 'var(--on-dark-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegativeActive = 'var(--on-dark-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола информация на темном фоне */
export var onDarkSurfaceInfo = 'var(--on-dark-surface-info, #118CDF)';
/** Цвет фона поверхности/контрола информация на темном фоне */
export var onDarkSurfaceInfoHover = 'var(--on-dark-surface-info-hover, #1798EEFF)';
/** Цвет фона поверхности/контрола информация на темном фоне */
export var onDarkSurfaceInfoActive = 'var(--on-dark-surface-info-active, #1086D5FF)';
/** Минорный цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositiveMinor = 'var(--on-dark-surface-positive-minor, #0A2B10)';
/** Минорный цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositiveMinorHover = 'var(--on-dark-surface-positive-minor-hover, #0E3A16FF)';
/** Минорный цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositiveMinorActive = 'var(--on-dark-surface-positive-minor-active, #08210CFF)';
/** Минорный цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceWarningMinor = 'var(--on-dark-surface-warning-minor, #3D1D0A)';
/** Минорный цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceWarningMinorHover = 'var(--on-dark-surface-warning-minor-hover, #4F250DFF)';
/** Минорный цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceWarningMinorActive = 'var(--on-dark-surface-warning-minor-active, #351909FF)';
/** Минорный цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegativeMinor = 'var(--on-dark-surface-negative-minor, #4A0D13)';
/** Минорный цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegativeMinorHover = 'var(--on-dark-surface-negative-minor-hover, #5B1018FF)';
/** Минорный цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegativeMinorActive = 'var(--on-dark-surface-negative-minor-active, #410B11FF)';
/** Минорный цвет фона поверхности/контрола информация на темном фоне */
export var onDarkSurfaceInfoMinor = 'var(--on-dark-surface-info-minor, #0C283B)';
/** Минорный цвет фона поверхности/контрола информация на темном фоне */
export var onDarkSurfaceInfoMinorHover = 'var(--on-dark-surface-info-minor-hover, #10344CFF)';
/** Минорный цвет фона поверхности/контрола информация на темном фоне */
export var onDarkSurfaceInfoMinorActive = 'var(--on-dark-surface-info-minor-active, #0A2333FF)';
/** Прозрачный цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfaceTransparentPositive = 'var(--on-dark-surface-transparent-positive, #1A9E3247)';
/** Прозрачный цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfaceTransparentPositiveHover = 'var(--on-dark-surface-transparent-positive-hover, #1A9E3266)';
/** Прозрачный цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfaceTransparentPositiveActive = 'var(--on-dark-surface-transparent-positive-active, #1A9E3238)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceTransparentWarning = 'var(--on-dark-surface-transparent-warning, #FA5F0547)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceTransparentWarningHover = 'var(--on-dark-surface-transparent-warning-hover, #FA5F0566)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceTransparentWarningActive = 'var(--on-dark-surface-transparent-warning-active, #FA5F0538)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceTransparentNegative = 'var(--on-dark-surface-transparent-negative, #FF293E47)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceTransparentNegativeHover = 'var(--on-dark-surface-transparent-negative-hover, #FF293E66)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceTransparentNegativeActive = 'var(--on-dark-surface-transparent-negative-active, #FF293E38)';
/** Прозрачный цвет фона поверхности/контрола информация на темном фоне */
export var onDarkSurfaceTransparentInfo = 'var(--on-dark-surface-transparent-info, #118CDF47)';
/** Прозрачный цвет фона поверхности/контрола информация на темном фоне */
export var onDarkSurfaceTransparentInfoHover = 'var(--on-dark-surface-transparent-info-hover, #118CDF66)';
/** Прозрачный цвет фона поверхности/контрола информация на темном фоне */
export var onDarkSurfaceTransparentInfoActive = 'var(--on-dark-surface-transparent-info-active, #118CDF38)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F5F5F5)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidPrimaryHover = 'var(--on-light-surface-solid-primary-hover, #FAFAFAFF)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidPrimaryActive = 'var(--on-light-surface-solid-primary-active, #F0F0F0FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #ECECEC)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondaryHover = 'var(--on-light-surface-solid-secondary-hover, #F7F7F7FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondaryActive = 'var(--on-light-surface-solid-secondary-active, #E8E8E8FF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #DDDDDD)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidTertiaryHover = 'var(--on-light-surface-solid-tertiary-hover, #E8E8E8FF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidTertiaryActive = 'var(--on-light-surface-solid-tertiary-active, #D9D9D9FF)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFF)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCardHover = 'var(--on-light-surface-solid-card-hover, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCardActive = 'var(--on-light-surface-solid-card-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #080808)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefaultHover = 'var(--on-light-surface-solid-default-hover, #0D0D0DFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefaultActive = 'var(--on-light-surface-solid-default-active, #030303FF)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, #08080808)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentPrimaryHover = 'var(--on-light-surface-transparent-primary-hover, #08080812)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentPrimaryActive = 'var(--on-light-surface-transparent-primary-active, #08080803)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, #0808080F)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondaryHover = 'var(--on-light-surface-transparent-secondary-hover, #0808081F)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondaryActive = 'var(--on-light-surface-transparent-secondary-active, #0808080A)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, #0808081F)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiaryHover = 'var(--on-light-surface-transparent-tertiary-hover, #0808083D)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiaryActive = 'var(--on-light-surface-transparent-tertiary-active, #0808080F)';
/** Глубокий прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentDeep = 'var(--on-light-surface-transparent-deep, #080808A3)';
/** Глубокий прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentDeepHover = 'var(--on-light-surface-transparent-deep-hover, #080808C2)';
/** Глубокий прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentDeepActive = 'var(--on-light-surface-transparent-deep-active, #08080894)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, #0808080F)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCardHover = 'var(--on-light-surface-transparent-card-hover, #0808081F)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCardActive = 'var(--on-light-surface-transparent-card-active, #0808080A)';
/** Фон поверхности/контрола без заливки на светлом фоне */
export var onLightSurfaceClear = 'var(--on-light-surface-clear, #FFFFFF00)';
/** Фон поверхности/контрола без заливки на светлом фоне */
export var onLightSurfaceClearHover = 'var(--on-light-surface-clear-hover, #FFFFFF00)';
/** Фон поверхности/контрола без заливки на светлом фоне */
export var onLightSurfaceClearActive = 'var(--on-light-surface-clear-active, #FFFFFF00)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccent = 'var(--on-light-surface-accent, #108E26)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentHover = 'var(--on-light-surface-accent-hover, #12A12AFF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentActive = 'var(--on-light-surface-accent-active, #0F8523FF)';
/** Акцентный минорный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentMinor = 'var(--on-light-surface-accent-minor, #77F38E)';
/** Акцентный минорный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentMinorHover = 'var(--on-light-surface-accent-minor-hover, #8AF59DFF)';
/** Акцентный минорный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentMinorActive = 'var(--on-light-surface-accent-minor-active, #6EF286FF)';
/** Прозрачный акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentAccent = 'var(--on-light-surface-transparent-accent, #108E2647)';
/** Прозрачный акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentAccentHover = 'var(--on-light-surface-transparent-accent-hover, #108E2566)';
/** Прозрачный акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentAccentActive = 'var(--on-light-surface-transparent-accent-active, #108E2538)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositive = 'var(--on-light-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositiveHover = 'var(--on-light-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositiveActive = 'var(--on-light-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceWarning = 'var(--on-light-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceWarningHover = 'var(--on-light-surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceWarningActive = 'var(--on-light-surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegative = 'var(--on-light-surface-negative, #FF293E)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeHover = 'var(--on-light-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeActive = 'var(--on-light-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола информация на светлом фоне */
export var onLightSurfaceInfo = 'var(--on-light-surface-info, #118CDF)';
/** Цвет фона поверхности/контрола информация на светлом фоне */
export var onLightSurfaceInfoHover = 'var(--on-light-surface-info-hover, #1798EEFF)';
/** Цвет фона поверхности/контрола информация на светлом фоне */
export var onLightSurfaceInfoActive = 'var(--on-light-surface-info-active, #1086D5FF)';
/** Минорный цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositiveMinor = 'var(--on-light-surface-positive-minor, #77F38E)';
/** Минорный цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositiveMinorHover = 'var(--on-light-surface-positive-minor-hover, #8AF59DFF)';
/** Минорный цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositiveMinorActive = 'var(--on-light-surface-positive-minor-active, #6EF286FF)';
/** Минорный цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceWarningMinor = 'var(--on-light-surface-warning-minor, #FFD1B8)';
/** Минорный цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceWarningMinorHover = 'var(--on-light-surface-warning-minor-hover, #FFDECCFF)';
/** Минорный цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceWarningMinorActive = 'var(--on-light-surface-warning-minor-active, #FFCAADFF)';
/** Минорный цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeMinor = 'var(--on-light-surface-negative-minor, #FECDD2)';
/** Минорный цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeMinorHover = 'var(--on-light-surface-negative-minor-hover, #FED7DBFF)';
/** Минорный цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeMinorActive = 'var(--on-light-surface-negative-minor-active, #FEC3C9FF)';
/** Минорный цвет фона поверхности/контрола информация на светлом фоне */
export var onLightSurfaceInfoMinor = 'var(--on-light-surface-info-minor, #B0DFFF)';
/** Минорный цвет фона поверхности/контрола информация на светлом фоне */
export var onLightSurfaceInfoMinorHover = 'var(--on-light-surface-info-minor-hover, #C7E9FFFF)';
/** Минорный цвет фона поверхности/контрола информация на светлом фоне */
export var onLightSurfaceInfoMinorActive = 'var(--on-light-surface-info-minor-active, #A8DCFFFF)';
/** Прозрачный цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfaceTransparentPositive = 'var(--on-light-surface-transparent-positive, #1A9E3247)';
/** Прозрачный цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfaceTransparentPositiveHover = 'var(--on-light-surface-transparent-positive-hover, #1A9E3266)';
/** Прозрачный цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfaceTransparentPositiveActive = 'var(--on-light-surface-transparent-positive-active, #1A9E3238)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceTransparentWarning = 'var(--on-light-surface-transparent-warning, #FA5F0547)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceTransparentWarningHover = 'var(--on-light-surface-transparent-warning-hover, #FA5F0566)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceTransparentWarningActive = 'var(--on-light-surface-transparent-warning-active, #FA5F0538)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceTransparentNegative = 'var(--on-light-surface-transparent-negative, #FF293E47)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceTransparentNegativeHover = 'var(--on-light-surface-transparent-negative-hover, #FF293E66)';
/** Прозрачный цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceTransparentNegativeActive = 'var(--on-light-surface-transparent-negative-active, #FF293E38)';
/** Прозрачный цвет фона поверхности/контрола информация на светлом фоне */
export var onLightSurfaceTransparentInfo = 'var(--on-light-surface-transparent-info, #118CDF47)';
/** Прозрачный цвет фона поверхности/контрола информация на светлом фоне */
export var onLightSurfaceTransparentInfoHover = 'var(--on-light-surface-transparent-info-hover, #118CDF66)';
/** Прозрачный цвет фона поверхности/контрола информация на светлом фоне */
export var onLightSurfaceTransparentInfoActive = 'var(--on-light-surface-transparent-info-active, #118CDF38)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidPrimary = 'var(--inverse-surface-solid-primary, #F5F5F5)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidPrimaryHover = 'var(--inverse-surface-solid-primary-hover, #FAFAFAFF)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidPrimaryActive = 'var(--inverse-surface-solid-primary-active, #F0F0F0FF)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidSecondary = 'var(--inverse-surface-solid-secondary, #ECECEC)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidSecondaryHover = 'var(--inverse-surface-solid-secondary-hover, #F7F7F7FF)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidSecondaryActive = 'var(--inverse-surface-solid-secondary-active, #E8E8E8FF)';
/** Инвертированный третичный непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidTertiary = 'var(--inverse-surface-solid-tertiary, #DDDDDD)';
/** Инвертированный третичный непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidTertiaryHover = 'var(--inverse-surface-solid-tertiary-hover, #E8E8E8FF)';
/** Инвертированный третичный непрозрачный фон поверхности/контрола */
export var inverseSurfaceSolidTertiaryActive = 'var(--inverse-surface-solid-tertiary-active, #D9D9D9FF)';
/** Инвертированный основной фон для карточек */
export var inverseSurfaceSolidCard = 'var(--inverse-surface-solid-card, #FFFFFF)';
/** Инвертированный основной фон для карточек */
export var inverseSurfaceSolidCardHover = 'var(--inverse-surface-solid-card-hover, #FFFFFFFF)';
/** Инвертированный основной фон для карточек */
export var inverseSurfaceSolidCardActive = 'var(--inverse-surface-solid-card-active, #FFFFFFFF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
export var inverseSurfaceSolidDefault = 'var(--inverse-surface-solid-default, #080808)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
export var inverseSurfaceSolidDefaultHover = 'var(--inverse-surface-solid-default-hover, #0D0D0DFF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
export var inverseSurfaceSolidDefaultActive = 'var(--inverse-surface-solid-default-active, #030303FF)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentPrimary = 'var(--inverse-surface-transparent-primary, #08080808)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentPrimaryHover = 'var(--inverse-surface-transparent-primary-hover, #08080812)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentPrimaryActive = 'var(--inverse-surface-transparent-primary-active, #08080803)';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentSecondary = 'var(--inverse-surface-transparent-secondary, #0808080F)';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentSecondaryHover = 'var(--inverse-surface-transparent-secondary-hover, #0808081F)';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentSecondaryActive = 'var(--inverse-surface-transparent-secondary-active, #0808080A)';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentTertiary = 'var(--inverse-surface-transparent-tertiary, #0808081F)';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentTertiaryHover = 'var(--inverse-surface-transparent-tertiary-hover, #0808083D)';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentTertiaryActive = 'var(--inverse-surface-transparent-tertiary-active, #0808080F)';
/** Инвертированный глубокий прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentDeep = 'var(--inverse-surface-transparent-deep, #080808A3)';
/** Инвертированный глубокий прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentDeepHover = 'var(--inverse-surface-transparent-deep-hover, #080808C2)';
/** Инвертированный глубокий прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentDeepActive = 'var(--inverse-surface-transparent-deep-active, #08080894)';
/** Инвертированный прозрачный фон для карточек */
export var inverseSurfaceTransparentCard = 'var(--inverse-surface-transparent-card, #0808080F)';
/** Инвертированный прозрачный фон для карточек */
export var inverseSurfaceTransparentCardHover = 'var(--inverse-surface-transparent-card-hover, #0808081F)';
/** Инвертированный прозрачный фон для карточек */
export var inverseSurfaceTransparentCardActive = 'var(--inverse-surface-transparent-card-active, #0808080A)';
/** Инвертированный фон поверхности/контрола без заливки */
export var inverseSurfaceClear = 'var(--inverse-surface-clear, #FFFFFF00)';
/** Инвертированный фон поверхности/контрола без заливки */
export var inverseSurfaceClearHover = 'var(--inverse-surface-clear-hover, #FFFFFF00)';
/** Инвертированный фон поверхности/контрола без заливки */
export var inverseSurfaceClearActive = 'var(--inverse-surface-clear-active, #FFFFFF00)';
/** Инвертированный акцентный фон поверхности/контрола */
export var inverseSurfaceAccent = 'var(--inverse-surface-accent, #108E26)';
/** Инвертированный акцентный фон поверхности/контрола */
export var inverseSurfaceAccentHover = 'var(--inverse-surface-accent-hover, #12A12AFF)';
/** Инвертированный акцентный фон поверхности/контрола */
export var inverseSurfaceAccentActive = 'var(--inverse-surface-accent-active, #0F8523FF)';
/** Инвертированный акцентный минорный непрозрачный фон поверхности/контрола */
export var inverseSurfaceAccentMinor = 'var(--inverse-surface-accent-minor, #77F38E)';
/** Инвертированный акцентный минорный непрозрачный фон поверхности/контрола */
export var inverseSurfaceAccentMinorHover = 'var(--inverse-surface-accent-minor-hover, #8AF59DFF)';
/** Инвертированный акцентный минорный непрозрачный фон поверхности/контрола */
export var inverseSurfaceAccentMinorActive = 'var(--inverse-surface-accent-minor-active, #6EF286FF)';
/** Прозрачный инвертированный акцентный фон поверхности/контрола */
export var inverseSurfaceTransparentAccent = 'var(--inverse-surface-transparent-accent, #108E2647)';
/** Прозрачный инвертированный акцентный фон поверхности/контрола */
export var inverseSurfaceTransparentAccentHover = 'var(--inverse-surface-transparent-accent-hover, #108E2566)';
/** Прозрачный инвертированный акцентный фон поверхности/контрола */
export var inverseSurfaceTransparentAccentActive = 'var(--inverse-surface-transparent-accent-active, #108E2538)';
/** Инвертированный цвет фона поверхности/контрола успех */
export var inverseSurfacePositive = 'var(--inverse-surface-positive, #1A9E32)';
/** Инвертированный цвет фона поверхности/контрола успех */
export var inverseSurfacePositiveHover = 'var(--inverse-surface-positive-hover, #1DAF37FF)';
/** Инвертированный цвет фона поверхности/контрола успех */
export var inverseSurfacePositiveActive = 'var(--inverse-surface-positive-active, #18952FFF)';
/** Инвертированный цвет фона поверхности/контрола предупреждение */
export var inverseSurfaceWarning = 'var(--inverse-surface-warning, #FA5F05)';
/** Инвертированный цвет фона поверхности/контрола предупреждение */
export var inverseSurfaceWarningHover = 'var(--inverse-surface-warning-hover, #FB7223FF)';
/** Инвертированный цвет фона поверхности/контрола предупреждение */
export var inverseSurfaceWarningActive = 'var(--inverse-surface-warning-active, #F05B05FF)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
export var inverseSurfaceNegative = 'var(--inverse-surface-negative, #FF293E)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
export var inverseSurfaceNegativeHover = 'var(--inverse-surface-negative-hover, #FF475AFF)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
export var inverseSurfaceNegativeActive = 'var(--inverse-surface-negative-active, #FF1F35FF)';
/** Инвертированный цвет фона поверхности/контрола информация */
export var inverseSurfaceInfo = 'var(--inverse-surface-info, #118CDF)';
/** Инвертированный цвет фона поверхности/контрола информация */
export var inverseSurfaceInfoHover = 'var(--inverse-surface-info-hover, #1798EEFF)';
/** Инвертированный цвет фона поверхности/контрола информация */
export var inverseSurfaceInfoActive = 'var(--inverse-surface-info-active, #1086D5FF)';
/** Инвертированный минорный цвет фона поверхности/контрола успех */
export var inverseSurfacePositiveMinor = 'var(--inverse-surface-positive-minor, #77F38E)';
/** Инвертированный минорный цвет фона поверхности/контрола успех */
export var inverseSurfacePositiveMinorHover = 'var(--inverse-surface-positive-minor-hover, #8AF59DFF)';
/** Инвертированный минорный цвет фона поверхности/контрола успех */
export var inverseSurfacePositiveMinorActive = 'var(--inverse-surface-positive-minor-active, #6EF286FF)';
/** Инвертированный минорный цвет фона поверхности/контрола предупреждение */
export var inverseSurfaceWarningMinor = 'var(--inverse-surface-warning-minor, #FFD1B8)';
/** Инвертированный минорный цвет фона поверхности/контрола предупреждение */
export var inverseSurfaceWarningMinorHover = 'var(--inverse-surface-warning-minor-hover, #FFDECCFF)';
/** Инвертированный минорный цвет фона поверхности/контрола предупреждение */
export var inverseSurfaceWarningMinorActive = 'var(--inverse-surface-warning-minor-active, #FFCAADFF)';
/** Инвертированный минорный цвет фона поверхности/контрола ошибка */
export var inverseSurfaceNegativeMinor = 'var(--inverse-surface-negative-minor, #FECDD2)';
/** Инвертированный минорный цвет фона поверхности/контрола ошибка */
export var inverseSurfaceNegativeMinorHover = 'var(--inverse-surface-negative-minor-hover, #FED7DBFF)';
/** Инвертированный минорный цвет фона поверхности/контрола ошибка */
export var inverseSurfaceNegativeMinorActive = 'var(--inverse-surface-negative-minor-active, #FEC3C9FF)';
/** Инвертированный минорный цвет фона поверхности/контрола информация */
export var inverseSurfaceInfoMinor = 'var(--inverse-surface-info-minor, #B0DFFF)';
/** Инвертированный минорный цвет фона поверхности/контрола информация */
export var inverseSurfaceInfoMinorHover = 'var(--inverse-surface-info-minor-hover, #C7E9FFFF)';
/** Инвертированный минорный цвет фона поверхности/контрола информация */
export var inverseSurfaceInfoMinorActive = 'var(--inverse-surface-info-minor-active, #A8DCFFFF)';
/** Прозрачный инвертированный цвет фона поверхности/контрола успех */
export var inverseSurfaceTransparentPositive = 'var(--inverse-surface-transparent-positive, #1A9E3247)';
/** Прозрачный инвертированный цвет фона поверхности/контрола успех */
export var inverseSurfaceTransparentPositiveHover = 'var(--inverse-surface-transparent-positive-hover, #1A9E3266)';
/** Прозрачный инвертированный цвет фона поверхности/контрола успех */
export var inverseSurfaceTransparentPositiveActive = 'var(--inverse-surface-transparent-positive-active, #1A9E3238)';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
export var inverseSurfaceTransparentWarning = 'var(--inverse-surface-transparent-warning, #FA5F0547)';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
export var inverseSurfaceTransparentWarningHover = 'var(--inverse-surface-transparent-warning-hover, #FA5F0566)';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
export var inverseSurfaceTransparentWarningActive = 'var(--inverse-surface-transparent-warning-active, #FA5F0538)';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
export var inverseSurfaceTransparentNegative = 'var(--inverse-surface-transparent-negative, #FF293E47)';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
export var inverseSurfaceTransparentNegativeHover = 'var(--inverse-surface-transparent-negative-hover, #FF293E66)';
/** Прозрачный инвертированный цвет фона поверхности/контрола предупреждение */
export var inverseSurfaceTransparentNegativeActive = 'var(--inverse-surface-transparent-negative-active, #FF293E38)';
/** Прозрачный инвертированный цвет фона поверхности/контрола информация */
export var inverseSurfaceTransparentInfo = 'var(--inverse-surface-transparent-info, #118CDF47)';
/** Прозрачный инвертированный цвет фона поверхности/контрола информация */
export var inverseSurfaceTransparentInfoHover = 'var(--inverse-surface-transparent-info-hover, #118CDF66)';
/** Прозрачный инвертированный цвет фона поверхности/контрола информация */
export var inverseSurfaceTransparentInfoActive = 'var(--inverse-surface-transparent-info-active, #118CDF38)';
/** Основной фон */
export var backgroundPrimary = 'var(--background-primary, #080808)';
/** Основной фон на темном фоне */
export var darkBackgroundPrimary = 'var(--dark-background-primary, #080808)';
/** Основной фон на светлом фоне */
export var lightBackgroundPrimary = 'var(--light-background-primary, #F9F9F9)';
/** Инвертированный основной фон */
export var inverseBackgroundPrimary = 'var(--inverse-background-primary, #F9F9F9)';
/** Цвет фона паранжи светлый */
export var overlaySoft = 'var(--overlay-soft, #0808088F)';
/** Цвет фона паранжи темный */
export var overlayHard = 'var(--overlay-hard, #080808F5)';
/** Цвет фона паранжи размытый */
export var overlayBlur = 'var(--overlay-blur, #08080833)';
/** Цвет фона паранжи светлый на темном фоне */
export var onDarkOverlaySoft = 'var(--on-dark-overlay-soft, #0808088F)';
/** Цвет фона паранжи темный на темном фоне */
export var onDarkOverlayHard = 'var(--on-dark-overlay-hard, #080808F5)';
/** Цвет фона паранжи размытый на темном фоне */
export var onDarkOverlayBlur = 'var(--on-dark-overlay-blur, #08080833)';
/** Цвет фона паранжи светлый на светлом фоне */
export var onLightOverlaySoft = 'var(--on-light-overlay-soft, #F9F9F98F)';
/** Цвет фона паранжи темный на светлом фоне */
export var onLightOverlayHard = 'var(--on-light-overlay-hard, #F9F9F9F5)';
/** Цвет фона паранжи размытый на светлом фоне */
export var onLightOverlayBlur = 'var(--on-light-overlay-blur, #F9F9F933)';
/** Инвертированный цвет фона паранжи светлый */
export var inverseOverlaySoft = 'var(--inverse-overlay-soft, #F9F9F98F)';
/** Инвертированный цвет фона паранжи темный */
export var inverseOverlayHard = 'var(--inverse-overlay-hard, #F9F9F9F5)';
/** Инвертированный цвет фона паранжи размытый */
export var inverseOverlayBlur = 'var(--inverse-overlay-blur, #F9F9F933)';
/** Основной непрозрачный цвет обводки */
export var outlineDefaultOutlineSolidPrimary = 'var(--outline-default-outline-solid-primary, #262626)';
/** Основной непрозрачный цвет обводки */
export var outlineDefaultOutlineSolidPrimaryHover = 'var(--outline-default-outline-solid-primary-hover, #FFFFFFFF)';
/** Основной непрозрачный цвет обводки */
export var outlineDefaultOutlineSolidPrimaryActive = 'var(--outline-default-outline-solid-primary-active, #ADADADFF)';
/** Вторичный непрозрачный цвет обводки */
export var outlineDefaultOutlineSolidSecondary = 'var(--outline-default-outline-solid-secondary, #363636)';
/** Вторичный непрозрачный цвет обводки */
export var outlineDefaultOutlineSolidSecondaryHover = 'var(--outline-default-outline-solid-secondary-hover, #FFFFFFFF)';
/** Вторичный непрозрачный цвет обводки */
export var outlineDefaultOutlineSolidSecondaryActive = 'var(--outline-default-outline-solid-secondary-active, #A1A1A1FF)';
/** Третичный непрозрачный цвет обводки */
export var outlineDefaultOutlineSolidTertiary = 'var(--outline-default-outline-solid-tertiary, #4E4E4E)';
/** Третичный непрозрачный цвет обводки */
export var outlineDefaultOutlineSolidTertiaryHover = 'var(--outline-default-outline-solid-tertiary-hover, #FFFFFFFF)';
/** Третичный непрозрачный цвет обводки */
export var outlineDefaultOutlineSolidTertiaryActive = 'var(--outline-default-outline-solid-tertiary-active, #8C8C8CFF)';
/** Прозрачный цвет обводки по умолчанию */
export var outlineDefaultOutlineTransparentDefault = 'var(--outline-default-outline-transparent-default, #F9F9F9F5)';
/** Прозрачный цвет обводки по умолчанию */
export var outlineDefaultOutlineTransparentDefaultHover = 'var(--outline-default-outline-transparent-default-hover, #FAFAFA93)';
/** Прозрачный цвет обводки по умолчанию */
export var outlineDefaultOutlineTransparentDefaultActive = 'var(--outline-default-outline-transparent-default-active, #FAFAFAC4)';
/** Основной прозрачный цвет обводки */
export var outlineDefaultOutlineTransparentPrimary = 'var(--outline-default-outline-transparent-primary, #F9F9F91F)';
/** Основной прозрачный цвет обводки */
export var outlineDefaultOutlineTransparentPrimaryHover = 'var(--outline-default-outline-transparent-primary-hover, #FAFAFAFF)';
/** Основной прозрачный цвет обводки */
export var outlineDefaultOutlineTransparentPrimaryActive = 'var(--outline-default-outline-transparent-primary-active, #FAFAFA25)';
/** Вторичный прозрачный цвет обводки */
export var outlineDefaultOutlineTransparentSecondary = 'var(--outline-default-outline-transparent-secondary, #F9F9F933)';
/** Вторичный прозрачный цвет обводки */
export var outlineDefaultOutlineTransparentSecondaryHover = 'var(--outline-default-outline-transparent-secondary-hover, #FAFAFAFF)';
/** Вторичный прозрачный цвет обводки */
export var outlineDefaultOutlineTransparentSecondaryActive = 'var(--outline-default-outline-transparent-secondary-active, #FAFAFA3D)';
/** Третичный прозрачный цвет обводки */
export var outlineDefaultOutlineTransparentTertiary = 'var(--outline-default-outline-transparent-tertiary, #F9F9F947)';
/** Третичный прозрачный цвет обводки */
export var outlineDefaultOutlineTransparentTertiaryHover = 'var(--outline-default-outline-transparent-tertiary-hover, #FAFAFAFF)';
/** Третичный прозрачный цвет обводки */
export var outlineDefaultOutlineTransparentTertiaryActive = 'var(--outline-default-outline-transparent-tertiary-active, #FAFAFA56)';
/** Бесцветная обводка */
export var outlineDefaultOutlineClear = 'var(--outline-default-outline-clear, #FFFFFF00)';
/** Бесцветная обводка */
export var outlineDefaultOutlineClearHover = 'var(--outline-default-outline-clear-hover, #FFFFFF00)';
/** Бесцветная обводка */
export var outlineDefaultOutlineClearActive = 'var(--outline-default-outline-clear-active, #FFFFFF00)';
/** Акцентный цвет обводки */
export var outlineDefaultOutlineAccent = 'var(--outline-default-outline-accent, #1A9E32)';
/** Акцентный цвет обводки */
export var outlineDefaultOutlineAccentHover = 'var(--outline-default-outline-accent-hover, #1EB83AFF)';
/** Акцентный цвет обводки */
export var outlineDefaultOutlineAccentActive = 'var(--outline-default-outline-accent-active, #15842AFF)';
/** Акцентный минорный непрозрачный цвет обводки */
export var outlineDefaultOutlineAccentMinor = 'var(--outline-default-outline-accent-minor, #0A4014)';
/** Акцентный минорный непрозрачный цвет обводки */
export var outlineDefaultOutlineAccentMinorHover = 'var(--outline-default-outline-accent-minor-hover, #137725FF)';
/** Акцентный минорный непрозрачный цвет обводки */
export var outlineDefaultOutlineAccentMinorActive = 'var(--outline-default-outline-accent-minor-active, #0E5D1DFF)';
/** Прозрачный акцентный цвет обводки */
export var outlineDefaultOutlineTransparentAccent = 'var(--outline-default-outline-transparent-accent, #1A9E3266)';
/** Прозрачный акцентный цвет обводки */
export var outlineDefaultOutlineTransparentAccentHover = 'var(--outline-default-outline-transparent-accent-hover, #1A9E32FF)';
/** Прозрачный акцентный цвет обводки */
export var outlineDefaultOutlineTransparentAccentActive = 'var(--outline-default-outline-transparent-accent-active, #1A9E327A)';
/** Цвет обводки успех */
export var outlineDefaultOutlinePositive = 'var(--outline-default-outline-positive, #1A9E32)';
/** Цвет обводки успех */
export var outlineDefaultOutlinePositiveHover = 'var(--outline-default-outline-positive-hover, #1EB83AFF)';
/** Цвет обводки успех */
export var outlineDefaultOutlinePositiveActive = 'var(--outline-default-outline-positive-active, #15842AFF)';
/** Цвет обводки предупреждение */
export var outlineDefaultOutlineWarning = 'var(--outline-default-outline-warning, #FA5F05)';
/** Цвет обводки предупреждение */
export var outlineDefaultOutlineWarningHover = 'var(--outline-default-outline-warning-hover, #FB7223FF)';
/** Цвет обводки предупреждение */
export var outlineDefaultOutlineWarningActive = 'var(--outline-default-outline-warning-active, #DC5304FF)';
/** Цвет обводки ошибка */
export var outlineDefaultOutlineNegative = 'var(--outline-default-outline-negative, #FF293E)';
/** Цвет обводки ошибка */
export var outlineDefaultOutlineNegativeHover = 'var(--outline-default-outline-negative-hover, #FF475AFF)';
/** Цвет обводки ошибка */
export var outlineDefaultOutlineNegativeActive = 'var(--outline-default-outline-negative-active, #FF0A23FF)';
/** Цвет обводки информация */
export var outlineDefaultOutlineInfo = 'var(--outline-default-outline-info, #118CDF)';
/** Цвет обводки информация */
export var outlineDefaultOutlineInfoHover = 'var(--outline-default-outline-info-hover, #50B1F2FF)';
/** Цвет обводки информация */
export var outlineDefaultOutlineInfoActive = 'var(--outline-default-outline-info-active, #0F7BC2FF)';
/** Минорный цвет обводки успех */
export var outlineDefaultOutlinePositiveMinor = 'var(--outline-default-outline-positive-minor, #0A4014)';
/** Минорный цвет обводки успех */
export var outlineDefaultOutlinePositiveMinorHover = 'var(--outline-default-outline-positive-minor-hover, #137725FF)';
/** Минорный цвет обводки успех */
export var outlineDefaultOutlinePositiveMinorActive = 'var(--outline-default-outline-positive-minor-active, #0E5D1DFF)';
/** Минорный цвет обводки предупреждение */
export var outlineDefaultOutlineWarningMinor = 'var(--outline-default-outline-warning-minor, #5C2709)';
/** Минорный цвет обводки предупреждение */
export var outlineDefaultOutlineWarningMinorHover = 'var(--outline-default-outline-warning-minor-hover, #95400FFF)';
/** Минорный цвет обводки предупреждение */
export var outlineDefaultOutlineWarningMinorActive = 'var(--outline-default-outline-warning-minor-active, #79340CFF)';
/** Минорный цвет обводки ошибка */
export var outlineDefaultOutlineNegativeMinor = 'var(--outline-default-outline-negative-minor, #6B1019)';
/** Минорный цвет обводки ошибка */
export var outlineDefaultOutlineNegativeMinorHover = 'var(--outline-default-outline-negative-minor-hover, #A01825FF)';
/** Минорный цвет обводки ошибка */
export var outlineDefaultOutlineNegativeMinorActive = 'var(--outline-default-outline-negative-minor-active, #85141FFF)';
/** Минорный цвет обводки информация */
export var outlineDefaultOutlineInfoMinor = 'var(--outline-default-outline-info-minor, #0D3B59)';
/** Минорный цвет обводки информация */
export var outlineDefaultOutlineInfoMinorHover = 'var(--outline-default-outline-info-minor-hover, #145E8FFF)';
/** Минорный цвет обводки информация */
export var outlineDefaultOutlineInfoMinorActive = 'var(--outline-default-outline-info-minor-active, #114C74FF)';
/** Прозрачный цвет обводки успех */
export var outlineDefaultOutlineTransparentPositive = 'var(--outline-default-outline-transparent-positive, #1A9E3247)';
/** Прозрачный цвет обводки успех */
export var outlineDefaultOutlineTransparentPositiveHover = 'var(--outline-default-outline-transparent-positive-hover, #1A9E32FF)';
/** Прозрачный цвет обводки успех */
export var outlineDefaultOutlineTransparentPositiveActive = 'var(--outline-default-outline-transparent-positive-active, #1A9E3256)';
/** Прозрачный цвет обводки предупреждение */
export var outlineDefaultOutlineTransparentWarning = 'var(--outline-default-outline-transparent-warning, #FA5F0547)';
/** Прозрачный цвет обводки предупреждение */
export var outlineDefaultOutlineTransparentWarningHover = 'var(--outline-default-outline-transparent-warning-hover, #FA5F05FF)';
/** Прозрачный цвет обводки предупреждение */
export var outlineDefaultOutlineTransparentWarningActive = 'var(--outline-default-outline-transparent-warning-active, #FA5F0556)';
/** Прозрачный цвет обводки предупреждение */
export var outlineDefaultOutlineTransparentNegative = 'var(--outline-default-outline-transparent-negative, #FF293E47)';
/** Прозрачный цвет обводки предупреждение */
export var outlineDefaultOutlineTransparentNegativeHover = 'var(--outline-default-outline-transparent-negative-hover, #FF293EFF)';
/** Прозрачный цвет обводки предупреждение */
export var outlineDefaultOutlineTransparentNegativeActive = 'var(--outline-default-outline-transparent-negative-active, #FF293E56)';
/** Прозрачный цвет обводки информация */
export var outlineDefaultOutlineTransparentInfo = 'var(--outline-default-outline-transparent-info, #118CDF47)';
/** Прозрачный цвет обводки информация */
export var outlineDefaultOutlineTransparentInfoHover = 'var(--outline-default-outline-transparent-info-hover, #118CDFFF)';
/** Прозрачный цвет обводки информация */
export var outlineDefaultOutlineTransparentInfoActive = 'var(--outline-default-outline-transparent-info-active, #118CDF56)';
/** Основной непрозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineSolidPrimary = 'var(--outline-on-dark-outline-solid-primary, #262626)';
/** Основной непрозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineSolidPrimaryHover = 'var(--outline-on-dark-outline-solid-primary-hover, #FFFFFFFF)';
/** Основной непрозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineSolidPrimaryActive = 'var(--outline-on-dark-outline-solid-primary-active, #ADADADFF)';
/** Вторичный непрозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineSolidSecondary = 'var(--outline-on-dark-outline-solid-secondary, #363636)';
/** Вторичный непрозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineSolidSecondaryHover = 'var(--outline-on-dark-outline-solid-secondary-hover, #FFFFFFFF)';
/** Вторичный непрозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineSolidSecondaryActive = 'var(--outline-on-dark-outline-solid-secondary-active, #A1A1A1FF)';
/** Третичный непрозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineSolidTertiary = 'var(--outline-on-dark-outline-solid-tertiary, #4E4E4E)';
/** Третичный непрозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineSolidTertiaryHover = 'var(--outline-on-dark-outline-solid-tertiary-hover, #FFFFFFFF)';
/** Третичный непрозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineSolidTertiaryActive = 'var(--outline-on-dark-outline-solid-tertiary-active, #8C8C8CFF)';
/** Прозрачный цвет обводки по умолчанию на темном фоне */
export var outlineOnDarkOutlineTransparentDefault = 'var(--outline-on-dark-outline-transparent-default, #F9F9F9F5)';
/** Прозрачный цвет обводки по умолчанию на темном фоне */
export var outlineOnDarkOutlineTransparentDefaultHover = 'var(--outline-on-dark-outline-transparent-default-hover, #FAFAFA93)';
/** Прозрачный цвет обводки по умолчанию на темном фоне */
export var outlineOnDarkOutlineTransparentDefaultActive = 'var(--outline-on-dark-outline-transparent-default-active, #FAFAFAC4)';
/** Основной прозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineTransparentPrimary = 'var(--outline-on-dark-outline-transparent-primary, #F9F9F91F)';
/** Основной прозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineTransparentPrimaryHover = 'var(--outline-on-dark-outline-transparent-primary-hover, #FAFAFAFF)';
/** Основной прозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineTransparentPrimaryActive = 'var(--outline-on-dark-outline-transparent-primary-active, #FAFAFA25)';
/** Вторичный прозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineTransparentSecondary = 'var(--outline-on-dark-outline-transparent-secondary, #F9F9F933)';
/** Вторичный прозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineTransparentSecondaryHover = 'var(--outline-on-dark-outline-transparent-secondary-hover, #FAFAFAFF)';
/** Вторичный прозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineTransparentSecondaryActive = 'var(--outline-on-dark-outline-transparent-secondary-active, #FAFAFA3D)';
/** Третичный прозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineTransparentTertiary = 'var(--outline-on-dark-outline-transparent-tertiary, #F9F9F947)';
/** Третичный прозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineTransparentTertiaryHover = 'var(--outline-on-dark-outline-transparent-tertiary-hover, #FAFAFAFF)';
/** Третичный прозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineTransparentTertiaryActive = 'var(--outline-on-dark-outline-transparent-tertiary-active, #FAFAFA56)';
/** Бесцветная обводка на темном фоне */
export var outlineOnDarkOutlineClear = 'var(--outline-on-dark-outline-clear, #FFFFFF00)';
/** Бесцветная обводка на темном фоне */
export var outlineOnDarkOutlineClearHover = 'var(--outline-on-dark-outline-clear-hover, #FFFFFF00)';
/** Бесцветная обводка на темном фоне */
export var outlineOnDarkOutlineClearActive = 'var(--outline-on-dark-outline-clear-active, #FFFFFF00)';
/** Акцентный цвет обводки на темном фоне */
export var outlineOnDarkOutlineAccent = 'var(--outline-on-dark-outline-accent, #1A9E32)';
/** Акцентный цвет обводки на темном фоне */
export var outlineOnDarkOutlineAccentHover = 'var(--outline-on-dark-outline-accent-hover, #1EB83AFF)';
/** Акцентный цвет обводки на темном фоне */
export var outlineOnDarkOutlineAccentActive = 'var(--outline-on-dark-outline-accent-active, #15842AFF)';
/** Акцентный минорный непрозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineAccentMinor = 'var(--outline-on-dark-outline-accent-minor, #0A4014)';
/** Акцентный минорный непрозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineAccentMinorHover = 'var(--outline-on-dark-outline-accent-minor-hover, #137725FF)';
/** Акцентный минорный непрозрачный цвет обводки на темном фоне */
export var outlineOnDarkOutlineAccentMinorActive = 'var(--outline-on-dark-outline-accent-minor-active, #0E5D1DFF)';
/** Прозрачный акцентный цвет обводки на темном фоне */
export var outlineOnDarkOutlineTransparentAccent = 'var(--outline-on-dark-outline-transparent-accent, #1A9E3266)';
/** Прозрачный акцентный цвет обводки на темном фоне */
export var outlineOnDarkOutlineTransparentAccentHover = 'var(--outline-on-dark-outline-transparent-accent-hover, #1A9E32FF)';
/** Прозрачный акцентный цвет обводки на темном фоне */
export var outlineOnDarkOutlineTransparentAccentActive = 'var(--outline-on-dark-outline-transparent-accent-active, #1A9E327A)';
/** Цвет обводки успех на темном фоне */
export var outlineOnDarkOutlinePositive = 'var(--outline-on-dark-outline-positive, #1A9E32)';
/** Цвет обводки успех на темном фоне */
export var outlineOnDarkOutlinePositiveHover = 'var(--outline-on-dark-outline-positive-hover, #1EB83AFF)';
/** Цвет обводки успех на темном фоне */
export var outlineOnDarkOutlinePositiveActive = 'var(--outline-on-dark-outline-positive-active, #15842AFF)';
/** Цвет обводки предупреждение на темном фоне */
export var outlineOnDarkOutlineWarning = 'var(--outline-on-dark-outline-warning, #FA5F05)';
/** Цвет обводки предупреждение на темном фоне */
export var outlineOnDarkOutlineWarningHover = 'var(--outline-on-dark-outline-warning-hover, #FB7223FF)';
/** Цвет обводки предупреждение на темном фоне */
export var outlineOnDarkOutlineWarningActive = 'var(--outline-on-dark-outline-warning-active, #DC5304FF)';
/** Цвет обводки ошибка на темном фоне */
export var outlineOnDarkOutlineNegative = 'var(--outline-on-dark-outline-negative, #FF293E)';
/** Цвет обводки ошибка на темном фоне */
export var outlineOnDarkOutlineNegativeHover = 'var(--outline-on-dark-outline-negative-hover, #FF475AFF)';
/** Цвет обводки ошибка на темном фоне */
export var outlineOnDarkOutlineNegativeActive = 'var(--outline-on-dark-outline-negative-active, #FF0A23FF)';
/** Цвет обводки информация на темном фоне */
export var outlineOnDarkOutlineInfo = 'var(--outline-on-dark-outline-info, #118CDF)';
/** Цвет обводки информация на темном фоне */
export var outlineOnDarkOutlineInfoHover = 'var(--outline-on-dark-outline-info-hover, #50B1F2FF)';
/** Цвет обводки информация на темном фоне */
export var outlineOnDarkOutlineInfoActive = 'var(--outline-on-dark-outline-info-active, #0F7BC2FF)';
/** Минорный цвет обводки успех на темном фоне */
export var outlineOnDarkOutlinePositiveMinor = 'var(--outline-on-dark-outline-positive-minor, #0A4014)';
/** Минорный цвет обводки успех на темном фоне */
export var outlineOnDarkOutlinePositiveMinorHover = 'var(--outline-on-dark-outline-positive-minor-hover, #137725FF)';
/** Минорный цвет обводки успех на темном фоне */
export var outlineOnDarkOutlinePositiveMinorActive = 'var(--outline-on-dark-outline-positive-minor-active, #0E5D1DFF)';
/** Минорный цвет обводки предупреждение на темном фоне */
export var outlineOnDarkOutlineWarningMinor = 'var(--outline-on-dark-outline-warning-minor, #5C2709)';
/** Минорный цвет обводки предупреждение на темном фоне */
export var outlineOnDarkOutlineWarningMinorHover = 'var(--outline-on-dark-outline-warning-minor-hover, #95400FFF)';
/** Минорный цвет обводки предупреждение на темном фоне */
export var outlineOnDarkOutlineWarningMinorActive = 'var(--outline-on-dark-outline-warning-minor-active, #79340CFF)';
/** Минорный цвет обводки ошибка на темном фоне */
export var outlineOnDarkOutlineNegativeMinor = 'var(--outline-on-dark-outline-negative-minor, #6B1019)';
/** Минорный цвет обводки ошибка на темном фоне */
export var outlineOnDarkOutlineNegativeMinorHover = 'var(--outline-on-dark-outline-negative-minor-hover, #A01825FF)';
/** Минорный цвет обводки ошибка на темном фоне */
export var outlineOnDarkOutlineNegativeMinorActive = 'var(--outline-on-dark-outline-negative-minor-active, #85141FFF)';
/** Минорный цвет обводки информация на темном фоне */
export var outlineOnDarkOutlineInfoMinor = 'var(--outline-on-dark-outline-info-minor, #0D3B59)';
/** Минорный цвет обводки информация на темном фоне */
export var outlineOnDarkOutlineInfoMinorHover = 'var(--outline-on-dark-outline-info-minor-hover, #145E8FFF)';
/** Минорный цвет обводки информация на темном фоне */
export var outlineOnDarkOutlineInfoMinorActive = 'var(--outline-on-dark-outline-info-minor-active, #114C74FF)';
/** Прозрачный цвет обводки успех на темном фоне */
export var outlineOnDarkOutlineTransparentPositive = 'var(--outline-on-dark-outline-transparent-positive, #1A9E3247)';
/** Прозрачный цвет обводки успех на темном фоне */
export var outlineOnDarkOutlineTransparentPositiveHover = 'var(--outline-on-dark-outline-transparent-positive-hover, #1A9E32FF)';
/** Прозрачный цвет обводки успех на темном фоне */
export var outlineOnDarkOutlineTransparentPositiveActive = 'var(--outline-on-dark-outline-transparent-positive-active, #1A9E3256)';
/** Прозрачный цвет обводки предупреждение на темном фоне */
export var outlineOnDarkOutlineTransparentWarning = 'var(--outline-on-dark-outline-transparent-warning, #FA5F0547)';
/** Прозрачный цвет обводки предупреждение на темном фоне */
export var outlineOnDarkOutlineTransparentWarningHover = 'var(--outline-on-dark-outline-transparent-warning-hover, #FA5F05FF)';
/** Прозрачный цвет обводки предупреждение на темном фоне */
export var outlineOnDarkOutlineTransparentWarningActive = 'var(--outline-on-dark-outline-transparent-warning-active, #FA5F0556)';
/** Прозрачный цвет обводки предупреждение на темном фоне */
export var outlineOnDarkOutlineTransparentNegative = 'var(--outline-on-dark-outline-transparent-negative, #FF293E47)';
/** Прозрачный цвет обводки предупреждение на темном фоне */
export var outlineOnDarkOutlineTransparentNegativeHover = 'var(--outline-on-dark-outline-transparent-negative-hover, #FF293EFF)';
/** Прозрачный цвет обводки предупреждение на темном фоне */
export var outlineOnDarkOutlineTransparentNegativeActive = 'var(--outline-on-dark-outline-transparent-negative-active, #FF293E56)';
/** Прозрачный цвет обводки информация на темном фоне */
export var outlineOnDarkOutlineTransparentInfo = 'var(--outline-on-dark-outline-transparent-info, #118CDF47)';
/** Прозрачный цвет обводки информация на темном фоне */
export var outlineOnDarkOutlineTransparentInfoHover = 'var(--outline-on-dark-outline-transparent-info-hover, #118CDFFF)';
/** Прозрачный цвет обводки информация на темном фоне */
export var outlineOnDarkOutlineTransparentInfoActive = 'var(--outline-on-dark-outline-transparent-info-active, #118CDF56)';
/** Основной непрозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineSolidPrimary = 'var(--outline-on-light-outline-solid-primary, #DDDDDD)';
/** Основной непрозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineSolidPrimaryHover = 'var(--outline-on-light-outline-solid-primary-hover, #ABABABFF)';
/** Основной непрозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineSolidPrimaryActive = 'var(--outline-on-light-outline-solid-primary-active, #C4C4C4FF)';
/** Вторичный непрозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineSolidSecondary = 'var(--outline-on-light-outline-solid-secondary, #B3B3B3)';
/** Вторичный непрозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineSolidSecondaryHover = 'var(--outline-on-light-outline-solid-secondary-hover, #FFFFFFFF)';
/** Вторичный непрозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineSolidSecondaryActive = 'var(--outline-on-light-outline-solid-secondary-active, #3D3D3DFF)';
/** Третичный непрозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineSolidTertiary = 'var(--outline-on-light-outline-solid-tertiary, #707070)';
/** Третичный непрозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineSolidTertiaryHover = 'var(--outline-on-light-outline-solid-tertiary-hover, #FFFFFFFF)';
/** Третичный непрозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineSolidTertiaryActive = 'var(--outline-on-light-outline-solid-tertiary-active, #737373FF)';
/** Прозрачный цвет обводки по умолчанию на светлом фоне */
export var outlineOnLightOutlineTransparentDefault = 'var(--outline-on-light-outline-transparent-default, #080808F5)';
/** Прозрачный цвет обводки по умолчанию на светлом фоне */
export var outlineOnLightOutlineTransparentDefaultHover = 'var(--outline-on-light-outline-transparent-default-hover, #08080893)';
/** Прозрачный цвет обводки по умолчанию на светлом фоне */
export var outlineOnLightOutlineTransparentDefaultActive = 'var(--outline-on-light-outline-transparent-default-active, #080808C4)';
/** Основной прозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineTransparentPrimary = 'var(--outline-on-light-outline-transparent-primary, #0808081F)';
/** Основной прозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineTransparentPrimaryHover = 'var(--outline-on-light-outline-transparent-primary-hover, #080808FF)';
/** Основной прозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineTransparentPrimaryActive = 'var(--outline-on-light-outline-transparent-primary-active, #08080825)';
/** Вторичный прозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineTransparentSecondary = 'var(--outline-on-light-outline-transparent-secondary, #08080847)';
/** Вторичный прозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineTransparentSecondaryHover = 'var(--outline-on-light-outline-transparent-secondary-hover, #080808FF)';
/** Вторичный прозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineTransparentSecondaryActive = 'var(--outline-on-light-outline-transparent-secondary-active, #08080856)';
/** Третичный прозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineTransparentTertiary = 'var(--outline-on-light-outline-transparent-tertiary, #0808088F)';
/** Третичный прозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineTransparentTertiaryHover = 'var(--outline-on-light-outline-transparent-tertiary-hover, #080808FF)';
/** Третичный прозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineTransparentTertiaryActive = 'var(--outline-on-light-outline-transparent-tertiary-active, #080808AB)';
/** Бесцветная обводка на светлом фоне */
export var outlineOnLightOutlineClear = 'var(--outline-on-light-outline-clear, #FFFFFF00)';
/** Бесцветная обводка на светлом фоне */
export var outlineOnLightOutlineClearHover = 'var(--outline-on-light-outline-clear-hover, #FFFFFF00)';
/** Бесцветная обводка на светлом фоне */
export var outlineOnLightOutlineClearActive = 'var(--outline-on-light-outline-clear-active, #FFFFFF00)';
/** Акцентный цвет обводки на светлом фоне */
export var outlineOnLightOutlineAccent = 'var(--outline-on-light-outline-accent, #108E26)';
/** Акцентный цвет обводки на светлом фоне */
export var outlineOnLightOutlineAccentHover = 'var(--outline-on-light-outline-accent-hover, #13AA2CFF)';
/** Акцентный цвет обводки на светлом фоне */
export var outlineOnLightOutlineAccentActive = 'var(--outline-on-light-outline-accent-active, #0D731EFF)';
/** Акцентный минорный непрозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineAccentMinor = 'var(--outline-on-light-outline-accent-minor, #45E362)';
/** Акцентный минорный непрозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineAccentMinorHover = 'var(--outline-on-light-outline-accent-minor-hover, #5FE778FF)';
/** Акцентный минорный непрозрачный цвет обводки на светлом фоне */
export var outlineOnLightOutlineAccentMinorActive = 'var(--outline-on-light-outline-accent-minor-active, #2ADF4BFF)';
/** Прозрачный акцентный цвет обводки на светлом фоне */
export var outlineOnLightOutlineTransparentAccent = 'var(--outline-on-light-outline-transparent-accent, #108E2666)';
/** Прозрачный акцентный цвет обводки на светлом фоне */
export var outlineOnLightOutlineTransparentAccentHover = 'var(--outline-on-light-outline-transparent-accent-hover, #108E25FF)';
/** Прозрачный акцентный цвет обводки на светлом фоне */
export var outlineOnLightOutlineTransparentAccentActive = 'var(--outline-on-light-outline-transparent-accent-active, #108E257A)';
/** Цвет обводки успех на светлом фоне */
export var outlineOnLightOutlinePositive = 'var(--outline-on-light-outline-positive, #108E26)';
/** Цвет обводки успех на светлом фоне */
export var outlineOnLightOutlinePositiveHover = 'var(--outline-on-light-outline-positive-hover, #13AA2CFF)';
/** Цвет обводки успех на светлом фоне */
export var outlineOnLightOutlinePositiveActive = 'var(--outline-on-light-outline-positive-active, #0D731EFF)';
/** Цвет обводки предупреждение на светлом фоне */
export var outlineOnLightOutlineWarning = 'var(--outline-on-light-outline-warning, #E85702)';
/** Цвет обводки предупреждение на светлом фоне */
export var outlineOnLightOutlineWarningHover = 'var(--outline-on-light-outline-warning-hover, #FD650DFF)';
/** Цвет обводки предупреждение на светлом фоне */
export var outlineOnLightOutlineWarningActive = 'var(--outline-on-light-outline-warning-active, #CA4B02FF)';
/** Цвет обводки ошибка на светлом фоне */
export var outlineOnLightOutlineNegative = 'var(--outline-on-light-outline-negative, #F31B31)';
/** Цвет обводки ошибка на светлом фоне */
export var outlineOnLightOutlineNegativeHover = 'var(--outline-on-light-outline-negative-hover, #F5384BFF)';
/** Цвет обводки ошибка на светлом фоне */
export var outlineOnLightOutlineNegativeActive = 'var(--outline-on-light-outline-negative-active, #E40C22FF)';
/** Цвет обводки информация на светлом фоне */
export var outlineOnLightOutlineInfo = 'var(--outline-on-light-outline-info, #0B7ECB)';
/** Цвет обводки информация на светлом фоне */
export var outlineOnLightOutlineInfoHover = 'var(--outline-on-light-outline-info-hover, #34A7F4FF)';
/** Цвет обводки информация на светлом фоне */
export var outlineOnLightOutlineInfoActive = 'var(--outline-on-light-outline-info-active, #096CAEFF)';
/** Минорный цвет обводки успех на светлом фоне */
export var outlineOnLightOutlinePositiveMinor = 'var(--outline-on-light-outline-positive-minor, #45E362)';
/** Минорный цвет обводки успех на светлом фоне */
export var outlineOnLightOutlinePositiveMinorHover = 'var(--outline-on-light-outline-positive-minor-hover, #5FE778FF)';
/** Минорный цвет обводки успех на светлом фоне */
export var outlineOnLightOutlinePositiveMinorActive = 'var(--outline-on-light-outline-positive-minor-active, #2ADF4BFF)';
/** Минорный цвет обводки предупреждение на светлом фоне */
export var outlineOnLightOutlineWarningMinor = 'var(--outline-on-light-outline-warning-minor, #FFB38A)';
/** Минорный цвет обводки предупреждение на светлом фоне */
export var outlineOnLightOutlineWarningMinorHover = 'var(--outline-on-light-outline-warning-minor-hover, #FFC7A8FF)';
/** Минорный цвет обводки предупреждение на светлом фоне */
export var outlineOnLightOutlineWarningMinorActive = 'var(--outline-on-light-outline-warning-minor-active, #FF9F6BFF)';
/** Минорный цвет обводки ошибка на светлом фоне */
export var outlineOnLightOutlineNegativeMinor = 'var(--outline-on-light-outline-negative-minor, #FEB3BB)';
/** Минорный цвет обводки ошибка на светлом фоне */
export var outlineOnLightOutlineNegativeMinorHover = 'var(--outline-on-light-outline-negative-minor-hover, #FD7785FF)';
/** Минорный цвет обводки ошибка на светлом фоне */
export var outlineOnLightOutlineNegativeMinorActive = 'var(--outline-on-light-outline-negative-minor-active, #FD96A0FF)';
/** Минорный цвет обводки информация на светлом фоне */
export var outlineOnLightOutlineInfoMinor = 'var(--outline-on-light-outline-info-minor, #85CEFF)';
/** Минорный цвет обводки информация на светлом фоне */
export var outlineOnLightOutlineInfoMinorHover = 'var(--outline-on-light-outline-info-minor-hover, #D6EFFFFF)';
/** Минорный цвет обводки информация на светлом фоне */
export var outlineOnLightOutlineInfoMinorActive = 'var(--outline-on-light-outline-info-minor-active, #66C2FFFF)';
/** Прозрачный цвет обводки успех на светлом фоне */
export var outlineOnLightOutlineTransparentPositive = 'var(--outline-on-light-outline-transparent-positive, #108E2647)';
/** Прозрачный цвет обводки успех на светлом фоне */
export var outlineOnLightOutlineTransparentPositiveHover = 'var(--outline-on-light-outline-transparent-positive-hover, #108E25FF)';
/** Прозрачный цвет обводки успех на светлом фоне */
export var outlineOnLightOutlineTransparentPositiveActive = 'var(--outline-on-light-outline-transparent-positive-active, #108E2556)';
/** Прозрачный цвет обводки предупреждение на светлом фоне */
export var outlineOnLightOutlineTransparentWarning = 'var(--outline-on-light-outline-transparent-warning, #E8570247)';
/** Прозрачный цвет обводки предупреждение на светлом фоне */
export var outlineOnLightOutlineTransparentWarningHover = 'var(--outline-on-light-outline-transparent-warning-hover, #E85702FF)';
/** Прозрачный цвет обводки предупреждение на светлом фоне */
export var outlineOnLightOutlineTransparentWarningActive = 'var(--outline-on-light-outline-transparent-warning-active, #E8570256)';
/** Прозрачный цвет обводки предупреждение на светлом фоне */
export var outlineOnLightOutlineTransparentNegative = 'var(--outline-on-light-outline-transparent-negative, #F31B3147)';
/** Прозрачный цвет обводки предупреждение на светлом фоне */
export var outlineOnLightOutlineTransparentNegativeHover = 'var(--outline-on-light-outline-transparent-negative-hover, #F31B31FF)';
/** Прозрачный цвет обводки предупреждение на светлом фоне */
export var outlineOnLightOutlineTransparentNegativeActive = 'var(--outline-on-light-outline-transparent-negative-active, #F31B3156)';
/** Прозрачный цвет обводки информация на светлом фоне */
export var outlineOnLightOutlineTransparentInfo = 'var(--outline-on-light-outline-transparent-info, #0B7ECB47)';
/** Прозрачный цвет обводки информация на светлом фоне */
export var outlineOnLightOutlineTransparentInfoHover = 'var(--outline-on-light-outline-transparent-info-hover, #0B7ECBFF)';
/** Прозрачный цвет обводки информация на светлом фоне */
export var outlineOnLightOutlineTransparentInfoActive = 'var(--outline-on-light-outline-transparent-info-active, #0B7ECB56)';
/** Инвертированный основной непрозрачный цвет обводки */
export var outlineInverseOutlineSolidPrimary = 'var(--outline-inverse-outline-solid-primary, #DDDDDD)';
/** Инвертированный основной непрозрачный цвет обводки */
export var outlineInverseOutlineSolidPrimaryHover = 'var(--outline-inverse-outline-solid-primary-hover, #ABABABFF)';
/** Инвертированный основной непрозрачный цвет обводки */
export var outlineInverseOutlineSolidPrimaryActive = 'var(--outline-inverse-outline-solid-primary-active, #C4C4C4FF)';
/** Инвертированный вторичный непрозрачный цвет обводки */
export var outlineInverseOutlineSolidSecondary = 'var(--outline-inverse-outline-solid-secondary, #B3B3B3)';
/** Инвертированный вторичный непрозрачный цвет обводки */
export var outlineInverseOutlineSolidSecondaryHover = 'var(--outline-inverse-outline-solid-secondary-hover, #FFFFFFFF)';
/** Инвертированный вторичный непрозрачный цвет обводки */
export var outlineInverseOutlineSolidSecondaryActive = 'var(--outline-inverse-outline-solid-secondary-active, #3D3D3DFF)';
/** Инвертированный третичный непрозрачный цвет обводки */
export var outlineInverseOutlineSolidTertiary = 'var(--outline-inverse-outline-solid-tertiary, #707070)';
/** Инвертированный третичный непрозрачный цвет обводки */
export var outlineInverseOutlineSolidTertiaryHover = 'var(--outline-inverse-outline-solid-tertiary-hover, #FFFFFFFF)';
/** Инвертированный третичный непрозрачный цвет обводки */
export var outlineInverseOutlineSolidTertiaryActive = 'var(--outline-inverse-outline-solid-tertiary-active, #737373FF)';
/** Инвертированный прозрачный цвет обводки по умолчанию */
export var outlineInverseOutlineTransparentDefault = 'var(--outline-inverse-outline-transparent-default, #080808F5)';
/** Инвертированный прозрачный цвет обводки по умолчанию */
export var outlineInverseOutlineTransparentDefaultHover = 'var(--outline-inverse-outline-transparent-default-hover, #08080893)';
/** Инвертированный прозрачный цвет обводки по умолчанию */
export var outlineInverseOutlineTransparentDefaultActive = 'var(--outline-inverse-outline-transparent-default-active, #080808C4)';
/** Инвертированный основной прозрачный цвет обводки */
export var outlineInverseOutlineTransparentPrimary = 'var(--outline-inverse-outline-transparent-primary, #0808081F)';
/** Инвертированный основной прозрачный цвет обводки */
export var outlineInverseOutlineTransparentPrimaryHover = 'var(--outline-inverse-outline-transparent-primary-hover, #080808FF)';
/** Инвертированный основной прозрачный цвет обводки */
export var outlineInverseOutlineTransparentPrimaryActive = 'var(--outline-inverse-outline-transparent-primary-active, #08080825)';
/** Инвертированный вторичный прозрачный цвет обводки */
export var outlineInverseOutlineTransparentSecondary = 'var(--outline-inverse-outline-transparent-secondary, #08080847)';
/** Инвертированный вторичный прозрачный цвет обводки */
export var outlineInverseOutlineTransparentSecondaryHover = 'var(--outline-inverse-outline-transparent-secondary-hover, #080808FF)';
/** Инвертированный вторичный прозрачный цвет обводки */
export var outlineInverseOutlineTransparentSecondaryActive = 'var(--outline-inverse-outline-transparent-secondary-active, #08080856)';
/** Инвертированный третичный прозрачный цвет обводки */
export var outlineInverseOutlineTransparentTertiary = 'var(--outline-inverse-outline-transparent-tertiary, #0808088F)';
/** Инвертированный третичный прозрачный цвет обводки */
export var outlineInverseOutlineTransparentTertiaryHover = 'var(--outline-inverse-outline-transparent-tertiary-hover, #080808FF)';
/** Инвертированный третичный прозрачный цвет обводки */
export var outlineInverseOutlineTransparentTertiaryActive = 'var(--outline-inverse-outline-transparent-tertiary-active, #080808AB)';
/** Инвертированная бесцветная обводка */
export var outlineInverseOutlineClear = 'var(--outline-inverse-outline-clear, #FFFFFF00)';
/** Инвертированная бесцветная обводка */
export var outlineInverseOutlineClearHover = 'var(--outline-inverse-outline-clear-hover, #FFFFFF00)';
/** Инвертированная бесцветная обводка */
export var outlineInverseOutlineClearActive = 'var(--outline-inverse-outline-clear-active, #FFFFFF00)';
/** Инвертированный акцентный цвет обводки */
export var outlineInverseOutlineAccent = 'var(--outline-inverse-outline-accent, #108E26)';
/** Инвертированный акцентный цвет обводки */
export var outlineInverseOutlineAccentHover = 'var(--outline-inverse-outline-accent-hover, #13AA2CFF)';
/** Инвертированный акцентный цвет обводки */
export var outlineInverseOutlineAccentActive = 'var(--outline-inverse-outline-accent-active, #0D731EFF)';
/** Инвертированный акцентный минорный непрозрачный цвет обводки */
export var outlineInverseOutlineAccentMinor = 'var(--outline-inverse-outline-accent-minor, #45E362)';
/** Инвертированный акцентный минорный непрозрачный цвет обводки */
export var outlineInverseOutlineAccentMinorHover = 'var(--outline-inverse-outline-accent-minor-hover, #5FE778FF)';
/** Инвертированный акцентный минорный непрозрачный цвет обводки */
export var outlineInverseOutlineAccentMinorActive = 'var(--outline-inverse-outline-accent-minor-active, #2ADF4BFF)';
/** Прозрачный инвертированный акцентный цвет обводки */
export var outlineInverseOutlineTransparentAccent = 'var(--outline-inverse-outline-transparent-accent, #108E2666)';
/** Прозрачный инвертированный акцентный цвет обводки */
export var outlineInverseOutlineTransparentAccentHover = 'var(--outline-inverse-outline-transparent-accent-hover, #108E25FF)';
/** Прозрачный инвертированный акцентный цвет обводки */
export var outlineInverseOutlineTransparentAccentActive = 'var(--outline-inverse-outline-transparent-accent-active, #108E257A)';
/** Инвертированный цвет обводки успех */
export var outlineInverseOutlinePositive = 'var(--outline-inverse-outline-positive, #108E26)';
/** Инвертированный цвет обводки успех */
export var outlineInverseOutlinePositiveHover = 'var(--outline-inverse-outline-positive-hover, #13AA2CFF)';
/** Инвертированный цвет обводки успех */
export var outlineInverseOutlinePositiveActive = 'var(--outline-inverse-outline-positive-active, #0D731EFF)';
/** Инвертированный цвет обводки предупреждение */
export var outlineInverseOutlineWarning = 'var(--outline-inverse-outline-warning, #E85702)';
/** Инвертированный цвет обводки предупреждение */
export var outlineInverseOutlineWarningHover = 'var(--outline-inverse-outline-warning-hover, #FD650DFF)';
/** Инвертированный цвет обводки предупреждение */
export var outlineInverseOutlineWarningActive = 'var(--outline-inverse-outline-warning-active, #CA4B02FF)';
/** Инвертированный цвет обводки ошибка */
export var outlineInverseOutlineNegative = 'var(--outline-inverse-outline-negative, #F31B31)';
/** Инвертированный цвет обводки ошибка */
export var outlineInverseOutlineNegativeHover = 'var(--outline-inverse-outline-negative-hover, #F5384BFF)';
/** Инвертированный цвет обводки ошибка */
export var outlineInverseOutlineNegativeActive = 'var(--outline-inverse-outline-negative-active, #E40C22FF)';
/** Инвертированный цвет обводки информация */
export var outlineInverseOutlineInfo = 'var(--outline-inverse-outline-info, #0B7ECB)';
/** Инвертированный цвет обводки информация */
export var outlineInverseOutlineInfoHover = 'var(--outline-inverse-outline-info-hover, #34A7F4FF)';
/** Инвертированный цвет обводки информация */
export var outlineInverseOutlineInfoActive = 'var(--outline-inverse-outline-info-active, #096CAEFF)';
/** Инвертированный минорный цвет обводки успех */
export var outlineInverseOutlinePositiveMinor = 'var(--outline-inverse-outline-positive-minor, #45E362)';
/** Инвертированный минорный цвет обводки успех */
export var outlineInverseOutlinePositiveMinorHover = 'var(--outline-inverse-outline-positive-minor-hover, #5FE778FF)';
/** Инвертированный минорный цвет обводки успех */
export var outlineInverseOutlinePositiveMinorActive = 'var(--outline-inverse-outline-positive-minor-active, #2ADF4BFF)';
/** Инвертированный минорный цвет обводки предупреждение */
export var outlineInverseOutlineWarningMinor = 'var(--outline-inverse-outline-warning-minor, #FFB38A)';
/** Инвертированный минорный цвет обводки предупреждение */
export var outlineInverseOutlineWarningMinorHover = 'var(--outline-inverse-outline-warning-minor-hover, #FFC7A8FF)';
/** Инвертированный минорный цвет обводки предупреждение */
export var outlineInverseOutlineWarningMinorActive = 'var(--outline-inverse-outline-warning-minor-active, #FF9F6BFF)';
/** Инвертированный минорный цвет обводки ошибка */
export var outlineInverseOutlineNegativeMinor = 'var(--outline-inverse-outline-negative-minor, #FEB3BB)';
/** Инвертированный минорный цвет обводки ошибка */
export var outlineInverseOutlineNegativeMinorHover = 'var(--outline-inverse-outline-negative-minor-hover, #FD7785FF)';
/** Инвертированный минорный цвет обводки ошибка */
export var outlineInverseOutlineNegativeMinorActive = 'var(--outline-inverse-outline-negative-minor-active, #FD96A0FF)';
/** Инвертированный минорный цвет обводки информация */
export var outlineInverseOutlineInfoMinor = 'var(--outline-inverse-outline-info-minor, #85CEFF)';
/** Инвертированный минорный цвет обводки информация */
export var outlineInverseOutlineInfoMinorHover = 'var(--outline-inverse-outline-info-minor-hover, #D6EFFFFF)';
/** Инвертированный минорный цвет обводки информация */
export var outlineInverseOutlineInfoMinorActive = 'var(--outline-inverse-outline-info-minor-active, #66C2FFFF)';
/** Прозрачный инвертированный цвет обводки успех */
export var outlineInverseOutlineTransparentPositive = 'var(--outline-inverse-outline-transparent-positive, #108E2647)';
/** Прозрачный инвертированный цвет обводки успех */
export var outlineInverseOutlineTransparentPositiveHover = 'var(--outline-inverse-outline-transparent-positive-hover, #108E25FF)';
/** Прозрачный инвертированный цвет обводки успех */
export var outlineInverseOutlineTransparentPositiveActive = 'var(--outline-inverse-outline-transparent-positive-active, #108E2556)';
/** Прозрачный инвертированный цвет обводки предупреждение */
export var outlineInverseOutlineTransparentWarning = 'var(--outline-inverse-outline-transparent-warning, #E8570247)';
/** Прозрачный инвертированный цвет обводки предупреждение */
export var outlineInverseOutlineTransparentWarningHover = 'var(--outline-inverse-outline-transparent-warning-hover, #E85702FF)';
/** Прозрачный инвертированный цвет обводки предупреждение */
export var outlineInverseOutlineTransparentWarningActive = 'var(--outline-inverse-outline-transparent-warning-active, #E8570256)';
/** Прозрачный инвертированный цвет обводки предупреждение */
export var outlineInverseOutlineTransparentNegative = 'var(--outline-inverse-outline-transparent-negative, #F31B3147)';
/** Прозрачный инвертированный цвет обводки предупреждение */
export var outlineInverseOutlineTransparentNegativeHover = 'var(--outline-inverse-outline-transparent-negative-hover, #F31B31FF)';
/** Прозрачный инвертированный цвет обводки предупреждение */
export var outlineInverseOutlineTransparentNegativeActive = 'var(--outline-inverse-outline-transparent-negative-active, #F31B3156)';
/** Прозрачный инвертированный цвет обводки информация */
export var outlineInverseOutlineTransparentInfo = 'var(--outline-inverse-outline-transparent-info, #0B7ECB47)';
/** Прозрачный инвертированный цвет обводки информация */
export var outlineInverseOutlineTransparentInfoHover = 'var(--outline-inverse-outline-transparent-info-hover, #0B7ECBFF)';
/** Прозрачный инвертированный цвет обводки информация */
export var outlineInverseOutlineTransparentInfoActive = 'var(--outline-inverse-outline-transparent-info-active, #0B7ECB56)';
export var skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
export var skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
export var whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
export var whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
export var whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
export var blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
export var blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
export var blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use onLightSurfaceSolidDefault */
export var buttonBlack = 'var(--plasma-colors-button-black, var(--on-light-surface-solid-default))';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
export var buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, var(--on-light-surface-transparent-secondary))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
export var buttonWhite = 'var(--plasma-colors-button-white, var(--on-dark-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
export var buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, var(--on-dark-surface-transparent-secondary))';
/** @deprecated instead use textPrimary */
export var text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
export var primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
export var secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
export var tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use textParagraph */
export var paragraph = 'var(--plasma-colors-paragraph, var(--text-paragraph))';
/** @deprecated instead use backgroundPrimary */
export var background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textAccent */
export var accent = 'var(--plasma-colors-accent, var(--text-accent))';
/** @deprecated instead use textPositive */
export var success = 'var(--plasma-colors-success, var(--text-positive))';
/** @deprecated instead use textWarning */
export var warning = 'var(--plasma-colors-warning, var(--text-warning))';
/** @deprecated instead use textNegative */
export var critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use overlaySoft */
export var overlay = 'var(--plasma-colors-overlay, var(--overlay-soft))';
/** @deprecated instead use surfaceTransparentPrimary */
export var surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, var(--surface-transparent-primary))';
/** @deprecated instead use surfaceTransparentSecondary */
export var surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, var(--surface-transparent-secondary))';
/** @deprecated instead use surfaceTransparentTertiary */
export var surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, var(--surface-transparent-tertiary))';
/** @deprecated instead use surfaceSolidPrimary */
export var surfaceSolid01 = 'var(--plasma-colors-surface-solid01, var(--surface-solid-primary))';
/** @deprecated instead use surfaceSolidSecondary */
export var surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
export var surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use surfaceTransparentCard */
export var surfaceCard = 'var(--plasma-colors-surface-card, var(--surface-transparent-card))';
/** @deprecated instead use surfaceTransparentSecondary */
export var buttonSecondary = 'var(--plasma-colors-button-secondary, var(--surface-transparent-secondary))';
/** @deprecated instead use textAccent */
export var buttonAccent = 'var(--plasma-colors-button-accent, var(--text-accent))';
/** @deprecated instead use surfacePositive */
export var buttonSuccess = 'var(--plasma-colors-button-success, var(--surface-positive))';
/** @deprecated instead use surfaceWarning */
export var buttonWarning = 'var(--plasma-colors-button-warning, var(--surface-warning))';
/** @deprecated instead use surfaceNegative */
export var buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
export var shadowDownSoftS = 'var(--shadow-down-soft-s, 0px 4px 14px -4px rgba(8, 8, 8, 0.08)), var(--shadow-down-soft-s, 0px 1px 4px -1px rgba(0, 0, 0, 0.04))';
export var shadowDownSoftM = 'var(--shadow-down-soft-m, 0px 24px 48px -8px rgba(0, 0, 0, 0.08))';
export var shadowDownSoftL = 'var(--shadow-down-soft-l, 0px 60px 112px -8px rgba(0, 0, 0, 0.08))';
export var shadowDownHardS = 'var(--shadow-down-hard-s, 0px 4px 12px -3px rgba(8, 8, 8, 0.16)), var(--shadow-down-hard-s, 0px 1px 4px -2px rgba(0, 0, 0, 0.08))';
export var shadowDownHardM = 'var(--shadow-down-hard-m, 0px 16px 32px -8px rgba(0, 0, 0, 0.24))';
export var shadowDownHardL = 'var(--shadow-down-hard-l, 0px 60px 112px -8px rgba(0, 0, 0, 0.40))';
export var shadowUpSoftS = 'var(--shadow-up-soft-s, 0px -4px 14px -4px rgba(8, 8, 8, 0.08)), var(--shadow-up-soft-s, 0px -1px 4px -1px rgba(0, 0, 0, 0.03))';
export var shadowUpSoftM = 'var(--shadow-up-soft-m, 0px -24px 48px -8px rgba(0, 0, 0, 0.08))';
export var shadowUpSoftL = 'var(--shadow-up-soft-l, 0px -60px 112px -8px rgba(0, 0, 0, 0.08))';
export var shadowUpHardS = 'var(--shadow-up-hard-s, 0px -4px 12px -3px rgba(8, 8, 8, 0.20)), var(--shadow-up-hard-s, 0px -1px 4px -1px rgba(0, 0, 0, 0.03))';
export var shadowUpHardM = 'var(--shadow-up-hard-m, 0px -16px 32px -8px rgba(0, 0, 0, 0.24))';
export var shadowUpHardL = 'var(--shadow-up-hard-l, 0px -60px 112px -8px rgba(0, 0, 0, 0.40))';
export var borderRadiusXxs = 'var(--border-radius-xxs, 4px)';
export var borderRadiusXs = 'var(--border-radius-xs, 6px)';
export var borderRadiusS = 'var(--border-radius-s, 8px)';
export var borderRadiusM = 'var(--border-radius-m, 12px)';
export var borderRadiusL = 'var(--border-radius-l, 16px)';
export var borderRadiusXl = 'var(--border-radius-xl, 20px)';
export var borderRadiusXxl = 'var(--border-radius-xxl, 40px)';
