// @salutejs/plasma-ui/components/TextField/TextField.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _Button = /*#__PURE__*/require("../Button");
var _ = /*#__PURE__*/require(".");
var _IconSleep, _ActionButton;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var defaultProps = {
  id: 'example-text-field',
  type: 'text',
  placeholder: 'Заполните поле',
  helperText: 'Подсказка к полю',
  disabled: false,
  readOnly: false,
  size: 'l',
  enableLeftIcon: true,
  enableRightIcon: true
};
var status = '';
var func = function func() {};
var Default = exports.Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(_.TextField, _extends({
    value: "\u0417\u043D\u0430\u0447\u0435\u043D\u0438\u0435 \u043F\u043E\u043B\u044F",
    status: status,
    contentLeft: _IconSleep || (_IconSleep = /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconSleep, {
      color: "inherit",
      size: "s"
    })),
    contentRight: _ActionButton || (_ActionButton = /*#__PURE__*/_react["default"].createElement(_Button.ActionButton, {
      view: "clear"
    }, /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconEye, {
      color: "inherit",
      size: "s"
    }))),
    onChange: func,
    onFocus: func,
    onBlur: func,
    onSearch: func
  }, defaultProps));
};
