// @salutejs/plasma-new-hope/styled-components/es/examples/typograpy/components/Old/Footnote/Footnote.config.js
import { css } from 'styled-components';
import { tokens } from '../../../../../components/Typography/tokens';
export var config = {
  defaults: {
    size: 'footnote1'
  },
  variations: {
    size: {
      footnote1: /*#__PURE__*/css(["", ":var(--plasma-typo-footnote1-font-family);", ":var(--plasma-typo-footnote1-font-size);", ":var(--plasma-typo-footnote1-font-style);", ":var(--plasma-typo-footnote1-font-weight);", ":var(--plasma-typo-footnote1-letter-spacing);", ":var(--plasma-typo-footnote1-line-height);"], tokens.typoFontFamily, tokens.typoFontSize, tokens.typoFontStyle, tokens.typoFontWeight, tokens.typoFontLetterSpacing, tokens.typoFontLineHeight),
      footnote2: /*#__PURE__*/css(["", ":var(--plasma-typo-footnote2-font-family);", ":var(--plasma-typo-footnote2-font-size);", ":var(--plasma-typo-footnote2-font-style);", ":var(--plasma-typo-footnote2-font-weight);", ":var(--plasma-typo-footnote2-letter-spacing);", ":var(--plasma-typo-footnote2-line-height);"], tokens.typoFontFamily, tokens.typoFontSize, tokens.typoFontStyle, tokens.typoFontWeight, tokens.typoFontLetterSpacing, tokens.typoFontLineHeight)
    }
  }
};
