// @salutejs/plasma-new-hope/es/components/Dropdown/utils/index.js
import { typeof as _typeof } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { Children, isValidElement, cloneElement } from 'react';

var getPlacement = function getPlacement(placement) {
  return "".concat(placement, "-start");
};
var getPlacements = function getPlacements(placements) {
  if (!placements) {
    return;
  }
  var isArray = Array.isArray(placements);
  if (!isArray) {
    return getPlacement(placements);
  }
  return (placements || []).map(function (placement) {
    return getPlacement(placement);
  });
};
var getCorrectHeight = function getCorrectHeight(listHeight) {
  if (Number.isNaN(Number(listHeight))) {
    return listHeight.toString();
  }
  return "".concat(listHeight, "rem");
};
var childrenWithProps = function childrenWithProps(children, additionalProps) {
  return Children.map(children, function (child) {
    if ( /*#__PURE__*/React.isValidElement(child)) {
      return /*#__PURE__*/React.cloneElement(child, additionalProps);
    }
    return child;
  });
};
var isReactObject = function isReactObject(element) {
  return _typeof(element) === 'object' || typeof element === 'function';
};
var getValidComponent = function getValidComponent(element, props) {
  if ( /*#__PURE__*/isValidElement(element)) {
    return /*#__PURE__*/cloneElement(element, props);
  }
  if (isReactObject(element)) {
    var Component = element;
    return /*#__PURE__*/React.createElement(Component, props);
  }
  return element;
};

export { childrenWithProps, getCorrectHeight, getPlacement, getPlacements, getValidComponent };
