// @salutejs/plasma-tokens-b2c/es/colors/values.js
// Generated by robots, do not change this manually!
/** Базовый белый цвет совпадает с whitePrimary */
export var white = 'rgba(255, 255, 255, 0.96)';
/** Первичный белый цвет */
export var whitePrimary = 'rgba(255, 255, 255, 0.96)';
/** Вторичный белый цвет */
export var whiteSecondary = 'rgba(255, 255, 255, 0.56)';
/** Третичный белый цвет */
export var whiteTertiary = 'rgba(255, 255, 255, 0.28)';
/** Базовый черный цвет совпадает с blackPrimary */
export var black = '#080808';
/** Первичный черный цвет */
export var blackPrimary = '#080808';
/** Вторичный черныйцвет */
export var blackSecondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный черный цвет */
export var blackTertiary = 'rgba(8, 8, 8, 0.28)';
/** Чёрный непрозрачный, не зависит от темы */
export var dark01 = '#171717';
/** Чёрный непрозрачный, не зависит от темы */
export var dark02 = '#232323';
/** Чёрный непрозрачный, не зависит от темы */
export var dark03 = '#363636';
/** Прозрачный цвет */
export var transparent = 'rgba(0, 0, 0, 0)';
/** Второстепенная прозрачная */
export var buttonClear = 'rgba(0, 0, 0, 0)';
/** Дефолтная чёрная, не зависит от темы */
export var buttonBlack = '#080808';
/** Второстепенная чёрная, не зависит от темы */
export var buttonBlackSecondary = 'rgba(8, 8, 8, 0.12)';
/** Чёрная для использования поверх картинок */
export var buttonBlackTransparent = 'rgba(8, 8, 8, 0.56)';
/** Дефолтная белая, не зависит от темы */
export var buttonWhite = '#FFFFFF';
/** Второстепенная белая, не зависит от темы */
export var buttonWhiteSecondary = 'rgba(255, 255, 255, 0.12)';
/** Базовый цвет текста, совпадает с primary */
export var text = '#080808';
export var link = '#2A72F8';
export var linkHover = '#4D89F9';
export var linkActive = '#0859F1';
export var linkVisited = '#2364DE';
export var linkVisitedHover = '#427AE3';
export var linkVisitedActive = '#1C54BC';
/** Основной текст в интерфейсе */
export var primary = '#080808';
/** Второстепенный текст */
export var secondary = 'rgba(8, 8, 8, 0.56)';
/** Третичный цвет текста */
export var tertiary = 'rgba(8, 8, 8, 0.28)';
export var paragraph = 'rgba(8, 8, 8, 0.8)';
export var inverse = 'rgba(255, 255, 255, 0.96)';
/** Основной бэкграунд всех приложений и смартапов */
export var background = '#FFFFFF';
/** Первичный цвет фона */
export var backgroundPrimary = '#FFFFFF';
/** Вторичный цвет фона */
export var backgroundSecondary = '#FFFFFF';
/** Третичный цвет фона */
export var backgroundTertiary = '#FFFFFF';
export var inputBorder = 'rgba(0, 0, 0, 0)';
export var inputBorderHover = 'rgba(0, 0, 0, 0)';
export var inputBorderFocus = 'rgba(0, 0, 0, 0)';
export var inputErrorBackground = 'rgba(243, 27, 48, 0.12)';
/** Акцентный цвет бренда */
export var accent = '#296AE3';
/** Обозначение успешного сценария */
export var success = '#0D8523';
/** Цвет предупреждения */
export var warning = '#D14D00';
/** Цвет ошибки */
export var critical = '#E31227';
/** Полупрозрачные поверхности */
export var surfaceLiquid01 = 'rgba(8, 8, 8, 0.02)';
/** Полупрозрачные поверхности */
export var surfaceLiquid02 = 'rgba(8, 8, 8, 0.06)';
/** Полупрозрачные поверхности */
export var surfaceLiquid03 = 'rgba(8, 8, 8, 0.12)';
/** Непрозрачные поверхности */
export var surfaceSolid01 = '#FAFAFA';
/** Непрозрачные поверхности */
export var surfaceSolid02 = '#F0F0F0';
/** Непрозрачные поверхности */
export var surfaceSolid03 = '#E1E1E1';
/** Основной фон для карточек */
export var surfaceCard = '#FFFFFF';
/** Первичный цвет контролов */
export var buttonPrimary = '#2A72F8';
export var buttonPrimaryHover = '#447ff9';
export var buttonPrimaryActive = '#2868e0';
/** Второстепенная кнопка */
export var buttonSecondary = 'rgba(8, 8, 8, 0.06)';
export var buttonSecondaryHover = 'rgba(8, 8, 8, 0.04)';
export var buttonSecondaryActive = 'rgba(8, 8, 8, 0.08)';
/** Акцентный цвет у контролов */
export var buttonAccent = '#296AE3';
/** Кнопка для успешного сценария */
export var buttonSuccess = '#148F2B';
export var buttonSuccessTransparent = 'rgba(20, 143, 43, 0.3)';
export var buttonSuccessHover = '#379740';
export var buttonSuccessActive = '#157e26';
export var buttonSuccessActiveTransparent = 'rgba(21, 126, 39, 0.35)';
/** Цвет предупреждения у контролов */
export var buttonWarning = '#E35502';
export var buttonWarningTransparent = 'rgba(227, 85, 2, 0.3)';
export var buttonWarningHover = '#f76d2f';
export var buttonWarningActive = '#de550a';
export var buttonWarningActiveTransparent = 'rgba(222, 85, 10, 0.35)';
/** Цвет ошибки у контролов */
export var buttonCritical = '#F31B30';
export var buttonCriticalTransparent = 'rgba(243, 27, 48, 0.3)';
export var buttonCriticalHover = '#f53b46';
export var buttonCriticalActive = '#dc1b2b';
export var buttonCriticalActiveTransparent = 'rgba(220, 27, 43, 0.35)';
/** Цвет зажатого контрола */
export var buttonChecked = '#080808';
export var buttonCheckedHover = '#080808';
export var buttonCheckedHoverColor = 'hsla(0, 0%, 100%, 0.9599999785423279)';
export var buttonCheckedActive = '#080808';
export var buttonCheckedActiveColor = 'hsla(0, 0%, 100%, 0.9599999785423279)';
/** Цвет рамки фокуса у контрола */
export var buttonFocused = '#296AE3';
/** Градиент для заливки основного фона */
export var gradient = 'linear-gradient(45deg, hsla(219.87577639751552, 100%, 68.43137254901961%, 1) 0%, hsla(190.1639344262295, 94.81865284974093%, 62.15686274509804%, 1) 100%)';
export var gradientDevice = '';
export var voicePhraseGradient = '';
export var skeletonGradient = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.09) 0%, rgba(8, 8, 8, 0.08) 6.25%, rgba(8, 8, 8, 0.05) 12.5%, rgba(8, 8, 8, 0.01) 25%, rgba(8, 8, 8, 0.05) 37.5%, rgba(8, 8, 8, 0.08) 43.75%, rgba(8, 8, 8, 0.09) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.05) 62.5%, rgba(8, 8, 8, 0.01) 75%, rgba(8, 8, 8, 0.05) 87.5%, rgba(8, 8, 8, 0.08) 93.75%, rgba(8, 8, 8, 0.09) 100% )';
export var skeletonGradientLighter = 'linear-gradient( 90deg, rgba(8, 8, 8, 0.36) 0%, rgba(8, 8, 8, 0.32) 6.25%, rgba(8, 8, 8, 0.20) 12.5%, rgba(8, 8, 8, 0.04) 25%, rgba(8, 8, 8, 0.20) 37.5%, rgba(8, 8, 8, 0.32) 43.75%, rgba(8, 8, 8, 0.36) 50%, rgba(8, 8, 8, 0.08) 56.25%, rgba(8, 8, 8, 0.20) 62.5%, rgba(8, 8, 8, 0.04) 75%, rgba(8, 8, 8, 0.20) 87.5%, rgba(8, 8, 8, 0.32) 93.75%, rgba(8, 8, 8, 0.36) 100% )';
export var speechBubbleSent = '';
export var speechBubbleReceived = '';
/** Цвет фона паранжи */
export var overlay = 'rgba(0, 0, 0, 0.8)';
export var overlaySoft = 'rgba(8, 8, 8, 0.56)';
export var overlayHard = 'rgba(8, 8, 8, 0.9)';
export var overlayBlur = 'rgba(35, 35, 35, 0.2)';
export var darkOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var darkOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var darkOverlayBlur = 'rgba(35, 35, 35, 0.2)';
export var lightOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var lightOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var lightOverlayBlur = 'rgba(35, 35, 35, 0.2)';
export var inverseOverlaySoft = 'rgba(8, 8, 8, 0.56)';
export var inverseOverlayHard = 'rgba(8, 8, 8, 0.9)';
export var inverseOverlayBlur = 'rgba(35, 35, 35, 0.2)';
