// @salutejs/plasma-new-hope/cjs/components/Popup/Popup.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var ReactDOM = require('react-dom');
var plasmaCore = require('@salutejs/plasma-core');
var index = require('../../utils/index.js');
var PopupContext = require('./PopupContext.js');
var PopupRoot = require('./PopupRoot.js');
var usePopup = require('./hooks/usePopup.js');
var Popup_tokens = require('./Popup.tokens.js');

var _excluded = ["id", "isOpen", "placement", "offset", "frame", "children", "overlay", "role", "zIndex", "popupInfo", "withAnimation", "className"];
var handlePosition = function handlePosition(placement, offset) {
  var x = '0rem';
  var y = '0rem';
  if (offset) {
    var _offset = _rollupPluginBabelHelpers.slicedToArray(offset, 2),
      _x = _offset[0],
      _y = _offset[1];
    x = typeof _x === 'number' ? "".concat(_x, "rem") : _x;
    y = typeof _y === 'number' ? "".concat(_y, "rem") : _y;
  }
  if (!placement || placement === 'center') {
    return {
      left: "calc(50% + ".concat(x, ")"),
      top: "calc(50% - ".concat(y, ")"),
      transform: 'translate(-50%, -50%)'
    };
  }
  var left;
  var right;
  var top;
  var bottom;
  var transform;
  var placements = placement.split('-');
  placements.forEach(function (placement) {
    switch (placement) {
      case 'left':
        left = x;
        break;
      case 'right':
        right = x;
        break;
      case 'top':
        top = y;
        break;
      case 'bottom':
        bottom = y;
        break;
    }
  });
  var isCenteredX = left === undefined && right === undefined;
  var isCenteredY = top === undefined && bottom === undefined;
  if (isCenteredX) {
    transform = 'translateX(-50%)';
  }
  if (isCenteredY) {
    transform = 'translateY(-50%)';
  }
  return {
    left: isCenteredX ? "calc(50% + ".concat(x, ")") : left,
    right: right,
    top: isCenteredY ? "calc(50% - ".concat(y, ")") : top,
    bottom: bottom,
    transform: transform
  };
};

/**
 * Базовый копмонент Popup.
 */
var popupRoot = function popupRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (_ref, outerRootRef) {
    var id = _ref.id,
      _ref$isOpen = _ref.isOpen,
      isOpen = _ref$isOpen === void 0 ? false : _ref$isOpen,
      _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? 'center' : _ref$placement,
      _ref$offset = _ref.offset,
      offset = _ref$offset === void 0 ? [0, 0] : _ref$offset,
      _ref$frame = _ref.frame,
      frame = _ref$frame === void 0 ? 'document' : _ref$frame,
      children = _ref.children,
      overlay = _ref.overlay,
      zIndex = _ref.zIndex,
      popupInfo = _ref.popupInfo,
      _ref$withAnimation = _ref.withAnimation,
      withAnimation = _ref$withAnimation === void 0 ? false : _ref$withAnimation,
      className = _ref.className,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
    var uniqId = plasmaCore.safeUseId();
    var innerId = id || uniqId;
    var _usePopup = usePopup.usePopup({
        isOpen: isOpen,
        id: innerId,
        popupInfo: popupInfo,
        withAnimation: withAnimation
      }),
      isVisible = _usePopup.isVisible,
      animationInfo = _usePopup.animationInfo,
      setVisible = _usePopup.setVisible;
    var portalRef = React.useRef(null);
    var contentRef = React.useRef(null);
    var innerRef = plasmaCore.useForkRef(contentRef, outerRootRef);
    var _useState = React.useState(false),
      _useState2 = _rollupPluginBabelHelpers.slicedToArray(_useState, 2),
      forceRender = _useState2[1];
    React.useEffect(function () {
      var portal = document.getElementById(PopupContext.POPUP_PORTAL_ID);
      if (typeof frame !== 'string' && frame && frame.current) {
        portal = frame.current;
      }
      if (!portal) {
        portal = document.createElement('div');
        portal.setAttribute('id', PopupContext.POPUP_PORTAL_ID);
        /**
         * Нужно для того, чтобы во фрейме не происходило скачков контента
         * при анимации через transform, если есть элемент с шириной/высотой в 100% (Overlay)
         */
        portal.style.width = '0';
        if (typeof frame === 'string' && frame !== 'document') {
          var _document$getElementB;
          (_document$getElementB = document.getElementById(frame)) === null || _document$getElementB === void 0 || _document$getElementB.appendChild(portal);
        } else {
          document.body.appendChild(portal);
        }
      }
      portalRef.current = portal;

      /**
       * Изменение стейта нужно для того, чтобы Popup
       * отобразился после записи DOM элемента в portalRef.current
       */
      forceRender(true);
    }, []);
    if (!isVisible && !isOpen) {
      return null;
    }
    var cls = index.cx(className, animationInfo !== null && animationInfo !== void 0 && animationInfo.endAnimation ? Popup_tokens.classes.endAnimation : '', animationInfo !== null && animationInfo !== void 0 && animationInfo.endTransition ? Popup_tokens.classes.endTransition : '');
    return /*#__PURE__*/React.createElement(React.Fragment, null, portalRef.current && /*#__PURE__*/ReactDOM.createPortal( /*#__PURE__*/React.createElement(Root, _rollupPluginBabelHelpers.extends({
      className: cls
    }, rest), overlay, /*#__PURE__*/React.createElement(PopupRoot.PopupRoot, {
      id: innerId,
      ref: innerRef,
      position: handlePosition(placement, offset),
      frame: frame,
      zIndex: zIndex,
      animationInfo: animationInfo,
      setVisible: setVisible
    }, children)), portalRef.current));
  });
};
var popupConfig = {
  name: 'Popup',
  tag: 'div',
  layout: popupRoot,
  base: '',
  variations: {},
  defaults: {}
};

exports.handlePosition = handlePosition;
exports.popupConfig = popupConfig;
exports.popupRoot = popupRoot;
