// @salutejs/plasma-new-hope/styled-components/es/components/Segment/ui/SegmentItem/SegmentItem.styles.js
import styled from 'styled-components';
import { css } from 'styled-components';
import { addFocus, applyEllipsis } from '../../../../mixins';
import { tokens } from '../../tokens';
export var base = /*#__PURE__*/css(["position:relative;display:inline-flex;align-items:center;justify-content:center;box-sizing:border-box;align-items:center;appearance:none;border:none;outline:none;cursor:pointer;margin-left:var(", ");-webkit-tap-highlight-color:rgba(0,0,0,0);&:first-child{margin-left:0;}", ";"], tokens.itemMarginLeft, /*#__PURE__*/addFocus({
  outlineSize: '0.063rem',
  outlineOffset: '-0.125rem',
  outlineColor: /*#__PURE__*/"var(".concat(tokens.outlineFocusColor, ")"),
  outlineRadius: 'calc(var(--plasma_private-outline-radius) + 0.063rem)',
  customFocusRules: /*#__PURE__*/"\n            &.focus-visible:focus,\n            &[data-focus-visible-added] {\n                &::before {\n                    z-index: 1;\n                    outline: none;\n                    box-shadow: 0 0 0 0.063rem var(".concat(tokens.outlineFocusColor, ");\n                }\n            }\n        ")
}));
export var StyledContent = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-m3m328-0"
})(["display:inline-block;width:fit-content;", ";"], /*#__PURE__*/applyEllipsis());
