// @salutejs/plasma-ui/es/utils/deviceDetection.js
export var deviceScales = {
  sberPortal: 2,
  sberBox: 2,
  mobile: 1
};
export var DeviceKindList = /*#__PURE__*/function (DeviceKindList) {
  DeviceKindList["sberBox"] = "sberBox";
  DeviceKindList["mobile"] = "mobile";
  DeviceKindList["sberPortal"] = "sberPortal";
  return DeviceKindList;
}({});

/**
 * Проверка в браузере на устройство "SberPortal".
 * @return {boolean}
 */
export var isSberPortal = function isSberPortal() {
  if (typeof navigator === 'undefined') {
    return false;
  }
  var ua = navigator.userAgent.toLowerCase();
  return ua.includes('sberportal') || ua.includes('stargate');
};

/**
 * Проверка в браузере на устройство "SberBox".
 * @return {boolean}
 */
export var isSberBox = function isSberBox() {
  if (typeof navigator === 'undefined') {
    return false;
  }
  var ua = navigator.userAgent.toLowerCase();
  return ua.includes('sberbox');
};

/**
 * Проверка в браузере на устройство "SberBoxTop".
 * @return {boolean}
 */
export var isSberBoxTop = function isSberBoxTop() {
  if (typeof navigator === 'undefined') {
    return false;
  }
  var ua = navigator.userAgent.toLowerCase();
  return ua.includes('satellite') || ua.includes('sberbox top');
};

/**
 * Проверка в браузере на устройство "SberBoxTV".
 * @return {boolean}
 */
export var isTV = function isTV() {
  if (typeof navigator === 'undefined') {
    return false;
  }
  var ua = navigator.userAgent.toLowerCase();
  return ua.includes('(tv; tv)') || ua.includes('(tv; huawei)') || ua.includes('(huawei-tv; huawei)') || ua.includes('(huawei-tv; huawei tv)');
};

/**
 * Вернет тип устройства, под которым запустилось приложение.
 * В случае запуска в серверном окружении, возвращает "sberBox".
 * @return {DeviceKind}
 */
export var detectDevice = function detectDevice() {
  if (typeof navigator === 'undefined') {
    return DeviceKindList.sberBox;
  }
  switch (true) {
    case isSberPortal():
      return DeviceKindList.sberPortal;
    case isSberBoxTop():
    case isSberBox():
    case isTV():
      return DeviceKindList.sberBox;
    default:
      return DeviceKindList.mobile;
  }
};
