// @salutejs/plasma-new-hope/cjs/components/Header/ui/HeaderLogo/HeaderLogo.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
require('@linaria/core');
var common = require('../../../../engines/common.js');
var Image$1 = require('../../../Image/Image.js');
var Header_tokens = require('../../Header.tokens.js');
var base = require('./variations/_size/base.js');

// INFO: Omit 'onResize' | 'onResizeCapture' | 'nonce'
// because this types coming with @types/react@18.0.25 and breaks react@17.0.2 with @types/react@18.0.18

// issue #823
var Image = /*#__PURE__*/common.component(Image$1.imageConfig);

/**
 * Компонент для размещения логотипа.
 */
var headerLogoRoot = function headerLogoRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (props, outerRootRef) {
    return /*#__PURE__*/React.createElement(Root, {
      ref: outerRootRef
    }, /*#__PURE__*/React.createElement(Image, _rollupPluginBabelHelpers.extends({
      className: Header_tokens.classes.headerLogoImage
    }, props)));
  });
};
var headerLogoConfig = {
  name: 'HeaderLogo',
  tag: 'div',
  layout: headerLogoRoot,
  base: '',
  variations: {
    size: {
      css: base.base
    }
  },
  defaults: {
    size: 'm'
  }
};

exports.headerLogoConfig = headerLogoConfig;
exports.headerLogoRoot = headerLogoRoot;
