// @salutejs/plasma-new-hope/es/components/Calendar/ui/CalendarYears/CalendarYears.js
import { slicedToArray as _slicedToArray } from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { useRef, useCallback, useEffect } from 'react';
import { cx } from '../../../../utils/index.js';
import { useYears } from '../../hooks/useYears.js';
import { classes } from '../../Calendar.tokens.js';
import { StyledCalendarYears, StyledFlex, StyledYearRoot, StyledYear } from './CalendarYears.styles.js';

/**
 * Компонент годов в календаре.
 */
var CalendarYears = function CalendarYears(_ref) {
  var currentDate = _ref.date,
    startYear = _ref.startYear,
    selectIndexes = _ref.selectIndexes,
    outerRefs = _ref.outerRefs,
    onChangeYear = _ref.onChangeYear,
    onSetSelected = _ref.onSetSelected,
    onKeyDown = _ref.onKeyDown;
  var _useYears = useYears(currentDate, startYear),
    _useYears2 = _slicedToArray(_useYears, 2),
    years = _useYears2[0],
    selected = _useYears2[1];
  var selectedRef = useRef(selected);
  var onSetSelectedRef = useRef(onSetSelected);
  var handleOnChangeYear = useCallback(function (event) {
    /**
     * нужно вызвать stopImmediatePropagation для случаев, когда
     * обработчик события onClick навешивается снаружи.
     * Как, например, в компоненте Popup
     */
    event.nativeEvent.stopImmediatePropagation();
    var year = event.currentTarget.dataset.year;
    onChangeYear(Number(year));
  }, [onChangeYear]);
  var getRefs = useCallback(function (element, i, j) {
    outerRefs.current[i][j] = element;
  }, [outerRefs]);
  useEffect(function () {
    if (selectedRef.current) {
      var _onSetSelectedRef$cur;
      (_onSetSelectedRef$cur = onSetSelectedRef.current) === null || _onSetSelectedRef$cur === void 0 || _onSetSelectedRef$cur.call(onSetSelectedRef, selectedRef.current);
    }
  }, []);
  return /*#__PURE__*/React.createElement(StyledCalendarYears, {
    role: "grid",
    "aria-labelledby": "id-grid-label",
    onKeyDown: onKeyDown
  }, years.map(function (year, i) {
    return /*#__PURE__*/React.createElement(StyledFlex, {
      role: "row",
      key: i
    }, year.map(function (_ref2, j) {
      var yearValue = _ref2.yearValue,
        isSelected = _ref2.isSelected,
        isCurrent = _ref2.isCurrent;
      var selectedClass = isSelected ? classes.selectedItem : undefined;
      var currentClass = !isSelected && isCurrent ? classes.currentItem : undefined;
      return /*#__PURE__*/React.createElement(StyledYearRoot, {
        className: cx(selectedClass, currentClass, classes.selectableItem),
        ref: function ref(element) {
          return getRefs(element, i, j);
        },
        tabIndex: i === (selectIndexes === null || selectIndexes === void 0 ? void 0 : selectIndexes[0]) && j === (selectIndexes === null || selectIndexes === void 0 ? void 0 : selectIndexes[1]) ? 0 : -1,
        onClick: handleOnChangeYear,
        "data-year": yearValue,
        "aria-selected": isSelected,
        role: "gridcell",
        key: "StyledYear-".concat(i, "-").concat(j),
        "aria-label": String(yearValue)
      }, /*#__PURE__*/React.createElement(StyledYear, null, yearValue));
    }));
  }));
};

export { CalendarYears };
