// @salutejs/plasma-web/css/cjs/components/Combobox/Combobox.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var plasmaNewHope = require('@salutejs/plasma-new-hope');
var Combobox_config = require('./Combobox.config.js');

var mergedConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.comboboxConfig, Combobox_config.config);
var Combobox = /*#__PURE__*/plasmaNewHope.component(mergedConfig);
var mergedItemConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.comboboxItemConfig);
var ComboboxItem = /*#__PURE__*/plasmaNewHope.component(mergedItemConfig);
var mergedDividerConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.comboboxDividerConfig);
var ComboboxDivider = /*#__PURE__*/plasmaNewHope.component(mergedDividerConfig);
var mergedFooterConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.comboboxFooterConfig);
var ComboboxFooter = /*#__PURE__*/plasmaNewHope.component(mergedFooterConfig);
var mergedGroupConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.comboboxGroupConfig);
var ComboboxGroup = /*#__PURE__*/plasmaNewHope.component(mergedGroupConfig);
var mergedHeaderConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.comboboxHeaderConfig);
var ComboboxHeader = /*#__PURE__*/plasmaNewHope.component(mergedHeaderConfig);

exports.Combobox = Combobox;
exports.ComboboxDivider = ComboboxDivider;
exports.ComboboxFooter = ComboboxFooter;
exports.ComboboxGroup = ComboboxGroup;
exports.ComboboxHeader = ComboboxHeader;
exports.ComboboxItem = ComboboxItem;
