// @salutejs/plasma-new-hope/es/components/Tabs/ui/Tabs/Tabs.styles.js
import './Tabs.styles_1b8qt33.css';
import { styled } from '@linaria/react';
import { tokens } from '../../tokens.js';

var base = "bu8v92r";
var StyledContent = /*#__PURE__*/styled('div')({
  name: "StyledContent",
  "class": "s8qwlso",
  propsAsIs: false
});
var StyledContentWrapper = /*#__PURE__*/styled('div')({
  name: "StyledContentWrapper",
  "class": "sh5e38r",
  propsAsIs: false
});
var _exp2 = function _exp2() {
  return function (_ref) {
    var isFilled = _ref.isFilled;
    return isFilled ? "var(".concat(tokens.arrowOuterPadding, ")") : '';
  };
};
var _exp3 = function _exp3() {
  return function (_ref2) {
    var isLeftArrow = _ref2.isLeftArrow;
    return isLeftArrow ? "var(".concat(tokens.arrowInnerPadding, ")") : 'var(--plasma_private-outer-padding)';
  };
};
var _exp4 = function _exp4() {
  return function (_ref3) {
    var isLeftArrow = _ref3.isLeftArrow;
    return isLeftArrow ? 'var(--plasma_private-outer-padding)' : "var(".concat(tokens.arrowInnerPadding, ")");
  };
};
var StyledArrow = /*#__PURE__*/styled('button')({
  name: "StyledArrow",
  "class": "s1ekwj92",
  propsAsIs: false,
  vars: {
    "s1ekwj92-0": [/*#__PURE__*/_exp2()],
    "s1ekwj92-1": [/*#__PURE__*/_exp3()],
    "s1ekwj92-2": [/*#__PURE__*/_exp4()]
  }
});

export { StyledArrow, StyledContent, StyledContentWrapper, base };
