// @salutejs/plasma-tokens/brands/sbermarket_metro/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.onDarkTextAccentGoldActive = exports.onDarkTextAccentGoldHover = exports.onDarkTextAccentDiscount = exports.onDarkTextAccentDiscountActive = exports.onDarkTextAccentDiscountHover = exports.onDarkTextNegative = exports.onDarkTextNegativeActive = exports.onDarkTextNegativeHover = exports.onDarkTextAccent = exports.onDarkTextAccentActive = exports.onDarkTextAccentHover = exports.onDarkTextParagraph = exports.onDarkTextParagraphActive = exports.onDarkTextParagraphHover = exports.onDarkTextTertiary = exports.onDarkTextTertiaryActive = exports.onDarkTextTertiaryHover = exports.onDarkTextSecondary = exports.onDarkTextSecondaryActive = exports.onDarkTextSecondaryHover = exports.onDarkTextPrimary = exports.onDarkTextPrimaryActive = exports.onDarkTextPrimaryHover = exports.textAccentB2B = exports.textAccentB2BActive = exports.textAccentB2BHover = exports.textAccentGold = exports.textAccentGoldActive = exports.textAccentGoldHover = exports.textAccentDiscount = exports.textAccentDiscountActive = exports.textAccentDiscountHover = exports.textNegative = exports.textNegativeActive = exports.textNegativeHover = exports.textAccent = exports.textAccentActive = exports.textAccentHover = exports.textParagraph = exports.textParagraphActive = exports.textParagraphHover = exports.textTertiary = exports.textTertiaryActive = exports.textTertiaryHover = exports.textSecondary = exports.textSecondaryActive = exports.textSecondaryHover = exports.textPrimary = exports.textPrimaryActive = exports.textPrimaryHover = void 0;
exports.surfaceNegativeHover = exports.surfaceAccentGradient = exports.surfaceAccentGradientActive = exports.surfaceAccentGradientHover = exports.surfaceAccent = exports.surfaceAccentActive = exports.surfaceAccentHover = exports.surfaceSolidCard = exports.surfaceSolidCardActive = exports.surfaceSolidCardHover = exports.surfaceSolidTertiary = exports.surfaceSolidTertiaryActive = exports.surfaceSolidTertiaryHover = exports.surfaceSolidSecondary = exports.surfaceSolidSecondaryActive = exports.surfaceSolidSecondaryHover = exports.inverseTextPrimary = exports.inverseTextPrimaryActive = exports.inverseTextPrimaryHover = exports.onLightTextAccentB2B = exports.onLightTextAccentB2BActive = exports.onLightTextAccentB2BHover = exports.onLightTextAccentGold = exports.onLightTextAccentGoldActive = exports.onLightTextAccentGoldHover = exports.onLightTextAccentDiscount = exports.onLightTextAccentDiscountActive = exports.onLightTextAccentDiscountHover = exports.onLightTextNegative = exports.onLightTextNegativeActive = exports.onLightTextNegativeHover = exports.onLightTextAccent = exports.onLightTextAccentActive = exports.onLightTextAccentHover = exports.onLightTextParagraph = exports.onLightTextParagraphActive = exports.onLightTextParagraphHover = exports.onLightTextTertiary = exports.onLightTextTertiaryActive = exports.onLightTextTertiaryHover = exports.onLightTextSecondary = exports.onLightTextSecondaryActive = exports.onLightTextSecondaryHover = exports.onLightTextPrimary = exports.onLightTextPrimaryActive = exports.onLightTextPrimaryHover = exports.onDarkTextAccentB2B = exports.onDarkTextAccentB2BActive = exports.onDarkTextAccentB2BHover = exports.onDarkTextAccentGold = void 0;
exports.onDarkSurfaceAdditional = exports.onDarkSurfaceAdditionalActive = exports.onDarkSurfaceAdditionalHover = exports.onDarkSurfaceAccentBlue = exports.onDarkSurfaceAccentBlueActive = exports.onDarkSurfaceAccentBlueHover = exports.onDarkSurfaceAccentGold = exports.onDarkSurfaceAccentGoldActive = exports.onDarkSurfaceAccentGoldHover = exports.onDarkSurfaceAccentDiscount = exports.onDarkSurfaceAccentDiscountActive = exports.onDarkSurfaceAccentDiscountHover = exports.onDarkSurfaceAccentB2B = exports.onDarkSurfaceAccentB2BActive = exports.onDarkSurfaceAccentB2BHover = exports.onDarkSurfaceNegative = exports.onDarkSurfaceNegativeActive = exports.onDarkSurfaceNegativeHover = exports.onDarkSurfaceAccentGradient = exports.onDarkSurfaceAccentGradientActive = exports.onDarkSurfaceAccentGradientHover = exports.onDarkSurfaceAccent = exports.onDarkSurfaceAccentActive = exports.onDarkSurfaceAccentHover = exports.onDarkSurfaceSolidCard = exports.onDarkSurfaceSolidCardActive = exports.onDarkSurfaceSolidCardHover = exports.onDarkSurfaceSolidTertiary = exports.onDarkSurfaceSolidTertiaryActive = exports.onDarkSurfaceSolidTertiaryHover = exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidSecondaryActive = exports.onDarkSurfaceSolidSecondaryHover = exports.surfaceAdditional = exports.surfaceAdditionalActive = exports.surfaceAdditionalHover = exports.surfaceAccentBlue = exports.surfaceAccentBlueActive = exports.surfaceAccentBlueHover = exports.surfaceAccentGold = exports.surfaceAccentGoldActive = exports.surfaceAccentGoldHover = exports.surfaceAccentDiscount = exports.surfaceAccentDiscountActive = exports.surfaceAccentDiscountHover = exports.surfaceAccentB2B = exports.surfaceAccentB2BActive = exports.surfaceAccentB2BHover = exports.surfaceNegative = exports.surfaceNegativeActive = void 0;
exports.primary = exports.text = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.skeletonGradientLighter = exports.skeletonGradient = exports.bannersDefaultBannerBlue = exports.bannersDefaultBannerMalachite = exports.bannersDefaultBannerGreen = exports.bannersDefaultBannerGold = exports.bannersDefaultBannerRed = exports.backgroundOverlay = exports.backgroundPrimary = exports.onLightSurfaceAdditional = exports.onLightSurfaceAdditionalActive = exports.onLightSurfaceAdditionalHover = exports.onLightSurfaceAccentBlue = exports.onLightSurfaceAccentBlueActive = exports.onLightSurfaceAccentBlueHover = exports.onLightSurfaceAccentGold = exports.onLightSurfaceAccentGoldActive = exports.onLightSurfaceAccentGoldHover = exports.onLightSurfaceAccentDiscount = exports.onLightSurfaceAccentDiscountActive = exports.onLightSurfaceAccentDiscountHover = exports.onLightSurfaceAccentB2B = exports.onLightSurfaceAccentB2BActive = exports.onLightSurfaceAccentB2BHover = exports.onLightSurfaceNegative = exports.onLightSurfaceNegativeActive = exports.onLightSurfaceNegativeHover = exports.onLightSurfaceAccentGradient = exports.onLightSurfaceAccentGradientActive = exports.onLightSurfaceAccentGradientHover = exports.onLightSurfaceAccent = exports.onLightSurfaceAccentActive = exports.onLightSurfaceAccentHover = exports.onLightSurfaceSolidCard = exports.onLightSurfaceSolidCardActive = exports.onLightSurfaceSolidCardHover = exports.onLightSurfaceSolidTertiary = exports.onLightSurfaceSolidTertiaryActive = exports.onLightSurfaceSolidTertiaryHover = exports.onLightSurfaceSolidSecondary = exports.onLightSurfaceSolidSecondaryActive = exports.onLightSurfaceSolidSecondaryHover = void 0;
exports.spacing60x = exports.spacing40x = exports.spacing32x = exports.spacing24x = exports.spacing20x = exports.spacing16x = exports.spacing12x = exports.spacing10x = exports.spacing8x = exports.spacing6x = exports.spacing4x = exports.spacing3x = exports.spacing2x = exports.spacing1x = exports.spacing0x = exports.borderRadiusXxl = exports.borderRadiusXl = exports.borderRadiusL = exports.borderRadiusM = exports.borderRadiusS = exports.borderRadiusXs = exports.borderRadiusXxs = exports.shadowDownHardL = exports.shadowDownHardM = exports.shadowDownHardS = exports.shadowDownSoftL = exports.shadowDownSoftM = exports.shadowDownSoftS = exports.buttonCritical = exports.buttonAccent = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.critical = exports.accent = exports.background = exports.paragraph = exports.tertiary = exports.secondary = void 0;
/** Основной цвет текста */
exports.textPrimaryHover = 'var(--text-primary-hover, #B3C5D0FF)';
/** Основной цвет текста */
exports.textPrimaryActive = 'var(--text-primary-active, #D3DDE4FF)';
/** Основной цвет текста */
exports.textPrimary = 'var(--text-primary, #F2F5F7)';
/** Вторичный цвет текста */
exports.textSecondaryHover = 'var(--text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста */
exports.textSecondaryActive = 'var(--text-secondary-active, #353E46FF)';
/** Вторичный цвет текста */
exports.textSecondary = 'var(--text-secondary, #A8B4BD)';
/** Третичный цвет текста */
exports.textTertiaryHover = 'var(--text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста */
exports.textTertiaryActive = 'var(--text-tertiary-active, #5B6267FF)';
/** Третичный цвет текста */
exports.textTertiary = 'var(--text-tertiary, #7D868C)';
/** Сплошной наборный текст */
exports.textParagraphHover = 'var(--text-paragraph-hover, #FFFFFFFF)';
/** Сплошной наборный текст */
exports.textParagraphActive = 'var(--text-paragraph-active, #262F36FF)';
/** Сплошной наборный текст */
exports.textParagraph = 'var(--text-paragraph, #BEC9D1)';
/** Акцентный цвет */
exports.textAccentHover = 'var(--text-accent-hover, #5297FFFF)';
/** Акцентный цвет */
exports.textAccentActive = 'var(--text-accent-active, #005AE0FF)';
/** Акцентный цвет */
exports.textAccent = 'var(--text-accent, #0064FE)';
/** Цвет ошибки */
exports.textNegativeHover = 'var(--text-negative-hover, #FF4D5EFF)';
/** Цвет ошибки */
exports.textNegativeActive = 'var(--text-negative-active, #FF0F27FF)';
/** Цвет ошибки */
exports.textNegative = 'var(--text-negative, #FF2E43)';
/** Дополнительный акцент */
exports.textAccentDiscountHover = 'var(--text-accent-discount-hover, #FF3D8BFF)';
/** Дополнительный акцент */
exports.textAccentDiscountActive = 'var(--text-accent-discount-active, #FF0066FF)';
/** Дополнительный акцент */
exports.textAccentDiscount = 'var(--text-accent-discount, #FF1F78)';
/** Дополнительный акцент */
exports.textAccentGoldHover = 'var(--text-accent-gold-hover, #FFDE5CFF)';
/** Дополнительный акцент */
exports.textAccentGoldActive = 'var(--text-accent-gold-active, #FFD21FFF)';
/** Дополнительный акцент */
exports.textAccentGold = 'var(--text-accent-gold, #FFD83D)';
/** Акцентный цвет бренда */
exports.textAccentB2BHover = 'var(--text-accent-b2-b-hover, #23F7FBFF)';
/** Акцентный цвет бренда */
exports.textAccentB2BActive = 'var(--text-accent-b2-b-active, #03A7AAFF)';
/** Акцентный цвет бренда */
exports.textAccentB2B = 'var(--text-accent-b2-b, #04C6C9)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimaryHover = 'var(--on-dark-text-primary-hover, #F2F5F793)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimaryActive = 'var(--on-dark-text-primary-active, #F2F5F7C4)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = 'var(--on-dark-text-primary, rgba(242, 245, 247, 0.96))';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondaryHover = 'var(--on-dark-text-secondary-hover, #F2F5F7FF)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondaryActive = 'var(--on-dark-text-secondary-active, #F2F5F7C4)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = 'var(--on-dark-text-secondary, rgba(242, 245, 247, 0.64))';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiaryHover = 'var(--on-dark-text-tertiary-hover, #F2F5F7FF)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiaryActive = 'var(--on-dark-text-tertiary-active, #F2F5F77A)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = 'var(--on-dark-text-tertiary, rgba(242, 245, 247, 0.4))';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraphHover = 'var(--on-dark-text-paragraph-hover, #F2F5F77A)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraphActive = 'var(--on-dark-text-paragraph-active, #F2F5F7A3)';
/** Сплошной наборный текст на темном фоне */
exports.onDarkTextParagraph = 'var(--on-dark-text-paragraph, rgba(242, 245, 247, 0.8))';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccentHover = 'var(--on-dark-text-accent-hover, #5297FFFF)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccentActive = 'var(--on-dark-text-accent-active, #005AE0FF)';
/** Акцентный цвет на темном фоне */
exports.onDarkTextAccent = 'var(--on-dark-text-accent, #0064FE)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegativeHover = 'var(--on-dark-text-negative-hover, #FF4D5EFF)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegativeActive = 'var(--on-dark-text-negative-active, #FF0F27FF)';
/** Цвет ошибки на темном фоне */
exports.onDarkTextNegative = 'var(--on-dark-text-negative, #FF2E43)';
/** Дополнительный акцент */
exports.onDarkTextAccentDiscountHover = 'var(--on-dark-text-accent-discount-hover, #FF3D8BFF)';
/** Дополнительный акцент */
exports.onDarkTextAccentDiscountActive = 'var(--on-dark-text-accent-discount-active, #FF0066FF)';
/** Дополнительный акцент */
exports.onDarkTextAccentDiscount = 'var(--on-dark-text-accent-discount, #FF1F78)';
/** Дополнительный акцент */
exports.onDarkTextAccentGoldHover = 'var(--on-dark-text-accent-gold-hover, #FFDE5CFF)';
/** Дополнительный акцент */
exports.onDarkTextAccentGoldActive = 'var(--on-dark-text-accent-gold-active, #FFD21FFF)';
/** Дополнительный акцент */
exports.onDarkTextAccentGold = 'var(--on-dark-text-accent-gold, #FFD83D)';
/** Акцентный цвет бренда */
exports.onDarkTextAccentB2BHover = 'var(--on-dark-text-accent-b2-b-hover, #23F7FBFF)';
/** Акцентный цвет бренда */
exports.onDarkTextAccentB2BActive = 'var(--on-dark-text-accent-b2-b-active, #03A7AAFF)';
/** Акцентный цвет бренда */
exports.onDarkTextAccentB2B = 'var(--on-dark-text-accent-b2-b, #04C6C9)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimaryHover = 'var(--on-light-text-primary-hover, #FFFFFFFF)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimaryActive = 'var(--on-light-text-primary-active, #A6AEB5FF)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = 'var(--on-light-text-primary, #24282B)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondaryHover = 'var(--on-light-text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondaryActive = 'var(--on-light-text-secondary-active, #51585DFF)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = 'var(--on-light-text-secondary, #8C959C)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiaryHover = 'var(--on-light-text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiaryActive = 'var(--on-light-text-tertiary-active, #262F36FF)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = 'var(--on-light-text-tertiary, #BEC9D1)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraphHover = 'var(--on-light-text-paragraph-hover, #FFFFFFFF)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraphActive = 'var(--on-light-text-paragraph-active, #A6AEB5FF)';
/** Сплошной наборный текст на светлом фоне */
exports.onLightTextParagraph = 'var(--on-light-text-paragraph, #24282B)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccentHover = 'var(--on-light-text-accent-hover, #5297FFFF)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccentActive = 'var(--on-light-text-accent-active, #005AE0FF)';
/** Акцентный цвет на светлом фоне */
exports.onLightTextAccent = 'var(--on-light-text-accent, #0064FE)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegativeHover = 'var(--on-light-text-negative-hover, #F5384BFF)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegativeActive = 'var(--on-light-text-negative-active, #E40C22FF)';
/** Цвет ошибки на светлом фоне */
exports.onLightTextNegative = 'var(--on-light-text-negative, #F31B31)';
/** Дополнительный акцент */
exports.onLightTextAccentDiscountHover = 'var(--on-light-text-accent-discount-hover, #FF3D8BFF)';
/** Дополнительный акцент */
exports.onLightTextAccentDiscountActive = 'var(--on-light-text-accent-discount-active, #FF0066FF)';
/** Дополнительный акцент */
exports.onLightTextAccentDiscount = 'var(--on-light-text-accent-discount, #FF1F78)';
/** Дополнительный акцент */
exports.onLightTextAccentGoldHover = 'var(--on-light-text-accent-gold-hover, #FFDE5CFF)';
/** Дополнительный акцент */
exports.onLightTextAccentGoldActive = 'var(--on-light-text-accent-gold-active, #FFD21FFF)';
/** Дополнительный акцент */
exports.onLightTextAccentGold = 'var(--on-light-text-accent-gold, #FFD83D)';
/** Акцентный цвет бренда */
exports.onLightTextAccentB2BHover = 'var(--on-light-text-accent-b2-b-hover, #11BDC0FF)';
/** Акцентный цвет бренда */
exports.onLightTextAccentB2BActive = 'var(--on-light-text-accent-b2-b-active, #0EA2A4FF)';
/** Акцентный цвет бренда */
exports.onLightTextAccentB2B = 'var(--on-light-text-accent-b2-b, #0C8688)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimaryHover = 'var(--inverse-text-primary-hover, #FFFFFFFF)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimaryActive = 'var(--inverse-text-primary-active, #A6AEB5FF)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimary = 'var(--inverse-text-primary, #24282B)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondaryHover = 'var(--surface-solid-secondary-hover, #2C3135FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondaryActive = 'var(--surface-solid-secondary-active, #1E2224FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondary = 'var(--surface-solid-secondary, #24282B)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiaryHover = 'var(--surface-solid-tertiary-hover, #3A4145FF)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiaryActive = 'var(--surface-solid-tertiary-active, #2C3135FF)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = 'var(--surface-solid-tertiary, #31373B)';
/** Основной фон для карточек */
exports.surfaceSolidCardHover = 'var(--surface-solid-card-hover, #191D1FFF)';
/** Основной фон для карточек */
exports.surfaceSolidCardActive = 'var(--surface-solid-card-active, #101214FF)';
/** Основной фон для карточек */
exports.surfaceSolidCard = 'var(--surface-solid-card, #15181A)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentHover = 'var(--surface-accent-hover, #1F78FFFF)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentActive = 'var(--surface-accent-active, #0062F5FF)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccent = 'var(--surface-accent, #0064FE)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradientHover = 'var(--surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradientActive = 'var(--surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%))';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeHover = 'var(--surface-negative-hover, #FF4D5EFF)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeActive = 'var(--surface-negative-active, #FF243AFF)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = 'var(--surface-negative, #FF2E43)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentB2BHover = 'var(--surface-accent-b2-b-hover, #0EBBBEFF)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentB2BActive = 'var(--surface-accent-b2-b-active, #0C9FA1FF)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentB2B = 'var(--surface-accent-b2-b, #0DA8AB)';
/** Дополнительный акцент для обозначения акций */
exports.surfaceAccentDiscountHover = 'var(--surface-accent-discount-hover, #FF3D8BFF)';
/** Дополнительный акцент для обозначения акций */
exports.surfaceAccentDiscountActive = 'var(--surface-accent-discount-active, #FF1472FF)';
/** Дополнительный акцент для обозначения акций */
exports.surfaceAccentDiscount = 'var(--surface-accent-discount, #FF1F78)';
/** Дополнительный акцент */
exports.surfaceAccentGoldHover = 'var(--surface-accent-gold-hover, #FFDE5CFF)';
/** Дополнительный акцент */
exports.surfaceAccentGoldActive = 'var(--surface-accent-gold-active, #FFD633FF)';
/** Дополнительный акцент */
exports.surfaceAccentGold = 'var(--surface-accent-gold, #FFD83D)';
/** Дополнительный акцент */
exports.surfaceAccentBlueHover = 'var(--surface-accent-blue-hover, #75A5FFFF)';
/** Дополнительный акцент */
exports.surfaceAccentBlueActive = 'var(--surface-accent-blue-active, #4D8BFFFF)';
/** Дополнительный акцент */
exports.surfaceAccentBlue = 'var(--surface-accent-blue, #5993FF)';
/** Пунктирные дивайдеры и неактивные контролы */
exports.surfaceAdditionalHover = 'var(--surface-additional-hover, #535A5FFF)';
/** Пунктирные дивайдеры и неактивные контролы */
exports.surfaceAdditionalActive = 'var(--surface-additional-active, #454A4FFF)';
/** Пунктирные дивайдеры и неактивные контролы */
exports.surfaceAdditional = 'var(--surface-additional, #494F54)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondaryHover = 'var(--on-dark-surface-solid-secondary-hover, #2C3135FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondaryActive = 'var(--on-dark-surface-solid-secondary-active, #1E2224FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #24282B)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiaryHover = 'var(--on-dark-surface-solid-tertiary-hover, #3A4145FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiaryActive = 'var(--on-dark-surface-solid-tertiary-active, #2C3135FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #31373B)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCardHover = 'var(--on-dark-surface-solid-card-hover, #191D1FFF)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCardActive = 'var(--on-dark-surface-solid-card-active, #101214FF)';
/** Основной фон для карточек на темном фоне */
exports.onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #15181A)';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccentHover = 'var(--on-dark-surface-accent-hover, #1F78FFFF)';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccentActive = 'var(--on-dark-surface-accent-active, #0062F5FF)';
/** Акцентный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #0064FE)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradientHover = 'var(--on-dark-surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradientActive = 'var(--on-dark-surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
exports.onDarkSurfaceAccentGradient = 'var(--on-dark-surface-accent-gradient, linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%))';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegativeHover = 'var(--on-dark-surface-negative-hover, #FF4D5EFF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegativeActive = 'var(--on-dark-surface-negative-active, #FF243AFF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
exports.onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF2E43)';
/** Акцентный фон поверхности/контрола */
exports.onDarkSurfaceAccentB2BHover = 'var(--on-dark-surface-accent-b2-b-hover, #0EBBBEFF)';
/** Акцентный фон поверхности/контрола */
exports.onDarkSurfaceAccentB2BActive = 'var(--on-dark-surface-accent-b2-b-active, #0C9FA1FF)';
/** Акцентный фон поверхности/контрола */
exports.onDarkSurfaceAccentB2B = 'var(--on-dark-surface-accent-b2-b, #0DA8AB)';
/** Дополнительный акцент для обозначения акций */
exports.onDarkSurfaceAccentDiscountHover = 'var(--on-dark-surface-accent-discount-hover, #FF3D8BFF)';
/** Дополнительный акцент для обозначения акций */
exports.onDarkSurfaceAccentDiscountActive = 'var(--on-dark-surface-accent-discount-active, #FF1472FF)';
/** Дополнительный акцент для обозначения акций */
exports.onDarkSurfaceAccentDiscount = 'var(--on-dark-surface-accent-discount, #FF1F78)';
/** Дополнительный акцент */
exports.onDarkSurfaceAccentGoldHover = 'var(--on-dark-surface-accent-gold-hover, #FFDE5CFF)';
/** Дополнительный акцент */
exports.onDarkSurfaceAccentGoldActive = 'var(--on-dark-surface-accent-gold-active, #FFD633FF)';
/** Дополнительный акцент */
exports.onDarkSurfaceAccentGold = 'var(--on-dark-surface-accent-gold, #FFD83D)';
/** Дополнительный акцент */
exports.onDarkSurfaceAccentBlueHover = 'var(--on-dark-surface-accent-blue-hover, #75A5FFFF)';
/** Дополнительный акцент */
exports.onDarkSurfaceAccentBlueActive = 'var(--on-dark-surface-accent-blue-active, #4D8BFFFF)';
/** Дополнительный акцент */
exports.onDarkSurfaceAccentBlue = 'var(--on-dark-surface-accent-blue, #5993FF)';
/** Пунктирные дивайдеры и неактивные контролы */
exports.onDarkSurfaceAdditionalHover = 'var(--on-dark-surface-additional-hover, #535A5FFF)';
/** Пунктирные дивайдеры и неактивные контролы */
exports.onDarkSurfaceAdditionalActive = 'var(--on-dark-surface-additional-active, #454A4FFF)';
/** Пунктирные дивайдеры и неактивные контролы */
exports.onDarkSurfaceAdditional = 'var(--on-dark-surface-additional, #494F54)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondaryHover = 'var(--on-light-surface-solid-secondary-hover, #F9FAFBFF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondaryActive = 'var(--on-light-surface-solid-secondary-active, #ECF0F3FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #F2F5F7)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidTertiaryHover = 'var(--on-light-surface-solid-tertiary-hover, #F2F5F8FF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidTertiaryActive = 'var(--on-light-surface-solid-tertiary-active, #DEE7EDFF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #E4EBF0)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCardHover = 'var(--on-light-surface-solid-card-hover, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCardActive = 'var(--on-light-surface-solid-card-active, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
exports.onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccentHover = 'var(--on-light-surface-accent-hover, #1F78FFFF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccentActive = 'var(--on-light-surface-accent-active, #0062F5FF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceAccent = 'var(--on-light-surface-accent, #0064FE)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradientHover = 'var(--on-light-surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradientActive = 'var(--on-light-surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
exports.onLightSurfaceAccentGradient = 'var(--on-light-surface-accent-gradient, linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%))';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegativeHover = 'var(--on-light-surface-negative-hover, #F5384BFF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegativeActive = 'var(--on-light-surface-negative-active, #F31228FF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
exports.onLightSurfaceNegative = 'var(--on-light-surface-negative, #F31B31)';
/** Акцентный фон поверхности/контрола */
exports.onLightSurfaceAccentB2BHover = 'var(--on-light-surface-accent-b2-b-hover, #0EBBBEFF)';
/** Акцентный фон поверхности/контрола */
exports.onLightSurfaceAccentB2BActive = 'var(--on-light-surface-accent-b2-b-active, #0C9FA1FF)';
/** Акцентный фон поверхности/контрола */
exports.onLightSurfaceAccentB2B = 'var(--on-light-surface-accent-b2-b, #0DA8AB)';
/** Дополнительный акцент для обозначения акций */
exports.onLightSurfaceAccentDiscountHover = 'var(--on-light-surface-accent-discount-hover, #FF3D8BFF)';
/** Дополнительный акцент для обозначения акций */
exports.onLightSurfaceAccentDiscountActive = 'var(--on-light-surface-accent-discount-active, #FF1472FF)';
/** Дополнительный акцент для обозначения акций */
exports.onLightSurfaceAccentDiscount = 'var(--on-light-surface-accent-discount, #FF1F78)';
/** Дополнительный акцент */
exports.onLightSurfaceAccentGoldHover = 'var(--on-light-surface-accent-gold-hover, #FFDE5CFF)';
/** Дополнительный акцент */
exports.onLightSurfaceAccentGoldActive = 'var(--on-light-surface-accent-gold-active, #FFD633FF)';
/** Дополнительный акцент */
exports.onLightSurfaceAccentGold = 'var(--on-light-surface-accent-gold, #FFD83D)';
/** Дополнительный акцент */
exports.onLightSurfaceAccentBlueHover = 'var(--on-light-surface-accent-blue-hover, #5D95FDFF)';
/** Дополнительный акцент */
exports.onLightSurfaceAccentBlueActive = 'var(--on-light-surface-accent-blue-active, #357BFDFF)';
/** Дополнительный акцент */
exports.onLightSurfaceAccentBlue = 'var(--on-light-surface-accent-blue, #3F81FD)';
/** Пунктирные дивайдеры и неактивные контролы */
exports.onLightSurfaceAdditionalHover = 'var(--on-light-surface-additional-hover, #DEE7EDFF)';
/** Пунктирные дивайдеры и неактивные контролы */
exports.onLightSurfaceAdditionalActive = 'var(--on-light-surface-additional-active, #CBD8E2FF)';
/** Пунктирные дивайдеры и неактивные контролы */
exports.onLightSurfaceAdditional = 'var(--on-light-surface-additional, #D1DDE5)';
/** Основной фон */
exports.backgroundPrimary = 'var(--background-primary, #08090A)';
/** Вторичный фон */
exports.backgroundOverlay = 'var(--background-overlay, rgba(21, 24, 26, 0.56))';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerRed = 'var(--banners-default-banner-red, #2E090D)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerGold = 'var(--banners-default-banner-gold, #1F1903)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerGreen = 'var(--banners-default-banner-green, #071F0C)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerMalachite = 'var(--banners-default-banner-malachite, #0A1C1C)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
exports.bannersDefaultBannerBlue = 'var(--banners-default-banner-blue, #0A162B)';
exports.skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
exports.skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use textPrimary */
exports.text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
exports.primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
exports.secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
exports.tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use textParagraph */
exports.paragraph = 'var(--plasma-colors-paragraph, var(--text-paragraph))';
/** @deprecated instead use backgroundPrimary */
exports.background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textAccent */
exports.accent = 'var(--plasma-colors-accent, var(--text-accent))';
/** @deprecated instead use textNegative */
exports.critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use textAccent */
exports.buttonAccent = 'var(--plasma-colors-button-accent, var(--text-accent))';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
exports.shadowDownSoftS = 'var(--shadow-down-soft-s, 0px 6px 12px rgba(0, 0, 0, 0.08))';
exports.shadowDownSoftM = 'var(--shadow-down-soft-m, 0 12px 24px rgba(0, 0, 0, 0.12))';
exports.shadowDownSoftL = 'var(--shadow-down-soft-l, 0 24px 48px rgba(0, 0, 0, 0.16))';
exports.shadowDownHardS = 'var(--shadow-down-hard-s, 0px 6px 12px rgba(0, 0, 0, 0.12))';
exports.shadowDownHardM = 'var(--shadow-down-hard-m, 0 12px 24px rgba(0, 0, 0, 0.16))';
exports.shadowDownHardL = 'var(--shadow-down-hard-l, 0 24px 48px rgba(0, 0, 0, 0.2))';
exports.borderRadiusXxs = 'var(--border-radius-xxs, 4px)';
exports.borderRadiusXs = 'var(--border-radius-xs, 8px)';
exports.borderRadiusS = 'var(--border-radius-s, 12px)';
exports.borderRadiusM = 'var(--border-radius-m, 16px)';
exports.borderRadiusL = 'var(--border-radius-l, 20px)';
exports.borderRadiusXl = 'var(--border-radius-xl, 24px)';
exports.borderRadiusXxl = 'var(--border-radius-xxl, 32px)';
exports.spacing0x = 'var(--spacing0x, 0px)';
exports.spacing1x = 'var(--spacing1x, 2px)';
exports.spacing2x = 'var(--spacing2x, 4px)';
exports.spacing3x = 'var(--spacing3x, 6px)';
exports.spacing4x = 'var(--spacing4x, 8px)';
exports.spacing6x = 'var(--spacing6x, 12px)';
exports.spacing8x = 'var(--spacing8x, 16px)';
exports.spacing10x = 'var(--spacing10x, 20px)';
exports.spacing12x = 'var(--spacing12x, 24px)';
exports.spacing16x = 'var(--spacing16x, 32px)';
exports.spacing20x = 'var(--spacing20x, 40px)';
exports.spacing24x = 'var(--spacing24x, 48px)';
exports.spacing32x = 'var(--spacing32x, 64px)';
exports.spacing40x = 'var(--spacing40x, 80px)';
exports.spacing60x = 'var(--spacing60x, 120px)';
