// @salutejs/plasma-hope/es/utils/mediaQuery.js
import { css } from 'styled-components';
export var breakpoints = {
  XXL: 1200,
  // Desktop large
  XL: 960,
  // TV
  L: 769,
  // Portal 8 cols
  M: 560,
  // Portal 6 cols
  S: 0 // Mobile
};
export var columns = {
  XXL: 16,
  XL: 12,
  L: 8,
  M: 6,
  S: 4
};
export var margins = {
  XXL: 4,
  XL: 4,
  L: 3.5,
  M: 3.5,
  S: 1
};
export var gutters = {
  XXL: 1,
  XL: 1,
  L: 1,
  M: 1,
  S: 0.5
};
export var sizes = /*#__PURE__*/Object.keys(breakpoints);

/**
 * Обертка над css-медиазапросами с заранее определенными брейкпоинтами.
 */
export var mediaQuery = function mediaQuery(breakpoint) {
  var deviceScale = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 1;
  var index = sizes.indexOf(breakpoint);
  var nextBreakpoint = sizes[index - 1];
  var min = breakpoints[breakpoint] > 0 ? breakpoints[breakpoint] * deviceScale : null;
  var max = breakpoints[nextBreakpoint] ? (breakpoints[nextBreakpoint] - 1) * deviceScale : null;
  if (min === null && max !== null) {
    return function (content) {
      return css(["@media (max-width:", "px){", "}"], max, content);
    };
  }
  if (min !== null && max !== null) {
    return function (content) {
      return css(["@media screen and (min-width:", "px) and (max-width:", "px){", "}"], min, max, content);
    };
  }
  return function (content) {
    return css(["@media (min-width:", "px){", "}"], min, content);
  };
};
