// @salutejs/plasma-new-hope/styled-components/cjs/engines/styled-components.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports._component = void 0;
Object.defineProperty(exports, "css", {
  enumerable: true,
  get: function get() {
    return _styledComponents.css;
  }
});
Object.defineProperty(exports, "styled", {
  enumerable: true,
  get: function get() {
    return _styledComponents["default"];
  }
});
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _common = /*#__PURE__*/require("./common");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var Root = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-1xug4g9-0"
})(["", ";", ";", ";"], function (_ref) {
  var base = _ref.base;
  return base;
}, function (_ref2) {
  var staticVariants = _ref2.staticVariants;
  return staticVariants;
}, function (_ref3) {
  var dynamicVariants = _ref3.dynamicVariants;
  return dynamicVariants;
});

/* eslint-disable no-underscore-dangle */
var _component = exports._component = function _component(componentConfig) {
  var tag = componentConfig.tag,
    base = componentConfig.base;
  var staticVariants = (0, _common.getStaticVariants)(componentConfig);
  var dynamicVariants = (0, _common.getDynamicVariants)(componentConfig);
  return /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
    return /*#__PURE__*/_react["default"].createElement(Root, _extends({
      as: tag,
      base: base,
      staticVariants: staticVariants,
      dynamicVariants: dynamicVariants
    }, props, {
      ref: ref
    }));
  });
};
