// @salutejs/plasma-new-hope/es/components/Pagination/Pagination.styles.js
import './Pagination.styles_y6gcva.css';
import { styled } from '@linaria/react';
import { buttonConfig } from '../Button/Button.js';
import { buttonGroupConfig } from '../ButtonGroup/ButtonGroup.js';
import '@linaria/core';
import { component, mergeConfig } from '../../engines/common.js';

var mergedButtonConfig = /*#__PURE__*/mergeConfig(buttonConfig);
var Button = /*#__PURE__*/component(mergedButtonConfig);
var mergedButtonGroupConfig = /*#__PURE__*/mergeConfig(buttonGroupConfig);
var ButtonGroup = /*#__PURE__*/component(mergedButtonGroupConfig);
var PaginationRoot = /*#__PURE__*/styled('div')({
  name: "PaginationRoot",
  "class": "p1m4gnut",
  propsAsIs: false
});
var PaginationPages = /*#__PURE__*/styled('div')({
  name: "PaginationPages",
  "class": "p195n7xj",
  propsAsIs: false
});
var PaginationActions = /*#__PURE__*/styled('div')({
  name: "PaginationActions",
  "class": "p19axp12",
  propsAsIs: false
});
var PaginationSection = /*#__PURE__*/styled('div')({
  name: "PaginationSection",
  "class": "p1rlzvhv",
  propsAsIs: false
});
var _exp7 = function _exp7() {
  return ButtonGroup;
};
var PaginationButtonGroup = /*#__PURE__*/styled(_exp7())({
  name: "PaginationButtonGroup",
  "class": "pivhg1z",
  propsAsIs: true
});
var _exp44 = function _exp44() {
  return Button;
};
var PaginationButton = /*#__PURE__*/styled(_exp44())({
  name: "PaginationButton",
  "class": "pshury3",
  propsAsIs: true
});
var _exp93 = function _exp93() {
  return PaginationButton;
};
var PaginationShorter = /*#__PURE__*/styled(_exp93())({
  name: "PaginationShorter",
  "class": "pwlctp1",
  propsAsIs: true
});

export { PaginationActions, PaginationButton, PaginationButtonGroup, PaginationPages, PaginationRoot, PaginationSection, PaginationShorter };
