// @salutejs/plasma-ui/components/Typography/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  BodyL: true,
  BodyM: true,
  BodyS: true,
  BodyXS: true,
  BodyXXS: true,
  DsplL: true,
  DsplM: true,
  DsplS: true,
  H1: true,
  H2: true,
  H3: true,
  H4: true,
  H5: true,
  TextL: true,
  TextM: true,
  TextS: true,
  TextXS: true
};
Object.defineProperty(exports, "BodyL", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.BodyL;
  }
});
Object.defineProperty(exports, "BodyM", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.BodyM;
  }
});
Object.defineProperty(exports, "BodyS", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.BodyS;
  }
});
Object.defineProperty(exports, "BodyXS", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.BodyXS;
  }
});
Object.defineProperty(exports, "BodyXXS", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.BodyXXS;
  }
});
Object.defineProperty(exports, "DsplL", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.DsplL;
  }
});
Object.defineProperty(exports, "DsplM", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.DsplM;
  }
});
Object.defineProperty(exports, "DsplS", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.DsplS;
  }
});
Object.defineProperty(exports, "H1", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.H1;
  }
});
Object.defineProperty(exports, "H2", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.H2;
  }
});
Object.defineProperty(exports, "H3", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.H3;
  }
});
Object.defineProperty(exports, "H4", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.H4;
  }
});
Object.defineProperty(exports, "H5", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.H5;
  }
});
Object.defineProperty(exports, "TextL", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.TextL;
  }
});
Object.defineProperty(exports, "TextM", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.TextM;
  }
});
Object.defineProperty(exports, "TextS", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.TextS;
  }
});
Object.defineProperty(exports, "TextXS", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.TextXS;
  }
});
var _plasmaTypo = /*#__PURE__*/require("@salutejs/plasma-typo");
var _Body = /*#__PURE__*/require("./Body");
Object.keys(_Body).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Body[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Body[key];
    }
  });
});
var _Button = /*#__PURE__*/require("./Button");
Object.keys(_Button).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Button[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Button[key];
    }
  });
});
var _Caption = /*#__PURE__*/require("./Caption");
Object.keys(_Caption).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Caption[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Caption[key];
    }
  });
});
var _Display = /*#__PURE__*/require("./Display");
Object.keys(_Display).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Display[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Display[key];
    }
  });
});
var _Footnote = /*#__PURE__*/require("./Footnote");
Object.keys(_Footnote).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Footnote[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Footnote[key];
    }
  });
});
var _Headline = /*#__PURE__*/require("./Headline");
Object.keys(_Headline).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Headline[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Headline[key];
    }
  });
});
var _Paragraph = /*#__PURE__*/require("./Paragraph");
Object.keys(_Paragraph).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Paragraph[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Paragraph[key];
    }
  });
});
var _Underline = /*#__PURE__*/require("./Underline");
Object.keys(_Underline).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Underline[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Underline[key];
    }
  });
});
