// @salutejs/plasma-new-hope/styled-components/cjs/components/Badge/Badge.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.badgeRoot = exports.badgeConfig = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _utils = /*#__PURE__*/require("../../utils");
var _base = /*#__PURE__*/require("./variations/_view/base");
var _base2 = /*#__PURE__*/require("./variations/_size/base");
var _base3 = /*#__PURE__*/require("./variations/_pilled/base");
var _base4 = /*#__PURE__*/require("./variations/_transparent/base");
var _Badge = /*#__PURE__*/require("./Badge.styles");
var _Badge2 = /*#__PURE__*/require("./Badge.tokens");
var _excluded = ["children", "className", "text", "contentLeft", "contentRight", "size", "view", "pilled", "transparent"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var badgeRoot = exports.badgeRoot = function badgeRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
    var children = props.children,
      className = props.className,
      text = props.text,
      contentLeft = props.contentLeft,
      contentRight = props.contentRight,
      size = props.size,
      view = props.view,
      _props$pilled = props.pilled,
      pilled = _props$pilled === void 0 ? false : _props$pilled,
      _props$transparent = props.transparent,
      transparent = _props$transparent === void 0 ? false : _props$transparent,
      rest = _objectWithoutProperties(props, _excluded);
    var pilledClass = pilled ? _Badge2.classes.badgePilled : undefined;
    var transparentClass = transparent ? _Badge2.classes.badgeTransparent : undefined;
    var txt = !text && typeof children === 'string' ? children : text;
    return /*#__PURE__*/_react["default"].createElement(Root, _extends({
      ref: ref,
      className: (0, _utils.cx)(pilledClass, transparentClass, className),
      view: view,
      size: size,
      pilled: pilled,
      transparent: transparent
    }, rest), contentLeft && /*#__PURE__*/_react["default"].createElement(_Badge.StyledContentLeft, null, contentLeft), txt ? /*#__PURE__*/_react["default"].createElement(_Badge.StyledContentMain, null, txt) : children, contentRight && /*#__PURE__*/_react["default"].createElement(_Badge.StyledContentRight, null, contentRight));
  });
};
var badgeConfig = exports.badgeConfig = {
  name: 'Badge',
  tag: 'div',
  layout: badgeRoot,
  base: _Badge.base,
  variations: {
    view: {
      css: _base.base
    },
    size: {
      css: _base2.base
    },
    pilled: {
      css: _base3.base,
      attrs: true
    },
    transparent: {
      css: _base4.base,
      attrs: true
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};
