// @salutejs/plasma-new-hope/cjs/components/Indicator/Indicator.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var Indicator_styles = require('./Indicator.styles.js');
var base$1 = require('./variations/_view/base.js');
var base = require('./variations/_size/base.js');

var _excluded = ["size", "view"];
var indicatorRoot = function indicatorRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (props, ref) {
    var size = props.size,
      view = props.view,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(props, _excluded);
    return /*#__PURE__*/React.createElement(Root, _rollupPluginBabelHelpers.extends({
      ref: ref,
      size: size,
      view: view
    }, rest));
  });
};
var indicatorConfig = {
  name: 'Indicator',
  tag: 'div',
  layout: indicatorRoot,
  base: Indicator_styles.base,
  variations: {
    size: {
      css: base.base
    },
    view: {
      css: base$1.base
    }
  },
  defaults: {
    size: 's',
    view: 'default'
  }
};

exports.indicatorConfig = indicatorConfig;
exports.indicatorRoot = indicatorRoot;
