// @salutejs/plasma-new-hope/cjs/components/Select/ui/SelectHeader/SelectHeader.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('react');
require('../../../../utils/react.js');
require('@salutejs/plasma-core');
require('@linaria/react');
var DropdownHeader = require('../../../Dropdown/ui/DropdownHeader/DropdownHeader.js');
require('@linaria/core');
require('focus-visible');
require('react-dom');
require('react-popper');
require('styled-components');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var SelectHeader_styles = require('./SelectHeader.styles.js');

var selectHeaderRoot = DropdownHeader.dropdownHeaderRoot;
var selectHeaderConfig = {
  name: 'SelectHeader',
  tag: 'div',
  layout: selectHeaderRoot,
  base: SelectHeader_styles.base,
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

exports.selectHeaderConfig = selectHeaderConfig;
exports.selectHeaderRoot = selectHeaderRoot;
