// @salutejs/plasma-new-hope/styled-components/cjs/examples/plasma_b2c/components/Toolbar/Toolbar.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _Toolbar = /*#__PURE__*/require("../../../../components/Toolbar");
var config = exports.config = {
  defaults: {
    view: 'default',
    size: 'm'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--shadow-down-soft-s);", ":var(--surface-solid-card);", ":var(--surface-transparent-tertiary);"], _Toolbar.toolbarTokens.boxShadow, _Toolbar.toolbarTokens.background, _Toolbar.toolbarTokens.dividerColor)
    },
    size: {
      xs: /*#__PURE__*/(0, _styledComponents.css)(["", ":2.5rem;", ":0.25rem;", ":0.75rem;", ":1.25rem;", ":0.5rem;", ":0.0625rem;"], _Toolbar.toolbarTokens.size, _Toolbar.toolbarTokens.padding, _Toolbar.toolbarTokens.borderRadius, _Toolbar.toolbarTokens.dividerSize, _Toolbar.toolbarTokens.dividerOffset, _Toolbar.toolbarTokens.dividerBorderRadius),
      s: /*#__PURE__*/(0, _styledComponents.css)(["", ":3rem;", ":0.25rem;", ":0.875rem;", ":1.5rem;", ":0.5rem;", ":0.0625rem;"], _Toolbar.toolbarTokens.size, _Toolbar.toolbarTokens.padding, _Toolbar.toolbarTokens.borderRadius, _Toolbar.toolbarTokens.dividerSize, _Toolbar.toolbarTokens.dividerOffset, _Toolbar.toolbarTokens.dividerBorderRadius),
      m: /*#__PURE__*/(0, _styledComponents.css)(["", ":3.75rem;", ":0.375rem;", ":1rem;", ":1.75rem;", ":0.625rem;", ":0.0625rem;"], _Toolbar.toolbarTokens.size, _Toolbar.toolbarTokens.padding, _Toolbar.toolbarTokens.borderRadius, _Toolbar.toolbarTokens.dividerSize, _Toolbar.toolbarTokens.dividerOffset, _Toolbar.toolbarTokens.dividerBorderRadius),
      l: /*#__PURE__*/(0, _styledComponents.css)(["", ":4.5rem;", ":0.5rem;", ":1.25rem;", ":2rem;", ":0.75rem;", ":0.0625rem;"], _Toolbar.toolbarTokens.size, _Toolbar.toolbarTokens.padding, _Toolbar.toolbarTokens.borderRadius, _Toolbar.toolbarTokens.dividerSize, _Toolbar.toolbarTokens.dividerOffset, _Toolbar.toolbarTokens.dividerBorderRadius)
    }
  }
};
