// @salutejs/plasma-hope/mixins/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "addFocus", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.addFocus;
  }
});
Object.defineProperty(exports, "applyBlur", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.applyBlur;
  }
});
Object.defineProperty(exports, "applyDisabled", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.applyDisabled;
  }
});
Object.defineProperty(exports, "applyEllipsis", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.applyEllipsis;
  }
});
Object.defineProperty(exports, "applyHyphens", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.applyHyphens;
  }
});
Object.defineProperty(exports, "applyMaxLines", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.applyMaxLines;
  }
});
Object.defineProperty(exports, "applyNoSelect", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.applyNoSelect;
  }
});
Object.defineProperty(exports, "applyRoundness", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.applyRoundness;
  }
});
Object.defineProperty(exports, "applyScrollSnap", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.applyScrollSnap;
  }
});
Object.defineProperty(exports, "applySpacing", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.applySpacing;
  }
});
Object.defineProperty(exports, "applyView", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.applyView;
  }
});
Object.defineProperty(exports, "blurs", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.blurs;
  }
});
Object.defineProperty(exports, "radiuses", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.radiuses;
  }
});
Object.defineProperty(exports, "shadows", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.shadows;
  }
});
Object.defineProperty(exports, "syntheticFocus", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.syntheticFocus;
  }
});
Object.defineProperty(exports, "views", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.views;
  }
});
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
