// @salutejs/plasma-new-hope/styled-components/cjs/components/TextField/TextField.styles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledTextBefore = exports.StyledTextAfter = exports.StyledContentRight = exports.StyledContentLeft = exports.StyledChips = exports.LeftHelper = exports.Label = exports.InputWrapper = exports.InputLabelWrapper = exports.Input = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _TextField = /*#__PURE__*/require("./TextField.tokens");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var InputWrapper = exports.InputWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-0"
})(["position:relative;display:flex;align-items:center;box-sizing:border-box;"]);
var InputLabelWrapper = exports.InputLabelWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-1"
})(["flex:1;overflow:scroll;position:relative;width:100%;display:inline-flex;align-items:center;::-webkit-scrollbar{display:none;}scrollbar-width:none;overscroll-behavior:contain;&.", "{height:var(", ");border-radius:var(", ");}"], /*#__PURE__*/String(_TextField.classes.hasChips), _TextField.tokens.chipHeight, _TextField.tokens.chipBorderRadius);
var StyledChips = exports.StyledChips = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-2"
})(["display:flex;gap:var(", ");margin-right:var(", ");user-select:none;"], _TextField.tokens.chipGap, _TextField.tokens.chipGap);
var Input = exports.Input = /*#__PURE__*/_styledComponents["default"].input.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-3"
})(["box-sizing:border-box;appearance:none;border:0;padding:0;background-color:transparent;outline:none;flex:1;min-width:60%;"]);
var Label = exports.Label = /*#__PURE__*/_styledComponents["default"].label.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-4"
})([""]);
var StyledContentLeft = exports.StyledContentLeft = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-5"
})(["margin:var(", ");"], _TextField.tokens.leftContentMargin);
var StyledContentRight = exports.StyledContentRight = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-6"
})(["margin:var(", ");"], _TextField.tokens.rightContentMargin);
var LeftHelper = exports.LeftHelper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-7"
})([""]);
var StyledTextBefore = exports.StyledTextBefore = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-8"
})([""]);
var StyledTextAfter = exports.StyledTextAfter = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-9"
})([""]);
