// @salutejs/plasma-new-hope/styled-components/cjs/components/TextField/TextField.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.textFieldRoot = exports.textFieldConfig = exports.base = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _styledComponents = /*#__PURE__*/require("styled-components");
var _utils = /*#__PURE__*/require("../../utils");
var _base = /*#__PURE__*/require("./variations/_size/base");
var _base2 = /*#__PURE__*/require("./variations/_view/base");
var _base3 = /*#__PURE__*/require("./variations/_disabled/base");
var _base4 = /*#__PURE__*/require("./variations/_read-only/base");
var _base5 = /*#__PURE__*/require("./variations/_label-placement/base");
var _TextField = /*#__PURE__*/require("./TextField.styles");
var _TextField2 = /*#__PURE__*/require("./TextField.tokens");
var _ui = /*#__PURE__*/require("./ui");
var _hooks = /*#__PURE__*/require("./hooks");
var _excluded = ["id", "className", "style", "contentLeft", "contentRight", "label", "labelPlacement", "textBefore", "textAfter", "placeholder", "leftHelper", "enumerationType", "view", "size", "readOnly", "disabled", "chips", "onChange", "onChangeChips", "onSearch", "onKeyDown"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var base = exports.base = /*#__PURE__*/(0, _styledComponents.css)(["display:block;overflow:hidden;"]);
var textFieldRoot = exports.textFieldRoot = function textFieldRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (_ref, ref) {
    var id = _ref.id,
      className = _ref.className,
      style = _ref.style,
      contentLeft = _ref.contentLeft,
      contentRight = _ref.contentRight,
      label = _ref.label,
      labelPlacement = _ref.labelPlacement,
      textBefore = _ref.textBefore,
      textAfter = _ref.textAfter,
      placeholder = _ref.placeholder,
      leftHelper = _ref.leftHelper,
      _ref$enumerationType = _ref.enumerationType,
      enumerationType = _ref$enumerationType === void 0 ? 'plain' : _ref$enumerationType,
      view = _ref.view,
      size = _ref.size,
      _ref$readOnly = _ref.readOnly,
      readOnly = _ref$readOnly === void 0 ? false : _ref$readOnly,
      _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? false : _ref$disabled,
      values = _ref.chips,
      onChange = _ref.onChange,
      onChangeChips = _ref.onChangeChips,
      onSearch = _ref.onSearch,
      onKeyDown = _ref.onKeyDown,
      rest = _objectWithoutProperties(_ref, _excluded);
    var contentRef = (0, _react.useRef)(null);
    var inputRef = (0, _react.useRef)(null);
    var inputForkRef = (0, _plasmaCore.useForkRef)(inputRef, ref);
    var chipsRefs = (0, _react.useRef)([]);
    var controlledRefs = {
      contentRef: contentRef,
      inputRef: inputRef,
      chipsRefs: chipsRefs
    };
    var _useState = (0, _react.useState)([]),
      _useState2 = _slicedToArray(_useState, 2),
      chips = _useState2[0],
      setChips = _useState2[1];
    var uniqId = (0, _plasmaCore.safeUseId)();
    var innerId = id || uniqId;
    var labelId = (0, _plasmaCore.safeUseId)();
    var helperTextId = (0, _plasmaCore.safeUseId)();
    var isChipEnumeration = enumerationType === 'chip';
    var hideLabel = (size === 'xs' || isChipEnumeration) && labelPlacement === 'inner';
    var labelInside = size !== 'xs' && labelPlacement === 'inner';
    var innerLabelPlacementValue = hideLabel ? 'outer' : labelPlacement;
    var innerPlaceholderValue = hideLabel ? label : placeholder;
    var innerLabelValue = hideLabel ? undefined : label;
    var isChipsVisible = isChipEnumeration && (chips === null || chips === void 0 ? void 0 : chips.length);
    var withHasChips = isChipsVisible ? _TextField2.classes.hasChips : undefined;
    var wrapperWithoutLeftContent = !contentLeft && isChipsVisible ? _TextField2.classes.hasEmptyContentLeft : undefined;
    var wrapperWithoutRightContent = !contentRight && isChipsVisible ? _TextField2.classes.hasEmptyContentRight : undefined;
    var handleChange = function handleChange(event) {
      if (disabled || readOnly) {
        return;
      }
      var _event$target = event.target,
        maxLength = _event$target.maxLength,
        value = _event$target.value;
      if (maxLength !== -1 && value.length > maxLength) {
        return;
      }
      onChange === null || onChange === void 0 || onChange(event);
    };
    var updateChips = function updateChips(newChips, newValues) {
      setChips(newChips);
      onChangeChips === null || onChangeChips === void 0 || onChangeChips(newValues);
    };
    var _useKeyNavigation = (0, _hooks.useKeyNavigation)({
        controlledRefs: controlledRefs,
        disabled: disabled,
        readOnly: readOnly,
        chips: chips,
        enumerationType: enumerationType,
        updateChips: updateChips,
        onSearch: onSearch,
        onChange: onChange
      }),
      handleInputKeydown = _useKeyNavigation.handleInputKeydown,
      handleChipKeyDown = _useKeyNavigation.handleChipKeyDown,
      onChipClear = _useKeyNavigation.onChipClear,
      handleContentKeyDown = _useKeyNavigation.handleContentKeyDown;
    var onChipClick = function onChipClick(event) {
      return event.stopPropagation();
    };
    var handleInputFocus = function handleInputFocus() {
      if (readOnly || disabled || !(inputRef !== null && inputRef !== void 0 && inputRef.current)) {
        return;
      }
      inputRef.current.scrollTo({
        top: 0,
        left: inputRef.current.offsetLeft,
        behavior: 'smooth'
      });
      inputRef.current.focus();
    };
    var getRef = function getRef(element, index) {
      if (element && chipsRefs !== null && chipsRefs !== void 0 && chipsRefs.current) {
        chipsRefs.current[index] = element;
      }
    };
    var handleOnKeyDown = function handleOnKeyDown(event) {
      handleInputKeydown(event);
      onKeyDown && onKeyDown(event);
    };
    (0, _react.useEffect)(function () {
      if (!isChipEnumeration && !(values !== null && values !== void 0 && values.length)) {
        return;
      }
      var newChips = (values === null || values === void 0 ? void 0 : values.map(function (value, index) {
        return {
          id: "".concat(index, "_").concat(value),
          text: value
        };
      })) || [];
      setChips(newChips);
    }, [isChipEnumeration, values]);
    return /*#__PURE__*/_react["default"].createElement(Root, {
      view: view,
      size: size,
      disabled: disabled,
      readOnly: !disabled && readOnly,
      labelPlacement: innerLabelPlacementValue,
      onClick: handleInputFocus,
      className: className,
      style: style
    }, labelInside || innerLabelValue && /*#__PURE__*/_react["default"].createElement(_TextField.Label, {
      id: labelId,
      htmlFor: id
    }, innerLabelValue), /*#__PURE__*/_react["default"].createElement(_TextField.InputWrapper, {
      className: (0, _utils.cx)(withHasChips, wrapperWithoutLeftContent, wrapperWithoutRightContent)
    }, contentLeft && /*#__PURE__*/_react["default"].createElement(_TextField.StyledContentLeft, null, contentLeft), /*#__PURE__*/_react["default"].createElement(_TextField.InputLabelWrapper, {
      tabIndex: -1,
      ref: contentRef,
      onKeyDown: handleContentKeyDown,
      className: withHasChips
    }, textBefore && /*#__PURE__*/_react["default"].createElement(_TextField.StyledTextBefore, null, textBefore), isChipEnumeration && Boolean(chips === null || chips === void 0 ? void 0 : chips.length) && /*#__PURE__*/_react["default"].createElement(_TextField.StyledChips, null, chips === null || chips === void 0 ? void 0 : chips.map(function (_ref2, index) {
      var chipId = _ref2.id,
        text = _ref2.text;
      return /*#__PURE__*/_react["default"].createElement(_ui.TextFieldChip, {
        id: chipId,
        ref: function ref(element) {
          return getRef(element, index);
        },
        key: "".concat(chipId, "_").concat(index),
        disabled: disabled,
        readOnly: readOnly,
        value: text,
        text: text,
        onKeyDown: function onKeyDown(event) {
          return handleChipKeyDown(event, chipId, index);
        },
        onClear: function onClear() {
          return onChipClear(chipId, index);
        },
        onClick: onChipClick
      });
    })), /*#__PURE__*/_react["default"].createElement(_TextField.Input, _extends({}, rest, {
      ref: inputForkRef,
      id: innerId,
      "aria-labelledby": labelId,
      "aria-describedby": helperTextId,
      placeholder: innerPlaceholderValue,
      disabled: disabled,
      readOnly: !disabled && readOnly,
      onChange: handleChange,
      onKeyDown: handleOnKeyDown
    })), labelInside && /*#__PURE__*/_react["default"].createElement(_TextField.Label, {
      id: labelId,
      htmlFor: innerId
    }, innerLabelValue), textAfter && /*#__PURE__*/_react["default"].createElement(_TextField.StyledTextAfter, null, textAfter)), contentRight && /*#__PURE__*/_react["default"].createElement(_TextField.StyledContentRight, null, contentRight)), leftHelper && /*#__PURE__*/_react["default"].createElement(_TextField.LeftHelper, {
      id: helperTextId
    }, leftHelper));
  });
};
var textFieldConfig = exports.textFieldConfig = {
  name: 'TextField',
  tag: 'div',
  layout: textFieldRoot,
  base: base,
  variations: {
    view: {
      css: _base2.base
    },
    size: {
      css: _base.base
    },
    disabled: {
      css: _base3.base,
      attrs: true
    },
    readOnly: {
      css: _base4.base,
      attrs: true
    },
    labelPlacement: {
      css: _base5.base
    }
  },
  defaults: {
    size: 'm',
    view: 'default'
  }
};
