// @salutejs/plasma-new-hope/es/components/Dropdown/Dropdown.styles.js
import './Dropdown.styles_1tihqkp.css';
import { styled } from '@linaria/react';
import '@linaria/core';
import { component } from '../../engines/common.js';
import { popoverConfig } from '../Popover/Popover.js';
import { getCorrectHeight } from './utils/index.js';
import { tokens } from './Dropdown.tokens.js';

var Popover = /*#__PURE__*/component(popoverConfig);
var _exp = function _exp() {
  return Popover;
};
var StyledPopover = /*#__PURE__*/styled(_exp())({
  name: "StyledPopover",
  "class": "sn7uamp",
  propsAsIs: true
});
var _exp7 = function _exp7() {
  return function (_ref) {
    var listWidth = _ref.listWidth;
    return listWidth || "var(".concat(tokens.width, ")");
  };
};
var _exp8 = function _exp8() {
  return function (_ref2) {
    var listHeight = _ref2.listHeight;
    return listHeight ? getCorrectHeight(listHeight) : 'auto';
  };
};
var _exp9 = function _exp9() {
  return function (_ref3) {
    var listOverflow = _ref3.listOverflow;
    return listOverflow || 'initial';
  };
};
var _exp10 = function _exp10() {
  return function (_ref4) {
    var isInnerUl = _ref4.isInnerUl;
    return isInnerUl ? "calc(var(".concat(tokens.padding, ") * -1) 0 0 var(").concat(tokens.padding, ")") : 0;
  };
};
var Ul = /*#__PURE__*/styled('ul')({
  name: "Ul",
  "class": "ujr7fmv",
  propsAsIs: false,
  vars: {
    "ujr7fmv-0": [/*#__PURE__*/_exp7()],
    "ujr7fmv-1": [/*#__PURE__*/_exp8()],
    "ujr7fmv-2": [/*#__PURE__*/_exp9()],
    "ujr7fmv-3": [/*#__PURE__*/_exp10()]
  }
});
var base = "byfocd2";

export { StyledPopover, Ul, base };
