// @salutejs/plasma-new-hope/styled-components/cjs/components/Panel/ui/PanelHeader/PanelHeader.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.panelHeaderRoot = exports.panelHeaderConfig = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _utils = /*#__PURE__*/require("../../../../utils");
var _Icon = /*#__PURE__*/require("../../../_Icon");
var _Panel = /*#__PURE__*/require("../../Panel.tokens");
var _PanelHeader = /*#__PURE__*/require("./PanelHeader.types");
var _PanelHeader2 = /*#__PURE__*/require("./PanelHeader.styles");
var _base = /*#__PURE__*/require("./variations/_view/base");
var _IconClose;
var _excluded = ["children", "actions", "view", "className", "onClose", "hasClose", "closePlacement"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var panelHeaderRoot = exports.panelHeaderRoot = function panelHeaderRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (_ref, outerRef) {
    var children = _ref.children,
      actions = _ref.actions,
      view = _ref.view,
      className = _ref.className,
      onClose = _ref.onClose,
      _ref$hasClose = _ref.hasClose,
      hasClose = _ref$hasClose === void 0 ? 'true' : _ref$hasClose,
      _ref$closePlacement = _ref.closePlacement,
      closePlacement = _ref$closePlacement === void 0 ? 'right' : _ref$closePlacement,
      rest = _objectWithoutProperties(_ref, _excluded);
    var hasHeaderClass = children ? undefined : _Panel.classes.hasHeader;
    var isRightCloseClass = closePlacement === _PanelHeader.placements.right ? _Panel.classes.isRightClose : undefined;
    return /*#__PURE__*/_react["default"].createElement(Root, _extends({
      ref: outerRef,
      view: view,
      className: (0, _utils.cx)(className, hasHeaderClass, isRightCloseClass)
    }, rest), children && /*#__PURE__*/_react["default"].createElement(_PanelHeader2.StyledHeader, null, children), actions && /*#__PURE__*/_react["default"].createElement(_PanelHeader2.ButtonWrapper, null, actions), hasClose && /*#__PURE__*/_react["default"].createElement(_PanelHeader2.ButtonWrapper, {
      placement: closePlacement
    }, /*#__PURE__*/_react["default"].createElement(_PanelHeader2.CloseIconWrapper, {
      view: "clear",
      size: "s",
      onClick: onClose
    }, _IconClose || (_IconClose = /*#__PURE__*/_react["default"].createElement(_Icon.IconClose, {
      size: "s",
      color: "inherit"
    })))));
  });
};
var panelHeaderConfig = exports.panelHeaderConfig = {
  name: 'PanelHeader',
  tag: 'div',
  layout: panelHeaderRoot,
  base: '',
  variations: {
    view: {
      css: _base.base
    }
  },
  defaults: {
    view: 'default'
  }
};
