// @salutejs/plasma-new-hope/styled-components/cjs/components/Button/Button.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.buttonRoot = exports.buttonConfig = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _roundness = /*#__PURE__*/require("../../utils/roundness");
var _utils = /*#__PURE__*/require("../../utils");
var _base = /*#__PURE__*/require("./variations/_view/base");
var _base2 = /*#__PURE__*/require("./variations/_size/base");
var _base3 = /*#__PURE__*/require("./variations/_disabled/base");
var _base4 = /*#__PURE__*/require("./variations/_focused/base");
var _base5 = /*#__PURE__*/require("./variations/_stretching/base");
var _base6 = /*#__PURE__*/require("./variations/_blurred/base");
var _Button = /*#__PURE__*/require("./Button.styles");
var _Button2 = /*#__PURE__*/require("./Button.tokens");
var _StyledSpinner;
var _excluded = ["children", "view", "size", "text", "value", "contentLeft", "contentRight", "contentPlacing", "isLoading", "loader", "stretch", "square", "pin", "disabled", "focused", "outlined", "className", "blur", "style", "stretching"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var buttonRoot = exports.buttonRoot = function buttonRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
    var children = props.children,
      view = props.view,
      size = props.size,
      text = props.text,
      value = props.value,
      contentLeft = props.contentLeft,
      contentRight = props.contentRight,
      _props$contentPlacing = props.contentPlacing,
      contentPlacing = _props$contentPlacing === void 0 ? 'default' : _props$contentPlacing,
      isLoading = props.isLoading,
      loader = props.loader,
      stretch = props.stretch,
      square = props.square,
      pin = props.pin,
      disabled = props.disabled,
      focused = props.focused,
      outlined = props.outlined,
      className = props.className,
      blur = props.blur,
      style = props.style,
      _props$stretching = props.stretching,
      stretching = _props$stretching === void 0 ? 'auto' : _props$stretching,
      rest = _objectWithoutProperties(props, _excluded);
    var txt = typeof children === 'string' ? children : text;
    var stretchingClass = stretch ? _Button2.classes.filledStretching : _Button2.classes["".concat(stretching, "Stretching")];
    var contentRelaxedClass = contentPlacing === 'relaxed' ? _Button2.classes.contentRelaxed : undefined;
    var squareClass = square ? _Button2.classes.buttonSquare : undefined;
    var buttonBorderRadius = pin ? (0, _roundness.convertRoundnessMatrix)(pin, "var(".concat(_Button2.tokens.buttonRadius, ")"), "var(".concat(_Button2.tokens.buttonHeight, ")")) : "var(".concat(_Button2.tokens.buttonRadius, ", calc(var(").concat(_Button2.tokens.buttonHeight, ") / 4))");
    var contentPlacingMapper = {
      "default": 'center',
      relaxed: 'space-between'
    };
    var contentPlacementValue = contentPlacing ? contentPlacingMapper[contentPlacing] : 'center';
    return /*#__PURE__*/_react["default"].createElement(Root, _extends({
      type: "button",
      ref: ref,
      view: view,
      size: size,
      value: value,
      disabled: disabled,
      focused: focused || outlined,
      contentPlacing: contentPlacing,
      className: (0, _utils.cx)(squareClass, stretchingClass, _Button2.classes.buttonItem, className),
      style: _objectSpread(_objectSpread({}, style), {}, {
        '--plasma_computed-btn-br': buttonBorderRadius,
        '--plasma_private-blur': blur
      })
    }, rest), /*#__PURE__*/_react["default"].createElement(_Button.LoadWrap, {
      contentPlacing: contentPlacementValue,
      isLoading: isLoading
    }, contentLeft && /*#__PURE__*/_react["default"].createElement(_Button.StyledContentLeft, null, contentLeft), txt ? /*#__PURE__*/_react["default"].createElement(_Button.ButtonText, {
      className: contentRelaxedClass
    }, txt) : children, value && /*#__PURE__*/_react["default"].createElement(_Button.ButtonValue, null, value), contentRight && /*#__PURE__*/_react["default"].createElement(_Button.StyledContentRight, null, contentRight)), isLoading && /*#__PURE__*/_react["default"].createElement(_Button.Loader, null, loader || _StyledSpinner || (_StyledSpinner = /*#__PURE__*/_react["default"].createElement(_Button.StyledSpinner, null))));
  });
};
var buttonConfig = exports.buttonConfig = {
  name: 'Button',
  tag: 'button',
  layout: buttonRoot,
  base: _Button.base,
  variations: {
    view: {
      css: _base.base
    },
    size: {
      css: _base2.base
    },
    disabled: {
      css: _base3.base,
      attrs: true
    },
    focused: {
      css: _base4.base
      // TODO: isLoading => disabled
    },
    blurred: {
      css: _base6.base
    },
    stretching: {
      css: _base5.base
    }
  },
  defaults: {
    view: 'secondary',
    size: 'm',
    stretching: 'auto'
  }
};
