// @salutejs/plasma-typo/lib/cjs/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.typography = exports.sbermarketTypoProperties = exports.sbermarketTypoObject = exports.sbermarket = exports.plasmaTypoProperties = exports.plasmaTypoObject = exports.plasma = exports.soulmateTypoProperties = exports.soulmateTypoObject = exports.soulmate = exports.rulerTypoProperties = exports.rulerTypoObject = exports.ruler = exports.sageTypoProperties = exports.sageTypoObject = exports.sage = exports.mageTypoProperties = exports.mageTypoObject = exports.mage = exports.standard = exports.compatibleNewTypoTokens = exports.compatible = exports.spacing = exports.applySpacing = exports.applyHyphens = void 0;
var typography = __importStar(require("./tokens"));
exports.typography = typography;
var mixins_1 = require("./mixins");
Object.defineProperty(exports, "applyHyphens", { enumerable: true, get: function () { return mixins_1.applyHyphens; } });
Object.defineProperty(exports, "applySpacing", { enumerable: true, get: function () { return mixins_1.applySpacing; } });
Object.defineProperty(exports, "spacing", { enumerable: true, get: function () { return mixins_1.spacing; } });
var compatible_1 = require("./compatible");
Object.defineProperty(exports, "compatible", { enumerable: true, get: function () { return compatible_1.compatible; } });
Object.defineProperty(exports, "compatibleNewTypoTokens", { enumerable: true, get: function () { return compatible_1.compatibleNewTypoTokens; } });
__exportStar(require("./components"), exports);
var standard_1 = require("./standard");
Object.defineProperty(exports, "standard", { enumerable: true, get: function () { return standard_1.standard; } });
var archetypes_1 = require("./archetypes");
Object.defineProperty(exports, "mage", { enumerable: true, get: function () { return archetypes_1.mage; } });
Object.defineProperty(exports, "mageTypoObject", { enumerable: true, get: function () { return archetypes_1.mageTypoObject; } });
Object.defineProperty(exports, "mageTypoProperties", { enumerable: true, get: function () { return archetypes_1.mageTypoProperties; } });
Object.defineProperty(exports, "sage", { enumerable: true, get: function () { return archetypes_1.sage; } });
Object.defineProperty(exports, "sageTypoObject", { enumerable: true, get: function () { return archetypes_1.sageTypoObject; } });
Object.defineProperty(exports, "sageTypoProperties", { enumerable: true, get: function () { return archetypes_1.sageTypoProperties; } });
Object.defineProperty(exports, "ruler", { enumerable: true, get: function () { return archetypes_1.ruler; } });
Object.defineProperty(exports, "rulerTypoObject", { enumerable: true, get: function () { return archetypes_1.rulerTypoObject; } });
Object.defineProperty(exports, "rulerTypoProperties", { enumerable: true, get: function () { return archetypes_1.rulerTypoProperties; } });
Object.defineProperty(exports, "soulmate", { enumerable: true, get: function () { return archetypes_1.soulmate; } });
Object.defineProperty(exports, "soulmateTypoObject", { enumerable: true, get: function () { return archetypes_1.soulmateTypoObject; } });
Object.defineProperty(exports, "soulmateTypoProperties", { enumerable: true, get: function () { return archetypes_1.soulmateTypoProperties; } });
Object.defineProperty(exports, "plasma", { enumerable: true, get: function () { return archetypes_1.plasma; } });
Object.defineProperty(exports, "plasmaTypoObject", { enumerable: true, get: function () { return archetypes_1.plasmaTypoObject; } });
Object.defineProperty(exports, "plasmaTypoProperties", { enumerable: true, get: function () { return archetypes_1.plasmaTypoProperties; } });
Object.defineProperty(exports, "sbermarket", { enumerable: true, get: function () { return archetypes_1.sbermarket; } });
Object.defineProperty(exports, "sbermarketTypoObject", { enumerable: true, get: function () { return archetypes_1.sbermarketTypoObject; } });
Object.defineProperty(exports, "sbermarketTypoProperties", { enumerable: true, get: function () { return archetypes_1.sbermarketTypoProperties; } });
__exportStar(require("./types"), exports);
__exportStar(require("./tokens"), exports);
