// @salutejs/plasma-typo/lib/cjs/tokens.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.textXSBold = exports.textXS = exports.textSBold = exports.textS = exports.textMBold = exports.textM = exports.textLBold = exports.textL = exports.bodyXXSBold = exports.bodyXXS = exports.bodyXSBold = exports.bodyXS = exports.bodySBold = exports.bodyS = exports.bodyMBold = exports.bodyM = exports.bodyLBold = exports.bodyL = exports.h5Bold = exports.h5 = exports.h4Bold = exports.h4 = exports.h3Bold = exports.h3 = exports.h2Bold = exports.h2 = exports.h1Bold = exports.h1 = exports.dsplSBold = exports.dsplS = exports.dsplMBold = exports.dsplM = exports.dsplLBold = exports.dsplL = void 0;
exports.dsplL = {
    fontFamily: 'var(--plasma-typo-dspl-l-font-family)',
    fontSize: 'var(--plasma-typo-dspl-l-font-size)',
    fontStyle: 'var(--plasma-typo-dspl-l-font-style)',
    fontWeight: 'var(--plasma-typo-dspl-l-font-weight)',
    letterSpacing: 'var(--plasma-typo-dspl-l-letter-spacing)',
    lineHeight: 'var(--plasma-typo-dspl-l-line-height)',
};
exports.dsplLBold = __assign(__assign({}, exports.dsplL), { fontWeight: 'var(--plasma-typo-dspl-l-bold-font-weight)' });
exports.dsplM = {
    fontFamily: 'var(--plasma-typo-dspl-m-font-family)',
    fontSize: 'var(--plasma-typo-dspl-m-font-size)',
    fontStyle: 'var(--plasma-typo-dspl-m-font-style)',
    fontWeight: 'var(--plasma-typo-dspl-m-font-weight)',
    letterSpacing: 'var(--plasma-typo-dspl-m-letter-spacing)',
    lineHeight: 'var(--plasma-typo-dspl-m-line-height)',
};
exports.dsplMBold = __assign(__assign({}, exports.dsplM), { fontWeight: 'var(--plasma-typo-dspl-m-bold-font-weight)' });
exports.dsplS = {
    fontFamily: 'var(--plasma-typo-dspl-s-font-family)',
    fontSize: 'var(--plasma-typo-dspl-s-font-size)',
    fontStyle: 'var(--plasma-typo-dspl-s-font-style)',
    fontWeight: 'var(--plasma-typo-dspl-s-font-weight)',
    letterSpacing: 'var(--plasma-typo-dspl-s-letter-spacing)',
    lineHeight: 'var(--plasma-typo-dspl-s-line-height)',
};
exports.dsplSBold = __assign(__assign({}, exports.dsplS), { fontWeight: 'var(--plasma-typo-dspl-s-bold-font-weight)' });
exports.h1 = {
    fontFamily: 'var(--plasma-typo-h1-font-family)',
    fontSize: 'var(--plasma-typo-h1-font-size)',
    fontStyle: 'var(--plasma-typo-h1-font-style)',
    fontWeight: 'var(--plasma-typo-h1-font-weight)',
    letterSpacing: 'var(--plasma-typo-h1-letter-spacing)',
    lineHeight: 'var(--plasma-typo-h1-line-height)',
};
exports.h1Bold = __assign(__assign({}, exports.h1), { fontWeight: 'var(--plasma-typo-h1-bold-font-weight)' });
exports.h2 = {
    fontFamily: 'var(--plasma-typo-h2-font-family)',
    fontSize: 'var(--plasma-typo-h2-font-size)',
    fontStyle: 'var(--plasma-typo-h2-font-style)',
    fontWeight: 'var(--plasma-typo-h2-font-weight)',
    letterSpacing: 'var(--plasma-typo-h2-letter-spacing)',
    lineHeight: 'var(--plasma-typo-h2-line-height)',
};
exports.h2Bold = __assign(__assign({}, exports.h2), { fontWeight: 'var(--plasma-typo-h2-bold-font-weight)' });
exports.h3 = {
    fontFamily: 'var(--plasma-typo-h3-font-family)',
    fontSize: 'var(--plasma-typo-h3-font-size)',
    fontStyle: 'var(--plasma-typo-h3-font-style)',
    fontWeight: 'var(--plasma-typo-h3-font-weight)',
    letterSpacing: 'var(--plasma-typo-h3-letter-spacing)',
    lineHeight: 'var(--plasma-typo-h3-line-height)',
};
exports.h3Bold = __assign(__assign({}, exports.h3), { fontWeight: 'var(--plasma-typo-h3-bold-font-weight)' });
exports.h4 = {
    fontFamily: 'var(--plasma-typo-h4-font-family)',
    fontSize: 'var(--plasma-typo-h4-font-size)',
    fontStyle: 'var(--plasma-typo-h4-font-style)',
    fontWeight: 'var(--plasma-typo-h4-font-weight)',
    letterSpacing: 'var(--plasma-typo-h4-letter-spacing)',
    lineHeight: 'var(--plasma-typo-h4-line-height)',
};
exports.h4Bold = __assign(__assign({}, exports.h4), { fontWeight: 'var(--plasma-typo-h4-bold-font-weight)' });
exports.h5 = {
    fontFamily: 'var(--plasma-typo-h5-font-family)',
    fontSize: 'var(--plasma-typo-h5-font-size)',
    fontStyle: 'var(--plasma-typo-h5-font-style)',
    fontWeight: 'var(--plasma-typo-h5-font-weight)',
    letterSpacing: 'var(--plasma-typo-h5-letter-spacing)',
    lineHeight: 'var(--plasma-typo-h5-line-height)',
};
exports.h5Bold = __assign(__assign({}, exports.h5), { fontWeight: 'var(--plasma-typo-h5-bold-font-weight)' });
exports.bodyL = {
    fontFamily: 'var(--plasma-typo-body-l-font-family)',
    fontSize: 'var(--plasma-typo-body-l-font-size)',
    fontStyle: 'var(--plasma-typo-body-l-font-style)',
    fontWeight: 'var(--plasma-typo-body-l-font-weight)',
    letterSpacing: 'var(--plasma-typo-body-l-letter-spacing)',
    lineHeight: 'var(--plasma-typo-body-l-line-height)',
};
exports.bodyLBold = __assign(__assign({}, exports.bodyL), { fontWeight: 'var(--plasma-typo-body-l-bold-font-weight)' });
exports.bodyM = {
    fontFamily: 'var(--plasma-typo-body-m-font-family)',
    fontSize: 'var(--plasma-typo-body-m-font-size)',
    fontStyle: 'var(--plasma-typo-body-m-font-style)',
    fontWeight: 'var(--plasma-typo-body-m-font-weight)',
    letterSpacing: 'var(--plasma-typo-body-m-letter-spacing)',
    lineHeight: 'var(--plasma-typo-body-m-line-height)',
};
exports.bodyMBold = __assign(__assign({}, exports.bodyM), { fontWeight: 'var(--plasma-typo-body-m-bold-font-weight)' });
exports.bodyS = {
    fontFamily: 'var(--plasma-typo-body-s-font-family)',
    fontSize: 'var(--plasma-typo-body-s-font-size)',
    fontStyle: 'var(--plasma-typo-body-s-font-style)',
    fontWeight: 'var(--plasma-typo-body-s-font-weight)',
    letterSpacing: 'var(--plasma-typo-body-s-letter-spacing)',
    lineHeight: 'var(--plasma-typo-body-s-line-height)',
};
exports.bodySBold = __assign(__assign({}, exports.bodyS), { fontWeight: 'var(--plasma-typo-body-s-bold-font-weight)' });
exports.bodyXS = {
    fontFamily: 'var(--plasma-typo-body-xs-font-family)',
    fontSize: 'var(--plasma-typo-body-xs-font-size)',
    fontStyle: 'var(--plasma-typo-body-xs-font-style)',
    fontWeight: 'var(--plasma-typo-body-xs-font-weight)',
    letterSpacing: 'var(--plasma-typo-body-xs-letter-spacing)',
    lineHeight: 'var(--plasma-typo-body-xs-line-height)',
};
exports.bodyXSBold = __assign(__assign({}, exports.bodyXS), { fontWeight: 'var(--plasma-typo-body-xs-bold-font-weight)' });
exports.bodyXXS = {
    fontFamily: 'var(--plasma-typo-body-xxs-font-family)',
    fontSize: 'var(--plasma-typo-body-xxs-font-size)',
    fontStyle: 'var(--plasma-typo-body-xxs-font-style)',
    fontWeight: 'var(--plasma-typo-body-xxs-font-weight)',
    letterSpacing: 'var(--plasma-typo-body-xxs-letter-spacing)',
    lineHeight: 'var(--plasma-typo-body-xxs-line-height)',
};
exports.bodyXXSBold = __assign(__assign({}, exports.bodyXXS), { fontWeight: 'var(--plasma-typo-body-xxs-bold-font-weight)' });
exports.textL = {
    fontFamily: 'var(--plasma-typo-text-l-font-family)',
    fontSize: 'var(--plasma-typo-text-l-font-size)',
    fontStyle: 'var(--plasma-typo-text-l-font-style)',
    fontWeight: 'var(--plasma-typo-text-l-font-weight)',
    letterSpacing: 'var(--plasma-typo-text-l-letter-spacing)',
    lineHeight: 'var(--plasma-typo-text-l-line-height)',
};
exports.textLBold = __assign(__assign({}, exports.textL), { fontWeight: 'var(--plasma-typo-text-l-bold-font-weight)' });
exports.textM = {
    fontFamily: 'var(--plasma-typo-text-m-font-family)',
    fontSize: 'var(--plasma-typo-text-m-font-size)',
    fontStyle: 'var(--plasma-typo-text-m-font-style)',
    fontWeight: 'var(--plasma-typo-text-m-font-weight)',
    letterSpacing: 'var(--plasma-typo-text-m-letter-spacing)',
    lineHeight: 'var(--plasma-typo-text-m-line-height)',
};
exports.textMBold = __assign(__assign({}, exports.textM), { fontWeight: 'var(--plasma-typo-text-m-bold-font-weight)' });
exports.textS = {
    fontFamily: 'var(--plasma-typo-text-s-font-family)',
    fontSize: 'var(--plasma-typo-text-s-font-size)',
    fontStyle: 'var(--plasma-typo-text-s-font-style)',
    fontWeight: 'var(--plasma-typo-text-s-font-weight)',
    letterSpacing: 'var(--plasma-typo-text-s-letter-spacing)',
    lineHeight: 'var(--plasma-typo-text-s-line-height)',
};
exports.textSBold = __assign(__assign({}, exports.textS), { fontWeight: 'var(--plasma-typo-text-s-bold-font-weight)' });
exports.textXS = {
    fontFamily: 'var(--plasma-typo-text-xs-font-family)',
    fontSize: 'var(--plasma-typo-text-xs-font-size)',
    fontStyle: 'var(--plasma-typo-text-xs-font-style)',
    fontWeight: 'var(--plasma-typo-text-xs-font-weight)',
    letterSpacing: 'var(--plasma-typo-text-xs-letter-spacing)',
    lineHeight: 'var(--plasma-typo-text-xs-line-height)',
};
exports.textXSBold = __assign(__assign({}, exports.textXS), { fontWeight: 'var(--plasma-typo-text-xs-bold-font-weight)' });
