// @salutejs/plasma-web/css/cjs/components/Typography/Old/Headline.config.js
import './Headline.config_aske57.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var configHeadline1 = {
  defaults: {
    size: 'headline1'
  },
  variations: {
    size: {
      headline1: "hsnb5hl"
    }
  }
};
var configHeadline2 = {
  defaults: {
    size: 'headline2'
  },
  variations: {
    size: {
      headline2: "h1ujk7jq"
    }
  }
};
var configHeadline3 = {
  defaults: {
    size: 'headline3'
  },
  variations: {
    size: {
      headline3: "hfun0z9"
    }
  }
};
var configHeadline4 = {
  defaults: {
    size: 'headline4'
  },
  variations: {
    size: {
      headline4: "h1br7d6f"
    }
  }
};
var configHeadline5 = {
  defaults: {
    size: 'headline5'
  },
  variations: {
    size: {
      headline5: "h1oxqle"
    }
  }
};

exports.configHeadline1 = configHeadline1;
exports.configHeadline2 = configHeadline2;
exports.configHeadline3 = configHeadline3;
exports.configHeadline4 = configHeadline4;
exports.configHeadline5 = configHeadline5;
