// @salutejs/plasma-web/css/cjs/components/Typography/Old/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var plasmaNewHope = require('@salutejs/plasma-new-hope');
var Body_config = require('./Body.config.js');
var Button_config = require('./Button.config.js');
var Caption_config = require('./Caption.config.js');
var Footnote_config = require('./Footnote.config.js');
var Headline_config = require('./Headline.config.js');
var Paragraph_config = require('./Paragraph.config.js');
var ParagraphText_config = require('./ParagraphText.config.js');
var Underline_config = require('./Underline.config.js');
var Subtitle_config = require('./Subtitle.config.js');

var body1Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.bodyConfig, Body_config.configBody1);
var Body1Component = /*#__PURE__*/plasmaNewHope.component(body1Config);
var body2Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.bodyConfig, Body_config.configBody2);
var Body2Component = /*#__PURE__*/plasmaNewHope.component(body2Config);
var button1Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.buttonTypographyConfig, Button_config.configButton1);
var Button1Component = /*#__PURE__*/plasmaNewHope.component(button1Config);
var button2Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.buttonTypographyConfig, Button_config.configButton2);
var Button2Component = /*#__PURE__*/plasmaNewHope.component(button2Config);
var captionConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.captionConfig, Caption_config.config);
var CaptionComponent = /*#__PURE__*/plasmaNewHope.component(captionConfig);
var footnote1Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.footnoteConfig, Footnote_config.configFootnote1);
var Footnote1Component = /*#__PURE__*/plasmaNewHope.component(footnote1Config);
var footnote2Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.footnoteConfig, Footnote_config.configFootnote2);
var Footnote2Component = /*#__PURE__*/plasmaNewHope.component(footnote2Config);
var headline1Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.headlineConfig, Headline_config.configHeadline1);
var Headline1Component = /*#__PURE__*/plasmaNewHope.component(headline1Config);
var headline2Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.headlineConfig, Headline_config.configHeadline2);
var Headline2Component = /*#__PURE__*/plasmaNewHope.component(headline2Config);
var headline3Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.headlineConfig, Headline_config.configHeadline3);
var Headline3Component = /*#__PURE__*/plasmaNewHope.component(headline3Config);
var headline4Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.headlineConfig, Headline_config.configHeadline4);
var Headline4Component = /*#__PURE__*/plasmaNewHope.component(headline4Config);
var headline5Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.headlineConfig, Headline_config.configHeadline5);
var Headline5Component = /*#__PURE__*/plasmaNewHope.component(headline5Config);
var underlineConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.underlineConfig, Underline_config.config);
var UnderlineComponent = /*#__PURE__*/plasmaNewHope.component(underlineConfig);
var subtitleConfig = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.subtitleConfig, Subtitle_config.config);
var SubtitleComponent = /*#__PURE__*/plasmaNewHope.component(subtitleConfig);
var paragraph1Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.paragraphConfig, Paragraph_config.configParagraph1);
var Paragraph1Component = /*#__PURE__*/plasmaNewHope.component(paragraph1Config);
var paragraph2Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.paragraphConfig, Paragraph_config.configParagraph2);
var Paragraph2Component = /*#__PURE__*/plasmaNewHope.component(paragraph2Config);
var paragraphText1Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.paragraphTextConfig, ParagraphText_config.configParagraphText1);
var ParagraphText1Component = /*#__PURE__*/plasmaNewHope.component(paragraphText1Config);
var paragraphText2Config = /*#__PURE__*/plasmaNewHope.mergeConfig(plasmaNewHope.paragraphTextConfig, ParagraphText_config.configParagraphText2);
var ParagraphText2Component = /*#__PURE__*/plasmaNewHope.component(paragraphText2Config);
var Body1 = Body1Component;
var Body2 = Body2Component;
var Button1 = Button1Component;
var Button2 = Button2Component;
var Caption = CaptionComponent;
var Footnote1 = Footnote1Component;
var Footnote2 = Footnote2Component;
var Headline1 = Headline1Component;
var Headline2 = Headline2Component;
var Headline3 = Headline3Component;
var Headline4 = Headline4Component;
var Headline5 = Headline5Component;
var P1 = Paragraph1Component;
var P2 = Paragraph2Component;
var ParagraphText1 = ParagraphText1Component;
var ParagraphText2 = ParagraphText2Component;
var Underline = UnderlineComponent;
var Subtitle = SubtitleComponent;

exports.Body1 = Body1;
exports.Body2 = Body2;
exports.Button1 = Button1;
exports.Button2 = Button2;
exports.Caption = Caption;
exports.Footnote1 = Footnote1;
exports.Footnote2 = Footnote2;
exports.Headline1 = Headline1;
exports.Headline2 = Headline2;
exports.Headline3 = Headline3;
exports.Headline4 = Headline4;
exports.Headline5 = Headline5;
exports.P1 = P1;
exports.P2 = P2;
exports.ParagraphText1 = ParagraphText1;
exports.ParagraphText2 = ParagraphText2;
exports.Subtitle = Subtitle;
exports.Underline = Underline;
