// @salutejs/plasma-core/hooks/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "useDebouncedFunction", {
  enumerable: true,
  get: function get() {
    return _useDebouncedFunction.useDebouncedFunction;
  }
});
Object.defineProperty(exports, "useFocusTrap", {
  enumerable: true,
  get: function get() {
    return _useFocusTrap.useFocusTrap;
  }
});
Object.defineProperty(exports, "useForkRef", {
  enumerable: true,
  get: function get() {
    return _useForkRef.useForkRef;
  }
});
Object.defineProperty(exports, "useIsomorphicLayoutEffect", {
  enumerable: true,
  get: function get() {
    return _useIsomorphicLayoutEffect.useIsomorphicLayoutEffect;
  }
});
Object.defineProperty(exports, "useResizeObserver", {
  enumerable: true,
  get: function get() {
    return _useResizeObserver.useResizeObserver;
  }
});
Object.defineProperty(exports, "useUniqId", {
  enumerable: true,
  get: function get() {
    return _useUniqId.useUniqId;
  }
});
var _useDebouncedFunction = /*#__PURE__*/require("./useDebouncedFunction");
var _useForkRef = /*#__PURE__*/require("./useForkRef");
var _useIsomorphicLayoutEffect = /*#__PURE__*/require("./useIsomorphicLayoutEffect");
var _useUniqId = /*#__PURE__*/require("./useUniqId");
var _useResizeObserver = /*#__PURE__*/require("./useResizeObserver");
var _useFocusTrap = /*#__PURE__*/require("./useFocusTrap");
