// @salutejs/plasma-new-hope/es/components/Dropdown/ui/DropdownFooter/DropdownFooter.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef } from 'react';
import { useUniqId } from '@salutejs/plasma-core';
import { base } from './variations/_view/base.js';
import { base as base$1 } from './variations/_size/base.js';
import { StyledDropdownFooter } from './DropdownFooter.styles.js';

var _excluded = ["id", "className", "children", "view", "size"];
/**
 * Элемент выпадающего списка
 */
var dropdownFooterRoot = function dropdownFooterRoot(Root) {
  return /*#__PURE__*/forwardRef(function (_ref, outerRootRef) {
    var id = _ref.id,
      className = _ref.className,
      children = _ref.children,
      view = _ref.view,
      size = _ref.size,
      rest = _objectWithoutProperties(_ref, _excluded);
    var uniqId = useUniqId();
    var innerId = id || uniqId;
    return /*#__PURE__*/React.createElement(Root, {
      id: innerId,
      ref: outerRootRef,
      view: view,
      size: size
    }, /*#__PURE__*/React.createElement(StyledDropdownFooter, _extends({
      className: className
    }, rest), children));
  });
};
var dropdownFooterConfig = {
  name: 'DropdownFooter',
  tag: 'div',
  layout: dropdownFooterRoot,
  base: '',
  variations: {
    view: {
      css: base
    },
    size: {
      css: base$1
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

export { dropdownFooterConfig, dropdownFooterRoot };
