// @salutejs/plasma-hope/components/Badge/Badge.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.badgeViews = exports.Badge = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
(0, _collectPackageInfo.collectPackageInfo)('Badge/Badge');
var badgeViews = exports.badgeViews = {
  primary: /*#__PURE__*/(0, _styledComponents.css)(["color:#226af1;background-color:rgba(34,106,241,0.09);"]),
  secondary: /*#__PURE__*/(0, _styledComponents.css)(["color:", ";background-color:", ";"], _plasmaCore.text, _plasmaCore.surfaceLiquid02),
  success: /*#__PURE__*/(0, _styledComponents.css)(["color:#09a552;background-color:rgba(9,165,82,0.08);"]),
  warning: /*#__PURE__*/(0, _styledComponents.css)(["color:#ee6820;background-color:rgba(238,104,32,0.08);"]),
  critical: /*#__PURE__*/(0, _styledComponents.css)(["color:#df2638;background-color:rgba(223,38,56,0.09);"])
};
/**
 * Небольшая бирка для ячеек и карточек.
 * Компонент может отображаться в нескольких размерах и цветах, может содержать текст и/или иконку.
 */
var Badge = exports.Badge = /*#__PURE__*/(0, _styledComponents["default"])(_plasmaCore.Badge).withConfig({
  componentId: "plasma-web-core__sc-50gfig-0"
})(["border-radius:0.25rem;", ""], function (_ref) {
  var _ref$view = _ref.view,
    view = _ref$view === void 0 ? 'primary' : _ref$view;
  return badgeViews[view];
});
