// @salutejs/plasma-ui/components/Typography/Headline/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "H1Compatible", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.H1;
  }
});
Object.defineProperty(exports, "H2Compatible", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.H2;
  }
});
Object.defineProperty(exports, "H3Compatible", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.H3;
  }
});
Object.defineProperty(exports, "H4Compatible", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.H4;
  }
});
Object.defineProperty(exports, "Headline1", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.Headline1;
  }
});
Object.defineProperty(exports, "Headline2", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.Headline2;
  }
});
Object.defineProperty(exports, "Headline3", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.Headline3;
  }
});
Object.defineProperty(exports, "Headline4", {
  enumerable: true,
  get: function get() {
    return _plasmaCore.Headline4;
  }
});
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
