// @salutejs/plasma-new-hope/styled-components/es/components/TextField/TextField.styles.js
import styled from 'styled-components';
import { classes, tokens } from './TextField.tokens';
export var InputWrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-0"
})(["position:relative;display:flex;align-items:center;box-sizing:border-box;"]);
export var InputLabelWrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-1"
})(["flex:1;overflow:scroll;position:relative;width:100%;display:inline-flex;align-items:center;::-webkit-scrollbar{display:none;}scrollbar-width:none;overscroll-behavior:contain;&.", "{height:var(", ");border-radius:var(", ");}"], /*#__PURE__*/String(classes.hasChips), tokens.chipHeight, tokens.chipBorderRadius);
export var StyledChips = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-2"
})(["display:flex;gap:var(", ");margin-right:var(", ");user-select:none;"], tokens.chipGap, tokens.chipGap);
export var Input = /*#__PURE__*/styled.input.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-3"
})(["box-sizing:border-box;appearance:none;border:0;padding:0;background-color:transparent;outline:none;flex:1;min-width:60%;"]);
export var Label = /*#__PURE__*/styled.label.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-4"
})([""]);
export var StyledContentLeft = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-5"
})(["margin:var(", ");"], tokens.leftContentMargin);
export var StyledContentRight = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-6"
})(["margin:var(", ");"], tokens.rightContentMargin);
export var LeftHelper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-7"
})([""]);
export var StyledTextBefore = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-8"
})([""]);
export var StyledTextAfter = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-g4vxbb-9"
})([""]);
