// @salutejs/plasma-new-hope/styled-components/cjs/examples/sds_engineer/components/Avatar/Avatar.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _Avatar = /*#__PURE__*/require("../../../../components/Avatar/Avatar.tokens");
var config = exports.config = {
  defaults: {
    view: 'default',
    size: 'xxl'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--text-accent-gradient);", ":linear-gradient(94deg,rgba(62,121,240,0.20) 6.49%,rgba(39,198,229,0.20) 93.51%);", ":var(--surface-positive);", ":var(--surface-solid-tertiary);", ":1.02;"], _Avatar.tokens.color, _Avatar.tokens.backgroundColor, _Avatar.tokens.statusOnlineColor, _Avatar.tokens.statusOfflineColor, _Avatar.tokens.scaleHover)
    },
    size: {
      xxl: /*#__PURE__*/(0, _styledComponents.css)(["", ":6.25rem;", ":2rem;", ":600;", ":2rem;", ":0.75rem;"], _Avatar.tokens.avatarSize, _Avatar.tokens.fontSize, _Avatar.tokens.fontWeight, _Avatar.tokens.lineHeight, _Avatar.tokens.statusIconSize),
      l: /*#__PURE__*/(0, _styledComponents.css)(["", ":3rem;", ":1.25rem;", ":600;", ":1.25rem;", ":0.5rem;"], _Avatar.tokens.avatarSize, _Avatar.tokens.fontSize, _Avatar.tokens.fontWeight, _Avatar.tokens.lineHeight, _Avatar.tokens.statusIconSize),
      m: /*#__PURE__*/(0, _styledComponents.css)(["", ":2.25rem;", ":0.875rem;", ":600;", ":0.875rem;", ":0.5rem;"], _Avatar.tokens.avatarSize, _Avatar.tokens.fontSize, _Avatar.tokens.fontWeight, _Avatar.tokens.lineHeight, _Avatar.tokens.statusIconSize),
      s: /*#__PURE__*/(0, _styledComponents.css)(["", ":1.5rem;", ":0.5rem;", ":600;", ":0.5rem;", ":0.375rem;"], _Avatar.tokens.avatarSize, _Avatar.tokens.fontSize, _Avatar.tokens.fontWeight, _Avatar.tokens.lineHeight, _Avatar.tokens.statusIconSize),
      fit: /*#__PURE__*/(0, _styledComponents.css)(["", ":100%;", ":0;", ":0;", ":0;", ":0;"], _Avatar.tokens.avatarSize, _Avatar.tokens.fontSize, _Avatar.tokens.fontWeight, _Avatar.tokens.lineHeight, _Avatar.tokens.statusIconSize)
    },
    focused: {
      "true": /*#__PURE__*/(0, _styledComponents.css)(["", ":0.1rem;", ":0.2rem;", ":var(--surface-accent);"], _Avatar.tokens.outlineSize, _Avatar.tokens.outlineOffset, _Avatar.tokens.outlineColor)
    }
  }
};
