// @salutejs/plasma-web/components/Typography/Old/Body.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.configBody3 = exports.configBody2 = exports.configBody1 = void 0;
var _styledComponents = /*#__PURE__*/require("@salutejs/plasma-new-hope/styled-components");
var _templateObject, _templateObject2, _templateObject3;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
var configBody1 = exports.configBody1 = {
  defaults: {
    size: 'body1'
  },
  variations: {
    size: {
      body1: /*#__PURE__*/(0, _styledComponents.css)(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-body1-font-family);\n                ", ": var(--plasma-typo-body1-font-size);\n                ", ": var(--plasma-typo-body1-font-style);\n                ", ": var(--plasma-typo-body1-font-weight);\n                ", ": var(--plasma-typo-body1-letter-spacing);\n                ", ": var(--plasma-typo-body1-line-height);\n            "])), _styledComponents.typographyTokens.typoFontFamily, _styledComponents.typographyTokens.typoFontSize, _styledComponents.typographyTokens.typoFontStyle, _styledComponents.typographyTokens.typoFontWeight, _styledComponents.typographyTokens.typoFontLetterSpacing, _styledComponents.typographyTokens.typoFontLineHeight)
    }
  }
};
var configBody2 = exports.configBody2 = {
  defaults: {
    size: 'body2'
  },
  variations: {
    size: {
      body2: /*#__PURE__*/(0, _styledComponents.css)(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-body2-font-family);\n                ", ": var(--plasma-typo-body2-font-size);\n                ", ": var(--plasma-typo-body2-font-style);\n                ", ": var(--plasma-typo-body2-font-weight);\n                ", ": var(--plasma-typo-body2-letter-spacing);\n                ", ": var(--plasma-typo-body2-line-height);\n            "])), _styledComponents.typographyTokens.typoFontFamily, _styledComponents.typographyTokens.typoFontSize, _styledComponents.typographyTokens.typoFontStyle, _styledComponents.typographyTokens.typoFontWeight, _styledComponents.typographyTokens.typoFontLetterSpacing, _styledComponents.typographyTokens.typoFontLineHeight)
    }
  }
};
var configBody3 = exports.configBody3 = {
  defaults: {
    size: 'body3'
  },
  variations: {
    size: {
      body3: /*#__PURE__*/(0, _styledComponents.css)(_templateObject3 || (_templateObject3 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-body3-font-family);\n                ", ": var(--plasma-typo-body3-font-size);\n                ", ": var(--plasma-typo-body3-font-style);\n                ", ": var(--plasma-typo-body3-font-weight);\n                ", ": var(--plasma-typo-body3-letter-spacing);\n                ", ": var(--plasma-typo-body3-line-height);\n            "])), _styledComponents.typographyTokens.typoFontFamily, _styledComponents.typographyTokens.typoFontSize, _styledComponents.typographyTokens.typoFontStyle, _styledComponents.typographyTokens.typoFontWeight, _styledComponents.typographyTokens.typoFontLetterSpacing, _styledComponents.typographyTokens.typoFontLineHeight)
    }
  }
};
