// @salutejs/plasma-web/components/Typography/Old/Button.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.configButton2 = exports.configButton1 = void 0;
var _styledComponents = /*#__PURE__*/require("@salutejs/plasma-new-hope/styled-components");
var _templateObject, _templateObject2;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
var configButton1 = exports.configButton1 = {
  defaults: {
    size: 'button1'
  },
  variations: {
    size: {
      button1: /*#__PURE__*/(0, _styledComponents.css)(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-button1-font-family);\n                ", ": var(--plasma-typo-button1-font-size);\n                ", ": var(--plasma-typo-button1-font-style);\n                ", ": var(--plasma-typo-button1-font-weight);\n                ", ": var(--plasma-typo-button1-letter-spacing);\n                ", ": var(--plasma-typo-button1-line-height);\n            "])), _styledComponents.typographyTokens.typoFontFamily, _styledComponents.typographyTokens.typoFontSize, _styledComponents.typographyTokens.typoFontStyle, _styledComponents.typographyTokens.typoFontWeight, _styledComponents.typographyTokens.typoFontLetterSpacing, _styledComponents.typographyTokens.typoFontLineHeight)
    }
  }
};
var configButton2 = exports.configButton2 = {
  defaults: {
    size: 'button2'
  },
  variations: {
    size: {
      button2: /*#__PURE__*/(0, _styledComponents.css)(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-button2-font-family);\n                ", ": var(--plasma-typo-button2-font-size);\n                ", ": var(--plasma-typo-button2-font-style);\n                ", ": var(--plasma-typo-button2-font-weight);\n                ", ": var(--plasma-typo-button2-letter-spacing);\n                ", ": var(--plasma-typo-button2-line-height);\n            "])), _styledComponents.typographyTokens.typoFontFamily, _styledComponents.typographyTokens.typoFontSize, _styledComponents.typographyTokens.typoFontStyle, _styledComponents.typographyTokens.typoFontWeight, _styledComponents.typographyTokens.typoFontLetterSpacing, _styledComponents.typographyTokens.typoFontLineHeight)
    }
  }
};
