// @salutejs/plasma-web/components/Typography/Old/Footnote.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.configFootnote2 = exports.configFootnote1 = void 0;
var _styledComponents = /*#__PURE__*/require("@salutejs/plasma-new-hope/styled-components");
var _templateObject, _templateObject2;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
var configFootnote1 = exports.configFootnote1 = {
  defaults: {
    size: 'footnote1'
  },
  variations: {
    size: {
      footnote1: /*#__PURE__*/(0, _styledComponents.css)(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-footnote1-font-family);\n                ", ": var(--plasma-typo-footnote1-font-size);\n                ", ": var(--plasma-typo-footnote1-font-style);\n                ", ": var(--plasma-typo-footnote1-font-weight);\n                ", ": var(--plasma-typo-footnote1-letter-spacing);\n                ", ": var(--plasma-typo-footnote1-line-height);\n            "])), _styledComponents.typographyTokens.typoFontFamily, _styledComponents.typographyTokens.typoFontSize, _styledComponents.typographyTokens.typoFontStyle, _styledComponents.typographyTokens.typoFontWeight, _styledComponents.typographyTokens.typoFontLetterSpacing, _styledComponents.typographyTokens.typoFontLineHeight)
    }
  }
};
var configFootnote2 = exports.configFootnote2 = {
  defaults: {
    size: 'footnote2'
  },
  variations: {
    size: {
      footnote2: /*#__PURE__*/(0, _styledComponents.css)(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-footnote2-font-family);\n                ", ": var(--plasma-typo-footnote2-font-size);\n                ", ": var(--plasma-typo-footnote2-font-style);\n                ", ": var(--plasma-typo-footnote2-font-weight);\n                ", ": var(--plasma-typo-footnote2-letter-spacing);\n                ", ": var(--plasma-typo-footnote2-line-height);\n            "])), _styledComponents.typographyTokens.typoFontFamily, _styledComponents.typographyTokens.typoFontSize, _styledComponents.typographyTokens.typoFontStyle, _styledComponents.typographyTokens.typoFontWeight, _styledComponents.typographyTokens.typoFontLetterSpacing, _styledComponents.typographyTokens.typoFontLineHeight)
    }
  }
};
