// @salutejs/plasma-web/components/Typography/Old/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Underline = exports.Subtitle = exports.ParagraphText2 = exports.ParagraphText1 = exports.P2 = exports.P1 = exports.Headline5 = exports.Headline4 = exports.Headline3 = exports.Headline2 = exports.Headline1 = exports.Footnote2 = exports.Footnote1 = exports.Display3 = exports.Display2 = exports.Display1 = exports.Caption = exports.Button2 = exports.Button1 = exports.Body3 = exports.Body2 = exports.Body1 = void 0;
var _styledComponents = /*#__PURE__*/require("@salutejs/plasma-new-hope/styled-components");
var _Body = /*#__PURE__*/require("./Body.config");
var _Button = /*#__PURE__*/require("./Button.config");
var _Caption = /*#__PURE__*/require("./Caption.config");
var _Display = /*#__PURE__*/require("./Display.config");
var _Footnote = /*#__PURE__*/require("./Footnote.config");
var _Headline = /*#__PURE__*/require("./Headline.config");
var _Paragraph = /*#__PURE__*/require("./Paragraph.config");
var _ParagraphText = /*#__PURE__*/require("./ParagraphText.config");
var _Underline = /*#__PURE__*/require("./Underline.config");
var _Subtitle = /*#__PURE__*/require("./Subtitle.config");
var body1Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.bodyConfig, _Body.configBody1);
var Body1Component = /*#__PURE__*/(0, _styledComponents.component)(body1Config);
var body2Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.bodyConfig, _Body.configBody2);
var Body2Component = /*#__PURE__*/(0, _styledComponents.component)(body2Config);
var body3Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.bodyConfig, _Body.configBody3);
var Body3Component = /*#__PURE__*/(0, _styledComponents.component)(body3Config);
var button1Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.buttonTypographyConfig, _Button.configButton1);
var Button1Component = /*#__PURE__*/(0, _styledComponents.component)(button1Config);
var button2Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.buttonTypographyConfig, _Button.configButton2);
var Button2Component = /*#__PURE__*/(0, _styledComponents.component)(button2Config);
var captionConfig = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.captionConfig, _Caption.config);
var CaptionComponent = /*#__PURE__*/(0, _styledComponents.component)(captionConfig);
var display1Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.displayConfig, _Display.configDisplay1);
var Display1Component = /*#__PURE__*/(0, _styledComponents.component)(display1Config);
var display2Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.displayConfig, _Display.configDisplay2);
var Display2Component = /*#__PURE__*/(0, _styledComponents.component)(display2Config);
var display3Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.displayConfig, _Display.configDisplay3);
var Display3Component = /*#__PURE__*/(0, _styledComponents.component)(display3Config);
var footnote1Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.footnoteConfig, _Footnote.configFootnote1);
var Footnote1Component = /*#__PURE__*/(0, _styledComponents.component)(footnote1Config);
var footnote2Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.footnoteConfig, _Footnote.configFootnote2);
var Footnote2Component = /*#__PURE__*/(0, _styledComponents.component)(footnote2Config);
var headline1Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.headlineConfig, _Headline.configHeadline1);
var Headline1Component = /*#__PURE__*/(0, _styledComponents.component)(headline1Config);
var headline2Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.headlineConfig, _Headline.configHeadline2);
var Headline2Component = /*#__PURE__*/(0, _styledComponents.component)(headline2Config);
var headline3Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.headlineConfig, _Headline.configHeadline3);
var Headline3Component = /*#__PURE__*/(0, _styledComponents.component)(headline3Config);
var headline4Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.headlineConfig, _Headline.configHeadline4);
var Headline4Component = /*#__PURE__*/(0, _styledComponents.component)(headline4Config);
var headline5Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.headlineConfig, _Headline.configHeadline5);
var Headline5Component = /*#__PURE__*/(0, _styledComponents.component)(headline5Config);
var underlineConfig = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.underlineConfig, _Underline.config);
var UnderlineComponent = /*#__PURE__*/(0, _styledComponents.component)(underlineConfig);
var subtitleConfig = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.subtitleConfig, _Subtitle.config);
var SubtitleComponent = /*#__PURE__*/(0, _styledComponents.component)(subtitleConfig);
var paragraph1Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.paragraphConfig, _Paragraph.configParagraph1);
var Paragraph1Component = /*#__PURE__*/(0, _styledComponents.component)(paragraph1Config);
var paragraph2Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.paragraphConfig, _Paragraph.configParagraph2);
var Paragraph2Component = /*#__PURE__*/(0, _styledComponents.component)(paragraph2Config);
var paragraphText1Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.paragraphTextConfig, _ParagraphText.configParagraphText1);
var ParagraphText1Component = /*#__PURE__*/(0, _styledComponents.component)(paragraphText1Config);
var paragraphText2Config = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.paragraphTextConfig, _ParagraphText.configParagraphText2);
var ParagraphText2Component = /*#__PURE__*/(0, _styledComponents.component)(paragraphText2Config);
var Body1 = exports.Body1 = Body1Component;
var Body2 = exports.Body2 = Body2Component;
var Body3 = exports.Body3 = Body3Component;
var Button1 = exports.Button1 = Button1Component;
var Button2 = exports.Button2 = Button2Component;
var Caption = exports.Caption = CaptionComponent;
var Display1 = exports.Display1 = Display1Component;
var Display2 = exports.Display2 = Display2Component;
var Display3 = exports.Display3 = Display3Component;
var Footnote1 = exports.Footnote1 = Footnote1Component;
var Footnote2 = exports.Footnote2 = Footnote2Component;
var Headline1 = exports.Headline1 = Headline1Component;
var Headline2 = exports.Headline2 = Headline2Component;
var Headline3 = exports.Headline3 = Headline3Component;
var Headline4 = exports.Headline4 = Headline4Component;
var Headline5 = exports.Headline5 = Headline5Component;
var P1 = exports.P1 = Paragraph1Component;
var P2 = exports.P2 = Paragraph2Component;
var ParagraphText1 = exports.ParagraphText1 = ParagraphText1Component;
var ParagraphText2 = exports.ParagraphText2 = ParagraphText2Component;
var Underline = exports.Underline = UnderlineComponent;
var Subtitle = exports.Subtitle = SubtitleComponent;
