// @salutejs/plasma-web/components/Typography/Old/Paragraph.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.configParagraph2 = exports.configParagraph1 = void 0;
var _styledComponents = /*#__PURE__*/require("@salutejs/plasma-new-hope/styled-components");
var _templateObject, _templateObject2;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
var configParagraph1 = exports.configParagraph1 = {
  defaults: {
    size: 'paragraph1'
  },
  variations: {
    size: {
      paragraph1: /*#__PURE__*/(0, _styledComponents.css)(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-paragraph1-font-family);\n                ", ": var(--plasma-typo-paragraph1-font-size);\n                ", ": var(--plasma-typo-paragraph1-font-style);\n                ", ": var(--plasma-typo-paragraph1-font-weight);\n                ", ": var(--plasma-typo-paragraph1-letter-spacing);\n                ", ": var(--plasma-typo-paragraph1-line-height);\n            "])), _styledComponents.typographyTokens.typoFontFamily, _styledComponents.typographyTokens.typoFontSize, _styledComponents.typographyTokens.typoFontStyle, _styledComponents.typographyTokens.typoFontWeight, _styledComponents.typographyTokens.typoFontLetterSpacing, _styledComponents.typographyTokens.typoFontLineHeight)
    }
  }
};
var configParagraph2 = exports.configParagraph2 = {
  defaults: {
    size: 'paragraph2'
  },
  variations: {
    size: {
      paragraph2: /*#__PURE__*/(0, _styledComponents.css)(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--plasma-typo-paragraph2-font-family);\n                ", ": var(--plasma-typo-paragraph2-font-size);\n                ", ": var(--plasma-typo-paragraph2-font-style);\n                ", ": var(--plasma-typo-paragraph2-font-weight);\n                ", ": var(--plasma-typo-paragraph2-letter-spacing);\n                ", ": var(--plasma-typo-paragraph2-line-height);\n            "])), _styledComponents.typographyTokens.typoFontFamily, _styledComponents.typographyTokens.typoFontSize, _styledComponents.typographyTokens.typoFontStyle, _styledComponents.typographyTokens.typoFontWeight, _styledComponents.typographyTokens.typoFontLetterSpacing, _styledComponents.typographyTokens.typoFontLineHeight)
    }
  }
};
