// @salutejs/plasma-ui/components/ProductCard/ProductCardPrice.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ProductCardPrice = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _Price = /*#__PURE__*/require("../Price");
var _Typography = /*#__PURE__*/require("../Typography");
var _types = /*#__PURE__*/require("./types");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
var StyledPrices = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-eqx89j-0"
})(["display:flex;flex-direction:column;justify-content:center;width:100%;min-height:2.375rem;"]);
var StyledPrice = /*#__PURE__*/(0, _styledComponents["default"])(_Price.Price).withConfig({
  componentId: "plasma-ui__sc-eqx89j-1"
})(["color:", ";", ""], function (_ref) {
  var $type = _ref.$type;
  switch ($type) {
    case _types.PriceTypes.NEW:
      return _plasmaCore.success;
    default:
      return _plasmaCore.black;
  }
}, function (_ref2) {
  var $type = _ref2.$type;
  return $type === _types.PriceTypes.OLD && (0, _styledComponents.css)(["opacity:0.56;"]);
});
var ProductCardPrice = exports.ProductCardPrice = /*#__PURE__*/(0, _react.memo)(function (_ref3) {
  var price = _ref3.price,
    oldPrice = _ref3.oldPrice,
    periodicity = _ref3.periodicity;
  if (!price) {
    return null;
  }
  return /*#__PURE__*/_react["default"].createElement(StyledPrices, null, /*#__PURE__*/_react["default"].createElement(StyledPrice, {
    $type: oldPrice !== undefined ? _types.PriceTypes.NEW : undefined,
    forwardedAs: _Typography.Body2,
    periodicity: periodicity
  }, price), oldPrice && /*#__PURE__*/_react["default"].createElement(StyledPrice, {
    $type: _types.PriceTypes.OLD,
    stroke: true,
    forwardedAs: _Typography.Caption,
    periodicity: periodicity
  }, oldPrice));
});
