// @salutejs/plasma-ui/components/ProductCard/ProductCard.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ProductCard = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _color = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("color"));
var _utils = /*#__PURE__*/require("../../utils");
var _Typography = /*#__PURE__*/require("../Typography");
var _Card = /*#__PURE__*/require("../Card/Card");
var _CardBody = /*#__PURE__*/require("../Card/CardBody");
var _CardContent = /*#__PURE__*/require("../Card/CardContent");
var _hooks = /*#__PURE__*/require("../../hooks");
var _ProductCardStepper = /*#__PURE__*/require("./ProductCardStepper");
var _ProductCardText = /*#__PURE__*/require("./ProductCardText");
var _ProductCardPrice = /*#__PURE__*/require("./ProductCardPrice");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _excluded = ["badge", "media", "text", "additionalInfo", "price", "oldPrice", "quantity", "quantityStep", "quantityMin", "quantityMax", "onQuantityChange", "disabled", "disabledAction", "readonly", "backgroundColor", "periodicity"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('ProductCard/ProductCard');
var StyledRoot = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-tygkir-0"
})(["position:relative;"]);
var StyledCard = /*#__PURE__*/(0, _styledComponents["default"])(_Card.Card).withConfig({
  componentId: "plasma-ui__sc-tygkir-1"
})(["height:100%;background-color:", ";color:", ";"], function (_ref) {
  var $backgroundColor = _ref.$backgroundColor;
  return $backgroundColor;
}, _plasmaCore.blackSecondary);
var StyledMediaSlot = /*#__PURE__*/(0, _react.memo)( /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-tygkir-2"
})(["height:100%;min-height:0;& img{display:block;height:100%;}"]));
var getGradient = function getGradient(backgroundColor) {
  var color = (0, _color["default"])(backgroundColor);
  return "linear-gradient(\n        180deg,\n        ".concat(color.alpha(0).string(), " 0%,\n        ").concat(color.alpha(0.0156863).string(), " 8.62%,\n        ").concat(color.alpha(0.054902).string(), " 16.56%,\n        ").concat(color.alpha(0.117647).string(), " 23.93%,\n        ").concat(color.alpha(0.2).string(), " 30.85%,\n        ").concat(color.alpha(0.290196).string(), " 37.42%,\n        ").concat(color.alpha(0.392157).string(), " 43.77%,\n        ").concat(color.alpha(0.501961).string(), " 50%,\n        ").concat(color.alpha(0.607843).string(), " 56.23%,\n        ").concat(color.alpha(0.709804).string(), " 62.58%,\n        ").concat(color.alpha(0.8).string(), " 69.15%,\n        ").concat(color.alpha(0.882353).string(), " 76.07%,\n        ").concat(color.alpha(0.945098).string(), " 83.44%,\n        ").concat(color.alpha(0.984314).string(), " 91.38%,\n        ").concat(backgroundColor, " 100%\n    )");
};
var StyledCardBody = /*#__PURE__*/(0, _styledComponents["default"])(_CardBody.CardBody).withConfig({
  componentId: "plasma-ui__sc-tygkir-3"
})(["justify-content:space-between;"]);

// TODO: https://github.com/salute-developers/plasma/issues/232
var StyledCardContent = /*#__PURE__*/(0, _styledComponents["default"])(_CardContent.CardContent).withConfig({
  componentId: "plasma-ui__sc-tygkir-4"
})(["padding:0 0.75rem 0.75rem;border-radius:0;background-color:", ";transition:", ";&::after{content:'';position:absolute;top:-2.5rem;left:0;right:0;height:2.5rem;transition:", ";background:", ";}", ""], function (_ref2) {
  var $backgroundColor = _ref2.$backgroundColor;
  return $backgroundColor;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.lowPerformance ? 'unset' : 'all 0.15s ease-in-out';
}, function (_ref4) {
  var theme = _ref4.theme;
  return theme.lowPerformance ? 'unset' : 'all 0.15s ease-in-out';
}, function (_ref5) {
  var $backgroundColor = _ref5.$backgroundColor;
  return $backgroundColor && getGradient($backgroundColor);
}, function (_ref6) {
  var $isValuePositive = _ref6.$isValuePositive;
  return $isValuePositive && (0, _styledComponents.css)(["margin-top:-3rem;"]);
});

// TODO: https://github.com/salute-developers/plasma/issues/232
var StyledAdditionalInfo = /*#__PURE__*/(0, _react.memo)( /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Footnote1).withConfig({
  componentId: "plasma-ui__sc-tygkir-5"
})(["overflow:hidden;text-overflow:ellipsis;white-space:nowrap;", ""], function (props) {
  return (0, _utils.mediaQuery)('XL', props.theme.deviceScale)((0, _styledComponents.css)(["", ""], _plasmaCore.body1));
}));
var StyledBottom = /*#__PURE__*/(0, _react.memo)( /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-tygkir-6"
})(["display:flex;align-items:center;flex-direction:column;margin-top:0.25rem;"]));

// TODO: https://github.com/salute-developers/plasma/issues/232
var StyledStepper = /*#__PURE__*/(0, _react.memo)( /*#__PURE__*/(0, _styledComponents["default"])(_ProductCardStepper.ProductCardStepper).withConfig({
  componentId: "plasma-ui__sc-tygkir-7"
})(["width:100%;transition:", ";", ""], function (_ref7) {
  var theme = _ref7.theme;
  return theme.lowPerformance ? 'unset' : 'all 0.15s ease-in-out';
}, function (_ref8) {
  var $onTop = _ref8.$onTop;
  return $onTop ? (0, _styledComponents.css)(["margin-top:-2.5rem;"]) : (0, _styledComponents.css)(["margin-top:0.5rem;"]);
}));
var StyledBadges = /*#__PURE__*/(0, _react.memo)( /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-tygkir-8"
})(["position:absolute;top:0.5rem;left:0.5rem;z-index:1;display:flex;flex-wrap:wrap;gap:0.25rem;max-width:calc(100% - 0.75rem);"]));

/**
 * Карточка продукта с возможностью указания картинки, текста, цены и выбора количества.
 */
// eslint-disable-next-line prefer-arrow-callback
var ProductCard = exports.ProductCard = /*#__PURE__*/(0, _react.forwardRef)(function ProductCard(_ref9, ref) {
  var badge = _ref9.badge,
    media = _ref9.media,
    text = _ref9.text,
    additionalInfo = _ref9.additionalInfo,
    price = _ref9.price,
    oldPrice = _ref9.oldPrice,
    quantity = _ref9.quantity,
    quantityStep = _ref9.quantityStep,
    quantityMin = _ref9.quantityMin,
    quantityMax = _ref9.quantityMax,
    onQuantityChange = _ref9.onQuantityChange,
    disabled = _ref9.disabled,
    disabledAction = _ref9.disabledAction,
    readonly = _ref9.readonly,
    _ref9$backgroundColor = _ref9.backgroundColor,
    backgroundColor = _ref9$backgroundColor === void 0 ? '#FFFFFF' : _ref9$backgroundColor,
    periodicity = _ref9.periodicity,
    rest = _objectWithoutProperties(_ref9, _excluded);
  var _useDeviceKind = (0, _hooks.useDeviceKind)(),
    isSberBox = _useDeviceKind.isSberBox;
  var isReadonly = isSberBox || readonly;
  var tabIndex = isSberBox ? 0 : -1;
  return /*#__PURE__*/_react["default"].createElement(StyledRoot, null, badge && /*#__PURE__*/_react["default"].createElement(StyledBadges, null, badge), /*#__PURE__*/_react["default"].createElement(StyledCard, _extends({}, rest, {
    ref: ref,
    tabIndex: tabIndex,
    outlined: isSberBox,
    disabled: disabled,
    $backgroundColor: backgroundColor
  }), /*#__PURE__*/_react["default"].createElement(StyledCardBody, null, media && /*#__PURE__*/_react["default"].createElement(StyledMediaSlot, null, media), /*#__PURE__*/_react["default"].createElement(StyledCardContent, {
    $isValuePositive: !isReadonly && !!quantity && quantity > 0,
    $backgroundColor: backgroundColor
  }, /*#__PURE__*/_react["default"].createElement(_ProductCardText.ProductCardText, {
    text: text
  }), additionalInfo && /*#__PURE__*/_react["default"].createElement(StyledAdditionalInfo, null, additionalInfo), (price !== undefined || quantity !== undefined) && /*#__PURE__*/_react["default"].createElement(StyledBottom, null, /*#__PURE__*/_react["default"].createElement(_ProductCardPrice.ProductCardPrice, {
    price: price,
    oldPrice: oldPrice,
    periodicity: periodicity
  }), !disabled && quantity !== undefined && /*#__PURE__*/_react["default"].createElement(StyledStepper, {
    readonly: isReadonly,
    value: quantity,
    step: quantityStep,
    min: quantityMin,
    max: quantityMax,
    onChange: onQuantityChange,
    $onTop: isReadonly || price !== undefined && quantity === 0,
    disabled: disabledAction || disabled
  }))))));
});
