// @salutejs/plasma-core/components/Image/Image.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Image = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _utils = /*#__PURE__*/require("../../utils");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _excluded = ["src", "srcSet", "sizes", "base", "alt", "width", "height"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('Image/Image');
var ratios = {
  '1 / 1': '100',
  '1/1': '100',
  '3 / 4': '133.3333',
  '3/4': '133.3333',
  '4 / 3': '75',
  '4/3': '75',
  '9 / 16': '177.7778',
  '9/16': '177.7778',
  '16 / 9': '56.25',
  '16/9': '56.25',
  '1 / 2': '200',
  '1/2': '200',
  '2 / 1': '50',
  '2/1': '50'
};
var StyledRoot = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-3h0amr-0"
})(["position:relative;display:block;box-sizing:border-box;overflow:hidden;width:100%;height:inherit;border-radius:inherit;", " ", ""], function (_ref) {
  var $ratio = _ref.$ratio,
    $customRatio = _ref.$customRatio;
  return ($ratio || $customRatio) && (0, _styledComponents.css)(["height:0;padding-bottom:", "%;"], $ratio ? ratios[$ratio] : $customRatio);
}, function (_ref2) {
  var $width = _ref2.$width,
    $height = _ref2.$height;
  return (0, _styledComponents.css)(["", " ", ""], $width && "width: ".concat((0, _utils.toCssSize)($width), ";"), $height && "height: ".concat((0, _utils.toCssSize)($height), ";"));
});
var StyledImg = /*#__PURE__*/_styledComponents["default"].img.withConfig({
  componentId: "plasma__sc-3h0amr-1"
})(["width:100%;", ";"], function (_ref3) {
  var src = _ref3.src,
    ratio = _ref3.ratio;
  return src && ratio && (0, _styledComponents.css)(["height:100%;object-fit:cover;position:absolute;"]);
});
var StyledDivImg = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma__sc-3h0amr-2"
})(["width:100%;height:100%;position:absolute;background-position:center;background-size:cover;"]);

/**
 * Компонент для отображения картинок.
 */
var ImageBase = function ImageBase(_ref4) {
  var src = _ref4.src,
    srcSet = _ref4.srcSet,
    sizes = _ref4.sizes,
    _ref4$base = _ref4.base,
    base = _ref4$base === void 0 ? 'img' : _ref4$base,
    alt = _ref4.alt,
    width = _ref4.width,
    height = _ref4.height,
    props = _objectWithoutProperties(_ref4, _excluded);
  var ratio = 'ratio' in props ? props.ratio : undefined;
  var customRatio = 'customRatio' in props ? props.customRatio : undefined;
  return /*#__PURE__*/_react["default"].createElement(StyledRoot, _extends({
    $ratio: ratio,
    $customRatio: customRatio,
    $width: width,
    $height: height
  }, props), base === 'img' && /*#__PURE__*/_react["default"].createElement(StyledImg, {
    src: src,
    srcSet: srcSet,
    sizes: sizes,
    alt: alt,
    ratio: ratio
  }), base === 'div' && /*#__PURE__*/_react["default"].createElement(StyledDivImg, {
    style: {
      backgroundImage: "url(".concat(src, ")")
    },
    role: "img",
    "aria-label": alt
  }));
};

/**
 * Компонент для отображения картинок.
 */
var Image = exports.Image = /*#__PURE__*/(0, _styledComponents["default"])(ImageBase).withConfig({
  componentId: "plasma__sc-3h0amr-3"
})([""]);
