// @salutejs/plasma-tokens/colors/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.overlaySoft = exports.overlay = exports.speechBubbleReceived = exports.speechBubbleSent = exports.skeletonGradientLighter = exports.skeletonGradient = exports.buttonChecked = exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonSecondary = exports.buttonPrimary = exports.surfaceCard = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.surfaceLiquid03 = exports.surfaceLiquid02 = exports.surfaceLiquid01 = exports.critical = exports.warning = exports.success = exports.backgroundTertiary = exports.backgroundSecondary = exports.backgroundPrimary = exports.background = exports.inverse = exports.paragraph = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlackTransparent = exports.buttonBlackSecondary = exports.buttonBlack = exports.buttonClear = exports.transparent = exports.dark03 = exports.dark02 = exports.dark01 = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.black = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = exports.white = void 0;
exports.voicePhraseGradient = exports.gradientDevice = exports.gradient = exports.buttonFocused = exports.buttonAccent = exports.accent = exports.inverseOverlayBlur = exports.inverseOverlayHard = exports.inverseOverlaySoft = exports.lightOverlayBlur = exports.lightOverlayHard = exports.lightOverlaySoft = exports.darkOverlayBlur = exports.darkOverlayHard = exports.darkOverlaySoft = exports.overlayBlur = exports.overlayHard = void 0;
/** Базовый белый цвет совпадает с whitePrimary */
exports.white = 'var(--plasma-colors-white, #FFFFFF)';
/** Первичный белый цвет */
exports.whitePrimary = 'var(--plasma-colors-white-primary, #FFFFFF)';
/** Вторичный белый цвет */
exports.whiteSecondary = 'var(--plasma-colors-white-secondary, rgba(255, 255, 255, 0.56))';
/** Третичный белый цвет */
exports.whiteTertiary = 'var(--plasma-colors-white-tertiary, rgba(255, 255, 255, 0.28))';
/** Базовый черный цвет совпадает с blackPrimary */
exports.black = 'var(--plasma-colors-black, #080808)';
/** Первичный черный цвет */
exports.blackPrimary = 'var(--plasma-colors-black-primary, #080808)';
/** Вторичный черныйцвет */
exports.blackSecondary = 'var(--plasma-colors-black-secondary, rgba(8, 8, 8, 0.56))';
/** Третичный черный цвет */
exports.blackTertiary = 'var(--plasma-colors-black-tertiary, rgba(8, 8, 8, 0.28))';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark01 = 'var(--plasma-colors-dark01, #171717)';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark02 = 'var(--plasma-colors-dark02, #232323)';
/** Чёрный непрозрачный, не зависит от темы */
exports.dark03 = 'var(--plasma-colors-dark03, #363636)';
/** Прозрачный цвет */
exports.transparent = 'var(--plasma-colors-transparent, rgba(0, 0, 0, 0))';
/** Второстепенная прозрачная */
exports.buttonClear = 'var(--plasma-colors-button-clear, rgba(0, 0, 0, 0))';
/** Дефолтная чёрная, не зависит от темы */
exports.buttonBlack = 'var(--plasma-colors-button-black, #080808)';
/** Второстепенная чёрная, не зависит от темы */
exports.buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, rgba(8, 8, 8, 0.12))';
/** Чёрная для использования поверх картинок */
exports.buttonBlackTransparent = 'var(--plasma-colors-button-black-transparent, rgba(8, 8, 8, 0.56))';
/** Дефолтная белая, не зависит от темы */
exports.buttonWhite = 'var(--plasma-colors-button-white, #FFFFFF)';
/** Второстепенная белая, не зависит от темы */
exports.buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, rgba(255, 255, 255, 0.12))';
/** Базовый цвет текста, совпадает с primary */
exports.text = 'var(--plasma-colors-text, rgba(255, 255, 255, 0.96))';
/** Основной текст в интерфейсе */
exports.primary = 'var(--plasma-colors-primary, rgba(255, 255, 255, 0.96))';
/** Второстепенный текст */
exports.secondary = 'var(--plasma-colors-secondary, rgba(255, 255, 255, 0.56))';
/** Третичный цвет текста */
exports.tertiary = 'var(--plasma-colors-tertiary, rgba(255, 255, 255, 0.28))';
/** Сплошной наборный текст */
exports.paragraph = 'var(--plasma-colors-paragraph, rgba(255, 255, 255, 0.96))';
/** Белый в светлой теме / Чёрный в тёмной теме */
exports.inverse = 'var(--plasma-colors-inverse, #080808)';
/** Основной бэкграунд всех приложений и смартапов */
exports.background = 'var(--plasma-colors-background, #080808)';
/** Первичный цвет фона */
exports.backgroundPrimary = 'var(--plasma-colors-background-primary, #171717)';
/** Вторичный цвет фона */
exports.backgroundSecondary = 'var(--plasma-colors-background-secondary, #232323)';
/** Третичный цвет фона */
exports.backgroundTertiary = 'var(--plasma-colors-background-tertiary, #363636)';
/** Обозначение успешного сценария */
exports.success = 'var(--plasma-colors-success, #24B23E)';
/** Цвет предупреждения */
exports.warning = 'var(--plasma-colors-warning, #FA6D20)';
/** Цвет ошибки */
exports.critical = 'var(--plasma-colors-critical, #FF4D5F)';
/** Полупрозрачные поверхности */
exports.surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, rgba(255, 255, 255, 0.06))';
/** Полупрозрачные поверхности */
exports.surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, rgba(255, 255, 255, 0.12))';
/** Полупрозрачные поверхности */
exports.surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, rgba(255, 255, 255, 0.2))';
/** Непрозрачные поверхности */
exports.surfaceSolid01 = 'var(--plasma-colors-surface-solid01, #171717)';
/** Непрозрачные поверхности */
exports.surfaceSolid02 = 'var(--plasma-colors-surface-solid02, #232323)';
/** Непрозрачные поверхности */
exports.surfaceSolid03 = 'var(--plasma-colors-surface-solid03, #363636)';
/** Основной фон для карточек */
exports.surfaceCard = 'var(--plasma-colors-surface-card, rgba(255, 255, 255, 0.12))';
/** Первичный цвет контролов */
exports.buttonPrimary = 'var(--plasma-colors-button-primary, #FFFFFF)';
/** Второстепенная кнопка */
exports.buttonSecondary = 'var(--plasma-colors-button-secondary, rgba(255, 255, 255, 0.12))';
/** Кнопка для успешного сценария */
exports.buttonSuccess = 'var(--plasma-colors-button-success, #21A038)';
/** Цвет предупреждения у контролов */
exports.buttonWarning = 'var(--plasma-colors-button-warning, #E35502)';
/** Цвет ошибки у контролов */
exports.buttonCritical = 'var(--plasma-colors-button-critical, #FF2E43)';
/** Цвет зажатого контрола */
exports.buttonChecked = 'var(--plasma-colors-button-checked, #FFFFFF)';
exports.skeletonGradient = 'var(--plasma-colors-skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
exports.skeletonGradientLighter = 'var(--plasma-colors-skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** Цвет фона баблов отправленный сообщений */
exports.speechBubbleSent = 'var(--plasma-colors-speech-bubble-sent, rgba(0, 0, 0, 0.28))';
/** Цвет фона баблов получнных сообщений */
exports.speechBubbleReceived = 'var(--plasma-colors-speech-bubble-received, rgba(255, 255, 255, 0.12))';
/** Цвет фона паранжи */
exports.overlay = 'var(--plasma-colors-overlay, rgba(0, 0, 0, 0.8))';
exports.overlaySoft = 'var(--plasma-colors-overlay-soft, rgba(8, 8, 8, 0.56))';
exports.overlayHard = 'var(--plasma-colors-overlay-hard, rgba(8, 8, 8, 0.9))';
exports.overlayBlur = 'var(--plasma-colors-overlay-blur, rgba(23, 23, 23, 0.28))';
exports.darkOverlaySoft = 'var(--plasma-colors-dark-overlay-soft, rgba(8, 8, 8, 0.56))';
exports.darkOverlayHard = 'var(--plasma-colors-dark-overlay-hard, rgba(8, 8, 8, 0.9))';
exports.darkOverlayBlur = 'var(--plasma-colors-dark-overlay-blur, rgba(23, 23, 23, 0.28))';
exports.lightOverlaySoft = 'var(--plasma-colors-light-overlay-soft, rgba(8, 8, 8, 0.56))';
exports.lightOverlayHard = 'var(--plasma-colors-light-overlay-hard, rgba(8, 8, 8, 0.9))';
exports.lightOverlayBlur = 'var(--plasma-colors-light-overlay-blur, rgba(23, 23, 23, 0.28))';
exports.inverseOverlaySoft = 'var(--plasma-colors-inverse-overlay-soft, rgba(8, 8, 8, 0.56))';
exports.inverseOverlayHard = 'var(--plasma-colors-inverse-overlay-hard, rgba(8, 8, 8, 0.9))';
exports.inverseOverlayBlur = 'var(--plasma-colors-inverse-overlay-blur, rgba(23, 23, 23, 0.28))';
/** Акцентный цвет бренда */
exports.accent = 'var(--plasma-colors-accent, #24B23E)';
/** Акцентный цвет у контролов */
exports.buttonAccent = 'var(--plasma-colors-button-accent, #21A038)';
/** Цвет рамки фокуса у контрола */
exports.buttonFocused = 'var(--plasma-colors-button-focused, rgba(255, 255, 255, 0.96))';
/** Градиент для заливки основного фона */
exports.gradient = 'var(--plasma-colors-gradient, linear-gradient(336.84deg, rgba(20, 116, 70, 0.6) 0%, rgba(8, 8, 8, 0) 64.88%), radial-gradient(100% 100% at 75.89% 100%, rgba(0, 133, 255, 0.24) 0%, rgba(0, 71, 255, 0.03) 100%), linear-gradient(180deg, rgba(8, 8, 8, 0) 50%, rgba(7, 71, 33, 0.3) 100%), linear-gradient(270deg, #061621 0%, rgba(8, 8, 8, 0) 100%))';
/** Градиент для заливки основного фона */
exports.gradientDevice = 'var(--plasma-colors-gradient-device, linear-gradient(26.05deg, rgba(15, 153, 24, 0.28) 0%, rgba(8, 8, 8, 0) 72.24%), radial-gradient(100% 100% at 0% 100%, rgba(0, 170, 255, 0.24) 0%, rgba(8, 8, 8, 0) 99.69%), radial-gradient(74.68% 149.35% at 50% 149.35%, rgba(0, 102, 255, 0.6) 0%, rgba(8, 8, 8, 0) 99.69%))';
/** Градиент подсказок о голосовых запросах */
exports.voicePhraseGradient = 'var(--plasma-colors-voice-phrase-gradient, linear-gradient(45deg, hsla(130, 75.63025210084032%, 46.666666666666664%, 1) 0%, hsla(189.78947368421052, 78.51239669421487%, 52.54901960784314%, 1) 100%))';
