// @salutejs/plasma-new-hope/es/components/Panel/ui/PanelHeader/PanelHeader.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef } from 'react';
import { cx } from '../../../../utils/index.js';
import '@linaria/react';
import { IconClose } from '../../../_Icon/Icons/IconClose.js';
import { classes } from '../../Panel.tokens.js';
import { placements } from './PanelHeader.types.js';
import { StyledHeader, ButtonWrapper, CloseIconWrapper } from './PanelHeader.styles.js';
import { base } from './variations/_view/base.js';

var _IconClose;
var _excluded = ["children", "actions", "view", "className", "onClose", "hasClose", "closePlacement"];
var panelHeaderRoot = function panelHeaderRoot(Root) {
  return /*#__PURE__*/forwardRef(function (_ref, outerRef) {
    var children = _ref.children,
      actions = _ref.actions,
      view = _ref.view,
      className = _ref.className,
      onClose = _ref.onClose,
      _ref$hasClose = _ref.hasClose,
      hasClose = _ref$hasClose === void 0 ? 'true' : _ref$hasClose,
      _ref$closePlacement = _ref.closePlacement,
      closePlacement = _ref$closePlacement === void 0 ? 'right' : _ref$closePlacement,
      rest = _objectWithoutProperties(_ref, _excluded);
    var hasHeaderClass = children ? undefined : classes.hasHeader;
    var isRightCloseClass = closePlacement === placements.right ? classes.isRightClose : undefined;
    return /*#__PURE__*/React.createElement(Root, _extends({
      ref: outerRef,
      view: view,
      className: cx(className, hasHeaderClass, isRightCloseClass)
    }, rest), children && /*#__PURE__*/React.createElement(StyledHeader, null, children), actions && /*#__PURE__*/React.createElement(ButtonWrapper, null, actions), hasClose && /*#__PURE__*/React.createElement(ButtonWrapper, {
      placement: closePlacement
    }, /*#__PURE__*/React.createElement(CloseIconWrapper, {
      view: "clear",
      size: "s",
      onClick: onClose
    }, _IconClose || (_IconClose = /*#__PURE__*/React.createElement(IconClose, {
      size: "s",
      color: "inherit"
    })))));
  });
};
var panelHeaderConfig = {
  name: 'PanelHeader',
  tag: 'div',
  layout: panelHeaderRoot,
  base: '',
  variations: {
    view: {
      css: base
    }
  },
  defaults: {
    view: 'default'
  }
};

export { panelHeaderConfig, panelHeaderRoot };
