// @salutejs/plasma-new-hope/es/components/Panel/ui/PanelHeader/PanelHeader.styles.js
import './PanelHeader.styles_1cid6t7.css';
import { styled } from '@linaria/react';
import '@linaria/core';
import { component, mergeConfig } from '../../../../engines/common.js';
import { buttonConfig } from '../../../Button/Button.js';
import { placements } from './PanelHeader.types.js';

var mergedButtonConfig = /*#__PURE__*/mergeConfig(buttonConfig);
var Button = /*#__PURE__*/component(mergedButtonConfig);
var _exp2 = function _exp2() {
  return function (_ref) {
    var placement = _ref.placement;
    return placement === placements.left ? -1 : 0;
  };
};
var _exp3 = function _exp3() {
  return function (_ref2) {
    var placement = _ref2.placement;
    return !placement || placement === placements.right ? '0.5rem' : '';
  };
};
var _exp4 = function _exp4() {
  return function (_ref3) {
    var placement = _ref3.placement;
    return placement === placements.left ? '0.5rem' : '';
  };
};
var ButtonWrapper = /*#__PURE__*/styled('div')({
  name: "ButtonWrapper",
  "class": "b5xb1jq",
  propsAsIs: false,
  vars: {
    "b5xb1jq-0": [/*#__PURE__*/_exp2()],
    "b5xb1jq-1": [/*#__PURE__*/_exp3()],
    "b5xb1jq-2": [/*#__PURE__*/_exp4()]
  }
});
var StyledHeader = /*#__PURE__*/styled('div')({
  name: "StyledHeader",
  "class": "s1hqmxl1",
  propsAsIs: false
});
var _exp5 = function _exp5() {
  return Button;
};
var CloseIconWrapper = /*#__PURE__*/styled(_exp5())({
  name: "CloseIconWrapper",
  "class": "c7hx8go",
  propsAsIs: true
});

export { ButtonWrapper, CloseIconWrapper, StyledHeader };
