// @salutejs/plasma-new-hope/styled-components/cjs/components/Dropdown/ui/DropdownOld/DropdownOld.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.dropdownOldRoot = exports.dropdownOldConfig = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _base = /*#__PURE__*/require("./variations/_view/base");
var _base2 = /*#__PURE__*/require("./variations/_size/base");
var _Dropdown = /*#__PURE__*/require("./Dropdown.styles");
var _utils = /*#__PURE__*/require("./utils");
var _excluded = ["id", "target", "children", "hasArrow", "role", "view", "size", "frame", "onToggle", "isFocusTrapped", "isOpen", "placement", "trigger", "offset", "preventOverflow", "closeOnOverlayClick", "closeOnEsc"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
/**
 * Выпадающий список без внешнего контроля видимости.
 */
var dropdownOldRoot = exports.dropdownOldRoot = function dropdownOldRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (_ref, outerRootRef) {
    var id = _ref.id,
      target = _ref.target,
      children = _ref.children,
      hasArrow = _ref.hasArrow,
      role = _ref.role,
      view = _ref.view,
      size = _ref.size,
      frame = _ref.frame,
      _onToggle = _ref.onToggle,
      _ref$isFocusTrapped = _ref.isFocusTrapped,
      isFocusTrapped = _ref$isFocusTrapped === void 0 ? true : _ref$isFocusTrapped,
      _ref$isOpen = _ref.isOpen,
      isOpen = _ref$isOpen === void 0 ? false : _ref$isOpen,
      _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? 'auto' : _ref$placement,
      _ref$trigger = _ref.trigger,
      trigger = _ref$trigger === void 0 ? 'click' : _ref$trigger,
      _ref$offset = _ref.offset,
      offset = _ref$offset === void 0 ? [0, 6] : _ref$offset,
      _ref$preventOverflow = _ref.preventOverflow,
      preventOverflow = _ref$preventOverflow === void 0 ? false : _ref$preventOverflow,
      _ref$closeOnOverlayCl = _ref.closeOnOverlayClick,
      closeOnOverlayClick = _ref$closeOnOverlayCl === void 0 ? false : _ref$closeOnOverlayCl,
      _ref$closeOnEsc = _ref.closeOnEsc,
      closeOnEsc = _ref$closeOnEsc === void 0 ? false : _ref$closeOnEsc,
      rest = _objectWithoutProperties(_ref, _excluded);
    var uniqId = (0, _plasmaCore.useUniqId)();
    var innerId = id || uniqId;
    var rootRef = (0, _react.useRef)(null);
    var dropdownRef = (0, _react.useRef)(null);
    var handleRef = (0, _plasmaCore.useForkRef)(rootRef, outerRootRef);
    var trapRef = (0, _plasmaCore.useFocusTrap)(isOpen && isFocusTrapped);
    var dropdownForkRef = (0, _plasmaCore.useForkRef)(dropdownRef, trapRef);
    return /*#__PURE__*/_react["default"].createElement(_Dropdown.StyledPopover, {
      role: role,
      isOpen: isOpen,
      usePortal: false,
      onToggle: function onToggle(is, event) {
        return _onToggle === null || _onToggle === void 0 ? void 0 : _onToggle(is, event);
      },
      id: innerId,
      ref: dropdownForkRef,
      target: target,
      offset: offset,
      preventOverflow: preventOverflow,
      hasArrow: hasArrow,
      placement: (0, _utils.getPlacements)(placement),
      trigger: trigger,
      closeOnOverlayClick: closeOnOverlayClick,
      closeOnEsc: closeOnEsc,
      isFocusTrapped: isFocusTrapped,
      frame: frame
    }, /*#__PURE__*/_react["default"].createElement(Root, _extends({
      ref: handleRef,
      view: view,
      size: size
    }, rest), /*#__PURE__*/_react["default"].createElement(_Dropdown.StyledDropdown, null, children)));
  });
};
var dropdownOldConfig = exports.dropdownOldConfig = {
  name: 'Dropdown',
  tag: 'div',
  layout: dropdownOldRoot,
  base: '',
  variations: {
    view: {
      css: _base.base
    },
    size: {
      css: _base2.base
    }
  },
  defaults: {
    view: 'default',
    size: 'm'
  }
};
