// @salutejs/plasma-tokens/es/brands/sbermarket_metro/index.js
// Generated by robots, do not change this manually!
/** Основной цвет текста */
export var textPrimaryHover = 'var(--text-primary-hover, #B3C5D0FF)';
/** Основной цвет текста */
export var textPrimaryActive = 'var(--text-primary-active, #D3DDE4FF)';
/** Основной цвет текста */
export var textPrimary = 'var(--text-primary, #F2F5F7)';
/** Вторичный цвет текста */
export var textSecondaryHover = 'var(--text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста */
export var textSecondaryActive = 'var(--text-secondary-active, #353E46FF)';
/** Вторичный цвет текста */
export var textSecondary = 'var(--text-secondary, #A8B4BD)';
/** Третичный цвет текста */
export var textTertiaryHover = 'var(--text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста */
export var textTertiaryActive = 'var(--text-tertiary-active, #5B6267FF)';
/** Третичный цвет текста */
export var textTertiary = 'var(--text-tertiary, #7D868C)';
/** Сплошной наборный текст */
export var textParagraphHover = 'var(--text-paragraph-hover, #FFFFFFFF)';
/** Сплошной наборный текст */
export var textParagraphActive = 'var(--text-paragraph-active, #262F36FF)';
/** Сплошной наборный текст */
export var textParagraph = 'var(--text-paragraph, #BEC9D1)';
/** Акцентный цвет */
export var textAccentHover = 'var(--text-accent-hover, #5297FFFF)';
/** Акцентный цвет */
export var textAccentActive = 'var(--text-accent-active, #005AE0FF)';
/** Акцентный цвет */
export var textAccent = 'var(--text-accent, #0064FE)';
/** Цвет ошибки */
export var textNegativeHover = 'var(--text-negative-hover, #FF4D5EFF)';
/** Цвет ошибки */
export var textNegativeActive = 'var(--text-negative-active, #FF0F27FF)';
/** Цвет ошибки */
export var textNegative = 'var(--text-negative, #FF2E43)';
/** Дополнительный акцент */
export var textAccentDiscountHover = 'var(--text-accent-discount-hover, #FF3D8BFF)';
/** Дополнительный акцент */
export var textAccentDiscountActive = 'var(--text-accent-discount-active, #FF0066FF)';
/** Дополнительный акцент */
export var textAccentDiscount = 'var(--text-accent-discount, #FF1F78)';
/** Дополнительный акцент */
export var textAccentGoldHover = 'var(--text-accent-gold-hover, #FFDE5CFF)';
/** Дополнительный акцент */
export var textAccentGoldActive = 'var(--text-accent-gold-active, #FFD21FFF)';
/** Дополнительный акцент */
export var textAccentGold = 'var(--text-accent-gold, #FFD83D)';
/** Акцентный цвет бренда */
export var textAccentB2BHover = 'var(--text-accent-b2-b-hover, #23F7FBFF)';
/** Акцентный цвет бренда */
export var textAccentB2BActive = 'var(--text-accent-b2-b-active, #03A7AAFF)';
/** Акцентный цвет бренда */
export var textAccentB2B = 'var(--text-accent-b2-b, #04C6C9)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimaryHover = 'var(--on-dark-text-primary-hover, #F2F5F793)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimaryActive = 'var(--on-dark-text-primary-active, #F2F5F7C4)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimary = 'var(--on-dark-text-primary, rgba(242, 245, 247, 0.96))';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondaryHover = 'var(--on-dark-text-secondary-hover, #F2F5F7FF)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondaryActive = 'var(--on-dark-text-secondary-active, #F2F5F7C4)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondary = 'var(--on-dark-text-secondary, rgba(242, 245, 247, 0.64))';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiaryHover = 'var(--on-dark-text-tertiary-hover, #F2F5F7FF)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiaryActive = 'var(--on-dark-text-tertiary-active, #F2F5F77A)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiary = 'var(--on-dark-text-tertiary, rgba(242, 245, 247, 0.4))';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraphHover = 'var(--on-dark-text-paragraph-hover, #F2F5F77A)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraphActive = 'var(--on-dark-text-paragraph-active, #F2F5F7A3)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraph = 'var(--on-dark-text-paragraph, rgba(242, 245, 247, 0.8))';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccentHover = 'var(--on-dark-text-accent-hover, #5297FFFF)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccentActive = 'var(--on-dark-text-accent-active, #005AE0FF)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccent = 'var(--on-dark-text-accent, #0064FE)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegativeHover = 'var(--on-dark-text-negative-hover, #FF4D5EFF)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegativeActive = 'var(--on-dark-text-negative-active, #FF0F27FF)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegative = 'var(--on-dark-text-negative, #FF2E43)';
/** Дополнительный акцент */
export var onDarkTextAccentDiscountHover = 'var(--on-dark-text-accent-discount-hover, #FF3D8BFF)';
/** Дополнительный акцент */
export var onDarkTextAccentDiscountActive = 'var(--on-dark-text-accent-discount-active, #FF0066FF)';
/** Дополнительный акцент */
export var onDarkTextAccentDiscount = 'var(--on-dark-text-accent-discount, #FF1F78)';
/** Дополнительный акцент */
export var onDarkTextAccentGoldHover = 'var(--on-dark-text-accent-gold-hover, #FFDE5CFF)';
/** Дополнительный акцент */
export var onDarkTextAccentGoldActive = 'var(--on-dark-text-accent-gold-active, #FFD21FFF)';
/** Дополнительный акцент */
export var onDarkTextAccentGold = 'var(--on-dark-text-accent-gold, #FFD83D)';
/** Акцентный цвет бренда */
export var onDarkTextAccentB2BHover = 'var(--on-dark-text-accent-b2-b-hover, #23F7FBFF)';
/** Акцентный цвет бренда */
export var onDarkTextAccentB2BActive = 'var(--on-dark-text-accent-b2-b-active, #03A7AAFF)';
/** Акцентный цвет бренда */
export var onDarkTextAccentB2B = 'var(--on-dark-text-accent-b2-b, #04C6C9)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimaryHover = 'var(--on-light-text-primary-hover, #FFFFFFFF)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimaryActive = 'var(--on-light-text-primary-active, #A6AEB5FF)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimary = 'var(--on-light-text-primary, #24282B)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondaryHover = 'var(--on-light-text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondaryActive = 'var(--on-light-text-secondary-active, #51585DFF)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondary = 'var(--on-light-text-secondary, #8C959C)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiaryHover = 'var(--on-light-text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiaryActive = 'var(--on-light-text-tertiary-active, #262F36FF)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiary = 'var(--on-light-text-tertiary, #BEC9D1)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraphHover = 'var(--on-light-text-paragraph-hover, #FFFFFFFF)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraphActive = 'var(--on-light-text-paragraph-active, #A6AEB5FF)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraph = 'var(--on-light-text-paragraph, #24282B)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccentHover = 'var(--on-light-text-accent-hover, #5297FFFF)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccentActive = 'var(--on-light-text-accent-active, #005AE0FF)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccent = 'var(--on-light-text-accent, #0064FE)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegativeHover = 'var(--on-light-text-negative-hover, #F5384BFF)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegativeActive = 'var(--on-light-text-negative-active, #E40C22FF)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegative = 'var(--on-light-text-negative, #F31B31)';
/** Дополнительный акцент */
export var onLightTextAccentDiscountHover = 'var(--on-light-text-accent-discount-hover, #FF3D8BFF)';
/** Дополнительный акцент */
export var onLightTextAccentDiscountActive = 'var(--on-light-text-accent-discount-active, #FF0066FF)';
/** Дополнительный акцент */
export var onLightTextAccentDiscount = 'var(--on-light-text-accent-discount, #FF1F78)';
/** Дополнительный акцент */
export var onLightTextAccentGoldHover = 'var(--on-light-text-accent-gold-hover, #FFDE5CFF)';
/** Дополнительный акцент */
export var onLightTextAccentGoldActive = 'var(--on-light-text-accent-gold-active, #FFD21FFF)';
/** Дополнительный акцент */
export var onLightTextAccentGold = 'var(--on-light-text-accent-gold, #FFD83D)';
/** Акцентный цвет бренда */
export var onLightTextAccentB2BHover = 'var(--on-light-text-accent-b2-b-hover, #11BDC0FF)';
/** Акцентный цвет бренда */
export var onLightTextAccentB2BActive = 'var(--on-light-text-accent-b2-b-active, #0EA2A4FF)';
/** Акцентный цвет бренда */
export var onLightTextAccentB2B = 'var(--on-light-text-accent-b2-b, #0C8688)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimaryHover = 'var(--inverse-text-primary-hover, #FFFFFFFF)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimaryActive = 'var(--inverse-text-primary-active, #A6AEB5FF)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimary = 'var(--inverse-text-primary, #24282B)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondaryHover = 'var(--surface-solid-secondary-hover, #2C3135FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondaryActive = 'var(--surface-solid-secondary-active, #1E2224FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondary = 'var(--surface-solid-secondary, #24282B)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiaryHover = 'var(--surface-solid-tertiary-hover, #3A4145FF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiaryActive = 'var(--surface-solid-tertiary-active, #2C3135FF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiary = 'var(--surface-solid-tertiary, #31373B)';
/** Основной фон для карточек */
export var surfaceSolidCardHover = 'var(--surface-solid-card-hover, #191D1FFF)';
/** Основной фон для карточек */
export var surfaceSolidCardActive = 'var(--surface-solid-card-active, #101214FF)';
/** Основной фон для карточек */
export var surfaceSolidCard = 'var(--surface-solid-card, #15181A)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentHover = 'var(--surface-accent-hover, #1F78FFFF)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentActive = 'var(--surface-accent-active, #0062F5FF)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccent = 'var(--surface-accent, #0064FE)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradientHover = 'var(--surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradientActive = 'var(--surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%))';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegativeHover = 'var(--surface-negative-hover, #FF4D5EFF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegativeActive = 'var(--surface-negative-active, #FF243AFF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegative = 'var(--surface-negative, #FF2E43)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentB2BHover = 'var(--surface-accent-b2-b-hover, #0EBBBEFF)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentB2BActive = 'var(--surface-accent-b2-b-active, #0C9FA1FF)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentB2B = 'var(--surface-accent-b2-b, #0DA8AB)';
/** Дополнительный акцент для обозначения акций */
export var surfaceAccentDiscountHover = 'var(--surface-accent-discount-hover, #FF3D8BFF)';
/** Дополнительный акцент для обозначения акций */
export var surfaceAccentDiscountActive = 'var(--surface-accent-discount-active, #FF1472FF)';
/** Дополнительный акцент для обозначения акций */
export var surfaceAccentDiscount = 'var(--surface-accent-discount, #FF1F78)';
/** Дополнительный акцент */
export var surfaceAccentGoldHover = 'var(--surface-accent-gold-hover, #FFDE5CFF)';
/** Дополнительный акцент */
export var surfaceAccentGoldActive = 'var(--surface-accent-gold-active, #FFD633FF)';
/** Дополнительный акцент */
export var surfaceAccentGold = 'var(--surface-accent-gold, #FFD83D)';
/** Дополнительный акцент */
export var surfaceAccentBlueHover = 'var(--surface-accent-blue-hover, #75A5FFFF)';
/** Дополнительный акцент */
export var surfaceAccentBlueActive = 'var(--surface-accent-blue-active, #4D8BFFFF)';
/** Дополнительный акцент */
export var surfaceAccentBlue = 'var(--surface-accent-blue, #5993FF)';
/** Пунктирные дивайдеры и неактивные контролы */
export var surfaceAdditionalHover = 'var(--surface-additional-hover, #535A5FFF)';
/** Пунктирные дивайдеры и неактивные контролы */
export var surfaceAdditionalActive = 'var(--surface-additional-active, #454A4FFF)';
/** Пунктирные дивайдеры и неактивные контролы */
export var surfaceAdditional = 'var(--surface-additional, #494F54)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondaryHover = 'var(--on-dark-surface-solid-secondary-hover, #2C3135FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondaryActive = 'var(--on-dark-surface-solid-secondary-active, #1E2224FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #24282B)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiaryHover = 'var(--on-dark-surface-solid-tertiary-hover, #3A4145FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiaryActive = 'var(--on-dark-surface-solid-tertiary-active, #2C3135FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #31373B)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCardHover = 'var(--on-dark-surface-solid-card-hover, #191D1FFF)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCardActive = 'var(--on-dark-surface-solid-card-active, #101214FF)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #15181A)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentHover = 'var(--on-dark-surface-accent-hover, #1F78FFFF)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentActive = 'var(--on-dark-surface-accent-active, #0062F5FF)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #0064FE)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradientHover = 'var(--on-dark-surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradientActive = 'var(--on-dark-surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradient = 'var(--on-dark-surface-accent-gradient, linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%))';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegativeHover = 'var(--on-dark-surface-negative-hover, #FF4D5EFF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegativeActive = 'var(--on-dark-surface-negative-active, #FF243AFF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF2E43)';
/** Акцентный фон поверхности/контрола */
export var onDarkSurfaceAccentB2BHover = 'var(--on-dark-surface-accent-b2-b-hover, #0EBBBEFF)';
/** Акцентный фон поверхности/контрола */
export var onDarkSurfaceAccentB2BActive = 'var(--on-dark-surface-accent-b2-b-active, #0C9FA1FF)';
/** Акцентный фон поверхности/контрола */
export var onDarkSurfaceAccentB2B = 'var(--on-dark-surface-accent-b2-b, #0DA8AB)';
/** Дополнительный акцент для обозначения акций */
export var onDarkSurfaceAccentDiscountHover = 'var(--on-dark-surface-accent-discount-hover, #FF3D8BFF)';
/** Дополнительный акцент для обозначения акций */
export var onDarkSurfaceAccentDiscountActive = 'var(--on-dark-surface-accent-discount-active, #FF1472FF)';
/** Дополнительный акцент для обозначения акций */
export var onDarkSurfaceAccentDiscount = 'var(--on-dark-surface-accent-discount, #FF1F78)';
/** Дополнительный акцент */
export var onDarkSurfaceAccentGoldHover = 'var(--on-dark-surface-accent-gold-hover, #FFDE5CFF)';
/** Дополнительный акцент */
export var onDarkSurfaceAccentGoldActive = 'var(--on-dark-surface-accent-gold-active, #FFD633FF)';
/** Дополнительный акцент */
export var onDarkSurfaceAccentGold = 'var(--on-dark-surface-accent-gold, #FFD83D)';
/** Дополнительный акцент */
export var onDarkSurfaceAccentBlueHover = 'var(--on-dark-surface-accent-blue-hover, #75A5FFFF)';
/** Дополнительный акцент */
export var onDarkSurfaceAccentBlueActive = 'var(--on-dark-surface-accent-blue-active, #4D8BFFFF)';
/** Дополнительный акцент */
export var onDarkSurfaceAccentBlue = 'var(--on-dark-surface-accent-blue, #5993FF)';
/** Пунктирные дивайдеры и неактивные контролы */
export var onDarkSurfaceAdditionalHover = 'var(--on-dark-surface-additional-hover, #535A5FFF)';
/** Пунктирные дивайдеры и неактивные контролы */
export var onDarkSurfaceAdditionalActive = 'var(--on-dark-surface-additional-active, #454A4FFF)';
/** Пунктирные дивайдеры и неактивные контролы */
export var onDarkSurfaceAdditional = 'var(--on-dark-surface-additional, #494F54)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondaryHover = 'var(--on-light-surface-solid-secondary-hover, #F9FAFBFF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondaryActive = 'var(--on-light-surface-solid-secondary-active, #ECF0F3FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #F2F5F7)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidTertiaryHover = 'var(--on-light-surface-solid-tertiary-hover, #F2F5F8FF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidTertiaryActive = 'var(--on-light-surface-solid-tertiary-active, #DEE7EDFF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #E4EBF0)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCardHover = 'var(--on-light-surface-solid-card-hover, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCardActive = 'var(--on-light-surface-solid-card-active, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentHover = 'var(--on-light-surface-accent-hover, #1F78FFFF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentActive = 'var(--on-light-surface-accent-active, #0062F5FF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccent = 'var(--on-light-surface-accent, #0064FE)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradientHover = 'var(--on-light-surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradientActive = 'var(--on-light-surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradient = 'var(--on-light-surface-accent-gradient, linear-gradient(122.62deg, #4817C1 0%, #C505D6 100%))';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeHover = 'var(--on-light-surface-negative-hover, #F5384BFF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeActive = 'var(--on-light-surface-negative-active, #F31228FF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegative = 'var(--on-light-surface-negative, #F31B31)';
/** Акцентный фон поверхности/контрола */
export var onLightSurfaceAccentB2BHover = 'var(--on-light-surface-accent-b2-b-hover, #0EBBBEFF)';
/** Акцентный фон поверхности/контрола */
export var onLightSurfaceAccentB2BActive = 'var(--on-light-surface-accent-b2-b-active, #0C9FA1FF)';
/** Акцентный фон поверхности/контрола */
export var onLightSurfaceAccentB2B = 'var(--on-light-surface-accent-b2-b, #0DA8AB)';
/** Дополнительный акцент для обозначения акций */
export var onLightSurfaceAccentDiscountHover = 'var(--on-light-surface-accent-discount-hover, #FF3D8BFF)';
/** Дополнительный акцент для обозначения акций */
export var onLightSurfaceAccentDiscountActive = 'var(--on-light-surface-accent-discount-active, #FF1472FF)';
/** Дополнительный акцент для обозначения акций */
export var onLightSurfaceAccentDiscount = 'var(--on-light-surface-accent-discount, #FF1F78)';
/** Дополнительный акцент */
export var onLightSurfaceAccentGoldHover = 'var(--on-light-surface-accent-gold-hover, #FFDE5CFF)';
/** Дополнительный акцент */
export var onLightSurfaceAccentGoldActive = 'var(--on-light-surface-accent-gold-active, #FFD633FF)';
/** Дополнительный акцент */
export var onLightSurfaceAccentGold = 'var(--on-light-surface-accent-gold, #FFD83D)';
/** Дополнительный акцент */
export var onLightSurfaceAccentBlueHover = 'var(--on-light-surface-accent-blue-hover, #5D95FDFF)';
/** Дополнительный акцент */
export var onLightSurfaceAccentBlueActive = 'var(--on-light-surface-accent-blue-active, #357BFDFF)';
/** Дополнительный акцент */
export var onLightSurfaceAccentBlue = 'var(--on-light-surface-accent-blue, #3F81FD)';
/** Пунктирные дивайдеры и неактивные контролы */
export var onLightSurfaceAdditionalHover = 'var(--on-light-surface-additional-hover, #DEE7EDFF)';
/** Пунктирные дивайдеры и неактивные контролы */
export var onLightSurfaceAdditionalActive = 'var(--on-light-surface-additional-active, #CBD8E2FF)';
/** Пунктирные дивайдеры и неактивные контролы */
export var onLightSurfaceAdditional = 'var(--on-light-surface-additional, #D1DDE5)';
/** Основной фон */
export var backgroundPrimary = 'var(--background-primary, #08090A)';
/** Вторичный фон */
export var backgroundOverlay = 'var(--background-overlay, rgba(21, 24, 26, 0.56))';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerRed = 'var(--banners-default-banner-red, #2E090D)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerGold = 'var(--banners-default-banner-gold, #1F1903)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerGreen = 'var(--banners-default-banner-green, #071F0C)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerMalachite = 'var(--banners-default-banner-malachite, #0A1C1C)';
/** Бэкграунд для баннеров и всплывающих нотификаций */
export var bannersDefaultBannerBlue = 'var(--banners-default-banner-blue, #0A162B)';
export var skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
export var skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
export var whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
export var whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
export var whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
export var blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
export var blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
export var blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use textPrimary */
export var text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
export var primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
export var secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
export var tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use textParagraph */
export var paragraph = 'var(--plasma-colors-paragraph, var(--text-paragraph))';
/** @deprecated instead use backgroundPrimary */
export var background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textAccent */
export var accent = 'var(--plasma-colors-accent, var(--text-accent))';
/** @deprecated instead use textNegative */
export var critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use surfaceSolidSecondary */
export var surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
export var surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use textAccent */
export var buttonAccent = 'var(--plasma-colors-button-accent, var(--text-accent))';
/** @deprecated instead use surfaceNegative */
export var buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
export var shadowDownSoftS = 'var(--shadow-down-soft-s, 0px 6px 12px rgba(0, 0, 0, 0.08))';
export var shadowDownSoftM = 'var(--shadow-down-soft-m, 0 12px 24px rgba(0, 0, 0, 0.12))';
export var shadowDownSoftL = 'var(--shadow-down-soft-l, 0 24px 48px rgba(0, 0, 0, 0.16))';
export var shadowDownHardS = 'var(--shadow-down-hard-s, 0px 6px 12px rgba(0, 0, 0, 0.12))';
export var shadowDownHardM = 'var(--shadow-down-hard-m, 0 12px 24px rgba(0, 0, 0, 0.16))';
export var shadowDownHardL = 'var(--shadow-down-hard-l, 0 24px 48px rgba(0, 0, 0, 0.2))';
export var borderRadiusXxs = 'var(--border-radius-xxs, 4px)';
export var borderRadiusXs = 'var(--border-radius-xs, 8px)';
export var borderRadiusS = 'var(--border-radius-s, 12px)';
export var borderRadiusM = 'var(--border-radius-m, 16px)';
export var borderRadiusL = 'var(--border-radius-l, 20px)';
export var borderRadiusXl = 'var(--border-radius-xl, 24px)';
export var borderRadiusXxl = 'var(--border-radius-xxl, 32px)';
export var spacing0x = 'var(--spacing0x, 0px)';
export var spacing1x = 'var(--spacing1x, 2px)';
export var spacing2x = 'var(--spacing2x, 4px)';
export var spacing3x = 'var(--spacing3x, 6px)';
export var spacing4x = 'var(--spacing4x, 8px)';
export var spacing6x = 'var(--spacing6x, 12px)';
export var spacing8x = 'var(--spacing8x, 16px)';
export var spacing10x = 'var(--spacing10x, 20px)';
export var spacing12x = 'var(--spacing12x, 24px)';
export var spacing16x = 'var(--spacing16x, 32px)';
export var spacing20x = 'var(--spacing20x, 40px)';
export var spacing24x = 'var(--spacing24x, 48px)';
export var spacing32x = 'var(--spacing32x, 64px)';
export var spacing40x = 'var(--spacing40x, 80px)';
export var spacing60x = 'var(--spacing60x, 120px)';
