// @salutejs/plasma-hope/es/components/AudioPlayer/AudioPlayer.js
var _IconPause, _IconPlay, _IconTrashFilled;
var _excluded = ["title", "duration", "url", "isPlaying", "isSelected", "canDelete", "onClick", "onDelete", "onPlay"];
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t["return"] && (u = t["return"](), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React, { useRef } from 'react';
import styled, { css } from 'styled-components';
import { primary, footnote1, surfaceLiquid03, surfaceLiquid02 } from '@salutejs/plasma-core';
import { IconPlay, IconPause, IconTrashFilled } from '@salutejs/plasma-icons';
import { useAudioPlayer, formatSecondsToMintues } from './utils';
import { collectPackageInfo } from '../../collectPackageInfo';
collectPackageInfo('AudioPlayer/AudioPlayer');
var ButtonDelete = /*#__PURE__*/styled.button.withConfig({
  componentId: "plasma-web-core__sc-1cldicm-0"
})(["position:absolute;right:1.25rem;top:1.25rem;display:none;padding-top:0;padding-bottom:0;line-height:0;background-color:transparent;border:0;cursor:pointer;svg{color:", ";}&:hover{svg{color:", ";}}"], surfaceLiquid03, primary);
export var StyledControl = /*#__PURE__*/styled.button.withConfig({
  componentId: "plasma-web-core__sc-1cldicm-1"
})(["background:transparent;border:0;cursor:pointer;display:flex;"]);
var StyledContainer = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-web-core__sc-1cldicm-2"
})(["position:relative;display:flex;align-items:center;box-sizing:border-box;width:100%;height:4rem;padding:1.25rem 1rem;appearance:none;border-radius:0.75rem;cursor:pointer;", " ", "{", "}"], function (_ref) {
  var isSelected = _ref.isSelected;
  return isSelected && css(["background-color:", ";"], surfaceLiquid03);
}, StyledControl, function (_ref2) {
  var isDisabled = _ref2.isDisabled;
  return isDisabled && css(["opacity:0.24;pointer-events:none;"]);
});
var StyledDuration = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-web-core__sc-1cldicm-3"
})(["position:absolute;right:1rem;text-align:right;color:", ";mix-blend-mode:normal;opacity:0.5;"], primary);
var StyledRoot = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-web-core__sc-1cldicm-4"
})(["", ";position:relative;width:100%;&:hover ", "{display:block;}&:hover ", "{display:none;}&:hover ", "{background-color:", ";}"], footnote1, ButtonDelete, StyledDuration, StyledContainer, surfaceLiquid02);
var StyledTitle = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-web-core__sc-1cldicm-5"
})(["margin-left:1rem;color:", ";"], primary);
/**
 * Компонент для воспроизведения аудио.
 */
export var AudioPlayer = function AudioPlayer(_ref3) {
  var title = _ref3.title,
    duration = _ref3.duration,
    url = _ref3.url,
    isPlaying = _ref3.isPlaying,
    isSelected = _ref3.isSelected,
    canDelete = _ref3.canDelete,
    onClick = _ref3.onClick,
    onDelete = _ref3.onDelete,
    onPlay = _ref3.onPlay,
    props = _objectWithoutProperties(_ref3, _excluded);
  var ref = useRef();
  var _useAudioPlayer = useAudioPlayer(url, isPlaying, ref),
    _useAudioPlayer2 = _slicedToArray(_useAudioPlayer, 1),
    canPlaying = _useAudioPlayer2[0];
  return /*#__PURE__*/React.createElement(StyledRoot, props, /*#__PURE__*/React.createElement(StyledContainer, {
    isSelected: isSelected,
    isDisabled: !canPlaying,
    onClick: onClick
  }, /*#__PURE__*/React.createElement(StyledControl, {
    onClick: onPlay
  }, isPlaying ? _IconPause || (_IconPause = /*#__PURE__*/React.createElement(IconPause, null)) : _IconPlay || (_IconPlay = /*#__PURE__*/React.createElement(IconPlay, null))), /*#__PURE__*/React.createElement(StyledTitle, null, title), duration && /*#__PURE__*/React.createElement(StyledDuration, null, formatSecondsToMintues(duration))), canDelete && /*#__PURE__*/React.createElement(ButtonDelete, {
    type: "button",
    onClick: onDelete
  }, _IconTrashFilled || (_IconTrashFilled = /*#__PURE__*/React.createElement(IconTrashFilled, null))));
};
