// @salutejs/plasma-web/components/Button/Button.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = exports.Anchor = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _plasmaSbUtils = /*#__PURE__*/require("@salutejs/plasma-sb-utils");
var _ = /*#__PURE__*/require(".");
var _IconPlaceholder, _IconPlaceholder2, _IconPlaceholder3, _IconPlaceholder4;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var defaultProps = {
  view: 'default',
  size: 'l',
  pin: 'square-square',
  disabled: false,
  outlined: true,
  focused: false,
  contentType: 'Text',
  isLoading: false,
  onClick: function onClick() {},
  onFocus: function onFocus() {},
  onBlur: function onBlur() {}
};
var text = 'Label';
var Default = exports.Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(_.Button, _extends({
    text: text,
    contentLeft: _IconPlaceholder || (_IconPlaceholder = /*#__PURE__*/_react["default"].createElement(_plasmaSbUtils.IconPlaceholder, null)),
    contentRight: _IconPlaceholder2 || (_IconPlaceholder2 = /*#__PURE__*/_react["default"].createElement(_plasmaSbUtils.IconPlaceholder, null))
  }, defaultProps));
};
var Anchor = exports.Anchor = function Anchor() {
  return /*#__PURE__*/_react["default"].createElement(_.Button, _extends({
    as: "a",
    text: text,
    contentLeft: _IconPlaceholder3 || (_IconPlaceholder3 = /*#__PURE__*/_react["default"].createElement(_plasmaSbUtils.IconPlaceholder, null)),
    contentRight: _IconPlaceholder4 || (_IconPlaceholder4 = /*#__PURE__*/_react["default"].createElement(_plasmaSbUtils.IconPlaceholder, null))
  }, defaultProps));
};
