// @salutejs/plasma-new-hope/cjs/components/Pagination/ui/PaginationSelectPerPage/PaginationSelectPerPage.styles.js
import './PaginationSelectPerPage.styles_rtehle.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
var Select$1 = require('../../../Select/Select.js');
require('react');
require('../../../../utils/react.js');
require('@salutejs/plasma-core');
require('@linaria/core');
var common = require('../../../../engines/common.js');
require('react-dom');
require('react-popper');
require('focus-visible');
require('styled-components');
var SelectItem$1 = require('../../../Select/ui/SelectItem/SelectItem.js');

var mergedSelectConfig = /*#__PURE__*/common.mergeConfig(Select$1.selectConfig);
var Select = /*#__PURE__*/common.component(mergedSelectConfig);
var mergedItemConfig = /*#__PURE__*/common.mergeConfig(SelectItem$1.selectItemConfig);
var SelectItem = /*#__PURE__*/common.component(mergedItemConfig);
var SelectPerPageRoot = /*#__PURE__*/react.styled('div')({
  name: "SelectPerPageRoot",
  "class": "s1e57mfu",
  propsAsIs: false
});
var _exp = function _exp() {
  return Select;
};
var SelectPerPageSelect = /*#__PURE__*/react.styled(_exp())({
  name: "SelectPerPageSelect",
  "class": "s5dck5y",
  propsAsIs: true
});
var _exp69 = function _exp69() {
  return SelectItem;
};
var SelectPerPageSelectItem = /*#__PURE__*/react.styled(_exp69())({
  name: "SelectPerPageSelectItem",
  "class": "squvg2e",
  propsAsIs: true
});
var SelectPerPageTypography = /*#__PURE__*/react.styled('div')({
  name: "SelectPerPageTypography",
  "class": "sbiht1s",
  propsAsIs: false
});

exports.SelectItem = SelectItem;
exports.SelectPerPageRoot = SelectPerPageRoot;
exports.SelectPerPageSelect = SelectPerPageSelect;
exports.SelectPerPageSelectItem = SelectPerPageSelectItem;
exports.SelectPerPageTypography = SelectPerPageTypography;
