// @salutejs/plasma-new-hope/es/components/Pagination/ui/PaginationSelectPerPage/PaginationSelectPerPage.styles.js
import './PaginationSelectPerPage.styles_rtehle.css';
import { styled } from '@linaria/react';
import { selectConfig } from '../../../Select/Select.js';
import 'react';
import '../../../../utils/react.js';
import '@salutejs/plasma-core';
import '@linaria/core';
import { component, mergeConfig } from '../../../../engines/common.js';
import 'react-dom';
import 'react-popper';
import 'focus-visible';
import 'styled-components';
import { selectItemConfig } from '../../../Select/ui/SelectItem/SelectItem.js';

var mergedSelectConfig = /*#__PURE__*/mergeConfig(selectConfig);
var Select = /*#__PURE__*/component(mergedSelectConfig);
var mergedItemConfig = /*#__PURE__*/mergeConfig(selectItemConfig);
var SelectItem = /*#__PURE__*/component(mergedItemConfig);
var SelectPerPageRoot = /*#__PURE__*/styled('div')({
  name: "SelectPerPageRoot",
  "class": "s1e57mfu",
  propsAsIs: false
});
var _exp = function _exp() {
  return Select;
};
var SelectPerPageSelect = /*#__PURE__*/styled(_exp())({
  name: "SelectPerPageSelect",
  "class": "s5dck5y",
  propsAsIs: true
});
var _exp69 = function _exp69() {
  return SelectItem;
};
var SelectPerPageSelectItem = /*#__PURE__*/styled(_exp69())({
  name: "SelectPerPageSelectItem",
  "class": "squvg2e",
  propsAsIs: true
});
var SelectPerPageTypography = /*#__PURE__*/styled('div')({
  name: "SelectPerPageTypography",
  "class": "sbiht1s",
  propsAsIs: false
});

export { SelectItem, SelectPerPageRoot, SelectPerPageSelect, SelectPerPageSelectItem, SelectPerPageTypography };
