// @salutejs/plasma-new-hope/es/components/Pagination/ui/PaginationSelectPerPage/PaginationSelectPerPage.js
import { objectWithoutProperties as _objectWithoutProperties, slicedToArray as _slicedToArray } from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { useState } from 'react';
import '@linaria/react';
import { IconDone } from '../../../_Icon/Icons/IconDone.js';
import { defaultValues } from '../../utils/index.js';
import { classes } from '../../Pagination.tokens.js';
import { SelectPerPageRoot, SelectPerPageTypography, SelectPerPageSelect, SelectPerPageSelectItem } from './PaginationSelectPerPage.styles.js';

var _IconDone;
var _excluded = ["value", "valuesList", "textPerPage", "onChangeValue"];
var PaginationSelectPerPage = function PaginationSelectPerPage(_ref) {
  var _ref$value = _ref.value,
    value = _ref$value === void 0 ? defaultValues.perPage : _ref$value,
    _ref$valuesList = _ref.valuesList,
    valuesList = _ref$valuesList === void 0 ? defaultValues.perPageList : _ref$valuesList,
    _ref$textPerPage = _ref.textPerPage,
    textPerPage = _ref$textPerPage === void 0 ? defaultValues.textPerPage : _ref$textPerPage,
    onChangeValue = _ref.onChangeValue,
    rest = _objectWithoutProperties(_ref, _excluded);
  var _useState = useState(value),
    _useState2 = _slicedToArray(_useState, 2),
    selectedValue = _useState2[0],
    setSelectedValue = _useState2[1];
  var handleSelectChange = function handleSelectChange(newValue) {
    if (newValue) {
      setSelectedValue(Number(newValue));
      onChangeValue === null || onChangeValue === void 0 || onChangeValue(Number(newValue));
    }
  };
  return /*#__PURE__*/React.createElement(SelectPerPageRoot, rest, /*#__PURE__*/React.createElement(SelectPerPageTypography, null, textPerPage), /*#__PURE__*/React.createElement(SelectPerPageSelect, {
    className: classes.selectWrapper,
    value: selectedValue,
    onChangeValue: handleSelectChange
  }, valuesList === null || valuesList === void 0 ? void 0 : valuesList.map(function (item) {
    return /*#__PURE__*/React.createElement(SelectPerPageSelectItem, {
      contentLeft: item === selectedValue ? _IconDone || (_IconDone = /*#__PURE__*/React.createElement(IconDone, {
        size: "xs",
        color: "inherit"
      })) : undefined,
      key: item,
      value: item,
      text: String(item)
    });
  })));
};

export { PaginationSelectPerPage };
