// @salutejs/plasma-hope/es/components/PreviewGallery/PreviewGalleryItemError.js
var _StyledErrorImage, _StyledIconDrag;
import React, { memo } from 'react';
import styled from 'styled-components';
import { surfaceSolid02, backgroundPrimary } from '@salutejs/plasma-core';
import { IconPictureFilled } from '@salutejs/plasma-icons';
import { StyledIconDrag, StyledItem, StyledTrashButton } from './PreviewGalleryItemBase';
var StyledErrorImage = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-web-core__sc-qjhbf7-0"
})(["position:relative;border-radius:0.75rem;padding-bottom:56.25%;background-color:", ";color:", ";"], surfaceSolid02, backgroundPrimary);
var StyledIconPictureFilled = /*#__PURE__*/styled(IconPictureFilled).withConfig({
  componentId: "plasma-web-core__sc-qjhbf7-1"
})(["position:absolute;top:50%;left:50%;margin-right:-50%;transform:translate(-50%,-50%);"]);

/**
 * Базовый компонент с ошибкой.
 */
export var PreviewGalleryItemError = /*#__PURE__*/memo(function (_ref) {
  var id = _ref.id,
    itemSize = _ref.itemSize,
    interactionType = _ref.interactionType,
    actionIcon = _ref.actionIcon,
    onItemAction = _ref.onItemAction;
  var itemRemoveHandle = function itemRemoveHandle(event) {
    event.stopPropagation();
    onItemAction === null || onItemAction === void 0 || onItemAction(id);
  };
  return /*#__PURE__*/React.createElement(StyledItem, {
    width: itemSize,
    key: id
  }, _StyledErrorImage || (_StyledErrorImage = /*#__PURE__*/React.createElement(StyledErrorImage, null, /*#__PURE__*/React.createElement(StyledIconPictureFilled, {
    size: "s",
    color: "inherit"
  }))), /*#__PURE__*/React.createElement(StyledTrashButton, {
    onClick: itemRemoveHandle
  }, actionIcon), interactionType !== 'selectable' && (_StyledIconDrag || (_StyledIconDrag = /*#__PURE__*/React.createElement(StyledIconDrag, {
    size: "xs",
    color: "inherit"
  }))));
});
