// @salutejs/plasma-new-hope/styled-components/cjs/components/Toast/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ToastControllerHoc", {
  enumerable: true,
  get: function get() {
    return _ToastController.ToastControllerHoc;
  }
});
Object.defineProperty(exports, "ToastProviderHoc", {
  enumerable: true,
  get: function get() {
    return _ToastProvider.ToastProviderHoc;
  }
});
Object.defineProperty(exports, "toastClasses", {
  enumerable: true,
  get: function get() {
    return _Toast2.classes;
  }
});
Object.defineProperty(exports, "toastConfig", {
  enumerable: true,
  get: function get() {
    return _Toast.toastConfig;
  }
});
Object.defineProperty(exports, "toastRoot", {
  enumerable: true,
  get: function get() {
    return _Toast.toastRoot;
  }
});
Object.defineProperty(exports, "toastTokens", {
  enumerable: true,
  get: function get() {
    return _Toast2.tokens;
  }
});
Object.defineProperty(exports, "useToast", {
  enumerable: true,
  get: function get() {
    return _ToastProvider.useToast;
  }
});
var _Toast = /*#__PURE__*/require("./Toast");
var _ToastController = /*#__PURE__*/require("./ToastController");
var _ToastProvider = /*#__PURE__*/require("./ToastProvider");
var _Toast2 = /*#__PURE__*/require("./Toast.tokens");
