// @salutejs/plasma-new-hope/styled-components/cjs/components/Toast/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.gradients = exports.TOAST_Z_INDEX = exports.TIMER_DELAY = exports.FADE_Z_INDEX = exports.DEFAULT_POSITION = exports.DEFAULT_FADE = exports.Button = void 0;
var _Button = /*#__PURE__*/require("../Button");
var _engines = /*#__PURE__*/require("../../engines");
var DEFAULT_FADE = exports.DEFAULT_FADE = true;
var DEFAULT_POSITION = exports.DEFAULT_POSITION = 'bottom';
var TIMER_DELAY = exports.TIMER_DELAY = 300;
var FADE_Z_INDEX = exports.FADE_Z_INDEX = '9900';
var TOAST_Z_INDEX = exports.TOAST_Z_INDEX = '9901';
var gradients = exports.gradients = {
  top: 'radial-gradient(200% 200% at 50% -100%, rgba(0, 0, 0, 0.8) 0%, rgba(0, 0, 0, 0) 100%)',
  bottom: 'radial-gradient(200% 200% at 50% 200%, rgba(0, 0, 0, 0.8) 0%, rgba(0, 0, 0, 0) 100%)'
};
var mergedButtonConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Button.buttonConfig);
var Button = exports.Button = /*#__PURE__*/(0, _engines.component)(mergedButtonConfig);
