// @salutejs/plasma-new-hope/cjs/components/Slider/components/Single/Single.styles.js
import './Single.styles_1r9f1e0.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');

var LabelWrapper = /*#__PURE__*/react.styled('div')({
  name: "LabelWrapper",
  "class": "l1p4kc08",
  propsAsIs: false
});
var LabelContentLeft = /*#__PURE__*/react.styled('div')({
  name: "LabelContentLeft",
  "class": "ly2gexi",
  propsAsIs: false
});
var Label = /*#__PURE__*/react.styled('label')({
  name: "Label",
  "class": "lozb535",
  propsAsIs: false
});
var StyledRangeValue = /*#__PURE__*/react.styled('span')({
  name: "StyledRangeValue",
  "class": "s1w9icbd",
  propsAsIs: false
});
var SliderBaseWrapper = /*#__PURE__*/react.styled('div')({
  name: "SliderBaseWrapper",
  "class": "s1d2sfgl",
  propsAsIs: false
});
var SingleWrapper = /*#__PURE__*/react.styled('div')({
  name: "SingleWrapper",
  "class": "ss3ujqx",
  propsAsIs: false
});

exports.Label = Label;
exports.LabelContentLeft = LabelContentLeft;
exports.LabelWrapper = LabelWrapper;
exports.SingleWrapper = SingleWrapper;
exports.SliderBaseWrapper = SliderBaseWrapper;
exports.StyledRangeValue = StyledRangeValue;
