// @salutejs/plasma-ui/components/TextBox/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "TextBox", {
  enumerable: true,
  get: function get() {
    return _TextBox.TextBox;
  }
});
Object.defineProperty(exports, "TextBoxBigTitle", {
  enumerable: true,
  get: function get() {
    return _TextBox.BigTitle;
  }
});
Object.defineProperty(exports, "TextBoxBiggerTitle", {
  enumerable: true,
  get: function get() {
    return _TextBox.BiggerTitle;
  }
});
Object.defineProperty(exports, "TextBoxCaption", {
  enumerable: true,
  get: function get() {
    return _TextBox.Caption;
  }
});
Object.defineProperty(exports, "TextBoxLabel", {
  enumerable: true,
  get: function get() {
    return _TextBox.Label;
  }
});
Object.defineProperty(exports, "TextBoxRoot", {
  enumerable: true,
  get: function get() {
    return _TextBox.TextBoxRoot;
  }
});
Object.defineProperty(exports, "TextBoxSubTitle", {
  enumerable: true,
  get: function get() {
    return _TextBox.SubTitle;
  }
});
Object.defineProperty(exports, "TextBoxTitle", {
  enumerable: true,
  get: function get() {
    return _TextBox.Title;
  }
});
var _TextBox = /*#__PURE__*/require("./TextBox");
