// @salutejs/plasma-ui/components/TextBox/TextBox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Title = exports.TextBoxRoot = exports.TextBox = exports.SubTitle = exports.Label = exports.Caption = exports.BiggerTitle = exports.BigTitle = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _plasmaTokens = /*#__PURE__*/require("@salutejs/plasma-tokens");
var _Typography = /*#__PURE__*/require("../Typography");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _excluded = ["label", "title", "subTitle", "caption", "size", "children"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('TextBox/TextBox');
var Title = exports.Title = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Body1).withConfig({
  componentId: "plasma-ui__sc-19dk7hy-0"
})(["margin:0.125rem 0;color:", ";"], _plasmaTokens.primary);
var BigTitle = exports.BigTitle = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Headline3).withConfig({
  componentId: "plasma-ui__sc-19dk7hy-1"
})(["margin:0.5rem 0;color:", ";"], _plasmaTokens.primary);
var BiggerTitle = exports.BiggerTitle = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Headline1).withConfig({
  componentId: "plasma-ui__sc-19dk7hy-2"
})(["margin:0.5rem 0;color:", ";"], _plasmaTokens.primary);
var Label = exports.Label = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-19dk7hy-3"
})(["margin:0.375rem 0;color:", ";"], function (_ref) {
  var color = _ref.color;
  return color || _plasmaTokens.secondary;
});
var SubTitle = exports.SubTitle = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Footnote1).withConfig({
  componentId: "plasma-ui__sc-19dk7hy-4"
})(["margin:0.125rem 0;color:", ";"], function (_ref2) {
  var color = _ref2.color;
  return color || _plasmaTokens.secondary;
});
var Caption = exports.Caption = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-19dk7hy-5"
})(["margin:0.25rem 0;color:", ";"], function (_ref3) {
  var color = _ref3.color;
  return color || _plasmaTokens.secondary;
});
var TextBoxRoot = exports.TextBoxRoot = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-ui__sc-19dk7hy-6"
})(["& >:first-child{margin-top:0;}& >:last-child{margin-bottom:0;}"]);
/**
 * Компонент для отображения текста в скомпанованном блоке.
 */
var TextBox = exports.TextBox = function TextBox(_ref4) {
  var label = _ref4.label,
    title = _ref4.title,
    subTitle = _ref4.subTitle,
    caption = _ref4.caption,
    _ref4$size = _ref4.size,
    size = _ref4$size === void 0 ? 'm' : _ref4$size,
    children = _ref4.children,
    rest = _objectWithoutProperties(_ref4, _excluded);
  if (children) {
    return /*#__PURE__*/_react["default"].createElement(TextBoxRoot, rest, children);
  }
  var T = size === 'm' ? Title : BigTitle;
  return /*#__PURE__*/_react["default"].createElement(TextBoxRoot, rest, label && /*#__PURE__*/_react["default"].createElement(Label, null, label), title && /*#__PURE__*/_react["default"].createElement(T, null, title), subTitle && /*#__PURE__*/_react["default"].createElement(SubTitle, null, subTitle), caption && /*#__PURE__*/_react["default"].createElement(Caption, null, caption));
};
