// @salutejs/plasma-web/components/Tabs/Tabs.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = exports.Arrows = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _plasmaIcons = /*#__PURE__*/require("@salutejs/plasma-icons");
var _2 = /*#__PURE__*/require(".");
var _IconClock;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var defaultProps = {
  itemsNumber: 4,
  disabled: false,
  stretch: true,
  label: 'Label'
};
var items = [0, 0, 0, 0];
var func = function func() {};
var Default = exports.Default = function Default() {
  var disabled = defaultProps.disabled,
    stretch = defaultProps.stretch,
    label = defaultProps.label;
  return /*#__PURE__*/_react["default"].createElement(_2.Tabs, {
    stretch: stretch,
    disabled: disabled
  }, items.map(function (_, i) {
    return /*#__PURE__*/_react["default"].createElement(_2.TabItem, {
      key: "item:".concat(i),
      isActive: i === 0,
      tabIndex: -1,
      contentLeft: _IconClock || (_IconClock = /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconClock, {
        color: "inherit"
      })),
      onClick: func,
      onFocus: func,
      onBlur: func
    }, label);
  }));
};
var arrowsProps = {
  disabled: false,
  stretch: true
};
var items2 = [{
  label: 'Label',
  contentLeft: /*#__PURE__*/_react["default"].createElement(_plasmaIcons.IconClock, {
    color: "inherit"
  })
}];
var Arrows = exports.Arrows = function Arrows() {
  var disabled = arrowsProps.disabled,
    stretch = arrowsProps.stretch;
  return /*#__PURE__*/_react["default"].createElement(_2.TabsController, {
    items: items2,
    index: 0,
    onIndexChange: func,
    stretch: stretch,
    disabled: disabled
  });
};
