// @salutejs/plasma-new-hope/styled-components/cjs/examples/plasma_web/components/Header/Header.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _Header = /*#__PURE__*/require("../../../../components/Header");
var config = exports.config = {
  defaults: {
    view: 'default',
    size: 'm'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--text-primary);", ":var(--surface-clear);", ":transform 0.15s ease-in-out;", ":var(--plasma-colors-primary);", ":var(--plasma-colors-primary);", ":var(--plasma-colors-primary);", ":var(--plasma-colors-secondary);", ":var(--plasma-colors-secondary);", ":var(--plasma-colors-secondary);"], _Header.headerTokens.headerArrowButtonColor, _Header.headerTokens.headerArrowButtonBackgroundColor, _Header.headerTokens.headerArrowTransition, _Header.headerTokens.headerTitleColor, _Header.headerTokens.headerBigTitleColor, _Header.headerTokens.headerBiggerTitleColor, _Header.headerTokens.headerLabelColor, _Header.headerTokens.headerSubTitleColor, _Header.headerTokens.headerCaptionColor)
    },
    size: {
      m: /*#__PURE__*/(0, _styledComponents.css)(["", ":2.25rem;", ":1.625rem;", ":0.625rem;", ":0.75rem;", ":1rem;", ":1.5rem;", ":0.125rem 0;", ":0.5rem 0;", ":0.5rem 0;", ":0.375rem 0;", ":0.125rem 0;", ":0.25rem 0;", ":2.25rem;", ":0.5rem;"], _Header.headerTokens.headerHeight, _Header.headerTokens.headerPaddingTop, _Header.headerTokens.headerPaddingBottom, _Header.headerTokens.headerLogoMarginRight, _Header.headerTokens.headerArrowMarginRight, _Header.headerTokens.headerArrowIconSize, _Header.headerTokens.headerTitleMargin, _Header.headerTokens.headerBigTitleMargin, _Header.headerTokens.headerBiggerTitleMargin, _Header.headerTokens.headerLabelMargin, _Header.headerTokens.headerSubTitleMargin, _Header.headerTokens.headerCaptionMargin, _Header.headerTokens.headerArrowHeight, _Header.headerTokens.headerArrowPadding)
    }
  }
};
