// @salutejs/plasma-tokens/es/brands/sberonline/index.js
// Generated by robots, do not change this manually!
/** Основной цвет текста */
export var textPrimaryHover = 'var(--text-primary-hover, #F5F5F593)';
/** Основной цвет текста */
export var textPrimaryActive = 'var(--text-primary-active, #F5F5F5C4)';
/** Основной цвет текста */
export var textPrimary = 'var(--text-primary, #f5f5f5F5)';
/** Вторичный цвет текста */
export var textSecondaryHover = 'var(--text-secondary-hover, #F5F5F5FF)';
/** Вторичный цвет текста */
export var textSecondaryActive = 'var(--text-secondary-active, #F5F5F5AB)';
/** Вторичный цвет текста */
export var textSecondary = 'var(--text-secondary, #F5f5f58F)';
/** Третичный цвет текста */
export var textTertiaryHover = 'var(--text-tertiary-hover, #F5F5F5FF)';
/** Третичный цвет текста */
export var textTertiaryActive = 'var(--text-tertiary-active, #F5F5F556)';
/** Третичный цвет текста */
export var textTertiary = 'var(--text-tertiary, #F5f5f548)';
/** Сплошной наборный текст */
export var textParagraphHover = 'var(--text-paragraph-hover, #F5F5F57A)';
/** Сплошной наборный текст */
export var textParagraphActive = 'var(--text-paragraph-active, #F5F5F5A3)';
/** Сплошной наборный текст */
export var textParagraph = 'var(--text-paragraph, #f5f5f5cc)';
/** Акцентный цвет */
export var textAccentHover = 'var(--text-accent-hover, #2ACB47FF)';
/** Акцентный цвет */
export var textAccentActive = 'var(--text-accent-active, #1F9835FF)';
/** Акцентный цвет */
export var textAccent = 'var(--text-accent, #24B23E)';
/** Акцентный цвет с градиентом */
export var textAccentGradientHover = 'var(--text-accent-gradient-hover, #CCCCCCFF)';
/** Акцентный цвет с градиентом */
export var textAccentGradientActive = 'var(--text-accent-gradient-active, #E6E6E6FF)';
/** Акцентный цвет с градиентом */
export var textAccentGradient = 'var(--text-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет успеха */
export var textPositiveHover = 'var(--text-positive-hover, #2ACB47FF)';
/** Цвет успеха */
export var textPositiveActive = 'var(--text-positive-active, #1F9835FF)';
/** Цвет успеха */
export var textPositive = 'var(--text-positive, #24B23E)';
/** Цвет предупреждения */
export var textWarningHover = 'var(--text-warning-hover, #FF8442FF)';
/** Цвет предупреждения */
export var textWarningActive = 'var(--text-warning-active, #FF5D05FF)';
/** Цвет предупреждения */
export var textWarning = 'var(--text-warning, #FF7024)';
/** Цвет ошибки */
export var textNegativeHover = 'var(--text-negative-hover, #FF5C6CFF)';
/** Цвет ошибки */
export var textNegativeActive = 'var(--text-negative-active, #FF1F35FF)';
/** Цвет ошибки */
export var textNegative = 'var(--text-negative, #FF3D51)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimaryHover = 'var(--on-dark-text-primary-hover, #F5F5F593)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimaryActive = 'var(--on-dark-text-primary-active, #F5F5F5C4)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimary = 'var(--on-dark-text-primary, #f5f5f5F5)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondaryHover = 'var(--on-dark-text-secondary-hover, #F5F5F5FF)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondaryActive = 'var(--on-dark-text-secondary-active, #F5F5F5AB)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondary = 'var(--on-dark-text-secondary, #F5f5f58F)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiaryHover = 'var(--on-dark-text-tertiary-hover, #F5F5F5FF)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiaryActive = 'var(--on-dark-text-tertiary-active, #F5F5F556)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiary = 'var(--on-dark-text-tertiary, #F5f5f548)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraphHover = 'var(--on-dark-text-paragraph-hover, #F5F5F57A)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraphActive = 'var(--on-dark-text-paragraph-active, #F5F5F5A3)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraph = 'var(--on-dark-text-paragraph, #f5f5f5CC)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccentHover = 'var(--on-dark-text-accent-hover, #2ACB47FF)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccentActive = 'var(--on-dark-text-accent-active, #1F9835FF)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccent = 'var(--on-dark-text-accent, #24B23E)';
/** Акцентный цвет с градиентом на темном фоне */
export var onDarkTextAccentGradientHover = 'var(--on-dark-text-accent-gradient-hover, #CCCCCCFF)';
/** Акцентный цвет с градиентом на темном фоне */
export var onDarkTextAccentGradientActive = 'var(--on-dark-text-accent-gradient-active, #E6E6E6FF)';
/** Акцентный цвет с градиентом на темном фоне */
export var onDarkTextAccentGradient = 'var(--on-dark-text-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет успеха на темном фоне */
export var onDarkTextPositiveHover = 'var(--on-dark-text-positive-hover, #2ACB47FF)';
/** Цвет успеха на темном фоне */
export var onDarkTextPositiveActive = 'var(--on-dark-text-positive-active, #1F9835FF)';
/** Цвет успеха на темном фоне */
export var onDarkTextPositive = 'var(--on-dark-text-positive, #24B23E)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarningHover = 'var(--on-dark-text-warning-hover, #FF8442FF)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarningActive = 'var(--on-dark-text-warning-active, #FF5D05FF)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarning = 'var(--on-dark-text-warning, #FF7024)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegativeHover = 'var(--on-dark-text-negative-hover, #FF5C6CFF)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegativeActive = 'var(--on-dark-text-negative-active, #FF1F35FF)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegative = 'var(--on-dark-text-negative, #FF3D51)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimaryHover = 'var(--on-light-text-primary-hover, #17171793)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimaryActive = 'var(--on-light-text-primary-active, #171717C4)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimary = 'var(--on-light-text-primary, #171717F5)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondaryHover = 'var(--on-light-text-secondary-hover, #171717FF)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondaryActive = 'var(--on-light-text-secondary-active, #171717AB)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondary = 'var(--on-light-text-secondary, #1717178F)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiaryHover = 'var(--on-light-text-tertiary-hover, #171717FF)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiaryActive = 'var(--on-light-text-tertiary-active, #17171756)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiary = 'var(--on-light-text-tertiary, #17171748)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraphHover = 'var(--on-light-text-paragraph-hover, #1717177A)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraphActive = 'var(--on-light-text-paragraph-active, #171717A3)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraph = 'var(--on-light-text-paragraph, #171717CC)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccentHover = 'var(--on-light-text-accent-hover, #13AA2CFF)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccentActive = 'var(--on-light-text-accent-active, #0D731EFF)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccent = 'var(--on-light-text-accent, #108E26)';
/** Акцентный цвет с градиентом на светлом фоне */
export var onLightTextAccentGradientHover = 'var(--on-light-text-accent-gradient-hover, #CCCCCCFF)';
/** Акцентный цвет с градиентом на светлом фоне */
export var onLightTextAccentGradientActive = 'var(--on-light-text-accent-gradient-active, #E6E6E6FF)';
/** Акцентный цвет с градиентом на светлом фоне */
export var onLightTextAccentGradient = 'var(--on-light-text-accent-gradient, linear-gradient(45deg, #0D8523 0%, #0DA8AB 100%))';
/** Цвет успеха на светлом фоне */
export var onLightTextPositiveHover = 'var(--on-light-text-positive-hover, #13AA2CFF)';
/** Цвет успеха на светлом фоне */
export var onLightTextPositiveActive = 'var(--on-light-text-positive-active, #0D731EFF)';
/** Цвет успеха на светлом фоне */
export var onLightTextPositive = 'var(--on-light-text-positive, #108E26)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarningHover = 'var(--on-light-text-warning-hover, #FD650DFF)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarningActive = 'var(--on-light-text-warning-active, #CA4B02FF)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarning = 'var(--on-light-text-warning, #E85702)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegativeHover = 'var(--on-light-text-negative-hover, #F5384BFF)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegativeActive = 'var(--on-light-text-negative-active, #E40C22FF)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegative = 'var(--on-light-text-negative, #F31B31)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimaryHover = 'var(--inverse-text-primary-hover, #17171793)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimaryActive = 'var(--inverse-text-primary-active, #171717C4)';
/** Инвертированный основной цвет текста */
export var inverseTextPrimary = 'var(--inverse-text-primary, #171717F5)';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondaryHover = 'var(--inverse-text-secondary-hover, #171717FF)';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondaryActive = 'var(--inverse-text-secondary-active, #171717AB)';
/** Инвертированный вторичный цвет текста */
export var inverseTextSecondary = 'var(--inverse-text-secondary, #1717178F)';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiaryHover = 'var(--inverse-text-tertiary-hover, #171717FF)';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiaryActive = 'var(--inverse-text-tertiary-active, #17171756)';
/** Инвертированный третичный цвет текста */
export var inverseTextTertiary = 'var(--inverse-text-tertiary, #17171748)';
/** Инвертированный cплошной наборный текст */
export var inverseTextParagraphHover = 'var(--inverse-text-paragraph-hover, #1717177A)';
/** Инвертированный cплошной наборный текст */
export var inverseTextParagraphActive = 'var(--inverse-text-paragraph-active, #171717A3)';
/** Инвертированный cплошной наборный текст */
export var inverseTextParagraph = 'var(--inverse-text-paragraph, #171717CC)';
/** Инвертированный акцентный цвет */
export var inverseTextAccentHover = 'var(--inverse-text-accent-hover, #13AA2CFF)';
/** Инвертированный акцентный цвет */
export var inverseTextAccentActive = 'var(--inverse-text-accent-active, #0D731EFF)';
/** Инвертированный акцентный цвет */
export var inverseTextAccent = 'var(--inverse-text-accent, #108E26)';
/** Инвертированный акцентный цвет с градиентом */
export var inverseTextAccentGradientHover = 'var(--inverse-text-accent-gradient-hover, #CCCCCCFF)';
/** Инвертированный акцентный цвет с градиентом */
export var inverseTextAccentGradientActive = 'var(--inverse-text-accent-gradient-active, #E6E6E6FF)';
/** Инвертированный акцентный цвет с градиентом */
export var inverseTextAccentGradient = 'var(--inverse-text-accent-gradient, linear-gradient(45deg, #0D8523 0%, #0DA8AB 100%))';
/** Инвертированный цвет успеха */
export var inverseTextPositiveHover = 'var(--inverse-text-positive-hover, #13AA2CFF)';
/** Инвертированный цвет успеха */
export var inverseTextPositiveActive = 'var(--inverse-text-positive-active, #0D731EFF)';
/** Инвертированный цвет успеха */
export var inverseTextPositive = 'var(--inverse-text-positive, #108E26)';
/** Инвертированный цвет предупреждения */
export var inverseTextWarningHover = 'var(--inverse-text-warning-hover, #FD650DFF)';
/** Инвертированный цвет предупреждения */
export var inverseTextWarningActive = 'var(--inverse-text-warning-active, #CA4B02FF)';
/** Инвертированный цвет предупреждения */
export var inverseTextWarning = 'var(--inverse-text-warning, #E85702)';
/** Инвертированный цвет ошибки */
export var inverseTextNegativeHover = 'var(--inverse-text-negative-hover, #F5384BFF)';
/** Инвертированный цвет ошибки */
export var inverseTextNegativeActive = 'var(--inverse-text-negative-active, #E40C22FF)';
/** Инвертированный цвет ошибки */
export var inverseTextNegative = 'var(--inverse-text-negative, #F31B31)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimaryHover = 'var(--surface-solid-primary-hover, #1C1C1CFF)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimaryActive = 'var(--surface-solid-primary-active, #121212FF)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimary = 'var(--surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondaryHover = 'var(--surface-solid-secondary-hover, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondaryActive = 'var(--surface-solid-secondary-active, #212121FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondary = 'var(--surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiaryHover = 'var(--surface-solid-tertiary-hover, #404040FF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiaryActive = 'var(--surface-solid-tertiary-active, #303030FF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiary = 'var(--surface-solid-tertiary, #363636)';
/** Основной фон для карточек */
export var surfaceSolidCardHover = 'var(--surface-solid-card-hover, #1C1C1CFF)';
/** Основной фон для карточек */
export var surfaceSolidCardActive = 'var(--surface-solid-card-active, #121212FF)';
/** Основной фон для карточек */
export var surfaceSolidCard = 'var(--surface-solid-card, #171717)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefaultHover = 'var(--surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefaultActive = 'var(--surface-solid-default-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefault = 'var(--surface-solid-default, #FFFFFFFF)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimaryHover = 'var(--surface-transparent-primary-hover, #FFFFFF1F)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimaryActive = 'var(--surface-transparent-primary-active, #FFFFFF0A)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimary = 'var(--surface-transparent-primary, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondaryHover = 'var(--surface-transparent-secondary-hover, #FFFFFF3D)';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondaryActive = 'var(--surface-transparent-secondary-active, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondary = 'var(--surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiaryHover = 'var(--surface-transparent-tertiary-hover, #FFFFFF52)';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiaryActive = 'var(--surface-transparent-tertiary-active, #FFFFFF24)';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон поверхности/контрола по умолчанию */
export var surfaceTransparentDeepHover = 'var(--surface-transparent-deep-hover, #FFFFFFC2)';
/** Прозрачный фон поверхности/контрола по умолчанию */
export var surfaceTransparentDeepActive = 'var(--surface-transparent-deep-active, #FFFFFF94)';
/** Прозрачный фон поверхности/контрола по умолчанию */
export var surfaceTransparentDeep = 'var(--surface-transparent-deep, #FFFFFFA3)';
/** Прозрачный фон для карточек */
export var surfaceTransparentCardHover = 'var(--surface-transparent-card-hover, #FFFFFF1F)';
/** Прозрачный фон для карточек */
export var surfaceTransparentCardActive = 'var(--surface-transparent-card-active, #FFFFFF0A)';
/** Прозрачный фон для карточек */
export var surfaceTransparentCard = 'var(--surface-transparent-card, #FFFFFF0F)';
/** Фон поверхности/контрола без заливки */
export var surfaceClearHover = 'var(--surface-clear-hover, #00000000)';
/** Фон поверхности/контрола без заливки */
export var surfaceClearActive = 'var(--surface-clear-active, #00000000)';
/** Фон поверхности/контрола без заливки */
export var surfaceClear = 'var(--surface-clear, rgba(0, 0, 0, 0))';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentHover = 'var(--surface-accent-hover, #1DAF37FF)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentActive = 'var(--surface-accent-active, #18952FFF)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccent = 'var(--surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradientHover = 'var(--surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradientActive = 'var(--surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет фона поверхности/контрола успех */
export var surfacePositiveHover = 'var(--surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositiveActive = 'var(--surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositive = 'var(--surface-positive, #1A9E32)';
/** Цвет фона поверхности предупреждение */
export var surfaceWarningHover = 'var(--surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности предупреждение */
export var surfaceWarningActive = 'var(--surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности предупреждение */
export var surfaceWarning = 'var(--surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegativeHover = 'var(--surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegativeActive = 'var(--surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegative = 'var(--surface-negative, #FF293E)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimaryHover = 'var(--on-dark-surface-solid-primary-hover, #1C1C1CFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimaryActive = 'var(--on-dark-surface-solid-primary-active, #121212FF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #171717)';
/** Вторичный непрозрачный фон поверхности на темном фоне */
export var onDarkSurfaceSolidSecondaryHover = 'var(--on-dark-surface-solid-secondary-hover, #303030FF)';
/** Вторичный непрозрачный фон поверхности на темном фоне */
export var onDarkSurfaceSolidSecondaryActive = 'var(--on-dark-surface-solid-secondary-active, #212121FF)';
/** Вторичный непрозрачный фон поверхности на темном фоне */
export var onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #262626)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiaryHover = 'var(--on-dark-surface-solid-tertiary-hover, #404040FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiaryActive = 'var(--on-dark-surface-solid-tertiary-active, #303030FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #363636)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCardHover = 'var(--on-dark-surface-solid-card-hover, #1C1C1CFF)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCardActive = 'var(--on-dark-surface-solid-card-active, #121212FF)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #171717)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefaultHover = 'var(--on-dark-surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefaultActive = 'var(--on-dark-surface-solid-default-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #FFFFFF)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimaryHover = 'var(--on-dark-surface-transparent-primary-hover, #FFFFFF1F)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimaryActive = 'var(--on-dark-surface-transparent-primary-active, #FFFFFF0A)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondaryHover = 'var(--on-dark-surface-transparent-secondary-hover, #FFFFFF3D)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondaryActive = 'var(--on-dark-surface-transparent-secondary-active, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiaryHover = 'var(--on-dark-surface-transparent-tertiary-hover, #FFFFFF52)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiaryActive = 'var(--on-dark-surface-transparent-tertiary-active, #FFFFFF24)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон поверхности по умолчанию на темном фоне */
export var onDarkSurfaceTransparentDeepHover = 'var(--on-dark-surface-transparent-deep-hover, #FFFFFFC2)';
/** Прозрачный фон поверхности по умолчанию на темном фоне */
export var onDarkSurfaceTransparentDeepActive = 'var(--on-dark-surface-transparent-deep-active, #FFFFFF94)';
/** Прозрачный фон поверхности по умолчанию на темном фоне */
export var onDarkSurfaceTransparentDeep = 'var(--on-dark-surface-transparent-deep, #FFFFFFA3)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCardHover = 'var(--on-dark-surface-transparent-card-hover, #FFFFFF1F)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCardActive = 'var(--on-dark-surface-transparent-card-active, #FFFFFF0A)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, #FFFFFF0F)';
/** Фон поверхности/контрола без заливки на темном фоне */
export var onDarkSurfaceClearHover = 'var(--on-dark-surface-clear-hover, #00000000)';
/** Фон поверхности/контрола без заливки на темном фоне */
export var onDarkSurfaceClearActive = 'var(--on-dark-surface-clear-active, #00000000)';
/** Фон поверхности/контрола без заливки на темном фоне */
export var onDarkSurfaceClear = 'var(--on-dark-surface-clear, rgba(0, 0, 0, 0))';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentHover = 'var(--on-dark-surface-accent-hover, #1DAF37FF)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentActive = 'var(--on-dark-surface-accent-active, #18952FFF)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradientHover = 'var(--on-dark-surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradientActive = 'var(--on-dark-surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradient = 'var(--on-dark-surface-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositiveHover = 'var(--on-dark-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositiveActive = 'var(--on-dark-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositive = 'var(--on-dark-surface-positive, #1A9E32)';
/** Цвет фона поверхности предупреждение на темном фоне */
export var onDarkSurfaceWarningHover = 'var(--on-dark-surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности предупреждение на темном фоне */
export var onDarkSurfaceWarningActive = 'var(--on-dark-surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности предупреждение на темном фоне */
export var onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegativeHover = 'var(--on-dark-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegativeActive = 'var(--on-dark-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293E)';
/** Основной непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidPrimaryHover = 'var(--on-light-surface-solid-primary-hover, #FAFAFAFF)';
/** Основной непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidPrimaryActive = 'var(--on-light-surface-solid-primary-active, #F0F0F0FF)';
/** Основной непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F5F5F5)';
/** Вторичный непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidSecondaryHover = 'var(--on-light-surface-solid-secondary-hover, #F7F7F7FF)';
/** Вторичный непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidSecondaryActive = 'var(--on-light-surface-solid-secondary-active, #E8E8E8FF)';
/** Вторичный непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #ECECEC)';
/** Третичный непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidTertiaryHover = 'var(--on-light-surface-solid-tertiary-hover, #E8E8E8FF)';
/** Третичный непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidTertiaryActive = 'var(--on-light-surface-solid-tertiary-active, #D9D9D9FF)';
/** Третичный непрозрачный фон поверхности на светлом фоне */
export var onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #DDDDDD)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCardHover = 'var(--on-light-surface-solid-card-hover, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCardActive = 'var(--on-light-surface-solid-card-active, #FFFFFFFF)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefaultHover = 'var(--on-light-surface-solid-default-hover, #0D0D0DFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefaultActive = 'var(--on-light-surface-solid-default-active, #030303FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #080808)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentPrimaryHover = 'var(--on-light-surface-transparent-primary-hover, #0808080F)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentPrimaryActive = 'var(--on-light-surface-transparent-primary-active, #08080800)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, rgba(8, 8, 8, 0.02))';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondaryHover = 'var(--on-light-surface-transparent-secondary-hover, #0808081F)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondaryActive = 'var(--on-light-surface-transparent-secondary-active, #0808080A)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiaryHover = 'var(--on-light-surface-transparent-tertiary-hover, #0808083D)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiaryActive = 'var(--on-light-surface-transparent-tertiary-active, #0808080F)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Прозрачный фон поверхности по умолчанию на светлом фоне */
export var onLightSurfaceTransparentDeepHover = 'var(--on-light-surface-transparent-deep-hover, #080808C2)';
/** Прозрачный фон поверхности по умолчанию на светлом фоне */
export var onLightSurfaceTransparentDeepActive = 'var(--on-light-surface-transparent-deep-active, #08080894)';
/** Прозрачный фон поверхности по умолчанию на светлом фоне */
export var onLightSurfaceTransparentDeep = 'var(--on-light-surface-transparent-deep, #080808A3)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCardHover = 'var(--on-light-surface-transparent-card-hover, #0808081F)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCardActive = 'var(--on-light-surface-transparent-card-active, #0808080A)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, rgba(8, 8, 8, 0.06))';
/** Фон поверхности/контрола без заливки на светлом фоне */
export var onLightSurfaceClearHover = 'var(--on-light-surface-clear-hover, #00000000)';
/** Фон поверхности/контрола без заливки на светлом фоне */
export var onLightSurfaceClearActive = 'var(--on-light-surface-clear-active, #00000000)';
/** Фон поверхности/контрола без заливки на светлом фоне */
export var onLightSurfaceClear = 'var(--on-light-surface-clear, rgba(0, 0, 0, 0))';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentHover = 'var(--on-light-surface-accent-hover, #1DAF37FF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentActive = 'var(--on-light-surface-accent-active, #18952FFF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccent = 'var(--on-light-surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradientHover = 'var(--on-light-surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradientActive = 'var(--on-light-surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradient = 'var(--on-light-surface-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositiveHover = 'var(--on-light-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositiveActive = 'var(--on-light-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositive = 'var(--on-light-surface-positive, #1A9E32)';
/** Цвет фона поверхности / предупреждение на светлом фоне */
export var onLightSurfaceWarningHover = 'var(--on-light-surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности / предупреждение на светлом фоне */
export var onLightSurfaceWarningActive = 'var(--on-light-surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности / предупреждение на светлом фоне */
export var onLightSurfaceWarning = 'var(--on-light-surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeHover = 'var(--on-light-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeActive = 'var(--on-light-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegative = 'var(--on-light-surface-negative, #FF293E)';
/** Инвертированный основной непрозрачный фон поверхности */
export var inverseSurfaceSolidPrimaryHover = 'var(--inverse-surface-solid-primary-hover, #FAFAFAFF)';
/** Инвертированный основной непрозрачный фон поверхности */
export var inverseSurfaceSolidPrimaryActive = 'var(--inverse-surface-solid-primary-active, #F0F0F0FF)';
/** Инвертированный основной непрозрачный фон поверхности */
export var inverseSurfaceSolidPrimary = 'var(--inverse-surface-solid-primary, #F5F5F5)';
/** Инвертированный вторичный непрозрачный фон поверхности */
export var inverseSurfaceSolidSecondaryHover = 'var(--inverse-surface-solid-secondary-hover, #F7F7F7FF)';
/** Инвертированный вторичный непрозрачный фон поверхности */
export var inverseSurfaceSolidSecondaryActive = 'var(--inverse-surface-solid-secondary-active, #E8E8E8FF)';
/** Инвертированный вторичный непрозрачный фон поверхности */
export var inverseSurfaceSolidSecondary = 'var(--inverse-surface-solid-secondary, #ECECEC)';
/** Инвертированный третичный непрозрачный фон поверхности */
export var inverseSurfaceSolidTertiaryHover = 'var(--inverse-surface-solid-tertiary-hover, #E8E8E8FF)';
/** Инвертированный третичный непрозрачный фон поверхности */
export var inverseSurfaceSolidTertiaryActive = 'var(--inverse-surface-solid-tertiary-active, #D9D9D9FF)';
/** Инвертированный третичный непрозрачный фон поверхности */
export var inverseSurfaceSolidTertiary = 'var(--inverse-surface-solid-tertiary, #DDDDDD)';
/** Инвертированный основной фон для карточек */
export var inverseSurfaceSolidCardHover = 'var(--inverse-surface-solid-card-hover, #FFFFFFFF)';
/** Инвертированный основной фон для карточек */
export var inverseSurfaceSolidCardActive = 'var(--inverse-surface-solid-card-active, #FFFFFFFF)';
/** Инвертированный основной фон для карточек */
export var inverseSurfaceSolidCard = 'var(--inverse-surface-solid-card, #FFFFFFFF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
export var inverseSurfaceSolidDefaultHover = 'var(--inverse-surface-solid-default-hover, #0D0D0DFF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
export var inverseSurfaceSolidDefaultActive = 'var(--inverse-surface-solid-default-active, #030303FF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
export var inverseSurfaceSolidDefault = 'var(--inverse-surface-solid-default, #080808)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentPrimaryHover = 'var(--inverse-surface-transparent-primary-hover, #0808080F)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentPrimaryActive = 'var(--inverse-surface-transparent-primary-active, #08080800)';
/** Инвертированный основной прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentPrimary = 'var(--inverse-surface-transparent-primary, rgba(8, 8, 8, 0.02))';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentSecondaryHover = 'var(--inverse-surface-transparent-secondary-hover, #0808081F)';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentSecondaryActive = 'var(--inverse-surface-transparent-secondary-active, #0808080A)';
/** Инвертированный вторичный прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentSecondary = 'var(--inverse-surface-transparent-secondary, rgba(8, 8, 8, 0.06))';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentTertiaryHover = 'var(--inverse-surface-transparent-tertiary-hover, #0808083D)';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentTertiaryActive = 'var(--inverse-surface-transparent-tertiary-active, #0808080F)';
/** Инвертированный третичный прозрачный фон поверхности/контрола */
export var inverseSurfaceTransparentTertiary = 'var(--inverse-surface-transparent-tertiary, rgba(8, 8, 8, 0.12))';
/** Инвертированный прозрачный фон поверхности по умолчанию */
export var inverseSurfaceTransparentDeepHover = 'var(--inverse-surface-transparent-deep-hover, #080808C2)';
/** Инвертированный прозрачный фон поверхности по умолчанию */
export var inverseSurfaceTransparentDeepActive = 'var(--inverse-surface-transparent-deep-active, #08080894)';
/** Инвертированный прозрачный фон поверхности по умолчанию */
export var inverseSurfaceTransparentDeep = 'var(--inverse-surface-transparent-deep, #080808A3)';
/** Инвертированный прозрачный фон для карточек */
export var inverseSurfaceTransparentCardHover = 'var(--inverse-surface-transparent-card-hover, #FFFFFFFF)';
/** Инвертированный прозрачный фон для карточек */
export var inverseSurfaceTransparentCardActive = 'var(--inverse-surface-transparent-card-active, #FFFFFFFF)';
/** Инвертированный прозрачный фон для карточек */
export var inverseSurfaceTransparentCard = 'var(--inverse-surface-transparent-card, #FFFFFFFF)';
/** Инвертированный фон поверхности/контрола без заливки */
export var inverseSurfaceClearHover = 'var(--inverse-surface-clear-hover, #00000000)';
/** Инвертированный фон поверхности/контрола без заливки */
export var inverseSurfaceClearActive = 'var(--inverse-surface-clear-active, #00000000)';
/** Инвертированный фон поверхности/контрола без заливки */
export var inverseSurfaceClear = 'var(--inverse-surface-clear, rgba(0, 0, 0, 0))';
/** Инвертированный акцентный фон поверхности/контрола */
export var inverseSurfaceAccentHover = 'var(--inverse-surface-accent-hover, #1DAF37FF)';
/** Инвертированный акцентный фон поверхности/контрола */
export var inverseSurfaceAccentActive = 'var(--inverse-surface-accent-active, #18952FFF)';
/** Инвертированный акцентный фон поверхности/контрола */
export var inverseSurfaceAccent = 'var(--inverse-surface-accent, #1A9E32)';
/** Инвертированный акцентный фон поверхности/контрола с градиентом */
export var inverseSurfaceAccentGradientHover = 'var(--inverse-surface-accent-gradient-hover, #FFFFFFFF)';
/** Инвертированный акцентный фон поверхности/контрола с градиентом */
export var inverseSurfaceAccentGradientActive = 'var(--inverse-surface-accent-gradient-active, #FFFFFFFF)';
/** Инвертированный акцентный фон поверхности/контрола с градиентом */
export var inverseSurfaceAccentGradient = 'var(--inverse-surface-accent-gradient, linear-gradient(45deg, #1A9E32 0%, #04C6C9 100%))';
/** Инвертированный цвет фона поверхности/контрола успех */
export var inverseSurfacePositiveHover = 'var(--inverse-surface-positive-hover, #1DAF37FF)';
/** Инвертированный цвет фона поверхности/контрола успех */
export var inverseSurfacePositiveActive = 'var(--inverse-surface-positive-active, #18952FFF)';
/** Инвертированный цвет фона поверхности/контрола успех */
export var inverseSurfacePositive = 'var(--inverse-surface-positive, #1A9E32)';
/** Инвертированный цвет фона поверхности / предупреждение */
export var inverseSurfaceWarningHover = 'var(--inverse-surface-warning-hover, #FB7223FF)';
/** Инвертированный цвет фона поверхности / предупреждение */
export var inverseSurfaceWarningActive = 'var(--inverse-surface-warning-active, #F05B05FF)';
/** Инвертированный цвет фона поверхности / предупреждение */
export var inverseSurfaceWarning = 'var(--inverse-surface-warning, #FA5F05)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
export var inverseSurfaceNegativeHover = 'var(--inverse-surface-negative-hover, #FF475AFF)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
export var inverseSurfaceNegativeActive = 'var(--inverse-surface-negative-active, #FF1F35FF)';
/** Инвертированный цвет фона поверхности/контрола ошибка */
export var inverseSurfaceNegative = 'var(--inverse-surface-negative, #FF293E)';
/** Основной фон */
export var backgroundPrimary = 'var(--background-primary, #080808)';
/** Основной фон на темном фоне */
export var darkBackgroundPrimary = 'var(--dark-background-primary, #080808)';
/** Основной фон на светлом фоне */
export var lightBackgroundPrimary = 'var(--light-background-primary, #F9F9F9)';
/** Инвертированный основной фон */
export var inverseBackgroundPrimary = 'var(--inverse-background-primary, #F9F9F9)';
/** Цвет фона паранжи светлый */
export var overlaySoft = 'var(--overlay-soft, #0808088F)';
/** Цвет фона паранжи темный */
export var overlayHard = 'var(--overlay-hard, #080808f5)';
/** Цвет фона паранжи светлый на темном фоне */
export var onDarkOverlaySoft = 'var(--on-dark-overlay-soft, #0808088f)';
/** Цвет фона паранжи темный на темном фоне */
export var onDarkOverlayHard = 'var(--on-dark-overlay-hard, #080808f5)';
/** Цвет фона паранжи светлый на светлом фоне */
export var onLightOverlaySoft = 'var(--on-light-overlay-soft, #f9f9f98f)';
/** Цвет фона паранжи темный на светлом фоне */
export var onLightOverlayHard = 'var(--on-light-overlay-hard, #f9f9f9f5)';
/** Инвертированный цвет фона паранжи светлый */
export var inverseOverlaySoft = 'var(--inverse-overlay-soft, #f9f9f98F)';
/** Инвертированный цвет фона паранжи темный */
export var inverseOverlayHard = 'var(--inverse-overlay-hard, #f9f9f9f5)';
export var skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
export var skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
export var whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
export var whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
export var whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
export var blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
export var blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
export var blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use onLightSurfaceSolidDefault */
export var buttonBlack = 'var(--plasma-colors-button-black, var(--on-light-surface-solid-default))';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
export var buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, var(--on-light-surface-transparent-secondary))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
export var buttonWhite = 'var(--plasma-colors-button-white, var(--on-dark-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
export var buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, var(--on-dark-surface-transparent-secondary))';
/** @deprecated instead use textPrimary */
export var text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
export var primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
export var secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
export var tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use textParagraph */
export var paragraph = 'var(--plasma-colors-paragraph, var(--text-paragraph))';
/** @deprecated instead use backgroundPrimary */
export var background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textAccent */
export var accent = 'var(--plasma-colors-accent, var(--text-accent))';
/** @deprecated instead use textPositive */
export var success = 'var(--plasma-colors-success, var(--text-positive))';
/** @deprecated instead use textWarning */
export var warning = 'var(--plasma-colors-warning, var(--text-warning))';
/** @deprecated instead use textNegative */
export var critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use overlaySoft */
export var overlay = 'var(--plasma-colors-overlay, var(--overlay-soft))';
/** @deprecated instead use surfaceTransparentPrimary */
export var surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, var(--surface-transparent-primary))';
/** @deprecated instead use surfaceTransparentSecondary */
export var surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, var(--surface-transparent-secondary))';
/** @deprecated instead use surfaceTransparentTertiary */
export var surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, var(--surface-transparent-tertiary))';
/** @deprecated instead use surfaceSolidPrimary */
export var surfaceSolid01 = 'var(--plasma-colors-surface-solid01, var(--surface-solid-primary))';
/** @deprecated instead use surfaceSolidSecondary */
export var surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
export var surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use surfaceTransparentCard */
export var surfaceCard = 'var(--plasma-colors-surface-card, var(--surface-transparent-card))';
/** @deprecated instead use surfaceTransparentSecondary */
export var buttonSecondary = 'var(--plasma-colors-button-secondary, var(--surface-transparent-secondary))';
/** @deprecated instead use textAccent */
export var buttonAccent = 'var(--plasma-colors-button-accent, var(--text-accent))';
/** @deprecated instead use surfacePositive */
export var buttonSuccess = 'var(--plasma-colors-button-success, var(--surface-positive))';
/** @deprecated instead use surfaceWarning */
export var buttonWarning = 'var(--plasma-colors-button-warning, var(--surface-warning))';
/** @deprecated instead use surfaceNegative */
export var buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
