// @salutejs/client/esm/assistant-972e648c.js
import { _ as __assign, b as __awaiter, c as __generator, d as createNanoEvents, a as __rest } from './common-ba25e019.js';
import { b as commonjsRequire, r as requireMinimal, c as commonjsGlobal, p as protoExports, M as MessageNames, d as createProtocol } from './sdk-0b376395.js';

var asr = {exports: {}};

/*eslint-disable block-scoped-var, id-length, no-control-regex, no-magic-numbers, no-prototype-builtins, no-redeclare, no-shadow, no-var, sort-vars*/

(function (module) {
	(function(global, factory) { /* global define, require, module */

	    /* AMD */ if (typeof commonjsRequire === 'function' && 'object' === 'object' && module && module.exports)
	        module.exports = factory(requireMinimal());

	})(commonjsGlobal, function($protobuf) {

	    // Common aliases
	    var $Reader = $protobuf.Reader, $util = $protobuf.util;
	    
	    // Exported root namespace
	    var $root = $protobuf.roots["default"] || ($protobuf.roots["default"] = {});
	    
	    $root.Variables = (function() {
	    
	        /**
	         * Properties of a Variables.
	         * @exports IVariables
	         * @interface IVariables
	         * @property {Object.<string,string>|null} [variables] Variables variables
	         */
	    
	        /**
	         * Constructs a new Variables.
	         * @exports Variables
	         * @classdesc Represents a Variables.
	         * @implements IVariables
	         * @constructor
	         * @param {IVariables=} [properties] Properties to set
	         */
	        function Variables(properties) {
	            this.variables = {};
	            if (properties)
	                for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                    if (properties[keys[i]] != null)
	                        this[keys[i]] = properties[keys[i]];
	        }
	    
	        /**
	         * Variables variables.
	         * @member {Object.<string,string>} variables
	         * @memberof Variables
	         * @instance
	         */
	        Variables.prototype.variables = $util.emptyObject;
	    
	        /**
	         * Decodes a Variables message from the specified reader or buffer.
	         * @function decode
	         * @memberof Variables
	         * @static
	         * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	         * @param {number} [length] Message length if known beforehand
	         * @returns {Variables} Variables
	         * @throws {Error} If the payload is not a reader or valid buffer
	         * @throws {$protobuf.util.ProtocolError} If required fields are missing
	         */
	        Variables.decode = function decode(reader, length) {
	            if (!(reader instanceof $Reader))
	                reader = $Reader.create(reader);
	            var end = length === undefined ? reader.len : reader.pos + length, message = new $root.Variables(), key, value;
	            while (reader.pos < end) {
	                var tag = reader.uint32();
	                switch (tag >>> 3) {
	                case 1: {
	                        if (message.variables === $util.emptyObject)
	                            message.variables = {};
	                        var end2 = reader.uint32() + reader.pos;
	                        key = "";
	                        value = "";
	                        while (reader.pos < end2) {
	                            var tag2 = reader.uint32();
	                            switch (tag2 >>> 3) {
	                            case 1:
	                                key = reader.string();
	                                break;
	                            case 2:
	                                value = reader.string();
	                                break;
	                            default:
	                                reader.skipType(tag2 & 7);
	                                break;
	                            }
	                        }
	                        message.variables[key] = value;
	                        break;
	                    }
	                default:
	                    reader.skipType(tag & 7);
	                    break;
	                }
	            }
	            return message;
	        };
	    
	        return Variables;
	    })();
	    
	    $root.UndecodedSeconds = (function() {
	    
	        /**
	         * Properties of an UndecodedSeconds.
	         * @exports IUndecodedSeconds
	         * @interface IUndecodedSeconds
	         * @property {number|null} [undecodedSeconds] UndecodedSeconds undecodedSeconds
	         */
	    
	        /**
	         * Constructs a new UndecodedSeconds.
	         * @exports UndecodedSeconds
	         * @classdesc Represents an UndecodedSeconds.
	         * @implements IUndecodedSeconds
	         * @constructor
	         * @param {IUndecodedSeconds=} [properties] Properties to set
	         */
	        function UndecodedSeconds(properties) {
	            if (properties)
	                for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                    if (properties[keys[i]] != null)
	                        this[keys[i]] = properties[keys[i]];
	        }
	    
	        /**
	         * UndecodedSeconds undecodedSeconds.
	         * @member {number} undecodedSeconds
	         * @memberof UndecodedSeconds
	         * @instance
	         */
	        UndecodedSeconds.prototype.undecodedSeconds = 0;
	    
	        /**
	         * Decodes an UndecodedSeconds message from the specified reader or buffer.
	         * @function decode
	         * @memberof UndecodedSeconds
	         * @static
	         * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	         * @param {number} [length] Message length if known beforehand
	         * @returns {UndecodedSeconds} UndecodedSeconds
	         * @throws {Error} If the payload is not a reader or valid buffer
	         * @throws {$protobuf.util.ProtocolError} If required fields are missing
	         */
	        UndecodedSeconds.decode = function decode(reader, length) {
	            if (!(reader instanceof $Reader))
	                reader = $Reader.create(reader);
	            var end = length === undefined ? reader.len : reader.pos + length, message = new $root.UndecodedSeconds();
	            while (reader.pos < end) {
	                var tag = reader.uint32();
	                switch (tag >>> 3) {
	                case 1: {
	                        message.undecodedSeconds = reader.float();
	                        break;
	                    }
	                default:
	                    reader.skipType(tag & 7);
	                    break;
	                }
	            }
	            return message;
	        };
	    
	        return UndecodedSeconds;
	    })();
	    
	    $root.FullyFinalized = (function() {
	    
	        /**
	         * Properties of a FullyFinalized.
	         * @exports IFullyFinalized
	         * @interface IFullyFinalized
	         */
	    
	        /**
	         * Constructs a new FullyFinalized.
	         * @exports FullyFinalized
	         * @classdesc Represents a FullyFinalized.
	         * @implements IFullyFinalized
	         * @constructor
	         * @param {IFullyFinalized=} [properties] Properties to set
	         */
	        function FullyFinalized(properties) {
	            if (properties)
	                for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                    if (properties[keys[i]] != null)
	                        this[keys[i]] = properties[keys[i]];
	        }
	    
	        /**
	         * Decodes a FullyFinalized message from the specified reader or buffer.
	         * @function decode
	         * @memberof FullyFinalized
	         * @static
	         * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	         * @param {number} [length] Message length if known beforehand
	         * @returns {FullyFinalized} FullyFinalized
	         * @throws {Error} If the payload is not a reader or valid buffer
	         * @throws {$protobuf.util.ProtocolError} If required fields are missing
	         */
	        FullyFinalized.decode = function decode(reader, length) {
	            if (!(reader instanceof $Reader))
	                reader = $Reader.create(reader);
	            var end = length === undefined ? reader.len : reader.pos + length, message = new $root.FullyFinalized();
	            while (reader.pos < end) {
	                var tag = reader.uint32();
	                switch (tag >>> 3) {
	                default:
	                    reader.skipType(tag & 7);
	                    break;
	                }
	            }
	            return message;
	        };
	    
	        return FullyFinalized;
	    })();
	    
	    $root.EmotionResult = (function() {
	    
	        /**
	         * Properties of an EmotionResult.
	         * @exports IEmotionResult
	         * @interface IEmotionResult
	         * @property {string|null} [name] EmotionResult name
	         * @property {number|null} [confidence] EmotionResult confidence
	         */
	    
	        /**
	         * Constructs a new EmotionResult.
	         * @exports EmotionResult
	         * @classdesc Represents an EmotionResult.
	         * @implements IEmotionResult
	         * @constructor
	         * @param {IEmotionResult=} [properties] Properties to set
	         */
	        function EmotionResult(properties) {
	            if (properties)
	                for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                    if (properties[keys[i]] != null)
	                        this[keys[i]] = properties[keys[i]];
	        }
	    
	        /**
	         * EmotionResult name.
	         * @member {string} name
	         * @memberof EmotionResult
	         * @instance
	         */
	        EmotionResult.prototype.name = "";
	    
	        /**
	         * EmotionResult confidence.
	         * @member {number} confidence
	         * @memberof EmotionResult
	         * @instance
	         */
	        EmotionResult.prototype.confidence = 0;
	    
	        /**
	         * Decodes an EmotionResult message from the specified reader or buffer.
	         * @function decode
	         * @memberof EmotionResult
	         * @static
	         * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	         * @param {number} [length] Message length if known beforehand
	         * @returns {EmotionResult} EmotionResult
	         * @throws {Error} If the payload is not a reader or valid buffer
	         * @throws {$protobuf.util.ProtocolError} If required fields are missing
	         */
	        EmotionResult.decode = function decode(reader, length) {
	            if (!(reader instanceof $Reader))
	                reader = $Reader.create(reader);
	            var end = length === undefined ? reader.len : reader.pos + length, message = new $root.EmotionResult();
	            while (reader.pos < end) {
	                var tag = reader.uint32();
	                switch (tag >>> 3) {
	                case 1: {
	                        message.name = reader.string();
	                        break;
	                    }
	                case 2: {
	                        message.confidence = reader.float();
	                        break;
	                    }
	                default:
	                    reader.skipType(tag & 7);
	                    break;
	                }
	            }
	            return message;
	        };
	    
	        return EmotionResult;
	    })();
	    
	    $root.Hypothesis = (function() {
	    
	        /**
	         * Properties of a Hypothesis.
	         * @exports IHypothesis
	         * @interface IHypothesis
	         * @property {string|null} [words] Hypothesis words
	         * @property {number|null} [acousticCost] Hypothesis acousticCost
	         * @property {number|null} [linguisticCost] Hypothesis linguisticCost
	         * @property {number|null} [finalCost] Hypothesis finalCost
	         * @property {number|null} [phraseStart] Hypothesis phraseStart
	         * @property {number|null} [phraseEnd] Hypothesis phraseEnd
	         * @property {string|null} [normalizedText] Hypothesis normalizedText
	         */
	    
	        /**
	         * Constructs a new Hypothesis.
	         * @exports Hypothesis
	         * @classdesc Represents a Hypothesis.
	         * @implements IHypothesis
	         * @constructor
	         * @param {IHypothesis=} [properties] Properties to set
	         */
	        function Hypothesis(properties) {
	            if (properties)
	                for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                    if (properties[keys[i]] != null)
	                        this[keys[i]] = properties[keys[i]];
	        }
	    
	        /**
	         * Hypothesis words.
	         * @member {string} words
	         * @memberof Hypothesis
	         * @instance
	         */
	        Hypothesis.prototype.words = "";
	    
	        /**
	         * Hypothesis acousticCost.
	         * @member {number} acousticCost
	         * @memberof Hypothesis
	         * @instance
	         */
	        Hypothesis.prototype.acousticCost = 0;
	    
	        /**
	         * Hypothesis linguisticCost.
	         * @member {number} linguisticCost
	         * @memberof Hypothesis
	         * @instance
	         */
	        Hypothesis.prototype.linguisticCost = 0;
	    
	        /**
	         * Hypothesis finalCost.
	         * @member {number} finalCost
	         * @memberof Hypothesis
	         * @instance
	         */
	        Hypothesis.prototype.finalCost = 0;
	    
	        /**
	         * Hypothesis phraseStart.
	         * @member {number} phraseStart
	         * @memberof Hypothesis
	         * @instance
	         */
	        Hypothesis.prototype.phraseStart = 0;
	    
	        /**
	         * Hypothesis phraseEnd.
	         * @member {number} phraseEnd
	         * @memberof Hypothesis
	         * @instance
	         */
	        Hypothesis.prototype.phraseEnd = 0;
	    
	        /**
	         * Hypothesis normalizedText.
	         * @member {string} normalizedText
	         * @memberof Hypothesis
	         * @instance
	         */
	        Hypothesis.prototype.normalizedText = "";
	    
	        /**
	         * Decodes a Hypothesis message from the specified reader or buffer.
	         * @function decode
	         * @memberof Hypothesis
	         * @static
	         * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	         * @param {number} [length] Message length if known beforehand
	         * @returns {Hypothesis} Hypothesis
	         * @throws {Error} If the payload is not a reader or valid buffer
	         * @throws {$protobuf.util.ProtocolError} If required fields are missing
	         */
	        Hypothesis.decode = function decode(reader, length) {
	            if (!(reader instanceof $Reader))
	                reader = $Reader.create(reader);
	            var end = length === undefined ? reader.len : reader.pos + length, message = new $root.Hypothesis();
	            while (reader.pos < end) {
	                var tag = reader.uint32();
	                switch (tag >>> 3) {
	                case 1: {
	                        message.words = reader.string();
	                        break;
	                    }
	                case 2: {
	                        message.acousticCost = reader.float();
	                        break;
	                    }
	                case 3: {
	                        message.linguisticCost = reader.float();
	                        break;
	                    }
	                case 4: {
	                        message.finalCost = reader.float();
	                        break;
	                    }
	                case 5: {
	                        message.phraseStart = reader.float();
	                        break;
	                    }
	                case 6: {
	                        message.phraseEnd = reader.float();
	                        break;
	                    }
	                case 7: {
	                        message.normalizedText = reader.string();
	                        break;
	                    }
	                default:
	                    reader.skipType(tag & 7);
	                    break;
	                }
	            }
	            return message;
	        };
	    
	        return Hypothesis;
	    })();
	    
	    $root.DecoderResult = (function() {
	    
	        /**
	         * Properties of a DecoderResult.
	         * @exports IDecoderResult
	         * @interface IDecoderResult
	         * @property {Array.<IHypothesis>|null} [hypothesis] DecoderResult hypothesis
	         * @property {number|null} [chunkStart] DecoderResult chunkStart
	         * @property {number|null} [chunkEnd] DecoderResult chunkEnd
	         * @property {number|null} [timeEndpointDetectionMs] DecoderResult timeEndpointDetectionMs
	         * @property {number|null} [timeDecodingMs] DecoderResult timeDecodingMs
	         * @property {IVariables|null} [variables] DecoderResult variables
	         * @property {boolean|null} [isFinal] DecoderResult isFinal
	         * @property {Array.<IEmotionResult>|null} [emotionResult] DecoderResult emotionResult
	         * @property {Array.<DecoderResult.IContextAnswer>|null} [contextAnswer] DecoderResult contextAnswer
	         */
	    
	        /**
	         * Constructs a new DecoderResult.
	         * @exports DecoderResult
	         * @classdesc Represents a DecoderResult.
	         * @implements IDecoderResult
	         * @constructor
	         * @param {IDecoderResult=} [properties] Properties to set
	         */
	        function DecoderResult(properties) {
	            this.hypothesis = [];
	            this.emotionResult = [];
	            this.contextAnswer = [];
	            if (properties)
	                for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                    if (properties[keys[i]] != null)
	                        this[keys[i]] = properties[keys[i]];
	        }
	    
	        /**
	         * DecoderResult hypothesis.
	         * @member {Array.<IHypothesis>} hypothesis
	         * @memberof DecoderResult
	         * @instance
	         */
	        DecoderResult.prototype.hypothesis = $util.emptyArray;
	    
	        /**
	         * DecoderResult chunkStart.
	         * @member {number} chunkStart
	         * @memberof DecoderResult
	         * @instance
	         */
	        DecoderResult.prototype.chunkStart = 0;
	    
	        /**
	         * DecoderResult chunkEnd.
	         * @member {number} chunkEnd
	         * @memberof DecoderResult
	         * @instance
	         */
	        DecoderResult.prototype.chunkEnd = 0;
	    
	        /**
	         * DecoderResult timeEndpointDetectionMs.
	         * @member {number} timeEndpointDetectionMs
	         * @memberof DecoderResult
	         * @instance
	         */
	        DecoderResult.prototype.timeEndpointDetectionMs = 0;
	    
	        /**
	         * DecoderResult timeDecodingMs.
	         * @member {number} timeDecodingMs
	         * @memberof DecoderResult
	         * @instance
	         */
	        DecoderResult.prototype.timeDecodingMs = 0;
	    
	        /**
	         * DecoderResult variables.
	         * @member {IVariables|null|undefined} variables
	         * @memberof DecoderResult
	         * @instance
	         */
	        DecoderResult.prototype.variables = null;
	    
	        /**
	         * DecoderResult isFinal.
	         * @member {boolean} isFinal
	         * @memberof DecoderResult
	         * @instance
	         */
	        DecoderResult.prototype.isFinal = false;
	    
	        /**
	         * DecoderResult emotionResult.
	         * @member {Array.<IEmotionResult>} emotionResult
	         * @memberof DecoderResult
	         * @instance
	         */
	        DecoderResult.prototype.emotionResult = $util.emptyArray;
	    
	        /**
	         * DecoderResult contextAnswer.
	         * @member {Array.<DecoderResult.IContextAnswer>} contextAnswer
	         * @memberof DecoderResult
	         * @instance
	         */
	        DecoderResult.prototype.contextAnswer = $util.emptyArray;
	    
	        /**
	         * Decodes a DecoderResult message from the specified reader or buffer.
	         * @function decode
	         * @memberof DecoderResult
	         * @static
	         * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	         * @param {number} [length] Message length if known beforehand
	         * @returns {DecoderResult} DecoderResult
	         * @throws {Error} If the payload is not a reader or valid buffer
	         * @throws {$protobuf.util.ProtocolError} If required fields are missing
	         */
	        DecoderResult.decode = function decode(reader, length) {
	            if (!(reader instanceof $Reader))
	                reader = $Reader.create(reader);
	            var end = length === undefined ? reader.len : reader.pos + length, message = new $root.DecoderResult();
	            while (reader.pos < end) {
	                var tag = reader.uint32();
	                switch (tag >>> 3) {
	                case 1: {
	                        if (!(message.hypothesis && message.hypothesis.length))
	                            message.hypothesis = [];
	                        message.hypothesis.push($root.Hypothesis.decode(reader, reader.uint32()));
	                        break;
	                    }
	                case 2: {
	                        message.chunkStart = reader.float();
	                        break;
	                    }
	                case 3: {
	                        message.chunkEnd = reader.float();
	                        break;
	                    }
	                case 4: {
	                        message.timeEndpointDetectionMs = reader.float();
	                        break;
	                    }
	                case 5: {
	                        message.timeDecodingMs = reader.float();
	                        break;
	                    }
	                case 6: {
	                        message.variables = $root.Variables.decode(reader, reader.uint32());
	                        break;
	                    }
	                case 7: {
	                        message.isFinal = reader.bool();
	                        break;
	                    }
	                case 8: {
	                        if (!(message.emotionResult && message.emotionResult.length))
	                            message.emotionResult = [];
	                        message.emotionResult.push($root.EmotionResult.decode(reader, reader.uint32()));
	                        break;
	                    }
	                case 9: {
	                        if (!(message.contextAnswer && message.contextAnswer.length))
	                            message.contextAnswer = [];
	                        message.contextAnswer.push($root.DecoderResult.ContextAnswer.decode(reader, reader.uint32()));
	                        break;
	                    }
	                default:
	                    reader.skipType(tag & 7);
	                    break;
	                }
	            }
	            return message;
	        };
	    
	        DecoderResult.ContextAnswer = (function() {
	    
	            /**
	             * Properties of a ContextAnswer.
	             * @memberof DecoderResult
	             * @interface IContextAnswer
	             * @property {Array.<DecoderResult.ContextAnswer.IContextRef>|null} [contextResult] ContextAnswer contextResult
	             */
	    
	            /**
	             * Constructs a new ContextAnswer.
	             * @memberof DecoderResult
	             * @classdesc Represents a ContextAnswer.
	             * @implements IContextAnswer
	             * @constructor
	             * @param {DecoderResult.IContextAnswer=} [properties] Properties to set
	             */
	            function ContextAnswer(properties) {
	                this.contextResult = [];
	                if (properties)
	                    for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                        if (properties[keys[i]] != null)
	                            this[keys[i]] = properties[keys[i]];
	            }
	    
	            /**
	             * ContextAnswer contextResult.
	             * @member {Array.<DecoderResult.ContextAnswer.IContextRef>} contextResult
	             * @memberof DecoderResult.ContextAnswer
	             * @instance
	             */
	            ContextAnswer.prototype.contextResult = $util.emptyArray;
	    
	            /**
	             * Decodes a ContextAnswer message from the specified reader or buffer.
	             * @function decode
	             * @memberof DecoderResult.ContextAnswer
	             * @static
	             * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	             * @param {number} [length] Message length if known beforehand
	             * @returns {DecoderResult.ContextAnswer} ContextAnswer
	             * @throws {Error} If the payload is not a reader or valid buffer
	             * @throws {$protobuf.util.ProtocolError} If required fields are missing
	             */
	            ContextAnswer.decode = function decode(reader, length) {
	                if (!(reader instanceof $Reader))
	                    reader = $Reader.create(reader);
	                var end = length === undefined ? reader.len : reader.pos + length, message = new $root.DecoderResult.ContextAnswer();
	                while (reader.pos < end) {
	                    var tag = reader.uint32();
	                    switch (tag >>> 3) {
	                    case 1: {
	                            if (!(message.contextResult && message.contextResult.length))
	                                message.contextResult = [];
	                            message.contextResult.push($root.DecoderResult.ContextAnswer.ContextRef.decode(reader, reader.uint32()));
	                            break;
	                        }
	                    default:
	                        reader.skipType(tag & 7);
	                        break;
	                    }
	                }
	                return message;
	            };
	    
	            ContextAnswer.ContextRef = (function() {
	    
	                /**
	                 * Properties of a ContextRef.
	                 * @memberof DecoderResult.ContextAnswer
	                 * @interface IContextRef
	                 * @property {string|null} [id] ContextRef id
	                 * @property {number|null} [index] ContextRef index
	                 * @property {string|null} [originalValue] ContextRef originalValue
	                 * @property {string|null} [predictedValue] ContextRef predictedValue
	                 * @property {number|null} [score] ContextRef score
	                 */
	    
	                /**
	                 * Constructs a new ContextRef.
	                 * @memberof DecoderResult.ContextAnswer
	                 * @classdesc Represents a ContextRef.
	                 * @implements IContextRef
	                 * @constructor
	                 * @param {DecoderResult.ContextAnswer.IContextRef=} [properties] Properties to set
	                 */
	                function ContextRef(properties) {
	                    if (properties)
	                        for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                            if (properties[keys[i]] != null)
	                                this[keys[i]] = properties[keys[i]];
	                }
	    
	                /**
	                 * ContextRef id.
	                 * @member {string} id
	                 * @memberof DecoderResult.ContextAnswer.ContextRef
	                 * @instance
	                 */
	                ContextRef.prototype.id = "";
	    
	                /**
	                 * ContextRef index.
	                 * @member {number} index
	                 * @memberof DecoderResult.ContextAnswer.ContextRef
	                 * @instance
	                 */
	                ContextRef.prototype.index = 0;
	    
	                /**
	                 * ContextRef originalValue.
	                 * @member {string} originalValue
	                 * @memberof DecoderResult.ContextAnswer.ContextRef
	                 * @instance
	                 */
	                ContextRef.prototype.originalValue = "";
	    
	                /**
	                 * ContextRef predictedValue.
	                 * @member {string} predictedValue
	                 * @memberof DecoderResult.ContextAnswer.ContextRef
	                 * @instance
	                 */
	                ContextRef.prototype.predictedValue = "";
	    
	                /**
	                 * ContextRef score.
	                 * @member {number} score
	                 * @memberof DecoderResult.ContextAnswer.ContextRef
	                 * @instance
	                 */
	                ContextRef.prototype.score = 0;
	    
	                /**
	                 * Decodes a ContextRef message from the specified reader or buffer.
	                 * @function decode
	                 * @memberof DecoderResult.ContextAnswer.ContextRef
	                 * @static
	                 * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	                 * @param {number} [length] Message length if known beforehand
	                 * @returns {DecoderResult.ContextAnswer.ContextRef} ContextRef
	                 * @throws {Error} If the payload is not a reader or valid buffer
	                 * @throws {$protobuf.util.ProtocolError} If required fields are missing
	                 */
	                ContextRef.decode = function decode(reader, length) {
	                    if (!(reader instanceof $Reader))
	                        reader = $Reader.create(reader);
	                    var end = length === undefined ? reader.len : reader.pos + length, message = new $root.DecoderResult.ContextAnswer.ContextRef();
	                    while (reader.pos < end) {
	                        var tag = reader.uint32();
	                        switch (tag >>> 3) {
	                        case 1: {
	                                message.id = reader.string();
	                                break;
	                            }
	                        case 2: {
	                                message.index = reader.int32();
	                                break;
	                            }
	                        case 3: {
	                                message.originalValue = reader.string();
	                                break;
	                            }
	                        case 4: {
	                                message.predictedValue = reader.string();
	                                break;
	                            }
	                        case 5: {
	                                message.score = reader.float();
	                                break;
	                            }
	                        default:
	                            reader.skipType(tag & 7);
	                            break;
	                        }
	                    }
	                    return message;
	                };
	    
	                return ContextRef;
	            })();
	    
	            return ContextAnswer;
	        })();
	    
	        return DecoderResult;
	    })();
	    
	    $root.ErrorResponse = (function() {
	    
	        /**
	         * Properties of an ErrorResponse.
	         * @exports IErrorResponse
	         * @interface IErrorResponse
	         * @property {string|null} [errorMessage] ErrorResponse errorMessage
	         */
	    
	        /**
	         * Constructs a new ErrorResponse.
	         * @exports ErrorResponse
	         * @classdesc Represents an ErrorResponse.
	         * @implements IErrorResponse
	         * @constructor
	         * @param {IErrorResponse=} [properties] Properties to set
	         */
	        function ErrorResponse(properties) {
	            if (properties)
	                for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                    if (properties[keys[i]] != null)
	                        this[keys[i]] = properties[keys[i]];
	        }
	    
	        /**
	         * ErrorResponse errorMessage.
	         * @member {string} errorMessage
	         * @memberof ErrorResponse
	         * @instance
	         */
	        ErrorResponse.prototype.errorMessage = "";
	    
	        /**
	         * Decodes an ErrorResponse message from the specified reader or buffer.
	         * @function decode
	         * @memberof ErrorResponse
	         * @static
	         * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	         * @param {number} [length] Message length if known beforehand
	         * @returns {ErrorResponse} ErrorResponse
	         * @throws {Error} If the payload is not a reader or valid buffer
	         * @throws {$protobuf.util.ProtocolError} If required fields are missing
	         */
	        ErrorResponse.decode = function decode(reader, length) {
	            if (!(reader instanceof $Reader))
	                reader = $Reader.create(reader);
	            var end = length === undefined ? reader.len : reader.pos + length, message = new $root.ErrorResponse();
	            while (reader.pos < end) {
	                var tag = reader.uint32();
	                switch (tag >>> 3) {
	                case 1: {
	                        message.errorMessage = reader.string();
	                        break;
	                    }
	                default:
	                    reader.skipType(tag & 7);
	                    break;
	                }
	            }
	            return message;
	        };
	    
	        return ErrorResponse;
	    })();
	    
	    $root.PacketWrapperFromServer = (function() {
	    
	        /**
	         * Properties of a PacketWrapperFromServer.
	         * @exports IPacketWrapperFromServer
	         * @interface IPacketWrapperFromServer
	         * @property {IUndecodedSeconds|null} [undecodedSecondsField] PacketWrapperFromServer undecodedSecondsField
	         * @property {IFullyFinalized|null} [fullyFinalizedField] PacketWrapperFromServer fullyFinalizedField
	         * @property {IDecoderResult|null} [decoderResultField] PacketWrapperFromServer decoderResultField
	         * @property {IErrorResponse|null} [errorResponse] PacketWrapperFromServer errorResponse
	         */
	    
	        /**
	         * Constructs a new PacketWrapperFromServer.
	         * @exports PacketWrapperFromServer
	         * @classdesc Represents a PacketWrapperFromServer.
	         * @implements IPacketWrapperFromServer
	         * @constructor
	         * @param {IPacketWrapperFromServer=} [properties] Properties to set
	         */
	        function PacketWrapperFromServer(properties) {
	            if (properties)
	                for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                    if (properties[keys[i]] != null)
	                        this[keys[i]] = properties[keys[i]];
	        }
	    
	        /**
	         * PacketWrapperFromServer undecodedSecondsField.
	         * @member {IUndecodedSeconds|null|undefined} undecodedSecondsField
	         * @memberof PacketWrapperFromServer
	         * @instance
	         */
	        PacketWrapperFromServer.prototype.undecodedSecondsField = null;
	    
	        /**
	         * PacketWrapperFromServer fullyFinalizedField.
	         * @member {IFullyFinalized|null|undefined} fullyFinalizedField
	         * @memberof PacketWrapperFromServer
	         * @instance
	         */
	        PacketWrapperFromServer.prototype.fullyFinalizedField = null;
	    
	        /**
	         * PacketWrapperFromServer decoderResultField.
	         * @member {IDecoderResult|null|undefined} decoderResultField
	         * @memberof PacketWrapperFromServer
	         * @instance
	         */
	        PacketWrapperFromServer.prototype.decoderResultField = null;
	    
	        /**
	         * PacketWrapperFromServer errorResponse.
	         * @member {IErrorResponse|null|undefined} errorResponse
	         * @memberof PacketWrapperFromServer
	         * @instance
	         */
	        PacketWrapperFromServer.prototype.errorResponse = null;
	    
	        // OneOf field names bound to virtual getters and setters
	        var $oneOfFields;
	    
	        /**
	         * PacketWrapperFromServer MessageType.
	         * @member {"undecodedSecondsField"|"fullyFinalizedField"|"decoderResultField"|"errorResponse"|undefined} MessageType
	         * @memberof PacketWrapperFromServer
	         * @instance
	         */
	        Object.defineProperty(PacketWrapperFromServer.prototype, "MessageType", {
	            get: $util.oneOfGetter($oneOfFields = ["undecodedSecondsField", "fullyFinalizedField", "decoderResultField", "errorResponse"]),
	            set: $util.oneOfSetter($oneOfFields)
	        });
	    
	        /**
	         * Decodes a PacketWrapperFromServer message from the specified reader or buffer.
	         * @function decode
	         * @memberof PacketWrapperFromServer
	         * @static
	         * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	         * @param {number} [length] Message length if known beforehand
	         * @returns {PacketWrapperFromServer} PacketWrapperFromServer
	         * @throws {Error} If the payload is not a reader or valid buffer
	         * @throws {$protobuf.util.ProtocolError} If required fields are missing
	         */
	        PacketWrapperFromServer.decode = function decode(reader, length) {
	            if (!(reader instanceof $Reader))
	                reader = $Reader.create(reader);
	            var end = length === undefined ? reader.len : reader.pos + length, message = new $root.PacketWrapperFromServer();
	            while (reader.pos < end) {
	                var tag = reader.uint32();
	                switch (tag >>> 3) {
	                case 1: {
	                        message.undecodedSecondsField = $root.UndecodedSeconds.decode(reader, reader.uint32());
	                        break;
	                    }
	                case 2: {
	                        message.fullyFinalizedField = $root.FullyFinalized.decode(reader, reader.uint32());
	                        break;
	                    }
	                case 4: {
	                        message.decoderResultField = $root.DecoderResult.decode(reader, reader.uint32());
	                        break;
	                    }
	                case 8: {
	                        message.errorResponse = $root.ErrorResponse.decode(reader, reader.uint32());
	                        break;
	                    }
	                default:
	                    reader.skipType(tag & 7);
	                    break;
	                }
	            }
	            return message;
	        };
	    
	        return PacketWrapperFromServer;
	    })();

	    return $root;
	}); 
} (asr));

var asrExports = asr.exports;

var mtt = {exports: {}};

/*eslint-disable block-scoped-var, id-length, no-control-regex, no-magic-numbers, no-prototype-builtins, no-redeclare, no-shadow, no-var, sort-vars*/

(function (module) {
	(function(global, factory) { /* global define, require, module */

	    /* AMD */ if (typeof commonjsRequire === 'function' && 'object' === 'object' && module && module.exports)
	        module.exports = factory(requireMinimal());

	})(commonjsGlobal, function($protobuf) {

	    // Common aliases
	    var $Reader = $protobuf.Reader, $util = $protobuf.util;
	    
	    // Exported root namespace
	    var $root = $protobuf.roots["default"] || ($protobuf.roots["default"] = {});
	    
	    $root.Music2TrackProtocol = (function() {
	    
	        /**
	         * Namespace Music2TrackProtocol.
	         * @exports Music2TrackProtocol
	         * @namespace
	         */
	        var Music2TrackProtocol = {};
	    
	        Music2TrackProtocol.DecoderResult = (function() {
	    
	            /**
	             * Properties of a DecoderResult.
	             * @memberof Music2TrackProtocol
	             * @interface IDecoderResult
	             * @property {string|null} [result] DecoderResult result
	             * @property {boolean|null} [isMusicFound] DecoderResult isMusicFound
	             * @property {boolean|null} [isFinal] DecoderResult isFinal
	             */
	    
	            /**
	             * Constructs a new DecoderResult.
	             * @memberof Music2TrackProtocol
	             * @classdesc Represents a DecoderResult.
	             * @implements IDecoderResult
	             * @constructor
	             * @param {Music2TrackProtocol.IDecoderResult=} [properties] Properties to set
	             */
	            function DecoderResult(properties) {
	                if (properties)
	                    for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                        if (properties[keys[i]] != null)
	                            this[keys[i]] = properties[keys[i]];
	            }
	    
	            /**
	             * DecoderResult result.
	             * @member {string} result
	             * @memberof Music2TrackProtocol.DecoderResult
	             * @instance
	             */
	            DecoderResult.prototype.result = "";
	    
	            /**
	             * DecoderResult isMusicFound.
	             * @member {boolean} isMusicFound
	             * @memberof Music2TrackProtocol.DecoderResult
	             * @instance
	             */
	            DecoderResult.prototype.isMusicFound = false;
	    
	            /**
	             * DecoderResult isFinal.
	             * @member {boolean} isFinal
	             * @memberof Music2TrackProtocol.DecoderResult
	             * @instance
	             */
	            DecoderResult.prototype.isFinal = false;
	    
	            /**
	             * Decodes a DecoderResult message from the specified reader or buffer.
	             * @function decode
	             * @memberof Music2TrackProtocol.DecoderResult
	             * @static
	             * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	             * @param {number} [length] Message length if known beforehand
	             * @returns {Music2TrackProtocol.DecoderResult} DecoderResult
	             * @throws {Error} If the payload is not a reader or valid buffer
	             * @throws {$protobuf.util.ProtocolError} If required fields are missing
	             */
	            DecoderResult.decode = function decode(reader, length) {
	                if (!(reader instanceof $Reader))
	                    reader = $Reader.create(reader);
	                var end = length === undefined ? reader.len : reader.pos + length, message = new $root.Music2TrackProtocol.DecoderResult();
	                while (reader.pos < end) {
	                    var tag = reader.uint32();
	                    switch (tag >>> 3) {
	                    case 1: {
	                            message.result = reader.string();
	                            break;
	                        }
	                    case 2: {
	                            message.isMusicFound = reader.bool();
	                            break;
	                        }
	                    case 3: {
	                            message.isFinal = reader.bool();
	                            break;
	                        }
	                    default:
	                        reader.skipType(tag & 7);
	                        break;
	                    }
	                }
	                return message;
	            };
	    
	            return DecoderResult;
	        })();
	    
	        Music2TrackProtocol.ErrorResponse = (function() {
	    
	            /**
	             * Properties of an ErrorResponse.
	             * @memberof Music2TrackProtocol
	             * @interface IErrorResponse
	             * @property {string|null} [errorMessage] ErrorResponse errorMessage
	             * @property {number|null} [errorCode] ErrorResponse errorCode
	             */
	    
	            /**
	             * Constructs a new ErrorResponse.
	             * @memberof Music2TrackProtocol
	             * @classdesc Represents an ErrorResponse.
	             * @implements IErrorResponse
	             * @constructor
	             * @param {Music2TrackProtocol.IErrorResponse=} [properties] Properties to set
	             */
	            function ErrorResponse(properties) {
	                if (properties)
	                    for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                        if (properties[keys[i]] != null)
	                            this[keys[i]] = properties[keys[i]];
	            }
	    
	            /**
	             * ErrorResponse errorMessage.
	             * @member {string} errorMessage
	             * @memberof Music2TrackProtocol.ErrorResponse
	             * @instance
	             */
	            ErrorResponse.prototype.errorMessage = "";
	    
	            /**
	             * ErrorResponse errorCode.
	             * @member {number} errorCode
	             * @memberof Music2TrackProtocol.ErrorResponse
	             * @instance
	             */
	            ErrorResponse.prototype.errorCode = 0;
	    
	            /**
	             * Decodes an ErrorResponse message from the specified reader or buffer.
	             * @function decode
	             * @memberof Music2TrackProtocol.ErrorResponse
	             * @static
	             * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	             * @param {number} [length] Message length if known beforehand
	             * @returns {Music2TrackProtocol.ErrorResponse} ErrorResponse
	             * @throws {Error} If the payload is not a reader or valid buffer
	             * @throws {$protobuf.util.ProtocolError} If required fields are missing
	             */
	            ErrorResponse.decode = function decode(reader, length) {
	                if (!(reader instanceof $Reader))
	                    reader = $Reader.create(reader);
	                var end = length === undefined ? reader.len : reader.pos + length, message = new $root.Music2TrackProtocol.ErrorResponse();
	                while (reader.pos < end) {
	                    var tag = reader.uint32();
	                    switch (tag >>> 3) {
	                    case 1: {
	                            message.errorMessage = reader.string();
	                            break;
	                        }
	                    case 2: {
	                            message.errorCode = reader.int32();
	                            break;
	                        }
	                    default:
	                        reader.skipType(tag & 7);
	                        break;
	                    }
	                }
	                return message;
	            };
	    
	            return ErrorResponse;
	        })();
	    
	        Music2TrackProtocol.MttResponse = (function() {
	    
	            /**
	             * Properties of a MttResponse.
	             * @memberof Music2TrackProtocol
	             * @interface IMttResponse
	             * @property {Music2TrackProtocol.IDecoderResult|null} [decoderResultField] MttResponse decoderResultField
	             * @property {Music2TrackProtocol.IErrorResponse|null} [errorResponse] MttResponse errorResponse
	             */
	    
	            /**
	             * Constructs a new MttResponse.
	             * @memberof Music2TrackProtocol
	             * @classdesc Represents a MttResponse.
	             * @implements IMttResponse
	             * @constructor
	             * @param {Music2TrackProtocol.IMttResponse=} [properties] Properties to set
	             */
	            function MttResponse(properties) {
	                if (properties)
	                    for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
	                        if (properties[keys[i]] != null)
	                            this[keys[i]] = properties[keys[i]];
	            }
	    
	            /**
	             * MttResponse decoderResultField.
	             * @member {Music2TrackProtocol.IDecoderResult|null|undefined} decoderResultField
	             * @memberof Music2TrackProtocol.MttResponse
	             * @instance
	             */
	            MttResponse.prototype.decoderResultField = null;
	    
	            /**
	             * MttResponse errorResponse.
	             * @member {Music2TrackProtocol.IErrorResponse|null|undefined} errorResponse
	             * @memberof Music2TrackProtocol.MttResponse
	             * @instance
	             */
	            MttResponse.prototype.errorResponse = null;
	    
	            // OneOf field names bound to virtual getters and setters
	            var $oneOfFields;
	    
	            /**
	             * MttResponse MessageType.
	             * @member {"decoderResultField"|"errorResponse"|undefined} MessageType
	             * @memberof Music2TrackProtocol.MttResponse
	             * @instance
	             */
	            Object.defineProperty(MttResponse.prototype, "MessageType", {
	                get: $util.oneOfGetter($oneOfFields = ["decoderResultField", "errorResponse"]),
	                set: $util.oneOfSetter($oneOfFields)
	            });
	    
	            /**
	             * Decodes a MttResponse message from the specified reader or buffer.
	             * @function decode
	             * @memberof Music2TrackProtocol.MttResponse
	             * @static
	             * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
	             * @param {number} [length] Message length if known beforehand
	             * @returns {Music2TrackProtocol.MttResponse} MttResponse
	             * @throws {Error} If the payload is not a reader or valid buffer
	             * @throws {$protobuf.util.ProtocolError} If required fields are missing
	             */
	            MttResponse.decode = function decode(reader, length) {
	                if (!(reader instanceof $Reader))
	                    reader = $Reader.create(reader);
	                var end = length === undefined ? reader.len : reader.pos + length, message = new $root.Music2TrackProtocol.MttResponse();
	                while (reader.pos < end) {
	                    var tag = reader.uint32();
	                    switch (tag >>> 3) {
	                    case 1: {
	                            message.decoderResultField = $root.Music2TrackProtocol.DecoderResult.decode(reader, reader.uint32());
	                            break;
	                        }
	                    case 2: {
	                            message.errorResponse = $root.Music2TrackProtocol.ErrorResponse.decode(reader, reader.uint32());
	                            break;
	                        }
	                    default:
	                        reader.skipType(tag & 7);
	                        break;
	                    }
	                }
	                return message;
	            };
	    
	            return MttResponse;
	        })();
	    
	        return Music2TrackProtocol;
	    })();

	    return $root;
	}); 
} (mtt));

var mttExports = mtt.exports;

var createClient = function (protocol, provideMeta, _a) {
    if (provideMeta === void 0) { provideMeta = undefined; }
    var getVoiceMeta = _a.getVoiceMeta;
    var _b = createNanoEvents(), on = _b.on, emit = _b.emit;
    /**          */
    var waitForAnswer = function (messageId) {
        return new Promise(function (resolve) {
            var off = on('systemMessage', function (systemMessageData, originalMessage) {
                if (originalMessage.messageId === messageId &&
                    (originalMessage.messageName === MessageNames.ANSWER_TO_USER ||
                        originalMessage.messageName === MessageNames.DO_NOTHING)) {
                    off();
                    resolve(systemMessageData);
                }
            });
        });
    };
    /**   systemMessage,    */
    var sendData = function (data, messageName, meta) {
        if (messageName === void 0) { messageName = ''; }
        var messageId = protocol.getMessageId();
        protocol.sendSystemMessage({
            data: data,
            messageName: messageName,
        }, true, messageId, { meta: meta || {} });
        return messageId;
    };
    /**  cancel   */
    var sendCancel = function (messageId) {
        protocol.sendCancel({}, true, messageId);
    };
    /**   */
    var sendOpenAssistant = function (_a) {
        var _b = _a === void 0 ? { isFirstSession: false } : _a, isFirstSession = _b.isFirstSession;
        return __awaiter(void 0, void 0, void 0, function () {
            var data, meta, _c;
            return __generator(this, function (_d) {
                switch (_d.label) {
                    case 0:
                        data = isFirstSession ? { is_first_session: true } : {};
                        if (!provideMeta) return [3 /*break*/, 2];
                        return [4 /*yield*/, provideMeta()];
                    case 1:
                        _c = _d.sent();
                        return [3 /*break*/, 3];
                    case 2:
                        _c = undefined;
                        _d.label = 3;
                    case 3:
                        meta = _c;
                        return [2 /*return*/, waitForAnswer(sendData(data, 'OPEN_ASSISTANT', meta))];
                }
            });
        });
    };
    /**  sendSystemMessage,    */
    var sendMeta = function (sendSystemMessage, additionalMeta) { return __awaiter(void 0, void 0, void 0, function () {
        var meta, _a;
        return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    if (!provideMeta) return [3 /*break*/, 2];
                    return [4 /*yield*/, provideMeta(additionalMeta)];
                case 1:
                    _a = _b.sent();
                    return [3 /*break*/, 3];
                case 2:
                    _a = {};
                    _b.label = 3;
                case 3:
                    meta = _a;
                    if (typeof meta !== 'undefined') {
                        sendSystemMessage({
                            data: {},
                            messageName: '',
                        }, false, {
                            meta: meta,
                        });
                    }
                    return [2 /*return*/];
            }
        });
    }); };
    /**     messageId */
    var sendMute = function (messageId) {
        protocol.sendMute({}, true, messageId);
    };
    /**  server_action   */
    var sendServerAction = function (serverAction, appInfo, messageName, mode) {
        if (messageName === void 0) { messageName = 'SERVER_ACTION'; }
        return __awaiter(void 0, void 0, void 0, function () {
            var messageId;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        messageId = protocol.getMessageId();
                        //   server_action    systemMessage
                        return [4 /*yield*/, sendMeta(function (data, _, _a) {
                                var _b = _a === void 0 ? {} : _a, meta = _b.meta;
                                var systemData = __rest(data, []);
                                protocol.sendSystemMessage({
                                    // eslint-disable-next-line camelcase
                                    data: __assign(__assign({}, systemData), { app_info: appInfo, server_action: serverAction, mode: mode }),
                                    messageName: messageName || 'SERVER_ACTION',
                                }, true, messageId, { meta: meta });
                            }, {
                                source: {
                                    sourceType: 'vps',
                                },
                            })];
                    case 1:
                        //   server_action    systemMessage
                        _a.sent();
                        return [2 /*return*/, messageId];
                }
            });
        });
    };
    /**      */
    var sendText = function (text, isSsml, shouldSendDisableDubbing, additionalMeta) {
        if (isSsml === void 0) { isSsml = false; }
        return __awaiter(void 0, void 0, void 0, function () {
            return __generator(this, function (_a) {
                if (text.trim() === '') {
                    return [2 /*return*/, undefined];
                }
                return [2 /*return*/, protocol.batch(function (_a) {
                        var sendSystemMessage = _a.sendSystemMessage, clientSendText = _a.sendText, sendSettings = _a.sendSettings, messageId = _a.messageId;
                        return __awaiter(void 0, void 0, void 0, function () {
                            var prevDubbing, sendDisableDubbing, isStillNeedReturnDubbing;
                            return __generator(this, function (_b) {
                                switch (_b.label) {
                                    case 0: return [4 /*yield*/, sendMeta(sendSystemMessage, additionalMeta)];
                                    case 1:
                                        _b.sent();
                                        prevDubbing = protocol.configuration.settings.dubbing;
                                        sendDisableDubbing = prevDubbing !== -1 && shouldSendDisableDubbing;
                                        if (!sendDisableDubbing) return [3 /*break*/, 3];
                                        return [4 /*yield*/, sendSettings({ dubbing: -1 }, false)];
                                    case 2:
                                        _b.sent();
                                        _b.label = 3;
                                    case 3:
                                        isSsml ? clientSendText(text, {}, 'application/ssml') : clientSendText(text, {});
                                        isStillNeedReturnDubbing = prevDubbing === protocol.configuration.settings.dubbing;
                                        if (sendDisableDubbing && isStillNeedReturnDubbing) {
                                            sendSettings({ dubbing: prevDubbing });
                                        }
                                        return [2 /*return*/, messageId];
                                }
                            });
                        });
                    })];
            });
        });
    };
    /**    , .   callback    ,
     *   */
    var createVoiceStream = function (callback, additionalMeta) {
        return protocol.batch(function (_a) {
            var sendSystemMessage = _a.sendSystemMessage, sendVoice = _a.sendVoice, messageId = _a.messageId;
            return __awaiter(void 0, void 0, void 0, function () {
                var first;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            first = true;
                            return [4 /*yield*/, sendMeta(sendSystemMessage, additionalMeta)];
                        case 1:
                            _b.sent();
                            return [4 /*yield*/, callback({
                                    sendVoice: function () {
                                        var args = [];
                                        for (var _i = 0; _i < arguments.length; _i++) {
                                            args[_i] = arguments[_i];
                                        }
                                        if (first) {
                                            //        
                                            sendVoice(args[0], args[1], args[2], __assign(__assign({}, args[3]), { meta: getVoiceMeta() }));
                                            first = false;
                                            return;
                                        }
                                        sendVoice.apply(void 0, args);
                                    },
                                    messageId: messageId,
                                })];
                        case 2:
                            _b.sent();
                            return [2 /*return*/];
                    }
                });
            });
        });
    };
    var off = protocol.on('incoming', function (message) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j;
        if ((_a = message.systemMessage) === null || _a === void 0 ? void 0 : _a.data) {
            emit('systemMessage', JSON.parse(message.systemMessage.data), message);
        }
        if (message.voice) {
            emit('voice', message.voice.data || new Uint8Array(), message);
        }
        if (message.status) {
            emit('status', message.status, message);
        }
        if (message.messageName === 'TAKE_HISTORY' && ((_b = message.bytes) === null || _b === void 0 ? void 0 : _b.data)) {
            var history_1 = protoExports.GetHistoryResponse.decode((_c = message.bytes) === null || _c === void 0 ? void 0 : _c.data).historyMessages;
            var parsedHistory = history_1.map(function (historyMessage) { return (__assign(__assign({}, historyMessage), { content: JSON.parse(historyMessage.content || '') })); });
            emit('history', parsedHistory, message);
        }
        if (message.messageName === MessageNames.STT && (message.text || ((_e = (_d = message.bytes) === null || _d === void 0 ? void 0 : _d.data) === null || _e === void 0 ? void 0 : _e.length))) {
            var response = ((_g = (_f = message.bytes) === null || _f === void 0 ? void 0 : _f.data) === null || _g === void 0 ? void 0 : _g.length)
                ? asrExports.PacketWrapperFromServer.decode(message.bytes.data)
                : undefined;
            emit('stt', { text: message.text, response: response }, message);
        }
        if (message.messageName === MessageNames.MTT && ((_j = (_h = message.bytes) === null || _h === void 0 ? void 0 : _h.data) === null || _j === void 0 ? void 0 : _j.length)) {
            emit('musicRecognition', mttExports.Music2TrackProtocol.MttResponse.decode(message.bytes.data), message);
        }
    });
    return {
        destroy: function () {
            off();
        },
        init: protocol.init,
        createVoiceStream: createVoiceStream,
        sendData: sendData,
        sendMeta: sendMeta,
        sendOpenAssistant: sendOpenAssistant,
        sendServerAction: sendServerAction,
        sendText: sendText,
        sendCancel: sendCancel,
        sendMute: sendMute,
        on: on,
        waitForAnswer: waitForAnswer,
    };
};

var RETRY_INTERVAL = 300; // ms
var defaultWSCreator = function (url) { return new WebSocket(url); };
var createTransport = function (_a) {
    var _b = _a.createWS, createWS = _b === void 0 ? defaultWSCreator : _b, checkCertUrl = _a.checkCertUrl;
    var _c = createNanoEvents(), on = _c.on, emit = _c.emit;
    var hasCert = !checkCertUrl;
    var retryTimeoutId = -1;
    var retries = 0;
    var status = 'closed';
    var webSocket;
    var stopped = true;
    var checkCert = function (checkUrl) {
        return new Promise(function (resolve) {
            window
                .fetch(checkUrl)
                .then(function () { return resolve(true); })
                .catch(function () { return resolve(false); });
        });
    };
    var close = function () {
        stopped = true;
        if (status === 'closing' || status === 'closed') {
            return;
        }
        status = 'closing';
        webSocket === null || webSocket === void 0 ? void 0 : webSocket.close();
    };
    var connect = function (url) { return __awaiter(void 0, void 0, void 0, function () {
        var okay;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    status = 'connecting';
                    emit('connecting');
                    if (!(!hasCert && window.navigator.onLine)) return [3 /*break*/, 2];
                    return [4 /*yield*/, checkCert(checkCertUrl)];
                case 1:
                    okay = _a.sent();
                    if (!okay) {
                        status = 'closed';
                        emit('close');
                        emit('error', new Error('Cert authority invalid'));
                        return [2 /*return*/];
                    }
                    hasCert = true;
                    _a.label = 2;
                case 2:
                    webSocket = createWS(url);
                    webSocket.binaryType = 'arraybuffer';
                    webSocket.addEventListener('open', function () {
                        if (webSocket.readyState !== 1) {
                            return;
                        }
                        window.clearTimeout(retryTimeoutId);
                        retries = 0;
                        status = 'open';
                        emit('open');
                    });
                    webSocket.addEventListener('close', function () {
                        status = 'closed';
                        emit('close');
                    });
                    webSocket.addEventListener('error', function (e) {
                        if (status !== 'connecting') {
                            throw e;
                        }
                        //  ,     
                        if (!webSocket || (webSocket.readyState === 3 && !stopped)) {
                            window.clearTimeout(retryTimeoutId);
                            if (retries < 2) {
                                retryTimeoutId = window.setTimeout(function () {
                                    // eslint-disable-next-line @typescript-eslint/no-use-before-define
                                    open(url);
                                    retries++;
                                }, RETRY_INTERVAL * retries);
                            }
                            else {
                                retries = 0;
                                emit('error', e);
                            }
                        }
                    });
                    webSocket.addEventListener('message', function (_a) {
                        var data = _a.data;
                        emit('message', data);
                    });
                    return [2 /*return*/];
            }
        });
    }); };
    var open = function (url) {
        if (status === 'connecting' || status === 'open') {
            return;
        }
        stopped = false;
        connect(url);
    };
    var reconnect = function (url) {
        if (status === 'closed') {
            open(url);
            return;
        }
        window.setTimeout(function () { return reconnect(url); });
        close();
    };
    var send = function (data) {
        if (!window.navigator.onLine) {
            close();
            emit('error');
            throw new Error('The client seems to be offline');
        }
        webSocket.send(data);
    };
    return {
        close: close,
        get isOnline() {
            return window.navigator.onLine;
        },
        on: on,
        open: open,
        reconnect: reconnect,
        send: send,
    };
};

var convertToMetaPermissions = function (permission) {
    return Object.keys(permission).map(function (key) { return ({
        type: key,
        status: permission[key],
    }); });
};
var getLocation = function () { return __awaiter(void 0, void 0, void 0, function () {
    return __generator(this, function (_a) {
        return [2 /*return*/, new Promise(function (resolve, reject) {
                navigator.geolocation.getCurrentPosition(function (_a) {
                    var coords = _a.coords, timestamp = _a.timestamp;
                    resolve({
                        lat: coords.latitude,
                        lon: coords.longitude,
                        accuracy: coords.accuracy,
                        timestamp: timestamp,
                    });
                }, reject, { timeout: 5000 });
            })];
    });
}); };
var getTime = function () { return ({
    //   , .. `Intl.DateTimeFormat().resolvedOptions().timeZone` -   
    timezone_id: Intl.DateTimeFormat().resolvedOptions().timeZone,
    timezone_offset_sec: -new Date().getTimezoneOffset() * 60,
    timestamp: Date.now(),
}); };
var getAnswerForRequestPermissions = function (requestMessageId, appInfo, items) { return __awaiter(void 0, void 0, void 0, function () {
    var permissions, response;
    return __generator(this, function (_a) {
        permissions = {
            record_audio: 'denied_once',
            geo: 'denied_once',
            read_contacts: 'denied_permanently',
            push: 'denied_once',
        };
        response = {
            app_info: appInfo,
            meta: {
                time: getTime(),
                permissions: [],
            },
            server_action: {
                action_id: 'command_response',
                request_message_id: requestMessageId,
                command_response: {
                    request_permissions: {
                        permissions: [],
                    },
                },
            },
        };
        return [2 /*return*/, Promise.all(items.map(function (permission) { return __awaiter(void 0, void 0, void 0, function () {
                var _a, location_1;
                var _c, _d, _e;
                return __generator(this, function (_f) {
                    switch (_f.label) {
                        case 0:
                            _a = permission;
                            switch (_a) {
                                case 'geo': return [3 /*break*/, 1];
                            }
                            return [3 /*break*/, 5];
                        case 1:
                            _f.trys.push([1, 3, , 4]);
                            return [4 /*yield*/, getLocation()];
                        case 2:
                            location_1 = _f.sent();
                            permissions.geo = 'granted';
                            response.meta.location = location_1;
                            (_c = response.server_action.command_response.request_permissions) === null || _c === void 0 ? void 0 : _c.permissions.push({
                                type: 'geo',
                                status: 'granted',
                            });
                            return [3 /*break*/, 4];
                        case 3:
                            _f.sent();
                            permissions.geo = 'denied_permanently';
                            (_d = response.server_action.command_response.request_permissions) === null || _d === void 0 ? void 0 : _d.permissions.push({
                                type: 'geo',
                                status: 'denied_permanently',
                            });
                            return [3 /*break*/, 4];
                        case 4: return [3 /*break*/, 6];
                        case 5:
                            //    
                            (_e = response.server_action.command_response.request_permissions) === null || _e === void 0 ? void 0 : _e.permissions.push({
                                type: permission,
                                status: 'denied_permanently',
                            });
                            return [3 /*break*/, 6];
                        case 6: return [2 /*return*/];
                    }
                });
            }); })).then(function () {
                response.meta.permissions = convertToMetaPermissions(permissions);
                return response;
            })];
    });
}); };

var IS_APPLE_MOBILE = typeof window !== 'undefined'
    ? (/iPad|iPhone|iPod/.test(navigator.platform) ||
        (navigator.platform === 'MacIntel' && navigator.maxTouchPoints > 1)) &&
        !window.MSStream
    : false;
function createSilentAudioFile(sampleRate) {
    var arrayBuffer = new ArrayBuffer(10);
    var dataView = new DataView(arrayBuffer);
    dataView.setUint32(0, sampleRate, true);
    dataView.setUint32(4, sampleRate, true);
    dataView.setUint16(8, 1, true);
    var missingCharacters = window
        .btoa(String.fromCharCode.apply(null, new Uint8Array(arrayBuffer)))
        .slice(0, 13);
    return "data:audio/wav;base64,UklGRisAAABXQVZFZm10IBAAAAABAAEA" + missingCharacters + "AgAZGF0YQcAAACAgICAgICAAAA=";
}
/**
 *  ,       silent mode.
 *     initialize()
 *  <audio>   ,  turnOn/turnOff.
 * @returns object
 */
var createIosSilentModePatch = function () {
    var audio = null;
    var destroy = function () {
        if (audio === null) {
            return;
        }
        audio.src = 'about:blank';
        audio.load();
        audio = null;
    };
    /**
     *  ,
     *        (click)
     */
    var initialize = function () {
        if (!IS_APPLE_MOBILE) {
            return;
        }
        destroy();
        audio = new Audio();
        audio.setAttribute('x-webkit-airplay', 'deny');
        audio.preload = 'auto';
        audio.loop = true;
        audio.src = createSilentAudioFile(16000);
        audio.load();
    };
    var turnOff = function () {
        audio === null || audio === void 0 ? void 0 : audio.pause();
    };
    return {
        destroy: destroy,
        initialize: initialize,
        turnOff: turnOff,
        turnOn: function () {
            audio === null || audio === void 0 ? void 0 : audio.play().catch(function (e) {
                console.error('ios audio patch excepted', e);
            });
        },
        get isActive() {
            return (audio === null || audio === void 0 ? void 0 : audio.paused) === false;
        },
    };
};
var iosSilentModePatch = createIosSilentModePatch();

var isAudioSupported = typeof window !== 'undefined' && (window.AudioContext || window.webkitAudioContext);
/**
 *    AudioContext  
 * @param options AudioContextOptions
 * @returns AudioContext
 */
function createAudioContext(options) {
    if (window.AudioContext) {
        return new AudioContext(options);
    }
    if (window.webkitAudioContext) {
        // eslint-disable-next-line new-cap
        return new window.webkitAudioContext();
    }
    throw new Error('Audio not supported');
}
var _a = createNanoEvents(), on = _a.on, emit = _a.emit;
var audioContext;
/**
 *      , ,    , AudioContext.
 *       AudioContext
 * @param onReady ,      AudioContext
 */
var resolveAudioContext = function (onReady) {
    if (!audioContext) {
        var isSafari_1 = navigator.vendor.search('Apple') >= 0;
        var context_1 = createAudioContext();
        audioContext = {
            context: context_1,
            ready: !isSafari_1 && context_1.state === 'running',
            on: on,
        };
        context_1.onstatechange = function () {
            if (audioContext.context.state === 'running') {
                audioContext.ready = true;
                emit('ready');
            }
        };
        ///          
        ///   -     -   
        if (!audioContext.ready) {
            var handleClick_1 = function () {
                document.removeEventListener('click', handleClick_1);
                iosSilentModePatch.initialize();
                if (isSafari_1) {
                    ///  , .  - ,
                    ///           
                    var oscillator = audioContext.context.createOscillator();
                    oscillator.frequency.value = 0;
                    oscillator.connect(audioContext.context.destination);
                    oscillator.start(0);
                    oscillator.stop(0.5);
                }
                if (audioContext.context.state === 'suspended') {
                    /// Developers who write games, WebRTC applications, or other websites that use the Web Audio API
                    /// should call context.resume() after the first user gesture (e.g. a click, or tap)
                    /// https://sites.google.com/a/chromium.org/dev/audio-video/autoplay
                    audioContext.context.resume();
                }
            };
            ///       ( ),
            ///     (  touch)
            document.addEventListener('click', handleClick_1);
        }
    }
    if (audioContext.ready) {
        onReady && onReady(audioContext.context, function () { return iosSilentModePatch.destroy(); });
    }
    else {
        var unsubscribe_1 = on('ready', function () {
            onReady(audioContext.context, function () { return iosSilentModePatch.destroy(); });
            unsubscribe_1();
        });
    }
};

/**
 *  sample rate c inSampleRate   outSampleRate   Float32Array  ArrayBuffer
 * @param buffer 
 * @param inSampleRate  sample rate
 * @param outSampleRate  sample rate
 * @returns    sample rate = outSampleRate
 */
var downsampleBuffer = function (buffer, inSampleRate, outSampleRate) {
    if (outSampleRate > inSampleRate) {
        throw new Error('downsampling rate show be smaller than original sample rate');
    }
    var sampleRateRatio = inSampleRate / outSampleRate;
    var newLength = Math.round(buffer.length / sampleRateRatio);
    var result = new Int16Array(newLength);
    var offsetResult = 0;
    var offsetBuffer = 0;
    while (offsetResult < result.length) {
        var nextOffsetBuffer = Math.round((offsetResult + 1) * sampleRateRatio);
        var accum = 0;
        var count = 0;
        for (var i = offsetBuffer; i < nextOffsetBuffer && i < buffer.length; i++) {
            accum += buffer[i];
            count++;
        }
        result[offsetResult] = Math.min(1, accum / count) * 0x7fff;
        offsetResult++;
        offsetBuffer = nextOffsetBuffer;
    }
    return result.buffer;
};
var TARGET_SAMPLE_RATE = 16000;
var IS_FIREFOX = typeof window !== 'undefined' && navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
var IS_SAFARI = typeof window !== 'undefined' && /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
var context;
var processor;
var analyser = null;
/**
 *  stream   (),     cb,
 *   ,      
 * @param stream -
 * @param cb callback,      
 * @returns ,     
 */
var createAudioRecorder = function (stream, cb, useAnalyser) {
    return new Promise(function (resolve) {
        var state = 'inactive';
        var input;
        var stop = function () {
            if (state === 'inactive') {
                return;
            }
            state = 'inactive';
            stream.getTracks().forEach(function (track) {
                track.stop();
            });
            input.disconnect();
        };
        var start = function () {
            if (state !== 'inactive') {
                throw new Error("Can't start not inactive recorder");
            }
            state = 'recording';
            if (!context) {
                context = createAudioContext({
                    // firefox    samplerate,    
                    sampleRate: IS_FIREFOX ? undefined : TARGET_SAMPLE_RATE,
                });
            }
            input = context.createMediaStreamSource(stream);
            if (!processor) {
                processor = context.createScriptProcessor(2048, 1, 1);
            }
            if (!analyser && useAnalyser) {
                analyser = context.createAnalyser();
            }
            var listener = function (e) {
                var buffer = e.inputBuffer.getChannelData(0);
                var data = downsampleBuffer(buffer, context.sampleRate, TARGET_SAMPLE_RATE);
                var last = state === 'inactive';
                //       voiceData > 0, ..
                //  safari      
                if (!IS_SAFARI || new Uint8Array(data).some(function (voiceData) { return voiceData > 0; })) {
                    var analyserArray = null;
                    if (analyser) {
                        analyserArray = new Uint8Array(analyser.frequencyBinCount);
                        analyser === null || analyser === void 0 ? void 0 : analyser.getByteTimeDomainData(analyserArray);
                    }
                    cb(data, analyserArray, last);
                    resolve(stop);
                }
                if (last) {
                    processor.removeEventListener('audioprocess', listener);
                }
            };
            processor.addEventListener('audioprocess', listener);
            input.connect(processor);
            if (analyser) {
                input.connect(analyser);
            }
            processor.connect(context.destination);
        };
        start();
    });
};
/**
 *         Promise,   .
 *   ,       cb -       .
 * @param cb Callback,       
 * @returns Promise,     
 */
var createNavigatorAudioProvider = function (cb, useAnalyser) {
    return navigator.mediaDevices
        .getUserMedia({
        audio: true,
    })
        .then(function (stream) {
        return createAudioRecorder(stream, cb, useAnalyser);
    })
        .catch(function (err) {
        if (window.location.protocol === 'http:') {
            throw new Error('Audio is supported only on a secure connection');
        }
        throw err;
    });
};

/**
 *  ,        .
 * @param createAudioProvider  
 * @returns Api     
 */
var createVoiceListener = function (createAudioProvider) {
    if (createAudioProvider === void 0) { createAudioProvider = createNavigatorAudioProvider; }
    var _a = createNanoEvents(), emit = _a.emit, on = _a.on;
    var stopRecord;
    var status = 'stopped';
    var cancelableToken = { current: false };
    var stop = function () {
        cancelableToken.current = true;
        cancelableToken = { current: false };
        status = 'stopped';
        stopRecord === null || stopRecord === void 0 ? void 0 : stopRecord();
        emit('status', 'stopped');
    };
    var listen = function (handleVoice) {
        cancelableToken = { current: false };
        var capturedToken = cancelableToken;
        status = 'started';
        emit('status', 'started');
        return createAudioProvider(function (data, analyser, last) { return handleVoice(new Uint8Array(data), analyser, last); })
            .then(function (recStop) {
            stopRecord = recStop;
        })
            .then(function () {
            if (capturedToken.current === true || status === 'stopped') {
                stopRecord();
            }
            else {
                status = 'listen';
                emit('status', 'listen');
            }
        })
            .catch(function (err) {
            status = 'stopped';
            emit('status', 'stopped');
            throw err;
        });
    };
    return {
        listen: listen,
        stop: stop,
        on: on,
        get status() {
            return status;
        },
    };
};

/**     */
var createTrackCollection = function () {
    var trackIds;
    var trackMap;
    var clear = function () {
        trackIds = new Array();
        trackMap = new Map();
    };
    var push = function (id, track) {
        if (trackMap.has(id)) {
            throw new Error('Track already exists');
        }
        trackMap.set(id, track);
        trackIds.push(id);
    };
    var has = function (id) { return trackMap.has(id); };
    var getById = function (id) {
        var track = trackMap.get(id);
        if (track === undefined) {
            throw new Error('Unknown track id');
        }
        return track;
    };
    var getByIndex = function (index) {
        if (index < 0 || index >= trackIds.length) {
            throw new Error('Index out of bounds');
        }
        var track = trackMap.get(trackIds[index]);
        if (track == null) {
            throw new Error('Something wrong...');
        }
        return track;
    };
    var some = function (predicate) { return trackIds.some(function (id) { return predicate(getById(id)); }); };
    clear();
    return {
        clear: clear,
        has: has,
        get: getById,
        getByIndex: getByIndex,
        push: push,
        some: some,
        get length() {
            return trackIds.length;
        },
    };
};

/**          */
var createChunkQueue = function () {
    var chunks = []; //  
    /**      */
    var push = function (chunk) {
        chunks.push(chunk);
    };
    /**      */
    var remove = function (chunk) {
        chunks.splice(chunks.indexOf(chunk), 1);
    };
    return {
        get chunks() {
            return chunks;
        },
        remove: remove,
        push: push,
        get length() {
            return chunks.length;
        },
        get ended() {
            //   ,   
            return chunks.length === 0;
        },
    };
};

var HZ_BYTES_COUNT = 2;
var from16BitToFloat32 = function (incomingData) {
    var l = incomingData.length;
    var outputData = new Float32Array(l);
    for (var i = 0; i < l; i += 1) {
        outputData[i] = incomingData[i] / 32768.0;
    }
    return outputData;
};
/**    ,     */
var createTrackStream = function (ctx, _a) {
    var _b = _a === void 0 ? {} : _a, _c = _b.sampleRate, sampleRate = _c === void 0 ? 24000 : _c, _d = _b.numberOfChannels, numberOfChannels = _d === void 0 ? 1 : _d, _e = _b.delay, delay = _e === void 0 ? 0 : _e, onPlay = _b.onPlay, onEnd = _b.onEnd, onStop = _b.onStop, trackStatus = _b.trackStatus;
    //    () 
    var queue = createChunkQueue();
    var buffer = new ArrayBuffer(0);
    var extraByte = null;
    var status = trackStatus || 'stop';
    var lastChunkOffset = 0;
    var startTime = 0;
    var firstChunk = true;
    var loaded = false;
    var end = function () {
        //      
        queue.chunks.forEach(function (chunk) {
            chunk.stop();
        });
        status = 'end';
        onEnd && onEnd();
        startTime = 0;
        lastChunkOffset = 0;
    };
    var stop = function () {
        onStop === null || onStop === void 0 ? void 0 : onStop();
        end();
    };
    var play = function () {
        var _a;
        if (status === 'end') {
            return;
        }
        if (status !== 'play') {
            status = 'play';
            onPlay && onPlay();
        }
        if (loaded && queue.ended) {
            end();
            return;
        }
        //  ,         
        if (loaded || buffer.byteLength / (sampleRate * HZ_BYTES_COUNT) >= delay) {
            if (buffer.byteLength < 1) {
                return;
            }
            var chunk = getChunkFromBuffer();
            startTime = queue.length === 0 ? ctx.currentTime : startTime;
            queue.push(chunk);
            chunk.start(startTime + lastChunkOffset);
            lastChunkOffset += ((_a = chunk.buffer) === null || _a === void 0 ? void 0 : _a.duration) || 0;
        }
    };
    /**       */
    var getExtraBytes = function (data, bytesArraysSizes) {
        if (extraByte == null && bytesArraysSizes.incomingMessageVoiceDataLength % 2) {
            extraByte = data[bytesArraysSizes.incomingMessageVoiceDataLength - 1];
            bytesArraysSizes.incomingMessageVoiceDataLength -= 1;
            bytesArraysSizes.sourceLen -= 1;
        }
        else if (extraByte != null) {
            bytesArraysSizes.prepend = extraByte;
            bytesArraysSizes.start = 1;
            if (bytesArraysSizes.incomingMessageVoiceDataLength % 2) {
                bytesArraysSizes.incomingMessageVoiceDataLength += 1;
                extraByte = null;
            }
            else {
                extraByte = data[bytesArraysSizes.incomingMessageVoiceDataLength - 1];
                bytesArraysSizes.sourceLen -= 1;
            }
        }
    };
    var createChunk = function (chunk) {
        var audioBuffer = ctx.createBuffer(numberOfChannels, chunk.length / numberOfChannels, sampleRate);
        for (var i = 0; i < numberOfChannels; i++) {
            var channelChunk = new Float32Array(chunk.length / numberOfChannels);
            var index = 0;
            for (var j = i; j < chunk.length; j += numberOfChannels) {
                channelChunk[index++] = chunk[j];
            }
            audioBuffer.getChannelData(i).set(channelChunk);
        }
        var source = ctx.createBufferSource();
        source.buffer = audioBuffer;
        source.connect(ctx.destination);
        source.onended = function () {
            queue.remove(source);
            if (queue.ended && status !== 'end') {
                status = 'end';
                onEnd && onEnd();
            }
        };
        return source;
    };
    /**     */
    var getChunkFromBuffer = function () {
        var tmp = buffer;
        buffer = new ArrayBuffer(0);
        var data = new Uint8Array(tmp);
        var bytesArraysSizes = {
            incomingMessageVoiceDataLength: data.length,
            sourceLen: data.length,
            start: 0,
            prepend: null,
        };
        //    
        getExtraBytes(data, bytesArraysSizes);
        var dataBuffer = new ArrayBuffer(bytesArraysSizes.incomingMessageVoiceDataLength);
        var bufferUi8 = new Uint8Array(dataBuffer);
        var bufferI16 = new Int16Array(dataBuffer);
        bufferUi8.set(data.slice(0, bytesArraysSizes.sourceLen), bytesArraysSizes.start);
        if (bytesArraysSizes.prepend != null) {
            bufferUi8[0] = bytesArraysSizes.prepend;
        }
        return createChunk(from16BitToFloat32(bufferI16));
    };
    /**       */
    var write = function (data) {
        // 44  -  
        var slicePoint = firstChunk ? 44 : 0;
        firstChunk = false;
        if (slicePoint >= data.length) {
            return;
        }
        var tmp = new Uint8Array(buffer.byteLength + data.length - slicePoint);
        tmp.set(new Uint8Array(buffer), 0);
        tmp.set(slicePoint ? data.slice(slicePoint) : data, buffer.byteLength);
        buffer = tmp;
        if (status === 'play') {
            play();
        }
    };
    return {
        get loaded() {
            return loaded;
        },
        setLoaded: function () {
            loaded = true;
            if (status === 'play') {
                play();
            }
        },
        write: write,
        get status() {
            return status;
        },
        play: play,
        stop: stop,
    };
};

var createVoicePlayer = function (actx, _a) {
    var _b = _a === void 0 ? {} : _a, _c = _b.startVoiceDelay, startVoiceDelay = _c === void 0 ? 0.2 : _c, sampleRate = _b.sampleRate, numberOfChannels = _b.numberOfChannels;
    var _d = createNanoEvents(), on = _d.on, emit = _d.emit;
    var tracks = createTrackCollection();
    // true -      (   ), false -    (  ..)
    var active = true;
    //     tracks
    var cursor = 0;
    var play = function () {
        if (cursor >= tracks.length) {
            if (tracks.some(function (track) { return !track.loaded; })) {
                return;
            }
            iosSilentModePatch.turnOff();
            //  ,     
            cursor = 0;
            tracks.clear();
            return;
        }
        //   silent mode ios
        if (!iosSilentModePatch.isActive) {
            iosSilentModePatch.turnOn();
        }
        //      
        var current = tracks.getByIndex(cursor);
        if (current.status === 'end') {
            if (cursor < tracks.length) {
                cursor++;
                play();
            }
        }
        else {
            current.play();
        }
    };
    var append = function (data, trackId, last) {
        if (last === void 0) { last = false; }
        var current = tracks.has(trackId) ? tracks.get(trackId) : undefined;
        if (current == null) {
            ///  trackId    -  
            ///    -   ,  play
            current = createTrackStream(actx, {
                sampleRate: sampleRate,
                numberOfChannels: numberOfChannels,
                delay: startVoiceDelay,
                onPlay: function () { return emit('play', trackId); },
                onStop: function () { return emit('stop', trackId); },
                onEnd: function () {
                    emit('end', trackId);
                    play();
                },
                trackStatus: active ? 'stop' : 'end',
            });
            tracks.push(trackId, current);
        }
        if (current.status !== 'end' && data.length) {
            current.write(data);
        }
        if (last) {
            //    
            current.setLoaded();
        }
        play();
    };
    var stop = function () {
        while (cursor < tracks.length) {
            var cur = cursor;
            cursor++;
            tracks.getByIndex(cur).stop();
        }
        iosSilentModePatch.turnOff();
    };
    return {
        append: append,
        setActive: function (value) {
            active = value;
            if (value) {
                play();
            }
            else {
                stop();
            }
        },
        on: on,
        stop: stop,
    };
};

/**   */
var filterEmptyChunks = function (chunksOriginal) {
    return chunksOriginal.reduce(function (acc, chunkOriginal) {
        var chunk = chunkOriginal.filter(function (int) { return int; });
        if (chunk.length) {
            acc.push(chunk);
        }
        return acc;
    }, []);
};
var createVoice = function (client, settings, emit, 
///  onReady  ,   
///   onReady,    
onReady, useAnalyser) {
    var voicePlayer;
    var listener = createVoiceListener(function (cb) { return createNavigatorAudioProvider(cb, useAnalyser); });
    var subscriptions = [];
    var appInfoDict = {};
    var mesIdQueue = [];
    /**     */
    var isRecognizeInitializing = false;
    /** /   */
    var isPlaying = false;
    /** id ,   ,    */
    var autolistenMessageId = null;
    /** id   ,     */
    var currentVoiceMessageId = null;
    /**   .   ,     */
    var streaming = null;
    /**  - */
    var destroyAudioContext = null;
    /**   ,  cancel.  true -     */
    var stopVoice = function (sendCancel) {
        if (sendCancel === void 0) { sendCancel = true; }
        autolistenMessageId = null;
        streaming = null;
        if (sendCancel && currentVoiceMessageId) {
            client.sendCancel(currentVoiceMessageId);
        }
        currentVoiceMessageId = null;
        if (listener.status === 'listen') {
            listener.stop();
            return true;
        }
        return false;
    };
    /**     */
    var stop = function () {
        //     
        stopVoice();
        voicePlayer === null || voicePlayer === void 0 ? void 0 : voicePlayer.stop();
    };
    var recognize = function (_a) {
        var _b = _a === void 0 ? {} : _a, begin = _b.begin, messageName = _b.messageName, _c = _b.isAutoListening, isAutoListening = _c === void 0 ? false : _c;
        return __awaiter(void 0, void 0, void 0, function () {
            var unsubscribe_1;
            return __generator(this, function (_d) {
                switch (_d.label) {
                    case 0:
                        if (stopVoice()) {
                            return [2 /*return*/];
                        }
                        if (isPlaying) {
                            voicePlayer === null || voicePlayer === void 0 ? void 0 : voicePlayer.stop();
                            return [2 /*return*/];
                        }
                        if (settings.current.disableListening) {
                            return [2 /*return*/];
                        }
                        if (!(listener.status === 'stopped' && !isRecognizeInitializing)) return [3 /*break*/, 2];
                        isRecognizeInitializing = true;
                        unsubscribe_1 = listener.on('status', function () {
                            isRecognizeInitializing = false;
                            unsubscribe_1();
                        });
                        return [4 /*yield*/, client.init().catch(function (error) {
                                isRecognizeInitializing = false;
                                throw error;
                            })];
                    case 1:
                        _d.sent();
                        return [2 /*return*/, client.createVoiceStream(function (_a) {
                                var sendVoice = _a.sendVoice, messageId = _a.messageId;
                                begin === null || begin === void 0 ? void 0 : begin.forEach(function (chunk) { return sendVoice(new Uint8Array(chunk), false); });
                                currentVoiceMessageId = messageId;
                                return listener.listen(function (chunk, analyser, last) {
                                    if (analyser) {
                                        emit({ voiceAnalyser: { data: analyser } });
                                    }
                                    sendVoice(chunk, last, messageName);
                                });
                            }, {
                                source: {
                                    sourceType: isAutoListening ? 'autoListening' : 'lavashar',
                                },
                            })];
                    case 2: return [2 /*return*/];
                }
            });
        });
    };
    /**
     *      VPS.
     *   last=true  3    Cancel.
     *        .
     *
     * @param chunks , sampleRate: 16000
     * @param last    ?
     * @param messageName ,    
     */
    var streamVoice = function (chunks, last, messageName) { return __awaiter(void 0, void 0, void 0, function () {
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    chunks = filterEmptyChunks(chunks);
                    if (streaming === null || streaming === void 0 ? void 0 : streaming(chunks, last)) {
                        return [2 /*return*/];
                    }
                    stopVoice();
                    if (isPlaying) {
                        voicePlayer === null || voicePlayer === void 0 ? void 0 : voicePlayer.stop();
                    }
                    if (!(!isRecognizeInitializing && chunks.length)) return [3 /*break*/, 2];
                    isRecognizeInitializing = true;
                    return [4 /*yield*/, client.init()];
                case 1:
                    _a.sent();
                    return [2 /*return*/, client.createVoiceStream(function (_a) {
                            var messageId = _a.messageId, sendVoice = _a.sendVoice;
                            return __awaiter(void 0, void 0, void 0, function () {
                                var cancelTimeoutId;
                                return __generator(this, function (_b) {
                                    cancelTimeoutId = -1;
                                    isRecognizeInitializing = false;
                                    currentVoiceMessageId = messageId;
                                    streaming = function (ch, l) {
                                        clearTimeout(cancelTimeoutId);
                                        ch.forEach(function (chunk) { return sendVoice(new Uint8Array(chunk), l, messageName); });
                                        if (l) {
                                            streaming = null;
                                        }
                                        else {
                                            cancelTimeoutId = setTimeout(function () {
                                                if (streaming) {
                                                    stopVoice();
                                                }
                                            }, 3000);
                                        }
                                        return true;
                                    };
                                    streaming(chunks, last);
                                    return [2 /*return*/];
                                });
                            });
                        }, {
                            source: {
                                sourceType: 'lavashar',
                            },
                        })];
                case 2: return [2 /*return*/];
            }
        });
    }); };
    /**
     *      VPS.
     *    (   last=true).
     *        .
     *
     * @param chunks , sampleRate: 16000
     * @param messageName ,    
     */
    var sendVoice = function (chunks, messageName) { return __awaiter(void 0, void 0, void 0, function () {
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    chunks = filterEmptyChunks(chunks);
                    stopVoice();
                    if (isPlaying) {
                        voicePlayer === null || voicePlayer === void 0 ? void 0 : voicePlayer.stop();
                    }
                    if (!(!isRecognizeInitializing && chunks.length)) return [3 /*break*/, 2];
                    isRecognizeInitializing = true;
                    return [4 /*yield*/, client.init()];
                case 1:
                    _a.sent();
                    return [2 /*return*/, client.createVoiceStream(function (_a) {
                            var messageId = _a.messageId, sendVoiceStream = _a.sendVoice;
                            isRecognizeInitializing = false;
                            currentVoiceMessageId = messageId;
                            chunks.forEach(function (chunk) { return sendVoiceStream(new Uint8Array(chunk), true, messageName); });
                            return Promise.resolve();
                        }, {
                            source: {
                                sourceType: 'lavashar',
                            },
                        })];
                case 2: return [2 /*return*/];
            }
        });
    }); };
    /**
     *   .
     *       - ,      .
     *
     * @param begin  , sampleRate: 16000      
     */
    var listen = function (_a, isAutoListening) {
        var _b = _a === void 0 ? {} : _a, begin = _b.begin;
        return recognize({ begin: begin, isAutoListening: isAutoListening });
    };
    /**
     *   .
     *        
     */
    var shazam = function () {
        return recognize({ messageName: MessageNames.MTT, isAutoListening: false });
    };
    if (isAudioSupported) {
        resolveAudioContext(function (context, destroy) {
            ///      
            ///     AudioContext
            voicePlayer = createVoicePlayer(context, { startVoiceDelay: 1 });
            destroyAudioContext = destroy;
            //   
            subscriptions.push(voicePlayer.on('play', function (mesId) {
                isPlaying = true;
                emit({ emotion: 'talk' });
                emit({ tts: { status: 'start', messageId: Number(mesId), appInfo: appInfoDict[mesId] } });
            }));
            subscriptions.push(voicePlayer.on('stop', function (mesId) {
                client.sendMute(Number(mesId));
            }));
            //   
            subscriptions.push(voicePlayer.on('end', function (mesId) {
                isPlaying = false;
                emit({ emotion: 'idle' });
                emit({ tts: { status: 'stop', messageId: Number(mesId), appInfo: appInfoDict[mesId] } });
                if (mesId === autolistenMessageId) {
                    listen();
                }
                //   appInfo  messageId
                var idx = 0;
                do {
                    delete appInfoDict[mesIdQueue[0]];
                } while (mesIdQueue[idx++] !== mesId && mesIdQueue.length > idx);
                mesIdQueue.splice(0, idx);
            }));
            //      
            onReady === null || onReady === void 0 ? void 0 : onReady();
        });
    }
    subscriptions.push(
    //   
    client.on('voice', function (data, message) {
        if (settings.current.disableDubbing) {
            return;
        }
        voicePlayer === null || voicePlayer === void 0 ? void 0 : voicePlayer.append(data, message.messageId.toString(), message.last === 1);
    }), 
    //   
    listener.on('status', function (status) {
        emit({ listener: { status: status } });
        if (status === 'listen') {
            voicePlayer === null || voicePlayer === void 0 ? void 0 : voicePlayer.setActive(false);
            emit({ emotion: 'listen' });
        }
        else if (status === 'stopped') {
            voicePlayer === null || voicePlayer === void 0 ? void 0 : voicePlayer.setActive(!settings.current.disableDubbing);
            emit({ asr: { text: '' }, emotion: 'idle' });
        }
    }), 
    //  
    client.on('systemMessage', function (systemMessage, originalMessage) {
        var autoListening = systemMessage.auto_listening;
        var messageId = originalMessage.messageId.toString();
        if (typeof systemMessage.app_info !== 'undefined') {
            appInfoDict[messageId] = systemMessage.app_info;
            mesIdQueue.push(messageId);
        }
        if (autoListening) {
            ///    -  mesId     
            ///    -   
            if (settings.current.disableDubbing === false) {
                autolistenMessageId = messageId;
            }
            else {
                listen({}, autoListening);
            }
        }
    }), client.on('status', function (_a) {
        var code = _a.code;
        if (code < 0) {
            stopVoice(false);
        }
    }), client.on('stt', function (_a, originalMessage) {
        var _b;
        var text = _a.text, response = _a.response;
        var listening = listener.status === 'listen' && !settings.current.disableListening;
        if (text) {
            var last = originalMessage.last === 1;
            if (last || listening) {
                emit({
                    asr: {
                        mid: originalMessage.messageId,
                        text: text.data || '',
                        last: last,
                    },
                });
            }
            if (last) {
                stopVoice(false);
            }
        }
        if (response) {
            var decoderResultField = response.decoderResultField, errorResponse = response.errorResponse;
            var last = !!(decoderResultField && (decoderResultField === null || decoderResultField === void 0 ? void 0 : decoderResultField.isFinal));
            if ((last || listening) && ((_b = decoderResultField === null || decoderResultField === void 0 ? void 0 : decoderResultField.hypothesis) === null || _b === void 0 ? void 0 : _b.length)) {
                emit({
                    asr: {
                        mid: originalMessage.messageId,
                        text: decoderResultField.hypothesis[0].normalizedText || '',
                        last: last,
                    },
                });
            }
            if (last || errorResponse) {
                stopVoice(false);
            }
        }
    }), client.on('musicRecognition', function (response, originalMessage) {
        var _a;
        emit({ mtt: { response: response, mid: originalMessage.messageId } });
        if (((_a = response.decoderResultField) === null || _a === void 0 ? void 0 : _a.isFinal) || response.errorResponse) {
            stopVoice(false);
        }
    }), settings.on('change-request', function (nextSettings) {
        var disableDubbing = nextSettings.disableDubbing, disableListening = nextSettings.disableListening;
        ///        
        ///  ,  
        disableListening && stopVoice();
        //   ,        (  ),  
        settings.current.disableDubbing !== disableDubbing && (voicePlayer === null || voicePlayer === void 0 ? void 0 : voicePlayer.setActive(!disableDubbing));
    }));
    return {
        destroy: function () {
            stopVoice();
            voicePlayer === null || voicePlayer === void 0 ? void 0 : voicePlayer.setActive(false);
            subscriptions.splice(0, subscriptions.length).map(function (unsubscribe) { return unsubscribe(); });
            destroyAudioContext === null || destroyAudioContext === void 0 ? void 0 : destroyAudioContext();
        },
        listen: listen,
        shazam: shazam,
        sendVoice: sendVoice,
        streamVoice: streamVoice,
        stop: stop,
        stopPlaying: function () {
            voicePlayer === null || voicePlayer === void 0 ? void 0 : voicePlayer.stop();
        },
    };
};

var createMutexedObject = function (initialObject) {
    var _a = createNanoEvents(), on = _a.on, emit = _a.emit;
    var object = __assign({}, initialObject);
    var nextObject = {};
    var mode = 'released';
    var tryApply = function () {
        if (mode === 'released') {
            var prevObject_1 = object;
            object = __assign(__assign({}, prevObject_1), nextObject);
            var isObjectChanged = Object.keys(nextObject).some(function (name) { return nextObject[name] !== prevObject_1[name]; });
            if (isObjectChanged) {
                emit('changed', object, prevObject_1);
            }
        }
    };
    var lock = function () {
        mode = 'locked';
    };
    var release = function () {
        mode = 'released';
        tryApply();
    };
    var change = function (setts) {
        nextObject = __assign(__assign({}, nextObject), setts);
        emit('change-request', setts);
        tryApply();
    };
    var current = {};
    Object.keys(initialObject).forEach(function (prop) {
        Object.defineProperty(current, prop, {
            get: function () {
                return object[prop];
            },
        });
    });
    return {
        on: on,
        lock: lock,
        release: release,
        change: change,
        current: current,
    };
};

var createMutexSwitcher = function (_a, initialDeps) {
    var lock = _a.lock, release = _a.release;
    var deps = __assign({}, initialDeps);
    return {
        change: function (nextDeps) {
            deps = __assign(__assign({}, deps), nextDeps);
            if (Object.values(deps).every(function (dep) { return dep; })) {
                release();
            }
            else {
                lock();
            }
        },
    };
};

var STATE_UPDATE_TIMEOUT = 200;
var DEFAULT_PROJECT_ID = 'd929986a-611a-2ba0-6174-1928c99600a5';
var DEFAULT_APPLICATION_ID = '7c4e23bf-cd93-b57e-874b-d9fc1b35f93d';
var DEFAULT_APP_VERSION_ID = '26d0bb2e-45d6-a276-f70e-6c016d1f9cff';
var DEFAULT_APP = {
    projectId: DEFAULT_PROJECT_ID,
    applicationId: DEFAULT_APPLICATION_ID,
    appversionId: DEFAULT_APP_VERSION_ID,
    frontendStateId: [DEFAULT_PROJECT_ID, DEFAULT_APPLICATION_ID, DEFAULT_APP_VERSION_ID].join('_'),
    frontendType: 'DIALOG',
    systemName: 'assistant',
    frontendEndpoint: 'None',
};
var BASIC_SMART_APP_COMMANDS_TYPES = ['smart_app_data', 'smart_app_error', 'start_smart_search', 'navigation'];
function convertFieldValuesToString(object) {
    return Object.keys(object).reduce(function (acc, key) {
        if (object[key]) {
            acc[key] =
                typeof object[key] === 'string' && object[key].startsWith('{')
                    ? object[key]
                    : JSON.stringify(object[key]);
        }
        return acc;
    }, {});
}
var isDefaultApp = function (appInfo) { return appInfo.frontendStateId === DEFAULT_APP.frontendStateId; };
var promiseTimeout = function (promise, timeout) {
    var timeoutId;
    return Promise.race([
        promise.then(function (v) {
            if (timeoutId) {
                window.clearTimeout(timeoutId);
            }
            return v;
        }),
        new Promise(function (_, reject) {
            timeoutId = window.setTimeout(function () {
                reject(new Error("Timed out in " + timeout + " ms."));
            }, timeout);
        }),
    ]);
};
var createAssistant = function (_a) {
    var _b;
    var getMeta = _a.getMeta, getInitialMeta = _a.getInitialMeta, getVoiceMeta = _a.getVoiceMeta, checkCertUrl = _a.checkCertUrl, useVoiceAnalyser = _a.useVoiceAnalyser, configuration = __rest(_a, ["getMeta", "getInitialMeta", "getVoiceMeta", "checkCertUrl", "useVoiceAnalyser"]);
    var _c = createNanoEvents(), on = _c.on, emit = _c.emit;
    // default_character      InitialSettings
    var defaultCharacter = 'sber';
    //  [messageId]: requestId,  requestId -   
    var requestIdMap = {};
    // mid   /  ( server_action)
    var lastMid = 0;
    var subscriptions = [];
    var backgroundApps = {};
    var settings = createMutexedObject({
        disableDubbing: configuration.settings.dubbing === -1,
        disableListening: false,
        sendTextAsSsml: false,
    });
    var settingsSwitcher = createMutexSwitcher(settings, { isListenerStopped: true, isVoicePlayerEnded: true });
    // /   
    var voiceReady = false;
    //  
    var app = { info: DEFAULT_APP };
    var sdkMeta = { theme: 'dark' };
    var metaProvider = function (additionalMeta) { return __awaiter(void 0, void 0, void 0, function () {
        var appState, _a, current_app, getBackgroundAppsMeta, background_apps;
        return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    if (!(app !== null && app.info.frontendType === 'WEB_APP' && app.getState)) return [3 /*break*/, 2];
                    return [4 /*yield*/, promiseTimeout(app.getState(), STATE_UPDATE_TIMEOUT).catch(function () {
                            // eslint-disable-next-line no-console
                            console.error('App-state wasn`t resolved, timeout had been expired');
                            return undefined;
                        })];
                case 1:
                    _a = _b.sent();
                    return [3 /*break*/, 3];
                case 2:
                    _a = undefined;
                    _b.label = 3;
                case 3:
                    appState = _a;
                    current_app = {
                        app_info: app.info,
                        state: appState || {},
                    };
                    getBackgroundAppsMeta = function () { return __awaiter(void 0, void 0, void 0, function () {
                        var apps, backgroundAppsIds, backgroundAppsMeta;
                        return __generator(this, function (_a) {
                            switch (_a.label) {
                                case 0:
                                    apps = __assign({}, backgroundApps);
                                    backgroundAppsIds = Object.keys(apps);
                                    backgroundAppsMeta = [];
                                    return [4 /*yield*/, Promise.all(backgroundAppsIds.map(function (applicationId) { return __awaiter(void 0, void 0, void 0, function () {
                                            var _a, getState;
                                            return __generator(this, function (_b) {
                                                _a = apps[applicationId].getState, getState = _a === void 0 ? function () { return Promise.resolve({}); } : _a;
                                                return [2 /*return*/, promiseTimeout(getState(), STATE_UPDATE_TIMEOUT).then(function (state) { return state; }, function () { return ({}); })];
                                            });
                                        }); })).then(function (results) {
                                            results.forEach(function (appResult, index) {
                                                var state = appResult;
                                                var applicationId = backgroundAppsIds[index];
                                                backgroundAppsMeta.push({
                                                    app_info: apps[applicationId].appInfo,
                                                    state: state,
                                                });
                                            });
                                        })];
                                case 1:
                                    _a.sent();
                                    return [2 /*return*/, backgroundAppsMeta];
                            }
                        });
                    }); };
                    return [4 /*yield*/, getBackgroundAppsMeta()];
                case 4:
                    background_apps = _b.sent();
                    return [2 /*return*/, convertFieldValuesToString(__assign(__assign(__assign(__assign({}, sdkMeta), { time: getTime(), current_app: current_app, background_apps: background_apps }), (additionalMeta || {})), (getMeta ? getMeta() : {})))];
            }
        });
    }); };
    var transport = createTransport({
        createWS: (_b = configuration.fakeVps) === null || _b === void 0 ? void 0 : _b.createFakeWS,
        checkCertUrl: checkCertUrl,
    });
    var protocol = createProtocol(transport, __assign(__assign({}, configuration), { getInitialMeta: typeof getInitialMeta !== 'undefined'
            ? function () {
                return getInitialMeta().then(function (meta) {
                    return convertFieldValuesToString(__assign(__assign({}, meta), { default_character: defaultCharacter }));
                });
            }
            : function () {
                return convertFieldValuesToString({
                    default_character: defaultCharacter,
                });
            }, 
        //    ,  
        settings: __assign(__assign({}, configuration.settings), { dubbing: -1 }) }));
    var client = createClient(protocol, metaProvider, {
        getVoiceMeta: function () { return (getVoiceMeta ? convertFieldValuesToString(getVoiceMeta()) : {}); },
    });
    var voice = createVoice(client, settings, function (event) {
        if (typeof event.tts !== 'undefined') {
            emit('tts', event.tts);
            settingsSwitcher.change({ isVoicePlayerEnded: event.tts.status === 'stop' });
            return;
        }
        if (typeof event.listener !== 'undefined') {
            settingsSwitcher.change({ isListenerStopped: event.listener.status === 'stopped' });
        }
        emit('assistant', event);
    }, function () {
        voiceReady = true;
        if (!settings.current.disableDubbing) {
            protocol.changeSettings({ dubbing: 1 });
        }
    }, useVoiceAnalyser);
    /**    */
    var closeApp = function (closing) {
        if (closing === void 0) { closing = app.info; }
        //    
        if (closing.applicationId === app.info.applicationId) {
            ///    ,
            ///      - 
            ///      DEFAULT_APP
            app = {
                info: DEFAULT_APP,
            };
        }
        if (!isDefaultApp(closing)) {
            emit('app', { type: 'close', app: closing });
        }
    };
    /**   */
    var sendText = function (text, shouldSendDisableDubbing, additionalMeta) {
        if (shouldSendDisableDubbing === void 0) { shouldSendDisableDubbing = false; }
        voice.stop();
        return client.sendText(text, settings.current.sendTextAsSsml, shouldSendDisableDubbing, additionalMeta);
    };
    /**  server_action */
    var sendServerAction = function (serverAction, messageName, requestId, actionApp, mode) {
        if (messageName === void 0) { messageName = 'SERVER_ACTION'; }
        if (requestId === void 0) { requestId = undefined; }
        if (actionApp === void 0) { actionApp = app.info; }
        client.sendServerAction(serverAction, actionApp, messageName, mode).then(function (messageId) {
            if (requestId && messageId) {
                requestIdMap[messageId.toString()] = requestId;
            }
        });
    };
    /**         .  */
    var sendMetaForPermissionRequest = function (requestMessageId, appInfo, items) { return __awaiter(void 0, void 0, void 0, function () {
        var _a, props, data, meta;
        return __generator(this, function (_b) {
            switch (_b.label) {
                case 0: return [4 /*yield*/, getAnswerForRequestPermissions(requestMessageId, appInfo, items)];
                case 1:
                    _a = _b.sent(), props = __rest(_a.meta, []), data = __rest(_a, ["meta"]);
                    return [4 /*yield*/, metaProvider()];
                case 2:
                    meta = _b.sent();
                    client.sendData(__assign({}, data), 'SERVER_ACTION', __assign(__assign({}, meta), convertFieldValuesToString(props)));
                    return [2 /*return*/];
            }
        });
    }); };
    subscriptions.push(protocol.on('ready', function () { return emit('vps', { type: 'ready' }); }));
    //  voicePlayer  ,     
    subscriptions.push(settings.on('changed', function (nextSettings, prevSettings) {
        if (nextSettings.disableDubbing !== prevSettings.disableDubbing) {
            voiceReady && protocol.changeSettings({ dubbing: nextSettings.disableDubbing ? -1 : 1 });
        }
    }));
    //     
    subscriptions.push(transport.on('error', function (error) {
        voice.stop();
        protocol.clearQueue();
        emit('vps', { type: 'error', error: error });
    }));
    //   
    subscriptions.push(protocol.on('outcoming', function (message) {
        if (message.text || message.voice) {
            ///      -
            /// ,       
            lastMid = message.messageId;
        }
        emit('vps', { type: 'outcoming', message: message });
    }));
    //  
    subscriptions.push(protocol.on('error', function (error) {
        emit('error', error);
    }));
    //   
    subscriptions.push(client.on('status', function (status, _a) {
        var messageId = _a.messageId;
        emit('status', status, messageId);
    }));
    //    GetHistoryRequest
    subscriptions.push(client.on('history', function (history) {
        emit('history', history);
    }));
    //   ,   
    subscriptions.push(client.on('systemMessage', function (systemMessage, originalMessage) {
        if (originalMessage.messageName === 'ANSWER_TO_USER') {
            var _a = systemMessage.answerId, answerId = _a === void 0 ? 0 : _a, activate_app_info = systemMessage.activate_app_info, _b = systemMessage.items, items = _b === void 0 ? [] : _b, mesAppInfo = systemMessage.app_info;
            var isChatApp = mesAppInfo && mesAppInfo.frontendType === 'CHAT_APP';
            var isDialog = mesAppInfo && mesAppInfo.frontendType === 'DIALOG';
            var isAppChanged = mesAppInfo && mesAppInfo.applicationId !== app.info.applicationId;
            if (
            // DIALOG    current_app
            !isDialog &&
                isAppChanged &&
                ///  activate_app_info  
                /// - activate_app_info: true
                (isChatApp || activate_app_info !== false)) {
                emit('app', { type: 'run', app: mesAppInfo });
            }
            if (isDialog && isAppChanged && app.info.applicationId !== DEFAULT_APPLICATION_ID) {
                emit('app', { type: 'run', app: DEFAULT_APP });
            }
            // cancel   
            if (answerId >= 2 && lastMid > originalMessage.messageId) {
                client.sendCancel(originalMessage.messageId);
            }
            //   ,    /
            if (lastMid < originalMessage.messageId && items.findIndex(function (_a) {
                var bubble = _a.bubble, card = _a.card;
                return bubble || card;
            }) >= 0) {
                lastMid = originalMessage.messageId;
            }
            if (items.length) {
                var _loop_1 = function (i) {
                    var command = items[i].command;
                    if (typeof command !== 'undefined') {
                        window.setTimeout(function () { return emit('command', command); });
                        if (command.type === 'start_music_recognition') {
                            voice.shazam();
                            return { value: void 0 };
                        }
                        if (command.type === 'request_permissions' && mesAppInfo) {
                            sendMetaForPermissionRequest(originalMessage.messageId, mesAppInfo, command.permissions);
                            return { value: void 0 };
                        }
                        if (command.type === 'action') {
                            emit('actionCommand', {
                                type: 'command',
                                command: command,
                                appInfo: mesAppInfo,
                            });
                        }
                        if (mesAppInfo && BASIC_SMART_APP_COMMANDS_TYPES.includes(command.type)) {
                            //   , .   
                            emit('app', {
                                type: 'command',
                                command: __assign(__assign({}, command), { sdk_meta: {
                                        mid: originalMessage.messageId.toString(),
                                        requestId: requestIdMap[originalMessage.messageId.toString()],
                                    } }),
                                app: mesAppInfo,
                            });
                        }
                        if (command.type === 'close_app' && !isDialog) {
                            closeApp(mesAppInfo);
                        }
                    }
                };
                for (var i = 0; i < items.length; i++) {
                    var state_1 = _loop_1(i);
                    if (typeof state_1 === "object")
                        return state_1.value;
                }
            }
            emit('vps', { type: 'incoming', systemMessage: systemMessage, originalMessage: originalMessage });
        }
    }));
    //   backgroundApp'   
    on('app', function (event) {
        var _a;
        if (event.type === 'command') {
            var backgroundAppOnCommand = (_a = backgroundApps[event.app.applicationId]) === null || _a === void 0 ? void 0 : _a.commandsSubscribers;
            if (Array.isArray(backgroundAppOnCommand)) {
                backgroundAppOnCommand.forEach(function (onCommand) {
                    var _a;
                    onCommand(event.command, (_a = event.command.sdk_meta) === null || _a === void 0 ? void 0 : _a.mid);
                });
            }
        }
    });
    /**  ,   */
    var destroy = function () {
        voice.destroy();
        client.destroy();
        protocol.destroy();
        subscriptions.splice(0, subscriptions.length).map(function (unsubscribe) { return unsubscribe(); });
    };
    /**   () */
    var start = function (_a) {
        var _b = _a === void 0 ? {} : _a, _c = _b.disableGreetings, disableGreetings = _c === void 0 ? false : _c, _d = _b.initPhrase, initPhrase = _d === void 0 ? undefined : _d, _e = _b.isFirstSession, isFirstSession = _e === void 0 ? false : _e;
        return __awaiter(void 0, void 0, void 0, function () {
            return __generator(this, function (_f) {
                switch (_f.label) {
                    case 0:
                        if (!(!disableGreetings && isDefaultApp(app.info))) return [3 /*break*/, 2];
                        return [4 /*yield*/, client.sendOpenAssistant({ isFirstSession: isFirstSession })];
                    case 1:
                        _f.sent();
                        _f.label = 2;
                    case 2:
                        if (initPhrase) {
                            return [2 /*return*/, client
                                    .sendText(initPhrase)
                                    .then(function (messageId) { return (messageId ? client.waitForAnswer(messageId) : undefined); })];
                        }
                        return [2 /*return*/, undefined];
                }
            });
        });
    };
    return {
        get activeApp() {
            return !isDefaultApp(app.info) ? app.info : null;
        },
        get settings() {
            return Object.create(Object.getPrototypeOf(settings.current), Object.getOwnPropertyDescriptors(settings.current));
        },
        destroy: destroy,
        closeApp: closeApp,
        listen: voice.listen,
        shazam: voice.shazam,
        sendServerAction: sendServerAction,
        getHistoryRequest: protocol.getHistoryRequest,
        sendText: sendText,
        sendVoice: voice.sendVoice,
        streamVoice: voice.streamVoice,
        start: start,
        stop: function () {
            voice.stop();
            if (lastMid !== 0) {
                client.sendCancel(lastMid);
            }
            setTimeout(function () {
                protocol.clearQueue();
                transport.close();
            });
        },
        stopTts: voice.stopPlaying,
        stopVoice: voice.stop,
        emit: emit,
        on: on,
        changeConfiguration: protocol.changeConfiguration,
        changeSettings: settings.change,
        changeSdkMeta: function (nextSdkMeta) {
            sdkMeta = __assign(__assign({}, sdkMeta), nextSdkMeta);
        },
        reconnect: protocol.reconnect,
        get protocol() {
            return protocol;
        },
        setActiveApp: function (info, getState) {
            app = { info: info, getState: getState };
        },
        addBackgroundApp: function (_a) {
            var appInfo = _a.appInfo, getState = _a.getState;
            backgroundApps[appInfo.applicationId] = {
                appInfo: appInfo,
                getState: getState,
                commandsSubscribers: [],
            };
            var remove = function () {
                delete backgroundApps[appInfo.applicationId];
            };
            var onCommand = function (subscriber) {
                var _a;
                (_a = backgroundApps[appInfo.applicationId]) === null || _a === void 0 ? void 0 : _a.commandsSubscribers.push(subscriber);
                return {
                    clearSubscribers: function () {
                        if (backgroundApps[appInfo.applicationId]) {
                            backgroundApps[appInfo.applicationId].commandsSubscribers = [];
                        }
                    },
                };
            };
            var sendServerActionToBackgroundApp = function (serverAction, messageName, requestId) {
                var _a;
                if (messageName === void 0) { messageName = 'SERVER_ACTION'; }
                if (requestId === void 0) { requestId = undefined; }
                return sendServerAction(serverAction, messageName, requestId, (_a = backgroundApps[appInfo.applicationId]) === null || _a === void 0 ? void 0 : _a.appInfo);
            };
            return {
                remove: remove,
                onCommand: onCommand,
                sendServerAction: sendServerActionToBackgroundApp,
            };
        },
        get status() {
            return protocol.status;
        },
        setDefaultCharacterMeta: function (characterId) {
            defaultCharacter = characterId;
        },
    };
};

export { createNavigatorAudioProvider as a, asrExports as b, createAssistant as c };
