// @salutejs/plasma-new-hope/styled-components/cjs/mixins/typography.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.underline = exports.textXSBold = exports.textXS = exports.textSBold = exports.textS = exports.textMBold = exports.textM = exports.textLBold = exports.textL = exports.paragraph2 = exports.paragraph1 = exports.headline4 = exports.headline3 = exports.headline2 = exports.headline1 = exports.h5Bold = exports.h5 = exports.h4Bold = exports.h4 = exports.h3Bold = exports.h3 = exports.h2Bold = exports.h2 = exports.h1Bold = exports.h1 = exports.footnote2 = exports.footnote1 = exports.dsplSBold = exports.dsplS = exports.dsplMBold = exports.dsplM = exports.dsplLBold = exports.dsplL = exports.display3 = exports.display2 = exports.display1 = exports.caption = exports.button2 = exports.button1 = exports.bodyXXSBold = exports.bodyXXS = exports.bodyXSBold = exports.bodyXS = exports.bodySBold = exports.bodyS = exports.bodyMBold = exports.bodyM = exports.bodyLBold = exports.bodyL = exports.body3 = exports.body2 = exports.body1 = void 0;
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
var dsplL = exports.dsplL = {
  fontFamily: 'var(--plasma-typo-dspl-l-font-family)',
  fontSize: 'var(--plasma-typo-dspl-l-font-size)',
  fontStyle: 'var(--plasma-typo-dspl-l-font-style)',
  fontWeight: 'var(--plasma-typo-dspl-l-font-weight)',
  letterSpacing: 'var(--plasma-typo-dspl-l-letter-spacing)',
  lineHeight: 'var(--plasma-typo-dspl-l-line-height)'
};
var dsplLBold = exports.dsplLBold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, dsplL), {}, {
  fontWeight: 'var(--plasma-typo-dspl-l-bold-font-weight)'
});
var dsplM = exports.dsplM = {
  fontFamily: 'var(--plasma-typo-dspl-m-font-family)',
  fontSize: 'var(--plasma-typo-dspl-m-font-size)',
  fontStyle: 'var(--plasma-typo-dspl-m-font-style)',
  fontWeight: 'var(--plasma-typo-dspl-m-font-weight)',
  letterSpacing: 'var(--plasma-typo-dspl-m-letter-spacing)',
  lineHeight: 'var(--plasma-typo-dspl-m-line-height)'
};
var dsplMBold = exports.dsplMBold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, dsplM), {}, {
  fontWeight: 'var(--plasma-typo-dspl-m-bold-font-weight)'
});
var dsplS = exports.dsplS = {
  fontFamily: 'var(--plasma-typo-dspl-s-font-family)',
  fontSize: 'var(--plasma-typo-dspl-s-font-size)',
  fontStyle: 'var(--plasma-typo-dspl-s-font-style)',
  fontWeight: 'var(--plasma-typo-dspl-s-font-weight)',
  letterSpacing: 'var(--plasma-typo-dspl-s-letter-spacing)',
  lineHeight: 'var(--plasma-typo-dspl-s-line-height)'
};
var dsplSBold = exports.dsplSBold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, dsplS), {}, {
  fontWeight: 'var(--plasma-typo-dspl-s-bold-font-weight)'
});
var h1 = exports.h1 = {
  fontFamily: 'var(--plasma-typo-h1-font-family)',
  fontSize: 'var(--plasma-typo-h1-font-size)',
  fontStyle: 'var(--plasma-typo-h1-font-style)',
  fontWeight: 'var(--plasma-typo-h1-font-weight)',
  letterSpacing: 'var(--plasma-typo-h1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-h1-line-height)'
};
var h1Bold = exports.h1Bold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, h1), {}, {
  fontWeight: 'var(--plasma-typo-h1-bold-font-weight)'
});
var h2 = exports.h2 = {
  fontFamily: 'var(--plasma-typo-h2-font-family)',
  fontSize: 'var(--plasma-typo-h2-font-size)',
  fontStyle: 'var(--plasma-typo-h2-font-style)',
  fontWeight: 'var(--plasma-typo-h2-font-weight)',
  letterSpacing: 'var(--plasma-typo-h2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-h2-line-height)'
};
var h2Bold = exports.h2Bold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, h2), {}, {
  fontWeight: 'var(--plasma-typo-h2-bold-font-weight)'
});
var h3 = exports.h3 = {
  fontFamily: 'var(--plasma-typo-h3-font-family)',
  fontSize: 'var(--plasma-typo-h3-font-size)',
  fontStyle: 'var(--plasma-typo-h3-font-style)',
  fontWeight: 'var(--plasma-typo-h3-font-weight)',
  letterSpacing: 'var(--plasma-typo-h3-letter-spacing)',
  lineHeight: 'var(--plasma-typo-h3-line-height)'
};
var h3Bold = exports.h3Bold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, h3), {}, {
  fontWeight: 'var(--plasma-typo-h3-bold-font-weight)'
});
var h4 = exports.h4 = {
  fontFamily: 'var(--plasma-typo-h4-font-family)',
  fontSize: 'var(--plasma-typo-h4-font-size)',
  fontStyle: 'var(--plasma-typo-h4-font-style)',
  fontWeight: 'var(--plasma-typo-h4-font-weight)',
  letterSpacing: 'var(--plasma-typo-h4-letter-spacing)',
  lineHeight: 'var(--plasma-typo-h4-line-height)'
};
var h4Bold = exports.h4Bold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, h4), {}, {
  fontWeight: 'var(--plasma-typo-h4-bold-font-weight)'
});
var h5 = exports.h5 = {
  fontFamily: 'var(--plasma-typo-h5-font-family)',
  fontSize: 'var(--plasma-typo-h5-font-size)',
  fontStyle: 'var(--plasma-typo-h5-font-style)',
  fontWeight: 'var(--plasma-typo-h5-font-weight)',
  letterSpacing: 'var(--plasma-typo-h5-letter-spacing)',
  lineHeight: 'var(--plasma-typo-h5-line-height)'
};
var h5Bold = exports.h5Bold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, h5), {}, {
  fontWeight: 'var(--plasma-typo-h5-bold-font-weight)'
});
var bodyL = exports.bodyL = {
  fontFamily: 'var(--plasma-typo-body-l-font-family)',
  fontSize: 'var(--plasma-typo-body-l-font-size)',
  fontStyle: 'var(--plasma-typo-body-l-font-style)',
  fontWeight: 'var(--plasma-typo-body-l-font-weight)',
  letterSpacing: 'var(--plasma-typo-body-l-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body-l-line-height)'
};
var bodyLBold = exports.bodyLBold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, bodyL), {}, {
  fontWeight: 'var(--plasma-typo-body-l-bold-font-weight)'
});
var bodyM = exports.bodyM = {
  fontFamily: 'var(--plasma-typo-body-m-font-family)',
  fontSize: 'var(--plasma-typo-body-m-font-size)',
  fontStyle: 'var(--plasma-typo-body-m-font-style)',
  fontWeight: 'var(--plasma-typo-body-m-font-weight)',
  letterSpacing: 'var(--plasma-typo-body-m-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body-m-line-height)'
};
var bodyMBold = exports.bodyMBold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, bodyM), {}, {
  fontWeight: 'var(--plasma-typo-body-m-bold-font-weight)'
});
var bodyS = exports.bodyS = {
  fontFamily: 'var(--plasma-typo-body-s-font-family)',
  fontSize: 'var(--plasma-typo-body-s-font-size)',
  fontStyle: 'var(--plasma-typo-body-s-font-style)',
  fontWeight: 'var(--plasma-typo-body-s-font-weight)',
  letterSpacing: 'var(--plasma-typo-body-s-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body-s-line-height)'
};
var bodySBold = exports.bodySBold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, bodyS), {}, {
  fontWeight: 'var(--plasma-typo-body-s-bold-font-weight)'
});
var bodyXS = exports.bodyXS = {
  fontFamily: 'var(--plasma-typo-body-xs-font-family)',
  fontSize: 'var(--plasma-typo-body-xs-font-size)',
  fontStyle: 'var(--plasma-typo-body-xs-font-style)',
  fontWeight: 'var(--plasma-typo-body-xs-font-weight)',
  letterSpacing: 'var(--plasma-typo-body-xs-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body-xs-line-height)'
};
var bodyXSBold = exports.bodyXSBold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, bodyXS), {}, {
  fontWeight: 'var(--plasma-typo-body-xs-bold-font-weight)'
});
var bodyXXS = exports.bodyXXS = {
  fontFamily: 'var(--plasma-typo-body-xxs-font-family)',
  fontSize: 'var(--plasma-typo-body-xxs-font-size)',
  fontStyle: 'var(--plasma-typo-body-xxs-font-style)',
  fontWeight: 'var(--plasma-typo-body-xxs-font-weight)',
  letterSpacing: 'var(--plasma-typo-body-xxs-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body-xxs-line-height)'
};
var bodyXXSBold = exports.bodyXXSBold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, bodyXXS), {}, {
  fontWeight: 'var(--plasma-typo-body-xxs-bold-font-weight)'
});
var textL = exports.textL = {
  fontFamily: 'var(--plasma-typo-text-l-font-family)',
  fontSize: 'var(--plasma-typo-text-l-font-size)',
  fontStyle: 'var(--plasma-typo-text-l-font-style)',
  fontWeight: 'var(--plasma-typo-text-l-font-weight)',
  letterSpacing: 'var(--plasma-typo-text-l-letter-spacing)',
  lineHeight: 'var(--plasma-typo-text-l-line-height)'
};
var textLBold = exports.textLBold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, textL), {}, {
  fontWeight: 'var(--plasma-typo-text-l-bold-font-weight)'
});
var textM = exports.textM = {
  fontFamily: 'var(--plasma-typo-text-m-font-family)',
  fontSize: 'var(--plasma-typo-text-m-font-size)',
  fontStyle: 'var(--plasma-typo-text-m-font-style)',
  fontWeight: 'var(--plasma-typo-text-m-font-weight)',
  letterSpacing: 'var(--plasma-typo-text-m-letter-spacing)',
  lineHeight: 'var(--plasma-typo-text-m-line-height)'
};
var textMBold = exports.textMBold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, textM), {}, {
  fontWeight: 'var(--plasma-typo-text-m-bold-font-weight)'
});
var textS = exports.textS = {
  fontFamily: 'var(--plasma-typo-text-s-font-family)',
  fontSize: 'var(--plasma-typo-text-s-font-size)',
  fontStyle: 'var(--plasma-typo-text-s-font-style)',
  fontWeight: 'var(--plasma-typo-text-s-font-weight)',
  letterSpacing: 'var(--plasma-typo-text-s-letter-spacing)',
  lineHeight: 'var(--plasma-typo-text-s-line-height)'
};
var textSBold = exports.textSBold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, textS), {}, {
  fontWeight: 'var(--plasma-typo-text-s-bold-font-weight)'
});
var textXS = exports.textXS = {
  fontFamily: 'var(--plasma-typo-text-xs-font-family)',
  fontSize: 'var(--plasma-typo-text-xs-font-size)',
  fontStyle: 'var(--plasma-typo-text-xs-font-style)',
  fontWeight: 'var(--plasma-typo-text-xs-font-weight)',
  letterSpacing: 'var(--plasma-typo-text-xs-letter-spacing)',
  lineHeight: 'var(--plasma-typo-text-xs-line-height)'
};
var textXSBold = exports.textXSBold = /*#__PURE__*/_objectSpread( /*#__PURE__*/_objectSpread({}, textXS), {}, {
  fontWeight: 'var(--plasma-typo-text-xs-bold-font-weight)'
});
var body1 = exports.body1 = {
  fontFamily: 'var(--plasma-typo-body1-font-family)',
  fontWeight: 'var(--plasma-typo-body1-font-weight)',
  fontStyle: 'var(--plasma-typo-body1-font-style)',
  fontSize: 'var(--plasma-typo-body1-font-size)',
  letterSpacing: 'var(--plasma-typo-body1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body1-line-height)'
};
var body2 = exports.body2 = {
  fontFamily: 'var(--plasma-typo-body2-font-family)',
  fontWeight: 'var(--plasma-typo-body2-font-weight)',
  fontStyle: 'var(--plasma-typo-body2-font-style)',
  fontSize: 'var(--plasma-typo-body2-font-size)',
  letterSpacing: 'var(--plasma-typo-body2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body2-line-height)'
};
var body3 = exports.body3 = {
  fontFamily: 'var(--plasma-typo-body3-font-family)',
  fontWeight: 'var(--plasma-typo-body3-font-weight)',
  fontStyle: 'var(--plasma-typo-body3-font-style)',
  fontSize: 'var(--plasma-typo-body3-font-size)',
  letterSpacing: 'var(--plasma-typo-body3-letter-spacing)',
  lineHeight: 'var(--plasma-typo-body3-line-height)'
};
var button1 = exports.button1 = {
  fontFamily: 'var(--plasma-typo-button1-font-family)',
  fontWeight: 'var(--plasma-typo-button1-font-weight)',
  fontStyle: 'var(--plasma-typo-button1-font-style)',
  fontSize: 'var(--plasma-typo-button1-font-size)',
  letterSpacing: 'var(--plasma-typo-button1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-button1-line-height)'
};
var button2 = exports.button2 = {
  fontFamily: 'var(--plasma-typo-button2-font-family)',
  fontWeight: 'var(--plasma-typo-button2-font-weight)',
  fontStyle: 'var(--plasma-typo-button2-font-style)',
  fontSize: 'var(--plasma-typo-button2-font-size)',
  letterSpacing: 'var(--plasma-typo-button2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-button2-line-height)'
};
var caption = exports.caption = {
  fontFamily: 'var(--plasma-typo-caption-font-family)',
  fontWeight: 'var(--plasma-typo-caption-font-weight)',
  fontStyle: 'var(--plasma-typo-caption-font-style)',
  fontSize: 'var(--plasma-typo-caption-font-size)',
  letterSpacing: 'var(--plasma-typo-caption-letter-spacing)',
  lineHeight: 'var(--plasma-typo-caption-line-height)'
};
var display1 = exports.display1 = {
  fontFamily: 'var(--plasma-typo-display1-font-family)',
  fontWeight: 'var(--plasma-typo-display1-font-weight)',
  fontStyle: 'var(--plasma-typo-display1-font-style)',
  fontSize: 'var(--plasma-typo-display1-font-size)',
  letterSpacing: 'var(--plasma-typo-display1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-display1-line-height)'
};
var display2 = exports.display2 = {
  fontFamily: 'var(--plasma-typo-display2-font-family)',
  fontWeight: 'var(--plasma-typo-display2-font-weight)',
  fontStyle: 'var(--plasma-typo-display2-font-style)',
  fontSize: 'var(--plasma-typo-display2-font-size)',
  letterSpacing: 'var(--plasma-typo-display2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-display2-line-height)'
};
var display3 = exports.display3 = {
  fontFamily: 'var(--plasma-typo-display3-font-family)',
  fontWeight: 'var(--plasma-typo-display3-font-weight)',
  fontStyle: 'var(--plasma-typo-display3-font-style)',
  fontSize: 'var(--plasma-typo-display3-font-size)',
  letterSpacing: 'var(--plasma-typo-display3-letter-spacing)',
  lineHeight: 'var(--plasma-typo-display3-line-height)'
};
var footnote1 = exports.footnote1 = {
  fontFamily: 'var(--plasma-typo-footnote1-font-family)',
  fontWeight: 'var(--plasma-typo-footnote1-font-weight)',
  fontStyle: 'var(--plasma-typo-footnote1-font-style)',
  fontSize: 'var(--plasma-typo-footnote1-font-size)',
  letterSpacing: 'var(--plasma-typo-footnote1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-footnote1-line-height)'
};
var footnote2 = exports.footnote2 = {
  fontFamily: 'var(--plasma-typo-footnote2-font-family)',
  fontWeight: 'var(--plasma-typo-footnote2-font-weight)',
  fontStyle: 'var(--plasma-typo-footnote2-font-style)',
  fontSize: 'var(--plasma-typo-footnote2-font-size)',
  letterSpacing: 'var(--plasma-typo-footnote2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-footnote2-line-height)'
};
var headline1 = exports.headline1 = {
  fontFamily: 'var(--plasma-typo-headline1-font-family)',
  fontWeight: 'var(--plasma-typo-headline1-font-weight)',
  fontStyle: 'var(--plasma-typo-headline1-font-style)',
  fontSize: 'var(--plasma-typo-headline1-font-size)',
  letterSpacing: 'var(--plasma-typo-headline1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-headline1-line-height)'
};
var headline2 = exports.headline2 = {
  fontFamily: 'var(--plasma-typo-headline2-font-family)',
  fontWeight: 'var(--plasma-typo-headline2-font-weight)',
  fontStyle: 'var(--plasma-typo-headline2-font-style)',
  fontSize: 'var(--plasma-typo-headline2-font-size)',
  letterSpacing: 'var(--plasma-typo-headline2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-headline2-line-height)'
};
var headline3 = exports.headline3 = {
  fontFamily: 'var(--plasma-typo-headline3-font-family)',
  fontWeight: 'var(--plasma-typo-headline3-font-weight)',
  fontStyle: 'var(--plasma-typo-headline3-font-style)',
  fontSize: 'var(--plasma-typo-headline3-font-size)',
  letterSpacing: 'var(--plasma-typo-headline3-letter-spacing)',
  lineHeight: 'var(--plasma-typo-headline3-line-height)'
};
var headline4 = exports.headline4 = {
  fontFamily: 'var(--plasma-typo-headline4-font-family)',
  fontWeight: 'var(--plasma-typo-headline4-font-weight)',
  fontStyle: 'var(--plasma-typo-headline4-font-style)',
  fontSize: 'var(--plasma-typo-headline4-font-size)',
  letterSpacing: 'var(--plasma-typo-headline4-letter-spacing)',
  lineHeight: 'var(--plasma-typo-headline4-line-height)'
};
var paragraph1 = exports.paragraph1 = {
  fontFamily: 'var(--plasma-typo-paragraph1-font-family)',
  fontWeight: 'var(--plasma-typo-paragraph1-font-weight)',
  fontStyle: 'var(--plasma-typo-paragraph1-font-style)',
  fontSize: 'var(--plasma-typo-paragraph1-font-size)',
  letterSpacing: 'var(--plasma-typo-paragraph1-letter-spacing)',
  lineHeight: 'var(--plasma-typo-paragraph1-line-height)'
};
var paragraph2 = exports.paragraph2 = {
  fontFamily: 'var(--plasma-typo-paragraph2-font-family)',
  fontWeight: 'var(--plasma-typo-paragraph2-font-weight)',
  fontStyle: 'var(--plasma-typo-paragraph2-font-style)',
  fontSize: 'var(--plasma-typo-paragraph2-font-size)',
  letterSpacing: 'var(--plasma-typo-paragraph2-letter-spacing)',
  lineHeight: 'var(--plasma-typo-paragraph2-line-height)'
};
var underline = exports.underline = {
  fontFamily: 'var(--plasma-typo-underline-font-family)',
  fontWeight: 'var(--plasma-typo-underline-font-weight)',
  fontStyle: 'var(--plasma-typo-underline-font-style)',
  fontSize: 'var(--plasma-typo-underline-font-size)',
  letterSpacing: 'var(--plasma-typo-underline-letter-spacing)',
  lineHeight: 'var(--plasma-typo-underline-line-height)'
};
