// @salutejs/plasma-new-hope/styled-components/cjs/components/Tabs/ui/TabItem/TabItem.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.tabItemRoot = exports.tabItemConfig = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _tokens = /*#__PURE__*/require("../../tokens");
var _utils = /*#__PURE__*/require("../../../../utils");
var _TabsContext = /*#__PURE__*/require("../../TabsContext");
var _base = /*#__PURE__*/require("./variations/_view/base");
var _base2 = /*#__PURE__*/require("./variations/_size/base");
var _base3 = /*#__PURE__*/require("./variations/_pilled/base");
var _base4 = /*#__PURE__*/require("./variations/_disabled/base");
var _TabItem = /*#__PURE__*/require("./TabItem.styles");
var _excluded = ["size", "view", "isActive", "selected", "disabled", "pilled", "children", "contentLeft", "contentRight", "animated", "onIndexChange", "itemIndex", "tabIndex", "className"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var tabItemRoot = exports.tabItemRoot = function tabItemRoot(Root) {
  return /*#__PURE__*/(0, _react.forwardRef)(function (props, outerRef) {
    var size = props.size,
      view = props.view,
      isActive = props.isActive,
      selected = props.selected,
      _props$disabled = props.disabled,
      disabled = _props$disabled === void 0 ? false : _props$disabled,
      _props$pilled = props.pilled,
      pilled = _props$pilled === void 0 ? false : _props$pilled,
      children = props.children,
      contentLeft = props.contentLeft,
      contentRight = props.contentRight,
      _props$animated = props.animated,
      animated = _props$animated === void 0 ? true : _props$animated,
      onIndexChange = props.onIndexChange,
      itemIndex = props.itemIndex,
      tabIndex = props.tabIndex,
      className = props.className,
      rest = _objectWithoutProperties(props, _excluded);
    var innerRef = (0, _react.useRef)(null);
    var ref = (0, _plasmaCore.useForkRef)(outerRef, innerRef);
    var refs = (0, _react.useContext)(_TabsContext.TabsContext);
    var role = 'tab';
    var pilledAttr = view !== 'clear' && pilled;
    var animatedClass = animated ? _tokens.classes.tabItemAnimated : undefined;
    var pilledClass = pilledAttr ? _tokens.classes.tabsPilled : undefined;
    var selectedClass = isActive || selected ? _tokens.classes.selectedTabsItem : undefined;
    var hasKeyNavigation = itemIndex !== undefined && onIndexChange !== undefined;
    var navigationTabIndex = !disabled && (refs === null || refs === void 0 ? void 0 : refs.current) === itemIndex ? 0 : -1;
    (0, _react.useEffect)(function () {
      if (!refs) {
        return;
      }
      refs.register(innerRef);
      return function () {
        return refs.unregister(innerRef);
      };
    }, [refs]);
    var onItemFocus = (0, _react.useCallback)(function (event) {
      if (!hasKeyNavigation && innerRef !== null && innerRef !== void 0 && innerRef.current) {
        innerRef.current.scrollTo({
          top: 0,
          left: innerRef.current.offsetLeft,
          behavior: 'smooth'
        });
        return;
      }
      if (disabled || !refs) {
        return;
      }
      var focusIndex = refs.items.findIndex(function (itemRef) {
        return itemRef.current === event.target;
      });
      if (focusIndex === refs.current) {
        return;
      }
      onIndexChange === null || onIndexChange === void 0 || onIndexChange(focusIndex);
      refs.setCurrent(focusIndex);
    }, [refs, innerRef, onIndexChange, disabled]);
    return /*#__PURE__*/_react["default"].createElement(Root, _extends({
      ref: ref,
      disabled: disabled,
      pilled: pilled,
      role: role,
      view: view,
      size: size,
      onFocus: onItemFocus,
      tabIndex: hasKeyNavigation ? navigationTabIndex : tabIndex,
      className: (0, _utils.cx)(pilledClass, selectedClass, animatedClass, className)
    }, rest), /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, contentLeft && /*#__PURE__*/_react["default"].createElement(_TabItem.LeftContent, {
      className: _tokens.classes.tabLeftContent
    }, contentLeft), /*#__PURE__*/_react["default"].createElement(_TabItem.StyledContent, {
      className: _tokens.classes.tabContent
    }, children), contentRight && /*#__PURE__*/_react["default"].createElement(_TabItem.RightContent, {
      className: _tokens.classes.tabRightContent
    }, contentRight)));
  });
};
var tabItemConfig = exports.tabItemConfig = {
  name: 'TabItem',
  tag: 'button',
  layout: tabItemRoot,
  base: _TabItem.base,
  variations: {
    size: {
      css: _base2.base
    },
    view: {
      css: _base.base
    },
    disabled: {
      css: _base4.base,
      attrs: true
    },
    pilled: {
      css: _base3.base
    }
  },
  defaults: {
    view: 'default'
  }
};
