// @salutejs/plasma-new-hope/es/components/Segment/SegmentProvider/SegmentProvider.js
import { slicedToArray as _slicedToArray, toConsumableArray as _toConsumableArray } from '../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { useState, useCallback, useEffect, createContext, useContext } from 'react';

var SegmentContext = /*#__PURE__*/createContext(undefined);
var useSegmentInner = function useSegmentInner() {
  var context = useContext(SegmentContext);
  if (!context) {
    throw new Error('useSegment must be used within a SegmentProvider');
  }
  return context;
};
var useSegment = function useSegment() {
  var _useSegmentInner = useSegmentInner(),
    selectedSegmentItems = _useSegmentInner.selectedSegmentItems;
  return {
    selectedSegmentItems: selectedSegmentItems
  };
};
var SegmentProvider = function SegmentProvider(_ref) {
  var children = _ref.children;
  var _useState = useState([]),
    _useState2 = _slicedToArray(_useState, 2),
    selectedSegmentItems = _useState2[0],
    setSelectedSegmentItems = _useState2[1];
  var _useState3 = useState('single'),
    _useState4 = _slicedToArray(_useState3, 2),
    selectionMode = _useState4[0],
    setSelectionMode = _useState4[1];
  var _useState5 = useState(false),
    _useState6 = _slicedToArray(_useState5, 2),
    disabledGroup = _useState6[0],
    setDisabledGroup = _useState6[1];
  var handleSelect = useCallback(function (label) {
    if (selectionMode !== 'multiple') {
      setSelectedSegmentItems(function (prevSelected) {
        return prevSelected.includes(label) ? [] : [label];
      });
      return;
    }
    setSelectedSegmentItems(function (prevSelected) {
      return prevSelected.includes(label) ? prevSelected.filter(function (item) {
        return item !== label;
      }) : [].concat(_toConsumableArray(prevSelected), [label]);
    });
  }, [selectionMode, setSelectedSegmentItems]);
  var contextValue = {
    selectedSegmentItems: selectedSegmentItems,
    handleSelect: handleSelect,
    selectionMode: selectionMode,
    setSelectionMode: setSelectionMode,
    disabledGroup: disabledGroup,
    setDisabledGroup: setDisabledGroup
  };
  useEffect(function () {
    setSelectedSegmentItems([]);
  }, [selectionMode]);
  return /*#__PURE__*/React.createElement(SegmentContext.Provider, {
    value: contextValue
  }, children);
};

export { SegmentContext, SegmentProvider, useSegment, useSegmentInner };
