// @salutejs/plasma-new-hope/es/components/Dropdown/ui/DropdownInner/DropdownInner.js
import React from 'react';
import '@salutejs/plasma-core';
import '@linaria/react';
import { DropdownItem } from '../DropdownItem/DropdownItem.js';
import { StyledPopover, Ul } from '../../Dropdown.styles.js';
import '../../../../utils/react.js';
import 'focus-visible';
import 'styled-components';
import '@linaria/core';
import 'react-dom';
import 'react-popper';

var DropdownInner = function DropdownInner(_ref) {
  var item = _ref.item,
    currentLevel = _ref.currentLevel,
    focusedPath = _ref.focusedPath,
    path = _ref.path,
    dispatchPath = _ref.dispatchPath,
    index = _ref.index,
    trigger = _ref.trigger,
    itemRole = _ref.itemRole,
    listHeight = _ref.listHeight,
    listOverflow = _ref.listOverflow,
    handleGlobalToggle = _ref.handleGlobalToggle,
    closeOnSelect = _ref.closeOnSelect,
    onHover = _ref.onHover,
    onItemSelect = _ref.onItemSelect,
    onItemClick = _ref.onItemClick,
    listWidth = _ref.listWidth,
    variant = _ref.variant,
    hasArrow = _ref.hasArrow;
  var handleToggle = function handleToggle(opened) {
    if (opened) {
      dispatchPath({
        type: 'changed_on_level',
        value: item.value.toString(),
        level: currentLevel + 1
      });
    } else {
      dispatchPath({
        type: 'cut_by_level',
        level: currentLevel + 1
      });
    }
  };
  var isCurrentListOpen = path[currentLevel + 1] === item.value.toString();
  var listId = "tree_level_".concat(currentLevel + 2);
  var nextLevel = currentLevel + 1;
  if (item !== null && item !== void 0 && item.items) {
    return /*#__PURE__*/React.createElement(StyledPopover, {
      isOpen: isCurrentListOpen,
      usePortal: false,
      placement: "right-start",
      trigger: trigger,
      target: /*#__PURE__*/React.createElement(DropdownItem, {
        item: item,
        index: index,
        path: path,
        focusedPath: focusedPath,
        currentLevel: currentLevel,
        itemRole: itemRole,
        onHover: onHover,
        onItemSelect: onItemSelect,
        onItemClick: onItemClick,
        variant: variant,
        hasArrow: hasArrow,
        ariaControls: listId,
        ariaExpanded: isCurrentListOpen,
        ariaLevel: nextLevel,
        ariaLabel: item.label
      }),
      onToggle: handleToggle,
      isFocusTrapped: false,
      preventOverflow: false
    }, /*#__PURE__*/React.createElement(Ul, {
      listHeight: listHeight,
      listOverflow: listOverflow,
      role: "group",
      id: listId,
      listWidth: listWidth,
      isInnerUl: true
    }, item.items.map(function (innerItem, innerIndex) {
      return /*#__PURE__*/React.createElement(DropdownInner, {
        key: "".concat(innerIndex, "/").concat(currentLevel),
        item: innerItem,
        currentLevel: nextLevel,
        focusedPath: focusedPath,
        path: path,
        dispatchPath: dispatchPath,
        index: innerIndex,
        trigger: trigger,
        itemRole: itemRole,
        listHeight: listHeight,
        listOverflow: listOverflow,
        handleGlobalToggle: handleGlobalToggle,
        closeOnSelect: closeOnSelect,
        onHover: onHover,
        onItemSelect: onItemSelect,
        onItemClick: onItemClick,
        listWidth: listWidth,
        variant: variant,
        hasArrow: hasArrow
      });
    })));
  }
  return /*#__PURE__*/React.createElement(DropdownItem, {
    item: item,
    index: index,
    focusedPath: focusedPath,
    currentLevel: currentLevel,
    itemRole: itemRole,
    handleGlobalToggle: handleGlobalToggle,
    closeOnSelect: closeOnSelect,
    onHover: onHover,
    onItemSelect: onItemSelect,
    onItemClick: onItemClick,
    variant: variant
  });
};

export { DropdownInner };
