// @salutejs/plasma-ui/components/Stepper/Stepper.hooks.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useStepper = void 0;
var _react = /*#__PURE__*/require("react");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
(0, _collectPackageInfo.collectPackageInfo)('Stepper/Stepper.hooks');
var useStepper = exports.useStepper = function useStepper(_ref) {
  var value = _ref.value,
    _ref$step = _ref.step,
    step = _ref$step === void 0 ? 1 : _ref$step,
    _ref$min = _ref.min,
    min = _ref$min === void 0 ? 0 : _ref$min,
    _ref$max = _ref.max,
    max = _ref$max === void 0 ? Infinity : _ref$max,
    onChange = _ref.onChange;
  var curValue = (0, _react.useRef)(value);
  (0, _react.useEffect)(function () {
    curValue.current = value;
  }, [value]);
  var onLessClick = (0, _react.useCallback)(function (event) {
    return onChange === null || onChange === void 0 ? void 0 : onChange(curValue.current - step, event);
  }, [step, onChange]);
  var onMoreClick = (0, _react.useCallback)(function (event) {
    return onChange === null || onChange === void 0 ? void 0 : onChange(curValue.current + step, event);
  }, [step, onChange]);
  var isMin = value <= min;
  var isMax = value >= max;
  var isLessDisabled = isMin || value - step < min;
  var isMoreDisabled = isMax || value + step > max;
  return {
    onLessClick: onLessClick,
    onMoreClick: onMoreClick,
    isMin: isMin,
    isMax: isMax,
    isLessDisabled: isLessDisabled,
    isMoreDisabled: isMoreDisabled
  };
};
