// @salutejs/plasma-new-hope/styled-components/es/components/Chip/Chip.js
var _excluded = ["children", "text", "contentLeft", "contentRight", "contentClearButton", "size", "className", "onClear", "onClick", "pilled", "readOnly", "disabled"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React, { forwardRef, useMemo } from 'react';
import { IconClose } from '../_Icon/Icons/IconClose';
import { cx } from '../../utils';
import { base as viewCSS } from './variations/_view/base';
import { base as sizeCSS } from './variations/_size/base';
import { base as disabledCSS } from './variations/_disabled/base';
import { base as readonlyCSS } from './variations/_readonly/base';
import { base as focusedCSS } from './variations/_focused/base';
import { base as pilledCSS } from './variations/_pilled/base';
import { StyledContentClear, StyledContentLeft, StyledContentMain, StyledContentRight, base } from './Chip.styles';
import { classes, tokens } from './Chip.tokens';
export var chipRoot = function chipRoot(Root) {
  return /*#__PURE__*/forwardRef(function (props, ref) {
    var children = props.children,
      text = props.text,
      contentLeft = props.contentLeft,
      contentRight = props.contentRight,
      contentClearButton = props.contentClearButton,
      size = props.size,
      className = props.className,
      onClear = props.onClear,
      onClick = props.onClick,
      _props$pilled = props.pilled,
      pilled = _props$pilled === void 0 ? false : _props$pilled,
      _props$readOnly = props.readOnly,
      readOnly = _props$readOnly === void 0 ? false : _props$readOnly,
      _props$disabled = props.disabled,
      disabled = _props$disabled === void 0 ? false : _props$disabled,
      rest = _objectWithoutProperties(props, _excluded);
    var txt = !text && typeof children === 'string' ? children : text;
    var pilledClass = pilled ? classes.pilled : undefined;
    var handleClick = function handleClick(event) {
      if (disabled || readOnly) {
        return;
      }
      onClick === null || onClick === void 0 || onClick(event);
      onClear === null || onClear === void 0 || onClear();
    };
    var ClearContent = useMemo(function () {
      return contentClearButton || /*#__PURE__*/React.createElement(StyledContentClear, null, /*#__PURE__*/React.createElement(IconClose, {
        sizeCustomProperty: tokens.closeIconSize,
        color: "inherit"
      }));
    }, [contentClearButton, tokens.closeIconSize]);
    return /*#__PURE__*/React.createElement(Root, _extends({
      type: "button",
      ref: ref,
      className: cx(pilledClass, className),
      tabIndex: readOnly ? -1 : 0,
      onClick: handleClick,
      disabled: disabled,
      pilled: pilled,
      readOnly: !disabled && readOnly,
      size: size
    }, rest), contentLeft && /*#__PURE__*/React.createElement(StyledContentLeft, null, contentLeft), txt ? /*#__PURE__*/React.createElement(StyledContentMain, null, txt) : children, contentRight && /*#__PURE__*/React.createElement(StyledContentRight, null, contentRight), ClearContent);
  });
};
export var chipConfig = {
  name: 'Chip',
  tag: 'button',
  layout: chipRoot,
  base: base,
  variations: {
    view: {
      css: viewCSS
    },
    size: {
      css: sizeCSS
    },
    disabled: {
      css: disabledCSS,
      attrs: true
    },
    readOnly: {
      css: readonlyCSS,
      attrs: true
    },
    focused: {
      css: focusedCSS
    },
    pilled: {
      css: pilledCSS,
      attrs: true
    }
  },
  defaults: {
    view: 'default',
    size: 'm'
  }
};
