// @salutejs/plasma-new-hope/styled-components/es/components/Chip/Chip.styles.js
import styled from 'styled-components';
import { css } from 'styled-components';
import { applyEllipsis } from '../../mixins';
import { tokens } from './Chip.tokens';
export var base = /*#__PURE__*/css(["position:relative;display:inline-flex;align-items:center;box-sizing:border-box;appearance:none;border:none;cursor:pointer;-webkit-tap-highlight-color:rgba(0,0,0,0);"]);
export var StyledContentLeft = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-bm1fxi-0"
})(["display:flex;margin-left:var(", ");margin-right:var(", ");"], tokens.leftContentMarginLeft, tokens.leftContentMarginRight);
export var StyledContentMain = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-bm1fxi-1"
})(["", ";"], /*#__PURE__*/applyEllipsis());
export var StyledContentRight = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-bm1fxi-2"
})(["display:flex;margin-left:var(", ");margin-right:var(", ");"], tokens.rightContentMarginLeft, tokens.rightContentMarginRight);
export var StyledContentClear = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-bm1fxi-3"
})(["display:flex;margin-left:var(", ");margin-right:var(", ");color:var(", ");"], tokens.clearContentMarginLeft, tokens.clearContentMarginRight, tokens.closeIconColor);
