// @salutejs/plasma-new-hope/styled-components/es/components/Drawer/Drawer.styles.js
import styled from 'styled-components';
import { component } from '../../engines';
import { popupClasses, popupConfig } from '../Popup';
import { classes } from './Drawer.tokens';
var Popup = /*#__PURE__*/component(popupConfig);
var animationVariants = {
  right: {
    show: {
      start: 'translate(100%, -50%)',
      end: 'translate(0, -50%)'
    },
    hide: {
      start: 'translate(0, -50%)',
      end: 'translate(100%, -50%)'
    }
  },
  bottom: {
    show: {
      start: 'translate(-50%, 100%)',
      end: 'translate(-50%, 0)'
    },
    hide: {
      start: 'translate(-50%, 0)',
      end: 'translate(-50%, 100%)'
    }
  },
  left: {
    show: {
      start: 'translate(-100%, -50%)',
      end: 'translate(0, -50%)'
    },
    hide: {
      start: 'translate(0, -50%)',
      end: 'translate(-100%, -50%)'
    }
  },
  top: {
    show: {
      start: 'translate(-50%, -100%)',
      end: 'translate(-50%, 0)'
    },
    hide: {
      start: 'translate(-50%, 0)',
      end: 'translate(-50%, -100%)'
    }
  }
};
export var StyledPopup = /*#__PURE__*/styled(Popup).withConfig({
  componentId: "plasma-new-hope__sc-cup208-0"
})(["&&.", "{animation:fadeIn 0.2s forwards;}&&.", " .", "{animation:fadeOut 0.2s forwards;}@keyframes fadeIn{from{opacity:0;}to{opacity:1;}}@keyframes fadeOut{from{opacity:1;}to{opacity:0;}}&& .", ",&& .", " > div{width:", ";height:", ";}&& .", "{animation:showAnimation 0.2s forwards;}&&.", " .", "{animation:hideAnimation 0.2s forwards;}@keyframes showAnimation{0%{transform:", ";opacity:0;}100%{transform:", ";opacity:1;}}@keyframes hideAnimation{0%{transform:", ";opacity:1;}100%{transform:", ";opacity:0;}}"], classes.overlay, popupClasses.endAnimation, classes.overlay, popupClasses.root, popupClasses.root, function (_ref) {
  var width = _ref.width;
  return width;
}, function (_ref2) {
  var height = _ref2.height;
  return height;
}, popupClasses.root, popupClasses.endAnimation, popupClasses.root, function (_ref3) {
  var placement = _ref3.placement;
  return animationVariants[placement].show.start;
}, function (_ref4) {
  var placement = _ref4.placement;
  return animationVariants[placement].show.end;
}, function (_ref5) {
  var placement = _ref5.placement;
  return animationVariants[placement].hide.start;
}, function (_ref6) {
  var placement = _ref6.placement;
  return animationVariants[placement].hide.end;
});
