// @salutejs/plasma-web/components/Calendar/Calendar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CalendarDoubleRange = exports.CalendarDouble = exports.CalendarBaseRange = exports.CalendarBase = exports.Calendar = void 0;
var _styledComponents = /*#__PURE__*/require("@salutejs/plasma-new-hope/styled-components");
var _CalendarBase = /*#__PURE__*/require("./CalendarBase.config");
var _CalendarDouble = /*#__PURE__*/require("./CalendarDouble.config");
var mergedCalendarBaseConfig = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.calendarBaseConfig, _CalendarBase.config);
var mergedCalendarDoubleConfig = /*#__PURE__*/(0, _styledComponents.mergeConfig)(_styledComponents.calendarDoubleConfig, _CalendarDouble.config);
var CalendarBase = exports.CalendarBase = /*#__PURE__*/(0, _styledComponents.component)(mergedCalendarBaseConfig);
var CalendarDouble = exports.CalendarDouble = /*#__PURE__*/(0, _styledComponents.component)(mergedCalendarDoubleConfig);
var CalendarBaseRange = exports.CalendarBaseRange = /*#__PURE__*/(0, _styledComponents.withRange)(CalendarBase);
var CalendarDoubleRange = exports.CalendarDoubleRange = /*#__PURE__*/(0, _styledComponents.withRange)(CalendarDouble);
var Calendar = exports.Calendar = /*#__PURE__*/(0, _styledComponents.boundCalendar)({
  base: CalendarBase,
  baseRange: CalendarBaseRange,
  "double": CalendarDouble,
  doubleRange: CalendarDoubleRange
});
