// @salutejs/plasma-ui/es/components/Cell/CellDisclosure.js
var _IconDisclosureRight;
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
import React from 'react';
import styled from 'styled-components';
import { IconDisclosureRight } from '@salutejs/plasma-icons';
import { Button } from '../Button';
import { collectPackageInfo } from '../../collectPackageInfo';
collectPackageInfo('Cell/CellDisclosure');
var DisclosureBtn = /*#__PURE__*/styled(Button).withConfig({
  componentId: "plasma-ui__sc-1hbfhpz-0"
})(["left:1rem;margin-left:-1rem;"]);
export var Disclosure = function Disclosure(props) {
  return /*#__PURE__*/React.createElement(DisclosureBtn, _extends({}, props, {
    size: "s",
    view: "clear",
    outlined: false,
    scaleOnInteraction: false,
    contentRight: _IconDisclosureRight || (_IconDisclosureRight = /*#__PURE__*/React.createElement(IconDisclosureRight, {
      size: "s"
    }))
  }));
};
