// @salutejs/plasma-web/components/Divider/Divider.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("@salutejs/plasma-new-hope/styled-components");
var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5, _templateObject6, _templateObject7;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
var config = exports.config = {
  defaults: {
    view: 'default',
    size: 'm',
    orientation: 'horizontal'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/(0, _styledComponents.css)(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--surface-transparent-tertiary);\n            "])), _styledComponents.dividerTokens.background),
      dark: /*#__PURE__*/(0, _styledComponents.css)(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--on-dark-surface-transparent-tertiary);\n            "])), _styledComponents.dividerTokens.background),
      light: /*#__PURE__*/(0, _styledComponents.css)(_templateObject3 || (_templateObject3 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--on-light-surface-transparent-tertiary);\n            "])), _styledComponents.dividerTokens.background),
      // TODO поменять на корректный токен --inverse-surface-transparent-tertiary когда он появится
      inverse: /*#__PURE__*/(0, _styledComponents.css)(_templateObject4 || (_templateObject4 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--inverse-text-tertiary);\n            "])), _styledComponents.dividerTokens.background)
    },
    size: {
      m: /*#__PURE__*/(0, _styledComponents.css)(_templateObject5 || (_templateObject5 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": 0.0625rem;\n            "])), _styledComponents.dividerTokens.borderRadius)
    },
    orientation: {
      horizontal: /*#__PURE__*/(0, _styledComponents.css)(_templateObject6 || (_templateObject6 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": 0.0625rem;\n            "])), _styledComponents.dividerTokens.baseSideSize),
      vertical: /*#__PURE__*/(0, _styledComponents.css)(_templateObject7 || (_templateObject7 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": 0.0625rem;\n            "])), _styledComponents.dividerTokens.baseSideSize)
    }
  }
};
