// @salutejs/plasma-web/components/Typography/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Body1", {
  enumerable: true,
  get: function get() {
    return _Old.Body1;
  }
});
Object.defineProperty(exports, "Body2", {
  enumerable: true,
  get: function get() {
    return _Old.Body2;
  }
});
Object.defineProperty(exports, "BodyL", {
  enumerable: true,
  get: function get() {
    return _Typography.BodyL;
  }
});
Object.defineProperty(exports, "BodyM", {
  enumerable: true,
  get: function get() {
    return _Typography.BodyM;
  }
});
Object.defineProperty(exports, "BodyS", {
  enumerable: true,
  get: function get() {
    return _Typography.BodyS;
  }
});
Object.defineProperty(exports, "BodyXS", {
  enumerable: true,
  get: function get() {
    return _Typography.BodyXS;
  }
});
Object.defineProperty(exports, "BodyXXS", {
  enumerable: true,
  get: function get() {
    return _Typography.BodyXXS;
  }
});
Object.defineProperty(exports, "Button1", {
  enumerable: true,
  get: function get() {
    return _Old.Button1;
  }
});
Object.defineProperty(exports, "Button2", {
  enumerable: true,
  get: function get() {
    return _Old.Button2;
  }
});
Object.defineProperty(exports, "Caption", {
  enumerable: true,
  get: function get() {
    return _Old.Caption;
  }
});
Object.defineProperty(exports, "DsplL", {
  enumerable: true,
  get: function get() {
    return _Typography.DsplL;
  }
});
Object.defineProperty(exports, "DsplM", {
  enumerable: true,
  get: function get() {
    return _Typography.DsplM;
  }
});
Object.defineProperty(exports, "DsplS", {
  enumerable: true,
  get: function get() {
    return _Typography.DsplS;
  }
});
Object.defineProperty(exports, "Footnote1", {
  enumerable: true,
  get: function get() {
    return _Old.Footnote1;
  }
});
Object.defineProperty(exports, "Footnote2", {
  enumerable: true,
  get: function get() {
    return _Old.Footnote2;
  }
});
Object.defineProperty(exports, "H1", {
  enumerable: true,
  get: function get() {
    return _Typography.H1;
  }
});
Object.defineProperty(exports, "H2", {
  enumerable: true,
  get: function get() {
    return _Typography.H2;
  }
});
Object.defineProperty(exports, "H3", {
  enumerable: true,
  get: function get() {
    return _Typography.H3;
  }
});
Object.defineProperty(exports, "H4", {
  enumerable: true,
  get: function get() {
    return _Typography.H4;
  }
});
Object.defineProperty(exports, "H5", {
  enumerable: true,
  get: function get() {
    return _Typography.H5;
  }
});
Object.defineProperty(exports, "Headline1", {
  enumerable: true,
  get: function get() {
    return _Old.Headline1;
  }
});
Object.defineProperty(exports, "Headline2", {
  enumerable: true,
  get: function get() {
    return _Old.Headline2;
  }
});
Object.defineProperty(exports, "Headline3", {
  enumerable: true,
  get: function get() {
    return _Old.Headline3;
  }
});
Object.defineProperty(exports, "Headline4", {
  enumerable: true,
  get: function get() {
    return _Old.Headline4;
  }
});
Object.defineProperty(exports, "Headline5", {
  enumerable: true,
  get: function get() {
    return _Old.Headline5;
  }
});
Object.defineProperty(exports, "P1", {
  enumerable: true,
  get: function get() {
    return _Old.P1;
  }
});
Object.defineProperty(exports, "P2", {
  enumerable: true,
  get: function get() {
    return _Old.P2;
  }
});
Object.defineProperty(exports, "ParagraphText1", {
  enumerable: true,
  get: function get() {
    return _Old.ParagraphText1;
  }
});
Object.defineProperty(exports, "ParagraphText2", {
  enumerable: true,
  get: function get() {
    return _Old.ParagraphText2;
  }
});
Object.defineProperty(exports, "Subtitle", {
  enumerable: true,
  get: function get() {
    return _Old.Subtitle;
  }
});
Object.defineProperty(exports, "TextL", {
  enumerable: true,
  get: function get() {
    return _Typography.TextL;
  }
});
Object.defineProperty(exports, "TextM", {
  enumerable: true,
  get: function get() {
    return _Typography.TextM;
  }
});
Object.defineProperty(exports, "TextS", {
  enumerable: true,
  get: function get() {
    return _Typography.TextS;
  }
});
Object.defineProperty(exports, "TextXS", {
  enumerable: true,
  get: function get() {
    return _Typography.TextXS;
  }
});
Object.defineProperty(exports, "Underline", {
  enumerable: true,
  get: function get() {
    return _Old.Underline;
  }
});
var _Typography = /*#__PURE__*/require("./Typography");
var _Old = /*#__PURE__*/require("./Old");
