// @salutejs/plasma-core/components/Card/Card.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledCard = exports.Card = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _tokens = /*#__PURE__*/require("../../tokens");
var _mixins = /*#__PURE__*/require("../../mixins");
var _Typography = /*#__PURE__*/require("../Typography");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
var _excluded = ["roundness", "children"]; // В этих константах задаем размеры в em, чтобы не зависеть напрямую от пикселей
// В то же время в числителях - значения в пикселях, взятые из макета
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
(0, _collectPackageInfo.collectPackageInfo)('Card/Card');
var fontSize = 16;
var shadowOffset = 8 / fontSize;
var shadowSize = 24 / fontSize;
var outlineSize = 2 / fontSize;
var DEFAULT_ROUNDNESS = 20;
var StyledCard = exports.StyledCard = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Body1).withConfig({
  componentId: "plasma__sc-1bj6k4p-0"
})(["", " ", ";position:relative;display:flex;box-sizing:border-box;flex-direction:column;flex-shrink:0;background:", ";box-shadow:0 ", "em ", "em rgba(0,0,0,0.1);transition:", ";", ""], _mixins.applyDisabled, _mixins.applyRoundness, _tokens.surfaceCard, shadowOffset, shadowSize, function (_ref) {
  var theme = _ref.theme;
  return theme.lowPerformance ? 'unset' : 'transform 0.4s ease-in-out';
}, function (_ref2) {
  var focused = _ref2.focused,
    outlined = _ref2.outlined,
    roundness = _ref2.roundness,
    scaleOnFocus = _ref2.scaleOnFocus,
    theme = _ref2.theme;
  return (0, _styledComponents.css)(["", " ", " &:focus{outline:none;}"], (0, _mixins.addFocus)({
    lowPerformance: theme.lowPerformance,
    focused: focused,
    outlined: outlined,
    outlineSize: "".concat(outlineSize, "rem"),
    outlineRadius: "".concat(_mixins.radiuses[roundness] + outlineSize, "rem")
  }), scaleOnFocus && (0, _mixins.syntheticFocus)((0, _styledComponents.css)(["&{transform:scale(1.04);}"]), focused));
});
/**
 * Контейнер со скругленными углами с возможностью фокусировки на нем.
 */
// eslint-disable-next-line prefer-arrow-callback
var Card = exports.Card = /*#__PURE__*/_react["default"].forwardRef(function Card(_ref3, ref) {
  var _ref3$roundness = _ref3.roundness,
    roundness = _ref3$roundness === void 0 ? DEFAULT_ROUNDNESS : _ref3$roundness,
    children = _ref3.children,
    rest = _objectWithoutProperties(_ref3, _excluded);
  return /*#__PURE__*/_react["default"].createElement(StyledCard, _extends({
    ref: ref,
    roundness: roundness
  }, rest), children);
});
