// @salutejs/plasma-core/components/Card/CardTypography.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CardParagraph2 = exports.CardParagraph1 = exports.CardHeadline3 = exports.CardHeadline2 = exports.CardHeadline1 = exports.CardFootnote2 = exports.CardFootnote1 = exports.CardBody2 = exports.CardBody1 = void 0;
var _styledComponents = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("styled-components"));
var _tokens = /*#__PURE__*/require("../../tokens");
var _Typography = /*#__PURE__*/require("../Typography");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
/* stylelint-disable */
// TODO:
(0, _collectPackageInfo.collectPackageInfo)('Card/CardTypography');
var views = {
  primary: _tokens.primary,
  secondary: _tokens.secondary,
  tertiary: _tokens.tertiary
};
var linesMixin = function linesMixin(_ref) {
  var _ref$lines = _ref.lines,
    lines = _ref$lines === void 0 ? 2 : _ref$lines;
  return (0, _styledComponents.css)(["display:-webkit-box;overflow:hidden;box-sizing:border-box;white-space:normal;-webkit-box-orient:vertical;-webkit-line-clamp:", ";"], lines);
};
var viewMixin = function viewMixin(_ref2) {
  var _ref2$view = _ref2.view,
    view = _ref2$view === void 0 ? 'primary' : _ref2$view;
  return (0, _styledComponents.css)(["transition:", ";color:", ";"], function (_ref3) {
    var theme = _ref3.theme;
    return theme.lowPerformance ? 'unset' : 'color 0.1s ease-in-out';
  }, views[view]);
};
var CardHeadline1 = exports.CardHeadline1 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Headline1).withConfig({
  componentId: "plasma__sc-zyrk75-0"
})(["", ";"], viewMixin);
var CardHeadline2 = exports.CardHeadline2 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Headline2).withConfig({
  componentId: "plasma__sc-zyrk75-1"
})(["", ";"], viewMixin);
var CardHeadline3 = exports.CardHeadline3 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Headline3).withConfig({
  componentId: "plasma__sc-zyrk75-2"
})(["", ";"], viewMixin);
var CardFootnote1 = exports.CardFootnote1 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Footnote1).withConfig({
  componentId: "plasma__sc-zyrk75-3"
})(["", ";"], viewMixin);
var CardFootnote2 = exports.CardFootnote2 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Footnote2).withConfig({
  componentId: "plasma__sc-zyrk75-4"
})(["", ";"], viewMixin);
var CardBody1 = exports.CardBody1 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Body1).withConfig({
  componentId: "plasma__sc-zyrk75-5"
})(["", ";", ";"], linesMixin, viewMixin);
var CardBody2 = exports.CardBody2 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.Body2).withConfig({
  componentId: "plasma__sc-zyrk75-6"
})(["", ";", ";"], linesMixin, viewMixin);
var CardParagraph1 = exports.CardParagraph1 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.ParagraphText1).withConfig({
  componentId: "plasma__sc-zyrk75-7"
})(["", ";", ";"], linesMixin, viewMixin);
var CardParagraph2 = exports.CardParagraph2 = /*#__PURE__*/(0, _styledComponents["default"])(_Typography.ParagraphText2).withConfig({
  componentId: "plasma__sc-zyrk75-8"
})(["", ";", ";"], linesMixin, viewMixin);
