// @salutejs/plasma-core/components/Card/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  Card: true,
  StyledCard: true,
  CardBody: true,
  CardContent: true,
  CardMedia: true,
  CardPrice: true
};
Object.defineProperty(exports, "Card", {
  enumerable: true,
  get: function get() {
    return _Card.Card;
  }
});
Object.defineProperty(exports, "CardBody", {
  enumerable: true,
  get: function get() {
    return _CardBody.CardBody;
  }
});
Object.defineProperty(exports, "CardContent", {
  enumerable: true,
  get: function get() {
    return _CardContent.CardContent;
  }
});
Object.defineProperty(exports, "CardMedia", {
  enumerable: true,
  get: function get() {
    return _CardMedia.CardMedia;
  }
});
Object.defineProperty(exports, "CardPrice", {
  enumerable: true,
  get: function get() {
    return _CardPrice.CardPrice;
  }
});
Object.defineProperty(exports, "StyledCard", {
  enumerable: true,
  get: function get() {
    return _Card.StyledCard;
  }
});
var _Card = /*#__PURE__*/require("./Card");
var _CardBody = /*#__PURE__*/require("./CardBody");
var _CardContent = /*#__PURE__*/require("./CardContent");
var _CardMedia = /*#__PURE__*/require("./CardMedia");
var _CardPrice = /*#__PURE__*/require("./CardPrice");
var _CardTypography = /*#__PURE__*/require("./CardTypography");
Object.keys(_CardTypography).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _CardTypography[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _CardTypography[key];
    }
  });
});
