// @salutejs/plasma-new-hope/styled-components/es/examples/typograpy/components/Old/Button/Button.config.js
import { css } from 'styled-components';
import { tokens } from '../../../../../components/Typography/tokens';
export var config = {
  defaults: {
    size: 'button1'
  },
  variations: {
    size: {
      button1: /*#__PURE__*/css(["", ":var(--plasma-typo-button1-font-family);", ":var(--plasma-typo-button1-font-size);", ":var(--plasma-typo-button1-font-style);", ":var(--plasma-typo-button1-font-weight);", ":var(--plasma-typo-button1-letter-spacing);", ":var(--plasma-typo-button1-line-height);"], tokens.typoFontFamily, tokens.typoFontSize, tokens.typoFontStyle, tokens.typoFontWeight, tokens.typoFontLetterSpacing, tokens.typoFontLineHeight),
      button2: /*#__PURE__*/css(["", ":var(--plasma-typo-button2-font-family);", ":var(--plasma-typo-button2-font-size);", ":var(--plasma-typo-button2-font-style);", ":var(--plasma-typo-button2-font-weight);", ":var(--plasma-typo-button2-letter-spacing);", ":var(--plasma-typo-button2-line-height);"], tokens.typoFontFamily, tokens.typoFontSize, tokens.typoFontStyle, tokens.typoFontWeight, tokens.typoFontLetterSpacing, tokens.typoFontLineHeight)
    }
  }
};
