// @salutejs/plasma-hope/components/Button/Button.props.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.buttonViews = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _plasmaCore = /*#__PURE__*/require("@salutejs/plasma-core");
var _tokens = /*#__PURE__*/require("../../tokens");
var _collectPackageInfo = /*#__PURE__*/require("../../collectPackageInfo");
/**
 * Views (colors) for both B2B and B2C
 */
(0, _collectPackageInfo.collectPackageInfo)('Button/Button.props');
var buttonViews = exports.buttonViews = {
  primary: /*#__PURE__*/(0, _styledComponents.css)(["", " &:hover{background-color:", ";color:", ";}&:active{background-color:", ";color:", ";}"], _plasmaCore.buttonViews.primary, _tokens.buttonPrimaryHover, _plasmaCore.buttonViews.primary.color, _tokens.buttonPrimaryActive, _plasmaCore.buttonViews.primary.color),
  success: /*#__PURE__*/(0, _styledComponents.css)(["", " &:hover{background-color:", ";color:", ";}&:active{background-color:", ";color:", ";}"], _plasmaCore.buttonViews.success, _tokens.buttonSuccessHover, _plasmaCore.buttonViews.success.color, _tokens.buttonSuccessActive, _plasmaCore.buttonViews.success.color),
  warning: /*#__PURE__*/(0, _styledComponents.css)(["", " &:hover{background-color:", ";color:", ";}&:active{background-color:", ";color:", ";}"], _plasmaCore.buttonViews.warning, _tokens.buttonWarningHover, _plasmaCore.buttonViews.warning.color, _tokens.buttonWarningActive, _plasmaCore.buttonViews.warning.color),
  critical: /*#__PURE__*/(0, _styledComponents.css)(["", " &:hover{background-color:", ";color:", ";}&:active{background-color:", ";color:", ";}"], _plasmaCore.buttonViews.critical, _tokens.buttonCriticalHover, _plasmaCore.buttonViews.critical.color, _tokens.buttonCriticalActive, _plasmaCore.buttonViews.critical.color),
  secondary: /*#__PURE__*/(0, _styledComponents.css)(["", " &:hover{background-color:", ";color:", ";}&:active{background-color:", ";color:", ";}"], _plasmaCore.buttonViews.secondary, _tokens.buttonSecondaryHover, _plasmaCore.buttonViews.secondary.color, _tokens.buttonSecondaryActive, _plasmaCore.buttonViews.secondary.color),
  checked: /*#__PURE__*/(0, _styledComponents.css)(["", " &:hover{background-color:", ";color:", ";}&:active{background-color:", ";color:", ";}"], _plasmaCore.buttonViews.checked, _tokens.buttonCheckedHover, _tokens.buttonCheckedHoverColor, _tokens.buttonCheckedActive, _tokens.buttonCheckedActiveColor),
  overlay: _plasmaCore.buttonViews.overlay,
  clear: _plasmaCore.buttonViews.clear
};

/**
 * @private
 */
