// @salutejs/plasma-web/css/es/components/ButtonGroup/ButtonGroup.config.js
import './ButtonGroup.config_y4g1d3.css';
var config = {
  defaults: {
    view: 'default',
    size: 'm'
  },
  variations: {
    view: {
      "default": "d1n1yjnm",
      /**
       * @deprecated
       * использовать `default`
       */
      primary: "pqck75c",
      accent: "az65cne",
      secondary: "s5fo1by",
      clear: "covvipf",
      success: "s1tax4m1",
      warning: "w1k2o0xu",
      critical: "c2bys7z",
      dark: "dp70rs0",
      black: "bmie5p2",
      white: "w1g3omzu"
    },
    size: {
      l: "l1116k6l",
      lr: "l15t0nuq",
      m: "m316r83",
      mr: "mzu697",
      s: "s16bzis9",
      sr: "s97067l",
      xs: "x1xcwbx9",
      xsr: "x1fuvc6m",
      xxs: "xsjqy1i"
    },
    orientation: {
      horizontal: "hmgmq8o",
      vertical: "v1udxooc"
    },
    gap: {
      none: "nbb1ace",
      dense: "dnyer5h",
      wide: "w1g2rt4j"
    },
    shape: {
      segmented: "s1xdd6tf",
      "default": "d1qv9uf"
    },
    stretching: {
      auto: "a192v9cb",
      filled: "fxjkdyx"
    }
  }
};

export { config };
