// @salutejs/plasma-new-hope/cjs/components/Calendar/CalendarBase/CalendarBase.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var reducer = require('../store/reducer.js');
var types = require('../store/types.js');
var index = require('../utils/index.js');
var useKeyNavigation = require('../hooks/useKeyNavigation.js');
var CalendarHeader = require('../ui/CalendarHeader/CalendarHeader.js');
var CalendarDays = require('../ui/CalendarDays/CalendarDays.js');
var CalendarMonths = require('../ui/CalendarMonths/CalendarMonths.js');
var CalendarYears = require('../ui/CalendarYears/CalendarYears.js');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var CalendarBase_styles = require('./CalendarBase.styles.js');

var _excluded = ["value", "min", "max", "includeEdgeDates", "type", "eventList", "disabledList", "onChangeValue"];
/**
 * Компонент календаря.
 */
var calendarBaseRoot = function calendarBaseRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (_ref, outerRootRef) {
    var externalValue = _ref.value,
      min = _ref.min,
      max = _ref.max,
      includeEdgeDates = _ref.includeEdgeDates,
      _ref$type = _ref.type,
      type = _ref$type === void 0 ? 'Days' : _ref$type,
      eventList = _ref.eventList,
      disabledList = _ref.disabledList,
      onChangeValue = _ref.onChangeValue,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
    var _useMemo = React.useMemo(function () {
        return Array.isArray(externalValue) ? externalValue : [externalValue];
      }, [externalValue]),
      _useMemo2 = _rollupPluginBabelHelpers.slicedToArray(_useMemo, 2),
      firstValue = _useMemo2[0],
      secondValue = _useMemo2[1];
    var value = secondValue || firstValue;
    var _useState = React.useState(),
      _useState2 = _rollupPluginBabelHelpers.slicedToArray(_useState, 2),
      hoveredDay = _useState2[0],
      setHoveredDay = _useState2[1];
    var _useState3 = React.useState(type),
      _useState4 = _rollupPluginBabelHelpers.slicedToArray(_useState3, 2),
      prevType = _useState4[0],
      setPrevType = _useState4[1];
    var _useState5 = React.useState(value),
      _useState6 = _rollupPluginBabelHelpers.slicedToArray(_useState5, 2),
      prevValue = _useState6[0],
      setPrevValue = _useState6[1];
    var _useState7 = React.useState(0),
      _useState8 = _rollupPluginBabelHelpers.slicedToArray(_useState7, 2),
      outOfRangeKey = _useState8[0],
      setOutOfRangeKey = _useState8[1];
    var _useReducer = React.useReducer(reducer.reducer, reducer.getInitialState(value, [5, 6], type)),
      _useReducer2 = _rollupPluginBabelHelpers.slicedToArray(_useReducer, 2),
      state = _useReducer2[0],
      dispatch = _useReducer2[1];
    var date = state.date,
      calendarState = state.calendarState,
      startYear = state.startYear,
      size = state.size;
    var handlePrev = React.useCallback(function () {
      var withShift = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
      if (calendarState === types.CalendarState.Days) {
        if (withShift) {
          dispatch({
            type: types.ActionType.PREVIOUS_YEAR,
            payload: {
              step: 1
            }
          });
          return;
        }
        dispatch({
          type: types.ActionType.PREVIOUS_MONTH,
          payload: {
            monthIndex: date.monthIndex,
            year: date.year
          }
        });
        return;
      }
      if (calendarState === types.CalendarState.Months) {
        dispatch({
          type: types.ActionType.PREVIOUS_YEAR,
          payload: {
            step: 1
          }
        });
        return;
      }
      if (calendarState === types.CalendarState.Years) {
        dispatch({
          type: types.ActionType.PREVIOUS_START_YEAR,
          payload: {
            yearsCount: index.YEAR_RENDER_COUNT
          }
        });
      }
    }, [date, calendarState]);
    var handleNext = React.useCallback(function () {
      var withShift = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
      if (calendarState === types.CalendarState.Days) {
        if (withShift) {
          dispatch({
            type: types.ActionType.NEXT_YEAR,
            payload: {
              step: 1
            }
          });
          return;
        }
        dispatch({
          type: types.ActionType.NEXT_MONTH,
          payload: {
            monthIndex: date.monthIndex,
            year: date.year
          }
        });
        return;
      }
      if (calendarState === types.CalendarState.Months) {
        dispatch({
          type: types.ActionType.NEXT_YEAR,
          payload: {
            step: 1
          }
        });
        return;
      }
      if (calendarState === types.CalendarState.Years) {
        dispatch({
          type: types.ActionType.NEXT_START_YEAR,
          payload: {
            yearsCount: index.YEAR_RENDER_COUNT
          }
        });
      }
    }, [date, calendarState]);
    var _useKeyNavigation = useKeyNavigation.useKeyNavigation({
        size: size,
        onNext: handleNext,
        onPrev: handlePrev
      }),
      _useKeyNavigation2 = _rollupPluginBabelHelpers.slicedToArray(_useKeyNavigation, 5),
      selectIndexes = _useKeyNavigation2[0],
      onKeyDown = _useKeyNavigation2[1],
      onSelectIndexes = _useKeyNavigation2[2],
      outerRefs = _useKeyNavigation2[3],
      isOutOfRange = _useKeyNavigation2[4];
    var handleOnChangeDay = React.useCallback(function (newDate, coord) {
      var newDay = new Date(newDate.year, newDate.monthIndex, newDate.day);
      onChangeValue === null || onChangeValue === void 0 || onChangeValue(newDay);
      onSelectIndexes(coord);
    }, [onChangeValue, onSelectIndexes]);
    var handleOnChangeMonth = React.useCallback(function (monthIndex) {
      dispatch({
        type: types.ActionType.UPDATE_MONTH,
        payload: {
          calendarState: types.CalendarState.Days,
          monthIndex: monthIndex,
          size: [5, 6]
        }
      });
    }, []);
    var handleOnChangeYear = React.useCallback(function (year) {
      dispatch({
        type: types.ActionType.UPDATE_YEAR,
        payload: {
          calendarState: types.CalendarState.Months,
          year: year
        }
      });
    }, []);
    var handleUpdateCalendarState = React.useCallback(function (newCalendarState, newSize) {
      dispatch({
        type: types.ActionType.UPDATE_CALENDAR_STATE,
        payload: {
          calendarState: newCalendarState,
          size: newSize
        }
      });
    }, []);
    if (value && prevValue && index.isValueUpdate(value, prevValue)) {
      dispatch({
        type: types.ActionType.UPDATE_DATE,
        payload: {
          value: value
        }
      });
      setPrevValue(value);
    }
    if (prevType !== type) {
      dispatch({
        type: types.ActionType.UPDATE_CALENDAR_STATE,
        payload: {
          calendarState: type
        }
      });
      setPrevType(type);
    }

    // Изменяем ключ каждый раз как пытаемся перейти на даты которые находятся за пределами min/max ограничений.
    // Это необходимо для того чтобы screen-reader корректно озвучивал уведомление aria-live="assertive"
    // о том что нет доступных дат
    var handleKeyDown = React.useCallback(function (event) {
      setOutOfRangeKey(function (previousState) {
        return Number(!previousState);
      });
      onKeyDown(event);
    }, [onKeyDown]);
    return /*#__PURE__*/React.createElement(Root, _rollupPluginBabelHelpers.extends({
      ref: outerRootRef,
      "aria-label": "\u0412\u044B\u0431\u043E\u0440 \u0434\u0430\u0442\u044B"
    }, rest), isOutOfRange && /*#__PURE__*/React.createElement(CalendarBase_styles.IsOutOfRange, {
      key: outOfRangeKey,
      "aria-atomic": "true",
      role: "alert",
      "aria-live": "assertive",
      "aria-relevant": "additions"
    }, "\u0414\u0430\u043B\u0435\u0435 \u043D\u0435\u0442 \u0434\u043E\u0441\u0442\u0443\u043F\u043D\u044B\u0445 \u0434\u0430\u0442."), /*#__PURE__*/React.createElement(CalendarHeader.CalendarHeader, {
      firstDate: date,
      startYear: startYear,
      type: calendarState,
      onPrev: handlePrev,
      onNext: handleNext,
      onUpdateCalendarState: handleUpdateCalendarState
    }), calendarState === types.CalendarState.Days && /*#__PURE__*/React.createElement(CalendarDays.CalendarDays, {
      eventList: eventList,
      disabledList: disabledList,
      min: min,
      max: max,
      includeEdgeDates: includeEdgeDates,
      value: externalValue,
      date: date,
      hoveredDay: hoveredDay,
      selectIndexes: selectIndexes,
      onChangeDay: handleOnChangeDay,
      onHoverDay: setHoveredDay,
      onSetSelected: onSelectIndexes,
      onKeyDown: handleKeyDown,
      outerRefs: outerRefs
    }), calendarState === types.CalendarState.Months && /*#__PURE__*/React.createElement(CalendarMonths.CalendarMonths, {
      date: date,
      selectIndexes: selectIndexes,
      onChangeMonth: handleOnChangeMonth,
      onSetSelected: onSelectIndexes,
      onKeyDown: onKeyDown,
      outerRefs: outerRefs
    }), calendarState === types.CalendarState.Years && /*#__PURE__*/React.createElement(CalendarYears.CalendarYears, {
      date: date,
      startYear: startYear,
      selectIndexes: selectIndexes,
      onChangeYear: handleOnChangeYear,
      onSetSelected: onSelectIndexes,
      onKeyDown: onKeyDown,
      outerRefs: outerRefs
    }));
  });
};
var calendarBaseConfig = {
  name: 'CalendarBase',
  tag: 'div',
  layout: calendarBaseRoot,
  base: CalendarBase_styles.StyledCalendar,
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

exports.calendarBaseConfig = calendarBaseConfig;
exports.calendarBaseRoot = calendarBaseRoot;
