// @salutejs/plasma-new-hope/cjs/components/Notification/Notification.styles.js
import './Notification.styles_7h8hp0.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
var Button$1 = require('../Button/Button.js');
require('@linaria/core');
var common = require('../../engines/common.js');
var Notification_tokens = require('./Notification.tokens.js');
var Notification_types = require('./Notification.types.js');

var StyledNotification = "sb12p";
var mergedButtonConfig = /*#__PURE__*/common.mergeConfig(Button$1.buttonConfig);
var Button = /*#__PURE__*/common.component(mergedButtonConfig);
var _exp = function _exp() {
  return Button;
};
var CloseIconWrapper = /*#__PURE__*/react.styled(_exp())({
  name: "CloseIconWrapper",
  "class": "crvd7hm",
  propsAsIs: true
});
var Wrapper = /*#__PURE__*/react.styled('div')({
  name: "Wrapper",
  "class": "wg1ga5v",
  propsAsIs: false
});
var _exp10 = function _exp10() {
  return function (_ref) {
    var iconPlacement = _ref.iconPlacement;
    return iconPlacement === Notification_types.placements.left ? "var(".concat(Notification_tokens.tokens.buttonsMarginLeft, ")") : 'unset';
  };
};
var ButtonsWrapper = /*#__PURE__*/react.styled('div')({
  name: "ButtonsWrapper",
  "class": "b1amsn8a",
  propsAsIs: false,
  vars: {
    "b1amsn8a-0": [/*#__PURE__*/_exp10()]
  }
});
var _exp14 = function _exp14() {
  return function (_ref2) {
    var iconPlacement = _ref2.iconPlacement;
    return iconPlacement === Notification_types.placements.left ? "var(".concat(Notification_tokens.tokens.contentLeftIconMargin, ")") : 'unset';
  };
};
var _exp15 = function _exp15() {
  return function (_ref3) {
    var iconPlacement = _ref3.iconPlacement;
    return iconPlacement === Notification_types.placements.top ? "var(".concat(Notification_tokens.tokens.contentTopIconMargin, ")") : 'unset';
  };
};
var IconWrapper = /*#__PURE__*/react.styled('div')({
  name: "IconWrapper",
  "class": "i14raqr9",
  propsAsIs: false,
  vars: {
    "i14raqr9-0": [/*#__PURE__*/_exp14()],
    "i14raqr9-1": [/*#__PURE__*/_exp15()]
  }
});
var _exp18 = function _exp18() {
  return function (_ref4) {
    var iconPlacement = _ref4.iconPlacement;
    return iconPlacement === Notification_types.placements.left ? 'flex' : 'block';
  };
};
var _exp19 = function _exp19() {
  return function (_ref5) {
    var iconPlacement = _ref5.iconPlacement;
    return iconPlacement ? "var(".concat(Notification_tokens.tokens.contentPaddingTop, ")") : "var(".concat(Notification_tokens.tokens.contentPaddingTopWithoutIcon, ")");
  };
};
var ContentBox = /*#__PURE__*/react.styled('div')({
  name: "ContentBox",
  "class": "c157peez",
  propsAsIs: false,
  vars: {
    "c157peez-0": [/*#__PURE__*/_exp18()],
    "c157peez-1": [/*#__PURE__*/_exp19()]
  }
});
var _exp25 = function _exp25() {
  return function (_ref6) {
    var iconPlacement = _ref6.iconPlacement;
    return !iconPlacement || iconPlacement === Notification_types.placements.top ? "var(".concat(Notification_tokens.tokens.textboxPaddingTopWithTopIcon, ")") : "var(".concat(Notification_tokens.tokens.textboxPaddingTop, ")");
  };
};
var _exp26 = function _exp26() {
  return function (_ref7) {
    var showCloseIcon = _ref7.showCloseIcon,
      iconPlacement = _ref7.iconPlacement;
    return showCloseIcon && iconPlacement !== Notification_types.placements.top ? "var(".concat(Notification_tokens.tokens.textboxPaddingRightWithCloseIcon, ")") : "var(".concat(Notification_tokens.tokens.textboxPaddingRight, ")");
  };
};
var TextBox = /*#__PURE__*/react.styled('div')({
  name: "TextBox",
  "class": "t1vmb0f2",
  propsAsIs: false,
  vars: {
    "t1vmb0f2-0": [/*#__PURE__*/_exp25()],
    "t1vmb0f2-1": [/*#__PURE__*/_exp26()]
  }
});
var StyledTitle = /*#__PURE__*/react.styled('div')({
  name: "StyledTitle",
  "class": "s1k4e58v",
  propsAsIs: false
});
var StyledContent = /*#__PURE__*/react.styled('div')({
  name: "StyledContent",
  "class": "s92kmpm",
  propsAsIs: false
});
var StyledItemWrapper = /*#__PURE__*/react.styled('div')({
  name: "StyledItemWrapper",
  "class": "s18m0x6v",
  propsAsIs: false
});
var StyledRoot = /*#__PURE__*/react.styled('div')({
  name: "StyledRoot",
  "class": "sfe5ql2",
  propsAsIs: false
});

exports.ButtonsWrapper = ButtonsWrapper;
exports.CloseIconWrapper = CloseIconWrapper;
exports.ContentBox = ContentBox;
exports.IconWrapper = IconWrapper;
exports.StyledContent = StyledContent;
exports.StyledItemWrapper = StyledItemWrapper;
exports.StyledNotification = StyledNotification;
exports.StyledRoot = StyledRoot;
exports.StyledTitle = StyledTitle;
exports.TextBox = TextBox;
exports.Wrapper = Wrapper;
