// @salutejs/plasma-new-hope/es/components/Cell/Cell.styles.js
import './Cell.styles_ygfo1w.css';
import styled from 'styled-components';
import { tokens } from './Cell.tokens.js';

var alignToFlex = {
  center: 'center',
  top: 'flex-start',
  bottom: 'flex-end'
};
var base = "b1wikybr";
var CellContentWrapper = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-taup5i-0"
})(["display:flex;flex:auto;align-items:stretch;justify-content:space-between;gap:var(", ");height:auto;"], tokens.cellGap);
var CellContent = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-taup5i-1"
})(["display:flex;align-items:center;height:auto;"]);
var CellLeft = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-taup5i-2"
})(["display:flex;align-items:", ";"], function (_ref) {
  var align = _ref.align;
  return alignToFlex[align];
});
var CellRight = /*#__PURE__*/styled.div.withConfig({
  componentId: "plasma-new-hope__sc-taup5i-3"
})(["display:flex;align-items:", ";text-align:right;"], function (_ref2) {
  var align = _ref2.align;
  return alignToFlex[align];
});

export { CellContent, CellContentWrapper, CellLeft, CellRight, base };
