// @salutejs/plasma-new-hope/cjs/components/Combobox/ui/ComboboxHeader/ComboboxHeader.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('react');
require('../../../../utils/react.js');
require('@salutejs/plasma-core');
require('@linaria/react');
var DropdownHeader = require('../../../Dropdown/ui/DropdownHeader/DropdownHeader.js');
require('@linaria/core');
require('focus-visible');
require('react-dom');
require('react-popper');
require('styled-components');
var base = require('./variations/_view/base.js');
var base$1 = require('./variations/_size/base.js');
var ComboboxHeader_styles = require('./ComboboxHeader.styles.js');

var comboboxHeaderRoot = DropdownHeader.dropdownHeaderRoot;
var comboboxHeaderConfig = {
  name: 'ComboboxHeader',
  tag: 'div',
  layout: comboboxHeaderRoot,
  base: ComboboxHeader_styles.base,
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    }
  },
  defaults: {
    view: 'primary',
    size: 'm'
  }
};

exports.comboboxHeaderConfig = comboboxHeaderConfig;
exports.comboboxHeaderRoot = comboboxHeaderRoot;
