// @salutejs/plasma-new-hope/cjs/components/Pagination/Pagination.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var index$1 = require('../../utils/index.js');
var Pagination_styles = require('./Pagination.styles.js');
var index = require('./utils/index.js');
var base = require('./variations/_view/base.js');
var base$2 = require('./variations/_type/base.js');
var base$1 = require('./variations/_size/base.js');
var Pagination_tokens = require('./Pagination.tokens.js');
var PaginationQuickJumpToPage = require('./ui/PaginationQuickJumpToPage/PaginationQuickJumpToPage.js');
var PaginationSelectPerPage = require('./ui/PaginationSelectPerPage/PaginationSelectPerPage.js');
var isNumber = require('../../utils/isNumber.js');

var _PaginationShorter;
var _excluded = ["value", "perPage", "slots", "view", "viewCurrentPage", "type", "size", "pilled", "square", "disabledPages", "count", "hasQuickJump", "hasPerPage", "perPageList", "isCommonButtonStyles", "placeholderQuickJump", "textQuickJump", "textPerPage", "leftContent", "rightContent", "onChangePageValue", "onChangePerPageValue"];
var paginationRoot = function paginationRoot(Root) {
  return /*#__PURE__*/React.forwardRef(function (_ref, ref) {
    var _ref$value = _ref.value,
      value = _ref$value === void 0 ? index.defaultValues.value : _ref$value,
      perPage = _ref.perPage,
      _ref$slots = _ref.slots,
      slots = _ref$slots === void 0 ? index.defaultValues.slots : _ref$slots,
      view = _ref.view,
      viewCurrentPage = _ref.viewCurrentPage,
      _ref$type = _ref.type,
      type = _ref$type === void 0 ? index.defaultValues.type : _ref$type,
      size = _ref.size,
      _ref$pilled = _ref.pilled,
      pilled = _ref$pilled === void 0 ? index.defaultValues.pilled : _ref$pilled,
      _ref$square = _ref.square,
      square = _ref$square === void 0 ? index.defaultValues.square : _ref$square,
      _ref$disabledPages = _ref.disabledPages,
      disabledPages = _ref$disabledPages === void 0 ? index.defaultValues.disabledPages : _ref$disabledPages,
      _ref$count = _ref.count,
      count = _ref$count === void 0 ? index.defaultValues.value : _ref$count,
      _ref$hasQuickJump = _ref.hasQuickJump,
      hasQuickJump = _ref$hasQuickJump === void 0 ? index.defaultValues.hasQuickJump : _ref$hasQuickJump,
      _ref$hasPerPage = _ref.hasPerPage,
      hasPerPage = _ref$hasPerPage === void 0 ? index.defaultValues.hasPerPage : _ref$hasPerPage,
      _ref$perPageList = _ref.perPageList,
      perPageList = _ref$perPageList === void 0 ? index.defaultValues.perPageList : _ref$perPageList,
      _ref$isCommonButtonSt = _ref.isCommonButtonStyles,
      isCommonButtonStyles = _ref$isCommonButtonSt === void 0 ? index.defaultValues.isCommonButtonStyles : _ref$isCommonButtonSt,
      _ref$placeholderQuick = _ref.placeholderQuickJump,
      placeholderQuickJump = _ref$placeholderQuick === void 0 ? index.defaultValues.placeholderQuickJump : _ref$placeholderQuick,
      _ref$textQuickJump = _ref.textQuickJump,
      textQuickJump = _ref$textQuickJump === void 0 ? index.defaultValues.textQuickJump : _ref$textQuickJump,
      _ref$textPerPage = _ref.textPerPage,
      textPerPage = _ref$textPerPage === void 0 ? index.defaultValues.textPerPage : _ref$textPerPage,
      leftContent = _ref.leftContent,
      rightContent = _ref.rightContent,
      onChangePageValue = _ref.onChangePageValue,
      onChangePerPageValue = _ref.onChangePerPageValue,
      rest = _rollupPluginBabelHelpers.objectWithoutProperties(_ref, _excluded);
    var _useState = React.useState(1),
      _useState2 = _rollupPluginBabelHelpers.slicedToArray(_useState, 2),
      page = _useState2[0],
      setPageValue = _useState2[1];
    var _useState3 = React.useState(perPage),
      _useState4 = _rollupPluginBabelHelpers.slicedToArray(_useState3, 2),
      perPageValue = _useState4[0],
      setPerPageValue = _useState4[1];
    var _useState5 = React.useState(value),
      _useState6 = _rollupPluginBabelHelpers.slicedToArray(_useState5, 2),
      pages = _useState6[0],
      setPagesValue = _useState6[1];
    var _useState7 = React.useState(null),
      _useState8 = _rollupPluginBabelHelpers.slicedToArray(_useState7, 2),
      sections = _useState8[0],
      setSections = _useState8[1];
    var typeClass = Pagination_tokens.classes["".concat(type, "Type")];
    var roundedClass = pilled ? 'circle-circle' : 'square-square';
    var withHasPerPageSelect = hasPerPage ? Pagination_tokens.classes.withHasPerPageSelect : undefined;
    var withHasQuickJump = hasQuickJump ? Pagination_tokens.classes.withHasQuickJump : undefined;
    var disabled = disabledPages.map(function (el) {
      return isNumber.isNumber(el) ? Number(el) : null;
    });
    var handlerSetPages = function handlerSetPages(newPerPage) {
      setPagesValue(Math.ceil(count / (newPerPage || (hasPerPage ? index.defaultValues.perPage : 1))));
    };
    var handlerSetPage = function handlerSetPage(newPageValue) {
      var pageValue = newPageValue !== null && newPageValue !== void 0 ? newPageValue : index.defaultValues.value;
      if (pageValue > pages) {
        pageValue = pages;
      }
      if (pageValue < 1) {
        pageValue = 1;
      }
      if (disabled.includes(pageValue)) {
        return;
      }
      setPageValue(pageValue);
      onChangePageValue === null || onChangePageValue === void 0 || onChangePageValue(pageValue);
    };
    var handlerSetPerPage = function handlerSetPerPage(newPerPageValue) {
      handlerSetPage(1);
      setPerPageValue(newPerPageValue);
      onChangePerPageValue === null || onChangePerPageValue === void 0 || onChangePerPageValue(newPerPageValue);
      handlerSetPages(newPerPageValue);
    };
    var isActiveButton = function isActiveButton(checkPage) {
      return page === checkPage ? Pagination_tokens.classes.paginationPageButtonActive : '';
    };
    React.useEffect(function () {
      handlerSetPages(perPage);
    }, [perPage, count, handlerSetPages]);
    React.useEffect(function () {
      handlerSetPage(value);
    }, [value]);
    React.useEffect(function () {
      setSections(index.getSections(page, pages, slots));
    }, [page, slots, pages, setSections, index.getSections]);
    return /*#__PURE__*/React.createElement(Root, _rollupPluginBabelHelpers.extends({
      size: size,
      view: view,
      viewCurrentPage: viewCurrentPage,
      className: index$1.cx(Pagination_tokens.classes.wrapper, typeClass),
      ref: ref
    }, rest), /*#__PURE__*/React.createElement(Pagination_styles.PaginationRoot, {
      className: typeClass
    }, /*#__PURE__*/React.createElement(Pagination_styles.PaginationPages, null, leftContent && /*#__PURE__*/React.createElement(Pagination_styles.PaginationButtonGroup, {
      isCommonButtonStyles: isCommonButtonStyles
    }, leftContent), sections && sections.map(function (section, indexSection) {
      return /*#__PURE__*/React.createElement(React.Fragment, null, indexSection !== 0 && (_PaginationShorter || (_PaginationShorter = /*#__PURE__*/React.createElement(Pagination_styles.PaginationShorter, {
        stretching: "fixed",
        disabled: true
      }, "..."))), /*#__PURE__*/React.createElement(Pagination_styles.PaginationSection, {
        key: indexSection
      }, section.map(function (pageValue) {
        return /*#__PURE__*/React.createElement(Pagination_styles.PaginationButton, {
          square: square,
          pin: roundedClass,
          key: pageValue,
          onClick: function onClick() {
            return handlerSetPage(pageValue);
          },
          disabled: disabled.includes(pageValue),
          className: index$1.cx(Pagination_tokens.classes.paginationPageButton, isActiveButton(pageValue))
        }, pageValue);
      })));
    }), rightContent && /*#__PURE__*/React.createElement(Pagination_styles.PaginationButtonGroup, {
      isCommonButtonStyles: isCommonButtonStyles
    }, rightContent)), /*#__PURE__*/React.createElement(Pagination_styles.PaginationActions, {
      className: index$1.cx(Pagination_tokens.classes.actions, typeClass, withHasPerPageSelect, withHasQuickJump)
    }, hasQuickJump && /*#__PURE__*/React.createElement(PaginationQuickJumpToPage.PaginationQuickJumpToPage, {
      placeholderQuickJump: placeholderQuickJump,
      textQuickJump: textQuickJump,
      onChangeValue: handlerSetPage
    }), hasPerPage && /*#__PURE__*/React.createElement(PaginationSelectPerPage.PaginationSelectPerPage, {
      textPerPage: textPerPage,
      value: perPageValue,
      valuesList: perPageList,
      onChangeValue: handlerSetPerPage
    }))));
  });
};
var paginationConfig = {
  name: 'Pagination',
  tag: 'div',
  layout: paginationRoot,
  base: '',
  variations: {
    view: {
      css: base.base
    },
    size: {
      css: base$1.base
    },
    type: {
      css: base$2.base
    }
  },
  defaults: {
    view: 'clear',
    size: 'm'
  }
};

exports.paginationConfig = paginationConfig;
exports.paginationRoot = paginationRoot;
