// @salutejs/plasma-new-hope/cjs/components/Pagination/Pagination.styles.js
import './Pagination.styles_y6gcva.css';
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('@linaria/react');
var Button$1 = require('../Button/Button.js');
var ButtonGroup$1 = require('../ButtonGroup/ButtonGroup.js');
require('@linaria/core');
var common = require('../../engines/common.js');

var mergedButtonConfig = /*#__PURE__*/common.mergeConfig(Button$1.buttonConfig);
var Button = /*#__PURE__*/common.component(mergedButtonConfig);
var mergedButtonGroupConfig = /*#__PURE__*/common.mergeConfig(ButtonGroup$1.buttonGroupConfig);
var ButtonGroup = /*#__PURE__*/common.component(mergedButtonGroupConfig);
var PaginationRoot = /*#__PURE__*/react.styled('div')({
  name: "PaginationRoot",
  "class": "p1m4gnut",
  propsAsIs: false
});
var PaginationPages = /*#__PURE__*/react.styled('div')({
  name: "PaginationPages",
  "class": "p195n7xj",
  propsAsIs: false
});
var PaginationActions = /*#__PURE__*/react.styled('div')({
  name: "PaginationActions",
  "class": "p19axp12",
  propsAsIs: false
});
var PaginationSection = /*#__PURE__*/react.styled('div')({
  name: "PaginationSection",
  "class": "p1rlzvhv",
  propsAsIs: false
});
var _exp7 = function _exp7() {
  return ButtonGroup;
};
var PaginationButtonGroup = /*#__PURE__*/react.styled(_exp7())({
  name: "PaginationButtonGroup",
  "class": "pivhg1z",
  propsAsIs: true
});
var _exp44 = function _exp44() {
  return Button;
};
var PaginationButton = /*#__PURE__*/react.styled(_exp44())({
  name: "PaginationButton",
  "class": "pshury3",
  propsAsIs: true
});
var _exp93 = function _exp93() {
  return PaginationButton;
};
var PaginationShorter = /*#__PURE__*/react.styled(_exp93())({
  name: "PaginationShorter",
  "class": "pwlctp1",
  propsAsIs: true
});

exports.PaginationActions = PaginationActions;
exports.PaginationButton = PaginationButton;
exports.PaginationButtonGroup = PaginationButtonGroup;
exports.PaginationPages = PaginationPages;
exports.PaginationRoot = PaginationRoot;
exports.PaginationSection = PaginationSection;
exports.PaginationShorter = PaginationShorter;
