// @salutejs/plasma-core/mixins/applyView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.views = exports.applyView = void 0;
var _tokens = /*#__PURE__*/require("../tokens");
/**
 * Общие цветовые стили оформления кнопок, баджей и т.п.
 */
var views = exports.views = {
  accent: {
    backgroundColor: _tokens.accent,
    color: _tokens.white
  },
  primary: {
    backgroundColor: _tokens.buttonAccent,
    color: _tokens.white
  },
  secondary: {
    backgroundColor: _tokens.buttonSecondary,
    color: _tokens.text
  },
  success: {
    backgroundColor: _tokens.buttonSuccess,
    color: _tokens.white
  },
  warning: {
    backgroundColor: _tokens.buttonWarning,
    color: _tokens.white
  },
  critical: {
    backgroundColor: _tokens.buttonCritical,
    color: _tokens.white
  },
  checked: {
    backgroundColor: _tokens.buttonChecked,
    /**
     * FixMe: Заменить на соответствующий цвет при исправлении токенов дизайнерами
     */
    color: _tokens.background
  },
  clear: {
    backgroundColor: _tokens.transparent,
    color: _tokens.text
  },
  overlay: {
    backgroundColor: _tokens.blackSecondary,
    color: _tokens.white
  }
};
/**
 * Миксин для применения общих цветовых стилей.
 */
var applyView = exports.applyView = function applyView(_ref) {
  var view = _ref.view;
  return views[view];
};
