// @salutejs/plasma-core/mixins/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  addFocus: true,
  syntheticFocus: true,
  applyBlur: true,
  blurs: true,
  applyDisabled: true,
  applyEllipsis: true,
  applyMaxLines: true,
  applyNoSelect: true,
  applyRoundness: true,
  radiuses: true,
  applySkeletonGradient: true,
  applyView: true,
  views: true,
  applyScrollSnap: true,
  applyHyphens: true,
  applySpacing: true,
  spacing: true
};
Object.defineProperty(exports, "addFocus", {
  enumerable: true,
  get: function get() {
    return _addFocus.addFocus;
  }
});
Object.defineProperty(exports, "applyBlur", {
  enumerable: true,
  get: function get() {
    return _blur.applyBlur;
  }
});
Object.defineProperty(exports, "applyDisabled", {
  enumerable: true,
  get: function get() {
    return _applyDisabled.applyDisabled;
  }
});
Object.defineProperty(exports, "applyEllipsis", {
  enumerable: true,
  get: function get() {
    return _applyEllipsis.applyEllipsis;
  }
});
Object.defineProperty(exports, "applyHyphens", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.applyHyphens;
  }
});
Object.defineProperty(exports, "applyMaxLines", {
  enumerable: true,
  get: function get() {
    return _applyMaxLines.applyMaxLines;
  }
});
Object.defineProperty(exports, "applyNoSelect", {
  enumerable: true,
  get: function get() {
    return _applyNoSelect.applyNoSelect;
  }
});
Object.defineProperty(exports, "applyRoundness", {
  enumerable: true,
  get: function get() {
    return _applyRoundness.applyRoundness;
  }
});
Object.defineProperty(exports, "applyScrollSnap", {
  enumerable: true,
  get: function get() {
    return _applyScrollSnap.applyScrollSnap;
  }
});
Object.defineProperty(exports, "applySkeletonGradient", {
  enumerable: true,
  get: function get() {
    return _applySkeletonGradient.applySkeletonGradient;
  }
});
Object.defineProperty(exports, "applySpacing", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.applySpacing;
  }
});
Object.defineProperty(exports, "applyView", {
  enumerable: true,
  get: function get() {
    return _applyView.applyView;
  }
});
Object.defineProperty(exports, "blurs", {
  enumerable: true,
  get: function get() {
    return _blur.blurs;
  }
});
Object.defineProperty(exports, "radiuses", {
  enumerable: true,
  get: function get() {
    return _applyRoundness.radiuses;
  }
});
Object.defineProperty(exports, "spacing", {
  enumerable: true,
  get: function get() {
    return _plasmaTypo.spacing;
  }
});
Object.defineProperty(exports, "syntheticFocus", {
  enumerable: true,
  get: function get() {
    return _addFocus.syntheticFocus;
  }
});
Object.defineProperty(exports, "views", {
  enumerable: true,
  get: function get() {
    return _applyView.views;
  }
});
var _addFocus = /*#__PURE__*/require("./addFocus");
var _blur = /*#__PURE__*/require("./blur");
var _applyDisabled = /*#__PURE__*/require("./applyDisabled");
var _applyEllipsis = /*#__PURE__*/require("./applyEllipsis");
var _applyMaxLines = /*#__PURE__*/require("./applyMaxLines");
var _applyNoSelect = /*#__PURE__*/require("./applyNoSelect");
var _applyRoundness = /*#__PURE__*/require("./applyRoundness");
var _applySkeletonGradient = /*#__PURE__*/require("./applySkeletonGradient");
var _applyView = /*#__PURE__*/require("./applyView");
var _applyScrollSnap = /*#__PURE__*/require("./applyScrollSnap");
var _shadows = /*#__PURE__*/require("./shadows");
Object.keys(_shadows).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _shadows[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _shadows[key];
    }
  });
});
var _plasmaTypo = /*#__PURE__*/require("@salutejs/plasma-typo");
