// @salutejs/plasma-new-hope/cjs/mixins/applySpacing.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../_virtual/_rollupPluginBabelHelpers.js');

var baseSpacing = {
  0: '0',
  2: '0.125rem',
  4: '0.25rem',
  8: '0.5rem',
  10: '0.625rem',
  12: '0.75rem',
  16: '1rem',
  18: '1.125rem',
  20: '1.25rem',
  24: '1.5rem',
  32: '2rem'
};
var spacing = /*#__PURE__*/_rollupPluginBabelHelpers.objectSpread2( /*#__PURE__*/_rollupPluginBabelHelpers.objectSpread2({}, baseSpacing), {}, {
  '0x': baseSpacing[0],
  '1x': baseSpacing[2],
  '2x': baseSpacing[4],
  '4x': baseSpacing[8],
  '5x': baseSpacing[10],
  '6x': baseSpacing[12],
  '8x': baseSpacing[16],
  '9x': baseSpacing[18],
  '10x': baseSpacing[20],
  '12x': baseSpacing[24],
  '16x': baseSpacing[32]
});
var createSpacingFn = function createSpacingFn(params) {
  return function (value) {
    return params.reduce(function (acc, cur) {
      return _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, acc), {}, _rollupPluginBabelHelpers.defineProperty({}, cur, spacing[value]));
    }, {});
  };
};
var m = /*#__PURE__*/createSpacingFn(['margin']);
var mt = /*#__PURE__*/createSpacingFn(['marginTop']);
var mr = /*#__PURE__*/createSpacingFn(['marginRight']);
var ml = /*#__PURE__*/createSpacingFn(['marginLeft']);
var mb = /*#__PURE__*/createSpacingFn(['marginBottom']);
var mx = /*#__PURE__*/createSpacingFn(['marginLeft', 'marginRight']);
var my = /*#__PURE__*/createSpacingFn(['marginTop', 'marginBottom']);
var p = /*#__PURE__*/createSpacingFn(['padding']);
var pt = /*#__PURE__*/createSpacingFn(['paddingTop']);
var pr = /*#__PURE__*/createSpacingFn(['paddingRight']);
var pl = /*#__PURE__*/createSpacingFn(['paddingLeft']);
var pb = /*#__PURE__*/createSpacingFn(['paddingBottom']);
var px = /*#__PURE__*/createSpacingFn(['paddingLeft', 'paddingRight']);
var py = /*#__PURE__*/createSpacingFn(['paddingTop', 'paddingBottom']);
var spacingFns = {
  m: m,
  mt: mt,
  mr: mr,
  ml: ml,
  mb: mb,
  mx: mx,
  my: my,
  p: p,
  pt: pt,
  pr: pr,
  pl: pl,
  pb: pb,
  px: px,
  py: py
};
var spacingFnsList = /*#__PURE__*/Object.entries(spacingFns);
/**
 * Миксин для добавления пропсов стандартных отступов в компонент.
 */
var applySpacing = function applySpacing(props) {
  return spacingFnsList.reduce(function (styles, _ref) {
    var _ref2 = _rollupPluginBabelHelpers.slicedToArray(_ref, 2),
      fnName = _ref2[0],
      fn = _ref2[1];
    if (fnName in props) {
      return _rollupPluginBabelHelpers.objectSpread2(_rollupPluginBabelHelpers.objectSpread2({}, styles), fn(props[fnName]));
    }
    return styles;
  }, {});
};

exports.applySpacing = applySpacing;
exports.spacing = spacing;
exports.spacingFns = spacingFns;
