// @salutejs/plasma-web/components/Drawer/Drawer.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("@salutejs/plasma-new-hope/styled-components");
var _templateObject, _templateObject2, _templateObject3, _templateObject4;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
var config = exports.config = {
  defaults: {
    view: 'default',
    size: 'm'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/(0, _styledComponents.css)(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": var(--surface-solid-card);\n                ", ": 0 3.75rem 7rem -0.5rem rgba(0, 0, 0, 0.08);\n                ", ": var(--surface-transparent-primary);\n                ", ": var(--overlay-blur);\n                ", ": var(--overlay-soft);\n            "])), _styledComponents.drawerTokens.background, _styledComponents.drawerTokens.shadow, _styledComponents.drawerTokens.contentBackgroundColor, _styledComponents.drawerTokens.drawerOverlayWithBlurColor, _styledComponents.drawerTokens.drawerOverlayColor)
    },
    size: {
      m: /*#__PURE__*/(0, _styledComponents.css)(_templateObject2 || (_templateObject2 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": 1.5rem;\n            "])), _styledComponents.drawerTokens.padding)
    },
    borderRadius: {
      none: /*#__PURE__*/(0, _styledComponents.css)(_templateObject3 || (_templateObject3 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": 0;\n            "])), _styledComponents.drawerTokens.borderRadius),
      "default": /*#__PURE__*/(0, _styledComponents.css)(_templateObject4 || (_templateObject4 = /*#__PURE__*/_taggedTemplateLiteral(["\n                ", ": 1.25rem;\n            "])), _styledComponents.drawerTokens.borderRadius)
    }
  }
};
