// @salutejs/plasma-new-hope/styled-components/cjs/components/Skeleton/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "RectSkeleton", {
  enumerable: true,
  get: function get() {
    return _RectSkeleton.RectSkeleton;
  }
});
Object.defineProperty(exports, "lineSkeletonConfig", {
  enumerable: true,
  get: function get() {
    return _LineSkeleton.lineSkeletonConfig;
  }
});
Object.defineProperty(exports, "lineSkeletonRoot", {
  enumerable: true,
  get: function get() {
    return _LineSkeleton.lineSkeletonRoot;
  }
});
Object.defineProperty(exports, "skeletonTokens", {
  enumerable: true,
  get: function get() {
    return _tokens.tokens;
  }
});
Object.defineProperty(exports, "textSkeleton", {
  enumerable: true,
  get: function get() {
    return _TextSkeleton.textSkeleton;
  }
});
Object.defineProperty(exports, "withSkeleton", {
  enumerable: true,
  get: function get() {
    return _withSkeleton.withSkeleton;
  }
});
var _LineSkeleton = /*#__PURE__*/require("./LineSkeleton");
var _TextSkeleton = /*#__PURE__*/require("./TextSkeleton");
var _RectSkeleton = /*#__PURE__*/require("./RectSkeleton");
var _withSkeleton = /*#__PURE__*/require("./hoc/withSkeleton");
var _tokens = /*#__PURE__*/require("./tokens");
