// @salutejs/plasma-new-hope/es/components/Typography/Old/Subtitle/Subtitle.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends, objectSpread2 as _objectSpread2 } from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef } from 'react';
import { applySpacing } from '../../../../mixins/applySpacing.js';
import { base as base$1 } from '../../variations/_size/base.js';
import { base } from './Subtitle.styles.js';

var _excluded = ["size", "children", "style"];
var subtitleRoot = function subtitleRoot(Root) {
  return /*#__PURE__*/forwardRef(function (props, ref) {
    var size = props.size,
      children = props.children,
      style = props.style,
      rest = _objectWithoutProperties(props, _excluded);
    return /*#__PURE__*/React.createElement(Root, _extends({
      size: size,
      ref: ref,
      style: _objectSpread2(_objectSpread2({}, style), applySpacing(rest))
    }, rest), children);
  });
};
var subtitleConfig = {
  name: 'Subtitle',
  tag: 'div',
  layout: subtitleRoot,
  base: base,
  variations: {
    size: {
      css: base$1
    }
  },
  defaults: {
    size: 'm'
  }
};

export { subtitleConfig, subtitleRoot };
