// @salutejs/plasma-new-hope/styled-components/cjs/examples/plasma_b2c/components/Select/Select.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectItem = exports.SelectHeader = exports.SelectGroup = exports.SelectFooter = exports.SelectDivider = exports.Select = void 0;
var _Select = /*#__PURE__*/require("../../../../components/Select");
var _engines = /*#__PURE__*/require("../../../../engines");
var _Select2 = /*#__PURE__*/require("./Select.config");
var mergedConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Select.selectConfig, _Select2.config);
var Select = exports.Select = /*#__PURE__*/(0, _engines.component)(mergedConfig);
var mergedItemConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Select.selectItemConfig);
var SelectItem = exports.SelectItem = /*#__PURE__*/(0, _engines.component)(mergedItemConfig);
var mergedDividerConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Select.selectDividerConfig);
var SelectDivider = exports.SelectDivider = /*#__PURE__*/(0, _engines.component)(mergedDividerConfig);
var mergedFooterConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Select.selectFooterConfig);
var SelectFooter = exports.SelectFooter = /*#__PURE__*/(0, _engines.component)(mergedFooterConfig);
var mergedGroupConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Select.selectGroupConfig);
var SelectGroup = exports.SelectGroup = /*#__PURE__*/(0, _engines.component)(mergedGroupConfig);
var mergedHeaderConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Select.selectHeaderConfig);
var SelectHeader = exports.SelectHeader = /*#__PURE__*/(0, _engines.component)(mergedHeaderConfig);
