// @salutejs/plasma-new-hope/styled-components/cjs/examples/plasma_b2c/components/Panel/Panel.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _Panel = /*#__PURE__*/require("../../../../components/Panel");
var config = exports.config = {
  defaults: {
    view: 'default',
    size: 'm',
    borderRadius: 'none'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--surface-solid-card);", ":0 3.75rem 7rem -0.5rem rgba(0,0,0,0.08);", ":var(--surface-transparent-primary);"], _Panel.panelTokens.background, _Panel.panelTokens.shadow, _Panel.panelTokens.contentBackgroundColor)
    },
    size: {
      m: /*#__PURE__*/(0, _styledComponents.css)(["", ":1.5rem;"], _Panel.panelTokens.padding)
    },
    borderRadius: {
      none: /*#__PURE__*/(0, _styledComponents.css)(["", ":0;"], _Panel.panelTokens.borderRadius),
      "default": /*#__PURE__*/(0, _styledComponents.css)(["", ":0.625rem;"], _Panel.panelTokens.borderRadius)
    }
  }
};
