// @salutejs/plasma-web/components/ElasticGrid/ElasticGrid.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _2 = /*#__PURE__*/require(".");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var StyledElasticGrid = /*#__PURE__*/(0, _styledComponents["default"])(_2.ElasticGrid).withConfig({
  componentId: "plasma-web__sc-11ngxdo-0"
})(["width:", ";"], function (_ref) {
  var $width = _ref.$width;
  return $width;
});
var Item = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-11ngxdo-1"
})(["width:100%;height:100px;background-color:#999;border-radius:10px;"]);
var defaultProps = {
  $width: '600px',
  itemsNumber: 5,
  minColWidth: 125,
  gapX: 8,
  gapY: 8
};
var arr = [0, 0, 0, 0, 0];
var Default = exports.Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(StyledElasticGrid, defaultProps, arr.map(function (_, k) {
    return /*#__PURE__*/_react["default"].createElement(Item, {
      key: k
    });
  }));
};
