// @salutejs/plasma-web/components/Switch/Switch.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _styledComponents = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("styled-components"));
var _ = /*#__PURE__*/require(".");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var func = function func() {};
var StyledWrapper = /*#__PURE__*/_styledComponents["default"].div.withConfig({
  componentId: "plasma-web__sc-1blafa5-0"
})(["width:13.75rem;"]);
var defaultProps = {
  label: 'Label',
  disabled: false
};
var Default = exports.Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(StyledWrapper, null, /*#__PURE__*/_react["default"].createElement(_.Switch, _extends({
    value: 0,
    checked: true,
    onChange: func,
    onFocus: func,
    onBlur: func
  }, defaultProps)));
};
