// @salutejs/plasma-new-hope/styled-components/cjs/examples/plasma_web/components/Drawer/Drawer.config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.config = void 0;
var _styledComponents = /*#__PURE__*/require("styled-components");
var _Drawer = /*#__PURE__*/require("../../../../components/Drawer");
var config = exports.config = {
  defaults: {
    view: 'default',
    size: 'm'
  },
  variations: {
    view: {
      "default": /*#__PURE__*/(0, _styledComponents.css)(["", ":var(--surface-solid-card);", ":0 3.75rem 7rem -0.5rem rgba(0,0,0,0.08);", ":var(--surface-transparent-primary);", ":var(--overlay-blur);", ":var(--overlay-soft);"], _Drawer.drawerTokens.background, _Drawer.drawerTokens.shadow, _Drawer.drawerTokens.contentBackgroundColor, _Drawer.drawerTokens.drawerOverlayWithBlurColor, _Drawer.drawerTokens.drawerOverlayColor)
    },
    size: {
      m: /*#__PURE__*/(0, _styledComponents.css)(["", ":1.5rem;"], _Drawer.drawerTokens.padding)
    },
    borderRadius: {
      none: /*#__PURE__*/(0, _styledComponents.css)(["", ":0;"], _Drawer.drawerTokens.borderRadius),
      "default": /*#__PURE__*/(0, _styledComponents.css)(["", ":1.25rem;"], _Drawer.drawerTokens.borderRadius)
    }
  }
};
