// @salutejs/plasma-new-hope/styled-components/cjs/examples/plasma_b2c/components/Header/Header.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HeaderArrow = void 0;
Object.defineProperty(exports, "HeaderContent", {
  enumerable: true,
  get: function get() {
    return _Header.HeaderContent;
  }
});
exports.HeaderRoot = exports.HeaderLogo = void 0;
Object.defineProperty(exports, "HeaderTitle", {
  enumerable: true,
  get: function get() {
    return _Header.HeaderTitle;
  }
});
exports.HeaderTitleWrapper = void 0;
var _Header = /*#__PURE__*/require("../../../../components/Header");
var _engines = /*#__PURE__*/require("../../../../engines");
var _Header2 = /*#__PURE__*/require("./Header.config");
var mergedConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Header.headerRootConfig, _Header2.config);
var mergedLogoConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Header.headerLogoConfig);
var mergedArrowConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Header.headerArrowConfig);
var mergedTitleWrapperConfig = /*#__PURE__*/(0, _engines.mergeConfig)(_Header.headerTitleWrapperConfig);
var HeaderRoot = exports.HeaderRoot = /*#__PURE__*/(0, _engines.component)(mergedConfig);
var HeaderLogo = exports.HeaderLogo = /*#__PURE__*/(0, _engines.component)(mergedLogoConfig);
var HeaderArrow = exports.HeaderArrow = /*#__PURE__*/(0, _engines.component)(mergedArrowConfig);
var HeaderTitleWrapper = exports.HeaderTitleWrapper = /*#__PURE__*/(0, _engines.component)(mergedTitleWrapperConfig);
