// @salutejs/plasma-new-hope/styled-components/cjs/components/Panel/ui/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "panelContentConfig", {
  enumerable: true,
  get: function get() {
    return _PanelContent.panelContentConfig;
  }
});
Object.defineProperty(exports, "panelContentRoot", {
  enumerable: true,
  get: function get() {
    return _PanelContent.panelContentRoot;
  }
});
Object.defineProperty(exports, "panelFooterConfig", {
  enumerable: true,
  get: function get() {
    return _PanelFooter.panelFooterConfig;
  }
});
Object.defineProperty(exports, "panelFooterRoot", {
  enumerable: true,
  get: function get() {
    return _PanelFooter.panelFooterRoot;
  }
});
Object.defineProperty(exports, "panelHeaderConfig", {
  enumerable: true,
  get: function get() {
    return _PanelHeader.panelHeaderConfig;
  }
});
Object.defineProperty(exports, "panelHeaderRoot", {
  enumerable: true,
  get: function get() {
    return _PanelHeader.panelHeaderRoot;
  }
});
var _PanelContent = /*#__PURE__*/require("./PanelContent/PanelContent");
var _PanelFooter = /*#__PURE__*/require("./PanelFooter/PanelFooter");
var _PanelHeader = /*#__PURE__*/require("./PanelHeader/PanelHeader");
