// @salutejs/plasma-new-hope/styled-components/es/components/Dropdown/utils/index.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
import React, { Children, cloneElement, isValidElement } from 'react';
export var getPlacement = function getPlacement(placement) {
  return "".concat(placement, "-start");
};
export var getPlacements = function getPlacements(placements) {
  if (!placements) {
    return;
  }
  var isArray = Array.isArray(placements);
  if (!isArray) {
    return getPlacement(placements);
  }
  return (placements || []).map(function (placement) {
    return getPlacement(placement);
  });
};
export var getCorrectHeight = function getCorrectHeight(listHeight) {
  if (Number.isNaN(Number(listHeight))) {
    return listHeight.toString();
  }
  return "".concat(listHeight, "rem");
};
export var childrenWithProps = function childrenWithProps(children, additionalProps) {
  return Children.map(children, function (child) {
    if ( /*#__PURE__*/React.isValidElement(child)) {
      return /*#__PURE__*/React.cloneElement(child, additionalProps);
    }
    return child;
  });
};
var isReactObject = function isReactObject(element) {
  return _typeof(element) === 'object' || typeof element === 'function';
};
export var getValidComponent = function getValidComponent(element, props) {
  if ( /*#__PURE__*/isValidElement(element)) {
    return /*#__PURE__*/cloneElement(element, props);
  }
  if (isReactObject(element)) {
    var Component = element;
    return /*#__PURE__*/React.createElement(Component, props);
  }
  return element;
};
