// @salutejs/plasma-core/utils/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "IS_REACT_18", {
  enumerable: true,
  get: function get() {
    return _react.IS_REACT_18;
  }
});
Object.defineProperty(exports, "animatedScrollToX", {
  enumerable: true,
  get: function get() {
    return _animatedScrollTo.animatedScrollToX;
  }
});
Object.defineProperty(exports, "animatedScrollToY", {
  enumerable: true,
  get: function get() {
    return _animatedScrollTo.animatedScrollToY;
  }
});
Object.defineProperty(exports, "canUseDOM", {
  enumerable: true,
  get: function get() {
    return _canUseDOM.canUseDOM;
  }
});
Object.defineProperty(exports, "convertRoundnessMatrix", {
  enumerable: true,
  get: function get() {
    return _roundness.convertRoundnessMatrix;
  }
});
Object.defineProperty(exports, "extractTextFrom", {
  enumerable: true,
  get: function get() {
    return _extractTextFrom.extractTextFrom;
  }
});
Object.defineProperty(exports, "formatCurrency", {
  enumerable: true,
  get: function get() {
    return _formatCurrency.formatCurrency;
  }
});
Object.defineProperty(exports, "monthLongName", {
  enumerable: true,
  get: function get() {
    return _formatters.monthLongName;
  }
});
Object.defineProperty(exports, "monthShortName", {
  enumerable: true,
  get: function get() {
    return _formatters.monthShortName;
  }
});
Object.defineProperty(exports, "padZeroNumber", {
  enumerable: true,
  get: function get() {
    return _formatters.padZeroNumber;
  }
});
Object.defineProperty(exports, "safeFlushSync", {
  enumerable: true,
  get: function get() {
    return _react.safeFlushSync;
  }
});
Object.defineProperty(exports, "safeUseId", {
  enumerable: true,
  get: function get() {
    return _react.safeUseId;
  }
});
Object.defineProperty(exports, "setRef", {
  enumerable: true,
  get: function get() {
    return _setRef.setRef;
  }
});
Object.defineProperty(exports, "toCssSize", {
  enumerable: true,
  get: function get() {
    return _toCssSize.toCssSize;
  }
});
Object.defineProperty(exports, "transformStyles", {
  enumerable: true,
  get: function get() {
    return _transformStyles.transformStyles;
  }
});
var _animatedScrollTo = /*#__PURE__*/require("./animatedScrollTo");
var _canUseDOM = /*#__PURE__*/require("./canUseDOM");
var _roundness = /*#__PURE__*/require("./roundness");
var _extractTextFrom = /*#__PURE__*/require("./extractTextFrom");
var _formatters = /*#__PURE__*/require("./formatters");
var _formatCurrency = /*#__PURE__*/require("./formatCurrency");
var _setRef = /*#__PURE__*/require("./setRef");
var _toCssSize = /*#__PURE__*/require("./toCssSize");
var _transformStyles = /*#__PURE__*/require("./transformStyles");
var _react = /*#__PURE__*/require("./react");
