// @salutejs/plasma-web/css/es/components/Typography/Old/index.js
import { component, mergeConfig, bodyConfig, buttonTypographyConfig, captionConfig as captionConfig$1, footnoteConfig, headlineConfig, underlineConfig as underlineConfig$1, subtitleConfig as subtitleConfig$1, paragraphConfig, paragraphTextConfig } from '@salutejs/plasma-new-hope';
import { configBody1, configBody2 } from './Body.config.js';
import { configButton1, configButton2 } from './Button.config.js';
import { config } from './Caption.config.js';
import { configFootnote1, configFootnote2 } from './Footnote.config.js';
import { configHeadline1, configHeadline2, configHeadline3, configHeadline4, configHeadline5 } from './Headline.config.js';
import { configParagraph1, configParagraph2 } from './Paragraph.config.js';
import { configParagraphText1, configParagraphText2 } from './ParagraphText.config.js';
import { config as config$1 } from './Underline.config.js';
import { config as config$2 } from './Subtitle.config.js';

var body1Config = /*#__PURE__*/mergeConfig(bodyConfig, configBody1);
var Body1Component = /*#__PURE__*/component(body1Config);
var body2Config = /*#__PURE__*/mergeConfig(bodyConfig, configBody2);
var Body2Component = /*#__PURE__*/component(body2Config);
var button1Config = /*#__PURE__*/mergeConfig(buttonTypographyConfig, configButton1);
var Button1Component = /*#__PURE__*/component(button1Config);
var button2Config = /*#__PURE__*/mergeConfig(buttonTypographyConfig, configButton2);
var Button2Component = /*#__PURE__*/component(button2Config);
var captionConfig = /*#__PURE__*/mergeConfig(captionConfig$1, config);
var CaptionComponent = /*#__PURE__*/component(captionConfig);
var footnote1Config = /*#__PURE__*/mergeConfig(footnoteConfig, configFootnote1);
var Footnote1Component = /*#__PURE__*/component(footnote1Config);
var footnote2Config = /*#__PURE__*/mergeConfig(footnoteConfig, configFootnote2);
var Footnote2Component = /*#__PURE__*/component(footnote2Config);
var headline1Config = /*#__PURE__*/mergeConfig(headlineConfig, configHeadline1);
var Headline1Component = /*#__PURE__*/component(headline1Config);
var headline2Config = /*#__PURE__*/mergeConfig(headlineConfig, configHeadline2);
var Headline2Component = /*#__PURE__*/component(headline2Config);
var headline3Config = /*#__PURE__*/mergeConfig(headlineConfig, configHeadline3);
var Headline3Component = /*#__PURE__*/component(headline3Config);
var headline4Config = /*#__PURE__*/mergeConfig(headlineConfig, configHeadline4);
var Headline4Component = /*#__PURE__*/component(headline4Config);
var headline5Config = /*#__PURE__*/mergeConfig(headlineConfig, configHeadline5);
var Headline5Component = /*#__PURE__*/component(headline5Config);
var underlineConfig = /*#__PURE__*/mergeConfig(underlineConfig$1, config$1);
var UnderlineComponent = /*#__PURE__*/component(underlineConfig);
var subtitleConfig = /*#__PURE__*/mergeConfig(subtitleConfig$1, config$2);
var SubtitleComponent = /*#__PURE__*/component(subtitleConfig);
var paragraph1Config = /*#__PURE__*/mergeConfig(paragraphConfig, configParagraph1);
var Paragraph1Component = /*#__PURE__*/component(paragraph1Config);
var paragraph2Config = /*#__PURE__*/mergeConfig(paragraphConfig, configParagraph2);
var Paragraph2Component = /*#__PURE__*/component(paragraph2Config);
var paragraphText1Config = /*#__PURE__*/mergeConfig(paragraphTextConfig, configParagraphText1);
var ParagraphText1Component = /*#__PURE__*/component(paragraphText1Config);
var paragraphText2Config = /*#__PURE__*/mergeConfig(paragraphTextConfig, configParagraphText2);
var ParagraphText2Component = /*#__PURE__*/component(paragraphText2Config);
var Body1 = Body1Component;
var Body2 = Body2Component;
var Button1 = Button1Component;
var Button2 = Button2Component;
var Caption = CaptionComponent;
var Footnote1 = Footnote1Component;
var Footnote2 = Footnote2Component;
var Headline1 = Headline1Component;
var Headline2 = Headline2Component;
var Headline3 = Headline3Component;
var Headline4 = Headline4Component;
var Headline5 = Headline5Component;
var P1 = Paragraph1Component;
var P2 = Paragraph2Component;
var ParagraphText1 = ParagraphText1Component;
var ParagraphText2 = ParagraphText2Component;
var Underline = UnderlineComponent;
var Subtitle = SubtitleComponent;

export { Body1, Body2, Button1, Button2, Caption, Footnote1, Footnote2, Headline1, Headline2, Headline3, Headline4, Headline5, P1, P2, ParagraphText1, ParagraphText2, Subtitle, Underline };
