// @salutejs/plasma-hope/es/components/Carousel/Carousel.js
var _excluded = ["index", "scrollSnapType", "scrollAlign", "detectActive", "detectThreshold", "scaleCallback", "scaleResetCallback", "onIndexChange", "paddingStart", "paddingEnd", "children", "ariaLive", "isDragScrollDisabled"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React from 'react';
import { useCarousel, Carousel as CarouselCore, CarouselTrack as StyledCarouselTrack, applyNoSelect } from '@salutejs/plasma-core';
import styled from 'styled-components';
import { useForkRef } from '../../hooks';
import { useDragScroll } from './hooks';
import { collectPackageInfo } from '../../collectPackageInfo';
collectPackageInfo('Carousel/Carousel');
var StyledCarousel = /*#__PURE__*/styled(CarouselCore).withConfig({
  componentId: "plasma-web-core__sc-zadi43-0"
})(["", ";"], applyNoSelect);

/**
 * Компонент для создания списков с прокруткой.
 */
// eslint-disable-next-line prefer-arrow-callback
export var Carousel = /*#__PURE__*/React.forwardRef(function Carousel(_ref, ref) {
  var _ref$index = _ref.index,
    index = _ref$index === void 0 ? 0 : _ref$index,
    _ref$scrollSnapType = _ref.scrollSnapType,
    scrollSnapType = _ref$scrollSnapType === void 0 ? 'mandatory' : _ref$scrollSnapType,
    scrollAlign = _ref.scrollAlign,
    detectActive = _ref.detectActive,
    detectThreshold = _ref.detectThreshold,
    scaleCallback = _ref.scaleCallback,
    scaleResetCallback = _ref.scaleResetCallback,
    onIndexChange = _ref.onIndexChange,
    paddingStart = _ref.paddingStart,
    paddingEnd = _ref.paddingEnd,
    children = _ref.children,
    _ref$ariaLive = _ref.ariaLive,
    ariaLive = _ref$ariaLive === void 0 ? 'off' : _ref$ariaLive,
    _ref$isDragScrollDisa = _ref.isDragScrollDisabled,
    isDragScrollDisabled = _ref$isDragScrollDisa === void 0 ? false : _ref$isDragScrollDisa,
    rest = _objectWithoutProperties(_ref, _excluded);
  var axis = 'x';
  var _useCarousel = useCarousel({
      index: index,
      axis: axis,
      scrollAlign: scrollAlign,
      detectActive: detectActive,
      detectThreshold: detectThreshold,
      scaleCallback: scaleCallback,
      scaleResetCallback: scaleResetCallback,
      onIndexChange: onIndexChange
    }),
    scrollRef = _useCarousel.scrollRef,
    trackRef = _useCarousel.trackRef;
  var handleRef = useForkRef(scrollRef, ref);
  useDragScroll(scrollRef, isDragScrollDisabled);
  return /*#__PURE__*/React.createElement(StyledCarousel, _extends({
    ref: handleRef,
    axis: axis,
    scrollSnapType: scrollSnapType
  }, rest), /*#__PURE__*/React.createElement(StyledCarouselTrack, {
    ref: trackRef,
    axis: axis,
    paddingStart: paddingStart,
    paddingEnd: paddingEnd,
    "aria-live": ariaLive
  }, children));
});
