// @salutejs/plasma-web/components/TextField/TextField.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DeferredValue = exports.Default = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _helpers = /*#__PURE__*/require("../../helpers");
var _ = /*#__PURE__*/require(".");
var _IconPlaceholder, _IconPlaceholder2, _TextField;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var func = function func() {};
var defaultProps = {
  id: 'example-text-field',
  size: 'l',
  label: 'Лейбл',
  animatedHint: undefined,
  placeholder: 'Заполните поле',
  helperText: 'Подсказка к полю',
  status: '',
  disabled: false,
  readOnly: false
};
var status = '';
var Default = exports.Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(_.TextField, _extends({}, defaultProps, {
    value: "\u0417\u043D\u0430\u0447\u0435\u043D\u0438\u0435 \u043F\u043E\u043B\u044F",
    contentLeft: _IconPlaceholder || (_IconPlaceholder = /*#__PURE__*/_react["default"].createElement(_helpers.IconPlaceholder, null)),
    contentRight: _IconPlaceholder2 || (_IconPlaceholder2 = /*#__PURE__*/_react["default"].createElement(_helpers.IconPlaceholder, null)),
    status: status,
    onChange: func,
    onFocus: func,
    onBlur: func,
    onSearch: func
  }));
};
var DeferredValue = exports.DeferredValue = function DeferredValue() {
  return _TextField || (_TextField = /*#__PURE__*/_react["default"].createElement(_.TextField, {
    placeholder: "Wait three seconds...",
    defaultValue: "",
    readOnly: true
  }));
};
