// @salutejs/plasma-web/components/Select/Select.js
"use strict";

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Select = void 0;
var _react = /*#__PURE__*/_interopRequireWildcard( /*#__PURE__*/require("react"));
var _plasmaHope = /*#__PURE__*/require("@salutejs/plasma-hope");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); } // import React, { forwardRef, CSSProperties, Ref } from 'react';
// import type { SelectProps, SelectRefElement } from '@salutejs/plasma-hope';
// import {
//     selectConfig,
//     component,
//     mergeConfig,
//     selectItemConfig,
//     cx,
//     popoverClasses,
// } from '@salutejs/plasma-new-hope/styled-components';
// import { IconDone } from '@salutejs/plasma-icons';
// import styled from 'styled-components';
//
// import { Checkbox } from '../Checkbox';
//
// import { config } from './Select.config';
// const mergedConfig = mergeConfig(selectConfig, config);
// export const SelectComponent = component(mergedConfig);
//
// const mergedItemConfig = mergeConfig(selectItemConfig);
// export const SelectItem = component(mergedItemConfig);
//
// const getColor = (status?: SelectProps['status']): string => {
//     switch (status) {
//         case 'success':
//             return '--text-positive';
//         case 'warning':
//             return '--text-warning';
//         case 'error':
//             return '--text-negative';
//         default:
//             return '--text-secondary';
//     }
// };
//
// const HelperText = styled.div<{ status?: SelectProps['status'] }>`
//     color: ${({ status }) => `var(${getColor(status)})`};
//     margin-top: 0.25rem;
//     font-family: var(--plasma-typo-body-xs-font-family), sans-serif;
//     font-size: var(--plasma-typo-body-xs-font-size);
//     font-style: var(--plasma-typo-body-xs-font-style);
//     font-weight: var(--plasma-typo-body-xs-font-weight);
//     letter-spacing: var(--plasma-typo-body-xs-letter-spacing);
//     line-height: var(--plasma-typo-body-xs-line-height);
// `;
//
// const SelectWrapper = styled.div<{
//     listHeight?: number | CSSProperties['height'];
// }>`
//     width: 100%;
//
//     & .select-height {
//         --plasma-select-height: ${({ listHeight }) =>
//             Number.isNaN(Number(listHeight)) ? listHeight : `${listHeight}rem`};
//     }
// `;
//
// /**
//  * Выпадающий список для использования в формах.
//  * Поддерживает выбор одного или нескольких значений.
//  */
// export const Select = forwardRef<SelectRefElement, SelectProps>(
//     (
//         {
//             value,
//             items,
//             onChange,
//             multiselect,
//             separator,
//             status,
//             placeholder,
//             helperText,
//             disabled,
//             onItemSelect,
//             isOpen,
//             listHeight,
//             listOverflow,
//             className,
//             ...rest
//         },
//         ref,
//     ) => {
//         const handleItemSelect: SelectProps['onItemSelect'] = (item, e) => {
//             if (onItemSelect) {
//                 onItemSelect(item, e);
//             }
//         };
//
//         return (
//             <SelectWrapper listHeight={listHeight}>
//                 <SelectComponent
//                     {...(rest as any)}
//                     frame="theme-root"
//                     value={value}
//                     valueType={multiselect ? 'multiple' : 'single'}
//                     placement="bottom"
//                     target="button-like"
//                     disabled={disabled}
//                     onChangeValue={onChange}
//                     label={placeholder}
//                     opened={isOpen}
//                     ref={ref as Ref<HTMLSelectElement>}
//                     className={cx(className, 'select-height')}
//                 >
//                     {items.map((item) => (
//                         <SelectItem
//                             key={item.value}
//                             value={item.value}
//                             text={item.label}
//                             onClick={(e) => handleItemSelect(item, e)}
//                             disabled={item.isDisabled}
//                             contentLeft={
//                                 // eslint-disable-next-line no-nested-ternary
//                                 multiselect ? (
//                                     <Checkbox size="m" />
//                                 ) : item.value === value ? (
//                                     <IconDone size="s" color="inherit" />
//                                 ) : undefined
//                             }
//                         />
//                     ))}
//                 </SelectComponent>
//
//                 {/* TODO: #1153 */}
//                 {helperText && <HelperText status={status}>{helperText}</HelperText>}
//             </SelectWrapper>
//         );
//     },
// );
// TODO: Удалить Select из hope с внедрением компонента SelectNested в new-hope

var SelectBase = /*#__PURE__*/(0, _plasmaHope.Select)({
  design: 'web'
});

/**
 * Выпадающий список для использования в формах.
 * Поддерживает выбор одного или нескольких значений.
 */
var Select = exports.Select = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  return /*#__PURE__*/_react["default"].createElement(SelectBase, _extends({}, props, {
    ref: ref
  }));
});
