// @salutejs/plasma-web/components/Select/Select.perftest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = void 0;
var _react = /*#__PURE__*/_interopRequireDefault( /*#__PURE__*/require("react"));
var _ = /*#__PURE__*/require(".");
var _Select, _Select2;
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
var items = [{
  value: 'each',
  label: 'Каждый'
}, {
  value: 'hunter',
  label: 'Охотник',
  isDisabled: true
}, {
  value: 'wants',
  label: 'Желает'
}, {
  value: 'toKnow',
  label: 'Знать'
}, {
  value: 'where',
  label: 'Где'
}, {
  value: 'is',
  label: 'Сидит'
}, {
  value: 'thePheasant',
  label: 'Фазан'
}, {
  value: 'fullText',
  label: 'Каждый охотник желает знать, где сидит фазан'
}];
var defaultProps = {
  id: 'example-select',
  multiselect: false,
  placeholder: 'Выберите пример',
  helperText: 'Заполните пример',
  status: '',
  disabled: false,
  listHeight: ''
};
var func = function func() {};
var itemsEmpty = [];
var Default = exports.Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(_.Select, _extends({
    value: "test",
    items: items,
    onChange: func,
    onFocus: func,
    onBlur: func
  }, defaultProps)), _Select || (_Select = /*#__PURE__*/_react["default"].createElement(_.Select, {
    id: "example-empty-select",
    value: null,
    items: itemsEmpty,
    placeholder: "\u041F\u0443\u0441\u0442\u043E\u0439 \u0441\u043F\u0438\u0441\u043E\u043A"
  })), _Select2 || (_Select2 = /*#__PURE__*/_react["default"].createElement(_.Select, {
    id: "example-disabled-select",
    value: null,
    items: items,
    placeholder: "\u0417\u0430\u0434\u0438\u0437\u0435\u0439\u0431\u043B\u0435\u043D\u043D\u044B\u0439 \u0441\u043F\u0438\u0441\u043E\u043A",
    disabled: true
  })));
};
