// @salutejs/plasma-tokens/es/brands/sberprime/index.js
// Generated by robots, do not change this manually!
/** Цвет фона паранжи светлый */
export var overlaySoft = 'var(--overlay-soft, rgba(8, 9, 10, 0.56))';
/** Цвет фона паранжи темный */
export var overlayHard = 'var(--overlay-hard, rgba(8, 9, 10, 0.96))';
/** Основной цвет текста */
export var textPrimaryHover = 'var(--text-primary-hover, #FFFFFF93)';
/** Основной цвет текста */
export var textPrimaryActive = 'var(--text-primary-active, #FFFFFFC4)';
/** Основной цвет текста */
export var textPrimary = 'var(--text-primary, rgba(255, 255, 255, 0.96))';
/** Вторичный цвет текста */
export var textSecondaryHover = 'var(--text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста */
export var textSecondaryActive = 'var(--text-secondary-active, #FFFFFFAB)';
/** Вторичный цвет текста */
export var textSecondary = 'var(--text-secondary, rgba(255, 255, 255, 0.56))';
/** Третичный цвет текста */
export var textTertiaryHover = 'var(--text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста */
export var textTertiaryActive = 'var(--text-tertiary-active, #FFFFFF56)';
/** Третичный цвет текста */
export var textTertiary = 'var(--text-tertiary, rgba(255, 255, 255, 0.28))';
/** Сплошной наборный текст */
export var textParagraphHover = 'var(--text-paragraph-hover, #FFFFFF7A)';
/** Сплошной наборный текст */
export var textParagraphActive = 'var(--text-paragraph-active, #FFFFFFA3)';
/** Сплошной наборный текст */
export var textParagraph = 'var(--text-paragraph, rgba(255, 255, 255, 0.8))';
/** Акцентный цвет */
export var textAccentHover = 'var(--text-accent-hover, #1CE91CFF)';
/** Акцентный цвет */
export var textAccentActive = 'var(--text-accent-active, #12B512FF)';
/** Акцентный цвет */
export var textAccent = 'var(--text-accent, #15D015)';
/** Акцентный цвет с градиентом */
export var textAccentGradientHover = 'var(--text-accent-gradient-hover, #CCCCCCFF)';
/** Акцентный цвет с градиентом */
export var textAccentGradientActive = 'var(--text-accent-gradient-active, #E6E6E6FF)';
/** Акцентный цвет с градиентом */
export var textAccentGradient = 'var(--text-accent-gradient, radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015)';
/** Цвет успеха */
export var textPositiveHover = 'var(--text-positive-hover, #1EB83AFF)';
/** Цвет успеха */
export var textPositiveActive = 'var(--text-positive-active, #15842AFF)';
/** Цвет успеха */
export var textPositive = 'var(--text-positive, #1A9E32)';
/** Цвет предупреждения */
export var textWarningHover = 'var(--text-warning-hover, #FA6F1EFF)';
/** Цвет предупреждения */
export var textWarningActive = 'var(--text-warning-active, #D75204FF)';
/** Цвет предупреждения */
export var textWarning = 'var(--text-warning, #F55D05)';
/** Цвет ошибки */
export var textNegativeHover = 'var(--text-negative-hover, #FF475AFF)';
/** Цвет ошибки */
export var textNegativeActive = 'var(--text-negative-active, #FF0A23FF)';
/** Цвет ошибки */
export var textNegative = 'var(--text-negative, #FF293E)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimaryHover = 'var(--on-dark-text-primary-hover, #FFFFFF93)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimaryActive = 'var(--on-dark-text-primary-active, #FFFFFFC4)';
/** Основной цвет текста на темном фоне */
export var onDarkTextPrimary = 'var(--on-dark-text-primary, rgba(255, 255, 255, 0.96))';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondaryHover = 'var(--on-dark-text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondaryActive = 'var(--on-dark-text-secondary-active, #FFFFFFAB)';
/** Вторичный цвет текста на темном фоне */
export var onDarkTextSecondary = 'var(--on-dark-text-secondary, rgba(255, 255, 255, 0.56))';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiaryHover = 'var(--on-dark-text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiaryActive = 'var(--on-dark-text-tertiary-active, #FFFFFF56)';
/** Третичный цвет текста на темном фоне */
export var onDarkTextTertiary = 'var(--on-dark-text-tertiary, rgba(255, 255, 255, 0.28))';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraphHover = 'var(--on-dark-text-paragraph-hover, #FFFFFF7A)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraphActive = 'var(--on-dark-text-paragraph-active, #FFFFFFA3)';
/** Сплошной наборный текст на темном фоне */
export var onDarkTextParagraph = 'var(--on-dark-text-paragraph, rgba(255, 255, 255, 0.8))';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccentHover = 'var(--on-dark-text-accent-hover, #1CE91CFF)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccentActive = 'var(--on-dark-text-accent-active, #12B512FF)';
/** Акцентный цвет на темном фоне */
export var onDarkTextAccent = 'var(--on-dark-text-accent, #15D015)';
/** Акцентный цвет с градиентом на темном фоне */
export var onDarkTextAccentGradientHover = 'var(--on-dark-text-accent-gradient-hover, #CCCCCCFF)';
/** Акцентный цвет с градиентом на темном фоне */
export var onDarkTextAccentGradientActive = 'var(--on-dark-text-accent-gradient-active, #E6E6E6FF)';
/** Акцентный цвет с градиентом на темном фоне */
export var onDarkTextAccentGradient = 'var(--on-dark-text-accent-gradient, radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015)';
/** Цвет успеха на темном фоне */
export var onDarkTextPositiveHover = 'var(--on-dark-text-positive-hover, #1EB83AFF)';
/** Цвет успеха на темном фоне */
export var onDarkTextPositiveActive = 'var(--on-dark-text-positive-active, #15842AFF)';
/** Цвет успеха на темном фоне */
export var onDarkTextPositive = 'var(--on-dark-text-positive, #1A9E32)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarningHover = 'var(--on-dark-text-warning-hover, #FA6F1EFF)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarningActive = 'var(--on-dark-text-warning-active, #D75204FF)';
/** Цвет предупреждения на темном фоне */
export var onDarkTextWarning = 'var(--on-dark-text-warning, #F55D05)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegativeHover = 'var(--on-dark-text-negative-hover, #FF475AFF)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegativeActive = 'var(--on-dark-text-negative-active, #FF0A23FF)';
/** Цвет ошибки на темном фоне */
export var onDarkTextNegative = 'var(--on-dark-text-negative, #FF293E)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimaryHover = 'var(--on-light-text-primary-hover, #23272A93)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimaryActive = 'var(--on-light-text-primary-active, #23272AC4)';
/** Основной цвет текста на светлом фоне */
export var onLightTextPrimary = 'var(--on-light-text-primary, rgba(36, 40, 43, 0.96))';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondaryHover = 'var(--on-light-text-secondary-hover, #23272AFF)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondaryActive = 'var(--on-light-text-secondary-active, #23272AAB)';
/** Вторичный цвет текста на светлом фоне */
export var onLightTextSecondary = 'var(--on-light-text-secondary, rgba(36, 40, 43, 0.56))';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiaryHover = 'var(--on-light-text-tertiary-hover, #23272AFF)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiaryActive = 'var(--on-light-text-tertiary-active, #23272A56)';
/** Третичный цвет текста на светлом фоне */
export var onLightTextTertiary = 'var(--on-light-text-tertiary, rgba(36, 40, 43, 0.28))';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraphHover = 'var(--on-light-text-paragraph-hover, #23272A7A)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraphActive = 'var(--on-light-text-paragraph-active, #23272AA3)';
/** Сплошной наборный текст на светлом фоне */
export var onLightTextParagraph = 'var(--on-light-text-paragraph, rgba(36, 40, 43, 0.8))';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccentHover = 'var(--on-light-text-accent-hover, #18CD18FF)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccentActive = 'var(--on-light-text-accent-active, #129712FF)';
/** Акцентный цвет на светлом фоне */
export var onLightTextAccent = 'var(--on-light-text-accent, #15B015)';
/** Акцентный цвет с градиентом на светлом фоне */
export var onLightTextAccentGradientHover = 'var(--on-light-text-accent-gradient-hover, #CCCCCCFF)';
/** Акцентный цвет с градиентом на светлом фоне */
export var onLightTextAccentGradientActive = 'var(--on-light-text-accent-gradient-active, #E6E6E6FF)';
/** Акцентный цвет с градиентом на светлом фоне */
export var onLightTextAccentGradient = 'var(--on-light-text-accent-gradient, radial-gradient(50% 200% at 0% 100%, rgba(250, 237, 5, 0.6) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(33, 202, 154, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(12, 156, 12, 0) 100%), #0C9C0C)';
/** Цвет успеха на светлом фоне */
export var onLightTextPositiveHover = 'var(--on-light-text-positive-hover, #1EB83AFF)';
/** Цвет успеха на светлом фоне */
export var onLightTextPositiveActive = 'var(--on-light-text-positive-active, #15842AFF)';
/** Цвет успеха на светлом фоне */
export var onLightTextPositive = 'var(--on-light-text-positive, #1A9E32)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarningHover = 'var(--on-light-text-warning-hover, #FA6F1EFF)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarningActive = 'var(--on-light-text-warning-active, #D75204FF)';
/** Цвет предупреждения на светлом фоне */
export var onLightTextWarning = 'var(--on-light-text-warning, #F55D05)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegativeHover = 'var(--on-light-text-negative-hover, #FF475AFF)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegativeActive = 'var(--on-light-text-negative-active, #FF0A23FF)';
/** Цвет ошибки на светлом фоне */
export var onLightTextNegative = 'var(--on-light-text-negative, #FF293E)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimaryHover = 'var(--surface-solid-primary-hover, #191D1FFF)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimaryActive = 'var(--surface-solid-primary-active, #101214FF)';
/** Основной непрозрачный фон поверхности/контрола */
export var surfaceSolidPrimary = 'var(--surface-solid-primary, #15181A)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondaryHover = 'var(--surface-solid-secondary-hover, #2C3135FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondaryActive = 'var(--surface-solid-secondary-active, #1E2224FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
export var surfaceSolidSecondary = 'var(--surface-solid-secondary, #24282B)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiaryHover = 'var(--surface-solid-tertiary-hover, #3A4145FF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiaryActive = 'var(--surface-solid-tertiary-active, #2C3135FF)';
/** Третичный непрозрачный фон поверхности/контрола */
export var surfaceSolidTertiary = 'var(--surface-solid-tertiary, #31373B)';
/** Основной фон для карточек */
export var surfaceSolidCardHover = 'var(--surface-solid-card-hover, #191D1FFF)';
/** Основной фон для карточек */
export var surfaceSolidCardActive = 'var(--surface-solid-card-active, #101214FF)';
/** Основной фон для карточек */
export var surfaceSolidCard = 'var(--surface-solid-card, #15181A)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefaultHover = 'var(--surface-solid-default-hover, #F9FAFBFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefaultActive = 'var(--surface-solid-default-active, #ECF0F3FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
export var surfaceSolidDefault = 'var(--surface-solid-default, #F2F5F7)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimaryHover = 'var(--surface-transparent-primary-hover, #F2F5F71F)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimaryActive = 'var(--surface-transparent-primary-active, #F2F5F70A)';
/** Основной прозрачный фон поверхности/контрола */
export var surfaceTransparentPrimary = 'var(--surface-transparent-primary, rgba(242, 245, 247, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondaryHover = 'var(--surface-transparent-secondary-hover, #F2F5F73D)';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondaryActive = 'var(--surface-transparent-secondary-active, #F2F5F70F)';
/** Вторичный прозрачный фон поверхности/контрола */
export var surfaceTransparentSecondary = 'var(--surface-transparent-secondary, rgba(242, 245, 247, 0.12))';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiaryHover = 'var(--surface-transparent-tertiary-hover, #F2F5F752)';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiaryActive = 'var(--surface-transparent-tertiary-active, #F2F5F724)';
/** Третичный прозрачный фон поверхности/контрола */
export var surfaceTransparentTertiary = 'var(--surface-transparent-tertiary, rgba(242, 245, 247, 0.2))';
/** Прозрачный фон для карточек */
export var surfaceTransparentCardHover = 'var(--surface-transparent-card-hover, #F2F5F71F)';
/** Прозрачный фон для карточек */
export var surfaceTransparentCardActive = 'var(--surface-transparent-card-active, #F2F5F70A)';
/** Прозрачный фон для карточек */
export var surfaceTransparentCard = 'var(--surface-transparent-card, rgba(242, 245, 247, 0.06))';
/** Фон поверхности/контрола без заливки */
export var surfaceClearHover = 'var(--surface-clear-hover, #00000000)';
/** Фон поверхности/контрола без заливки */
export var surfaceClearActive = 'var(--surface-clear-active, #00000000)';
/** Фон поверхности/контрола без заливки */
export var surfaceClear = 'var(--surface-clear, rgba(0, 0, 0, 0))';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentHover = 'var(--surface-accent-hover, #17C417FF)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccentActive = 'var(--surface-accent-active, #14A914FF)';
/** Акцентный фон поверхности/контрола */
export var surfaceAccent = 'var(--surface-accent, #15B015)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradientHover = 'var(--surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradientActive = 'var(--surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
export var surfaceAccentGradient = 'var(--surface-accent-gradient, radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositiveHover = 'var(--surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositiveActive = 'var(--surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех */
export var surfacePositive = 'var(--surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarningHover = 'var(--surface-warning-hover, #FA6814FF)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarningActive = 'var(--surface-warning-active, #EB5905FF)';
/** Цвет фона поверхности/контрола предупреждение */
export var surfaceWarning = 'var(--surface-warning, #F55D05)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegativeHover = 'var(--surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegativeActive = 'var(--surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка */
export var surfaceNegative = 'var(--surface-negative, #FF293E)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimaryHover = 'var(--on-dark-surface-solid-primary-hover, #191D1FFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimaryActive = 'var(--on-dark-surface-solid-primary-active, #101214FF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #15181A)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondaryHover = 'var(--on-dark-surface-solid-secondary-hover, #2C3135FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondaryActive = 'var(--on-dark-surface-solid-secondary-active, #1E2224FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #24282B)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiaryHover = 'var(--on-dark-surface-solid-tertiary-hover, #3A4145FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiaryActive = 'var(--on-dark-surface-solid-tertiary-active, #2C3135FF)';
/** Третичный непрозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceSolidTertiary = 'var(--on-dark-surface-solid-tertiary, #31373B)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCardHover = 'var(--on-dark-surface-solid-card-hover, #191D1FFF)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCardActive = 'var(--on-dark-surface-solid-card-active, #101214FF)';
/** Основной фон для карточек на темном фоне */
export var onDarkSurfaceSolidCard = 'var(--on-dark-surface-solid-card, #15181A)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefaultHover = 'var(--on-dark-surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefaultActive = 'var(--on-dark-surface-solid-default-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
export var onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #FFFFFF)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimaryHover = 'var(--on-dark-surface-transparent-primary-hover, #FFFFFF1F)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimaryActive = 'var(--on-dark-surface-transparent-primary-active, #FFFFFF0A)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, rgba(255, 255, 255, 0.06))';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondaryHover = 'var(--on-dark-surface-transparent-secondary-hover, #FFFFFF3D)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondaryActive = 'var(--on-dark-surface-transparent-secondary-active, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, rgba(255, 255, 255, 0.12))';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiaryHover = 'var(--on-dark-surface-transparent-tertiary-hover, #FFFFFF52)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiaryActive = 'var(--on-dark-surface-transparent-tertiary-active, #FFFFFF24)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, rgba(255, 255, 255, 0.2))';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCardHover = 'var(--on-dark-surface-transparent-card-hover, #FFFFFF1F)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCardActive = 'var(--on-dark-surface-transparent-card-active, #FFFFFF0A)';
/** Прозрачный фон для карточек на темном фоне */
export var onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, rgba(255, 255, 255, 0.06))';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentHover = 'var(--on-dark-surface-accent-hover, #17C417FF)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccentActive = 'var(--on-dark-surface-accent-active, #14A914FF)';
/** Акцентный фон поверхности/контрола на темном фоне */
export var onDarkSurfaceAccent = 'var(--on-dark-surface-accent, #15B015)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradientHover = 'var(--on-dark-surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradientActive = 'var(--on-dark-surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на темном фоне */
export var onDarkSurfaceAccentGradient = 'var(--on-dark-surface-accent-gradient, radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015)';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositiveHover = 'var(--on-dark-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositiveActive = 'var(--on-dark-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех на темном фоне */
export var onDarkSurfacePositive = 'var(--on-dark-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceWarningHover = 'var(--on-dark-surface-warning-hover, #FA6814FF)';
/** Цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceWarningActive = 'var(--on-dark-surface-warning-active, #EB5905FF)';
/** Цвет фона поверхности/контрола предупреждение на темном фоне */
export var onDarkSurfaceWarning = 'var(--on-dark-surface-warning, #F55D05)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegativeHover = 'var(--on-dark-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegativeActive = 'var(--on-dark-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка на темном фоне */
export var onDarkSurfaceNegative = 'var(--on-dark-surface-negative, #FF293E)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidPrimaryHover = 'var(--on-light-surface-solid-primary-hover, #F9FAFBFF)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidPrimaryActive = 'var(--on-light-surface-solid-primary-active, #ECF0F3FF)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #F2F5F7)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondaryHover = 'var(--on-light-surface-solid-secondary-hover, #F2F5F8FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondaryActive = 'var(--on-light-surface-solid-secondary-active, #DEE7EDFF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #E4EBF0)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidTertiaryHover = 'var(--on-light-surface-solid-tertiary-hover, #DEE7EDFF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidTertiaryActive = 'var(--on-light-surface-solid-tertiary-active, #CBD8E2FF)';
/** Третичный непрозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceSolidTertiary = 'var(--on-light-surface-solid-tertiary, #D1DDE5)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCardHover = 'var(--on-light-surface-solid-card-hover, #F9FAFBFF)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCardActive = 'var(--on-light-surface-solid-card-active, #ECF0F3FF)';
/** Основной фон для карточек на светлом фоне */
export var onLightSurfaceSolidCard = 'var(--on-light-surface-solid-card, #F2F5F7)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefaultHover = 'var(--on-light-surface-solid-default-hover, #0E0F11FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefaultActive = 'var(--on-light-surface-solid-default-active, #050506FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
export var onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #08090A)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentPrimaryHover = 'var(--on-light-surface-transparent-primary-hover, #090A0B12)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentPrimaryActive = 'var(--on-light-surface-transparent-primary-active, #090A0B03)';
/** Основной прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentPrimary = 'var(--on-light-surface-transparent-primary, rgba(8, 9, 10, 0.03))';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondaryHover = 'var(--on-light-surface-transparent-secondary-hover, #090A0B1F)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondaryActive = 'var(--on-light-surface-transparent-secondary-active, #090A0B0A)';
/** Вторичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentSecondary = 'var(--on-light-surface-transparent-secondary, rgba(8, 9, 10, 0.06))';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiaryHover = 'var(--on-light-surface-transparent-tertiary-hover, #090A0B3D)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiaryActive = 'var(--on-light-surface-transparent-tertiary-active, #090A0B0F)';
/** Третичный прозрачный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceTransparentTertiary = 'var(--on-light-surface-transparent-tertiary, rgba(8, 9, 10, 0.12))';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCardHover = 'var(--on-light-surface-transparent-card-hover, #090A0B12)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCardActive = 'var(--on-light-surface-transparent-card-active, #090A0B03)';
/** Прозрачный фон для карточек на светлом фоне */
export var onLightSurfaceTransparentCard = 'var(--on-light-surface-transparent-card, rgba(8, 9, 10, 0.03))';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentHover = 'var(--on-light-surface-accent-hover, #17C417FF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccentActive = 'var(--on-light-surface-accent-active, #14A914FF)';
/** Акцентный фон поверхности/контрола на светлом фоне */
export var onLightSurfaceAccent = 'var(--on-light-surface-accent, #15B015)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradientHover = 'var(--on-light-surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradientActive = 'var(--on-light-surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом на светлом фоне */
export var onLightSurfaceAccentGradient = 'var(--on-light-surface-accent-gradient, radial-gradient(50% 150% at 0% 100%, rgba(250, 237, 5, 0.9) 0%, rgba(21, 208, 21, 0.12) 100%), radial-gradient(100% 200% at 0% 50%, #15D015 0%, rgba(21, 208, 21, 0) 100%), radial-gradient(100% 320% at 100% 0%, #2A72F8 0%, rgba(21, 208, 21, 0) 100%), #15D015)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositiveHover = 'var(--on-light-surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositiveActive = 'var(--on-light-surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех на светлом фоне */
export var onLightSurfacePositive = 'var(--on-light-surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceWarningHover = 'var(--on-light-surface-warning-hover, #FA6814FF)';
/** Цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceWarningActive = 'var(--on-light-surface-warning-active, #EB5905FF)';
/** Цвет фона поверхности/контрола предупреждение на светлом фоне */
export var onLightSurfaceWarning = 'var(--on-light-surface-warning, #F55D05)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeHover = 'var(--on-light-surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegativeActive = 'var(--on-light-surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка на светлом фоне */
export var onLightSurfaceNegative = 'var(--on-light-surface-negative, #FF293E)';
/** Основной фон */
export var backgroundPrimary = 'var(--background-primary, #08090A)';
/** Основной фон на темном фоне */
export var darkBackgroundPrimary = 'var(--dark-background-primary, #08090A)';
/** Основной фон на светлом фоне */
export var lightBackgroundPrimary = 'var(--light-background-primary, #FFFFFF)';
export var skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
export var skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
export var whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
export var whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
export var whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
export var blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
export var blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
export var blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use onLightSurfaceSolidDefault */
export var buttonBlack = 'var(--plasma-colors-button-black, var(--on-light-surface-solid-default))';
/** @deprecated instead use onLightSurfaceTransparentSecondary */
export var buttonBlackSecondary = 'var(--plasma-colors-button-black-secondary, var(--on-light-surface-transparent-secondary))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
export var buttonWhite = 'var(--plasma-colors-button-white, var(--on-dark-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
export var buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, var(--on-dark-surface-transparent-secondary))';
/** @deprecated instead use textPrimary */
export var text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
export var primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
export var secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
export var tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use textParagraph */
export var paragraph = 'var(--plasma-colors-paragraph, var(--text-paragraph))';
/** @deprecated instead use backgroundPrimary */
export var background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textAccent */
export var accent = 'var(--plasma-colors-accent, var(--text-accent))';
/** @deprecated instead use textPositive */
export var success = 'var(--plasma-colors-success, var(--text-positive))';
/** @deprecated instead use textWarning */
export var warning = 'var(--plasma-colors-warning, var(--text-warning))';
/** @deprecated instead use textNegative */
export var critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use overlaySoft */
export var overlay = 'var(--plasma-colors-overlay, var(--overlay-soft))';
/** @deprecated instead use surfaceTransparentPrimary */
export var surfaceLiquid01 = 'var(--plasma-colors-surface-liquid01, var(--surface-transparent-primary))';
/** @deprecated instead use surfaceTransparentSecondary */
export var surfaceLiquid02 = 'var(--plasma-colors-surface-liquid02, var(--surface-transparent-secondary))';
/** @deprecated instead use surfaceTransparentTertiary */
export var surfaceLiquid03 = 'var(--plasma-colors-surface-liquid03, var(--surface-transparent-tertiary))';
/** @deprecated instead use surfaceSolidPrimary */
export var surfaceSolid01 = 'var(--plasma-colors-surface-solid01, var(--surface-solid-primary))';
/** @deprecated instead use surfaceSolidSecondary */
export var surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
export var surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use surfaceTransparentCard */
export var surfaceCard = 'var(--plasma-colors-surface-card, var(--surface-transparent-card))';
/** @deprecated instead use surfaceTransparentSecondary */
export var buttonSecondary = 'var(--plasma-colors-button-secondary, var(--surface-transparent-secondary))';
/** @deprecated instead use textAccent */
export var buttonAccent = 'var(--plasma-colors-button-accent, var(--text-accent))';
/** @deprecated instead use surfacePositive */
export var buttonSuccess = 'var(--plasma-colors-button-success, var(--surface-positive))';
/** @deprecated instead use surfaceWarning */
export var buttonWarning = 'var(--plasma-colors-button-warning, var(--surface-warning))';
/** @deprecated instead use surfaceNegative */
export var buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
export var borderRadiusXxs = 'var(--border-radius-xxs, 4px)';
export var borderRadiusXs = 'var(--border-radius-xs, 6px)';
export var borderRadiusS = 'var(--border-radius-s, 8px)';
export var borderRadiusM = 'var(--border-radius-m, 12px)';
export var borderRadiusL = 'var(--border-radius-l, 16px)';
export var borderRadiusXl = 'var(--border-radius-xl, 20px)';
export var borderRadiusXxl = 'var(--border-radius-xxl, 32px)';
