// @salutejs/plasma-core/components/Carousel/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Carousel", {
  enumerable: true,
  get: function get() {
    return _Carousel.Carousel;
  }
});
Object.defineProperty(exports, "CarouselGridWrapper", {
  enumerable: true,
  get: function get() {
    return _Carousel.CarouselGridWrapper;
  }
});
Object.defineProperty(exports, "CarouselItem", {
  enumerable: true,
  get: function get() {
    return _CarouselItem.CarouselItem;
  }
});
Object.defineProperty(exports, "CarouselItemVirtual", {
  enumerable: true,
  get: function get() {
    return _CarouselItem.CarouselItemVirtual;
  }
});
Object.defineProperty(exports, "CarouselTrack", {
  enumerable: true,
  get: function get() {
    return _Carousel.CarouselTrack;
  }
});
Object.defineProperty(exports, "useCarousel", {
  enumerable: true,
  get: function get() {
    return _hooks.useCarousel;
  }
});
Object.defineProperty(exports, "useCarouselLite", {
  enumerable: true,
  get: function get() {
    return _hooks.useCarouselLite;
  }
});
var _Carousel = /*#__PURE__*/require("./Carousel");
var _CarouselItem = /*#__PURE__*/require("./CarouselItem");
var _hooks = /*#__PURE__*/require("./hooks");
