// @salutejs/plasma-tokens/brands/flamingo/index.js
"use strict";
// Generated by robots, do not change this manually!
Object.defineProperty(exports, "__esModule", { value: true });
exports.inverseTextSecondaryActive = exports.inverseTextSecondaryHover = exports.inverseTextPrimary = exports.inverseTextPrimaryActive = exports.inverseTextPrimaryHover = exports.onLightTextTertiary = exports.onLightTextTertiaryActive = exports.onLightTextTertiaryHover = exports.onLightTextSecondary = exports.onLightTextSecondaryActive = exports.onLightTextSecondaryHover = exports.onLightTextPrimary = exports.onLightTextPrimaryActive = exports.onLightTextPrimaryHover = exports.onDarkTextTertiary = exports.onDarkTextTertiaryActive = exports.onDarkTextTertiaryHover = exports.onDarkTextSecondary = exports.onDarkTextSecondaryActive = exports.onDarkTextSecondaryHover = exports.onDarkTextPrimary = exports.onDarkTextPrimaryActive = exports.onDarkTextPrimaryHover = exports.textInfo = exports.textInfoActive = exports.textInfoHover = exports.textNegative = exports.textNegativeActive = exports.textNegativeHover = exports.textWarning = exports.textWarningActive = exports.textWarningHover = exports.textPositive = exports.textPositiveActive = exports.textPositiveHover = exports.textAccentGradient = exports.textAccentGradientActive = exports.textAccentGradientHover = exports.textAccent = exports.textAccentActive = exports.textAccentHover = exports.textTertiary = exports.textTertiaryActive = exports.textTertiaryHover = exports.textSecondary = exports.textSecondaryActive = exports.textSecondaryHover = exports.textPrimary = exports.textPrimaryActive = exports.textPrimaryHover = void 0;
exports.surfaceWarningHover = exports.surfacePositive = exports.surfacePositiveActive = exports.surfacePositiveHover = exports.surfaceAccentGradient = exports.surfaceAccentGradientActive = exports.surfaceAccentGradientHover = exports.surfaceAccent = exports.surfaceAccentActive = exports.surfaceAccentHover = exports.surfaceClear = exports.surfaceClearActive = exports.surfaceClearHover = exports.surfaceSolidDefault = exports.surfaceSolidDefaultActive = exports.surfaceSolidDefaultHover = exports.surfaceSolidCard = exports.surfaceSolidCardActive = exports.surfaceSolidCardHover = exports.surfaceSolidTertiary = exports.surfaceSolidTertiaryActive = exports.surfaceSolidTertiaryHover = exports.surfaceSolidSecondary = exports.surfaceSolidSecondaryActive = exports.surfaceSolidSecondaryHover = exports.surfaceSolidPrimary = exports.surfaceSolidPrimaryActive = exports.surfaceSolidPrimaryHover = exports.inverseTextInfo = exports.inverseTextInfoActive = exports.inverseTextInfoHover = exports.inverseTextNegative = exports.inverseTextNegativeActive = exports.inverseTextNegativeHover = exports.inverseTextWarning = exports.inverseTextWarningActive = exports.inverseTextWarningHover = exports.inverseTextPositive = exports.inverseTextPositiveActive = exports.inverseTextPositiveHover = exports.inverseTextAccentGradient = exports.inverseTextAccentGradientActive = exports.inverseTextAccentGradientHover = exports.inverseTextAccent = exports.inverseTextAccentActive = exports.inverseTextAccentHover = exports.inverseTextTertiary = exports.inverseTextTertiaryActive = exports.inverseTextTertiaryHover = exports.inverseTextSecondary = void 0;
exports.onDarkSurfaceSolidDefault = exports.onDarkSurfaceSolidDefaultActive = exports.onDarkSurfaceSolidDefaultHover = exports.onDarkSurfaceSolidSecondary = exports.onDarkSurfaceSolidSecondaryActive = exports.onDarkSurfaceSolidSecondaryHover = exports.onDarkSurfaceSolidPrimary = exports.onDarkSurfaceSolidPrimaryActive = exports.onDarkSurfaceSolidPrimaryHover = exports.surfaceTransparentRating = exports.surfaceTransparentRatingActive = exports.surfaceTransparentRatingHover = exports.surfaceRatingMinor = exports.surfaceRatingMinorActive = exports.surfaceRatingMinorHover = exports.surfaceRating = exports.surfaceRatingActive = exports.surfaceRatingHover = exports.surfaceTransparentInfo = exports.surfaceTransparentInfoActive = exports.surfaceTransparentInfoHover = exports.surfaceTransparentNegative = exports.surfaceTransparentNegativeActive = exports.surfaceTransparentNegativeHover = exports.surfaceTransparentWarning = exports.surfaceTransparentWarningActive = exports.surfaceTransparentWarningHover = exports.surfaceTransparentPositive = exports.surfaceTransparentPositiveActive = exports.surfaceTransparentPositiveHover = exports.surfaceInfoMinor = exports.surfaceInfoMinorActive = exports.surfaceInfoMinorHover = exports.surfaceNegativeMinor = exports.surfaceNegativeMinorActive = exports.surfaceNegativeMinorHover = exports.surfaceWarningMinor = exports.surfaceWarningMinorActive = exports.surfaceWarningMinorHover = exports.surfacePositiveMinor = exports.surfacePositiveMinorActive = exports.surfacePositiveMinorHover = exports.surfaceInfo = exports.surfaceInfoActive = exports.surfaceInfoHover = exports.surfaceNegative = exports.surfaceNegativeActive = exports.surfaceNegativeHover = exports.surfaceWarning = exports.surfaceWarningActive = void 0;
exports.outlineDefaultOutlineWarning = exports.outlineDefaultOutlineWarningActive = exports.outlineDefaultOutlineWarningHover = exports.outlineDefaultOutlinePositive = exports.outlineDefaultOutlinePositiveActive = exports.outlineDefaultOutlinePositiveHover = exports.overlayBlur = exports.overlayHard = exports.overlaySoft = exports.backgroundGradientPurple = exports.backgroundGradientPink = exports.backgroundGradientRed = exports.backgroundGradientBlue = exports.backgroundGradientYellow = exports.backgroundGradientOrange = exports.backgroundGradientGreen = exports.backgroundPrimary = exports.inverseSurfaceSolidDefault = exports.inverseSurfaceSolidDefaultActive = exports.inverseSurfaceSolidDefaultHover = exports.inverseSurfaceSolidSecondary = exports.inverseSurfaceSolidSecondaryActive = exports.inverseSurfaceSolidSecondaryHover = exports.inverseSurfaceSolidPrimary = exports.inverseSurfaceSolidPrimaryActive = exports.inverseSurfaceSolidPrimaryHover = exports.onLightSurfaceSolidDefault = exports.onLightSurfaceSolidDefaultActive = exports.onLightSurfaceSolidDefaultHover = exports.onLightSurfaceSolidSecondary = exports.onLightSurfaceSolidSecondaryActive = exports.onLightSurfaceSolidSecondaryHover = exports.onLightSurfaceSolidPrimary = exports.onLightSurfaceSolidPrimaryActive = exports.onLightSurfaceSolidPrimaryHover = exports.onDarkSurfaceTransparentCard = exports.onDarkSurfaceTransparentCardActive = exports.onDarkSurfaceTransparentCardHover = exports.onDarkSurfaceTransparentDeep = exports.onDarkSurfaceTransparentDeepActive = exports.onDarkSurfaceTransparentDeepHover = exports.onDarkSurfaceTransparentTertiary = exports.onDarkSurfaceTransparentTertiaryActive = exports.onDarkSurfaceTransparentTertiaryHover = exports.onDarkSurfaceTransparentSecondary = exports.onDarkSurfaceTransparentSecondaryActive = exports.onDarkSurfaceTransparentSecondaryHover = exports.onDarkSurfaceTransparentPrimary = exports.onDarkSurfaceTransparentPrimaryActive = exports.onDarkSurfaceTransparentPrimaryHover = void 0;
exports.skeletonGradientLighter = exports.skeletonGradient = exports.outlineDefaultOutlineRatingTransparent = exports.outlineDefaultOutlineRatingTransparentActive = exports.outlineDefaultOutlineRatingTransparentHover = exports.outlineDefaultOutlineInfoTransparent = exports.outlineDefaultOutlineInfoTransparentActive = exports.outlineDefaultOutlineInfoTransparentHover = exports.outlineDefaultOutlineNegativeTransparent = exports.outlineDefaultOutlineNegativeTransparentActive = exports.outlineDefaultOutlineNegativeTransparentHover = exports.outlineDefaultOutlineWarningTransparent = exports.outlineDefaultOutlineWarningTransparentActive = exports.outlineDefaultOutlineWarningTransparentHover = exports.outlineDefaultOutlinePositiveTransparent = exports.outlineDefaultOutlinePositiveTransparentActive = exports.outlineDefaultOutlinePositiveTransparentHover = exports.outlineDefaultOutlineRatingMinor = exports.outlineDefaultOutlineRatingMinorActive = exports.outlineDefaultOutlineRatingMinorHover = exports.outlineDefaultOutlineRating = exports.outlineDefaultOutlineRatingActive = exports.outlineDefaultOutlineRatingHover = exports.outlineDefaultOutlineTertiary = exports.outlineDefaultOutlineTertiaryActive = exports.outlineDefaultOutlineTertiaryHover = exports.outlineDefaultOutlineSecondary = exports.outlineDefaultOutlineSecondaryActive = exports.outlineDefaultOutlineSecondaryHover = exports.outlineDefaultOutlinePrimary = exports.outlineDefaultOutlinePrimaryActive = exports.outlineDefaultOutlinePrimaryHover = exports.outlineDefaultOutlineInfoMinor = exports.outlineDefaultOutlineInfoMinorActive = exports.outlineDefaultOutlineInfoMinorHover = exports.outlineDefaultOutlineNegativeMinor = exports.outlineDefaultOutlineNegativeMinorActive = exports.outlineDefaultOutlineNegativeMinorHover = exports.outlineDefaultOutlineWarningMinor = exports.outlineDefaultOutlineWarningMinorActive = exports.outlineDefaultOutlineWarningMinorHover = exports.outlineDefaultOutlinePositiveMinor = exports.outlineDefaultOutlinePositiveMinorActive = exports.outlineDefaultOutlinePositiveMinorHover = exports.outlineDefaultOutlineInfo = exports.outlineDefaultOutlineInfoActive = exports.outlineDefaultOutlineInfoHover = exports.outlineDefaultOutlineNegative = exports.outlineDefaultOutlineNegativeActive = exports.outlineDefaultOutlineNegativeHover = void 0;
exports.buttonCritical = exports.buttonWarning = exports.buttonSuccess = exports.buttonAccent = exports.surfaceSolid03 = exports.surfaceSolid02 = exports.surfaceSolid01 = exports.overlay = exports.critical = exports.warning = exports.success = exports.accent = exports.background = exports.tertiary = exports.secondary = exports.primary = exports.text = exports.buttonWhiteSecondary = exports.buttonWhite = exports.buttonBlack = exports.blackTertiary = exports.blackSecondary = exports.blackPrimary = exports.whiteTertiary = exports.whiteSecondary = exports.whitePrimary = void 0;
/** Основной цвет текста */
exports.textPrimaryHover = 'var(--text-primary-hover, #CCCCCCFF)';
/** Основной цвет текста */
exports.textPrimaryActive = 'var(--text-primary-active, #E6E6E6FF)';
/** Основной цвет текста */
exports.textPrimary = 'var(--text-primary, #FFFFFF)';
/** Вторичный цвет текста */
exports.textSecondaryHover = 'var(--text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста */
exports.textSecondaryActive = 'var(--text-secondary-active, #3D3D3DFF)';
/** Вторичный цвет текста */
exports.textSecondary = 'var(--text-secondary, #B3B3B3)';
/** Третичный цвет текста */
exports.textTertiaryHover = 'var(--text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста */
exports.textTertiaryActive = 'var(--text-tertiary-active, #737373FF)';
/** Третичный цвет текста */
exports.textTertiary = 'var(--text-tertiary, #707070)';
/** Акцентный цвет */
exports.textAccentHover = 'var(--text-accent-hover, #2ACB47FF)';
/** Акцентный цвет */
exports.textAccentActive = 'var(--text-accent-active, #1F9835FF)';
/** Акцентный цвет */
exports.textAccent = 'var(--text-accent, #24B23E)';
/** Акцентный цвет с градиентом */
exports.textAccentGradientHover = 'var(--text-accent-gradient-hover, #CCCCCCFF)';
/** Акцентный цвет с градиентом */
exports.textAccentGradientActive = 'var(--text-accent-gradient-active, #E6E6E6FF)';
/** Акцентный цвет с градиентом */
exports.textAccentGradient = 'var(--text-accent-gradient, linear-gradient(45deg, #0B8C0B 0%, #14CC98 100%))';
/** Цвет успеха */
exports.textPositiveHover = 'var(--text-positive-hover, #1EB83AFF)';
/** Цвет успеха */
exports.textPositiveActive = 'var(--text-positive-active, #15842AFF)';
/** Цвет успеха */
exports.textPositive = 'var(--text-positive, #1A9E32)';
/** Цвет предупреждения */
exports.textWarningHover = 'var(--text-warning-hover, #FB7223FF)';
/** Цвет предупреждения */
exports.textWarningActive = 'var(--text-warning-active, #DC5304FF)';
/** Цвет предупреждения */
exports.textWarning = 'var(--text-warning, #FA5F05)';
/** Цвет ошибки */
exports.textNegativeHover = 'var(--text-negative-hover, #FF475AFF)';
/** Цвет ошибки */
exports.textNegativeActive = 'var(--text-negative-active, #FF0A23FF)';
/** Цвет ошибки */
exports.textNegative = 'var(--text-negative, #FF293E)';
exports.textInfoHover = 'var(--text-info-hover, #50B1F2FF)';
exports.textInfoActive = 'var(--text-info-active, #0F7BC2FF)';
exports.textInfo = 'var(--text-info, #118CDF)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimaryHover = 'var(--on-dark-text-primary-hover, #CCCCCCFF)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimaryActive = 'var(--on-dark-text-primary-active, #E6E6E6FF)';
/** Основной цвет текста на темном фоне */
exports.onDarkTextPrimary = 'var(--on-dark-text-primary, #FFFFFF)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondaryHover = 'var(--on-dark-text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondaryActive = 'var(--on-dark-text-secondary-active, #3D3D3DFF)';
/** Вторичный цвет текста на темном фоне */
exports.onDarkTextSecondary = 'var(--on-dark-text-secondary, #B3B3B3)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiaryHover = 'var(--on-dark-text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiaryActive = 'var(--on-dark-text-tertiary-active, #737373FF)';
/** Третичный цвет текста на темном фоне */
exports.onDarkTextTertiary = 'var(--on-dark-text-tertiary, #707070)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimaryHover = 'var(--on-light-text-primary-hover, #FFFFFFFF)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimaryActive = 'var(--on-light-text-primary-active, #ADADADFF)';
/** Основной цвет текста на светлом фоне */
exports.onLightTextPrimary = 'var(--on-light-text-primary, #262626)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondaryHover = 'var(--on-light-text-secondary-hover, #FFFFFFFF)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondaryActive = 'var(--on-light-text-secondary-active, #8C8C8CFF)';
/** Вторичный цвет текста на светлом фоне */
exports.onLightTextSecondary = 'var(--on-light-text-secondary, #4E4E4E)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiaryHover = 'var(--on-light-text-tertiary-hover, #FFFFFFFF)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiaryActive = 'var(--on-light-text-tertiary-active, #575757FF)';
/** Третичный цвет текста на светлом фоне */
exports.onLightTextTertiary = 'var(--on-light-text-tertiary, #949494)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimaryHover = 'var(--inverse-text-primary-hover, #FFFFFFFF)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimaryActive = 'var(--inverse-text-primary-active, #ADADADFF)';
/** Инвертированный основной цвет текста */
exports.inverseTextPrimary = 'var(--inverse-text-primary, #262626)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondaryHover = 'var(--inverse-text-secondary-hover, #FFFFFFFF)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondaryActive = 'var(--inverse-text-secondary-active, #8C8C8CFF)';
/** Инвертированный вторичный цвет текста */
exports.inverseTextSecondary = 'var(--inverse-text-secondary, #4E4E4E)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiaryHover = 'var(--inverse-text-tertiary-hover, #FFFFFFFF)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiaryActive = 'var(--inverse-text-tertiary-active, #575757FF)';
/** Инвертированный третичный цвет текста */
exports.inverseTextTertiary = 'var(--inverse-text-tertiary, #949494)';
/** Инвертированный акцентный цвет */
exports.inverseTextAccentHover = 'var(--inverse-text-accent-hover, #13AA2CFF)';
/** Инвертированный акцентный цвет */
exports.inverseTextAccentActive = 'var(--inverse-text-accent-active, #0D731EFF)';
/** Инвертированный акцентный цвет */
exports.inverseTextAccent = 'var(--inverse-text-accent, #108E26)';
/** Инвертированный акцентный цвет с градиентом */
exports.inverseTextAccentGradientHover = 'var(--inverse-text-accent-gradient-hover, #CCCCCCFF)';
/** Инвертированный акцентный цвет с градиентом */
exports.inverseTextAccentGradientActive = 'var(--inverse-text-accent-gradient-active, #E6E6E6FF)';
/** Инвертированный акцентный цвет с градиентом */
exports.inverseTextAccentGradient = 'var(--inverse-text-accent-gradient, linear-gradient(45deg, #108210 0%, #14CC98 100%))';
/** Инвертированный цвет успеха */
exports.inverseTextPositiveHover = 'var(--inverse-text-positive-hover, #13AA2CFF)';
/** Инвертированный цвет успеха */
exports.inverseTextPositiveActive = 'var(--inverse-text-positive-active, #0D731EFF)';
/** Инвертированный цвет успеха */
exports.inverseTextPositive = 'var(--inverse-text-positive, #108E26)';
/** Инвертированный цвет предупреждения */
exports.inverseTextWarningHover = 'var(--inverse-text-warning-hover, #FD650DFF)';
/** Инвертированный цвет предупреждения */
exports.inverseTextWarningActive = 'var(--inverse-text-warning-active, #CA4B02FF)';
/** Инвертированный цвет предупреждения */
exports.inverseTextWarning = 'var(--inverse-text-warning, #E85702)';
/** Инвертированный цвет ошибки */
exports.inverseTextNegativeHover = 'var(--inverse-text-negative-hover, #ED263AFF)';
/** Инвертированный цвет ошибки */
exports.inverseTextNegativeActive = 'var(--inverse-text-negative-active, #C61022FF)';
/** Инвертированный цвет ошибки */
exports.inverseTextNegative = 'var(--inverse-text-negative, #E31227)';
exports.inverseTextInfoHover = 'var(--inverse-text-info-hover, #229EF1FF)';
exports.inverseTextInfoActive = 'var(--inverse-text-info-active, #0A6099FF)';
exports.inverseTextInfo = 'var(--inverse-text-info, #0C72B6)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimaryHover = 'var(--surface-solid-primary-hover, #303030FF)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimaryActive = 'var(--surface-solid-primary-active, #212121FF)';
/** Основной непрозрачный фон поверхности/контрола */
exports.surfaceSolidPrimary = 'var(--surface-solid-primary, #262626)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondaryHover = 'var(--surface-solid-secondary-hover, #404040FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondaryActive = 'var(--surface-solid-secondary-active, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidSecondary = 'var(--surface-solid-secondary, #363636)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiaryHover = 'var(--surface-solid-tertiary-hover, #595959FF)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiaryActive = 'var(--surface-solid-tertiary-active, #4A4A4AFF)';
/** Третичный непрозрачный фон поверхности/контрола */
exports.surfaceSolidTertiary = 'var(--surface-solid-tertiary, #4E4E4E)';
/** Основной фон для карточек */
exports.surfaceSolidCardHover = 'var(--surface-solid-card-hover, #303030FF)';
/** Основной фон для карточек */
exports.surfaceSolidCardActive = 'var(--surface-solid-card-active, #212121FF)';
/** Основной фон для карточек */
exports.surfaceSolidCard = 'var(--surface-solid-card, #262626)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefaultHover = 'var(--surface-solid-default-hover, #0D0D0DFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefaultActive = 'var(--surface-solid-default-active, #030303FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию */
exports.surfaceSolidDefault = 'var(--surface-solid-default, #080808)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClearHover = 'var(--surface-clear-hover, #00000000)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClearActive = 'var(--surface-clear-active, #00000000)';
/** Фон поверхности/контрола без заливки */
exports.surfaceClear = 'var(--surface-clear, rgba(0,0,0,0))';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentHover = 'var(--surface-accent-hover, #1DAF37FF)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccentActive = 'var(--surface-accent-active, #18952FFF)';
/** Акцентный фон поверхности/контрола */
exports.surfaceAccent = 'var(--surface-accent, #1A9E32)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradientHover = 'var(--surface-accent-gradient-hover, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradientActive = 'var(--surface-accent-gradient-active, #FFFFFFFF)';
/** Акцентный фон поверхности/контрола с градиентом */
exports.surfaceAccentGradient = 'var(--surface-accent-gradient, linear-gradient(45deg, #108210 0%, #14CC98 100%))';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositiveHover = 'var(--surface-positive-hover, #1DAF37FF)';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositiveActive = 'var(--surface-positive-active, #18952FFF)';
/** Цвет фона поверхности/контрола успех */
exports.surfacePositive = 'var(--surface-positive, #1A9E32)';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarningHover = 'var(--surface-warning-hover, #FB7223FF)';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarningActive = 'var(--surface-warning-active, #F05B05FF)';
/** Цвет фона поверхности/контрола предупреждение */
exports.surfaceWarning = 'var(--surface-warning, #FA5F05)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeHover = 'var(--surface-negative-hover, #FF475AFF)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegativeActive = 'var(--surface-negative-active, #FF1F35FF)';
/** Цвет фона поверхности/контрола ошибка */
exports.surfaceNegative = 'var(--surface-negative, #FF293E)';
exports.surfaceInfoHover = 'var(--surface-info-hover, #1798EEFF)';
exports.surfaceInfoActive = 'var(--surface-info-active, #1086D5FF)';
exports.surfaceInfo = 'var(--surface-info, #118CDF)';
exports.surfacePositiveMinorHover = 'var(--surface-positive-minor-hover, #0E3A16FF)';
exports.surfacePositiveMinorActive = 'var(--surface-positive-minor-active, #08210CFF)';
exports.surfacePositiveMinor = 'var(--surface-positive-minor, #0A2B10)';
exports.surfaceWarningMinorHover = 'var(--surface-warning-minor-hover, #4F250DFF)';
exports.surfaceWarningMinorActive = 'var(--surface-warning-minor-active, #351909FF)';
exports.surfaceWarningMinor = 'var(--surface-warning-minor, #3D1D0A)';
exports.surfaceNegativeMinorHover = 'var(--surface-negative-minor-hover, #5B1018FF)';
exports.surfaceNegativeMinorActive = 'var(--surface-negative-minor-active, #410B11FF)';
exports.surfaceNegativeMinor = 'var(--surface-negative-minor, #4A0D13)';
exports.surfaceInfoMinorHover = 'var(--surface-info-minor-hover, #10344CFF)';
exports.surfaceInfoMinorActive = 'var(--surface-info-minor-active, #0A2333FF)';
exports.surfaceInfoMinor = 'var(--surface-info-minor, #0C283B)';
/** Прозрачный цвет фона поверхности/контрола успех */
exports.surfaceTransparentPositiveHover = 'var(--surface-transparent-positive-hover, #1A9E3224)';
/** Прозрачный цвет фона поверхности/контрола успех */
exports.surfaceTransparentPositiveActive = 'var(--surface-transparent-positive-active, #1A9E320F)';
/** Прозрачный цвет фона поверхности/контрола успех */
exports.surfaceTransparentPositive = 'var(--surface-transparent-positive, rgba(26,158,50,0.08))';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
exports.surfaceTransparentWarningHover = 'var(--surface-transparent-warning-hover, #FF702424)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
exports.surfaceTransparentWarningActive = 'var(--surface-transparent-warning-active, #FF70240F)';
/** Прозрачный цвет фона поверхности/контрола предупреждение */
exports.surfaceTransparentWarning = 'var(--surface-transparent-warning, rgba(255,112,36,0.08))';
/** Прозрачный цвет фона поверхности/контрола ошибка */
exports.surfaceTransparentNegativeHover = 'var(--surface-transparent-negative-hover, #FF3D5124)';
/** Прозрачный цвет фона поверхности/контрола ошибка */
exports.surfaceTransparentNegativeActive = 'var(--surface-transparent-negative-active, #FF3D510F)';
/** Прозрачный цвет фона поверхности/контрола ошибка */
exports.surfaceTransparentNegative = 'var(--surface-transparent-negative, rgba(255,61,81,0.08))';
exports.surfaceTransparentInfoHover = 'var(--surface-transparent-info-hover, #199AF024)';
exports.surfaceTransparentInfoActive = 'var(--surface-transparent-info-active, #199AF00F)';
exports.surfaceTransparentInfo = 'var(--surface-transparent-info, rgba(25,154,240,0.08))';
exports.surfaceRatingHover = 'var(--surface-rating-hover, #FFDE5CFF)';
exports.surfaceRatingActive = 'var(--surface-rating-active, #FFD633FF)';
exports.surfaceRating = 'var(--surface-rating, #FFD83D)';
exports.surfaceRatingMinorHover = 'var(--surface-rating-minor-hover, #3C320BFF)';
exports.surfaceRatingMinorActive = 'var(--surface-rating-minor-active, #221D06FF)';
exports.surfaceRatingMinor = 'var(--surface-rating-minor, #2B2408)';
exports.surfaceTransparentRatingHover = 'var(--surface-transparent-rating-hover, #B7920124)';
exports.surfaceTransparentRatingActive = 'var(--surface-transparent-rating-active, #B792010F)';
exports.surfaceTransparentRating = 'var(--surface-transparent-rating, rgba(183,146,1,0.08))';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimaryHover = 'var(--on-dark-surface-solid-primary-hover, #FAFAFAFF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimaryActive = 'var(--on-dark-surface-solid-primary-active, #F0F0F0FF)';
/** Основной непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidPrimary = 'var(--on-dark-surface-solid-primary, #F5F5F5)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondaryHover = 'var(--on-dark-surface-solid-secondary-hover, #F7F7F7FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondaryActive = 'var(--on-dark-surface-solid-secondary-active, #E8E8E8FF)';
/** Вторичный непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceSolidSecondary = 'var(--on-dark-surface-solid-secondary, #ECECEC)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefaultHover = 'var(--on-dark-surface-solid-default-hover, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefaultActive = 'var(--on-dark-surface-solid-default-active, #FFFFFFFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на темном фоне */
exports.onDarkSurfaceSolidDefault = 'var(--on-dark-surface-solid-default, #FFFFFF)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimaryHover = 'var(--on-dark-surface-transparent-primary-hover, #FFFFFF1F)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimaryActive = 'var(--on-dark-surface-transparent-primary-active, #FFFFFF0A)';
/** Основной прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentPrimary = 'var(--on-dark-surface-transparent-primary, rgba(255,255,255,0.06))';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondaryHover = 'var(--on-dark-surface-transparent-secondary-hover, #FFFFFF3D)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondaryActive = 'var(--on-dark-surface-transparent-secondary-active, #FFFFFF0F)';
/** Вторичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentSecondary = 'var(--on-dark-surface-transparent-secondary, rgba(255,255,255,0.12))';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiaryHover = 'var(--on-dark-surface-transparent-tertiary-hover, #FFFFFF52)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiaryActive = 'var(--on-dark-surface-transparent-tertiary-active, #FFFFFF24)';
/** Третичный прозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentTertiary = 'var(--on-dark-surface-transparent-tertiary, rgba(255,255,255,0.2))';
/** Непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentDeepHover = 'var(--on-dark-surface-transparent-deep-hover, #FFFFFFC2)';
/** Непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentDeepActive = 'var(--on-dark-surface-transparent-deep-active, #FFFFFF94)';
/** Непрозрачный фон поверхности/контрола на темном фоне */
exports.onDarkSurfaceTransparentDeep = 'var(--on-dark-surface-transparent-deep, rgba(255,255,255,0.64))';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCardHover = 'var(--on-dark-surface-transparent-card-hover, #FFFFFF1F)';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCardActive = 'var(--on-dark-surface-transparent-card-active, #FFFFFF0A)';
/** Прозрачный фон для карточек на темном фоне */
exports.onDarkSurfaceTransparentCard = 'var(--on-dark-surface-transparent-card, rgba(255,255,255,0.06))';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidPrimaryHover = 'var(--on-light-surface-solid-primary-hover, #303030FF)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidPrimaryActive = 'var(--on-light-surface-solid-primary-active, #212121FF)';
/** Основной непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidPrimary = 'var(--on-light-surface-solid-primary, #262626)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondaryHover = 'var(--on-light-surface-solid-secondary-hover, #404040FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondaryActive = 'var(--on-light-surface-solid-secondary-active, #303030FF)';
/** Вторичный непрозрачный фон поверхности/контрола на светлом фоне */
exports.onLightSurfaceSolidSecondary = 'var(--on-light-surface-solid-secondary, #363636)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefaultHover = 'var(--on-light-surface-solid-default-hover, #0D0D0DFF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefaultActive = 'var(--on-light-surface-solid-default-active, #030303FF)';
/** Непрозрачный фон поверхности/контрола по умолчанию на светлом фоне */
exports.onLightSurfaceSolidDefault = 'var(--on-light-surface-solid-default, #080808)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidPrimaryHover = 'var(--inverse-surface-solid-primary-hover, #FAFAFAFF)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidPrimaryActive = 'var(--inverse-surface-solid-primary-active, #F0F0F0FF)';
/** Инвертированный основной непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidPrimary = 'var(--inverse-surface-solid-primary, #F5F5F5)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidSecondaryHover = 'var(--inverse-surface-solid-secondary-hover, #F7F7F7FF)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidSecondaryActive = 'var(--inverse-surface-solid-secondary-active, #E8E8E8FF)';
/** Инвертированный вторичный непрозрачный фон поверхности/контрола */
exports.inverseSurfaceSolidSecondary = 'var(--inverse-surface-solid-secondary, #ECECEC)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
exports.inverseSurfaceSolidDefaultHover = 'var(--inverse-surface-solid-default-hover, #FFFFFFFF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
exports.inverseSurfaceSolidDefaultActive = 'var(--inverse-surface-solid-default-active, #FFFFFFFF)';
/** Инвертированный непрозрачный фон поверхности/контрола по умолчанию */
exports.inverseSurfaceSolidDefault = 'var(--inverse-surface-solid-default, #FFFFFF)';
/** Основной фон */
exports.backgroundPrimary = 'var(--background-primary, #080808)';
exports.backgroundGradientGreen = 'var(--background-gradient-green, linear-gradient(36.09deg, #24B23E 1.152%, #AFED00 95.819%))';
exports.backgroundGradientOrange = 'var(--background-gradient-orange, linear-gradient(36.09deg, #FA6D20 1.152%, #FA9D6B 95.819%))';
exports.backgroundGradientYellow = 'var(--background-gradient-yellow, linear-gradient(36.09deg, #F2C202 1.152%, #FFD83D 95.819%))';
exports.backgroundGradientBlue = 'var(--background-gradient-blue, linear-gradient(36.09deg, #129DFA 1.152%, #52BAFF 95.819%))';
exports.backgroundGradientRed = 'var(--background-gradient-red, linear-gradient(36.09deg, #FF2E43 1.152%, #FD96A0 95.819%))';
exports.backgroundGradientPink = 'var(--background-gradient-pink, linear-gradient(34.38deg, #FF1F78 7.846%, #FF8FBC 90.55%))';
exports.backgroundGradientPurple = 'var(--background-gradient-purple, linear-gradient(34.38deg, #AD42F5 7.846%, #D39CF7 90.55%))';
/** Цвет фона паранжи светлый */
exports.overlaySoft = 'var(--overlay-soft, rgba(8,8,8,0.56))';
/** Цвет фона паранжи темный */
exports.overlayHard = 'var(--overlay-hard, rgba(8,8,8,0.96))';
exports.overlayBlur = 'var(--overlay-blur, rgba(8,8,8,0.20))';
exports.outlineDefaultOutlinePositiveHover = 'var(--outline-default-outline-positive-hover, #2ACB47FF)';
exports.outlineDefaultOutlinePositiveActive = 'var(--outline-default-outline-positive-active, #1F9835FF)';
exports.outlineDefaultOutlinePositive = 'var(--outline-default-outline-positive, #24B23E)';
exports.outlineDefaultOutlineWarningHover = 'var(--outline-default-outline-warning-hover, #FF8442FF)';
exports.outlineDefaultOutlineWarningActive = 'var(--outline-default-outline-warning-active, #FF5D05FF)';
exports.outlineDefaultOutlineWarning = 'var(--outline-default-outline-warning, #FF7024)';
exports.outlineDefaultOutlineNegativeHover = 'var(--outline-default-outline-negative-hover, #FF5C6CFF)';
exports.outlineDefaultOutlineNegativeActive = 'var(--outline-default-outline-negative-active, #FF1F35FF)';
exports.outlineDefaultOutlineNegative = 'var(--outline-default-outline-negative, #FF3D51)';
exports.outlineDefaultOutlineInfoHover = 'var(--outline-default-outline-info-hover, #66BCF5FF)';
exports.outlineDefaultOutlineInfoActive = 'var(--outline-default-outline-info-active, #0E8ADDFF)';
exports.outlineDefaultOutlineInfo = 'var(--outline-default-outline-info, #199AF0)';
exports.outlineDefaultOutlinePositiveMinorHover = 'var(--outline-default-outline-positive-minor-hover, #0F760FFF)';
exports.outlineDefaultOutlinePositiveMinorActive = 'var(--outline-default-outline-positive-minor-active, #0B5B0BFF)';
exports.outlineDefaultOutlinePositiveMinor = 'var(--outline-default-outline-positive-minor, #084008)';
exports.outlineDefaultOutlineWarningMinorHover = 'var(--outline-default-outline-warning-minor-hover, #95400FFF)';
exports.outlineDefaultOutlineWarningMinorActive = 'var(--outline-default-outline-warning-minor-active, #79340CFF)';
exports.outlineDefaultOutlineWarningMinor = 'var(--outline-default-outline-warning-minor, #5C2709)';
exports.outlineDefaultOutlineNegativeMinorHover = 'var(--outline-default-outline-negative-minor-hover, #A01825FF)';
exports.outlineDefaultOutlineNegativeMinorActive = 'var(--outline-default-outline-negative-minor-active, #85141FFF)';
exports.outlineDefaultOutlineNegativeMinor = 'var(--outline-default-outline-negative-minor, #6B1019)';
exports.outlineDefaultOutlineInfoMinorHover = 'var(--outline-default-outline-info-minor-hover, #145E8FFF)';
exports.outlineDefaultOutlineInfoMinorActive = 'var(--outline-default-outline-info-minor-active, #114C74FF)';
exports.outlineDefaultOutlineInfoMinor = 'var(--outline-default-outline-info-minor, #0D3B59)';
exports.outlineDefaultOutlinePrimaryHover = 'var(--outline-default-outline-primary-hover, #FFFFFFFF)';
exports.outlineDefaultOutlinePrimaryActive = 'var(--outline-default-outline-primary-active, #8C8C8CFF)';
exports.outlineDefaultOutlinePrimary = 'var(--outline-default-outline-primary, #4E4E4E)';
exports.outlineDefaultOutlineSecondaryHover = 'var(--outline-default-outline-secondary-hover, #FFFFFFFF)';
exports.outlineDefaultOutlineSecondaryActive = 'var(--outline-default-outline-secondary-active, #737373FF)';
exports.outlineDefaultOutlineSecondary = 'var(--outline-default-outline-secondary, #707070)';
exports.outlineDefaultOutlineTertiaryHover = 'var(--outline-default-outline-tertiary-hover, #FFFFFFFF)';
exports.outlineDefaultOutlineTertiaryActive = 'var(--outline-default-outline-tertiary-active, #575757FF)';
exports.outlineDefaultOutlineTertiary = 'var(--outline-default-outline-tertiary, #949494)';
exports.outlineDefaultOutlineRatingHover = 'var(--outline-default-outline-rating-hover, #FFDE5CFF)';
exports.outlineDefaultOutlineRatingActive = 'var(--outline-default-outline-rating-active, #FFD21FFF)';
exports.outlineDefaultOutlineRating = 'var(--outline-default-outline-rating, #FFD83D)';
exports.outlineDefaultOutlineRatingMinorHover = 'var(--outline-default-outline-rating-minor-hover, #77620DFF)';
exports.outlineDefaultOutlineRatingMinorActive = 'var(--outline-default-outline-rating-minor-active, #5C4B0AFF)';
exports.outlineDefaultOutlineRatingMinor = 'var(--outline-default-outline-rating-minor, #403507)';
exports.outlineDefaultOutlinePositiveTransparentHover = 'var(--outline-default-outline-positive-transparent-hover, #24B23EFF)';
exports.outlineDefaultOutlinePositiveTransparentActive = 'var(--outline-default-outline-positive-transparent-active, #24B23E31)';
exports.outlineDefaultOutlinePositiveTransparent = 'var(--outline-default-outline-positive-transparent, rgba(36,178,62,0.16))';
exports.outlineDefaultOutlineWarningTransparentHover = 'var(--outline-default-outline-warning-transparent-hover, #FF7024FF)';
exports.outlineDefaultOutlineWarningTransparentActive = 'var(--outline-default-outline-warning-transparent-active, #FF702431)';
exports.outlineDefaultOutlineWarningTransparent = 'var(--outline-default-outline-warning-transparent, rgba(255,112,36,0.16))';
exports.outlineDefaultOutlineNegativeTransparentHover = 'var(--outline-default-outline-negative-transparent-hover, #FF3D51FF)';
exports.outlineDefaultOutlineNegativeTransparentActive = 'var(--outline-default-outline-negative-transparent-active, #FF3D5131)';
exports.outlineDefaultOutlineNegativeTransparent = 'var(--outline-default-outline-negative-transparent, rgba(255,61,81,0.16))';
exports.outlineDefaultOutlineInfoTransparentHover = 'var(--outline-default-outline-info-transparent-hover, #528EFFFF)';
exports.outlineDefaultOutlineInfoTransparentActive = 'var(--outline-default-outline-info-transparent-active, #528EFF31)';
exports.outlineDefaultOutlineInfoTransparent = 'var(--outline-default-outline-info-transparent, rgba(82,142,255,0.16))';
exports.outlineDefaultOutlineRatingTransparentHover = 'var(--outline-default-outline-rating-transparent-hover, #FFD83DFF)';
exports.outlineDefaultOutlineRatingTransparentActive = 'var(--outline-default-outline-rating-transparent-active, #FFD83D31)';
exports.outlineDefaultOutlineRatingTransparent = 'var(--outline-default-outline-rating-transparent, rgba(255,216,61,0.16))';
exports.skeletonGradient = 'var(--skeleton-gradient, linear-gradient( 90deg, rgba(255, 255, 255, 0.09) 0%, rgba(255, 255, 255, 0.08) 6.25%, rgba(255, 255, 255, 0.05) 12.5%, rgba(255, 255, 255, 0.01) 25%, rgba(255, 255, 255, 0.05) 37.5%, rgba(255, 255, 255, 0.08) 43.75%, rgba(255, 255, 255, 0.09) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.05) 62.5%, rgba(255, 255, 255, 0.01) 75%, rgba(255, 255, 255, 0.05) 87.5%, rgba(255, 255, 255, 0.08) 93.75%, rgba(255, 255, 255, 0.09) 100% ))';
exports.skeletonGradientLighter = 'var(--skeleton-gradient-lighter, linear-gradient( 90deg, rgba(255, 255, 255, 0.36) 0%, rgba(255, 255, 255, 0.32) 6.25%, rgba(255, 255, 255, 0.20) 12.5%, rgba(255, 255, 255, 0.04) 25%, rgba(255, 255, 255, 0.20) 37.5%, rgba(255, 255, 255, 0.32) 43.75%, rgba(255, 255, 255, 0.36) 50%, rgba(255, 255, 255, 0.08) 56.25%, rgba(255, 255, 255, 0.20) 62.5%, rgba(255, 255, 255, 0.04) 75%, rgba(255, 255, 255, 0.20) 87.5%, rgba(255, 255, 255, 0.32) 93.75%, rgba(255, 255, 255, 0.36) 100% ))';
/** @deprecated instead use onDarkTextPrimary */
exports.whitePrimary = 'var(--plasma-colors-white-primary, var(--on-dark-text-primary))';
/** @deprecated instead use onDarkTextSecondary */
exports.whiteSecondary = 'var(--plasma-colors-white-secondary, var(--on-dark-text-secondary))';
/** @deprecated instead use onDarkTextTertiary */
exports.whiteTertiary = 'var(--plasma-colors-white-tertiary, var(--on-dark-text-tertiary))';
/** @deprecated instead use onLightTextPrimary */
exports.blackPrimary = 'var(--plasma-colors-black-primary, var(--on-light-text-primary))';
/** @deprecated instead use onLightTextSecondary */
exports.blackSecondary = 'var(--plasma-colors-black-secondary, var(--on-light-text-secondary))';
/** @deprecated instead use onLightTextTertiary */
exports.blackTertiary = 'var(--plasma-colors-black-tertiary, var(--on-light-text-tertiary))';
/** @deprecated instead use onLightSurfaceSolidDefault */
exports.buttonBlack = 'var(--plasma-colors-button-black, var(--on-light-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceSolidDefault */
exports.buttonWhite = 'var(--plasma-colors-button-white, var(--on-dark-surface-solid-default))';
/** @deprecated instead use onDarkSurfaceTransparentSecondary */
exports.buttonWhiteSecondary = 'var(--plasma-colors-button-white-secondary, var(--on-dark-surface-transparent-secondary))';
/** @deprecated instead use textPrimary */
exports.text = 'var(--plasma-colors-text, var(--text-primary))';
/** @deprecated instead use textPrimary */
exports.primary = 'var(--plasma-colors-primary, var(--text-primary))';
/** @deprecated instead use textSecondary */
exports.secondary = 'var(--plasma-colors-secondary, var(--text-secondary))';
/** @deprecated instead use textTertiary */
exports.tertiary = 'var(--plasma-colors-tertiary, var(--text-tertiary))';
/** @deprecated instead use backgroundPrimary */
exports.background = 'var(--plasma-colors-background, var(--background-primary))';
/** @deprecated instead use textAccent */
exports.accent = 'var(--plasma-colors-accent, var(--text-accent))';
/** @deprecated instead use textPositive */
exports.success = 'var(--plasma-colors-success, var(--text-positive))';
/** @deprecated instead use textWarning */
exports.warning = 'var(--plasma-colors-warning, var(--text-warning))';
/** @deprecated instead use textNegative */
exports.critical = 'var(--plasma-colors-critical, var(--text-negative))';
/** @deprecated instead use overlaySoft */
exports.overlay = 'var(--plasma-colors-overlay, var(--overlay-soft))';
/** @deprecated instead use surfaceSolidPrimary */
exports.surfaceSolid01 = 'var(--plasma-colors-surface-solid01, var(--surface-solid-primary))';
/** @deprecated instead use surfaceSolidSecondary */
exports.surfaceSolid02 = 'var(--plasma-colors-surface-solid02, var(--surface-solid-secondary))';
/** @deprecated instead use surfaceSolidTertiary */
exports.surfaceSolid03 = 'var(--plasma-colors-surface-solid03, var(--surface-solid-tertiary))';
/** @deprecated instead use textAccent */
exports.buttonAccent = 'var(--plasma-colors-button-accent, var(--text-accent))';
/** @deprecated instead use surfacePositive */
exports.buttonSuccess = 'var(--plasma-colors-button-success, var(--surface-positive))';
/** @deprecated instead use surfaceWarning */
exports.buttonWarning = 'var(--plasma-colors-button-warning, var(--surface-warning))';
/** @deprecated instead use surfaceNegative */
exports.buttonCritical = 'var(--plasma-colors-button-critical, var(--surface-negative))';
