// @salutejs/plasma-new-hope/styled-components/es/examples/typograpy/components/Old/Paragraph/Paragraph.config.js
import { css } from 'styled-components';
import { tokens } from '../../../../../components/Typography/tokens';
export var config = {
  defaults: {
    size: 'paragraph1'
  },
  variations: {
    size: {
      paragraph1: /*#__PURE__*/css(["", ":var(--plasma-typo-paragraph1-font-family);", ":var(--plasma-typo-paragraph1-font-size);", ":var(--plasma-typo-paragraph1-font-style);", ":var(--plasma-typo-paragraph1-font-weight);", ":var(--plasma-typo-paragraph1-letter-spacing);", ":var(--plasma-typo-paragraph1-line-height);"], tokens.typoFontFamily, tokens.typoFontSize, tokens.typoFontStyle, tokens.typoFontWeight, tokens.typoFontLetterSpacing, tokens.typoFontLineHeight),
      paragraph2: /*#__PURE__*/css(["", ":var(--plasma-typo-paragraph2-font-family);", ":var(--plasma-typo-paragraph2-font-size);", ":var(--plasma-typo-paragraph2-font-style);", ":var(--plasma-typo-paragraph2-font-weight);", ":var(--plasma-typo-paragraph2-letter-spacing);", ":var(--plasma-typo-paragraph2-line-height);"], tokens.typoFontFamily, tokens.typoFontSize, tokens.typoFontStyle, tokens.typoFontWeight, tokens.typoFontLetterSpacing, tokens.typoFontLineHeight)
    }
  }
};
