// @salutejs/plasma-new-hope/styled-components/es/components/Checkbox/variations/_focused/base.js
import { css } from 'styled-components';
import { addFocus } from '../../../../mixins';
import { classes, tokens } from '../../Checkbox.tokens';

/**
 * INFO: на 36 строке нет возможности использвать .${classes.checkboxTrigger},
 * потому что линтер правит селектор и он применяется некорректно
 */

export var base = /*#__PURE__*/css([".", "{", "}input[data-focus-visible-added] + label .", "{border-color:transparent;", "}input[data-focus-visible-added]:checked + label .checkbox-trigger,input[type='checkbox'][data-focus-visible-added]:indeterminate + label .", "{", "}"], classes.checkboxTrigger, /*#__PURE__*/addFocus({
  outlineOffset: '0',
  outlineSize: '0.125rem',
  outlineRadius: /*#__PURE__*/"var(".concat(tokens.triggerBorderRadius, ")"),
  outlineColor: 'transparent'
}), classes.checkboxTrigger, /*#__PURE__*/addFocus({
  outlineOffset: '-0.125rem',
  outlineSize: '0.125rem',
  outlineRadius: /*#__PURE__*/"var(".concat(tokens.triggerBorderRadius, ")"),
  outlineColor: /*#__PURE__*/"var(".concat(tokens.focusColor, ")"),
  customFocusRules: /*#__PURE__*/"\n                &::before {\n                    box-shadow: 0 0 0 0.125rem var(".concat(tokens.focusColor, ");\n                }\n            ")
}), classes.checkboxTrigger, /*#__PURE__*/addFocus({
  outlineOffset: '-0.125rem',
  outlineSize: '0.063rem',
  outlineRadius: /*#__PURE__*/"calc(var(".concat(tokens.triggerBorderRadius, ") + 0.125rem)"),
  outlineColor: /*#__PURE__*/"var(".concat(tokens.focusColor, ")"),
  customFocusRules: /*#__PURE__*/"\n                &::before {\n                    box-shadow: 0 0 0 0.063rem var(".concat(tokens.focusColor, ");\n                }\n            ")
}));
