// @salutejs/plasma-new-hope/es/components/Spinner/Spinner.js
import { objectWithoutProperties as _objectWithoutProperties, extends as _extends } from '../../_virtual/_rollupPluginBabelHelpers.js';
import React, { forwardRef } from 'react';
import { safeUseId } from '@salutejs/plasma-core';
import { base as base$2 } from './variations/_view/base.js';
import { base as base$1 } from './variations/_size/base.js';
import { SpinnerSvg } from './SpinnerSvg.js';
import { SpinnerWrapper, base } from './Spinner.styles.js';
import { useInnerProps } from './hooks/index.js';

var _excluded = ["id"];
var spinnerRoot = function spinnerRoot(Root) {
  return /*#__PURE__*/forwardRef(function (props, ref) {
    var id = props.id,
      rest = _objectWithoutProperties(props, _excluded);
    var innerProps = useInnerProps(props, {
      wrapperWidth: 'auto',
      wrapperHeight: 'auto',
      svgWidth: 'var(--plasma-spinner-size)',
      svgHeight: 'var(--plasma-spinner-size)',
      color: 'currentColor'
    });
    var uniqId = safeUseId;
    var innerId = id || uniqId();
    return /*#__PURE__*/React.createElement(Root, _extends({
      ref: ref,
      id: innerId
    }, rest), /*#__PURE__*/React.createElement(SpinnerWrapper, {
      width: innerProps.wrapperWidth,
      height: innerProps.wrapperHeight
    }, /*#__PURE__*/React.createElement(SpinnerSvg, {
      id: innerId,
      width: innerProps.svgWidth,
      height: innerProps.svgHeight,
      color: innerProps.color
    })));
  });
};
var spinnerConfig = {
  name: 'Spinner',
  tag: 'div',
  layout: spinnerRoot,
  base: base,
  variations: {
    size: {
      css: base$1
    },
    view: {
      css: base$2
    }
  },
  defaults: {
    size: 'm',
    view: 'accent'
  }
};

export { spinnerConfig, spinnerRoot };
