// @salutejs/plasma-new-hope/styled-components/es/components/Dropdown/ui/DropdownOld/DropdownOld.js
var _excluded = ["id", "target", "children", "hasArrow", "role", "view", "size", "frame", "onToggle", "isFocusTrapped", "isOpen", "placement", "trigger", "offset", "preventOverflow", "closeOnOverlayClick", "closeOnEsc"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import React, { forwardRef, useRef } from 'react';
import { useFocusTrap, useForkRef, useUniqId } from '@salutejs/plasma-core';
import { base as viewCSS } from './variations/_view/base';
import { base as sizeCSS } from './variations/_size/base';
import { StyledDropdown, StyledPopover } from './Dropdown.styles';
import { getPlacements } from './utils';
/**
 * Выпадающий список без внешнего контроля видимости.
 */
export var dropdownOldRoot = function dropdownOldRoot(Root) {
  return /*#__PURE__*/forwardRef(function (_ref, outerRootRef) {
    var id = _ref.id,
      target = _ref.target,
      children = _ref.children,
      hasArrow = _ref.hasArrow,
      role = _ref.role,
      view = _ref.view,
      size = _ref.size,
      frame = _ref.frame,
      _onToggle = _ref.onToggle,
      _ref$isFocusTrapped = _ref.isFocusTrapped,
      isFocusTrapped = _ref$isFocusTrapped === void 0 ? true : _ref$isFocusTrapped,
      _ref$isOpen = _ref.isOpen,
      isOpen = _ref$isOpen === void 0 ? false : _ref$isOpen,
      _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? 'auto' : _ref$placement,
      _ref$trigger = _ref.trigger,
      trigger = _ref$trigger === void 0 ? 'click' : _ref$trigger,
      _ref$offset = _ref.offset,
      offset = _ref$offset === void 0 ? [0, 6] : _ref$offset,
      _ref$preventOverflow = _ref.preventOverflow,
      preventOverflow = _ref$preventOverflow === void 0 ? false : _ref$preventOverflow,
      _ref$closeOnOverlayCl = _ref.closeOnOverlayClick,
      closeOnOverlayClick = _ref$closeOnOverlayCl === void 0 ? false : _ref$closeOnOverlayCl,
      _ref$closeOnEsc = _ref.closeOnEsc,
      closeOnEsc = _ref$closeOnEsc === void 0 ? false : _ref$closeOnEsc,
      rest = _objectWithoutProperties(_ref, _excluded);
    var uniqId = useUniqId();
    var innerId = id || uniqId;
    var rootRef = useRef(null);
    var dropdownRef = useRef(null);
    var handleRef = useForkRef(rootRef, outerRootRef);
    var trapRef = useFocusTrap(isOpen && isFocusTrapped);
    var dropdownForkRef = useForkRef(dropdownRef, trapRef);
    return /*#__PURE__*/React.createElement(StyledPopover, {
      role: role,
      isOpen: isOpen,
      usePortal: false,
      onToggle: function onToggle(is, event) {
        return _onToggle === null || _onToggle === void 0 ? void 0 : _onToggle(is, event);
      },
      id: innerId,
      ref: dropdownForkRef,
      target: target,
      offset: offset,
      preventOverflow: preventOverflow,
      hasArrow: hasArrow,
      placement: getPlacements(placement),
      trigger: trigger,
      closeOnOverlayClick: closeOnOverlayClick,
      closeOnEsc: closeOnEsc,
      isFocusTrapped: isFocusTrapped,
      frame: frame
    }, /*#__PURE__*/React.createElement(Root, _extends({
      ref: handleRef,
      view: view,
      size: size
    }, rest), /*#__PURE__*/React.createElement(StyledDropdown, null, children)));
  });
};
export var dropdownOldConfig = {
  name: 'Dropdown',
  tag: 'div',
  layout: dropdownOldRoot,
  base: '',
  variations: {
    view: {
      css: viewCSS
    },
    size: {
      css: sizeCSS
    }
  },
  defaults: {
    view: 'default',
    size: 'm'
  }
};
